<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> * Created on 01-21-2010</a>
<span class="sourceLineNo">021</span><a id="line.21"> */</a>
<span class="sourceLineNo">022</span><a id="line.22">package org.biojava.nbio.core.sequence.io;</a>
<span class="sourceLineNo">023</span><a id="line.23"></a>
<span class="sourceLineNo">024</span><a id="line.24">import org.biojava.nbio.core.sequence.DNASequence;</a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.nbio.core.sequence.GeneSequence;</a>
<span class="sourceLineNo">026</span><a id="line.26">import org.biojava.nbio.core.sequence.ProteinSequence;</a>
<span class="sourceLineNo">027</span><a id="line.27">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</a>
<span class="sourceLineNo">028</span><a id="line.28">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.biojava.nbio.core.sequence.io.template.FastaHeaderFormatInterface;</a>
<span class="sourceLineNo">030</span><a id="line.30">import org.biojava.nbio.core.sequence.template.Compound;</a>
<span class="sourceLineNo">031</span><a id="line.31">import org.biojava.nbio.core.sequence.template.Sequence;</a>
<span class="sourceLineNo">032</span><a id="line.32"></a>
<span class="sourceLineNo">033</span><a id="line.33">import java.io.BufferedOutputStream;</a>
<span class="sourceLineNo">034</span><a id="line.34">import java.io.File;</a>
<span class="sourceLineNo">035</span><a id="line.35">import java.io.FileOutputStream;</a>
<span class="sourceLineNo">036</span><a id="line.36">import java.io.OutputStream;</a>
<span class="sourceLineNo">037</span><a id="line.37">import java.util.ArrayList;</a>
<span class="sourceLineNo">038</span><a id="line.38">import java.util.Collection;</a>
<span class="sourceLineNo">039</span><a id="line.39"></a>
<span class="sourceLineNo">040</span><a id="line.40">/**</a>
<span class="sourceLineNo">041</span><a id="line.41"> * The class that should be used to write out fasta file of a sequence collection</a>
<span class="sourceLineNo">042</span><a id="line.42"> * @author Scooter Willis &lt;willishf at gmail dot com&gt;</a>
<span class="sourceLineNo">043</span><a id="line.43"> */</a>
<span class="sourceLineNo">044</span><a id="line.44">public class FastaWriterHelper {</a>
<span class="sourceLineNo">045</span><a id="line.45"></a>
<span class="sourceLineNo">046</span><a id="line.46"></a>
<span class="sourceLineNo">047</span><a id="line.47"></a>
<span class="sourceLineNo">048</span><a id="line.48">        /**</a>
<span class="sourceLineNo">049</span><a id="line.49">         * Write collection of protein sequences to a file</a>
<span class="sourceLineNo">050</span><a id="line.50">         *</a>
<span class="sourceLineNo">051</span><a id="line.51">         * @param file</a>
<span class="sourceLineNo">052</span><a id="line.52">         * @param proteinSequences</a>
<span class="sourceLineNo">053</span><a id="line.53">         * @throws Exception</a>
<span class="sourceLineNo">054</span><a id="line.54">         */</a>
<span class="sourceLineNo">055</span><a id="line.55">        public static void writeProteinSequence(File file,</a>
<span class="sourceLineNo">056</span><a id="line.56">                        Collection&lt;ProteinSequence&gt; proteinSequences) throws Exception {</a>
<span class="sourceLineNo">057</span><a id="line.57">                FileOutputStream outputStream = new FileOutputStream(file);</a>
<span class="sourceLineNo">058</span><a id="line.58">                BufferedOutputStream bo = new BufferedOutputStream(outputStream);</a>
<span class="sourceLineNo">059</span><a id="line.59">                writeProteinSequence(bo, proteinSequences);</a>
<span class="sourceLineNo">060</span><a id="line.60">                bo.close();</a>
<span class="sourceLineNo">061</span><a id="line.61">                outputStream.close();</a>
<span class="sourceLineNo">062</span><a id="line.62">        }</a>
<span class="sourceLineNo">063</span><a id="line.63"></a>
<span class="sourceLineNo">064</span><a id="line.64">        /**</a>
<span class="sourceLineNo">065</span><a id="line.65">         * Write collection of protein sequences to a stream</a>
<span class="sourceLineNo">066</span><a id="line.66">         * @param outputStream</a>
<span class="sourceLineNo">067</span><a id="line.67">         * @param proteinSequences</a>
<span class="sourceLineNo">068</span><a id="line.68">         * @throws Exception</a>
<span class="sourceLineNo">069</span><a id="line.69">         */</a>
<span class="sourceLineNo">070</span><a id="line.70"></a>
<span class="sourceLineNo">071</span><a id="line.71">        public static void writeProteinSequence(OutputStream outputStream,</a>
<span class="sourceLineNo">072</span><a id="line.72">                        Collection&lt;ProteinSequence&gt; proteinSequences) throws Exception {</a>
<span class="sourceLineNo">073</span><a id="line.73"></a>
<span class="sourceLineNo">074</span><a id="line.74">                FastaWriter&lt;ProteinSequence, AminoAcidCompound&gt; fastaWriter = new FastaWriter&lt;ProteinSequence, AminoAcidCompound&gt;(</a>
<span class="sourceLineNo">075</span><a id="line.75">                                outputStream, proteinSequences,</a>
<span class="sourceLineNo">076</span><a id="line.76">                                new GenericFastaHeaderFormat&lt;ProteinSequence, AminoAcidCompound&gt;());</a>
<span class="sourceLineNo">077</span><a id="line.77">                fastaWriter.process();</a>
<span class="sourceLineNo">078</span><a id="line.78"></a>
<span class="sourceLineNo">079</span><a id="line.79">        }</a>
<span class="sourceLineNo">080</span><a id="line.80"></a>
<span class="sourceLineNo">081</span><a id="line.81">                /**</a>
<span class="sourceLineNo">082</span><a id="line.82">         * Write a collection of GeneSequences to a file where if the gene is negative strand it will flip and complement the sequence</a>
<span class="sourceLineNo">083</span><a id="line.83">         * @param file</a>
<span class="sourceLineNo">084</span><a id="line.84">         * @param geneSequences</a>
<span class="sourceLineNo">085</span><a id="line.85">         * @throws Exception</a>
<span class="sourceLineNo">086</span><a id="line.86">         */</a>
<span class="sourceLineNo">087</span><a id="line.87"></a>
<span class="sourceLineNo">088</span><a id="line.88">        public static void writeGeneSequence(File file, Collection&lt;GeneSequence&gt; geneSequences,boolean showExonUppercase) throws Exception {</a>
<span class="sourceLineNo">089</span><a id="line.89">                FileOutputStream outputStream = new FileOutputStream(file);</a>
<span class="sourceLineNo">090</span><a id="line.90">                BufferedOutputStream bo = new BufferedOutputStream(outputStream);</a>
<span class="sourceLineNo">091</span><a id="line.91">                writeGeneSequence(bo, geneSequences,showExonUppercase);</a>
<span class="sourceLineNo">092</span><a id="line.92">                bo.close();</a>
<span class="sourceLineNo">093</span><a id="line.93">                outputStream.close();</a>
<span class="sourceLineNo">094</span><a id="line.94">        }</a>
<span class="sourceLineNo">095</span><a id="line.95"></a>
<span class="sourceLineNo">096</span><a id="line.96">        /**</a>
<span class="sourceLineNo">097</span><a id="line.97">         * Write a collection of GeneSequences to a file where if the gene is negative strand it will flip and complement the sequence</a>
<span class="sourceLineNo">098</span><a id="line.98">         * @param outputStream</a>
<span class="sourceLineNo">099</span><a id="line.99">         * @param dnaSequences</a>
<span class="sourceLineNo">100</span><a id="line.100">         * @throws Exception</a>
<span class="sourceLineNo">101</span><a id="line.101">         */</a>
<span class="sourceLineNo">102</span><a id="line.102"></a>
<span class="sourceLineNo">103</span><a id="line.103">        public static void writeGeneSequence(OutputStream outputStream, Collection&lt;GeneSequence&gt; geneSequences,boolean showExonUppercase) throws Exception {</a>
<span class="sourceLineNo">104</span><a id="line.104">                FastaGeneWriter fastaWriter = new FastaGeneWriter(</a>
<span class="sourceLineNo">105</span><a id="line.105">                                outputStream, geneSequences,</a>
<span class="sourceLineNo">106</span><a id="line.106">                                new GenericFastaHeaderFormat&lt;GeneSequence, NucleotideCompound&gt;(),showExonUppercase);</a>
<span class="sourceLineNo">107</span><a id="line.107">                fastaWriter.process();</a>
<span class="sourceLineNo">108</span><a id="line.108"></a>
<span class="sourceLineNo">109</span><a id="line.109">        }</a>
<span class="sourceLineNo">110</span><a id="line.110"></a>
<span class="sourceLineNo">111</span><a id="line.111"></a>
<span class="sourceLineNo">112</span><a id="line.112">        /**</a>
<span class="sourceLineNo">113</span><a id="line.113">         * Write a collection of NucleotideSequences to a file</a>
<span class="sourceLineNo">114</span><a id="line.114">         * @param file</a>
<span class="sourceLineNo">115</span><a id="line.115">         * @param dnaSequences</a>
<span class="sourceLineNo">116</span><a id="line.116">         * @throws Exception</a>
<span class="sourceLineNo">117</span><a id="line.117">         */</a>
<span class="sourceLineNo">118</span><a id="line.118"></a>
<span class="sourceLineNo">119</span><a id="line.119">        public static void writeNucleotideSequence(File file, Collection&lt;DNASequence&gt; dnaSequences) throws Exception {</a>
<span class="sourceLineNo">120</span><a id="line.120">                FileOutputStream outputStream = new FileOutputStream(file);</a>
<span class="sourceLineNo">121</span><a id="line.121">                BufferedOutputStream bo = new BufferedOutputStream(outputStream);</a>
<span class="sourceLineNo">122</span><a id="line.122">                writeNucleotideSequence(bo, dnaSequences);</a>
<span class="sourceLineNo">123</span><a id="line.123">                bo.close();</a>
<span class="sourceLineNo">124</span><a id="line.124">                outputStream.close();</a>
<span class="sourceLineNo">125</span><a id="line.125">        }</a>
<span class="sourceLineNo">126</span><a id="line.126"></a>
<span class="sourceLineNo">127</span><a id="line.127">        /**</a>
<span class="sourceLineNo">128</span><a id="line.128">         * Write a collection of NucleotideSequences to a file</a>
<span class="sourceLineNo">129</span><a id="line.129">         * @param outputStream</a>
<span class="sourceLineNo">130</span><a id="line.130">         * @param dnaSequences</a>
<span class="sourceLineNo">131</span><a id="line.131">         * @throws Exception</a>
<span class="sourceLineNo">132</span><a id="line.132">         */</a>
<span class="sourceLineNo">133</span><a id="line.133"></a>
<span class="sourceLineNo">134</span><a id="line.134">        public static void writeNucleotideSequence(OutputStream outputStream, Collection&lt;DNASequence&gt; dnaSequences) throws Exception {</a>
<span class="sourceLineNo">135</span><a id="line.135">                FastaWriter&lt;DNASequence, NucleotideCompound&gt; fastaWriter = new FastaWriter&lt;DNASequence, NucleotideCompound&gt;(</a>
<span class="sourceLineNo">136</span><a id="line.136">                                outputStream, dnaSequences,</a>
<span class="sourceLineNo">137</span><a id="line.137">                                new GenericFastaHeaderFormat&lt;DNASequence, NucleotideCompound&gt;());</a>
<span class="sourceLineNo">138</span><a id="line.138">                fastaWriter.process();</a>
<span class="sourceLineNo">139</span><a id="line.139"></a>
<span class="sourceLineNo">140</span><a id="line.140">        }</a>
<span class="sourceLineNo">141</span><a id="line.141"></a>
<span class="sourceLineNo">142</span><a id="line.142">        /**</a>
<span class="sourceLineNo">143</span><a id="line.143">         * Write a sequence to a file</a>
<span class="sourceLineNo">144</span><a id="line.144">         * @param file</a>
<span class="sourceLineNo">145</span><a id="line.145">         * @param sequence</a>
<span class="sourceLineNo">146</span><a id="line.146">         * @throws Exception</a>
<span class="sourceLineNo">147</span><a id="line.147">         */</a>
<span class="sourceLineNo">148</span><a id="line.148">        public static void writeSequence(File file, Sequence&lt;?&gt; sequence) throws Exception {</a>
<span class="sourceLineNo">149</span><a id="line.149">                FileOutputStream outputStream = new FileOutputStream(file);</a>
<span class="sourceLineNo">150</span><a id="line.150">                BufferedOutputStream bo = new BufferedOutputStream(outputStream);</a>
<span class="sourceLineNo">151</span><a id="line.151">                writeSequences(bo, singleSeqToCollection(sequence));</a>
<span class="sourceLineNo">152</span><a id="line.152">                bo.close();</a>
<span class="sourceLineNo">153</span><a id="line.153">                outputStream.close();</a>
<span class="sourceLineNo">154</span><a id="line.154">        }</a>
<span class="sourceLineNo">155</span><a id="line.155"></a>
<span class="sourceLineNo">156</span><a id="line.156">        /**</a>
<span class="sourceLineNo">157</span><a id="line.157">         * Write a sequence to OutputStream</a>
<span class="sourceLineNo">158</span><a id="line.158">         * @param outputStream</a>
<span class="sourceLineNo">159</span><a id="line.159">         * @param sequence</a>
<span class="sourceLineNo">160</span><a id="line.160">         * @throws Exception</a>
<span class="sourceLineNo">161</span><a id="line.161">         */</a>
<span class="sourceLineNo">162</span><a id="line.162">        public static void writeSequence(OutputStream outputStream, Sequence&lt;?&gt; sequence) throws Exception {</a>
<span class="sourceLineNo">163</span><a id="line.163">                writeSequences(outputStream, singleSeqToCollection(sequence));</a>
<span class="sourceLineNo">164</span><a id="line.164">        }</a>
<span class="sourceLineNo">165</span><a id="line.165"></a>
<span class="sourceLineNo">166</span><a id="line.166">        /**</a>
<span class="sourceLineNo">167</span><a id="line.167">         *</a>
<span class="sourceLineNo">168</span><a id="line.168">         * @param sequence</a>
<span class="sourceLineNo">169</span><a id="line.169">         * @return</a>
<span class="sourceLineNo">170</span><a id="line.170">         */</a>
<span class="sourceLineNo">171</span><a id="line.171"></a>
<span class="sourceLineNo">172</span><a id="line.172">        private static Collection&lt;Sequence&lt;?&gt;&gt; singleSeqToCollection(Sequence&lt;?&gt; sequence) {</a>
<span class="sourceLineNo">173</span><a id="line.173">                Collection&lt;Sequence&lt;?&gt;&gt; sequences = new ArrayList&lt;Sequence&lt;?&gt;&gt;();</a>
<span class="sourceLineNo">174</span><a id="line.174">                sequences.add(sequence);</a>
<span class="sourceLineNo">175</span><a id="line.175">                return sequences;</a>
<span class="sourceLineNo">176</span><a id="line.176">        }</a>
<span class="sourceLineNo">177</span><a id="line.177"></a>
<span class="sourceLineNo">178</span><a id="line.178">        /**</a>
<span class="sourceLineNo">179</span><a id="line.179">         * Method which will write your given Sequences to the specified</a>
<span class="sourceLineNo">180</span><a id="line.180">         * {@link OutputStream}. This is a very generic method which writes just the</a>
<span class="sourceLineNo">181</span><a id="line.181">         * AccessionID of the Sequence as the FASTA header.</a>
<span class="sourceLineNo">182</span><a id="line.182">         *</a>
<span class="sourceLineNo">183</span><a id="line.183">         * @param outputStream Stream to write to; can be System.out</a>
<span class="sourceLineNo">184</span><a id="line.184">         * @param sequences The sequences to write out</a>
<span class="sourceLineNo">185</span><a id="line.185">         * @throws Exception Thrown normally thanks to IO problems</a>
<span class="sourceLineNo">186</span><a id="line.186">         */</a>
<span class="sourceLineNo">187</span><a id="line.187">        public static void writeSequences(OutputStream outputStream,</a>
<span class="sourceLineNo">188</span><a id="line.188">                        Collection&lt;Sequence&lt;?&gt;&gt; sequences) throws Exception {</a>
<span class="sourceLineNo">189</span><a id="line.189"></a>
<span class="sourceLineNo">190</span><a id="line.190">                FastaHeaderFormatInterface&lt;Sequence&lt;?&gt;, Compound&gt; fhfi =</a>
<span class="sourceLineNo">191</span><a id="line.191">                                new FastaHeaderFormatInterface&lt;Sequence&lt;?&gt;, Compound&gt;() {</a>
<span class="sourceLineNo">192</span><a id="line.192"></a>
<span class="sourceLineNo">193</span><a id="line.193">                                        @Override</a>
<span class="sourceLineNo">194</span><a id="line.194">                                        public String getHeader(Sequence&lt;?&gt; sequence) {</a>
<span class="sourceLineNo">195</span><a id="line.195">                                                return sequence.getAccession().toString();</a>
<span class="sourceLineNo">196</span><a id="line.196">                                        }</a>
<span class="sourceLineNo">197</span><a id="line.197"></a>
<span class="sourceLineNo">198</span><a id="line.198">                                        ;</a>
<span class="sourceLineNo">199</span><a id="line.199">                                };</a>
<span class="sourceLineNo">200</span><a id="line.200"></a>
<span class="sourceLineNo">201</span><a id="line.201">                FastaWriter&lt;Sequence&lt;?&gt;, Compound&gt; fastaWriter =</a>
<span class="sourceLineNo">202</span><a id="line.202">                                new FastaWriter&lt;Sequence&lt;?&gt;, Compound&gt;(outputStream,</a>
<span class="sourceLineNo">203</span><a id="line.203">                                sequences, fhfi);</a>
<span class="sourceLineNo">204</span><a id="line.204"></a>
<span class="sourceLineNo">205</span><a id="line.205">                fastaWriter.process();</a>
<span class="sourceLineNo">206</span><a id="line.206">        }</a>
<span class="sourceLineNo">207</span><a id="line.207">}</a>




























































</pre>
</div>
</main>
</body>
</html>
