<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.1) on Thu Jan 03 22:38:55 CET 2019 -->
<title>Uses of Interface org.biojava.nbio.core.sequence.template.Sequence (biojava 5.2.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2019-01-03">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Interface org.biojava.nbio.core.sequence.template.Sequence (biojava 5.2.0 API)";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Interface org.biojava.nbio.core.sequence.template.Sequence" class="title">Uses of Interface<br>org.biojava.nbio.core.sequence.template.Sequence</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.aaproperties.xml">org.biojava.nbio.aaproperties.xml</a></th>
<td class="colLast">
<div class="block">Set of classes that handles the reading and writing of xml files.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.alignment">org.biojava.nbio.alignment</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.alignment.routines">org.biojava.nbio.alignment.routines</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.alignment.template">org.biojava.nbio.alignment.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.alignment">org.biojava.nbio.core.alignment</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.alignment.template">org.biojava.nbio.core.alignment.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.search.io">org.biojava.nbio.core.search.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.search.io.blast">org.biojava.nbio.core.search.io.blast</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.compound">org.biojava.nbio.core.sequence.compound</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.edits">org.biojava.nbio.core.sequence.edits</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.io.template">org.biojava.nbio.core.sequence.io.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.io.util">org.biojava.nbio.core.sequence.io.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.loader">org.biojava.nbio.core.sequence.loader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.location.template">org.biojava.nbio.core.sequence.location.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.storage">org.biojava.nbio.core.sequence.storage</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.transcription">org.biojava.nbio.core.sequence.transcription</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.views">org.biojava.nbio.core.sequence.views</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.util">org.biojava.nbio.core.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.phylo">org.biojava.nbio.phylo</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure">org.biojava.nbio.structure</a></th>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></th>
<td class="colLast">
<div class="block">Input and Output of Structures</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.ws.alignment">org.biojava.nbio.ws.alignment</a></th>
<td class="colLast">
<div class="block">This package contains the interfaces that need to be implemented by discrete alignment services.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.ws.alignment.qblast">org.biojava.nbio.ws.alignment.qblast</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.aaproperties.xml">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../aaproperties/xml/package-summary.html">org.biojava.nbio.aaproperties.xml</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../aaproperties/xml/package-summary.html">org.biojava.nbio.aaproperties.xml</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CaseFreeAminoAcidCompoundSet.</span><code><span class="memberNameLink"><a href="../../../../aaproperties/xml/CaseFreeAminoAcidCompoundSet.html#isValidSequence(org.biojava.nbio.core.sequence.template.Sequence)">isValidSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ModifiedAminoAcidCompoundSet.</span><code><span class="memberNameLink"><a href="../../../../aaproperties/xml/ModifiedAminoAcidCompoundSet.html#isValidSequence(org.biojava.nbio.core.sequence.template.Sequence)">isValidSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../alignment/package-summary.html">org.biojava.nbio.alignment</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../../alignment/package-summary.html">org.biojava.nbio.alignment</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/FractionalIdentityInProfileScorer.html" title="class in org.biojava.nbio.alignment">FractionalIdentityInProfileScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair picked from an alignment
 <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/FractionalIdentityScorer.html" title="class in org.biojava.nbio.alignment">FractionalIdentityScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/FractionalSimilarityInProfileScorer.html" title="class in org.biojava.nbio.alignment">FractionalSimilarityInProfileScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair picked from an alignment
 <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/FractionalSimilarityScorer.html" title="class in org.biojava.nbio.alignment">FractionalSimilarityScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/GuideTree.html" title="class in org.biojava.nbio.alignment">GuideTree</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a data structure for a guide tree used during progressive multiple sequence alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/NeedlemanWunsch.html" title="class in org.biojava.nbio.alignment">NeedlemanWunsch</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Needleman and Wunsch defined an algorithm for pairwise global sequence alignments (from the first until the last
 <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/SimpleProfileProfileAligner.html" title="class in org.biojava.nbio.alignment">SimpleProfileProfileAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a simple (naive) <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/SmithWaterman.html" title="class in org.biojava.nbio.alignment">SmithWaterman</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Smith and Waterman defined an algorithm for pairwise local sequence alignments (best match of sections from each
 <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/StandardRescoreRefiner.html" title="class in org.biojava.nbio.alignment">StandardRescoreRefiner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/SubstitutionMatrixScorer.html" title="class in org.biojava.nbio.alignment">SubstitutionMatrixScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Scores using a substitution matrix.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../alignment/package-summary.html">org.biojava.nbio.alignment</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;S,&#8203;C&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getAllPairsAlignments(java.util.List,org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getAllPairsAlignments</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences,
                     <a href="../../../../alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;type,
                     <a href="../../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                     <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code></th>
<td class="colLast">
<div class="block">Factory method which computes a sequence alignment for all <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pairs in the given <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../../alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceScorer</a>&lt;S,&#8203;C&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getAllPairsScorers(java.util.List,org.biojava.nbio.alignment.Alignments.PairwiseSequenceScorerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getAllPairsScorers</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences,
                  <a href="../../../../alignment/Alignments.PairwiseSequenceScorerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceScorerType</a>&nbsp;type,
                  <a href="../../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                  <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code></th>
<td class="colLast">
<div class="block">Factory method which sets up a sequence pair scorer for all <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pairs in the given <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>double[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getAllPairsScores(java.util.List,org.biojava.nbio.alignment.Alignments.PairwiseSequenceScorerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getAllPairsScores</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences,
                 <a href="../../../../alignment/Alignments.PairwiseSequenceScorerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceScorerType</a>&nbsp;type,
                 <a href="../../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                 <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code></th>
<td class="colLast">
<div class="block">Factory method which computes a sequence pair score for all <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pairs in the given <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;S,&#8203;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getMultipleSequenceAlignment(java.util.List,java.lang.Object...)">getMultipleSequenceAlignment</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences,
                            <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>...&nbsp;settings)</code></th>
<td class="colLast">
<div class="block">Factory method which computes a multiple sequence alignment for the given <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a> of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../../../../alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceAligner</a>&lt;S,&#8203;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getPairwiseAligner(S,S,org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getPairwiseAligner</a></span>&#8203;(S&nbsp;query,
                  S&nbsp;target,
                  <a href="../../../../alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;type,
                  <a href="../../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                  <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code></th>
<td class="colLast">
<div class="block">Factory method which constructs a pairwise sequence aligner.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../../../alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;S,&#8203;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getPairwiseAlignment(S,S,org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getPairwiseAlignment</a></span>&#8203;(S&nbsp;query,
                    S&nbsp;target,
                    <a href="../../../../alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;type,
                    <a href="../../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                    <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code></th>
<td class="colLast">
<div class="block">Factory method which computes a sequence alignment for the given <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pair.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;S,&#8203;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getProgressiveAlignment(org.biojava.nbio.alignment.GuideTree,org.biojava.nbio.alignment.Alignments.ProfileProfileAlignerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getProgressiveAlignment</a></span>&#8203;(<a href="../../../../alignment/GuideTree.html" title="class in org.biojava.nbio.alignment">GuideTree</a>&lt;S,&#8203;C&gt;&nbsp;tree,
                       <a href="../../../../alignment/Alignments.ProfileProfileAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.ProfileProfileAlignerType</a>&nbsp;type,
                       <a href="../../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                       <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code></th>
<td class="colLast">
<div class="block">Factory method to run the profile-profile alignments of a progressive multiple sequence alignment concurrently.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>double[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#runPairwiseScorers(java.util.List)">runPairwiseScorers</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../../alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceScorer</a>&lt;S,&#8203;C&gt;&gt;&nbsp;scorers)</code></th>
<td class="colLast">
<div class="block">Factory method to run a list of scorers concurrently.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.alignment.routines">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../alignment/routines/package-summary.html">org.biojava.nbio.alignment.routines</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../../alignment/routines/package-summary.html">org.biojava.nbio.alignment.routines</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/routines/AnchoredPairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.routines">AnchoredPairwiseSequenceAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">This algorithm uses a divide-and-conquer approach to find optimal pairwise global sequence alignments (from the
 first until the last <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>) with the restriction that any alignment produced
 will connect the query sequence to the target sequence at the <em>anchors</em>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/routines/GuanUberbacher.html" title="class in org.biojava.nbio.alignment.routines">GuanUberbacher</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Guan and Uberbacher defined an algorithm for pairwise global sequence alignments (from the first until the last
 <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.alignment.template">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../alignment/template/package-summary.html">org.biojava.nbio.alignment.template</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../../alignment/template/package-summary.html">org.biojava.nbio.alignment.template</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/AbstractMatrixAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractMatrixAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements common code for an <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> which builds a score matrix during computation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/AbstractPairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractPairwiseSequenceAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements common code for an <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/AbstractProfileProfileAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractProfileProfileAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements common code for an <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template">Aligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an algorithm which computes an alignment <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a> from a list of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/CallablePairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.template">CallablePairwiseSequenceAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a concurrency wrapper for a <a href="../../../../alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template"><code>PairwiseSequenceAligner</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/CallablePairwiseSequenceScorer.html" title="class in org.biojava.nbio.alignment.template">CallablePairwiseSequenceScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a concurrency wrapper for a <a href="../../../../alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template"><code>PairwiseSequenceScorer</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/CallableProfileProfileAligner.html" title="class in org.biojava.nbio.alignment.template">CallableProfileProfileAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a concurrency wrapper for a <a href="../../../../alignment/template/ProfileProfileAligner.html" title="interface in org.biojava.nbio.alignment.template"><code>ProfileProfileAligner</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/GuideTreeNode.html" title="interface in org.biojava.nbio.alignment.template">GuideTreeNode</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for the node in a guide tree used during progressive multiple sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/HierarchicalClusterer.html" title="interface in org.biojava.nbio.alignment.template">HierarchicalClusterer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a clustering algorithm that converts a distance matrix into a tree.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/MatrixAligner.html" title="interface in org.biojava.nbio.alignment.template">MatrixAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> which builds a score matrix during computation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/PairInProfileScorer.html" title="interface in org.biojava.nbio.alignment.template">PairInProfileScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an algorithm which computes a score for a sequence alignment pair picked from an alignment <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an algorithm which computes a score for a pair of sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/PartitionRefiner.html" title="interface in org.biojava.nbio.alignment.template">PartitionRefiner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an algorithm which computes a new alignment <code>Profile</code> by splitting a current alignment and realigning.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/ProfileProfileAligner.html" title="interface in org.biojava.nbio.alignment.template">ProfileProfileAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <code>Profile</code>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/ProfileProfileScorer.html" title="interface in org.biojava.nbio.alignment.template">ProfileProfileScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an algorithm which computes a score for a pairing of alignment profiles.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/RescoreRefiner.html" title="interface in org.biojava.nbio.alignment.template">RescoreRefiner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an algorithm which computes a new alignment <code>Profile</code> by rescoring all pairs in an alignment and
 realigning.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../alignment/package-summary.html">org.biojava.nbio.core.alignment</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../alignment/package-summary.html">org.biojava.nbio.core.alignment</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/SimpleAlignedSequence.html" title="class in org.biojava.nbio.core.alignment">SimpleAlignedSequence</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a data structure for a <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/SimpleProfile.html" title="class in org.biojava.nbio.core.alignment">SimpleProfile</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a data structure for the results of sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/SimpleProfilePair.html" title="class in org.biojava.nbio.core.alignment">SimpleProfilePair</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a data structure for the results of the alignment of a pair of <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/SimpleSequencePair.html" title="class in org.biojava.nbio.core.alignment">SimpleSequencePair</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a data structure for the results of pairwise sequence alignment.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Classes in <a href="../../../alignment/package-summary.html">org.biojava.nbio.core.alignment</a> that implement <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/SimpleAlignedSequence.html" title="class in org.biojava.nbio.core.alignment">SimpleAlignedSequence</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a data structure for a <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../alignment/package-summary.html">org.biojava.nbio.core.alignment</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">AlignedSequence</a>&lt;<a href="../../../alignment/SimpleProfile.html" title="type parameter in SimpleProfile">S</a>,&#8203;<a href="../../../alignment/SimpleProfile.html" title="type parameter in SimpleProfile">C</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SimpleProfile.</span><code><span class="memberNameLink"><a href="../../../alignment/SimpleProfile.html#getAlignedSequences(S...)">getAlignedSequences</a></span>&#8203;(<a href="../../../alignment/SimpleProfile.html" title="type parameter in SimpleProfile">S</a>...&nbsp;sequences)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.alignment.template">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../alignment/template/package-summary.html">org.biojava.nbio.core.alignment.template</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../alignment/template/package-summary.html">org.biojava.nbio.core.alignment.template</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">AlignedSequence</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for a <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/MutableAlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">MutableAlignedSequence</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a mutable (editable) data structure for an <a href="../../../alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template"><code>AlignedSequence</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/MutableProfile.html" title="interface in org.biojava.nbio.core.alignment.template">MutableProfile</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a mutable (editable) data structure for a <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/MutableProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">MutableProfilePair</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a mutable (editable) data structure for a <a href="../../../alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template"><code>ProfilePair</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/MutableSequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">MutableSequencePair</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a mutable (editable) data structure for the results of pairwise sequence alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for the results of sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for the results of the alignment of a pair of <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/ProfileView.html" title="interface in org.biojava.nbio.core.alignment.template">ProfileView</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for a view of sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for the results of pairwise sequence alignment.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Subinterfaces of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../alignment/template/package-summary.html">org.biojava.nbio.core.alignment.template</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">AlignedSequence</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for a <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/MutableAlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">MutableAlignedSequence</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a mutable (editable) data structure for an <a href="../../../alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template"><code>AlignedSequence</code></a>.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../alignment/template/package-summary.html">org.biojava.nbio.core.alignment.template</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">AlignedSequence</a>&lt;<a href="../../../alignment/template/Profile.html" title="type parameter in Profile">S</a>,&#8203;<a href="../../../alignment/template/Profile.html" title="type parameter in Profile">C</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Profile.</span><code><span class="memberNameLink"><a href="../../../alignment/template/Profile.html#getAlignedSequences(S...)">getAlignedSequences</a></span>&#8203;(<a href="../../../alignment/template/Profile.html" title="type parameter in Profile">S</a>...&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Returns a <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a> containing some of the individual <a href="../../../alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template"><code>AlignedSequence</code></a>s of this alignment.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.search.io">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../search/io/package-summary.html">org.biojava.nbio.core.search.io</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../search/io/package-summary.html">org.biojava.nbio.core.search.io</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../search/io/Hsp.html" title="class in org.biojava.nbio.core.search.io">Hsp</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">This class models a search Hsp.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../search/io/package-summary.html">org.biojava.nbio.core.search.io</a> that return <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Hit.</span><code><span class="memberNameLink"><a href="../../../search/io/Hit.html#getHitSequence()">getHitSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">returns the reference to the original and whole sequence hit in the database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Result.</span><code><span class="memberNameLink"><a href="../../../search/io/Result.html#getQuerySequence()">getQuerySequence</a></span>()</code></th>
<td class="colLast">
<div class="block">returns the reference to the original and whole sequence used to query the database.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../search/io/package-summary.html">org.biojava.nbio.core.search.io</a> with type arguments of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ResultFactory.</span><code><span class="memberNameLink"><a href="../../../search/io/ResultFactory.html#setDatabaseReferences(java.util.List)">setDatabaseReferences</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Specify the collection of sequences objects used as database in the Search run.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ResultFactory.</span><code><span class="memberNameLink"><a href="../../../search/io/ResultFactory.html#setQueryReferences(java.util.List)">setQueryReferences</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Specify the collection of sequences objects used as queries in the Search run.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../search/io/package-summary.html">org.biojava.nbio.core.search.io</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../search/io/Hit.html#%3Cinit%3E(int,java.lang.String,java.lang.String,java.lang.String,int,java.util.List,org.biojava.nbio.core.sequence.template.Sequence)">Hit</a></span>&#8203;(int&nbsp;hitNum,
   <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;hitId,
   <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;hitDef,
   <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;hitAccession,
   int&nbsp;hitLen,
   <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../search/io/Hsp.html" title="class in org.biojava.nbio.core.search.io">Hsp</a>&gt;&nbsp;hsps,
   <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&nbsp;hitSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../search/io/Result.html#%3Cinit%3E(java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.util.HashMap,int,java.lang.String,java.lang.String,int,java.util.List,org.biojava.nbio.core.sequence.template.Sequence)">Result</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;program,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;version,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;reference,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;dbFile,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">HashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;&nbsp;programSpecificParameters,
      int&nbsp;iterationNumber,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;queryID,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;queryDef,
      int&nbsp;queryLength,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../search/io/Hit.html" title="class in org.biojava.nbio.core.search.io">Hit</a>&gt;&nbsp;hits,
      <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&nbsp;querySequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.search.io.blast">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../search/io/blast/package-summary.html">org.biojava.nbio.core.search.io.blast</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../search/io/blast/package-summary.html">org.biojava.nbio.core.search.io.blast</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../search/io/blast/BlastHitBuilder.html" title="class in org.biojava.nbio.core.search.io.blast">BlastHitBuilder</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BlastHitBuilder.</span><code><span class="memberNameLink"><a href="../../../search/io/blast/BlastHitBuilder.html#setHitSequence(org.biojava.nbio.core.sequence.template.Sequence)">setHitSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&nbsp;s)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../search/io/blast/BlastResultBuilder.html" title="class in org.biojava.nbio.core.search.io.blast">BlastResultBuilder</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BlastResultBuilder.</span><code><span class="memberNameLink"><a href="../../../search/io/blast/BlastResultBuilder.html#setQuerySequence(org.biojava.nbio.core.sequence.template.Sequence)">setQuerySequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&nbsp;s)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../search/io/blast/package-summary.html">org.biojava.nbio.core.search.io.blast</a> with type arguments of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BlastTabularParser.</span><code><span class="memberNameLink"><a href="../../../search/io/blast/BlastTabularParser.html#setDatabaseReferences(java.util.List)">setDatabaseReferences</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Intended for use with run module.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BlastXMLParser.</span><code><span class="memberNameLink"><a href="../../../search/io/blast/BlastXMLParser.html#setDatabaseReferences(java.util.List)">setDatabaseReferences</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&gt;&nbsp;sequences)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BlastTabularParser.</span><code><span class="memberNameLink"><a href="../../../search/io/blast/BlastTabularParser.html#setQueryReferences(java.util.List)">setQueryReferences</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Intended for use with run module.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BlastXMLParser.</span><code><span class="memberNameLink"><a href="../../../search/io/blast/BlastXMLParser.html#setQueryReferences(java.util.List)">setQueryReferences</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&gt;&nbsp;sequences)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../search/io/blast/package-summary.html">org.biojava.nbio.core.search.io.blast</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../search/io/blast/BlastResult.html#%3Cinit%3E(java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.util.HashMap,int,java.lang.String,java.lang.String,int,java.util.List,org.biojava.nbio.core.sequence.template.Sequence)">BlastResult</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;program,
           <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;version,
           <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;reference,
           <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;dbFile,
           <a href="https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">HashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;&nbsp;programSpecificParameters,
           int&nbsp;iterationNumber,
           <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;queryID,
           <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;queryDef,
           int&nbsp;queryLength,
           <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../search/io/Hit.html" title="class in org.biojava.nbio.core.search.io">Hit</a>&gt;&nbsp;hits,
           <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&nbsp;querySequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../package-summary.html">org.biojava.nbio.core.sequence</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../package-summary.html">org.biojava.nbio.core.sequence</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a minimal data structure for reading and writing a sequence alignment.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Classes in <a href="../../package-summary.html">org.biojava.nbio.core.sequence</a> that implement <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../BasicSequence.html" title="class in org.biojava.nbio.core.sequence">BasicSequence</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Bare bones version of the Sequence object to be used sparingly.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a></span></code></th>
<td class="colLast">
<div class="block">Represents a exon or coding sequence in a gene.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a></span></code></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span></code></th>
<td class="colLast">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a></span></code></th>
<td class="colLast">
<div class="block">A gene contains a collection of Exon sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span></code></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../RNASequence.html" title="class in org.biojava.nbio.core.sequence">RNASequence</a></span></code></th>
<td class="colLast">
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../StartCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StartCodonSequence</a></span></code></th>
<td class="colLast">
<div class="block">Used to map the start codon feature on a gene</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../StopCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StopCodonSequence</a></span></code></th>
<td class="colLast">
<div class="block">Used to map the stop codon sequence on a gene</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a></span></code></th>
<td class="colLast">
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.compound">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../compound/package-summary.html">org.biojava.nbio.core.sequence.compound</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../compound/package-summary.html">org.biojava.nbio.core.sequence.compound</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompoundSet.</span><code><span class="memberNameLink"><a href="../../compound/AminoAcidCompoundSet.html#isValidSequence(org.biojava.nbio.core.sequence.template.Sequence)">isValidSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.edits">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../edits/package-summary.html">org.biojava.nbio.core.sequence.edits</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../edits/package-summary.html">org.biojava.nbio.core.sequence.edits</a> that return <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.AbstractEdit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html#edit(org.biojava.nbio.core.sequence.template.Sequence)">edit</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.html" title="type parameter in Edit">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.html#edit(org.biojava.nbio.core.sequence.template.Sequence)">edit</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.html" title="type parameter in Edit">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.AbstractEdit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html#getEmptySequence(org.biojava.nbio.core.sequence.template.Sequence)">getEmptySequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">
<div class="block">Returns an empty sequence with the given compound set of the editing
 sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected abstract <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.AbstractEdit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">
<div class="block">Should return the 5-prime end of the given Sequence according to
 the edit.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Delete.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Delete.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Insert.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Insert.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Substitute.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Substitute.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.AbstractEdit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html#getTargetSequence(org.biojava.nbio.core.sequence.template.Sequence)">getTargetSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">
<div class="block">Returns the Sequence which is our edit.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected abstract <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.AbstractEdit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">
<div class="block">Should return the 3-prime end of the given Sequence according to
 the edit.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Delete.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Delete.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Insert.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Insert.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Substitute.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Substitute.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../edits/package-summary.html">org.biojava.nbio.core.sequence.edits</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.AbstractEdit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html#edit(org.biojava.nbio.core.sequence.template.Sequence)">edit</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.html" title="type parameter in Edit">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.html#edit(org.biojava.nbio.core.sequence.template.Sequence)">edit</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.html" title="type parameter in Edit">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.AbstractEdit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html#getEmptySequence(org.biojava.nbio.core.sequence.template.Sequence)">getEmptySequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">
<div class="block">Returns an empty sequence with the given compound set of the editing
 sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Substitute.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Substitute.html#getEnd(org.biojava.nbio.core.sequence.template.Sequence)">getEnd</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Must use this rather than the no-args getEnd as this can return
 -1 and the length of a sub is dependent on the length of the
 Sequence; we cannot assume 1:1 mapping between characters in a
 String and the number of compounds we will have to insert.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected abstract <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.AbstractEdit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">
<div class="block">Should return the 5-prime end of the given Sequence according to
 the edit.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Delete.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Delete.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Insert.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Insert.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Substitute.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Substitute.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.AbstractEdit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html#getTargetSequence(org.biojava.nbio.core.sequence.template.Sequence)">getTargetSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">
<div class="block">Returns the Sequence which is our edit.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected abstract <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.AbstractEdit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">
<div class="block">Should return the 3-prime end of the given Sequence according to
 the edit.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Delete.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Delete.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Insert.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Insert.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.Substitute.</span><code><span class="memberNameLink"><a href="../../edits/Edit.Substitute.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;&nbsp;editingSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Edit.AbstractEdit.</span><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html#setSequence(org.biojava.nbio.core.sequence.template.Sequence)">setSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../edits/package-summary.html">org.biojava.nbio.core.sequence.edits</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.Insert.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence,int)">Insert</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;&nbsp;sequence,
      int&nbsp;position)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.Insert.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence,int,int)">Insert</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;&nbsp;sequence,
      int&nbsp;start,
      int&nbsp;stop)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.Substitute.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence,int)">Substitute</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;&nbsp;sequence,
          int&nbsp;position)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../io/package-summary.html">org.biojava.nbio.core.sequence.io</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/FastaReader.html" title="class in org.biojava.nbio.core.sequence.io">FastaReader</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be the
 primary class used to read Fasta files</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/FastaWriter.html" title="class in org.biojava.nbio.core.sequence.io">FastaWriter</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The FastaWriter writes a collection of sequences to an outputStream.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenbankWriter.html" title="class in org.biojava.nbio.core.sequence.io">GenbankWriter</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaWriterHelper.</span><code><span class="memberNameLink"><a href="../../io/FastaWriterHelper.html#writeSequence(java.io.File,org.biojava.nbio.core.sequence.template.Sequence)">writeSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file,
             <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Write a sequence to a file</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaWriterHelper.</span><code><span class="memberNameLink"><a href="../../io/FastaWriterHelper.html#writeSequence(java.io.OutputStream,org.biojava.nbio.core.sequence.template.Sequence)">writeSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">OutputStream</a>&nbsp;outputStream,
             <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Write a sequence to OutputStream</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankWriterHelper.</span><code><span class="memberNameLink"><a href="../../io/GenbankWriterHelper.html#writeSequence(java.io.File,org.biojava.nbio.core.sequence.template.Sequence)">writeSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file,
             <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Write a sequence to a file</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankWriterHelper.</span><code><span class="memberNameLink"><a href="../../io/GenbankWriterHelper.html#writeSequence(java.io.OutputStream,org.biojava.nbio.core.sequence.template.Sequence)">writeSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">OutputStream</a>&nbsp;outputStream,
             <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Write a sequence to OutputStream</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with type arguments of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaWriterHelper.</span><code><span class="memberNameLink"><a href="../../io/FastaWriterHelper.html#writeSequences(java.io.OutputStream,java.util.Collection)">writeSequences</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">OutputStream</a>&nbsp;outputStream,
              <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Method which will write your given Sequences to the specified
 <a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink"><code>OutputStream</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankWriterHelper.</span><code><span class="memberNameLink"><a href="../../io/GenbankWriterHelper.html#writeSequences(java.io.OutputStream,java.util.Collection)">writeSequences</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">OutputStream</a>&nbsp;outputStream,
              <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Method which will write your given Sequences to the specified
 <a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink"><code>OutputStream</code></a>.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.io.template">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/template/FastaHeaderFormatInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">FastaHeaderFormatInterface</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/template/GenbankHeaderFormatInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">GenbankHeaderFormatInterface</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.io.util">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../io/util/package-summary.html">org.biojava.nbio.core.sequence.io.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/util/package-summary.html">org.biojava.nbio.core.sequence.io.util</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getGCGChecksum(java.util.List)">getGCGChecksum</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Calculates GCG checksum for entire list of sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getGCGChecksum(S)">getGCGChecksum</a></span>&#8203;(S&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Calculates GCG checksum for a given sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getGCGHeader(java.util.List)">getGCGHeader</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Assembles a GCG file header</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getIDFormat(java.util.List)">getIDFormat</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Creates format String for accession IDs</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.loader">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a> that implement <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/ArrayListProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">ArrayListProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/GenbankProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">GenbankProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/SequenceFileProxyLoader.html" title="class in org.biojava.nbio.core.sequence.loader">SequenceFileProxyLoader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">This class represents the storage container of a sequence stored in a fasta file where
 the initial parsing of the file we store the offset and length of the sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/StringProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">StringProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">An example of a ProxySequenceReader that is created from a String.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/UniprotProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">UniprotProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Pass in a Uniprot ID and this ProxySequenceReader when passed to a ProteinSequence will get the sequence data and other data elements
 associated with the ProteinSequence by Uniprot.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.location.template">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../location/template/package-summary.html">org.biojava.nbio.core.sequence.location.template</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../location/template/package-summary.html">org.biojava.nbio.core.sequence.location.template</a> that return <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#getRelevantSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getRelevantSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Location.</span><code><span class="memberNameLink"><a href="../../location/template/Location.html#getRelevantSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getRelevantSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Will return a SequenceReader object which offers a view of all resolved
 locations i.e. those locations which are not complex and define the
 true Sequence represented</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#getSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">If circular this will return the sequence represented by the sub
 locations joined.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Location.</span><code><span class="memberNameLink"><a href="../../location/template/Location.html#getSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Will return a SequenceReader object which represents the outer bounds
 of this Location</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#reverseSequence(org.biojava.nbio.core.sequence.template.Sequence)">reverseSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Reverses and (if possible) complements the Sequence so as to represent
 the reverse strand (if one exists).</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../location/template/package-summary.html">org.biojava.nbio.core.sequence.location.template</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#canComplement(org.biojava.nbio.core.sequence.template.Sequence)">canComplement</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Uses the Sequence's CompoundSet to decide if a compound can
 be assgined to ComplementCompound meaning it can complement</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#getRelevantSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getRelevantSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Location.</span><code><span class="memberNameLink"><a href="../../location/template/Location.html#getRelevantSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getRelevantSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Will return a SequenceReader object which offers a view of all resolved
 locations i.e. those locations which are not complex and define the
 true Sequence represented</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#getSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">If circular this will return the sequence represented by the sub
 locations joined.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Location.</span><code><span class="memberNameLink"><a href="../../location/template/Location.html#getSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Will return a SequenceReader object which represents the outer bounds
 of this Location</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#reverseSequence(org.biojava.nbio.core.sequence.template.Sequence)">reverseSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Reverses and (if possible) complements the Sequence so as to represent
 the reverse strand (if one exists).</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.storage">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> that implement <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/ArrayListSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">ArrayListSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Stores a Sequence as a collection of compounds in an ArrayList</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">BitSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">An implementation of the popular bit encodings.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/FourBitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">FourBitSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Four bit encoding of the bit formats.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">JoiningSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">This reader actually proxies onto multiple types of sequence in order
 to allow a number of sequence objects to act as if they are one sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/SingleCompoundSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">SingleCompoundSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">An implementation of the SequenceReader interface which for every
 call will return only 1 compound (given to it during construction; a String
 is also valid but will require a CompoundSet).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/TwoBitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">TwoBitSequenceReader</a>&lt;C extends <a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implementation of the 2bit encoding.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SingleCompoundSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/SingleCompoundSequenceReader.html#equals(org.biojava.nbio.core.sequence.template.Sequence)">equals</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&gt;&nbsp;o)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BitSequenceReader.BitArrayWorker.</span><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.BitArrayWorker.html#populate(org.biojava.nbio.core.sequence.template.Sequence)">populate</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Loops through the Compounds in a Sequence and passes them onto
 <a href="../../storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(C,int)"><code>BitSequenceReader.BitArrayWorker.setCompoundAt(Compound, int)</code></a></div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.BitArrayWorker.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence)">BitArrayWorker</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence)">FourBitArrayWorker</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/FourBitSequenceReader.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence)">FourBitSequenceReader</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.CompoundSet,org.biojava.nbio.core.sequence.template.Sequence...)">JoiningSequenceReader</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&nbsp;compoundSet,
                     <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;...&nbsp;sequences)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence...)">JoiningSequenceReader</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;...&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Allows creation of the store from Vargs Sequence<C> objects.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence)">TwoBitArrayWorker</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/TwoBitSequenceReader.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence)">TwoBitSequenceReader</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructor parameters in <a href="../../storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> with type arguments of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html#%3Cinit%3E(java.util.List)">JoiningSequenceReader</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Allows creation of the store from List<Sequence<C>>.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.CompoundSet,java.util.List)">JoiningSequenceReader</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&nbsp;compoundSet,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&gt;&nbsp;sequences)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.template">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../LightweightProfile.html" title="interface in org.biojava.nbio.core.sequence.template">LightweightProfile</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a minimal data structure for reading and writing a sequence alignment.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Subinterfaces of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Classes in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> that implement <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The base class for DNA, RNA and Protein sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceProxyView.html" title="class in org.biojava.nbio.core.sequence.template">SequenceProxyView</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> that return <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractCompoundTranslator.</span><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence)">createSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CompoundTranslator.</span><code><span class="memberNameLink"><a href="../CompoundTranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence)">createSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceProxyView.</span><code><span class="memberNameLink"><a href="../SequenceProxyView.html#getViewedSequence()">getViewedSequence</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../SequenceView.html" title="type parameter in SequenceView">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceView.</span><code><span class="memberNameLink"><a href="../SequenceView.html#getViewedSequence()">getViewedSequence</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#shuffle(org.biojava.nbio.core.sequence.template.Sequence)">shuffle</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Implements sequence shuffling by first materializing the given
 <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> into a <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a>, applying
 <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html?is-external=true#shuffle(java.util.List)" title="class or interface in java.util" class="externalLink"><code>Collections.shuffle(List)</code></a> and then returning the shuffled
 elements in a new instance of <code>SequenceBackingStore</code> which behaves
 as a <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> that return types with arguments of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractCompoundTranslator.</span><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CompoundTranslator.</span><code><span class="memberNameLink"><a href="../CompoundTranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractCompoundTranslator.</span><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html#workingListToSequences(java.util.List)">workingListToSequences</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;&gt;&nbsp;workingList)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#checksum(org.biojava.nbio.core.sequence.template.Sequence)">checksum</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Performs a simple CRC64 checksum on any given sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#countAT(org.biojava.nbio.core.sequence.template.Sequence)">countAT</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Returns the count of AT in the given sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#countCompounds(org.biojava.nbio.core.sequence.template.Sequence,C...)">countCompounds</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
              C...&nbsp;compounds)</code></th>
<td class="colLast">
<div class="block">For the given vargs of compounds this method counts the number of
 times those compounds appear in the given sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#countGC(org.biojava.nbio.core.sequence.template.Sequence)">countGC</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Returns the count of GC in the given sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html?is-external=true" title="class or interface in java.util" class="externalLink">Iterator</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#createIterator(org.biojava.nbio.core.sequence.template.Sequence)">createIterator</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Creates a simple sequence iterator which moves through a sequence going
 from 1 to the length of the Sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractCompoundTranslator.</span><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence)">createSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CompoundTranslator.</span><code><span class="memberNameLink"><a href="../CompoundTranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence)">createSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractCompoundTranslator.</span><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CompoundTranslator.</span><code><span class="memberNameLink"><a href="../CompoundTranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#createSubSequence(org.biojava.nbio.core.sequence.template.Sequence,int,int)">createSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                 int&nbsp;start,
                 int&nbsp;end)</code></th>
<td class="colLast">
<div class="block">Creates a simple sub sequence view delimited by the given start and end.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;C,&#8203;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#getComposition(org.biojava.nbio.core.sequence.template.Sequence)">getComposition</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Does a linear scan over the given Sequence and records the number of
 times each base appears.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;C,&#8203;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang" class="externalLink">Double</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#getDistribution(org.biojava.nbio.core.sequence.template.Sequence)">getDistribution</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Analogous to <a href="../SequenceMixin.html#getComposition(org.biojava.nbio.core.sequence.template.Sequence)"><code>SequenceMixin.getComposition(Sequence)</code></a> but returns the
 distribution of that <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> over the given sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#indexOf(org.biojava.nbio.core.sequence.template.Sequence,C)">indexOf</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
       C&nbsp;compound)</code></th>
<td class="colLast">
<div class="block">Performs a linear search of the given Sequence for the given compound.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#inverse(org.biojava.nbio.core.sequence.template.Sequence)">inverse</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">A method which attempts to do the right thing when is comes to a
 reverse/reverse complement</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractCompoundSet.</span><code><span class="memberNameLink"><a href="../AbstractCompoundSet.html#isValidSequence(org.biojava.nbio.core.sequence.template.Sequence)">isValidSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../AbstractCompoundSet.html" title="type parameter in AbstractCompoundSet">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CompoundSet.</span><code><span class="memberNameLink"><a href="../CompoundSet.html#isValidSequence(org.biojava.nbio.core.sequence.template.Sequence)">isValidSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../CompoundSet.html" title="type parameter in CompoundSet">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#lastIndexOf(org.biojava.nbio.core.sequence.template.Sequence,C)">lastIndexOf</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
           C&nbsp;compound)</code></th>
<td class="colLast">
<div class="block">Performs a reversed linear search of the given Sequence by wrapping
 it in a <a href="../../views/ReversedSequenceView.html" title="class in org.biojava.nbio.core.sequence.views"><code>ReversedSequenceView</code></a> and passing it into
 <a href="../SequenceMixin.html#indexOf(org.biojava.nbio.core.sequence.template.Sequence,C)"><code>SequenceMixin.indexOf(Sequence, Compound)</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#nonOverlappingKmers(org.biojava.nbio.core.sequence.template.Sequence,int)">nonOverlappingKmers</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                   int&nbsp;kmer)</code></th>
<td class="colLast">
<div class="block">Produces kmers of the specified size e.g.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#overlappingKmers(org.biojava.nbio.core.sequence.template.Sequence,int)">overlappingKmers</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                int&nbsp;kmer)</code></th>
<td class="colLast">
<div class="block">Used to generate overlapping k-mers such i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#sequenceEquality(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.core.sequence.template.Sequence)">sequenceEquality</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;source,
                <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;target)</code></th>
<td class="colLast">
<div class="block">A case-sensitive manner of comparing two sequence objects together.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#sequenceEqualityIgnoreCase(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.core.sequence.template.Sequence)">sequenceEqualityIgnoreCase</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;source,
                          <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;target)</code></th>
<td class="colLast">
<div class="block">A case-insensitive manner of comparing two sequence objects together.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#shuffle(org.biojava.nbio.core.sequence.template.Sequence)">shuffle</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Implements sequence shuffling by first materializing the given
 <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> into a <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a>, applying
 <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html?is-external=true#shuffle(java.util.List)" title="class or interface in java.util" class="externalLink"><code>Collections.shuffle(List)</code></a> and then returning the shuffled
 elements in a new instance of <code>SequenceBackingStore</code> which behaves
 as a <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#toList(org.biojava.nbio.core.sequence.template.Sequence)">toList</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">For the given <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> this will return a <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a> filled with
 the Compounds of that <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#toString(org.biojava.nbio.core.sequence.template.Sequence)">toString</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Shortcut to <a href="../SequenceMixin.html#toStringBuilder(org.biojava.nbio.core.sequence.template.Sequence)"><code>SequenceMixin.toStringBuilder(org.biojava.nbio.core.sequence.template.Sequence)</code></a>
 which calls toString() on the resulting object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html?is-external=true" title="class or interface in java.lang" class="externalLink">StringBuilder</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#toStringBuilder(org.biojava.nbio.core.sequence.template.Sequence)">toStringBuilder</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">For the given Sequence this will return a <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html?is-external=true" title="class or interface in java.lang" class="externalLink"><code>StringBuilder</code></a> object
 filled with the results of <code>Compound#toString()</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#write(java.lang.Appendable,org.biojava.nbio.core.sequence.template.Sequence)">write</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Appendable.html?is-external=true" title="class or interface in java.lang" class="externalLink">Appendable</a>&nbsp;appendable,
     <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Used as a way of sending a Sequence to a writer without the cost of
 converting to a full length String and then writing the data out</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceMixin.SequenceIterator.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence)">SequenceIterator</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../SequenceMixin.SequenceIterator.html" title="type parameter in SequenceMixin.SequenceIterator">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceProxyView.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence)">SequenceProxyView</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceProxyView.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence,java.lang.Integer,java.lang.Integer)">SequenceProxyView</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a>&gt;&nbsp;sequence,
                 <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&nbsp;bioStart,
                 <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&nbsp;bioEnd)</code></th>
<td class="colLast">
<div class="block">Main constructor for working with SequenceProxyViews</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.transcription">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../transcription/package-summary.html">org.biojava.nbio.core.sequence.transcription</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../transcription/package-summary.html">org.biojava.nbio.core.sequence.transcription</a> that return <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DNAToRNATranslator.</span><code><span class="memberNameLink"><a href="../../transcription/DNAToRNATranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence)">createSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DNAToRNATranslator.</span><code><span class="memberNameLink"><a href="../../transcription/DNAToRNATranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.core.sequence.transcription.Frame)">createSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence,
              <a href="../../transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a>&nbsp;frame)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.html#translate(org.biojava.nbio.core.sequence.template.Sequence)">translate</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna)</code></th>
<td class="colLast">
<div class="block">Quick method to let you go from a CDS to a Peptide quickly.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Frame.</span><code><span class="memberNameLink"><a href="../../transcription/Frame.html#wrap(org.biojava.nbio.core.sequence.template.Sequence)">wrap</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;incoming)</code></th>
<td class="colLast">
<div class="block">Optionally wraps a Sequence in a reverse complementing view (if the
 frame is on the reverse strand) and creates a sub sequence view if
 it is required.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../transcription/package-summary.html">org.biojava.nbio.core.sequence.transcription</a> that return types with arguments of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DNAToRNATranslator.</span><code><span class="memberNameLink"><a href="../../transcription/DNAToRNATranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">
<div class="block">Overloaded local version which delegates to an optional translator
 when told to (specified during construction).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RNAToAminoAcidTranslator.</span><code><span class="memberNameLink"><a href="../../transcription/RNAToAminoAcidTranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">
<div class="block">Performs the core conversion of RNA to Peptide.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="../../transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a>,&#8203;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.html#multipleFrameTranslation(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.core.sequence.transcription.Frame...)">multipleFrameTranslation</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna,
                        <a href="../../transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a>...&nbsp;frames)</code></th>
<td class="colLast">
<div class="block">A way of translating DNA in a number of frames</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../transcription/package-summary.html">org.biojava.nbio.core.sequence.transcription</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DNAToRNATranslator.</span><code><span class="memberNameLink"><a href="../../transcription/DNAToRNATranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence)">createSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DNAToRNATranslator.</span><code><span class="memberNameLink"><a href="../../transcription/DNAToRNATranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.core.sequence.transcription.Frame)">createSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence,
              <a href="../../transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a>&nbsp;frame)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DNAToRNATranslator.</span><code><span class="memberNameLink"><a href="../../transcription/DNAToRNATranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">
<div class="block">Overloaded local version which delegates to an optional translator
 when told to (specified during construction).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RNAToAminoAcidTranslator.</span><code><span class="memberNameLink"><a href="../../transcription/RNAToAminoAcidTranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code></th>
<td class="colLast">
<div class="block">Performs the core conversion of RNA to Peptide.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="../../transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a>,&#8203;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.html#multipleFrameTranslation(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.core.sequence.transcription.Frame...)">multipleFrameTranslation</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna,
                        <a href="../../transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a>...&nbsp;frames)</code></th>
<td class="colLast">
<div class="block">A way of translating DNA in a number of frames</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptionEngine.</span><code><span class="memberNameLink"><a href="../../transcription/TranscriptionEngine.html#translate(org.biojava.nbio.core.sequence.template.Sequence)">translate</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna)</code></th>
<td class="colLast">
<div class="block">Quick method to let you go from a CDS to a Peptide quickly.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Frame.</span><code><span class="memberNameLink"><a href="../../transcription/Frame.html#wrap(org.biojava.nbio.core.sequence.template.Sequence)">wrap</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;incoming)</code></th>
<td class="colLast">
<div class="block">Optionally wraps a Sequence in a reverse complementing view (if the
 frame is on the reverse strand) and creates a sub sequence view if
 it is required.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../RNASequence.html" title="class in org.biojava.nbio.core.sequence">RNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DNAToRNATranslator.</span><code><span class="memberNameLink"><a href="../../transcription/DNAToRNATranslator.html#wrapToRna(org.biojava.nbio.core.sequence.template.Sequence)">wrapToRna</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna)</code></th>
<td class="colLast">
<div class="block">Takes in the given DNA Sequence and returns an instance of RNASequence
 which is using <a href="../../views/RnaSequenceView.html" title="class in org.biojava.nbio.core.sequence.views"><code>RnaSequenceView</code></a> as a
 <a href="../ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template"><code>ProxySequenceReader</code></a>.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.views">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../views/package-summary.html">org.biojava.nbio.core.sequence.views</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../views/package-summary.html">org.biojava.nbio.core.sequence.views</a> that implement <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/ComplementSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ComplementSequenceView</a>&lt;C extends <a href="../ComplementCompound.html" title="interface in org.biojava.nbio.core.sequence.template">ComplementCompound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">For a given sequence this class will create a view over the top of it
 and for every request the code will return the complement of the underlying
 base e.g. base A will become base T</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/ReversedSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ReversedSequenceView</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">For a given sequence this class will return the base at the reversed
 position i.e. in a sequence of size 10, if you request base 2 you will get
 back the base at position 9.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/RnaSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">RnaSequenceView</a></span></code></th>
<td class="colLast">
<div class="block">Attempts to do on the fly translation of RNA by not requesting the compounds
 until asked.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../views/package-summary.html">org.biojava.nbio.core.sequence.views</a> that return <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../views/WindowedSequence.html" title="type parameter in WindowedSequence">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">WindowedSequence.</span><code><span class="memberNameLink"><a href="../../views/WindowedSequence.html#getBackingSequence()">getBackingSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Access the sequence which backs this window</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../views/package-summary.html">org.biojava.nbio.core.sequence.views</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/ComplementSequenceView.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence)">ComplementSequenceView</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../views/ComplementSequenceView.html" title="type parameter in ComplementSequenceView">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/ReversedSequenceView.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence)">ReversedSequenceView</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../views/ReversedSequenceView.html" title="type parameter in ReversedSequenceView">C</a>&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/RnaSequenceView.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence)">RnaSequenceView</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;sourceDna)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/RnaSequenceView.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.core.sequence.template.CompoundSet)">RnaSequenceView</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;sourceDna,
               <a href="../CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCompounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/WindowedSequence.html#%3Cinit%3E(org.biojava.nbio.core.sequence.template.Sequence,int)">WindowedSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../views/WindowedSequence.html" title="type parameter in WindowedSequence">C</a>&gt;&nbsp;sequence,
                int&nbsp;windowSize)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.util">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../util/package-summary.html">org.biojava.nbio.core.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../util/package-summary.html">org.biojava.nbio.core.util</a> that return <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceTools.</span><code><span class="memberNameLink"><a href="../../../util/SequenceTools.html#getSequenceFromString(java.lang.String)">getSequenceFromString</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.phylo">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../phylo/package-summary.html">org.biojava.nbio.phylo</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../phylo/package-summary.html">org.biojava.nbio.phylo</a> with type parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.msa.Msa</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ForesterWrapper.</span><code><span class="memberNameLink"><a href="../../../../phylo/ForesterWrapper.html#convert(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">convert</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">Convert a BioJava <a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence"><code>MultipleSequenceAlignment</code></a> to a forester
 <code>Msa</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#dissimilarityScore(org.biojava.nbio.core.sequence.MultipleSequenceAlignment,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">dissimilarityScore</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa,
                  <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;D&gt;&nbsp;M)</code></th>
<td class="colLast">
<div class="block">The dissimilarity score is the additive inverse of the similarity score
 (sum of scores) between two aligned sequences using a substitution model
 (Substitution Matrix).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#fractionalDissimilarity(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">fractionalDissimilarity</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">The fractional dissimilarity (D) is defined as the percentage of sites
 that differ between two aligned sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#fractionalDissimilarityScore(org.biojava.nbio.core.sequence.MultipleSequenceAlignment,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">fractionalDissimilarityScore</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa,
                            <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;D&gt;&nbsp;M)</code></th>
<td class="colLast">
<div class="block">The fractional dissimilarity score (Ds) is a relative measure of the
 dissimilarity between two aligned sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#kimuraDistance(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">kimuraDistance</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">The Kimura evolutionary distance (d) is a correction of the fractional
 dissimilarity (D) specially needed for large evolutionary distances.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#pamMLdistance(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">pamMLdistance</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">The PAM (Point Accepted Mutations) distance is a measure of evolutionary
 distance in protein sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#percentageIdentity(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">percentageIdentity</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">BioJava implementation for percentage of identity (PID).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#poissonDistance(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">poissonDistance</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">The Poisson (correction) evolutionary distance (d) is a function of the
 fractional dissimilarity (D), given by:</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#structuralDistance(double%5B%5D%5B%5D,double,double,double)">structuralDistance</a></span>&#8203;(double[][]&nbsp;rmsdMat,
                  double&nbsp;alpha,
                  double&nbsp;rmsdMax,
                  double&nbsp;rmsd0)</code></th>
<td class="colLast">
<div class="block">The structural distance (d<sub>S</sub>) uses the structural similarity
 (or dissimilarity) from a the structural alignment of two protein
 strutures.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../structure/package-summary.html">org.biojava.nbio.structure</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../structure/package-summary.html">org.biojava.nbio.structure</a> that return <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chain.</span><code><span class="memberNameLink"><a href="../../../../structure/Chain.html#getBJSequence()">getBJSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Converts the SEQRES groups of a Chain to a Biojava Sequence object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChainImpl.</span><code><span class="memberNameLink"><a href="../../../../structure/ChainImpl.html#getBJSequence()">getBJSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Converts the SEQRES groups of a Chain to a Biojava Sequence object.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.io">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../structure/io/package-summary.html">org.biojava.nbio.structure.io</a></h3>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../../structure/io/package-summary.html">org.biojava.nbio.structure.io</a> with type arguments of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../structure/io/FastaAFPChainConverter.html#fastaToAfpChain(org.biojava.nbio.core.alignment.template.SequencePair,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></span>&#8203;(<a href="../../../alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;,&#8203;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;alignment,
               <a href="../../../../structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code></th>
<td class="colLast">
<div class="block">Provided only for convenience.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.ws.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../ws/alignment/package-summary.html">org.biojava.nbio.ws.alignment</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../ws/alignment/package-summary.html">org.biojava.nbio.ws.alignment</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RemotePairwiseAlignmentService.</span><code><span class="memberNameLink"><a href="../../../../ws/alignment/RemotePairwiseAlignmentService.html#sendAlignmentRequest(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.ws.alignment.RemotePairwiseAlignmentProperties)">sendAlignmentRequest</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;seq,
                    <a href="../../../../ws/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojava.nbio.ws.alignment">RemotePairwiseAlignmentProperties</a>&nbsp;rpa)</code></th>
<td class="colLast">
<div class="block">Doing the actual analysis on the instantiated service using specified parameters and the RichSequence object</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.ws.alignment.qblast">
<!--   -->
</a>
<h3>Uses of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../ws/alignment/qblast/package-summary.html">org.biojava.nbio.ws.alignment.qblast</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../ws/alignment/qblast/package-summary.html">org.biojava.nbio.ws.alignment.qblast</a> with parameters of type <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NCBIQBlastService.</span><code><span class="memberNameLink"><a href="../../../../ws/alignment/qblast/NCBIQBlastService.html#sendAlignmentRequest(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.ws.alignment.RemotePairwiseAlignmentProperties)">sendAlignmentRequest</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;seq,
                    <a href="../../../../ws/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojava.nbio.ws.alignment">RemotePairwiseAlignmentProperties</a>&nbsp;rpa)</code></th>
<td class="colLast">
<div class="block">Converts given sequence to String and calls
 <a href="../../../../ws/alignment/qblast/NCBIQBlastService.html#sendAlignmentRequest(java.lang.String,org.biojava.nbio.ws.alignment.RemotePairwiseAlignmentProperties)"><code>NCBIQBlastService.sendAlignmentRequest(String, RemotePairwiseAlignmentProperties)</code></a></div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2019 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
