<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span>package org.biojava.nbio.structure.io;<a name="line.21"></a>
<span class="sourceLineNo">022</span><a name="line.22"></a>
<span class="sourceLineNo">023</span>import org.biojava.nbio.structure.ResidueNumber;<a name="line.23"></a>
<span class="sourceLineNo">024</span>import org.biojava.nbio.structure.Structure;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.nbio.structure.StructureException;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.structure.align.util.AtomCache;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.nbio.core.sequence.ProteinSequence;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.biojava.nbio.core.sequence.io.FastaReader;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface;<a name="line.31"></a>
<span class="sourceLineNo">032</span><a name="line.32"></a>
<span class="sourceLineNo">033</span>import java.io.File;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import java.io.FileNotFoundException;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import java.io.IOException;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import java.io.InputStream;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import java.util.LinkedHashMap;<a name="line.37"></a>
<span class="sourceLineNo">038</span><a name="line.38"></a>
<span class="sourceLineNo">039</span><a name="line.39"></a>
<span class="sourceLineNo">040</span>/**<a name="line.40"></a>
<span class="sourceLineNo">041</span> * Reads a protein sequence from a fasta file and attempts to match it to a<a name="line.41"></a>
<span class="sourceLineNo">042</span> * 3D structure. Any gaps ('-') in the fasta file are preserved as null atoms in<a name="line.42"></a>
<span class="sourceLineNo">043</span> * the output, allowing structural alignments to be read from fasta files.<a name="line.43"></a>
<span class="sourceLineNo">044</span> *<a name="line.44"></a>
<span class="sourceLineNo">045</span> * &lt;p&gt;Structures are loaded from an AtomCache. For this to work, the accession<a name="line.45"></a>
<span class="sourceLineNo">046</span> * for each protein should be parsed from the fasta header line into a form<a name="line.46"></a>
<span class="sourceLineNo">047</span> * understood by {@link AtomCache#getStructure(String)}.<a name="line.47"></a>
<span class="sourceLineNo">048</span> *<a name="line.48"></a>
<span class="sourceLineNo">049</span> * &lt;p&gt;Lowercase letters are sometimes used to specify unaligned residues.<a name="line.49"></a>
<span class="sourceLineNo">050</span> * This information can be preserved by using a CasePreservingSequenceCreator,<a name="line.50"></a>
<span class="sourceLineNo">051</span> * which allows the case of residues to be accessed through the<a name="line.51"></a>
<span class="sourceLineNo">052</span> * {@link ProteinSequence#getUserCollection()} method.<a name="line.52"></a>
<span class="sourceLineNo">053</span> *<a name="line.53"></a>
<span class="sourceLineNo">054</span> * @author Spencer Bliven<a name="line.54"></a>
<span class="sourceLineNo">055</span> *<a name="line.55"></a>
<span class="sourceLineNo">056</span> */<a name="line.56"></a>
<span class="sourceLineNo">057</span>public class FastaStructureParser {<a name="line.57"></a>
<span class="sourceLineNo">058</span><a name="line.58"></a>
<span class="sourceLineNo">059</span>        // inputs<a name="line.59"></a>
<span class="sourceLineNo">060</span>        private FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; reader;<a name="line.60"></a>
<span class="sourceLineNo">061</span>        private AtomCache cache;<a name="line.61"></a>
<span class="sourceLineNo">062</span><a name="line.62"></a>
<span class="sourceLineNo">063</span>        // cache processed data<a name="line.63"></a>
<span class="sourceLineNo">064</span>        private String[] accessions;<a name="line.64"></a>
<span class="sourceLineNo">065</span>        private ProteinSequence[] sequences;<a name="line.65"></a>
<span class="sourceLineNo">066</span>        private Structure[] structures;<a name="line.66"></a>
<span class="sourceLineNo">067</span>        private ResidueNumber[][] residues;<a name="line.67"></a>
<span class="sourceLineNo">068</span><a name="line.68"></a>
<span class="sourceLineNo">069</span>        public FastaStructureParser(InputStream is,<a name="line.69"></a>
<span class="sourceLineNo">070</span>                        SequenceHeaderParserInterface&lt;ProteinSequence, AminoAcidCompound&gt; headerParser,<a name="line.70"></a>
<span class="sourceLineNo">071</span>                        SequenceCreatorInterface&lt;AminoAcidCompound&gt; sequenceCreator,<a name="line.71"></a>
<span class="sourceLineNo">072</span>                        AtomCache cache)<a name="line.72"></a>
<span class="sourceLineNo">073</span>        {<a name="line.73"></a>
<span class="sourceLineNo">074</span>                this(new FastaReader&lt;ProteinSequence, AminoAcidCompound&gt;(<a name="line.74"></a>
<span class="sourceLineNo">075</span>                                is, headerParser, sequenceCreator),cache);<a name="line.75"></a>
<span class="sourceLineNo">076</span>        }<a name="line.76"></a>
<span class="sourceLineNo">077</span><a name="line.77"></a>
<span class="sourceLineNo">078</span>        public FastaStructureParser(File file,<a name="line.78"></a>
<span class="sourceLineNo">079</span>                        SequenceHeaderParserInterface&lt;ProteinSequence, AminoAcidCompound&gt; headerParser,<a name="line.79"></a>
<span class="sourceLineNo">080</span>                        SequenceCreatorInterface&lt;AminoAcidCompound&gt; sequenceCreator,<a name="line.80"></a>
<span class="sourceLineNo">081</span>                        AtomCache cache) throws FileNotFoundException<a name="line.81"></a>
<span class="sourceLineNo">082</span>        {<a name="line.82"></a>
<span class="sourceLineNo">083</span>                this(new FastaReader&lt;ProteinSequence, AminoAcidCompound&gt;(<a name="line.83"></a>
<span class="sourceLineNo">084</span>                                file, headerParser, sequenceCreator), cache);<a name="line.84"></a>
<span class="sourceLineNo">085</span>        }<a name="line.85"></a>
<span class="sourceLineNo">086</span><a name="line.86"></a>
<span class="sourceLineNo">087</span>        public FastaStructureParser(FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; reader,<a name="line.87"></a>
<span class="sourceLineNo">088</span>                        AtomCache cache) {<a name="line.88"></a>
<span class="sourceLineNo">089</span>                this.reader = reader;<a name="line.89"></a>
<span class="sourceLineNo">090</span>                this.cache = cache;<a name="line.90"></a>
<span class="sourceLineNo">091</span>                this.accessions = null;<a name="line.91"></a>
<span class="sourceLineNo">092</span>                this.sequences = null;<a name="line.92"></a>
<span class="sourceLineNo">093</span>                this.structures = null;<a name="line.93"></a>
<span class="sourceLineNo">094</span>                this.residues = null;<a name="line.94"></a>
<span class="sourceLineNo">095</span>        }<a name="line.95"></a>
<span class="sourceLineNo">096</span><a name="line.96"></a>
<span class="sourceLineNo">097</span><a name="line.97"></a>
<span class="sourceLineNo">098</span>        /**<a name="line.98"></a>
<span class="sourceLineNo">099</span>         * Parses the fasta file and loads it into memory.<a name="line.99"></a>
<span class="sourceLineNo">100</span>         *<a name="line.100"></a>
<span class="sourceLineNo">101</span>         * Information can be subsequently accessed through<a name="line.101"></a>
<span class="sourceLineNo">102</span>         * {@link #getSequences()},<a name="line.102"></a>
<span class="sourceLineNo">103</span>         * {@link #getStructures()},<a name="line.103"></a>
<span class="sourceLineNo">104</span>         * {@link #getResidues()}, and<a name="line.104"></a>
<span class="sourceLineNo">105</span>         * {@link #getAccessions()}.<a name="line.105"></a>
<span class="sourceLineNo">106</span>         *<a name="line.106"></a>
<span class="sourceLineNo">107</span>         * @throws IOException<a name="line.107"></a>
<span class="sourceLineNo">108</span>         * @throws StructureException<a name="line.108"></a>
<span class="sourceLineNo">109</span>         */<a name="line.109"></a>
<span class="sourceLineNo">110</span>        public void process() throws IOException, StructureException {<a name="line.110"></a>
<span class="sourceLineNo">111</span>                if(sequences == null) { // only process once, then return cached values<a name="line.111"></a>
<span class="sourceLineNo">112</span>                        LinkedHashMap&lt;String, ProteinSequence&gt; sequenceMap = reader.process();<a name="line.112"></a>
<span class="sourceLineNo">113</span><a name="line.113"></a>
<span class="sourceLineNo">114</span>                        sequences = sequenceMap.values().toArray(new ProteinSequence[0]);<a name="line.114"></a>
<span class="sourceLineNo">115</span>                        accessions = new String[sequences.length];<a name="line.115"></a>
<span class="sourceLineNo">116</span>                        structures = new Structure[sequences.length];<a name="line.116"></a>
<span class="sourceLineNo">117</span>                        residues = new ResidueNumber[sequences.length][];<a name="line.117"></a>
<span class="sourceLineNo">118</span><a name="line.118"></a>
<span class="sourceLineNo">119</span>                        // Match each sequence  to a series of PDB Residue numbers<a name="line.119"></a>
<span class="sourceLineNo">120</span>                        for(int i=0;i&lt;sequences.length;i++) {<a name="line.120"></a>
<span class="sourceLineNo">121</span>                                accessions[i] = sequences[i].getAccession().getID();<a name="line.121"></a>
<span class="sourceLineNo">122</span><a name="line.122"></a>
<span class="sourceLineNo">123</span>                                //System.out.println("Fetching "+accession);<a name="line.123"></a>
<span class="sourceLineNo">124</span>                                structures[i] = cache.getStructure(accessions[i]);<a name="line.124"></a>
<span class="sourceLineNo">125</span><a name="line.125"></a>
<span class="sourceLineNo">126</span>                                residues[i] = StructureSequenceMatcher.matchSequenceToStructure(sequences[i], structures[i]);<a name="line.126"></a>
<span class="sourceLineNo">127</span><a name="line.127"></a>
<span class="sourceLineNo">128</span>                                assert( residues[i].length == sequences[i].getLength());<a name="line.128"></a>
<span class="sourceLineNo">129</span>                        }<a name="line.129"></a>
<span class="sourceLineNo">130</span>                }<a name="line.130"></a>
<span class="sourceLineNo">131</span>        }<a name="line.131"></a>
<span class="sourceLineNo">132</span><a name="line.132"></a>
<span class="sourceLineNo">133</span><a name="line.133"></a>
<span class="sourceLineNo">134</span>        /**<a name="line.134"></a>
<span class="sourceLineNo">135</span>         * Gets the protein sequences read from the Fasta file.<a name="line.135"></a>
<span class="sourceLineNo">136</span>         * Returns null if {@link #process()} has not been called.<a name="line.136"></a>
<span class="sourceLineNo">137</span>         * @return An array ProteinSequences from<a name="line.137"></a>
<span class="sourceLineNo">138</span>         *  parsing the fasta file, or null if process() hasn't been called.<a name="line.138"></a>
<span class="sourceLineNo">139</span>         */<a name="line.139"></a>
<span class="sourceLineNo">140</span>        public ProteinSequence[] getSequences() {<a name="line.140"></a>
<span class="sourceLineNo">141</span>                return sequences;<a name="line.141"></a>
<span class="sourceLineNo">142</span>        }<a name="line.142"></a>
<span class="sourceLineNo">143</span><a name="line.143"></a>
<span class="sourceLineNo">144</span>        /**<a name="line.144"></a>
<span class="sourceLineNo">145</span>         * Gets the protein structures mapped from the Fasta file.<a name="line.145"></a>
<span class="sourceLineNo">146</span>         * Returns null if {@link #process()} has not been called.<a name="line.146"></a>
<span class="sourceLineNo">147</span>         * @return An array of Structures for each protein<a name="line.147"></a>
<span class="sourceLineNo">148</span>         *  in the fasta file, or null if process() hasn't been called.<a name="line.148"></a>
<span class="sourceLineNo">149</span>         */<a name="line.149"></a>
<span class="sourceLineNo">150</span>        public Structure[] getStructures() {<a name="line.150"></a>
<span class="sourceLineNo">151</span>                return structures;<a name="line.151"></a>
<span class="sourceLineNo">152</span>        }<a name="line.152"></a>
<span class="sourceLineNo">153</span><a name="line.153"></a>
<span class="sourceLineNo">154</span>        /**<a name="line.154"></a>
<span class="sourceLineNo">155</span>         * For each residue in the fasta file, return the ResidueNumber in the<a name="line.155"></a>
<span class="sourceLineNo">156</span>         * corresponding structure. If the residue cannot be found in the structure,<a name="line.156"></a>
<span class="sourceLineNo">157</span>         * that entry will be null. This can happen if that residue was not included<a name="line.157"></a>
<span class="sourceLineNo">158</span>         * in the PDB file (eg disordered residues), if the fasta sequence does not<a name="line.158"></a>
<span class="sourceLineNo">159</span>         * match the PDB sequence, or if errors occur during the matching process.<a name="line.159"></a>
<span class="sourceLineNo">160</span>         * @return A 2D array of ResidueNumbers, or null if process() hasn't been called.<a name="line.160"></a>
<span class="sourceLineNo">161</span>         * @see StructureSequenceMatcher#matchSequenceToStructure(ProteinSequence, Structure)<a name="line.161"></a>
<span class="sourceLineNo">162</span>         */<a name="line.162"></a>
<span class="sourceLineNo">163</span>        public ResidueNumber[][] getResidues() {<a name="line.163"></a>
<span class="sourceLineNo">164</span>                return residues;<a name="line.164"></a>
<span class="sourceLineNo">165</span>        }<a name="line.165"></a>
<span class="sourceLineNo">166</span><a name="line.166"></a>
<span class="sourceLineNo">167</span>        /**<a name="line.167"></a>
<span class="sourceLineNo">168</span>         * Gets the protein accessions mapped from the Fasta file.<a name="line.168"></a>
<span class="sourceLineNo">169</span>         * Returns null if {@link #process()} has not been called.<a name="line.169"></a>
<span class="sourceLineNo">170</span>         * @return An array of Structures for each protein<a name="line.170"></a>
<span class="sourceLineNo">171</span>         *  in the fasta file, or null if process() hasn't been called.<a name="line.171"></a>
<span class="sourceLineNo">172</span>         */<a name="line.172"></a>
<span class="sourceLineNo">173</span>        public String[] getAccessions() {<a name="line.173"></a>
<span class="sourceLineNo">174</span>                return accessions;<a name="line.174"></a>
<span class="sourceLineNo">175</span>        }<a name="line.175"></a>
<span class="sourceLineNo">176</span>}<a name="line.176"></a>




























































</pre>
</div>
</body>
</html>
