<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.11) on Fri Oct 29 16:26:12 PDT 2021 -->
<title>org.biojava.nbio.structure (biojava 6.0.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2021-10-29">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="org.biojava.nbio.structure (biojava 6.0.0 API)";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li class="navBarCell1Rev">Package</li>
<li>Class</li>
<li><a href="package-use.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h1 title="Package" class="title">Package&nbsp;org.biojava.nbio.structure</h1>
</div>
<div class="contentContainer">
<section role="region"><a id="package.description">
<!--   -->
</a>
<div class="block"><BODY>
 <P>
 Interfaces and classes for protein structure (PDB).
 </P>
 <p>
 See also the <a href="https://github.com/biojava/biojava3-tutorial/blob/master/structure/README.md">BioJava 3 tutorial</a> for more information on the protein structure modules.
 </p>
 <h2>Parse PDB files</h2>
 To load a PDB file see the <a href="io/PDBFileReader.html">PDBFileReader</a> class in the
 <a href="io/package-summary.html">IO subpackage</a>.
 <h2>Parse mmCif files</h2>
 To laod a mmCif file see the <a href="io/MMCIFFileReader.html">MMCIFFileReader</a> class.
 <h2>The Structure object</h2>
 The <a href="Structure.html">Structure</a> object allows to access the PDB header information as well
 as to the data from the ATOM records. The header information is currently available through the following objects:
 <ul>
 <li><a href="PDBHeader.html">PDBHeader</a></li>
 <li><a href="DBRef.html">DBRef</a></li>
 <li><a href="Compound.html">Compound</a></li>
 </ul>
 The structure object provides access to the data from the ATOM records through
 a hierarchy of sub-object:
 <pre>
 <a href="Structure.html">Structure</a>
 |
 <a href="Chain.html">Chain</a>
 |
 <a href="Group.html">Group</a>
 |
 <a href="Atom.html">Atom</a>
 </pre>
 Learn more <a href="http://biojava.org/wiki/BioJava:CookBook:PDB:groups">how to work with groups</a>.
 <h2>Other Features</h2>
 <ul>
 <li>Calculate <a href="align/ce/CeMain.html">protein structure alignments with CE and FATCAT.</a></li>
 <li><a href="http://www.spice-3d.org/hibernatePDB/" target="_top">Serialize PDB files to databases using Hibernate</a></li>
 <li><a href="Calc.html">Tools for performing calculations</a>
 <li><a href="gui/BiojavaJmol.html">Display structures in Jmol</a></li>
 </ul>
 <p>
 For more documentation on how to work with the Structure API please
 see <a href="http://biojava.org/wiki/BioJava:CookBook#Protein_Structure" target="_top">
 http://biojava.org/wiki/BioJava:CookBook#Protein_Structure</a>
 </p>
 </BODY></div>
<dl>
<dt><span class="simpleTagLabel">Since:</span></dt>
<dd>1.5</dd>
</dl>
</section>
<ul class="blockList">
<li class="blockList">
<table class="typeSummary">
<caption><span>Interface Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a></th>
<td class="colLast">
<div class="block">
 A <a href="Group.html" title="interface in org.biojava.nbio.structure"><code>Group</code></a> that represents an AminoAcid.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></th>
<td class="colLast">
<div class="block">A simple interface for an Atom.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="AtomPositionMap.GroupMatcher.html" title="interface in org.biojava.nbio.structure">AtomPositionMap.GroupMatcher</a></th>
<td class="colLast">
<div class="block">Used as a Predicate to indicate whether a particular Atom should be mapped</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="Bond.html" title="interface in org.biojava.nbio.structure">Bond</a></th>
<td class="colLast">
<div class="block">A simple bond -- it stores information about two atoms as well as information
 about its bond order.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="Chain.html" title="interface in org.biojava.nbio.structure">Chain</a></th>
<td class="colLast">
<div class="block">
 Defines the interface for a Chain.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></th>
<td class="colLast">
<div class="block">This is the data structure for a single Group of atoms.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="PDBRecord.html" title="interface in org.biojava.nbio.structure">PDBRecord</a></th>
<td class="colLast">
<div class="block">An interface implemented by all classes that represent PDB records.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a></th>
<td class="colLast">
<div class="block">Created by andreas on 9/11/15.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></th>
<td class="colLast">
<div class="block">Interface for a structure object.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a></th>
<td class="colLast">
<div class="block">An identifier that <em>uniquely</em> identifies a whole <a href="Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> or
 arbitrary substructure.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<table class="typeSummary">
<caption><span>Class Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="AminoAcidImpl.html" title="class in org.biojava.nbio.structure">AminoAcidImpl</a></th>
<td class="colLast">
<div class="block">AminoAcid inherits most from Hetatom.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="AtomImpl.html" title="class in org.biojava.nbio.structure">AtomImpl</a></th>
<td class="colLast">
<div class="block">Implementation of an Atom of a PDB file.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="AtomIterator.html" title="class in org.biojava.nbio.structure">AtomIterator</a></th>
<td class="colLast">
<div class="block">an iterator over all atoms of a structure / group.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="AtomPositionMap.html" title="class in org.biojava.nbio.structure">AtomPositionMap</a></th>
<td class="colLast">
<div class="block">A map from <a href="ResidueNumber.html" title="class in org.biojava.nbio.structure"><code>ResidueNumbers</code></a> to ATOM record positions in a PDB file.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="AugmentedResidueRange.html" title="class in org.biojava.nbio.structure">AugmentedResidueRange</a></th>
<td class="colLast">
<div class="block">Created by douglas on 1/23/15.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="Author.html" title="class in org.biojava.nbio.structure">Author</a></th>
<td class="colLast">
<div class="block">Describes author attributes for author information in a PDB file.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="BioAssemblyIdentifier.html" title="class in org.biojava.nbio.structure">BioAssemblyIdentifier</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="BondImpl.html" title="class in org.biojava.nbio.structure">BondImpl</a></th>
<td class="colLast">
<div class="block">A simple bond -- it stores information about two atoms as well as information
 about its bond order.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="Calc.html" title="class in org.biojava.nbio.structure">Calc</a></th>
<td class="colLast">
<div class="block">Utility operations on Atoms, AminoAcids, Matrices, Point3d, etc.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="ChainImpl.html" title="class in org.biojava.nbio.structure">ChainImpl</a></th>
<td class="colLast">
<div class="block">A Chain in a PDB file.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="DatabasePDBRevRecord.html" title="class in org.biojava.nbio.structure">DatabasePDBRevRecord</a></th>
<td class="colLast">
<div class="block">Represents revision records for use by <a href="PDBHeader.html" title="class in org.biojava.nbio.structure"><code>PDBHeader</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="DBRef.html" title="class in org.biojava.nbio.structure">DBRef</a></th>
<td class="colLast">
<div class="block">A class to represent database cross references.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="EntityInfo.html" title="class in org.biojava.nbio.structure">EntityInfo</a></th>
<td class="colLast">
<div class="block">An object to contain the info from the PDB header for a Molecule.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="GroupIterator.html" title="class in org.biojava.nbio.structure">GroupIterator</a></th>
<td class="colLast">
<div class="block">An iterator over all groups of a structure.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="HetatomImpl.html" title="class in org.biojava.nbio.structure">HetatomImpl</a></th>
<td class="colLast">
<div class="block">Generic Implementation of a Group interface.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="JournalArticle.html" title="class in org.biojava.nbio.structure">JournalArticle</a></th>
<td class="colLast">
<div class="block">PDB-specific</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="Model.html" title="class in org.biojava.nbio.structure">Model</a></th>
<td class="colLast">
<div class="block">An internal utility class for StructureImpl to make it easier to manage poly and nonpoly chains.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="Mutator.html" title="class in org.biojava.nbio.structure">Mutator</a></th>
<td class="colLast">
<div class="block">A class that can change one amino acid to another.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="NucleotideImpl.html" title="class in org.biojava.nbio.structure">NucleotideImpl</a></th>
<td class="colLast">
<div class="block">A nucleotide group is almost the same as a Hetatm group.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="PassthroughIdentifier.html" title="class in org.biojava.nbio.structure">PassthroughIdentifier</a></th>
<td class="colLast">
<div class="block">A stub StructureIdentifier, representing the full structure in all cases.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="PDBCrystallographicInfo.html" title="class in org.biojava.nbio.structure">PDBCrystallographicInfo</a></th>
<td class="colLast">
<div class="block">A class to hold crystallographic information about a PDB structure.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="PDBHeader.html" title="class in org.biojava.nbio.structure">PDBHeader</a></th>
<td class="colLast">
<div class="block">A class that contains PDB Header information.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="PdbId.html" title="class in org.biojava.nbio.structure">PdbId</a></th>
<td class="colLast">
<div class="block">A wrapper class for the PDB identifier.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="PDBStatus.html" title="class in org.biojava.nbio.structure">PDBStatus</a></th>
<td class="colLast">
<div class="block">Methods for getting the status of a PDB file (current, removed, unreleased)
 and for accessing different versions of the structure.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a></th>
<td class="colLast">
<div class="block">Everything that is needed to uniquely describe a residue position</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="ResidueRange.html" title="class in org.biojava.nbio.structure">ResidueRange</a></th>
<td class="colLast">
<div class="block">A chainName, a start residue, and an end residue.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="ResidueRangeAndLength.html" title="class in org.biojava.nbio.structure">ResidueRangeAndLength</a></th>
<td class="colLast">
<div class="block">A chain, a start residue, and an end residue.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="SeqMisMatchImpl.html" title="class in org.biojava.nbio.structure">SeqMisMatchImpl</a></th>
<td class="colLast">
<div class="block">Created by andreas on 9/11/15.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="Site.html" title="class in org.biojava.nbio.structure">Site</a></th>
<td class="colLast">
<div class="block">Holds the data of sites presented in PDB files.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="StandardAminoAcid.html" title="class in org.biojava.nbio.structure">StandardAminoAcid</a></th>
<td class="colLast">
<div class="block">A class that provides a set of standard amino acids.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="StructureImpl.html" title="class in org.biojava.nbio.structure">StructureImpl</a></th>
<td class="colLast">
<div class="block">Implementation of a PDB Structure.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="StructureIO.html" title="class in org.biojava.nbio.structure">StructureIO</a></th>
<td class="colLast">
<div class="block">A class that provides static access methods for easy lookup of protein structure related components</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="StructureTools.html" title="class in org.biojava.nbio.structure">StructureTools</a></th>
<td class="colLast">
<div class="block">A class that provides some tool methods.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="SubstructureIdentifier.html" title="class in org.biojava.nbio.structure">SubstructureIdentifier</a></th>
<td class="colLast">
<div class="block">This is the canonical way to identify a part of a structure.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="URLIdentifier.html" title="class in org.biojava.nbio.structure">URLIdentifier</a></th>
<td class="colLast">
<div class="block">Represents a structure loaded from a URL (including a file URL)

 A few custom query parameters are supported:

 
 <tt>format=[pdb|cif]</tt> Specify the file format (will otherwise be
     guessed from the extension)
 <tt>pdbId=[String]</tt> Specify the PDB ID (also guessed from the filename)
 <tt>chainID=[String]</tt> A single chain from the structure
 <tt>residues=[String]</tt> Residue ranges, in a form understood by
     <a href="SubstructureIdentifier.html" title="class in org.biojava.nbio.structure"><code>SubstructureIdentifier</code></a>
 </div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<table class="typeSummary">
<caption><span>Enum Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Enum</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="BondType.html" title="enum in org.biojava.nbio.structure">BondType</a></th>
<td class="colLast">
<div class="block">Work in progress - NOT final!</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="Element.html" title="enum in org.biojava.nbio.structure">Element</a></th>
<td class="colLast">
<div class="block">Element is an enumeration of the elements of the periodic table.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="ElementType.html" title="enum in org.biojava.nbio.structure">ElementType</a></th>
<td class="colLast">
<div class="block">ElementType is an enumeration of the types of elements found in the periodic table.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="EntityType.html" title="enum in org.biojava.nbio.structure">EntityType</a></th>
<td class="colLast">
<div class="block">The type of entity (polymer, non-polymer, water, macrolide)
 as defined in the mmCIF dictionary: <a href="http://mmcif.wwpdb.org/dictionaries/mmcif_pdbx_v40.dic/Items/_entity.type.html"></a></div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="ExperimentalTechnique.html" title="enum in org.biojava.nbio.structure">ExperimentalTechnique</a></th>
<td class="colLast">
<div class="block">An enum to represent the experimental technique of a PDB structure</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a></th>
<td class="colLast">
<div class="block">This contains basic categories for Group types.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="HetatomImpl.PerformanceBehavior.html" title="enum in org.biojava.nbio.structure">HetatomImpl.PerformanceBehavior</a></th>
<td class="colLast">
<div class="block">Behaviors for how to balance memory vs. performance.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="PDBStatus.Status.html" title="enum in org.biojava.nbio.structure">PDBStatus.Status</a></th>
<td class="colLast">
<div class="block">Represents a simplified 3 state status of PDB IDs.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<table class="typeSummary">
<caption><span>Exception Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Exception</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></th>
<td class="colLast">
<div class="block">An exception during the parsing of a PDB file.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li class="navBarCell1Rev">Package</li>
<li>Class</li>
<li><a href="package-use.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2021 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
