<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.16) on Fri Oct 28 16:15:48 PDT 2022 -->
<title>Chain (biojava 6.1.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2022-10-28">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Chain (biojava 6.1.0 API)";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":6,"i1":6,"i2":6,"i3":6,"i4":6,"i5":6,"i6":6,"i7":6,"i8":6,"i9":6,"i10":6,"i11":6,"i12":6,"i13":6,"i14":6,"i15":6,"i16":6,"i17":6,"i18":6,"i19":6,"i20":6,"i21":6,"i22":6,"i23":6,"i24":6,"i25":6,"i26":6,"i27":6,"i28":6,"i29":6,"i30":6,"i31":6,"i32":6,"i33":6,"i34":6,"i35":6,"i36":6};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],4:["t3","Abstract Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Chain.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">org.biojava.nbio.structure</a></div>
<h2 title="Interface Chain" class="title">Interface Chain</h2>
</div>
<div class="contentContainer">
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Superinterfaces:</dt>
<dd><code><a href="https://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io" class="externalLink">Serializable</a></code></dd>
</dl>
<dl>
<dt>All Known Implementing Classes:</dt>
<dd><code><a href="ChainImpl.html" title="class in org.biojava.nbio.structure">ChainImpl</a></code></dd>
</dl>
<hr>
<pre>public interface <a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.51">Chain</a>
extends <a href="https://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io" class="externalLink">Serializable</a></pre>
<div class="block"><p>
 Defines the interface for a Chain. A Chain corresponds to a Chain in a PDB file.
 A chain consists of a list of <a href="Group.html" title="interface in org.biojava.nbio.structure"><code>Group</code></a> objects. A Group can either be
 an <a href="AminoAcid.html" title="interface in org.biojava.nbio.structure"><code>AminoAcid</code></a>, <a href="HetatomImpl.html" title="class in org.biojava.nbio.structure"><code>Hetatom</code></a> or <a href="NucleotideImpl.html" title="class in org.biojava.nbio.structure"><code>Nucleotide</code></a>.
 </p>

 <p>
 The BioJava API provides access to both the ATOM and SEQRES records in a PDB file.
 During parsing of a PDB file it aligns the ATOM and SEQRES groups and joins them.
 The SEQRES sequence can be accessed via  <a href="#getSeqResGroups()"><code>getSeqResGroups()</code></a> and the
 ATOM groups via <a href="#getAtomGroups()"><code>getAtomGroups()</code></a>. Groups that have been observed
 (i.e. they are in the ATOM records) can be detected by <a href="Group.html" title="interface in org.biojava.nbio.structure"><code>Group</code></a>.has3D()
  </p></div>
<dl>
<dt><span class="simpleTagLabel">Since:</span></dt>
<dd>1.4</dd>
<dt><span class="simpleTagLabel">Version:</span></dt>
<dd>%I% %G%</dd>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Andreas Prlic</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t3" class="tableTab"><span><a href="javascript:show(4);">Abstract Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addGroup(org.biojava.nbio.structure.Group)">addGroup</a></span>&#8203;(<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group)</code></th>
<td class="colLast">
<div class="block">Add a group to the list of ATOM record group of this chain.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#clone()">clone</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns an identical copy of this Chain.</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAtomGroup(int)">getAtomGroup</a></span>&#8203;(int&nbsp;position)</code></th>
<td class="colLast">
<div class="block">Return the Group at given position,
 from within Groups with observed density in the chain, i.e.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAtomGroups()">getAtomGroups</a></span>()</code></th>
<td class="colLast">
<div class="block">Return all Groups with observed density in the chain, i.e.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAtomGroups(org.biojava.nbio.structure.GroupType)">getAtomGroups</a></span>&#8203;(<a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;type)</code></th>
<td class="colLast">
<div class="block">Return a List of all (observed) Groups of a special type, one of: <a href="GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a>,
 <a href="GroupType.html#HETATM"><code>GroupType.HETATM</code></a> or <a href="GroupType.html#NUCLEOTIDE"><code>GroupType.NUCLEOTIDE</code></a>.</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAtomLength()">getAtomLength</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the number of Groups with observed density in the chain, i.e.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAtomSequence()">getAtomSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the sequence of amino acids as it has been provided in the ATOM records.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code><a href="../core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBJSequence()">getBJSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Converts the SEQRES groups of a Chain to a Biojava Sequence object.</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code><a href="EntityInfo.html" title="class in org.biojava.nbio.structure">EntityInfo</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getEntityInfo()">getEntityInfo</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the EntityInfo for this chain.</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code><a href="EntityType.html" title="enum in org.biojava.nbio.structure">EntityType</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getEntityType()">getEntityType</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the EntityType of this chain.</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGroupByPDB(org.biojava.nbio.structure.ResidueNumber)">getGroupByPDB</a></span>&#8203;(<a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;resNum)</code></th>
<td class="colLast">
<div class="block">Get a group by its PDB residue numbering.</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,org.biojava.nbio.structure.ResidueNumber)">getGroupsByPDB</a></span>&#8203;(<a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
              <a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd)</code></th>
<td class="colLast">
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,org.biojava.nbio.structure.ResidueNumber,boolean)">getGroupsByPDB</a></span>&#8203;(<a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
              <a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd,
              boolean&nbsp;ignoreMissing)</code></th>
<td class="colLast">
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getId()">getId</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the 'private' asymId (internal chain IDs in mmCif) for this chain.</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getName()">getName</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the 'public' authId (chain ID in PDB file)</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code><a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getPredominantGroupType()">getPredominantGroupType</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the predominant <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> for a given Chain, following these
 rules: if the ratio of number of residues of a certain
 <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> to total non-water residues is above the threshold
 <a href="StructureTools.html#RATIO_RESIDUES_TO_TOTAL">0.95</a>, then that <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> is
 returned if there is no <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> that is above the
 threshold then the <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> with most members is chosen, logging
 it</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSeqMisMatches()">getSeqMisMatches</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets annotated sequence mismatches for this chain.</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSeqResGroup(int)">getSeqResGroup</a></span>&#8203;(int&nbsp;position)</code></th>
<td class="colLast">
<div class="block">Return the Group at given position,
 from within groups in the SEQRES records of the chain, i.e.</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSeqResGroups()">getSeqResGroups</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns a list of all groups in SEQRES records of the chain, i.e.</div>
</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSeqResGroups(org.biojava.nbio.structure.GroupType)">getSeqResGroups</a></span>&#8203;(<a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;type)</code></th>
<td class="colLast">
<div class="block">Returns a List of all SEQRES groups of a special type, one of: <a href="GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a>,
 <a href="GroupType.html#HETATM"><code>GroupType.HETATM</code></a> or <a href="GroupType.html#NUCLEOTIDE"><code>GroupType.NUCLEOTIDE</code></a>.</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSeqResLength()">getSeqResLength</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the number of groups in the SEQRES records of the chain, i.e.</div>
</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSeqResSequence()">getSeqResSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the PDB SEQRES sequence as a one-letter sequence string.</div>
</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code><a href="Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getStructure()">getStructure</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the parent Structure of this chain.</div>
</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isNucleicAcid()">isNucleicAcid</a></span>()</code></th>
<td class="colLast">
<div class="block">Tell whether given chain is DNA or RNA</div>
</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isProtein()">isProtein</a></span>()</code></th>
<td class="colLast">
<div class="block">Tell whether given chain is a protein chain</div>
</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isPureNonPolymer()">isPureNonPolymer</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns true if the given chain is composed of non-polymeric (including water) groups only.</div>
</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#isWaterOnly()">isWaterOnly</a></span>()</code></th>
<td class="colLast">
<div class="block">Tests if a chain is consisting of water molecules only</div>
</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setAtomGroups(java.util.List)">setAtomGroups</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;groups)</code></th>
<td class="colLast">
<div class="block">Set all Groups with observed density in the chain, i.e.</div>
</td>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setEntityInfo(org.biojava.nbio.structure.EntityInfo)">setEntityInfo</a></span>&#8203;(<a href="EntityInfo.html" title="class in org.biojava.nbio.structure">EntityInfo</a>&nbsp;entityInfo)</code></th>
<td class="colLast">
<div class="block">Sets the Entity information</div>
</td>
</tr>
<tr id="i29" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setId(java.lang.String)">setId</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;asymId)</code></th>
<td class="colLast">
<div class="block">Set the 'private' asymId (internal chain IDs in mmCif) for this chain.</div>
</td>
</tr>
<tr id="i30" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setName(java.lang.String)">setName</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;authId)</code></th>
<td class="colLast">
<div class="block">Set the 'public' authId (chain ID in PDB file)</div>
</td>
</tr>
<tr id="i31" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setSeqMisMatches(java.util.List)">setSeqMisMatches</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a>&gt;&nbsp;seqMisMatches)</code></th>
<td class="colLast">
<div class="block">Sets annotated sequence mismatches for this chain.</div>
</td>
</tr>
<tr id="i32" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setSeqResGroups(java.util.List)">setSeqResGroups</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;seqResGroups)</code></th>
<td class="colLast">
<div class="block">Sets the list of SeqResGroups for this chain.</div>
</td>
</tr>
<tr id="i33" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setStructure(org.biojava.nbio.structure.Structure)">setStructure</a></span>&#8203;(<a href="Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;parent)</code></th>
<td class="colLast">
<div class="block">Sets the back-reference to its parent Structure.</div>
</td>
</tr>
<tr id="i34" class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#toMMCIF()">toMMCIF</a></span>()</code></th>
<td class="colLast">
<div class="block">Convert this Chain to a String in mmCIF format</div>
</td>
</tr>
<tr id="i35" class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#toPDB()">toPDB</a></span>()</code></th>
<td class="colLast">
<div class="block">Convert this Chain to a String in PDB format</div>
</td>
</tr>
<tr id="i36" class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#toString()">toString</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="clone()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>clone</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.57">clone</a>()</pre>
<div class="block">Returns an identical copy of this Chain.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>an identical copy of this Chain</dd>
</dl>
</li>
</ul>
<a id="addGroup(org.biojava.nbio.structure.Group)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addGroup</h4>
<pre class="methodSignature">void&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.65">addGroup</a>&#8203;(<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group)</pre>
<div class="block">Add a group to the list of ATOM record group of this chain.
 To add SEQRES records a more complex alignment between ATOM and SEQRES residues
 is required, please see SeqRes2AtomAligner for more details on that.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>group</code> - a Group object</dd>
</dl>
</li>
</ul>
<a id="getId()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getId</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.74">getId</a>()</pre>
<div class="block">Get the 'private' asymId (internal chain IDs in mmCif) for this chain.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the asymId</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#setId(java.lang.String)"><code>setId(String)</code></a>, 
<a href="#getName()"><code>getName()</code></a></dd>
</dl>
</li>
</ul>
<a id="setId(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setId</h4>
<pre class="methodSignature">void&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.82">setId</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;asymId)</pre>
<div class="block">Set the 'private' asymId (internal chain IDs in mmCif) for this chain.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>asymId</code> - the internal chain Id</dd>
</dl>
</li>
</ul>
<a id="setName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setName</h4>
<pre class="methodSignature">void&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.91">setName</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;authId)</pre>
<div class="block">Set the 'public' authId (chain ID in PDB file)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>authId</code> - the 'public' authId (chain ID in PDB file)</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getId()"><code>getId()</code></a></dd>
</dl>
</li>
</ul>
<a id="getName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getName</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.99">getName</a>()</pre>
<div class="block">Get the 'public' authId (chain ID in PDB file)</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the authId for this chain.</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getId()"><code>getId()</code></a></dd>
</dl>
</li>
</ul>
<a id="getAtomGroup(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAtomGroup</h4>
<pre class="methodSignature"><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.112">getAtomGroup</a>&#8203;(int&nbsp;position)</pre>
<div class="block">Return the Group at given position,
 from within Groups with observed density in the chain, i.e.
 those with coordinates in ATOM and HETATMS (including waters) records.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>position</code> - an int</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a Group object</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getAtomLength()"><code>getAtomLength()</code></a>, 
<a href="#getAtomGroups()"><code>getAtomGroups()</code></a>, 
<a href="#getSeqResGroup(int)"><code>getSeqResGroup(int)</code></a></dd>
</dl>
</li>
</ul>
<a id="getSeqResGroup(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSeqResGroup</h4>
<pre class="methodSignature"><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.124">getSeqResGroup</a>&#8203;(int&nbsp;position)</pre>
<div class="block">Return the Group at given position,
 from within groups in the SEQRES records of the chain, i.e.
 the aminoacids/nucleotides in the construct.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>position</code> - an int</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a Group object</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getSeqResLength()"><code>getSeqResLength()</code></a>, 
<a href="#getSeqResGroups()"><code>getSeqResGroups()</code></a>, 
<a href="#getAtomGroup(int)"><code>getAtomGroup(int)</code></a></dd>
</dl>
</li>
</ul>
<a id="getAtomGroups()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAtomGroups</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.136">getAtomGroups</a>()</pre>
<div class="block">Return all Groups with observed density in the chain, i.e.
 those with coordinates in ATOM and HETATMS (including waters) records.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a List object representing the Groups of this Chain.</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#setAtomGroups(java.util.List)"><code>setAtomGroups(List)</code></a>, 
<a href="#getAtomLength()"><code>getAtomLength()</code></a>, 
<a href="#getSeqResGroups()"><code>getSeqResGroups()</code></a></dd>
</dl>
</li>
</ul>
<a id="setAtomGroups(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAtomGroups</h4>
<pre class="methodSignature">void&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.144">setAtomGroups</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;groups)</pre>
<div class="block">Set all Groups with observed density in the chain, i.e.
 those with coordinates in ATOM and HETATMs (including waters) records.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>groups</code> - a List object representing the Groups of this Chain.</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getAtomGroups()"><code>getAtomGroups()</code></a></dd>
</dl>
</li>
</ul>
<a id="getAtomGroups(org.biojava.nbio.structure.GroupType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAtomGroups</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.155">getAtomGroups</a>&#8203;(<a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;type)</pre>
<div class="block">Return a List of all (observed) Groups of a special type, one of: <a href="GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a>,
 <a href="GroupType.html#HETATM"><code>GroupType.HETATM</code></a> or <a href="GroupType.html#NUCLEOTIDE"><code>GroupType.NUCLEOTIDE</code></a>.
 Note that if a standard aminoacid appears as a HETATM (because it is part of a ligand) then
 it is still considered as <a href="GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a> and not as <a href="GroupType.html#HETATM"><code>GroupType.HETATM</code></a>.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>type</code> - GroupType</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a List object</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#setAtomGroups(java.util.List)"><code>setAtomGroups(List)</code></a></dd>
</dl>
</li>
</ul>
<a id="getGroupByPDB(org.biojava.nbio.structure.ResidueNumber)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGroupByPDB</h4>
<pre class="methodSignature"><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.166">getGroupByPDB</a>&#8203;(<a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;resNum)
             throws <a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></pre>
<div class="block">Get a group by its PDB residue numbering. If the PDB residue number is not known,
 throws a StructureException.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>resNum</code> - the PDB residue number of the group</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the matching group</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd>
</dl>
</li>
</ul>
<a id="getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,org.biojava.nbio.structure.ResidueNumber)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGroupsByPDB</h4>
<pre class="methodSignature"><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.176">getGroupsByPDB</a>&#8203;(<a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
                       <a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd)
                throws <a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></pre>
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>pdbresnumStart</code> - PDB residue number of start. If null, defaults to the chain start.</dd>
<dd><code>pdbresnumEnd</code> - PDB residue number of end. If null, defaults to the chain end.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>Groups in between. or throws a StructureException if either start or end can not be found,</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd>
</dl>
</li>
</ul>
<a id="getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,org.biojava.nbio.structure.ResidueNumber,boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGroupsByPDB</h4>
<pre class="methodSignature"><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.192">getGroupsByPDB</a>&#8203;(<a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
                       <a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd,
                       boolean&nbsp;ignoreMissing)
                throws <a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></pre>
<div class="block">Get all groups that are located between two PDB residue numbers. In contrast to getGroupsByPDB
 this method call ignores if the exact outer groups are not found. This is useful e.g. when requesting the range
 of groups as specified by the DBREF records - these frequently are rather inaccurate.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>pdbresnumStart</code> - PDB residue number of start. If null, defaults to the chain start.</dd>
<dd><code>pdbresnumEnd</code> - PDB residue number of end. If null, defaults to the chain end.</dd>
<dd><code>ignoreMissing</code> - ignore missing groups in this range.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>Groups in between. or throws a StructureException if either start or end can not be found,</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd>
</dl>
</li>
</ul>
<a id="getAtomLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAtomLength</h4>
<pre class="methodSignature">int&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.204">getAtomLength</a>()</pre>
<div class="block">Returns the number of Groups with observed density in the chain, i.e.
 those with coordinates in ATOM and HETATMs (including waters) records</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the length</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getAtomGroup(int)"><code>getAtomGroup(int)</code></a>, 
<a href="#getAtomGroups()"><code>getAtomGroups()</code></a></dd>
</dl>
</li>
</ul>
<a id="getSeqResLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSeqResLength</h4>
<pre class="methodSignature">int&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.215">getSeqResLength</a>()</pre>
<div class="block">Returns the number of groups in the SEQRES records of the chain, i.e.
 the number of aminoacids/nucleotides in the construct</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the length</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getSeqResGroup(int)"><code>getSeqResGroup(int)</code></a>, 
<a href="#getSeqResGroups()"><code>getSeqResGroups()</code></a>, 
<a href="#getAtomLength()"><code>getAtomLength()</code></a></dd>
</dl>
</li>
</ul>
<a id="setEntityInfo(org.biojava.nbio.structure.EntityInfo)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setEntityInfo</h4>
<pre class="methodSignature">void&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.222">setEntityInfo</a>&#8203;(<a href="EntityInfo.html" title="class in org.biojava.nbio.structure">EntityInfo</a>&nbsp;entityInfo)</pre>
<div class="block">Sets the Entity information</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>entityInfo</code> - the EntityInfo</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getEntityInfo()"><code>getEntityInfo()</code></a></dd>
</dl>
</li>
</ul>
<a id="getEntityInfo()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getEntityInfo</h4>
<pre class="methodSignature"><a href="EntityInfo.html" title="class in org.biojava.nbio.structure">EntityInfo</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.230">getEntityInfo</a>()</pre>
<div class="block">Returns the EntityInfo for this chain.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the EntityInfo object</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#setEntityInfo(org.biojava.nbio.structure.EntityInfo)"><code>setEntityInfo(EntityInfo)</code></a></dd>
</dl>
</li>
</ul>
<a id="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.232">toString</a>()</pre>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang" class="externalLink">toString</a></code>&nbsp;in class&nbsp;<code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a></code></dd>
</dl>
</li>
</ul>
<a id="getBJSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBJSequence</h4>
<pre class="methodSignature"><a href="../core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.240">getBJSequence</a>()</pre>
<div class="block">Converts the SEQRES groups of a Chain to a Biojava Sequence object.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the SEQRES groups of the Chain as a Sequence object.</dd>
</dl>
</li>
</ul>
<a id="getAtomSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAtomSequence</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.247">getAtomSequence</a>()</pre>
<div class="block">Returns the sequence of amino acids as it has been provided in the ATOM records.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>amino acid sequence as string</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getSeqResSequence()"><code>getSeqResSequence()</code></a></dd>
</dl>
</li>
</ul>
<a id="getSeqResSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSeqResSequence</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.255">getSeqResSequence</a>()</pre>
<div class="block">Returns the PDB SEQRES sequence as a one-letter sequence string.
 Non-standard residues are represented by an "X".</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>one-letter PDB SEQRES sequence as string</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getAtomSequence()"><code>getAtomSequence()</code></a></dd>
</dl>
</li>
</ul>
<a id="getSeqResGroups(org.biojava.nbio.structure.GroupType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSeqResGroups</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.264">getSeqResGroups</a>&#8203;(<a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;type)</pre>
<div class="block">Returns a List of all SEQRES groups of a special type, one of: <a href="GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a>,
 <a href="GroupType.html#HETATM"><code>GroupType.HETATM</code></a> or <a href="GroupType.html#NUCLEOTIDE"><code>GroupType.NUCLEOTIDE</code></a>.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>type</code> - a GroupType</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>an List object</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#setSeqResGroups(java.util.List)"><code>setSeqResGroups(List)</code></a></dd>
</dl>
</li>
</ul>
<a id="getSeqResGroups()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSeqResGroups</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.274">getSeqResGroups</a>()</pre>
<div class="block">Returns a list of all groups in SEQRES records of the chain, i.e.
 the aminoacids/nucleotides in the construct.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a List of all Group objects of this chain</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#setSeqResGroups(java.util.List)"><code>setSeqResGroups(List)</code></a>, 
<a href="#getSeqResLength()"><code>getSeqResLength()</code></a>, 
<a href="#getAtomGroups()"><code>getAtomGroups()</code></a></dd>
</dl>
</li>
</ul>
<a id="setSeqResGroups(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSeqResGroups</h4>
<pre class="methodSignature">void&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.282">setSeqResGroups</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;seqResGroups)</pre>
<div class="block">Sets the list of SeqResGroups for this chain.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>seqResGroups</code> - a List of Group objects that from the SEQRES groups of this chain.</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getSeqResGroups()"><code>getSeqResGroups()</code></a></dd>
</dl>
</li>
</ul>
<a id="setStructure(org.biojava.nbio.structure.Structure)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setStructure</h4>
<pre class="methodSignature">void&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.289">setStructure</a>&#8203;(<a href="Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;parent)</pre>
<div class="block">Sets the back-reference to its parent Structure.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>parent</code> - </dd>
</dl>
</li>
</ul>
<a id="getStructure()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStructure</h4>
<pre class="methodSignature"><a href="Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.297">getStructure</a>()</pre>
<div class="block">Returns the parent Structure of this chain.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the parent Structure object</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#setStructure(org.biojava.nbio.structure.Structure)"><code>setStructure(Structure)</code></a></dd>
</dl>
</li>
</ul>
<a id="toPDB()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toPDB</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.303">toPDB</a>()</pre>
<div class="block">Convert this Chain to a String in PDB format</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="toMMCIF()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toMMCIF</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.309">toMMCIF</a>()</pre>
<div class="block">Convert this Chain to a String in mmCIF format</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="setSeqMisMatches(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSeqMisMatches</h4>
<pre class="methodSignature">void&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.317">setSeqMisMatches</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a>&gt;&nbsp;seqMisMatches)</pre>
<div class="block">Sets annotated sequence mismatches for this chain. This is based on the STRUCT_REF_SEQ_DIF mmCif category</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>seqMisMatches</code> - </dd>
</dl>
</li>
</ul>
<a id="getSeqMisMatches()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSeqMisMatches</h4>
<pre class="methodSignature"><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a>&gt;&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.324">getSeqMisMatches</a>()</pre>
<div class="block">Gets annotated sequence mismatches for this chain. This is based on the STRUCT_REF_SEQ_DIF mmCif category</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a list of sequence mismatches (or null if none found)</dd>
</dl>
</li>
</ul>
<a id="getEntityType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getEntityType</h4>
<pre class="methodSignature"><a href="EntityType.html" title="enum in org.biojava.nbio.structure">EntityType</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.331">getEntityType</a>()</pre>
<div class="block">Returns the EntityType of this chain. Equivalent to getEntityInfo().getType()</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="EntityType.html" title="enum in org.biojava.nbio.structure"><code>EntityType</code></a></dd>
</dl>
</li>
</ul>
<a id="isWaterOnly()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isWaterOnly</h4>
<pre class="methodSignature">boolean&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.338">isWaterOnly</a>()</pre>
<div class="block">Tests if a chain is consisting of water molecules only</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>true if there are only solvent molecules in this chain.</dd>
</dl>
</li>
</ul>
<a id="isPureNonPolymer()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isPureNonPolymer</h4>
<pre class="methodSignature">boolean&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.345">isPureNonPolymer</a>()</pre>
<div class="block">Returns true if the given chain is composed of non-polymeric (including water) groups only.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>true if only non-polymeric groups in this chain.</dd>
</dl>
</li>
</ul>
<a id="getPredominantGroupType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPredominantGroupType</h4>
<pre class="methodSignature"><a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.364">getPredominantGroupType</a>()</pre>
<div class="block">Get the predominant <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> for a given Chain, following these
 rules: <li>if the ratio of number of residues of a certain
 <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> to total non-water residues is above the threshold
 <a href="StructureTools.html#RATIO_RESIDUES_TO_TOTAL">0.95</a>, then that <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> is
 returned</li> <li>if there is no <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> that is above the
 threshold then the <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> with most members is chosen, logging
 it</li>
 <p>
 See also <a href="chem/ChemComp.html#getPolymerType()"><code>ChemComp.getPolymerType()</code></a> and
 <a href="chem/ChemComp.html#getResidueType()"><code>ChemComp.getResidueType()</code></a> which follow the PDB chemical component
 dictionary and provide a much more accurate description of groups and
 their linking.
 </p></div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the predominant group type</dd>
</dl>
</li>
</ul>
<a id="isProtein()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isProtein</h4>
<pre class="methodSignature">boolean&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.372">isProtein</a>()</pre>
<div class="block">Tell whether given chain is a protein chain</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>true if protein, false if nucleotide or ligand</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getPredominantGroupType()"><code>getPredominantGroupType()</code></a></dd>
</dl>
</li>
</ul>
<a id="isNucleicAcid()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>isNucleicAcid</h4>
<pre class="methodSignature">boolean&nbsp;<a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line.380">isNucleicAcid</a>()</pre>
<div class="block">Tell whether given chain is DNA or RNA</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>true if nucleic acid, false if protein or ligand</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getPredominantGroupType()"><code>getPredominantGroupType()</code></a></dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Chain.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2022 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
