<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.16) on Fri Oct 28 16:15:48 PDT 2022 -->
<title>All Classes (biojava 6.1.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2022-10-28">
<link rel="stylesheet" type="text/css" href="stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="script.js"></script>
<script type="text/javascript" src="jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="All Classes (biojava 6.1.0 API)";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":2,"i1":2,"i2":1,"i3":2,"i4":2,"i5":2,"i6":2,"i7":2,"i8":2,"i9":2,"i10":2,"i11":2,"i12":2,"i13":2,"i14":2,"i15":2,"i16":2,"i17":2,"i18":2,"i19":2,"i20":2,"i21":2,"i22":2,"i23":4,"i24":2,"i25":2,"i26":2,"i27":1,"i28":1,"i29":2,"i30":2,"i31":2,"i32":2,"i33":2,"i34":2,"i35":2,"i36":2,"i37":2,"i38":2,"i39":2,"i40":2,"i41":2,"i42":2,"i43":2,"i44":2,"i45":2,"i46":2,"i47":1,"i48":2,"i49":1,"i50":4,"i51":1,"i52":2,"i53":2,"i54":2,"i55":2,"i56":4,"i57":2,"i58":2,"i59":2,"i60":1,"i61":2,"i62":2,"i63":1,"i64":2,"i65":2,"i66":2,"i67":4,"i68":4,"i69":4,"i70":4,"i71":4,"i72":2,"i73":2,"i74":2,"i75":2,"i76":2,"i77":2,"i78":2,"i79":2,"i80":2,"i81":8,"i82":2,"i83":2,"i84":2,"i85":2,"i86":2,"i87":2,"i88":1,"i89":2,"i90":2,"i91":2,"i92":2,"i93":2,"i94":2,"i95":2,"i96":2,"i97":1,"i98":1,"i99":2,"i100":2,"i101":2,"i102":2,"i103":2,"i104":16,"i105":2,"i106":4,"i107":1,"i108":2,"i109":2,"i110":2,"i111":2,"i112":2,"i113":2,"i114":2,"i115":2,"i116":2,"i117":1,"i118":2,"i119":2,"i120":1,"i121":2,"i122":2,"i123":2,"i124":2,"i125":2,"i126":2,"i127":2,"i128":2,"i129":2,"i130":2,"i131":2,"i132":2,"i133":2,"i134":2,"i135":2,"i136":2,"i137":4,"i138":2,"i139":2,"i140":2,"i141":2,"i142":2,"i143":4,"i144":4,"i145":4,"i146":4,"i147":4,"i148":2,"i149":2,"i150":2,"i151":2,"i152":2,"i153":1,"i154":2,"i155":1,"i156":2,"i157":1,"i158":2,"i159":2,"i160":4,"i161":2,"i162":4,"i163":4,"i164":2,"i165":2,"i166":2,"i167":2,"i168":2,"i169":2,"i170":2,"i171":2,"i172":2,"i173":2,"i174":2,"i175":2,"i176":2,"i177":2,"i178":4,"i179":1,"i180":2,"i181":2,"i182":2,"i183":2,"i184":2,"i185":2,"i186":2,"i187":2,"i188":2,"i189":2,"i190":2,"i191":2,"i192":2,"i193":4,"i194":2,"i195":2,"i196":2,"i197":1,"i198":2,"i199":4,"i200":2,"i201":2,"i202":2,"i203":2,"i204":2,"i205":4,"i206":4,"i207":4,"i208":2,"i209":2,"i210":2,"i211":1,"i212":2,"i213":2,"i214":1,"i215":2,"i216":2,"i217":2,"i218":2,"i219":1,"i220":2,"i221":2,"i222":2,"i223":2,"i224":2,"i225":1,"i226":2,"i227":2,"i228":2,"i229":2,"i230":2,"i231":2,"i232":2,"i233":2,"i234":2,"i235":2,"i236":1,"i237":2,"i238":1,"i239":2,"i240":1,"i241":1,"i242":2,"i243":2,"i244":2,"i245":2,"i246":2,"i247":2,"i248":2,"i249":2,"i250":2,"i251":2,"i252":2,"i253":1,"i254":2,"i255":2,"i256":4,"i257":2,"i258":2,"i259":1,"i260":2,"i261":2,"i262":2,"i263":1,"i264":8,"i265":1,"i266":1,"i267":2,"i268":2,"i269":1,"i270":8,"i271":2,"i272":2,"i273":2,"i274":1,"i275":2,"i276":2,"i277":2,"i278":2,"i279":2,"i280":2,"i281":2,"i282":2,"i283":2,"i284":2,"i285":2,"i286":2,"i287":2,"i288":2,"i289":2,"i290":1,"i291":2,"i292":2,"i293":2,"i294":4,"i295":2,"i296":2,"i297":2,"i298":2,"i299":2,"i300":2,"i301":2,"i302":2,"i303":2,"i304":2,"i305":2,"i306":2,"i307":2,"i308":2,"i309":2,"i310":2,"i311":1,"i312":2,"i313":4,"i314":2,"i315":2,"i316":2,"i317":2,"i318":2,"i319":2,"i320":2,"i321":2,"i322":2,"i323":2,"i324":2,"i325":2,"i326":2,"i327":2,"i328":2,"i329":2,"i330":2,"i331":2,"i332":2,"i333":2,"i334":2,"i335":2,"i336":2,"i337":2,"i338":2,"i339":2,"i340":2,"i341":2,"i342":2,"i343":2,"i344":2,"i345":2,"i346":2,"i347":2,"i348":2,"i349":2,"i350":2,"i351":2,"i352":2,"i353":2,"i354":2,"i355":2,"i356":2,"i357":1,"i358":2,"i359":2,"i360":2,"i361":2,"i362":2,"i363":2,"i364":4,"i365":2,"i366":2,"i367":2,"i368":1,"i369":2,"i370":2,"i371":2,"i372":1,"i373":2,"i374":2,"i375":2,"i376":2,"i377":1,"i378":2,"i379":2,"i380":2,"i381":2,"i382":2,"i383":2,"i384":4,"i385":2,"i386":4,"i387":2,"i388":2,"i389":2,"i390":2,"i391":2,"i392":2,"i393":4,"i394":2,"i395":2,"i396":2,"i397":4,"i398":2,"i399":2,"i400":2,"i401":1,"i402":2,"i403":2,"i404":2,"i405":2,"i406":2,"i407":2,"i408":2,"i409":2,"i410":2,"i411":1,"i412":2,"i413":4,"i414":1,"i415":2,"i416":2,"i417":2,"i418":2,"i419":2,"i420":2,"i421":2,"i422":2,"i423":2,"i424":2,"i425":1,"i426":1,"i427":2,"i428":1,"i429":1,"i430":2,"i431":2,"i432":2,"i433":2,"i434":2,"i435":2,"i436":2,"i437":2,"i438":2,"i439":2,"i440":2,"i441":2,"i442":2,"i443":2,"i444":2,"i445":2,"i446":4,"i447":2,"i448":2,"i449":1,"i450":4,"i451":2,"i452":1,"i453":2,"i454":2,"i455":2,"i456":2,"i457":2,"i458":2,"i459":2,"i460":2,"i461":2,"i462":2,"i463":2,"i464":2,"i465":2,"i466":2,"i467":2,"i468":2,"i469":2,"i470":2,"i471":2,"i472":2,"i473":2,"i474":2,"i475":2,"i476":2,"i477":2,"i478":2,"i479":2,"i480":2,"i481":2,"i482":2,"i483":2,"i484":2,"i485":2,"i486":1,"i487":2,"i488":2,"i489":2,"i490":2,"i491":2,"i492":2,"i493":4,"i494":2,"i495":2,"i496":1,"i497":2,"i498":2,"i499":2,"i500":2,"i501":2,"i502":2,"i503":2,"i504":2,"i505":2,"i506":2,"i507":2,"i508":2,"i509":2,"i510":4,"i511":1,"i512":2,"i513":2,"i514":2,"i515":2,"i516":1,"i517":2,"i518":2,"i519":2,"i520":2,"i521":2,"i522":2,"i523":2,"i524":2,"i525":2,"i526":2,"i527":2,"i528":2,"i529":2,"i530":2,"i531":2,"i532":2,"i533":2,"i534":2,"i535":8,"i536":2,"i537":1,"i538":1,"i539":1,"i540":4,"i541":4,"i542":4,"i543":4,"i544":2,"i545":2,"i546":2,"i547":2,"i548":2,"i549":2,"i550":2,"i551":2,"i552":2,"i553":2,"i554":2,"i555":2,"i556":2,"i557":2,"i558":2,"i559":2,"i560":2,"i561":2,"i562":2,"i563":2,"i564":2,"i565":2,"i566":2,"i567":2,"i568":2,"i569":2,"i570":2,"i571":2,"i572":2,"i573":2,"i574":1,"i575":4,"i576":2,"i577":4,"i578":2,"i579":4,"i580":4,"i581":2,"i582":1,"i583":1,"i584":2,"i585":2,"i586":2,"i587":2,"i588":2,"i589":2,"i590":2,"i591":2,"i592":2,"i593":2,"i594":2,"i595":1,"i596":1,"i597":2,"i598":2,"i599":2,"i600":2,"i601":2,"i602":2,"i603":2,"i604":2,"i605":1,"i606":2,"i607":2,"i608":2,"i609":2,"i610":2,"i611":2,"i612":2,"i613":2,"i614":2,"i615":2,"i616":2,"i617":2,"i618":2,"i619":2,"i620":4,"i621":1,"i622":2,"i623":2,"i624":4,"i625":2,"i626":1,"i627":2,"i628":2,"i629":2,"i630":4,"i631":1,"i632":2,"i633":2,"i634":2,"i635":1,"i636":2,"i637":2,"i638":2,"i639":2,"i640":2,"i641":2,"i642":2,"i643":2,"i644":2,"i645":2,"i646":2,"i647":2,"i648":2,"i649":2,"i650":2,"i651":2,"i652":2,"i653":1,"i654":1,"i655":1,"i656":1,"i657":1,"i658":1,"i659":2,"i660":2,"i661":2,"i662":2,"i663":2,"i664":2,"i665":2,"i666":2,"i667":2,"i668":2,"i669":2,"i670":2,"i671":2,"i672":2,"i673":2,"i674":2,"i675":2,"i676":2,"i677":2,"i678":1,"i679":2,"i680":2,"i681":2,"i682":2,"i683":2,"i684":1,"i685":2,"i686":8,"i687":1,"i688":1,"i689":1,"i690":2,"i691":2,"i692":2,"i693":2,"i694":2,"i695":1,"i696":2,"i697":2,"i698":2,"i699":2,"i700":2,"i701":1,"i702":1,"i703":1,"i704":2,"i705":2,"i706":2,"i707":1,"i708":8,"i709":1,"i710":2,"i711":2,"i712":2,"i713":2,"i714":2,"i715":2,"i716":2,"i717":2,"i718":2,"i719":2,"i720":2,"i721":8,"i722":1,"i723":2,"i724":2,"i725":4,"i726":2,"i727":2,"i728":2,"i729":2,"i730":2,"i731":2,"i732":4,"i733":2,"i734":2,"i735":2,"i736":2,"i737":1,"i738":1,"i739":1,"i740":4,"i741":2,"i742":2,"i743":2,"i744":2,"i745":2,"i746":2,"i747":1,"i748":4,"i749":1,"i750":1,"i751":1,"i752":1,"i753":2,"i754":2,"i755":1,"i756":2,"i757":2,"i758":2,"i759":2,"i760":2,"i761":2,"i762":2,"i763":1,"i764":2,"i765":2,"i766":2,"i767":2,"i768":4,"i769":2,"i770":2,"i771":2,"i772":2,"i773":2,"i774":2,"i775":2,"i776":2,"i777":1,"i778":2,"i779":2,"i780":2,"i781":2,"i782":1,"i783":2,"i784":8,"i785":2,"i786":1,"i787":1,"i788":1,"i789":1,"i790":2,"i791":4,"i792":1,"i793":2,"i794":4,"i795":2,"i796":2,"i797":2,"i798":2,"i799":4,"i800":2,"i801":2,"i802":1,"i803":2,"i804":2,"i805":2,"i806":2,"i807":2,"i808":2,"i809":2,"i810":2,"i811":4,"i812":2,"i813":2,"i814":2,"i815":2,"i816":2,"i817":2,"i818":2,"i819":2,"i820":2,"i821":2,"i822":2,"i823":4,"i824":1,"i825":2,"i826":2,"i827":2,"i828":2,"i829":2,"i830":8,"i831":2,"i832":2,"i833":2,"i834":1,"i835":1,"i836":2,"i837":2,"i838":2,"i839":2,"i840":2,"i841":2,"i842":4,"i843":2,"i844":2,"i845":1,"i846":2,"i847":2,"i848":2,"i849":1,"i850":2,"i851":2,"i852":1,"i853":2,"i854":2,"i855":1,"i856":2,"i857":2,"i858":2,"i859":2,"i860":1,"i861":2,"i862":2,"i863":2,"i864":2,"i865":2,"i866":4,"i867":4,"i868":1,"i869":1,"i870":2,"i871":1,"i872":2,"i873":2,"i874":2,"i875":1,"i876":2,"i877":2,"i878":2,"i879":2,"i880":2,"i881":2,"i882":2,"i883":2,"i884":2,"i885":2,"i886":2,"i887":2,"i888":2,"i889":2,"i890":2,"i891":2,"i892":2,"i893":2,"i894":2,"i895":2,"i896":2,"i897":2,"i898":2,"i899":2,"i900":2,"i901":2,"i902":2,"i903":2,"i904":2,"i905":2,"i906":2,"i907":2,"i908":2,"i909":2,"i910":2,"i911":2,"i912":2,"i913":2,"i914":2,"i915":2,"i916":2,"i917":2,"i918":2,"i919":2,"i920":2,"i921":4,"i922":2,"i923":2,"i924":2,"i925":2,"i926":2,"i927":2,"i928":2,"i929":2,"i930":2,"i931":2,"i932":2,"i933":2,"i934":2,"i935":4,"i936":2,"i937":4,"i938":2,"i939":2,"i940":1,"i941":2,"i942":2,"i943":2,"i944":1,"i945":1,"i946":2,"i947":2,"i948":2,"i949":2,"i950":2,"i951":2,"i952":2,"i953":8,"i954":4,"i955":2,"i956":2,"i957":1,"i958":2,"i959":2,"i960":2,"i961":2,"i962":2,"i963":1,"i964":2,"i965":2,"i966":4,"i967":2,"i968":1,"i969":1,"i970":2,"i971":2,"i972":2,"i973":1,"i974":1,"i975":2,"i976":2,"i977":2,"i978":2,"i979":2,"i980":2,"i981":4,"i982":2,"i983":2,"i984":1,"i985":2,"i986":2,"i987":2,"i988":2,"i989":2,"i990":2,"i991":2,"i992":4,"i993":4,"i994":4,"i995":4,"i996":2,"i997":2,"i998":2,"i999":2,"i1000":2,"i1001":2,"i1002":2,"i1003":2,"i1004":2,"i1005":2,"i1006":2,"i1007":2,"i1008":4,"i1009":1,"i1010":2,"i1011":2,"i1012":2,"i1013":2,"i1014":2,"i1015":2,"i1016":2,"i1017":2,"i1018":1,"i1019":2,"i1020":2,"i1021":2,"i1022":1,"i1023":2,"i1024":2,"i1025":2,"i1026":2,"i1027":2,"i1028":2,"i1029":2,"i1030":2,"i1031":2,"i1032":2,"i1033":4,"i1034":8,"i1035":2,"i1036":4,"i1037":2,"i1038":4,"i1039":1,"i1040":2,"i1041":2,"i1042":2,"i1043":2,"i1044":2,"i1045":2,"i1046":2,"i1047":2,"i1048":2,"i1049":2,"i1050":2,"i1051":2,"i1052":2,"i1053":1,"i1054":2,"i1055":2,"i1056":1,"i1057":2,"i1058":2,"i1059":2,"i1060":2,"i1061":2,"i1062":2,"i1063":2,"i1064":2,"i1065":2,"i1066":2,"i1067":1,"i1068":2};
var tabs = {65535:["t0","All Classes"],1:["t1","Interface Summary"],2:["t2","Class Summary"],4:["t3","Enum Summary"],8:["t4","Exception Summary"],16:["t5","Error Summary"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "./";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li><a href="index-all.html">Index</a></li>
<li><a href="help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h1 title="All&amp;nbsp;Classes" class="title">All&nbsp;Classes</h1>
</div>
<div class="allClassesContainer">
<ul class="blockList">
<li class="blockList">
<table class="typeSummary">
<caption><span id="t0" class="activeTableTab"><span>All Classes</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Interface Summary</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Class Summary</a></span><span class="tabEnd">&nbsp;</span></span><span id="t3" class="tableTab"><span><a href="javascript:show(4);">Enum Summary</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Exception Summary</a></span><span class="tabEnd">&nbsp;</span></span><span id="t5" class="tableTab"><span><a href="javascript:show(16);">Error Summary</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/matrices/AAindexFactory.html" title="class in org.biojava.nbio.core.alignment.matrices">AAindexFactory</a></td>
<th class="colLast" scope="row">
<div class="block">Factory class to get Providers for substitution matrices that are provided by
 the AAINDEX database.</div>
</th>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/matrices/AAIndexFileParser.html" title="class in org.biojava.nbio.core.alignment.matrices">AAIndexFileParser</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/matrices/AAIndexProvider.html" title="interface in org.biojava.nbio.core.alignment.matrices">AAIndexProvider</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/ABITrace.html" title="class in org.biojava.nbio.core.sequence.io">ABITrace</a></td>
<th class="colLast" scope="row">
<div class="block">Title: ABITrace</div>
</th>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/compound/ABITracerCompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">ABITracerCompoundSet</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/AboutDialog.html" title="class in org.biojava.nbio.structure.align.gui">AboutDialog</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a></td>
<th class="colLast" scope="row">
<div class="block">An Abstract Class to generalize the visualization of AFP and
 MultipleAlignment structure alignments in Jmol.</div>
</th>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/utils/AbstractAnnotation.html" title="class in org.biojava.nbio.ontology.utils">AbstractAnnotation</a></td>
<th class="colLast" scope="row">
<div class="block">A utility class to ease the problem of implementing an Annotation to that of
 providing an apropreate implementation of Map.</div>
</th>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/AbstractCifFileSupplier.html" title="class in org.biojava.nbio.structure.io.cif">AbstractCifFileSupplier</a>&lt;<a href="org/biojava/nbio/structure/io/cif/AbstractCifFileSupplier.html" title="type parameter in AbstractCifFileSupplier">S</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Convert a BioJava object to a CifFile.</div>
</th>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/AbstractCifFileSupplier.WrappedAtom.html" title="class in org.biojava.nbio.structure.io.cif">AbstractCifFileSupplier.WrappedAtom</a></td>
<th class="colLast" scope="row">
<div class="block">Wrapped atoms represent individual atoms enriched with model- and chain-level information.</div>
</th>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/AbstractCompound.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompound</a></td>
<th class="colLast" scope="row">
<div class="block">The details of a Compound</div>
</th>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompoundSet</a>&lt;<a href="org/biojava/nbio/core/sequence/template/AbstractCompoundSet.html" title="type parameter in AbstractCompoundSet">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompoundTranslator</a>&lt;<a href="org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>,&#8203;<a href="org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/features/AbstractFeature.html" title="class in org.biojava.nbio.core.sequence.features">AbstractFeature</a>&lt;<a href="org/biojava/nbio/core/sequence/features/AbstractFeature.html" title="type parameter in AbstractFeature">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/features/AbstractFeature.html" title="type parameter in AbstractFeature">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/features/AbstractFeature.html" title="type parameter in AbstractFeature">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</th>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava.nbio.core.sequence.location.template">AbstractLocation</a></td>
<th class="colLast" scope="row">
<div class="block">Base abstraction of a location which encodes for the majority of important
 features about a location such as the start, end and strand</div>
</th>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/AbstractMatrixAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractMatrixAligner</a>&lt;<a href="org/biojava/nbio/alignment/template/AbstractMatrixAligner.html" title="type parameter in AbstractMatrixAligner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/AbstractMatrixAligner.html" title="type parameter in AbstractMatrixAligner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/AbstractMatrixAligner.html" title="type parameter in AbstractMatrixAligner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements common code for an <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> which builds a score matrix during computation.</div>
</th>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava.nbio.core.sequence.template">AbstractNucleotideCompoundSet</a>&lt;<a href="org/biojava/nbio/core/sequence/template/AbstractNucleotideCompoundSet.html" title="type parameter in AbstractNucleotideCompoundSet">C</a> extends <a href="org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractPairwiseSequenceAligner</a>&lt;<a href="org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html" title="type parameter in AbstractPairwiseSequenceAligner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html" title="type parameter in AbstractPairwiseSequenceAligner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html" title="type parameter in AbstractPairwiseSequenceAligner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements common code for an <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</th>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractProfileProfileAligner</a>&lt;<a href="org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements common code for an <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</th>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/reference/AbstractReference.html" title="class in org.biojava.nbio.core.sequence.reference">AbstractReference</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/AbstractScorer.html" title="class in org.biojava.nbio.alignment.template">AbstractScorer</a></td>
<th class="colLast" scope="row">
<div class="block">Implements common code for algorithms which compute a score.</div>
</th>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/AbstractScoresCache.html" title="class in org.biojava.nbio.structure.align.multiple">AbstractScoresCache</a></td>
<th class="colLast" scope="row">
<div class="block">Abstact implementation of the <a href="org/biojava/nbio/structure/align/multiple/ScoresCache.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>ScoresCache</code></a> with the shared code used
 in all objects with a variables cache.</div>
</th>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">The base class for DNA, RNA and Protein sequences.</div>
</th>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template">AbstractSequence.AnnotationType</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/AbstractStructureAlignment.html" title="class in org.biojava.nbio.structure.align">AbstractStructureAlignment</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/AbstractTerm.html" title="class in org.biojava.nbio.ontology">AbstractTerm</a></td>
<th class="colLast" scope="row">
<div class="block">Abstract implementation of term

 This provides basic change-forwarding functionality from
                the annotation and ontology properties.</div>
</th>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/AbstractUserArgumentProcessor.html" title="class in org.biojava.nbio.structure.align.ce">AbstractUserArgumentProcessor</a></td>
<th class="colLast" scope="row">
<div class="block">Base class for a new structure alignment CLI.</div>
</th>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/template/AccesionedLocation.html" title="interface in org.biojava.nbio.core.sequence.location.template">AccesionedLocation</a></td>
<th class="colLast" scope="row">
<div class="block">A location which is bound to an AccessionID.</div>
</th>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/Accessioned.html" title="interface in org.biojava.nbio.core.sequence.template">Accessioned</a></td>
<th class="colLast" scope="row">
<div class="block">Indicates an entity is accessioned</div>
</th>
</tr>
<tr id="i29" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a></td>
<th class="colLast" scope="row">
<div class="block">Used in Sequences as the unique identifier.</div>
</th>
</tr>
<tr id="i30" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/model/AFP.html" title="class in org.biojava.nbio.structure.align.model">AFP</a></td>
<th class="colLast" scope="row">
<div class="block">A class to represent a FATCAT aligned fragment pair (AFP)</div>
</th>
</tr>
<tr id="i31" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html" title="class in org.biojava.nbio.structure.align.util">AFPAlignmentDisplay</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i32" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/calc/AFPCalculator.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">AFPCalculator</a></td>
<th class="colLast" scope="row">
<div class="block">A class that performs calculations on AFPChains</div>
</th>
</tr>
<tr id="i33" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></td>
<th class="colLast" scope="row">
<div class="block">A bean to contain the core of a structure alignment.</div>
</th>
</tr>
<tr id="i34" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/aligpanel/AFPChainCoordManager.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">AFPChainCoordManager</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i35" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/calc/AFPChainer.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">AFPChainer</a></td>
<th class="colLast" scope="row">
<div class="block">a class to chain AFPs to an alignment</div>
</th>
</tr>
<tr id="i36" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/xml/AFPChainFlipper.html" title="class in org.biojava.nbio.structure.align.xml">AFPChainFlipper</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i37" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/AFPChainScorer.html" title="class in org.biojava.nbio.structure.align.util">AFPChainScorer</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i38" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/model/AfpChainWriter.html" title="class in org.biojava.nbio.structure.align.model">AfpChainWriter</a></td>
<th class="colLast" scope="row">
<div class="block">A class to convert the data in an AfpChain object to various String outputs.</div>
</th>
</tr>
<tr id="i39" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html" title="class in org.biojava.nbio.structure.align.xml">AFPChainXMLConverter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i40" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html" title="class in org.biojava.nbio.structure.align.xml">AFPChainXMLParser</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i41" class="rowColor">
<td class="colFirst"><a href="demo/AFPFromFasta.html" title="class in demo">AFPFromFasta</a></td>
<th class="colLast" scope="row">
<div class="block">Demo displaying a structural alignment from a FASTA file using <a href="org/biojava/nbio/structure/io/FastaAFPChainConverter.html" title="class in org.biojava.nbio.structure.io"><code>FastaAFPChainConverter</code></a>.</div>
</th>
</tr>
<tr id="i42" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/calc/AFPOptimizer.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">AFPOptimizer</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i43" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/calc/AFPPostProcessor.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">AFPPostProcessor</a></td>
<th class="colLast" scope="row">
<div class="block">does post processing after alignment chaingin</div>
</th>
</tr>
<tr id="i44" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/AFPTwister.html" title="class in org.biojava.nbio.structure.align">AFPTwister</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i45" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/stats/AgScore.html" title="class in org.biojava.nbio.survival.cox.stats">AgScore</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i46" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/helper/AligMatEl.html" title="class in org.biojava.nbio.structure.align.helper">AligMatEl</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i47" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/pairwise/Alignable.html" title="interface in org.biojava.nbio.structure.align.pairwise">Alignable</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i48" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/AlignedPosition.html" title="class in org.biojava.nbio.structure.gui.util">AlignedPosition</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i49" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">AlignedSequence</a>&lt;<a href="org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="type parameter in AlignedSequence">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="type parameter in AlignedSequence">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="type parameter in AlignedSequence">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a data structure for a <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</th>
</tr>
<tr id="i50" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/template/AlignedSequence.Step.html" title="enum in org.biojava.nbio.core.alignment.template">AlignedSequence.Step</a></td>
<th class="colLast" scope="row">
<div class="block">Defines an alignment step in order to pass alignment information from an <code>Aligner</code> to a constructor.</div>
</th>
</tr>
<tr id="i51" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template">Aligner</a>&lt;<a href="org/biojava/nbio/alignment/template/Aligner.html" title="type parameter in Aligner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/Aligner.html" title="type parameter in Aligner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/Aligner.html" title="type parameter in Aligner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines an algorithm which computes an alignment <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a> from a list of <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</th>
</tr>
<tr id="i52" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/routines/AlignerHelper.html" title="class in org.biojava.nbio.alignment.routines">AlignerHelper</a></td>
<th class="colLast" scope="row">
<div class="block">Static utility to construct alignment routines from a common library of methods.</div>
</th>
</tr>
<tr id="i53" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/routines/AlignerHelper.Anchor.html" title="class in org.biojava.nbio.alignment.routines">AlignerHelper.Anchor</a></td>
<th class="colLast" scope="row">
<div class="block">Compounds in query and target sequences that must align</div>
</th>
</tr>
<tr id="i54" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/routines/AlignerHelper.Anchor.QueryIndexComparator.html" title="class in org.biojava.nbio.alignment.routines">AlignerHelper.Anchor.QueryIndexComparator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i55" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/routines/AlignerHelper.Cut.html" title="class in org.biojava.nbio.alignment.routines">AlignerHelper.Cut</a></td>
<th class="colLast" scope="row">
<div class="block">Defines a 'cut' row for divide-and-conquer alignment in which a new anchor is found.</div>
</th>
</tr>
<tr id="i56" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/routines/AlignerHelper.Last.html" title="enum in org.biojava.nbio.alignment.routines">AlignerHelper.Last</a></td>
<th class="colLast" scope="row">
<div class="block">Define a traceback pointer for the three edit operations: substitution (match/replacement of a query compound
 with a target compound), deletion (removal of a query compound leaving a gap in the target sequence), and
 insertion (addition of a target compound opening a gap in the query sequence).</div>
</th>
</tr>
<tr id="i57" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/routines/AlignerHelper.Subproblem.html" title="class in org.biojava.nbio.alignment.routines">AlignerHelper.Subproblem</a></td>
<th class="colLast" scope="row">
<div class="block">Alignment subproblem.</div>
</th>
</tr>
<tr id="i58" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/AlignmentCalc.html" title="class in org.biojava.nbio.structure.align.gui">AlignmentCalc</a></td>
<th class="colLast" scope="row">
<div class="block">A class that obtains two structures via DAS and aligns them
  This is done in a separate thread.</div>
</th>
</tr>
<tr id="i59" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/AlignmentCalc.html" title="class in org.biojava.nbio.structure.gui.util">AlignmentCalc</a></td>
<th class="colLast" scope="row">
<div class="block">A class that obtains two structures via DAS and aligns them
  This is done in a separate thread.</div>
</th>
</tr>
<tr id="i60" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/AlignmentCalculationRunnable.html" title="interface in org.biojava.nbio.structure.align.gui">AlignmentCalculationRunnable</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i61" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/AlignmentGui.html" title="class in org.biojava.nbio.structure.align.gui">AlignmentGui</a></td>
<th class="colLast" scope="row">
<div class="block">A JFrame that allows to trigger a pairwise structure alignment,
 either from files in a directory,
 or after manual upload.</div>
</th>
</tr>
<tr id="i62" class="altColor">
<td class="colFirst"><a href="demo/AlignmentGuiDemo.html" title="class in demo">AlignmentGuiDemo</a></td>
<th class="colLast" scope="row">
<div class="block">Get an instance of the two alignment GUIs.</div>
</th>
</tr>
<tr id="i63" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/events/AlignmentPositionListener.html" title="interface in org.biojava.nbio.structure.gui.events">AlignmentPositionListener</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i64" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/pairwise/AlignmentProgressListener.html" title="class in org.biojava.nbio.structure.align.pairwise">AlignmentProgressListener</a></td>
<th class="colLast" scope="row">
<div class="block">A class to listen to progress of the structure alignment calculations</div>
</th>
</tr>
<tr id="i65" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/pairwise/AlignmentResult.html" title="class in org.biojava.nbio.structure.align.pairwise">AlignmentResult</a></td>
<th class="colLast" scope="row">
<div class="block">A class to track the alignment results in a flat file</div>
</th>
</tr>
<tr id="i66" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/Alignments.html" title="class in org.biojava.nbio.alignment">Alignments</a></td>
<th class="colLast" scope="row">
<div class="block">Static utility to easily run alignment routines.</div>
</th>
</tr>
<tr id="i67" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/Alignments.PairInProfileScorerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairInProfileScorerType</a></td>
<th class="colLast" scope="row">
<div class="block">List of implemented sequence pair in a profile scoring routines.</div>
</th>
</tr>
<tr id="i68" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a></td>
<th class="colLast" scope="row">
<div class="block">List of implemented pairwise sequence alignment routines.</div>
</th>
</tr>
<tr id="i69" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/Alignments.PairwiseSequenceScorerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceScorerType</a></td>
<th class="colLast" scope="row">
<div class="block">List of implemented pairwise sequence scoring routines.</div>
</th>
</tr>
<tr id="i70" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/Alignments.ProfileProfileAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.ProfileProfileAlignerType</a></td>
<th class="colLast" scope="row">
<div class="block">List of implemented profile-profile alignment routines.</div>
</th>
</tr>
<tr id="i71" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/Alignments.RefinerType.html" title="enum in org.biojava.nbio.alignment">Alignments.RefinerType</a></td>
<th class="colLast" scope="row">
<div class="block">List of implemented profile refinement routines.</div>
</th>
</tr>
<tr id="i72" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/AlignmentTextPanel.html" title="class in org.biojava.nbio.structure.align.gui">AlignmentTextPanel</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i73" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/AlignmentTools.html" title="class in org.biojava.nbio.structure.align.util">AlignmentTools</a></td>
<th class="colLast" scope="row">
<div class="block">Methods for analyzing and manipulating AFPChains and for
 other pairwise alignment utilities.</div>
</th>
</tr>
<tr id="i74" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/AlignmentTools.IdentityMap.html" title="class in org.biojava.nbio.structure.align.util">AlignmentTools.IdentityMap</a>&lt;<a href="org/biojava/nbio/structure/align/util/AlignmentTools.IdentityMap.html" title="type parameter in AlignmentTools.IdentityMap">K</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A Map<K,V> can be viewed as a function from K to V.</div>
</th>
</tr>
<tr id="i75" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/pairwise/AligNPE.html" title="class in org.biojava.nbio.structure.align.pairwise">AligNPE</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i76" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/helper/AlignUtils.html" title="class in org.biojava.nbio.structure.align.helper">AlignUtils</a></td>
<th class="colLast" scope="row">
<div class="block">Low level helper methods for CE and FATCAT algorithms.</div>
</th>
</tr>
<tr id="i77" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/aligpanel/AligPanel.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">AligPanel</a></td>
<th class="colLast" scope="row">
<div class="block">A JPanel that can display an AFPChain in a nice way and interact with Jmol.</div>
</th>
</tr>
<tr id="i78" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/aligpanel/AligPanelMouseMotionListener.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">AligPanelMouseMotionListener</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i79" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/webstart/AligUIManager.html" title="class in org.biojava.nbio.structure.align.webstart">AligUIManager</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i80" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/AllChemCompProvider.html" title="class in org.biojava.nbio.structure.chem">AllChemCompProvider</a></td>
<th class="colLast" scope="row">
<div class="block">A ChemComp provider that downloads and caches the components.cif file from the wwPDB site.</div>
</th>
</tr>
<tr id="i81" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/AlreadyExistsException.html" title="class in org.biojava.nbio.ontology">AlreadyExistsException</a></td>
<th class="colLast" scope="row">
<div class="block">Thrown to indicate that a term or triple can't be added to an ontology
 because it is already present.</div>
</th>
</tr>
<tr id="i82" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/pairwise/AltAligComparator.html" title="class in org.biojava.nbio.structure.align.pairwise">AltAligComparator</a></td>
<th class="colLast" scope="row">
<div class="block">A comparator to sort AlternativeAlignments based on their number of equivalent residues
 and RMSD.</div>
</th>
</tr>
<tr id="i83" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html" title="class in org.biojava.nbio.structure.align.pairwise">AlternativeAlignment</a></td>
<th class="colLast" scope="row">
<div class="block">Implements a class which handles one possible (alternative) solution.</div>
</th>
</tr>
<tr id="i84" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/AlternativeAlignmentFrame.html" title="class in org.biojava.nbio.structure.gui.util">AlternativeAlignmentFrame</a></td>
<th class="colLast" scope="row">
<div class="block">a frame showing the alternative alignments, which are the result of a structure superimposition</div>
</th>
</tr>
<tr id="i85" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/compound/AmbiguityDNACompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">AmbiguityDNACompoundSet</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i86" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/compound/AmbiguityDNARNAHybridCompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">AmbiguityDNARNAHybridCompoundSet</a></td>
<th class="colLast" scope="row">
<div class="block">Ambiguity set for hybrid DNA/RNA sequences.</div>
</th>
</tr>
<tr id="i87" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/compound/AmbiguityRNACompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">AmbiguityRNACompoundSet</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i88" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a></td>
<th class="colLast" scope="row">
<div class="block">
 A <a href="org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure"><code>Group</code></a> that represents an AminoAcid.</div>
</th>
</tr>
<tr id="i89" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/xml/AminoAcidComposition.html" title="class in org.biojava.nbio.aaproperties.xml">AminoAcidComposition</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i90" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/xml/AminoAcidCompositionTable.html" title="class in org.biojava.nbio.aaproperties.xml">AminoAcidCompositionTable</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i91" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a></td>
<th class="colLast" scope="row">
<div class="block">Used to describe an Amino Acid.</div>
</th>
</tr>
<tr id="i92" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompoundSet</a></td>
<th class="colLast" scope="row">
<div class="block">Set of proteinogenic amino acids.</div>
</th>
</tr>
<tr id="i93" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/AminoAcidImpl.html" title="class in org.biojava.nbio.structure">AminoAcidImpl</a></td>
<th class="colLast" scope="row">
<div class="block">AminoAcid inherits most from Hetatom.</div>
</th>
</tr>
<tr id="i94" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/AminoAcidProperties.html" title="class in org.biojava.nbio.aaproperties">AminoAcidProperties</a></td>
<th class="colLast" scope="row">
<div class="block">This class provides the protein properties at the level of individual amino acids.</div>
</th>
</tr>
<tr id="i95" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.routines">AnchoredPairwiseSequenceAligner</a>&lt;<a href="org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html" title="type parameter in AnchoredPairwiseSequenceAligner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html" title="type parameter in AnchoredPairwiseSequenceAligner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html" title="type parameter in AnchoredPairwiseSequenceAligner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">This algorithm uses a divide-and-conquer approach to find optimal pairwise global sequence alignments (from the
 first until the last <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>) with the restriction that any alignment produced
 will connect the query sequence to the target sequence at the <em>anchors</em>.</div>
</th>
</tr>
<tr id="i96" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/AngleOrderDetectorPlus.html" title="class in org.biojava.nbio.structure.symmetry.internal">AngleOrderDetectorPlus</a></td>
<th class="colLast" scope="row">
<div class="block">Guesses an order of rotational symmetry from the angle.</div>
</th>
</tr>
<tr id="i97" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/utils/Annotatable.html" title="interface in org.biojava.nbio.ontology.utils">Annotatable</a></td>
<th class="colLast" scope="row">
<div class="block">Indicates that an object has an associated annotation.</div>
</th>
</tr>
<tr id="i98" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/utils/Annotation.html" title="interface in org.biojava.nbio.ontology.utils">Annotation</a></td>
<th class="colLast" scope="row">
<div class="block">
 Arbitrary annotation associated with one or more objects.</div>
</th>
</tr>
<tr id="i99" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/App.html" title="class in org.biojava.nbio.genome">App</a></td>
<th class="colLast" scope="row">
<div class="block">Hello world!</div>
</th>
</tr>
<tr id="i100" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/App.html" title="class in org.biojava.nbio.ontology">App</a></td>
<th class="colLast" scope="row">
<div class="block">Hello world!</div>
</th>
</tr>
<tr id="i101" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/loader/ArrayListProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">ArrayListProxySequenceReader</a>&lt;<a href="org/biojava/nbio/core/sequence/loader/ArrayListProxySequenceReader.html" title="type parameter in ArrayListProxySequenceReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i102" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">ArrayListSequenceReader</a>&lt;<a href="org/biojava/nbio/core/sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Stores a Sequence as a collection of compounds in an ArrayList</div>
</th>
</tr>
<tr id="i103" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/asa/AsaCalculator.html" title="class in org.biojava.nbio.structure.asa">AsaCalculator</a></td>
<th class="colLast" scope="row">
<div class="block">Class to calculate Accessible Surface Areas based on
 the rolling ball algorithm by Shrake and Rupley.</div>
</th>
</tr>
<tr id="i104" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/utils/AssertionFailure.html" title="class in org.biojava.nbio.ontology.utils">AssertionFailure</a></td>
<th class="colLast" scope="row">
<div class="block">An unchecked exception representing an Assertion failure.</div>
</th>
</tr>
<tr id="i105" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/Astral.html" title="class in org.biojava.nbio.structure.scop">Astral</a></td>
<th class="colLast" scope="row">
<div class="block">Provides programmatic access to ASTRAL representative sets.</div>
</th>
</tr>
<tr id="i106" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/Astral.AstralSet.html" title="enum in org.biojava.nbio.structure.scop">Astral.AstralSet</a></td>
<th class="colLast" scope="row">
<div class="block">An ASTRAL sequence-identity cutoff with an identifier such as:</div>
</th>
</tr>
<tr id="i107" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></td>
<th class="colLast" scope="row">
<div class="block">A simple interface for an Atom.</div>
</th>
</tr>
<tr id="i108" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a></td>
<th class="colLast" scope="row">
<div class="block">A utility class that provides easy access to Structure objects.</div>
</th>
</tr>
<tr id="i109" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/AtomContact.html" title="class in org.biojava.nbio.structure.contact">AtomContact</a></td>
<th class="colLast" scope="row">
<div class="block">A pair of atoms that are in contact</div>
</th>
</tr>
<tr id="i110" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/AtomContactSet.html" title="class in org.biojava.nbio.structure.contact">AtomContactSet</a></td>
<th class="colLast" scope="row">
<div class="block">A set of atom-atom contacts to hold the results of intra and inter-chain contact calculations</div>
</th>
</tr>
<tr id="i111" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/AtomIdentifier.html" title="class in org.biojava.nbio.structure.contact">AtomIdentifier</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i112" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/AtomImpl.html" title="class in org.biojava.nbio.structure">AtomImpl</a></td>
<th class="colLast" scope="row">
<div class="block">Implementation of an Atom of a PDB file.</div>
</th>
</tr>
<tr id="i113" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/jmol/AtomInfo.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AtomInfo</a></td>
<th class="colLast" scope="row">
<div class="block">This class uniquely describes an atom</div>
</th>
</tr>
<tr id="i114" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/jmol/AtomInfoParser.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AtomInfoParser</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i115" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/AtomIterator.html" title="class in org.biojava.nbio.structure">AtomIterator</a></td>
<th class="colLast" scope="row">
<div class="block">an iterator over all atoms of a structure / group.</div>
</th>
</tr>
<tr id="i116" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/AtomPositionMap.html" title="class in org.biojava.nbio.structure">AtomPositionMap</a></td>
<th class="colLast" scope="row">
<div class="block">A map from <a href="org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure"><code>ResidueNumbers</code></a> to ATOM record positions in a PDB file.</div>
</th>
</tr>
<tr id="i117" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/AtomPositionMap.GroupMatcher.html" title="interface in org.biojava.nbio.structure">AtomPositionMap.GroupMatcher</a></td>
<th class="colLast" scope="row">
<div class="block">Used as a Predicate to indicate whether a particular Atom should be mapped</div>
</th>
</tr>
<tr id="i118" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/AugmentedResidueRange.html" title="class in org.biojava.nbio.structure">AugmentedResidueRange</a></td>
<th class="colLast" scope="row">
<div class="block">Created by douglas on 1/23/15.</div>
</th>
</tr>
<tr id="i119" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/Author.html" title="class in org.biojava.nbio.structure">Author</a></td>
<th class="colLast" scope="row">
<div class="block">Describes author attributes for author information in a PDB file.</div>
</th>
</tr>
<tr id="i120" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/autosuggest/AutoSuggestProvider.html" title="interface in org.biojava.nbio.structure.align.gui.autosuggest">AutoSuggestProvider</a></td>
<th class="colLast" scope="row">
<div class="block">A class that provides auto-completion suggestions for JAutoSuggest</div>
</th>
</tr>
<tr id="i121" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/axis/AxisAligner.html" title="class in org.biojava.nbio.structure.symmetry.axis">AxisAligner</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i122" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/basepairs/BasePairParameters.html" title="class in org.biojava.nbio.structure.basepairs">BasePairParameters</a></td>
<th class="colLast" scope="row">
<div class="block">This module calculates the el Hassan-Calladine Base Pairing and Base-pair Step Parameters for any nucleic
 acid containing structure that has the information about the core base-pair rings.</div>
</th>
</tr>
<tr id="i123" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/BasicSequence.html" title="class in org.biojava.nbio.core.sequence">BasicSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/BasicSequence.html" title="type parameter in BasicSequence">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Bare bones version of the Sequence object to be used sparingly.</div>
</th>
</tr>
<tr id="i124" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/BcifFileReader.html" title="class in org.biojava.nbio.structure.io">BcifFileReader</a></td>
<th class="colLast" scope="row">
<div class="block">Parse binary Cif files and provide capabilities to store them locally.</div>
</th>
</tr>
<tr id="i125" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/BerkeleyScopInstallation.html" title="class in org.biojava.nbio.structure.scop">BerkeleyScopInstallation</a></td>
<th class="colLast" scope="row">
<div class="block">SCOPe:

 The Structural Classification of Proteins (extended) at Berkeley Lab and UC Berkeley
 (<a href="http://scop.berkeley.edu/">http://scop.berkeley.edu/</a>).</div>
</th>
</tr>
<tr id="i126" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/secstruc/BetaBridge.html" title="class in org.biojava.nbio.structure.secstruc">BetaBridge</a></td>
<th class="colLast" scope="row">
<div class="block">Container that represents a beta Bridge between two residues.</div>
</th>
</tr>
<tr id="i127" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/BioAssemblyIdentifier.html" title="class in org.biojava.nbio.structure">BioAssemblyIdentifier</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i128" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/quaternary/BioAssemblyInfo.html" title="class in org.biojava.nbio.structure.quaternary">BioAssemblyInfo</a></td>
<th class="colLast" scope="row">
<div class="block">Representation of a Biological Assembly annotation as provided by the PDB.</div>
</th>
</tr>
<tr id="i129" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/quaternary/BioAssemblyTools.html" title="class in org.biojava.nbio.structure.quaternary">BioAssemblyTools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i130" class="altColor">
<td class="colFirst"><a href="demo/BioJavaAADemo.html" title="class in demo">BioJavaAADemo</a></td>
<th class="colLast" scope="row">
<div class="block">Created by andreas on 8/9/14.</div>
</th>
</tr>
<tr id="i131" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/BiojavaJmol.html" title="class in org.biojava.nbio.structure.gui">BiojavaJmol</a></td>
<th class="colLast" scope="row">
<div class="block">A class that provides a simple GUI for Jmol</div>
</th>
</tr>
<tr id="i132" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/BioJavaStructureAlignment.html" title="class in org.biojava.nbio.structure.align">BioJavaStructureAlignment</a></td>
<th class="colLast" scope="row">
<div class="block">Wrapper for the BioJava Structure Alignment Implementation</div>
</th>
</tr>
<tr id="i133" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/quaternary/BiologicalAssemblyBuilder.html" title="class in org.biojava.nbio.structure.quaternary">BiologicalAssemblyBuilder</a></td>
<th class="colLast" scope="row">
<div class="block">Reconstructs the quaternary structure of a protein from an asymmetric unit</div>
</th>
</tr>
<tr id="i134" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/quaternary/BiologicalAssemblyTransformation.html" title="class in org.biojava.nbio.structure.quaternary">BiologicalAssemblyTransformation</a></td>
<th class="colLast" scope="row">
<div class="block">The transformation needed for generation of biological assemblies
 from the contents of a PDB/mmCIF file.</div>
</th>
</tr>
<tr id="i135" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">BitSequenceReader</a>&lt;<a href="org/biojava/nbio/core/sequence/storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">An implementation of the popular bit encodings.</div>
</th>
</tr>
<tr id="i136" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava.nbio.core.sequence.storage">BitSequenceReader.BitArrayWorker</a>&lt;<a href="org/biojava/nbio/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">The logic of working with a bit has been separated out into this class
 to help developers create the bit data structures without having to
 put the code into an intermediate format and to also use the format
 without the need to copy this code.</div>
</th>
</tr>
<tr id="i137" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/qblast/BlastAlignmentParameterEnum.html" title="enum in org.biojava.nbio.ws.alignment.qblast">BlastAlignmentParameterEnum</a></td>
<th class="colLast" scope="row">
<div class="block">Alignment request parameters accepted by QBlast service.<br/>
 Not all are mandatory.</div>
</th>
</tr>
<tr id="i138" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/utils/BlastClustReader.html" title="class in org.biojava.nbio.structure.symmetry.utils">BlastClustReader</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i139" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/search/io/blast/BlastHitBuilder.html" title="class in org.biojava.nbio.core.search.io.blast">BlastHitBuilder</a></td>
<th class="colLast" scope="row">
<div class="block">Designed by Paolo Pavan.</div>
</th>
</tr>
<tr id="i140" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/homology/BlastHomologyHits.html" title="class in org.biojava.nbio.genome.homology">BlastHomologyHits</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i141" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/search/io/blast/BlastHspBuilder.html" title="class in org.biojava.nbio.core.search.io.blast">BlastHspBuilder</a></td>
<th class="colLast" scope="row">
<div class="block">Designed by Paolo Pavan.</div>
</th>
</tr>
<tr id="i142" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/qblast/BlastJob.html" title="class in org.biojava.nbio.ws.alignment.qblast">BlastJob</a></td>
<th class="colLast" scope="row">
<div class="block">Information about QBlast search job</div>
</th>
</tr>
<tr id="i143" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/qblast/BlastMatrixEnum.html" title="enum in org.biojava.nbio.ws.alignment.qblast">BlastMatrixEnum</a></td>
<th class="colLast" scope="row">
<div class="block">Enum representing matrices supported by QBlast</div>
</th>
</tr>
<tr id="i144" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/qblast/BlastOutputAlignmentFormatEnum.html" title="enum in org.biojava.nbio.ws.alignment.qblast">BlastOutputAlignmentFormatEnum</a></td>
<th class="colLast" scope="row">
<div class="block">Enum representing available output alignment types.</div>
</th>
</tr>
<tr id="i145" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/qblast/BlastOutputFormatEnum.html" title="enum in org.biojava.nbio.ws.alignment.qblast">BlastOutputFormatEnum</a></td>
<th class="colLast" scope="row">
<div class="block">Enum representing available output formats.</div>
</th>
</tr>
<tr id="i146" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/qblast/BlastOutputParameterEnum.html" title="enum in org.biojava.nbio.ws.alignment.qblast">BlastOutputParameterEnum</a></td>
<th class="colLast" scope="row">
<div class="block">Output parameters accepted by QBlast service.</div>
</th>
</tr>
<tr id="i147" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/qblast/BlastProgramEnum.html" title="enum in org.biojava.nbio.ws.alignment.qblast">BlastProgramEnum</a></td>
<th class="colLast" scope="row">
<div class="block">Enum representing available blast programs.</div>
</th>
</tr>
<tr id="i148" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/search/io/blast/BlastResult.html" title="class in org.biojava.nbio.core.search.io.blast">BlastResult</a></td>
<th class="colLast" scope="row">
<div class="block">This class models a Blast/Blast plus result.</div>
</th>
</tr>
<tr id="i149" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/search/io/blast/BlastResultBuilder.html" title="class in org.biojava.nbio.core.search.io.blast">BlastResultBuilder</a></td>
<th class="colLast" scope="row">
<div class="block">Designed by Paolo Pavan.</div>
</th>
</tr>
<tr id="i150" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/search/io/blast/BlastTabularParser.html" title="class in org.biojava.nbio.core.search.io.blast">BlastTabularParser</a></td>
<th class="colLast" scope="row">
<div class="block">Designed by Paolo Pavan.</div>
</th>
</tr>
<tr id="i151" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/search/io/blast/BlastXMLParser.html" title="class in org.biojava.nbio.core.search.io.blast">BlastXMLParser</a></td>
<th class="colLast" scope="row">
<div class="block">Re-designed by Paolo Pavan on the footprint of:
 org.biojava.nbio.genome.query.BlastXMLQuery by Scooter Willis <willishf at gmail dot com>

 You may want to find my contacts on Github and LinkedIn for code info
 or discuss major changes.</div>
</th>
</tr>
<tr id="i152" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/query/BlastXMLQuery.html" title="class in org.biojava.nbio.genome.query">BlastXMLQuery</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i153" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/Block.html" title="interface in org.biojava.nbio.structure.align.multiple">Block</a></td>
<th class="colLast" scope="row">
<div class="block">A Block is a Data Structure that stores aligned positions of a
 <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> with the condition that residues are in a
 sequential order.</div>
</th>
</tr>
<tr id="i154" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/BlockImpl.html" title="class in org.biojava.nbio.structure.align.multiple">BlockImpl</a></td>
<th class="colLast" scope="row">
<div class="block">General implementation of a <a href="org/biojava/nbio/structure/align/multiple/Block.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>Block</code></a> that supports any type of
 sequential alignment with gaps.</div>
</th>
</tr>
<tr id="i155" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/BlockSet.html" title="interface in org.biojava.nbio.structure.align.multiple">BlockSet</a></td>
<th class="colLast" scope="row">
<div class="block">A BlockSet is a Data Structure to store a flexible alignment part of a
 multiple alignment.</div>
</th>
</tr>
<tr id="i156" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/BlockSetImpl.html" title="class in org.biojava.nbio.structure.align.multiple">BlockSetImpl</a></td>
<th class="colLast" scope="row">
<div class="block">A general implementation of a BlockSet to store a flexible part of a multiple
 alignment.</div>
</th>
</tr>
<tr id="i157" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/Bond.html" title="interface in org.biojava.nbio.structure">Bond</a></td>
<th class="colLast" scope="row">
<div class="block">A simple bond -- it stores information about two atoms as well as information
 about its bond order.</div>
</th>
</tr>
<tr id="i158" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/BondImpl.html" title="class in org.biojava.nbio.structure">BondImpl</a></td>
<th class="colLast" scope="row">
<div class="block">A simple bond -- it stores information about two atoms as well as information
 about its bond order.</div>
</th>
</tr>
<tr id="i159" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/BondMaker.html" title="class in org.biojava.nbio.structure.io">BondMaker</a></td>
<th class="colLast" scope="row">
<div class="block">Adds polymer bonds for peptides and nucleotides based on distance cutoffs and
 intra-group (residue) bonds based on data from the Chemical Component Dictionary
 to the Structure object.</div>
</th>
</tr>
<tr id="i160" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/BondType.html" title="enum in org.biojava.nbio.structure">BondType</a></td>
<th class="colLast" scope="row">
<div class="block">Work in progress - NOT final!</div>
</th>
</tr>
<tr id="i161" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a></td>
<th class="colLast" scope="row">
<div class="block">A bounding box for short cutting some geometrical calculations.</div>
</th>
</tr>
<tr id="i162" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/xtal/BravaisLattice.html" title="enum in org.biojava.nbio.structure.xtal">BravaisLattice</a></td>
<th class="colLast" scope="row">
<div class="block">An enum to represent the 7 Bravais lattices</div>
</th>
</tr>
<tr id="i163" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/secstruc/BridgeType.html" title="enum in org.biojava.nbio.structure.secstruc">BridgeType</a></td>
<th class="colLast" scope="row">
<div class="block">A bridge is formed by two non-overlapping stretches of three residues each
 (i-1,i,i+1) and (j-1,j,j+1), where i<j.</div>
</th>
</tr>
<tr id="i164" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/webstart/BrowserOpener.html" title="class in org.biojava.nbio.structure.align.webstart">BrowserOpener</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i165" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava.nbio.core.sequence.io">BufferedReaderBytesRead</a></td>
<th class="colLast" scope="row">
<div class="block">Need to keep track of actual bytes read and take advantage of buffered reader
 performance.</div>
</th>
</tr>
<tr id="i166" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/C2RotationSolver.html" title="class in org.biojava.nbio.structure.symmetry.core">C2RotationSolver</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i167" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/CAConverter.html" title="class in org.biojava.nbio.structure.io">CAConverter</a></td>
<th class="colLast" scope="row">
<div class="block">Converts full atom representations to Calpha only ones.</div>
</th>
</tr>
<tr id="i168" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/Calc.html" title="class in org.biojava.nbio.structure">Calc</a></td>
<th class="colLast" scope="row">
<div class="block">Utility operations on Atoms, AminoAcids, Matrices, Point3d, etc.</div>
</th>
</tr>
<tr id="i169" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/geometry/CalcPoint.html" title="class in org.biojava.nbio.structure.geometry">CalcPoint</a></td>
<th class="colLast" scope="row">
<div class="block">Utility operations on Point3d.</div>
</th>
</tr>
<tr id="i170" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/CallablePairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.template">CallablePairwiseSequenceAligner</a>&lt;<a href="org/biojava/nbio/alignment/template/CallablePairwiseSequenceAligner.html" title="type parameter in CallablePairwiseSequenceAligner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/CallablePairwiseSequenceAligner.html" title="type parameter in CallablePairwiseSequenceAligner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/CallablePairwiseSequenceAligner.html" title="type parameter in CallablePairwiseSequenceAligner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements a concurrency wrapper for a <a href="org/biojava/nbio/alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template"><code>PairwiseSequenceAligner</code></a>.</div>
</th>
</tr>
<tr id="i171" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/CallablePairwiseSequenceScorer.html" title="class in org.biojava.nbio.alignment.template">CallablePairwiseSequenceScorer</a>&lt;<a href="org/biojava/nbio/alignment/template/CallablePairwiseSequenceScorer.html" title="type parameter in CallablePairwiseSequenceScorer">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/CallablePairwiseSequenceScorer.html" title="type parameter in CallablePairwiseSequenceScorer">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/CallablePairwiseSequenceScorer.html" title="type parameter in CallablePairwiseSequenceScorer">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements a concurrency wrapper for a <a href="org/biojava/nbio/alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template"><code>PairwiseSequenceScorer</code></a>.</div>
</th>
</tr>
<tr id="i172" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/CallableProfileProfileAligner.html" title="class in org.biojava.nbio.alignment.template">CallableProfileProfileAligner</a>&lt;<a href="org/biojava/nbio/alignment/template/CallableProfileProfileAligner.html" title="type parameter in CallableProfileProfileAligner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/CallableProfileProfileAligner.html" title="type parameter in CallableProfileProfileAligner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/CallableProfileProfileAligner.html" title="type parameter in CallableProfileProfileAligner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements a concurrency wrapper for a <a href="org/biojava/nbio/alignment/template/ProfileProfileAligner.html" title="interface in org.biojava.nbio.alignment.template"><code>ProfileProfileAligner</code></a>.</div>
</th>
</tr>
<tr id="i173" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/CallableStructureAlignment.html" title="class in org.biojava.nbio.structure.align">CallableStructureAlignment</a></td>
<th class="colLast" scope="row">
<div class="block">Simple Callable Class that calculates a pairwise alignment in a different
 thread, so that multiple pairwise alignments can be run in parallel
 (examples: all-to-all alignments, DB search alignments).</div>
</th>
</tr>
<tr id="i174" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/quaternary/CartesianProduct.html" title="class in org.biojava.nbio.structure.quaternary">CartesianProduct</a>&lt;<a href="org/biojava/nbio/structure/quaternary/CartesianProduct.html" title="type parameter in CartesianProduct">T</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A cartesian product between two lists A and B is the set of all ordered pairs
 of the elements of both sets.</div>
</th>
</tr>
<tr id="i175" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/xml/CaseFreeAminoAcidCompoundSet.html" title="class in org.biojava.nbio.aaproperties.xml">CaseFreeAminoAcidCompoundSet</a></td>
<th class="colLast" scope="row">
<div class="block">Set of proteinogenic amino acids.</div>
</th>
</tr>
<tr id="i176" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava.nbio.core.sequence.transcription">CaseInsensitiveCompound</a></td>
<th class="colLast" scope="row">
<div class="block">Attempts to wrap compounds so it is possible to view them
 in a case insensitive manner</div>
</th>
</tr>
<tr id="i177" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">CasePreservingProteinSequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">A sequence creator which preserves the case of its input string in
 the user collection of the returned ProteinSequence.</div>
</th>
</tr>
<tr id="i178" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cath/CathCategory.html" title="enum in org.biojava.nbio.structure.cath">CathCategory</a></td>
<th class="colLast" scope="row">
<div class="block">The categories found within CATH.</div>
</th>
</tr>
<tr id="i179" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath">CathDatabase</a></td>
<th class="colLast" scope="row">
<div class="block">General API for interacting with CATH.</div>
</th>
</tr>
<tr id="i180" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cath/CathDomain.html" title="class in org.biojava.nbio.structure.cath">CathDomain</a></td>
<th class="colLast" scope="row">
<div class="block">A class which represents a single CATH domain.</div>
</th>
</tr>
<tr id="i181" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cath/CathFactory.html" title="class in org.biojava.nbio.structure.cath">CathFactory</a></td>
<th class="colLast" scope="row">
<div class="block">Controls global <a href="org/biojava/nbio/structure/cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath"><code>CathDatabases</code></a> being used.</div>
</th>
</tr>
<tr id="i182" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cath/CathFragment.html" title="class in org.biojava.nbio.structure.cath">CathFragment</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i183" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cath/CathInstallation.html" title="class in org.biojava.nbio.structure.cath">CathInstallation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i184" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cath/CathNode.html" title="class in org.biojava.nbio.structure.cath">CathNode</a></td>
<th class="colLast" scope="row">
<div class="block">Represents a node in the CATH hierarchy.</div>
</th>
</tr>
<tr id="i185" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cath/CathSegment.html" title="class in org.biojava.nbio.structure.cath">CathSegment</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i186" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/CDSComparator.html" title="class in org.biojava.nbio.core.sequence">CDSComparator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i187" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a></td>
<th class="colLast" scope="row">
<div class="block">Represents a exon or coding sequence in a gene.</div>
</th>
</tr>
<tr id="i188" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a></td>
<th class="colLast" scope="row">
<div class="block">This is based on the original Combinatorial Extension (CE) source code from 2003 or 2004 (CE version 2.3),
 as has been originally developed by I.</div>
</th>
</tr>
<tr id="i189" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html" title="class in org.biojava.nbio.structure.align.ce">CeCalculatorEnhanced</a></td>
<th class="colLast" scope="row">
<div class="block">This is based on the original Combinatorial Extension (CE) source code from 2003 or 2004 (CE version 2.3),
 as has been originally developed by I.</div>
</th>
</tr>
<tr id="i190" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CeCPMain.html" title="class in org.biojava.nbio.structure.align.ce">CeCPMain</a></td>
<th class="colLast" scope="row">
<div class="block">A wrapper for <a href="org/biojava/nbio/structure/align/ce/CeMain.html" title="class in org.biojava.nbio.structure.align.ce"><code>CeMain</code></a> which sets default parameters to be appropriate for finding
 circular permutations.</div>
</th>
</tr>
<tr id="i191" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CeCPMain.CPRange.html" title="class in org.biojava.nbio.structure.align.ce">CeCPMain.CPRange</a></td>
<th class="colLast" scope="row">
<div class="block">Tiny wrapper for the disallowed regions of an alignment.</div>
</th>
</tr>
<tr id="i192" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a></td>
<th class="colLast" scope="row">
<div class="block">Provides parameters to <a href="org/biojava/nbio/structure/align/ce/CeCPMain.html" title="class in org.biojava.nbio.structure.align.ce"><code>CeCPMain</code></a></div>
</th>
</tr>
<tr id="i193" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CECPParameters.DuplicationHint.html" title="enum in org.biojava.nbio.structure.align.ce">CECPParameters.DuplicationHint</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i194" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CeCPUserArgumentProcessor.html" title="class in org.biojava.nbio.structure.align.ce">CeCPUserArgumentProcessor</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i195" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CeMain.html" title="class in org.biojava.nbio.structure.align.ce">CeMain</a></td>
<th class="colLast" scope="row">
<div class="block">The main class of the Java implementation of the Combinatorial Extension Algorithm (CE),
 as has been originally developed by I.</div>
</th>
</tr>
<tr id="i196" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/figure/CensorStatus.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">CensorStatus</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i197" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/figure/CensorStatusSelect.html" title="interface in org.biojava.nbio.survival.kaplanmeier.figure">CensorStatusSelect</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i198" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CeParameters.html" title="class in org.biojava.nbio.structure.align.ce">CeParameters</a></td>
<th class="colLast" scope="row">
<div class="block">Contains the parameters that can be sent to CE</div>
</th>
</tr>
<tr id="i199" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CeParameters.ScoringStrategy.html" title="enum in org.biojava.nbio.structure.align.ce">CeParameters.ScoringStrategy</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i200" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CeSideChainMain.html" title="class in org.biojava.nbio.structure.align.ce">CeSideChainMain</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i201" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CeSideChainUserArgumentProcessor.html" title="class in org.biojava.nbio.structure.align.ce">CeSideChainUserArgumentProcessor</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i202" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/CeSymm.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymm</a></td>
<th class="colLast" scope="row">
<div class="block">Identify the symmetries in a structure by running an alignment of the
 structure against itself disabling the diagonal of the identity alignment.</div>
</th>
</tr>
<tr id="i203" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/CeSymmIterative.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmIterative</a></td>
<th class="colLast" scope="row">
<div class="block">Iterative version of CeSymm that aims at identifying all symmetry axis of a
 structure.</div>
</th>
</tr>
<tr id="i204" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a></td>
<th class="colLast" scope="row">
<div class="block">Provides parameters to <a href="org/biojava/nbio/structure/symmetry/internal/CeSymm.html" title="class in org.biojava.nbio.structure.symmetry.internal"><code>CeSymm</code></a>.</div>
</th>
</tr>
<tr id="i205" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/CESymmParameters.OrderDetectorMethod.html" title="enum in org.biojava.nbio.structure.symmetry.internal">CESymmParameters.OrderDetectorMethod</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i206" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/CESymmParameters.RefineMethod.html" title="enum in org.biojava.nbio.structure.symmetry.internal">CESymmParameters.RefineMethod</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i207" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/CESymmParameters.SymmetryType.html" title="enum in org.biojava.nbio.structure.symmetry.internal">CESymmParameters.SymmetryType</a></td>
<th class="colLast" scope="row">
<div class="block">The internal symmetry detection can be divided into two types: CLOSE:
 includes the circular and dihedral symmetries, and OPEN: includes the
 helical and protein repeats symmetries.</div>
</th>
</tr>
<tr id="i208" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></td>
<th class="colLast" scope="row">
<div class="block">This Class stores all the relevant information of an internal symmetry result
 obtained with CeSymm.</div>
</th>
</tr>
<tr id="i209" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CeUserArgumentProcessor.html" title="class in org.biojava.nbio.structure.align.ce">CeUserArgumentProcessor</a></td>
<th class="colLast" scope="row">
<div class="block">process the arguments from command line</div>
</th>
</tr>
<tr id="i210" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/CeUserArgumentProcessor.CeStartupParams.html" title="class in org.biojava.nbio.structure.align.ce">CeUserArgumentProcessor.CeStartupParams</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i211" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a></td>
<th class="colLast" scope="row">
<div class="block">
 Defines the interface for a Chain.</div>
</th>
</tr>
<tr id="i212" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/ChainImpl.html" title="class in org.biojava.nbio.structure">ChainImpl</a></td>
<th class="colLast" scope="row">
<div class="block">A Chain in a PDB file.</div>
</th>
</tr>
<tr id="i213" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/misc/ChainSignature.html" title="class in org.biojava.nbio.structure.symmetry.misc">ChainSignature</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i214" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/data/ChangeValue.html" title="interface in org.biojava.nbio.survival.data">ChangeValue</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i215" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/ChargeAdder.html" title="class in org.biojava.nbio.structure.io">ChargeAdder</a></td>
<th class="colLast" scope="row">
<div class="block">A class to add appropriate charge information to a structure.</div>
</th>
</tr>
<tr id="i216" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/ChemComp.html" title="class in org.biojava.nbio.structure.chem">ChemComp</a></td>
<th class="colLast" scope="row">
<div class="block">Properties of a chemical component.</div>
</th>
</tr>
<tr id="i217" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/ChemCompAtom.html" title="class in org.biojava.nbio.structure.chem">ChemCompAtom</a></td>
<th class="colLast" scope="row">
<div class="block">Properties of an atom of a chemical component.</div>
</th>
</tr>
<tr id="i218" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/ChemCompBond.html" title="class in org.biojava.nbio.structure.chem">ChemCompBond</a></td>
<th class="colLast" scope="row">
<div class="block">Properties of a bond in a chemical component.</div>
</th>
</tr>
<tr id="i219" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/ChemCompConsumer.html" title="interface in org.biojava.nbio.structure.io.cif">ChemCompConsumer</a></td>
<th class="colLast" scope="row">
<div class="block">Create the <a href="org/biojava/nbio/structure/chem/ChemicalComponentDictionary.html" title="class in org.biojava.nbio.structure.chem"><code>ChemicalComponentDictionary</code></a> from CIF data.</div>
</th>
</tr>
<tr id="i220" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/ChemCompConsumerImpl.html" title="class in org.biojava.nbio.structure.io.cif">ChemCompConsumerImpl</a></td>
<th class="colLast" scope="row">
<div class="block">Consumes a CCD file to create the <a href="org/biojava/nbio/structure/chem/ChemicalComponentDictionary.html" title="class in org.biojava.nbio.structure.chem"><code>ChemicalComponentDictionary</code></a>.</div>
</th>
</tr>
<tr id="i221" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/ChemCompConverter.html" title="class in org.biojava.nbio.structure.io.cif">ChemCompConverter</a></td>
<th class="colLast" scope="row">
<div class="block">Convert CifFiles to chem comps.</div>
</th>
</tr>
<tr id="i222" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/ChemCompDescriptor.html" title="class in org.biojava.nbio.structure.chem">ChemCompDescriptor</a></td>
<th class="colLast" scope="row">
<div class="block">Properties of the chemical component descriptor.</div>
</th>
</tr>
<tr id="i223" class="rowColor">
<td class="colFirst"><a href="demo/ChemCompDistribution.html" title="class in demo">ChemCompDistribution</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i224" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/ChemCompGroupFactory.html" title="class in org.biojava.nbio.structure.chem">ChemCompGroupFactory</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i225" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/ChemCompProvider.html" title="interface in org.biojava.nbio.structure.chem">ChemCompProvider</a></td>
<th class="colLast" scope="row">
<div class="block">Interface that is implemented by all classes that can provide <a href="org/biojava/nbio/structure/chem/ChemComp.html" title="class in org.biojava.nbio.structure.chem"><code>ChemComp</code></a> definitions.</div>
</th>
</tr>
<tr id="i226" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/ChemCompTools.html" title="class in org.biojava.nbio.structure.chem">ChemCompTools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i227" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/ChemicalComponentDictionary.html" title="class in org.biojava.nbio.structure.chem">ChemicalComponentDictionary</a></td>
<th class="colLast" scope="row">
<div class="block">A representation of the Chemical Component Dictionary.</div>
</th>
</tr>
<tr id="i228" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/stats/ChiSq.html" title="class in org.biojava.nbio.survival.cox.stats">ChiSq</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i229" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/stats/Cholesky2.html" title="class in org.biojava.nbio.survival.cox.stats">Cholesky2</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i230" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/jama/CholeskyDecomposition.html" title="class in org.biojava.nbio.structure.jama">CholeskyDecomposition</a></td>
<th class="colLast" scope="row">
<div class="block">Cholesky Decomposition.</div>
</th>
</tr>
<tr id="i231" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/ChooseDirAction.html" title="class in org.biojava.nbio.structure.align.gui">ChooseDirAction</a></td>
<th class="colLast" scope="row">
<div class="block">Ask the user to provide a directory containting PDB files.</div>
</th>
</tr>
<tr id="i232" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/util/ChromosomeMappingTools.html" title="class in org.biojava.nbio.genome.util">ChromosomeMappingTools</a></td>
<th class="colLast" scope="row">
<div class="block">A class that can map chromosomal positions to mRNA (coding sequence) positions.</div>
</th>
</tr>
<tr id="i233" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a></td>
<th class="colLast" scope="row">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</th>
</tr>
<tr id="i234" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a></td>
<th class="colLast" scope="row">
<div class="block">Created by ap3 on 27/10/2014.</div>
</th>
</tr>
<tr id="i235" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/stats/Chsolve2.html" title="class in org.biojava.nbio.survival.cox.stats">Chsolve2</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i236" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/CifBean.html" title="interface in org.biojava.nbio.structure.io.cif">CifBean</a></td>
<th class="colLast" scope="row">
<div class="block">Flag for BioJava beans that resemble categories defined by the mmCIF schema.</div>
</th>
</tr>
<tr id="i237" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/CifChainSupplierImpl.html" title="class in org.biojava.nbio.structure.io.cif">CifChainSupplierImpl</a></td>
<th class="colLast" scope="row">
<div class="block">Convert a chain to a <code>CifFile</code>.</div>
</th>
</tr>
<tr id="i238" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/CifFileConsumer.html" title="interface in org.biojava.nbio.structure.io.cif">CifFileConsumer</a>&lt;<a href="org/biojava/nbio/structure/io/cif/CifFileConsumer.html" title="type parameter in CifFileConsumer">S</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a rather generic interface which allows to populate some data structure with data parsed from a CIF file.</div>
</th>
</tr>
<tr id="i239" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/CifFileReader.html" title="class in org.biojava.nbio.structure.io">CifFileReader</a></td>
<th class="colLast" scope="row">
<div class="block">Parse text Cif files and provide capabilities to store them locally.</div>
</th>
</tr>
<tr id="i240" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/CifFileSupplier.html" title="interface in org.biojava.nbio.structure.io.cif">CifFileSupplier</a>&lt;<a href="org/biojava/nbio/structure/io/cif/CifFileSupplier.html" title="type parameter in CifFileSupplier">S</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Create a CifFile instance for a given container of structure data.</div>
</th>
</tr>
<tr id="i241" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/CifStructureConsumer.html" title="interface in org.biojava.nbio.structure.io.cif">CifStructureConsumer</a></td>
<th class="colLast" scope="row">
<div class="block">Defines the categories to consume during CIF parsing.</div>
</th>
</tr>
<tr id="i242" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/CifStructureConsumerImpl.html" title="class in org.biojava.nbio.structure.io.cif">CifStructureConsumerImpl</a></td>
<th class="colLast" scope="row">
<div class="block">An implementation of a CifFileConsumer for BioJava.</div>
</th>
</tr>
<tr id="i243" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/CifStructureConverter.html" title="class in org.biojava.nbio.structure.io.cif">CifStructureConverter</a></td>
<th class="colLast" scope="row">
<div class="block">Convert BioJava structures to CifFiles and vice versa.</div>
</th>
</tr>
<tr id="i244" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/CifStructureSupplierImpl.html" title="class in org.biojava.nbio.structure.io.cif">CifStructureSupplierImpl</a></td>
<th class="colLast" scope="row">
<div class="block">Convert a structure to a CifFile.</div>
</th>
</tr>
<tr id="i245" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava.nbio.core.sequence.io.util">ClasspathResource</a></td>
<th class="colLast" scope="row">
<div class="block">This object represents a classpath resource on the local system.</div>
</th>
</tr>
<tr id="i246" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/metadata/ClinicalMetaDataOutcome.html" title="class in org.biojava.nbio.survival.kaplanmeier.metadata">ClinicalMetaDataOutcome</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i247" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/CliTools.html" title="class in org.biojava.nbio.structure.align.util">CliTools</a></td>
<th class="colLast" scope="row">
<div class="block">Utilities for autoconfiguring javabeans based on command line arguments.</div>
</th>
</tr>
<tr id="i248" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ClusterAltAligs.html" title="class in org.biojava.nbio.structure.align">ClusterAltAligs</a></td>
<th class="colLast" scope="row">
<div class="block">A class that clusters alternative alignments according to their
 similarity.</div>
</th>
</tr>
<tr id="i249" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/pdp/ClusterDomains.html" title="class in org.biojava.nbio.structure.domain.pdp">ClusterDomains</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i250" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/compound/CodonCompound.html" title="class in org.biojava.nbio.core.sequence.compound">CodonCompound</a></td>
<th class="colLast" scope="row">
<div class="block">Define a codon</div>
</th>
</tr>
<tr id="i251" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/CollectionTools.html" title="class in org.biojava.nbio.structure.align.util">CollectionTools</a></td>
<th class="colLast" scope="row">
<div class="block">Utilities for working with collections.</div>
</th>
</tr>
<tr id="i252" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/jmolScript/ColorConverter.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">ColorConverter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i253" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/color/ColorInterpolator.html" title="interface in org.biojava.nbio.structure.gui.util.color">ColorInterpolator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i254" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/color/ColorUtils.html" title="class in org.biojava.nbio.structure.gui.util.color">ColorUtils</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i255" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/CommandPrompt.html" title="class in org.biojava.nbio.aaproperties">CommandPrompt</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i256" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/CommandPrompt.PropertyName.html" title="enum in org.biojava.nbio.aaproperties">CommandPrompt.PropertyName</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i257" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/data/CompactCharSequence.html" title="class in org.biojava.nbio.survival.data">CompactCharSequence</a></td>
<th class="colLast" scope="row">
<div class="block">http://www.javamex.com/tutorials/memory/ascii_charsequence.shtml</div>
</th>
</tr>
<tr id="i258" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/phylo/Comparison.html" title="class in org.biojava.nbio.phylo">Comparison</a></td>
<th class="colLast" scope="row">
<div class="block">This class provides static methods for the calculation of the percentage of
 identity between two aligned sequences.</div>
</th>
</tr>
<tr id="i259" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/ComplementCompound.html" title="interface in org.biojava.nbio.core.sequence.template">ComplementCompound</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i260" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ComplementSequenceView</a>&lt;<a href="org/biojava/nbio/core/sequence/views/ComplementSequenceView.html" title="type parameter in ComplementSequenceView">C</a> extends <a href="org/biojava/nbio/core/sequence/template/ComplementCompound.html" title="interface in org.biojava.nbio.core.sequence.template">ComplementCompound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">For a given sequence this class will create a view over the top of it
 and for every request the code will return the complement of the underlying
 base e.g. base A will become base T</div>
</th>
</tr>
<tr id="i261" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/Component.html" title="class in org.biojava.nbio.protmod">Component</a></td>
<th class="colLast" scope="row">
<div class="block">contains information about a certain Component.</div>
</th>
</tr>
<tr id="i262" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/io/ComponentXMLConverter.html" title="class in org.biojava.nbio.protmod.io">ComponentXMLConverter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i263" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i264" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i265" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="org/biojava/nbio/core/sequence/template/CompoundSet.html" title="type parameter in CompoundSet">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i266" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundTranslator</a>&lt;<a href="org/biojava/nbio/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>,&#8203;<a href="org/biojava/nbio/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i267" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/ConcurrencyTools.html" title="class in org.biojava.nbio.core.util">ConcurrencyTools</a></td>
<th class="colLast" scope="row">
<div class="block">Static utility to easily share a thread pool for concurrent/parallel/lazy execution.</div>
</th>
</tr>
<tr id="i268" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/ConfigPDBInstallPanel.html" title="class in org.biojava.nbio.structure.align.gui">ConfigPDBInstallPanel</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i269" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/ConfigStrucAligParams.html" title="interface in org.biojava.nbio.structure.align.ce">ConfigStrucAligParams</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i270" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/ConfigurationException.html" title="class in org.biojava.nbio.structure.align.util">ConfigurationException</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i271" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/webstart/ConfigXMLHandler.html" title="class in org.biojava.nbio.structure.align.webstart">ConfigXMLHandler</a></td>
<th class="colLast" scope="row">
<div class="block">XML content handler for serialisation of RegistryConfiguration class</div>
</th>
</tr>
<tr id="i272" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/Constraints.html" title="class in org.biojava.nbio.aaproperties">Constraints</a></td>
<th class="colLast" scope="row">
<div class="block">This class is used to support the implementation of properties stated in IPeptideProperties.</div>
</th>
</tr>
<tr id="i273" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/Contact.html" title="class in org.biojava.nbio.structure.contact">Contact</a></td>
<th class="colLast" scope="row">
<div class="block">A simple class to store contacts in the form of pairs of indices and a distance associated to them.</div>
</th>
</tr>
<tr id="i274" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/color/ContinuousColorMapper.html" title="interface in org.biojava.nbio.structure.gui.util.color">ContinuousColorMapper</a></td>
<th class="colLast" scope="row">
<div class="block">Provides a mapping between real numbers and Colors.</div>
</th>
</tr>
<tr id="i275" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/color/ContinuousColorMapperTransform.html" title="class in org.biojava.nbio.structure.gui.util.color">ContinuousColorMapperTransform</a></td>
<th class="colLast" scope="row">
<div class="block">Maps colors by performing a transform of the input data and then passing
 the transformed value to a ContinuousColorMapper for rendering.</div>
</th>
</tr>
<tr id="i276" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2Charge.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2Charge</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i277" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2Hydrophobicity.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2Hydrophobicity</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i278" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2NormalizedVanDerWaalsVolume.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2NormalizedVanDerWaalsVolume</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i279" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2Polarity.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2Polarity</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i280" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2Polarizability.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2Polarizability</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i281" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2SecondaryStructure.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2SecondaryStructure</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i282" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2SolventAccessibility.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2SolventAccessibility</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i283" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convertor.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convertor</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i284" class="altColor">
<td class="colFirst"><a href="demo/CookBook.html" title="class in demo">CookBook</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i285" class="rowColor">
<td class="colFirst"><a href="demo/CookbookMSA.html" title="class in demo">CookbookMSA</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i286" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/CoordManager.html" title="class in org.biojava.nbio.structure.gui.util">CoordManager</a></td>
<th class="colLast" scope="row">
<div class="block">a class that manages the conversion of sequence coordinate system to
 JPanel drawing coordinates</div>
</th>
</tr>
<tr id="i287" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/util/CoreSuperimposer.html" title="class in org.biojava.nbio.structure.align.multiple.util">CoreSuperimposer</a></td>
<th class="colLast" scope="row">
<div class="block">Superimposes the core aligned residues of every structure in a
 <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> onto a reference structure.</div>
</th>
</tr>
<tr id="i288" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/CoxCC.html" title="class in org.biojava.nbio.survival.cox">CoxCC</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i289" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/CoxCoefficient.html" title="class in org.biojava.nbio.survival.cox">CoxCoefficient</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i290" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/comparators/CoxComparatorInterface.html" title="interface in org.biojava.nbio.survival.cox.comparators">CoxComparatorInterface</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i291" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/CoxHelper.html" title="class in org.biojava.nbio.survival.cox">CoxHelper</a></td>
<th class="colLast" scope="row">
<div class="block">The CoxHelper class is provided to start with a tab delimited file in a similar process in R and return the results as a CoxInfo class.</div>
</th>
</tr>
<tr id="i292" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/CoxInfo.html" title="class in org.biojava.nbio.survival.cox">CoxInfo</a></td>
<th class="colLast" scope="row">
<div class="block">Holds the results of a cox analysis where calling dump(), toString() will give an output similar to R</div>
</th>
</tr>
<tr id="i293" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/CoxMart.html" title="class in org.biojava.nbio.survival.cox">CoxMart</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i294" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/CoxMethod.html" title="enum in org.biojava.nbio.survival.cox">CoxMethod</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i295" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/CoxR.html" title="class in org.biojava.nbio.survival.cox">CoxR</a></td>
<th class="colLast" scope="row">
<div class="block">This is a port of the R survival code used for doing Cox Regression.</div>
</th>
</tr>
<tr id="i296" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/CoxScore.html" title="class in org.biojava.nbio.survival.cox">CoxScore</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i297" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/CoxVariables.html" title="class in org.biojava.nbio.survival.cox">CoxVariables</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i298" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/comparators/CoxVariablesOverallModelFitComparator.html" title="class in org.biojava.nbio.survival.cox.comparators">CoxVariablesOverallModelFitComparator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i299" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/comparators/CoxVariablesVariableComparator.html" title="class in org.biojava.nbio.survival.cox.comparators">CoxVariablesVariableComparator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i300" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/CRC64Checksum.html" title="class in org.biojava.nbio.core.util">CRC64Checksum</a></td>
<th class="colLast" scope="row">
<div class="block">Utility class that calculates a CRC64 checksum on a stream of bytes.</div>
</th>
</tr>
<tr id="i301" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/xtal/CrystalBuilder.html" title="class in org.biojava.nbio.structure.xtal">CrystalBuilder</a></td>
<th class="colLast" scope="row">
<div class="block">A class containing methods to find interfaces in a given crystallographic Structure by
 reconstructing the crystal lattice through application of symmetry operators</div>
</th>
</tr>
<tr id="i302" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/xtal/CrystalCell.html" title="class in org.biojava.nbio.structure.xtal">CrystalCell</a></td>
<th class="colLast" scope="row">
<div class="block">A crystal cell's parameters.</div>
</th>
</tr>
<tr id="i303" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/xtal/CrystalTransform.html" title="class in org.biojava.nbio.structure.xtal">CrystalTransform</a></td>
<th class="colLast" scope="row">
<div class="block">Representation of a transformation in a crystal:
 - a transformation id (each of the transformations in a space group, 0 to m)
 - a crystal translation
 The transformation matrix in crystal basis is stored, representing the basic
 transformation together with the crystal translation.</div>
</th>
</tr>
<tr id="i304" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/pdp/Cut.html" title="class in org.biojava.nbio.structure.domain.pdp">Cut</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i305" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/pdp/CutDomain.html" title="class in org.biojava.nbio.structure.domain.pdp">CutDomain</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i306" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/pdp/CutSites.html" title="class in org.biojava.nbio.structure.domain.pdp">CutSites</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i307" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/pdp/CutValues.html" title="class in org.biojava.nbio.structure.domain.pdp">CutValues</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i308" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/cytoband/Cytoband.html" title="class in org.biojava.nbio.genome.parsers.cytoband">Cytoband</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i309" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/cytoband/CytobandParser.html" title="class in org.biojava.nbio.genome.parsers.cytoband">CytobandParser</a></td>
<th class="colLast" scope="row">
<div class="block">Parses the cytoband (karyotype) file from UCSC.</div>
</th>
</tr>
<tr id="i310" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/DatabasePDBRevRecord.html" title="class in org.biojava.nbio.structure">DatabasePDBRevRecord</a></td>
<th class="colLast" scope="row">
<div class="block">Represents revision records for use by <a href="org/biojava/nbio/structure/PDBHeader.html" title="class in org.biojava.nbio.structure"><code>PDBHeader</code></a>.</div>
</th>
</tr>
<tr id="i311" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava.nbio.core.sequence.features">DatabaseReferenceInterface</a></td>
<th class="colLast" scope="row">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list of cross reference id(s)</div>
</th>
</tr>
<tr id="i312" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/phosphosite/Dataset.html" title="class in org.biojava.nbio.phosphosite">Dataset</a></td>
<th class="colLast" scope="row">
<div class="block">Phosphosite is available under the PhosphoSitePlus® is licensed under Creative Commons Attribution-NonCommercial-ShareAlike 3.0 Unported License and is freely available for non-commercial purposes from

 http://www.phosphosite.org/staticDownloads.do

 Please acknowledge PhosphoSitePlus®, www.phosphosite.org" at appropriate locations.</div>
</th>
</tr>
<tr id="i313" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/DataSource.html" title="enum in org.biojava.nbio.core.sequence">DataSource</a></td>
<th class="colLast" scope="row">
<div class="block">
 GenBank                           gi|gi-number|gb|accession|locus
 ENA Data Library                  gi|gi-number|emb|accession|locus
 DDBJ, DNA Database of Japan       gi|gi-number|dbj|accession|locus
 NBRF PIR                          pir||entry
 Protein Research Foundation       prf||name
 SWISS-PROT UNIPROT                sp|accession|name
 Brookhaven Protein Data Bank (1)  pdb|entry|chain
 Brookhaven Protein Data Bank (2)  entry:chain|PDBID|CHAIN|SEQUENCE
 Patents                           pat|country|number
 GenInfo Backbone Id               bbs|number
 General database identifier       gnl|database|identifier
 NCBI Reference Sequence           ref|accession|locus
 Local Sequence identifier         lcl|identifier
 </pr></div>
</th>
</tr>
<tr id="i314" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/DBRef.html" title="class in org.biojava.nbio.structure">DBRef</a></td>
<th class="colLast" scope="row">
<div class="block">A class to represent database cross references.</div>
</th>
</tr>
<tr id="i315" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava.nbio.core.sequence.features">DBReferenceInfo</a></td>
<th class="colLast" scope="row">
<div class="block">If you have a uniprot ID then it is possible to get a collection
 of other id(s) that the protein is known by.</div>
</th>
</tr>
<tr id="i316" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/matrices/DefaultAAIndexProvider.html" title="class in org.biojava.nbio.core.alignment.matrices">DefaultAAIndexProvider</a></td>
<th class="colLast" scope="row">
<div class="block">The default provider for AAINDEX loads substitution matrices from the AAINDEX file in the resources directory</div>
</th>
</tr>
<tr id="i317" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/autosuggest/DefaultAutoSuggestProvider.html" title="class in org.biojava.nbio.structure.align.gui.autosuggest">DefaultAutoSuggestProvider</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i318" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/color/DefaultMatrixMapper.html" title="class in org.biojava.nbio.structure.gui.util.color">DefaultMatrixMapper</a></td>
<th class="colLast" scope="row">
<div class="block">Color Mapper which mimics the default coloring of JMatrixPanel pixels.</div>
</th>
</tr>
<tr id="i319" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/DefaultOps.html" title="class in org.biojava.nbio.ontology">DefaultOps</a></td>
<th class="colLast" scope="row">
<div class="block">Default implementation of OntologyOps.</div>
</th>
</tr>
<tr id="i320" class="altColor">
<td class="colFirst"><a href="demo/DemoAlignmentFromFasta.html" title="class in demo">DemoAlignmentFromFasta</a></td>
<th class="colLast" scope="row">
<div class="block">Demo of how to use the <a href="org/biojava/nbio/structure/io/FastaStructureParser.html" title="class in org.biojava.nbio.structure.io"><code>FastaStructureParser</code></a> class to read protein
 structures from a FASTA file.</div>
</th>
</tr>
<tr id="i321" class="rowColor">
<td class="colFirst"><a href="demo/DemoAlignProteins.html" title="class in demo">DemoAlignProteins</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i322" class="altColor">
<td class="colFirst"><a href="demo/DemoAsa.html" title="class in demo">DemoAsa</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i323" class="rowColor">
<td class="colFirst"><a href="demo/DemoAtomCache.html" title="class in demo">DemoAtomCache</a></td>
<th class="colLast" scope="row">
<div class="block">Example of how to load PDB files using the AtomCache class.</div>
</th>
</tr>
<tr id="i324" class="altColor">
<td class="colFirst"><a href="demo/DemoBerkeleyScop.html" title="class in demo">DemoBerkeleyScop</a></td>
<th class="colLast" scope="row">
<div class="block">A demo for how to use the Berkeley version of SCOP instead of the default UK-SCOP</div>
</th>
</tr>
<tr id="i325" class="rowColor">
<td class="colFirst"><a href="demo/DemoBioAssemblies.html" title="class in demo">DemoBioAssemblies</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i326" class="altColor">
<td class="colFirst"><a href="demo/DemoCATH.html" title="class in demo">DemoCATH</a></td>
<th class="colLast" scope="row">
<div class="block">An example for how to access CATH data.</div>
</th>
</tr>
<tr id="i327" class="rowColor">
<td class="colFirst"><a href="demo/DemoCE.html" title="class in demo">DemoCE</a></td>
<th class="colLast" scope="row">
<div class="block">Example of how to run a structure alignment using the CE algorithm.</div>
</th>
</tr>
<tr id="i328" class="altColor">
<td class="colFirst"><a href="demo/DemoCeSymm.html" title="class in demo">DemoCeSymm</a></td>
<th class="colLast" scope="row">
<div class="block">Quick demo of how to call CE-Symm programmatically.</div>
</th>
</tr>
<tr id="i329" class="rowColor">
<td class="colFirst"><a href="demo/DemoChangeChemCompProvider.html" title="class in demo">DemoChangeChemCompProvider</a></td>
<th class="colLast" scope="row">
<div class="block">This demo shows how to use an alternative ChemCompProvider.</div>
</th>
</tr>
<tr id="i330" class="altColor">
<td class="colFirst"><a href="demo/DemoCommandLineStartup.html" title="class in demo">DemoCommandLineStartup</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i331" class="rowColor">
<td class="colFirst"><a href="demo/DemoContacts.html" title="class in demo">DemoContacts</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i332" class="altColor">
<td class="colFirst"><a href="demo/DemoCrystalInterfaces.html" title="class in demo">DemoCrystalInterfaces</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i333" class="rowColor">
<td class="colFirst"><a href="demo/DemoDistanceTree.html" title="class in demo">DemoDistanceTree</a></td>
<th class="colLast" scope="row">
<div class="block">This demo contains the CookBook example to create a phylogenetic tree from a
 multiple sequence alignment (MSA).</div>
</th>
</tr>
<tr id="i334" class="altColor">
<td class="colFirst"><a href="demo/DemoDomainsplit.html" title="class in demo">DemoDomainsplit</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i335" class="rowColor">
<td class="colFirst"><a href="demo/DemoFATCAT.html" title="class in demo">DemoFATCAT</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i336" class="altColor">
<td class="colFirst"><a href="demo/DemoLoadSecStruc.html" title="class in demo">DemoLoadSecStruc</a></td>
<th class="colLast" scope="row">
<div class="block">Demonstration of how to load a Structure with the SS information, either from
 the PDB file annotation (Author's assignment) or from the DSSP file in the
 PDB servers (DSSP assignment).</div>
</th>
</tr>
<tr id="i337" class="rowColor">
<td class="colFirst"><a href="demo/DemoLoadStructure.html" title="class in demo">DemoLoadStructure</a></td>
<th class="colLast" scope="row">
<div class="block">Example for how to load protein structures (from PDB files).</div>
</th>
</tr>
<tr id="i338" class="altColor">
<td class="colFirst"><a href="demo/DemoLoadSubstMax.html" title="class in demo">DemoLoadSubstMax</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i339" class="rowColor">
<td class="colFirst"><a href="demo/DemoMMCIFReader.html" title="class in demo">DemoMMCIFReader</a></td>
<th class="colLast" scope="row">
<div class="block">An example of how to read MMcif files</div>
</th>
</tr>
<tr id="i340" class="altColor">
<td class="colFirst"><a href="demo/DemoMmcifToPdbConverter.html" title="class in demo">DemoMmcifToPdbConverter</a></td>
<th class="colLast" scope="row">
<div class="block">An example of how to convert mmCIF file to PDB file</div>
</th>
</tr>
<tr id="i341" class="rowColor">
<td class="colFirst"><a href="demo/DemoMmtfReader.html" title="class in demo">DemoMmtfReader</a></td>
<th class="colLast" scope="row">
<div class="block">Class to show how to read a Biojava structure using MMTF</div>
</th>
</tr>
<tr id="i342" class="altColor">
<td class="colFirst"><a href="demo/DemoMmtfWriter.html" title="class in demo">DemoMmtfWriter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i343" class="rowColor">
<td class="colFirst"><a href="demo/DemoMultipleMC.html" title="class in demo">DemoMultipleMC</a></td>
<th class="colLast" scope="row">
<div class="block">Demo for running the MultipleMC Algorithm on a protein family.</div>
</th>
</tr>
<tr id="i344" class="altColor">
<td class="colFirst"><a href="demo/DemoNWALignment.html" title="class in demo">DemoNWALignment</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i345" class="rowColor">
<td class="colFirst"><a href="demo/DemoQsAlign.html" title="class in demo">DemoQsAlign</a></td>
<th class="colLast" scope="row">
<div class="block">Demo on how to use programatically <a href="org/biojava/nbio/structure/align/quaternary/QsAlign.html" title="class in org.biojava.nbio.structure.align.quaternary"><code>QsAlign</code></a> for the alignment of
 quaternary structures.</div>
</th>
</tr>
<tr id="i346" class="altColor">
<td class="colFirst"><a href="demo/DemoQuatSymmetryJmol.html" title="class in demo">DemoQuatSymmetryJmol</a></td>
<th class="colLast" scope="row">
<div class="block">This demo shows how to display the <a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html" title="class in org.biojava.nbio.structure.symmetry.core"><code>QuatSymmetryResults</code></a> of a
 structure.</div>
</th>
</tr>
<tr id="i347" class="rowColor">
<td class="colFirst"><a href="demo/DemoRotationAxis.html" title="class in demo">DemoRotationAxis</a></td>
<th class="colLast" scope="row">
<div class="block">A demo for how to use <a href="org/biojava/nbio/structure/align/util/RotationAxis.html" title="class in org.biojava.nbio.structure.align.util"><code>RotationAxis</code></a> to display the rotation for an
 alignment.</div>
</th>
</tr>
<tr id="i348" class="altColor">
<td class="colFirst"><a href="demo/DemoSCOP.html" title="class in demo">DemoSCOP</a></td>
<th class="colLast" scope="row">
<div class="block">A class demonstrating the use of the SCOP parsing tools</div>
</th>
</tr>
<tr id="i349" class="rowColor">
<td class="colFirst"><a href="demo/DemoSecStrucCalc.html" title="class in demo">DemoSecStrucCalc</a></td>
<th class="colLast" scope="row">
<div class="block">Demonstration on how to use the Secondary Structure Prediction (DSSP)
 implementation in BioJava and obtain different SS representations and
 outputs.</div>
</th>
</tr>
<tr id="i350" class="altColor">
<td class="colFirst"><a href="demo/DemoShowBiolAssembly.html" title="class in demo">DemoShowBiolAssembly</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i351" class="rowColor">
<td class="colFirst"><a href="demo/DemoShowCATHDomain.html" title="class in demo">DemoShowCATHDomain</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i352" class="altColor">
<td class="colFirst"><a href="demo/DemoShowLargeAssembly.html" title="class in demo">DemoShowLargeAssembly</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i353" class="rowColor">
<td class="colFirst"><a href="demo/DemoSixFrameTranslation.html" title="class in demo">DemoSixFrameTranslation</a></td>
<th class="colLast" scope="row">
<div class="block">Created by andreas on 8/10/15.</div>
</th>
</tr>
<tr id="i354" class="altColor">
<td class="colFirst"><a href="demo/DemoStructureFromFasta.html" title="class in demo">DemoStructureFromFasta</a></td>
<th class="colLast" scope="row">
<div class="block">Demo of how to use the <a href="org/biojava/nbio/structure/io/FastaStructureParser.html" title="class in org.biojava.nbio.structure.io"><code>FastaStructureParser</code></a> class to read protein
 structures from a FASTA file.</div>
</th>
</tr>
<tr id="i355" class="rowColor">
<td class="colFirst"><a href="demo/DemoSW3DAligner.html" title="class in demo">DemoSW3DAligner</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i356" class="altColor">
<td class="colFirst"><a href="demo/DemoSymmetry.html" title="class in demo">DemoSymmetry</a></td>
<th class="colLast" scope="row">
<div class="block">A demo on how to use the quaternary symmetry detection algorithms.</div>
</th>
</tr>
<tr id="i357" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/metadata/DiscreteQuantizerInterface.html" title="interface in org.biojava.nbio.survival.kaplanmeier.metadata">DiscreteQuantizerInterface</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i358" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/DisplayAFP.html" title="class in org.biojava.nbio.structure.align.gui">DisplayAFP</a></td>
<th class="colLast" scope="row">
<div class="block">A utility class for visualistion of structure alignments</div>
</th>
</tr>
<tr id="i359" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/geometry/DistanceBox.html" title="class in org.biojava.nbio.structure.symmetry.geometry">DistanceBox</a>&lt;<a href="org/biojava/nbio/structure/symmetry/geometry/DistanceBox.html" title="type parameter in DistanceBox">T</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i360" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/phylo/DistanceMatrixCalculator.html" title="class in org.biojava.nbio.phylo">DistanceMatrixCalculator</a></td>
<th class="colLast" scope="row">
<div class="block">The DistanceMatrixCalculator methods generate a <code>DistanceMatrix</code> from a
 <a href="org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence"><code>MultipleSequenceAlignment</code></a> or other indirect distance infomation (RMSD).</div>
</th>
</tr>
<tr id="i361" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/phylo/DistanceTreeEvaluator.html" title="class in org.biojava.nbio.phylo">DistanceTreeEvaluator</a></td>
<th class="colLast" scope="row">
<div class="block">Check the accuracy of a Distance Tree by least squares error (LSE) of the
 Tree branch lengths and the original Distance Matrix.</div>
</th>
</tr>
<tr id="i362" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/compound/DNACompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">DNACompoundSet</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i363" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></td>
<th class="colLast" scope="row">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</th>
</tr>
<tr id="i364" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava.nbio.core.sequence">DNASequence.DNAType</a></td>
<th class="colLast" scope="row">
<div class="block">The type of DNA sequence</div>
</th>
</tr>
<tr id="i365" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">DNASequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">A helper class that allows different ways to read a string and create a DNA sequence.</div>
</th>
</tr>
<tr id="i366" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava.nbio.core.sequence.transcription">DNAToRNATranslator</a></td>
<th class="colLast" scope="row">
<div class="block">Performs the first stage of transcription by going from DNA to RNA.</div>
</th>
</tr>
<tr id="i367" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i368" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/DomainProvider.html" title="interface in org.biojava.nbio.structure.domain">DomainProvider</a></td>
<th class="colLast" scope="row">
<div class="block">Decomposes a structure from the PDB into representative domains</div>
</th>
</tr>
<tr id="i369" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/DotPlotPanel.html" title="class in org.biojava.nbio.structure.align.gui">DotPlotPanel</a></td>
<th class="colLast" scope="row">
<div class="block">Displays the dot plot trace for an alignment.</div>
</th>
</tr>
<tr id="i370" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/DownloadChemCompProvider.html" title="class in org.biojava.nbio.structure.chem">DownloadChemCompProvider</a></td>
<th class="colLast" scope="row">
<div class="block">This provider of chemical components can download and cache chemical component definition files from the RCSB PDB web
 site.</div>
</th>
</tr>
<tr id="i371" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/secstruc/DSSPParser.html" title="class in org.biojava.nbio.structure.secstruc">DSSPParser</a></td>
<th class="colLast" scope="row">
<div class="block">Class to parse a DSSP file (output of the DSSP program),
 that contains the secondary structure assignment of a structure.</div>
</th>
</tr>
<tr id="i372" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/ecod/EcodDatabase.html" title="interface in org.biojava.nbio.structure.ecod">EcodDatabase</a></td>
<th class="colLast" scope="row">
<div class="block">General API for interacting with ECOD.</div>
</th>
</tr>
<tr id="i373" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/ecod/EcodDomain.html" title="class in org.biojava.nbio.structure.ecod">EcodDomain</a></td>
<th class="colLast" scope="row">
<div class="block">An EcodDomain contains all the information of the ECOD database: id,
 classification groups (from higher to lower in the tree: X,H,T,F), PDB code,
 chain, residue ranges and status (manual or automatic classification).</div>
</th>
</tr>
<tr id="i374" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/ecod/EcodFactory.html" title="class in org.biojava.nbio.structure.ecod">EcodFactory</a></td>
<th class="colLast" scope="row">
<div class="block">Controls global <a href="org/biojava/nbio/structure/ecod/EcodDatabase.html" title="interface in org.biojava.nbio.structure.ecod"><code>EcodDatabases</code></a> being used.</div>
</th>
</tr>
<tr id="i375" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/ecod/EcodInstallation.html" title="class in org.biojava.nbio.structure.ecod">EcodInstallation</a></td>
<th class="colLast" scope="row">
<div class="block">Provides access to the Evolutionary Classification of Protein Domains (ECOD).</div>
</th>
</tr>
<tr id="i376" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/ecod/EcodInstallation.EcodParser.html" title="class in org.biojava.nbio.structure.ecod">EcodInstallation.EcodParser</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i377" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/edits/Edit.html" title="interface in org.biojava.nbio.core.sequence.edits">Edit</a>&lt;<a href="org/biojava/nbio/core/sequence/edits/Edit.html" title="type parameter in Edit">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Interface for carrying out edit operations on a Sequence.</div>
</th>
</tr>
<tr id="i378" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava.nbio.core.sequence.edits">Edit.AbstractEdit</a>&lt;<a href="org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Abstract class which defines all edit operations as a call to discover
 what 5' and 3' ends of an editing Sequence should be joined together
 with a target Sequence.</div>
</th>
</tr>
<tr id="i379" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/edits/Edit.Delete.html" title="class in org.biojava.nbio.core.sequence.edits">Edit.Delete</a>&lt;<a href="org/biojava/nbio/core/sequence/edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implementation which allows for the deletion of bases from a Sequence</div>
</th>
</tr>
<tr id="i380" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="class in org.biojava.nbio.core.sequence.edits">Edit.Insert</a>&lt;<a href="org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Edit implementation which allows us to insert a base at any position
 in a Sequence.</div>
</th>
</tr>
<tr id="i381" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="class in org.biojava.nbio.core.sequence.edits">Edit.Substitute</a>&lt;<a href="org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Allows for the substitution of bases into an existing Sequence.</div>
</th>
</tr>
<tr id="i382" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/jama/EigenvalueDecomposition.html" title="class in org.biojava.nbio.structure.jama">EigenvalueDecomposition</a></td>
<th class="colLast" scope="row">
<div class="block">Eigenvalues and eigenvectors of a real matrix.</div>
</th>
</tr>
<tr id="i383" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/xml/Element.html" title="class in org.biojava.nbio.aaproperties.xml">Element</a></td>
<th class="colLast" scope="row">
<div class="block">One way to model the elements</div>
</th>
</tr>
<tr id="i384" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/Element.html" title="enum in org.biojava.nbio.structure">Element</a></td>
<th class="colLast" scope="row">
<div class="block">Element is an enumeration of the elements of the periodic table.</div>
</th>
</tr>
<tr id="i385" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/xml/ElementTable.html" title="class in org.biojava.nbio.aaproperties.xml">ElementTable</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i386" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/ElementType.html" title="enum in org.biojava.nbio.structure">ElementType</a></td>
<th class="colLast" scope="row">
<div class="block">ElementType is an enumeration of the types of elements found in the periodic table.</div>
</th>
</tr>
<tr id="i387" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/embl/EmblId.html" title="class in org.biojava.nbio.core.sequence.io.embl">EmblId</a></td>
<th class="colLast" scope="row">
<div class="block">This class contains the processed data of embl file
 Primary accession number
 Sequence version number
 Topology: 'circular' or 'linear'
 Molecule type
 Data class
 Taxonomic division
 Sequence length</div>
</th>
</tr>
<tr id="i388" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/embl/EmblReader.html" title="class in org.biojava.nbio.core.sequence.io.embl">EmblReader</a></td>
<th class="colLast" scope="row">
<div class="block">This class should process the data of embl file</div>
</th>
</tr>
<tr id="i389" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/embl/EmblRecord.html" title="class in org.biojava.nbio.core.sequence.io.embl">EmblRecord</a></td>
<th class="colLast" scope="row">
<div class="block">this class contains the parsed data of embl file</div>
</th>
</tr>
<tr id="i390" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/embl/EmblReference.html" title="class in org.biojava.nbio.core.sequence.io.embl">EmblReference</a></td>
<th class="colLast" scope="row">
<div class="block">This class contains the processed data of embl file that
 contains the referenceNumber, referenceComment, referencePosition
 referenceCrossReference, referenceGroup, referenceAuthor
 referenceTitle, referenceLocation</div>
</th>
</tr>
<tr id="i391" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/EntityFinder.html" title="class in org.biojava.nbio.structure.io">EntityFinder</a></td>
<th class="colLast" scope="row">
<div class="block">Heuristical finding of Entities (called Compounds in legacy PDB format)
 in a given Structure.</div>
</th>
</tr>
<tr id="i392" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/EntityInfo.html" title="class in org.biojava.nbio.structure">EntityInfo</a></td>
<th class="colLast" scope="row">
<div class="block">An object to contain the info from the PDB header for a Molecule.</div>
</th>
</tr>
<tr id="i393" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/EntityType.html" title="enum in org.biojava.nbio.structure">EntityType</a></td>
<th class="colLast" scope="row">
<div class="block">The type of entity (polymer, non-polymer, water, macrolide)
 as defined in the mmCIF dictionary: <a href="http://mmcif.wwpdb.org/dictionaries/mmcif_pdbx_v40.dic/Items/_entity.type.html"></a></div>
</th>
</tr>
<tr id="i394" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/Equals.html" title="class in org.biojava.nbio.core.util">Equals</a></td>
<th class="colLast" scope="row">
<div class="block">A set of helper methods which return true if the two parameters are
 equal to each other.</div>
</th>
</tr>
<tr id="i395" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/ExonComparator.html" title="class in org.biojava.nbio.core.sequence">ExonComparator</a></td>
<th class="colLast" scope="row">
<div class="block">Sort Exon where it is a little confusing if exons should always be ordered left to right
 where a negative stranded gene should go the other direction.</div>
</th>
</tr>
<tr id="i396" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a></td>
<th class="colLast" scope="row">
<div class="block">A gene contains a collection of Exon sequences</div>
</th>
</tr>
<tr id="i397" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/ExperimentalTechnique.html" title="enum in org.biojava.nbio.structure">ExperimentalTechnique</a></td>
<th class="colLast" scope="row">
<div class="block">An enum to represent the experimental technique of a PDB structure</div>
</th>
</tr>
<tr id="i398" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/figure/ExpressionFigure.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">ExpressionFigure</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i399" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/FastaAFPChainConverter.html" title="class in org.biojava.nbio.structure.io">FastaAFPChainConverter</a></td>
<th class="colLast" scope="row">
<div class="block">A collection of static utilities to convert between <a href="org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model"><code>AFPChains</code></a> and <code>FastaSequences</code>.</div>
</th>
</tr>
<tr id="i400" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava.nbio.core.sequence.io">FastaGeneWriter</a></td>
<th class="colLast" scope="row">
<div class="block">A Gene sequence has a Positive or Negative Strand where we want to write out to a stream the 5 to 3 prime version.</div>
</th>
</tr>
<tr id="i401" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/template/FastaHeaderFormatInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">FastaHeaderFormatInterface</a>&lt;<a href="org/biojava/nbio/core/sequence/io/template/FastaHeaderFormatInterface.html" title="type parameter in FastaHeaderFormatInterface">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/template/FastaHeaderFormatInterface.html" title="type parameter in FastaHeaderFormatInterface">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i402" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/FastaReader.html" title="class in org.biojava.nbio.core.sequence.io">FastaReader</a>&lt;<a href="org/biojava/nbio/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be the
 primary class used to read Fasta files</div>
</th>
</tr>
<tr id="i403" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava.nbio.core.sequence.io">FastaReaderHelper</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i404" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a></td>
<th class="colLast" scope="row">
<div class="block">A FASTA formatted sequence.</div>
</th>
</tr>
<tr id="i405" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/FastaSequenceParser.html" title="class in org.biojava.nbio.core.sequence.io">FastaSequenceParser</a></td>
<th class="colLast" scope="row">
<div class="block">Used to parse a stream of a fasta file to get the sequence</div>
</th>
</tr>
<tr id="i406" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/FastaStructureParser.html" title="class in org.biojava.nbio.structure.io">FastaStructureParser</a></td>
<th class="colLast" scope="row">
<div class="block">Reads a protein sequence from a fasta file and attempts to match it to a
 3D structure.</div>
</th>
</tr>
<tr id="i407" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/FastaWriter.html" title="class in org.biojava.nbio.core.sequence.io">FastaWriter</a>&lt;<a href="org/biojava/nbio/core/sequence/io/FastaWriter.html" title="type parameter in FastaWriter">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/FastaWriter.html" title="type parameter in FastaWriter">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">The FastaWriter writes a collection of sequences to an outputStream.</div>
</th>
</tr>
<tr id="i408" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/FastaWriterHelper.html" title="class in org.biojava.nbio.core.sequence.io">FastaWriterHelper</a></td>
<th class="colLast" scope="row">
<div class="block">The class that should be used to write out fasta file of a sequence collection</div>
</th>
</tr>
<tr id="i409" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a></td>
<th class="colLast" scope="row">
<div class="block">FASTQ formatted sequence.</div>
</th>
</tr>
<tr id="i410" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/FastqBuilder.html" title="class in org.biojava.nbio.genome.io.fastq">FastqBuilder</a></td>
<th class="colLast" scope="row">
<div class="block">Fluent builder API for creating FASTQ formatted sequences.</div>
</th>
</tr>
<tr id="i411" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/FastqReader.html" title="interface in org.biojava.nbio.genome.io.fastq">FastqReader</a></td>
<th class="colLast" scope="row">
<div class="block">Reader for FASTQ formatted sequences.</div>
</th>
</tr>
<tr id="i412" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/FastqTools.html" title="class in org.biojava.nbio.genome.io.fastq">FastqTools</a></td>
<th class="colLast" scope="row">
<div class="block">Utility methods for FASTQ formatted sequences.</div>
</th>
</tr>
<tr id="i413" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html" title="enum in org.biojava.nbio.genome.io.fastq">FastqVariant</a></td>
<th class="colLast" scope="row">
<div class="block">FASTQ sequence format variant.</div>
</th>
</tr>
<tr id="i414" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/FastqWriter.html" title="interface in org.biojava.nbio.genome.io.fastq">FastqWriter</a></td>
<th class="colLast" scope="row">
<div class="block">Writer for FASTQ formatted sequences.</div>
</th>
</tr>
<tr id="i415" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/FatCat.html" title="class in org.biojava.nbio.structure.align.fatcat">FatCat</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i416" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/calc/FatCatAligner.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatAligner</a></td>
<th class="colLast" scope="row">
<div class="block">A class that does calculations on an AFPChain</div>
</th>
</tr>
<tr id="i417" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/FatCatFlexible.html" title="class in org.biojava.nbio.structure.align.fatcat">FatCatFlexible</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i418" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i419" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/FatCatRigid.html" title="class in org.biojava.nbio.structure.align.fatcat">FatCatRigid</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i420" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/FatCatUserArgumentProcessor.html" title="class in org.biojava.nbio.structure.align.fatcat">FatCatUserArgumentProcessor</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i421" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/calc/FCAlignHelper.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FCAlignHelper</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i422" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/gff/Feature.html" title="class in org.biojava.nbio.genome.parsers.gff">Feature</a></td>
<th class="colLast" scope="row">
<div class="block">A Feature corresponds to a single row in a GFF file.</div>
</th>
</tr>
<tr id="i423" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/features/FeatureDbReferenceInfo.html" title="class in org.biojava.nbio.core.sequence.features">FeatureDbReferenceInfo</a>&lt;<a href="org/biojava/nbio/core/sequence/features/FeatureDbReferenceInfo.html" title="type parameter in FeatureDbReferenceInfo">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/features/FeatureDbReferenceInfo.html" title="type parameter in FeatureDbReferenceInfo">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/features/FeatureDbReferenceInfo.html" title="type parameter in FeatureDbReferenceInfo">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">It is <a href="org/biojava/nbio/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava.nbio.core.sequence.features"><code>DBReferenceInfo</code></a> which implements <a href="org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features"><code>FeatureInterface</code></a>.</div>
</th>
</tr>
<tr id="i424" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/gff/FeatureHelper.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureHelper</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i425" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/gff/FeatureI.html" title="interface in org.biojava.nbio.genome.parsers.gff">FeatureI</a></td>
<th class="colLast" scope="row">
<div class="block">A feature on a sequence (for example, an exon or a gene), defined by a location
 and a set of attributes encoded as key/value pairs.</div>
</th>
</tr>
<tr id="i426" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="type parameter in FeatureInterface">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="type parameter in FeatureInterface">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="type parameter in FeatureInterface">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Interface class to handle describing arbitrary features.</div>
</th>
</tr>
<tr id="i427" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></td>
<th class="colLast" scope="row">
<div class="block">A list of FeatureI objects implemented using a Java ArrayList; corresponds to a GFF file.</div>
</th>
</tr>
<tr id="i428" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/features/FeatureRetriever.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureRetriever</a>&lt;<a href="org/biojava/nbio/core/sequence/features/FeatureRetriever.html" title="type parameter in FeatureRetriever">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list features</div>
</th>
</tr>
<tr id="i429" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeaturesKeyWordInterface</a></td>
<th class="colLast" scope="row">
<div class="block">Models the keywords that are annotated for a protein sequence at Uniprot.</div>
</th>
</tr>
<tr id="i430" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/FileConvert.html" title="class in org.biojava.nbio.structure.io">FileConvert</a></td>
<th class="colLast" scope="row">
<div class="block">Methods to convert a structure object into different file formats.</div>
</th>
</tr>
<tr id="i431" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/FileDownloadUtils.html" title="class in org.biojava.nbio.core.util">FileDownloadUtils</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i432" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/FileParsingParameters.html" title="class in org.biojava.nbio.structure.io">FileParsingParameters</a></td>
<th class="colLast" scope="row">
<div class="block">A class that configures parameters that can be sent to the PDB file parsers

 
  <a href="org/biojava/nbio/structure/io/FileParsingParameters.html#setParseCAOnly(boolean)"><code>FileParsingParameters.setParseCAOnly(boolean)</code></a> - parse only the Atom records for C-alpha atoms
  <a href="org/biojava/nbio/structure/io/FileParsingParameters.html#setParseSecStruc(boolean)"><code>FileParsingParameters.setParseSecStruc(boolean)</code></a> - a flag if the secondary structure information from the PDB file (author's assignment) should be parsed.</div>
</th>
</tr>
<tr id="i433" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/FileProxyDNASequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">FileProxyDNASequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the stream
 the sequence and the offset index are passed to create a Protein sequence that will be loaded in lazily.</div>
</th>
</tr>
<tr id="i434" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/FileProxyProteinSequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">FileProxyProteinSequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the stream
 the sequence and the offset index are passed to create a Protein sequence that will be loaded in lazily.</div>
</th>
</tr>
<tr id="i435" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/FileProxyRNASequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">FileProxyRNASequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the stream
 the sequence and the offset index are passed to create a Protein sequence that will be loaded in lazily.</div>
</th>
</tr>
<tr id="i436" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/FlatFileCache.html" title="class in org.biojava.nbio.core.util">FlatFileCache</a></td>
<th class="colLast" scope="row">
<div class="block">Provides a cache for storing multiple small files in memory.</div>
</th>
</tr>
<tr id="i437" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/phylo/ForesterWrapper.html" title="class in org.biojava.nbio.phylo">ForesterWrapper</a></td>
<th class="colLast" scope="row">
<div class="block">This class contains wrapper methods for communication between BioJava and
 forester (e.g, Data Structure conversion).</div>
</th>
</tr>
<tr id="i438" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">FourBitSequenceReader</a>&lt;<a href="org/biojava/nbio/core/sequence/storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Four bit encoding of the bit formats.</div>
</th>
</tr>
<tr id="i439" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="class in org.biojava.nbio.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker</a>&lt;<a href="org/biojava/nbio/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A four bit per compound implementation of the bit array worker code.</div>
</th>
</tr>
<tr id="i440" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/FractionalIdentityInProfileScorer.html" title="class in org.biojava.nbio.alignment">FractionalIdentityInProfileScorer</a>&lt;<a href="org/biojava/nbio/alignment/FractionalIdentityInProfileScorer.html" title="type parameter in FractionalIdentityInProfileScorer">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/FractionalIdentityInProfileScorer.html" title="type parameter in FractionalIdentityInProfileScorer">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/FractionalIdentityInProfileScorer.html" title="type parameter in FractionalIdentityInProfileScorer">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair picked from an alignment
 <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</th>
</tr>
<tr id="i441" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/FractionalIdentityScorer.html" title="class in org.biojava.nbio.alignment">FractionalIdentityScorer</a>&lt;<a href="org/biojava/nbio/alignment/FractionalIdentityScorer.html" title="type parameter in FractionalIdentityScorer">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/FractionalIdentityScorer.html" title="type parameter in FractionalIdentityScorer">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/FractionalIdentityScorer.html" title="type parameter in FractionalIdentityScorer">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair.</div>
</th>
</tr>
<tr id="i442" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/FractionalSimilarityInProfileScorer.html" title="class in org.biojava.nbio.alignment">FractionalSimilarityInProfileScorer</a>&lt;<a href="org/biojava/nbio/alignment/FractionalSimilarityInProfileScorer.html" title="type parameter in FractionalSimilarityInProfileScorer">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/FractionalSimilarityInProfileScorer.html" title="type parameter in FractionalSimilarityInProfileScorer">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/FractionalSimilarityInProfileScorer.html" title="type parameter in FractionalSimilarityInProfileScorer">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair picked from an alignment
 <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</th>
</tr>
<tr id="i443" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/FractionalSimilarityScorer.html" title="class in org.biojava.nbio.alignment">FractionalSimilarityScorer</a>&lt;<a href="org/biojava/nbio/alignment/FractionalSimilarityScorer.html" title="type parameter in FractionalSimilarityScorer">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/FractionalSimilarityScorer.html" title="type parameter in FractionalSimilarityScorer">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/FractionalSimilarityScorer.html" title="type parameter in FractionalSimilarityScorer">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair.</div>
</th>
</tr>
<tr id="i444" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html" title="class in org.biojava.nbio.structure.align.pairwise">FragmentJoiner</a></td>
<th class="colLast" scope="row">
<div class="block">Joins the initial Fragments together to larger Fragments</div>
</th>
</tr>
<tr id="i445" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/pairwise/FragmentPair.html" title="class in org.biojava.nbio.structure.align.pairwise">FragmentPair</a></td>
<th class="colLast" scope="row">
<div class="block">a pair of fragments of two protein structures</div>
</th>
</tr>
<tr id="i446" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a></td>
<th class="colLast" scope="row">
<div class="block">Indicates a way of translating a sequence.</div>
</th>
</tr>
<tr id="i447" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/FuzzyPoint.html" title="class in org.biojava.nbio.core.sequence.location">FuzzyPoint</a></td>
<th class="colLast" scope="row">
<div class="block">Implementation for resolving fuzzy locations.</div>
</th>
</tr>
<tr id="i448" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/helper/GapArray.html" title="class in org.biojava.nbio.structure.align.helper">GapArray</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i449" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a></td>
<th class="colLast" scope="row">
<div class="block">Defines a data structure for the gap penalties used during a sequence alignment routine.</div>
</th>
</tr>
<tr id="i450" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/GapPenalty.Type.html" title="enum in org.biojava.nbio.alignment.template">GapPenalty.Type</a></td>
<th class="colLast" scope="row">
<div class="block">Defines the possible types of gap penalties.</div>
</th>
</tr>
<tr id="i451" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/gff/GCStats.html" title="class in org.biojava.nbio.genome.parsers.gff">GCStats</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i452" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/template/GenbankHeaderFormatInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">GenbankHeaderFormatInterface</a>&lt;<a href="org/biojava/nbio/core/sequence/io/template/GenbankHeaderFormatInterface.html" title="type parameter in GenbankHeaderFormatInterface">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/template/GenbankHeaderFormatInterface.html" title="type parameter in GenbankHeaderFormatInterface">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i453" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/loader/GenbankProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">GenbankProxySequenceReader</a>&lt;<a href="org/biojava/nbio/core/sequence/loader/GenbankProxySequenceReader.html" title="type parameter in GenbankProxySequenceReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i454" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/GenbankReader.html" title="class in org.biojava.nbio.core.sequence.io">GenbankReader</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenbankReader.html" title="type parameter in GenbankReader">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenbankReader.html" title="type parameter in GenbankReader">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/GenbankReader.html" title="type parameter in GenbankReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Use <a href="org/biojava/nbio/core/sequence/io/GenbankReaderHelper.html" title="class in org.biojava.nbio.core.sequence.io"><code>GenbankReaderHelper</code></a> as an example of how to use this class where <a href="org/biojava/nbio/core/sequence/io/GenbankReaderHelper.html" title="class in org.biojava.nbio.core.sequence.io"><code>GenbankReaderHelper</code></a> should be the
 primary class used to read Genbank files</div>
</th>
</tr>
<tr id="i455" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/GenbankReaderHelper.html" title="class in org.biojava.nbio.core.sequence.io">GenbankReaderHelper</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i456" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/reference/GenbankReference.html" title="class in org.biojava.nbio.core.sequence.reference">GenbankReference</a></td>
<th class="colLast" scope="row">
<div class="block">For Genbank format file only.</div>
</th>
</tr>
<tr id="i457" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/GenbankSequenceParser.html" title="class in org.biojava.nbio.core.sequence.io">GenbankSequenceParser</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenbankSequenceParser.html" title="type parameter in GenbankSequenceParser">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenbankSequenceParser.html" title="type parameter in GenbankSequenceParser">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/GenbankSequenceParser.html" title="type parameter in GenbankSequenceParser">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i458" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/GenbankWriter.html" title="class in org.biojava.nbio.core.sequence.io">GenbankWriter</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenbankWriter.html" title="type parameter in GenbankWriter">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/GenbankWriter.html" title="type parameter in GenbankWriter">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i459" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/GenbankWriterHelper.html" title="class in org.biojava.nbio.core.sequence.io">GenbankWriterHelper</a></td>
<th class="colLast" scope="row">
<div class="block">The class that should be used to write out genbank file of a sequence
 collection</div>
</th>
</tr>
<tr id="i460" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i461" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/genename/GeneChromosomePositionParser.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePositionParser</a></td>
<th class="colLast" scope="row">
<div class="block">A parser that parses a file from the UCSC genome browser that contains mapping of gene name to chromosome positions</div>
</th>
</tr>
<tr id="i462" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/GeneFeatureHelper.html" title="class in org.biojava.nbio.genome">GeneFeatureHelper</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i463" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/gff/GeneIDGFF2Reader.html" title="class in org.biojava.nbio.genome.parsers.gff">GeneIDGFF2Reader</a></td>
<th class="colLast" scope="row">
<div class="block">http://www.bioperl.org/wiki/GTF
 Read and write FeatureLists as GFF/GTF formatted files.</div>
</th>
</tr>
<tr id="i464" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava.nbio.genome.parsers.geneid">GeneIDXMLReader</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i465" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/gff/GeneMarkGTFReader.html" title="class in org.biojava.nbio.genome.parsers.gff">GeneMarkGTFReader</a></td>
<th class="colLast" scope="row">
<div class="block">http://www.bioperl.org/wiki/GTF
 Read and write FeatureLists as GFF/GTF formatted files.</div>
</th>
</tr>
<tr id="i466" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/genename/GeneName.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneName</a></td>
<th class="colLast" scope="row">
<div class="block">A simple bean that contains gene name information as available from www.genenames.org</div>
</th>
</tr>
<tr id="i467" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/genename/GeneNamesParser.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneNamesParser</a></td>
<th class="colLast" scope="row">
<div class="block">Parses a file from the www.genenames.org website that contains a mapping of human gene names to other databases</div>
</th>
</tr>
<tr id="i468" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/GenericFastaHeaderFormat.html" title="class in org.biojava.nbio.core.sequence.io">GenericFastaHeaderFormat</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenericFastaHeaderFormat.html" title="type parameter in GenericFastaHeaderFormat">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/GenericFastaHeaderFormat.html" title="type parameter in GenericFastaHeaderFormat">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">We store the original header if the sequence is parsed from a fasta file and will use that exact
 sequence if we write out the sequences to a fasta file.</div>
</th>
</tr>
<tr id="i469" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">GenericFastaHeaderParser</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/GenericFastaHeaderParser.html" title="type parameter in GenericFastaHeaderParser">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">The default fasta header parser where some headers are well defined based on the source
 database which allows us to set the source of the protein sequence and the identifier
 that can be used in future implementations to load features from external sources

 If the user has a custom header with local data then they can create their own implementation
 of a FastaHeaderParserInterface</div>
</th>
</tr>
<tr id="i470" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/GenericGenbankHeaderFormat.html" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderFormat</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenericGenbankHeaderFormat.html" title="type parameter in GenericGenbankHeaderFormat">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenericGenbankHeaderFormat.html" title="type parameter in GenericGenbankHeaderFormat">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/GenericGenbankHeaderFormat.html" title="type parameter in GenericGenbankHeaderFormat">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i471" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/GenericGenbankHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderParser</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenericGenbankHeaderParser.html" title="type parameter in GenericGenbankHeaderParser">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenericGenbankHeaderParser.html" title="type parameter in GenericGenbankHeaderParser">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/GenericGenbankHeaderParser.html" title="type parameter in GenericGenbankHeaderParser">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i472" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/GenericInsdcHeaderFormat.html" title="class in org.biojava.nbio.core.sequence.io">GenericInsdcHeaderFormat</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenericInsdcHeaderFormat.html" title="type parameter in GenericInsdcHeaderFormat">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/io/GenericInsdcHeaderFormat.html" title="type parameter in GenericInsdcHeaderFormat">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/GenericInsdcHeaderFormat.html" title="type parameter in GenericInsdcHeaderFormat">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i473" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i474" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/pdp/GetDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">GetDistanceMatrix</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i475" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/homology/GFF3FromUniprotBlastHits.html" title="class in org.biojava.nbio.genome.homology">GFF3FromUniprotBlastHits</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i476" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/gff/GFF3Reader.html" title="class in org.biojava.nbio.genome.parsers.gff">GFF3Reader</a></td>
<th class="colLast" scope="row">
<div class="block">http://www.bioperl.org/wiki/GTF
 Read and write FeatureLists as GFF/GTF formatted files.</div>
</th>
</tr>
<tr id="i477" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/gff/GFF3Writer.html" title="class in org.biojava.nbio.genome.parsers.gff">GFF3Writer</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i478" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/io/GOParser.html" title="class in org.biojava.nbio.ontology.io">GOParser</a></td>
<th class="colLast" scope="row">
<div class="block">Simple parser for the Gene Ontology (GO) flatfile format.</div>
</th>
</tr>
<tr id="i479" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/pairwise/Gotoh.html" title="class in org.biojava.nbio.structure.align.pairwise">Gotoh</a></td>
<th class="colLast" scope="row">
<div class="block">a class to perform Gotoh algorithm</div>
</th>
</tr>
<tr id="i480" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/color/GradientMapper.html" title="class in org.biojava.nbio.structure.gui.util.color">GradientMapper</a></td>
<th class="colLast" scope="row">
<div class="block">Maps a set of real values onto a gradient.</div>
</th>
</tr>
<tr id="i481" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/color/GradientPanel.html" title="class in org.biojava.nbio.structure.gui.util.color">GradientPanel</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i482" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/GraphComponentOrderDetector.html" title="class in org.biojava.nbio.structure.symmetry.internal">GraphComponentOrderDetector</a></td>
<th class="colLast" scope="row">
<div class="block">The GraphOrderDetector transforms the self-alignment into a Graph and
 extracts its maximally connected Components.</div>
</th>
</tr>
<tr id="i483" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/GraphComponentRefiner.html" title="class in org.biojava.nbio.structure.symmetry.internal">GraphComponentRefiner</a></td>
<th class="colLast" scope="row">
<div class="block">The GraphRefiner transforms the self-alignment into a Graph and extracts its
 maximally connected Components.</div>
</th>
</tr>
<tr id="i484" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/Grid.html" title="class in org.biojava.nbio.structure.contact">Grid</a></td>
<th class="colLast" scope="row">
<div class="block">A grid to be used for calculating atom contacts through a spatial hashing algorithm.</div>
</th>
</tr>
<tr id="i485" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/GridCell.html" title="class in org.biojava.nbio.structure.contact">GridCell</a></td>
<th class="colLast" scope="row">
<div class="block">A grid cell to be used in contact calculation via spatial hashing algorithm.</div>
</th>
</tr>
<tr id="i486" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></td>
<th class="colLast" scope="row">
<div class="block">This is the data structure for a single Group of atoms.</div>
</th>
</tr>
<tr id="i487" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/asa/GroupAsa.html" title="class in org.biojava.nbio.structure.asa">GroupAsa</a></td>
<th class="colLast" scope="row">
<div class="block">A class to store the results of ASA calculations, it can
 hold ASA values per atom present in <a href="org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure"><code>Group</code></a></div>
</th>
</tr>
<tr id="i488" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/GroupContact.html" title="class in org.biojava.nbio.structure.contact">GroupContact</a></td>
<th class="colLast" scope="row">
<div class="block">A pair of residues that are in contact</div>
</th>
</tr>
<tr id="i489" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/GroupContactSet.html" title="class in org.biojava.nbio.structure.contact">GroupContactSet</a></td>
<th class="colLast" scope="row">
<div class="block">A set of residue-residue contacts.</div>
</th>
</tr>
<tr id="i490" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/GroupIterator.html" title="class in org.biojava.nbio.structure">GroupIterator</a></td>
<th class="colLast" scope="row">
<div class="block">An iterator over all groups of a structure.</div>
</th>
</tr>
<tr id="i491" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/figure/GroupResults.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">GroupResults</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i492" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/GroupToSDF.html" title="class in org.biojava.nbio.structure.io">GroupToSDF</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i493" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a></td>
<th class="colLast" scope="row">
<div class="block">This contains basic categories for Group types.</div>
</th>
</tr>
<tr id="i494" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/routines/GuanUberbacher.html" title="class in org.biojava.nbio.alignment.routines">GuanUberbacher</a>&lt;<a href="org/biojava/nbio/alignment/routines/GuanUberbacher.html" title="type parameter in GuanUberbacher">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/routines/GuanUberbacher.html" title="type parameter in GuanUberbacher">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/routines/GuanUberbacher.html" title="type parameter in GuanUberbacher">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Guan and Uberbacher defined an algorithm for pairwise global sequence alignments (from the first until the last
 <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</th>
</tr>
<tr id="i495" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/GuideTree.html" title="class in org.biojava.nbio.alignment">GuideTree</a>&lt;<a href="org/biojava/nbio/alignment/GuideTree.html" title="type parameter in GuideTree">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/GuideTree.html" title="type parameter in GuideTree">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/GuideTree.html" title="type parameter in GuideTree">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements a data structure for a guide tree used during progressive multiple sequence alignment.</div>
</th>
</tr>
<tr id="i496" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/GuideTreeNode.html" title="interface in org.biojava.nbio.alignment.template">GuideTreeNode</a>&lt;<a href="org/biojava/nbio/alignment/template/GuideTreeNode.html" title="type parameter in GuideTreeNode">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/GuideTreeNode.html" title="type parameter in GuideTreeNode">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/GuideTreeNode.html" title="type parameter in GuideTreeNode">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a data structure for the node in a guide tree used during progressive multiple sequence alignment.</div>
</th>
</tr>
<tr id="i497" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/GuiWrapper.html" title="class in org.biojava.nbio.structure.align.ce">GuiWrapper</a></td>
<th class="colLast" scope="row">
<div class="block">A class to wrap some of the strucutre.gui classes using Reflection</div>
</th>
</tr>
<tr id="i498" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/Hashcoder.html" title="class in org.biojava.nbio.core.util">Hashcoder</a></td>
<th class="colLast" scope="row">
<div class="block">Contains helper methods for generating a HashCode without having to resort to
 the commons lang hashcode builders.</div>
</th>
</tr>
<tr id="i499" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/secstruc/HBond.html" title="class in org.biojava.nbio.structure.secstruc">HBond</a></td>
<th class="colLast" scope="row">
<div class="block">Container that represents a hidrogen bond.</div>
</th>
</tr>
<tr id="i500" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/data/HeaderInfo.html" title="class in org.biojava.nbio.survival.data">HeaderInfo</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i501" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/HelicalRepeatUnit.html" title="class in org.biojava.nbio.structure.symmetry.core">HelicalRepeatUnit</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i502" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/Helix.html" title="class in org.biojava.nbio.structure.symmetry.core">Helix</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i503" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/axis/HelixAxisAligner.html" title="class in org.biojava.nbio.structure.symmetry.axis">HelixAxisAligner</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i504" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/HelixExtender.html" title="class in org.biojava.nbio.structure.symmetry.core">HelixExtender</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i505" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/HelixLayers.html" title="class in org.biojava.nbio.structure.symmetry.core">HelixLayers</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i506" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/HelixSolver.html" title="class in org.biojava.nbio.structure.symmetry.core">HelixSolver</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i507" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/proteincomparisontool/HelloWorld.html" title="class in org.biojava.nbio.proteincomparisontool">HelloWorld</a></td>
<th class="colLast" scope="row">
<div class="block">A dummy class that does nothing.</div>
</th>
</tr>
<tr id="i508" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/HelpDialog.html" title="class in org.biojava.nbio.structure.align.gui">HelpDialog</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i509" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/HetatomImpl.html" title="class in org.biojava.nbio.structure">HetatomImpl</a></td>
<th class="colLast" scope="row">
<div class="block">Generic Implementation of a Group interface.</div>
</th>
</tr>
<tr id="i510" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/HetatomImpl.PerformanceBehavior.html" title="enum in org.biojava.nbio.structure">HetatomImpl.PerformanceBehavior</a></td>
<th class="colLast" scope="row">
<div class="block">Behaviors for how to balance memory vs. performance.</div>
</th>
</tr>
<tr id="i511" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/HierarchicalClusterer.html" title="interface in org.biojava.nbio.alignment.template">HierarchicalClusterer</a>&lt;<a href="org/biojava/nbio/alignment/template/HierarchicalClusterer.html" title="type parameter in HierarchicalClusterer">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/HierarchicalClusterer.html" title="type parameter in HierarchicalClusterer">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/HierarchicalClusterer.html" title="type parameter in HierarchicalClusterer">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a clustering algorithm that converts a distance matrix into a tree.</div>
</th>
</tr>
<tr id="i512" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/search/io/Hit.html" title="class in org.biojava.nbio.core.search.io">Hit</a></td>
<th class="colLast" scope="row">
<div class="block">This class models a search Hit.</div>
</th>
</tr>
<tr id="i513" class="rowColor">
<td class="colFirst"><a href="demo/HmmerDemo.html" title="class in demo">HmmerDemo</a></td>
<th class="colLast" scope="row">
<div class="block">The cookbook recipe for how to request Pfam annotations for a protein sequence using the Hmmer3 service</div>
</th>
</tr>
<tr id="i514" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/hmmer/HmmerDomain.html" title="class in org.biojava.nbio.ws.hmmer">HmmerDomain</a></td>
<th class="colLast" scope="row">
<div class="block">Provides the details of a domain hit</div>
</th>
</tr>
<tr id="i515" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/hmmer/HmmerResult.html" title="class in org.biojava.nbio.ws.hmmer">HmmerResult</a></td>
<th class="colLast" scope="row">
<div class="block">The results of a Hmmer search for a single sequence</div>
</th>
</tr>
<tr id="i516" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/hmmer/HmmerScan.html" title="interface in org.biojava.nbio.ws.hmmer">HmmerScan</a></td>
<th class="colLast" scope="row">
<div class="block">Interface for performing Hmmscans on sequences.</div>
</th>
</tr>
<tr id="i517" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/search/io/Hsp.html" title="class in org.biojava.nbio.core.search.io">Hsp</a>&lt;<a href="org/biojava/nbio/core/search/io/Hsp.html" title="type parameter in Hsp">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/search/io/Hsp.html" title="type parameter in Hsp">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/search/io/Hsp.html" title="type parameter in Hsp">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">This class models a search Hsp.</div>
</th>
</tr>
<tr id="i518" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/color/HSVColorSpace.html" title="class in org.biojava.nbio.structure.gui.util.color">HSVColorSpace</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i519" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/geometry/IcosahedralSampler.html" title="class in org.biojava.nbio.structure.symmetry.geometry">IcosahedralSampler</a></td>
<th class="colLast" scope="row">
<div class="block">Represents an even coverage of quaternion space by 60 points.</div>
</th>
</tr>
<tr id="i520" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/geometry/Icosahedron.html" title="class in org.biojava.nbio.structure.symmetry.geometry">Icosahedron</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i521" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/helper/IdxComparator.html" title="class in org.biojava.nbio.structure.align.helper">IdxComparator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i522" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/IlluminaFastqReader.html" title="class in org.biojava.nbio.genome.io.fastq">IlluminaFastqReader</a></td>
<th class="colLast" scope="row">
<div class="block">Reader for <a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html#FASTQ_ILLUMINA"><code>FastqVariant.FASTQ_ILLUMINA</code></a> formatted sequences.</div>
</th>
</tr>
<tr id="i523" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/IlluminaFastqWriter.html" title="class in org.biojava.nbio.genome.io.fastq">IlluminaFastqWriter</a></td>
<th class="colLast" scope="row">
<div class="block">Writer for <a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html#FASTQ_ILLUMINA"><code>FastqVariant.FASTQ_ILLUMINA</code></a> formatted sequences.</div>
</th>
</tr>
<tr id="i524" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/helper/IndexPair.html" title="class in org.biojava.nbio.structure.align.helper">IndexPair</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i525" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/InputStreamProvider.html" title="class in org.biojava.nbio.core.util">InputStreamProvider</a></td>
<th class="colLast" scope="row">
<div class="block">A class that provides an InputStream from a File.</div>
</th>
</tr>
<tr id="i526" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/InsdcLocations.html" title="class in org.biojava.nbio.core.sequence.location">InsdcLocations</a></td>
<th class="colLast" scope="row">
<div class="block">A collection of locations which are used whenever we work with INSDC; some
 of which could be deprecated (from INSDC's point of view) yet appear
 in records.</div>
</th>
</tr>
<tr id="i527" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/InsdcLocations.BondLocation.html" title="class in org.biojava.nbio.core.sequence.location">InsdcLocations.BondLocation</a></td>
<th class="colLast" scope="row">
<div class="block">Used to represent bond locations equivalent to bond(7,8) or bond(7).</div>
</th>
</tr>
<tr id="i528" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava.nbio.core.sequence.location">InsdcLocations.GroupLocation</a></td>
<th class="colLast" scope="row">
<div class="block">Deprecated in INSDC yet still appears; equivalent to the order()
 directive except no 5' to 3' ordering is defined.</div>
</th>
</tr>
<tr id="i529" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/InsdcLocations.OneOfLocation.html" title="class in org.biojava.nbio.core.sequence.location">InsdcLocations.OneOfLocation</a></td>
<th class="colLast" scope="row">
<div class="block">Deprecated in INSDC; refers to a set of locations of which one
 location could be valid e.g. one-of(location, location, location).</div>
</th>
</tr>
<tr id="i530" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava.nbio.core.sequence.location">InsdcLocations.OrderLocation</a></td>
<th class="colLast" scope="row">
<div class="block">Used to describe a 5' to 3' ordering but no firm assurance it is correct</div>
</th>
</tr>
<tr id="i531" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/InsdcParser.html" title="class in org.biojava.nbio.core.sequence.location">InsdcParser</a></td>
<th class="colLast" scope="row">
<div class="block">Parser for working with INSDC style locations.</div>
</th>
</tr>
<tr id="i532" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/IntegerOntology.html" title="class in org.biojava.nbio.ontology">IntegerOntology</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i533" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/InterfaceFinder.html" title="class in org.biojava.nbio.structure.contact">InterfaceFinder</a></td>
<th class="colLast" scope="row">
<div class="block">A class containing methods to find interfaces in a given structure.</div>
</th>
</tr>
<tr id="i534" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i535" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/InvalidTermException.html" title="class in org.biojava.nbio.ontology">InvalidTermException</a></td>
<th class="colLast" scope="row">
<div class="block">Thrown to indicate that an ontology term is not acceptable or
 appropriate in a given context</div>
</th>
</tr>
<tr id="i536" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/util/IOUtils.html" title="class in org.biojava.nbio.core.sequence.io.util">IOUtils</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i537" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/util/IOUtils.ReaderProcessor.html" title="interface in org.biojava.nbio.core.sequence.io.util">IOUtils.ReaderProcessor</a></td>
<th class="colLast" scope="row">
<div class="block">Closure interface used when working with
 <code>IOUtils#processReader(String)</code>.</div>
</th>
</tr>
<tr id="i538" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/IPeptideProperties.html" title="interface in org.biojava.nbio.aaproperties">IPeptideProperties</a></td>
<th class="colLast" scope="row">
<div class="block">An interface to generate some basic physico-chemical properties of protein sequences.<br/>
 The following properties could be generated:</div>
</th>
</tr>
<tr id="i539" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/IProfeatProperties.html" title="interface in org.biojava.nbio.aaproperties.profeat">IProfeatProperties</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i540" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a></td>
<th class="colLast" scope="row">
<div class="block">Enumeration of the seven different attributes</div>
</th>
</tr>
<tr id="i541" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a></td>
<th class="colLast" scope="row">
<div class="block">Enumeration of the distribution for the first, first 25%, first 50%, first 75% and 100% of the grouping</div>
</th>
</tr>
<tr id="i542" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a></td>
<th class="colLast" scope="row">
<div class="block">Enumeration of the three different groupings for each attributes</div>
</th>
</tr>
<tr id="i543" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/IProfeatProperties.TRANSITION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.TRANSITION</a></td>
<th class="colLast" scope="row">
<div class="block">Enumeration of the transition between groupA and groupB</div>
</th>
</tr>
<tr id="i544" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/xml/Isotope.html" title="class in org.biojava.nbio.aaproperties.xml">Isotope</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i545" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/IUPACParser.html" title="class in org.biojava.nbio.core.sequence.io">IUPACParser</a></td>
<th class="colLast" scope="row">
<div class="block">Available translations

 
 1 - UNIVERSAL
 2 - VERTEBRATE_MITOCHONDRIAL
 3 - YEAST_MITOCHONDRIAL
 4 - MOLD_MITOCHONDRIAL
 5 - INVERTEBRATE_MITOCHONDRIAL
 6 - CILIATE_NUCLEAR
 9 - ECHINODERM_MITOCHONDRIAL
 10 - EUPLOTID_NUCLEAR
 11 - BACTERIAL
 12 - ALTERNATIVE_YEAST_NUCLEAR
 13 - ASCIDIAN_MITOCHONDRIAL
 14 - FLATWORM_MITOCHONDRIAL
 15 - BLEPHARISMA_MACRONUCLEAR
 16 - 2CHLOROPHYCEAN_MITOCHONDRIAL
 21 - TREMATODE_MITOCHONDRIAL
 23 - SCENEDESMUS_MITOCHONDRIAL
 

 Taken from <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c">NCBI</a> with slight modification and put into the classpath resource.</div>
</th>
</tr>
<tr id="i546" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava.nbio.core.sequence.io">IUPACParser.IUPACTable</a></td>
<th class="colLast" scope="row">
<div class="block">Holds the concept of a codon table from the IUPAC format</div>
</th>
</tr>
<tr id="i547" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/autosuggest/JAutoSuggest.html" title="class in org.biojava.nbio.structure.align.gui.autosuggest">JAutoSuggest</a></td>
<th class="colLast" scope="row">
<div class="block">A JTextField that can make suggestions for auto-complete.</div>
</th>
</tr>
<tr id="i548" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/JMatrixPanel.html" title="class in org.biojava.nbio.structure.gui">JMatrixPanel</a></td>
<th class="colLast" scope="row">
<div class="block">a JPanel that can display a difference of distance matrix and paths that have been
 taken for the alignment</div>
</th>
</tr>
<tr id="i549" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/events/JmolAlignedPositionListener.html" title="class in org.biojava.nbio.structure.gui.events">JmolAlignedPositionListener</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i550" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/jmol/JmolPanel.html" title="class in org.biojava.nbio.structure.align.gui.jmol">JmolPanel</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i551" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/jmol/JmolPanel.JmolLoggerAdapter.html" title="class in org.biojava.nbio.structure.align.gui.jmol">JmolPanel.JmolLoggerAdapter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i552" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGenerator.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGenerator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i553" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorC1.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorC1</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i554" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorCn.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorCn</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i555" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorDn.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorDn</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i556" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorH.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorH</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i557" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorI.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorI</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i558" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorO.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorO</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i559" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorPointGroup.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorPointGroup</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i560" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorT.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorT</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i561" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/jmol/JmolTools.html" title="class in org.biojava.nbio.structure.align.gui.jmol">JmolTools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i562" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/JmolViewerImpl.html" title="class in org.biojava.nbio.structure.gui">JmolViewerImpl</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i563" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/webstart/JNLPProxy.html" title="class in org.biojava.nbio.structure.align.webstart">JNLPProxy</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i564" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">JoiningSequenceReader</a>&lt;<a href="org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">This reader actually proxies onto multiple types of sequence in order
 to allow a number of sequence objects to act as if they are one sequence.</div>
</th>
</tr>
<tr id="i565" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a></td>
<th class="colLast" scope="row">
<div class="block">A utility class that defines which set of atoms are considered
 to be on equivalent positions.</div>
</th>
</tr>
<tr id="i566" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/JournalArticle.html" title="class in org.biojava.nbio.structure">JournalArticle</a></td>
<th class="colLast" scope="row">
<div class="block">PDB-specific</div>
</th>
</tr>
<tr id="i567" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/JPrintPanel.html" title="class in org.biojava.nbio.structure.align.gui">JPrintPanel</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i568" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ronn/Jronn.html" title="class in org.biojava.nbio.ronn">Jronn</a></td>
<th class="colLast" scope="row">
<div class="block">This class gives public API to RONN functions.</div>
</th>
</tr>
<tr id="i569" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a></td>
<th class="colLast" scope="row">
<div class="block">Holder for the ranges, contain pointers to starting and ending position
 on the sequence which comprises a disordered region.</div>
</th>
</tr>
<tr id="i570" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/figure/KaplanMeierFigure.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">KaplanMeierFigure</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i571" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/utils/KeyedWeakReference.html" title="class in org.biojava.nbio.ontology.utils">KeyedWeakReference</a></td>
<th class="colLast" scope="row">
<div class="block">Subclass of WeakReference which includes and extra field (the key)
 which can be used to help cleanup once this reference has been
 enqueued.</div>
</th>
</tr>
<tr id="i572" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/figure/KMFigureInfo.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">KMFigureInfo</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i573" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/secstruc/Ladder.html" title="class in org.biojava.nbio.structure.secstruc">Ladder</a></td>
<th class="colLast" scope="row">
<div class="block">A Ladder is a set of one or more consecutive bridges of identical type.</div>
</th>
</tr>
<tr id="i574" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/LightweightProfile.html" title="interface in org.biojava.nbio.core.sequence.template">LightweightProfile</a>&lt;<a href="org/biojava/nbio/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/template/LightweightProfile.html" title="type parameter in LightweightProfile">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a minimal data structure for reading and writing a sequence alignment.</div>
</th>
</tr>
<tr id="i575" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava.nbio.core.sequence.template">LightweightProfile.StringFormat</a></td>
<th class="colLast" scope="row">
<div class="block">List of output formats.</div>
</th>
</tr>
<tr id="i576" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/color/LinearColorInterpolator.html" title="class in org.biojava.nbio.structure.gui.util.color">LinearColorInterpolator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i577" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/color/LinearColorInterpolator.InterpolationDirection.html" title="enum in org.biojava.nbio.structure.gui.util.color">LinearColorInterpolator.InterpolationDirection</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i578" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/LocalPDBDirectory.html" title="class in org.biojava.nbio.structure.io">LocalPDBDirectory</a></td>
<th class="colLast" scope="row">
<div class="block">Superclass for classes which download and interact with the PDB's FTP server,
 specifically <a href="org/biojava/nbio/structure/io/PDBFileReader.html" title="class in org.biojava.nbio.structure.io"><code>PDBFileReader</code></a> and <a href="org/biojava/nbio/structure/io/CifFileReader.html" title="class in org.biojava.nbio.structure.io"><code>CifFileReader</code></a>.</div>
</th>
</tr>
<tr id="i579" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/LocalPDBDirectory.FetchBehavior.html" title="enum in org.biojava.nbio.structure.io">LocalPDBDirectory.FetchBehavior</a></td>
<th class="colLast" scope="row">
<div class="block">Controls when the class should fetch files from the ftp server</div>
</th>
</tr>
<tr id="i580" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/LocalPDBDirectory.ObsoleteBehavior.html" title="enum in org.biojava.nbio.structure.io">LocalPDBDirectory.ObsoleteBehavior</a></td>
<th class="colLast" scope="row">
<div class="block">Behaviors for when an obsolete structure is requested.</div>
</th>
</tr>
<tr id="i581" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/LocalProteinDomainParser.html" title="class in org.biojava.nbio.structure.domain">LocalProteinDomainParser</a></td>
<th class="colLast" scope="row">
<div class="block">Protein Domain Parser is a an algorithm that attempts at assigning domains for 3D protein structures.</div>
</th>
</tr>
<tr id="i582" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/LocalScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">LocalScopDatabase</a></td>
<th class="colLast" scope="row">
<div class="block">Classes which implement ScopDatabase in a way which allows them to serve queries
 without accessing the internet should implement this interface instead.</div>
</th>
</tr>
<tr id="i583" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/template/Location.html" title="interface in org.biojava.nbio.core.sequence.location.template">Location</a></td>
<th class="colLast" scope="row">
<div class="block">Sets of integers used to represent the location of features on sequence.</div>
</th>
</tr>
<tr id="i584" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/gff/Location.html" title="class in org.biojava.nbio.genome.parsers.gff">Location</a></td>
<th class="colLast" scope="row">
<div class="block">A location on a sequence.</div>
</th>
</tr>
<tr id="i585" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/template/Location.Tools.html" title="class in org.biojava.nbio.core.sequence.location.template">Location.Tools</a></td>
<th class="colLast" scope="row">
<div class="block">Helper methods for use with the Location classes.</div>
</th>
</tr>
<tr id="i586" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/LocationHelper.html" title="class in org.biojava.nbio.core.sequence.location">LocationHelper</a></td>
<th class="colLast" scope="row">
<div class="block">Helper methods for use with the Location classes.</div>
</th>
</tr>
<tr id="i587" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/gff/LocIterator.html" title="class in org.biojava.nbio.genome.parsers.gff">LocIterator</a></td>
<th class="colLast" scope="row">
<div class="block">Move a sliding window over a Location.</div>
</th>
</tr>
<tr id="i588" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/color/LogColorMapper.html" title="class in org.biojava.nbio.structure.gui.util.color">LogColorMapper</a></td>
<th class="colLast" scope="row">
<div class="block">Performs a log10 transform on input before passing the values off to another
 colorMapper.</div>
</th>
</tr>
<tr id="i589" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/jama/LUDecomposition.html" title="class in org.biojava.nbio.structure.jama">LUDecomposition</a></td>
<th class="colLast" scope="row">
<div class="block">LU Decomposition.</div>
</th>
</tr>
<tr id="i590" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/qblast/MapToStringTransformer.html" title="class in org.biojava.nbio.ws.alignment.qblast">MapToStringTransformer</a></td>
<th class="colLast" scope="row">
<div class="block">Transforms Map to String.</div>
</th>
</tr>
<tr id="i591" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/jama/Maths.html" title="class in org.biojava.nbio.structure.jama">Maths</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i592" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/geometry/Matrices.html" title="class in org.biojava.nbio.structure.geometry">Matrices</a></td>
<th class="colLast" scope="row">
<div class="block">Matrices contains static methods to operate and transform matrices used in 3D
 geometry (transformation matrices and rotation matrices).</div>
</th>
</tr>
<tr id="i593" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></td>
<th class="colLast" scope="row">
<div class="block">Jama = Java Matrix class.</div>
</th>
</tr>
<tr id="i594" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/matrix/Matrix.html" title="class in org.biojava.nbio.survival.cox.matrix">Matrix</a></td>
<th class="colLast" scope="row">
<div class="block">***********************************************************************
 Compilation: javac Matrix.java Execution: java Matrix

 A bare-bones collection of static methods for manipulating matrices.</div>
</th>
</tr>
<tr id="i595" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/MatrixAligner.html" title="interface in org.biojava.nbio.alignment.template">MatrixAligner</a>&lt;<a href="org/biojava/nbio/alignment/template/MatrixAligner.html" title="type parameter in MatrixAligner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/MatrixAligner.html" title="type parameter in MatrixAligner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/MatrixAligner.html" title="type parameter in MatrixAligner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines an <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> which builds a score matrix during computation.</div>
</th>
</tr>
<tr id="i596" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/MatrixListener.html" title="interface in org.biojava.nbio.structure.align.ce">MatrixListener</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i597" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/comparators/MeanModelComparator.html" title="class in org.biojava.nbio.survival.cox.comparators">MeanModelComparator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i598" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/metadata/MeanQuantizer.html" title="class in org.biojava.nbio.survival.kaplanmeier.metadata">MeanQuantizer</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i599" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/MemoryMonitor.html" title="class in org.biojava.nbio.structure.align.gui">MemoryMonitor</a></td>
<th class="colLast" scope="row">
<div class="block">Tracks Memory allocated & used, displayed in graph form.</div>
</th>
</tr>
<tr id="i600" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/MenuCreator.html" title="class in org.biojava.nbio.structure.align.gui">MenuCreator</a></td>
<th class="colLast" scope="row">
<div class="block">Create the menus for structure alignment GUI windows (JFrames).</div>
</th>
</tr>
<tr id="i601" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/MenuCreator.html" title="class in org.biojava.nbio.structure.gui.util">MenuCreator</a></td>
<th class="colLast" scope="row">
<div class="block">Create the menu for BiojavaJmol</div>
</th>
</tr>
<tr id="i602" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/MenuCreator.DotPlotListener.html" title="class in org.biojava.nbio.structure.align.gui">MenuCreator.DotPlotListener</a></td>
<th class="colLast" scope="row">
<div class="block">Creates a frame to display a DotPlotPanel.</div>
</th>
</tr>
<tr id="i603" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/exceptions/Messages.html" title="class in org.biojava.nbio.core.exceptions">Messages</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i604" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/metadata/MetaDataInfo.html" title="class in org.biojava.nbio.survival.kaplanmeier.metadata">MetaDataInfo</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i605" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/MetalBondConsumer.html" title="interface in org.biojava.nbio.structure.io.cif">MetalBondConsumer</a></td>
<th class="colLast" scope="row">
<div class="block">Consume metal bond data.</div>
</th>
</tr>
<tr id="i606" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/MetalBondConsumerImpl.html" title="class in org.biojava.nbio.structure.io.cif">MetalBondConsumerImpl</a></td>
<th class="colLast" scope="row">
<div class="block">Created by andreas on 6/9/16.</div>
</th>
</tr>
<tr id="i607" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/cif/MetalBondConverter.html" title="class in org.biojava.nbio.structure.io.cif">MetalBondConverter</a></td>
<th class="colLast" scope="row">
<div class="block">Created by andreas on 6/6/16.</div>
</th>
</tr>
<tr id="i608" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/MetalBondDistance.html" title="class in org.biojava.nbio.structure.chem">MetalBondDistance</a></td>
<th class="colLast" scope="row">
<div class="block">A bean that contains cutoffs for correctly detecting metal bonds.</div>
</th>
</tr>
<tr id="i609" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/basepairs/MismatchedBasePairParameters.html" title="class in org.biojava.nbio.structure.basepairs">MismatchedBasePairParameters</a></td>
<th class="colLast" scope="row">
<div class="block">This class allows for finding inter-strand base pairs that are not necessarily canonical Watson-Crick pairs.</div>
</th>
</tr>
<tr id="i610" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/mmtf/MmtfActions.html" title="class in org.biojava.nbio.structure.io.mmtf">MmtfActions</a></td>
<th class="colLast" scope="row">
<div class="block">A class of functions for reading and writing Biojava structures using MMTF</div>
</th>
</tr>
<tr id="i611" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/MMTFFileReader.html" title="class in org.biojava.nbio.structure.io">MMTFFileReader</a></td>
<th class="colLast" scope="row">
<div class="block">A class to read MMTF files and cache them locally.</div>
</th>
</tr>
<tr id="i612" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/mmtf/MmtfStructureReader.html" title="class in org.biojava.nbio.structure.io.mmtf">MmtfStructureReader</a></td>
<th class="colLast" scope="row">
<div class="block">A biojava specific structure inflator for MMTF.</div>
</th>
</tr>
<tr id="i613" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/mmtf/MmtfStructureWriter.html" title="class in org.biojava.nbio.structure.io.mmtf">MmtfStructureWriter</a></td>
<th class="colLast" scope="row">
<div class="block">Class to take Biojava structure data and covert to the DataApi for encoding.</div>
</th>
</tr>
<tr id="i614" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/mmtf/MmtfSummaryDataBean.html" title="class in org.biojava.nbio.structure.io.mmtf">MmtfSummaryDataBean</a></td>
<th class="colLast" scope="row">
<div class="block">Class to store the summary data for a given structure.</div>
</th>
</tr>
<tr id="i615" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/mmtf/MmtfUtils.html" title="class in org.biojava.nbio.structure.io.mmtf">MmtfUtils</a></td>
<th class="colLast" scope="row">
<div class="block">A utils class of functions needed for Biojava to read and write to mmtf.</div>
</th>
</tr>
<tr id="i616" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/Model.html" title="class in org.biojava.nbio.structure">Model</a></td>
<th class="colLast" scope="row">
<div class="block">An internal utility class for StructureImpl to make it easier to manage poly and nonpoly chains.</div>
</th>
</tr>
<tr id="i617" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ronn/ModelLoader.html" title="class in org.biojava.nbio.ronn">ModelLoader</a></td>
<th class="colLast" scope="row">
<div class="block">Class that loads data from the model files into <a href="org/biojava/nbio/ronn/ModelLoader.Model.html" title="class in org.biojava.nbio.ronn"><code>ModelLoader.Model</code></a> objects</div>
</th>
</tr>
<tr id="i618" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ronn/ModelLoader.Model.html" title="class in org.biojava.nbio.ronn">ModelLoader.Model</a></td>
<th class="colLast" scope="row">
<div class="block">Represent a RONN model</div>
</th>
</tr>
<tr id="i619" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ronn/ModelLoader.Threshold.html" title="class in org.biojava.nbio.ronn">ModelLoader.Threshold</a></td>
<th class="colLast" scope="row">
<div class="block">Represents a Threshold</div>
</th>
</tr>
<tr id="i620" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/ModificationCategory.html" title="enum in org.biojava.nbio.protmod">ModificationCategory</a></td>
<th class="colLast" scope="row">
<div class="block">define modification categories.</div>
</th>
</tr>
<tr id="i621" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/ModificationCondition.html" title="interface in org.biojava.nbio.protmod">ModificationCondition</a></td>
<th class="colLast" scope="row">
<div class="block">Conditions of a protein modification, e.g. components and atoms.</div>
</th>
</tr>
<tr id="i622" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/ModificationConditionImpl.html" title="class in org.biojava.nbio.protmod">ModificationConditionImpl</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i623" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/ModificationLinkage.html" title="class in org.biojava.nbio.protmod">ModificationLinkage</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i624" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/ModificationOccurrenceType.html" title="enum in org.biojava.nbio.protmod">ModificationOccurrenceType</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i625" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/xml/ModifiedAminoAcidCompoundSet.html" title="class in org.biojava.nbio.aaproperties.xml">ModifiedAminoAcidCompoundSet</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i626" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/structure/ModifiedCompound.html" title="interface in org.biojava.nbio.protmod.structure">ModifiedCompound</a></td>
<th class="colLast" scope="row">
<div class="block">Root interface for all modifications in structure.</div>
</th>
</tr>
<tr id="i627" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/structure/ModifiedCompoundImpl.html" title="class in org.biojava.nbio.protmod.structure">ModifiedCompoundImpl</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i628" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/io/ModifiedCompoundXMLConverter.html" title="class in org.biojava.nbio.protmod.io">ModifiedCompoundXMLConverter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i629" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/geometry/MomentsOfInertia.html" title="class in org.biojava.nbio.structure.geometry">MomentsOfInertia</a></td>
<th class="colLast" scope="row">
<div class="block">The moment of inertia, otherwise known as the angular mass or rotational
 inertia, of a rigid body determines the torque needed for a desired angular
 acceleration about a rotational axis.</div>
</th>
</tr>
<tr id="i630" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/geometry/MomentsOfInertia.SymmetryClass.html" title="enum in org.biojava.nbio.structure.geometry">MomentsOfInertia.SymmetryClass</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i631" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></td>
<th class="colLast" scope="row">
<div class="block">A MultipleAlignment is a Data Structure to store the core information of a
 multiple structure alignment, as a return type.</div>
</th>
</tr>
<tr id="i632" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/MultipleAlignmentCalc.html" title="class in org.biojava.nbio.structure.align.gui">MultipleAlignmentCalc</a></td>
<th class="colLast" scope="row">
<div class="block">A class that obtains structures via DAS and aligns them.</div>
</th>
</tr>
<tr id="i633" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleAlignmentCoordManager.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">MultipleAlignmentCoordManager</a></td>
<th class="colLast" scope="row">
<div class="block">Generalization of the Coodinate Manager to include an arbitrary number of
 sequences (lines) for MultipleAlignment visualization.</div>
</th>
</tr>
<tr id="i634" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentDisplay.html" title="class in org.biojava.nbio.structure.align.multiple.util">MultipleAlignmentDisplay</a></td>
<th class="colLast" scope="row">
<div class="block">Utility functions to generalize the visualization of MultipleAlignments in
 molecular viewers.</div>
</th>
</tr>
<tr id="i635" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsemble.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignmentEnsemble</a></td>
<th class="colLast" scope="row">
<div class="block">A MultipleAlignmentEnsemble is a collection of <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>s
 that share the same structures (Atoms) and creation properties (algorithm,
 version, creation time, etc.).</div>
</th>
</tr>
<tr id="i636" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsembleImpl.html" title="class in org.biojava.nbio.structure.align.multiple">MultipleAlignmentEnsembleImpl</a></td>
<th class="colLast" scope="row">
<div class="block">A general implementation of a <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsemble.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignmentEnsemble</code></a>.</div>
</th>
</tr>
<tr id="i637" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/MultipleAlignmentGUI.html" title="class in org.biojava.nbio.structure.align.gui">MultipleAlignmentGUI</a></td>
<th class="colLast" scope="row">
<div class="block">A JFrame that allows to trigger a multiple structure alignment,
 either from files in a directory or after manual upload.</div>
</th>
</tr>
<tr id="i638" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/MultipleAlignmentImpl.html" title="class in org.biojava.nbio.structure.align.multiple">MultipleAlignmentImpl</a></td>
<th class="colLast" scope="row">
<div class="block">A general implementation of a <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</th>
</tr>
<tr id="i639" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MultipleAlignmentJmol</a></td>
<th class="colLast" scope="row">
<div class="block">A class that provides a 3D visualization Frame in Jmol for
 <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>s.</div>
</th>
</tr>
<tr id="i640" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/MultipleAlignmentJmolDisplay.html" title="class in org.biojava.nbio.structure.align.gui">MultipleAlignmentJmolDisplay</a></td>
<th class="colLast" scope="row">
<div class="block">Utility Class that provides helper methods for the visualization of
 <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>s.</div>
</th>
</tr>
<tr id="i641" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html" title="class in org.biojava.nbio.structure.align.multiple.util">MultipleAlignmentScorer</a></td>
<th class="colLast" scope="row">
<div class="block">Utility class for calculating common scores of <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>s.</div>
</th>
</tr>
<tr id="i642" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html" title="class in org.biojava.nbio.structure.align.multiple.util">MultipleAlignmentTools</a></td>
<th class="colLast" scope="row">
<div class="block">Utility functions for working with <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</th>
</tr>
<tr id="i643" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html" title="class in org.biojava.nbio.structure.align.multiple.util">MultipleAlignmentWriter</a></td>
<th class="colLast" scope="row">
<div class="block">This class contains functions for the conversion of <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>
 to various String outputs.</div>
</th>
</tr>
<tr id="i644" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/xml/MultipleAlignmentXMLConverter.html" title="class in org.biojava.nbio.structure.align.xml">MultipleAlignmentXMLConverter</a></td>
<th class="colLast" scope="row">
<div class="block">Helper methods to convert all the hierarchy levels of a MultipleAlignment
 into an XML format.</div>
</th>
</tr>
<tr id="i645" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/xml/MultipleAlignmentXMLParser.html" title="class in org.biojava.nbio.structure.align.xml">MultipleAlignmentXMLParser</a></td>
<th class="colLast" scope="row">
<div class="block">Parse an XML file representing a <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsemble.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignmentEnsemble</code></a>, so
 that the original alignment can be recovered.</div>
</th>
</tr>
<tr id="i646" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanel.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">MultipleAligPanel</a></td>
<th class="colLast" scope="row">
<div class="block">A JPanel that can display the sequence alignment of a
 <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> in a nice way and interact with Jmol by
 selecting the aligned atoms of the  sequence selection.</div>
</th>
</tr>
<tr id="i647" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanelMouseMotionListener.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">MultipleAligPanelMouseMotionListener</a></td>
<th class="colLast" scope="row">
<div class="block">Mouse Motion Listener for the <a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanel.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel"><code>MultipleAligPanel</code></a>,
 which provides methods to obtain positions of the mouse
 and connect them to the sequence alignment positions using
 the information in <a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleAlignmentCoordManager.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel"><code>MultipleAlignmentCoordManager</code></a>.</div>
</th>
</tr>
<tr id="i648" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/mc/MultipleMcMain.html" title="class in org.biojava.nbio.structure.align.multiple.mc">MultipleMcMain</a></td>
<th class="colLast" scope="row">
<div class="block">Main class of the Java implementation of the Combinatorial Extension -
 Monte Carlo (CEMC) Algorithm,
 as it was originally described by C.Guda, E.D.Scheeff, P.E.</div>
</th>
</tr>
<tr id="i649" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/mc/MultipleMcOptimizer.html" title="class in org.biojava.nbio.structure.align.multiple.mc">MultipleMcOptimizer</a></td>
<th class="colLast" scope="row">
<div class="block">This class takes a MultipleAlignment seed previously generated and runs a
 Monte Carlo optimization in order to improve the overall score and highlight
 common structural motifs.</div>
</th>
</tr>
<tr id="i650" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/mc/MultipleMcParameters.html" title="class in org.biojava.nbio.structure.align.multiple.mc">MultipleMcParameters</a></td>
<th class="colLast" scope="row">
<div class="block">Contains the parameters to be sent to the MC optimization.</div>
</th>
</tr>
<tr id="i651" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;<a href="org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="type parameter in MultipleSequenceAlignment">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements a minimal data structure for reading and writing a sequence alignment.</div>
</th>
</tr>
<tr id="i652" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleStatusDisplay.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">MultipleStatusDisplay</a></td>
<th class="colLast" scope="row">
<div class="block">This class provides information of the selected positions in the
 <a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanel.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel"><code>MultipleAligPanel</code></a>.</div>
</th>
</tr>
<tr id="i653" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/MultipleStructureAligner.html" title="interface in org.biojava.nbio.structure.align">MultipleStructureAligner</a></td>
<th class="colLast" scope="row">
<div class="block">Interface for the Multiple Structure Alignment Algorithms.</div>
</th>
</tr>
<tr id="i654" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/util/MultipleSuperimposer.html" title="interface in org.biojava.nbio.structure.align.multiple.util">MultipleSuperimposer</a></td>
<th class="colLast" scope="row">
<div class="block">Interface for Multiple Alignment superposition algorithms.</div>
</th>
</tr>
<tr id="i655" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/template/MutableAlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">MutableAlignedSequence</a>&lt;<a href="org/biojava/nbio/core/alignment/template/MutableAlignedSequence.html" title="type parameter in MutableAlignedSequence">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/template/MutableAlignedSequence.html" title="type parameter in MutableAlignedSequence">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/template/MutableAlignedSequence.html" title="type parameter in MutableAlignedSequence">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a mutable (editable) data structure for an <a href="org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template"><code>AlignedSequence</code></a>.</div>
</th>
</tr>
<tr id="i656" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/template/MutableProfile.html" title="interface in org.biojava.nbio.core.alignment.template">MutableProfile</a>&lt;<a href="org/biojava/nbio/core/alignment/template/MutableProfile.html" title="type parameter in MutableProfile">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/template/MutableProfile.html" title="type parameter in MutableProfile">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/template/MutableProfile.html" title="type parameter in MutableProfile">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a mutable (editable) data structure for a <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</th>
</tr>
<tr id="i657" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/template/MutableProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">MutableProfilePair</a>&lt;<a href="org/biojava/nbio/core/alignment/template/MutableProfilePair.html" title="type parameter in MutableProfilePair">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/template/MutableProfilePair.html" title="type parameter in MutableProfilePair">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/template/MutableProfilePair.html" title="type parameter in MutableProfilePair">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a mutable (editable) data structure for a <a href="org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template"><code>ProfilePair</code></a>.</div>
</th>
</tr>
<tr id="i658" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/template/MutableSequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">MutableSequencePair</a>&lt;<a href="org/biojava/nbio/core/alignment/template/MutableSequencePair.html" title="type parameter in MutableSequencePair">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/template/MutableSequencePair.html" title="type parameter in MutableSequencePair">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/template/MutableSequencePair.html" title="type parameter in MutableSequencePair">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a mutable (editable) data structure for the results of pairwise sequence alignment.</div>
</th>
</tr>
<tr id="i659" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/Mutator.html" title="class in org.biojava.nbio.structure">Mutator</a></td>
<th class="colLast" scope="row">
<div class="block">A class that can change one amino acid to another.</div>
</th>
</tr>
<tr id="i660" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/MyAlignmentLoadListener.html" title="class in org.biojava.nbio.structure.align.gui">MyAlignmentLoadListener</a></td>
<th class="colLast" scope="row">
<div class="block">Loads an alignment in an XML format and displays its content in a
 new Jmol panel.</div>
</th>
</tr>
<tr id="i661" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/MyDistMaxListener.html" title="class in org.biojava.nbio.structure.align.gui">MyDistMaxListener</a></td>
<th class="colLast" scope="row">
<div class="block">Shows the interatomic Distance Matrices of all the Structures aligned in different Frames.</div>
</th>
</tr>
<tr id="i662" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/MyExportListener.html" title="class in org.biojava.nbio.structure.align.gui">MyExportListener</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i663" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/jmol/MyJmolStatusListener.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MyJmolStatusListener</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i664" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/MyOpenPdbFileListener.html" title="class in org.biojava.nbio.structure.align.gui">MyOpenPdbFileListener</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i665" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/MySaveFileListener.html" title="class in org.biojava.nbio.structure.align.gui">MySaveFileListener</a></td>
<th class="colLast" scope="row">
<div class="block">Save an alignment to a specified File by the user.</div>
</th>
</tr>
<tr id="i666" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/MyTableRowSorter.html" title="class in org.biojava.nbio.structure.align.gui">MyTableRowSorter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i667" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/xml/MyValidationEventHandler.html" title="class in org.biojava.nbio.aaproperties.xml">MyValidationEventHandler</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i668" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/xml/Name2Count.html" title="class in org.biojava.nbio.aaproperties.xml">Name2Count</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i669" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/qblast/NCBIQBlastAlignmentProperties.html" title="class in org.biojava.nbio.ws.alignment.qblast">NCBIQBlastAlignmentProperties</a></td>
<th class="colLast" scope="row">
<div class="block">This class wraps a QBlast search request parameter <code>Map</code> by adding several convenient parameter addition
 methods.</div>
</th>
</tr>
<tr id="i670" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/qblast/NCBIQBlastOutputProperties.html" title="class in org.biojava.nbio.ws.alignment.qblast">NCBIQBlastOutputProperties</a></td>
<th class="colLast" scope="row">
<div class="block">This class wraps a QBlast output parameter <code>Map</code> by adding several convenient parameter addition methods.</div>
</th>
</tr>
<tr id="i671" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/qblast/NCBIQBlastService.html" title="class in org.biojava.nbio.ws.alignment.qblast">NCBIQBlastService</a></td>
<th class="colLast" scope="row">
<div class="block">Provides a simple way of submitting BLAST request to the QBlast service at NCBI.</div>
</th>
</tr>
<tr id="i672" class="altColor">
<td class="colFirst"><a href="demo/NCBIQBlastServiceDemo.html" title="class in demo">NCBIQBlastServiceDemo</a></td>
<th class="colLast" scope="row">
<div class="block">A simple demo showing <a href="org/biojava/nbio/ws/alignment/qblast/NCBIQBlastService.html" title="class in org.biojava.nbio.ws.alignment.qblast"><code>NCBIQBlastService</code></a> usage</div>
</th>
</tr>
<tr id="i673" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/NeedlemanWunsch.html" title="class in org.biojava.nbio.alignment">NeedlemanWunsch</a>&lt;<a href="org/biojava/nbio/alignment/NeedlemanWunsch.html" title="type parameter in NeedlemanWunsch">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/NeedlemanWunsch.html" title="type parameter in NeedlemanWunsch">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/NeedlemanWunsch.html" title="type parameter in NeedlemanWunsch">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Needleman and Wunsch defined an algorithm for pairwise global sequence alignments (from the first until the last
 <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</th>
</tr>
<tr id="i674" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i675" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/NucleotideImpl.html" title="class in org.biojava.nbio.structure">NucleotideImpl</a></td>
<th class="colLast" scope="row">
<div class="block">A nucleotide group is almost the same as a Hetatm group.</div>
</th>
</tr>
<tr id="i676" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ronn/NullOutputStream.html" title="class in org.biojava.nbio.ronn">NullOutputStream</a></td>
<th class="colLast" scope="row">
<div class="block">The stream that void its input</div>
</th>
</tr>
<tr id="i677" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/figure/NumbersAtRiskPanel.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">NumbersAtRiskPanel</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i678" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/obo/OboFileEventListener.html" title="interface in org.biojava.nbio.ontology.obo">OboFileEventListener</a></td>
<th class="colLast" scope="row">
<div class="block">an interface for events that occur during parsing of .obo files</div>
</th>
</tr>
<tr id="i679" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/obo/OboFileHandler.html" title="class in org.biojava.nbio.ontology.obo">OboFileHandler</a></td>
<th class="colLast" scope="row">
<div class="block">A  file handler for .obo files</div>
</th>
</tr>
<tr id="i680" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/obo/OboFileParser.html" title="class in org.biojava.nbio.ontology.obo">OboFileParser</a></td>
<th class="colLast" scope="row">
<div class="block">A class to parse the content of an OBO file.</div>
</th>
</tr>
<tr id="i681" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/obo/OboFileParser.SOPair.html" title="class in org.biojava.nbio.ontology.obo">OboFileParser.SOPair</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i682" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/io/OboParser.html" title="class in org.biojava.nbio.ontology.io">OboParser</a></td>
<th class="colLast" scope="row">
<div class="block">Parses an OBO file.</div>
</th>
</tr>
<tr id="i683" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/geometry/Octahedron.html" title="class in org.biojava.nbio.structure.symmetry.geometry">Octahedron</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i684" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/Ontology.html" title="interface in org.biojava.nbio.ontology">Ontology</a></td>
<th class="colLast" scope="row">
<div class="block">An ontology.</div>
</th>
</tr>
<tr id="i685" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/Ontology.Impl.html" title="class in org.biojava.nbio.ontology">Ontology.Impl</a></td>
<th class="colLast" scope="row">
<div class="block">A basic in-memory implementation of an ontology</div>
</th>
</tr>
<tr id="i686" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/OntologyException.html" title="class in org.biojava.nbio.ontology">OntologyException</a></td>
<th class="colLast" scope="row">
<div class="block">Thrown to indicate an error in an Ontology object</div>
</th>
</tr>
<tr id="i687" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/OntologyFactory.html" title="interface in org.biojava.nbio.ontology">OntologyFactory</a></td>
<th class="colLast" scope="row">
<div class="block">A factory for Ontology instances.</div>
</th>
</tr>
<tr id="i688" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/OntologyOps.html" title="interface in org.biojava.nbio.ontology">OntologyOps</a></td>
<th class="colLast" scope="row">
<div class="block">This is an interface for optimizing ontology operators.</div>
</th>
</tr>
<tr id="i689" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/OntologyTerm.html" title="interface in org.biojava.nbio.ontology">OntologyTerm</a></td>
<th class="colLast" scope="row">
<div class="block">A term in an ontology which identifies another ontology.</div>
</th>
</tr>
<tr id="i690" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/OntologyTerm.Impl.html" title="class in org.biojava.nbio.ontology">OntologyTerm.Impl</a></td>
<th class="colLast" scope="row">
<div class="block">Simple in-memory implementation of a remote ontology term.</div>
</th>
</tr>
<tr id="i691" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/OntoTools.html" title="class in org.biojava.nbio.ontology">OntoTools</a></td>
<th class="colLast" scope="row">
<div class="block">Tools for manipulating ontologies.</div>
</th>
</tr>
<tr id="i692" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/quaternary/OperatorResolver.html" title="class in org.biojava.nbio.structure.quaternary">OperatorResolver</a></td>
<th class="colLast" scope="row">
<div class="block">A class to resolve the operators for transformations</div>
</th>
</tr>
<tr id="i693" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/OptimalCECPMain.html" title="class in org.biojava.nbio.structure.align.ce">OptimalCECPMain</a></td>
<th class="colLast" scope="row">
<div class="block">A wrapper for <a href="org/biojava/nbio/structure/align/ce/CeMain.html" title="class in org.biojava.nbio.structure.align.ce"><code>CeMain</code></a> which sets default parameters to be appropriate for finding
 circular permutations.</div>
</th>
</tr>
<tr id="i694" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/OptimalCECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">OptimalCECPParameters</a></td>
<th class="colLast" scope="row">
<div class="block">Contains the parameters that can be sent to CE</div>
</th>
</tr>
<tr id="i695" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/OrderDetector.html" title="interface in org.biojava.nbio.structure.symmetry.internal">OrderDetector</a></td>
<th class="colLast" scope="row">
<div class="block">A method to decide the order of symmetry (number of subunits)
 given a structure self-alignment, calculated by CE-Symm.</div>
</th>
</tr>
<tr id="i696" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/quaternary/OrderedPair.html" title="class in org.biojava.nbio.structure.quaternary">OrderedPair</a>&lt;<a href="org/biojava/nbio/structure/quaternary/OrderedPair.html" title="type parameter in OrderedPair">T</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">An ordered pair represents a component of a cartesian product.</div>
</th>
</tr>
<tr id="i697" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ronn/ORonn.html" title="class in org.biojava.nbio.ronn">ORonn</a></td>
<th class="colLast" scope="row">
<div class="block">Fully re-factored and enhanced version of RONN.</div>
</th>
</tr>
<tr id="i698" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ronn/ORonnModel.html" title="class in org.biojava.nbio.ronn">ORonnModel</a></td>
<th class="colLast" scope="row">
<div class="block">Fully re-factored version of RONN model.</div>
</th>
</tr>
<tr id="i699" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/query/OutputHitsGFF.html" title="class in org.biojava.nbio.genome.query">OutputHitsGFF</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i700" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a>&lt;<a href="org/biojava/nbio/structure/contact/Pair.html" title="type parameter in Pair">T</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A Pair of objects.</div>
</th>
</tr>
<tr id="i701" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/PairInProfileScorer.html" title="interface in org.biojava.nbio.alignment.template">PairInProfileScorer</a>&lt;<a href="org/biojava/nbio/alignment/template/PairInProfileScorer.html" title="type parameter in PairInProfileScorer">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/PairInProfileScorer.html" title="type parameter in PairInProfileScorer">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/PairInProfileScorer.html" title="type parameter in PairInProfileScorer">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines an algorithm which computes a score for a sequence alignment pair picked from an alignment <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</th>
</tr>
<tr id="i702" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceAligner</a>&lt;<a href="org/biojava/nbio/alignment/template/PairwiseSequenceAligner.html" title="type parameter in PairwiseSequenceAligner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/PairwiseSequenceAligner.html" title="type parameter in PairwiseSequenceAligner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/PairwiseSequenceAligner.html" title="type parameter in PairwiseSequenceAligner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines an <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</th>
</tr>
<tr id="i703" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceScorer</a>&lt;<a href="org/biojava/nbio/alignment/template/PairwiseSequenceScorer.html" title="type parameter in PairwiseSequenceScorer">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/PairwiseSequenceScorer.html" title="type parameter in PairwiseSequenceScorer">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/PairwiseSequenceScorer.html" title="type parameter in PairwiseSequenceScorer">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines an algorithm which computes a score for a pair of sequences.</div>
</th>
</tr>
<tr id="i704" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/ParameterGUI.html" title="class in org.biojava.nbio.structure.align.gui">ParameterGUI</a></td>
<th class="colLast" scope="row">
<div class="block">UI for <a href="org/biojava/nbio/structure/align/ce/ConfigStrucAligParams.html" title="interface in org.biojava.nbio.structure.align.ce"><code>ConfigStrucAligParams</code></a>, for the AlignmentGUI.</div>
</th>
</tr>
<tr id="i705" class="rowColor">
<td class="colFirst"><a href="demo/ParseFastaFileDemo.html" title="class in demo">ParseFastaFileDemo</a></td>
<th class="colLast" scope="row">
<div class="block">Created by andreas on 6/17/15.</div>
</th>
</tr>
<tr id="i706" class="altColor">
<td class="colFirst"><a href="demo/ParseGO.html" title="class in demo">ParseGO</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i707" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/ParseListener.html" title="interface in org.biojava.nbio.genome.io.fastq">ParseListener</a></td>
<th class="colLast" scope="row">
<div class="block">Low-level event based parser callback.</div>
</th>
</tr>
<tr id="i708" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/exceptions/ParserException.html" title="class in org.biojava.nbio.core.exceptions">ParserException</a></td>
<th class="colLast" scope="row">
<div class="block">General abstraction of different parsing errors</div>
</th>
</tr>
<tr id="i709" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/PartitionRefiner.html" title="interface in org.biojava.nbio.alignment.template">PartitionRefiner</a>&lt;<a href="org/biojava/nbio/alignment/template/PartitionRefiner.html" title="type parameter in PartitionRefiner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/PartitionRefiner.html" title="type parameter in PartitionRefiner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/PartitionRefiner.html" title="type parameter in PartitionRefiner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines an algorithm which computes a new alignment <code>Profile</code> by splitting a current alignment and realigning.</div>
</th>
</tr>
<tr id="i710" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/PassthroughIdentifier.html" title="class in org.biojava.nbio.structure">PassthroughIdentifier</a></td>
<th class="colLast" scope="row">
<div class="block">A stub StructureIdentifier, representing the full structure in all cases.</div>
</th>
</tr>
<tr id="i711" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/PDBBioAssemblyParser.html" title="class in org.biojava.nbio.structure.io">PDBBioAssemblyParser</a></td>
<th class="colLast" scope="row">
<div class="block">Parses REMARK 350 records in a PDB file and creates transformations to
 construct the quaternary structure of a protein from an asymmetric unit</div>
</th>
</tr>
<tr id="i712" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/PDBCrystallographicInfo.html" title="class in org.biojava.nbio.structure">PDBCrystallographicInfo</a></td>
<th class="colLast" scope="row">
<div class="block">A class to hold crystallographic information about a PDB structure.</div>
</th>
</tr>
<tr id="i713" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/PDBDirPanel.html" title="class in org.biojava.nbio.structure.gui.util">PDBDirPanel</a></td>
<th class="colLast" scope="row">
<div class="block">A class to define where a structure for the alignment is coming from</div>
</th>
</tr>
<tr id="i714" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/PDBFileParser.html" title="class in org.biojava.nbio.structure.io">PDBFileParser</a></td>
<th class="colLast" scope="row">
<div class="block">This class implements the actual PDB file parsing.</div>
</th>
</tr>
<tr id="i715" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/PDBFileReader.html" title="class in org.biojava.nbio.structure.io">PDBFileReader</a></td>
<th class="colLast" scope="row">
<div class="block">
  The wrapper class for parsing a PDB file.</div>
</th>
</tr>
<tr id="i716" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/PDBHeader.html" title="class in org.biojava.nbio.structure">PDBHeader</a></td>
<th class="colLast" scope="row">
<div class="block">A class that contains PDB Header information.</div>
</th>
</tr>
<tr id="i717" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/PdbId.html" title="class in org.biojava.nbio.structure">PdbId</a></td>
<th class="colLast" scope="row">
<div class="block">A wrapper class for the PDB identifier.</div>
</th>
</tr>
<tr id="i718" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/client/PdbPair.html" title="class in org.biojava.nbio.structure.align.client">PdbPair</a></td>
<th class="colLast" scope="row">
<div class="block">A pair for structure alignment.</div>
</th>
</tr>
<tr id="i719" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/xml/PdbPairsMessage.html" title="class in org.biojava.nbio.structure.align.xml">PdbPairsMessage</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i720" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/xml/PdbPairXMLConverter.html" title="class in org.biojava.nbio.structure.align.xml">PdbPairXMLConverter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i721" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/PDBParseException.html" title="class in org.biojava.nbio.structure.io">PDBParseException</a></td>
<th class="colLast" scope="row">
<div class="block">An exception during the parsing of a PDB file.</div>
</th>
</tr>
<tr id="i722" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/PDBRecord.html" title="interface in org.biojava.nbio.structure">PDBRecord</a></td>
<th class="colLast" scope="row">
<div class="block">An interface implemented by all classes that represent PDB records.</div>
</th>
</tr>
<tr id="i723" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/PDBServerPanel.html" title="class in org.biojava.nbio.structure.gui.util">PDBServerPanel</a></td>
<th class="colLast" scope="row">
<div class="block">A class to define where a structure for the alignment is coming from</div>
</th>
</tr>
<tr id="i724" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/PDBStatus.html" title="class in org.biojava.nbio.structure">PDBStatus</a></td>
<th class="colLast" scope="row">
<div class="block">Methods for getting the status of a PDB file (current, removed, unreleased)
 and for accessing different versions of the structure.</div>
</th>
</tr>
<tr id="i725" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/PDBStatus.Status.html" title="enum in org.biojava.nbio.structure">PDBStatus.Status</a></td>
<th class="colLast" scope="row">
<div class="block">Represents a simplified 3 state status of PDB IDs.</div>
</th>
</tr>
<tr id="i726" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/util/PDBTemporaryStorageUtils.html" title="class in org.biojava.nbio.structure.io.util">PDBTemporaryStorageUtils</a></td>
<th class="colLast" scope="row">
<div class="block"><b>Internal use only.</div>
</th>
</tr>
<tr id="i727" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/util/PDBTemporaryStorageUtils.LinkRecord.html" title="class in org.biojava.nbio.structure.io.util">PDBTemporaryStorageUtils.LinkRecord</a></td>
<th class="colLast" scope="row">
<div class="block">Temporary data storage for LINK records.</div>
</th>
</tr>
<tr id="i728" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/PDBUploadPanel.html" title="class in org.biojava.nbio.structure.gui.util">PDBUploadPanel</a></td>
<th class="colLast" scope="row">
<div class="block">A JPanel to upload 2 custom PDB files.</div>
</th>
</tr>
<tr id="i729" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i730" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/pdp/PDPParameters.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPParameters</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i731" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/PeptideProperties.html" title="class in org.biojava.nbio.aaproperties">PeptideProperties</a></td>
<th class="colLast" scope="row">
<div class="block">This is an adaptor class which enable the ease of generating protein properties.</div>
</th>
</tr>
<tr id="i732" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/PeptideProperties.SingleLetterAACode.html" title="enum in org.biojava.nbio.aaproperties">PeptideProperties.SingleLetterAACode</a></td>
<th class="colLast" scope="row">
<div class="block">Enumeration of 20 standard amino acid code</div>
</th>
</tr>
<tr id="i733" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html" title="class in org.biojava.nbio.aaproperties">PeptidePropertiesImpl</a></td>
<th class="colLast" scope="row">
<div class="block">This class contains the actual implementation of IPeptideProperties and is wrapped around by PeptideProperties for ease of use.</div>
</th>
</tr>
<tr id="i734" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/utils/PermutationGenerator.html" title="class in org.biojava.nbio.structure.symmetry.utils">PermutationGenerator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i735" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/PermutationGroup.html" title="class in org.biojava.nbio.structure.symmetry.core">PermutationGroup</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i736" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/PlainFastaHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">PlainFastaHeaderParser</a>&lt;<a href="org/biojava/nbio/core/sequence/io/PlainFastaHeaderParser.html" title="type parameter in PlainFastaHeaderParser">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/io/PlainFastaHeaderParser.html" title="type parameter in PlainFastaHeaderParser">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/PlainFastaHeaderParser.html" title="type parameter in PlainFastaHeaderParser">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">The plain fasta header takes everything in the header as a single entity.</div>
</th>
</tr>
<tr id="i737" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/template/Point.html" title="interface in org.biojava.nbio.core.sequence.location.template">Point</a></td>
<th class="colLast" scope="row">
<div class="block">Holds a single point part of a location</div>
</th>
</tr>
<tr id="i738" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/template/Point.Resolver.html" title="interface in org.biojava.nbio.core.sequence.location.template">Point.Resolver</a>&lt;<a href="org/biojava/nbio/core/sequence/location/template/Point.Resolver.html" title="type parameter in Point.Resolver">T</a> extends <a href="org/biojava/nbio/core/sequence/location/template/Point.html" title="interface in org.biojava.nbio.core.sequence.location.template">Point</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Used to resolve a position about a point</div>
</th>
</tr>
<tr id="i739" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/geometry/Polyhedron.html" title="interface in org.biojava.nbio.structure.symmetry.geometry">Polyhedron</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i740" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/PolymerType.html" title="enum in org.biojava.nbio.structure.chem">PolymerType</a></td>
<th class="colLast" scope="row">
<div class="block">Enumerates the classification of polymers.</div>
</th>
</tr>
<tr id="i741" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/utils/PowerSet.html" title="class in org.biojava.nbio.structure.symmetry.utils">PowerSet</a>&lt;<a href="org/biojava/nbio/structure/symmetry/utils/PowerSet.html" title="type parameter in PowerSet">T</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">In mathematics, the power set (or powerset) of any set S, written P(S), is
 the set of all subsets of S, including the empty set and S itself.</div>
</th>
</tr>
<tr id="i742" class="altColor">
<td class="colFirst"><a href="demo/PredictDisorder.html" title="class in demo">PredictDisorder</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i743" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/PrettyXMLWriter.html" title="class in org.biojava.nbio.core.util">PrettyXMLWriter</a></td>
<th class="colLast" scope="row">
<div class="block">Implementation of XMLWriter which emits nicely formatted documents
 to a PrintWriter.</div>
</th>
</tr>
<tr id="i744" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/geometry/Prism.html" title="class in org.biojava.nbio.structure.symmetry.geometry">Prism</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i745" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/ProfeatProperties.html" title="class in org.biojava.nbio.aaproperties.profeat">ProfeatProperties</a></td>
<th class="colLast" scope="row">
<div class="block">This is an adaptor class which enable the ease of generating profeat properties.</div>
</th>
</tr>
<tr id="i746" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/profeat/ProfeatPropertiesImpl.html" title="class in org.biojava.nbio.aaproperties.profeat">ProfeatPropertiesImpl</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i747" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;<a href="org/biojava/nbio/core/alignment/template/Profile.html" title="type parameter in Profile">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/template/Profile.html" title="type parameter in Profile">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/template/Profile.html" title="type parameter in Profile">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a data structure for the results of sequence alignment.</div>
</th>
</tr>
<tr id="i748" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/template/Profile.StringFormat.html" title="enum in org.biojava.nbio.core.alignment.template">Profile.StringFormat</a></td>
<th class="colLast" scope="row">
<div class="block">List of output formats.</div>
</th>
</tr>
<tr id="i749" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair</a>&lt;<a href="org/biojava/nbio/core/alignment/template/ProfilePair.html" title="type parameter in ProfilePair">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/template/ProfilePair.html" title="type parameter in ProfilePair">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/template/ProfilePair.html" title="type parameter in ProfilePair">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a data structure for the results of the alignment of a pair of <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</th>
</tr>
<tr id="i750" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/ProfileProfileAligner.html" title="interface in org.biojava.nbio.alignment.template">ProfileProfileAligner</a>&lt;<a href="org/biojava/nbio/alignment/template/ProfileProfileAligner.html" title="type parameter in ProfileProfileAligner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/ProfileProfileAligner.html" title="type parameter in ProfileProfileAligner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/ProfileProfileAligner.html" title="type parameter in ProfileProfileAligner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines an <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <code>Profile</code>s.</div>
</th>
</tr>
<tr id="i751" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/ProfileProfileScorer.html" title="interface in org.biojava.nbio.alignment.template">ProfileProfileScorer</a>&lt;<a href="org/biojava/nbio/alignment/template/ProfileProfileScorer.html" title="type parameter in ProfileProfileScorer">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/ProfileProfileScorer.html" title="type parameter in ProfileProfileScorer">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/ProfileProfileScorer.html" title="type parameter in ProfileProfileScorer">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines an algorithm which computes a score for a pairing of alignment profiles.</div>
</th>
</tr>
<tr id="i752" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/template/ProfileView.html" title="interface in org.biojava.nbio.core.alignment.template">ProfileView</a>&lt;<a href="org/biojava/nbio/core/alignment/template/ProfileView.html" title="type parameter in ProfileView">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/template/ProfileView.html" title="type parameter in ProfileView">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/template/ProfileView.html" title="type parameter in ProfileView">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a data structure for a view of sequence alignment.</div>
</th>
</tr>
<tr id="i753" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/misc/ProteinComplexSignature.html" title="class in org.biojava.nbio.structure.symmetry.misc">ProteinComplexSignature</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i754" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/util/ProteinMappingTools.html" title="class in org.biojava.nbio.genome.util">ProteinMappingTools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i755" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/ProteinModification.html" title="interface in org.biojava.nbio.protmod">ProteinModification</a></td>
<th class="colLast" scope="row">
<div class="block">This interface defines information about a specific protein
 modification.</div>
</th>
</tr>
<tr id="i756" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/structure/ProteinModificationIdentifier.html" title="class in org.biojava.nbio.protmod.structure">ProteinModificationIdentifier</a></td>
<th class="colLast" scope="row">
<div class="block">Identify attachment modification in a 3-D structure.</div>
</th>
</tr>
<tr id="i757" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/ProteinModificationImpl.html" title="class in org.biojava.nbio.protmod">ProteinModificationImpl</a></td>
<th class="colLast" scope="row">
<div class="block">This class contains information about a specific protein
 modification.</div>
</th>
</tr>
<tr id="i758" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/ProteinModificationImpl.Builder.html" title="class in org.biojava.nbio.protmod">ProteinModificationImpl.Builder</a></td>
<th class="colLast" scope="row">
<div class="block">Uses Builder pattern to build a ProteinModification.</div>
</th>
</tr>
<tr id="i759" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/ProteinModificationRegistry.html" title="class in org.biojava.nbio.protmod">ProteinModificationRegistry</a></td>
<th class="colLast" scope="row">
<div class="block">This class serves as a instance registry by maintaining
 a pool of ProteinModification instances.</div>
</th>
</tr>
<tr id="i760" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/io/ProteinModificationXmlReader.html" title="class in org.biojava.nbio.protmod.io">ProteinModificationXmlReader</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i761" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></td>
<th class="colLast" scope="row">
<div class="block">The representation of a ProteinSequence</div>
</th>
</tr>
<tr id="i762" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">ProteinSequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">Used to create a ProteinSequence from a String to allow for details
 about the location of the sequence etc.</div>
</th>
</tr>
<tr id="i763" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="type parameter in ProxySequenceReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i764" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/jama/QRDecomposition.html" title="class in org.biojava.nbio.structure.jama">QRDecomposition</a></td>
<th class="colLast" scope="row">
<div class="block">QR Decomposition.</div>
</th>
</tr>
<tr id="i765" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/quaternary/QsAlign.html" title="class in org.biojava.nbio.structure.align.quaternary">QsAlign</a></td>
<th class="colLast" scope="row">
<div class="block">Quaternary Structure Alignment (QS-Align).</div>
</th>
</tr>
<tr id="i766" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/quaternary/QsAlignParameters.html" title="class in org.biojava.nbio.structure.align.quaternary">QsAlignParameters</a></td>
<th class="colLast" scope="row">
<div class="block">The parameter bean for the <a href="org/biojava/nbio/structure/align/quaternary/QsAlign.html" title="class in org.biojava.nbio.structure.align.quaternary"><code>QsAlign</code></a> algorithm.</div>
</th>
</tr>
<tr id="i767" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/quaternary/QsAlignResult.html" title="class in org.biojava.nbio.structure.align.quaternary">QsAlignResult</a></td>
<th class="colLast" scope="row">
<div class="block">Result of a Quaternary Structure Alignment <a href="org/biojava/nbio/structure/align/quaternary/QsAlign.html" title="class in org.biojava.nbio.structure.align.quaternary"><code>QsAlign</code></a>.</div>
</th>
</tr>
<tr id="i768" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/quaternary/QsRelation.html" title="enum in org.biojava.nbio.structure.align.quaternary">QsRelation</a></td>
<th class="colLast" scope="row">
<div class="block">The Quaternary Structure Relation describes the pairwise relation between two
 quaternary structures.</div>
</th>
</tr>
<tr id="i769" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/features/Qualifier.html" title="class in org.biojava.nbio.core.sequence.features">Qualifier</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i770" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QualityFeature</a>&lt;<a href="org/biojava/nbio/core/sequence/features/QualityFeature.html" title="type parameter in QualityFeature">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/features/QualityFeature.html" title="type parameter in QualityFeature">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/features/QualityFeature.html" title="type parameter in QualityFeature">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">DNA Sequences produced by modern sequencers usually have quality informaion
 attached to them.</div>
</th>
</tr>
<tr id="i771" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QuantityFeature</a>&lt;<a href="org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="type parameter in QuantityFeature">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="type parameter in QuantityFeature">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="type parameter in QuantityFeature">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">It is common to have a numerical value or values associated with a feature.</div>
</th>
</tr>
<tr id="i772" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/QuatSuperpositionScorer.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSuperpositionScorer</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i773" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryDetector.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryDetector</a></td>
<th class="colLast" scope="row">
<div class="block">Detects the symmetry (global, pseudo, internal and local) of protein
 structures.</div>
</th>
</tr>
<tr id="i774" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryParameters.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryParameters</a></td>
<th class="colLast" scope="row">
<div class="block">The QuatSymmetryParameters specify the options used for the detection of
 quaternary symmetry in structures using the <a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryDetector.html" title="class in org.biojava.nbio.structure.symmetry.core"><code>QuatSymmetryDetector</code></a>.</div>
</th>
</tr>
<tr id="i775" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryResults</a></td>
<th class="colLast" scope="row">
<div class="block">Holds the results of quaternary symmetry perception obtained with
 <a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryDetector.html" title="class in org.biojava.nbio.structure.symmetry.core"><code>QuatSymmetryDetector</code></a>.</div>
</th>
</tr>
<tr id="i776" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryScores.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryScores</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i777" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetrySolver.html" title="interface in org.biojava.nbio.structure.symmetry.core">QuatSymmetrySolver</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i778" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetrySubunits.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetrySubunits</a></td>
<th class="colLast" scope="row">
<div class="block">A bean to represent information about the set of <a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a>s being
 considered for symmetry detection.</div>
</th>
</tr>
<tr id="i779" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/jmol/RasmolCommandListener.html" title="class in org.biojava.nbio.structure.align.gui.jmol">RasmolCommandListener</a></td>
<th class="colLast" scope="row">
<div class="block">a utility class that listens to Ramsol script commands in the @link <a href="org/biojava/nbio/structure/gui/BiojavaJmol.html" title="class in org.biojava.nbio.structure.gui"><code>BiojavaJmol</code></a> class</div>
</th>
</tr>
<tr id="i780" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/geometry/RectangularPrism.html" title="class in org.biojava.nbio.structure.symmetry.geometry">RectangularPrism</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i781" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/ReducedChemCompProvider.html" title="class in org.biojava.nbio.structure.chem">ReducedChemCompProvider</a></td>
<th class="colLast" scope="row">
<div class="block">Unlike the <a href="org/biojava/nbio/structure/chem/DownloadChemCompProvider.html" title="class in org.biojava.nbio.structure.chem"><code>DownloadChemCompProvider</code></a>, this  <a href="org/biojava/nbio/structure/chem/ChemCompProvider.html" title="interface in org.biojava.nbio.structure.chem"><code>ChemCompProvider</code></a> does not download any chem comp
 definitions.</div>
</th>
</tr>
<tr id="i782" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/reference/ReferenceInterface.html" title="interface in org.biojava.nbio.core.sequence.reference">ReferenceInterface</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i783" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/util/ReferenceSuperimposer.html" title="class in org.biojava.nbio.structure.align.multiple.util">ReferenceSuperimposer</a></td>
<th class="colLast" scope="row">
<div class="block">Superimposes each structure in a <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> onto a reference
 structure.</div>
</th>
</tr>
<tr id="i784" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/RefinerFailedException.html" title="class in org.biojava.nbio.structure.symmetry.internal">RefinerFailedException</a></td>
<th class="colLast" scope="row">
<div class="block">Refinement of the self-alignment failed.</div>
</th>
</tr>
<tr id="i785" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/hmmer/RemoteHmmerScan.html" title="class in org.biojava.nbio.ws.hmmer">RemoteHmmerScan</a></td>
<th class="colLast" scope="row">
<div class="block">Makes remote calls to the HMMER web service at the EBI web site and returns Pfam domain annotations for an input protein sequence.</div>
</th>
</tr>
<tr id="i786" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/RemotePairwiseAlignmentOutputProperties.html" title="interface in org.biojava.nbio.ws.alignment">RemotePairwiseAlignmentOutputProperties</a></td>
<th class="colLast" scope="row">
<div class="block">RemotePairwiseAlignmentOutputProperties: the simplest representation of an object capable of holding
 output formatting informations to be fed to a RemotePairwiseAlignmentService-implemented object.</div>
</th>
</tr>
<tr id="i787" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojava.nbio.ws.alignment">RemotePairwiseAlignmentProperties</a></td>
<th class="colLast" scope="row">
<div class="block">RemotePairwiseAlignmentProperties is a interface that contains the barest of
 methods for setting and getting Alignment properties.</div>
</th>
</tr>
<tr id="i788" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ws/alignment/RemotePairwiseAlignmentService.html" title="interface in org.biojava.nbio.ws.alignment">RemotePairwiseAlignmentService</a></td>
<th class="colLast" scope="row">
<div class="block">This interface specifies minimal information needed to execute a pairwise alignment on a remote service.</div>
</th>
</tr>
<tr id="i789" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/RemoteTerm.html" title="interface in org.biojava.nbio.ontology">RemoteTerm</a></td>
<th class="colLast" scope="row">
<div class="block">A term in another ontology.</div>
</th>
</tr>
<tr id="i790" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/RemoteTerm.Impl.html" title="class in org.biojava.nbio.ontology">RemoteTerm.Impl</a></td>
<th class="colLast" scope="row">
<div class="block">Simple in-memory implementation of a remote ontology term.</div>
</th>
</tr>
<tr id="i791" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/RenderStyle.html" title="enum in org.biojava.nbio.structure.gui">RenderStyle</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i792" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/RescoreRefiner.html" title="interface in org.biojava.nbio.alignment.template">RescoreRefiner</a>&lt;<a href="org/biojava/nbio/alignment/template/RescoreRefiner.html" title="type parameter in RescoreRefiner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/template/RescoreRefiner.html" title="type parameter in RescoreRefiner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/template/RescoreRefiner.html" title="type parameter in RescoreRefiner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines an algorithm which computes a new alignment <code>Profile</code> by rescoring all pairs in an alignment and
 realigning.</div>
</th>
</tr>
<tr id="i793" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/ResidualsCoxph.html" title="class in org.biojava.nbio.survival.cox">ResidualsCoxph</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i794" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/ResidualsCoxph.Type.html" title="enum in org.biojava.nbio.survival.cox">ResidualsCoxph.Type</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i795" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/ResidueGroup.html" title="class in org.biojava.nbio.structure.symmetry.internal">ResidueGroup</a></td>
<th class="colLast" scope="row">
<div class="block">A ResidueGroup is a set of residues that are part of a maximally connected
 component of the self-Alignment Graph in symmetry analysis.</div>
</th>
</tr>
<tr id="i796" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a></td>
<th class="colLast" scope="row">
<div class="block">Everything that is needed to uniquely describe a residue position</div>
</th>
</tr>
<tr id="i797" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/ResidueRange.html" title="class in org.biojava.nbio.structure">ResidueRange</a></td>
<th class="colLast" scope="row">
<div class="block">A chainName, a start residue, and an end residue.</div>
</th>
</tr>
<tr id="i798" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/ResidueRangeAndLength.html" title="class in org.biojava.nbio.structure">ResidueRangeAndLength</a></td>
<th class="colLast" scope="row">
<div class="block">A chain, a start residue, and an end residue.</div>
</th>
</tr>
<tr id="i799" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/ResidueType.html" title="enum in org.biojava.nbio.structure.chem">ResidueType</a></td>
<th class="colLast" scope="row">
<div class="block">Enumerates the possible classifications of residues.</div>
</th>
</tr>
<tr id="i800" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/ResourceManager.html" title="class in org.biojava.nbio.structure.align.util">ResourceManager</a></td>
<th class="colLast" scope="row">
<div class="block">A class that manages the Strings that are defined in the spice.properties file.</div>
</th>
</tr>
<tr id="i801" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/search/io/Result.html" title="class in org.biojava.nbio.core.search.io">Result</a></td>
<th class="colLast" scope="row">
<div class="block">This class models a search result.</div>
</th>
</tr>
<tr id="i802" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/search/io/ResultFactory.html" title="interface in org.biojava.nbio.core.search.io">ResultFactory</a></td>
<th class="colLast" scope="row">
<div class="block">Designed by Paolo Pavan.</div>
</th>
</tr>
<tr id="i803" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ReversedSequenceView</a>&lt;<a href="org/biojava/nbio/core/sequence/views/ReversedSequenceView.html" title="type parameter in ReversedSequenceView">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">For a given sequence this class will return the base at the reversed
 position i.e. in a sequence of size 10, if you request base 2 you will get
 back the base at position 9.</div>
</th>
</tr>
<tr id="i804" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/RiskInfo.html" title="class in org.biojava.nbio.survival.cox">RiskInfo</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i805" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/compound/RNACompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">RNACompoundSet</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i806" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/RNASequence.html" title="class in org.biojava.nbio.core.sequence">RNASequence</a></td>
<th class="colLast" scope="row">
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</th>
</tr>
<tr id="i807" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">RNASequenceCreator</a></td>
<th class="colLast" scope="row">
<div class="block">Used to create a RNA sequence</div>
</th>
</tr>
<tr id="i808" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/views/RnaSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">RnaSequenceView</a></td>
<th class="colLast" scope="row">
<div class="block">Attempts to do on the fly translation of RNA by not requesting the compounds
 until asked.</div>
</th>
</tr>
<tr id="i809" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava.nbio.core.sequence.transcription">RNAToAminoAcidTranslator</a></td>
<th class="colLast" scope="row">
<div class="block">Takes a <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> of <a href="org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound"><code>NucleotideCompound</code></a> which should represent
 an RNA sequence (<a href="org/biojava/nbio/core/sequence/RNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>RNASequence</code></a> is good for this) and returns a list of
 <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> which hold <a href="org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound"><code>AminoAcidCompound</code></a>.</div>
</th>
</tr>
<tr id="i810" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ronn/RonnConstraint.html" title="class in org.biojava.nbio.ronn">RonnConstraint</a></td>
<th class="colLast" scope="row">
<div class="block">A collection of various constrain values used by RONN</div>
</th>
</tr>
<tr id="i811" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ronn/RonnConstraint.Threshold.html" title="enum in org.biojava.nbio.ronn">RonnConstraint.Threshold</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i812" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/Rotation.html" title="class in org.biojava.nbio.structure.symmetry.core">Rotation</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i813" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/RotationAxis.html" title="class in org.biojava.nbio.structure.align.util">RotationAxis</a></td>
<th class="colLast" scope="row">
<div class="block">Calculates the rotation axis for an alignment</div>
</th>
</tr>
<tr id="i814" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/axis/RotationAxisAligner.html" title="class in org.biojava.nbio.structure.symmetry.axis">RotationAxisAligner</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i815" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/RotationGroup.html" title="class in org.biojava.nbio.structure.symmetry.core">RotationGroup</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i816" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/RotationSolver.html" title="class in org.biojava.nbio.structure.symmetry.core">RotationSolver</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i817" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/SangerFastqReader.html" title="class in org.biojava.nbio.genome.io.fastq">SangerFastqReader</a></td>
<th class="colLast" scope="row">
<div class="block">Reader for <a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html#FASTQ_SANGER"><code>FastqVariant.FASTQ_SANGER</code></a> formatted sequences.</div>
</th>
</tr>
<tr id="i818" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/SangerFastqWriter.html" title="class in org.biojava.nbio.genome.io.fastq">SangerFastqWriter</a></td>
<th class="colLast" scope="row">
<div class="block">Writer for <a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html#FASTQ_SANGER"><code>FastqVariant.FASTQ_SANGER</code></a> formatted sequences.</div>
</th>
</tr>
<tr id="i819" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/ScaleableMatrixPanel.html" title="class in org.biojava.nbio.structure.gui">ScaleableMatrixPanel</a></td>
<th class="colLast" scope="row">
<div class="block">A JPanel that can display the underlying distance matrix
 data of the protein structure alignment algorithm.</div>
</th>
</tr>
<tr id="i820" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/matrices/ScaledSubstitutionMatrix.html" title="class in org.biojava.nbio.core.alignment.matrices">ScaledSubstitutionMatrix</a></td>
<th class="colLast" scope="row">
<div class="block">The biojava-alignment module represents substitution matrices with short
 values.</div>
</th>
</tr>
<tr id="i821" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/xml/SchemaGenerator.html" title="class in org.biojava.nbio.aaproperties.xml">SchemaGenerator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i822" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/autosuggest/SCOPAutoSuggestProvider.html" title="class in org.biojava.nbio.structure.align.gui.autosuggest">SCOPAutoSuggestProvider</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i823" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/ScopCategory.html" title="enum in org.biojava.nbio.structure.scop">ScopCategory</a></td>
<th class="colLast" scope="row">
<div class="block">The various categories provided by SCOP.</div>
</th>
</tr>
<tr id="i824" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/ScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">ScopDatabase</a></td>
<th class="colLast" scope="row">
<div class="block">General API how to interact with SCOP</div>
</th>
</tr>
<tr id="i825" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/ScopDescription.html" title="class in org.biojava.nbio.structure.scop">ScopDescription</a></td>
<th class="colLast" scope="row">
<div class="block">Contains data from
 dir.des.scop.txt_1.75</div>
</th>
</tr>
<tr id="i826" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop">ScopDomain</a></td>
<th class="colLast" scope="row">
<div class="block">Container for the information for a domain.</div>
</th>
</tr>
<tr id="i827" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/ScopFactory.html" title="class in org.biojava.nbio.structure.scop">ScopFactory</a></td>
<th class="colLast" scope="row">
<div class="block">Controls the global ScopDatabase being used.</div>
</th>
</tr>
<tr id="i828" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/ScopInstallation.html" title="class in org.biojava.nbio.structure.scop">ScopInstallation</a></td>
<th class="colLast" scope="row">
<div class="block">This class provides access to the SCOP protein structure classification.</div>
</th>
</tr>
<tr id="i829" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/ScopInstallationInstance.html" title="class in org.biojava.nbio.structure.gui.util">ScopInstallationInstance</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i830" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/ScopIOException.html" title="class in org.biojava.nbio.structure.scop">ScopIOException</a></td>
<th class="colLast" scope="row">
<div class="block">Indicates that an I/O error occurred with SCOP lazy initialization.</div>
</th>
</tr>
<tr id="i831" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/ScopMirror.html" title="class in org.biojava.nbio.structure.scop">ScopMirror</a></td>
<th class="colLast" scope="row">
<div class="block">Helper class to store paths to the four SCOP files

 The string "%s" is replaced with the version number.</div>
</th>
</tr>
<tr id="i832" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/scop/ScopNode.html" title="class in org.biojava.nbio.structure.scop">ScopNode</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i833" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/ScopSelectPanel.html" title="class in org.biojava.nbio.structure.gui.util">ScopSelectPanel</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i834" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/template/Scorer.html" title="interface in org.biojava.nbio.alignment.template">Scorer</a></td>
<th class="colLast" scope="row">
<div class="block">Defines an algorithm which computes a score.</div>
</th>
</tr>
<tr id="i835" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/multiple/ScoresCache.html" title="interface in org.biojava.nbio.structure.align.multiple">ScoresCache</a></td>
<th class="colLast" scope="row">
<div class="block">Interface for classes which implement a temporary cache for various numeric
 scores, e.g.</div>
</th>
</tr>
<tr id="i836" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/search/io/SearchIO.html" title="class in org.biojava.nbio.core.search.io">SearchIO</a></td>
<th class="colLast" scope="row">
<div class="block">Designed by Paolo Pavan.</div>
</th>
</tr>
<tr id="i837" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/secstruc/SecStrucCalc.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucCalc</a></td>
<th class="colLast" scope="row">
<div class="block">Calculate and assign the secondary structure (SS) to the
 Groups of a Structure object.</div>
</th>
</tr>
<tr id="i838" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/secstruc/SecStrucElement.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucElement</a></td>
<th class="colLast" scope="row">
<div class="block">A secondary structure element (SSE) is an object representing a block of
 sequential residues that share the same secondary structure type.</div>
</th>
</tr>
<tr id="i839" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/secstruc/SecStrucInfo.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucInfo</a></td>
<th class="colLast" scope="row">
<div class="block">Container for the secondary structure information of a single residue.</div>
</th>
</tr>
<tr id="i840" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a></td>
<th class="colLast" scope="row">
<div class="block">This class extends the basic container for secondary structure annotation,
 including all the information used in the DSSP algorithm.</div>
</th>
</tr>
<tr id="i841" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/secstruc/SecStrucTools.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucTools</a></td>
<th class="colLast" scope="row">
<div class="block">This class contains methods for obtaining and converting secondary structure
 information from BioJava <a href="org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a>s.</div>
</th>
</tr>
<tr id="i842" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/secstruc/SecStrucType.html" title="enum in org.biojava.nbio.structure.secstruc">SecStrucType</a></td>
<th class="colLast" scope="row">
<div class="block">This enum contains all of the secondary structure types found in the DSSP
 output.</div>
</th>
</tr>
<tr id="i843" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/pdp/Segment.html" title="class in org.biojava.nbio.structure.domain.pdp">Segment</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i844" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/pdp/SegmentComparator.html" title="class in org.biojava.nbio.structure.domain.pdp">SegmentComparator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i845" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/Selection.html" title="interface in org.biojava.nbio.structure.gui">Selection</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i846" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/SelectionImpl.html" title="class in org.biojava.nbio.structure.gui">SelectionImpl</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i847" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/SelectMultiplePanel.html" title="class in org.biojava.nbio.structure.gui.util">SelectMultiplePanel</a></td>
<th class="colLast" scope="row">
<div class="block">A Text Panel that allows the user to specify multiple structure
 identifiers, space separated.</div>
</th>
</tr>
<tr id="i848" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/SelectPDBPanel.html" title="class in org.biojava.nbio.structure.align.gui">SelectPDBPanel</a></td>
<th class="colLast" scope="row">
<div class="block">A Panel that allows user to specify PDB & chain ID, as well as sub-ranges</div>
</th>
</tr>
<tr id="i849" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a></td>
<th class="colLast" scope="row">
<div class="block">Created by andreas on 9/11/15.</div>
</th>
</tr>
<tr id="i850" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/SeqMisMatchImpl.html" title="class in org.biojava.nbio.structure">SeqMisMatchImpl</a></td>
<th class="colLast" scope="row">
<div class="block">Created by andreas on 9/11/15.</div>
</th>
</tr>
<tr id="i851" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/SeqRes2AtomAligner.html" title="class in org.biojava.nbio.structure.io">SeqRes2AtomAligner</a></td>
<th class="colLast" scope="row">
<div class="block">Aligns the SEQRES residues to the ATOM residues.</div>
</th>
</tr>
<tr id="i852" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Main interface for defining a collection of Compounds and accessing them
 using biological indexes</div>
</th>
</tr>
<tr id="i853" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/storage/SequenceAsStringHelper.html" title="class in org.biojava.nbio.core.sequence.storage">SequenceAsStringHelper</a>&lt;<a href="org/biojava/nbio/core/sequence/storage/SequenceAsStringHelper.html" title="type parameter in SequenceAsStringHelper">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">This is a common method that can be used across multiple storage/proxy implementations to
 handle Negative strand and other interesting elements of sequence data.</div>
</th>
</tr>
<tr id="i854" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/SequenceComparator.html" title="class in org.biojava.nbio.core.sequence">SequenceComparator</a></td>
<th class="colLast" scope="row">
<div class="block">Used to sort sequences in ascending order of  bioBegin property.</div>
</th>
</tr>
<tr id="i855" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i856" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/SequenceDisplay.html" title="class in org.biojava.nbio.structure.gui">SequenceDisplay</a></td>
<th class="colLast" scope="row">
<div class="block">A sequence display that can show the results of a protein structure alignment.</div>
</th>
</tr>
<tr id="i857" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava.nbio.core.sequence.loader">SequenceFileProxyLoader</a>&lt;<a href="org/biojava/nbio/core/sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">This class represents the storage container of a sequence stored in a fasta file where
 the initial parsing of the file we store the offset and length of the sequence.</div>
</th>
</tr>
<tr id="i858" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/SequenceFunctionOrderDetector.html" title="class in org.biojava.nbio.structure.symmetry.internal">SequenceFunctionOrderDetector</a></td>
<th class="colLast" scope="row">
<div class="block">Calls Spencer's method for determining order.</div>
</th>
</tr>
<tr id="i859" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/SequenceFunctionRefiner.html" title="class in org.biojava.nbio.structure.symmetry.internal">SequenceFunctionRefiner</a></td>
<th class="colLast" scope="row">
<div class="block">Creates a refined alignment with the CE-Symm alternative self-alignment.</div>
</th>
</tr>
<tr id="i860" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="type parameter in SequenceHeaderParserInterface">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="type parameter in SequenceHeaderParserInterface">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i861" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/SequenceLocation.html" title="class in org.biojava.nbio.core.sequence.location">SequenceLocation</a>&lt;<a href="org/biojava/nbio/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/location/SequenceLocation.html" title="type parameter in SequenceLocation">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A location in a sequence that keeps a reference to its parent sequence</div>
</th>
</tr>
<tr id="i862" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/SequenceMixin.html" title="class in org.biojava.nbio.core.sequence.template">SequenceMixin</a></td>
<th class="colLast" scope="row">
<div class="block">Provides a set of static methods to be used as static imports when needed
 across multiple Sequence implementations but inheritance gets in the way.</div>
</th>
</tr>
<tr id="i863" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava.nbio.core.sequence.template">SequenceMixin.SequenceIterator</a>&lt;<a href="org/biojava/nbio/core/sequence/template/SequenceMixin.SequenceIterator.html" title="type parameter in SequenceMixin.SequenceIterator">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A basic sequence iterator which iterates over the given Sequence by
 biological index.</div>
</th>
</tr>
<tr id="i864" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/SequenceMouseListener.html" title="class in org.biojava.nbio.structure.gui.util">SequenceMouseListener</a></td>
<th class="colLast" scope="row">
<div class="block">a mouse listener for the AbstractChainRenderer class
 it listens to all mouse events and triggeres appropriate
 SequenceListener and FeatureListener events</div>
</th>
</tr>
<tr id="i865" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava.nbio.core.sequence">SequenceOptimizationHints</a></td>
<th class="colLast" scope="row">
<div class="block">A static class that provides optimization hints for memory or performance handling of sequence data.</div>
</th>
</tr>
<tr id="i866" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/SequenceOptimizationHints.SequenceCollection.html" title="enum in org.biojava.nbio.core.sequence">SequenceOptimizationHints.SequenceCollection</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i867" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/SequenceOptimizationHints.SequenceUsage.html" title="enum in org.biojava.nbio.core.sequence">SequenceOptimizationHints.SequenceUsage</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i868" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;<a href="org/biojava/nbio/core/alignment/template/SequencePair.html" title="type parameter in SequencePair">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/template/SequencePair.html" title="type parameter in SequencePair">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/template/SequencePair.html" title="type parameter in SequencePair">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a data structure for the results of pairwise sequence alignment.</div>
</th>
</tr>
<tr id="i869" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/io/template/SequenceParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceParserInterface</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i870" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/SequenceProxyView.html" title="class in org.biojava.nbio.core.sequence.template">SequenceProxyView</a>&lt;<a href="org/biojava/nbio/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i871" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="org/biojava/nbio/core/sequence/template/SequenceReader.html" title="type parameter in SequenceReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i872" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/SequenceScalePanel.html" title="class in org.biojava.nbio.structure.gui.util">SequenceScalePanel</a></td>
<th class="colLast" scope="row">
<div class="block">A class that draws a Sequence as a rectangle, a scale display over it.</div>
</th>
</tr>
<tr id="i873" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/SequenceTools.html" title="class in org.biojava.nbio.core.util">SequenceTools</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i874" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/data/sequence/SequenceUtil.html" title="class in org.biojava.nbio.data.sequence">SequenceUtil</a></td>
<th class="colLast" scope="row">
<div class="block">Utility class for operations on sequences</div>
</th>
</tr>
<tr id="i875" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="org/biojava/nbio/core/sequence/template/SequenceView.html" title="type parameter in SequenceView">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i876" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/SerializableCache.html" title="class in org.biojava.nbio.structure.domain">SerializableCache</a>&lt;<a href="org/biojava/nbio/structure/domain/SerializableCache.html" title="type parameter in SerializableCache">K</a>,&#8203;<a href="org/biojava/nbio/structure/domain/SerializableCache.html" title="type parameter in SerializableCache">V</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A class that provides all that is necessary to create a Serializable Cache</div>
</th>
</tr>
<tr id="i877" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/domain/pdp/ShortSegmentRemover.html" title="class in org.biojava.nbio.structure.domain.pdp">ShortSegmentRemover</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i878" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/ShowPDBIDListener.html" title="class in org.biojava.nbio.structure.align.gui">ShowPDBIDListener</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i879" class="rowColor">
<td class="colFirst"><a href="demo/ShowStructureInJmol.html" title="class in demo">ShowStructureInJmol</a></td>
<th class="colLast" scope="row">
<div class="block">Demo how to load and display a structure in Jmol</div>
</th>
</tr>
<tr id="i880" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/sifts/SiftsChainEntry.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsChainEntry</a></td>
<th class="colLast" scope="row">
<div class="block">An entry in the chain-level SIFTS mapping between UniProt and the PDB.</div>
</th>
</tr>
<tr id="i881" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/sifts/SiftsChainToUniprotMapping.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsChainToUniprotMapping</a></td>
<th class="colLast" scope="row">
<div class="block">A mapping between UniProt entries and PDB chains.</div>
</th>
</tr>
<tr id="i882" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/sifts/SiftsEntity.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsEntity</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i883" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/sifts/SiftsMappingProvider.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsMappingProvider</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i884" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/sifts/SiftsResidue.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsResidue</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i885" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/sifts/SiftsSegment.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsSegment</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i886" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/sifts/SiftsXMLParser.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsXMLParser</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i887" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/calc/SigEva.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">SigEva</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i888" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/SimpleAlignedSequence.html" title="class in org.biojava.nbio.core.alignment">SimpleAlignedSequence</a>&lt;<a href="org/biojava/nbio/core/alignment/SimpleAlignedSequence.html" title="type parameter in SimpleAlignedSequence">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/SimpleAlignedSequence.html" title="type parameter in SimpleAlignedSequence">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/SimpleAlignedSequence.html" title="type parameter in SimpleAlignedSequence">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements a data structure for a <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</th>
</tr>
<tr id="i889" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/SimpleGapPenalty.html" title="class in org.biojava.nbio.alignment">SimpleGapPenalty</a></td>
<th class="colLast" scope="row">
<div class="block">Implements a data structure for the gap penalties used during a sequence alignment routine.</div>
</th>
</tr>
<tr id="i890" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/SimpleLocation.html" title="class in org.biojava.nbio.core.sequence.location">SimpleLocation</a></td>
<th class="colLast" scope="row">
<div class="block">Very basic implementation of the Location interface which defines a series
 of simple constructors.</div>
</th>
</tr>
<tr id="i891" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/location/SimplePoint.html" title="class in org.biojava.nbio.core.sequence.location">SimplePoint</a></td>
<th class="colLast" scope="row">
<div class="block">Basic implementation of the Point interface.</div>
</th>
</tr>
<tr id="i892" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/SimpleProfile.html" title="class in org.biojava.nbio.core.alignment">SimpleProfile</a>&lt;<a href="org/biojava/nbio/core/alignment/SimpleProfile.html" title="type parameter in SimpleProfile">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/SimpleProfile.html" title="type parameter in SimpleProfile">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/SimpleProfile.html" title="type parameter in SimpleProfile">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements a data structure for the results of sequence alignment.</div>
</th>
</tr>
<tr id="i893" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/SimpleProfilePair.html" title="class in org.biojava.nbio.core.alignment">SimpleProfilePair</a>&lt;<a href="org/biojava/nbio/core/alignment/SimpleProfilePair.html" title="type parameter in SimpleProfilePair">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/SimpleProfilePair.html" title="type parameter in SimpleProfilePair">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/SimpleProfilePair.html" title="type parameter in SimpleProfilePair">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements a data structure for the results of the alignment of a pair of <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</th>
</tr>
<tr id="i894" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="class in org.biojava.nbio.alignment">SimpleProfileProfileAligner</a>&lt;<a href="org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements a simple (naive) <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</th>
</tr>
<tr id="i895" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/SimpleSequencePair.html" title="class in org.biojava.nbio.core.alignment">SimpleSequencePair</a>&lt;<a href="org/biojava/nbio/core/alignment/SimpleSequencePair.html" title="type parameter in SimpleSequencePair">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/core/alignment/SimpleSequencePair.html" title="type parameter in SimpleSequencePair">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/alignment/SimpleSequencePair.html" title="type parameter in SimpleSequencePair">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements a data structure for the results of pairwise sequence alignment.</div>
</th>
</tr>
<tr id="i896" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/matrices/SimpleSubstitutionMatrix.html" title="class in org.biojava.nbio.core.alignment.matrices">SimpleSubstitutionMatrix</a>&lt;<a href="org/biojava/nbio/core/alignment/matrices/SimpleSubstitutionMatrix.html" title="type parameter in SimpleSubstitutionMatrix">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implements a data structure which holds the score (penalty or bonus) given during alignment for the exchange of one
 <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> in a sequence for another.</div>
</th>
</tr>
<tr id="i897" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">SingleCompoundSequenceReader</a>&lt;<a href="org/biojava/nbio/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">An implementation of the SequenceReader interface which for every
 call will return only 1 compound (given to it during construction; a String
 is also valid but will require a CompoundSet).</div>
</th>
</tr>
<tr id="i898" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/SingleLinkageClusterer.html" title="class in org.biojava.nbio.core.util">SingleLinkageClusterer</a></td>
<th class="colLast" scope="row">
<div class="block">An implementation of a single linkage clusterer

 See http://en.wikipedia.org/wiki/Single-linkage_clustering</div>
</th>
</tr>
<tr id="i899" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/jama/SingularValueDecomposition.html" title="class in org.biojava.nbio.structure.jama">SingularValueDecomposition</a></td>
<th class="colLast" scope="row">
<div class="block">Singular Value Decomposition.</div>
</th>
</tr>
<tr id="i900" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/phosphosite/Site.html" title="class in org.biojava.nbio.phosphosite">Site</a></td>
<th class="colLast" scope="row">
<div class="block">Created by ap3 on 31/10/2014.</div>
</th>
</tr>
<tr id="i901" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/Site.html" title="class in org.biojava.nbio.structure">Site</a></td>
<th class="colLast" scope="row">
<div class="block">Holds the data of sites presented in PDB files.</div>
</th>
</tr>
<tr id="i902" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/utils/SmallAnnotation.html" title="class in org.biojava.nbio.ontology.utils">SmallAnnotation</a></td>
<th class="colLast" scope="row">
<div class="block">Annotation that is optimized for memory usage.</div>
</th>
</tr>
<tr id="i903" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/utils/SmallMap.html" title="class in org.biojava.nbio.ontology.utils">SmallMap</a></td>
<th class="colLast" scope="row">
<div class="block">Lightweight implementation of Map which uses little memory to store a
 small number of mappings, at the expense of scalability.</div>
</th>
</tr>
<tr id="i904" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/SmithWaterman.html" title="class in org.biojava.nbio.alignment">SmithWaterman</a>&lt;<a href="org/biojava/nbio/alignment/SmithWaterman.html" title="type parameter in SmithWaterman">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/SmithWaterman.html" title="type parameter in SmithWaterman">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/SmithWaterman.html" title="type parameter in SmithWaterman">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Smith and Waterman defined an algorithm for pairwise local sequence alignments (best match of sections from each
 <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</th>
</tr>
<tr id="i905" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/seq/SmithWaterman3Daligner.html" title="class in org.biojava.nbio.structure.align.seq">SmithWaterman3Daligner</a></td>
<th class="colLast" scope="row">
<div class="block">Provides a 3D superimposition of two structures based on their sequence
 alignment.</div>
</th>
</tr>
<tr id="i906" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/seq/SmithWaterman3DParameters.html" title="class in org.biojava.nbio.structure.align.seq">SmithWaterman3DParameters</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i907" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/seq/SmithWatermanUserArgumentProcessor.html" title="class in org.biojava.nbio.structure.align.seq">SmithWatermanUserArgumentProcessor</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i908" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/seq/SmithWatermanUserArgumentProcessor.SmithWatermanStartupParams.html" title="class in org.biojava.nbio.structure.align.seq">SmithWatermanUserArgumentProcessor.SmithWatermanStartupParams</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i909" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/SoftHashMap.html" title="class in org.biojava.nbio.core.util">SoftHashMap</a>&lt;<a href="org/biojava/nbio/core/util/SoftHashMap.html" title="type parameter in SoftHashMap">K</a>,&#8203;<a href="org/biojava/nbio/core/util/SoftHashMap.html" title="type parameter in SoftHashMap">V</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A in memory cache using soft references.</div>
</th>
</tr>
<tr id="i910" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/SolexaFastqReader.html" title="class in org.biojava.nbio.genome.io.fastq">SolexaFastqReader</a></td>
<th class="colLast" scope="row">
<div class="block">Reader for <a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html#FASTQ_SOLEXA"><code>FastqVariant.FASTQ_SOLEXA</code></a> formatted sequences.</div>
</th>
</tr>
<tr id="i911" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/SolexaFastqWriter.html" title="class in org.biojava.nbio.genome.io.fastq">SolexaFastqWriter</a></td>
<th class="colLast" scope="row">
<div class="block">Writer for <a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html#FASTQ_SOLEXA"><code>FastqVariant.FASTQ_SOLEXA</code></a> formatted sequences.</div>
</th>
</tr>
<tr id="i912" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/xtal/SpaceGroup.html" title="class in org.biojava.nbio.structure.xtal">SpaceGroup</a></td>
<th class="colLast" scope="row">
<div class="block">A crystallographic space group.</div>
</th>
</tr>
<tr id="i913" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/xtal/io/SpaceGroupMapElements.html" title="class in org.biojava.nbio.structure.xtal.io">SpaceGroupMapElements</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i914" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/xtal/io/SpaceGroupMapRoot.html" title="class in org.biojava.nbio.structure.xtal.io">SpaceGroupMapRoot</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i915" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/math/SparseSquareMatrix.html" title="class in org.biojava.nbio.structure.math">SparseSquareMatrix</a></td>
<th class="colLast" scope="row">
<div class="block">A sparse, square matrix, implementing using two arrays of sparse
  vectors, one representation for the rows and one for the columns.</div>
</th>
</tr>
<tr id="i916" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/math/SparseVector.html" title="class in org.biojava.nbio.structure.math">SparseVector</a></td>
<th class="colLast" scope="row">
<div class="block">A sparse vector, implemented using a symbol table.</div>
</th>
</tr>
<tr id="i917" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/geometry/SphereSampler.html" title="class in org.biojava.nbio.structure.symmetry.geometry">SphereSampler</a></td>
<th class="colLast" scope="row">
<div class="block">Sample possible orientations.</div>
</th>
</tr>
<tr id="i918" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/util/SplitFasta.html" title="class in org.biojava.nbio.genome.util">SplitFasta</a></td>
<th class="colLast" scope="row">
<div class="block">Utility to write each Fasta entry to a unique file</div>
</th>
</tr>
<tr id="i919" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/color/SqrtColorMapper.html" title="class in org.biojava.nbio.structure.gui.util.color">SqrtColorMapper</a></td>
<th class="colLast" scope="row">
<div class="block">Performs a sqrt transform on input before passing the values off to another
 colorMapper.</div>
</th>
</tr>
<tr id="i920" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/SSBondImpl.html" title="class in org.biojava.nbio.structure.io">SSBondImpl</a></td>
<th class="colLast" scope="row">
<div class="block">A simple bean to store disulfide bridge information, the SSBOND records in the PDB files.</div>
</th>
</tr>
<tr id="i921" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/cytoband/StainType.html" title="enum in org.biojava.nbio.genome.parsers.cytoband">StainType</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i922" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/StandardAminoAcid.html" title="class in org.biojava.nbio.structure">StandardAminoAcid</a></td>
<th class="colLast" scope="row">
<div class="block">A class that provides a set of standard amino acids.</div>
</th>
</tr>
<tr id="i923" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/StandardRescoreRefiner.html" title="class in org.biojava.nbio.alignment">StandardRescoreRefiner</a>&lt;<a href="org/biojava/nbio/alignment/StandardRescoreRefiner.html" title="type parameter in StandardRescoreRefiner">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/StandardRescoreRefiner.html" title="type parameter in StandardRescoreRefiner">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/StandardRescoreRefiner.html" title="type parameter in StandardRescoreRefiner">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i924" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/StartCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StartCodonSequence</a></td>
<th class="colLast" scope="row">
<div class="block">Used to map the start codon feature on a gene</div>
</th>
</tr>
<tr id="i925" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/StartupParameters.html" title="class in org.biojava.nbio.structure.align.ce">StartupParameters</a></td>
<th class="colLast" scope="row">
<div class="block">A simple bean that contains the parameters that can get set at startup</div>
</th>
</tr>
<tr id="i926" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/utils/StaticMemberPlaceHolder.html" title="class in org.biojava.nbio.ontology.utils">StaticMemberPlaceHolder</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i927" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/aligpanel/StatusDisplay.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">StatusDisplay</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i928" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/matrix/StdArrayIO.html" title="class in org.biojava.nbio.survival.cox.matrix">StdArrayIO</a></td>
<th class="colLast" scope="row">
<div class="block"><i>Standard array IO</i>.</div>
</th>
</tr>
<tr id="i929" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/io/StockholmFileAnnotation.html" title="class in org.biojava.nbio.alignment.io">StockholmFileAnnotation</a></td>
<th class="colLast" scope="row">
<div class="block">Stores all the content parsed from the #=GF lines</div>
</th>
</tr>
<tr id="i930" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/io/StockholmFileAnnotation.StockholmFileAnnotationReference.html" title="class in org.biojava.nbio.alignment.io">StockholmFileAnnotation.StockholmFileAnnotationReference</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i931" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/io/StockholmFileParser.html" title="class in org.biojava.nbio.alignment.io">StockholmFileParser</a></td>
<th class="colLast" scope="row">
<div class="block">Stockholm file parser.<br>
 for more information about the format refer to
 
 <a href="./org/biojava/nbio/alignment/io/ftp://ftp.sanger.ac.uk/pub/databases/Pfam/current_release/userman.txt">ftp://ftp.sanger.ac.uk/pub/databases
 /Pfam/current_release/userman.txt</a>.
 <a href="./org/biojava/nbio/alignment/io/ftp://ftp.sanger.ac.uk/pub/databases/Rfam/CURRENT/USERMAN">ftp://ftp.sanger.ac.uk/pub/databases/Rfam/CURRENT
 /USERMAN</a>.
 <a href="http://sonnhammer.sbc.su.se/Stockholm.html">http://sonnhammer.sbc.su.se/Stockholm.html</a>.
 </div>
</th>
</tr>
<tr id="i932" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/io/StockholmStructure.html" title="class in org.biojava.nbio.alignment.io">StockholmStructure</a></td>
<th class="colLast" scope="row">
<div class="block">Stores all the content of a Stockholm file.</div>
</th>
</tr>
<tr id="i933" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/io/StockholmStructure.DatabaseReference.html" title="class in org.biojava.nbio.alignment.io">StockholmStructure.DatabaseReference</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i934" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/Stoichiometry.html" title="class in org.biojava.nbio.structure.symmetry.core">Stoichiometry</a></td>
<th class="colLast" scope="row">
<div class="block">A utility object that describes Stoichiometry (composition of a protein assembly),
 determined via clustering procedure <a href="org/biojava/nbio/structure/cluster/SubunitClusterer.html" title="class in org.biojava.nbio.structure.cluster"><code>SubunitClusterer</code></a>,
 and implements human-readable representation using various strategies.</div>
</th>
</tr>
<tr id="i935" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/Stoichiometry.StringOverflowStrategy.html" title="enum in org.biojava.nbio.structure.symmetry.core">Stoichiometry.StringOverflowStrategy</a></td>
<th class="colLast" scope="row">
<div class="block">What to do when the number of <a href="org/biojava/nbio/structure/cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster"><code>SubunitCluster</code></a> exceeds the length of the alphabet.</div>
</th>
</tr>
<tr id="i936" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/StopCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StopCodonSequence</a></td>
<th class="colLast" scope="row">
<div class="block">Used to map the stop codon sequence on a gene</div>
</th>
</tr>
<tr id="i937" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a></td>
<th class="colLast" scope="row">
<div class="block">Provides a way of representing the strand of a sequence, location
 hit or feature.</div>
</th>
</tr>
<tr id="i938" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/StrataInfo.html" title="class in org.biojava.nbio.survival.cox">StrataInfo</a></td>
<th class="colLast" scope="row">
<div class="block">Information needed to represent a survival curve</div>
</th>
</tr>
<tr id="i939" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/pairwise/StrCompAlignment.html" title="class in org.biojava.nbio.structure.align.pairwise">StrCompAlignment</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i940" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/io/fastq/StreamListener.html" title="interface in org.biojava.nbio.genome.io.fastq">StreamListener</a></td>
<th class="colLast" scope="row">
<div class="block">Event based parser callback.</div>
</th>
</tr>
<tr id="i941" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/StringManipulationHelper.html" title="class in org.biojava.nbio.core.util">StringManipulationHelper</a></td>
<th class="colLast" scope="row">
<div class="block">A utility class for common <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink"><code>String</code></a> manipulation tasks.</div>
</th>
</tr>
<tr id="i942" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">StringProxySequenceReader</a>&lt;<a href="org/biojava/nbio/core/sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">An example of a ProxySequenceReader that is created from a String.</div>
</th>
</tr>
<tr id="i943" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a></td>
<th class="colLast" scope="row">
<div class="block">A class that contains all the parameters of the structure alignment algorithm.</div>
</th>
</tr>
<tr id="i944" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></td>
<th class="colLast" scope="row">
<div class="block">Interface for a structure object.</div>
</th>
</tr>
<tr id="i945" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/StructureAlignment.html" title="interface in org.biojava.nbio.structure.align">StructureAlignment</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i946" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html" title="class in org.biojava.nbio.structure.align.gui">StructureAlignmentDisplay</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i947" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/StructureAlignmentFactory.html" title="class in org.biojava.nbio.structure.align">StructureAlignmentFactory</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i948" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></td>
<th class="colLast" scope="row">
<div class="block">A class that provides a simple GUI for Jmol</div>
</th>
</tr>
<tr id="i949" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/fatcat/calc/StructureAlignmentOptimizer.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">StructureAlignmentOptimizer</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i950" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/structure/StructureAtom.html" title="class in org.biojava.nbio.protmod.structure">StructureAtom</a></td>
<th class="colLast" scope="row">
<div class="block">Everything that is needed to uniquely describe a atom.</div>
</th>
</tr>
<tr id="i951" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/structure/StructureAtomLinkage.html" title="class in org.biojava.nbio.protmod.structure">StructureAtomLinkage</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i952" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/io/StructureAtomXMLConverter.html" title="class in org.biojava.nbio.protmod.io">StructureAtomXMLConverter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i953" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></td>
<th class="colLast" scope="row">
<div class="block">An exception during the parsing of a PDB file.</div>
</th>
</tr>
<tr id="i954" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/StructureFiletype.html" title="enum in org.biojava.nbio.structure.io">StructureFiletype</a></td>
<th class="colLast" scope="row">
<div class="block">An enum of supported file formats.</div>
</th>
</tr>
<tr id="i955" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/structure/StructureGroup.html" title="class in org.biojava.nbio.protmod.structure">StructureGroup</a></td>
<th class="colLast" scope="row">
<div class="block">Information of a group (residue or ligand) involved in a modification.</div>
</th>
</tr>
<tr id="i956" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/io/StructureGroupXMLConverter.html" title="class in org.biojava.nbio.protmod.io">StructureGroupXMLConverter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i957" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a></td>
<th class="colLast" scope="row">
<div class="block">An identifier that <em>uniquely</em> identifies a whole <a href="org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> or
 arbitrary substructure.</div>
</th>
</tr>
<tr id="i958" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/StructureImpl.html" title="class in org.biojava.nbio.structure">StructureImpl</a></td>
<th class="colLast" scope="row">
<div class="block">Implementation of a PDB Structure.</div>
</th>
</tr>
<tr id="i959" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/StructureInterface.html" title="class in org.biojava.nbio.structure.contact">StructureInterface</a></td>
<th class="colLast" scope="row">
<div class="block">An interface between 2 molecules (2 sets of atoms).</div>
</th>
</tr>
<tr id="i960" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/StructureInterfaceCluster.html" title="class in org.biojava.nbio.structure.contact">StructureInterfaceCluster</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i961" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/contact/StructureInterfaceList.html" title="class in org.biojava.nbio.structure.contact">StructureInterfaceList</a></td>
<th class="colLast" scope="row">
<div class="block">A list of interfaces between 2 molecules (2 sets of atoms)</div>
</th>
</tr>
<tr id="i962" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/StructureIO.html" title="class in org.biojava.nbio.structure">StructureIO</a></td>
<th class="colLast" scope="row">
<div class="block">A class that provides static access methods for easy lookup of protein structure related components</div>
</th>
</tr>
<tr id="i963" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/StructureIOFile.html" title="interface in org.biojava.nbio.structure.io">StructureIOFile</a></td>
<th class="colLast" scope="row">
<div class="block">StructureIOFile extends StructureProvider with methods specific to
  parsing files from the filesystem.</div>
</th>
</tr>
<tr id="i964" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/StructureLoaderThread.html" title="class in org.biojava.nbio.structure.align.gui">StructureLoaderThread</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i965" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/client/StructureName.html" title="class in org.biojava.nbio.structure.align.client">StructureName</a></td>
<th class="colLast" scope="row">
<div class="block">A utility class that makes working with names of structures, domains and ranges easier.</div>
</th>
</tr>
<tr id="i966" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/client/StructureName.Source.html" title="enum in org.biojava.nbio.structure.align.client">StructureName.Source</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i967" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/StructurePairAligner.html" title="class in org.biojava.nbio.structure.align">StructurePairAligner</a></td>
<th class="colLast" scope="row">
<div class="block">Perform a pairwise protein structure superimposition.</div>
</th>
</tr>
<tr id="i968" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/util/StructurePairSelector.html" title="interface in org.biojava.nbio.structure.gui.util">StructurePairSelector</a></td>
<th class="colLast" scope="row">
<div class="block">To be implemented by JPanels that are part of the GUI to trigger structure aligmnents.</div>
</th>
</tr>
<tr id="i969" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/StructureProvider.html" title="interface in org.biojava.nbio.structure.io">StructureProvider</a></td>
<th class="colLast" scope="row">
<div class="block">A class that can provide a protein structure object from somewhere.</div>
</th>
</tr>
<tr id="i970" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/io/StructureSequenceMatcher.html" title="class in org.biojava.nbio.structure.io">StructureSequenceMatcher</a></td>
<th class="colLast" scope="row">
<div class="block">A utility class with methods for matching ProteinSequences with
 Structures.</div>
</th>
</tr>
<tr id="i971" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/StructureTools.html" title="class in org.biojava.nbio.structure">StructureTools</a></td>
<th class="colLast" scope="row">
<div class="block">A class that provides some tool methods.</div>
</th>
</tr>
<tr id="i972" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/protmod/structure/StructureUtil.html" title="class in org.biojava.nbio.protmod.structure">StructureUtil</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i973" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/StructureViewer.html" title="interface in org.biojava.nbio.structure.gui">StructureViewer</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i974" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="type parameter in SubstitutionMatrix">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Defines a data structure which holds the score (penalty or bonus) given during alignment for the exchange of one
 <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> in a sequence for another.</div>
</th>
</tr>
<tr id="i975" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html" title="class in org.biojava.nbio.core.alignment.matrices">SubstitutionMatrixHelper</a></td>
<th class="colLast" scope="row">
<div class="block">Static utility to access substitution matrices that come bundled with BioJava.</div>
</th>
</tr>
<tr id="i976" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/alignment/SubstitutionMatrixScorer.html" title="class in org.biojava.nbio.alignment">SubstitutionMatrixScorer</a>&lt;<a href="org/biojava/nbio/alignment/SubstitutionMatrixScorer.html" title="type parameter in SubstitutionMatrixScorer">S</a> extends <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="org/biojava/nbio/alignment/SubstitutionMatrixScorer.html" title="type parameter in SubstitutionMatrixScorer">C</a>&gt;,&#8203;<a href="org/biojava/nbio/alignment/SubstitutionMatrixScorer.html" title="type parameter in SubstitutionMatrixScorer">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Scores using a substitution matrix.</div>
</th>
</tr>
<tr id="i977" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/SubstructureIdentifier.html" title="class in org.biojava.nbio.structure">SubstructureIdentifier</a></td>
<th class="colLast" scope="row">
<div class="block">This is the canonical way to identify a part of a structure.</div>
</th>
</tr>
<tr id="i978" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster">Subunit</a></td>
<th class="colLast" scope="row">
<div class="block">A Subunit consists of a set of residues from a Structure, which may
 correspond to an entire Chain, a Domain, or any subset or combination of
 residues from them.</div>
</th>
</tr>
<tr id="i979" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster">SubunitCluster</a></td>
<th class="colLast" scope="row">
<div class="block">A SubunitCluster contains a set of equivalent <a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetrySubunits.html" title="class in org.biojava.nbio.structure.symmetry.core"><code>QuatSymmetrySubunits</code></a>,
 the set of equivalent residues (EQR) between <a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a> and a
 <a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a> representative.</div>
</th>
</tr>
<tr id="i980" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cluster/SubunitClusterer.html" title="class in org.biojava.nbio.structure.cluster">SubunitClusterer</a></td>
<th class="colLast" scope="row">
<div class="block">The SubunitClusterer takes as input a collection of <a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a> and
 returns a collection of <a href="org/biojava/nbio/structure/cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster"><code>SubunitCluster</code></a>.</div>
</th>
</tr>
<tr id="i981" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cluster/SubunitClustererMethod.html" title="enum in org.biojava.nbio.structure.cluster">SubunitClustererMethod</a></td>
<th class="colLast" scope="row">
<div class="block">The SubunitClustererMethod ennummerates all methods that can be used to
 cluster <a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a> in the <a href="org/biojava/nbio/structure/cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster"><code>SubunitCluster</code></a>.</div>
</th>
</tr>
<tr id="i982" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a></td>
<th class="colLast" scope="row">
<div class="block">The SubunitClustererParameters specifies the options used for the clustering
 of the subunits in structures using the <a href="org/biojava/nbio/structure/cluster/SubunitClusterer.html" title="class in org.biojava.nbio.structure.cluster"><code>SubunitClusterer</code></a>.</div>
</th>
</tr>
<tr id="i983" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/cluster/SubunitExtractor.html" title="class in org.biojava.nbio.structure.cluster">SubunitExtractor</a></td>
<th class="colLast" scope="row">
<div class="block">The SubunitExtractor extracts the information of each protein <a href="org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure"><code>Chain</code></a>
 in a <a href="org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> and converts them into a List of <a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a>.</div>
</th>
</tr>
<tr id="i984" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/geometry/SuperPosition.html" title="interface in org.biojava.nbio.structure.geometry">SuperPosition</a></td>
<th class="colLast" scope="row">
<div class="block">The SuperPosition interface defines and documents the required methods for
 any superpostion algorithm implementation, so that the input and expected
 output are uniform.</div>
</th>
</tr>
<tr id="i985" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/geometry/SuperPositionAbstract.html" title="class in org.biojava.nbio.structure.geometry">SuperPositionAbstract</a></td>
<th class="colLast" scope="row">
<div class="block">The SuperPositionAbstract contains common code shared by all SuperPosition
 algorithm implementations.</div>
</th>
</tr>
<tr id="i986" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/geometry/SuperPositionQCP.html" title="class in org.biojava.nbio.structure.geometry">SuperPositionQCP</a></td>
<th class="colLast" scope="row">
<div class="block">Implementation of the Quaternion-Based Characteristic Polynomial algorithm
 for RMSD and Superposition calculations.</div>
</th>
</tr>
<tr id="i987" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/geometry/SuperPositionQuat.html" title="class in org.biojava.nbio.structure.geometry">SuperPositionQuat</a></td>
<th class="colLast" scope="row">
<div class="block">The SuperPositionQuat implements a quaternion based algorithm to superpose
 arrays of Points in 3D.</div>
</th>
</tr>
<tr id="i988" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/geometry/SuperPositions.html" title="class in org.biojava.nbio.structure.geometry">SuperPositions</a></td>
<th class="colLast" scope="row">
<div class="block">SuperPositions is a Class that provides static helper methods and an easy
 access to the whole family of <a href="org/biojava/nbio/structure/geometry/SuperPosition.html" title="interface in org.biojava.nbio.structure.geometry"><code>SuperPosition</code></a> algorithms.</div>
</th>
</tr>
<tr id="i989" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/geometry/SuperPositionSVD.html" title="class in org.biojava.nbio.structure.geometry">SuperPositionSVD</a></td>
<th class="colLast" scope="row">
<div class="block">A class that calculates the superposition between two sets of points using an
 SVD Matrix Decomposition.</div>
</th>
</tr>
<tr id="i990" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/SurvFitInfo.html" title="class in org.biojava.nbio.survival.cox">SurvFitInfo</a></td>
<th class="colLast" scope="row">
<div class="block">Contains info for graphing km figures</div>
</th>
</tr>
<tr id="i991" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/figure/SurvFitKM.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">SurvFitKM</a></td>
<th class="colLast" scope="row">
<div class="block">Ported from survfitKM.S When combining multiple entries with same time not
 sure how the weighting adds up</div>
</th>
</tr>
<tr id="i992" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/figure/SurvFitKM.ConfLower.html" title="enum in org.biojava.nbio.survival.kaplanmeier.figure">SurvFitKM.ConfLower</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i993" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/figure/SurvFitKM.ConfType.html" title="enum in org.biojava.nbio.survival.kaplanmeier.figure">SurvFitKM.ConfType</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i994" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/figure/SurvFitKM.Error.html" title="enum in org.biojava.nbio.survival.kaplanmeier.figure">SurvFitKM.Error</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i995" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/kaplanmeier/figure/SurvFitKM.Method.html" title="enum in org.biojava.nbio.survival.kaplanmeier.figure">SurvFitKM.Method</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i996" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/SurvivalInfo.html" title="class in org.biojava.nbio.survival.cox">SurvivalInfo</a></td>
<th class="colLast" scope="row">
<div class="block">Data class to represent a single sample where time and event/censor status is required
 Additionally each variable and data associated with that variable.</div>
</th>
</tr>
<tr id="i997" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/comparators/SurvivalInfoComparator.html" title="class in org.biojava.nbio.survival.cox.comparators">SurvivalInfoComparator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i998" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/SurvivalInfoHelper.html" title="class in org.biojava.nbio.survival.cox">SurvivalInfoHelper</a></td>
<th class="colLast" scope="row">
<div class="block">Used to work with SurvivalInfo</div>
</th>
</tr>
<tr id="i999" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/SurvivalInfoIndex.html" title="class in org.biojava.nbio.survival.cox">SurvivalInfoIndex</a></td>
<th class="colLast" scope="row">
<div class="block">Not used and probably should be deleted</div>
</th>
</tr>
<tr id="i1000" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/comparators/SurvivalInfoValueComparator.html" title="class in org.biojava.nbio.survival.cox.comparators">SurvivalInfoValueComparator</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1001" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/math/SymbolTable.html" title="class in org.biojava.nbio.structure.math">SymbolTable</a>&lt;<a href="org/biojava/nbio/structure/math/SymbolTable.html" title="type parameter in SymbolTable">Key</a> extends <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Comparable.html?is-external=true" title="class or interface in java.lang" class="externalLink">Comparable</a>&lt;<a href="org/biojava/nbio/structure/math/SymbolTable.html" title="type parameter in SymbolTable">Key</a>&gt;,&#8203;<a href="org/biojava/nbio/structure/math/SymbolTable.html" title="type parameter in SymbolTable">Value</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Sorted symbol table implementation using a java.util.TreeMap.</div>
</th>
</tr>
<tr id="i1002" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/SymmetryAxes.html" title="class in org.biojava.nbio.structure.symmetry.internal">SymmetryAxes</a></td>
<th class="colLast" scope="row">
<div class="block">Data Structure that stores all the symmetry axis that describe
 the symmetry of a structure.</div>
</th>
</tr>
<tr id="i1003" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/SymmetryAxes.Axis.html" title="class in org.biojava.nbio.structure.symmetry.internal">SymmetryAxes.Axis</a></td>
<th class="colLast" scope="row">
<div class="block">Represents an axis of symmetry</div>
</th>
</tr>
<tr id="i1004" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/gui/SymmetryCalc.html" title="class in org.biojava.nbio.structure.symmetry.gui">SymmetryCalc</a></td>
<th class="colLast" scope="row">
<div class="block">Calculates a symmetry analysis and displays the results.</div>
</th>
</tr>
<tr id="i1005" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/gui/SymmetryDisplay.html" title="class in org.biojava.nbio.structure.symmetry.gui">SymmetryDisplay</a></td>
<th class="colLast" scope="row">
<div class="block">Class that provides visualizations methods for symmetry alignments.</div>
</th>
</tr>
<tr id="i1006" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/gui/SymmetryGui.html" title="class in org.biojava.nbio.structure.symmetry.gui">SymmetryGui</a></td>
<th class="colLast" scope="row">
<div class="block">A JFrame that allows to trigger a symmetry analysis, either from files
 in a directory or after manual upload
 Adapted from the AlignmentGui class in biojava.</div>
</th>
</tr>
<tr id="i1007" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/gui/SymmetryListener.html" title="class in org.biojava.nbio.structure.symmetry.gui">SymmetryListener</a></td>
<th class="colLast" scope="row">
<div class="block">Action Listener for the symmetry menu.</div>
</th>
</tr>
<tr id="i1008" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/SymmetryPerceptionMethod.html" title="enum in org.biojava.nbio.structure.symmetry.core">SymmetryPerceptionMethod</a></td>
<th class="colLast" scope="row">
<div class="block">Method used for symmetry perception in the <a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryDetector.html" title="class in org.biojava.nbio.structure.symmetry.core"><code>QuatSymmetryDetector</code></a>.</div>
</th>
</tr>
<tr id="i1009" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/SymmetryRefiner.html" title="interface in org.biojava.nbio.structure.symmetry.internal">SymmetryRefiner</a></td>
<th class="colLast" scope="row">
<div class="block">Interface for all symmetry refinement implementations.</div>
</th>
</tr>
<tr id="i1010" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html" title="class in org.biojava.nbio.structure.symmetry.utils">SymmetryTools</a></td>
<th class="colLast" scope="row">
<div class="block">Utility methods for symmetry (quaternary and internal) detection and result
 manipulation.</div>
</th>
</tr>
<tr id="i1011" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/internal/SymmOptimizer.html" title="class in org.biojava.nbio.structure.symmetry.internal">SymmOptimizer</a></td>
<th class="colLast" scope="row">
<div class="block">Optimizes a symmetry alignment by a Monte Carlo score optimization of the
 repeat multiple alignment.</div>
</th>
</tr>
<tr id="i1012" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/xtal/SymoplibParser.html" title="class in org.biojava.nbio.structure.xtal">SymoplibParser</a></td>
<th class="colLast" scope="row">
<div class="block">A class containing static methods to parse the symop.lib file from the
 CCP4 package.</div>
</th>
</tr>
<tr id="i1013" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/SynchronizedOutFile.html" title="class in org.biojava.nbio.structure.align.util">SynchronizedOutFile</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1014" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/Synonym.html" title="class in org.biojava.nbio.ontology">Synonym</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1015" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/core/SystematicSolver.html" title="class in org.biojava.nbio.structure.symmetry.core">SystematicSolver</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1016" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/gui/SystemInfo.html" title="class in org.biojava.nbio.structure.align.gui">SystemInfo</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1017" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/io/TabDelimParser.html" title="class in org.biojava.nbio.ontology.io">TabDelimParser</a></td>
<th class="colLast" scope="row">
<div class="block">Parse tab-delimited ontology files into Ontology objects.</div>
</th>
</tr>
<tr id="i1018" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/transcription/Table.html" title="interface in org.biojava.nbio.core.sequence.transcription">Table</a></td>
<th class="colLast" scope="row">
<div class="block">Provides a way of separating us from the specific <a href="org/biojava/nbio/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava.nbio.core.sequence.io"><code>IUPACParser.IUPACTable</code></a> even
 though this is the only implementing class for the interface.</div>
</th>
</tr>
<tr id="i1019" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava.nbio.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></td>
<th class="colLast" scope="row">
<div class="block">Class used to hold three nucleotides together and allow for equality
 to be assessed in a case insensitive manner.</div>
</th>
</tr>
<tr id="i1020" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/transcription/Table.Codon.html" title="class in org.biojava.nbio.core.sequence.transcription">Table.Codon</a></td>
<th class="colLast" scope="row">
<div class="block">Instance of a Codon which is 3 <a href="org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound"><code>NucleotideCompound</code></a>s, its
 corresponding <a href="org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound"><code>AminoAcidCompound</code></a> and if it is a start or stop codon.</div>
</th>
</tr>
<tr id="i1021" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/TaxonomyID.html" title="class in org.biojava.nbio.core.sequence">TaxonomyID</a></td>
<th class="colLast" scope="row">
<div class="block">A sequence can be associated with a species or Taxonomy ID</div>
</th>
</tr>
<tr id="i1022" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/Term.html" title="interface in org.biojava.nbio.ontology">Term</a></td>
<th class="colLast" scope="row">
<div class="block">A term in an ontology.</div>
</th>
</tr>
<tr id="i1023" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/Term.Impl.html" title="class in org.biojava.nbio.ontology">Term.Impl</a></td>
<th class="colLast" scope="row">
<div class="block">Simple in-memory implementation of an ontology term.</div>
</th>
</tr>
<tr id="i1024" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/basepairs/TertiaryBasePairParameters.html" title="class in org.biojava.nbio.structure.basepairs">TertiaryBasePairParameters</a></td>
<th class="colLast" scope="row">
<div class="block">This class also finds the base pairing and base-pair step parameters but has a broader definition
 of a base pair so that non-canonical-WC base pairs will be detected and reported.</div>
</th>
</tr>
<tr id="i1025" class="rowColor">
<td class="colFirst"><a href="demo/TestDNANeedlemanWunsch.html" title="class in demo">TestDNANeedlemanWunsch</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1026" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/symmetry/geometry/Tetrahedron.html" title="class in org.biojava.nbio.structure.symmetry.geometry">Tetrahedron</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1027" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/features/TextFeature.html" title="class in org.biojava.nbio.core.sequence.features">TextFeature</a>&lt;<a href="org/biojava/nbio/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">S</a> extends <a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">C</a>&gt;,&#8203;<a href="org/biojava/nbio/core/sequence/features/TextFeature.html" title="type parameter in TextFeature">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A implmentation of AbstractFeature</div>
</th>
</tr>
<tr id="i1028" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ronn/Timer.html" title="class in org.biojava.nbio.ronn">Timer</a></td>
<th class="colLast" scope="row">
<div class="block">A simple timer, calculates the time interval between two events.</div>
</th>
</tr>
<tr id="i1029" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava.nbio.core.sequence.transcription">TranscriptionEngine</a></td>
<th class="colLast" scope="row">
<div class="block">Used as a way of encapsulating the data structures required to parse DNA to a
 Protein sequence.</div>
</th>
</tr>
<tr id="i1030" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava.nbio.core.sequence.transcription">TranscriptionEngine.Builder</a></td>
<th class="colLast" scope="row">
<div class="block">This class is the way to create a <code>TranslationEngine</code>.</div>
</th>
</tr>
<tr id="i1031" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a></td>
<th class="colLast" scope="row">
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</th>
</tr>
<tr id="i1032" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/xtal/io/TransfAlgebraicAdapter.html" title="class in org.biojava.nbio.structure.xtal.io">TransfAlgebraicAdapter</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1033" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/xtal/TransformType.html" title="enum in org.biojava.nbio.structure.xtal">TransformType</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1034" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/exceptions/TranslationException.html" title="class in org.biojava.nbio.core.exceptions">TranslationException</a></td>
<th class="colLast" scope="row">
<div class="block">Thrown from AbstractCompundTranslator</div>
</th>
</tr>
<tr id="i1035" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/phylo/TreeConstructor.html" title="class in org.biojava.nbio.phylo">TreeConstructor</a></td>
<th class="colLast" scope="row">
<div class="block">The TreeConstructor uses the forester library to build different types of
 phylogenetic trees.</div>
</th>
</tr>
<tr id="i1036" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/phylo/TreeConstructorType.html" title="enum in org.biojava.nbio.phylo">TreeConstructorType</a></td>
<th class="colLast" scope="row">
<div class="block">The TreeConstructorType specifies the aligorithm used to construct the tree
 (clustering algorithm).</div>
</th>
</tr>
<tr id="i1037" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/xtal/io/TreeMapSpaceGroupWrapper.html" title="class in org.biojava.nbio.structure.xtal.io">TreeMapSpaceGroupWrapper</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1038" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/phylo/TreeType.html" title="enum in org.biojava.nbio.phylo">TreeType</a></td>
<th class="colLast" scope="row">
<div class="block">The TreeType specifies the optimization criteria used to generate the tree.</div>
</th>
</tr>
<tr id="i1039" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/Triple.html" title="interface in org.biojava.nbio.ontology">Triple</a></td>
<th class="colLast" scope="row">
<div class="block">A triple in an ontology.</div>
</th>
</tr>
<tr id="i1040" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/Triple.Impl.html" title="class in org.biojava.nbio.ontology">Triple.Impl</a></td>
<th class="colLast" scope="row">
<div class="block">Basic in-memory implementation of a Triple in an ontology

 This can be used to implement Ontology.createTriple</div>
</th>
</tr>
<tr id="i1041" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a></td>
<th class="colLast" scope="row">
<div class="block">A facade that makes it easier to work with a 2bit file.</div>
</th>
</tr>
<tr id="i1042" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/parsers/twobit/TwoBitParser.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitParser</a></td>
<th class="colLast" scope="row">
<div class="block">downloaded from http://storage.bioinf.fbb.msu.ru/~roman/TwoBitParser.java

 Class is a parser of UCSC Genome Browser file format .2bit used to store
 nucleotide sequence information.</div>
</th>
</tr>
<tr id="i1043" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">TwoBitSequenceReader</a>&lt;<a href="org/biojava/nbio/core/sequence/storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</a> extends <a href="org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Implementation of the 2bit encoding.</div>
</th>
</tr>
<tr id="i1044" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava.nbio.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker</a>&lt;<a href="org/biojava/nbio/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</a> extends <a href="org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Extension of the BitArrayWorker which provides the 2bit implementation
 code.</div>
</th>
</tr>
<tr id="i1045" class="rowColor">
<td class="colFirst"><a href="demo/UncompressFile.html" title="class in demo">UncompressFile</a></td>
<th class="colLast" scope="row">
<div class="block">Uncompresses a single tarred or zipped file, writing output to stdandard out</div>
</th>
</tr>
<tr id="i1046" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/UncompressInputStream.html" title="class in org.biojava.nbio.core.util">UncompressInputStream</a></td>
<th class="colLast" scope="row">
<div class="block">This class decompresses an input stream containing data compressed with
 the unix "compress" utility (LZC, a LZW variant).</div>
</th>
</tr>
<tr id="i1047" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">UniprotProxySequenceReader</a>&lt;<a href="org/biojava/nbio/core/sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">Pass in a Uniprot ID and this ProxySequenceReader when passed to a ProteinSequence will get the sequence data and other data elements
 associated with the ProteinSequence by Uniprot.</div>
</th>
</tr>
<tr id="i1048" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/genome/uniprot/UniprotToFasta.html" title="class in org.biojava.nbio.genome.uniprot">UniprotToFasta</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1049" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/xtal/UnitCellBoundingBox.html" title="class in org.biojava.nbio.structure.xtal">UnitCellBoundingBox</a></td>
<th class="colLast" scope="row">
<div class="block">A class to contain the BoundingBoxes of all polymeric molecules in a full unit cell.</div>
</th>
</tr>
<tr id="i1050" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/geometry/UnitQuaternions.html" title="class in org.biojava.nbio.structure.geometry">UnitQuaternions</a></td>
<th class="colLast" scope="row">
<div class="block">UnitQuaternions is a static Class that contains methods for calculating and
 using unit quaternions.</div>
</th>
</tr>
<tr id="i1051" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/URLConnectionTools.html" title="class in org.biojava.nbio.structure.align.util">URLConnectionTools</a></td>
<th class="colLast" scope="row">
<div class="block">A class that takes care about opening URLConnections and sets the proper timeouts</div>
</th>
</tr>
<tr id="i1052" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/URLIdentifier.html" title="class in org.biojava.nbio.structure">URLIdentifier</a></td>
<th class="colLast" scope="row">
<div class="block">Represents a structure loaded from a URL (including a file URL)

 A few custom query parameters are supported:

 
 <tt>format=[pdb|cif]</tt> Specify the file format (will otherwise be
     guessed from the extension)
 <tt>pdbId=[String]</tt> Specify the PDB ID (also guessed from the filename)
 <tt>chainID=[String]</tt> A single chain from the structure
 <tt>residues=[String]</tt> Residue ranges, in a form understood by
     <a href="org/biojava/nbio/structure/SubstructureIdentifier.html" title="class in org.biojava.nbio.structure"><code>SubstructureIdentifier</code></a>
 </div>
</th>
</tr>
<tr id="i1053" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/ce/UserArgumentProcessor.html" title="interface in org.biojava.nbio.structure.align.ce">UserArgumentProcessor</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1054" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/util/UserConfiguration.html" title="class in org.biojava.nbio.structure.align.util">UserConfiguration</a></td>
<th class="colLast" scope="row">
<div class="block">A container to persist config to the file system</div>
</th>
</tr>
<tr id="i1055" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/aaproperties/Utils.html" title="class in org.biojava.nbio.aaproperties">Utils</a></td>
<th class="colLast" scope="row">
<div class="block">This is a utility class that contains utility methods which will facilitates the coding of other methods</div>
</th>
</tr>
<tr id="i1056" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/Variable.html" title="interface in org.biojava.nbio.ontology">Variable</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1057" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/Variable.Impl.html" title="class in org.biojava.nbio.ontology">Variable.Impl</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1058" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/WaldTest.html" title="class in org.biojava.nbio.survival.cox">WaldTest</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1059" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/cox/WaldTestInfo.html" title="class in org.biojava.nbio.survival.cox">WaldTestInfo</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1060" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/ontology/utils/WeakValueHashMap.html" title="class in org.biojava.nbio.ontology.utils">WeakValueHashMap</a></td>
<th class="colLast" scope="row">
<div class="block">Map implementation which keeps weak references to values.</div>
</th>
</tr>
<tr id="i1061" class="rowColor">
<td class="colFirst"><a href="demo/WebStartClientDemo.html" title="class in demo">WebStartClientDemo</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1062" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/align/webstart/WebStartMain.html" title="class in org.biojava.nbio.structure.align.webstart">WebStartMain</a></td>
<th class="colLast" scope="row">&nbsp;</th>
</tr>
<tr id="i1063" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/sequence/views/WindowedSequence.html" title="class in org.biojava.nbio.core.sequence.views">WindowedSequence</a>&lt;<a href="org/biojava/nbio/core/sequence/views/WindowedSequence.html" title="type parameter in WindowedSequence">C</a> extends <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</td>
<th class="colLast" scope="row">
<div class="block">A sliding window view of a sequence which does not implement any
 interfaces like <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> because they do not fit how this works.</div>
</th>
</tr>
<tr id="i1064" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/survival/data/WorkSheet.html" title="class in org.biojava.nbio.survival.data">WorkSheet</a></td>
<th class="colLast" scope="row">
<div class="block">Need to handle very large spreadsheets of expression data so keep memory
 footprint low</div>
</th>
</tr>
<tr id="i1065" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/gui/WrapLayout.html" title="class in org.biojava.nbio.structure.gui">WrapLayout</a></td>
<th class="colLast" scope="row">
<div class="block">FlowLayout subclass that fully supports wrapping of components.</div>
</th>
</tr>
<tr id="i1066" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/XMLHelper.html" title="class in org.biojava.nbio.core.util">XMLHelper</a></td>
<th class="colLast" scope="row">
<div class="block">Helper methods to simplify boilerplate XML parsing code for  org.w3c.dom XML objects</div>
</th>
</tr>
<tr id="i1067" class="rowColor">
<td class="colFirst"><a href="org/biojava/nbio/core/util/XMLWriter.html" title="interface in org.biojava.nbio.core.util">XMLWriter</a></td>
<th class="colLast" scope="row">
<div class="block">Simple interface for building XML documents.</div>
</th>
</tr>
<tr id="i1068" class="altColor">
<td class="colFirst"><a href="org/biojava/nbio/structure/chem/ZipChemCompProvider.html" title="class in org.biojava.nbio.structure.chem">ZipChemCompProvider</a></td>
<th class="colLast" scope="row">
<div class="block">This chemical component provider retrieves and caches chemical component definition files from a
 zip archive specified in its construction.</div>
</th>
</tr>
</table>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li><a href="index-all.html">Index</a></li>
<li><a href="help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2022 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
