<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> * created at Apr 26, 2008</a>
<span class="sourceLineNo">021</span><a id="line.21"> */</a>
<span class="sourceLineNo">022</span><a id="line.22">package org.biojava.nbio.structure.io.mmcif;</a>
<span class="sourceLineNo">023</span><a id="line.23"></a>
<span class="sourceLineNo">024</span><a id="line.24">import java.text.ParseException;</a>
<span class="sourceLineNo">025</span><a id="line.25">import java.text.SimpleDateFormat;</a>
<span class="sourceLineNo">026</span><a id="line.26">import java.util.ArrayList;</a>
<span class="sourceLineNo">027</span><a id="line.27">import java.util.Date;</a>
<span class="sourceLineNo">028</span><a id="line.28">import java.util.HashMap;</a>
<span class="sourceLineNo">029</span><a id="line.29">import java.util.LinkedHashMap;</a>
<span class="sourceLineNo">030</span><a id="line.30">import java.util.List;</a>
<span class="sourceLineNo">031</span><a id="line.31">import java.util.Locale;</a>
<span class="sourceLineNo">032</span><a id="line.32">import java.util.Map;</a>
<span class="sourceLineNo">033</span><a id="line.33"></a>
<span class="sourceLineNo">034</span><a id="line.34">import javax.vecmath.Matrix4d;</a>
<span class="sourceLineNo">035</span><a id="line.35"></a>
<span class="sourceLineNo">036</span><a id="line.36">import org.biojava.nbio.structure.AminoAcid;</a>
<span class="sourceLineNo">037</span><a id="line.37">import org.biojava.nbio.structure.AminoAcidImpl;</a>
<span class="sourceLineNo">038</span><a id="line.38">import org.biojava.nbio.structure.Atom;</a>
<span class="sourceLineNo">039</span><a id="line.39">import org.biojava.nbio.structure.AtomImpl;</a>
<span class="sourceLineNo">040</span><a id="line.40">import org.biojava.nbio.structure.Chain;</a>
<span class="sourceLineNo">041</span><a id="line.41">import org.biojava.nbio.structure.ChainImpl;</a>
<span class="sourceLineNo">042</span><a id="line.42">import org.biojava.nbio.structure.EntityInfo;</a>
<span class="sourceLineNo">043</span><a id="line.43">import org.biojava.nbio.structure.EntityType;</a>
<span class="sourceLineNo">044</span><a id="line.44">import org.biojava.nbio.structure.DBRef;</a>
<span class="sourceLineNo">045</span><a id="line.45">import org.biojava.nbio.structure.Element;</a>
<span class="sourceLineNo">046</span><a id="line.46">import org.biojava.nbio.structure.Group;</a>
<span class="sourceLineNo">047</span><a id="line.47">import org.biojava.nbio.structure.GroupType;</a>
<span class="sourceLineNo">048</span><a id="line.48">import org.biojava.nbio.structure.HetatomImpl;</a>
<span class="sourceLineNo">049</span><a id="line.49">import org.biojava.nbio.structure.NucleotideImpl;</a>
<span class="sourceLineNo">050</span><a id="line.50">import org.biojava.nbio.structure.PDBCrystallographicInfo;</a>
<span class="sourceLineNo">051</span><a id="line.51">import org.biojava.nbio.structure.PDBHeader;</a>
<span class="sourceLineNo">052</span><a id="line.52">import org.biojava.nbio.structure.ResidueNumber;</a>
<span class="sourceLineNo">053</span><a id="line.53">import org.biojava.nbio.structure.SeqMisMatch;</a>
<span class="sourceLineNo">054</span><a id="line.54">import org.biojava.nbio.structure.SeqMisMatchImpl;</a>
<span class="sourceLineNo">055</span><a id="line.55">import org.biojava.nbio.structure.Site;</a>
<span class="sourceLineNo">056</span><a id="line.56">import org.biojava.nbio.structure.Structure;</a>
<span class="sourceLineNo">057</span><a id="line.57">import org.biojava.nbio.structure.StructureException;</a>
<span class="sourceLineNo">058</span><a id="line.58">import org.biojava.nbio.structure.StructureImpl;</a>
<span class="sourceLineNo">059</span><a id="line.59">import org.biojava.nbio.structure.StructureTools;</a>
<span class="sourceLineNo">060</span><a id="line.60">import org.biojava.nbio.structure.io.BondMaker;</a>
<span class="sourceLineNo">061</span><a id="line.61">import org.biojava.nbio.structure.io.ChargeAdder;</a>
<span class="sourceLineNo">062</span><a id="line.62">import org.biojava.nbio.structure.io.EntityFinder;</a>
<span class="sourceLineNo">063</span><a id="line.63">import org.biojava.nbio.structure.io.FileParsingParameters;</a>
<span class="sourceLineNo">064</span><a id="line.64">import org.biojava.nbio.structure.io.SeqRes2AtomAligner;</a>
<span class="sourceLineNo">065</span><a id="line.65">import org.biojava.nbio.structure.io.mmcif.model.AtomSite;</a>
<span class="sourceLineNo">066</span><a id="line.66">import org.biojava.nbio.structure.io.mmcif.model.AtomSites;</a>
<span class="sourceLineNo">067</span><a id="line.67">import org.biojava.nbio.structure.io.mmcif.model.AuditAuthor;</a>
<span class="sourceLineNo">068</span><a id="line.68">import org.biojava.nbio.structure.io.mmcif.model.Cell;</a>
<span class="sourceLineNo">069</span><a id="line.69">import org.biojava.nbio.structure.io.mmcif.model.ChemComp;</a>
<span class="sourceLineNo">070</span><a id="line.70">import org.biojava.nbio.structure.io.mmcif.model.ChemCompAtom;</a>
<span class="sourceLineNo">071</span><a id="line.71">import org.biojava.nbio.structure.io.mmcif.model.ChemCompBond;</a>
<span class="sourceLineNo">072</span><a id="line.72">import org.biojava.nbio.structure.io.mmcif.model.ChemCompDescriptor;</a>
<span class="sourceLineNo">073</span><a id="line.73">import org.biojava.nbio.structure.io.mmcif.model.DatabasePDBremark;</a>
<span class="sourceLineNo">074</span><a id="line.74">import org.biojava.nbio.structure.io.mmcif.model.DatabasePDBrev;</a>
<span class="sourceLineNo">075</span><a id="line.75">import org.biojava.nbio.structure.io.mmcif.model.DatabasePdbrevRecord;</a>
<span class="sourceLineNo">076</span><a id="line.76">import org.biojava.nbio.structure.io.mmcif.model.Entity;</a>
<span class="sourceLineNo">077</span><a id="line.77">import org.biojava.nbio.structure.io.mmcif.model.EntityPoly;</a>
<span class="sourceLineNo">078</span><a id="line.78">import org.biojava.nbio.structure.io.mmcif.model.EntityPolySeq;</a>
<span class="sourceLineNo">079</span><a id="line.79">import org.biojava.nbio.structure.io.mmcif.model.EntitySrcGen;</a>
<span class="sourceLineNo">080</span><a id="line.80">import org.biojava.nbio.structure.io.mmcif.model.EntitySrcNat;</a>
<span class="sourceLineNo">081</span><a id="line.81">import org.biojava.nbio.structure.io.mmcif.model.EntitySrcSyn;</a>
<span class="sourceLineNo">082</span><a id="line.82">import org.biojava.nbio.structure.io.mmcif.model.Exptl;</a>
<span class="sourceLineNo">083</span><a id="line.83">import org.biojava.nbio.structure.io.mmcif.model.PdbxAuditRevisionHistory;</a>
<span class="sourceLineNo">084</span><a id="line.84">import org.biojava.nbio.structure.io.mmcif.model.PdbxChemCompDescriptor;</a>
<span class="sourceLineNo">085</span><a id="line.85">import org.biojava.nbio.structure.io.mmcif.model.PdbxChemCompIdentifier;</a>
<span class="sourceLineNo">086</span><a id="line.86">import org.biojava.nbio.structure.io.mmcif.model.PdbxDatabaseStatus;</a>
<span class="sourceLineNo">087</span><a id="line.87">import org.biojava.nbio.structure.io.mmcif.model.PdbxEntityNonPoly;</a>
<span class="sourceLineNo">088</span><a id="line.88">import org.biojava.nbio.structure.io.mmcif.model.PdbxNonPolyScheme;</a>
<span class="sourceLineNo">089</span><a id="line.89">import org.biojava.nbio.structure.io.mmcif.model.PdbxPolySeqScheme;</a>
<span class="sourceLineNo">090</span><a id="line.90">import org.biojava.nbio.structure.io.mmcif.model.PdbxStructAssembly;</a>
<span class="sourceLineNo">091</span><a id="line.91">import org.biojava.nbio.structure.io.mmcif.model.PdbxStructAssemblyGen;</a>
<span class="sourceLineNo">092</span><a id="line.92">import org.biojava.nbio.structure.io.mmcif.model.PdbxStructOperList;</a>
<span class="sourceLineNo">093</span><a id="line.93">import org.biojava.nbio.structure.io.mmcif.model.Refine;</a>
<span class="sourceLineNo">094</span><a id="line.94">import org.biojava.nbio.structure.io.mmcif.model.Struct;</a>
<span class="sourceLineNo">095</span><a id="line.95">import org.biojava.nbio.structure.io.mmcif.model.StructAsym;</a>
<span class="sourceLineNo">096</span><a id="line.96">import org.biojava.nbio.structure.io.mmcif.model.StructConn;</a>
<span class="sourceLineNo">097</span><a id="line.97">import org.biojava.nbio.structure.io.mmcif.model.StructKeywords;</a>
<span class="sourceLineNo">098</span><a id="line.98">import org.biojava.nbio.structure.io.mmcif.model.StructNcsOper;</a>
<span class="sourceLineNo">099</span><a id="line.99">import org.biojava.nbio.structure.io.mmcif.model.StructRef;</a>
<span class="sourceLineNo">100</span><a id="line.100">import org.biojava.nbio.structure.io.mmcif.model.StructRefSeq;</a>
<span class="sourceLineNo">101</span><a id="line.101">import org.biojava.nbio.structure.io.mmcif.model.StructRefSeqDif;</a>
<span class="sourceLineNo">102</span><a id="line.102">import org.biojava.nbio.structure.io.mmcif.model.StructSite;</a>
<span class="sourceLineNo">103</span><a id="line.103">import org.biojava.nbio.structure.io.mmcif.model.StructSiteGen;</a>
<span class="sourceLineNo">104</span><a id="line.104">import org.biojava.nbio.structure.io.mmcif.model.Symmetry;</a>
<span class="sourceLineNo">105</span><a id="line.105">import org.biojava.nbio.structure.quaternary.BioAssemblyInfo;</a>
<span class="sourceLineNo">106</span><a id="line.106">import org.biojava.nbio.structure.quaternary.BiologicalAssemblyBuilder;</a>
<span class="sourceLineNo">107</span><a id="line.107">import org.biojava.nbio.structure.quaternary.BiologicalAssemblyTransformation;</a>
<span class="sourceLineNo">108</span><a id="line.108">import org.biojava.nbio.structure.xtal.CrystalCell;</a>
<span class="sourceLineNo">109</span><a id="line.109">import org.biojava.nbio.structure.xtal.SpaceGroup;</a>
<span class="sourceLineNo">110</span><a id="line.110">import org.biojava.nbio.structure.xtal.SymoplibParser;</a>
<span class="sourceLineNo">111</span><a id="line.111">import org.slf4j.Logger;</a>
<span class="sourceLineNo">112</span><a id="line.112">import org.slf4j.LoggerFactory;</a>
<span class="sourceLineNo">113</span><a id="line.113"></a>
<span class="sourceLineNo">114</span><a id="line.114">/**</a>
<span class="sourceLineNo">115</span><a id="line.115"> * A MMcifConsumer implementation that builds an in-memory representation of the</a>
<span class="sourceLineNo">116</span><a id="line.116"> * content of a mmcif file as a BioJava Structure object.</a>
<span class="sourceLineNo">117</span><a id="line.117"> *</a>
<span class="sourceLineNo">118</span><a id="line.118"> * @author Andreas Prlic</a>
<span class="sourceLineNo">119</span><a id="line.119"> * @since 1.7</a>
<span class="sourceLineNo">120</span><a id="line.120"> */</a>
<span class="sourceLineNo">121</span><a id="line.121"></a>
<span class="sourceLineNo">122</span><a id="line.122">public class SimpleMMcifConsumer implements MMcifConsumer {</a>
<span class="sourceLineNo">123</span><a id="line.123"></a>
<span class="sourceLineNo">124</span><a id="line.124">        private static final Logger logger = LoggerFactory.getLogger(SimpleMMcifConsumer.class);</a>
<span class="sourceLineNo">125</span><a id="line.125"></a>
<span class="sourceLineNo">126</span><a id="line.126">        private Structure structure;</a>
<span class="sourceLineNo">127</span><a id="line.127">        private Chain currentChain;</a>
<span class="sourceLineNo">128</span><a id="line.128">        private Group currentGroup;</a>
<span class="sourceLineNo">129</span><a id="line.129"></a>
<span class="sourceLineNo">130</span><a id="line.130">        /**</a>
<span class="sourceLineNo">131</span><a id="line.131">         * A temporary data structure to hold all parsed chains</a>
<span class="sourceLineNo">132</span><a id="line.132">         */</a>
<span class="sourceLineNo">133</span><a id="line.133">        private ArrayList&lt;List&lt;Chain&gt;&gt; allModels;</a>
<span class="sourceLineNo">134</span><a id="line.134">        /**</a>
<span class="sourceLineNo">135</span><a id="line.135">         * The current set of chains per model</a>
<span class="sourceLineNo">136</span><a id="line.136">         */</a>
<span class="sourceLineNo">137</span><a id="line.137">        private List&lt;Chain&gt;      currentModel;</a>
<span class="sourceLineNo">138</span><a id="line.138">        private List&lt;Entity&gt;     entities;</a>
<span class="sourceLineNo">139</span><a id="line.139">        /**</a>
<span class="sourceLineNo">140</span><a id="line.140">         * Needed in header only mode to get mapping between asym ids and author ids</a>
<span class="sourceLineNo">141</span><a id="line.141">         */</a>
<span class="sourceLineNo">142</span><a id="line.142">        private List&lt;EntityPoly&gt; entityPolys;</a>
<span class="sourceLineNo">143</span><a id="line.143">        private List&lt;StructRef&gt;  strucRefs;</a>
<span class="sourceLineNo">144</span><a id="line.144">        private List&lt;Chain&gt;      seqResChains;</a>
<span class="sourceLineNo">145</span><a id="line.145">        private List&lt;Chain&gt;      entityChains; // needed to link entities, chains and compounds...</a>
<span class="sourceLineNo">146</span><a id="line.146">        private List&lt;StructAsym&gt; structAsyms;  // needed to link entities, chains and compounds...</a>
<span class="sourceLineNo">147</span><a id="line.147">        private List&lt;PdbxStructOperList&gt; structOpers ; //</a>
<span class="sourceLineNo">148</span><a id="line.148">        private List&lt;PdbxStructAssembly&gt; strucAssemblies;</a>
<span class="sourceLineNo">149</span><a id="line.149">        private List&lt;PdbxStructAssemblyGen&gt; strucAssemblyGens;</a>
<span class="sourceLineNo">150</span><a id="line.150">        private List&lt;EntitySrcGen&gt; entitySrcGens;</a>
<span class="sourceLineNo">151</span><a id="line.151">        private List&lt;EntitySrcNat&gt; entitySrcNats;</a>
<span class="sourceLineNo">152</span><a id="line.152">        private List&lt;EntitySrcSyn&gt; entitySrcSyns;</a>
<span class="sourceLineNo">153</span><a id="line.153">        private List&lt;StructConn&gt; structConn;</a>
<span class="sourceLineNo">154</span><a id="line.154">        private List&lt;StructNcsOper&gt; structNcsOper;</a>
<span class="sourceLineNo">155</span><a id="line.155">        private List&lt;StructRefSeqDif&gt; sequenceDifs;</a>
<span class="sourceLineNo">156</span><a id="line.156">        private List&lt;StructSiteGen&gt; structSiteGens;</a>
<span class="sourceLineNo">157</span><a id="line.157"></a>
<span class="sourceLineNo">158</span><a id="line.158">        private Matrix4d parsedScaleMatrix;</a>
<span class="sourceLineNo">159</span><a id="line.159"></a>
<span class="sourceLineNo">160</span><a id="line.160"></a>
<span class="sourceLineNo">161</span><a id="line.161"></a>
<span class="sourceLineNo">162</span><a id="line.162">        /**</a>
<span class="sourceLineNo">163</span><a id="line.163">         * A map of asym ids (internal chain ids) to entity ids extracted from</a>
<span class="sourceLineNo">164</span><a id="line.164">         * the _struct_asym category</a>
<span class="sourceLineNo">165</span><a id="line.165">         */</a>
<span class="sourceLineNo">166</span><a id="line.166">        private Map&lt;String,String&gt; asymId2entityId;</a>
<span class="sourceLineNo">167</span><a id="line.167"></a>
<span class="sourceLineNo">168</span><a id="line.168">        /**</a>
<span class="sourceLineNo">169</span><a id="line.169">         * A map of asym ids (internal chain ids) to author ids extracted from</a>
<span class="sourceLineNo">170</span><a id="line.170">         * the _entity_poly category. Used in header only parsing.</a>
<span class="sourceLineNo">171</span><a id="line.171">         */</a>
<span class="sourceLineNo">172</span><a id="line.172">        private Map&lt;String,String&gt; asymId2authorId;</a>
<span class="sourceLineNo">173</span><a id="line.173"></a>
<span class="sourceLineNo">174</span><a id="line.174">        private String currentNmrModelNumber ;</a>
<span class="sourceLineNo">175</span><a id="line.175"></a>
<span class="sourceLineNo">176</span><a id="line.176">        private FileParsingParameters params;</a>
<span class="sourceLineNo">177</span><a id="line.177"></a>
<span class="sourceLineNo">178</span><a id="line.178">        public  SimpleMMcifConsumer(){</a>
<span class="sourceLineNo">179</span><a id="line.179">                params = new FileParsingParameters();</a>
<span class="sourceLineNo">180</span><a id="line.180">                documentStart();</a>
<span class="sourceLineNo">181</span><a id="line.181"></a>
<span class="sourceLineNo">182</span><a id="line.182">        }</a>
<span class="sourceLineNo">183</span><a id="line.183"></a>
<span class="sourceLineNo">184</span><a id="line.184">        @Override</a>
<span class="sourceLineNo">185</span><a id="line.185">        public void newEntity(Entity entity) {</a>
<span class="sourceLineNo">186</span><a id="line.186">                logger.debug("New entity: {}",entity.toString());</a>
<span class="sourceLineNo">187</span><a id="line.187">                entities.add(entity);</a>
<span class="sourceLineNo">188</span><a id="line.188">        }</a>
<span class="sourceLineNo">189</span><a id="line.189"></a>
<span class="sourceLineNo">190</span><a id="line.190">        @Override</a>
<span class="sourceLineNo">191</span><a id="line.191">        public void newEntityPoly(EntityPoly entityPoly) {</a>
<span class="sourceLineNo">192</span><a id="line.192">                entityPolys.add(entityPoly);</a>
<span class="sourceLineNo">193</span><a id="line.193">        }</a>
<span class="sourceLineNo">194</span><a id="line.194"></a>
<span class="sourceLineNo">195</span><a id="line.195">        @Override</a>
<span class="sourceLineNo">196</span><a id="line.196">        public void newPdbxStructOperList(PdbxStructOperList structOper){</a>
<span class="sourceLineNo">197</span><a id="line.197"></a>
<span class="sourceLineNo">198</span><a id="line.198">                structOpers.add(structOper);</a>
<span class="sourceLineNo">199</span><a id="line.199">        }</a>
<span class="sourceLineNo">200</span><a id="line.200"></a>
<span class="sourceLineNo">201</span><a id="line.201">        @Override</a>
<span class="sourceLineNo">202</span><a id="line.202">        public void newStructAsym(StructAsym sasym){</a>
<span class="sourceLineNo">203</span><a id="line.203"></a>
<span class="sourceLineNo">204</span><a id="line.204">                structAsyms.add(sasym);</a>
<span class="sourceLineNo">205</span><a id="line.205">        }</a>
<span class="sourceLineNo">206</span><a id="line.206"></a>
<span class="sourceLineNo">207</span><a id="line.207">        private Entity getEntity(int entity_id){</a>
<span class="sourceLineNo">208</span><a id="line.208">                try {</a>
<span class="sourceLineNo">209</span><a id="line.209">                        for (Entity e: entities){</a>
<span class="sourceLineNo">210</span><a id="line.210">                                int eId = Integer.parseInt(e.getId());</a>
<span class="sourceLineNo">211</span><a id="line.211">                                if  (eId== entity_id){</a>
<span class="sourceLineNo">212</span><a id="line.212">                                        return e;</a>
<span class="sourceLineNo">213</span><a id="line.213">                                }</a>
<span class="sourceLineNo">214</span><a id="line.214">                        }</a>
<span class="sourceLineNo">215</span><a id="line.215">                } catch (NumberFormatException e) {</a>
<span class="sourceLineNo">216</span><a id="line.216">                        logger.warn("Entity id does not look like a number:", e.getMessage());</a>
<span class="sourceLineNo">217</span><a id="line.217">                }</a>
<span class="sourceLineNo">218</span><a id="line.218">                return null;</a>
<span class="sourceLineNo">219</span><a id="line.219">        }</a>
<span class="sourceLineNo">220</span><a id="line.220"></a>
<span class="sourceLineNo">221</span><a id="line.221">        @Override</a>
<span class="sourceLineNo">222</span><a id="line.222">        public void newStructKeywords(StructKeywords kw){</a>
<span class="sourceLineNo">223</span><a id="line.223">                PDBHeader header = structure.getPDBHeader();</a>
<span class="sourceLineNo">224</span><a id="line.224">                if ( header == null)</a>
<span class="sourceLineNo">225</span><a id="line.225">                        header = new PDBHeader();</a>
<span class="sourceLineNo">226</span><a id="line.226">                header.setDescription(kw.getPdbx_keywords());</a>
<span class="sourceLineNo">227</span><a id="line.227">                header.setClassification(kw.getPdbx_keywords());</a>
<span class="sourceLineNo">228</span><a id="line.228">        }</a>
<span class="sourceLineNo">229</span><a id="line.229"></a>
<span class="sourceLineNo">230</span><a id="line.230">        @Override</a>
<span class="sourceLineNo">231</span><a id="line.231">        public void setStruct(Struct struct) {</a>
<span class="sourceLineNo">232</span><a id="line.232"></a>
<span class="sourceLineNo">233</span><a id="line.233">                PDBHeader header = structure.getPDBHeader();</a>
<span class="sourceLineNo">234</span><a id="line.234">                if ( header == null)</a>
<span class="sourceLineNo">235</span><a id="line.235">                        header = new PDBHeader();</a>
<span class="sourceLineNo">236</span><a id="line.236"></a>
<span class="sourceLineNo">237</span><a id="line.237">                header.setTitle(struct.getTitle());</a>
<span class="sourceLineNo">238</span><a id="line.238">                header.setIdCode(struct.getEntry_id());</a>
<span class="sourceLineNo">239</span><a id="line.239">                //header.setDescription(struct.getPdbx_descriptor());</a>
<span class="sourceLineNo">240</span><a id="line.240">                //header.setClassification(struct.getPdbx_descriptor());</a>
<span class="sourceLineNo">241</span><a id="line.241">                //header.setDescription(struct.getPdbx_descriptor());</a>
<span class="sourceLineNo">242</span><a id="line.242"></a>
<span class="sourceLineNo">243</span><a id="line.243"></a>
<span class="sourceLineNo">244</span><a id="line.244"></a>
<span class="sourceLineNo">245</span><a id="line.245">                structure.setPDBHeader(header);</a>
<span class="sourceLineNo">246</span><a id="line.246">                structure.setPDBCode(struct.getEntry_id());</a>
<span class="sourceLineNo">247</span><a id="line.247">        }</a>
<span class="sourceLineNo">248</span><a id="line.248"></a>
<span class="sourceLineNo">249</span><a id="line.249">        /** initiate new group, either Hetatom, Nucleotide, or AminoAcid */</a>
<span class="sourceLineNo">250</span><a id="line.250">        private Group getNewGroup(String recordName,Character aminoCode1, long seq_id,String groupCode3) {</a>
<span class="sourceLineNo">251</span><a id="line.251"></a>
<span class="sourceLineNo">252</span><a id="line.252">                Group g = ChemCompGroupFactory.getGroupFromChemCompDictionary(groupCode3);</a>
<span class="sourceLineNo">253</span><a id="line.253">                if ( g != null &amp;&amp; !g.getChemComp().isEmpty()) {</a>
<span class="sourceLineNo">254</span><a id="line.254">                        if ( g instanceof AminoAcidImpl) {</a>
<span class="sourceLineNo">255</span><a id="line.255">                                AminoAcidImpl aa = (AminoAcidImpl) g;</a>
<span class="sourceLineNo">256</span><a id="line.256">                                aa.setId(seq_id);</a>
<span class="sourceLineNo">257</span><a id="line.257">                        } else if ( g instanceof NucleotideImpl) {</a>
<span class="sourceLineNo">258</span><a id="line.258">                                NucleotideImpl nuc =  (NucleotideImpl) g;</a>
<span class="sourceLineNo">259</span><a id="line.259">                                nuc.setId(seq_id);</a>
<span class="sourceLineNo">260</span><a id="line.260">                        } else if ( g instanceof HetatomImpl) {</a>
<span class="sourceLineNo">261</span><a id="line.261">                                HetatomImpl het = (HetatomImpl)g;</a>
<span class="sourceLineNo">262</span><a id="line.262">                                het.setId(seq_id);</a>
<span class="sourceLineNo">263</span><a id="line.263">                        }</a>
<span class="sourceLineNo">264</span><a id="line.264">                        return g;</a>
<span class="sourceLineNo">265</span><a id="line.265">                }</a>
<span class="sourceLineNo">266</span><a id="line.266"></a>
<span class="sourceLineNo">267</span><a id="line.267"></a>
<span class="sourceLineNo">268</span><a id="line.268"></a>
<span class="sourceLineNo">269</span><a id="line.269">                Group group;</a>
<span class="sourceLineNo">270</span><a id="line.270">                if ( recordName.equals("ATOM") ) {</a>
<span class="sourceLineNo">271</span><a id="line.271">                        if (StructureTools.isNucleotide(groupCode3))  {</a>
<span class="sourceLineNo">272</span><a id="line.272">                                // it is a nucleotide</a>
<span class="sourceLineNo">273</span><a id="line.273">                                NucleotideImpl nu = new NucleotideImpl();</a>
<span class="sourceLineNo">274</span><a id="line.274">                                group = nu;</a>
<span class="sourceLineNo">275</span><a id="line.275">                                nu.setId(seq_id);</a>
<span class="sourceLineNo">276</span><a id="line.276"></a>
<span class="sourceLineNo">277</span><a id="line.277">                        } else if (aminoCode1==null || aminoCode1 == StructureTools.UNKNOWN_GROUP_LABEL){</a>
<span class="sourceLineNo">278</span><a id="line.278">                                HetatomImpl h = new HetatomImpl();</a>
<span class="sourceLineNo">279</span><a id="line.279">                                h.setId(seq_id);</a>
<span class="sourceLineNo">280</span><a id="line.280">                                group = h;</a>
<span class="sourceLineNo">281</span><a id="line.281"></a>
<span class="sourceLineNo">282</span><a id="line.282">                        } else {</a>
<span class="sourceLineNo">283</span><a id="line.283">                                AminoAcidImpl aa = new AminoAcidImpl() ;</a>
<span class="sourceLineNo">284</span><a id="line.284">                                aa.setAminoType(aminoCode1);</a>
<span class="sourceLineNo">285</span><a id="line.285">                                aa.setId(seq_id);</a>
<span class="sourceLineNo">286</span><a id="line.286">                                group = aa ;</a>
<span class="sourceLineNo">287</span><a id="line.287">                        }</a>
<span class="sourceLineNo">288</span><a id="line.288">                }</a>
<span class="sourceLineNo">289</span><a id="line.289">                else {</a>
<span class="sourceLineNo">290</span><a id="line.290">                        if (StructureTools.isNucleotide(groupCode3))  {</a>
<span class="sourceLineNo">291</span><a id="line.291">                                // it is a nucleotide</a>
<span class="sourceLineNo">292</span><a id="line.292">                                NucleotideImpl nu = new NucleotideImpl();</a>
<span class="sourceLineNo">293</span><a id="line.293">                                group = nu;</a>
<span class="sourceLineNo">294</span><a id="line.294">                                nu.setId(seq_id);</a>
<span class="sourceLineNo">295</span><a id="line.295">                        }</a>
<span class="sourceLineNo">296</span><a id="line.296">                        else if (aminoCode1 != null ) {</a>
<span class="sourceLineNo">297</span><a id="line.297">                                AminoAcidImpl aa = new AminoAcidImpl() ;</a>
<span class="sourceLineNo">298</span><a id="line.298">                                aa.setAminoType(aminoCode1);</a>
<span class="sourceLineNo">299</span><a id="line.299">                                aa.setId(seq_id);</a>
<span class="sourceLineNo">300</span><a id="line.300">                                group = aa ;</a>
<span class="sourceLineNo">301</span><a id="line.301">                        } else {</a>
<span class="sourceLineNo">302</span><a id="line.302">                                HetatomImpl h = new HetatomImpl();</a>
<span class="sourceLineNo">303</span><a id="line.303">                                h.setId(seq_id);</a>
<span class="sourceLineNo">304</span><a id="line.304">                                group = h;</a>
<span class="sourceLineNo">305</span><a id="line.305">                        }</a>
<span class="sourceLineNo">306</span><a id="line.306">                }</a>
<span class="sourceLineNo">307</span><a id="line.307">                return  group ;</a>
<span class="sourceLineNo">308</span><a id="line.308">        }</a>
<span class="sourceLineNo">309</span><a id="line.309"></a>
<span class="sourceLineNo">310</span><a id="line.310">        /**</a>
<span class="sourceLineNo">311</span><a id="line.311">         * Test if the given asymId is already present in the list of chains given. If yes, returns the chain</a>
<span class="sourceLineNo">312</span><a id="line.312">         * otherwise returns null.</a>
<span class="sourceLineNo">313</span><a id="line.313">         */</a>
<span class="sourceLineNo">314</span><a id="line.314">        private static Chain isKnownChain(String asymId, List&lt;Chain&gt; chains){</a>
<span class="sourceLineNo">315</span><a id="line.315"></a>
<span class="sourceLineNo">316</span><a id="line.316">                for (int i = 0; i&lt; chains.size();i++){</a>
<span class="sourceLineNo">317</span><a id="line.317">                        Chain testchain =  chains.get(i);</a>
<span class="sourceLineNo">318</span><a id="line.318">                        //System.out.println("comparing chainID &gt;"+chainID+"&lt; against testchain " + i+" &gt;" +testchain.getName()+"&lt;");</a>
<span class="sourceLineNo">319</span><a id="line.319">                        if (asymId.equals(testchain.getId())) {</a>
<span class="sourceLineNo">320</span><a id="line.320">                                //System.out.println("chain "+ chainID+" already known ...");</a>
<span class="sourceLineNo">321</span><a id="line.321">                                return testchain;</a>
<span class="sourceLineNo">322</span><a id="line.322">                        }</a>
<span class="sourceLineNo">323</span><a id="line.323">                }</a>
<span class="sourceLineNo">324</span><a id="line.324"></a>
<span class="sourceLineNo">325</span><a id="line.325">                return null;</a>
<span class="sourceLineNo">326</span><a id="line.326">        }</a>
<span class="sourceLineNo">327</span><a id="line.327"></a>
<span class="sourceLineNo">328</span><a id="line.328">        @Override</a>
<span class="sourceLineNo">329</span><a id="line.329">        public void newAtomSite(AtomSite atom) {</a>
<span class="sourceLineNo">330</span><a id="line.330"></a>
<span class="sourceLineNo">331</span><a id="line.331">                if (params.isHeaderOnly()) return;</a>
<span class="sourceLineNo">332</span><a id="line.332"></a>
<span class="sourceLineNo">333</span><a id="line.333">                // Warning: getLabel_asym_id is not the "chain id" in the PDB file</a>
<span class="sourceLineNo">334</span><a id="line.334">                // it is the internally used chain id.</a>
<span class="sourceLineNo">335</span><a id="line.335">                // later on we will fix this...</a>
<span class="sourceLineNo">336</span><a id="line.336"></a>
<span class="sourceLineNo">337</span><a id="line.337">                // later one needs to map the asym id to the pdb_strand_id</a>
<span class="sourceLineNo">338</span><a id="line.338"></a>
<span class="sourceLineNo">339</span><a id="line.339">                //TODO: add support for FileParsingParams.getMaxAtoms()</a>
<span class="sourceLineNo">340</span><a id="line.340"></a>
<span class="sourceLineNo">341</span><a id="line.341">                boolean startOfNewChain = false;</a>
<span class="sourceLineNo">342</span><a id="line.342"></a>
<span class="sourceLineNo">343</span><a id="line.343">                String asymId = atom.getLabel_asym_id();</a>
<span class="sourceLineNo">344</span><a id="line.344">                String authId = atom.getAuth_asym_id();</a>
<span class="sourceLineNo">345</span><a id="line.345"></a>
<span class="sourceLineNo">346</span><a id="line.346">                String recordName    = atom.getGroup_PDB();</a>
<span class="sourceLineNo">347</span><a id="line.347">                String residueNumberS = atom.getAuth_seq_id();</a>
<span class="sourceLineNo">348</span><a id="line.348">                Integer residueNrInt = Integer.parseInt(residueNumberS);</a>
<span class="sourceLineNo">349</span><a id="line.349"></a>
<span class="sourceLineNo">350</span><a id="line.350">                // the 3-letter name of the group:</a>
<span class="sourceLineNo">351</span><a id="line.351">                String groupCode3    = atom.getLabel_comp_id();</a>
<span class="sourceLineNo">352</span><a id="line.352"></a>
<span class="sourceLineNo">353</span><a id="line.353">                boolean isHetAtomInFile = false;</a>
<span class="sourceLineNo">354</span><a id="line.354"></a>
<span class="sourceLineNo">355</span><a id="line.355">                Character aminoCode1 = null;</a>
<span class="sourceLineNo">356</span><a id="line.356">                if ( recordName.equals("ATOM") )</a>
<span class="sourceLineNo">357</span><a id="line.357">                        aminoCode1 = StructureTools.get1LetterCodeAmino(groupCode3);</a>
<span class="sourceLineNo">358</span><a id="line.358">                else {</a>
<span class="sourceLineNo">359</span><a id="line.359">                        aminoCode1 = StructureTools.get1LetterCodeAmino(groupCode3);</a>
<span class="sourceLineNo">360</span><a id="line.360"></a>
<span class="sourceLineNo">361</span><a id="line.361">                        // for nucleotides this will be null..</a>
<span class="sourceLineNo">362</span><a id="line.362">                        if (aminoCode1 != null &amp;&amp;  aminoCode1.equals(StructureTools.UNKNOWN_GROUP_LABEL))</a>
<span class="sourceLineNo">363</span><a id="line.363">                                aminoCode1 = null;</a>
<span class="sourceLineNo">364</span><a id="line.364"></a>
<span class="sourceLineNo">365</span><a id="line.365">                        isHetAtomInFile = true;</a>
<span class="sourceLineNo">366</span><a id="line.366">                }</a>
<span class="sourceLineNo">367</span><a id="line.367">                String insCodeS = atom.getPdbx_PDB_ins_code();</a>
<span class="sourceLineNo">368</span><a id="line.368">                Character insCode = null;</a>
<span class="sourceLineNo">369</span><a id="line.369">                if (!  insCodeS.equals("?")) {</a>
<span class="sourceLineNo">370</span><a id="line.370">                        insCode = insCodeS.charAt(0);</a>
<span class="sourceLineNo">371</span><a id="line.371">                }</a>
<span class="sourceLineNo">372</span><a id="line.372">                // we store the internal seq id in the Atom._id field</a>
<span class="sourceLineNo">373</span><a id="line.373">                // this is not a PDB file field but we need this to internally assign the insertion codes later</a>
<span class="sourceLineNo">374</span><a id="line.374">                // from the pdbx_poly_seq entries..</a>
<span class="sourceLineNo">375</span><a id="line.375"></a>
<span class="sourceLineNo">376</span><a id="line.376">                long seq_id = -1;</a>
<span class="sourceLineNo">377</span><a id="line.377">                try {</a>
<span class="sourceLineNo">378</span><a id="line.378">                        seq_id = Long.parseLong(atom.getLabel_seq_id());</a>
<span class="sourceLineNo">379</span><a id="line.379">                } catch (NumberFormatException e){</a>
<span class="sourceLineNo">380</span><a id="line.380">                        // non polymer chains (ligands and small molecules) will have a label_seq_id set to '.', thus it is ok to</a>
<span class="sourceLineNo">381</span><a id="line.381">                        // silently ignore this</a>
<span class="sourceLineNo">382</span><a id="line.382">                        //logger.debug("Could not parse number for _atom_site.label_seq_id: "+e.getMessage());</a>
<span class="sourceLineNo">383</span><a id="line.383">                }</a>
<span class="sourceLineNo">384</span><a id="line.384"></a>
<span class="sourceLineNo">385</span><a id="line.385">                String nmrModelNumber = atom.getPdbx_PDB_model_num();</a>
<span class="sourceLineNo">386</span><a id="line.386"></a>
<span class="sourceLineNo">387</span><a id="line.387">                if ( currentNmrModelNumber == null) {</a>
<span class="sourceLineNo">388</span><a id="line.388">                        currentNmrModelNumber = nmrModelNumber;</a>
<span class="sourceLineNo">389</span><a id="line.389">                }</a>
<span class="sourceLineNo">390</span><a id="line.390"></a>
<span class="sourceLineNo">391</span><a id="line.391">                if (! currentNmrModelNumber.equals(nmrModelNumber)){</a>
<span class="sourceLineNo">392</span><a id="line.392">                        currentNmrModelNumber = nmrModelNumber;</a>
<span class="sourceLineNo">393</span><a id="line.393"></a>
<span class="sourceLineNo">394</span><a id="line.394">                        // add previous data</a>
<span class="sourceLineNo">395</span><a id="line.395">                        if ( currentChain != null ) {</a>
<span class="sourceLineNo">396</span><a id="line.396">                                currentChain.addGroup(currentGroup);</a>
<span class="sourceLineNo">397</span><a id="line.397">                                currentGroup.trimToSize();</a>
<span class="sourceLineNo">398</span><a id="line.398">                        }</a>
<span class="sourceLineNo">399</span><a id="line.399"></a>
<span class="sourceLineNo">400</span><a id="line.400">                        // we came to the beginning of a new NMR model</a>
<span class="sourceLineNo">401</span><a id="line.401">                        allModels.add(currentModel);</a>
<span class="sourceLineNo">402</span><a id="line.402">                        currentModel = new ArrayList&lt;Chain&gt;();</a>
<span class="sourceLineNo">403</span><a id="line.403">                        currentChain = null;</a>
<span class="sourceLineNo">404</span><a id="line.404">                        currentGroup = null;</a>
<span class="sourceLineNo">405</span><a id="line.405">                }</a>
<span class="sourceLineNo">406</span><a id="line.406"></a>
<span class="sourceLineNo">407</span><a id="line.407"></a>
<span class="sourceLineNo">408</span><a id="line.408">                if (currentChain == null) {</a>
<span class="sourceLineNo">409</span><a id="line.409"></a>
<span class="sourceLineNo">410</span><a id="line.410">                        currentChain = new ChainImpl();</a>
<span class="sourceLineNo">411</span><a id="line.411">                        currentChain.setName(authId);</a>
<span class="sourceLineNo">412</span><a id="line.412">                        currentChain.setId(asymId);</a>
<span class="sourceLineNo">413</span><a id="line.413">                        currentModel.add(currentChain);</a>
<span class="sourceLineNo">414</span><a id="line.414">                        startOfNewChain = true;</a>
<span class="sourceLineNo">415</span><a id="line.415">                }</a>
<span class="sourceLineNo">416</span><a id="line.416"></a>
<span class="sourceLineNo">417</span><a id="line.417">                //System.out.println("BEFORE: " + chain_id + " " + current_chain.getName());</a>
<span class="sourceLineNo">418</span><a id="line.418">                if ( ! asymId.equals(currentChain.getId()) ) {</a>
<span class="sourceLineNo">419</span><a id="line.419">                        //logger.info("unknown chain. creating new chain. authId:" + authId + " asymId: " + asymId);</a>
<span class="sourceLineNo">420</span><a id="line.420">                        startOfNewChain = true;</a>
<span class="sourceLineNo">421</span><a id="line.421"></a>
<span class="sourceLineNo">422</span><a id="line.422">                        // end up old chain...</a>
<span class="sourceLineNo">423</span><a id="line.423">                        currentChain.addGroup(currentGroup);</a>
<span class="sourceLineNo">424</span><a id="line.424"></a>
<span class="sourceLineNo">425</span><a id="line.425">                        // see if old chain is known ...</a>
<span class="sourceLineNo">426</span><a id="line.426">                        Chain testchain = isKnownChain(asymId,currentModel);</a>
<span class="sourceLineNo">427</span><a id="line.427"></a>
<span class="sourceLineNo">428</span><a id="line.428">                        if ( testchain == null) {</a>
<span class="sourceLineNo">429</span><a id="line.429">                                //logger.info("unknown chain. creating new chain. authId:" + authId + " asymId: " + asymId);</a>
<span class="sourceLineNo">430</span><a id="line.430"></a>
<span class="sourceLineNo">431</span><a id="line.431">                                currentChain = new ChainImpl();</a>
<span class="sourceLineNo">432</span><a id="line.432">                                currentChain.setName(authId);</a>
<span class="sourceLineNo">433</span><a id="line.433">                                currentChain.setId(asymId);</a>
<span class="sourceLineNo">434</span><a id="line.434"></a>
<span class="sourceLineNo">435</span><a id="line.435">                        }   else {</a>
<span class="sourceLineNo">436</span><a id="line.436">                                currentChain = testchain;</a>
<span class="sourceLineNo">437</span><a id="line.437">                        }</a>
<span class="sourceLineNo">438</span><a id="line.438"></a>
<span class="sourceLineNo">439</span><a id="line.439">                        if ( ! currentModel.contains(currentChain))</a>
<span class="sourceLineNo">440</span><a id="line.440">                                currentModel.add(currentChain);</a>
<span class="sourceLineNo">441</span><a id="line.441"></a>
<span class="sourceLineNo">442</span><a id="line.442">                }</a>
<span class="sourceLineNo">443</span><a id="line.443"></a>
<span class="sourceLineNo">444</span><a id="line.444"></a>
<span class="sourceLineNo">445</span><a id="line.445">                ResidueNumber residueNumber = new ResidueNumber(authId,residueNrInt, insCode);</a>
<span class="sourceLineNo">446</span><a id="line.446"></a>
<span class="sourceLineNo">447</span><a id="line.447">                if (currentGroup == null) {</a>
<span class="sourceLineNo">448</span><a id="line.448"></a>
<span class="sourceLineNo">449</span><a id="line.449"></a>
<span class="sourceLineNo">450</span><a id="line.450">                        currentGroup = getNewGroup(recordName,aminoCode1,seq_id, groupCode3);</a>
<span class="sourceLineNo">451</span><a id="line.451"></a>
<span class="sourceLineNo">452</span><a id="line.452">                        currentGroup.setResidueNumber(residueNumber);</a>
<span class="sourceLineNo">453</span><a id="line.453">                        currentGroup.setPDBName(groupCode3);</a>
<span class="sourceLineNo">454</span><a id="line.454">                        currentGroup.setHetAtomInFile(isHetAtomInFile);</a>
<span class="sourceLineNo">455</span><a id="line.455">                }</a>
<span class="sourceLineNo">456</span><a id="line.456"></a>
<span class="sourceLineNo">457</span><a id="line.457">                // SET UP THE ALT LOC GROUP</a>
<span class="sourceLineNo">458</span><a id="line.458">                Group altGroup = null;</a>
<span class="sourceLineNo">459</span><a id="line.459">                String altLocS = atom.getLabel_alt_id();</a>
<span class="sourceLineNo">460</span><a id="line.460">                Character altLoc = ' ';</a>
<span class="sourceLineNo">461</span><a id="line.461">                if ( altLocS.length()&gt;0) {</a>
<span class="sourceLineNo">462</span><a id="line.462">                        altLoc = altLocS.charAt(0);</a>
<span class="sourceLineNo">463</span><a id="line.463">                        if ( altLoc.equals('.') )</a>
<span class="sourceLineNo">464</span><a id="line.464">                                altLoc = ' ';</a>
<span class="sourceLineNo">465</span><a id="line.465"></a>
<span class="sourceLineNo">466</span><a id="line.466">                }</a>
<span class="sourceLineNo">467</span><a id="line.467">                // If it's the start of the new chain</a>
<span class="sourceLineNo">468</span><a id="line.468">                if ( startOfNewChain){</a>
<span class="sourceLineNo">469</span><a id="line.469">                        currentGroup = getNewGroup(recordName,aminoCode1,seq_id, groupCode3);</a>
<span class="sourceLineNo">470</span><a id="line.470">                        currentGroup.setResidueNumber(residueNumber);</a>
<span class="sourceLineNo">471</span><a id="line.471">                        currentGroup.setPDBName(groupCode3);</a>
<span class="sourceLineNo">472</span><a id="line.472">                        currentGroup.setHetAtomInFile(isHetAtomInFile);</a>
<span class="sourceLineNo">473</span><a id="line.473">                }</a>
<span class="sourceLineNo">474</span><a id="line.474">                // ANTHONY BRADLEY ADDED THIS -&gt; WE ONLY WAN'T TO CHECK FOR ALT LOCS WHEN IT's NOT THE FIRST GROUP IN CHAIN</a>
<span class="sourceLineNo">475</span><a id="line.475">                else{</a>
<span class="sourceLineNo">476</span><a id="line.476">                // check if residue number is the same ...</a>
<span class="sourceLineNo">477</span><a id="line.477">                // insertion code is part of residue number</a>
<span class="sourceLineNo">478</span><a id="line.478">                        if ( ! residueNumber.equals(currentGroup.getResidueNumber())) {</a>
<span class="sourceLineNo">479</span><a id="line.479">                        //System.out.println("end of residue: "+current_group.getPDBCode()+" "+residueNrInt);</a>
<span class="sourceLineNo">480</span><a id="line.480">                                currentChain.addGroup(currentGroup);</a>
<span class="sourceLineNo">481</span><a id="line.481">                                currentGroup.trimToSize();</a>
<span class="sourceLineNo">482</span><a id="line.482">                                currentGroup = getNewGroup(recordName,aminoCode1,seq_id,groupCode3);</a>
<span class="sourceLineNo">483</span><a id="line.483">                                currentGroup.setPDBName(groupCode3);</a>
<span class="sourceLineNo">484</span><a id="line.484">                                currentGroup.setResidueNumber(residueNumber);</a>
<span class="sourceLineNo">485</span><a id="line.485">                                currentGroup.setHetAtomInFile(isHetAtomInFile);</a>
<span class="sourceLineNo">486</span><a id="line.486"></a>
<span class="sourceLineNo">487</span><a id="line.487"></a>
<span class="sourceLineNo">488</span><a id="line.488">                } else {</a>
<span class="sourceLineNo">489</span><a id="line.489">                        // same residueNumber, but altLocs...</a>
<span class="sourceLineNo">490</span><a id="line.490">                        // test altLoc</a>
<span class="sourceLineNo">491</span><a id="line.491"></a>
<span class="sourceLineNo">492</span><a id="line.492">                        if ( ! altLoc.equals(' ') &amp;&amp; ( ! altLoc.equals('.'))) {</a>
<span class="sourceLineNo">493</span><a id="line.493">                                        logger.debug("found altLoc! " + altLoc + " " + currentGroup + " " + altGroup);</a>
<span class="sourceLineNo">494</span><a id="line.494">                                altGroup = getCorrectAltLocGroup( altLoc,recordName,aminoCode1,groupCode3, seq_id);</a>
<span class="sourceLineNo">495</span><a id="line.495">                                if (altGroup.getChain()==null) {</a>
<span class="sourceLineNo">496</span><a id="line.496">                                                altGroup.setChain(currentChain);</a>
<span class="sourceLineNo">497</span><a id="line.497">                                        }</a>
<span class="sourceLineNo">498</span><a id="line.498">                                }</a>
<span class="sourceLineNo">499</span><a id="line.499">                        }</a>
<span class="sourceLineNo">500</span><a id="line.500">                }</a>
<span class="sourceLineNo">501</span><a id="line.501">                //atomCount++;</a>
<span class="sourceLineNo">502</span><a id="line.502">                //System.out.println("fixing atom name for  &gt;" + atom.getLabel_atom_id() + "&lt; &gt;" + fullname + "&lt;");</a>
<span class="sourceLineNo">503</span><a id="line.503"></a>
<span class="sourceLineNo">504</span><a id="line.504"></a>
<span class="sourceLineNo">505</span><a id="line.505">                if ( params.isParseCAOnly() ){</a>
<span class="sourceLineNo">506</span><a id="line.506">                        // yes , user wants to get CA only</a>
<span class="sourceLineNo">507</span><a id="line.507">                        // only parse CA atoms...</a>
<span class="sourceLineNo">508</span><a id="line.508">                        if (! (atom.getLabel_atom_id().equals(StructureTools.CA_ATOM_NAME) &amp;&amp; atom.getType_symbol().equals("C"))) {</a>
<span class="sourceLineNo">509</span><a id="line.509">                                //System.out.println("ignoring " + line);</a>
<span class="sourceLineNo">510</span><a id="line.510">                                //atomCount--;</a>
<span class="sourceLineNo">511</span><a id="line.511">                                return;</a>
<span class="sourceLineNo">512</span><a id="line.512">                        }</a>
<span class="sourceLineNo">513</span><a id="line.513">                }</a>
<span class="sourceLineNo">514</span><a id="line.514"></a>
<span class="sourceLineNo">515</span><a id="line.515">                //see if chain_id is one of the previous chains ...</a>
<span class="sourceLineNo">516</span><a id="line.516"></a>
<span class="sourceLineNo">517</span><a id="line.517">                Atom a = convertAtom(atom);</a>
<span class="sourceLineNo">518</span><a id="line.518"></a>
<span class="sourceLineNo">519</span><a id="line.519">                //see if chain_id is one of the previous chains ...</a>
<span class="sourceLineNo">520</span><a id="line.520">                if ( altGroup != null) {</a>
<span class="sourceLineNo">521</span><a id="line.521">                        altGroup.addAtom(a);</a>
<span class="sourceLineNo">522</span><a id="line.522">                        altGroup = null;</a>
<span class="sourceLineNo">523</span><a id="line.523">                }</a>
<span class="sourceLineNo">524</span><a id="line.524">                else {</a>
<span class="sourceLineNo">525</span><a id="line.525">                        currentGroup.addAtom(a);</a>
<span class="sourceLineNo">526</span><a id="line.526">                }</a>
<span class="sourceLineNo">527</span><a id="line.527"></a>
<span class="sourceLineNo">528</span><a id="line.528"></a>
<span class="sourceLineNo">529</span><a id="line.529">                String atomName = a.getName();</a>
<span class="sourceLineNo">530</span><a id="line.530">                // make sure that main group has all atoms</a>
<span class="sourceLineNo">531</span><a id="line.531">                // GitHub issue: #76</a>
<span class="sourceLineNo">532</span><a id="line.532">                if ( ! currentGroup.hasAtom(atomName)) {</a>
<span class="sourceLineNo">533</span><a id="line.533">                        // Unless it's microheterogenity https://github.com/rcsb/codec-devel/issues/81</a>
<span class="sourceLineNo">534</span><a id="line.534">                        if (currentGroup.getPDBName().equals(a.getGroup().getPDBName())) {</a>
<span class="sourceLineNo">535</span><a id="line.535">                                if(!StructureTools.hasNonDeuteratedEquiv(a,currentGroup)){</a>
<span class="sourceLineNo">536</span><a id="line.536">                                        currentGroup.addAtom(a);</a>
<span class="sourceLineNo">537</span><a id="line.537">                                }</a>
<span class="sourceLineNo">538</span><a id="line.538">                        }</a>
<span class="sourceLineNo">539</span><a id="line.539"></a>
<span class="sourceLineNo">540</span><a id="line.540">                }</a>
<span class="sourceLineNo">541</span><a id="line.541">        }</a>
<span class="sourceLineNo">542</span><a id="line.542"></a>
<span class="sourceLineNo">543</span><a id="line.543">        /**</a>
<span class="sourceLineNo">544</span><a id="line.544">         * Convert a mmCIF AtomSite object to a BioJava Atom object</a>
<span class="sourceLineNo">545</span><a id="line.545">         *</a>
<span class="sourceLineNo">546</span><a id="line.546">         * @param atom the mmmcif AtomSite record</a>
<span class="sourceLineNo">547</span><a id="line.547">         * @return an Atom</a>
<span class="sourceLineNo">548</span><a id="line.548">         */</a>
<span class="sourceLineNo">549</span><a id="line.549">        private Atom convertAtom(AtomSite atom){</a>
<span class="sourceLineNo">550</span><a id="line.550"></a>
<span class="sourceLineNo">551</span><a id="line.551"></a>
<span class="sourceLineNo">552</span><a id="line.552">                Atom a = new AtomImpl();</a>
<span class="sourceLineNo">553</span><a id="line.553"></a>
<span class="sourceLineNo">554</span><a id="line.554">                a.setPDBserial(Integer.parseInt(atom.getId()));</a>
<span class="sourceLineNo">555</span><a id="line.555">                a.setName(atom.getLabel_atom_id());</a>
<span class="sourceLineNo">556</span><a id="line.556"></a>
<span class="sourceLineNo">557</span><a id="line.557">                double x = Double.parseDouble (atom.getCartn_x());</a>
<span class="sourceLineNo">558</span><a id="line.558">                double y = Double.parseDouble (atom.getCartn_y());</a>
<span class="sourceLineNo">559</span><a id="line.559">                double z = Double.parseDouble (atom.getCartn_z());</a>
<span class="sourceLineNo">560</span><a id="line.560">                a.setX(x);</a>
<span class="sourceLineNo">561</span><a id="line.561">                a.setY(y);</a>
<span class="sourceLineNo">562</span><a id="line.562">                a.setZ(z);</a>
<span class="sourceLineNo">563</span><a id="line.563"></a>
<span class="sourceLineNo">564</span><a id="line.564">                float occupancy = Float.parseFloat (atom.getOccupancy());</a>
<span class="sourceLineNo">565</span><a id="line.565">                a.setOccupancy(occupancy);</a>
<span class="sourceLineNo">566</span><a id="line.566"></a>
<span class="sourceLineNo">567</span><a id="line.567">                float temp = Float.parseFloat (atom.getB_iso_or_equiv());</a>
<span class="sourceLineNo">568</span><a id="line.568">                a.setTempFactor(temp);</a>
<span class="sourceLineNo">569</span><a id="line.569"></a>
<span class="sourceLineNo">570</span><a id="line.570">                String alt = atom.getLabel_alt_id();</a>
<span class="sourceLineNo">571</span><a id="line.571">                if (( alt != null ) &amp;&amp; ( alt.length() &gt; 0) &amp;&amp; (! alt.equals("."))){</a>
<span class="sourceLineNo">572</span><a id="line.572">                        a.setAltLoc(new Character(alt.charAt(0)));</a>
<span class="sourceLineNo">573</span><a id="line.573">                } else {</a>
<span class="sourceLineNo">574</span><a id="line.574">                        a.setAltLoc(new Character(' '));</a>
<span class="sourceLineNo">575</span><a id="line.575">                }</a>
<span class="sourceLineNo">576</span><a id="line.576"></a>
<span class="sourceLineNo">577</span><a id="line.577">                Element element = Element.R;</a>
<span class="sourceLineNo">578</span><a id="line.578">                try {</a>
<span class="sourceLineNo">579</span><a id="line.579">                        element = Element.valueOfIgnoreCase(atom.getType_symbol());</a>
<span class="sourceLineNo">580</span><a id="line.580">                }  catch (IllegalArgumentException e) {</a>
<span class="sourceLineNo">581</span><a id="line.581">                        logger.info("Element {} was not recognised as a BioJava-known element, the element will be represented as the generic element {}", atom.getType_symbol(), Element.R.name());</a>
<span class="sourceLineNo">582</span><a id="line.582">                }</a>
<span class="sourceLineNo">583</span><a id="line.583">                a.setElement(element);</a>
<span class="sourceLineNo">584</span><a id="line.584"></a>
<span class="sourceLineNo">585</span><a id="line.585">                return a;</a>
<span class="sourceLineNo">586</span><a id="line.586"></a>
<span class="sourceLineNo">587</span><a id="line.587">        }</a>
<span class="sourceLineNo">588</span><a id="line.588"></a>
<span class="sourceLineNo">589</span><a id="line.589"></a>
<span class="sourceLineNo">590</span><a id="line.590">        private Group getCorrectAltLocGroup( Character altLoc,</a>
<span class="sourceLineNo">591</span><a id="line.591">                        String recordName,</a>
<span class="sourceLineNo">592</span><a id="line.592">                        Character aminoCode1,</a>
<span class="sourceLineNo">593</span><a id="line.593">                        String groupCode3,</a>
<span class="sourceLineNo">594</span><a id="line.594">                        long seq_id) {</a>
<span class="sourceLineNo">595</span><a id="line.595"></a>
<span class="sourceLineNo">596</span><a id="line.596">                // see if we know this altLoc already;</a>
<span class="sourceLineNo">597</span><a id="line.597">                List&lt;Atom&gt; atoms = currentGroup.getAtoms();</a>
<span class="sourceLineNo">598</span><a id="line.598">                if ( atoms.size() &gt; 0) {</a>
<span class="sourceLineNo">599</span><a id="line.599">                        Atom a1 = atoms.get(0);</a>
<span class="sourceLineNo">600</span><a id="line.600">                        // we are just adding atoms to the current group</a>
<span class="sourceLineNo">601</span><a id="line.601">                        // probably there is a second group following later...</a>
<span class="sourceLineNo">602</span><a id="line.602">                        if (a1.getAltLoc().equals(altLoc)) {</a>
<span class="sourceLineNo">603</span><a id="line.603"></a>
<span class="sourceLineNo">604</span><a id="line.604">                                return currentGroup;</a>
<span class="sourceLineNo">605</span><a id="line.605">                        }</a>
<span class="sourceLineNo">606</span><a id="line.606">                }</a>
<span class="sourceLineNo">607</span><a id="line.607"></a>
<span class="sourceLineNo">608</span><a id="line.608">                List&lt;Group&gt; altLocs = currentGroup.getAltLocs();</a>
<span class="sourceLineNo">609</span><a id="line.609">                for ( Group altLocG : altLocs ){</a>
<span class="sourceLineNo">610</span><a id="line.610">                        atoms = altLocG.getAtoms();</a>
<span class="sourceLineNo">611</span><a id="line.611">                        if ( atoms.size() &gt; 0) {</a>
<span class="sourceLineNo">612</span><a id="line.612">                                for ( Atom a1 : atoms) {</a>
<span class="sourceLineNo">613</span><a id="line.613">                                        if (a1.getAltLoc().equals( altLoc)) {</a>
<span class="sourceLineNo">614</span><a id="line.614"></a>
<span class="sourceLineNo">615</span><a id="line.615">                                                return altLocG;</a>
<span class="sourceLineNo">616</span><a id="line.616">                                        }</a>
<span class="sourceLineNo">617</span><a id="line.617">                                }</a>
<span class="sourceLineNo">618</span><a id="line.618">                        }</a>
<span class="sourceLineNo">619</span><a id="line.619">                }</a>
<span class="sourceLineNo">620</span><a id="line.620"></a>
<span class="sourceLineNo">621</span><a id="line.621">                // no matching altLoc group found.</a>
<span class="sourceLineNo">622</span><a id="line.622">                // build it up.</a>
<span class="sourceLineNo">623</span><a id="line.623"></a>
<span class="sourceLineNo">624</span><a id="line.624">                if ( groupCode3.equals(currentGroup.getPDBName())) {</a>
<span class="sourceLineNo">625</span><a id="line.625">                        if ( currentGroup.getAtoms().size() == 0) {</a>
<span class="sourceLineNo">626</span><a id="line.626">                                //System.out.println("current group is empty " + current_group + " " + altLoc);</a>
<span class="sourceLineNo">627</span><a id="line.627">                                return currentGroup;</a>
<span class="sourceLineNo">628</span><a id="line.628">                        }</a>
<span class="sourceLineNo">629</span><a id="line.629">                        //System.out.println("cloning current group " + current_group + " " + current_group.getAtoms().get(0).getAltLoc() + " altLoc " + altLoc);</a>
<span class="sourceLineNo">630</span><a id="line.630">                        Group altLocG = (Group) currentGroup.clone();</a>
<span class="sourceLineNo">631</span><a id="line.631">                        // drop atoms from cloned group...</a>
<span class="sourceLineNo">632</span><a id="line.632">                        // https://redmine.open-bio.org/issues/3307</a>
<span class="sourceLineNo">633</span><a id="line.633">                        altLocG.setAtoms(new ArrayList&lt;Atom&gt;());</a>
<span class="sourceLineNo">634</span><a id="line.634">                        altLocG.getAltLocs().clear();</a>
<span class="sourceLineNo">635</span><a id="line.635">                        currentGroup.addAltLoc(altLocG);</a>
<span class="sourceLineNo">636</span><a id="line.636">                        return altLocG;</a>
<span class="sourceLineNo">637</span><a id="line.637">                }</a>
<span class="sourceLineNo">638</span><a id="line.638"></a>
<span class="sourceLineNo">639</span><a id="line.639">                //      System.out.println("new  group " + recordName + " " + aminoCode1 + " " +groupCode3);</a>
<span class="sourceLineNo">640</span><a id="line.640">                //String recordName,Character aminoCode1, long seq_id,String groupCode3) {</a>
<span class="sourceLineNo">641</span><a id="line.641">                Group altLocG = getNewGroup(recordName,aminoCode1,seq_id,groupCode3);</a>
<span class="sourceLineNo">642</span><a id="line.642"></a>
<span class="sourceLineNo">643</span><a id="line.643">                altLocG.setPDBName(groupCode3);</a>
<span class="sourceLineNo">644</span><a id="line.644">                altLocG.setResidueNumber(currentGroup.getResidueNumber());</a>
<span class="sourceLineNo">645</span><a id="line.645">                currentGroup.addAltLoc(altLocG);</a>
<span class="sourceLineNo">646</span><a id="line.646">                return altLocG;</a>
<span class="sourceLineNo">647</span><a id="line.647">        }</a>
<span class="sourceLineNo">648</span><a id="line.648"></a>
<span class="sourceLineNo">649</span><a id="line.649">        /**</a>
<span class="sourceLineNo">650</span><a id="line.650">         * Start the parsing</a>
<span class="sourceLineNo">651</span><a id="line.651">         */</a>
<span class="sourceLineNo">652</span><a id="line.652">        @Override</a>
<span class="sourceLineNo">653</span><a id="line.653">        public void documentStart() {</a>
<span class="sourceLineNo">654</span><a id="line.654">                structure = new StructureImpl();</a>
<span class="sourceLineNo">655</span><a id="line.655"></a>
<span class="sourceLineNo">656</span><a id="line.656">                currentChain        = null;</a>
<span class="sourceLineNo">657</span><a id="line.657">                currentGroup            = null;</a>
<span class="sourceLineNo">658</span><a id="line.658">                currentNmrModelNumber   = null;</a>
<span class="sourceLineNo">659</span><a id="line.659">                //atomCount                     = 0;</a>
<span class="sourceLineNo">660</span><a id="line.660"></a>
<span class="sourceLineNo">661</span><a id="line.661">                allModels     = new ArrayList&lt;List&lt;Chain&gt;&gt;();</a>
<span class="sourceLineNo">662</span><a id="line.662">                currentModel  = new ArrayList&lt;Chain&gt;();</a>
<span class="sourceLineNo">663</span><a id="line.663">                entities      = new ArrayList&lt;Entity&gt;();</a>
<span class="sourceLineNo">664</span><a id="line.664">                entityPolys   = new ArrayList&lt;&gt;();</a>
<span class="sourceLineNo">665</span><a id="line.665">                strucRefs     = new ArrayList&lt;StructRef&gt;();</a>
<span class="sourceLineNo">666</span><a id="line.666">                seqResChains  = new ArrayList&lt;Chain&gt;();</a>
<span class="sourceLineNo">667</span><a id="line.667">                entityChains  = new ArrayList&lt;Chain&gt;();</a>
<span class="sourceLineNo">668</span><a id="line.668">                structAsyms   = new ArrayList&lt;StructAsym&gt;();</a>
<span class="sourceLineNo">669</span><a id="line.669"></a>
<span class="sourceLineNo">670</span><a id="line.670">                asymId2entityId = new HashMap&lt;String,String&gt;();</a>
<span class="sourceLineNo">671</span><a id="line.671">                asymId2authorId = new HashMap&lt;&gt;();</a>
<span class="sourceLineNo">672</span><a id="line.672">                structOpers   = new ArrayList&lt;PdbxStructOperList&gt;();</a>
<span class="sourceLineNo">673</span><a id="line.673">                strucAssemblies = new ArrayList&lt;PdbxStructAssembly&gt;();</a>
<span class="sourceLineNo">674</span><a id="line.674">                strucAssemblyGens = new ArrayList&lt;PdbxStructAssemblyGen&gt;();</a>
<span class="sourceLineNo">675</span><a id="line.675">                entitySrcGens = new ArrayList&lt;EntitySrcGen&gt;();</a>
<span class="sourceLineNo">676</span><a id="line.676">                entitySrcNats = new ArrayList&lt;EntitySrcNat&gt;();</a>
<span class="sourceLineNo">677</span><a id="line.677">                entitySrcSyns = new ArrayList&lt;EntitySrcSyn&gt;();</a>
<span class="sourceLineNo">678</span><a id="line.678">                structConn = new ArrayList&lt;StructConn&gt;();</a>
<span class="sourceLineNo">679</span><a id="line.679">                structNcsOper = new ArrayList&lt;StructNcsOper&gt;();</a>
<span class="sourceLineNo">680</span><a id="line.680">                sequenceDifs = new ArrayList&lt;StructRefSeqDif&gt;();</a>
<span class="sourceLineNo">681</span><a id="line.681">                structSiteGens = new ArrayList&lt;StructSiteGen&gt;();</a>
<span class="sourceLineNo">682</span><a id="line.682">        }</a>
<span class="sourceLineNo">683</span><a id="line.683"></a>
<span class="sourceLineNo">684</span><a id="line.684"></a>
<span class="sourceLineNo">685</span><a id="line.685">        @Override</a>
<span class="sourceLineNo">686</span><a id="line.686">        public void documentEnd() {</a>
<span class="sourceLineNo">687</span><a id="line.687"></a>
<span class="sourceLineNo">688</span><a id="line.688">                // Expected that there is one current_chain that needs to be added to the model</a>
<span class="sourceLineNo">689</span><a id="line.689">                // When in headerOnly mode, no Atoms are read, and there will not be an active</a>
<span class="sourceLineNo">690</span><a id="line.690">                // current_chain.</a>
<span class="sourceLineNo">691</span><a id="line.691">                if ( currentChain != null ) {</a>
<span class="sourceLineNo">692</span><a id="line.692"></a>
<span class="sourceLineNo">693</span><a id="line.693">                        currentChain.addGroup(currentGroup);</a>
<span class="sourceLineNo">694</span><a id="line.694">                        if (isKnownChain(currentChain.getId(),currentModel) == null) {</a>
<span class="sourceLineNo">695</span><a id="line.695">                                currentModel.add(currentChain);</a>
<span class="sourceLineNo">696</span><a id="line.696">                        }</a>
<span class="sourceLineNo">697</span><a id="line.697">                } else if (!params.isHeaderOnly()){</a>
<span class="sourceLineNo">698</span><a id="line.698">                        logger.warn("current chain is null at end of document.");</a>
<span class="sourceLineNo">699</span><a id="line.699">                }</a>
<span class="sourceLineNo">700</span><a id="line.700"></a>
<span class="sourceLineNo">701</span><a id="line.701">                allModels.add(currentModel);</a>
<span class="sourceLineNo">702</span><a id="line.702"></a>
<span class="sourceLineNo">703</span><a id="line.703">                // this populates the asymId2authorId and asymId2entityId maps, needed in header only mode to get the mapping</a>
<span class="sourceLineNo">704</span><a id="line.704">                // between the 2 chain identifiers.</a>
<span class="sourceLineNo">705</span><a id="line.705">                initMaps();</a>
<span class="sourceLineNo">706</span><a id="line.706"></a>
<span class="sourceLineNo">707</span><a id="line.707">                for (StructAsym asym : structAsyms) {</a>
<span class="sourceLineNo">708</span><a id="line.708"></a>
<span class="sourceLineNo">709</span><a id="line.709">                        logger.debug("Entity {} matches asym_id: {}", asym.getEntity_id(), asym.getId() );</a>
<span class="sourceLineNo">710</span><a id="line.710"></a>
<span class="sourceLineNo">711</span><a id="line.711">                        Chain s = getEntityChain(asym.getEntity_id());</a>
<span class="sourceLineNo">712</span><a id="line.712">                        Chain seqres = (Chain)s.clone();</a>
<span class="sourceLineNo">713</span><a id="line.713">                        // to solve issue #160 (e.g. 3u7t)</a>
<span class="sourceLineNo">714</span><a id="line.714">                        seqres = removeSeqResHeterogeneity(seqres);</a>
<span class="sourceLineNo">715</span><a id="line.715">                        seqres.setId(asym.getId());</a>
<span class="sourceLineNo">716</span><a id="line.716">                        if (asymId2authorId.get(asym.getId()) !=null ){</a>
<span class="sourceLineNo">717</span><a id="line.717">                                seqres.setName(asymId2authorId.get(asym.getId()));</a>
<span class="sourceLineNo">718</span><a id="line.718">                        } else {</a>
<span class="sourceLineNo">719</span><a id="line.719">                                seqres.setName(asym.getId());</a>
<span class="sourceLineNo">720</span><a id="line.720">                        }</a>
<span class="sourceLineNo">721</span><a id="line.721"></a>
<span class="sourceLineNo">722</span><a id="line.722">                        EntityType type = null;</a>
<span class="sourceLineNo">723</span><a id="line.723">                        try {</a>
<span class="sourceLineNo">724</span><a id="line.724">                                Entity ent = getEntity(Integer.parseInt(asym.getEntity_id()));</a>
<span class="sourceLineNo">725</span><a id="line.725">                                type = EntityType.entityTypeFromString(ent.getType());</a>
<span class="sourceLineNo">726</span><a id="line.726">                        } catch (NumberFormatException e) {</a>
<span class="sourceLineNo">727</span><a id="line.727">                                logger.debug("Could not parse integer from entity id field {}", asym.getEntity_id());</a>
<span class="sourceLineNo">728</span><a id="line.728">                        }</a>
<span class="sourceLineNo">729</span><a id="line.729"></a>
<span class="sourceLineNo">730</span><a id="line.730">                        // we'll only add seqres chains that are polymeric or unknown</a>
<span class="sourceLineNo">731</span><a id="line.731">                        if (type==null || type==EntityType.POLYMER ) {</a>
<span class="sourceLineNo">732</span><a id="line.732">                        seqResChains.add(seqres);</a>
<span class="sourceLineNo">733</span><a id="line.733">                        }</a>
<span class="sourceLineNo">734</span><a id="line.734"></a>
<span class="sourceLineNo">735</span><a id="line.735">                        logger.debug(" seqres: " + asym.getId() + " " + seqres + "&lt;") ;</a>
<span class="sourceLineNo">736</span><a id="line.736">                        // adding the entities to structure</a>
<span class="sourceLineNo">737</span><a id="line.737">                        addEntities(asym);</a>
<span class="sourceLineNo">738</span><a id="line.738"></a>
<span class="sourceLineNo">739</span><a id="line.739">                }</a>
<span class="sourceLineNo">740</span><a id="line.740"></a>
<span class="sourceLineNo">741</span><a id="line.741">                if (structAsyms.isEmpty()) {</a>
<span class="sourceLineNo">742</span><a id="line.742">                        logger.warn("No _struct_asym category in file, no SEQRES groups will be added.");</a>
<span class="sourceLineNo">743</span><a id="line.743">                }</a>
<span class="sourceLineNo">744</span><a id="line.744"></a>
<span class="sourceLineNo">745</span><a id="line.745">                // entities</a>
<span class="sourceLineNo">746</span><a id="line.746">                // In addEntities above we created the entities if they were present in the file</a>
<span class="sourceLineNo">747</span><a id="line.747">                // Now we need to make sure that they are linked to chains and also that if they are not present in the file we need to add them now</a>
<span class="sourceLineNo">748</span><a id="line.748">                linkEntities();</a>
<span class="sourceLineNo">749</span><a id="line.749"></a>
<span class="sourceLineNo">750</span><a id="line.750">                // now that we know the entities, we can add all chains to structure so that they are stored</a>
<span class="sourceLineNo">751</span><a id="line.751">                // properly as polymer/nonpolymer/water chains inside structure</a>
<span class="sourceLineNo">752</span><a id="line.752">                for (List&lt;Chain&gt; model:allModels) {</a>
<span class="sourceLineNo">753</span><a id="line.753">                        structure.addModel(model);</a>
<span class="sourceLineNo">754</span><a id="line.754">                }</a>
<span class="sourceLineNo">755</span><a id="line.755"></a>
<span class="sourceLineNo">756</span><a id="line.756">                // Only align if requested (default) and not when headerOnly mode with no Atoms.</a>
<span class="sourceLineNo">757</span><a id="line.757">                // Otherwise, we store the empty SeqRes Groups unchanged in the right chains.</a>
<span class="sourceLineNo">758</span><a id="line.758">                if ( params.isAlignSeqRes() &amp;&amp; !params.isHeaderOnly() ){</a>
<span class="sourceLineNo">759</span><a id="line.759">                        logger.debug("Parsing mode align_seqres, will parse SEQRES and align to ATOM sequence");</a>
<span class="sourceLineNo">760</span><a id="line.760">                        alignSeqRes();</a>
<span class="sourceLineNo">761</span><a id="line.761">                } else {</a>
<span class="sourceLineNo">762</span><a id="line.762">                        logger.debug("Parsing mode unalign_seqres, will parse SEQRES but not align it to ATOM sequence");</a>
<span class="sourceLineNo">763</span><a id="line.763">                        SeqRes2AtomAligner.storeUnAlignedSeqRes(structure, seqResChains, params.isHeaderOnly());</a>
<span class="sourceLineNo">764</span><a id="line.764">                }</a>
<span class="sourceLineNo">765</span><a id="line.765"></a>
<span class="sourceLineNo">766</span><a id="line.766"></a>
<span class="sourceLineNo">767</span><a id="line.767">                // Now make sure all altlocgroups have all the atoms in all the groups</a>
<span class="sourceLineNo">768</span><a id="line.768">                StructureTools.cleanUpAltLocs(structure);</a>
<span class="sourceLineNo">769</span><a id="line.769"></a>
<span class="sourceLineNo">770</span><a id="line.770">                // NOTE bonds and charges can only be done at this point that the chain id mapping is properly sorted out</a>
<span class="sourceLineNo">771</span><a id="line.771">                if (!params.isHeaderOnly()) {</a>
<span class="sourceLineNo">772</span><a id="line.772">                        if ( params.shouldCreateAtomBonds()) {</a>
<span class="sourceLineNo">773</span><a id="line.773">                                addBonds();</a>
<span class="sourceLineNo">774</span><a id="line.774">                        }</a>
<span class="sourceLineNo">775</span><a id="line.775"></a>
<span class="sourceLineNo">776</span><a id="line.776">                        if ( params.shouldCreateAtomCharges()) {</a>
<span class="sourceLineNo">777</span><a id="line.777">                                addCharges();</a>
<span class="sourceLineNo">778</span><a id="line.778">                        }</a>
<span class="sourceLineNo">779</span><a id="line.779">                }</a>
<span class="sourceLineNo">780</span><a id="line.780"></a>
<span class="sourceLineNo">781</span><a id="line.781">                if (!params.isHeaderOnly()) {</a>
<span class="sourceLineNo">782</span><a id="line.782"></a>
<span class="sourceLineNo">783</span><a id="line.783">                        // Do structure.setSites(sites) after any chain renaming to be like PDB.</a>
<span class="sourceLineNo">784</span><a id="line.784">                        addSites();</a>
<span class="sourceLineNo">785</span><a id="line.785">                }</a>
<span class="sourceLineNo">786</span><a id="line.786"></a>
<span class="sourceLineNo">787</span><a id="line.787"></a>
<span class="sourceLineNo">788</span><a id="line.788"></a>
<span class="sourceLineNo">789</span><a id="line.789">                // set the oligomeric state info in the header...</a>
<span class="sourceLineNo">790</span><a id="line.790">                if (params.isParseBioAssembly()) {</a>
<span class="sourceLineNo">791</span><a id="line.791"></a>
<span class="sourceLineNo">792</span><a id="line.792">                        // the more detailed mapping of chains to rotation operations happens in StructureIO...</a>
<span class="sourceLineNo">793</span><a id="line.793"></a>
<span class="sourceLineNo">794</span><a id="line.794">                        Map&lt;Integer,BioAssemblyInfo&gt; bioAssemblies = new LinkedHashMap&lt;Integer, BioAssemblyInfo&gt;();</a>
<span class="sourceLineNo">795</span><a id="line.795"></a>
<span class="sourceLineNo">796</span><a id="line.796">                        for ( PdbxStructAssembly psa : strucAssemblies){</a>
<span class="sourceLineNo">797</span><a id="line.797"></a>
<span class="sourceLineNo">798</span><a id="line.798">                                List&lt;PdbxStructAssemblyGen&gt; psags = new ArrayList&lt;PdbxStructAssemblyGen&gt;(1);</a>
<span class="sourceLineNo">799</span><a id="line.799"></a>
<span class="sourceLineNo">800</span><a id="line.800">                                for ( PdbxStructAssemblyGen psag: strucAssemblyGens ) {</a>
<span class="sourceLineNo">801</span><a id="line.801">                                        if ( psag.getAssembly_id().equals(psa.getId())) {</a>
<span class="sourceLineNo">802</span><a id="line.802">                                                psags.add(psag);</a>
<span class="sourceLineNo">803</span><a id="line.803">                                        }</a>
<span class="sourceLineNo">804</span><a id="line.804">                                }</a>
<span class="sourceLineNo">805</span><a id="line.805"></a>
<span class="sourceLineNo">806</span><a id="line.806">                                BiologicalAssemblyBuilder builder = new BiologicalAssemblyBuilder();</a>
<span class="sourceLineNo">807</span><a id="line.807"></a>
<span class="sourceLineNo">808</span><a id="line.808">                                // these are the transformations that need to be applied to our model</a>
<span class="sourceLineNo">809</span><a id="line.809">                                List&lt;BiologicalAssemblyTransformation&gt; transformations = builder.getBioUnitTransformationList(psa, psags, structOpers);</a>
<span class="sourceLineNo">810</span><a id="line.810"></a>
<span class="sourceLineNo">811</span><a id="line.811">                                int bioAssemblyId = -1;</a>
<span class="sourceLineNo">812</span><a id="line.812">                                try {</a>
<span class="sourceLineNo">813</span><a id="line.813">                                        bioAssemblyId = Integer.parseInt(psa.getId());</a>
<span class="sourceLineNo">814</span><a id="line.814">                                } catch (NumberFormatException e) {</a>
<span class="sourceLineNo">815</span><a id="line.815">                                        logger.info("Could not parse a numerical bio assembly id from '{}'",psa.getId());</a>
<span class="sourceLineNo">816</span><a id="line.816">                                }</a>
<span class="sourceLineNo">817</span><a id="line.817"></a>
<span class="sourceLineNo">818</span><a id="line.818">                                // if bioassembly id is not numerical we throw it away</a>
<span class="sourceLineNo">819</span><a id="line.819">                                // this happens usually for viral capsid entries, like 1ei7</a>
<span class="sourceLineNo">820</span><a id="line.820">                                // see issue #230 in github</a>
<span class="sourceLineNo">821</span><a id="line.821">                                if (bioAssemblyId!=-1) {</a>
<span class="sourceLineNo">822</span><a id="line.822">                                        int mmSize = 0;</a>
<span class="sourceLineNo">823</span><a id="line.823">                                        // note that the transforms contain asym ids of both polymers and non-polymers</a>
<span class="sourceLineNo">824</span><a id="line.824">                                        // For the mmsize, we are only interested in the polymers</a>
<span class="sourceLineNo">825</span><a id="line.825">                                        for (BiologicalAssemblyTransformation transf:transformations) {</a>
<span class="sourceLineNo">826</span><a id="line.826">                                                Chain c = structure.getChain(transf.getChainId());</a>
<span class="sourceLineNo">827</span><a id="line.827">                                                if (c==null) {</a>
<span class="sourceLineNo">828</span><a id="line.828">                                                        logger.info("Could not find asym id {} specified in struct_assembly_gen", transf.getChainId());</a>
<span class="sourceLineNo">829</span><a id="line.829">                                                        continue;</a>
<span class="sourceLineNo">830</span><a id="line.830">                                                }</a>
<span class="sourceLineNo">831</span><a id="line.831">                                                if (c.getEntityType() == EntityType.POLYMER &amp;&amp;</a>
<span class="sourceLineNo">832</span><a id="line.832">                                                        // for entries like 4kro, sugars are annotated as polymers but we</a>
<span class="sourceLineNo">833</span><a id="line.833">                                                        // don't want them in the macromolecularSize count</a>
<span class="sourceLineNo">834</span><a id="line.834">                                                        !c.getEntityInfo().getDescription().contains("SUGAR") ) {</a>
<span class="sourceLineNo">835</span><a id="line.835"></a>
<span class="sourceLineNo">836</span><a id="line.836">                                                                mmSize++;</a>
<span class="sourceLineNo">837</span><a id="line.837">                                                        }</a>
<span class="sourceLineNo">838</span><a id="line.838">                                        }</a>
<span class="sourceLineNo">839</span><a id="line.839"></a>
<span class="sourceLineNo">840</span><a id="line.840">                                        BioAssemblyInfo bioAssembly = new BioAssemblyInfo();</a>
<span class="sourceLineNo">841</span><a id="line.841">                                        bioAssembly.setId(bioAssemblyId);</a>
<span class="sourceLineNo">842</span><a id="line.842">                                        bioAssembly.setMacromolecularSize(mmSize);</a>
<span class="sourceLineNo">843</span><a id="line.843">                                        bioAssembly.setTransforms(transformations);</a>
<span class="sourceLineNo">844</span><a id="line.844">                                        bioAssemblies.put(bioAssemblyId,bioAssembly);</a>
<span class="sourceLineNo">845</span><a id="line.845">                                }</a>
<span class="sourceLineNo">846</span><a id="line.846"></a>
<span class="sourceLineNo">847</span><a id="line.847">                        }</a>
<span class="sourceLineNo">848</span><a id="line.848">                        structure.getPDBHeader().setBioAssemblies(bioAssemblies);</a>
<span class="sourceLineNo">849</span><a id="line.849">                }</a>
<span class="sourceLineNo">850</span><a id="line.850"></a>
<span class="sourceLineNo">851</span><a id="line.851">                setStructNcsOps();</a>
<span class="sourceLineNo">852</span><a id="line.852"></a>
<span class="sourceLineNo">853</span><a id="line.853">                setCrystallographicInfoMetadata();</a>
<span class="sourceLineNo">854</span><a id="line.854"></a>
<span class="sourceLineNo">855</span><a id="line.855"></a>
<span class="sourceLineNo">856</span><a id="line.856">                Map&lt;String,List&lt;SeqMisMatch&gt;&gt; misMatchMap = new HashMap&lt;String, List&lt;SeqMisMatch&gt;&gt;();</a>
<span class="sourceLineNo">857</span><a id="line.857">                for (StructRefSeqDif sdif : sequenceDifs) {</a>
<span class="sourceLineNo">858</span><a id="line.858">                        SeqMisMatch misMatch = new SeqMisMatchImpl();</a>
<span class="sourceLineNo">859</span><a id="line.859">                        misMatch.setDetails(sdif.getDetails());</a>
<span class="sourceLineNo">860</span><a id="line.860"></a>
<span class="sourceLineNo">861</span><a id="line.861">                        String insCode = sdif.getPdbx_pdb_ins_code();</a>
<span class="sourceLineNo">862</span><a id="line.862">                        if ( insCode != null &amp;&amp; insCode.equals("?"))</a>
<span class="sourceLineNo">863</span><a id="line.863">                                insCode = null;</a>
<span class="sourceLineNo">864</span><a id="line.864">                        misMatch.setInsCode(insCode);</a>
<span class="sourceLineNo">865</span><a id="line.865">                        misMatch.setOrigGroup(sdif.getDb_mon_id());</a>
<span class="sourceLineNo">866</span><a id="line.866">                        misMatch.setPdbGroup(sdif.getMon_id());</a>
<span class="sourceLineNo">867</span><a id="line.867">                        misMatch.setPdbResNum(sdif.getPdbx_auth_seq_num());</a>
<span class="sourceLineNo">868</span><a id="line.868">                        misMatch.setUniProtId(sdif.getPdbx_seq_db_accession_code());</a>
<span class="sourceLineNo">869</span><a id="line.869">                        misMatch.setSeqNum(sdif.getSeq_num());</a>
<span class="sourceLineNo">870</span><a id="line.870"></a>
<span class="sourceLineNo">871</span><a id="line.871"></a>
<span class="sourceLineNo">872</span><a id="line.872">                        List&lt;SeqMisMatch&gt; mms = misMatchMap.get(sdif.getPdbx_pdb_strand_id());</a>
<span class="sourceLineNo">873</span><a id="line.873">                        if ( mms == null) {</a>
<span class="sourceLineNo">874</span><a id="line.874">                                mms = new ArrayList&lt;SeqMisMatch&gt;();</a>
<span class="sourceLineNo">875</span><a id="line.875">                                misMatchMap.put(sdif.getPdbx_pdb_strand_id(),mms);</a>
<span class="sourceLineNo">876</span><a id="line.876">                        }</a>
<span class="sourceLineNo">877</span><a id="line.877">                        mms.add(misMatch);</a>
<span class="sourceLineNo">878</span><a id="line.878"></a>
<span class="sourceLineNo">879</span><a id="line.879">                }</a>
<span class="sourceLineNo">880</span><a id="line.880"></a>
<span class="sourceLineNo">881</span><a id="line.881">                for (String chainId : misMatchMap.keySet()){</a>
<span class="sourceLineNo">882</span><a id="line.882"></a>
<span class="sourceLineNo">883</span><a id="line.883">                        Chain chain = structure.getPolyChainByPDB(chainId);</a>
<span class="sourceLineNo">884</span><a id="line.884"></a>
<span class="sourceLineNo">885</span><a id="line.885">                        if ( chain == null) {</a>
<span class="sourceLineNo">886</span><a id="line.886">                                logger.warn("Could not set mismatches for chain with author id" + chainId);</a>
<span class="sourceLineNo">887</span><a id="line.887">                                continue;</a>
<span class="sourceLineNo">888</span><a id="line.888">                        }</a>
<span class="sourceLineNo">889</span><a id="line.889"></a>
<span class="sourceLineNo">890</span><a id="line.890">                        chain.setSeqMisMatches(misMatchMap.get(chainId));</a>
<span class="sourceLineNo">891</span><a id="line.891"></a>
<span class="sourceLineNo">892</span><a id="line.892"></a>
<span class="sourceLineNo">893</span><a id="line.893">                }</a>
<span class="sourceLineNo">894</span><a id="line.894"></a>
<span class="sourceLineNo">895</span><a id="line.895">        }</a>
<span class="sourceLineNo">896</span><a id="line.896"></a>
<span class="sourceLineNo">897</span><a id="line.897">        /**</a>
<span class="sourceLineNo">898</span><a id="line.898">         * Here we link entities to chains.</a>
<span class="sourceLineNo">899</span><a id="line.899">         * Also if entities are not present in file, this initialises the entities with some heuristics, see {@link org.biojava.nbio.structure.io.EntityFinder}</a>
<span class="sourceLineNo">900</span><a id="line.900">         */</a>
<span class="sourceLineNo">901</span><a id="line.901">        private void linkEntities() {</a>
<span class="sourceLineNo">902</span><a id="line.902"></a>
<span class="sourceLineNo">903</span><a id="line.903">                for (int i =0; i&lt; allModels.size() ; i++){</a>
<span class="sourceLineNo">904</span><a id="line.904">                        for (Chain chain : allModels.get(i)) {</a>
<span class="sourceLineNo">905</span><a id="line.905">                                //logger.info("linking entities for " + chain.getId() + " "  + chain.getName());</a>
<span class="sourceLineNo">906</span><a id="line.906">                                String entityId = asymId2entityId.get(chain.getId());</a>
<span class="sourceLineNo">907</span><a id="line.907"></a>
<span class="sourceLineNo">908</span><a id="line.908">                                if (entityId==null) {</a>
<span class="sourceLineNo">909</span><a id="line.909">                                        // this can happen for instance if the cif file didn't have _struct_asym category at all</a>
<span class="sourceLineNo">910</span><a id="line.910">                                        // and thus we have no asymId2entityId mapping at all</a>
<span class="sourceLineNo">911</span><a id="line.911">                                        logger.info("No entity id could be found for chain {}", chain.getId());</a>
<span class="sourceLineNo">912</span><a id="line.912">                                        continue;</a>
<span class="sourceLineNo">913</span><a id="line.913">                                }</a>
<span class="sourceLineNo">914</span><a id="line.914">                                int eId = Integer.parseInt(entityId);</a>
<span class="sourceLineNo">915</span><a id="line.915"></a>
<span class="sourceLineNo">916</span><a id="line.916">                                // Entities are not added for non-polymeric entities, if a chain is non-polymeric its entity won't be found.</a>
<span class="sourceLineNo">917</span><a id="line.917">                                // TODO: add all entities and unique compounds and add methods to directly get polymer or non-polymer</a>
<span class="sourceLineNo">918</span><a id="line.918">                                // asyms (chains).  Either create a unique StructureImpl or modify existing for a better representation of the</a>
<span class="sourceLineNo">919</span><a id="line.919">                                // mmCIF internal data structures but is compatible with Structure interface.</a>
<span class="sourceLineNo">920</span><a id="line.920">                                // Some examples of PDB entries with this kind of problem:</a>
<span class="sourceLineNo">921</span><a id="line.921">                                //   - 2uub: asym_id X, chainName Z, entity_id 24: fully non-polymeric but still with its own chainName</a>
<span class="sourceLineNo">922</span><a id="line.922">                                //   - 3o6j: asym_id K, chainName Z, entity_id 6 : a single water molecule</a>
<span class="sourceLineNo">923</span><a id="line.923">                                //   - 1dz9: asym_id K, chainName K, entity_id 6 : a potassium ion alone</a>
<span class="sourceLineNo">924</span><a id="line.924"></a>
<span class="sourceLineNo">925</span><a id="line.925">                                EntityInfo entityInfo = structure.getEntityById(eId);</a>
<span class="sourceLineNo">926</span><a id="line.926">                                if (entityInfo==null) {</a>
<span class="sourceLineNo">927</span><a id="line.927">                                        // Supports the case where the only chain members were from non-polymeric entity that is missing.</a>
<span class="sourceLineNo">928</span><a id="line.928">                                        // Solved by creating a new Compound(entity) to which this chain will belong.</a>
<span class="sourceLineNo">929</span><a id="line.929">                                        logger.info("Could not find an Entity for entity_id {}, for chain id {}, creating a new Entity.",</a>
<span class="sourceLineNo">930</span><a id="line.930">                                                        eId, chain.getId());</a>
<span class="sourceLineNo">931</span><a id="line.931">                                        entityInfo = new EntityInfo();</a>
<span class="sourceLineNo">932</span><a id="line.932">                                        entityInfo.setMolId(eId);</a>
<span class="sourceLineNo">933</span><a id="line.933">                                        entityInfo.addChain(chain);</a>
<span class="sourceLineNo">934</span><a id="line.934">                                        if (chain.isWaterOnly()) {</a>
<span class="sourceLineNo">935</span><a id="line.935">                                                entityInfo.setType(EntityType.WATER);</a>
<span class="sourceLineNo">936</span><a id="line.936">                                        } else {</a>
<span class="sourceLineNo">937</span><a id="line.937">                                                entityInfo.setType(EntityType.NONPOLYMER);</a>
<span class="sourceLineNo">938</span><a id="line.938">                                        }</a>
<span class="sourceLineNo">939</span><a id="line.939">                                        chain.setEntityInfo(entityInfo);</a>
<span class="sourceLineNo">940</span><a id="line.940">                                        structure.addEntityInfo(entityInfo);</a>
<span class="sourceLineNo">941</span><a id="line.941">                                } else {</a>
<span class="sourceLineNo">942</span><a id="line.942">                                        logger.debug("Adding chain with chain id {} (auth id {}) to Entity with entity_id {}",</a>
<span class="sourceLineNo">943</span><a id="line.943">                                                        chain.getId(), chain.getName(), eId);</a>
<span class="sourceLineNo">944</span><a id="line.944">                                        entityInfo.addChain(chain);</a>
<span class="sourceLineNo">945</span><a id="line.945">                                        chain.setEntityInfo(entityInfo);</a>
<span class="sourceLineNo">946</span><a id="line.946">                                }</a>
<span class="sourceLineNo">947</span><a id="line.947"></a>
<span class="sourceLineNo">948</span><a id="line.948">                        }</a>
<span class="sourceLineNo">949</span><a id="line.949"></a>
<span class="sourceLineNo">950</span><a id="line.950">                }</a>
<span class="sourceLineNo">951</span><a id="line.951"></a>
<span class="sourceLineNo">952</span><a id="line.952">                // if no entity information was present in file we then go and find the entities heuristically with EntityFinder</a>
<span class="sourceLineNo">953</span><a id="line.953">                List&lt;EntityInfo&gt; entityInfos = structure.getEntityInfos();</a>
<span class="sourceLineNo">954</span><a id="line.954">                if (entityInfos==null || entityInfos.isEmpty()) {</a>
<span class="sourceLineNo">955</span><a id="line.955"></a>
<span class="sourceLineNo">956</span><a id="line.956">                        List&lt;List&lt;Chain&gt;&gt; polyModels = new ArrayList&lt;&gt;();</a>
<span class="sourceLineNo">957</span><a id="line.957">                        List&lt;List&lt;Chain&gt;&gt; nonPolyModels = new ArrayList&lt;&gt;();</a>
<span class="sourceLineNo">958</span><a id="line.958">                        List&lt;List&lt;Chain&gt;&gt; waterModels = new ArrayList&lt;&gt;();</a>
<span class="sourceLineNo">959</span><a id="line.959"></a>
<span class="sourceLineNo">960</span><a id="line.960">                        for (List&lt;Chain&gt; model:allModels) {</a>
<span class="sourceLineNo">961</span><a id="line.961"></a>
<span class="sourceLineNo">962</span><a id="line.962">                                List&lt;Chain&gt; polyChains = new ArrayList&lt;&gt;();</a>
<span class="sourceLineNo">963</span><a id="line.963">                                List&lt;Chain&gt; nonPolyChains = new ArrayList&lt;&gt;();</a>
<span class="sourceLineNo">964</span><a id="line.964">                                List&lt;Chain&gt; waterChains = new ArrayList&lt;&gt;();</a>
<span class="sourceLineNo">965</span><a id="line.965"></a>
<span class="sourceLineNo">966</span><a id="line.966">                                polyModels.add(polyChains);</a>
<span class="sourceLineNo">967</span><a id="line.967">                                nonPolyModels.add(nonPolyChains);</a>
<span class="sourceLineNo">968</span><a id="line.968">                                waterModels.add(waterChains);</a>
<span class="sourceLineNo">969</span><a id="line.969"></a>
<span class="sourceLineNo">970</span><a id="line.970">                                for (Chain c:model) {</a>
<span class="sourceLineNo">971</span><a id="line.971"></a>
<span class="sourceLineNo">972</span><a id="line.972">                                        // we only have entities for polymeric chains, all others are ignored for assigning entities</a>
<span class="sourceLineNo">973</span><a id="line.973">                                        if (c.isWaterOnly()) {</a>
<span class="sourceLineNo">974</span><a id="line.974">                                                waterChains.add(c);</a>
<span class="sourceLineNo">975</span><a id="line.975"></a>
<span class="sourceLineNo">976</span><a id="line.976">                                        } else if (c.isPureNonPolymer()) {</a>
<span class="sourceLineNo">977</span><a id="line.977">                                                nonPolyChains.add(c);</a>
<span class="sourceLineNo">978</span><a id="line.978"></a>
<span class="sourceLineNo">979</span><a id="line.979">                                        } else {</a>
<span class="sourceLineNo">980</span><a id="line.980">                                                polyChains.add(c);</a>
<span class="sourceLineNo">981</span><a id="line.981">                                        }</a>
<span class="sourceLineNo">982</span><a id="line.982">                                }</a>
<span class="sourceLineNo">983</span><a id="line.983">                        }</a>
<span class="sourceLineNo">984</span><a id="line.984"></a>
<span class="sourceLineNo">985</span><a id="line.985">                        entityInfos = EntityFinder.findPolyEntities(polyModels);</a>
<span class="sourceLineNo">986</span><a id="line.986">                        EntityFinder.createPurelyNonPolyEntities(nonPolyModels, waterModels, entityInfos);</a>
<span class="sourceLineNo">987</span><a id="line.987"></a>
<span class="sourceLineNo">988</span><a id="line.988"></a>
<span class="sourceLineNo">989</span><a id="line.989">                        structure.setEntityInfos(entityInfos);</a>
<span class="sourceLineNo">990</span><a id="line.990">                }</a>
<span class="sourceLineNo">991</span><a id="line.991"></a>
<span class="sourceLineNo">992</span><a id="line.992">                // final sanity check: it can happen that from the annotated entities some are not linked to any chains</a>
<span class="sourceLineNo">993</span><a id="line.993">                // e.g. 3s26: a sugar entity does not have any chains associated to it (it seems to be happening with many sugar compounds)</a>
<span class="sourceLineNo">994</span><a id="line.994">                // we simply log it, this can sign some other problems if the entities are used down the line</a>
<span class="sourceLineNo">995</span><a id="line.995">                for (EntityInfo e:entityInfos) {</a>
<span class="sourceLineNo">996</span><a id="line.996">                        if (e.getChains().isEmpty()) {</a>
<span class="sourceLineNo">997</span><a id="line.997">                                logger.info("Entity {} '{}' has no chains associated to it",</a>
<span class="sourceLineNo">998</span><a id="line.998">                                                e.getMolId()&lt;0?"with no entity id":e.getMolId(), e.getDescription());</a>
<span class="sourceLineNo">999</span><a id="line.999">                        }</a>
<span class="sourceLineNo">1000</span><a id="line.1000">                }</a>
<span class="sourceLineNo">1001</span><a id="line.1001"></a>
<span class="sourceLineNo">1002</span><a id="line.1002">        }</a>
<span class="sourceLineNo">1003</span><a id="line.1003"></a>
<span class="sourceLineNo">1004</span><a id="line.1004">        private void addCharges() {</a>
<span class="sourceLineNo">1005</span><a id="line.1005">                ChargeAdder.addCharges(structure);</a>
<span class="sourceLineNo">1006</span><a id="line.1006">        }</a>
<span class="sourceLineNo">1007</span><a id="line.1007"></a>
<span class="sourceLineNo">1008</span><a id="line.1008">        /**</a>
<span class="sourceLineNo">1009</span><a id="line.1009">         * The method will return a new reference to a Chain with any consecutive groups</a>
<span class="sourceLineNo">1010</span><a id="line.1010">         * having same residue numbers removed.</a>
<span class="sourceLineNo">1011</span><a id="line.1011">         * This is necessary to solve the microheterogeneity issue in entries like 3u7t (see github issue #160)</a>
<span class="sourceLineNo">1012</span><a id="line.1012">         * @param c</a>
<span class="sourceLineNo">1013</span><a id="line.1013">         * @return</a>
<span class="sourceLineNo">1014</span><a id="line.1014">         */</a>
<span class="sourceLineNo">1015</span><a id="line.1015">        private static Chain removeSeqResHeterogeneity(Chain c) {</a>
<span class="sourceLineNo">1016</span><a id="line.1016"></a>
<span class="sourceLineNo">1017</span><a id="line.1017">                Chain trimmedChain = new ChainImpl();</a>
<span class="sourceLineNo">1018</span><a id="line.1018"></a>
<span class="sourceLineNo">1019</span><a id="line.1019">                ResidueNumber lastResNum = null;</a>
<span class="sourceLineNo">1020</span><a id="line.1020"></a>
<span class="sourceLineNo">1021</span><a id="line.1021">                for (Group g:c.getAtomGroups()) {</a>
<span class="sourceLineNo">1022</span><a id="line.1022"></a>
<span class="sourceLineNo">1023</span><a id="line.1023">                        // note we have to deep copy this, otherwise they stay linked and would get altered in addGroup(g)</a>
<span class="sourceLineNo">1024</span><a id="line.1024">                        ResidueNumber currentResNum = new ResidueNumber(</a>
<span class="sourceLineNo">1025</span><a id="line.1025">                                        g.getResidueNumber().getChainName(),</a>
<span class="sourceLineNo">1026</span><a id="line.1026">                                        g.getResidueNumber().getSeqNum(),</a>
<span class="sourceLineNo">1027</span><a id="line.1027">                                        g.getResidueNumber().getInsCode());</a>
<span class="sourceLineNo">1028</span><a id="line.1028"></a>
<span class="sourceLineNo">1029</span><a id="line.1029">                        if (lastResNum == null || !lastResNum.equals(currentResNum) ) {</a>
<span class="sourceLineNo">1030</span><a id="line.1030">                                trimmedChain.addGroup(g);</a>
<span class="sourceLineNo">1031</span><a id="line.1031">                        } else {</a>
<span class="sourceLineNo">1032</span><a id="line.1032">                                logger.debug("Removing seqres group because it seems to be repeated in entity_poly_seq, most likely has hetero='y': "+g);</a>
<span class="sourceLineNo">1033</span><a id="line.1033">                        }</a>
<span class="sourceLineNo">1034</span><a id="line.1034"></a>
<span class="sourceLineNo">1035</span><a id="line.1035">                        lastResNum = currentResNum;</a>
<span class="sourceLineNo">1036</span><a id="line.1036"></a>
<span class="sourceLineNo">1037</span><a id="line.1037">                }</a>
<span class="sourceLineNo">1038</span><a id="line.1038">                return trimmedChain;</a>
<span class="sourceLineNo">1039</span><a id="line.1039">        }</a>
<span class="sourceLineNo">1040</span><a id="line.1040"></a>
<span class="sourceLineNo">1041</span><a id="line.1041">        private void addBonds() {</a>
<span class="sourceLineNo">1042</span><a id="line.1042">                BondMaker maker = new BondMaker(structure, params);</a>
<span class="sourceLineNo">1043</span><a id="line.1043">                maker.makeBonds();</a>
<span class="sourceLineNo">1044</span><a id="line.1044">                maker.formBondsFromStructConn(structConn);</a>
<span class="sourceLineNo">1045</span><a id="line.1045">        }</a>
<span class="sourceLineNo">1046</span><a id="line.1046"></a>
<span class="sourceLineNo">1047</span><a id="line.1047">        private void alignSeqRes() {</a>
<span class="sourceLineNo">1048</span><a id="line.1048"></a>
<span class="sourceLineNo">1049</span><a id="line.1049">                logger.debug("Parsing mode align_seqres, will align to ATOM to SEQRES sequence");</a>
<span class="sourceLineNo">1050</span><a id="line.1050"></a>
<span class="sourceLineNo">1051</span><a id="line.1051">                // fix SEQRES residue numbering for all models</a>
<span class="sourceLineNo">1052</span><a id="line.1052"></a>
<span class="sourceLineNo">1053</span><a id="line.1053">                for (int model=0;model&lt;structure.nrModels();model++) {</a>
<span class="sourceLineNo">1054</span><a id="line.1054"></a>
<span class="sourceLineNo">1055</span><a id="line.1055">                        List&lt;Chain&gt; atomList   = structure.getModel(model);</a>
<span class="sourceLineNo">1056</span><a id="line.1056"></a>
<span class="sourceLineNo">1057</span><a id="line.1057">                        for (Chain seqResChain: seqResChains){</a>
<span class="sourceLineNo">1058</span><a id="line.1058"></a>
<span class="sourceLineNo">1059</span><a id="line.1059">                                // this extracts the matching atom chain from atomList</a>
<span class="sourceLineNo">1060</span><a id="line.1060">                                Chain atomChain = SeqRes2AtomAligner.getMatchingAtomRes(seqResChain, atomList, true);</a>
<span class="sourceLineNo">1061</span><a id="line.1061"></a>
<span class="sourceLineNo">1062</span><a id="line.1062">                                if (atomChain == null) {</a>
<span class="sourceLineNo">1063</span><a id="line.1063">                                        // most likely there's no observed residues at all for the seqres chain: can't map</a>
<span class="sourceLineNo">1064</span><a id="line.1064">                                        // e.g. 3zyb: chains with asym_id L,M,N,O,P have no observed residues</a>
<span class="sourceLineNo">1065</span><a id="line.1065">                                        logger.info("Could not map SEQRES chain with asym_id={} to any ATOM chain. Most likely there's no observed residues in the chain.",</a>
<span class="sourceLineNo">1066</span><a id="line.1066">                                                        seqResChain.getId());</a>
<span class="sourceLineNo">1067</span><a id="line.1067">                                        continue;</a>
<span class="sourceLineNo">1068</span><a id="line.1068">                                }</a>
<span class="sourceLineNo">1069</span><a id="line.1069"></a>
<span class="sourceLineNo">1070</span><a id="line.1070">                                //map the atoms to the seqres...</a>
<span class="sourceLineNo">1071</span><a id="line.1071"></a>
<span class="sourceLineNo">1072</span><a id="line.1072">                                // we need to first clone the seqres so that they stay independent for different models</a>
<span class="sourceLineNo">1073</span><a id="line.1073">                                List&lt;Group&gt; seqResGroups = new ArrayList&lt;Group&gt;();</a>
<span class="sourceLineNo">1074</span><a id="line.1074">                                for (int i=0;i&lt;seqResChain.getAtomGroups().size();i++) {</a>
<span class="sourceLineNo">1075</span><a id="line.1075">                                        seqResGroups.add((Group)seqResChain.getAtomGroups().get(i).clone());</a>
<span class="sourceLineNo">1076</span><a id="line.1076">                                }</a>
<span class="sourceLineNo">1077</span><a id="line.1077"></a>
<span class="sourceLineNo">1078</span><a id="line.1078">                                for ( int seqResPos = 0 ; seqResPos &lt; seqResGroups.size(); seqResPos++) {</a>
<span class="sourceLineNo">1079</span><a id="line.1079">                                        Group seqresG = seqResGroups.get(seqResPos);</a>
<span class="sourceLineNo">1080</span><a id="line.1080">                                        boolean found = false;</a>
<span class="sourceLineNo">1081</span><a id="line.1081">                                        for ( Group atomG: atomChain.getAtomGroups()) {</a>
<span class="sourceLineNo">1082</span><a id="line.1082"></a>
<span class="sourceLineNo">1083</span><a id="line.1083">                                                int internalNr = getInternalNr (atomG);</a>
<span class="sourceLineNo">1084</span><a id="line.1084"></a>
<span class="sourceLineNo">1085</span><a id="line.1085">                                                if (seqresG.getResidueNumber().getSeqNum() == internalNr ) {</a>
<span class="sourceLineNo">1086</span><a id="line.1086">                                                        seqResGroups.set(seqResPos, atomG);</a>
<span class="sourceLineNo">1087</span><a id="line.1087">                                                        found = true;</a>
<span class="sourceLineNo">1088</span><a id="line.1088">                                                        break;</a>
<span class="sourceLineNo">1089</span><a id="line.1089">                                                }</a>
<span class="sourceLineNo">1090</span><a id="line.1090"></a>
<span class="sourceLineNo">1091</span><a id="line.1091"></a>
<span class="sourceLineNo">1092</span><a id="line.1092">                                        }</a>
<span class="sourceLineNo">1093</span><a id="line.1093">                                        if ( ! found)</a>
<span class="sourceLineNo">1094</span><a id="line.1094">                                                // so far the residue number has tracked internal numbering.</a>
<span class="sourceLineNo">1095</span><a id="line.1095">                                                // however there are no atom records, as such this can't be a PDB residue number...</a>
<span class="sourceLineNo">1096</span><a id="line.1096">                                                seqresG.setResidueNumber(null);</a>
<span class="sourceLineNo">1097</span><a id="line.1097">                                }</a>
<span class="sourceLineNo">1098</span><a id="line.1098">                                atomChain.setSeqResGroups(seqResGroups);</a>
<span class="sourceLineNo">1099</span><a id="line.1099"></a>
<span class="sourceLineNo">1100</span><a id="line.1100">                        }</a>
<span class="sourceLineNo">1101</span><a id="line.1101">                }</a>
<span class="sourceLineNo">1102</span><a id="line.1102">        }</a>
<span class="sourceLineNo">1103</span><a id="line.1103"></a>
<span class="sourceLineNo">1104</span><a id="line.1104">        private int getInternalNr(Group atomG) {</a>
<span class="sourceLineNo">1105</span><a id="line.1105">                if ( atomG.getType().equals(GroupType.AMINOACID)) {</a>
<span class="sourceLineNo">1106</span><a id="line.1106">                        AminoAcidImpl aa = (AminoAcidImpl) atomG;</a>
<span class="sourceLineNo">1107</span><a id="line.1107">                        return new Long(aa.getId()).intValue();</a>
<span class="sourceLineNo">1108</span><a id="line.1108">                } else if ( atomG.getType().equals(GroupType.NUCLEOTIDE)) {</a>
<span class="sourceLineNo">1109</span><a id="line.1109">                        NucleotideImpl nu = (NucleotideImpl) atomG;</a>
<span class="sourceLineNo">1110</span><a id="line.1110">                        return new Long(nu.getId()).intValue();</a>
<span class="sourceLineNo">1111</span><a id="line.1111">                } else {</a>
<span class="sourceLineNo">1112</span><a id="line.1112">                        HetatomImpl he = (HetatomImpl) atomG;</a>
<span class="sourceLineNo">1113</span><a id="line.1113">                        return new Long(he.getId()).intValue();</a>
<span class="sourceLineNo">1114</span><a id="line.1114">                }</a>
<span class="sourceLineNo">1115</span><a id="line.1115">        }</a>
<span class="sourceLineNo">1116</span><a id="line.1116"></a>
<span class="sourceLineNo">1117</span><a id="line.1117">        private void addEntities(StructAsym asym) {</a>
<span class="sourceLineNo">1118</span><a id="line.1118">                int eId = 0;</a>
<span class="sourceLineNo">1119</span><a id="line.1119">                try {</a>
<span class="sourceLineNo">1120</span><a id="line.1120">                        eId = Integer.parseInt(asym.getEntity_id());</a>
<span class="sourceLineNo">1121</span><a id="line.1121">                } catch (NumberFormatException e) {</a>
<span class="sourceLineNo">1122</span><a id="line.1122">                        logger.warn("Could not parse mol_id from string {}. Will use 0 for creating Entity",asym.getEntity_id());</a>
<span class="sourceLineNo">1123</span><a id="line.1123">                }</a>
<span class="sourceLineNo">1124</span><a id="line.1124">                Entity e = getEntity(eId);</a>
<span class="sourceLineNo">1125</span><a id="line.1125"></a>
<span class="sourceLineNo">1126</span><a id="line.1126">                // for some mmCIF files like 1yrm all 3 of _entity_src_gen, _entity_src_nat and _pdbx_entity_src_syn are missing</a>
<span class="sourceLineNo">1127</span><a id="line.1127">                // we need to fill the Compounds in some other way:</a>
<span class="sourceLineNo">1128</span><a id="line.1128"></a>
<span class="sourceLineNo">1129</span><a id="line.1129">                EntityInfo entityInfo = structure.getEntityById(eId);</a>
<span class="sourceLineNo">1130</span><a id="line.1130"></a>
<span class="sourceLineNo">1131</span><a id="line.1131">                if (entityInfo==null) {</a>
<span class="sourceLineNo">1132</span><a id="line.1132">                        //logger.info("Creating new EntityInfo " + eId + " " + e.getId() + " " + e.getPdbx_description());</a>
<span class="sourceLineNo">1133</span><a id="line.1133">                        entityInfo = new EntityInfo();</a>
<span class="sourceLineNo">1134</span><a id="line.1134">                        entityInfo.setMolId(eId);</a>
<span class="sourceLineNo">1135</span><a id="line.1135">                        // we only add the compound if a polymeric one (to match what the PDB parser does)</a>
<span class="sourceLineNo">1136</span><a id="line.1136">                        if (e!=null) {</a>
<span class="sourceLineNo">1137</span><a id="line.1137">                                entityInfo.setDescription(e.getPdbx_description());</a>
<span class="sourceLineNo">1138</span><a id="line.1138"></a>
<span class="sourceLineNo">1139</span><a id="line.1139">                                EntityType eType = EntityType.entityTypeFromString(e.getType());</a>
<span class="sourceLineNo">1140</span><a id="line.1140">                                if (eType!=null) {</a>
<span class="sourceLineNo">1141</span><a id="line.1141">                                        entityInfo.setType(eType);</a>
<span class="sourceLineNo">1142</span><a id="line.1142">                                } else {</a>
<span class="sourceLineNo">1143</span><a id="line.1143">                                        logger.warn("Type '{}' is not recognised as a valid entity type for entity {}", e.getType(), eId);</a>
<span class="sourceLineNo">1144</span><a id="line.1144">                                }</a>
<span class="sourceLineNo">1145</span><a id="line.1145">                                addAncilliaryEntityData(asym, eId, e, entityInfo);</a>
<span class="sourceLineNo">1146</span><a id="line.1146">                                structure.addEntityInfo(entityInfo);</a>
<span class="sourceLineNo">1147</span><a id="line.1147">                                logger.debug("Adding Entity with entity id {} from _entity, with name: {}",eId, entityInfo.getDescription());</a>
<span class="sourceLineNo">1148</span><a id="line.1148">                        }</a>
<span class="sourceLineNo">1149</span><a id="line.1149">                }</a>
<span class="sourceLineNo">1150</span><a id="line.1150">        }</a>
<span class="sourceLineNo">1151</span><a id="line.1151"></a>
<span class="sourceLineNo">1152</span><a id="line.1152"></a>
<span class="sourceLineNo">1153</span><a id="line.1153">        /**</a>
<span class="sourceLineNo">1154</span><a id="line.1154">         * Add any extra information to the entity information.</a>
<span class="sourceLineNo">1155</span><a id="line.1155">         * @param asym</a>
<span class="sourceLineNo">1156</span><a id="line.1156">         * @param entityId</a>
<span class="sourceLineNo">1157</span><a id="line.1157">         * @param entity</a>
<span class="sourceLineNo">1158</span><a id="line.1158">         * @param entityInfo</a>
<span class="sourceLineNo">1159</span><a id="line.1159">         */</a>
<span class="sourceLineNo">1160</span><a id="line.1160">        private void addAncilliaryEntityData(StructAsym asym, int entityId, Entity entity, EntityInfo entityInfo) {</a>
<span class="sourceLineNo">1161</span><a id="line.1161">                // Loop through each of the entity types and add the corresponding data</a>
<span class="sourceLineNo">1162</span><a id="line.1162">                // We're assuming if data is duplicated between sources it is consistent</a>
<span class="sourceLineNo">1163</span><a id="line.1163">                // This is a potentially huge assumption...</a>
<span class="sourceLineNo">1164</span><a id="line.1164"></a>
<span class="sourceLineNo">1165</span><a id="line.1165"></a>
<span class="sourceLineNo">1166</span><a id="line.1166">                for (EntitySrcGen esg : entitySrcGens) {</a>
<span class="sourceLineNo">1167</span><a id="line.1167"></a>
<span class="sourceLineNo">1168</span><a id="line.1168">                        if (! esg.getEntity_id().equals(asym.getEntity_id()))</a>
<span class="sourceLineNo">1169</span><a id="line.1169">                                continue;</a>
<span class="sourceLineNo">1170</span><a id="line.1170"></a>
<span class="sourceLineNo">1171</span><a id="line.1171">                        addInformationFromESG(esg, entityId, entityInfo);</a>
<span class="sourceLineNo">1172</span><a id="line.1172"></a>
<span class="sourceLineNo">1173</span><a id="line.1173">                }</a>
<span class="sourceLineNo">1174</span><a id="line.1174"></a>
<span class="sourceLineNo">1175</span><a id="line.1175">                for (EntitySrcNat esn : entitySrcNats) {</a>
<span class="sourceLineNo">1176</span><a id="line.1176">                        if (! esn.getEntity_id().equals(asym.getEntity_id()))</a>
<span class="sourceLineNo">1177</span><a id="line.1177">                                continue;</a>
<span class="sourceLineNo">1178</span><a id="line.1178">                        addInformationFromESN(esn, entityId, entityInfo);</a>
<span class="sourceLineNo">1179</span><a id="line.1179"></a>
<span class="sourceLineNo">1180</span><a id="line.1180">                }</a>
<span class="sourceLineNo">1181</span><a id="line.1181"></a>
<span class="sourceLineNo">1182</span><a id="line.1182">                for (EntitySrcSyn ess : entitySrcSyns) {</a>
<span class="sourceLineNo">1183</span><a id="line.1183">                        if (! ess.getEntity_id().equals(asym.getEntity_id()))</a>
<span class="sourceLineNo">1184</span><a id="line.1184">                                continue;</a>
<span class="sourceLineNo">1185</span><a id="line.1185">                        addInfoFromESS(ess, entityId, entityInfo);</a>
<span class="sourceLineNo">1186</span><a id="line.1186"></a>
<span class="sourceLineNo">1187</span><a id="line.1187">                }</a>
<span class="sourceLineNo">1188</span><a id="line.1188">        }</a>
<span class="sourceLineNo">1189</span><a id="line.1189"></a>
<span class="sourceLineNo">1190</span><a id="line.1190">        /**</a>
<span class="sourceLineNo">1191</span><a id="line.1191">         * Add the information from an ESG to a compound.</a>
<span class="sourceLineNo">1192</span><a id="line.1192">         * @param entitySrcInfo</a>
<span class="sourceLineNo">1193</span><a id="line.1193">         * @param entityId</a>
<span class="sourceLineNo">1194</span><a id="line.1194">         * @param c</a>
<span class="sourceLineNo">1195</span><a id="line.1195">         */</a>
<span class="sourceLineNo">1196</span><a id="line.1196">        private void addInformationFromESG(EntitySrcGen entitySrcInfo, int entityId, EntityInfo c) {</a>
<span class="sourceLineNo">1197</span><a id="line.1197">                c.setAtcc(entitySrcInfo.getPdbx_gene_src_atcc());</a>
<span class="sourceLineNo">1198</span><a id="line.1198">                c.setCell(entitySrcInfo.getPdbx_gene_src_cell());</a>
<span class="sourceLineNo">1199</span><a id="line.1199">                c.setOrganismCommon(entitySrcInfo.getGene_src_common_name());</a>
<span class="sourceLineNo">1200</span><a id="line.1200">                c.setOrganismScientific(entitySrcInfo.getPdbx_gene_src_scientific_name());</a>
<span class="sourceLineNo">1201</span><a id="line.1201">                c.setOrganismTaxId(entitySrcInfo.getPdbx_gene_src_ncbi_taxonomy_id());</a>
<span class="sourceLineNo">1202</span><a id="line.1202">                c.setExpressionSystemTaxId(entitySrcInfo.getPdbx_host_org_ncbi_taxonomy_id());</a>
<span class="sourceLineNo">1203</span><a id="line.1203">                c.setExpressionSystem(entitySrcInfo.getPdbx_host_org_scientific_name());</a>
<span class="sourceLineNo">1204</span><a id="line.1204">        }</a>
<span class="sourceLineNo">1205</span><a id="line.1205"></a>
<span class="sourceLineNo">1206</span><a id="line.1206">        /**</a>
<span class="sourceLineNo">1207</span><a id="line.1207">         * Add the information to entity info from ESN.</a>
<span class="sourceLineNo">1208</span><a id="line.1208">         * @param esn</a>
<span class="sourceLineNo">1209</span><a id="line.1209">         * @param eId</a>
<span class="sourceLineNo">1210</span><a id="line.1210">         * @param c</a>
<span class="sourceLineNo">1211</span><a id="line.1211">         */</a>
<span class="sourceLineNo">1212</span><a id="line.1212">        private void addInformationFromESN(EntitySrcNat esn, int eId, EntityInfo c) {</a>
<span class="sourceLineNo">1213</span><a id="line.1213"></a>
<span class="sourceLineNo">1214</span><a id="line.1214">                c.setAtcc(esn.getPdbx_atcc());</a>
<span class="sourceLineNo">1215</span><a id="line.1215">                c.setCell(esn.getPdbx_cell());</a>
<span class="sourceLineNo">1216</span><a id="line.1216">                c.setOrganismCommon(esn.getCommon_name());</a>
<span class="sourceLineNo">1217</span><a id="line.1217">                c.setOrganismScientific(esn.getPdbx_organism_scientific());</a>
<span class="sourceLineNo">1218</span><a id="line.1218">                c.setOrganismTaxId(esn.getPdbx_ncbi_taxonomy_id());</a>
<span class="sourceLineNo">1219</span><a id="line.1219"></a>
<span class="sourceLineNo">1220</span><a id="line.1220">        }</a>
<span class="sourceLineNo">1221</span><a id="line.1221">        /**</a>
<span class="sourceLineNo">1222</span><a id="line.1222">         * Add the information from ESS to Entity info.</a>
<span class="sourceLineNo">1223</span><a id="line.1223">         * @param ess</a>
<span class="sourceLineNo">1224</span><a id="line.1224">         * @param eId</a>
<span class="sourceLineNo">1225</span><a id="line.1225">         * @param c</a>
<span class="sourceLineNo">1226</span><a id="line.1226">         */</a>
<span class="sourceLineNo">1227</span><a id="line.1227">        private void addInfoFromESS(EntitySrcSyn ess, int eId, EntityInfo c) {</a>
<span class="sourceLineNo">1228</span><a id="line.1228">                c.setOrganismCommon(ess.getOrganism_common_name());</a>
<span class="sourceLineNo">1229</span><a id="line.1229">                c.setOrganismScientific(ess.getOrganism_scientific());</a>
<span class="sourceLineNo">1230</span><a id="line.1230">                c.setOrganismTaxId(ess.getNcbi_taxonomy_id());</a>
<span class="sourceLineNo">1231</span><a id="line.1231"></a>
<span class="sourceLineNo">1232</span><a id="line.1232">        }</a>
<span class="sourceLineNo">1233</span><a id="line.1233"></a>
<span class="sourceLineNo">1234</span><a id="line.1234">        private void initMaps() {</a>
<span class="sourceLineNo">1235</span><a id="line.1235"></a>
<span class="sourceLineNo">1236</span><a id="line.1236"></a>
<span class="sourceLineNo">1237</span><a id="line.1237">                if (structAsyms == null || structAsyms.isEmpty()) {</a>
<span class="sourceLineNo">1238</span><a id="line.1238">                        logger.info("No _struct_asym category found in file. No asym id to entity_id mapping will be available");</a>
<span class="sourceLineNo">1239</span><a id="line.1239">                        return;</a>
<span class="sourceLineNo">1240</span><a id="line.1240">                }</a>
<span class="sourceLineNo">1241</span><a id="line.1241"></a>
<span class="sourceLineNo">1242</span><a id="line.1242">                Map&lt;String, List&lt;String&gt;&gt; entityId2asymId = new HashMap&lt;&gt;();</a>
<span class="sourceLineNo">1243</span><a id="line.1243"></a>
<span class="sourceLineNo">1244</span><a id="line.1244">                for (StructAsym asym : structAsyms) {</a>
<span class="sourceLineNo">1245</span><a id="line.1245"></a>
<span class="sourceLineNo">1246</span><a id="line.1246">                        logger.debug("Entity {} matches asym_id: {}", asym.getEntity_id(), asym.getId() );</a>
<span class="sourceLineNo">1247</span><a id="line.1247"></a>
<span class="sourceLineNo">1248</span><a id="line.1248">                        asymId2entityId.put(asym.getId(), asym.getEntity_id());</a>
<span class="sourceLineNo">1249</span><a id="line.1249"></a>
<span class="sourceLineNo">1250</span><a id="line.1250">                        if (entityId2asymId.containsKey(asym.getEntity_id())) {</a>
<span class="sourceLineNo">1251</span><a id="line.1251">                                List&lt;String&gt; asymIds = entityId2asymId.get(asym.getEntity_id());</a>
<span class="sourceLineNo">1252</span><a id="line.1252">                                asymIds.add(asym.getId());</a>
<span class="sourceLineNo">1253</span><a id="line.1253">                        } else {</a>
<span class="sourceLineNo">1254</span><a id="line.1254">                                List&lt;String&gt; asymIds = new ArrayList&lt;&gt;();</a>
<span class="sourceLineNo">1255</span><a id="line.1255">                                asymIds.add(asym.getId());</a>
<span class="sourceLineNo">1256</span><a id="line.1256">                                entityId2asymId.put(asym.getEntity_id(), asymIds);</a>
<span class="sourceLineNo">1257</span><a id="line.1257">                        }</a>
<span class="sourceLineNo">1258</span><a id="line.1258">                }</a>
<span class="sourceLineNo">1259</span><a id="line.1259"></a>
<span class="sourceLineNo">1260</span><a id="line.1260">                if (entityPolys==null || entityPolys.isEmpty()) {</a>
<span class="sourceLineNo">1261</span><a id="line.1261">                        logger.info("No _entity_poly category found in file. No asym id to author id mapping will be available for header only parsing");</a>
<span class="sourceLineNo">1262</span><a id="line.1262">                        return;</a>
<span class="sourceLineNo">1263</span><a id="line.1263">                }</a>
<span class="sourceLineNo">1264</span><a id="line.1264"></a>
<span class="sourceLineNo">1265</span><a id="line.1265">                for (EntityPoly ep:entityPolys) {</a>
<span class="sourceLineNo">1266</span><a id="line.1266">                        if (ep.getPdbx_strand_id()==null) {</a>
<span class="sourceLineNo">1267</span><a id="line.1267">                                logger.info("_entity_poly.pdbx_strand_id is null for entity {}. Won't be able to map asym ids to author ids for this entity.", ep.getEntity_id());</a>
<span class="sourceLineNo">1268</span><a id="line.1268">                                continue;</a>
<span class="sourceLineNo">1269</span><a id="line.1269">                        }</a>
<span class="sourceLineNo">1270</span><a id="line.1270">                        String[] chainNames = ep.getPdbx_strand_id().split(",");</a>
<span class="sourceLineNo">1271</span><a id="line.1271">                        List&lt;String&gt; asymIds = entityId2asymId.get(ep.getEntity_id());</a>
<span class="sourceLineNo">1272</span><a id="line.1272">                        if (chainNames.length!=asymIds.size()) {</a>
<span class="sourceLineNo">1273</span><a id="line.1273">                                logger.warn("The list of asym ids (from _struct_asym) and the list of author ids (from _entity_poly) for entity {} have different lengths! Can't provide a mapping from asym ids to author chain ids", ep.getEntity_id());</a>
<span class="sourceLineNo">1274</span><a id="line.1274">                                continue;</a>
<span class="sourceLineNo">1275</span><a id="line.1275">                        }</a>
<span class="sourceLineNo">1276</span><a id="line.1276">                        for (int i=0; i&lt;chainNames.length; i++) {</a>
<span class="sourceLineNo">1277</span><a id="line.1277">                                asymId2authorId.put(asymIds.get(i), chainNames[i]);</a>
<span class="sourceLineNo">1278</span><a id="line.1278">                        }</a>
<span class="sourceLineNo">1279</span><a id="line.1279">                }</a>
<span class="sourceLineNo">1280</span><a id="line.1280">        }</a>
<span class="sourceLineNo">1281</span><a id="line.1281"></a>
<span class="sourceLineNo">1282</span><a id="line.1282">        private void setStructNcsOps() {</a>
<span class="sourceLineNo">1283</span><a id="line.1283"></a>
<span class="sourceLineNo">1284</span><a id="line.1284">                ArrayList&lt;Matrix4d&gt; ncsOperators = new ArrayList&lt;Matrix4d&gt;();</a>
<span class="sourceLineNo">1285</span><a id="line.1285"></a>
<span class="sourceLineNo">1286</span><a id="line.1286">                for (StructNcsOper sNcsOper:structNcsOper) {</a>
<span class="sourceLineNo">1287</span><a id="line.1287"></a>
<span class="sourceLineNo">1288</span><a id="line.1288">                        if (!sNcsOper.getCode().equals("generate")) continue;</a>
<span class="sourceLineNo">1289</span><a id="line.1289"></a>
<span class="sourceLineNo">1290</span><a id="line.1290">                        try {</a>
<span class="sourceLineNo">1291</span><a id="line.1291">                                Matrix4d op = new Matrix4d();</a>
<span class="sourceLineNo">1292</span><a id="line.1292">                                op.setElement(3, 0, 0.0);</a>
<span class="sourceLineNo">1293</span><a id="line.1293">                                op.setElement(3, 1, 0.0);</a>
<span class="sourceLineNo">1294</span><a id="line.1294">                                op.setElement(3, 2, 0.0);</a>
<span class="sourceLineNo">1295</span><a id="line.1295">                                op.setElement(3, 3, 1.0);</a>
<span class="sourceLineNo">1296</span><a id="line.1296"></a>
<span class="sourceLineNo">1297</span><a id="line.1297"></a>
<span class="sourceLineNo">1298</span><a id="line.1298">                                op.setElement(0, 0, Double.parseDouble(sNcsOper.getMatrix11()));</a>
<span class="sourceLineNo">1299</span><a id="line.1299">                                op.setElement(0, 1, Double.parseDouble(sNcsOper.getMatrix12()));</a>
<span class="sourceLineNo">1300</span><a id="line.1300">                                op.setElement(0, 2, Double.parseDouble(sNcsOper.getMatrix13()));</a>
<span class="sourceLineNo">1301</span><a id="line.1301"></a>
<span class="sourceLineNo">1302</span><a id="line.1302">                                op.setElement(1, 0, Double.parseDouble(sNcsOper.getMatrix21()));</a>
<span class="sourceLineNo">1303</span><a id="line.1303">                                op.setElement(1, 1, Double.parseDouble(sNcsOper.getMatrix22()));</a>
<span class="sourceLineNo">1304</span><a id="line.1304">                                op.setElement(1, 2, Double.parseDouble(sNcsOper.getMatrix23()));</a>
<span class="sourceLineNo">1305</span><a id="line.1305"></a>
<span class="sourceLineNo">1306</span><a id="line.1306">                                op.setElement(2, 0, Double.parseDouble(sNcsOper.getMatrix31()));</a>
<span class="sourceLineNo">1307</span><a id="line.1307">                                op.setElement(2, 1, Double.parseDouble(sNcsOper.getMatrix32()));</a>
<span class="sourceLineNo">1308</span><a id="line.1308">                                op.setElement(2, 2, Double.parseDouble(sNcsOper.getMatrix33()));</a>
<span class="sourceLineNo">1309</span><a id="line.1309"></a>
<span class="sourceLineNo">1310</span><a id="line.1310">                                op.setElement(0, 3, Double.parseDouble(sNcsOper.getVector1()));</a>
<span class="sourceLineNo">1311</span><a id="line.1311">                                op.setElement(1, 3, Double.parseDouble(sNcsOper.getVector2()));</a>
<span class="sourceLineNo">1312</span><a id="line.1312">                                op.setElement(2, 3, Double.parseDouble(sNcsOper.getVector3()));</a>
<span class="sourceLineNo">1313</span><a id="line.1313"></a>
<span class="sourceLineNo">1314</span><a id="line.1314">                                ncsOperators.add(op);</a>
<span class="sourceLineNo">1315</span><a id="line.1315"></a>
<span class="sourceLineNo">1316</span><a id="line.1316">                        } catch (NumberFormatException e) {</a>
<span class="sourceLineNo">1317</span><a id="line.1317">                                logger.warn("Error parsing doubles in NCS operator list, skipping operator {}", structNcsOper.indexOf(sNcsOper)+1);</a>
<span class="sourceLineNo">1318</span><a id="line.1318">                        }</a>
<span class="sourceLineNo">1319</span><a id="line.1319"></a>
<span class="sourceLineNo">1320</span><a id="line.1320">                }</a>
<span class="sourceLineNo">1321</span><a id="line.1321"></a>
<span class="sourceLineNo">1322</span><a id="line.1322">                // we only set it if not empty, otherwise remains null</a>
<span class="sourceLineNo">1323</span><a id="line.1323">                if (ncsOperators.size()&gt;0) {</a>
<span class="sourceLineNo">1324</span><a id="line.1324">                        structure.getCrystallographicInfo().setNcsOperators(</a>
<span class="sourceLineNo">1325</span><a id="line.1325">                                        ncsOperators.toArray(new Matrix4d[ncsOperators.size()]));</a>
<span class="sourceLineNo">1326</span><a id="line.1326">                }</a>
<span class="sourceLineNo">1327</span><a id="line.1327">        }</a>
<span class="sourceLineNo">1328</span><a id="line.1328"></a>
<span class="sourceLineNo">1329</span><a id="line.1329">        private void setCrystallographicInfoMetadata() {</a>
<span class="sourceLineNo">1330</span><a id="line.1330">                if (parsedScaleMatrix!=null) {</a>
<span class="sourceLineNo">1331</span><a id="line.1331"></a>
<span class="sourceLineNo">1332</span><a id="line.1332">                        PDBCrystallographicInfo crystalInfo = structure.getCrystallographicInfo();</a>
<span class="sourceLineNo">1333</span><a id="line.1333"></a>
<span class="sourceLineNo">1334</span><a id="line.1334">                        boolean nonStd = false;</a>
<span class="sourceLineNo">1335</span><a id="line.1335">                        if (crystalInfo.getCrystalCell()!=null &amp;&amp; !crystalInfo.getCrystalCell().checkScaleMatrix(parsedScaleMatrix)) {</a>
<span class="sourceLineNo">1336</span><a id="line.1336">                                nonStd = true;</a>
<span class="sourceLineNo">1337</span><a id="line.1337">                        }</a>
<span class="sourceLineNo">1338</span><a id="line.1338"></a>
<span class="sourceLineNo">1339</span><a id="line.1339">                        crystalInfo.setNonStandardCoordFrameConvention(nonStd);</a>
<span class="sourceLineNo">1340</span><a id="line.1340">                }</a>
<span class="sourceLineNo">1341</span><a id="line.1341">        }</a>
<span class="sourceLineNo">1342</span><a id="line.1342"></a>
<span class="sourceLineNo">1343</span><a id="line.1343"></a>
<span class="sourceLineNo">1344</span><a id="line.1344">        /** This method will return the parsed protein structure, once the parsing has been finished</a>
<span class="sourceLineNo">1345</span><a id="line.1345">         *</a>
<span class="sourceLineNo">1346</span><a id="line.1346">         * @return a BioJava protein structure object</a>
<span class="sourceLineNo">1347</span><a id="line.1347">         */</a>
<span class="sourceLineNo">1348</span><a id="line.1348">        public Structure getStructure() {</a>
<span class="sourceLineNo">1349</span><a id="line.1349"></a>
<span class="sourceLineNo">1350</span><a id="line.1350">                return structure;</a>
<span class="sourceLineNo">1351</span><a id="line.1351">        }</a>
<span class="sourceLineNo">1352</span><a id="line.1352"></a>
<span class="sourceLineNo">1353</span><a id="line.1353">        @Override</a>
<span class="sourceLineNo">1354</span><a id="line.1354">        public void newDatabasePDBrevRecord(DatabasePdbrevRecord record) {</a>
<span class="sourceLineNo">1355</span><a id="line.1355"></a>
<span class="sourceLineNo">1356</span><a id="line.1356">                PDBHeader header = structure.getPDBHeader();</a>
<span class="sourceLineNo">1357</span><a id="line.1357"></a>
<span class="sourceLineNo">1358</span><a id="line.1358">                if ( header == null) {</a>
<span class="sourceLineNo">1359</span><a id="line.1359">                        header = new PDBHeader();</a>
<span class="sourceLineNo">1360</span><a id="line.1360">                        structure.setPDBHeader(header);</a>
<span class="sourceLineNo">1361</span><a id="line.1361">                }</a>
<span class="sourceLineNo">1362</span><a id="line.1362"></a>
<span class="sourceLineNo">1363</span><a id="line.1363">                List&lt;DatabasePdbrevRecord&gt; revRecords = header.getRevisionRecords();</a>
<span class="sourceLineNo">1364</span><a id="line.1364">                if ( revRecords == null) {</a>
<span class="sourceLineNo">1365</span><a id="line.1365">                        revRecords = new ArrayList&lt;DatabasePdbrevRecord&gt;();</a>
<span class="sourceLineNo">1366</span><a id="line.1366">                        header.setRevisionRecords(revRecords);</a>
<span class="sourceLineNo">1367</span><a id="line.1367">                }</a>
<span class="sourceLineNo">1368</span><a id="line.1368">                revRecords.add(record);</a>
<span class="sourceLineNo">1369</span><a id="line.1369"></a>
<span class="sourceLineNo">1370</span><a id="line.1370"></a>
<span class="sourceLineNo">1371</span><a id="line.1371">        }</a>
<span class="sourceLineNo">1372</span><a id="line.1372"></a>
<span class="sourceLineNo">1373</span><a id="line.1373"></a>
<span class="sourceLineNo">1374</span><a id="line.1374">        @Override</a>
<span class="sourceLineNo">1375</span><a id="line.1375">        public void newDatabasePDBrev(DatabasePDBrev dbrev) {</a>
<span class="sourceLineNo">1376</span><a id="line.1376"></a>
<span class="sourceLineNo">1377</span><a id="line.1377">                logger.debug("got a database revision:" + dbrev);</a>
<span class="sourceLineNo">1378</span><a id="line.1378"></a>
<span class="sourceLineNo">1379</span><a id="line.1379">                SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd",Locale.US);</a>
<span class="sourceLineNo">1380</span><a id="line.1380">                PDBHeader header = structure.getPDBHeader();</a>
<span class="sourceLineNo">1381</span><a id="line.1381"></a>
<span class="sourceLineNo">1382</span><a id="line.1382">                if ( header == null) {</a>
<span class="sourceLineNo">1383</span><a id="line.1383">                        header = new PDBHeader();</a>
<span class="sourceLineNo">1384</span><a id="line.1384">                }</a>
<span class="sourceLineNo">1385</span><a id="line.1385"></a>
<span class="sourceLineNo">1386</span><a id="line.1386">                if (dbrev.getNum().equals("1")){</a>
<span class="sourceLineNo">1387</span><a id="line.1387"></a>
<span class="sourceLineNo">1388</span><a id="line.1388">                        try {</a>
<span class="sourceLineNo">1389</span><a id="line.1389">                                Date dep = dateFormat.parse(dbrev.getDate_original());</a>
<span class="sourceLineNo">1390</span><a id="line.1390">                                header.setDepDate(dep);</a>
<span class="sourceLineNo">1391</span><a id="line.1391"></a>
<span class="sourceLineNo">1392</span><a id="line.1392">                        } catch (ParseException e){</a>
<span class="sourceLineNo">1393</span><a id="line.1393">                                logger.warn("Could not parse date string '{}', deposition date will be unavailable", dbrev.getDate_original());</a>
<span class="sourceLineNo">1394</span><a id="line.1394">                        }</a>
<span class="sourceLineNo">1395</span><a id="line.1395"></a>
<span class="sourceLineNo">1396</span><a id="line.1396">                        try {</a>
<span class="sourceLineNo">1397</span><a id="line.1397">                                Date rel = dateFormat.parse(dbrev.getDate());</a>
<span class="sourceLineNo">1398</span><a id="line.1398">                                header.setRelDate(rel);</a>
<span class="sourceLineNo">1399</span><a id="line.1399"></a>
<span class="sourceLineNo">1400</span><a id="line.1400">                        } catch (ParseException e){</a>
<span class="sourceLineNo">1401</span><a id="line.1401">                                logger.warn("Could not parse date string '{}', modification date will be unavailable", dbrev.getDate());</a>
<span class="sourceLineNo">1402</span><a id="line.1402">                        }</a>
<span class="sourceLineNo">1403</span><a id="line.1403"></a>
<span class="sourceLineNo">1404</span><a id="line.1404"></a>
<span class="sourceLineNo">1405</span><a id="line.1405">                } else {</a>
<span class="sourceLineNo">1406</span><a id="line.1406">                        try {</a>
<span class="sourceLineNo">1407</span><a id="line.1407"></a>
<span class="sourceLineNo">1408</span><a id="line.1408">                                Date mod = dateFormat.parse(dbrev.getDate());</a>
<span class="sourceLineNo">1409</span><a id="line.1409">                                header.setModDate(mod);</a>
<span class="sourceLineNo">1410</span><a id="line.1410"></a>
<span class="sourceLineNo">1411</span><a id="line.1411">                        } catch (ParseException e){</a>
<span class="sourceLineNo">1412</span><a id="line.1412">                                logger.warn("Could not parse date string '{}', modification date will be unavailable", dbrev.getDate());</a>
<span class="sourceLineNo">1413</span><a id="line.1413">                        }</a>
<span class="sourceLineNo">1414</span><a id="line.1414">                }</a>
<span class="sourceLineNo">1415</span><a id="line.1415"></a>
<span class="sourceLineNo">1416</span><a id="line.1416">                structure.setPDBHeader(header);</a>
<span class="sourceLineNo">1417</span><a id="line.1417">        }</a>
<span class="sourceLineNo">1418</span><a id="line.1418"></a>
<span class="sourceLineNo">1419</span><a id="line.1419">        @Override</a>
<span class="sourceLineNo">1420</span><a id="line.1420">        public void newPdbxAuditRevisionHistory(PdbxAuditRevisionHistory history) {</a>
<span class="sourceLineNo">1421</span><a id="line.1421"></a>
<span class="sourceLineNo">1422</span><a id="line.1422">                SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd",Locale.US);</a>
<span class="sourceLineNo">1423</span><a id="line.1423">                PDBHeader header = structure.getPDBHeader();</a>
<span class="sourceLineNo">1424</span><a id="line.1424"></a>
<span class="sourceLineNo">1425</span><a id="line.1425">                if ( header == null) {</a>
<span class="sourceLineNo">1426</span><a id="line.1426">                        header = new PDBHeader();</a>
<span class="sourceLineNo">1427</span><a id="line.1427">                }</a>
<span class="sourceLineNo">1428</span><a id="line.1428"></a>
<span class="sourceLineNo">1429</span><a id="line.1429">        // first entry in revision history is the release date</a>
<span class="sourceLineNo">1430</span><a id="line.1430">                if (history.getOrdinal().equals("1")){</a>
<span class="sourceLineNo">1431</span><a id="line.1431">                        try {</a>
<span class="sourceLineNo">1432</span><a id="line.1432">                                Date releaseDate = dateFormat.parse(history.getRevision_date());</a>
<span class="sourceLineNo">1433</span><a id="line.1433">                                header.setRelDate(releaseDate);</a>
<span class="sourceLineNo">1434</span><a id="line.1434"></a>
<span class="sourceLineNo">1435</span><a id="line.1435">                        } catch (ParseException e){</a>
<span class="sourceLineNo">1436</span><a id="line.1436">                                logger.warn("Could not parse date string '{}', release date will be unavailable", history.getRevision_date());</a>
<span class="sourceLineNo">1437</span><a id="line.1437">                        }</a>
<span class="sourceLineNo">1438</span><a id="line.1438">                } else {</a>
<span class="sourceLineNo">1439</span><a id="line.1439">                        // all other dates are revision dates;</a>
<span class="sourceLineNo">1440</span><a id="line.1440">                        // since this method may be called multiple times,</a>
<span class="sourceLineNo">1441</span><a id="line.1441">                        // the last revision date will "stick"</a>
<span class="sourceLineNo">1442</span><a id="line.1442">                        try {</a>
<span class="sourceLineNo">1443</span><a id="line.1443">                                Date revisionDate = dateFormat.parse(history.getRevision_date());</a>
<span class="sourceLineNo">1444</span><a id="line.1444">                                header.setModDate(revisionDate);</a>
<span class="sourceLineNo">1445</span><a id="line.1445">                        } catch (ParseException e){</a>
<span class="sourceLineNo">1446</span><a id="line.1446">                                logger.warn("Could not parse date string '{}', revision date will be unavailable", history.getRevision_date());</a>
<span class="sourceLineNo">1447</span><a id="line.1447">                        }</a>
<span class="sourceLineNo">1448</span><a id="line.1448">                }</a>
<span class="sourceLineNo">1449</span><a id="line.1449"></a>
<span class="sourceLineNo">1450</span><a id="line.1450">                structure.setPDBHeader(header);</a>
<span class="sourceLineNo">1451</span><a id="line.1451">        }</a>
<span class="sourceLineNo">1452</span><a id="line.1452"></a>
<span class="sourceLineNo">1453</span><a id="line.1453">        @Override</a>
<span class="sourceLineNo">1454</span><a id="line.1454">        public void newPdbxDatabaseStatus(PdbxDatabaseStatus status) {</a>
<span class="sourceLineNo">1455</span><a id="line.1455"></a>
<span class="sourceLineNo">1456</span><a id="line.1456">                // the deposition date field is only available in mmCIF 5.0</a>
<span class="sourceLineNo">1457</span><a id="line.1457"></a>
<span class="sourceLineNo">1458</span><a id="line.1458">                if (status.getRecvd_initial_deposition_date() == null) {</a>
<span class="sourceLineNo">1459</span><a id="line.1459">                        // skip this method for older mmCIF versions</a>
<span class="sourceLineNo">1460</span><a id="line.1460">                        return;</a>
<span class="sourceLineNo">1461</span><a id="line.1461">                }</a>
<span class="sourceLineNo">1462</span><a id="line.1462"></a>
<span class="sourceLineNo">1463</span><a id="line.1463">                SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd",Locale.US);</a>
<span class="sourceLineNo">1464</span><a id="line.1464">                PDBHeader header = structure.getPDBHeader();</a>
<span class="sourceLineNo">1465</span><a id="line.1465"></a>
<span class="sourceLineNo">1466</span><a id="line.1466">                if (header == null) {</a>
<span class="sourceLineNo">1467</span><a id="line.1467">                        header = new PDBHeader();</a>
<span class="sourceLineNo">1468</span><a id="line.1468">                }</a>
<span class="sourceLineNo">1469</span><a id="line.1469"></a>
<span class="sourceLineNo">1470</span><a id="line.1470">                try {</a>
<span class="sourceLineNo">1471</span><a id="line.1471">                        Date depositionDate = dateFormat.parse(status.getRecvd_initial_deposition_date());</a>
<span class="sourceLineNo">1472</span><a id="line.1472">                        header.setDepDate(depositionDate);</a>
<span class="sourceLineNo">1473</span><a id="line.1473">                } catch (ParseException e){</a>
<span class="sourceLineNo">1474</span><a id="line.1474">                        logger.warn("Could not parse date string '{}', deposition date will be unavailable", status.getRecvd_initial_deposition_date());</a>
<span class="sourceLineNo">1475</span><a id="line.1475">                }</a>
<span class="sourceLineNo">1476</span><a id="line.1476"></a>
<span class="sourceLineNo">1477</span><a id="line.1477">                structure.setPDBHeader(header);</a>
<span class="sourceLineNo">1478</span><a id="line.1478">        }</a>
<span class="sourceLineNo">1479</span><a id="line.1479"></a>
<span class="sourceLineNo">1480</span><a id="line.1480">        @Override</a>
<span class="sourceLineNo">1481</span><a id="line.1481">        public void newDatabasePDBremark(DatabasePDBremark remark) {</a>
<span class="sourceLineNo">1482</span><a id="line.1482">                //System.out.println(remark);</a>
<span class="sourceLineNo">1483</span><a id="line.1483">                String id = remark.getId();</a>
<span class="sourceLineNo">1484</span><a id="line.1484">                if (id.equals("2")){</a>
<span class="sourceLineNo">1485</span><a id="line.1485"></a>
<span class="sourceLineNo">1486</span><a id="line.1486">                        //this remark field contains the resolution information:</a>
<span class="sourceLineNo">1487</span><a id="line.1487">                        String line = remark.getText();</a>
<span class="sourceLineNo">1488</span><a id="line.1488"></a>
<span class="sourceLineNo">1489</span><a id="line.1489">                        int i = line.indexOf("ANGSTROM");</a>
<span class="sourceLineNo">1490</span><a id="line.1490">                        if ( i &gt; 5) {</a>
<span class="sourceLineNo">1491</span><a id="line.1491">                                // line contains ANGSTROM info...</a>
<span class="sourceLineNo">1492</span><a id="line.1492">                                String resolution = line.substring(i-5,i).trim();</a>
<span class="sourceLineNo">1493</span><a id="line.1493">                                // convert string to float</a>
<span class="sourceLineNo">1494</span><a id="line.1494">                                float res = 99 ;</a>
<span class="sourceLineNo">1495</span><a id="line.1495">                                try {</a>
<span class="sourceLineNo">1496</span><a id="line.1496">                                        res = Float.parseFloat(resolution);</a>
<span class="sourceLineNo">1497</span><a id="line.1497"></a>
<span class="sourceLineNo">1498</span><a id="line.1498">                                } catch (NumberFormatException e) {</a>
<span class="sourceLineNo">1499</span><a id="line.1499">                                        logger.info("could not parse resolution from line and ignoring it " + line);</a>
<span class="sourceLineNo">1500</span><a id="line.1500">                                        return ;</a>
<span class="sourceLineNo">1501</span><a id="line.1501"></a>
<span class="sourceLineNo">1502</span><a id="line.1502"></a>
<span class="sourceLineNo">1503</span><a id="line.1503">                                }</a>
<span class="sourceLineNo">1504</span><a id="line.1504">                                // support for old style header</a>
<span class="sourceLineNo">1505</span><a id="line.1505"></a>
<span class="sourceLineNo">1506</span><a id="line.1506">                                PDBHeader pdbHeader = structure.getPDBHeader();</a>
<span class="sourceLineNo">1507</span><a id="line.1507">                                pdbHeader.setResolution(res);</a>
<span class="sourceLineNo">1508</span><a id="line.1508"></a>
<span class="sourceLineNo">1509</span><a id="line.1509">                        }</a>
<span class="sourceLineNo">1510</span><a id="line.1510"></a>
<span class="sourceLineNo">1511</span><a id="line.1511">                }</a>
<span class="sourceLineNo">1512</span><a id="line.1512">        }</a>
<span class="sourceLineNo">1513</span><a id="line.1513"></a>
<span class="sourceLineNo">1514</span><a id="line.1514">        @Override</a>
<span class="sourceLineNo">1515</span><a id="line.1515">        public void newRefine(Refine r){</a>
<span class="sourceLineNo">1516</span><a id="line.1516"></a>
<span class="sourceLineNo">1517</span><a id="line.1517">                PDBHeader pdbHeader = structure.getPDBHeader();</a>
<span class="sourceLineNo">1518</span><a id="line.1518">                // RESOLUTION</a>
<span class="sourceLineNo">1519</span><a id="line.1519">                // in very rare cases (for instance hybrid methods x-ray + neutron diffraction, e.g. 3ins, 4n9m)</a>
<span class="sourceLineNo">1520</span><a id="line.1520">                // there are 2 resolution values, one for each method</a>
<span class="sourceLineNo">1521</span><a id="line.1521">                // we take the last one found so that behaviour is like in PDB file parsing</a>
<span class="sourceLineNo">1522</span><a id="line.1522">                if (pdbHeader.getResolution()!=PDBHeader.DEFAULT_RESOLUTION) {</a>
<span class="sourceLineNo">1523</span><a id="line.1523">                        logger.warn("More than 1 resolution value present, will use last one {} and discard previous {} "</a>
<span class="sourceLineNo">1524</span><a id="line.1524">                                        ,r.getLs_d_res_high(), String.format("%4.2f",pdbHeader.getResolution()));</a>
<span class="sourceLineNo">1525</span><a id="line.1525">                }</a>
<span class="sourceLineNo">1526</span><a id="line.1526">                try {</a>
<span class="sourceLineNo">1527</span><a id="line.1527">                        pdbHeader.setResolution(Float.parseFloat(r.getLs_d_res_high()));</a>
<span class="sourceLineNo">1528</span><a id="line.1528">                } catch (NumberFormatException e){</a>
<span class="sourceLineNo">1529</span><a id="line.1529">                        logger.info("Could not parse resolution from " + r.getLs_d_res_high() + " " + e.getMessage());</a>
<span class="sourceLineNo">1530</span><a id="line.1530">                }</a>
<span class="sourceLineNo">1531</span><a id="line.1531"></a>
<span class="sourceLineNo">1532</span><a id="line.1532"></a>
<span class="sourceLineNo">1533</span><a id="line.1533">                // RFREE</a>
<span class="sourceLineNo">1534</span><a id="line.1534">                if (pdbHeader.getRfree()!=PDBHeader.DEFAULT_RFREE) {</a>
<span class="sourceLineNo">1535</span><a id="line.1535">                        logger.warn("More than 1 Rfree value present, will use last one {} and discard previous {} ",</a>
<span class="sourceLineNo">1536</span><a id="line.1536">                                        r.getLs_R_factor_R_free(), String.format("%4.2f",pdbHeader.getRfree()));</a>
<span class="sourceLineNo">1537</span><a id="line.1537">                }</a>
<span class="sourceLineNo">1538</span><a id="line.1538">                if (r.getLs_R_factor_R_free()==null) {</a>
<span class="sourceLineNo">1539</span><a id="line.1539">                        // some entries like 2ifo haven't got this field at all</a>
<span class="sourceLineNo">1540</span><a id="line.1540">                        logger.info("_refine.ls_R_factor_R_free not present, not parsing Rfree value");</a>
<span class="sourceLineNo">1541</span><a id="line.1541">                } else {</a>
<span class="sourceLineNo">1542</span><a id="line.1542">                        try {</a>
<span class="sourceLineNo">1543</span><a id="line.1543">                                pdbHeader.setRfree(Float.parseFloat(r.getLs_R_factor_R_free()));</a>
<span class="sourceLineNo">1544</span><a id="line.1544">                        } catch (NumberFormatException e){</a>
<span class="sourceLineNo">1545</span><a id="line.1545">                                // no rfree present ('?') is very usual, that's why we set it to debug</a>
<span class="sourceLineNo">1546</span><a id="line.1546">                                logger.debug("Could not parse Rfree from string '{}'", r.getLs_R_factor_R_free());</a>
<span class="sourceLineNo">1547</span><a id="line.1547">                        }</a>
<span class="sourceLineNo">1548</span><a id="line.1548">                }</a>
<span class="sourceLineNo">1549</span><a id="line.1549"></a>
<span class="sourceLineNo">1550</span><a id="line.1550">                // RWORK</a>
<span class="sourceLineNo">1551</span><a id="line.1551">                if(pdbHeader.getRwork()!=PDBHeader.DEFAULT_RFREE) {</a>
<span class="sourceLineNo">1552</span><a id="line.1552">                        logger.warn("More than 1 R work value present, will use last one {} and discard previous {} ",</a>
<span class="sourceLineNo">1553</span><a id="line.1553">                                        r.getLs_R_factor_R_work(), String.format("%4.2f",pdbHeader.getRwork()));</a>
<span class="sourceLineNo">1554</span><a id="line.1554">                }</a>
<span class="sourceLineNo">1555</span><a id="line.1555">                if(r.getLs_R_factor_R_work()==null){</a>
<span class="sourceLineNo">1556</span><a id="line.1556">                        logger.info("_refine.ls_R_factor_R_work not present, not parsing R-work value");</a>
<span class="sourceLineNo">1557</span><a id="line.1557">                }</a>
<span class="sourceLineNo">1558</span><a id="line.1558">                else{</a>
<span class="sourceLineNo">1559</span><a id="line.1559">                        try{</a>
<span class="sourceLineNo">1560</span><a id="line.1560">                                pdbHeader.setRwork(Float.parseFloat(r.getLs_R_factor_R_work()));</a>
<span class="sourceLineNo">1561</span><a id="line.1561">                        }</a>
<span class="sourceLineNo">1562</span><a id="line.1562">                        catch (NumberFormatException e){</a>
<span class="sourceLineNo">1563</span><a id="line.1563">                                logger.debug("Could not parse R-work from string '{}'", r.getLs_R_factor_R_work());</a>
<span class="sourceLineNo">1564</span><a id="line.1564">                        }</a>
<span class="sourceLineNo">1565</span><a id="line.1565"></a>
<span class="sourceLineNo">1566</span><a id="line.1566">                }</a>
<span class="sourceLineNo">1567</span><a id="line.1567"></a>
<span class="sourceLineNo">1568</span><a id="line.1568">        }</a>
<span class="sourceLineNo">1569</span><a id="line.1569"></a>
<span class="sourceLineNo">1570</span><a id="line.1570"></a>
<span class="sourceLineNo">1571</span><a id="line.1571">        @Override</a>
<span class="sourceLineNo">1572</span><a id="line.1572">        public void newAuditAuthor(AuditAuthor aa){</a>
<span class="sourceLineNo">1573</span><a id="line.1573"></a>
<span class="sourceLineNo">1574</span><a id="line.1574">                String name =  aa.getName();</a>
<span class="sourceLineNo">1575</span><a id="line.1575"></a>
<span class="sourceLineNo">1576</span><a id="line.1576">                StringBuffer famName = new StringBuffer();</a>
<span class="sourceLineNo">1577</span><a id="line.1577">                StringBuffer initials = new StringBuffer();</a>
<span class="sourceLineNo">1578</span><a id="line.1578">                boolean afterComma = false;</a>
<span class="sourceLineNo">1579</span><a id="line.1579">                for ( char c: name.toCharArray()) {</a>
<span class="sourceLineNo">1580</span><a id="line.1580">                        if ( c == ' ')</a>
<span class="sourceLineNo">1581</span><a id="line.1581">                                continue;</a>
<span class="sourceLineNo">1582</span><a id="line.1582">                        if ( c == ','){</a>
<span class="sourceLineNo">1583</span><a id="line.1583">                                afterComma = true;</a>
<span class="sourceLineNo">1584</span><a id="line.1584">                                continue;</a>
<span class="sourceLineNo">1585</span><a id="line.1585">                        }</a>
<span class="sourceLineNo">1586</span><a id="line.1586"></a>
<span class="sourceLineNo">1587</span><a id="line.1587">                        if ( afterComma)</a>
<span class="sourceLineNo">1588</span><a id="line.1588">                                initials.append(c);</a>
<span class="sourceLineNo">1589</span><a id="line.1589">                        else</a>
<span class="sourceLineNo">1590</span><a id="line.1590">                                famName.append(c);</a>
<span class="sourceLineNo">1591</span><a id="line.1591">                }</a>
<span class="sourceLineNo">1592</span><a id="line.1592"></a>
<span class="sourceLineNo">1593</span><a id="line.1593">                StringBuffer newaa = new StringBuffer();</a>
<span class="sourceLineNo">1594</span><a id="line.1594">                newaa.append(initials);</a>
<span class="sourceLineNo">1595</span><a id="line.1595">                newaa.append(famName);</a>
<span class="sourceLineNo">1596</span><a id="line.1596"></a>
<span class="sourceLineNo">1597</span><a id="line.1597">                PDBHeader header = structure.getPDBHeader();</a>
<span class="sourceLineNo">1598</span><a id="line.1598">                String auth = header.getAuthors();</a>
<span class="sourceLineNo">1599</span><a id="line.1599">                if (auth == null) {</a>
<span class="sourceLineNo">1600</span><a id="line.1600">                        header.setAuthors(newaa.toString());</a>
<span class="sourceLineNo">1601</span><a id="line.1601">                }else {</a>
<span class="sourceLineNo">1602</span><a id="line.1602">                        auth += "," + newaa.toString();</a>
<span class="sourceLineNo">1603</span><a id="line.1603">                        header.setAuthors(auth);</a>
<span class="sourceLineNo">1604</span><a id="line.1604"></a>
<span class="sourceLineNo">1605</span><a id="line.1605">                }</a>
<span class="sourceLineNo">1606</span><a id="line.1606">        }</a>
<span class="sourceLineNo">1607</span><a id="line.1607"></a>
<span class="sourceLineNo">1608</span><a id="line.1608">        @Override</a>
<span class="sourceLineNo">1609</span><a id="line.1609">        public void newExptl(Exptl exptl) {</a>
<span class="sourceLineNo">1610</span><a id="line.1610"></a>
<span class="sourceLineNo">1611</span><a id="line.1611">                PDBHeader pdbHeader = structure.getPDBHeader();</a>
<span class="sourceLineNo">1612</span><a id="line.1612">                String method = exptl.getMethod();</a>
<span class="sourceLineNo">1613</span><a id="line.1613">                pdbHeader.setExperimentalTechnique(method);</a>
<span class="sourceLineNo">1614</span><a id="line.1614"></a>
<span class="sourceLineNo">1615</span><a id="line.1615">        }</a>
<span class="sourceLineNo">1616</span><a id="line.1616"></a>
<span class="sourceLineNo">1617</span><a id="line.1617">        @Override</a>
<span class="sourceLineNo">1618</span><a id="line.1618">        public void newCell(Cell cell) {</a>
<span class="sourceLineNo">1619</span><a id="line.1619"></a>
<span class="sourceLineNo">1620</span><a id="line.1620">                try {</a>
<span class="sourceLineNo">1621</span><a id="line.1621">                        float a = Float.parseFloat(cell.getLength_a());</a>
<span class="sourceLineNo">1622</span><a id="line.1622">                        float b = Float.parseFloat(cell.getLength_b());</a>
<span class="sourceLineNo">1623</span><a id="line.1623">                        float c = Float.parseFloat(cell.getLength_c());</a>
<span class="sourceLineNo">1624</span><a id="line.1624">                        float alpha = Float.parseFloat(cell.getAngle_alpha());</a>
<span class="sourceLineNo">1625</span><a id="line.1625">                        float beta = Float.parseFloat(cell.getAngle_beta());</a>
<span class="sourceLineNo">1626</span><a id="line.1626">                        float gamma = Float.parseFloat(cell.getAngle_gamma());</a>
<span class="sourceLineNo">1627</span><a id="line.1627"></a>
<span class="sourceLineNo">1628</span><a id="line.1628">                        CrystalCell xtalCell = new CrystalCell();</a>
<span class="sourceLineNo">1629</span><a id="line.1629">                        xtalCell.setA(a);</a>
<span class="sourceLineNo">1630</span><a id="line.1630">                        xtalCell.setB(b);</a>
<span class="sourceLineNo">1631</span><a id="line.1631">                        xtalCell.setC(c);</a>
<span class="sourceLineNo">1632</span><a id="line.1632">                        xtalCell.setAlpha(alpha);</a>
<span class="sourceLineNo">1633</span><a id="line.1633">                        xtalCell.setBeta(beta);</a>
<span class="sourceLineNo">1634</span><a id="line.1634">                        xtalCell.setGamma(gamma);</a>
<span class="sourceLineNo">1635</span><a id="line.1635"></a>
<span class="sourceLineNo">1636</span><a id="line.1636">                        if (!xtalCell.isCellReasonable()) {</a>
<span class="sourceLineNo">1637</span><a id="line.1637">                                // If the entry describes a structure determined by a technique other than X-ray crystallography,</a>
<span class="sourceLineNo">1638</span><a id="line.1638">                            // cell is (sometimes!) a = b = c = 1.0, alpha = beta = gamma = 90 degrees</a>
<span class="sourceLineNo">1639</span><a id="line.1639">                                // if so we don't add and CrystalCell will be null</a>
<span class="sourceLineNo">1640</span><a id="line.1640">                                logger.debug("The crystal cell read from file does not have reasonable dimensions (at least one dimension is below {}), discarding it.",</a>
<span class="sourceLineNo">1641</span><a id="line.1641">                                                CrystalCell.MIN_VALID_CELL_SIZE);</a>
<span class="sourceLineNo">1642</span><a id="line.1642">                                return;</a>
<span class="sourceLineNo">1643</span><a id="line.1643">                        }</a>
<span class="sourceLineNo">1644</span><a id="line.1644"></a>
<span class="sourceLineNo">1645</span><a id="line.1645">                        structure.getPDBHeader().getCrystallographicInfo().setCrystalCell(xtalCell);</a>
<span class="sourceLineNo">1646</span><a id="line.1646"></a>
<span class="sourceLineNo">1647</span><a id="line.1647">                } catch (NumberFormatException e){</a>
<span class="sourceLineNo">1648</span><a id="line.1648">                        structure.getPDBHeader().getCrystallographicInfo().setCrystalCell(null);</a>
<span class="sourceLineNo">1649</span><a id="line.1649">                        logger.info("could not parse some cell parameters ("+e.getMessage()+"), ignoring _cell ");</a>
<span class="sourceLineNo">1650</span><a id="line.1650">                }</a>
<span class="sourceLineNo">1651</span><a id="line.1651">        }</a>
<span class="sourceLineNo">1652</span><a id="line.1652"></a>
<span class="sourceLineNo">1653</span><a id="line.1653">        @Override</a>
<span class="sourceLineNo">1654</span><a id="line.1654">        public void newSymmetry(Symmetry symmetry) {</a>
<span class="sourceLineNo">1655</span><a id="line.1655">                String spaceGroup = symmetry.getSpace_group_name_H_M();</a>
<span class="sourceLineNo">1656</span><a id="line.1656">                SpaceGroup sg = SymoplibParser.getSpaceGroup(spaceGroup);</a>
<span class="sourceLineNo">1657</span><a id="line.1657">                if (sg==null) {</a>
<span class="sourceLineNo">1658</span><a id="line.1658">                        logger.warn("Space group '"+spaceGroup+"' not recognised as a standard space group");</a>
<span class="sourceLineNo">1659</span><a id="line.1659">                        structure.getPDBHeader().getCrystallographicInfo().setNonStandardSg(true);</a>
<span class="sourceLineNo">1660</span><a id="line.1660">                } else {</a>
<span class="sourceLineNo">1661</span><a id="line.1661">                        structure.getPDBHeader().getCrystallographicInfo().setSpaceGroup(sg);</a>
<span class="sourceLineNo">1662</span><a id="line.1662">                        structure.getPDBHeader().getCrystallographicInfo().setNonStandardSg(false);</a>
<span class="sourceLineNo">1663</span><a id="line.1663">                }</a>
<span class="sourceLineNo">1664</span><a id="line.1664">        }</a>
<span class="sourceLineNo">1665</span><a id="line.1665"></a>
<span class="sourceLineNo">1666</span><a id="line.1666">        @Override</a>
<span class="sourceLineNo">1667</span><a id="line.1667">        public void newStructNcsOper(StructNcsOper sNcsOper) {</a>
<span class="sourceLineNo">1668</span><a id="line.1668">                structNcsOper.add(sNcsOper);</a>
<span class="sourceLineNo">1669</span><a id="line.1669">        }</a>
<span class="sourceLineNo">1670</span><a id="line.1670"></a>
<span class="sourceLineNo">1671</span><a id="line.1671">        public void newAtomSites(AtomSites atomSites) {</a>
<span class="sourceLineNo">1672</span><a id="line.1672"></a>
<span class="sourceLineNo">1673</span><a id="line.1673">                try {</a>
<span class="sourceLineNo">1674</span><a id="line.1674">                        Matrix4d m = new Matrix4d(</a>
<span class="sourceLineNo">1675</span><a id="line.1675">                                Double.parseDouble(atomSites.getFract_transf_matrix11()), Double.parseDouble(atomSites.getFract_transf_matrix12()), Double.parseDouble(atomSites.getFract_transf_matrix13()), Double.parseDouble(atomSites.getFract_transf_vector1()),</a>
<span class="sourceLineNo">1676</span><a id="line.1676">                                Double.parseDouble(atomSites.getFract_transf_matrix21()), Double.parseDouble(atomSites.getFract_transf_matrix22()), Double.parseDouble(atomSites.getFract_transf_matrix23()), Double.parseDouble(atomSites.getFract_transf_vector2()),</a>
<span class="sourceLineNo">1677</span><a id="line.1677">                                Double.parseDouble(atomSites.getFract_transf_matrix31()), Double.parseDouble(atomSites.getFract_transf_matrix32()), Double.parseDouble(atomSites.getFract_transf_matrix33()), Double.parseDouble(atomSites.getFract_transf_vector3()),</a>
<span class="sourceLineNo">1678</span><a id="line.1678">                                0,0,0,1);</a>
<span class="sourceLineNo">1679</span><a id="line.1679"></a>
<span class="sourceLineNo">1680</span><a id="line.1680">                        parsedScaleMatrix = m;</a>
<span class="sourceLineNo">1681</span><a id="line.1681"></a>
<span class="sourceLineNo">1682</span><a id="line.1682">                } catch (NumberFormatException e) {</a>
<span class="sourceLineNo">1683</span><a id="line.1683">                        logger.warn("Some values in _atom_sites.fract_transf_matrix or _atom_sites.fract_transf_vector could not be parsed as numbers. Can't check whether coordinate frame convention is correct! Error: {}", e.getMessage());</a>
<span class="sourceLineNo">1684</span><a id="line.1684">                        structure.getPDBHeader().getCrystallographicInfo().setNonStandardCoordFrameConvention(false);</a>
<span class="sourceLineNo">1685</span><a id="line.1685"></a>
<span class="sourceLineNo">1686</span><a id="line.1686">                        // in this case parsedScaleMatrix stays null and can't be used in documentEnd()</a>
<span class="sourceLineNo">1687</span><a id="line.1687">                }</a>
<span class="sourceLineNo">1688</span><a id="line.1688">        }</a>
<span class="sourceLineNo">1689</span><a id="line.1689"></a>
<span class="sourceLineNo">1690</span><a id="line.1690">        @Override</a>
<span class="sourceLineNo">1691</span><a id="line.1691">        public void newStructRef(StructRef sref) {</a>
<span class="sourceLineNo">1692</span><a id="line.1692">                logger.debug(sref.toString());</a>
<span class="sourceLineNo">1693</span><a id="line.1693">                strucRefs.add(sref);</a>
<span class="sourceLineNo">1694</span><a id="line.1694">        }</a>
<span class="sourceLineNo">1695</span><a id="line.1695"></a>
<span class="sourceLineNo">1696</span><a id="line.1696">        private StructRef getStructRef(String ref_id){</a>
<span class="sourceLineNo">1697</span><a id="line.1697">                for (StructRef structRef : strucRefs) {</a>
<span class="sourceLineNo">1698</span><a id="line.1698"></a>
<span class="sourceLineNo">1699</span><a id="line.1699">                        if (structRef.getId().equals(ref_id)){</a>
<span class="sourceLineNo">1700</span><a id="line.1700">                                return structRef;</a>
<span class="sourceLineNo">1701</span><a id="line.1701">                        }</a>
<span class="sourceLineNo">1702</span><a id="line.1702"></a>
<span class="sourceLineNo">1703</span><a id="line.1703">                }</a>
<span class="sourceLineNo">1704</span><a id="line.1704">                return null;</a>
<span class="sourceLineNo">1705</span><a id="line.1705"></a>
<span class="sourceLineNo">1706</span><a id="line.1706">        }</a>
<span class="sourceLineNo">1707</span><a id="line.1707"></a>
<span class="sourceLineNo">1708</span><a id="line.1708">        /**</a>
<span class="sourceLineNo">1709</span><a id="line.1709">         * create a DBRef record from the StrucRefSeq record:</a>
<span class="sourceLineNo">1710</span><a id="line.1710">         * &lt;pre&gt;</a>
<span class="sourceLineNo">1711</span><a id="line.1711">         * PDB record                    DBREF</a>
<span class="sourceLineNo">1712</span><a id="line.1712">         * Field Name                    mmCIF Data Item</a>
<span class="sourceLineNo">1713</span><a id="line.1713">         * Section                       n.a.</a>
<span class="sourceLineNo">1714</span><a id="line.1714">         * PDB_ID_Code                   _struct_ref_seq.pdbx_PDB_id_code</a>
<span class="sourceLineNo">1715</span><a id="line.1715">         * Strand_ID                     _struct_ref_seq.pdbx_strand_id</a>
<span class="sourceLineNo">1716</span><a id="line.1716">         * Begin_Residue_Number          _struct_ref_seq.pdbx_auth_seq_align_beg</a>
<span class="sourceLineNo">1717</span><a id="line.1717">         * Begin_Ins_Code                _struct_ref_seq.pdbx_seq_align_beg_ins_code</a>
<span class="sourceLineNo">1718</span><a id="line.1718">         * End_Residue_Number            _struct_ref_seq.pdbx_auth_seq_align_end</a>
<span class="sourceLineNo">1719</span><a id="line.1719">         * End_Ins_Code                  _struct_ref_seq.pdbx_seq_align_end_ins_code</a>
<span class="sourceLineNo">1720</span><a id="line.1720">         * Database                      _struct_ref.db_name</a>
<span class="sourceLineNo">1721</span><a id="line.1721">         * Database_Accession_No         _struct_ref_seq.pdbx_db_accession</a>
<span class="sourceLineNo">1722</span><a id="line.1722">         * Database_ID_Code              _struct_ref.db_code</a>
<span class="sourceLineNo">1723</span><a id="line.1723">         * Database_Begin_Residue_Number _struct_ref_seq.db_align_beg</a>
<span class="sourceLineNo">1724</span><a id="line.1724">         * Databaes_Begin_Ins_Code       _struct_ref_seq.pdbx_db_align_beg_ins_code</a>
<span class="sourceLineNo">1725</span><a id="line.1725">         * Database_End_Residue_Number   _struct_ref_seq.db_align_end</a>
<span class="sourceLineNo">1726</span><a id="line.1726">         * Databaes_End_Ins_Code         _struct_ref_seq.pdbx_db_align_end_ins_code</a>
<span class="sourceLineNo">1727</span><a id="line.1727">         * &lt;/pre&gt;</a>
<span class="sourceLineNo">1728</span><a id="line.1728">         *</a>
<span class="sourceLineNo">1729</span><a id="line.1729">         *</a>
<span class="sourceLineNo">1730</span><a id="line.1730">         */</a>
<span class="sourceLineNo">1731</span><a id="line.1731">        @Override</a>
<span class="sourceLineNo">1732</span><a id="line.1732">        public void newStructRefSeq(StructRefSeq sref) {</a>
<span class="sourceLineNo">1733</span><a id="line.1733">                DBRef r = new DBRef();</a>
<span class="sourceLineNo">1734</span><a id="line.1734"></a>
<span class="sourceLineNo">1735</span><a id="line.1735">                r.setIdCode(sref.getPdbx_PDB_id_code());</a>
<span class="sourceLineNo">1736</span><a id="line.1736">                r.setDbAccession(sref.getPdbx_db_accession());</a>
<span class="sourceLineNo">1737</span><a id="line.1737">                r.setDbIdCode(sref.getPdbx_db_accession());</a>
<span class="sourceLineNo">1738</span><a id="line.1738"></a>
<span class="sourceLineNo">1739</span><a id="line.1739">                r.setChainName(sref.getPdbx_strand_id());</a>
<span class="sourceLineNo">1740</span><a id="line.1740">                StructRef structRef = getStructRef(sref.getRef_id());</a>
<span class="sourceLineNo">1741</span><a id="line.1741">                if (structRef == null){</a>
<span class="sourceLineNo">1742</span><a id="line.1742">                        logger.info("could not find StructRef " + sref.getRef_id() + " for StructRefSeq " + sref);</a>
<span class="sourceLineNo">1743</span><a id="line.1743">                } else {</a>
<span class="sourceLineNo">1744</span><a id="line.1744">                        r.setDatabase(structRef.getDb_name());</a>
<span class="sourceLineNo">1745</span><a id="line.1745">                        r.setDbIdCode(structRef.getDb_code());</a>
<span class="sourceLineNo">1746</span><a id="line.1746">                }</a>
<span class="sourceLineNo">1747</span><a id="line.1747"></a>
<span class="sourceLineNo">1748</span><a id="line.1748">                int seqbegin;</a>
<span class="sourceLineNo">1749</span><a id="line.1749">                int seqend;</a>
<span class="sourceLineNo">1750</span><a id="line.1750">                try{</a>
<span class="sourceLineNo">1751</span><a id="line.1751">                        seqbegin = Integer.parseInt(sref.getPdbx_auth_seq_align_beg());</a>
<span class="sourceLineNo">1752</span><a id="line.1752">                        seqend   = Integer.parseInt(sref.getPdbx_auth_seq_align_end());</a>
<span class="sourceLineNo">1753</span><a id="line.1753">                }</a>
<span class="sourceLineNo">1754</span><a id="line.1754">                catch(NumberFormatException e){</a>
<span class="sourceLineNo">1755</span><a id="line.1755">                        // this happens in a few entries, annotation error? e.g. 6eoj</a>
<span class="sourceLineNo">1756</span><a id="line.1756">                        logger.warn("Couldn't parse pdbx_auth_seq_align_beg/end in _struct_ref_seq. Will not store dbref alignment info for accession {}. Error: {}", r.getDbAccession(), e.getMessage());</a>
<span class="sourceLineNo">1757</span><a id="line.1757">                        return;</a>
<span class="sourceLineNo">1758</span><a id="line.1758">                }</a>
<span class="sourceLineNo">1759</span><a id="line.1759"></a>
<span class="sourceLineNo">1760</span><a id="line.1760">                Character begin_ins_code = ' ';</a>
<span class="sourceLineNo">1761</span><a id="line.1761">                if (sref.getPdbx_seq_align_beg_ins_code() != null ) {</a>
<span class="sourceLineNo">1762</span><a id="line.1762">                    begin_ins_code = new Character(sref.getPdbx_seq_align_beg_ins_code().charAt(0));</a>
<span class="sourceLineNo">1763</span><a id="line.1763">                }</a>
<span class="sourceLineNo">1764</span><a id="line.1764"></a>
<span class="sourceLineNo">1765</span><a id="line.1765">                Character end_ins_code = ' ';</a>
<span class="sourceLineNo">1766</span><a id="line.1766">                if (sref.getPdbx_seq_align_end_ins_code() != null) {</a>
<span class="sourceLineNo">1767</span><a id="line.1767">                    end_ins_code = new Character(sref.getPdbx_seq_align_end_ins_code().charAt(0));</a>
<span class="sourceLineNo">1768</span><a id="line.1768">                }</a>
<span class="sourceLineNo">1769</span><a id="line.1769"></a>
<span class="sourceLineNo">1770</span><a id="line.1770">                if (begin_ins_code == '?')</a>
<span class="sourceLineNo">1771</span><a id="line.1771">                        begin_ins_code = ' ';</a>
<span class="sourceLineNo">1772</span><a id="line.1772"></a>
<span class="sourceLineNo">1773</span><a id="line.1773">                if (end_ins_code == '?')</a>
<span class="sourceLineNo">1774</span><a id="line.1774">                        end_ins_code = ' ';</a>
<span class="sourceLineNo">1775</span><a id="line.1775"></a>
<span class="sourceLineNo">1776</span><a id="line.1776">                r.setSeqBegin(seqbegin);</a>
<span class="sourceLineNo">1777</span><a id="line.1777">                r.setInsertBegin(begin_ins_code);</a>
<span class="sourceLineNo">1778</span><a id="line.1778"></a>
<span class="sourceLineNo">1779</span><a id="line.1779">                r.setSeqEnd(seqend);</a>
<span class="sourceLineNo">1780</span><a id="line.1780">                r.setInsertEnd(end_ins_code);</a>
<span class="sourceLineNo">1781</span><a id="line.1781"></a>
<span class="sourceLineNo">1782</span><a id="line.1782">                int dbseqbegin = Integer.parseInt(sref.getDb_align_beg());</a>
<span class="sourceLineNo">1783</span><a id="line.1783">                int dbseqend   = Integer.parseInt(sref.getDb_align_end());</a>
<span class="sourceLineNo">1784</span><a id="line.1784"></a>
<span class="sourceLineNo">1785</span><a id="line.1785">                Character db_begin_in_code = ' ';</a>
<span class="sourceLineNo">1786</span><a id="line.1786">                if (sref.getPdbx_db_align_beg_ins_code() != null) {</a>
<span class="sourceLineNo">1787</span><a id="line.1787">                    db_begin_in_code = new Character(sref.getPdbx_db_align_beg_ins_code().charAt(0));</a>
<span class="sourceLineNo">1788</span><a id="line.1788">                }</a>
<span class="sourceLineNo">1789</span><a id="line.1789"></a>
<span class="sourceLineNo">1790</span><a id="line.1790">                Character db_end_in_code = ' ';</a>
<span class="sourceLineNo">1791</span><a id="line.1791">                if (sref.getPdbx_db_align_end_ins_code() != null) {</a>
<span class="sourceLineNo">1792</span><a id="line.1792">                    db_end_in_code = new Character(sref.getPdbx_db_align_end_ins_code().charAt(0));</a>
<span class="sourceLineNo">1793</span><a id="line.1793">                }</a>
<span class="sourceLineNo">1794</span><a id="line.1794"></a>
<span class="sourceLineNo">1795</span><a id="line.1795">                if (db_begin_in_code == '?')</a>
<span class="sourceLineNo">1796</span><a id="line.1796">                        db_begin_in_code = ' ';</a>
<span class="sourceLineNo">1797</span><a id="line.1797"></a>
<span class="sourceLineNo">1798</span><a id="line.1798">                if (db_end_in_code == '?')</a>
<span class="sourceLineNo">1799</span><a id="line.1799">                        db_end_in_code = ' ';</a>
<span class="sourceLineNo">1800</span><a id="line.1800"></a>
<span class="sourceLineNo">1801</span><a id="line.1801"></a>
<span class="sourceLineNo">1802</span><a id="line.1802">                r.setDbSeqBegin(dbseqbegin);</a>
<span class="sourceLineNo">1803</span><a id="line.1803">                r.setIdbnsBegin(db_begin_in_code);</a>
<span class="sourceLineNo">1804</span><a id="line.1804"></a>
<span class="sourceLineNo">1805</span><a id="line.1805">                r.setDbSeqEnd(dbseqend);</a>
<span class="sourceLineNo">1806</span><a id="line.1806">                r.setIdbnsEnd(db_end_in_code);</a>
<span class="sourceLineNo">1807</span><a id="line.1807"></a>
<span class="sourceLineNo">1808</span><a id="line.1808">                List&lt;DBRef&gt; dbrefs = structure.getDBRefs();</a>
<span class="sourceLineNo">1809</span><a id="line.1809">                if ( dbrefs == null)</a>
<span class="sourceLineNo">1810</span><a id="line.1810">                        dbrefs = new ArrayList&lt;DBRef&gt;();</a>
<span class="sourceLineNo">1811</span><a id="line.1811">                dbrefs.add(r);</a>
<span class="sourceLineNo">1812</span><a id="line.1812"></a>
<span class="sourceLineNo">1813</span><a id="line.1813">                logger.debug(r.toPDB());</a>
<span class="sourceLineNo">1814</span><a id="line.1814"></a>
<span class="sourceLineNo">1815</span><a id="line.1815">                structure.setDBRefs(dbrefs);</a>
<span class="sourceLineNo">1816</span><a id="line.1816"></a>
<span class="sourceLineNo">1817</span><a id="line.1817">        }</a>
<span class="sourceLineNo">1818</span><a id="line.1818"></a>
<span class="sourceLineNo">1819</span><a id="line.1819">        @Override</a>
<span class="sourceLineNo">1820</span><a id="line.1820">        public void newStructRefSeqDif(StructRefSeqDif sref) {</a>
<span class="sourceLineNo">1821</span><a id="line.1821">                sequenceDifs.add(sref);</a>
<span class="sourceLineNo">1822</span><a id="line.1822">        }</a>
<span class="sourceLineNo">1823</span><a id="line.1823"></a>
<span class="sourceLineNo">1824</span><a id="line.1824">        private Chain getEntityChain(String entity_id){</a>
<span class="sourceLineNo">1825</span><a id="line.1825"></a>
<span class="sourceLineNo">1826</span><a id="line.1826">                for (Chain chain : entityChains) {</a>
<span class="sourceLineNo">1827</span><a id="line.1827">                        if ( chain.getId().equals(entity_id)){</a>
<span class="sourceLineNo">1828</span><a id="line.1828"></a>
<span class="sourceLineNo">1829</span><a id="line.1829">                                return chain;</a>
<span class="sourceLineNo">1830</span><a id="line.1830">                        }</a>
<span class="sourceLineNo">1831</span><a id="line.1831">                }</a>
<span class="sourceLineNo">1832</span><a id="line.1832">                // does not exist yet, so create...</a>
<span class="sourceLineNo">1833</span><a id="line.1833"></a>
<span class="sourceLineNo">1834</span><a id="line.1834">                Chain   chain = new ChainImpl();</a>
<span class="sourceLineNo">1835</span><a id="line.1835">                chain.setId(entity_id);</a>
<span class="sourceLineNo">1836</span><a id="line.1836">                entityChains.add(chain);</a>
<span class="sourceLineNo">1837</span><a id="line.1837"></a>
<span class="sourceLineNo">1838</span><a id="line.1838">                return chain;</a>
<span class="sourceLineNo">1839</span><a id="line.1839"></a>
<span class="sourceLineNo">1840</span><a id="line.1840">        }</a>
<span class="sourceLineNo">1841</span><a id="line.1841"></a>
<span class="sourceLineNo">1842</span><a id="line.1842">        //private Chain getSeqResChain(String chainID){</a>
<span class="sourceLineNo">1843</span><a id="line.1843">        //      return getChainFromList(seqResChains, chainID);</a>
<span class="sourceLineNo">1844</span><a id="line.1844">        //}</a>
<span class="sourceLineNo">1845</span><a id="line.1845"></a>
<span class="sourceLineNo">1846</span><a id="line.1846"></a>
<span class="sourceLineNo">1847</span><a id="line.1847">        /**</a>
<span class="sourceLineNo">1848</span><a id="line.1848">         * Data items in the ENTITY_SRC_GEN category record details of</a>
<span class="sourceLineNo">1849</span><a id="line.1849">         * the source from which the entity was obtained in cases</a>
<span class="sourceLineNo">1850</span><a id="line.1850">         * where the source was genetically manipulated.  The</a>
<span class="sourceLineNo">1851</span><a id="line.1851">         * following are treated separately:  items pertaining to the tissue</a>
<span class="sourceLineNo">1852</span><a id="line.1852">         * from which the gene was obtained, items pertaining to the host</a>
<span class="sourceLineNo">1853</span><a id="line.1853">         * organism for gene expression and items pertaining to the actual</a>
<span class="sourceLineNo">1854</span><a id="line.1854">         * producing organism (plasmid).</a>
<span class="sourceLineNo">1855</span><a id="line.1855">         */</a>
<span class="sourceLineNo">1856</span><a id="line.1856">        @Override</a>
<span class="sourceLineNo">1857</span><a id="line.1857">        public void newEntitySrcGen(EntitySrcGen entitySrcGen){</a>
<span class="sourceLineNo">1858</span><a id="line.1858"></a>
<span class="sourceLineNo">1859</span><a id="line.1859">                // add to internal list. Map to Compound object later on...</a>
<span class="sourceLineNo">1860</span><a id="line.1860">                entitySrcGens.add(entitySrcGen);</a>
<span class="sourceLineNo">1861</span><a id="line.1861">        }</a>
<span class="sourceLineNo">1862</span><a id="line.1862"></a>
<span class="sourceLineNo">1863</span><a id="line.1863">        @Override</a>
<span class="sourceLineNo">1864</span><a id="line.1864">        public void newEntitySrcNat(EntitySrcNat entitySrcNat){</a>
<span class="sourceLineNo">1865</span><a id="line.1865"></a>
<span class="sourceLineNo">1866</span><a id="line.1866">                // add to internal list. Map to Compound object later on...</a>
<span class="sourceLineNo">1867</span><a id="line.1867">                entitySrcNats.add(entitySrcNat);</a>
<span class="sourceLineNo">1868</span><a id="line.1868">        }</a>
<span class="sourceLineNo">1869</span><a id="line.1869"></a>
<span class="sourceLineNo">1870</span><a id="line.1870">        @Override</a>
<span class="sourceLineNo">1871</span><a id="line.1871">        public void newEntitySrcSyn(EntitySrcSyn entitySrcSyn){</a>
<span class="sourceLineNo">1872</span><a id="line.1872"></a>
<span class="sourceLineNo">1873</span><a id="line.1873">                // add to internal list. Map to Compound object later on...</a>
<span class="sourceLineNo">1874</span><a id="line.1874">                entitySrcSyns.add(entitySrcSyn);</a>
<span class="sourceLineNo">1875</span><a id="line.1875">        }</a>
<span class="sourceLineNo">1876</span><a id="line.1876"></a>
<span class="sourceLineNo">1877</span><a id="line.1877">        /**</a>
<span class="sourceLineNo">1878</span><a id="line.1878">         * The EntityPolySeq object provide the amino acid sequence objects for the Entities.</a>
<span class="sourceLineNo">1879</span><a id="line.1879">         * Later on the entities are mapped to the BioJava {@link Chain} and {@link EntityInfo} objects.</a>
<span class="sourceLineNo">1880</span><a id="line.1880">         * @param epolseq the EntityPolySeq record for one amino acid</a>
<span class="sourceLineNo">1881</span><a id="line.1881">         */</a>
<span class="sourceLineNo">1882</span><a id="line.1882">        @Override</a>
<span class="sourceLineNo">1883</span><a id="line.1883">        public void newEntityPolySeq(EntityPolySeq epolseq) {</a>
<span class="sourceLineNo">1884</span><a id="line.1884"></a>
<span class="sourceLineNo">1885</span><a id="line.1885">                logger.debug("NEW entity poly seq " + epolseq);</a>
<span class="sourceLineNo">1886</span><a id="line.1886"></a>
<span class="sourceLineNo">1887</span><a id="line.1887">                int eId = -1;</a>
<span class="sourceLineNo">1888</span><a id="line.1888">                try {</a>
<span class="sourceLineNo">1889</span><a id="line.1889">                        eId = Integer.parseInt(epolseq.getEntity_id());</a>
<span class="sourceLineNo">1890</span><a id="line.1890">                } catch (NumberFormatException e) {</a>
<span class="sourceLineNo">1891</span><a id="line.1891">                        logger.warn("Could not parse entity id from EntityPolySeq: "+e.getMessage());</a>
<span class="sourceLineNo">1892</span><a id="line.1892">                }</a>
<span class="sourceLineNo">1893</span><a id="line.1893">                Entity e = getEntity(eId);</a>
<span class="sourceLineNo">1894</span><a id="line.1894"></a>
<span class="sourceLineNo">1895</span><a id="line.1895">                if (e == null){</a>
<span class="sourceLineNo">1896</span><a id="line.1896">                        logger.info("Could not find entity "+ epolseq.getEntity_id()+". Can not match sequence to it.");</a>
<span class="sourceLineNo">1897</span><a id="line.1897">                        return;</a>
<span class="sourceLineNo">1898</span><a id="line.1898">                }</a>
<span class="sourceLineNo">1899</span><a id="line.1899"></a>
<span class="sourceLineNo">1900</span><a id="line.1900">                Chain entityChain = getEntityChain(epolseq.getEntity_id());</a>
<span class="sourceLineNo">1901</span><a id="line.1901"></a>
<span class="sourceLineNo">1902</span><a id="line.1902">                // first we check through the chemcomp provider, if it fails we do some heuristics to guess the type of group</a>
<span class="sourceLineNo">1903</span><a id="line.1903">                // TODO some of this code is analogous to getNewGroup() and we should try to unify them - JD 2016-03-08</a>
<span class="sourceLineNo">1904</span><a id="line.1904"></a>
<span class="sourceLineNo">1905</span><a id="line.1905">                Group g = ChemCompGroupFactory.getGroupFromChemCompDictionary(epolseq.getMon_id());</a>
<span class="sourceLineNo">1906</span><a id="line.1906">                //int seqId = Integer.parseInt(epolseq.getNum());</a>
<span class="sourceLineNo">1907</span><a id="line.1907">                if ( g != null &amp;&amp; !g.getChemComp().isEmpty()) {</a>
<span class="sourceLineNo">1908</span><a id="line.1908">                        if ( g instanceof AminoAcidImpl) {</a>
<span class="sourceLineNo">1909</span><a id="line.1909">                                AminoAcidImpl aa = (AminoAcidImpl) g;</a>
<span class="sourceLineNo">1910</span><a id="line.1910">                                aa.setRecordType(AminoAcid.SEQRESRECORD);</a>
<span class="sourceLineNo">1911</span><a id="line.1911">                                //aa.setId(seqId);</a>
<span class="sourceLineNo">1912</span><a id="line.1912">                        }</a>
<span class="sourceLineNo">1913</span><a id="line.1913">                } else {</a>
<span class="sourceLineNo">1914</span><a id="line.1914"></a>
<span class="sourceLineNo">1915</span><a id="line.1915">                        if (epolseq.getMon_id().length()==3 &amp;&amp; StructureTools.get1LetterCodeAmino(epolseq.getMon_id())!=null){</a>
<span class="sourceLineNo">1916</span><a id="line.1916">                                AminoAcidImpl a = new AminoAcidImpl();</a>
<span class="sourceLineNo">1917</span><a id="line.1917">                                a.setRecordType(AminoAcid.SEQRESRECORD);</a>
<span class="sourceLineNo">1918</span><a id="line.1918">                                Character code1 = StructureTools.get1LetterCodeAmino(epolseq.getMon_id());</a>
<span class="sourceLineNo">1919</span><a id="line.1919">                                a.setAminoType(code1);</a>
<span class="sourceLineNo">1920</span><a id="line.1920">                                g = a;</a>
<span class="sourceLineNo">1921</span><a id="line.1921"></a>
<span class="sourceLineNo">1922</span><a id="line.1922">                        } else if ( StructureTools.isNucleotide(epolseq.getMon_id())) {</a>
<span class="sourceLineNo">1923</span><a id="line.1923">                                // the group is actually a nucleotide group...</a>
<span class="sourceLineNo">1924</span><a id="line.1924">                                NucleotideImpl n = new NucleotideImpl();</a>
<span class="sourceLineNo">1925</span><a id="line.1925">                                g = n;</a>
<span class="sourceLineNo">1926</span><a id="line.1926"></a>
<span class="sourceLineNo">1927</span><a id="line.1927">                        } else {</a>
<span class="sourceLineNo">1928</span><a id="line.1928">                                logger.debug("Residue {} {} is not a standard aminoacid or nucleotide, will create a het group for it", epolseq.getNum(),epolseq.getMon_id());</a>
<span class="sourceLineNo">1929</span><a id="line.1929">                                HetatomImpl h = new HetatomImpl();</a>
<span class="sourceLineNo">1930</span><a id="line.1930">                                g = h;</a>
<span class="sourceLineNo">1931</span><a id="line.1931"></a>
<span class="sourceLineNo">1932</span><a id="line.1932">                        }</a>
<span class="sourceLineNo">1933</span><a id="line.1933"></a>
<span class="sourceLineNo">1934</span><a id="line.1934"></a>
<span class="sourceLineNo">1935</span><a id="line.1935">                }</a>
<span class="sourceLineNo">1936</span><a id="line.1936">                // at this stage we don't know about author residue numbers (insertion codes)</a>
<span class="sourceLineNo">1937</span><a id="line.1937">                // we abuse now the ResidueNumber field setting the internal residue numbers (label_seq_id, strictly sequential and follow the seqres sequence 1 to n)</a>
<span class="sourceLineNo">1938</span><a id="line.1938">                // later the actual ResidueNumbers (author residue numbers) have to be corrected in alignSeqRes()</a>
<span class="sourceLineNo">1939</span><a id="line.1939">                g.setResidueNumber(ResidueNumber.fromString(epolseq.getNum()));</a>
<span class="sourceLineNo">1940</span><a id="line.1940"></a>
<span class="sourceLineNo">1941</span><a id="line.1941">                g.setPDBName(epolseq.getMon_id());</a>
<span class="sourceLineNo">1942</span><a id="line.1942"></a>
<span class="sourceLineNo">1943</span><a id="line.1943">                entityChain.addGroup(g);</a>
<span class="sourceLineNo">1944</span><a id="line.1944"></a>
<span class="sourceLineNo">1945</span><a id="line.1945">        }</a>
<span class="sourceLineNo">1946</span><a id="line.1946"></a>
<span class="sourceLineNo">1947</span><a id="line.1947">        @Override</a>
<span class="sourceLineNo">1948</span><a id="line.1948">        public void newPdbxPolySeqScheme(PdbxPolySeqScheme ppss) {</a>
<span class="sourceLineNo">1949</span><a id="line.1949"></a>
<span class="sourceLineNo">1950</span><a id="line.1950">                //if ( headerOnly)</a>
<span class="sourceLineNo">1951</span><a id="line.1951">                //      return;</a>
<span class="sourceLineNo">1952</span><a id="line.1952"></a>
<span class="sourceLineNo">1953</span><a id="line.1953">                // replace the group asym ids with the real PDB ids!</a>
<span class="sourceLineNo">1954</span><a id="line.1954">                // replaceGroupSeqPos(ppss);  // This might be incorrect in some pdb, to use auth_seq_id of the pdbx_poly_seq_scheme.</a>
<span class="sourceLineNo">1955</span><a id="line.1955"></a>
<span class="sourceLineNo">1956</span><a id="line.1956"></a>
<span class="sourceLineNo">1957</span><a id="line.1957">        }</a>
<span class="sourceLineNo">1958</span><a id="line.1958"></a>
<span class="sourceLineNo">1959</span><a id="line.1959"></a>
<span class="sourceLineNo">1960</span><a id="line.1960">        @Override</a>
<span class="sourceLineNo">1961</span><a id="line.1961">        public void newPdbxNonPolyScheme(PdbxNonPolyScheme ppss) {</a>
<span class="sourceLineNo">1962</span><a id="line.1962"></a>
<span class="sourceLineNo">1963</span><a id="line.1963">                //if (headerOnly)</a>
<span class="sourceLineNo">1964</span><a id="line.1964">                //      return;</a>
<span class="sourceLineNo">1965</span><a id="line.1965"></a>
<span class="sourceLineNo">1966</span><a id="line.1966">                // merge the EntityPolySeq info and the AtomSite chains into one...</a>
<span class="sourceLineNo">1967</span><a id="line.1967">                //already known ignore:</a>
<span class="sourceLineNo">1968</span><a id="line.1968"></a>
<span class="sourceLineNo">1969</span><a id="line.1969">        }</a>
<span class="sourceLineNo">1970</span><a id="line.1970"></a>
<span class="sourceLineNo">1971</span><a id="line.1971">        @Override</a>
<span class="sourceLineNo">1972</span><a id="line.1972">        public void newPdbxEntityNonPoly(PdbxEntityNonPoly pen){</a>
<span class="sourceLineNo">1973</span><a id="line.1973">                // TODO: do something with them...</a>
<span class="sourceLineNo">1974</span><a id="line.1974">                // not implemented yet...</a>
<span class="sourceLineNo">1975</span><a id="line.1975">                logger.debug(pen.getEntity_id() + " " + pen.getName() + " " + pen.getComp_id());</a>
<span class="sourceLineNo">1976</span><a id="line.1976"></a>
<span class="sourceLineNo">1977</span><a id="line.1977">        }</a>
<span class="sourceLineNo">1978</span><a id="line.1978"></a>
<span class="sourceLineNo">1979</span><a id="line.1979">        @Override</a>
<span class="sourceLineNo">1980</span><a id="line.1980">        public void newChemComp(ChemComp c) {</a>
<span class="sourceLineNo">1981</span><a id="line.1981">                // TODO: do something with them...</a>
<span class="sourceLineNo">1982</span><a id="line.1982"></a>
<span class="sourceLineNo">1983</span><a id="line.1983">        }</a>
<span class="sourceLineNo">1984</span><a id="line.1984"></a>
<span class="sourceLineNo">1985</span><a id="line.1985">        @Override</a>
<span class="sourceLineNo">1986</span><a id="line.1986">        public void newGenericData(String category, List&lt;String&gt; loopFields,</a>
<span class="sourceLineNo">1987</span><a id="line.1987">                        List&lt;String&gt; lineData) {</a>
<span class="sourceLineNo">1988</span><a id="line.1988"></a>
<span class="sourceLineNo">1989</span><a id="line.1989">                //logger.debug("unhandled category so far: " + category);</a>
<span class="sourceLineNo">1990</span><a id="line.1990">        }</a>
<span class="sourceLineNo">1991</span><a id="line.1991"></a>
<span class="sourceLineNo">1992</span><a id="line.1992">        @Override</a>
<span class="sourceLineNo">1993</span><a id="line.1993">        public FileParsingParameters getFileParsingParameters()</a>
<span class="sourceLineNo">1994</span><a id="line.1994">        {</a>
<span class="sourceLineNo">1995</span><a id="line.1995">                return params;</a>
<span class="sourceLineNo">1996</span><a id="line.1996">        }</a>
<span class="sourceLineNo">1997</span><a id="line.1997"></a>
<span class="sourceLineNo">1998</span><a id="line.1998">        @Override</a>
<span class="sourceLineNo">1999</span><a id="line.1999">        public void setFileParsingParameters(FileParsingParameters params)</a>
<span class="sourceLineNo">2000</span><a id="line.2000">        {</a>
<span class="sourceLineNo">2001</span><a id="line.2001">                this.params = params;</a>
<span class="sourceLineNo">2002</span><a id="line.2002"></a>
<span class="sourceLineNo">2003</span><a id="line.2003">        }</a>
<span class="sourceLineNo">2004</span><a id="line.2004"></a>
<span class="sourceLineNo">2005</span><a id="line.2005">        @Override</a>
<span class="sourceLineNo">2006</span><a id="line.2006">        public void newChemCompDescriptor(ChemCompDescriptor ccd) {</a>
<span class="sourceLineNo">2007</span><a id="line.2007"></a>
<span class="sourceLineNo">2008</span><a id="line.2008">                // TODO nothing happening here yet.</a>
<span class="sourceLineNo">2009</span><a id="line.2009"></a>
<span class="sourceLineNo">2010</span><a id="line.2010">        }</a>
<span class="sourceLineNo">2011</span><a id="line.2011"></a>
<span class="sourceLineNo">2012</span><a id="line.2012"></a>
<span class="sourceLineNo">2013</span><a id="line.2013"></a>
<span class="sourceLineNo">2014</span><a id="line.2014">        public List&lt;PdbxStructOperList&gt; getStructOpers() {</a>
<span class="sourceLineNo">2015</span><a id="line.2015">                return structOpers;</a>
<span class="sourceLineNo">2016</span><a id="line.2016">        }</a>
<span class="sourceLineNo">2017</span><a id="line.2017"></a>
<span class="sourceLineNo">2018</span><a id="line.2018">        @Override</a>
<span class="sourceLineNo">2019</span><a id="line.2019">        public void newPdbxStrucAssembly(PdbxStructAssembly strucAssembly) {</a>
<span class="sourceLineNo">2020</span><a id="line.2020">                strucAssemblies.add(strucAssembly);</a>
<span class="sourceLineNo">2021</span><a id="line.2021"></a>
<span class="sourceLineNo">2022</span><a id="line.2022">        }</a>
<span class="sourceLineNo">2023</span><a id="line.2023"></a>
<span class="sourceLineNo">2024</span><a id="line.2024">        public List&lt;PdbxStructAssembly&gt; getStructAssemblies(){</a>
<span class="sourceLineNo">2025</span><a id="line.2025">                return strucAssemblies;</a>
<span class="sourceLineNo">2026</span><a id="line.2026">        }</a>
<span class="sourceLineNo">2027</span><a id="line.2027"></a>
<span class="sourceLineNo">2028</span><a id="line.2028">        @Override</a>
<span class="sourceLineNo">2029</span><a id="line.2029">        public void newPdbxStrucAssemblyGen(PdbxStructAssemblyGen strucAssembly) {</a>
<span class="sourceLineNo">2030</span><a id="line.2030">                strucAssemblyGens.add(strucAssembly);</a>
<span class="sourceLineNo">2031</span><a id="line.2031"></a>
<span class="sourceLineNo">2032</span><a id="line.2032">        }</a>
<span class="sourceLineNo">2033</span><a id="line.2033"></a>
<span class="sourceLineNo">2034</span><a id="line.2034">        public List&lt;PdbxStructAssemblyGen&gt; getStructAssemblyGens(){</a>
<span class="sourceLineNo">2035</span><a id="line.2035">                return strucAssemblyGens;</a>
<span class="sourceLineNo">2036</span><a id="line.2036">        }</a>
<span class="sourceLineNo">2037</span><a id="line.2037"></a>
<span class="sourceLineNo">2038</span><a id="line.2038">        @Override</a>
<span class="sourceLineNo">2039</span><a id="line.2039">        public void newChemCompAtom(ChemCompAtom atom) {</a>
<span class="sourceLineNo">2040</span><a id="line.2040"></a>
<span class="sourceLineNo">2041</span><a id="line.2041">        }</a>
<span class="sourceLineNo">2042</span><a id="line.2042"></a>
<span class="sourceLineNo">2043</span><a id="line.2043">        @Override</a>
<span class="sourceLineNo">2044</span><a id="line.2044">        public void newPdbxChemCompIndentifier(PdbxChemCompIdentifier id) {</a>
<span class="sourceLineNo">2045</span><a id="line.2045"></a>
<span class="sourceLineNo">2046</span><a id="line.2046">        }</a>
<span class="sourceLineNo">2047</span><a id="line.2047"></a>
<span class="sourceLineNo">2048</span><a id="line.2048">        @Override</a>
<span class="sourceLineNo">2049</span><a id="line.2049">        public void newChemCompBond(ChemCompBond bond) {</a>
<span class="sourceLineNo">2050</span><a id="line.2050"></a>
<span class="sourceLineNo">2051</span><a id="line.2051">        }</a>
<span class="sourceLineNo">2052</span><a id="line.2052"></a>
<span class="sourceLineNo">2053</span><a id="line.2053">        @Override</a>
<span class="sourceLineNo">2054</span><a id="line.2054">        public void newPdbxChemCompDescriptor(PdbxChemCompDescriptor desc) {</a>
<span class="sourceLineNo">2055</span><a id="line.2055"></a>
<span class="sourceLineNo">2056</span><a id="line.2056">        }</a>
<span class="sourceLineNo">2057</span><a id="line.2057"></a>
<span class="sourceLineNo">2058</span><a id="line.2058">        @Override</a>
<span class="sourceLineNo">2059</span><a id="line.2059">        public void newStructConn(StructConn structConn) {</a>
<span class="sourceLineNo">2060</span><a id="line.2060">                this.structConn.add(structConn);</a>
<span class="sourceLineNo">2061</span><a id="line.2061">        }</a>
<span class="sourceLineNo">2062</span><a id="line.2062"></a>
<span class="sourceLineNo">2063</span><a id="line.2063">        @Override</a>
<span class="sourceLineNo">2064</span><a id="line.2064">        public void newStructSiteGen(StructSiteGen siteGen) { this.structSiteGens.add(siteGen); }</a>
<span class="sourceLineNo">2065</span><a id="line.2065"></a>
<span class="sourceLineNo">2066</span><a id="line.2066">        @Override</a>
<span class="sourceLineNo">2067</span><a id="line.2067">        public void newStructSite(StructSite structSite) {</a>
<span class="sourceLineNo">2068</span><a id="line.2068"></a>
<span class="sourceLineNo">2069</span><a id="line.2069">                if (params.isHeaderOnly()) {</a>
<span class="sourceLineNo">2070</span><a id="line.2070">                        return;</a>
<span class="sourceLineNo">2071</span><a id="line.2071">                }</a>
<span class="sourceLineNo">2072</span><a id="line.2072"></a>
<span class="sourceLineNo">2073</span><a id="line.2073">                // Simply implement the method.</a>
<span class="sourceLineNo">2074</span><a id="line.2074">                List&lt;Site&gt; sites = structure.getSites();</a>
<span class="sourceLineNo">2075</span><a id="line.2075">                if (sites == null) sites = new ArrayList&lt;Site&gt;();</a>
<span class="sourceLineNo">2076</span><a id="line.2076"></a>
<span class="sourceLineNo">2077</span><a id="line.2077">                Site site = null;</a>
<span class="sourceLineNo">2078</span><a id="line.2078">                for (Site asite : sites) {</a>
<span class="sourceLineNo">2079</span><a id="line.2079">                        if (asite.getSiteID().equals(structSite.getId())) {</a>
<span class="sourceLineNo">2080</span><a id="line.2080">                                site = asite;           // Prevent duplicate siteIds</a>
<span class="sourceLineNo">2081</span><a id="line.2081">                        }</a>
<span class="sourceLineNo">2082</span><a id="line.2082">                }</a>
<span class="sourceLineNo">2083</span><a id="line.2083">                boolean addSite = false;</a>
<span class="sourceLineNo">2084</span><a id="line.2084">                if (site == null) { site = new Site(); addSite = true; }</a>
<span class="sourceLineNo">2085</span><a id="line.2085">                site.setSiteID(structSite.getId());</a>
<span class="sourceLineNo">2086</span><a id="line.2086">                site.setDescription(structSite.getDetails());</a>
<span class="sourceLineNo">2087</span><a id="line.2087">                // site.setPdbxEvidenceCode(structSite.getPdbxEvidenceCode()); // TODO - add addition fields in Sites</a>
<span class="sourceLineNo">2088</span><a id="line.2088">                if (addSite) sites.add(site);</a>
<span class="sourceLineNo">2089</span><a id="line.2089"></a>
<span class="sourceLineNo">2090</span><a id="line.2090">                structure.setSites(sites);</a>
<span class="sourceLineNo">2091</span><a id="line.2091">        }</a>
<span class="sourceLineNo">2092</span><a id="line.2092"></a>
<span class="sourceLineNo">2093</span><a id="line.2093">        /**</a>
<span class="sourceLineNo">2094</span><a id="line.2094">         * Build sites in a BioJava Structure using the original author chain id &amp; residue numbers.</a>
<span class="sourceLineNo">2095</span><a id="line.2095">         * Sites are built from struct_site_gen records that have been parsed.</a>
<span class="sourceLineNo">2096</span><a id="line.2096">         */</a>
<span class="sourceLineNo">2097</span><a id="line.2097">        private void addSites() {</a>
<span class="sourceLineNo">2098</span><a id="line.2098">                List&lt;Site&gt; sites = structure.getSites();</a>
<span class="sourceLineNo">2099</span><a id="line.2099">                if (sites == null) sites = new ArrayList&lt;Site&gt;();</a>
<span class="sourceLineNo">2100</span><a id="line.2100"></a>
<span class="sourceLineNo">2101</span><a id="line.2101">                for (StructSiteGen siteGen : structSiteGens) {</a>
<span class="sourceLineNo">2102</span><a id="line.2102">                                // For each StructSiteGen, find the residues involved, if they exist then</a>
<span class="sourceLineNo">2103</span><a id="line.2103">                                String site_id = siteGen.getSite_id(); // multiple could be in same site.</a>
<span class="sourceLineNo">2104</span><a id="line.2104">                                if (site_id == null) site_id = "";</a>
<span class="sourceLineNo">2105</span><a id="line.2105">                                String comp_id = siteGen.getLabel_comp_id();  // PDBName</a>
<span class="sourceLineNo">2106</span><a id="line.2106"></a>
<span class="sourceLineNo">2107</span><a id="line.2107">                                // Assumption: the author chain ID and residue number for the site is consistent with the original</a>
<span class="sourceLineNo">2108</span><a id="line.2108">                                // author chain id and residue numbers.</a>
<span class="sourceLineNo">2109</span><a id="line.2109"></a>
<span class="sourceLineNo">2110</span><a id="line.2110">                        String asymId = siteGen.getLabel_asym_id(); // chain name</a>
<span class="sourceLineNo">2111</span><a id="line.2111">                        String authId = siteGen.getAuth_asym_id(); // chain Id</a>
<span class="sourceLineNo">2112</span><a id="line.2112">                                String auth_seq_id = siteGen.getAuth_seq_id(); // Res num</a>
<span class="sourceLineNo">2113</span><a id="line.2113"></a>
<span class="sourceLineNo">2114</span><a id="line.2114">                                String insCode = siteGen.getPdbx_auth_ins_code();</a>
<span class="sourceLineNo">2115</span><a id="line.2115">                                if ( insCode != null &amp;&amp; insCode.equals("?"))</a>
<span class="sourceLineNo">2116</span><a id="line.2116">                                        insCode = null;</a>
<span class="sourceLineNo">2117</span><a id="line.2117"></a>
<span class="sourceLineNo">2118</span><a id="line.2118">                                // Look for asymID = chainID and seqID = seq_ID.  Check that comp_id matches the resname.</a>
<span class="sourceLineNo">2119</span><a id="line.2119">                                Group g = null;</a>
<span class="sourceLineNo">2120</span><a id="line.2120">                                try {</a>
<span class="sourceLineNo">2121</span><a id="line.2121">                                Chain chain = structure.getChain(asymId);</a>
<span class="sourceLineNo">2122</span><a id="line.2122"></a>
<span class="sourceLineNo">2123</span><a id="line.2123">                                        if (null != chain) {</a>
<span class="sourceLineNo">2124</span><a id="line.2124">                                                try {</a>
<span class="sourceLineNo">2125</span><a id="line.2125">                                                        Character insChar = null;</a>
<span class="sourceLineNo">2126</span><a id="line.2126">                                                        if (null != insCode &amp;&amp; insCode.length() &gt; 0) insChar = insCode.charAt(0);</a>
<span class="sourceLineNo">2127</span><a id="line.2127">                                                g = chain.getGroupByPDB(new ResidueNumber(null, Integer.parseInt(auth_seq_id), insChar));</a>
<span class="sourceLineNo">2128</span><a id="line.2128">                                                } catch (NumberFormatException e) {</a>
<span class="sourceLineNo">2129</span><a id="line.2129">                                                logger.warn("Could not lookup residue : " + authId + auth_seq_id);</a>
<span class="sourceLineNo">2130</span><a id="line.2130">                                                }</a>
<span class="sourceLineNo">2131</span><a id="line.2131">                                        }</a>
<span class="sourceLineNo">2132</span><a id="line.2132">                                } catch (StructureException e) {</a>
<span class="sourceLineNo">2133</span><a id="line.2133">                                        logger.warn("Problem finding residue in site entry " + siteGen.getSite_id() + " - " + e.getMessage(), e.getMessage());</a>
<span class="sourceLineNo">2134</span><a id="line.2134">                                }</a>
<span class="sourceLineNo">2135</span><a id="line.2135"></a>
<span class="sourceLineNo">2136</span><a id="line.2136">                                if (g != null) {</a>
<span class="sourceLineNo">2137</span><a id="line.2137">                                        // 2. find the site_id, if not existing, create anew.</a>
<span class="sourceLineNo">2138</span><a id="line.2138">                                        Site site = null;</a>
<span class="sourceLineNo">2139</span><a id="line.2139">                                        for (Site asite: sites) {</a>
<span class="sourceLineNo">2140</span><a id="line.2140">                                                if (site_id.equals(asite.getSiteID())) site = asite;</a>
<span class="sourceLineNo">2141</span><a id="line.2141">                                        }</a>
<span class="sourceLineNo">2142</span><a id="line.2142"></a>
<span class="sourceLineNo">2143</span><a id="line.2143">                                        boolean addSite = false;</a>
<span class="sourceLineNo">2144</span><a id="line.2144"></a>
<span class="sourceLineNo">2145</span><a id="line.2145">                                        // 3. add this residue to the site.</a>
<span class="sourceLineNo">2146</span><a id="line.2146">                                        if (site == null) {</a>
<span class="sourceLineNo">2147</span><a id="line.2147">                                                addSite = true;</a>
<span class="sourceLineNo">2148</span><a id="line.2148">                                                site = new Site();</a>
<span class="sourceLineNo">2149</span><a id="line.2149">                                                site.setSiteID(site_id);</a>
<span class="sourceLineNo">2150</span><a id="line.2150">                                        }</a>
<span class="sourceLineNo">2151</span><a id="line.2151"></a>
<span class="sourceLineNo">2152</span><a id="line.2152">                                        List&lt;Group&gt; groups = site.getGroups();</a>
<span class="sourceLineNo">2153</span><a id="line.2153">                                        if (groups == null) groups = new ArrayList&lt;Group&gt;();</a>
<span class="sourceLineNo">2154</span><a id="line.2154"></a>
<span class="sourceLineNo">2155</span><a id="line.2155">                                        // Check the self-consistency of the residue reference from auth_seq_id and chain_id</a>
<span class="sourceLineNo">2156</span><a id="line.2156">                                        if (!comp_id.equals(g.getPDBName())) {</a>
<span class="sourceLineNo">2157</span><a id="line.2157">                                        logger.warn("comp_id doesn't match the residue at " + authId + " " + auth_seq_id + " - skipping");</a>
<span class="sourceLineNo">2158</span><a id="line.2158">                                        } else {</a>
<span class="sourceLineNo">2159</span><a id="line.2159">                                                groups.add(g);</a>
<span class="sourceLineNo">2160</span><a id="line.2160">                                                site.setGroups(groups);</a>
<span class="sourceLineNo">2161</span><a id="line.2161">                                        }</a>
<span class="sourceLineNo">2162</span><a id="line.2162">                                        if (addSite) sites.add(site);</a>
<span class="sourceLineNo">2163</span><a id="line.2163">                                }</a>
<span class="sourceLineNo">2164</span><a id="line.2164">                }</a>
<span class="sourceLineNo">2165</span><a id="line.2165">                structure.setSites(sites);</a>
<span class="sourceLineNo">2166</span><a id="line.2166">        }</a>
<span class="sourceLineNo">2167</span><a id="line.2167">}</a>




























































</pre>
</div>
</main>
</body>
</html>
