<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_162) on Fri Apr 06 16:19:08 PDT 2018 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Class org.biojava.nbio.core.sequence.ProteinSequence (biojava 5.0.0 API)</title>
<meta name="date" content="2018-04-06">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class org.biojava.nbio.core.sequence.ProteinSequence (biojava 5.0.0 API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/nbio/core/sequence/class-use/ProteinSequence.html" target="_top">Frames</a></li>
<li><a href="ProteinSequence.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Class org.biojava.nbio.core.sequence.ProteinSequence" class="title">Uses of Class<br>org.biojava.nbio.core.sequence.ProteinSequence</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.aaproperties">org.biojava.nbio.aaproperties</a></td>
<td class="colLast">
<div class="block">Set of classes that responsible for providing APIs and Command Prompt access to generate some basic physico-chemical properties of protein sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.aaproperties.profeat">org.biojava.nbio.aaproperties.profeat</a></td>
<td class="colLast">
<div class="block">Set of classes that responsible for providing APIs access to generate some PROFEAT properties of protein sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.aaproperties.profeat.convertor">org.biojava.nbio.aaproperties.profeat.convertor</a></td>
<td class="colLast">
<div class="block">Set of classes that enable the conversion protein sequences into various attributes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.util">org.biojava.nbio.core.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.genome">org.biojava.nbio.genome</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.genome.parsers.geneid">org.biojava.nbio.genome.parsers.geneid</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.genome.util">org.biojava.nbio.genome.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.ronn">org.biojava.nbio.ronn</a></td>
<td class="colLast">
<div class="block">
 BioJava provide a module biojava-protein-disorder for prediction disordered regions
 from a protein sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.cluster">org.biojava.nbio.structure.cluster</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></td>
<td class="colLast">
<div class="block">Input and Output of Structures</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.ws.hmmer">org.biojava.nbio.ws.hmmer</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.nbio.aaproperties">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/aaproperties/package-summary.html">org.biojava.nbio.aaproperties</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/aaproperties/package-summary.html">org.biojava.nbio.aaproperties</a> with parameters of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getAAComposition-org.biojava.nbio.core.sequence.ProteinSequence-">getAAComposition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getAAComposition-org.biojava.nbio.core.sequence.ProteinSequence-">getAAComposition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>
<div class="block">Returns the composition of the 20 standard amino acid in the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getAbsorbance-org.biojava.nbio.core.sequence.ProteinSequence-boolean-">getAbsorbance</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
             boolean&nbsp;assumeCysReduced)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getAbsorbance-org.biojava.nbio.core.sequence.ProteinSequence-boolean-">getAbsorbance</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
             boolean&nbsp;assumeCysReduced)</code>
<div class="block">Returns the absorbance (optical density) of sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getApliphaticIndex-org.biojava.nbio.core.sequence.ProteinSequence-">getApliphaticIndex</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getApliphaticIndex-org.biojava.nbio.core.sequence.ProteinSequence-">getApliphaticIndex</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>
<div class="block">Returns the apliphatic index of sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getAvgHydropathy-org.biojava.nbio.core.sequence.ProteinSequence-">getAvgHydropathy</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getAvgHydropathy-org.biojava.nbio.core.sequence.ProteinSequence-">getAvgHydropathy</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>
<div class="block">Returns the average hydropathy value of sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getEnrichment-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.core.sequence.compound.AminoAcidCompound-">getEnrichment</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
             <a href="../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&nbsp;aminoAcidCode)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getEnrichment-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.core.sequence.compound.AminoAcidCompound-">getEnrichment</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
             <a href="../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&nbsp;aminoAcidCode)</code>
<div class="block">Returns the composition of specified amino acid in the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getExtinctionCoefficient-org.biojava.nbio.core.sequence.ProteinSequence-boolean-">getExtinctionCoefficient</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                        boolean&nbsp;assumeCysReduced)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getExtinctionCoefficient-org.biojava.nbio.core.sequence.ProteinSequence-boolean-">getExtinctionCoefficient</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                        boolean&nbsp;assumeCysReduced)</code>
<div class="block">Returns the extinction coefficient of sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getInstabilityIndex-org.biojava.nbio.core.sequence.ProteinSequence-">getInstabilityIndex</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getInstabilityIndex-org.biojava.nbio.core.sequence.ProteinSequence-">getInstabilityIndex</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>
<div class="block">Returns the instability index of sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getIsoelectricPoint-org.biojava.nbio.core.sequence.ProteinSequence-">getIsoelectricPoint</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getIsoelectricPoint-org.biojava.nbio.core.sequence.ProteinSequence-">getIsoelectricPoint</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;seuqence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getIsoelectricPoint-org.biojava.nbio.core.sequence.ProteinSequence-boolean-">getIsoelectricPoint</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                   boolean&nbsp;useExpasyValues)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getIsoelectricPoint-org.biojava.nbio.core.sequence.ProteinSequence-boolean-">getIsoelectricPoint</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                   boolean&nbsp;useExpasyValues)</code>
<div class="block">Returns the isoelectric point of sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getMolecularWeight-org.biojava.nbio.core.sequence.ProteinSequence-">getMolecularWeight</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getMolecularWeight-org.biojava.nbio.core.sequence.ProteinSequence-">getMolecularWeight</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>
<div class="block">Returns the molecular weight of sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getMolecularWeight-org.biojava.nbio.core.sequence.ProteinSequence-java.io.File-">getMolecularWeight</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;aminoAcidCompositionFile)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getMolecularWeight-org.biojava.nbio.core.sequence.ProteinSequence-java.io.File-">getMolecularWeight</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;aminoAcidCompositionFile)</code>
<div class="block">Returns the molecular weight of sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getMolecularWeight-org.biojava.nbio.core.sequence.ProteinSequence-java.io.File-java.io.File-">getMolecularWeight</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;elementMassFile,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;aminoAcidCompositionFile)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getMolecularWeight-org.biojava.nbio.core.sequence.ProteinSequence-java.io.File-java.io.File-">getMolecularWeight</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;elementMassFile,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;aminoAcidCompositionFile)</code>
<div class="block">Returns the molecular weight of sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getMolecularWeightBasedOnXML-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.xml.AminoAcidCompositionTable-">getMolecularWeightBasedOnXML</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                            <a href="../../../../../../org/biojava/nbio/aaproperties/xml/AminoAcidCompositionTable.html" title="class in org.biojava.nbio.aaproperties.xml">AminoAcidCompositionTable</a>&nbsp;aminoAcidCompositionTable)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getMolecularWeightBasedOnXML-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.xml.AminoAcidCompositionTable-">getMolecularWeightBasedOnXML</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                            <a href="../../../../../../org/biojava/nbio/aaproperties/xml/AminoAcidCompositionTable.html" title="class in org.biojava.nbio.aaproperties.xml">AminoAcidCompositionTable</a>&nbsp;aminoAcidCompositionTable)</code>
<div class="block">Returns the molecular weight of sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getNetCharge-org.biojava.nbio.core.sequence.ProteinSequence-">getNetCharge</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getNetCharge-org.biojava.nbio.core.sequence.ProteinSequence-">getNetCharge</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getNetCharge-org.biojava.nbio.core.sequence.ProteinSequence-boolean-">getNetCharge</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
            boolean&nbsp;useExpasyValues)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getNetCharge-org.biojava.nbio.core.sequence.ProteinSequence-boolean-">getNetCharge</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
            boolean&nbsp;useExpasyValues)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">PeptidePropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html#getNetCharge-org.biojava.nbio.core.sequence.ProteinSequence-boolean-double-">getNetCharge</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
            boolean&nbsp;useExpasyValues,
            double&nbsp;pHPoint)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IPeptideProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/IPeptideProperties.html#getNetCharge-org.biojava.nbio.core.sequence.ProteinSequence-boolean-double-">getNetCharge</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
            boolean&nbsp;useExpasyValues,
            double&nbsp;pHPoint)</code>
<div class="block">Returns the net charge of sequence at pH 7.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.aaproperties.profeat">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/package-summary.html">org.biojava.nbio.aaproperties.profeat</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/package-summary.html">org.biojava.nbio.aaproperties.profeat</a> with parameters of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatProperties.html#getComposition-org.biojava.nbio.core.sequence.ProteinSequence-">getComposition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatPropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatPropertiesImpl.html#getComposition-org.biojava.nbio.core.sequence.ProteinSequence-">getComposition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">IProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.html#getComposition-org.biojava.nbio.core.sequence.ProteinSequence-">getComposition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatProperties.html#getComposition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-">getComposition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
              <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatPropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatPropertiesImpl.html#getComposition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-">getComposition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
              <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">IProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.html#getComposition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-">getComposition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
              <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatProperties.html#getComposition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.GROUPING-">getComposition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
              <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute,
              <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>&nbsp;group)</code>
<div class="block">An adaptor method which returns the composition of the specific grouping for the given attribute.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatPropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatPropertiesImpl.html#getComposition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.GROUPING-">getComposition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
              <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute,
              <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>&nbsp;group)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.html#getComposition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.GROUPING-">getComposition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
              <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute,
              <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>&nbsp;group)</code>
<div class="block">Returns the composition of the specific grouping for the given attribute.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatProperties.html#getDistributionPosition-org.biojava.nbio.core.sequence.ProteinSequence-">getDistributionPosition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatPropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatPropertiesImpl.html#getDistributionPosition-org.biojava.nbio.core.sequence.ProteinSequence-">getDistributionPosition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">IProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.html#getDistributionPosition-org.biojava.nbio.core.sequence.ProteinSequence-">getDistributionPosition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatProperties.html#getDistributionPosition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-">getDistributionPosition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatPropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatPropertiesImpl.html#getDistributionPosition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-">getDistributionPosition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">IProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.html#getDistributionPosition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-">getDistributionPosition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatProperties.html#getDistributionPosition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.GROUPING-">getDistributionPosition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>&nbsp;group)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatPropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatPropertiesImpl.html#getDistributionPosition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.GROUPING-">getDistributionPosition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>&nbsp;group)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">IProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.html#getDistributionPosition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.GROUPING-">getDistributionPosition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>&nbsp;group)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatProperties.html#getDistributionPosition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.GROUPING-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.DISTRIBUTION-">getDistributionPosition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>&nbsp;group,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a>&nbsp;distribution)</code>
<div class="block">An adaptor method which computes and return the position with respect to the sequence where the given distribution of the grouping can be found.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatPropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatPropertiesImpl.html#getDistributionPosition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.GROUPING-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.DISTRIBUTION-">getDistributionPosition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>&nbsp;group,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a>&nbsp;distribution)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.html#getDistributionPosition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.GROUPING-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.DISTRIBUTION-">getDistributionPosition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a>&nbsp;group,
                       <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a>&nbsp;distribution)</code>
<div class="block">Computes and return the position with respect to the sequence where the given distribution of the grouping can be found.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.TRANSITION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.TRANSITION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatProperties.html#getTransition-org.biojava.nbio.core.sequence.ProteinSequence-">getTransition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.TRANSITION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.TRANSITION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatPropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatPropertiesImpl.html#getTransition-org.biojava.nbio.core.sequence.ProteinSequence-">getTransition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.TRANSITION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.TRANSITION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">IProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.html#getTransition-org.biojava.nbio.core.sequence.ProteinSequence-">getTransition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.TRANSITION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.TRANSITION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatProperties.html#getTransition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-">getTransition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
             <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.TRANSITION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.TRANSITION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatPropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatPropertiesImpl.html#getTransition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-">getTransition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
             <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.TRANSITION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.TRANSITION</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">IProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.html#getTransition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-">getTransition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
             <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatProperties.html#getTransition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.TRANSITION-">getTransition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
             <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute,
             <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.TRANSITION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.TRANSITION</a>&nbsp;transition)</code>
<div class="block">An adaptor method which returns the number of transition between the specified groups for the given attribute with respect to the length of sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">ProfeatPropertiesImpl.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/ProfeatPropertiesImpl.html#getTransition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.TRANSITION-">getTransition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
             <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute,
             <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.TRANSITION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.TRANSITION</a>&nbsp;transition)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="typeNameLabel">IProfeatProperties.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.html#getTransition-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.ATTRIBUTE-org.biojava.nbio.aaproperties.profeat.IProfeatProperties.TRANSITION-">getTransition</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
             <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a>&nbsp;attribute,
             <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/IProfeatProperties.TRANSITION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.TRANSITION</a>&nbsp;transition)</code>
<div class="block">Returns the number of transition between the specified groups for the given attribute with respect to the length of sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.aaproperties.profeat.convertor">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/convertor/package-summary.html">org.biojava.nbio.aaproperties.profeat.convertor</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/aaproperties/profeat/convertor/package-summary.html">org.biojava.nbio.aaproperties.profeat.convertor</a> with parameters of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="typeNameLabel">Convertor.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/aaproperties/profeat/convertor/Convertor.html#convert-org.biojava.nbio.core.sequence.ProteinSequence-">convert</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>
<div class="block">Returns the converted sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a> that return <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><span class="typeNameLabel">TranscriptSequence.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getProteinSequence--">getProteinSequence</a></span>()</code>
<div class="block">Get the protein sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><span class="typeNameLabel">RNASequence.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/RNASequence.html#getProteinSequence--">getProteinSequence</a></span>()</code>
<div class="block">Get the ProteinSequence from the RNA sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><span class="typeNameLabel">TranscriptSequence.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getProteinSequence-org.biojava.nbio.core.sequence.transcription.TranscriptionEngine-">getProteinSequence</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava.nbio.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine)</code>
<div class="block">Get the protein sequence with user defined TranscriptEngine</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><span class="typeNameLabel">RNASequence.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/RNASequence.html#getProteinSequence-org.biojava.nbio.core.sequence.transcription.TranscriptionEngine-">getProteinSequence</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava.nbio.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine)</code>
<div class="block">Get the ProteinSequene from the RNA sequence with user defined
 transcription engine</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a> that return types with arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">TranscriptSequence.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getProteinCDSSequences--">getProteinCDSSequences</a></span>()</code>
<div class="block">Return a list of protein sequences based on each CDS sequence
 where the phase shift between two CDS sequences is assigned to the
 CDS sequence that starts the triplet.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> that return types with arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FastaReaderHelper.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/io/FastaReaderHelper.html#readFastaProteinSequence-java.io.File-">readFastaProteinSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file)</code>
<div class="block">Read a fasta file containing amino acids with setup that would handle most
 cases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FastaReaderHelper.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/io/FastaReaderHelper.html#readFastaProteinSequence-java.io.InputStream-">readFastaProteinSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;inStream)</code>
<div class="block">Read a fasta file containing amino acids with setup that would handle most
 cases.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">GenbankReaderHelper.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/io/GenbankReaderHelper.html#readGenbankProteinSequence-java.io.File-">readGenbankProteinSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file)</code>
<div class="block">Read a Genbank file containing amino acids with setup that would handle most
 cases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">GenbankReaderHelper.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/io/GenbankReaderHelper.html#readGenbankProteinSequence-java.io.File-boolean-">readGenbankProteinSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
                          boolean&nbsp;lazySequenceLoad)</code>
<div class="block">Selecting lazySequenceLoad=true will parse the Genbank file and figure out the accessionid and offsets and return sequence objects
 that can in the future read the sequence from the disk.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">GenbankReaderHelper.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/io/GenbankReaderHelper.html#readGenbankProteinSequence-java.io.InputStream-">readGenbankProteinSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;inStream)</code>
<div class="block">Read a Genbank file containing amino acids with setup that would handle most
 cases.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with parameters of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/io/CasePreservingProteinSequenceCreator.html#setLowercaseToNull-org.biojava.nbio.core.sequence.ProteinSequence-java.lang.Object:A-">setLowercaseToNull</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;seq,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>[]&nbsp;out)</code>
<div class="block">Takes a <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence"><code>ProteinSequence</code></a> which was created by a
 <a href="../../../../../../org/biojava/nbio/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io"><code>CasePreservingProteinSequenceCreator</code></a>.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with type arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">GenbankWriterHelper.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/io/GenbankWriterHelper.html#writeProteinSequence-java.io.File-java.util.Collection-">writeProteinSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
                    <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;&nbsp;proteinSequences)</code>
<div class="block">Write collection of protein sequences to a file</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">FastaWriterHelper.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/io/FastaWriterHelper.html#writeProteinSequence-java.io.File-java.util.Collection-">writeProteinSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
                    <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;&nbsp;proteinSequences)</code>
<div class="block">Write collection of protein sequences to a file</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">GenbankWriterHelper.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/io/GenbankWriterHelper.html#writeProteinSequence-java.io.OutputStream-java.util.Collection-">writeProteinSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;outputStream,
                    <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;&nbsp;proteinSequences)</code>
<div class="block">Write collection of protein sequences to a stream</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">FastaWriterHelper.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/io/FastaWriterHelper.html#writeProteinSequence-java.io.OutputStream-java.util.Collection-">writeProteinSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;outputStream,
                    <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;&nbsp;proteinSequences)</code>
<div class="block">Write collection of protein sequences to a stream</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/core/util/package-summary.html">org.biojava.nbio.core.util</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/core/util/package-summary.html">org.biojava.nbio.core.util</a> with parameters of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><span class="typeNameLabel">SequenceTools.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/util/SequenceTools.html#equalLengthSequences-org.biojava.nbio.core.sequence.ProteinSequence:A-">equalLengthSequences</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>[]&nbsp;sequences)</code>
<div class="block">A method to check whether an array of sequences contains at least two sequences having an equal length.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.genome">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/genome/package-summary.html">org.biojava.nbio.genome</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/genome/package-summary.html">org.biojava.nbio.genome</a> that return types with arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">GeneFeatureHelper.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/genome/GeneFeatureHelper.html#getProteinSequences-java.util.Collection-">getProteinSequences</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequences)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.genome.parsers.geneid">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/genome/parsers/geneid/package-summary.html">org.biojava.nbio.genome.parsers.geneid</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/genome/parsers/geneid/package-summary.html">org.biojava.nbio.genome.parsers.geneid</a> that return types with arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">GeneIDXMLReader.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/genome/parsers/geneid/GeneIDXMLReader.html#getProteinSequences--">getProteinSequences</a></span>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.genome.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/genome/util/package-summary.html">org.biojava.nbio.genome.util</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/genome/util/package-summary.html">org.biojava.nbio.genome.util</a> that return <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><span class="typeNameLabel">ProteinMappingTools.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/genome/util/ProteinMappingTools.html#convertDNAtoProteinSequence-org.biojava.nbio.core.sequence.DNASequence-">convertDNAtoProteinSequence</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;dnaSequence)</code>
<div class="block">Converts the DNA sequence to protein sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><span class="typeNameLabel">ProteinMappingTools.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/genome/util/ProteinMappingTools.html#convertDNAtoProteinSequence-java.lang.String-">convertDNAtoProteinSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dnaSequence)</code>
<div class="block">Converts the DNA sequence to protein sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.ronn">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/ronn/package-summary.html">org.biojava.nbio.ronn</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/ronn/package-summary.html">org.biojava.nbio.ronn</a> with parameters of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a></code></td>
<td class="colLast"><span class="typeNameLabel">Jronn.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/ronn/Jronn.html#convertProteinSequencetoFasta-org.biojava.nbio.core.sequence.ProteinSequence-">convertProteinSequencetoFasta</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>
<div class="block">Utility method to convert a BioJava ProteinSequence object to the FastaSequence
  object used internally in JRonn.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static float[]</code></td>
<td class="colLast"><span class="typeNameLabel">Jronn.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/ronn/Jronn.html#getDisorderScores-org.biojava.nbio.core.sequence.ProteinSequence-">getDisorderScores</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>
<div class="block">Calculates the probability value for each residue in the protein sequence,
 telling the probability that the residue belongs to disordered region.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.multiple.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a> that return types with arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#toProteinMSA-org.biojava.nbio.structure.align.multiple.MultipleAlignment-">toProteinMSA</a></span>(<a href="../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code>
<div class="block">Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid
 residues.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.cluster">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/structure/cluster/package-summary.html">org.biojava.nbio.structure.cluster</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/structure/cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> that return <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><span class="typeNameLabel">Subunit.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/cluster/Subunit.html#getProteinSequence--">getProteinSequence</a></span>()</code>
<div class="block">Get the protein sequence of the Subunit.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a> that return <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureSequenceMatcher.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/io/StructureSequenceMatcher.html#getProteinSequenceForStructure-org.biojava.nbio.structure.Structure-java.util.Map-">getProteinSequenceForStructure</a></span>(<a href="../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struct,
                              <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>,<a href="../../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;groupIndexPosition)</code>
<div class="block">Generates a ProteinSequence corresponding to the sequence of struct,
 and maintains a mapping from the sequence back to the original groups.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">FastaStructureParser.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#getSequences--">getSequences</a></span>()</code>
<div class="block">Gets the protein sequences read from the Fasta file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureSequenceMatcher.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/io/StructureSequenceMatcher.html#removeGaps-org.biojava.nbio.core.sequence.ProteinSequence-">removeGaps</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;gapped)</code>
<div class="block">Removes all gaps ('-') from a protein sequence</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a> with parameters of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#cpFastaToAfpChain-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.structure.Structure-int-">cpFastaToAfpChain</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;first,
                 <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;second,
                 <a href="../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code>
<div class="block">Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site
 <code>cpSite</code> residues to the right.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">fastaToAfpChain</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence1,
               <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence2,
               <a href="../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Returns an AFPChain corresponding to the alignment between <code>structure1</code> and <code>structure2</code>, which is given by the gapped protein sequences <code>sequence1</code> and <code>sequence2</code>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureSequenceMatcher.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/io/StructureSequenceMatcher.html#getSubstructureMatchingProteinSequence-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.structure.Structure-">getSubstructureMatchingProteinSequence</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                                      <a href="../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;wholeStructure)</code>
<div class="block">Get a substructure of <code>wholeStructure</code> containing only the <a href="../../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure"><code>Groups</code></a> that are included in
 <code>sequence</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">StructureSequenceMatcher.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/io/StructureSequenceMatcher.html#matchSequenceToStructure-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.structure.Structure-">matchSequenceToStructure</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;seq,
                        <a href="../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struct)</code>
<div class="block">Given a sequence and the corresponding Structure, get the ResidueNumber
 for each residue in the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureSequenceMatcher.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/io/StructureSequenceMatcher.html#removeGaps-org.biojava.nbio.core.sequence.ProteinSequence-">removeGaps</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;gapped)</code>
<div class="block">Removes all gaps ('-') from a protein sequence</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a> with type arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain-java.util.Map-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">fastaToAfpChain</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;&nbsp;sequences,
               <a href="../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Uses two sequences each with a corresponding structure to create an AFPChain corresponding to the alignment.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructor parameters in <a href="../../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a> with type arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#FastaStructureParser-org.biojava.nbio.core.sequence.io.FastaReader-org.biojava.nbio.structure.align.util.AtomCache-">FastaStructureParser</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/io/FastaReader.html" title="class in org.biojava.nbio.core.sequence.io">FastaReader</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;reader,
                    <a href="../../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#FastaStructureParser-java.io.File-org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface-org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface-org.biojava.nbio.structure.align.util.AtomCache-">FastaStructureParser</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
                    <a href="../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
                    <a href="../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
                    <a href="../../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#FastaStructureParser-java.io.InputStream-org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface-org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface-org.biojava.nbio.structure.align.util.AtomCache-">FastaStructureParser</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;is,
                    <a href="../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
                    <a href="../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
                    <a href="../../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.ws.hmmer">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a> in <a href="../../../../../../org/biojava/nbio/ws/hmmer/package-summary.html">org.biojava.nbio.ws.hmmer</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/ws/hmmer/package-summary.html">org.biojava.nbio.ws.hmmer</a> with parameters of type <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/SortedSet.html?is-external=true" title="class or interface in java.util">SortedSet</a>&lt;<a href="../../../../../../org/biojava/nbio/ws/hmmer/HmmerResult.html" title="class in org.biojava.nbio.ws.hmmer">HmmerResult</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">RemoteHmmerScan.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/ws/hmmer/RemoteHmmerScan.html#scan-org.biojava.nbio.core.sequence.ProteinSequence-">scan</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/SortedSet.html?is-external=true" title="class or interface in java.util">SortedSet</a>&lt;<a href="../../../../../../org/biojava/nbio/ws/hmmer/HmmerResult.html" title="class in org.biojava.nbio.ws.hmmer">HmmerResult</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">HmmerScan.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/ws/hmmer/HmmerScan.html#scan-org.biojava.nbio.core.sequence.ProteinSequence-">scan</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/SortedSet.html?is-external=true" title="class or interface in java.util">SortedSet</a>&lt;<a href="../../../../../../org/biojava/nbio/ws/hmmer/HmmerResult.html" title="class in org.biojava.nbio.ws.hmmer">HmmerResult</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">RemoteHmmerScan.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/ws/hmmer/RemoteHmmerScan.html#scan-org.biojava.nbio.core.sequence.ProteinSequence-java.net.URL-">scan</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
    <a href="https://docs.oracle.com/javase/8/docs/api/java/net/URL.html?is-external=true" title="class or interface in java.net">URL</a>&nbsp;serviceLocation)</code>
<div class="block">Scans a protein sequence for Pfam profile matches.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/nbio/core/sequence/class-use/ProteinSequence.html" target="_top">Frames</a></li>
<li><a href="ProteinSequence.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2018 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
