<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> * Created on DATE</a>
<span class="sourceLineNo">021</span><a id="line.21"> *</a>
<span class="sourceLineNo">022</span><a id="line.22"> */</a>
<span class="sourceLineNo">023</span><a id="line.23">package org.biojava.nbio.core.sequence;</a>
<span class="sourceLineNo">024</span><a id="line.24"></a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</a>
<span class="sourceLineNo">026</span><a id="line.26">import org.biojava.nbio.core.sequence.compound.DNACompoundSet;</a>
<span class="sourceLineNo">027</span><a id="line.27">import org.biojava.nbio.core.sequence.transcription.TranscriptionEngine;</a>
<span class="sourceLineNo">028</span><a id="line.28">import org.slf4j.Logger;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.slf4j.LoggerFactory;</a>
<span class="sourceLineNo">030</span><a id="line.30"></a>
<span class="sourceLineNo">031</span><a id="line.31">import java.util.ArrayList;</a>
<span class="sourceLineNo">032</span><a id="line.32">import java.util.Collections;</a>
<span class="sourceLineNo">033</span><a id="line.33">import java.util.LinkedHashMap;</a>
<span class="sourceLineNo">034</span><a id="line.34"></a>
<span class="sourceLineNo">035</span><a id="line.35">/**</a>
<span class="sourceLineNo">036</span><a id="line.36"> * This is the sequence if you want to go from a gene sequence to a protein sequence. Need to start with a</a>
<span class="sourceLineNo">037</span><a id="line.37"> * ChromosomeSequence then getting a GeneSequence and then a TranscriptSequence</a>
<span class="sourceLineNo">038</span><a id="line.38"> * @author Scooter Willis</a>
<span class="sourceLineNo">039</span><a id="line.39"> */</a>
<span class="sourceLineNo">040</span><a id="line.40">public class TranscriptSequence extends DNASequence {</a>
<span class="sourceLineNo">041</span><a id="line.41"></a>
<span class="sourceLineNo">042</span><a id="line.42">        private final static Logger logger = LoggerFactory.getLogger(TranscriptSequence.class);</a>
<span class="sourceLineNo">043</span><a id="line.43"></a>
<span class="sourceLineNo">044</span><a id="line.44">        private final ArrayList&lt;CDSSequence&gt; cdsSequenceList = new ArrayList&lt;CDSSequence&gt;();</a>
<span class="sourceLineNo">045</span><a id="line.45">        private final LinkedHashMap&lt;String, CDSSequence&gt; cdsSequenceHashMap = new LinkedHashMap&lt;String, CDSSequence&gt;();</a>
<span class="sourceLineNo">046</span><a id="line.46">        private StartCodonSequence startCodonSequence = null;</a>
<span class="sourceLineNo">047</span><a id="line.47">        private StopCodonSequence stopCodonSequence = null;</a>
<span class="sourceLineNo">048</span><a id="line.48">        private GeneSequence parentGeneSequence = null;</a>
<span class="sourceLineNo">049</span><a id="line.49"></a>
<span class="sourceLineNo">050</span><a id="line.50">        /**</a>
<span class="sourceLineNo">051</span><a id="line.51">         *</a>
<span class="sourceLineNo">052</span><a id="line.52">         * @param parentDNASequence</a>
<span class="sourceLineNo">053</span><a id="line.53">         * @param begin</a>
<span class="sourceLineNo">054</span><a id="line.54">         * @param end inclusive of end</a>
<span class="sourceLineNo">055</span><a id="line.55">         */</a>
<span class="sourceLineNo">056</span><a id="line.56">        public TranscriptSequence(GeneSequence parentDNASequence, int begin, int end) {</a>
<span class="sourceLineNo">057</span><a id="line.57">                setParentSequence(parentDNASequence);</a>
<span class="sourceLineNo">058</span><a id="line.58">                this.parentGeneSequence = parentDNASequence;</a>
<span class="sourceLineNo">059</span><a id="line.59">                setBioBegin(begin);</a>
<span class="sourceLineNo">060</span><a id="line.60">                setBioEnd(end);</a>
<span class="sourceLineNo">061</span><a id="line.61">                this.setCompoundSet(DNACompoundSet.getDNACompoundSet());</a>
<span class="sourceLineNo">062</span><a id="line.62"></a>
<span class="sourceLineNo">063</span><a id="line.63">        }</a>
<span class="sourceLineNo">064</span><a id="line.64"></a>
<span class="sourceLineNo">065</span><a id="line.65">                @Override</a>
<span class="sourceLineNo">066</span><a id="line.66">        public int getLength() {</a>
<span class="sourceLineNo">067</span><a id="line.67">                return Math.abs(this.getBioEnd() - this.getBioBegin()) + 1;</a>
<span class="sourceLineNo">068</span><a id="line.68">        }</a>
<span class="sourceLineNo">069</span><a id="line.69"></a>
<span class="sourceLineNo">070</span><a id="line.70">        /**</a>
<span class="sourceLineNo">071</span><a id="line.71">         * @return the strand</a>
<span class="sourceLineNo">072</span><a id="line.72">         */</a>
<span class="sourceLineNo">073</span><a id="line.73">        public Strand getStrand() {</a>
<span class="sourceLineNo">074</span><a id="line.74">                return parentGeneSequence.getStrand();</a>
<span class="sourceLineNo">075</span><a id="line.75">        }</a>
<span class="sourceLineNo">076</span><a id="line.76"></a>
<span class="sourceLineNo">077</span><a id="line.77">        /**</a>
<span class="sourceLineNo">078</span><a id="line.78">         * Remove a CDS or coding sequence from the transcript sequence</a>
<span class="sourceLineNo">079</span><a id="line.79">         * @param accession</a>
<span class="sourceLineNo">080</span><a id="line.80">         * @return</a>
<span class="sourceLineNo">081</span><a id="line.81">         */</a>
<span class="sourceLineNo">082</span><a id="line.82">        public CDSSequence removeCDS(String accession) {</a>
<span class="sourceLineNo">083</span><a id="line.83">                for (CDSSequence cdsSequence : cdsSequenceList) {</a>
<span class="sourceLineNo">084</span><a id="line.84">                        if (cdsSequence.getAccession().getID().equals(accession)) {</a>
<span class="sourceLineNo">085</span><a id="line.85">                                cdsSequenceList.remove(cdsSequence);</a>
<span class="sourceLineNo">086</span><a id="line.86">                                cdsSequenceHashMap.remove(accession);</a>
<span class="sourceLineNo">087</span><a id="line.87">                                return cdsSequence;</a>
<span class="sourceLineNo">088</span><a id="line.88">                        }</a>
<span class="sourceLineNo">089</span><a id="line.89">                }</a>
<span class="sourceLineNo">090</span><a id="line.90">                return null;</a>
<span class="sourceLineNo">091</span><a id="line.91">        }</a>
<span class="sourceLineNo">092</span><a id="line.92"></a>
<span class="sourceLineNo">093</span><a id="line.93">        /**</a>
<span class="sourceLineNo">094</span><a id="line.94">         * Get the CDS sequences that have been added to the TranscriptSequences</a>
<span class="sourceLineNo">095</span><a id="line.95">         * @return</a>
<span class="sourceLineNo">096</span><a id="line.96">         */</a>
<span class="sourceLineNo">097</span><a id="line.97">        public LinkedHashMap&lt;String, CDSSequence&gt; getCDSSequences() {</a>
<span class="sourceLineNo">098</span><a id="line.98">                return cdsSequenceHashMap;</a>
<span class="sourceLineNo">099</span><a id="line.99">        }</a>
<span class="sourceLineNo">100</span><a id="line.100"></a>
<span class="sourceLineNo">101</span><a id="line.101">        /**</a>
<span class="sourceLineNo">102</span><a id="line.102">         * Add a Coding Sequence region with phase to the transcript sequence</a>
<span class="sourceLineNo">103</span><a id="line.103">         * @param accession</a>
<span class="sourceLineNo">104</span><a id="line.104">         * @param begin</a>
<span class="sourceLineNo">105</span><a id="line.105">         * @param end</a>
<span class="sourceLineNo">106</span><a id="line.106">         * @param phase 0,1,2</a>
<span class="sourceLineNo">107</span><a id="line.107">         * @return</a>
<span class="sourceLineNo">108</span><a id="line.108">         */</a>
<span class="sourceLineNo">109</span><a id="line.109">        public CDSSequence addCDS(AccessionID accession, int begin, int end, int phase) throws Exception {</a>
<span class="sourceLineNo">110</span><a id="line.110">                if (cdsSequenceHashMap.containsKey(accession.getID())) {</a>
<span class="sourceLineNo">111</span><a id="line.111">                        throw new Exception("Duplicate accesion id " + accession.getID());</a>
<span class="sourceLineNo">112</span><a id="line.112">                }</a>
<span class="sourceLineNo">113</span><a id="line.113">                CDSSequence cdsSequence = new CDSSequence(this, begin, end, phase); //sense should be the same as parent</a>
<span class="sourceLineNo">114</span><a id="line.114">                cdsSequence.setAccession(accession);</a>
<span class="sourceLineNo">115</span><a id="line.115">                cdsSequenceList.add(cdsSequence);</a>
<span class="sourceLineNo">116</span><a id="line.116">                Collections.sort(cdsSequenceList, new CDSComparator());</a>
<span class="sourceLineNo">117</span><a id="line.117">                cdsSequenceHashMap.put(accession.getID(), cdsSequence);</a>
<span class="sourceLineNo">118</span><a id="line.118">                return cdsSequence;</a>
<span class="sourceLineNo">119</span><a id="line.119">        }</a>
<span class="sourceLineNo">120</span><a id="line.120"></a>
<span class="sourceLineNo">121</span><a id="line.121">        /**</a>
<span class="sourceLineNo">122</span><a id="line.122">         * http://www.sequenceontology.org/gff3.shtml</a>
<span class="sourceLineNo">123</span><a id="line.123">         * http://biowiki.org/~yam/bioe131/GFF.ppt</a>
<span class="sourceLineNo">124</span><a id="line.124">         * @return</a>
<span class="sourceLineNo">125</span><a id="line.125">         */</a>
<span class="sourceLineNo">126</span><a id="line.126">        /**</a>
<span class="sourceLineNo">127</span><a id="line.127">         * Return a list of protein sequences based on each CDS sequence</a>
<span class="sourceLineNo">128</span><a id="line.128">         * where the phase shift between two CDS sequences is assigned to the</a>
<span class="sourceLineNo">129</span><a id="line.129">         * CDS sequence that starts the triplet. This can be used to map</a>
<span class="sourceLineNo">130</span><a id="line.130">         * a CDS/exon region of a protein sequence back to the DNA sequence</a>
<span class="sourceLineNo">131</span><a id="line.131">         * If you have a protein sequence and a predicted gene you can take the</a>
<span class="sourceLineNo">132</span><a id="line.132">         * predict CDS protein sequences and align back to the protein sequence.</a>
<span class="sourceLineNo">133</span><a id="line.133">         * If you have errors in mapping the predicted protein CDS regions to</a>
<span class="sourceLineNo">134</span><a id="line.134">         * an the known protein sequence then you can identify possible errors</a>
<span class="sourceLineNo">135</span><a id="line.135">         * in the prediction</a>
<span class="sourceLineNo">136</span><a id="line.136">         *</a>
<span class="sourceLineNo">137</span><a id="line.137">         * @return</a>
<span class="sourceLineNo">138</span><a id="line.138">         */</a>
<span class="sourceLineNo">139</span><a id="line.139">        public ArrayList&lt;ProteinSequence&gt; getProteinCDSSequences() {</a>
<span class="sourceLineNo">140</span><a id="line.140">                ArrayList&lt;ProteinSequence&gt; proteinSequenceList = new ArrayList&lt;ProteinSequence&gt;();</a>
<span class="sourceLineNo">141</span><a id="line.141">                for (int i = 0; i &lt; cdsSequenceList.size(); i++) {</a>
<span class="sourceLineNo">142</span><a id="line.142">                        CDSSequence cdsSequence = cdsSequenceList.get(i);</a>
<span class="sourceLineNo">143</span><a id="line.143">                        String codingSequence = cdsSequence.getCodingSequence();</a>
<span class="sourceLineNo">144</span><a id="line.144">                        //          logger.debug("CDS {} {} = {}", getStrand(), cdsSequence.getPhase(), codingSequence);</a>
<span class="sourceLineNo">145</span><a id="line.145">                        if (this.getStrand() == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">146</span><a id="line.146">                                if (cdsSequence.phase == 1) {</a>
<span class="sourceLineNo">147</span><a id="line.147">                                        codingSequence = codingSequence.substring(1, codingSequence.length());</a>
<span class="sourceLineNo">148</span><a id="line.148">                                } else if (cdsSequence.phase == 2) {</a>
<span class="sourceLineNo">149</span><a id="line.149">                                        codingSequence = codingSequence.substring(2, codingSequence.length());</a>
<span class="sourceLineNo">150</span><a id="line.150">                                }</a>
<span class="sourceLineNo">151</span><a id="line.151">                                if (i &lt; cdsSequenceList.size() - 1) {</a>
<span class="sourceLineNo">152</span><a id="line.152">                                        CDSSequence nextCDSSequence = cdsSequenceList.get(i + 1);</a>
<span class="sourceLineNo">153</span><a id="line.153">                                        if (nextCDSSequence.phase == 1) {</a>
<span class="sourceLineNo">154</span><a id="line.154">                                                String nextCodingSequence = nextCDSSequence.getCodingSequence();</a>
<span class="sourceLineNo">155</span><a id="line.155">                                                codingSequence = codingSequence + nextCodingSequence.substring(0, 1);</a>
<span class="sourceLineNo">156</span><a id="line.156">                                        } else if (nextCDSSequence.phase == 2) {</a>
<span class="sourceLineNo">157</span><a id="line.157">                                                String nextCodingSequence = nextCDSSequence.getCodingSequence();</a>
<span class="sourceLineNo">158</span><a id="line.158">                                                codingSequence = codingSequence + nextCodingSequence.substring(0, 2);</a>
<span class="sourceLineNo">159</span><a id="line.159">                                        }</a>
<span class="sourceLineNo">160</span><a id="line.160">                                }</a>
<span class="sourceLineNo">161</span><a id="line.161">                        } else {</a>
<span class="sourceLineNo">162</span><a id="line.162">                                if (cdsSequence.phase == 1) {</a>
<span class="sourceLineNo">163</span><a id="line.163">                                        codingSequence = codingSequence.substring(1, codingSequence.length());</a>
<span class="sourceLineNo">164</span><a id="line.164">                                } else if (cdsSequence.phase == 2) {</a>
<span class="sourceLineNo">165</span><a id="line.165">                                        codingSequence = codingSequence.substring(2, codingSequence.length());</a>
<span class="sourceLineNo">166</span><a id="line.166">                                }</a>
<span class="sourceLineNo">167</span><a id="line.167">                                if (i &lt; cdsSequenceList.size() - 1) {</a>
<span class="sourceLineNo">168</span><a id="line.168">                                        CDSSequence nextCDSSequence = cdsSequenceList.get(i + 1);</a>
<span class="sourceLineNo">169</span><a id="line.169">                                        if (nextCDSSequence.phase == 1) {</a>
<span class="sourceLineNo">170</span><a id="line.170">                                                String nextCodingSequence = nextCDSSequence.getCodingSequence();</a>
<span class="sourceLineNo">171</span><a id="line.171">                                                codingSequence = codingSequence + nextCodingSequence.substring(0, 1);</a>
<span class="sourceLineNo">172</span><a id="line.172">                                        } else if (nextCDSSequence.phase == 2) {</a>
<span class="sourceLineNo">173</span><a id="line.173">                                                String nextCodingSequence = nextCDSSequence.getCodingSequence();</a>
<span class="sourceLineNo">174</span><a id="line.174">                                                codingSequence = codingSequence + nextCodingSequence.substring(0, 2);</a>
<span class="sourceLineNo">175</span><a id="line.175">                                        }</a>
<span class="sourceLineNo">176</span><a id="line.176">                                }</a>
<span class="sourceLineNo">177</span><a id="line.177">                        }</a>
<span class="sourceLineNo">178</span><a id="line.178"></a>
<span class="sourceLineNo">179</span><a id="line.179"></a>
<span class="sourceLineNo">180</span><a id="line.180">                        //    logger.debug("Coding Sequence: {}", codingSequence);</a>
<span class="sourceLineNo">181</span><a id="line.181"></a>
<span class="sourceLineNo">182</span><a id="line.182">                        DNASequence dnaCodingSequence = null;</a>
<span class="sourceLineNo">183</span><a id="line.183">                        try {</a>
<span class="sourceLineNo">184</span><a id="line.184">                                dnaCodingSequence = new DNASequence(codingSequence.toUpperCase());</a>
<span class="sourceLineNo">185</span><a id="line.185">                        } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">186</span><a id="line.186">                                // if I understand this should not happen, please correct if I'm wrong - JD 2014-10-24</a>
<span class="sourceLineNo">187</span><a id="line.187">                                logger.error("Could not create DNA coding sequence, {}. This is most likely a bug.", e.getMessage());</a>
<span class="sourceLineNo">188</span><a id="line.188">                        }</a>
<span class="sourceLineNo">189</span><a id="line.189">                        RNASequence rnaCodingSequence = dnaCodingSequence.getRNASequence(TranscriptionEngine.getDefault());</a>
<span class="sourceLineNo">190</span><a id="line.190">                        ProteinSequence proteinSequence = rnaCodingSequence.getProteinSequence(TranscriptionEngine.getDefault());</a>
<span class="sourceLineNo">191</span><a id="line.191">                        proteinSequence.setAccession(new AccessionID(cdsSequence.getAccession().getID()));</a>
<span class="sourceLineNo">192</span><a id="line.192">                        proteinSequence.setParentDNASequence(cdsSequence, 1, cdsSequence.getLength());</a>
<span class="sourceLineNo">193</span><a id="line.193">                        proteinSequenceList.add(proteinSequence);</a>
<span class="sourceLineNo">194</span><a id="line.194">                }</a>
<span class="sourceLineNo">195</span><a id="line.195">                return proteinSequenceList;</a>
<span class="sourceLineNo">196</span><a id="line.196">        }</a>
<span class="sourceLineNo">197</span><a id="line.197"></a>
<span class="sourceLineNo">198</span><a id="line.198">        /**</a>
<span class="sourceLineNo">199</span><a id="line.199">         * Get the stitched together CDS sequences then maps to the cDNA</a>
<span class="sourceLineNo">200</span><a id="line.200">         * @return</a>
<span class="sourceLineNo">201</span><a id="line.201">         */</a>
<span class="sourceLineNo">202</span><a id="line.202">        public DNASequence getDNACodingSequence() {</a>
<span class="sourceLineNo">203</span><a id="line.203">                StringBuilder sb = new StringBuilder();</a>
<span class="sourceLineNo">204</span><a id="line.204">                for (CDSSequence cdsSequence : cdsSequenceList) {</a>
<span class="sourceLineNo">205</span><a id="line.205">                        sb.append(cdsSequence.getCodingSequence());</a>
<span class="sourceLineNo">206</span><a id="line.206">                }</a>
<span class="sourceLineNo">207</span><a id="line.207"></a>
<span class="sourceLineNo">208</span><a id="line.208">                DNASequence dnaSequence = null;</a>
<span class="sourceLineNo">209</span><a id="line.209">                try {</a>
<span class="sourceLineNo">210</span><a id="line.210">                        dnaSequence = new DNASequence(sb.toString().toUpperCase());</a>
<span class="sourceLineNo">211</span><a id="line.211">                } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">212</span><a id="line.212">                        // if I understand this should not happen, please correct if I'm wrong - JD 2014-10-24</a>
<span class="sourceLineNo">213</span><a id="line.213">                        logger.error("Could not create DNA coding sequence, {}. This is most likely a bug.", e.getMessage());</a>
<span class="sourceLineNo">214</span><a id="line.214">                }</a>
<span class="sourceLineNo">215</span><a id="line.215">                dnaSequence.setAccession(new AccessionID(this.getAccession().getID()));</a>
<span class="sourceLineNo">216</span><a id="line.216">                return dnaSequence;</a>
<span class="sourceLineNo">217</span><a id="line.217">        }</a>
<span class="sourceLineNo">218</span><a id="line.218"></a>
<span class="sourceLineNo">219</span><a id="line.219">        /**</a>
<span class="sourceLineNo">220</span><a id="line.220">         * Get the protein sequence</a>
<span class="sourceLineNo">221</span><a id="line.221">         * @return</a>
<span class="sourceLineNo">222</span><a id="line.222">         */</a>
<span class="sourceLineNo">223</span><a id="line.223">        public ProteinSequence getProteinSequence() {</a>
<span class="sourceLineNo">224</span><a id="line.224">                return getProteinSequence(TranscriptionEngine.getDefault());</a>
<span class="sourceLineNo">225</span><a id="line.225">        }</a>
<span class="sourceLineNo">226</span><a id="line.226"></a>
<span class="sourceLineNo">227</span><a id="line.227">        /**</a>
<span class="sourceLineNo">228</span><a id="line.228">         * Get the protein sequence with user defined TranscriptEngine</a>
<span class="sourceLineNo">229</span><a id="line.229">         * @param engine</a>
<span class="sourceLineNo">230</span><a id="line.230">         * @return</a>
<span class="sourceLineNo">231</span><a id="line.231">         */</a>
<span class="sourceLineNo">232</span><a id="line.232">        public ProteinSequence getProteinSequence(TranscriptionEngine engine) {</a>
<span class="sourceLineNo">233</span><a id="line.233">                DNASequence dnaCodingSequence = getDNACodingSequence();</a>
<span class="sourceLineNo">234</span><a id="line.234">                RNASequence rnaCodingSequence = dnaCodingSequence.getRNASequence(engine);</a>
<span class="sourceLineNo">235</span><a id="line.235">                ProteinSequence proteinSequence = rnaCodingSequence.getProteinSequence(engine);</a>
<span class="sourceLineNo">236</span><a id="line.236">                proteinSequence.setAccession(new AccessionID(this.getAccession().getID()));</a>
<span class="sourceLineNo">237</span><a id="line.237"></a>
<span class="sourceLineNo">238</span><a id="line.238">                return proteinSequence;</a>
<span class="sourceLineNo">239</span><a id="line.239">        }</a>
<span class="sourceLineNo">240</span><a id="line.240"></a>
<span class="sourceLineNo">241</span><a id="line.241">        /**</a>
<span class="sourceLineNo">242</span><a id="line.242">         * @return the startCodonSequence</a>
<span class="sourceLineNo">243</span><a id="line.243">         */</a>
<span class="sourceLineNo">244</span><a id="line.244">        public StartCodonSequence getStartCodonSequence() {</a>
<span class="sourceLineNo">245</span><a id="line.245">                return startCodonSequence;</a>
<span class="sourceLineNo">246</span><a id="line.246">        }</a>
<span class="sourceLineNo">247</span><a id="line.247"></a>
<span class="sourceLineNo">248</span><a id="line.248">        /**</a>
<span class="sourceLineNo">249</span><a id="line.249">         * @param startCodonSequence the startCodonSequence to set</a>
<span class="sourceLineNo">250</span><a id="line.250">         */</a>
<span class="sourceLineNo">251</span><a id="line.251">        public void addStartCodonSequence(AccessionID accession, int begin, int end) {</a>
<span class="sourceLineNo">252</span><a id="line.252">                this.startCodonSequence = new StartCodonSequence(this, begin, end);</a>
<span class="sourceLineNo">253</span><a id="line.253">                startCodonSequence.setAccession(accession);</a>
<span class="sourceLineNo">254</span><a id="line.254">        }</a>
<span class="sourceLineNo">255</span><a id="line.255"></a>
<span class="sourceLineNo">256</span><a id="line.256">        /**</a>
<span class="sourceLineNo">257</span><a id="line.257">         * @return the stopCodonSequence</a>
<span class="sourceLineNo">258</span><a id="line.258">         */</a>
<span class="sourceLineNo">259</span><a id="line.259">        public StopCodonSequence getStopCodonSequence() {</a>
<span class="sourceLineNo">260</span><a id="line.260">                return stopCodonSequence;</a>
<span class="sourceLineNo">261</span><a id="line.261">        }</a>
<span class="sourceLineNo">262</span><a id="line.262"></a>
<span class="sourceLineNo">263</span><a id="line.263">        /**</a>
<span class="sourceLineNo">264</span><a id="line.264">         * @param stopCodonSequence the stopCodonSequence to set</a>
<span class="sourceLineNo">265</span><a id="line.265">         */</a>
<span class="sourceLineNo">266</span><a id="line.266">        public void addStopCodonSequence(AccessionID accession, int begin, int end) {</a>
<span class="sourceLineNo">267</span><a id="line.267">                this.stopCodonSequence = new StopCodonSequence(this, begin, end);</a>
<span class="sourceLineNo">268</span><a id="line.268">                stopCodonSequence.setAccession(accession);</a>
<span class="sourceLineNo">269</span><a id="line.269">        }</a>
<span class="sourceLineNo">270</span><a id="line.270">}</a>




























































</pre>
</div>
</main>
</body>
</html>
