<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> * created at Apr 26, 2008<a name="line.20"></a>
<span class="sourceLineNo">021</span> */<a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojava.nbio.structure.io.mmcif;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import java.text.ParseException;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import java.text.SimpleDateFormat;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import java.util.ArrayList;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import java.util.Date;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import java.util.HashMap;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import java.util.List;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import java.util.Locale;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import java.util.Map;<a name="line.31"></a>
<span class="sourceLineNo">032</span><a name="line.32"></a>
<span class="sourceLineNo">033</span>import javax.vecmath.Matrix4d;<a name="line.33"></a>
<span class="sourceLineNo">034</span><a name="line.34"></a>
<span class="sourceLineNo">035</span>import org.biojava.nbio.structure.AminoAcid;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import org.biojava.nbio.structure.AminoAcidImpl;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import org.biojava.nbio.structure.Atom;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import org.biojava.nbio.structure.AtomImpl;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import org.biojava.nbio.structure.Chain;<a name="line.39"></a>
<span class="sourceLineNo">040</span>import org.biojava.nbio.structure.ChainImpl;<a name="line.40"></a>
<span class="sourceLineNo">041</span>import org.biojava.nbio.structure.EntityInfo;<a name="line.41"></a>
<span class="sourceLineNo">042</span>import org.biojava.nbio.structure.EntityType;<a name="line.42"></a>
<span class="sourceLineNo">043</span>import org.biojava.nbio.structure.DBRef;<a name="line.43"></a>
<span class="sourceLineNo">044</span>import org.biojava.nbio.structure.Element;<a name="line.44"></a>
<span class="sourceLineNo">045</span>import org.biojava.nbio.structure.Group;<a name="line.45"></a>
<span class="sourceLineNo">046</span>import org.biojava.nbio.structure.GroupType;<a name="line.46"></a>
<span class="sourceLineNo">047</span>import org.biojava.nbio.structure.HetatomImpl;<a name="line.47"></a>
<span class="sourceLineNo">048</span>import org.biojava.nbio.structure.NucleotideImpl;<a name="line.48"></a>
<span class="sourceLineNo">049</span>import org.biojava.nbio.structure.PDBCrystallographicInfo;<a name="line.49"></a>
<span class="sourceLineNo">050</span>import org.biojava.nbio.structure.PDBHeader;<a name="line.50"></a>
<span class="sourceLineNo">051</span>import org.biojava.nbio.structure.ResidueNumber;<a name="line.51"></a>
<span class="sourceLineNo">052</span>import org.biojava.nbio.structure.SeqMisMatch;<a name="line.52"></a>
<span class="sourceLineNo">053</span>import org.biojava.nbio.structure.SeqMisMatchImpl;<a name="line.53"></a>
<span class="sourceLineNo">054</span>import org.biojava.nbio.structure.Site;<a name="line.54"></a>
<span class="sourceLineNo">055</span>import org.biojava.nbio.structure.Structure;<a name="line.55"></a>
<span class="sourceLineNo">056</span>import org.biojava.nbio.structure.StructureException;<a name="line.56"></a>
<span class="sourceLineNo">057</span>import org.biojava.nbio.structure.StructureImpl;<a name="line.57"></a>
<span class="sourceLineNo">058</span>import org.biojava.nbio.structure.StructureTools;<a name="line.58"></a>
<span class="sourceLineNo">059</span>import org.biojava.nbio.structure.io.BondMaker;<a name="line.59"></a>
<span class="sourceLineNo">060</span>import org.biojava.nbio.structure.io.ChargeAdder;<a name="line.60"></a>
<span class="sourceLineNo">061</span>import org.biojava.nbio.structure.io.EntityFinder;<a name="line.61"></a>
<span class="sourceLineNo">062</span>import org.biojava.nbio.structure.io.FileParsingParameters;<a name="line.62"></a>
<span class="sourceLineNo">063</span>import org.biojava.nbio.structure.io.SeqRes2AtomAligner;<a name="line.63"></a>
<span class="sourceLineNo">064</span>import org.biojava.nbio.structure.io.mmcif.model.AtomSite;<a name="line.64"></a>
<span class="sourceLineNo">065</span>import org.biojava.nbio.structure.io.mmcif.model.AtomSites;<a name="line.65"></a>
<span class="sourceLineNo">066</span>import org.biojava.nbio.structure.io.mmcif.model.AuditAuthor;<a name="line.66"></a>
<span class="sourceLineNo">067</span>import org.biojava.nbio.structure.io.mmcif.model.Cell;<a name="line.67"></a>
<span class="sourceLineNo">068</span>import org.biojava.nbio.structure.io.mmcif.model.ChemComp;<a name="line.68"></a>
<span class="sourceLineNo">069</span>import org.biojava.nbio.structure.io.mmcif.model.ChemCompAtom;<a name="line.69"></a>
<span class="sourceLineNo">070</span>import org.biojava.nbio.structure.io.mmcif.model.ChemCompBond;<a name="line.70"></a>
<span class="sourceLineNo">071</span>import org.biojava.nbio.structure.io.mmcif.model.ChemCompDescriptor;<a name="line.71"></a>
<span class="sourceLineNo">072</span>import org.biojava.nbio.structure.io.mmcif.model.DatabasePDBremark;<a name="line.72"></a>
<span class="sourceLineNo">073</span>import org.biojava.nbio.structure.io.mmcif.model.DatabasePDBrev;<a name="line.73"></a>
<span class="sourceLineNo">074</span>import org.biojava.nbio.structure.io.mmcif.model.DatabasePdbrevRecord;<a name="line.74"></a>
<span class="sourceLineNo">075</span>import org.biojava.nbio.structure.io.mmcif.model.Entity;<a name="line.75"></a>
<span class="sourceLineNo">076</span>import org.biojava.nbio.structure.io.mmcif.model.EntityPoly;<a name="line.76"></a>
<span class="sourceLineNo">077</span>import org.biojava.nbio.structure.io.mmcif.model.EntityPolySeq;<a name="line.77"></a>
<span class="sourceLineNo">078</span>import org.biojava.nbio.structure.io.mmcif.model.EntitySrcGen;<a name="line.78"></a>
<span class="sourceLineNo">079</span>import org.biojava.nbio.structure.io.mmcif.model.EntitySrcNat;<a name="line.79"></a>
<span class="sourceLineNo">080</span>import org.biojava.nbio.structure.io.mmcif.model.EntitySrcSyn;<a name="line.80"></a>
<span class="sourceLineNo">081</span>import org.biojava.nbio.structure.io.mmcif.model.Exptl;<a name="line.81"></a>
<span class="sourceLineNo">082</span>import org.biojava.nbio.structure.io.mmcif.model.PdbxAuditRevisionHistory;<a name="line.82"></a>
<span class="sourceLineNo">083</span>import org.biojava.nbio.structure.io.mmcif.model.PdbxChemCompDescriptor;<a name="line.83"></a>
<span class="sourceLineNo">084</span>import org.biojava.nbio.structure.io.mmcif.model.PdbxChemCompIdentifier;<a name="line.84"></a>
<span class="sourceLineNo">085</span>import org.biojava.nbio.structure.io.mmcif.model.PdbxDatabaseStatus;<a name="line.85"></a>
<span class="sourceLineNo">086</span>import org.biojava.nbio.structure.io.mmcif.model.PdbxEntityNonPoly;<a name="line.86"></a>
<span class="sourceLineNo">087</span>import org.biojava.nbio.structure.io.mmcif.model.PdbxNonPolyScheme;<a name="line.87"></a>
<span class="sourceLineNo">088</span>import org.biojava.nbio.structure.io.mmcif.model.PdbxPolySeqScheme;<a name="line.88"></a>
<span class="sourceLineNo">089</span>import org.biojava.nbio.structure.io.mmcif.model.PdbxStructAssembly;<a name="line.89"></a>
<span class="sourceLineNo">090</span>import org.biojava.nbio.structure.io.mmcif.model.PdbxStructAssemblyGen;<a name="line.90"></a>
<span class="sourceLineNo">091</span>import org.biojava.nbio.structure.io.mmcif.model.PdbxStructOperList;<a name="line.91"></a>
<span class="sourceLineNo">092</span>import org.biojava.nbio.structure.io.mmcif.model.Refine;<a name="line.92"></a>
<span class="sourceLineNo">093</span>import org.biojava.nbio.structure.io.mmcif.model.Struct;<a name="line.93"></a>
<span class="sourceLineNo">094</span>import org.biojava.nbio.structure.io.mmcif.model.StructAsym;<a name="line.94"></a>
<span class="sourceLineNo">095</span>import org.biojava.nbio.structure.io.mmcif.model.StructConn;<a name="line.95"></a>
<span class="sourceLineNo">096</span>import org.biojava.nbio.structure.io.mmcif.model.StructKeywords;<a name="line.96"></a>
<span class="sourceLineNo">097</span>import org.biojava.nbio.structure.io.mmcif.model.StructNcsOper;<a name="line.97"></a>
<span class="sourceLineNo">098</span>import org.biojava.nbio.structure.io.mmcif.model.StructRef;<a name="line.98"></a>
<span class="sourceLineNo">099</span>import org.biojava.nbio.structure.io.mmcif.model.StructRefSeq;<a name="line.99"></a>
<span class="sourceLineNo">100</span>import org.biojava.nbio.structure.io.mmcif.model.StructRefSeqDif;<a name="line.100"></a>
<span class="sourceLineNo">101</span>import org.biojava.nbio.structure.io.mmcif.model.StructSite;<a name="line.101"></a>
<span class="sourceLineNo">102</span>import org.biojava.nbio.structure.io.mmcif.model.StructSiteGen;<a name="line.102"></a>
<span class="sourceLineNo">103</span>import org.biojava.nbio.structure.io.mmcif.model.Symmetry;<a name="line.103"></a>
<span class="sourceLineNo">104</span>import org.biojava.nbio.structure.quaternary.BioAssemblyInfo;<a name="line.104"></a>
<span class="sourceLineNo">105</span>import org.biojava.nbio.structure.quaternary.BiologicalAssemblyBuilder;<a name="line.105"></a>
<span class="sourceLineNo">106</span>import org.biojava.nbio.structure.quaternary.BiologicalAssemblyTransformation;<a name="line.106"></a>
<span class="sourceLineNo">107</span>import org.biojava.nbio.structure.xtal.CrystalCell;<a name="line.107"></a>
<span class="sourceLineNo">108</span>import org.biojava.nbio.structure.xtal.SpaceGroup;<a name="line.108"></a>
<span class="sourceLineNo">109</span>import org.biojava.nbio.structure.xtal.SymoplibParser;<a name="line.109"></a>
<span class="sourceLineNo">110</span>import org.slf4j.Logger;<a name="line.110"></a>
<span class="sourceLineNo">111</span>import org.slf4j.LoggerFactory;<a name="line.111"></a>
<span class="sourceLineNo">112</span><a name="line.112"></a>
<span class="sourceLineNo">113</span>/**<a name="line.113"></a>
<span class="sourceLineNo">114</span> * A MMcifConsumer implementation that builds an in-memory representation of the<a name="line.114"></a>
<span class="sourceLineNo">115</span> * content of a mmcif file as a BioJava Structure object.<a name="line.115"></a>
<span class="sourceLineNo">116</span> *<a name="line.116"></a>
<span class="sourceLineNo">117</span> * @author Andreas Prlic<a name="line.117"></a>
<span class="sourceLineNo">118</span> * @since 1.7<a name="line.118"></a>
<span class="sourceLineNo">119</span> */<a name="line.119"></a>
<span class="sourceLineNo">120</span><a name="line.120"></a>
<span class="sourceLineNo">121</span>public class SimpleMMcifConsumer implements MMcifConsumer {<a name="line.121"></a>
<span class="sourceLineNo">122</span><a name="line.122"></a>
<span class="sourceLineNo">123</span>        private static final Logger logger = LoggerFactory.getLogger(SimpleMMcifConsumer.class);<a name="line.123"></a>
<span class="sourceLineNo">124</span><a name="line.124"></a>
<span class="sourceLineNo">125</span>        private Structure structure;<a name="line.125"></a>
<span class="sourceLineNo">126</span>        private Chain currentChain;<a name="line.126"></a>
<span class="sourceLineNo">127</span>        private Group currentGroup;<a name="line.127"></a>
<span class="sourceLineNo">128</span><a name="line.128"></a>
<span class="sourceLineNo">129</span>        /**<a name="line.129"></a>
<span class="sourceLineNo">130</span>         * A temporary data structure to hold all parsed chains<a name="line.130"></a>
<span class="sourceLineNo">131</span>         */<a name="line.131"></a>
<span class="sourceLineNo">132</span>        private ArrayList&lt;List&lt;Chain&gt;&gt; allModels; <a name="line.132"></a>
<span class="sourceLineNo">133</span>        /**<a name="line.133"></a>
<span class="sourceLineNo">134</span>         * The current set of chains per model<a name="line.134"></a>
<span class="sourceLineNo">135</span>         */<a name="line.135"></a>
<span class="sourceLineNo">136</span>        private List&lt;Chain&gt;      currentModel;<a name="line.136"></a>
<span class="sourceLineNo">137</span>        private List&lt;Entity&gt;     entities;<a name="line.137"></a>
<span class="sourceLineNo">138</span>        /**<a name="line.138"></a>
<span class="sourceLineNo">139</span>         * Needed in header only mode to get mapping between asym ids and author ids<a name="line.139"></a>
<span class="sourceLineNo">140</span>         */<a name="line.140"></a>
<span class="sourceLineNo">141</span>        private List&lt;EntityPoly&gt; entityPolys;<a name="line.141"></a>
<span class="sourceLineNo">142</span>        private List&lt;StructRef&gt;  strucRefs;<a name="line.142"></a>
<span class="sourceLineNo">143</span>        private List&lt;Chain&gt;      seqResChains;<a name="line.143"></a>
<span class="sourceLineNo">144</span>        private List&lt;Chain&gt;      entityChains; // needed to link entities, chains and compounds...<a name="line.144"></a>
<span class="sourceLineNo">145</span>        private List&lt;StructAsym&gt; structAsyms;  // needed to link entities, chains and compounds...<a name="line.145"></a>
<span class="sourceLineNo">146</span>        private List&lt;PdbxStructOperList&gt; structOpers ; //<a name="line.146"></a>
<span class="sourceLineNo">147</span>        private List&lt;PdbxStructAssembly&gt; strucAssemblies;<a name="line.147"></a>
<span class="sourceLineNo">148</span>        private List&lt;PdbxStructAssemblyGen&gt; strucAssemblyGens;<a name="line.148"></a>
<span class="sourceLineNo">149</span>        private List&lt;EntitySrcGen&gt; entitySrcGens;<a name="line.149"></a>
<span class="sourceLineNo">150</span>        private List&lt;EntitySrcNat&gt; entitySrcNats;<a name="line.150"></a>
<span class="sourceLineNo">151</span>        private List&lt;EntitySrcSyn&gt; entitySrcSyns;<a name="line.151"></a>
<span class="sourceLineNo">152</span>        private List&lt;StructConn&gt; structConn;<a name="line.152"></a>
<span class="sourceLineNo">153</span>        private List&lt;StructNcsOper&gt; structNcsOper;<a name="line.153"></a>
<span class="sourceLineNo">154</span>        private List&lt;StructRefSeqDif&gt; sequenceDifs;<a name="line.154"></a>
<span class="sourceLineNo">155</span>        private List&lt;StructSiteGen&gt; structSiteGens;<a name="line.155"></a>
<span class="sourceLineNo">156</span>        <a name="line.156"></a>
<span class="sourceLineNo">157</span>        private Matrix4d parsedScaleMatrix;<a name="line.157"></a>
<span class="sourceLineNo">158</span><a name="line.158"></a>
<span class="sourceLineNo">159</span><a name="line.159"></a>
<span class="sourceLineNo">160</span><a name="line.160"></a>
<span class="sourceLineNo">161</span>        /**<a name="line.161"></a>
<span class="sourceLineNo">162</span>         * A map of asym ids (internal chain ids) to entity ids extracted from<a name="line.162"></a>
<span class="sourceLineNo">163</span>         * the _struct_asym category<a name="line.163"></a>
<span class="sourceLineNo">164</span>         */<a name="line.164"></a>
<span class="sourceLineNo">165</span>        private Map&lt;String,String&gt; asymId2entityId;<a name="line.165"></a>
<span class="sourceLineNo">166</span><a name="line.166"></a>
<span class="sourceLineNo">167</span>        /**<a name="line.167"></a>
<span class="sourceLineNo">168</span>         * A map of asym ids (internal chain ids) to author ids extracted from <a name="line.168"></a>
<span class="sourceLineNo">169</span>         * the _entity_poly category. Used in header only parsing.<a name="line.169"></a>
<span class="sourceLineNo">170</span>         */<a name="line.170"></a>
<span class="sourceLineNo">171</span>        private Map&lt;String,String&gt; asymId2authorId;<a name="line.171"></a>
<span class="sourceLineNo">172</span><a name="line.172"></a>
<span class="sourceLineNo">173</span>        private String currentNmrModelNumber ;<a name="line.173"></a>
<span class="sourceLineNo">174</span><a name="line.174"></a>
<span class="sourceLineNo">175</span>        private FileParsingParameters params;<a name="line.175"></a>
<span class="sourceLineNo">176</span><a name="line.176"></a>
<span class="sourceLineNo">177</span>        public  SimpleMMcifConsumer(){<a name="line.177"></a>
<span class="sourceLineNo">178</span>                params = new FileParsingParameters();<a name="line.178"></a>
<span class="sourceLineNo">179</span>                documentStart();<a name="line.179"></a>
<span class="sourceLineNo">180</span><a name="line.180"></a>
<span class="sourceLineNo">181</span>        }<a name="line.181"></a>
<span class="sourceLineNo">182</span><a name="line.182"></a>
<span class="sourceLineNo">183</span>        @Override<a name="line.183"></a>
<span class="sourceLineNo">184</span>        public void newEntity(Entity entity) {<a name="line.184"></a>
<span class="sourceLineNo">185</span>                logger.debug("New entity: {}",entity.toString());<a name="line.185"></a>
<span class="sourceLineNo">186</span>                entities.add(entity);<a name="line.186"></a>
<span class="sourceLineNo">187</span>        }<a name="line.187"></a>
<span class="sourceLineNo">188</span><a name="line.188"></a>
<span class="sourceLineNo">189</span>        @Override<a name="line.189"></a>
<span class="sourceLineNo">190</span>        public void newEntityPoly(EntityPoly entityPoly) {<a name="line.190"></a>
<span class="sourceLineNo">191</span>                entityPolys.add(entityPoly);<a name="line.191"></a>
<span class="sourceLineNo">192</span>        }<a name="line.192"></a>
<span class="sourceLineNo">193</span><a name="line.193"></a>
<span class="sourceLineNo">194</span>        @Override<a name="line.194"></a>
<span class="sourceLineNo">195</span>        public void newPdbxStructOperList(PdbxStructOperList structOper){<a name="line.195"></a>
<span class="sourceLineNo">196</span><a name="line.196"></a>
<span class="sourceLineNo">197</span>                structOpers.add(structOper);<a name="line.197"></a>
<span class="sourceLineNo">198</span>        }<a name="line.198"></a>
<span class="sourceLineNo">199</span><a name="line.199"></a>
<span class="sourceLineNo">200</span>        @Override<a name="line.200"></a>
<span class="sourceLineNo">201</span>        public void newStructAsym(StructAsym sasym){<a name="line.201"></a>
<span class="sourceLineNo">202</span><a name="line.202"></a>
<span class="sourceLineNo">203</span>                structAsyms.add(sasym);<a name="line.203"></a>
<span class="sourceLineNo">204</span>        }<a name="line.204"></a>
<span class="sourceLineNo">205</span><a name="line.205"></a>
<span class="sourceLineNo">206</span>        private Entity getEntity(int entity_id){<a name="line.206"></a>
<span class="sourceLineNo">207</span>                try {<a name="line.207"></a>
<span class="sourceLineNo">208</span>                        for (Entity e: entities){<a name="line.208"></a>
<span class="sourceLineNo">209</span>                                int eId = Integer.parseInt(e.getId());<a name="line.209"></a>
<span class="sourceLineNo">210</span>                                if  (eId== entity_id){<a name="line.210"></a>
<span class="sourceLineNo">211</span>                                        return e;<a name="line.211"></a>
<span class="sourceLineNo">212</span>                                }<a name="line.212"></a>
<span class="sourceLineNo">213</span>                        }<a name="line.213"></a>
<span class="sourceLineNo">214</span>                } catch (NumberFormatException e) {<a name="line.214"></a>
<span class="sourceLineNo">215</span>                        logger.warn("Entity id does not look like a number:", e.getMessage());<a name="line.215"></a>
<span class="sourceLineNo">216</span>                }<a name="line.216"></a>
<span class="sourceLineNo">217</span>                return null;<a name="line.217"></a>
<span class="sourceLineNo">218</span>        }<a name="line.218"></a>
<span class="sourceLineNo">219</span><a name="line.219"></a>
<span class="sourceLineNo">220</span>        @Override<a name="line.220"></a>
<span class="sourceLineNo">221</span>        public void newStructKeywords(StructKeywords kw){<a name="line.221"></a>
<span class="sourceLineNo">222</span>                PDBHeader header = structure.getPDBHeader();<a name="line.222"></a>
<span class="sourceLineNo">223</span>                if ( header == null)<a name="line.223"></a>
<span class="sourceLineNo">224</span>                        header = new PDBHeader();<a name="line.224"></a>
<span class="sourceLineNo">225</span>                header.setDescription(kw.getPdbx_keywords());<a name="line.225"></a>
<span class="sourceLineNo">226</span>                header.setClassification(kw.getPdbx_keywords());<a name="line.226"></a>
<span class="sourceLineNo">227</span>        }<a name="line.227"></a>
<span class="sourceLineNo">228</span><a name="line.228"></a>
<span class="sourceLineNo">229</span>        @Override<a name="line.229"></a>
<span class="sourceLineNo">230</span>        public void setStruct(Struct struct) {<a name="line.230"></a>
<span class="sourceLineNo">231</span><a name="line.231"></a>
<span class="sourceLineNo">232</span>                PDBHeader header = structure.getPDBHeader();<a name="line.232"></a>
<span class="sourceLineNo">233</span>                if ( header == null)<a name="line.233"></a>
<span class="sourceLineNo">234</span>                        header = new PDBHeader();<a name="line.234"></a>
<span class="sourceLineNo">235</span><a name="line.235"></a>
<span class="sourceLineNo">236</span>                header.setTitle(struct.getTitle());<a name="line.236"></a>
<span class="sourceLineNo">237</span>                header.setIdCode(struct.getEntry_id());<a name="line.237"></a>
<span class="sourceLineNo">238</span>                //header.setDescription(struct.getPdbx_descriptor());<a name="line.238"></a>
<span class="sourceLineNo">239</span>                //header.setClassification(struct.getPdbx_descriptor());<a name="line.239"></a>
<span class="sourceLineNo">240</span>                //header.setDescription(struct.getPdbx_descriptor());<a name="line.240"></a>
<span class="sourceLineNo">241</span><a name="line.241"></a>
<span class="sourceLineNo">242</span><a name="line.242"></a>
<span class="sourceLineNo">243</span><a name="line.243"></a>
<span class="sourceLineNo">244</span>                structure.setPDBHeader(header);<a name="line.244"></a>
<span class="sourceLineNo">245</span>                structure.setPDBCode(struct.getEntry_id());<a name="line.245"></a>
<span class="sourceLineNo">246</span>        }<a name="line.246"></a>
<span class="sourceLineNo">247</span><a name="line.247"></a>
<span class="sourceLineNo">248</span>        /** initiate new group, either Hetatom, Nucleotide, or AminoAcid */<a name="line.248"></a>
<span class="sourceLineNo">249</span>        private Group getNewGroup(String recordName,Character aminoCode1, long seq_id,String groupCode3) {<a name="line.249"></a>
<span class="sourceLineNo">250</span><a name="line.250"></a>
<span class="sourceLineNo">251</span>                Group g = ChemCompGroupFactory.getGroupFromChemCompDictionary(groupCode3);<a name="line.251"></a>
<span class="sourceLineNo">252</span>                if ( g != null &amp;&amp; !g.getChemComp().isEmpty()) {<a name="line.252"></a>
<span class="sourceLineNo">253</span>                        if ( g instanceof AminoAcidImpl) {<a name="line.253"></a>
<span class="sourceLineNo">254</span>                                AminoAcidImpl aa = (AminoAcidImpl) g;<a name="line.254"></a>
<span class="sourceLineNo">255</span>                                aa.setId(seq_id);<a name="line.255"></a>
<span class="sourceLineNo">256</span>                        } else if ( g instanceof NucleotideImpl) {<a name="line.256"></a>
<span class="sourceLineNo">257</span>                                NucleotideImpl nuc =  (NucleotideImpl) g;<a name="line.257"></a>
<span class="sourceLineNo">258</span>                                nuc.setId(seq_id);<a name="line.258"></a>
<span class="sourceLineNo">259</span>                        } else if ( g instanceof HetatomImpl) {<a name="line.259"></a>
<span class="sourceLineNo">260</span>                                HetatomImpl het = (HetatomImpl)g;<a name="line.260"></a>
<span class="sourceLineNo">261</span>                                het.setId(seq_id);<a name="line.261"></a>
<span class="sourceLineNo">262</span>                        }<a name="line.262"></a>
<span class="sourceLineNo">263</span>                        return g;<a name="line.263"></a>
<span class="sourceLineNo">264</span>                }<a name="line.264"></a>
<span class="sourceLineNo">265</span><a name="line.265"></a>
<span class="sourceLineNo">266</span><a name="line.266"></a>
<span class="sourceLineNo">267</span><a name="line.267"></a>
<span class="sourceLineNo">268</span>                Group group;<a name="line.268"></a>
<span class="sourceLineNo">269</span>                if ( recordName.equals("ATOM") ) {<a name="line.269"></a>
<span class="sourceLineNo">270</span>                        if (StructureTools.isNucleotide(groupCode3))  {<a name="line.270"></a>
<span class="sourceLineNo">271</span>                                // it is a nucleotide<a name="line.271"></a>
<span class="sourceLineNo">272</span>                                NucleotideImpl nu = new NucleotideImpl();<a name="line.272"></a>
<span class="sourceLineNo">273</span>                                group = nu;<a name="line.273"></a>
<span class="sourceLineNo">274</span>                                nu.setId(seq_id);<a name="line.274"></a>
<span class="sourceLineNo">275</span><a name="line.275"></a>
<span class="sourceLineNo">276</span>                        } else if (aminoCode1==null || aminoCode1 == StructureTools.UNKNOWN_GROUP_LABEL){<a name="line.276"></a>
<span class="sourceLineNo">277</span>                                HetatomImpl h = new HetatomImpl();<a name="line.277"></a>
<span class="sourceLineNo">278</span>                                h.setId(seq_id);<a name="line.278"></a>
<span class="sourceLineNo">279</span>                                group = h;<a name="line.279"></a>
<span class="sourceLineNo">280</span><a name="line.280"></a>
<span class="sourceLineNo">281</span>                        } else {<a name="line.281"></a>
<span class="sourceLineNo">282</span>                                AminoAcidImpl aa = new AminoAcidImpl() ;<a name="line.282"></a>
<span class="sourceLineNo">283</span>                                aa.setAminoType(aminoCode1);<a name="line.283"></a>
<span class="sourceLineNo">284</span>                                aa.setId(seq_id);<a name="line.284"></a>
<span class="sourceLineNo">285</span>                                group = aa ;<a name="line.285"></a>
<span class="sourceLineNo">286</span>                        }<a name="line.286"></a>
<span class="sourceLineNo">287</span>                }<a name="line.287"></a>
<span class="sourceLineNo">288</span>                else {<a name="line.288"></a>
<span class="sourceLineNo">289</span>                        if (StructureTools.isNucleotide(groupCode3))  {<a name="line.289"></a>
<span class="sourceLineNo">290</span>                                // it is a nucleotide<a name="line.290"></a>
<span class="sourceLineNo">291</span>                                NucleotideImpl nu = new NucleotideImpl();<a name="line.291"></a>
<span class="sourceLineNo">292</span>                                group = nu;<a name="line.292"></a>
<span class="sourceLineNo">293</span>                                nu.setId(seq_id);<a name="line.293"></a>
<span class="sourceLineNo">294</span>                        }<a name="line.294"></a>
<span class="sourceLineNo">295</span>                        else if (aminoCode1 != null ) {<a name="line.295"></a>
<span class="sourceLineNo">296</span>                                AminoAcidImpl aa = new AminoAcidImpl() ;<a name="line.296"></a>
<span class="sourceLineNo">297</span>                                aa.setAminoType(aminoCode1);<a name="line.297"></a>
<span class="sourceLineNo">298</span>                                aa.setId(seq_id);<a name="line.298"></a>
<span class="sourceLineNo">299</span>                                group = aa ;<a name="line.299"></a>
<span class="sourceLineNo">300</span>                        } else {<a name="line.300"></a>
<span class="sourceLineNo">301</span>                                HetatomImpl h = new HetatomImpl();<a name="line.301"></a>
<span class="sourceLineNo">302</span>                                h.setId(seq_id);<a name="line.302"></a>
<span class="sourceLineNo">303</span>                                group = h;<a name="line.303"></a>
<span class="sourceLineNo">304</span>                        }<a name="line.304"></a>
<span class="sourceLineNo">305</span>                }<a name="line.305"></a>
<span class="sourceLineNo">306</span>                return  group ;<a name="line.306"></a>
<span class="sourceLineNo">307</span>        }<a name="line.307"></a>
<span class="sourceLineNo">308</span><a name="line.308"></a>
<span class="sourceLineNo">309</span>        /**<a name="line.309"></a>
<span class="sourceLineNo">310</span>         * Test if the given asymId is already present in the list of chains given. If yes, returns the chain<a name="line.310"></a>
<span class="sourceLineNo">311</span>         * otherwise returns null.<a name="line.311"></a>
<span class="sourceLineNo">312</span>         */<a name="line.312"></a>
<span class="sourceLineNo">313</span>        private static Chain isKnownChain(String asymId, List&lt;Chain&gt; chains){<a name="line.313"></a>
<span class="sourceLineNo">314</span><a name="line.314"></a>
<span class="sourceLineNo">315</span>                for (int i = 0; i&lt; chains.size();i++){<a name="line.315"></a>
<span class="sourceLineNo">316</span>                        Chain testchain =  chains.get(i);<a name="line.316"></a>
<span class="sourceLineNo">317</span>                        //System.out.println("comparing chainID &gt;"+chainID+"&lt; against testchain " + i+" &gt;" +testchain.getName()+"&lt;");<a name="line.317"></a>
<span class="sourceLineNo">318</span>                        if (asymId.equals(testchain.getId())) {<a name="line.318"></a>
<span class="sourceLineNo">319</span>                                //System.out.println("chain "+ chainID+" already known ...");<a name="line.319"></a>
<span class="sourceLineNo">320</span>                                return testchain;<a name="line.320"></a>
<span class="sourceLineNo">321</span>                        }<a name="line.321"></a>
<span class="sourceLineNo">322</span>                }<a name="line.322"></a>
<span class="sourceLineNo">323</span><a name="line.323"></a>
<span class="sourceLineNo">324</span>                return null;<a name="line.324"></a>
<span class="sourceLineNo">325</span>        }<a name="line.325"></a>
<span class="sourceLineNo">326</span><a name="line.326"></a>
<span class="sourceLineNo">327</span>        @Override<a name="line.327"></a>
<span class="sourceLineNo">328</span>        public void newAtomSite(AtomSite atom) {<a name="line.328"></a>
<span class="sourceLineNo">329</span><a name="line.329"></a>
<span class="sourceLineNo">330</span>                if (params.isHeaderOnly()) return;<a name="line.330"></a>
<span class="sourceLineNo">331</span><a name="line.331"></a>
<span class="sourceLineNo">332</span>                // Warning: getLabel_asym_id is not the "chain id" in the PDB file<a name="line.332"></a>
<span class="sourceLineNo">333</span>                // it is the internally used chain id.<a name="line.333"></a>
<span class="sourceLineNo">334</span>                // later on we will fix this...<a name="line.334"></a>
<span class="sourceLineNo">335</span><a name="line.335"></a>
<span class="sourceLineNo">336</span>                // later one needs to map the asym id to the pdb_strand_id<a name="line.336"></a>
<span class="sourceLineNo">337</span><a name="line.337"></a>
<span class="sourceLineNo">338</span>                //TODO: add support for FileParsingParams.getMaxAtoms()<a name="line.338"></a>
<span class="sourceLineNo">339</span><a name="line.339"></a>
<span class="sourceLineNo">340</span>                boolean startOfNewChain = false;<a name="line.340"></a>
<span class="sourceLineNo">341</span><a name="line.341"></a>
<span class="sourceLineNo">342</span>                String asymId = atom.getLabel_asym_id();<a name="line.342"></a>
<span class="sourceLineNo">343</span>                String authId = atom.getAuth_asym_id();<a name="line.343"></a>
<span class="sourceLineNo">344</span><a name="line.344"></a>
<span class="sourceLineNo">345</span>                String recordName    = atom.getGroup_PDB();<a name="line.345"></a>
<span class="sourceLineNo">346</span>                String residueNumberS = atom.getAuth_seq_id();<a name="line.346"></a>
<span class="sourceLineNo">347</span>                Integer residueNrInt = Integer.parseInt(residueNumberS);<a name="line.347"></a>
<span class="sourceLineNo">348</span><a name="line.348"></a>
<span class="sourceLineNo">349</span>                // the 3-letter name of the group:<a name="line.349"></a>
<span class="sourceLineNo">350</span>                String groupCode3    = atom.getLabel_comp_id();<a name="line.350"></a>
<span class="sourceLineNo">351</span><a name="line.351"></a>
<span class="sourceLineNo">352</span>                boolean isHetAtomInFile = false;<a name="line.352"></a>
<span class="sourceLineNo">353</span><a name="line.353"></a>
<span class="sourceLineNo">354</span>                Character aminoCode1 = null;<a name="line.354"></a>
<span class="sourceLineNo">355</span>                if ( recordName.equals("ATOM") )<a name="line.355"></a>
<span class="sourceLineNo">356</span>                        aminoCode1 = StructureTools.get1LetterCodeAmino(groupCode3);<a name="line.356"></a>
<span class="sourceLineNo">357</span>                else {<a name="line.357"></a>
<span class="sourceLineNo">358</span>                        aminoCode1 = StructureTools.get1LetterCodeAmino(groupCode3);<a name="line.358"></a>
<span class="sourceLineNo">359</span><a name="line.359"></a>
<span class="sourceLineNo">360</span>                        // for nucleotides this will be null..<a name="line.360"></a>
<span class="sourceLineNo">361</span>                        if (aminoCode1 != null &amp;&amp;  aminoCode1.equals(StructureTools.UNKNOWN_GROUP_LABEL))<a name="line.361"></a>
<span class="sourceLineNo">362</span>                                aminoCode1 = null;<a name="line.362"></a>
<span class="sourceLineNo">363</span><a name="line.363"></a>
<span class="sourceLineNo">364</span>                        isHetAtomInFile = true;<a name="line.364"></a>
<span class="sourceLineNo">365</span>                }<a name="line.365"></a>
<span class="sourceLineNo">366</span>                String insCodeS = atom.getPdbx_PDB_ins_code();<a name="line.366"></a>
<span class="sourceLineNo">367</span>                Character insCode = null;<a name="line.367"></a>
<span class="sourceLineNo">368</span>                if (!  insCodeS.equals("?")) {<a name="line.368"></a>
<span class="sourceLineNo">369</span>                        insCode = insCodeS.charAt(0);<a name="line.369"></a>
<span class="sourceLineNo">370</span>                }<a name="line.370"></a>
<span class="sourceLineNo">371</span>                // we store the internal seq id in the Atom._id field<a name="line.371"></a>
<span class="sourceLineNo">372</span>                // this is not a PDB file field but we need this to internally assign the insertion codes later<a name="line.372"></a>
<span class="sourceLineNo">373</span>                // from the pdbx_poly_seq entries..<a name="line.373"></a>
<span class="sourceLineNo">374</span><a name="line.374"></a>
<span class="sourceLineNo">375</span>                long seq_id = -1;<a name="line.375"></a>
<span class="sourceLineNo">376</span>                try {<a name="line.376"></a>
<span class="sourceLineNo">377</span>                        seq_id = Long.parseLong(atom.getLabel_seq_id());<a name="line.377"></a>
<span class="sourceLineNo">378</span>                } catch (NumberFormatException e){<a name="line.378"></a>
<span class="sourceLineNo">379</span>                        // non polymer chains (ligands and small molecules) will have a label_seq_id set to '.', thus it is ok to<a name="line.379"></a>
<span class="sourceLineNo">380</span>                        // silently ignore this<a name="line.380"></a>
<span class="sourceLineNo">381</span>                        //logger.debug("Could not parse number for _atom_site.label_seq_id: "+e.getMessage());<a name="line.381"></a>
<span class="sourceLineNo">382</span>                }<a name="line.382"></a>
<span class="sourceLineNo">383</span><a name="line.383"></a>
<span class="sourceLineNo">384</span>                String nmrModelNumber = atom.getPdbx_PDB_model_num();<a name="line.384"></a>
<span class="sourceLineNo">385</span><a name="line.385"></a>
<span class="sourceLineNo">386</span>                if ( currentNmrModelNumber == null) {<a name="line.386"></a>
<span class="sourceLineNo">387</span>                        currentNmrModelNumber = nmrModelNumber;<a name="line.387"></a>
<span class="sourceLineNo">388</span>                }<a name="line.388"></a>
<span class="sourceLineNo">389</span><a name="line.389"></a>
<span class="sourceLineNo">390</span>                if (! currentNmrModelNumber.equals(nmrModelNumber)){<a name="line.390"></a>
<span class="sourceLineNo">391</span>                        currentNmrModelNumber = nmrModelNumber;<a name="line.391"></a>
<span class="sourceLineNo">392</span><a name="line.392"></a>
<span class="sourceLineNo">393</span>                        // add previous data<a name="line.393"></a>
<span class="sourceLineNo">394</span>                        if ( currentChain != null ) {<a name="line.394"></a>
<span class="sourceLineNo">395</span>                                currentChain.addGroup(currentGroup);<a name="line.395"></a>
<span class="sourceLineNo">396</span>                                currentGroup.trimToSize();<a name="line.396"></a>
<span class="sourceLineNo">397</span>                        }<a name="line.397"></a>
<span class="sourceLineNo">398</span><a name="line.398"></a>
<span class="sourceLineNo">399</span>                        // we came to the beginning of a new NMR model<a name="line.399"></a>
<span class="sourceLineNo">400</span>                        allModels.add(currentModel);<a name="line.400"></a>
<span class="sourceLineNo">401</span>                        currentModel = new ArrayList&lt;Chain&gt;();<a name="line.401"></a>
<span class="sourceLineNo">402</span>                        currentChain = null;<a name="line.402"></a>
<span class="sourceLineNo">403</span>                        currentGroup = null;<a name="line.403"></a>
<span class="sourceLineNo">404</span>                }<a name="line.404"></a>
<span class="sourceLineNo">405</span><a name="line.405"></a>
<span class="sourceLineNo">406</span><a name="line.406"></a>
<span class="sourceLineNo">407</span>                if (currentChain == null) {<a name="line.407"></a>
<span class="sourceLineNo">408</span><a name="line.408"></a>
<span class="sourceLineNo">409</span>                        currentChain = new ChainImpl();<a name="line.409"></a>
<span class="sourceLineNo">410</span>                        currentChain.setName(authId);<a name="line.410"></a>
<span class="sourceLineNo">411</span>                        currentChain.setId(asymId);<a name="line.411"></a>
<span class="sourceLineNo">412</span>                        currentModel.add(currentChain);<a name="line.412"></a>
<span class="sourceLineNo">413</span>                        startOfNewChain = true;<a name="line.413"></a>
<span class="sourceLineNo">414</span>                }<a name="line.414"></a>
<span class="sourceLineNo">415</span><a name="line.415"></a>
<span class="sourceLineNo">416</span>                //System.out.println("BEFORE: " + chain_id + " " + current_chain.getName());<a name="line.416"></a>
<span class="sourceLineNo">417</span>                if ( ! asymId.equals(currentChain.getId()) ) {<a name="line.417"></a>
<span class="sourceLineNo">418</span>                        //logger.info("unknown chain. creating new chain. authId:" + authId + " asymId: " + asymId);<a name="line.418"></a>
<span class="sourceLineNo">419</span>                        startOfNewChain = true;<a name="line.419"></a>
<span class="sourceLineNo">420</span><a name="line.420"></a>
<span class="sourceLineNo">421</span>                        // end up old chain...<a name="line.421"></a>
<span class="sourceLineNo">422</span>                        currentChain.addGroup(currentGroup);<a name="line.422"></a>
<span class="sourceLineNo">423</span><a name="line.423"></a>
<span class="sourceLineNo">424</span>                        // see if old chain is known ...<a name="line.424"></a>
<span class="sourceLineNo">425</span>                        Chain testchain = isKnownChain(asymId,currentModel);<a name="line.425"></a>
<span class="sourceLineNo">426</span><a name="line.426"></a>
<span class="sourceLineNo">427</span>                        if ( testchain == null) {<a name="line.427"></a>
<span class="sourceLineNo">428</span>                                //logger.info("unknown chain. creating new chain. authId:" + authId + " asymId: " + asymId);<a name="line.428"></a>
<span class="sourceLineNo">429</span><a name="line.429"></a>
<span class="sourceLineNo">430</span>                                currentChain = new ChainImpl();<a name="line.430"></a>
<span class="sourceLineNo">431</span>                                currentChain.setName(authId);<a name="line.431"></a>
<span class="sourceLineNo">432</span>                                currentChain.setId(asymId);<a name="line.432"></a>
<span class="sourceLineNo">433</span><a name="line.433"></a>
<span class="sourceLineNo">434</span>                        }   else {<a name="line.434"></a>
<span class="sourceLineNo">435</span>                                currentChain = testchain;<a name="line.435"></a>
<span class="sourceLineNo">436</span>                        }<a name="line.436"></a>
<span class="sourceLineNo">437</span><a name="line.437"></a>
<span class="sourceLineNo">438</span>                        if ( ! currentModel.contains(currentChain))<a name="line.438"></a>
<span class="sourceLineNo">439</span>                                currentModel.add(currentChain);<a name="line.439"></a>
<span class="sourceLineNo">440</span><a name="line.440"></a>
<span class="sourceLineNo">441</span>                }<a name="line.441"></a>
<span class="sourceLineNo">442</span><a name="line.442"></a>
<span class="sourceLineNo">443</span><a name="line.443"></a>
<span class="sourceLineNo">444</span>                ResidueNumber residueNumber = new ResidueNumber(authId,residueNrInt, insCode);<a name="line.444"></a>
<span class="sourceLineNo">445</span><a name="line.445"></a>
<span class="sourceLineNo">446</span>                if (currentGroup == null) {<a name="line.446"></a>
<span class="sourceLineNo">447</span><a name="line.447"></a>
<span class="sourceLineNo">448</span><a name="line.448"></a>
<span class="sourceLineNo">449</span>                        currentGroup = getNewGroup(recordName,aminoCode1,seq_id, groupCode3);<a name="line.449"></a>
<span class="sourceLineNo">450</span><a name="line.450"></a>
<span class="sourceLineNo">451</span>                        currentGroup.setResidueNumber(residueNumber);<a name="line.451"></a>
<span class="sourceLineNo">452</span>                        currentGroup.setPDBName(groupCode3);<a name="line.452"></a>
<span class="sourceLineNo">453</span>                        currentGroup.setHetAtomInFile(isHetAtomInFile);<a name="line.453"></a>
<span class="sourceLineNo">454</span>                }<a name="line.454"></a>
<span class="sourceLineNo">455</span><a name="line.455"></a>
<span class="sourceLineNo">456</span>                // SET UP THE ALT LOC GROUP<a name="line.456"></a>
<span class="sourceLineNo">457</span>                Group altGroup = null;<a name="line.457"></a>
<span class="sourceLineNo">458</span>                String altLocS = atom.getLabel_alt_id();<a name="line.458"></a>
<span class="sourceLineNo">459</span>                Character altLoc = ' ';<a name="line.459"></a>
<span class="sourceLineNo">460</span>                if ( altLocS.length()&gt;0) {<a name="line.460"></a>
<span class="sourceLineNo">461</span>                        altLoc = altLocS.charAt(0);<a name="line.461"></a>
<span class="sourceLineNo">462</span>                        if ( altLoc.equals('.') )<a name="line.462"></a>
<span class="sourceLineNo">463</span>                                altLoc = ' ';<a name="line.463"></a>
<span class="sourceLineNo">464</span><a name="line.464"></a>
<span class="sourceLineNo">465</span>                }<a name="line.465"></a>
<span class="sourceLineNo">466</span>                // If it's the start of the new chain <a name="line.466"></a>
<span class="sourceLineNo">467</span>                if ( startOfNewChain){<a name="line.467"></a>
<span class="sourceLineNo">468</span>                        currentGroup = getNewGroup(recordName,aminoCode1,seq_id, groupCode3);<a name="line.468"></a>
<span class="sourceLineNo">469</span>                        currentGroup.setResidueNumber(residueNumber);<a name="line.469"></a>
<span class="sourceLineNo">470</span>                        currentGroup.setPDBName(groupCode3);<a name="line.470"></a>
<span class="sourceLineNo">471</span>                        currentGroup.setHetAtomInFile(isHetAtomInFile);<a name="line.471"></a>
<span class="sourceLineNo">472</span>                }<a name="line.472"></a>
<span class="sourceLineNo">473</span>                // ANTHONY BRADLEY ADDED THIS -&gt; WE ONLY WAN'T TO CHECK FOR ALT LOCS WHEN IT's NOT THE FIRST GROUP IN CHAIN<a name="line.473"></a>
<span class="sourceLineNo">474</span>                else{<a name="line.474"></a>
<span class="sourceLineNo">475</span>                        // check if residue number is the same ...<a name="line.475"></a>
<span class="sourceLineNo">476</span>                        // insertion code is part of residue number<a name="line.476"></a>
<span class="sourceLineNo">477</span>                        if ( ! residueNumber.equals(currentGroup.getResidueNumber())) {<a name="line.477"></a>
<span class="sourceLineNo">478</span>                                //System.out.println("end of residue: "+current_group.getPDBCode()+" "+residueNrInt);<a name="line.478"></a>
<span class="sourceLineNo">479</span>                                currentChain.addGroup(currentGroup);<a name="line.479"></a>
<span class="sourceLineNo">480</span>                                currentGroup.trimToSize();<a name="line.480"></a>
<span class="sourceLineNo">481</span>                                currentGroup = getNewGroup(recordName,aminoCode1,seq_id,groupCode3);<a name="line.481"></a>
<span class="sourceLineNo">482</span>                                currentGroup.setPDBName(groupCode3);<a name="line.482"></a>
<span class="sourceLineNo">483</span>                                currentGroup.setResidueNumber(residueNumber);<a name="line.483"></a>
<span class="sourceLineNo">484</span>                                currentGroup.setHetAtomInFile(isHetAtomInFile);<a name="line.484"></a>
<span class="sourceLineNo">485</span><a name="line.485"></a>
<span class="sourceLineNo">486</span><a name="line.486"></a>
<span class="sourceLineNo">487</span>                        } else {<a name="line.487"></a>
<span class="sourceLineNo">488</span>                                // same residueNumber, but altLocs...<a name="line.488"></a>
<span class="sourceLineNo">489</span>                                // test altLoc<a name="line.489"></a>
<span class="sourceLineNo">490</span><a name="line.490"></a>
<span class="sourceLineNo">491</span>                                if ( ! altLoc.equals(' ') &amp;&amp; ( ! altLoc.equals('.'))) {<a name="line.491"></a>
<span class="sourceLineNo">492</span>                                        logger.debug("found altLoc! " + altLoc + " " + currentGroup + " " + altGroup);<a name="line.492"></a>
<span class="sourceLineNo">493</span>                                        altGroup = getCorrectAltLocGroup( altLoc,recordName,aminoCode1,groupCode3, seq_id);<a name="line.493"></a>
<span class="sourceLineNo">494</span>                                        if (altGroup.getChain()==null) {<a name="line.494"></a>
<span class="sourceLineNo">495</span>                                                altGroup.setChain(currentChain);<a name="line.495"></a>
<span class="sourceLineNo">496</span>                                        }<a name="line.496"></a>
<span class="sourceLineNo">497</span>                                }<a name="line.497"></a>
<span class="sourceLineNo">498</span>                        }<a name="line.498"></a>
<span class="sourceLineNo">499</span>                }<a name="line.499"></a>
<span class="sourceLineNo">500</span>                //atomCount++;<a name="line.500"></a>
<span class="sourceLineNo">501</span>                //System.out.println("fixing atom name for  &gt;" + atom.getLabel_atom_id() + "&lt; &gt;" + fullname + "&lt;");<a name="line.501"></a>
<span class="sourceLineNo">502</span><a name="line.502"></a>
<span class="sourceLineNo">503</span><a name="line.503"></a>
<span class="sourceLineNo">504</span>                if ( params.isParseCAOnly() ){<a name="line.504"></a>
<span class="sourceLineNo">505</span>                        // yes , user wants to get CA only<a name="line.505"></a>
<span class="sourceLineNo">506</span>                        // only parse CA atoms...<a name="line.506"></a>
<span class="sourceLineNo">507</span>                        if (! (atom.getLabel_atom_id().equals(StructureTools.CA_ATOM_NAME) &amp;&amp; atom.getType_symbol().equals("C"))) {<a name="line.507"></a>
<span class="sourceLineNo">508</span>                                //System.out.println("ignoring " + line);<a name="line.508"></a>
<span class="sourceLineNo">509</span>                                //atomCount--;<a name="line.509"></a>
<span class="sourceLineNo">510</span>                                return;<a name="line.510"></a>
<span class="sourceLineNo">511</span>                        }<a name="line.511"></a>
<span class="sourceLineNo">512</span>                }<a name="line.512"></a>
<span class="sourceLineNo">513</span><a name="line.513"></a>
<span class="sourceLineNo">514</span>                //see if chain_id is one of the previous chains ...<a name="line.514"></a>
<span class="sourceLineNo">515</span><a name="line.515"></a>
<span class="sourceLineNo">516</span>                Atom a = convertAtom(atom);<a name="line.516"></a>
<span class="sourceLineNo">517</span><a name="line.517"></a>
<span class="sourceLineNo">518</span>                //see if chain_id is one of the previous chains ...<a name="line.518"></a>
<span class="sourceLineNo">519</span>                if ( altGroup != null) {<a name="line.519"></a>
<span class="sourceLineNo">520</span>                        altGroup.addAtom(a);<a name="line.520"></a>
<span class="sourceLineNo">521</span>                        altGroup = null;<a name="line.521"></a>
<span class="sourceLineNo">522</span>                }<a name="line.522"></a>
<span class="sourceLineNo">523</span>                else {<a name="line.523"></a>
<span class="sourceLineNo">524</span>                        currentGroup.addAtom(a);<a name="line.524"></a>
<span class="sourceLineNo">525</span>                }<a name="line.525"></a>
<span class="sourceLineNo">526</span><a name="line.526"></a>
<span class="sourceLineNo">527</span><a name="line.527"></a>
<span class="sourceLineNo">528</span>                String atomName = a.getName();<a name="line.528"></a>
<span class="sourceLineNo">529</span>                // make sure that main group has all atoms <a name="line.529"></a>
<span class="sourceLineNo">530</span>                // GitHub issue: #76<a name="line.530"></a>
<span class="sourceLineNo">531</span>                if ( ! currentGroup.hasAtom(atomName)) {<a name="line.531"></a>
<span class="sourceLineNo">532</span>                        // Unless it's microheterogenity https://github.com/rcsb/codec-devel/issues/81<a name="line.532"></a>
<span class="sourceLineNo">533</span>                        if (currentGroup.getPDBName().equals(a.getGroup().getPDBName())) {<a name="line.533"></a>
<span class="sourceLineNo">534</span>                                if(!StructureTools.hasNonDeuteratedEquiv(a,currentGroup)){<a name="line.534"></a>
<span class="sourceLineNo">535</span>                                        currentGroup.addAtom(a);<a name="line.535"></a>
<span class="sourceLineNo">536</span>                                }<a name="line.536"></a>
<span class="sourceLineNo">537</span>                        }<a name="line.537"></a>
<span class="sourceLineNo">538</span><a name="line.538"></a>
<span class="sourceLineNo">539</span>                }<a name="line.539"></a>
<span class="sourceLineNo">540</span>        }<a name="line.540"></a>
<span class="sourceLineNo">541</span><a name="line.541"></a>
<span class="sourceLineNo">542</span>        /** <a name="line.542"></a>
<span class="sourceLineNo">543</span>         * Convert a mmCIF AtomSite object to a BioJava Atom object<a name="line.543"></a>
<span class="sourceLineNo">544</span>         *<a name="line.544"></a>
<span class="sourceLineNo">545</span>         * @param atom the mmmcif AtomSite record<a name="line.545"></a>
<span class="sourceLineNo">546</span>         * @return an Atom<a name="line.546"></a>
<span class="sourceLineNo">547</span>         */<a name="line.547"></a>
<span class="sourceLineNo">548</span>        private Atom convertAtom(AtomSite atom){<a name="line.548"></a>
<span class="sourceLineNo">549</span><a name="line.549"></a>
<span class="sourceLineNo">550</span><a name="line.550"></a>
<span class="sourceLineNo">551</span>                Atom a = new AtomImpl();<a name="line.551"></a>
<span class="sourceLineNo">552</span><a name="line.552"></a>
<span class="sourceLineNo">553</span>                a.setPDBserial(Integer.parseInt(atom.getId()));<a name="line.553"></a>
<span class="sourceLineNo">554</span>                a.setName(atom.getLabel_atom_id());<a name="line.554"></a>
<span class="sourceLineNo">555</span><a name="line.555"></a>
<span class="sourceLineNo">556</span>                double x = Double.parseDouble (atom.getCartn_x());<a name="line.556"></a>
<span class="sourceLineNo">557</span>                double y = Double.parseDouble (atom.getCartn_y());<a name="line.557"></a>
<span class="sourceLineNo">558</span>                double z = Double.parseDouble (atom.getCartn_z());<a name="line.558"></a>
<span class="sourceLineNo">559</span>                a.setX(x);<a name="line.559"></a>
<span class="sourceLineNo">560</span>                a.setY(y);<a name="line.560"></a>
<span class="sourceLineNo">561</span>                a.setZ(z);<a name="line.561"></a>
<span class="sourceLineNo">562</span><a name="line.562"></a>
<span class="sourceLineNo">563</span>                float occupancy = Float.parseFloat (atom.getOccupancy());<a name="line.563"></a>
<span class="sourceLineNo">564</span>                a.setOccupancy(occupancy);<a name="line.564"></a>
<span class="sourceLineNo">565</span><a name="line.565"></a>
<span class="sourceLineNo">566</span>                float temp = Float.parseFloat (atom.getB_iso_or_equiv());<a name="line.566"></a>
<span class="sourceLineNo">567</span>                a.setTempFactor(temp);<a name="line.567"></a>
<span class="sourceLineNo">568</span><a name="line.568"></a>
<span class="sourceLineNo">569</span>                String alt = atom.getLabel_alt_id();<a name="line.569"></a>
<span class="sourceLineNo">570</span>                if (( alt != null ) &amp;&amp; ( alt.length() &gt; 0) &amp;&amp; (! alt.equals("."))){<a name="line.570"></a>
<span class="sourceLineNo">571</span>                        a.setAltLoc(new Character(alt.charAt(0)));<a name="line.571"></a>
<span class="sourceLineNo">572</span>                } else {<a name="line.572"></a>
<span class="sourceLineNo">573</span>                        a.setAltLoc(new Character(' '));<a name="line.573"></a>
<span class="sourceLineNo">574</span>                }<a name="line.574"></a>
<span class="sourceLineNo">575</span><a name="line.575"></a>
<span class="sourceLineNo">576</span>                Element element = Element.R;<a name="line.576"></a>
<span class="sourceLineNo">577</span>                try {<a name="line.577"></a>
<span class="sourceLineNo">578</span>                        element = Element.valueOfIgnoreCase(atom.getType_symbol());<a name="line.578"></a>
<span class="sourceLineNo">579</span>                }  catch (IllegalArgumentException e) {<a name="line.579"></a>
<span class="sourceLineNo">580</span>                        logger.info("Element {} was not recognised as a BioJava-known element, the element will be represented as the generic element {}", atom.getType_symbol(), Element.R.name());<a name="line.580"></a>
<span class="sourceLineNo">581</span>                }<a name="line.581"></a>
<span class="sourceLineNo">582</span>                a.setElement(element);<a name="line.582"></a>
<span class="sourceLineNo">583</span><a name="line.583"></a>
<span class="sourceLineNo">584</span>                return a;<a name="line.584"></a>
<span class="sourceLineNo">585</span><a name="line.585"></a>
<span class="sourceLineNo">586</span>        }<a name="line.586"></a>
<span class="sourceLineNo">587</span><a name="line.587"></a>
<span class="sourceLineNo">588</span><a name="line.588"></a>
<span class="sourceLineNo">589</span>        private Group getCorrectAltLocGroup( Character altLoc,<a name="line.589"></a>
<span class="sourceLineNo">590</span>                        String recordName,<a name="line.590"></a>
<span class="sourceLineNo">591</span>                        Character aminoCode1,<a name="line.591"></a>
<span class="sourceLineNo">592</span>                        String groupCode3,<a name="line.592"></a>
<span class="sourceLineNo">593</span>                        long seq_id) {<a name="line.593"></a>
<span class="sourceLineNo">594</span><a name="line.594"></a>
<span class="sourceLineNo">595</span>                // see if we know this altLoc already;<a name="line.595"></a>
<span class="sourceLineNo">596</span>                List&lt;Atom&gt; atoms = currentGroup.getAtoms();<a name="line.596"></a>
<span class="sourceLineNo">597</span>                if ( atoms.size() &gt; 0) {<a name="line.597"></a>
<span class="sourceLineNo">598</span>                        Atom a1 = atoms.get(0);<a name="line.598"></a>
<span class="sourceLineNo">599</span>                        // we are just adding atoms to the current group<a name="line.599"></a>
<span class="sourceLineNo">600</span>                        // probably there is a second group following later...<a name="line.600"></a>
<span class="sourceLineNo">601</span>                        if (a1.getAltLoc().equals(altLoc)) {<a name="line.601"></a>
<span class="sourceLineNo">602</span><a name="line.602"></a>
<span class="sourceLineNo">603</span>                                return currentGroup;<a name="line.603"></a>
<span class="sourceLineNo">604</span>                        }<a name="line.604"></a>
<span class="sourceLineNo">605</span>                }<a name="line.605"></a>
<span class="sourceLineNo">606</span><a name="line.606"></a>
<span class="sourceLineNo">607</span>                List&lt;Group&gt; altLocs = currentGroup.getAltLocs();<a name="line.607"></a>
<span class="sourceLineNo">608</span>                for ( Group altLocG : altLocs ){<a name="line.608"></a>
<span class="sourceLineNo">609</span>                        atoms = altLocG.getAtoms();<a name="line.609"></a>
<span class="sourceLineNo">610</span>                        if ( atoms.size() &gt; 0) {<a name="line.610"></a>
<span class="sourceLineNo">611</span>                                for ( Atom a1 : atoms) {<a name="line.611"></a>
<span class="sourceLineNo">612</span>                                        if (a1.getAltLoc().equals( altLoc)) {<a name="line.612"></a>
<span class="sourceLineNo">613</span><a name="line.613"></a>
<span class="sourceLineNo">614</span>                                                return altLocG;<a name="line.614"></a>
<span class="sourceLineNo">615</span>                                        }<a name="line.615"></a>
<span class="sourceLineNo">616</span>                                }<a name="line.616"></a>
<span class="sourceLineNo">617</span>                        }<a name="line.617"></a>
<span class="sourceLineNo">618</span>                }<a name="line.618"></a>
<span class="sourceLineNo">619</span><a name="line.619"></a>
<span class="sourceLineNo">620</span>                // no matching altLoc group found.<a name="line.620"></a>
<span class="sourceLineNo">621</span>                // build it up.<a name="line.621"></a>
<span class="sourceLineNo">622</span><a name="line.622"></a>
<span class="sourceLineNo">623</span>                if ( groupCode3.equals(currentGroup.getPDBName())) {<a name="line.623"></a>
<span class="sourceLineNo">624</span>                        if ( currentGroup.getAtoms().size() == 0) {<a name="line.624"></a>
<span class="sourceLineNo">625</span>                                //System.out.println("current group is empty " + current_group + " " + altLoc);<a name="line.625"></a>
<span class="sourceLineNo">626</span>                                return currentGroup;<a name="line.626"></a>
<span class="sourceLineNo">627</span>                        }<a name="line.627"></a>
<span class="sourceLineNo">628</span>                        //System.out.println("cloning current group " + current_group + " " + current_group.getAtoms().get(0).getAltLoc() + " altLoc " + altLoc);<a name="line.628"></a>
<span class="sourceLineNo">629</span>                        Group altLocG = (Group) currentGroup.clone();<a name="line.629"></a>
<span class="sourceLineNo">630</span>                        // drop atoms from cloned group...<a name="line.630"></a>
<span class="sourceLineNo">631</span>                        // https://redmine.open-bio.org/issues/3307<a name="line.631"></a>
<span class="sourceLineNo">632</span>                        altLocG.setAtoms(new ArrayList&lt;Atom&gt;());<a name="line.632"></a>
<span class="sourceLineNo">633</span>                        altLocG.getAltLocs().clear();<a name="line.633"></a>
<span class="sourceLineNo">634</span>                        currentGroup.addAltLoc(altLocG);<a name="line.634"></a>
<span class="sourceLineNo">635</span>                        return altLocG;<a name="line.635"></a>
<span class="sourceLineNo">636</span>                }<a name="line.636"></a>
<span class="sourceLineNo">637</span><a name="line.637"></a>
<span class="sourceLineNo">638</span>                //      System.out.println("new  group " + recordName + " " + aminoCode1 + " " +groupCode3);<a name="line.638"></a>
<span class="sourceLineNo">639</span>                //String recordName,Character aminoCode1, long seq_id,String groupCode3) {<a name="line.639"></a>
<span class="sourceLineNo">640</span>                Group altLocG = getNewGroup(recordName,aminoCode1,seq_id,groupCode3);<a name="line.640"></a>
<span class="sourceLineNo">641</span><a name="line.641"></a>
<span class="sourceLineNo">642</span>                altLocG.setPDBName(groupCode3);<a name="line.642"></a>
<span class="sourceLineNo">643</span>                altLocG.setResidueNumber(currentGroup.getResidueNumber());<a name="line.643"></a>
<span class="sourceLineNo">644</span>                currentGroup.addAltLoc(altLocG);<a name="line.644"></a>
<span class="sourceLineNo">645</span>                return altLocG;<a name="line.645"></a>
<span class="sourceLineNo">646</span>        }<a name="line.646"></a>
<span class="sourceLineNo">647</span><a name="line.647"></a>
<span class="sourceLineNo">648</span>        /** <a name="line.648"></a>
<span class="sourceLineNo">649</span>         * Start the parsing<a name="line.649"></a>
<span class="sourceLineNo">650</span>         */<a name="line.650"></a>
<span class="sourceLineNo">651</span>        @Override<a name="line.651"></a>
<span class="sourceLineNo">652</span>        public void documentStart() {<a name="line.652"></a>
<span class="sourceLineNo">653</span>                structure = new StructureImpl();<a name="line.653"></a>
<span class="sourceLineNo">654</span><a name="line.654"></a>
<span class="sourceLineNo">655</span>                currentChain        = null;<a name="line.655"></a>
<span class="sourceLineNo">656</span>                currentGroup            = null;<a name="line.656"></a>
<span class="sourceLineNo">657</span>                currentNmrModelNumber   = null;<a name="line.657"></a>
<span class="sourceLineNo">658</span>                //atomCount                     = 0;<a name="line.658"></a>
<span class="sourceLineNo">659</span><a name="line.659"></a>
<span class="sourceLineNo">660</span>                allModels     = new ArrayList&lt;List&lt;Chain&gt;&gt;();<a name="line.660"></a>
<span class="sourceLineNo">661</span>                currentModel  = new ArrayList&lt;Chain&gt;();<a name="line.661"></a>
<span class="sourceLineNo">662</span>                entities      = new ArrayList&lt;Entity&gt;();<a name="line.662"></a>
<span class="sourceLineNo">663</span>                entityPolys   = new ArrayList&lt;&gt;();<a name="line.663"></a>
<span class="sourceLineNo">664</span>                strucRefs     = new ArrayList&lt;StructRef&gt;();<a name="line.664"></a>
<span class="sourceLineNo">665</span>                seqResChains  = new ArrayList&lt;Chain&gt;();<a name="line.665"></a>
<span class="sourceLineNo">666</span>                entityChains  = new ArrayList&lt;Chain&gt;();<a name="line.666"></a>
<span class="sourceLineNo">667</span>                structAsyms   = new ArrayList&lt;StructAsym&gt;();<a name="line.667"></a>
<span class="sourceLineNo">668</span><a name="line.668"></a>
<span class="sourceLineNo">669</span>                asymId2entityId = new HashMap&lt;String,String&gt;();<a name="line.669"></a>
<span class="sourceLineNo">670</span>                asymId2authorId = new HashMap&lt;&gt;();<a name="line.670"></a>
<span class="sourceLineNo">671</span>                structOpers   = new ArrayList&lt;PdbxStructOperList&gt;();<a name="line.671"></a>
<span class="sourceLineNo">672</span>                strucAssemblies = new ArrayList&lt;PdbxStructAssembly&gt;();<a name="line.672"></a>
<span class="sourceLineNo">673</span>                strucAssemblyGens = new ArrayList&lt;PdbxStructAssemblyGen&gt;();<a name="line.673"></a>
<span class="sourceLineNo">674</span>                entitySrcGens = new ArrayList&lt;EntitySrcGen&gt;();<a name="line.674"></a>
<span class="sourceLineNo">675</span>                entitySrcNats = new ArrayList&lt;EntitySrcNat&gt;();<a name="line.675"></a>
<span class="sourceLineNo">676</span>                entitySrcSyns = new ArrayList&lt;EntitySrcSyn&gt;();<a name="line.676"></a>
<span class="sourceLineNo">677</span>                structConn = new ArrayList&lt;StructConn&gt;();<a name="line.677"></a>
<span class="sourceLineNo">678</span>                structNcsOper = new ArrayList&lt;StructNcsOper&gt;();<a name="line.678"></a>
<span class="sourceLineNo">679</span>                sequenceDifs = new ArrayList&lt;StructRefSeqDif&gt;();<a name="line.679"></a>
<span class="sourceLineNo">680</span>                structSiteGens = new ArrayList&lt;StructSiteGen&gt;();<a name="line.680"></a>
<span class="sourceLineNo">681</span>        }<a name="line.681"></a>
<span class="sourceLineNo">682</span><a name="line.682"></a>
<span class="sourceLineNo">683</span><a name="line.683"></a>
<span class="sourceLineNo">684</span>        @Override<a name="line.684"></a>
<span class="sourceLineNo">685</span>        public void documentEnd() {<a name="line.685"></a>
<span class="sourceLineNo">686</span><a name="line.686"></a>
<span class="sourceLineNo">687</span>                // Expected that there is one current_chain that needs to be added to the model<a name="line.687"></a>
<span class="sourceLineNo">688</span>                // When in headerOnly mode, no Atoms are read, and there will not be an active<a name="line.688"></a>
<span class="sourceLineNo">689</span>                // current_chain.<a name="line.689"></a>
<span class="sourceLineNo">690</span>                if ( currentChain != null ) {<a name="line.690"></a>
<span class="sourceLineNo">691</span><a name="line.691"></a>
<span class="sourceLineNo">692</span>                        currentChain.addGroup(currentGroup);<a name="line.692"></a>
<span class="sourceLineNo">693</span>                        if (isKnownChain(currentChain.getId(),currentModel) == null) {<a name="line.693"></a>
<span class="sourceLineNo">694</span>                                currentModel.add(currentChain);<a name="line.694"></a>
<span class="sourceLineNo">695</span>                        }<a name="line.695"></a>
<span class="sourceLineNo">696</span>                } else if (!params.isHeaderOnly()){<a name="line.696"></a>
<span class="sourceLineNo">697</span>                        logger.warn("current chain is null at end of document.");<a name="line.697"></a>
<span class="sourceLineNo">698</span>                }<a name="line.698"></a>
<span class="sourceLineNo">699</span><a name="line.699"></a>
<span class="sourceLineNo">700</span>                allModels.add(currentModel);<a name="line.700"></a>
<span class="sourceLineNo">701</span><a name="line.701"></a>
<span class="sourceLineNo">702</span>                // this populates the asymId2authorId and asymId2entityId maps, needed in header only mode to get the mapping <a name="line.702"></a>
<span class="sourceLineNo">703</span>                // between the 2 chain identifiers.<a name="line.703"></a>
<span class="sourceLineNo">704</span>                initMaps();<a name="line.704"></a>
<span class="sourceLineNo">705</span><a name="line.705"></a>
<span class="sourceLineNo">706</span>                for (StructAsym asym : structAsyms) {<a name="line.706"></a>
<span class="sourceLineNo">707</span><a name="line.707"></a>
<span class="sourceLineNo">708</span>                        logger.debug("Entity {} matches asym_id: {}", asym.getEntity_id(), asym.getId() );<a name="line.708"></a>
<span class="sourceLineNo">709</span><a name="line.709"></a>
<span class="sourceLineNo">710</span>                        Chain s = getEntityChain(asym.getEntity_id());<a name="line.710"></a>
<span class="sourceLineNo">711</span>                        Chain seqres = (Chain)s.clone();<a name="line.711"></a>
<span class="sourceLineNo">712</span>                        // to solve issue #160 (e.g. 3u7t)<a name="line.712"></a>
<span class="sourceLineNo">713</span>                        seqres = removeSeqResHeterogeneity(seqres);<a name="line.713"></a>
<span class="sourceLineNo">714</span>                        seqres.setId(asym.getId());<a name="line.714"></a>
<span class="sourceLineNo">715</span>                        if (asymId2authorId.get(asym.getId()) !=null ){ <a name="line.715"></a>
<span class="sourceLineNo">716</span>                                seqres.setName(asymId2authorId.get(asym.getId()));<a name="line.716"></a>
<span class="sourceLineNo">717</span>                        } else {<a name="line.717"></a>
<span class="sourceLineNo">718</span>                                seqres.setName(asym.getId());<a name="line.718"></a>
<span class="sourceLineNo">719</span>                        }<a name="line.719"></a>
<span class="sourceLineNo">720</span><a name="line.720"></a>
<span class="sourceLineNo">721</span>                        EntityType type = null;<a name="line.721"></a>
<span class="sourceLineNo">722</span>                        try {<a name="line.722"></a>
<span class="sourceLineNo">723</span>                                Entity ent = getEntity(Integer.parseInt(asym.getEntity_id()));<a name="line.723"></a>
<span class="sourceLineNo">724</span>                                type = EntityType.entityTypeFromString(ent.getType());<a name="line.724"></a>
<span class="sourceLineNo">725</span>                        } catch (NumberFormatException e) {<a name="line.725"></a>
<span class="sourceLineNo">726</span>                                logger.debug("Could not parse integer from entity id field {}", asym.getEntity_id());<a name="line.726"></a>
<span class="sourceLineNo">727</span>                        }<a name="line.727"></a>
<span class="sourceLineNo">728</span><a name="line.728"></a>
<span class="sourceLineNo">729</span>                        // we'll only add seqres chains that are polymeric or unknown<a name="line.729"></a>
<span class="sourceLineNo">730</span>                        if (type==null || type==EntityType.POLYMER ) {<a name="line.730"></a>
<span class="sourceLineNo">731</span>                                seqResChains.add(seqres);       <a name="line.731"></a>
<span class="sourceLineNo">732</span>                        }<a name="line.732"></a>
<span class="sourceLineNo">733</span><a name="line.733"></a>
<span class="sourceLineNo">734</span>                        logger.debug(" seqres: " + asym.getId() + " " + seqres + "&lt;") ;<a name="line.734"></a>
<span class="sourceLineNo">735</span>                        // adding the entities to structure<a name="line.735"></a>
<span class="sourceLineNo">736</span>                        addEntities(asym);<a name="line.736"></a>
<span class="sourceLineNo">737</span><a name="line.737"></a>
<span class="sourceLineNo">738</span>                }<a name="line.738"></a>
<span class="sourceLineNo">739</span><a name="line.739"></a>
<span class="sourceLineNo">740</span>                if (structAsyms.isEmpty()) {<a name="line.740"></a>
<span class="sourceLineNo">741</span>                        logger.warn("No _struct_asym category in file, no SEQRES groups will be added.");<a name="line.741"></a>
<span class="sourceLineNo">742</span>                }<a name="line.742"></a>
<span class="sourceLineNo">743</span><a name="line.743"></a>
<span class="sourceLineNo">744</span>                // entities<a name="line.744"></a>
<span class="sourceLineNo">745</span>                // In addEntities above we created the entities if they were present in the file<a name="line.745"></a>
<span class="sourceLineNo">746</span>                // Now we need to make sure that they are linked to chains and also that if they are not present in the file we need to add them now<a name="line.746"></a>
<span class="sourceLineNo">747</span>                linkEntities();<a name="line.747"></a>
<span class="sourceLineNo">748</span><a name="line.748"></a>
<span class="sourceLineNo">749</span>                // now that we know the entities, we can add all chains to structure so that they are stored<a name="line.749"></a>
<span class="sourceLineNo">750</span>                // properly as polymer/nonpolymer/water chains inside structure<a name="line.750"></a>
<span class="sourceLineNo">751</span>                for (List&lt;Chain&gt; model:allModels) {<a name="line.751"></a>
<span class="sourceLineNo">752</span>                        structure.addModel(model);<a name="line.752"></a>
<span class="sourceLineNo">753</span>                }<a name="line.753"></a>
<span class="sourceLineNo">754</span><a name="line.754"></a>
<span class="sourceLineNo">755</span>                // Only align if requested (default) and not when headerOnly mode with no Atoms.<a name="line.755"></a>
<span class="sourceLineNo">756</span>                // Otherwise, we store the empty SeqRes Groups unchanged in the right chains.<a name="line.756"></a>
<span class="sourceLineNo">757</span>                if ( params.isAlignSeqRes() &amp;&amp; !params.isHeaderOnly() ){<a name="line.757"></a>
<span class="sourceLineNo">758</span>                        logger.debug("Parsing mode align_seqres, will parse SEQRES and align to ATOM sequence");<a name="line.758"></a>
<span class="sourceLineNo">759</span>                        alignSeqRes();<a name="line.759"></a>
<span class="sourceLineNo">760</span>                } else {<a name="line.760"></a>
<span class="sourceLineNo">761</span>                        logger.debug("Parsing mode unalign_seqres, will parse SEQRES but not align it to ATOM sequence");<a name="line.761"></a>
<span class="sourceLineNo">762</span>                        SeqRes2AtomAligner.storeUnAlignedSeqRes(structure, seqResChains, params.isHeaderOnly());<a name="line.762"></a>
<span class="sourceLineNo">763</span>                }<a name="line.763"></a>
<span class="sourceLineNo">764</span><a name="line.764"></a>
<span class="sourceLineNo">765</span><a name="line.765"></a>
<span class="sourceLineNo">766</span>                // Now make sure all altlocgroups have all the atoms in all the groups<a name="line.766"></a>
<span class="sourceLineNo">767</span>                StructureTools.cleanUpAltLocs(structure);<a name="line.767"></a>
<span class="sourceLineNo">768</span><a name="line.768"></a>
<span class="sourceLineNo">769</span><a name="line.769"></a>
<span class="sourceLineNo">770</span>                // NOTE bonds and charges can only be done at this point that the chain id mapping is properly sorted out<a name="line.770"></a>
<span class="sourceLineNo">771</span>                if (!params.isHeaderOnly()) {<a name="line.771"></a>
<span class="sourceLineNo">772</span>                        if ( params.shouldCreateAtomBonds()) {<a name="line.772"></a>
<span class="sourceLineNo">773</span>                                addBonds();<a name="line.773"></a>
<span class="sourceLineNo">774</span>                        }<a name="line.774"></a>
<span class="sourceLineNo">775</span><a name="line.775"></a>
<span class="sourceLineNo">776</span>                        if ( params.shouldCreateAtomCharges()) {<a name="line.776"></a>
<span class="sourceLineNo">777</span>                                addCharges();<a name="line.777"></a>
<span class="sourceLineNo">778</span>                        }<a name="line.778"></a>
<span class="sourceLineNo">779</span>                }<a name="line.779"></a>
<span class="sourceLineNo">780</span><a name="line.780"></a>
<span class="sourceLineNo">781</span>                if (!params.isHeaderOnly()) {<a name="line.781"></a>
<span class="sourceLineNo">782</span><a name="line.782"></a>
<span class="sourceLineNo">783</span>                        // Do structure.setSites(sites) after any chain renaming to be like PDB.<a name="line.783"></a>
<span class="sourceLineNo">784</span>                        addSites();<a name="line.784"></a>
<span class="sourceLineNo">785</span>                }<a name="line.785"></a>
<span class="sourceLineNo">786</span><a name="line.786"></a>
<span class="sourceLineNo">787</span><a name="line.787"></a>
<span class="sourceLineNo">788</span><a name="line.788"></a>
<span class="sourceLineNo">789</span>                // set the oligomeric state info in the header...<a name="line.789"></a>
<span class="sourceLineNo">790</span>                if (params.isParseBioAssembly()) {<a name="line.790"></a>
<span class="sourceLineNo">791</span><a name="line.791"></a>
<span class="sourceLineNo">792</span>                        // the more detailed mapping of chains to rotation operations happens in StructureIO...<a name="line.792"></a>
<span class="sourceLineNo">793</span><a name="line.793"></a>
<span class="sourceLineNo">794</span>                        Map&lt;Integer,BioAssemblyInfo&gt; bioAssemblies = new HashMap&lt;Integer, BioAssemblyInfo&gt;();<a name="line.794"></a>
<span class="sourceLineNo">795</span><a name="line.795"></a>
<span class="sourceLineNo">796</span>                        for ( PdbxStructAssembly psa : strucAssemblies){<a name="line.796"></a>
<span class="sourceLineNo">797</span><a name="line.797"></a>
<span class="sourceLineNo">798</span>                                List&lt;PdbxStructAssemblyGen&gt; psags = new ArrayList&lt;PdbxStructAssemblyGen&gt;(1);<a name="line.798"></a>
<span class="sourceLineNo">799</span><a name="line.799"></a>
<span class="sourceLineNo">800</span>                                for ( PdbxStructAssemblyGen psag: strucAssemblyGens ) {<a name="line.800"></a>
<span class="sourceLineNo">801</span>                                        if ( psag.getAssembly_id().equals(psa.getId())) {<a name="line.801"></a>
<span class="sourceLineNo">802</span>                                                psags.add(psag);<a name="line.802"></a>
<span class="sourceLineNo">803</span>                                        }<a name="line.803"></a>
<span class="sourceLineNo">804</span>                                }<a name="line.804"></a>
<span class="sourceLineNo">805</span><a name="line.805"></a>
<span class="sourceLineNo">806</span>                                BiologicalAssemblyBuilder builder = new BiologicalAssemblyBuilder();<a name="line.806"></a>
<span class="sourceLineNo">807</span><a name="line.807"></a>
<span class="sourceLineNo">808</span>                                // these are the transformations that need to be applied to our model<a name="line.808"></a>
<span class="sourceLineNo">809</span>                                List&lt;BiologicalAssemblyTransformation&gt; transformations = builder.getBioUnitTransformationList(psa, psags, structOpers);<a name="line.809"></a>
<span class="sourceLineNo">810</span><a name="line.810"></a>
<span class="sourceLineNo">811</span>                                int bioAssemblyId = -1;<a name="line.811"></a>
<span class="sourceLineNo">812</span>                                try {<a name="line.812"></a>
<span class="sourceLineNo">813</span>                                        bioAssemblyId = Integer.parseInt(psa.getId());<a name="line.813"></a>
<span class="sourceLineNo">814</span>                                } catch (NumberFormatException e) {<a name="line.814"></a>
<span class="sourceLineNo">815</span>                                        logger.info("Could not parse a numerical bio assembly id from '{}'",psa.getId());<a name="line.815"></a>
<span class="sourceLineNo">816</span>                                }<a name="line.816"></a>
<span class="sourceLineNo">817</span><a name="line.817"></a>
<span class="sourceLineNo">818</span>                                // if bioassembly id is not numerical we throw it away<a name="line.818"></a>
<span class="sourceLineNo">819</span>                                // this happens usually for viral capsid entries, like 1ei7<a name="line.819"></a>
<span class="sourceLineNo">820</span>                                // see issue #230 in github<a name="line.820"></a>
<span class="sourceLineNo">821</span>                                if (bioAssemblyId!=-1) {<a name="line.821"></a>
<span class="sourceLineNo">822</span>                                        int mmSize = 0;<a name="line.822"></a>
<span class="sourceLineNo">823</span>                                        // note that the transforms contain asym ids of both polymers and non-polymers<a name="line.823"></a>
<span class="sourceLineNo">824</span>                                        // For the mmsize, we are only interested in the polymers<a name="line.824"></a>
<span class="sourceLineNo">825</span>                                        for (BiologicalAssemblyTransformation transf:transformations) {<a name="line.825"></a>
<span class="sourceLineNo">826</span>                                                Chain c = structure.getChain(transf.getChainId());<a name="line.826"></a>
<span class="sourceLineNo">827</span>                                                if (c==null) {<a name="line.827"></a>
<span class="sourceLineNo">828</span>                                                        logger.info("Could not find asym id {} specified in struct_assembly_gen", transf.getChainId());<a name="line.828"></a>
<span class="sourceLineNo">829</span>                                                        continue;<a name="line.829"></a>
<span class="sourceLineNo">830</span>                                                }<a name="line.830"></a>
<span class="sourceLineNo">831</span>                                                if (c.getEntityType() == EntityType.POLYMER &amp;&amp;<a name="line.831"></a>
<span class="sourceLineNo">832</span>                                                        // for entries like 4kro, sugars are annotated as polymers but we<a name="line.832"></a>
<span class="sourceLineNo">833</span>                                                        // don't want them in the macromolecularSize count<a name="line.833"></a>
<span class="sourceLineNo">834</span>                                                        !c.getEntityInfo().getDescription().contains("SUGAR") ) {<a name="line.834"></a>
<span class="sourceLineNo">835</span>                                                                <a name="line.835"></a>
<span class="sourceLineNo">836</span>                                                                mmSize++;<a name="line.836"></a>
<span class="sourceLineNo">837</span>                                                        }<a name="line.837"></a>
<span class="sourceLineNo">838</span>                                        }<a name="line.838"></a>
<span class="sourceLineNo">839</span>                                        <a name="line.839"></a>
<span class="sourceLineNo">840</span>                                        BioAssemblyInfo bioAssembly = new BioAssemblyInfo();<a name="line.840"></a>
<span class="sourceLineNo">841</span>                                        bioAssembly.setId(bioAssemblyId);<a name="line.841"></a>
<span class="sourceLineNo">842</span>                                        bioAssembly.setMacromolecularSize(mmSize);<a name="line.842"></a>
<span class="sourceLineNo">843</span>                                        bioAssembly.setTransforms(transformations);<a name="line.843"></a>
<span class="sourceLineNo">844</span>                                        bioAssemblies.put(bioAssemblyId,bioAssembly);<a name="line.844"></a>
<span class="sourceLineNo">845</span>                                }<a name="line.845"></a>
<span class="sourceLineNo">846</span><a name="line.846"></a>
<span class="sourceLineNo">847</span>                        }<a name="line.847"></a>
<span class="sourceLineNo">848</span>                        structure.getPDBHeader().setBioAssemblies(bioAssemblies);<a name="line.848"></a>
<span class="sourceLineNo">849</span>                }<a name="line.849"></a>
<span class="sourceLineNo">850</span><a name="line.850"></a>
<span class="sourceLineNo">851</span>                setStructNcsOps();<a name="line.851"></a>
<span class="sourceLineNo">852</span>                <a name="line.852"></a>
<span class="sourceLineNo">853</span>                setCrystallographicInfoMetadata();<a name="line.853"></a>
<span class="sourceLineNo">854</span><a name="line.854"></a>
<span class="sourceLineNo">855</span><a name="line.855"></a>
<span class="sourceLineNo">856</span>                Map&lt;String,List&lt;SeqMisMatch&gt;&gt; misMatchMap = new HashMap&lt;String, List&lt;SeqMisMatch&gt;&gt;();<a name="line.856"></a>
<span class="sourceLineNo">857</span>                for (StructRefSeqDif sdif : sequenceDifs) {<a name="line.857"></a>
<span class="sourceLineNo">858</span>                        SeqMisMatch misMatch = new SeqMisMatchImpl();<a name="line.858"></a>
<span class="sourceLineNo">859</span>                        misMatch.setDetails(sdif.getDetails());<a name="line.859"></a>
<span class="sourceLineNo">860</span><a name="line.860"></a>
<span class="sourceLineNo">861</span>                        String insCode = sdif.getPdbx_pdb_ins_code();<a name="line.861"></a>
<span class="sourceLineNo">862</span>                        if ( insCode != null &amp;&amp; insCode.equals("?"))<a name="line.862"></a>
<span class="sourceLineNo">863</span>                                insCode = null;<a name="line.863"></a>
<span class="sourceLineNo">864</span>                        misMatch.setInsCode(insCode);<a name="line.864"></a>
<span class="sourceLineNo">865</span>                        misMatch.setOrigGroup(sdif.getDb_mon_id());<a name="line.865"></a>
<span class="sourceLineNo">866</span>                        misMatch.setPdbGroup(sdif.getMon_id());<a name="line.866"></a>
<span class="sourceLineNo">867</span>                        misMatch.setPdbResNum(sdif.getPdbx_auth_seq_num());<a name="line.867"></a>
<span class="sourceLineNo">868</span>                        misMatch.setUniProtId(sdif.getPdbx_seq_db_accession_code());<a name="line.868"></a>
<span class="sourceLineNo">869</span>                        misMatch.setSeqNum(sdif.getSeq_num());<a name="line.869"></a>
<span class="sourceLineNo">870</span><a name="line.870"></a>
<span class="sourceLineNo">871</span><a name="line.871"></a>
<span class="sourceLineNo">872</span>                        List&lt;SeqMisMatch&gt; mms = misMatchMap.get(sdif.getPdbx_pdb_strand_id());<a name="line.872"></a>
<span class="sourceLineNo">873</span>                        if ( mms == null) {<a name="line.873"></a>
<span class="sourceLineNo">874</span>                                mms = new ArrayList&lt;SeqMisMatch&gt;();<a name="line.874"></a>
<span class="sourceLineNo">875</span>                                misMatchMap.put(sdif.getPdbx_pdb_strand_id(),mms);<a name="line.875"></a>
<span class="sourceLineNo">876</span>                        }<a name="line.876"></a>
<span class="sourceLineNo">877</span>                        mms.add(misMatch);<a name="line.877"></a>
<span class="sourceLineNo">878</span><a name="line.878"></a>
<span class="sourceLineNo">879</span>                }<a name="line.879"></a>
<span class="sourceLineNo">880</span><a name="line.880"></a>
<span class="sourceLineNo">881</span>                for (String chainId : misMatchMap.keySet()){<a name="line.881"></a>
<span class="sourceLineNo">882</span><a name="line.882"></a>
<span class="sourceLineNo">883</span>                        Chain chain = structure.getPolyChainByPDB(chainId);<a name="line.883"></a>
<span class="sourceLineNo">884</span><a name="line.884"></a>
<span class="sourceLineNo">885</span>                        if ( chain == null) {<a name="line.885"></a>
<span class="sourceLineNo">886</span>                                logger.warn("Could not set mismatches for chain with author id" + chainId);<a name="line.886"></a>
<span class="sourceLineNo">887</span>                                continue;<a name="line.887"></a>
<span class="sourceLineNo">888</span>                        }<a name="line.888"></a>
<span class="sourceLineNo">889</span><a name="line.889"></a>
<span class="sourceLineNo">890</span>                        chain.setSeqMisMatches(misMatchMap.get(chainId));<a name="line.890"></a>
<span class="sourceLineNo">891</span><a name="line.891"></a>
<span class="sourceLineNo">892</span><a name="line.892"></a>
<span class="sourceLineNo">893</span>                }<a name="line.893"></a>
<span class="sourceLineNo">894</span><a name="line.894"></a>
<span class="sourceLineNo">895</span>        }<a name="line.895"></a>
<span class="sourceLineNo">896</span><a name="line.896"></a>
<span class="sourceLineNo">897</span>        /**<a name="line.897"></a>
<span class="sourceLineNo">898</span>         * Here we link entities to chains.<a name="line.898"></a>
<span class="sourceLineNo">899</span>         * Also if entities are not present in file, this initialises the entities with some heuristics, see {@link org.biojava.nbio.structure.io.EntityFinder}<a name="line.899"></a>
<span class="sourceLineNo">900</span>         */<a name="line.900"></a>
<span class="sourceLineNo">901</span>        private void linkEntities() {<a name="line.901"></a>
<span class="sourceLineNo">902</span><a name="line.902"></a>
<span class="sourceLineNo">903</span>                for (int i =0; i&lt; allModels.size() ; i++){<a name="line.903"></a>
<span class="sourceLineNo">904</span>                        for (Chain chain : allModels.get(i)) {<a name="line.904"></a>
<span class="sourceLineNo">905</span>                                //logger.info("linking entities for " + chain.getId() + " "  + chain.getName());<a name="line.905"></a>
<span class="sourceLineNo">906</span>                                String entityId = asymId2entityId.get(chain.getId());<a name="line.906"></a>
<span class="sourceLineNo">907</span><a name="line.907"></a>
<span class="sourceLineNo">908</span>                                if (entityId==null) {<a name="line.908"></a>
<span class="sourceLineNo">909</span>                                        // this can happen for instance if the cif file didn't have _struct_asym category at all<a name="line.909"></a>
<span class="sourceLineNo">910</span>                                        // and thus we have no asymId2entityId mapping at all<a name="line.910"></a>
<span class="sourceLineNo">911</span>                                        logger.info("No entity id could be found for chain {}", chain.getId());<a name="line.911"></a>
<span class="sourceLineNo">912</span>                                        continue;<a name="line.912"></a>
<span class="sourceLineNo">913</span>                                }<a name="line.913"></a>
<span class="sourceLineNo">914</span>                                int eId = Integer.parseInt(entityId);<a name="line.914"></a>
<span class="sourceLineNo">915</span><a name="line.915"></a>
<span class="sourceLineNo">916</span>                                // Entities are not added for non-polymeric entities, if a chain is non-polymeric its entity won't be found.<a name="line.916"></a>
<span class="sourceLineNo">917</span>                                // TODO: add all entities and unique compounds and add methods to directly get polymer or non-polymer<a name="line.917"></a>
<span class="sourceLineNo">918</span>                                // asyms (chains).  Either create a unique StructureImpl or modify existing for a better representation of the<a name="line.918"></a>
<span class="sourceLineNo">919</span>                                // mmCIF internal data structures but is compatible with Structure interface.<a name="line.919"></a>
<span class="sourceLineNo">920</span>                                // Some examples of PDB entries with this kind of problem:<a name="line.920"></a>
<span class="sourceLineNo">921</span>                                //   - 2uub: asym_id X, chainName Z, entity_id 24: fully non-polymeric but still with its own chainName<a name="line.921"></a>
<span class="sourceLineNo">922</span>                                //   - 3o6j: asym_id K, chainName Z, entity_id 6 : a single water molecule<a name="line.922"></a>
<span class="sourceLineNo">923</span>                                //   - 1dz9: asym_id K, chainName K, entity_id 6 : a potassium ion alone<a name="line.923"></a>
<span class="sourceLineNo">924</span><a name="line.924"></a>
<span class="sourceLineNo">925</span>                                EntityInfo entityInfo = structure.getEntityById(eId);<a name="line.925"></a>
<span class="sourceLineNo">926</span>                                if (entityInfo==null) {<a name="line.926"></a>
<span class="sourceLineNo">927</span>                                        // Supports the case where the only chain members were from non-polymeric entity that is missing.<a name="line.927"></a>
<span class="sourceLineNo">928</span>                                        // Solved by creating a new Compound(entity) to which this chain will belong.<a name="line.928"></a>
<span class="sourceLineNo">929</span>                                        logger.info("Could not find an Entity for entity_id {}, for chain id {}, creating a new Entity.",<a name="line.929"></a>
<span class="sourceLineNo">930</span>                                                        eId, chain.getId());<a name="line.930"></a>
<span class="sourceLineNo">931</span>                                        entityInfo = new EntityInfo();<a name="line.931"></a>
<span class="sourceLineNo">932</span>                                        entityInfo.setMolId(eId);<a name="line.932"></a>
<span class="sourceLineNo">933</span>                                        entityInfo.addChain(chain);<a name="line.933"></a>
<span class="sourceLineNo">934</span>                                        if (chain.isWaterOnly()) {<a name="line.934"></a>
<span class="sourceLineNo">935</span>                                                entityInfo.setType(EntityType.WATER);<a name="line.935"></a>
<span class="sourceLineNo">936</span>                                        } else {<a name="line.936"></a>
<span class="sourceLineNo">937</span>                                                entityInfo.setType(EntityType.NONPOLYMER);<a name="line.937"></a>
<span class="sourceLineNo">938</span>                                        }<a name="line.938"></a>
<span class="sourceLineNo">939</span>                                        chain.setEntityInfo(entityInfo);<a name="line.939"></a>
<span class="sourceLineNo">940</span>                                        structure.addEntityInfo(entityInfo);<a name="line.940"></a>
<span class="sourceLineNo">941</span>                                } else {<a name="line.941"></a>
<span class="sourceLineNo">942</span>                                        logger.debug("Adding chain with chain id {} (auth id {}) to Entity with entity_id {}",<a name="line.942"></a>
<span class="sourceLineNo">943</span>                                                        chain.getId(), chain.getName(), eId);<a name="line.943"></a>
<span class="sourceLineNo">944</span>                                        entityInfo.addChain(chain);<a name="line.944"></a>
<span class="sourceLineNo">945</span>                                        chain.setEntityInfo(entityInfo);<a name="line.945"></a>
<span class="sourceLineNo">946</span>                                }<a name="line.946"></a>
<span class="sourceLineNo">947</span><a name="line.947"></a>
<span class="sourceLineNo">948</span>                        }<a name="line.948"></a>
<span class="sourceLineNo">949</span><a name="line.949"></a>
<span class="sourceLineNo">950</span>                }<a name="line.950"></a>
<span class="sourceLineNo">951</span><a name="line.951"></a>
<span class="sourceLineNo">952</span>                // if no entity information was present in file we then go and find the entities heuristically with EntityFinder<a name="line.952"></a>
<span class="sourceLineNo">953</span>                List&lt;EntityInfo&gt; entityInfos = structure.getEntityInfos();<a name="line.953"></a>
<span class="sourceLineNo">954</span>                if (entityInfos==null || entityInfos.isEmpty()) {<a name="line.954"></a>
<span class="sourceLineNo">955</span><a name="line.955"></a>
<span class="sourceLineNo">956</span>                        List&lt;List&lt;Chain&gt;&gt; polyModels = new ArrayList&lt;&gt;();<a name="line.956"></a>
<span class="sourceLineNo">957</span>                        List&lt;List&lt;Chain&gt;&gt; nonPolyModels = new ArrayList&lt;&gt;();<a name="line.957"></a>
<span class="sourceLineNo">958</span>                        List&lt;List&lt;Chain&gt;&gt; waterModels = new ArrayList&lt;&gt;();<a name="line.958"></a>
<span class="sourceLineNo">959</span><a name="line.959"></a>
<span class="sourceLineNo">960</span>                        for (List&lt;Chain&gt; model:allModels) {<a name="line.960"></a>
<span class="sourceLineNo">961</span><a name="line.961"></a>
<span class="sourceLineNo">962</span>                                List&lt;Chain&gt; polyChains = new ArrayList&lt;&gt;();<a name="line.962"></a>
<span class="sourceLineNo">963</span>                                List&lt;Chain&gt; nonPolyChains = new ArrayList&lt;&gt;();<a name="line.963"></a>
<span class="sourceLineNo">964</span>                                List&lt;Chain&gt; waterChains = new ArrayList&lt;&gt;();<a name="line.964"></a>
<span class="sourceLineNo">965</span><a name="line.965"></a>
<span class="sourceLineNo">966</span>                                polyModels.add(polyChains);<a name="line.966"></a>
<span class="sourceLineNo">967</span>                                nonPolyModels.add(nonPolyChains);<a name="line.967"></a>
<span class="sourceLineNo">968</span>                                waterModels.add(waterChains);<a name="line.968"></a>
<span class="sourceLineNo">969</span><a name="line.969"></a>
<span class="sourceLineNo">970</span>                                for (Chain c:model) {<a name="line.970"></a>
<span class="sourceLineNo">971</span><a name="line.971"></a>
<span class="sourceLineNo">972</span>                                        // we only have entities for polymeric chains, all others are ignored for assigning entities<a name="line.972"></a>
<span class="sourceLineNo">973</span>                                        if (c.isWaterOnly()) {<a name="line.973"></a>
<span class="sourceLineNo">974</span>                                                waterChains.add(c);<a name="line.974"></a>
<span class="sourceLineNo">975</span><a name="line.975"></a>
<span class="sourceLineNo">976</span>                                        } else if (c.isPureNonPolymer()) {<a name="line.976"></a>
<span class="sourceLineNo">977</span>                                                nonPolyChains.add(c);<a name="line.977"></a>
<span class="sourceLineNo">978</span><a name="line.978"></a>
<span class="sourceLineNo">979</span>                                        } else {<a name="line.979"></a>
<span class="sourceLineNo">980</span>                                                polyChains.add(c);<a name="line.980"></a>
<span class="sourceLineNo">981</span>                                        }<a name="line.981"></a>
<span class="sourceLineNo">982</span>                                }<a name="line.982"></a>
<span class="sourceLineNo">983</span>                        }<a name="line.983"></a>
<span class="sourceLineNo">984</span><a name="line.984"></a>
<span class="sourceLineNo">985</span>                        entityInfos = EntityFinder.findPolyEntities(polyModels);<a name="line.985"></a>
<span class="sourceLineNo">986</span>                        EntityFinder.createPurelyNonPolyEntities(nonPolyModels, waterModels, entityInfos);<a name="line.986"></a>
<span class="sourceLineNo">987</span><a name="line.987"></a>
<span class="sourceLineNo">988</span><a name="line.988"></a>
<span class="sourceLineNo">989</span>                        structure.setEntityInfos(entityInfos);<a name="line.989"></a>
<span class="sourceLineNo">990</span>                }<a name="line.990"></a>
<span class="sourceLineNo">991</span><a name="line.991"></a>
<span class="sourceLineNo">992</span>                // final sanity check: it can happen that from the annotated entities some are not linked to any chains<a name="line.992"></a>
<span class="sourceLineNo">993</span>                // e.g. 3s26: a sugar entity does not have any chains associated to it (it seems to be happening with many sugar compounds)<a name="line.993"></a>
<span class="sourceLineNo">994</span>                // we simply log it, this can sign some other problems if the entities are used down the line<a name="line.994"></a>
<span class="sourceLineNo">995</span>                for (EntityInfo e:entityInfos) {<a name="line.995"></a>
<span class="sourceLineNo">996</span>                        if (e.getChains().isEmpty()) {<a name="line.996"></a>
<span class="sourceLineNo">997</span>                                logger.info("Entity {} '{}' has no chains associated to it",<a name="line.997"></a>
<span class="sourceLineNo">998</span>                                                e.getMolId()&lt;0?"with no entity id":e.getMolId(), e.getDescription());<a name="line.998"></a>
<span class="sourceLineNo">999</span>                        }<a name="line.999"></a>
<span class="sourceLineNo">1000</span>                }<a name="line.1000"></a>
<span class="sourceLineNo">1001</span><a name="line.1001"></a>
<span class="sourceLineNo">1002</span>        }<a name="line.1002"></a>
<span class="sourceLineNo">1003</span><a name="line.1003"></a>
<span class="sourceLineNo">1004</span>        private void addCharges() {<a name="line.1004"></a>
<span class="sourceLineNo">1005</span>                ChargeAdder.addCharges(structure);<a name="line.1005"></a>
<span class="sourceLineNo">1006</span>        }<a name="line.1006"></a>
<span class="sourceLineNo">1007</span><a name="line.1007"></a>
<span class="sourceLineNo">1008</span>        /**<a name="line.1008"></a>
<span class="sourceLineNo">1009</span>         * The method will return a new reference to a Chain with any consecutive groups<a name="line.1009"></a>
<span class="sourceLineNo">1010</span>         * having same residue numbers removed.<a name="line.1010"></a>
<span class="sourceLineNo">1011</span>         * This is necessary to solve the microheterogeneity issue in entries like 3u7t (see github issue #160)<a name="line.1011"></a>
<span class="sourceLineNo">1012</span>         * @param c<a name="line.1012"></a>
<span class="sourceLineNo">1013</span>         * @return<a name="line.1013"></a>
<span class="sourceLineNo">1014</span>         */<a name="line.1014"></a>
<span class="sourceLineNo">1015</span>        private static Chain removeSeqResHeterogeneity(Chain c) {<a name="line.1015"></a>
<span class="sourceLineNo">1016</span><a name="line.1016"></a>
<span class="sourceLineNo">1017</span>                Chain trimmedChain = new ChainImpl();<a name="line.1017"></a>
<span class="sourceLineNo">1018</span><a name="line.1018"></a>
<span class="sourceLineNo">1019</span>                ResidueNumber lastResNum = null;<a name="line.1019"></a>
<span class="sourceLineNo">1020</span><a name="line.1020"></a>
<span class="sourceLineNo">1021</span>                for (Group g:c.getAtomGroups()) {<a name="line.1021"></a>
<span class="sourceLineNo">1022</span><a name="line.1022"></a>
<span class="sourceLineNo">1023</span>                        // note we have to deep copy this, otherwise they stay linked and would get altered in addGroup(g)<a name="line.1023"></a>
<span class="sourceLineNo">1024</span>                        ResidueNumber currentResNum = new ResidueNumber(<a name="line.1024"></a>
<span class="sourceLineNo">1025</span>                                        g.getResidueNumber().getChainName(),<a name="line.1025"></a>
<span class="sourceLineNo">1026</span>                                        g.getResidueNumber().getSeqNum(),<a name="line.1026"></a>
<span class="sourceLineNo">1027</span>                                        g.getResidueNumber().getInsCode());<a name="line.1027"></a>
<span class="sourceLineNo">1028</span><a name="line.1028"></a>
<span class="sourceLineNo">1029</span>                        if (lastResNum == null || !lastResNum.equals(currentResNum) ) {<a name="line.1029"></a>
<span class="sourceLineNo">1030</span>                                trimmedChain.addGroup(g);<a name="line.1030"></a>
<span class="sourceLineNo">1031</span>                        } else {<a name="line.1031"></a>
<span class="sourceLineNo">1032</span>                                logger.debug("Removing seqres group because it seems to be repeated in entity_poly_seq, most likely has hetero='y': "+g);<a name="line.1032"></a>
<span class="sourceLineNo">1033</span>                        }<a name="line.1033"></a>
<span class="sourceLineNo">1034</span><a name="line.1034"></a>
<span class="sourceLineNo">1035</span>                        lastResNum = currentResNum;<a name="line.1035"></a>
<span class="sourceLineNo">1036</span><a name="line.1036"></a>
<span class="sourceLineNo">1037</span>                }<a name="line.1037"></a>
<span class="sourceLineNo">1038</span>                return trimmedChain;<a name="line.1038"></a>
<span class="sourceLineNo">1039</span>        }<a name="line.1039"></a>
<span class="sourceLineNo">1040</span><a name="line.1040"></a>
<span class="sourceLineNo">1041</span>        private void addBonds() {<a name="line.1041"></a>
<span class="sourceLineNo">1042</span>                BondMaker maker = new BondMaker(structure, params);<a name="line.1042"></a>
<span class="sourceLineNo">1043</span>                maker.makeBonds();<a name="line.1043"></a>
<span class="sourceLineNo">1044</span>                maker.formBondsFromStructConn(structConn);<a name="line.1044"></a>
<span class="sourceLineNo">1045</span>        }<a name="line.1045"></a>
<span class="sourceLineNo">1046</span><a name="line.1046"></a>
<span class="sourceLineNo">1047</span>        private void alignSeqRes() {<a name="line.1047"></a>
<span class="sourceLineNo">1048</span><a name="line.1048"></a>
<span class="sourceLineNo">1049</span>                logger.debug("Parsing mode align_seqres, will align to ATOM to SEQRES sequence");<a name="line.1049"></a>
<span class="sourceLineNo">1050</span><a name="line.1050"></a>
<span class="sourceLineNo">1051</span>                // fix SEQRES residue numbering for all models<a name="line.1051"></a>
<span class="sourceLineNo">1052</span><a name="line.1052"></a>
<span class="sourceLineNo">1053</span>                for (int model=0;model&lt;structure.nrModels();model++) {<a name="line.1053"></a>
<span class="sourceLineNo">1054</span><a name="line.1054"></a>
<span class="sourceLineNo">1055</span>                        List&lt;Chain&gt; atomList   = structure.getModel(model);<a name="line.1055"></a>
<span class="sourceLineNo">1056</span><a name="line.1056"></a>
<span class="sourceLineNo">1057</span>                        for (Chain seqResChain: seqResChains){<a name="line.1057"></a>
<span class="sourceLineNo">1058</span><a name="line.1058"></a>
<span class="sourceLineNo">1059</span>                                // this extracts the matching atom chain from atomList<a name="line.1059"></a>
<span class="sourceLineNo">1060</span>                                Chain atomChain = SeqRes2AtomAligner.getMatchingAtomRes(seqResChain, atomList, true);<a name="line.1060"></a>
<span class="sourceLineNo">1061</span><a name="line.1061"></a>
<span class="sourceLineNo">1062</span>                                if (atomChain == null) {<a name="line.1062"></a>
<span class="sourceLineNo">1063</span>                                        // most likely there's no observed residues at all for the seqres chain: can't map<a name="line.1063"></a>
<span class="sourceLineNo">1064</span>                                        // e.g. 3zyb: chains with asym_id L,M,N,O,P have no observed residues<a name="line.1064"></a>
<span class="sourceLineNo">1065</span>                                        logger.info("Could not map SEQRES chain with asym_id={} to any ATOM chain. Most likely there's no observed residues in the chain.",<a name="line.1065"></a>
<span class="sourceLineNo">1066</span>                                                        seqResChain.getId());<a name="line.1066"></a>
<span class="sourceLineNo">1067</span>                                        continue;<a name="line.1067"></a>
<span class="sourceLineNo">1068</span>                                }<a name="line.1068"></a>
<span class="sourceLineNo">1069</span><a name="line.1069"></a>
<span class="sourceLineNo">1070</span>                                //map the atoms to the seqres...<a name="line.1070"></a>
<span class="sourceLineNo">1071</span><a name="line.1071"></a>
<span class="sourceLineNo">1072</span>                                // we need to first clone the seqres so that they stay independent for different models<a name="line.1072"></a>
<span class="sourceLineNo">1073</span>                                List&lt;Group&gt; seqResGroups = new ArrayList&lt;Group&gt;();<a name="line.1073"></a>
<span class="sourceLineNo">1074</span>                                for (int i=0;i&lt;seqResChain.getAtomGroups().size();i++) {<a name="line.1074"></a>
<span class="sourceLineNo">1075</span>                                        seqResGroups.add((Group)seqResChain.getAtomGroups().get(i).clone());<a name="line.1075"></a>
<span class="sourceLineNo">1076</span>                                }<a name="line.1076"></a>
<span class="sourceLineNo">1077</span><a name="line.1077"></a>
<span class="sourceLineNo">1078</span>                                for ( int seqResPos = 0 ; seqResPos &lt; seqResGroups.size(); seqResPos++) {<a name="line.1078"></a>
<span class="sourceLineNo">1079</span>                                        Group seqresG = seqResGroups.get(seqResPos);<a name="line.1079"></a>
<span class="sourceLineNo">1080</span>                                        boolean found = false;<a name="line.1080"></a>
<span class="sourceLineNo">1081</span>                                        for ( Group atomG: atomChain.getAtomGroups()) {<a name="line.1081"></a>
<span class="sourceLineNo">1082</span><a name="line.1082"></a>
<span class="sourceLineNo">1083</span>                                                int internalNr = getInternalNr (atomG);<a name="line.1083"></a>
<span class="sourceLineNo">1084</span><a name="line.1084"></a>
<span class="sourceLineNo">1085</span>                                                if (seqresG.getResidueNumber().getSeqNum() == internalNr ) {<a name="line.1085"></a>
<span class="sourceLineNo">1086</span>                                                        seqResGroups.set(seqResPos, atomG);<a name="line.1086"></a>
<span class="sourceLineNo">1087</span>                                                        found = true;<a name="line.1087"></a>
<span class="sourceLineNo">1088</span>                                                        break;<a name="line.1088"></a>
<span class="sourceLineNo">1089</span>                                                }<a name="line.1089"></a>
<span class="sourceLineNo">1090</span><a name="line.1090"></a>
<span class="sourceLineNo">1091</span><a name="line.1091"></a>
<span class="sourceLineNo">1092</span>                                        }<a name="line.1092"></a>
<span class="sourceLineNo">1093</span>                                        if ( ! found)<a name="line.1093"></a>
<span class="sourceLineNo">1094</span>                                                // so far the residue number has tracked internal numbering.<a name="line.1094"></a>
<span class="sourceLineNo">1095</span>                                                // however there are no atom records, as such this can't be a PDB residue number...<a name="line.1095"></a>
<span class="sourceLineNo">1096</span>                                                seqresG.setResidueNumber(null);<a name="line.1096"></a>
<span class="sourceLineNo">1097</span>                                }<a name="line.1097"></a>
<span class="sourceLineNo">1098</span>                                atomChain.setSeqResGroups(seqResGroups);<a name="line.1098"></a>
<span class="sourceLineNo">1099</span><a name="line.1099"></a>
<span class="sourceLineNo">1100</span>                        }<a name="line.1100"></a>
<span class="sourceLineNo">1101</span>                }<a name="line.1101"></a>
<span class="sourceLineNo">1102</span>        }<a name="line.1102"></a>
<span class="sourceLineNo">1103</span><a name="line.1103"></a>
<span class="sourceLineNo">1104</span>        private int getInternalNr(Group atomG) {<a name="line.1104"></a>
<span class="sourceLineNo">1105</span>                if ( atomG.getType().equals(GroupType.AMINOACID)) {<a name="line.1105"></a>
<span class="sourceLineNo">1106</span>                        AminoAcidImpl aa = (AminoAcidImpl) atomG;<a name="line.1106"></a>
<span class="sourceLineNo">1107</span>                        return new Long(aa.getId()).intValue();<a name="line.1107"></a>
<span class="sourceLineNo">1108</span>                } else if ( atomG.getType().equals(GroupType.NUCLEOTIDE)) {<a name="line.1108"></a>
<span class="sourceLineNo">1109</span>                        NucleotideImpl nu = (NucleotideImpl) atomG;<a name="line.1109"></a>
<span class="sourceLineNo">1110</span>                        return new Long(nu.getId()).intValue();<a name="line.1110"></a>
<span class="sourceLineNo">1111</span>                } else {<a name="line.1111"></a>
<span class="sourceLineNo">1112</span>                        HetatomImpl he = (HetatomImpl) atomG;<a name="line.1112"></a>
<span class="sourceLineNo">1113</span>                        return new Long(he.getId()).intValue();<a name="line.1113"></a>
<span class="sourceLineNo">1114</span>                }<a name="line.1114"></a>
<span class="sourceLineNo">1115</span>        }<a name="line.1115"></a>
<span class="sourceLineNo">1116</span><a name="line.1116"></a>
<span class="sourceLineNo">1117</span>        private void addEntities(StructAsym asym) {<a name="line.1117"></a>
<span class="sourceLineNo">1118</span>                int eId = 0;<a name="line.1118"></a>
<span class="sourceLineNo">1119</span>                try {<a name="line.1119"></a>
<span class="sourceLineNo">1120</span>                        eId = Integer.parseInt(asym.getEntity_id());<a name="line.1120"></a>
<span class="sourceLineNo">1121</span>                } catch (NumberFormatException e) {<a name="line.1121"></a>
<span class="sourceLineNo">1122</span>                        logger.warn("Could not parse mol_id from string {}. Will use 0 for creating Entity",asym.getEntity_id());<a name="line.1122"></a>
<span class="sourceLineNo">1123</span>                }<a name="line.1123"></a>
<span class="sourceLineNo">1124</span>                Entity e = getEntity(eId);<a name="line.1124"></a>
<span class="sourceLineNo">1125</span><a name="line.1125"></a>
<span class="sourceLineNo">1126</span>                // for some mmCIF files like 1yrm all 3 of _entity_src_gen, _entity_src_nat and _pdbx_entity_src_syn are missing<a name="line.1126"></a>
<span class="sourceLineNo">1127</span>                // we need to fill the Compounds in some other way:<a name="line.1127"></a>
<span class="sourceLineNo">1128</span><a name="line.1128"></a>
<span class="sourceLineNo">1129</span>                EntityInfo entityInfo = structure.getEntityById(eId);<a name="line.1129"></a>
<span class="sourceLineNo">1130</span><a name="line.1130"></a>
<span class="sourceLineNo">1131</span>                if (entityInfo==null) {<a name="line.1131"></a>
<span class="sourceLineNo">1132</span>                        //logger.info("Creating new EntityInfo " + eId + " " + e.getId() + " " + e.getPdbx_description());<a name="line.1132"></a>
<span class="sourceLineNo">1133</span>                        entityInfo = new EntityInfo();<a name="line.1133"></a>
<span class="sourceLineNo">1134</span>                        entityInfo.setMolId(eId);<a name="line.1134"></a>
<span class="sourceLineNo">1135</span>                        // we only add the compound if a polymeric one (to match what the PDB parser does)<a name="line.1135"></a>
<span class="sourceLineNo">1136</span>                        if (e!=null) {<a name="line.1136"></a>
<span class="sourceLineNo">1137</span>                                entityInfo.setDescription(e.getPdbx_description());<a name="line.1137"></a>
<span class="sourceLineNo">1138</span><a name="line.1138"></a>
<span class="sourceLineNo">1139</span>                                EntityType eType = EntityType.entityTypeFromString(e.getType());<a name="line.1139"></a>
<span class="sourceLineNo">1140</span>                                if (eType!=null) {<a name="line.1140"></a>
<span class="sourceLineNo">1141</span>                                        entityInfo.setType(eType);<a name="line.1141"></a>
<span class="sourceLineNo">1142</span>                                } else {<a name="line.1142"></a>
<span class="sourceLineNo">1143</span>                                        logger.warn("Type '{}' is not recognised as a valid entity type for entity {}", e.getType(), eId);<a name="line.1143"></a>
<span class="sourceLineNo">1144</span>                                }<a name="line.1144"></a>
<span class="sourceLineNo">1145</span>                                addAncilliaryEntityData(asym, eId, e, entityInfo);<a name="line.1145"></a>
<span class="sourceLineNo">1146</span>                                structure.addEntityInfo(entityInfo);<a name="line.1146"></a>
<span class="sourceLineNo">1147</span>                                logger.debug("Adding Entity with entity id {} from _entity, with name: {}",eId, entityInfo.getDescription());<a name="line.1147"></a>
<span class="sourceLineNo">1148</span>                        }<a name="line.1148"></a>
<span class="sourceLineNo">1149</span>                }<a name="line.1149"></a>
<span class="sourceLineNo">1150</span>        }<a name="line.1150"></a>
<span class="sourceLineNo">1151</span><a name="line.1151"></a>
<span class="sourceLineNo">1152</span><a name="line.1152"></a>
<span class="sourceLineNo">1153</span>        /**<a name="line.1153"></a>
<span class="sourceLineNo">1154</span>         * Add any extra information to the entity information.<a name="line.1154"></a>
<span class="sourceLineNo">1155</span>         * @param asym <a name="line.1155"></a>
<span class="sourceLineNo">1156</span>         * @param entityId <a name="line.1156"></a>
<span class="sourceLineNo">1157</span>         * @param entity <a name="line.1157"></a>
<span class="sourceLineNo">1158</span>         * @param entityInfo <a name="line.1158"></a>
<span class="sourceLineNo">1159</span>         */<a name="line.1159"></a>
<span class="sourceLineNo">1160</span>        private void addAncilliaryEntityData(StructAsym asym, int entityId, Entity entity, EntityInfo entityInfo) {<a name="line.1160"></a>
<span class="sourceLineNo">1161</span>                // Loop through each of the entity types and add the corresponding data<a name="line.1161"></a>
<span class="sourceLineNo">1162</span>                // We're assuming if data is duplicated between sources it is consistent<a name="line.1162"></a>
<span class="sourceLineNo">1163</span>                // This is a potentially huge assumption...<a name="line.1163"></a>
<span class="sourceLineNo">1164</span><a name="line.1164"></a>
<span class="sourceLineNo">1165</span><a name="line.1165"></a>
<span class="sourceLineNo">1166</span>                for (EntitySrcGen esg : entitySrcGens) {<a name="line.1166"></a>
<span class="sourceLineNo">1167</span><a name="line.1167"></a>
<span class="sourceLineNo">1168</span>                        if (! esg.getEntity_id().equals(asym.getEntity_id()))<a name="line.1168"></a>
<span class="sourceLineNo">1169</span>                                continue;<a name="line.1169"></a>
<span class="sourceLineNo">1170</span><a name="line.1170"></a>
<span class="sourceLineNo">1171</span>                        addInformationFromESG(esg, entityId, entityInfo);<a name="line.1171"></a>
<span class="sourceLineNo">1172</span><a name="line.1172"></a>
<span class="sourceLineNo">1173</span>                }<a name="line.1173"></a>
<span class="sourceLineNo">1174</span><a name="line.1174"></a>
<span class="sourceLineNo">1175</span>                for (EntitySrcNat esn : entitySrcNats) {<a name="line.1175"></a>
<span class="sourceLineNo">1176</span>                        if (! esn.getEntity_id().equals(asym.getEntity_id()))<a name="line.1176"></a>
<span class="sourceLineNo">1177</span>                                continue;<a name="line.1177"></a>
<span class="sourceLineNo">1178</span>                        addInformationFromESN(esn, entityId, entityInfo);<a name="line.1178"></a>
<span class="sourceLineNo">1179</span><a name="line.1179"></a>
<span class="sourceLineNo">1180</span>                }<a name="line.1180"></a>
<span class="sourceLineNo">1181</span><a name="line.1181"></a>
<span class="sourceLineNo">1182</span>                for (EntitySrcSyn ess : entitySrcSyns) {<a name="line.1182"></a>
<span class="sourceLineNo">1183</span>                        if (! ess.getEntity_id().equals(asym.getEntity_id()))<a name="line.1183"></a>
<span class="sourceLineNo">1184</span>                                continue;<a name="line.1184"></a>
<span class="sourceLineNo">1185</span>                        addInfoFromESS(ess, entityId, entityInfo);<a name="line.1185"></a>
<span class="sourceLineNo">1186</span><a name="line.1186"></a>
<span class="sourceLineNo">1187</span>                }               <a name="line.1187"></a>
<span class="sourceLineNo">1188</span>        }<a name="line.1188"></a>
<span class="sourceLineNo">1189</span><a name="line.1189"></a>
<span class="sourceLineNo">1190</span>        /**<a name="line.1190"></a>
<span class="sourceLineNo">1191</span>         * Add the information from an ESG to a compound.<a name="line.1191"></a>
<span class="sourceLineNo">1192</span>         * @param entitySrcInfo<a name="line.1192"></a>
<span class="sourceLineNo">1193</span>         * @param entityId<a name="line.1193"></a>
<span class="sourceLineNo">1194</span>         * @param c<a name="line.1194"></a>
<span class="sourceLineNo">1195</span>         */<a name="line.1195"></a>
<span class="sourceLineNo">1196</span>        private void addInformationFromESG(EntitySrcGen entitySrcInfo, int entityId, EntityInfo c) {<a name="line.1196"></a>
<span class="sourceLineNo">1197</span>                c.setAtcc(entitySrcInfo.getPdbx_gene_src_atcc());<a name="line.1197"></a>
<span class="sourceLineNo">1198</span>                c.setCell(entitySrcInfo.getPdbx_gene_src_cell());<a name="line.1198"></a>
<span class="sourceLineNo">1199</span>                c.setOrganismCommon(entitySrcInfo.getGene_src_common_name());<a name="line.1199"></a>
<span class="sourceLineNo">1200</span>                c.setOrganismScientific(entitySrcInfo.getPdbx_gene_src_scientific_name());<a name="line.1200"></a>
<span class="sourceLineNo">1201</span>                c.setOrganismTaxId(entitySrcInfo.getPdbx_gene_src_ncbi_taxonomy_id());<a name="line.1201"></a>
<span class="sourceLineNo">1202</span>                c.setExpressionSystemTaxId(entitySrcInfo.getPdbx_host_org_ncbi_taxonomy_id());<a name="line.1202"></a>
<span class="sourceLineNo">1203</span>                c.setExpressionSystem(entitySrcInfo.getPdbx_host_org_scientific_name());<a name="line.1203"></a>
<span class="sourceLineNo">1204</span>        }<a name="line.1204"></a>
<span class="sourceLineNo">1205</span><a name="line.1205"></a>
<span class="sourceLineNo">1206</span>        /**<a name="line.1206"></a>
<span class="sourceLineNo">1207</span>         * Add the information to entity info from ESN.<a name="line.1207"></a>
<span class="sourceLineNo">1208</span>         * @param esn<a name="line.1208"></a>
<span class="sourceLineNo">1209</span>         * @param eId<a name="line.1209"></a>
<span class="sourceLineNo">1210</span>         * @param c<a name="line.1210"></a>
<span class="sourceLineNo">1211</span>         */<a name="line.1211"></a>
<span class="sourceLineNo">1212</span>        private void addInformationFromESN(EntitySrcNat esn, int eId, EntityInfo c) {<a name="line.1212"></a>
<span class="sourceLineNo">1213</span><a name="line.1213"></a>
<span class="sourceLineNo">1214</span>                c.setAtcc(esn.getPdbx_atcc());<a name="line.1214"></a>
<span class="sourceLineNo">1215</span>                c.setCell(esn.getPdbx_cell());<a name="line.1215"></a>
<span class="sourceLineNo">1216</span>                c.setOrganismCommon(esn.getCommon_name());<a name="line.1216"></a>
<span class="sourceLineNo">1217</span>                c.setOrganismScientific(esn.getPdbx_organism_scientific());<a name="line.1217"></a>
<span class="sourceLineNo">1218</span>                c.setOrganismTaxId(esn.getPdbx_ncbi_taxonomy_id());<a name="line.1218"></a>
<span class="sourceLineNo">1219</span><a name="line.1219"></a>
<span class="sourceLineNo">1220</span>        }<a name="line.1220"></a>
<span class="sourceLineNo">1221</span>        /**<a name="line.1221"></a>
<span class="sourceLineNo">1222</span>         * Add the information from ESS to Entity info.<a name="line.1222"></a>
<span class="sourceLineNo">1223</span>         * @param ess<a name="line.1223"></a>
<span class="sourceLineNo">1224</span>         * @param eId<a name="line.1224"></a>
<span class="sourceLineNo">1225</span>         * @param c<a name="line.1225"></a>
<span class="sourceLineNo">1226</span>         */<a name="line.1226"></a>
<span class="sourceLineNo">1227</span>        private void addInfoFromESS(EntitySrcSyn ess, int eId, EntityInfo c) {<a name="line.1227"></a>
<span class="sourceLineNo">1228</span>                c.setOrganismCommon(ess.getOrganism_common_name());<a name="line.1228"></a>
<span class="sourceLineNo">1229</span>                c.setOrganismScientific(ess.getOrganism_scientific());<a name="line.1229"></a>
<span class="sourceLineNo">1230</span>                c.setOrganismTaxId(ess.getNcbi_taxonomy_id());<a name="line.1230"></a>
<span class="sourceLineNo">1231</span><a name="line.1231"></a>
<span class="sourceLineNo">1232</span>        }<a name="line.1232"></a>
<span class="sourceLineNo">1233</span><a name="line.1233"></a>
<span class="sourceLineNo">1234</span>        private void initMaps() {<a name="line.1234"></a>
<span class="sourceLineNo">1235</span><a name="line.1235"></a>
<span class="sourceLineNo">1236</span><a name="line.1236"></a>
<span class="sourceLineNo">1237</span>                if (structAsyms == null || structAsyms.isEmpty()) {<a name="line.1237"></a>
<span class="sourceLineNo">1238</span>                        logger.info("No _struct_asym category found in file. No asym id to entity_id mapping will be available");<a name="line.1238"></a>
<span class="sourceLineNo">1239</span>                        return;<a name="line.1239"></a>
<span class="sourceLineNo">1240</span>                }<a name="line.1240"></a>
<span class="sourceLineNo">1241</span><a name="line.1241"></a>
<span class="sourceLineNo">1242</span>                Map&lt;String, List&lt;String&gt;&gt; entityId2asymId = new HashMap&lt;&gt;();<a name="line.1242"></a>
<span class="sourceLineNo">1243</span><a name="line.1243"></a>
<span class="sourceLineNo">1244</span>                for (StructAsym asym : structAsyms) {<a name="line.1244"></a>
<span class="sourceLineNo">1245</span><a name="line.1245"></a>
<span class="sourceLineNo">1246</span>                        logger.debug("Entity {} matches asym_id: {}", asym.getEntity_id(), asym.getId() );<a name="line.1246"></a>
<span class="sourceLineNo">1247</span><a name="line.1247"></a>
<span class="sourceLineNo">1248</span>                        asymId2entityId.put(asym.getId(), asym.getEntity_id());<a name="line.1248"></a>
<span class="sourceLineNo">1249</span><a name="line.1249"></a>
<span class="sourceLineNo">1250</span>                        if (entityId2asymId.containsKey(asym.getEntity_id())) {<a name="line.1250"></a>
<span class="sourceLineNo">1251</span>                                List&lt;String&gt; asymIds = entityId2asymId.get(asym.getEntity_id());<a name="line.1251"></a>
<span class="sourceLineNo">1252</span>                                asymIds.add(asym.getId());<a name="line.1252"></a>
<span class="sourceLineNo">1253</span>                        } else {<a name="line.1253"></a>
<span class="sourceLineNo">1254</span>                                List&lt;String&gt; asymIds = new ArrayList&lt;&gt;();<a name="line.1254"></a>
<span class="sourceLineNo">1255</span>                                asymIds.add(asym.getId());<a name="line.1255"></a>
<span class="sourceLineNo">1256</span>                                entityId2asymId.put(asym.getEntity_id(), asymIds);<a name="line.1256"></a>
<span class="sourceLineNo">1257</span>                        }<a name="line.1257"></a>
<span class="sourceLineNo">1258</span>                }<a name="line.1258"></a>
<span class="sourceLineNo">1259</span><a name="line.1259"></a>
<span class="sourceLineNo">1260</span>                if (entityPolys==null || entityPolys.isEmpty()) {<a name="line.1260"></a>
<span class="sourceLineNo">1261</span>                        logger.info("No _entity_poly category found in file. No asym id to author id mapping will be available for header only parsing");<a name="line.1261"></a>
<span class="sourceLineNo">1262</span>                        return;<a name="line.1262"></a>
<span class="sourceLineNo">1263</span>                }<a name="line.1263"></a>
<span class="sourceLineNo">1264</span><a name="line.1264"></a>
<span class="sourceLineNo">1265</span>                for (EntityPoly ep:entityPolys) {<a name="line.1265"></a>
<span class="sourceLineNo">1266</span>                        if (ep.getPdbx_strand_id()==null) {<a name="line.1266"></a>
<span class="sourceLineNo">1267</span>                                logger.info("_entity_poly.pdbx_strand_id is null for entity {}. Won't be able to map asym ids to author ids for this entity.", ep.getEntity_id());<a name="line.1267"></a>
<span class="sourceLineNo">1268</span>                                continue;<a name="line.1268"></a>
<span class="sourceLineNo">1269</span>                        }<a name="line.1269"></a>
<span class="sourceLineNo">1270</span>                        String[] chainNames = ep.getPdbx_strand_id().split(",");<a name="line.1270"></a>
<span class="sourceLineNo">1271</span>                        List&lt;String&gt; asymIds = entityId2asymId.get(ep.getEntity_id());<a name="line.1271"></a>
<span class="sourceLineNo">1272</span>                        if (chainNames.length!=asymIds.size()) {<a name="line.1272"></a>
<span class="sourceLineNo">1273</span>                                logger.warn("The list of asym ids (from _struct_asym) and the list of author ids (from _entity_poly) for entity {} have different lengths! Can't provide a mapping from asym ids to author chain ids", ep.getEntity_id());<a name="line.1273"></a>
<span class="sourceLineNo">1274</span>                                continue;<a name="line.1274"></a>
<span class="sourceLineNo">1275</span>                        }<a name="line.1275"></a>
<span class="sourceLineNo">1276</span>                        for (int i=0; i&lt;chainNames.length; i++) {<a name="line.1276"></a>
<span class="sourceLineNo">1277</span>                                asymId2authorId.put(asymIds.get(i), chainNames[i]);<a name="line.1277"></a>
<span class="sourceLineNo">1278</span>                        }<a name="line.1278"></a>
<span class="sourceLineNo">1279</span>                }<a name="line.1279"></a>
<span class="sourceLineNo">1280</span>        }<a name="line.1280"></a>
<span class="sourceLineNo">1281</span>        <a name="line.1281"></a>
<span class="sourceLineNo">1282</span>        private void setStructNcsOps() {<a name="line.1282"></a>
<span class="sourceLineNo">1283</span>                <a name="line.1283"></a>
<span class="sourceLineNo">1284</span>                ArrayList&lt;Matrix4d&gt; ncsOperators = new ArrayList&lt;Matrix4d&gt;();<a name="line.1284"></a>
<span class="sourceLineNo">1285</span>                <a name="line.1285"></a>
<span class="sourceLineNo">1286</span>                for (StructNcsOper sNcsOper:structNcsOper) {<a name="line.1286"></a>
<span class="sourceLineNo">1287</span>                        <a name="line.1287"></a>
<span class="sourceLineNo">1288</span>                        if (!sNcsOper.getCode().equals("generate")) continue;<a name="line.1288"></a>
<span class="sourceLineNo">1289</span>                        <a name="line.1289"></a>
<span class="sourceLineNo">1290</span>                        try {<a name="line.1290"></a>
<span class="sourceLineNo">1291</span>                                Matrix4d op = new Matrix4d();<a name="line.1291"></a>
<span class="sourceLineNo">1292</span>                                op.setElement(3, 0, 0.0);<a name="line.1292"></a>
<span class="sourceLineNo">1293</span>                                op.setElement(3, 1, 0.0);<a name="line.1293"></a>
<span class="sourceLineNo">1294</span>                                op.setElement(3, 2, 0.0);<a name="line.1294"></a>
<span class="sourceLineNo">1295</span>                                op.setElement(3, 3, 1.0);<a name="line.1295"></a>
<span class="sourceLineNo">1296</span><a name="line.1296"></a>
<span class="sourceLineNo">1297</span><a name="line.1297"></a>
<span class="sourceLineNo">1298</span>                                op.setElement(0, 0, Double.parseDouble(sNcsOper.getMatrix11()));<a name="line.1298"></a>
<span class="sourceLineNo">1299</span>                                op.setElement(0, 1, Double.parseDouble(sNcsOper.getMatrix12()));<a name="line.1299"></a>
<span class="sourceLineNo">1300</span>                                op.setElement(0, 2, Double.parseDouble(sNcsOper.getMatrix13()));<a name="line.1300"></a>
<span class="sourceLineNo">1301</span><a name="line.1301"></a>
<span class="sourceLineNo">1302</span>                                op.setElement(1, 0, Double.parseDouble(sNcsOper.getMatrix21()));<a name="line.1302"></a>
<span class="sourceLineNo">1303</span>                                op.setElement(1, 1, Double.parseDouble(sNcsOper.getMatrix22()));<a name="line.1303"></a>
<span class="sourceLineNo">1304</span>                                op.setElement(1, 2, Double.parseDouble(sNcsOper.getMatrix23()));<a name="line.1304"></a>
<span class="sourceLineNo">1305</span><a name="line.1305"></a>
<span class="sourceLineNo">1306</span>                                op.setElement(2, 0, Double.parseDouble(sNcsOper.getMatrix31()));<a name="line.1306"></a>
<span class="sourceLineNo">1307</span>                                op.setElement(2, 1, Double.parseDouble(sNcsOper.getMatrix32()));<a name="line.1307"></a>
<span class="sourceLineNo">1308</span>                                op.setElement(2, 2, Double.parseDouble(sNcsOper.getMatrix33()));<a name="line.1308"></a>
<span class="sourceLineNo">1309</span><a name="line.1309"></a>
<span class="sourceLineNo">1310</span>                                op.setElement(0, 3, Double.parseDouble(sNcsOper.getVector1()));<a name="line.1310"></a>
<span class="sourceLineNo">1311</span>                                op.setElement(1, 3, Double.parseDouble(sNcsOper.getVector2()));<a name="line.1311"></a>
<span class="sourceLineNo">1312</span>                                op.setElement(2, 3, Double.parseDouble(sNcsOper.getVector3()));<a name="line.1312"></a>
<span class="sourceLineNo">1313</span><a name="line.1313"></a>
<span class="sourceLineNo">1314</span>                                ncsOperators.add(op);<a name="line.1314"></a>
<span class="sourceLineNo">1315</span>                                <a name="line.1315"></a>
<span class="sourceLineNo">1316</span>                        } catch (NumberFormatException e) {<a name="line.1316"></a>
<span class="sourceLineNo">1317</span>                                logger.warn("Error parsing doubles in NCS operator list, skipping operator {}", structNcsOper.indexOf(sNcsOper)+1); <a name="line.1317"></a>
<span class="sourceLineNo">1318</span>                        }<a name="line.1318"></a>
<span class="sourceLineNo">1319</span><a name="line.1319"></a>
<span class="sourceLineNo">1320</span>                }<a name="line.1320"></a>
<span class="sourceLineNo">1321</span>                <a name="line.1321"></a>
<span class="sourceLineNo">1322</span>                // we only set it if not empty, otherwise remains null<a name="line.1322"></a>
<span class="sourceLineNo">1323</span>                if (ncsOperators.size()&gt;0) {<a name="line.1323"></a>
<span class="sourceLineNo">1324</span>                        structure.getCrystallographicInfo().setNcsOperators(<a name="line.1324"></a>
<span class="sourceLineNo">1325</span>                                        ncsOperators.toArray(new Matrix4d[ncsOperators.size()]));<a name="line.1325"></a>
<span class="sourceLineNo">1326</span>                }<a name="line.1326"></a>
<span class="sourceLineNo">1327</span>        }<a name="line.1327"></a>
<span class="sourceLineNo">1328</span>        <a name="line.1328"></a>
<span class="sourceLineNo">1329</span>        private void setCrystallographicInfoMetadata() {<a name="line.1329"></a>
<span class="sourceLineNo">1330</span>                if (parsedScaleMatrix!=null) {<a name="line.1330"></a>
<span class="sourceLineNo">1331</span>                        <a name="line.1331"></a>
<span class="sourceLineNo">1332</span>                        PDBCrystallographicInfo crystalInfo = structure.getCrystallographicInfo();<a name="line.1332"></a>
<span class="sourceLineNo">1333</span>                        <a name="line.1333"></a>
<span class="sourceLineNo">1334</span>                        boolean nonStd = false;<a name="line.1334"></a>
<span class="sourceLineNo">1335</span>                        if (crystalInfo.getCrystalCell()!=null &amp;&amp; !crystalInfo.getCrystalCell().checkScaleMatrix(parsedScaleMatrix)) {<a name="line.1335"></a>
<span class="sourceLineNo">1336</span>                                nonStd = true;<a name="line.1336"></a>
<span class="sourceLineNo">1337</span>                        }<a name="line.1337"></a>
<span class="sourceLineNo">1338</span>                        <a name="line.1338"></a>
<span class="sourceLineNo">1339</span>                        crystalInfo.setNonStandardCoordFrameConvention(nonStd); <a name="line.1339"></a>
<span class="sourceLineNo">1340</span>                }<a name="line.1340"></a>
<span class="sourceLineNo">1341</span>        }<a name="line.1341"></a>
<span class="sourceLineNo">1342</span><a name="line.1342"></a>
<span class="sourceLineNo">1343</span><a name="line.1343"></a>
<span class="sourceLineNo">1344</span>        /** This method will return the parsed protein structure, once the parsing has been finished<a name="line.1344"></a>
<span class="sourceLineNo">1345</span>         *<a name="line.1345"></a>
<span class="sourceLineNo">1346</span>         * @return a BioJava protein structure object<a name="line.1346"></a>
<span class="sourceLineNo">1347</span>         */<a name="line.1347"></a>
<span class="sourceLineNo">1348</span>        public Structure getStructure() {<a name="line.1348"></a>
<span class="sourceLineNo">1349</span><a name="line.1349"></a>
<span class="sourceLineNo">1350</span>                return structure;<a name="line.1350"></a>
<span class="sourceLineNo">1351</span>        }<a name="line.1351"></a>
<span class="sourceLineNo">1352</span><a name="line.1352"></a>
<span class="sourceLineNo">1353</span>        @Override<a name="line.1353"></a>
<span class="sourceLineNo">1354</span>        public void newDatabasePDBrevRecord(DatabasePdbrevRecord record) {<a name="line.1354"></a>
<span class="sourceLineNo">1355</span><a name="line.1355"></a>
<span class="sourceLineNo">1356</span>                PDBHeader header = structure.getPDBHeader();<a name="line.1356"></a>
<span class="sourceLineNo">1357</span><a name="line.1357"></a>
<span class="sourceLineNo">1358</span>                if ( header == null) {<a name="line.1358"></a>
<span class="sourceLineNo">1359</span>                        header = new PDBHeader();<a name="line.1359"></a>
<span class="sourceLineNo">1360</span>                        structure.setPDBHeader(header);<a name="line.1360"></a>
<span class="sourceLineNo">1361</span>                }<a name="line.1361"></a>
<span class="sourceLineNo">1362</span><a name="line.1362"></a>
<span class="sourceLineNo">1363</span>                List&lt;DatabasePdbrevRecord&gt; revRecords = header.getRevisionRecords();<a name="line.1363"></a>
<span class="sourceLineNo">1364</span>                if ( revRecords == null) {<a name="line.1364"></a>
<span class="sourceLineNo">1365</span>                        revRecords = new ArrayList&lt;DatabasePdbrevRecord&gt;();<a name="line.1365"></a>
<span class="sourceLineNo">1366</span>                        header.setRevisionRecords(revRecords);<a name="line.1366"></a>
<span class="sourceLineNo">1367</span>                }<a name="line.1367"></a>
<span class="sourceLineNo">1368</span>                revRecords.add(record);<a name="line.1368"></a>
<span class="sourceLineNo">1369</span><a name="line.1369"></a>
<span class="sourceLineNo">1370</span><a name="line.1370"></a>
<span class="sourceLineNo">1371</span>        }<a name="line.1371"></a>
<span class="sourceLineNo">1372</span><a name="line.1372"></a>
<span class="sourceLineNo">1373</span><a name="line.1373"></a>
<span class="sourceLineNo">1374</span>        @Override<a name="line.1374"></a>
<span class="sourceLineNo">1375</span>        public void newDatabasePDBrev(DatabasePDBrev dbrev) {<a name="line.1375"></a>
<span class="sourceLineNo">1376</span>                <a name="line.1376"></a>
<span class="sourceLineNo">1377</span>                logger.debug("got a database revision:" + dbrev);<a name="line.1377"></a>
<span class="sourceLineNo">1378</span>                <a name="line.1378"></a>
<span class="sourceLineNo">1379</span>                SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd",Locale.US);<a name="line.1379"></a>
<span class="sourceLineNo">1380</span>                PDBHeader header = structure.getPDBHeader();<a name="line.1380"></a>
<span class="sourceLineNo">1381</span><a name="line.1381"></a>
<span class="sourceLineNo">1382</span>                if ( header == null) {<a name="line.1382"></a>
<span class="sourceLineNo">1383</span>                        header = new PDBHeader();<a name="line.1383"></a>
<span class="sourceLineNo">1384</span>                }<a name="line.1384"></a>
<span class="sourceLineNo">1385</span><a name="line.1385"></a>
<span class="sourceLineNo">1386</span>                if (dbrev.getNum().equals("1")){<a name="line.1386"></a>
<span class="sourceLineNo">1387</span><a name="line.1387"></a>
<span class="sourceLineNo">1388</span>                        try {<a name="line.1388"></a>
<span class="sourceLineNo">1389</span>                                Date dep = dateFormat.parse(dbrev.getDate_original());<a name="line.1389"></a>
<span class="sourceLineNo">1390</span>                                header.setDepDate(dep);<a name="line.1390"></a>
<span class="sourceLineNo">1391</span><a name="line.1391"></a>
<span class="sourceLineNo">1392</span>                        } catch (ParseException e){<a name="line.1392"></a>
<span class="sourceLineNo">1393</span>                                logger.warn("Could not parse date string '{}', deposition date will be unavailable", dbrev.getDate_original());<a name="line.1393"></a>
<span class="sourceLineNo">1394</span>                        }<a name="line.1394"></a>
<span class="sourceLineNo">1395</span><a name="line.1395"></a>
<span class="sourceLineNo">1396</span>                        try {<a name="line.1396"></a>
<span class="sourceLineNo">1397</span>                                Date rel = dateFormat.parse(dbrev.getDate());<a name="line.1397"></a>
<span class="sourceLineNo">1398</span>                                header.setRelDate(rel);<a name="line.1398"></a>
<span class="sourceLineNo">1399</span><a name="line.1399"></a>
<span class="sourceLineNo">1400</span>                        } catch (ParseException e){<a name="line.1400"></a>
<span class="sourceLineNo">1401</span>                                logger.warn("Could not parse date string '{}', modification date will be unavailable", dbrev.getDate());<a name="line.1401"></a>
<span class="sourceLineNo">1402</span>                        }<a name="line.1402"></a>
<span class="sourceLineNo">1403</span><a name="line.1403"></a>
<span class="sourceLineNo">1404</span><a name="line.1404"></a>
<span class="sourceLineNo">1405</span>                } else {<a name="line.1405"></a>
<span class="sourceLineNo">1406</span>                        try {<a name="line.1406"></a>
<span class="sourceLineNo">1407</span><a name="line.1407"></a>
<span class="sourceLineNo">1408</span>                                Date mod = dateFormat.parse(dbrev.getDate());<a name="line.1408"></a>
<span class="sourceLineNo">1409</span>                                header.setModDate(mod);<a name="line.1409"></a>
<span class="sourceLineNo">1410</span><a name="line.1410"></a>
<span class="sourceLineNo">1411</span>                        } catch (ParseException e){<a name="line.1411"></a>
<span class="sourceLineNo">1412</span>                                logger.warn("Could not parse date string '{}', modification date will be unavailable", dbrev.getDate());<a name="line.1412"></a>
<span class="sourceLineNo">1413</span>                        }<a name="line.1413"></a>
<span class="sourceLineNo">1414</span>                }<a name="line.1414"></a>
<span class="sourceLineNo">1415</span><a name="line.1415"></a>
<span class="sourceLineNo">1416</span>                structure.setPDBHeader(header);<a name="line.1416"></a>
<span class="sourceLineNo">1417</span>        }<a name="line.1417"></a>
<span class="sourceLineNo">1418</span>        <a name="line.1418"></a>
<span class="sourceLineNo">1419</span>        @Override<a name="line.1419"></a>
<span class="sourceLineNo">1420</span>        public void newPdbxAuditRevisionHistory(PdbxAuditRevisionHistory history) {<a name="line.1420"></a>
<span class="sourceLineNo">1421</span>                <a name="line.1421"></a>
<span class="sourceLineNo">1422</span>                SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd",Locale.US);<a name="line.1422"></a>
<span class="sourceLineNo">1423</span>                PDBHeader header = structure.getPDBHeader();<a name="line.1423"></a>
<span class="sourceLineNo">1424</span><a name="line.1424"></a>
<span class="sourceLineNo">1425</span>                if ( header == null) {<a name="line.1425"></a>
<span class="sourceLineNo">1426</span>                        header = new PDBHeader();<a name="line.1426"></a>
<span class="sourceLineNo">1427</span>                }<a name="line.1427"></a>
<span class="sourceLineNo">1428</span><a name="line.1428"></a>
<span class="sourceLineNo">1429</span>        // first entry in revision history is the release date<a name="line.1429"></a>
<span class="sourceLineNo">1430</span>                if (history.getOrdinal().equals("1")){<a name="line.1430"></a>
<span class="sourceLineNo">1431</span>                        try {<a name="line.1431"></a>
<span class="sourceLineNo">1432</span>                                Date releaseDate = dateFormat.parse(history.getRevision_date());<a name="line.1432"></a>
<span class="sourceLineNo">1433</span>                                header.setRelDate(releaseDate);<a name="line.1433"></a>
<span class="sourceLineNo">1434</span>                                <a name="line.1434"></a>
<span class="sourceLineNo">1435</span>                        } catch (ParseException e){<a name="line.1435"></a>
<span class="sourceLineNo">1436</span>                                logger.warn("Could not parse date string '{}', release date will be unavailable", history.getRevision_date());<a name="line.1436"></a>
<span class="sourceLineNo">1437</span>                        }<a name="line.1437"></a>
<span class="sourceLineNo">1438</span>                } else {<a name="line.1438"></a>
<span class="sourceLineNo">1439</span>                        // all other dates are revision dates;<a name="line.1439"></a>
<span class="sourceLineNo">1440</span>                        // since this method may be called multiple times,<a name="line.1440"></a>
<span class="sourceLineNo">1441</span>                        // the last revision date will "stick"<a name="line.1441"></a>
<span class="sourceLineNo">1442</span>                        try {<a name="line.1442"></a>
<span class="sourceLineNo">1443</span>                                Date revisionDate = dateFormat.parse(history.getRevision_date());<a name="line.1443"></a>
<span class="sourceLineNo">1444</span>                                header.setModDate(revisionDate);<a name="line.1444"></a>
<span class="sourceLineNo">1445</span>                        } catch (ParseException e){<a name="line.1445"></a>
<span class="sourceLineNo">1446</span>                                logger.warn("Could not parse date string '{}', revision date will be unavailable", history.getRevision_date());<a name="line.1446"></a>
<span class="sourceLineNo">1447</span>                        }<a name="line.1447"></a>
<span class="sourceLineNo">1448</span>                }<a name="line.1448"></a>
<span class="sourceLineNo">1449</span><a name="line.1449"></a>
<span class="sourceLineNo">1450</span>                structure.setPDBHeader(header);<a name="line.1450"></a>
<span class="sourceLineNo">1451</span>        }<a name="line.1451"></a>
<span class="sourceLineNo">1452</span>        <a name="line.1452"></a>
<span class="sourceLineNo">1453</span>        @Override<a name="line.1453"></a>
<span class="sourceLineNo">1454</span>        public void newPdbxDatabaseStatus(PdbxDatabaseStatus status) {<a name="line.1454"></a>
<span class="sourceLineNo">1455</span><a name="line.1455"></a>
<span class="sourceLineNo">1456</span>                // the deposition date field is only available in mmCIF 5.0<a name="line.1456"></a>
<span class="sourceLineNo">1457</span><a name="line.1457"></a>
<span class="sourceLineNo">1458</span>                if (status.getRecvd_initial_deposition_date() == null) {<a name="line.1458"></a>
<span class="sourceLineNo">1459</span>                        // skip this method for older mmCIF versions<a name="line.1459"></a>
<span class="sourceLineNo">1460</span>                        return;<a name="line.1460"></a>
<span class="sourceLineNo">1461</span>                }<a name="line.1461"></a>
<span class="sourceLineNo">1462</span>                <a name="line.1462"></a>
<span class="sourceLineNo">1463</span>                SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd",Locale.US);<a name="line.1463"></a>
<span class="sourceLineNo">1464</span>                PDBHeader header = structure.getPDBHeader();<a name="line.1464"></a>
<span class="sourceLineNo">1465</span><a name="line.1465"></a>
<span class="sourceLineNo">1466</span>                if (header == null) {<a name="line.1466"></a>
<span class="sourceLineNo">1467</span>                        header = new PDBHeader();<a name="line.1467"></a>
<span class="sourceLineNo">1468</span>                }<a name="line.1468"></a>
<span class="sourceLineNo">1469</span><a name="line.1469"></a>
<span class="sourceLineNo">1470</span>                try {<a name="line.1470"></a>
<span class="sourceLineNo">1471</span>                        Date depositionDate = dateFormat.parse(status.getRecvd_initial_deposition_date());<a name="line.1471"></a>
<span class="sourceLineNo">1472</span>                        header.setDepDate(depositionDate);<a name="line.1472"></a>
<span class="sourceLineNo">1473</span>                } catch (ParseException e){<a name="line.1473"></a>
<span class="sourceLineNo">1474</span>                        logger.warn("Could not parse date string '{}', deposition date will be unavailable", status.getRecvd_initial_deposition_date());<a name="line.1474"></a>
<span class="sourceLineNo">1475</span>                }<a name="line.1475"></a>
<span class="sourceLineNo">1476</span><a name="line.1476"></a>
<span class="sourceLineNo">1477</span>                structure.setPDBHeader(header);<a name="line.1477"></a>
<span class="sourceLineNo">1478</span>        }<a name="line.1478"></a>
<span class="sourceLineNo">1479</span><a name="line.1479"></a>
<span class="sourceLineNo">1480</span>        @Override<a name="line.1480"></a>
<span class="sourceLineNo">1481</span>        public void newDatabasePDBremark(DatabasePDBremark remark) {<a name="line.1481"></a>
<span class="sourceLineNo">1482</span>                //System.out.println(remark);<a name="line.1482"></a>
<span class="sourceLineNo">1483</span>                String id = remark.getId();<a name="line.1483"></a>
<span class="sourceLineNo">1484</span>                if (id.equals("2")){<a name="line.1484"></a>
<span class="sourceLineNo">1485</span><a name="line.1485"></a>
<span class="sourceLineNo">1486</span>                        //this remark field contains the resolution information:<a name="line.1486"></a>
<span class="sourceLineNo">1487</span>                        String line = remark.getText();<a name="line.1487"></a>
<span class="sourceLineNo">1488</span><a name="line.1488"></a>
<span class="sourceLineNo">1489</span>                        int i = line.indexOf("ANGSTROM");<a name="line.1489"></a>
<span class="sourceLineNo">1490</span>                        if ( i &gt; 5) {<a name="line.1490"></a>
<span class="sourceLineNo">1491</span>                                // line contains ANGSTROM info...<a name="line.1491"></a>
<span class="sourceLineNo">1492</span>                                String resolution = line.substring(i-5,i).trim();<a name="line.1492"></a>
<span class="sourceLineNo">1493</span>                                // convert string to float<a name="line.1493"></a>
<span class="sourceLineNo">1494</span>                                float res = 99 ;<a name="line.1494"></a>
<span class="sourceLineNo">1495</span>                                try {<a name="line.1495"></a>
<span class="sourceLineNo">1496</span>                                        res = Float.parseFloat(resolution);<a name="line.1496"></a>
<span class="sourceLineNo">1497</span><a name="line.1497"></a>
<span class="sourceLineNo">1498</span>                                } catch (NumberFormatException e) {<a name="line.1498"></a>
<span class="sourceLineNo">1499</span>                                        logger.info("could not parse resolution from line and ignoring it " + line);<a name="line.1499"></a>
<span class="sourceLineNo">1500</span>                                        return ;<a name="line.1500"></a>
<span class="sourceLineNo">1501</span><a name="line.1501"></a>
<span class="sourceLineNo">1502</span><a name="line.1502"></a>
<span class="sourceLineNo">1503</span>                                }<a name="line.1503"></a>
<span class="sourceLineNo">1504</span>                                // support for old style header<a name="line.1504"></a>
<span class="sourceLineNo">1505</span><a name="line.1505"></a>
<span class="sourceLineNo">1506</span>                                PDBHeader pdbHeader = structure.getPDBHeader();<a name="line.1506"></a>
<span class="sourceLineNo">1507</span>                                pdbHeader.setResolution(res);<a name="line.1507"></a>
<span class="sourceLineNo">1508</span><a name="line.1508"></a>
<span class="sourceLineNo">1509</span>                        }<a name="line.1509"></a>
<span class="sourceLineNo">1510</span><a name="line.1510"></a>
<span class="sourceLineNo">1511</span>                }<a name="line.1511"></a>
<span class="sourceLineNo">1512</span>        }<a name="line.1512"></a>
<span class="sourceLineNo">1513</span><a name="line.1513"></a>
<span class="sourceLineNo">1514</span>        @Override<a name="line.1514"></a>
<span class="sourceLineNo">1515</span>        public void newRefine(Refine r){<a name="line.1515"></a>
<span class="sourceLineNo">1516</span><a name="line.1516"></a>
<span class="sourceLineNo">1517</span>                PDBHeader pdbHeader = structure.getPDBHeader();<a name="line.1517"></a>
<span class="sourceLineNo">1518</span>                // RESOLUTION<a name="line.1518"></a>
<span class="sourceLineNo">1519</span>                // in very rare cases (for instance hybrid methods x-ray + neutron diffraction, e.g. 3ins, 4n9m)<a name="line.1519"></a>
<span class="sourceLineNo">1520</span>                // there are 2 resolution values, one for each method<a name="line.1520"></a>
<span class="sourceLineNo">1521</span>                // we take the last one found so that behaviour is like in PDB file parsing<a name="line.1521"></a>
<span class="sourceLineNo">1522</span>                if (pdbHeader.getResolution()!=PDBHeader.DEFAULT_RESOLUTION) {<a name="line.1522"></a>
<span class="sourceLineNo">1523</span>                        logger.warn("More than 1 resolution value present, will use last one {} and discard previous {} "<a name="line.1523"></a>
<span class="sourceLineNo">1524</span>                                        ,r.getLs_d_res_high(), String.format("%4.2f",pdbHeader.getResolution()));<a name="line.1524"></a>
<span class="sourceLineNo">1525</span>                }<a name="line.1525"></a>
<span class="sourceLineNo">1526</span>                try {<a name="line.1526"></a>
<span class="sourceLineNo">1527</span>                        pdbHeader.setResolution(Float.parseFloat(r.getLs_d_res_high()));<a name="line.1527"></a>
<span class="sourceLineNo">1528</span>                } catch (NumberFormatException e){<a name="line.1528"></a>
<span class="sourceLineNo">1529</span>                        logger.info("Could not parse resolution from " + r.getLs_d_res_high() + " " + e.getMessage());<a name="line.1529"></a>
<span class="sourceLineNo">1530</span>                }<a name="line.1530"></a>
<span class="sourceLineNo">1531</span><a name="line.1531"></a>
<span class="sourceLineNo">1532</span><a name="line.1532"></a>
<span class="sourceLineNo">1533</span>                // RFREE<a name="line.1533"></a>
<span class="sourceLineNo">1534</span>                if (pdbHeader.getRfree()!=PDBHeader.DEFAULT_RFREE) {<a name="line.1534"></a>
<span class="sourceLineNo">1535</span>                        logger.warn("More than 1 Rfree value present, will use last one {} and discard previous {} ",<a name="line.1535"></a>
<span class="sourceLineNo">1536</span>                                        r.getLs_R_factor_R_free(), String.format("%4.2f",pdbHeader.getRfree()));<a name="line.1536"></a>
<span class="sourceLineNo">1537</span>                }<a name="line.1537"></a>
<span class="sourceLineNo">1538</span>                if (r.getLs_R_factor_R_free()==null) {<a name="line.1538"></a>
<span class="sourceLineNo">1539</span>                        // some entries like 2ifo haven't got this field at all<a name="line.1539"></a>
<span class="sourceLineNo">1540</span>                        logger.info("_refine.ls_R_factor_R_free not present, not parsing Rfree value");<a name="line.1540"></a>
<span class="sourceLineNo">1541</span>                } else {<a name="line.1541"></a>
<span class="sourceLineNo">1542</span>                        try {<a name="line.1542"></a>
<span class="sourceLineNo">1543</span>                                pdbHeader.setRfree(Float.parseFloat(r.getLs_R_factor_R_free()));<a name="line.1543"></a>
<span class="sourceLineNo">1544</span>                        } catch (NumberFormatException e){<a name="line.1544"></a>
<span class="sourceLineNo">1545</span>                                // no rfree present ('?') is very usual, that's why we set it to debug<a name="line.1545"></a>
<span class="sourceLineNo">1546</span>                                logger.debug("Could not parse Rfree from string '{}'", r.getLs_R_factor_R_free());<a name="line.1546"></a>
<span class="sourceLineNo">1547</span>                        }<a name="line.1547"></a>
<span class="sourceLineNo">1548</span>                }<a name="line.1548"></a>
<span class="sourceLineNo">1549</span><a name="line.1549"></a>
<span class="sourceLineNo">1550</span>                // RWORK<a name="line.1550"></a>
<span class="sourceLineNo">1551</span>                if(pdbHeader.getRwork()!=PDBHeader.DEFAULT_RFREE) {<a name="line.1551"></a>
<span class="sourceLineNo">1552</span>                        logger.warn("More than 1 R work value present, will use last one {} and discard previous {} ",<a name="line.1552"></a>
<span class="sourceLineNo">1553</span>                                        r.getLs_R_factor_R_work(), String.format("%4.2f",pdbHeader.getRwork()));<a name="line.1553"></a>
<span class="sourceLineNo">1554</span>                }<a name="line.1554"></a>
<span class="sourceLineNo">1555</span>                if(r.getLs_R_factor_R_work()==null){<a name="line.1555"></a>
<span class="sourceLineNo">1556</span>                        logger.info("_refine.ls_R_factor_R_work not present, not parsing R-work value");<a name="line.1556"></a>
<span class="sourceLineNo">1557</span>                }<a name="line.1557"></a>
<span class="sourceLineNo">1558</span>                else{<a name="line.1558"></a>
<span class="sourceLineNo">1559</span>                        try{<a name="line.1559"></a>
<span class="sourceLineNo">1560</span>                                pdbHeader.setRwork(Float.parseFloat(r.getLs_R_factor_R_work()));<a name="line.1560"></a>
<span class="sourceLineNo">1561</span>                        }<a name="line.1561"></a>
<span class="sourceLineNo">1562</span>                        catch (NumberFormatException e){<a name="line.1562"></a>
<span class="sourceLineNo">1563</span>                                logger.debug("Could not parse R-work from string '{}'", r.getLs_R_factor_R_work());<a name="line.1563"></a>
<span class="sourceLineNo">1564</span>                        }<a name="line.1564"></a>
<span class="sourceLineNo">1565</span><a name="line.1565"></a>
<span class="sourceLineNo">1566</span>                }<a name="line.1566"></a>
<span class="sourceLineNo">1567</span><a name="line.1567"></a>
<span class="sourceLineNo">1568</span>        }<a name="line.1568"></a>
<span class="sourceLineNo">1569</span><a name="line.1569"></a>
<span class="sourceLineNo">1570</span><a name="line.1570"></a>
<span class="sourceLineNo">1571</span>        @Override<a name="line.1571"></a>
<span class="sourceLineNo">1572</span>        public void newAuditAuthor(AuditAuthor aa){<a name="line.1572"></a>
<span class="sourceLineNo">1573</span><a name="line.1573"></a>
<span class="sourceLineNo">1574</span>                String name =  aa.getName();<a name="line.1574"></a>
<span class="sourceLineNo">1575</span><a name="line.1575"></a>
<span class="sourceLineNo">1576</span>                StringBuffer famName = new StringBuffer();<a name="line.1576"></a>
<span class="sourceLineNo">1577</span>                StringBuffer initials = new StringBuffer();<a name="line.1577"></a>
<span class="sourceLineNo">1578</span>                boolean afterComma = false;<a name="line.1578"></a>
<span class="sourceLineNo">1579</span>                for ( char c: name.toCharArray()) {<a name="line.1579"></a>
<span class="sourceLineNo">1580</span>                        if ( c == ' ')<a name="line.1580"></a>
<span class="sourceLineNo">1581</span>                                continue;<a name="line.1581"></a>
<span class="sourceLineNo">1582</span>                        if ( c == ','){<a name="line.1582"></a>
<span class="sourceLineNo">1583</span>                                afterComma = true;<a name="line.1583"></a>
<span class="sourceLineNo">1584</span>                                continue;<a name="line.1584"></a>
<span class="sourceLineNo">1585</span>                        }<a name="line.1585"></a>
<span class="sourceLineNo">1586</span><a name="line.1586"></a>
<span class="sourceLineNo">1587</span>                        if ( afterComma)<a name="line.1587"></a>
<span class="sourceLineNo">1588</span>                                initials.append(c);<a name="line.1588"></a>
<span class="sourceLineNo">1589</span>                        else<a name="line.1589"></a>
<span class="sourceLineNo">1590</span>                                famName.append(c);<a name="line.1590"></a>
<span class="sourceLineNo">1591</span>                }<a name="line.1591"></a>
<span class="sourceLineNo">1592</span><a name="line.1592"></a>
<span class="sourceLineNo">1593</span>                StringBuffer newaa = new StringBuffer();<a name="line.1593"></a>
<span class="sourceLineNo">1594</span>                newaa.append(initials);<a name="line.1594"></a>
<span class="sourceLineNo">1595</span>                newaa.append(famName);<a name="line.1595"></a>
<span class="sourceLineNo">1596</span><a name="line.1596"></a>
<span class="sourceLineNo">1597</span>                PDBHeader header = structure.getPDBHeader();<a name="line.1597"></a>
<span class="sourceLineNo">1598</span>                String auth = header.getAuthors();<a name="line.1598"></a>
<span class="sourceLineNo">1599</span>                if (auth == null) {<a name="line.1599"></a>
<span class="sourceLineNo">1600</span>                        header.setAuthors(newaa.toString());<a name="line.1600"></a>
<span class="sourceLineNo">1601</span>                }else {<a name="line.1601"></a>
<span class="sourceLineNo">1602</span>                        auth += "," + newaa.toString();<a name="line.1602"></a>
<span class="sourceLineNo">1603</span>                        header.setAuthors(auth);<a name="line.1603"></a>
<span class="sourceLineNo">1604</span><a name="line.1604"></a>
<span class="sourceLineNo">1605</span>                }<a name="line.1605"></a>
<span class="sourceLineNo">1606</span>        }<a name="line.1606"></a>
<span class="sourceLineNo">1607</span><a name="line.1607"></a>
<span class="sourceLineNo">1608</span>        @Override<a name="line.1608"></a>
<span class="sourceLineNo">1609</span>        public void newExptl(Exptl exptl) {<a name="line.1609"></a>
<span class="sourceLineNo">1610</span><a name="line.1610"></a>
<span class="sourceLineNo">1611</span>                PDBHeader pdbHeader = structure.getPDBHeader();<a name="line.1611"></a>
<span class="sourceLineNo">1612</span>                String method = exptl.getMethod();<a name="line.1612"></a>
<span class="sourceLineNo">1613</span>                pdbHeader.setExperimentalTechnique(method);<a name="line.1613"></a>
<span class="sourceLineNo">1614</span><a name="line.1614"></a>
<span class="sourceLineNo">1615</span>        }<a name="line.1615"></a>
<span class="sourceLineNo">1616</span><a name="line.1616"></a>
<span class="sourceLineNo">1617</span>        @Override<a name="line.1617"></a>
<span class="sourceLineNo">1618</span>        public void newCell(Cell cell) {<a name="line.1618"></a>
<span class="sourceLineNo">1619</span><a name="line.1619"></a>
<span class="sourceLineNo">1620</span>                try {<a name="line.1620"></a>
<span class="sourceLineNo">1621</span>                        float a = Float.parseFloat(cell.getLength_a());<a name="line.1621"></a>
<span class="sourceLineNo">1622</span>                        float b = Float.parseFloat(cell.getLength_b());<a name="line.1622"></a>
<span class="sourceLineNo">1623</span>                        float c = Float.parseFloat(cell.getLength_c());<a name="line.1623"></a>
<span class="sourceLineNo">1624</span>                        float alpha = Float.parseFloat(cell.getAngle_alpha());<a name="line.1624"></a>
<span class="sourceLineNo">1625</span>                        float beta = Float.parseFloat(cell.getAngle_beta());<a name="line.1625"></a>
<span class="sourceLineNo">1626</span>                        float gamma = Float.parseFloat(cell.getAngle_gamma());<a name="line.1626"></a>
<span class="sourceLineNo">1627</span><a name="line.1627"></a>
<span class="sourceLineNo">1628</span>                        CrystalCell xtalCell = new CrystalCell();<a name="line.1628"></a>
<span class="sourceLineNo">1629</span>                        xtalCell.setA(a);<a name="line.1629"></a>
<span class="sourceLineNo">1630</span>                        xtalCell.setB(b);<a name="line.1630"></a>
<span class="sourceLineNo">1631</span>                        xtalCell.setC(c);<a name="line.1631"></a>
<span class="sourceLineNo">1632</span>                        xtalCell.setAlpha(alpha);<a name="line.1632"></a>
<span class="sourceLineNo">1633</span>                        xtalCell.setBeta(beta);<a name="line.1633"></a>
<span class="sourceLineNo">1634</span>                        xtalCell.setGamma(gamma);<a name="line.1634"></a>
<span class="sourceLineNo">1635</span><a name="line.1635"></a>
<span class="sourceLineNo">1636</span>                        if (!xtalCell.isCellReasonable()) {<a name="line.1636"></a>
<span class="sourceLineNo">1637</span>                                // If the entry describes a structure determined by a technique other than X-ray crystallography,<a name="line.1637"></a>
<span class="sourceLineNo">1638</span>                                // cell is (sometimes!) a = b = c = 1.0, alpha = beta = gamma = 90 degrees<a name="line.1638"></a>
<span class="sourceLineNo">1639</span>                                // if so we don't add and CrystalCell will be null<a name="line.1639"></a>
<span class="sourceLineNo">1640</span>                                logger.debug("The crystal cell read from file does not have reasonable dimensions (at least one dimension is below {}), discarding it.",<a name="line.1640"></a>
<span class="sourceLineNo">1641</span>                                                CrystalCell.MIN_VALID_CELL_SIZE);<a name="line.1641"></a>
<span class="sourceLineNo">1642</span>                                return;<a name="line.1642"></a>
<span class="sourceLineNo">1643</span>                        }<a name="line.1643"></a>
<span class="sourceLineNo">1644</span><a name="line.1644"></a>
<span class="sourceLineNo">1645</span>                        structure.getPDBHeader().getCrystallographicInfo().setCrystalCell(xtalCell);<a name="line.1645"></a>
<span class="sourceLineNo">1646</span><a name="line.1646"></a>
<span class="sourceLineNo">1647</span>                } catch (NumberFormatException e){<a name="line.1647"></a>
<span class="sourceLineNo">1648</span>                        structure.getPDBHeader().getCrystallographicInfo().setCrystalCell(null);<a name="line.1648"></a>
<span class="sourceLineNo">1649</span>                        logger.info("could not parse some cell parameters ("+e.getMessage()+"), ignoring _cell ");<a name="line.1649"></a>
<span class="sourceLineNo">1650</span>                }<a name="line.1650"></a>
<span class="sourceLineNo">1651</span>        }<a name="line.1651"></a>
<span class="sourceLineNo">1652</span><a name="line.1652"></a>
<span class="sourceLineNo">1653</span>        @Override<a name="line.1653"></a>
<span class="sourceLineNo">1654</span>        public void newSymmetry(Symmetry symmetry) {<a name="line.1654"></a>
<span class="sourceLineNo">1655</span>                String spaceGroup = symmetry.getSpace_group_name_H_M();<a name="line.1655"></a>
<span class="sourceLineNo">1656</span>                SpaceGroup sg = SymoplibParser.getSpaceGroup(spaceGroup);<a name="line.1656"></a>
<span class="sourceLineNo">1657</span>                if (sg==null) {<a name="line.1657"></a>
<span class="sourceLineNo">1658</span>                        logger.warn("Space group '"+spaceGroup+"' not recognised as a standard space group");<a name="line.1658"></a>
<span class="sourceLineNo">1659</span>                        structure.getPDBHeader().getCrystallographicInfo().setNonStandardSg(true);<a name="line.1659"></a>
<span class="sourceLineNo">1660</span>                } else {<a name="line.1660"></a>
<span class="sourceLineNo">1661</span>                        structure.getPDBHeader().getCrystallographicInfo().setSpaceGroup(sg);<a name="line.1661"></a>
<span class="sourceLineNo">1662</span>                        structure.getPDBHeader().getCrystallographicInfo().setNonStandardSg(false);<a name="line.1662"></a>
<span class="sourceLineNo">1663</span>                }<a name="line.1663"></a>
<span class="sourceLineNo">1664</span>        }<a name="line.1664"></a>
<span class="sourceLineNo">1665</span><a name="line.1665"></a>
<span class="sourceLineNo">1666</span>        @Override<a name="line.1666"></a>
<span class="sourceLineNo">1667</span>        public void newStructNcsOper(StructNcsOper sNcsOper) {<a name="line.1667"></a>
<span class="sourceLineNo">1668</span>                structNcsOper.add(sNcsOper);<a name="line.1668"></a>
<span class="sourceLineNo">1669</span>        }<a name="line.1669"></a>
<span class="sourceLineNo">1670</span>        <a name="line.1670"></a>
<span class="sourceLineNo">1671</span>        public void newAtomSites(AtomSites atomSites) {<a name="line.1671"></a>
<span class="sourceLineNo">1672</span>                <a name="line.1672"></a>
<span class="sourceLineNo">1673</span>                try {<a name="line.1673"></a>
<span class="sourceLineNo">1674</span>                        Matrix4d m = new Matrix4d(<a name="line.1674"></a>
<span class="sourceLineNo">1675</span>                                Double.parseDouble(atomSites.getFract_transf_matrix11()), Double.parseDouble(atomSites.getFract_transf_matrix12()), Double.parseDouble(atomSites.getFract_transf_matrix13()), Double.parseDouble(atomSites.getFract_transf_vector1()),<a name="line.1675"></a>
<span class="sourceLineNo">1676</span>                                Double.parseDouble(atomSites.getFract_transf_matrix21()), Double.parseDouble(atomSites.getFract_transf_matrix22()), Double.parseDouble(atomSites.getFract_transf_matrix23()), Double.parseDouble(atomSites.getFract_transf_vector2()),<a name="line.1676"></a>
<span class="sourceLineNo">1677</span>                                Double.parseDouble(atomSites.getFract_transf_matrix31()), Double.parseDouble(atomSites.getFract_transf_matrix32()), Double.parseDouble(atomSites.getFract_transf_matrix33()), Double.parseDouble(atomSites.getFract_transf_vector3()),<a name="line.1677"></a>
<span class="sourceLineNo">1678</span>                                0,0,0,1);<a name="line.1678"></a>
<span class="sourceLineNo">1679</span><a name="line.1679"></a>
<span class="sourceLineNo">1680</span>                        parsedScaleMatrix = m;<a name="line.1680"></a>
<span class="sourceLineNo">1681</span>                <a name="line.1681"></a>
<span class="sourceLineNo">1682</span>                } catch (NumberFormatException e) {<a name="line.1682"></a>
<span class="sourceLineNo">1683</span>                        logger.warn("Some values in _atom_sites.fract_transf_matrix or _atom_sites.fract_transf_vector could not be parsed as numbers. Can't check whether coordinate frame convention is correct! Error: {}", e.getMessage());<a name="line.1683"></a>
<span class="sourceLineNo">1684</span>                        structure.getPDBHeader().getCrystallographicInfo().setNonStandardCoordFrameConvention(false);<a name="line.1684"></a>
<span class="sourceLineNo">1685</span>                        <a name="line.1685"></a>
<span class="sourceLineNo">1686</span>                        // in this case parsedScaleMatrix stays null and can't be used in documentEnd()<a name="line.1686"></a>
<span class="sourceLineNo">1687</span>                }<a name="line.1687"></a>
<span class="sourceLineNo">1688</span>        }<a name="line.1688"></a>
<span class="sourceLineNo">1689</span><a name="line.1689"></a>
<span class="sourceLineNo">1690</span>        @Override<a name="line.1690"></a>
<span class="sourceLineNo">1691</span>        public void newStructRef(StructRef sref) {<a name="line.1691"></a>
<span class="sourceLineNo">1692</span>                logger.debug(sref.toString());<a name="line.1692"></a>
<span class="sourceLineNo">1693</span>                strucRefs.add(sref);<a name="line.1693"></a>
<span class="sourceLineNo">1694</span>        }<a name="line.1694"></a>
<span class="sourceLineNo">1695</span><a name="line.1695"></a>
<span class="sourceLineNo">1696</span>        private StructRef getStructRef(String ref_id){<a name="line.1696"></a>
<span class="sourceLineNo">1697</span>                for (StructRef structRef : strucRefs) {<a name="line.1697"></a>
<span class="sourceLineNo">1698</span><a name="line.1698"></a>
<span class="sourceLineNo">1699</span>                        if (structRef.getId().equals(ref_id)){<a name="line.1699"></a>
<span class="sourceLineNo">1700</span>                                return structRef;<a name="line.1700"></a>
<span class="sourceLineNo">1701</span>                        }<a name="line.1701"></a>
<span class="sourceLineNo">1702</span><a name="line.1702"></a>
<span class="sourceLineNo">1703</span>                }<a name="line.1703"></a>
<span class="sourceLineNo">1704</span>                return null;<a name="line.1704"></a>
<span class="sourceLineNo">1705</span><a name="line.1705"></a>
<span class="sourceLineNo">1706</span>        }<a name="line.1706"></a>
<span class="sourceLineNo">1707</span><a name="line.1707"></a>
<span class="sourceLineNo">1708</span>        /**<a name="line.1708"></a>
<span class="sourceLineNo">1709</span>         * create a DBRef record from the StrucRefSeq record:<a name="line.1709"></a>
<span class="sourceLineNo">1710</span>         * &lt;pre&gt;<a name="line.1710"></a>
<span class="sourceLineNo">1711</span>         * PDB record                    DBREF<a name="line.1711"></a>
<span class="sourceLineNo">1712</span>         * Field Name                    mmCIF Data Item<a name="line.1712"></a>
<span class="sourceLineNo">1713</span>         * Section                       n.a.<a name="line.1713"></a>
<span class="sourceLineNo">1714</span>         * PDB_ID_Code                   _struct_ref_seq.pdbx_PDB_id_code<a name="line.1714"></a>
<span class="sourceLineNo">1715</span>         * Strand_ID                     _struct_ref_seq.pdbx_strand_id<a name="line.1715"></a>
<span class="sourceLineNo">1716</span>         * Begin_Residue_Number          _struct_ref_seq.pdbx_auth_seq_align_beg<a name="line.1716"></a>
<span class="sourceLineNo">1717</span>         * Begin_Ins_Code                _struct_ref_seq.pdbx_seq_align_beg_ins_code<a name="line.1717"></a>
<span class="sourceLineNo">1718</span>         * End_Residue_Number            _struct_ref_seq.pdbx_auth_seq_align_end<a name="line.1718"></a>
<span class="sourceLineNo">1719</span>         * End_Ins_Code                  _struct_ref_seq.pdbx_seq_align_end_ins_code<a name="line.1719"></a>
<span class="sourceLineNo">1720</span>         * Database                      _struct_ref.db_name<a name="line.1720"></a>
<span class="sourceLineNo">1721</span>         * Database_Accession_No         _struct_ref_seq.pdbx_db_accession<a name="line.1721"></a>
<span class="sourceLineNo">1722</span>         * Database_ID_Code              _struct_ref.db_code<a name="line.1722"></a>
<span class="sourceLineNo">1723</span>         * Database_Begin_Residue_Number _struct_ref_seq.db_align_beg<a name="line.1723"></a>
<span class="sourceLineNo">1724</span>         * Databaes_Begin_Ins_Code       _struct_ref_seq.pdbx_db_align_beg_ins_code<a name="line.1724"></a>
<span class="sourceLineNo">1725</span>         * Database_End_Residue_Number   _struct_ref_seq.db_align_end<a name="line.1725"></a>
<span class="sourceLineNo">1726</span>         * Databaes_End_Ins_Code         _struct_ref_seq.pdbx_db_align_end_ins_code<a name="line.1726"></a>
<span class="sourceLineNo">1727</span>         * &lt;/pre&gt;<a name="line.1727"></a>
<span class="sourceLineNo">1728</span>         *<a name="line.1728"></a>
<span class="sourceLineNo">1729</span>         *<a name="line.1729"></a>
<span class="sourceLineNo">1730</span>         */<a name="line.1730"></a>
<span class="sourceLineNo">1731</span>        @Override<a name="line.1731"></a>
<span class="sourceLineNo">1732</span>        public void newStructRefSeq(StructRefSeq sref) {<a name="line.1732"></a>
<span class="sourceLineNo">1733</span>                DBRef r = new DBRef();<a name="line.1733"></a>
<span class="sourceLineNo">1734</span><a name="line.1734"></a>
<span class="sourceLineNo">1735</span>                r.setIdCode(sref.getPdbx_PDB_id_code());<a name="line.1735"></a>
<span class="sourceLineNo">1736</span>                r.setDbAccession(sref.getPdbx_db_accession());<a name="line.1736"></a>
<span class="sourceLineNo">1737</span>                r.setDbIdCode(sref.getPdbx_db_accession());<a name="line.1737"></a>
<span class="sourceLineNo">1738</span><a name="line.1738"></a>
<span class="sourceLineNo">1739</span>                r.setChainName(sref.getPdbx_strand_id());<a name="line.1739"></a>
<span class="sourceLineNo">1740</span>                StructRef structRef = getStructRef(sref.getRef_id());<a name="line.1740"></a>
<span class="sourceLineNo">1741</span>                if (structRef == null){<a name="line.1741"></a>
<span class="sourceLineNo">1742</span>                        logger.info("could not find StructRef " + sref.getRef_id() + " for StructRefSeq " + sref);<a name="line.1742"></a>
<span class="sourceLineNo">1743</span>                } else {<a name="line.1743"></a>
<span class="sourceLineNo">1744</span>                        r.setDatabase(structRef.getDb_name());<a name="line.1744"></a>
<span class="sourceLineNo">1745</span>                        r.setDbIdCode(structRef.getDb_code());<a name="line.1745"></a>
<span class="sourceLineNo">1746</span>                }<a name="line.1746"></a>
<span class="sourceLineNo">1747</span><a name="line.1747"></a>
<span class="sourceLineNo">1748</span>                int seqbegin;<a name="line.1748"></a>
<span class="sourceLineNo">1749</span>                int seqend;<a name="line.1749"></a>
<span class="sourceLineNo">1750</span>                try{<a name="line.1750"></a>
<span class="sourceLineNo">1751</span>                        seqbegin = Integer.parseInt(sref.getPdbx_auth_seq_align_beg());<a name="line.1751"></a>
<span class="sourceLineNo">1752</span>                        seqend   = Integer.parseInt(sref.getPdbx_auth_seq_align_end());<a name="line.1752"></a>
<span class="sourceLineNo">1753</span>                }<a name="line.1753"></a>
<span class="sourceLineNo">1754</span>                catch(NumberFormatException e){<a name="line.1754"></a>
<span class="sourceLineNo">1755</span>                        // this happens in a few entries, annotation error? e.g. 6eoj<a name="line.1755"></a>
<span class="sourceLineNo">1756</span>                        logger.warn("Couldn't parse pdbx_auth_seq_align_beg/end in _struct_ref_seq. Will not store dbref alignment info for accession {}. Error: {}", r.getDbAccession(), e.getMessage());<a name="line.1756"></a>
<span class="sourceLineNo">1757</span>                        return;<a name="line.1757"></a>
<span class="sourceLineNo">1758</span>                }<a name="line.1758"></a>
<span class="sourceLineNo">1759</span>                <a name="line.1759"></a>
<span class="sourceLineNo">1760</span>                Character begin_ins_code = ' ';<a name="line.1760"></a>
<span class="sourceLineNo">1761</span>                if (sref.getPdbx_seq_align_beg_ins_code() != null ) {<a name="line.1761"></a>
<span class="sourceLineNo">1762</span>                    begin_ins_code = new Character(sref.getPdbx_seq_align_beg_ins_code().charAt(0));<a name="line.1762"></a>
<span class="sourceLineNo">1763</span>                }<a name="line.1763"></a>
<span class="sourceLineNo">1764</span>                <a name="line.1764"></a>
<span class="sourceLineNo">1765</span>                Character end_ins_code = ' ';<a name="line.1765"></a>
<span class="sourceLineNo">1766</span>                if (sref.getPdbx_seq_align_end_ins_code() != null) {<a name="line.1766"></a>
<span class="sourceLineNo">1767</span>                    end_ins_code = new Character(sref.getPdbx_seq_align_end_ins_code().charAt(0));<a name="line.1767"></a>
<span class="sourceLineNo">1768</span>                }<a name="line.1768"></a>
<span class="sourceLineNo">1769</span><a name="line.1769"></a>
<span class="sourceLineNo">1770</span>                if (begin_ins_code == '?')<a name="line.1770"></a>
<span class="sourceLineNo">1771</span>                        begin_ins_code = ' ';<a name="line.1771"></a>
<span class="sourceLineNo">1772</span><a name="line.1772"></a>
<span class="sourceLineNo">1773</span>                if (end_ins_code == '?')<a name="line.1773"></a>
<span class="sourceLineNo">1774</span>                        end_ins_code = ' ';<a name="line.1774"></a>
<span class="sourceLineNo">1775</span><a name="line.1775"></a>
<span class="sourceLineNo">1776</span>                r.setSeqBegin(seqbegin);<a name="line.1776"></a>
<span class="sourceLineNo">1777</span>                r.setInsertBegin(begin_ins_code);<a name="line.1777"></a>
<span class="sourceLineNo">1778</span><a name="line.1778"></a>
<span class="sourceLineNo">1779</span>                r.setSeqEnd(seqend);<a name="line.1779"></a>
<span class="sourceLineNo">1780</span>                r.setInsertEnd(end_ins_code);<a name="line.1780"></a>
<span class="sourceLineNo">1781</span><a name="line.1781"></a>
<span class="sourceLineNo">1782</span>                int dbseqbegin = Integer.parseInt(sref.getDb_align_beg());<a name="line.1782"></a>
<span class="sourceLineNo">1783</span>                int dbseqend   = Integer.parseInt(sref.getDb_align_end());<a name="line.1783"></a>
<span class="sourceLineNo">1784</span>                <a name="line.1784"></a>
<span class="sourceLineNo">1785</span>                Character db_begin_in_code = ' ';<a name="line.1785"></a>
<span class="sourceLineNo">1786</span>                if (sref.getPdbx_db_align_beg_ins_code() != null) {<a name="line.1786"></a>
<span class="sourceLineNo">1787</span>                    db_begin_in_code = new Character(sref.getPdbx_db_align_beg_ins_code().charAt(0));<a name="line.1787"></a>
<span class="sourceLineNo">1788</span>                }<a name="line.1788"></a>
<span class="sourceLineNo">1789</span>                <a name="line.1789"></a>
<span class="sourceLineNo">1790</span>                Character db_end_in_code = ' ';<a name="line.1790"></a>
<span class="sourceLineNo">1791</span>                if (sref.getPdbx_db_align_end_ins_code() != null) {<a name="line.1791"></a>
<span class="sourceLineNo">1792</span>                    db_end_in_code = new Character(sref.getPdbx_db_align_end_ins_code().charAt(0));<a name="line.1792"></a>
<span class="sourceLineNo">1793</span>                }<a name="line.1793"></a>
<span class="sourceLineNo">1794</span><a name="line.1794"></a>
<span class="sourceLineNo">1795</span>                if (db_begin_in_code == '?')<a name="line.1795"></a>
<span class="sourceLineNo">1796</span>                        db_begin_in_code = ' ';<a name="line.1796"></a>
<span class="sourceLineNo">1797</span><a name="line.1797"></a>
<span class="sourceLineNo">1798</span>                if (db_end_in_code == '?')<a name="line.1798"></a>
<span class="sourceLineNo">1799</span>                        db_end_in_code = ' ';<a name="line.1799"></a>
<span class="sourceLineNo">1800</span><a name="line.1800"></a>
<span class="sourceLineNo">1801</span><a name="line.1801"></a>
<span class="sourceLineNo">1802</span>                r.setDbSeqBegin(dbseqbegin);<a name="line.1802"></a>
<span class="sourceLineNo">1803</span>                r.setIdbnsBegin(db_begin_in_code);<a name="line.1803"></a>
<span class="sourceLineNo">1804</span><a name="line.1804"></a>
<span class="sourceLineNo">1805</span>                r.setDbSeqEnd(dbseqend);<a name="line.1805"></a>
<span class="sourceLineNo">1806</span>                r.setIdbnsEnd(db_end_in_code);<a name="line.1806"></a>
<span class="sourceLineNo">1807</span><a name="line.1807"></a>
<span class="sourceLineNo">1808</span>                List&lt;DBRef&gt; dbrefs = structure.getDBRefs();<a name="line.1808"></a>
<span class="sourceLineNo">1809</span>                if ( dbrefs == null)<a name="line.1809"></a>
<span class="sourceLineNo">1810</span>                        dbrefs = new ArrayList&lt;DBRef&gt;();<a name="line.1810"></a>
<span class="sourceLineNo">1811</span>                dbrefs.add(r);<a name="line.1811"></a>
<span class="sourceLineNo">1812</span><a name="line.1812"></a>
<span class="sourceLineNo">1813</span>                logger.debug(r.toPDB());<a name="line.1813"></a>
<span class="sourceLineNo">1814</span><a name="line.1814"></a>
<span class="sourceLineNo">1815</span>                structure.setDBRefs(dbrefs);<a name="line.1815"></a>
<span class="sourceLineNo">1816</span><a name="line.1816"></a>
<span class="sourceLineNo">1817</span>        }<a name="line.1817"></a>
<span class="sourceLineNo">1818</span><a name="line.1818"></a>
<span class="sourceLineNo">1819</span>        @Override<a name="line.1819"></a>
<span class="sourceLineNo">1820</span>        public void newStructRefSeqDif(StructRefSeqDif sref) {<a name="line.1820"></a>
<span class="sourceLineNo">1821</span>                sequenceDifs.add(sref);<a name="line.1821"></a>
<span class="sourceLineNo">1822</span>        }<a name="line.1822"></a>
<span class="sourceLineNo">1823</span><a name="line.1823"></a>
<span class="sourceLineNo">1824</span>        private Chain getEntityChain(String entity_id){<a name="line.1824"></a>
<span class="sourceLineNo">1825</span><a name="line.1825"></a>
<span class="sourceLineNo">1826</span>                for (Chain chain : entityChains) {<a name="line.1826"></a>
<span class="sourceLineNo">1827</span>                        if ( chain.getId().equals(entity_id)){<a name="line.1827"></a>
<span class="sourceLineNo">1828</span><a name="line.1828"></a>
<span class="sourceLineNo">1829</span>                                return chain;<a name="line.1829"></a>
<span class="sourceLineNo">1830</span>                        }<a name="line.1830"></a>
<span class="sourceLineNo">1831</span>                }<a name="line.1831"></a>
<span class="sourceLineNo">1832</span>                // does not exist yet, so create...<a name="line.1832"></a>
<span class="sourceLineNo">1833</span><a name="line.1833"></a>
<span class="sourceLineNo">1834</span>                Chain   chain = new ChainImpl();<a name="line.1834"></a>
<span class="sourceLineNo">1835</span>                chain.setId(entity_id);<a name="line.1835"></a>
<span class="sourceLineNo">1836</span>                entityChains.add(chain);<a name="line.1836"></a>
<span class="sourceLineNo">1837</span><a name="line.1837"></a>
<span class="sourceLineNo">1838</span>                return chain;<a name="line.1838"></a>
<span class="sourceLineNo">1839</span><a name="line.1839"></a>
<span class="sourceLineNo">1840</span>        }<a name="line.1840"></a>
<span class="sourceLineNo">1841</span><a name="line.1841"></a>
<span class="sourceLineNo">1842</span>        //private Chain getSeqResChain(String chainID){<a name="line.1842"></a>
<span class="sourceLineNo">1843</span>        //      return getChainFromList(seqResChains, chainID);<a name="line.1843"></a>
<span class="sourceLineNo">1844</span>        //}<a name="line.1844"></a>
<span class="sourceLineNo">1845</span><a name="line.1845"></a>
<span class="sourceLineNo">1846</span><a name="line.1846"></a>
<span class="sourceLineNo">1847</span>        /**<a name="line.1847"></a>
<span class="sourceLineNo">1848</span>         * Data items in the ENTITY_SRC_GEN category record details of<a name="line.1848"></a>
<span class="sourceLineNo">1849</span>         * the source from which the entity was obtained in cases<a name="line.1849"></a>
<span class="sourceLineNo">1850</span>         * where the source was genetically manipulated.  The<a name="line.1850"></a>
<span class="sourceLineNo">1851</span>         * following are treated separately:  items pertaining to the tissue<a name="line.1851"></a>
<span class="sourceLineNo">1852</span>         * from which the gene was obtained, items pertaining to the host<a name="line.1852"></a>
<span class="sourceLineNo">1853</span>         * organism for gene expression and items pertaining to the actual<a name="line.1853"></a>
<span class="sourceLineNo">1854</span>         * producing organism (plasmid).<a name="line.1854"></a>
<span class="sourceLineNo">1855</span>         */<a name="line.1855"></a>
<span class="sourceLineNo">1856</span>        @Override<a name="line.1856"></a>
<span class="sourceLineNo">1857</span>        public void newEntitySrcGen(EntitySrcGen entitySrcGen){<a name="line.1857"></a>
<span class="sourceLineNo">1858</span><a name="line.1858"></a>
<span class="sourceLineNo">1859</span>                // add to internal list. Map to Compound object later on...<a name="line.1859"></a>
<span class="sourceLineNo">1860</span>                entitySrcGens.add(entitySrcGen);<a name="line.1860"></a>
<span class="sourceLineNo">1861</span>        }<a name="line.1861"></a>
<span class="sourceLineNo">1862</span><a name="line.1862"></a>
<span class="sourceLineNo">1863</span>        @Override<a name="line.1863"></a>
<span class="sourceLineNo">1864</span>        public void newEntitySrcNat(EntitySrcNat entitySrcNat){<a name="line.1864"></a>
<span class="sourceLineNo">1865</span><a name="line.1865"></a>
<span class="sourceLineNo">1866</span>                // add to internal list. Map to Compound object later on...<a name="line.1866"></a>
<span class="sourceLineNo">1867</span>                entitySrcNats.add(entitySrcNat);<a name="line.1867"></a>
<span class="sourceLineNo">1868</span>        }<a name="line.1868"></a>
<span class="sourceLineNo">1869</span><a name="line.1869"></a>
<span class="sourceLineNo">1870</span>        @Override<a name="line.1870"></a>
<span class="sourceLineNo">1871</span>        public void newEntitySrcSyn(EntitySrcSyn entitySrcSyn){<a name="line.1871"></a>
<span class="sourceLineNo">1872</span><a name="line.1872"></a>
<span class="sourceLineNo">1873</span>                // add to internal list. Map to Compound object later on...<a name="line.1873"></a>
<span class="sourceLineNo">1874</span>                entitySrcSyns.add(entitySrcSyn);<a name="line.1874"></a>
<span class="sourceLineNo">1875</span>        }<a name="line.1875"></a>
<span class="sourceLineNo">1876</span><a name="line.1876"></a>
<span class="sourceLineNo">1877</span>        /**<a name="line.1877"></a>
<span class="sourceLineNo">1878</span>         * The EntityPolySeq object provide the amino acid sequence objects for the Entities.<a name="line.1878"></a>
<span class="sourceLineNo">1879</span>         * Later on the entities are mapped to the BioJava {@link Chain} and {@link EntityInfo} objects.<a name="line.1879"></a>
<span class="sourceLineNo">1880</span>         * @param epolseq the EntityPolySeq record for one amino acid<a name="line.1880"></a>
<span class="sourceLineNo">1881</span>         */<a name="line.1881"></a>
<span class="sourceLineNo">1882</span>        @Override<a name="line.1882"></a>
<span class="sourceLineNo">1883</span>        public void newEntityPolySeq(EntityPolySeq epolseq) {<a name="line.1883"></a>
<span class="sourceLineNo">1884</span><a name="line.1884"></a>
<span class="sourceLineNo">1885</span>                logger.debug("NEW entity poly seq " + epolseq);<a name="line.1885"></a>
<span class="sourceLineNo">1886</span><a name="line.1886"></a>
<span class="sourceLineNo">1887</span>                int eId = -1;<a name="line.1887"></a>
<span class="sourceLineNo">1888</span>                try {<a name="line.1888"></a>
<span class="sourceLineNo">1889</span>                        eId = Integer.parseInt(epolseq.getEntity_id());<a name="line.1889"></a>
<span class="sourceLineNo">1890</span>                } catch (NumberFormatException e) {<a name="line.1890"></a>
<span class="sourceLineNo">1891</span>                        logger.warn("Could not parse entity id from EntityPolySeq: "+e.getMessage());<a name="line.1891"></a>
<span class="sourceLineNo">1892</span>                }<a name="line.1892"></a>
<span class="sourceLineNo">1893</span>                Entity e = getEntity(eId);<a name="line.1893"></a>
<span class="sourceLineNo">1894</span><a name="line.1894"></a>
<span class="sourceLineNo">1895</span>                if (e == null){<a name="line.1895"></a>
<span class="sourceLineNo">1896</span>                        logger.info("Could not find entity "+ epolseq.getEntity_id()+". Can not match sequence to it.");<a name="line.1896"></a>
<span class="sourceLineNo">1897</span>                        return;<a name="line.1897"></a>
<span class="sourceLineNo">1898</span>                }<a name="line.1898"></a>
<span class="sourceLineNo">1899</span><a name="line.1899"></a>
<span class="sourceLineNo">1900</span>                Chain entityChain = getEntityChain(epolseq.getEntity_id());<a name="line.1900"></a>
<span class="sourceLineNo">1901</span><a name="line.1901"></a>
<span class="sourceLineNo">1902</span>                // first we check through the chemcomp provider, if it fails we do some heuristics to guess the type of group<a name="line.1902"></a>
<span class="sourceLineNo">1903</span>                // TODO some of this code is analogous to getNewGroup() and we should try to unify them - JD 2016-03-08<a name="line.1903"></a>
<span class="sourceLineNo">1904</span><a name="line.1904"></a>
<span class="sourceLineNo">1905</span>                Group g = ChemCompGroupFactory.getGroupFromChemCompDictionary(epolseq.getMon_id());<a name="line.1905"></a>
<span class="sourceLineNo">1906</span>                //int seqId = Integer.parseInt(epolseq.getNum());<a name="line.1906"></a>
<span class="sourceLineNo">1907</span>                if ( g != null &amp;&amp; !g.getChemComp().isEmpty()) {<a name="line.1907"></a>
<span class="sourceLineNo">1908</span>                        if ( g instanceof AminoAcidImpl) {<a name="line.1908"></a>
<span class="sourceLineNo">1909</span>                                AminoAcidImpl aa = (AminoAcidImpl) g;<a name="line.1909"></a>
<span class="sourceLineNo">1910</span>                                aa.setRecordType(AminoAcid.SEQRESRECORD);<a name="line.1910"></a>
<span class="sourceLineNo">1911</span>                                //aa.setId(seqId);<a name="line.1911"></a>
<span class="sourceLineNo">1912</span>                        }<a name="line.1912"></a>
<span class="sourceLineNo">1913</span>                } else {<a name="line.1913"></a>
<span class="sourceLineNo">1914</span><a name="line.1914"></a>
<span class="sourceLineNo">1915</span>                        if (epolseq.getMon_id().length()==3 &amp;&amp; StructureTools.get1LetterCodeAmino(epolseq.getMon_id())!=null){<a name="line.1915"></a>
<span class="sourceLineNo">1916</span>                                AminoAcidImpl a = new AminoAcidImpl();<a name="line.1916"></a>
<span class="sourceLineNo">1917</span>                                a.setRecordType(AminoAcid.SEQRESRECORD);<a name="line.1917"></a>
<span class="sourceLineNo">1918</span>                                Character code1 = StructureTools.get1LetterCodeAmino(epolseq.getMon_id());<a name="line.1918"></a>
<span class="sourceLineNo">1919</span>                                a.setAminoType(code1);<a name="line.1919"></a>
<span class="sourceLineNo">1920</span>                                g = a;<a name="line.1920"></a>
<span class="sourceLineNo">1921</span><a name="line.1921"></a>
<span class="sourceLineNo">1922</span>                        } else if ( StructureTools.isNucleotide(epolseq.getMon_id())) {<a name="line.1922"></a>
<span class="sourceLineNo">1923</span>                                // the group is actually a nucleotide group...<a name="line.1923"></a>
<span class="sourceLineNo">1924</span>                                NucleotideImpl n = new NucleotideImpl();<a name="line.1924"></a>
<span class="sourceLineNo">1925</span>                                g = n;<a name="line.1925"></a>
<span class="sourceLineNo">1926</span><a name="line.1926"></a>
<span class="sourceLineNo">1927</span>                        } else {<a name="line.1927"></a>
<span class="sourceLineNo">1928</span>                                logger.debug("Residue {} {} is not a standard aminoacid or nucleotide, will create a het group for it", epolseq.getNum(),epolseq.getMon_id());<a name="line.1928"></a>
<span class="sourceLineNo">1929</span>                                HetatomImpl h = new HetatomImpl();<a name="line.1929"></a>
<span class="sourceLineNo">1930</span>                                g = h;<a name="line.1930"></a>
<span class="sourceLineNo">1931</span><a name="line.1931"></a>
<span class="sourceLineNo">1932</span>                        }<a name="line.1932"></a>
<span class="sourceLineNo">1933</span><a name="line.1933"></a>
<span class="sourceLineNo">1934</span><a name="line.1934"></a>
<span class="sourceLineNo">1935</span>                }<a name="line.1935"></a>
<span class="sourceLineNo">1936</span>                // at this stage we don't know about author residue numbers (insertion codes)<a name="line.1936"></a>
<span class="sourceLineNo">1937</span>                // we abuse now the ResidueNumber field setting the internal residue numbers (label_seq_id, strictly sequential and follow the seqres sequence 1 to n)<a name="line.1937"></a>
<span class="sourceLineNo">1938</span>                // later the actual ResidueNumbers (author residue numbers) have to be corrected in alignSeqRes()<a name="line.1938"></a>
<span class="sourceLineNo">1939</span>                g.setResidueNumber(ResidueNumber.fromString(epolseq.getNum()));<a name="line.1939"></a>
<span class="sourceLineNo">1940</span><a name="line.1940"></a>
<span class="sourceLineNo">1941</span>                g.setPDBName(epolseq.getMon_id());<a name="line.1941"></a>
<span class="sourceLineNo">1942</span><a name="line.1942"></a>
<span class="sourceLineNo">1943</span>                entityChain.addGroup(g);<a name="line.1943"></a>
<span class="sourceLineNo">1944</span><a name="line.1944"></a>
<span class="sourceLineNo">1945</span>        }<a name="line.1945"></a>
<span class="sourceLineNo">1946</span><a name="line.1946"></a>
<span class="sourceLineNo">1947</span>        @Override<a name="line.1947"></a>
<span class="sourceLineNo">1948</span>        public void newPdbxPolySeqScheme(PdbxPolySeqScheme ppss) {<a name="line.1948"></a>
<span class="sourceLineNo">1949</span><a name="line.1949"></a>
<span class="sourceLineNo">1950</span>                //if ( headerOnly)<a name="line.1950"></a>
<span class="sourceLineNo">1951</span>                //      return;<a name="line.1951"></a>
<span class="sourceLineNo">1952</span><a name="line.1952"></a>
<span class="sourceLineNo">1953</span>                // replace the group asym ids with the real PDB ids!<a name="line.1953"></a>
<span class="sourceLineNo">1954</span>                // replaceGroupSeqPos(ppss);  // This might be incorrect in some pdb, to use auth_seq_id of the pdbx_poly_seq_scheme.<a name="line.1954"></a>
<span class="sourceLineNo">1955</span><a name="line.1955"></a>
<span class="sourceLineNo">1956</span><a name="line.1956"></a>
<span class="sourceLineNo">1957</span>        }<a name="line.1957"></a>
<span class="sourceLineNo">1958</span><a name="line.1958"></a>
<span class="sourceLineNo">1959</span><a name="line.1959"></a>
<span class="sourceLineNo">1960</span>        @Override<a name="line.1960"></a>
<span class="sourceLineNo">1961</span>        public void newPdbxNonPolyScheme(PdbxNonPolyScheme ppss) {<a name="line.1961"></a>
<span class="sourceLineNo">1962</span><a name="line.1962"></a>
<span class="sourceLineNo">1963</span>                //if (headerOnly)<a name="line.1963"></a>
<span class="sourceLineNo">1964</span>                //      return;<a name="line.1964"></a>
<span class="sourceLineNo">1965</span><a name="line.1965"></a>
<span class="sourceLineNo">1966</span>                // merge the EntityPolySeq info and the AtomSite chains into one...<a name="line.1966"></a>
<span class="sourceLineNo">1967</span>                //already known ignore:<a name="line.1967"></a>
<span class="sourceLineNo">1968</span><a name="line.1968"></a>
<span class="sourceLineNo">1969</span>        }<a name="line.1969"></a>
<span class="sourceLineNo">1970</span><a name="line.1970"></a>
<span class="sourceLineNo">1971</span>        @Override<a name="line.1971"></a>
<span class="sourceLineNo">1972</span>        public void newPdbxEntityNonPoly(PdbxEntityNonPoly pen){<a name="line.1972"></a>
<span class="sourceLineNo">1973</span>                // TODO: do something with them...<a name="line.1973"></a>
<span class="sourceLineNo">1974</span>                // not implemented yet...<a name="line.1974"></a>
<span class="sourceLineNo">1975</span>                logger.debug(pen.getEntity_id() + " " + pen.getName() + " " + pen.getComp_id());<a name="line.1975"></a>
<span class="sourceLineNo">1976</span><a name="line.1976"></a>
<span class="sourceLineNo">1977</span>        }<a name="line.1977"></a>
<span class="sourceLineNo">1978</span><a name="line.1978"></a>
<span class="sourceLineNo">1979</span>        @Override<a name="line.1979"></a>
<span class="sourceLineNo">1980</span>        public void newChemComp(ChemComp c) {<a name="line.1980"></a>
<span class="sourceLineNo">1981</span>                // TODO: do something with them...<a name="line.1981"></a>
<span class="sourceLineNo">1982</span><a name="line.1982"></a>
<span class="sourceLineNo">1983</span>        }<a name="line.1983"></a>
<span class="sourceLineNo">1984</span><a name="line.1984"></a>
<span class="sourceLineNo">1985</span>        @Override<a name="line.1985"></a>
<span class="sourceLineNo">1986</span>        public void newGenericData(String category, List&lt;String&gt; loopFields,<a name="line.1986"></a>
<span class="sourceLineNo">1987</span>                        List&lt;String&gt; lineData) {<a name="line.1987"></a>
<span class="sourceLineNo">1988</span><a name="line.1988"></a>
<span class="sourceLineNo">1989</span>                //logger.debug("unhandled category so far: " + category);<a name="line.1989"></a>
<span class="sourceLineNo">1990</span>        }<a name="line.1990"></a>
<span class="sourceLineNo">1991</span><a name="line.1991"></a>
<span class="sourceLineNo">1992</span>        @Override<a name="line.1992"></a>
<span class="sourceLineNo">1993</span>        public FileParsingParameters getFileParsingParameters()<a name="line.1993"></a>
<span class="sourceLineNo">1994</span>        {<a name="line.1994"></a>
<span class="sourceLineNo">1995</span>                return params;<a name="line.1995"></a>
<span class="sourceLineNo">1996</span>        }<a name="line.1996"></a>
<span class="sourceLineNo">1997</span><a name="line.1997"></a>
<span class="sourceLineNo">1998</span>        @Override<a name="line.1998"></a>
<span class="sourceLineNo">1999</span>        public void setFileParsingParameters(FileParsingParameters params)<a name="line.1999"></a>
<span class="sourceLineNo">2000</span>        {<a name="line.2000"></a>
<span class="sourceLineNo">2001</span>                this.params = params;<a name="line.2001"></a>
<span class="sourceLineNo">2002</span><a name="line.2002"></a>
<span class="sourceLineNo">2003</span>        }<a name="line.2003"></a>
<span class="sourceLineNo">2004</span><a name="line.2004"></a>
<span class="sourceLineNo">2005</span>        @Override<a name="line.2005"></a>
<span class="sourceLineNo">2006</span>        public void newChemCompDescriptor(ChemCompDescriptor ccd) {<a name="line.2006"></a>
<span class="sourceLineNo">2007</span><a name="line.2007"></a>
<span class="sourceLineNo">2008</span>                // TODO nothing happening here yet.<a name="line.2008"></a>
<span class="sourceLineNo">2009</span><a name="line.2009"></a>
<span class="sourceLineNo">2010</span>        }<a name="line.2010"></a>
<span class="sourceLineNo">2011</span><a name="line.2011"></a>
<span class="sourceLineNo">2012</span><a name="line.2012"></a>
<span class="sourceLineNo">2013</span><a name="line.2013"></a>
<span class="sourceLineNo">2014</span>        public List&lt;PdbxStructOperList&gt; getStructOpers() {<a name="line.2014"></a>
<span class="sourceLineNo">2015</span>                return structOpers;<a name="line.2015"></a>
<span class="sourceLineNo">2016</span>        }<a name="line.2016"></a>
<span class="sourceLineNo">2017</span><a name="line.2017"></a>
<span class="sourceLineNo">2018</span>        @Override<a name="line.2018"></a>
<span class="sourceLineNo">2019</span>        public void newPdbxStrucAssembly(PdbxStructAssembly strucAssembly) {<a name="line.2019"></a>
<span class="sourceLineNo">2020</span>                strucAssemblies.add(strucAssembly);<a name="line.2020"></a>
<span class="sourceLineNo">2021</span><a name="line.2021"></a>
<span class="sourceLineNo">2022</span>        }<a name="line.2022"></a>
<span class="sourceLineNo">2023</span><a name="line.2023"></a>
<span class="sourceLineNo">2024</span>        public List&lt;PdbxStructAssembly&gt; getStructAssemblies(){<a name="line.2024"></a>
<span class="sourceLineNo">2025</span>                return strucAssemblies;<a name="line.2025"></a>
<span class="sourceLineNo">2026</span>        }<a name="line.2026"></a>
<span class="sourceLineNo">2027</span><a name="line.2027"></a>
<span class="sourceLineNo">2028</span>        @Override<a name="line.2028"></a>
<span class="sourceLineNo">2029</span>        public void newPdbxStrucAssemblyGen(PdbxStructAssemblyGen strucAssembly) {<a name="line.2029"></a>
<span class="sourceLineNo">2030</span>                strucAssemblyGens.add(strucAssembly);<a name="line.2030"></a>
<span class="sourceLineNo">2031</span><a name="line.2031"></a>
<span class="sourceLineNo">2032</span>        }<a name="line.2032"></a>
<span class="sourceLineNo">2033</span><a name="line.2033"></a>
<span class="sourceLineNo">2034</span>        public List&lt;PdbxStructAssemblyGen&gt; getStructAssemblyGens(){<a name="line.2034"></a>
<span class="sourceLineNo">2035</span>                return strucAssemblyGens;<a name="line.2035"></a>
<span class="sourceLineNo">2036</span>        }<a name="line.2036"></a>
<span class="sourceLineNo">2037</span><a name="line.2037"></a>
<span class="sourceLineNo">2038</span>        @Override<a name="line.2038"></a>
<span class="sourceLineNo">2039</span>        public void newChemCompAtom(ChemCompAtom atom) {<a name="line.2039"></a>
<span class="sourceLineNo">2040</span><a name="line.2040"></a>
<span class="sourceLineNo">2041</span>        }<a name="line.2041"></a>
<span class="sourceLineNo">2042</span><a name="line.2042"></a>
<span class="sourceLineNo">2043</span>        @Override<a name="line.2043"></a>
<span class="sourceLineNo">2044</span>        public void newPdbxChemCompIndentifier(PdbxChemCompIdentifier id) {<a name="line.2044"></a>
<span class="sourceLineNo">2045</span><a name="line.2045"></a>
<span class="sourceLineNo">2046</span>        }<a name="line.2046"></a>
<span class="sourceLineNo">2047</span><a name="line.2047"></a>
<span class="sourceLineNo">2048</span>        @Override<a name="line.2048"></a>
<span class="sourceLineNo">2049</span>        public void newChemCompBond(ChemCompBond bond) {<a name="line.2049"></a>
<span class="sourceLineNo">2050</span><a name="line.2050"></a>
<span class="sourceLineNo">2051</span>        }<a name="line.2051"></a>
<span class="sourceLineNo">2052</span><a name="line.2052"></a>
<span class="sourceLineNo">2053</span>        @Override<a name="line.2053"></a>
<span class="sourceLineNo">2054</span>        public void newPdbxChemCompDescriptor(PdbxChemCompDescriptor desc) {<a name="line.2054"></a>
<span class="sourceLineNo">2055</span><a name="line.2055"></a>
<span class="sourceLineNo">2056</span>        }<a name="line.2056"></a>
<span class="sourceLineNo">2057</span><a name="line.2057"></a>
<span class="sourceLineNo">2058</span>        @Override<a name="line.2058"></a>
<span class="sourceLineNo">2059</span>        public void newStructConn(StructConn structConn) {<a name="line.2059"></a>
<span class="sourceLineNo">2060</span>                this.structConn.add(structConn);<a name="line.2060"></a>
<span class="sourceLineNo">2061</span>        }<a name="line.2061"></a>
<span class="sourceLineNo">2062</span><a name="line.2062"></a>
<span class="sourceLineNo">2063</span>        @Override<a name="line.2063"></a>
<span class="sourceLineNo">2064</span>        public void newStructSiteGen(StructSiteGen siteGen) { this.structSiteGens.add(siteGen); }<a name="line.2064"></a>
<span class="sourceLineNo">2065</span><a name="line.2065"></a>
<span class="sourceLineNo">2066</span>        @Override<a name="line.2066"></a>
<span class="sourceLineNo">2067</span>        public void newStructSite(StructSite structSite) {<a name="line.2067"></a>
<span class="sourceLineNo">2068</span><a name="line.2068"></a>
<span class="sourceLineNo">2069</span>                if (params.isHeaderOnly()) {<a name="line.2069"></a>
<span class="sourceLineNo">2070</span>                        return;<a name="line.2070"></a>
<span class="sourceLineNo">2071</span>                }<a name="line.2071"></a>
<span class="sourceLineNo">2072</span><a name="line.2072"></a>
<span class="sourceLineNo">2073</span>                // Simply implement the method.<a name="line.2073"></a>
<span class="sourceLineNo">2074</span>                List&lt;Site&gt; sites = structure.getSites();<a name="line.2074"></a>
<span class="sourceLineNo">2075</span>                if (sites == null) sites = new ArrayList&lt;Site&gt;();<a name="line.2075"></a>
<span class="sourceLineNo">2076</span><a name="line.2076"></a>
<span class="sourceLineNo">2077</span>                Site site = null;<a name="line.2077"></a>
<span class="sourceLineNo">2078</span>                for (Site asite : sites) {<a name="line.2078"></a>
<span class="sourceLineNo">2079</span>                        if (asite.getSiteID().equals(structSite.getId())) {<a name="line.2079"></a>
<span class="sourceLineNo">2080</span>                                site = asite;           // Prevent duplicate siteIds<a name="line.2080"></a>
<span class="sourceLineNo">2081</span>                        }<a name="line.2081"></a>
<span class="sourceLineNo">2082</span>                }<a name="line.2082"></a>
<span class="sourceLineNo">2083</span>                boolean addSite = false;<a name="line.2083"></a>
<span class="sourceLineNo">2084</span>                if (site == null) { site = new Site(); addSite = true; }<a name="line.2084"></a>
<span class="sourceLineNo">2085</span>                site.setSiteID(structSite.getId());<a name="line.2085"></a>
<span class="sourceLineNo">2086</span>                site.setDescription(structSite.getDetails());<a name="line.2086"></a>
<span class="sourceLineNo">2087</span>                // site.setPdbxEvidenceCode(structSite.getPdbxEvidenceCode()); // TODO - add addition fields in Sites<a name="line.2087"></a>
<span class="sourceLineNo">2088</span>                if (addSite) sites.add(site);<a name="line.2088"></a>
<span class="sourceLineNo">2089</span><a name="line.2089"></a>
<span class="sourceLineNo">2090</span>                structure.setSites(sites);<a name="line.2090"></a>
<span class="sourceLineNo">2091</span>        }<a name="line.2091"></a>
<span class="sourceLineNo">2092</span><a name="line.2092"></a>
<span class="sourceLineNo">2093</span>        /**<a name="line.2093"></a>
<span class="sourceLineNo">2094</span>         * Build sites in a BioJava Structure using the original author chain id &amp; residue numbers.<a name="line.2094"></a>
<span class="sourceLineNo">2095</span>         * Sites are built from struct_site_gen records that have been parsed.<a name="line.2095"></a>
<span class="sourceLineNo">2096</span>         */<a name="line.2096"></a>
<span class="sourceLineNo">2097</span>        private void addSites() {<a name="line.2097"></a>
<span class="sourceLineNo">2098</span>                List&lt;Site&gt; sites = structure.getSites();<a name="line.2098"></a>
<span class="sourceLineNo">2099</span>                if (sites == null) sites = new ArrayList&lt;Site&gt;();<a name="line.2099"></a>
<span class="sourceLineNo">2100</span><a name="line.2100"></a>
<span class="sourceLineNo">2101</span>                for (StructSiteGen siteGen : structSiteGens) {<a name="line.2101"></a>
<span class="sourceLineNo">2102</span>                        // For each StructSiteGen, find the residues involved, if they exist then<a name="line.2102"></a>
<span class="sourceLineNo">2103</span>                        String site_id = siteGen.getSite_id(); // multiple could be in same site.<a name="line.2103"></a>
<span class="sourceLineNo">2104</span>                        if (site_id == null) site_id = "";<a name="line.2104"></a>
<span class="sourceLineNo">2105</span>                        String comp_id = siteGen.getLabel_comp_id();  // PDBName<a name="line.2105"></a>
<span class="sourceLineNo">2106</span><a name="line.2106"></a>
<span class="sourceLineNo">2107</span>                        // Assumption: the author chain ID and residue number for the site is consistent with the original<a name="line.2107"></a>
<span class="sourceLineNo">2108</span>                        // author chain id and residue numbers.<a name="line.2108"></a>
<span class="sourceLineNo">2109</span><a name="line.2109"></a>
<span class="sourceLineNo">2110</span>                        String asymId = siteGen.getLabel_asym_id(); // chain name<a name="line.2110"></a>
<span class="sourceLineNo">2111</span>                        String authId = siteGen.getAuth_asym_id(); // chain Id<a name="line.2111"></a>
<span class="sourceLineNo">2112</span>                        String auth_seq_id = siteGen.getAuth_seq_id(); // Res num<a name="line.2112"></a>
<span class="sourceLineNo">2113</span><a name="line.2113"></a>
<span class="sourceLineNo">2114</span>                        String insCode = siteGen.getPdbx_auth_ins_code();<a name="line.2114"></a>
<span class="sourceLineNo">2115</span>                        if ( insCode != null &amp;&amp; insCode.equals("?"))<a name="line.2115"></a>
<span class="sourceLineNo">2116</span>                                insCode = null;<a name="line.2116"></a>
<span class="sourceLineNo">2117</span><a name="line.2117"></a>
<span class="sourceLineNo">2118</span>                        // Look for asymID = chainID and seqID = seq_ID.  Check that comp_id matches the resname.<a name="line.2118"></a>
<span class="sourceLineNo">2119</span>                        Group g = null;<a name="line.2119"></a>
<span class="sourceLineNo">2120</span>                        try {<a name="line.2120"></a>
<span class="sourceLineNo">2121</span>                                Chain chain = structure.getChain(asymId);<a name="line.2121"></a>
<span class="sourceLineNo">2122</span><a name="line.2122"></a>
<span class="sourceLineNo">2123</span>                                if (null != chain) {<a name="line.2123"></a>
<span class="sourceLineNo">2124</span>                                        try {<a name="line.2124"></a>
<span class="sourceLineNo">2125</span>                                                Character insChar = null;<a name="line.2125"></a>
<span class="sourceLineNo">2126</span>                                                if (null != insCode &amp;&amp; insCode.length() &gt; 0) insChar = insCode.charAt(0);<a name="line.2126"></a>
<span class="sourceLineNo">2127</span>                                                g = chain.getGroupByPDB(new ResidueNumber(null, Integer.parseInt(auth_seq_id), insChar));<a name="line.2127"></a>
<span class="sourceLineNo">2128</span>                                        } catch (NumberFormatException e) {<a name="line.2128"></a>
<span class="sourceLineNo">2129</span>                                                logger.warn("Could not lookup residue : " + authId + auth_seq_id);<a name="line.2129"></a>
<span class="sourceLineNo">2130</span>                                        }<a name="line.2130"></a>
<span class="sourceLineNo">2131</span>                                }<a name="line.2131"></a>
<span class="sourceLineNo">2132</span>                        } catch (StructureException e) {<a name="line.2132"></a>
<span class="sourceLineNo">2133</span>                                logger.warn("Problem finding residue in site entry " + siteGen.getSite_id() + " - " + e.getMessage(), e.getMessage());<a name="line.2133"></a>
<span class="sourceLineNo">2134</span>                        }<a name="line.2134"></a>
<span class="sourceLineNo">2135</span><a name="line.2135"></a>
<span class="sourceLineNo">2136</span>                        if (g != null) {<a name="line.2136"></a>
<span class="sourceLineNo">2137</span>                                // 2. find the site_id, if not existing, create anew.<a name="line.2137"></a>
<span class="sourceLineNo">2138</span>                                Site site = null;<a name="line.2138"></a>
<span class="sourceLineNo">2139</span>                                for (Site asite: sites) {<a name="line.2139"></a>
<span class="sourceLineNo">2140</span>                                        if (site_id.equals(asite.getSiteID())) site = asite;<a name="line.2140"></a>
<span class="sourceLineNo">2141</span>                                }<a name="line.2141"></a>
<span class="sourceLineNo">2142</span><a name="line.2142"></a>
<span class="sourceLineNo">2143</span>                                boolean addSite = false;<a name="line.2143"></a>
<span class="sourceLineNo">2144</span><a name="line.2144"></a>
<span class="sourceLineNo">2145</span>                                // 3. add this residue to the site.<a name="line.2145"></a>
<span class="sourceLineNo">2146</span>                                if (site == null) {<a name="line.2146"></a>
<span class="sourceLineNo">2147</span>                                        addSite = true;<a name="line.2147"></a>
<span class="sourceLineNo">2148</span>                                        site = new Site();<a name="line.2148"></a>
<span class="sourceLineNo">2149</span>                                        site.setSiteID(site_id);<a name="line.2149"></a>
<span class="sourceLineNo">2150</span>                                }<a name="line.2150"></a>
<span class="sourceLineNo">2151</span><a name="line.2151"></a>
<span class="sourceLineNo">2152</span>                                List&lt;Group&gt; groups = site.getGroups();<a name="line.2152"></a>
<span class="sourceLineNo">2153</span>                                if (groups == null) groups = new ArrayList&lt;Group&gt;();<a name="line.2153"></a>
<span class="sourceLineNo">2154</span><a name="line.2154"></a>
<span class="sourceLineNo">2155</span>                                // Check the self-consistency of the residue reference from auth_seq_id and chain_id<a name="line.2155"></a>
<span class="sourceLineNo">2156</span>                                if (!comp_id.equals(g.getPDBName())) {<a name="line.2156"></a>
<span class="sourceLineNo">2157</span>                                        logger.warn("comp_id doesn't match the residue at " + authId + " " + auth_seq_id + " - skipping");<a name="line.2157"></a>
<span class="sourceLineNo">2158</span>                                } else {<a name="line.2158"></a>
<span class="sourceLineNo">2159</span>                                        groups.add(g);<a name="line.2159"></a>
<span class="sourceLineNo">2160</span>                                        site.setGroups(groups);<a name="line.2160"></a>
<span class="sourceLineNo">2161</span>                                }<a name="line.2161"></a>
<span class="sourceLineNo">2162</span>                                if (addSite) sites.add(site);<a name="line.2162"></a>
<span class="sourceLineNo">2163</span>                        }<a name="line.2163"></a>
<span class="sourceLineNo">2164</span>                }<a name="line.2164"></a>
<span class="sourceLineNo">2165</span>                structure.setSites(sites);<a name="line.2165"></a>
<span class="sourceLineNo">2166</span>        }<a name="line.2166"></a>
<span class="sourceLineNo">2167</span>}<a name="line.2167"></a>




























































</pre>
</div>
</body>
</html>
