<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.11) on Fri Oct 29 16:26:12 PDT 2021 -->
<title>Uses of Class org.biojava.nbio.core.exceptions.CompoundNotFoundException (biojava 6.0.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2021-10-29">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class org.biojava.nbio.core.exceptions.CompoundNotFoundException (biojava 6.0.0 API)";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Class org.biojava.nbio.core.exceptions.CompoundNotFoundException" class="title">Uses of Class<br>org.biojava.nbio.core.exceptions.CompoundNotFoundException</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#demo">demo</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.io.template">org.biojava.nbio.core.sequence.io.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.loader">org.biojava.nbio.core.sequence.loader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.storage">org.biojava.nbio.core.sequence.storage</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.views">org.biojava.nbio.core.sequence.views</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.util">org.biojava.nbio.core.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome.io.fastq">org.biojava.nbio.genome.io.fastq</a></th>
<td class="colLast">
<div class="block">FASTQ and variants sequence format I/O.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome.util">org.biojava.nbio.genome.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.cluster">org.biojava.nbio.structure.cluster</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></th>
<td class="colLast">
<div class="block">Input and Output of Structures</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="demo">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../../../../../demo/package-summary.html">demo</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../../../demo/package-summary.html">demo</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BioJavaAADemo.</span><code><span class="memberNameLink"><a href="../../../../../../demo/BioJavaAADemo.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/package-summary.html">org.biojava.nbio.core.sequence</a></h3>
<table class="useSummary">
<caption><span>Constructors in <a href="../../sequence/package-summary.html">org.biojava.nbio.core.sequence</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/BasicSequence.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)">BasicSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence,
             <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/BasicSequence.html" title="type parameter in BasicSequence">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/ChromosomeSequence.html#%3Cinit%3E(java.lang.String)">ChromosomeSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;seqString)</code></th>
<td class="colLast">
<div class="block">String is king and assume DNA</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/ChromosomeSequence.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)">ChromosomeSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;seqString,
                  <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Allows the creation of a ChromosomeSequence using String for the sequence with a custom CompoundSet</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/DNASequence.html#%3Cinit%3E(java.lang.String)">DNASequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;seqString)</code></th>
<td class="colLast">
<div class="block">String is king and create a sequence from DNA with default DNA compound set</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/DNASequence.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)">DNASequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;seqString,
           <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a sequence from a string with user defined compound set</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/ProteinSequence.html#%3Cinit%3E(java.lang.String)">ProteinSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;seqString)</code></th>
<td class="colLast">
<div class="block">Create a protein from a string</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/ProteinSequence.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)">ProteinSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;seqString,
               <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a protein from a string with a user defined set of amino acids</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/RNASequence.html#%3Cinit%3E(java.lang.String)">RNASequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;seqString)</code></th>
<td class="colLast">
<div class="block">Create a RNA sequence from a String</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/RNASequence.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)">RNASequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;seqString,
           <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a RNA sequence from a string with a user defined RNA compound set</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ABITrace.</span><code><span class="memberNameLink"><a href="../../sequence/io/ABITrace.html#getSequence()">getSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the original programmatically determined (unedited) sequence as a <code>AbstractSequence<NucleotideCompound></code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../sequence/io/CasePreservingProteinSequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../sequence/io/DNASequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyDNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../sequence/io/FileProxyDNASequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence and the offset
 index in the fasta file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../sequence/io/FileProxyProteinSequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence and the offset
 index in the fasta file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileProxyRNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../sequence/io/FileProxyRNASequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence and the offset
 index in the fasta file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../sequence/io/ProteinSequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../sequence/io/RNASequenceCreator.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ABITrace.</span><code><span class="memberNameLink"><a href="../../sequence/io/ABITrace.html#getTrace(java.lang.String)">getTrace</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;base)</code></th>
<td class="colLast">
<div class="block">Returns one of the four traces - all of the y-coordinate values,
 each of which correspond to a single x-coordinate relative to the
 position in the array, so that if element 4 in the array is 972, then
 x is 4 and y is 972 for that point.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="../../sequence/io/GenbankReader.html" title="type parameter in GenbankReader">S</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankReader.</span><code><span class="memberNameLink"><a href="../../sequence/io/GenbankReader.html#process()">process</a></span>()</code></th>
<td class="colLast">
<div class="block">The parsing is done in this method.<br>
 This method will return all the available Genbank records
 in the File or InputStream, closes the underlying resource,
 and return the results in <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>LinkedHashMap</code></a>.<br>
 You don't need to call <a href="../../sequence/io/GenbankReader.html#close()"><code>GenbankReader.close()</code></a> after calling this method.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="../../sequence/io/GenbankReader.html" title="type parameter in GenbankReader">S</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankReader.</span><code><span class="memberNameLink"><a href="../../sequence/io/GenbankReader.html#process(int)">process</a></span>&#8203;(int&nbsp;max)</code></th>
<td class="colLast">
<div class="block">This method tries to parse maximum <code>max</code> records from
 the open File or InputStream, and leaves the underlying resource open.<br>

 Subsequent calls to the same method continue parsing the rest of the file.<br>
 This is particularly useful when dealing with very big data files,
 (e.g.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.io.template">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../sequence/io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceCreatorInterface.</span><code><span class="memberNameLink"><a href="../../sequence/io/template/SequenceCreatorInterface.html#getSequence(java.lang.String,long)">getSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence,
           long&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.loader">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceFileProxyLoader.</span><code><span class="memberNameLink"><a href="../../sequence/loader/SequenceFileProxyLoader.html#setContents(java.lang.String)">setContents</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StringProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../sequence/loader/StringProxySequenceReader.html#setContents(java.lang.String)">setContents</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StringProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../sequence/loader/StringProxySequenceReader.html#setContents(java.lang.String,java.util.ArrayList)">setContents</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence,
           <a href="https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&nbsp;features)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">UniprotProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../sequence/loader/UniprotProxySequenceReader.html#setContents(java.lang.String)">setContents</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Once the sequence is retrieved set the contents and make sure everything this is valid
 Some uniprot records contain white space in the sequence.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/loader/GenbankProxySequenceReader.html#%3Cinit%3E(java.lang.String,java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)">GenbankProxySequenceReader</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;genbankDirectoryCache,
                          <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accessionID,
                          <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/loader/GenbankProxySequenceReader.html" title="type parameter in GenbankProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/loader/SequenceFileProxyLoader.html#%3Cinit%3E(java.io.File,org.biojava.nbio.core.sequence.io.template.SequenceParserInterface,long,int,org.biojava.nbio.core.sequence.template.CompoundSet)">SequenceFileProxyLoader</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file,
                       <a href="../../sequence/io/template/SequenceParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceParserInterface</a>&nbsp;sequenceParser,
                       long&nbsp;sequenceStartIndex,
                       int&nbsp;sequenceLength,
                       <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/loader/StringProxySequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)">StringProxySequenceReader</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence,
                         <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/loader/UniprotProxySequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)">UniprotProxySequenceReader</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession,
                          <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">The UniProt id is used to retrieve the UniProt XML which is then parsed as a DOM object
 so we know everything about the protein.</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/loader/UniprotProxySequenceReader.html#%3Cinit%3E(org.w3c.dom.Document,org.biojava.nbio.core.sequence.template.CompoundSet)">UniprotProxySequenceReader</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/org/w3c/dom/Document.html?is-external=true" title="class or interface in org.w3c.dom" class="externalLink">Document</a>&nbsp;document,
                          <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">The xml is passed in as a DOM object so we know everything about the protein.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.storage">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ArrayListSequenceReader.</span><code><span class="memberNameLink"><a href="../../sequence/storage/ArrayListSequenceReader.html#setContents(java.lang.String)">setContents</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BitSequenceReader.</span><code><span class="memberNameLink"><a href="../../sequence/storage/BitSequenceReader.html#setContents(java.lang.String)">setContents</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Class is immutable & so this is unsupported</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">JoiningSequenceReader.</span><code><span class="memberNameLink"><a href="../../sequence/storage/JoiningSequenceReader.html#setContents(java.lang.String)">setContents</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SingleCompoundSequenceReader.</span><code><span class="memberNameLink"><a href="../../sequence/storage/SingleCompoundSequenceReader.html#setContents(java.lang.String)">setContents</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Unsupported</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/storage/ArrayListSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)">ArrayListSequenceReader</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence,
                       <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.template">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceReader.</span><code><span class="memberNameLink"><a href="../../sequence/template/SequenceReader.html#setContents(java.lang.String)">setContents</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../sequence/template/AbstractSequence.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)">AbstractSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;seqString,
                <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">Create a Sequence from a simple string where the values should be found in compoundSet</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.views">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/views/package-summary.html">org.biojava.nbio.core.sequence.views</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../sequence/views/package-summary.html">org.biojava.nbio.core.sequence.views</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RnaSequenceView.</span><code><span class="memberNameLink"><a href="../../sequence/views/RnaSequenceView.html#setContents(java.lang.String)">setContents</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.util">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../util/package-summary.html">org.biojava.nbio.core.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../util/package-summary.html">org.biojava.nbio.core.util</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceTools.</span><code><span class="memberNameLink"><a href="../../util/SequenceTools.html#getSequenceFromString(java.lang.String)">getSequenceFromString</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Attempts to parse String as a DNA sequence first.<br/>
 If this fails it tries to  parse as a ProteinSequence.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.genome.io.fastq">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../../genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastqTools.</span><code><span class="memberNameLink"><a href="../../../genome/io/fastq/FastqTools.html#createDNASequence(org.biojava.nbio.genome.io.fastq.Fastq)">createDNASequence</a></span>&#8203;(<a href="../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></th>
<td class="colLast">
<div class="block">Create and return a new <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> from the specified FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastqTools.</span><code><span class="memberNameLink"><a href="../../../genome/io/fastq/FastqTools.html#createDNASequenceWithErrorProbabilities(org.biojava.nbio.genome.io.fastq.Fastq)">createDNASequenceWithErrorProbabilities</a></span>&#8203;(<a href="../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></th>
<td class="colLast">
<div class="block">Create and return a new <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with error probabilities from the specified
 FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastqTools.</span><code><span class="memberNameLink"><a href="../../../genome/io/fastq/FastqTools.html#createDNASequenceWithQualityScores(org.biojava.nbio.genome.io.fastq.Fastq)">createDNASequenceWithQualityScores</a></span>&#8203;(<a href="../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></th>
<td class="colLast">
<div class="block">Create and return a new <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores from the specified
 FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastqTools.</span><code><span class="memberNameLink"><a href="../../../genome/io/fastq/FastqTools.html#createDNASequenceWithQualityScoresAndErrorProbabilities(org.biojava.nbio.genome.io.fastq.Fastq)">createDNASequenceWithQualityScoresAndErrorProbabilities</a></span>&#8203;(<a href="../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></th>
<td class="colLast">
<div class="block">Create and return a new <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores and error probabilities from the
 specified FASTQ formatted sequence.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.genome.util">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../../genome/util/package-summary.html">org.biojava.nbio.genome.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genome/util/package-summary.html">org.biojava.nbio.genome.util</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinMappingTools.</span><code><span class="memberNameLink"><a href="../../../genome/util/ProteinMappingTools.html#convertDNAtoProteinSequence(java.lang.String)">convertDNAtoProteinSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;dnaSequence)</code></th>
<td class="colLast">
<div class="block">Converts the DNA sequence to protein sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinMappingTools.</span><code><span class="memberNameLink"><a href="../../../genome/util/ProteinMappingTools.html#convertDNAtoProteinSequence(org.biojava.nbio.core.sequence.DNASequence)">convertDNAtoProteinSequence</a></span>&#8203;(<a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;dnaSequence)</code></th>
<td class="colLast">
<div class="block">Converts the DNA sequence to protein sequence.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.multiple.util">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../../structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static org.forester.phylogeny.Phylogeny</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../../../structure/align/multiple/util/MultipleAlignmentTools.html#getHSDMTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getHSDMTree</a></span>&#8203;(<a href="../../../structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></th>
<td class="colLast">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using
 dissimilarity scores (DS), based in SDM Substitution Matrix (ideal for
 distantly related proteins, structure-derived) and the Neighbor Joining
 algorithm from forester.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static org.forester.phylogeny.Phylogeny</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../../../structure/align/multiple/util/MultipleAlignmentTools.html#getKimuraTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getKimuraTree</a></span>&#8203;(<a href="../../../structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></th>
<td class="colLast">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using Kimura
 distances and the Neighbor Joining algorithm from forester.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;<a href="../../sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,&#8203;<a href="../../sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../../../structure/align/multiple/util/MultipleAlignmentTools.html#toProteinMSA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">toProteinMSA</a></span>&#8203;(<a href="../../../structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></th>
<td class="colLast">
<div class="block">Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid
 residues.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.cluster">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../../structure/cluster/package-summary.html">org.biojava.nbio.structure.cluster</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../structure/cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Subunit.</span><code><span class="memberNameLink"><a href="../../../structure/cluster/Subunit.html#getProteinSequence()">getProteinSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the protein sequence of the Subunit.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SubunitCluster.</span><code><span class="memberNameLink"><a href="../../../structure/cluster/SubunitCluster.html#mergeSequence(org.biojava.nbio.structure.cluster.SubunitCluster,org.biojava.nbio.structure.cluster.SubunitClustererParameters)">mergeSequence</a></span>&#8203;(<a href="../../../structure/cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster">SubunitCluster</a>&nbsp;other,
             <a href="../../../structure/cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Merges the other SubunitCluster into this one if their representatives
 sequences are similar (according to the criteria in params).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SubunitCluster.</span><code><span class="memberNameLink"><a href="../../../structure/cluster/SubunitCluster.html#mergeSequence(org.biojava.nbio.structure.cluster.SubunitCluster,org.biojava.nbio.structure.cluster.SubunitClustererParameters,org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">mergeSequence</a></span>&#8203;(<a href="../../../structure/cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster">SubunitCluster</a>&nbsp;other,
             <a href="../../../structure/cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a>&nbsp;params,
             <a href="../../../alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;alignerType,
             <a href="../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
             <a href="../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;subsMatrix)</code></th>
<td class="colLast">
<div class="block">Merges the other SubunitCluster into this one if their representatives
 sequences are similar (according to the criteria in params).</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.io">
<!--   -->
</a>
<h3>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../../structure/io/package-summary.html">org.biojava.nbio.structure.io</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../structure/io/package-summary.html">org.biojava.nbio.structure.io</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../structure/io/FastaAFPChainConverter.html#cpFastaToAfpChain(java.lang.String,java.lang.String,org.biojava.nbio.structure.Structure,int)">cpFastaToAfpChain</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;first,
                 <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;second,
                 <a href="../../../structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../structure/io/FastaAFPChainConverter.html#fastaStringToAfpChain(java.lang.String,java.lang.String,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure)">fastaStringToAfpChain</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence1,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence2,
                     <a href="../../../structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
                     <a href="../../../structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code></th>
<td class="colLast">
<div class="block">Returns an AFPChain corresponding to the alignment between <code>structure1</code> and <code>structure2</code>, which is given by the gapped protein sequences <code>sequence1</code> and <code>sequence2</code>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../structure/io/FastaAFPChainConverter.html#fastaToAfpChain(java.lang.String,java.lang.String,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence1,
               <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence2,
               <a href="../../../structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code></th>
<td class="colLast">
<div class="block">TODO Write comment</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2021 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
