<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.16) on Fri Oct 28 16:15:48 PDT 2022 -->
<title>Uses of Package org.biojava.nbio.core.sequence (biojava 6.1.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2022-10-28">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Package org.biojava.nbio.core.sequence (biojava 6.1.0 API)";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h1 title="Uses of Package org.biojava.nbio.core.sequence" class="title">Uses of Package<br>org.biojava.nbio.core.sequence</h1>
</div>
<div class="contentContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="package-summary.html">org.biojava.nbio.core.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.aaproperties">org.biojava.nbio.aaproperties</a></th>
<td class="colLast">
<div class="block">Set of classes that responsible for providing APIs and Command Prompt access to generate some basic physico-chemical properties of protein sequences.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.aaproperties.profeat">org.biojava.nbio.aaproperties.profeat</a></th>
<td class="colLast">
<div class="block">Set of classes that responsible for providing APIs access to generate some PROFEAT properties of protein sequences.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.aaproperties.profeat.convertor">org.biojava.nbio.aaproperties.profeat.convertor</a></th>
<td class="colLast">
<div class="block">Set of classes that enable the conversion protein sequences into various attributes.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.alignment">org.biojava.nbio.core.alignment</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.loader">org.biojava.nbio.core.sequence.loader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.location">org.biojava.nbio.core.sequence.location</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.location.template">org.biojava.nbio.core.sequence.location.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.storage">org.biojava.nbio.core.sequence.storage</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.transcription">org.biojava.nbio.core.sequence.transcription</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.util">org.biojava.nbio.core.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome">org.biojava.nbio.genome</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome.homology">org.biojava.nbio.genome.homology</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome.io.fastq">org.biojava.nbio.genome.io.fastq</a></th>
<td class="colLast">
<div class="block">FASTQ and variants sequence format I/O.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome.parsers.geneid">org.biojava.nbio.genome.parsers.geneid</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome.parsers.gff">org.biojava.nbio.genome.parsers.gff</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome.util">org.biojava.nbio.genome.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.phylo">org.biojava.nbio.phylo</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.ronn">org.biojava.nbio.ronn</a></th>
<td class="colLast">
<div class="block">
 BioJava provide a module biojava-protein-disorder for prediction disordered regions
 from a protein sequence.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.cluster">org.biojava.nbio.structure.cluster</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></th>
<td class="colLast">
<div class="block">Input and Output of Structures</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.ws.hmmer">org.biojava.nbio.ws.hmmer</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.aaproperties">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../aaproperties/package-summary.html">org.biojava.nbio.aaproperties</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.aaproperties">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.aaproperties.profeat">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../aaproperties/profeat/package-summary.html">org.biojava.nbio.aaproperties.profeat</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.aaproperties.profeat">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.aaproperties.profeat.convertor">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../aaproperties/profeat/convertor/package-summary.html">org.biojava.nbio.aaproperties.profeat.convertor</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.aaproperties.profeat.convertor">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.alignment">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../alignment/package-summary.html">org.biojava.nbio.core.alignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/AccessionID.html#org.biojava.nbio.core.alignment">AccessionID</a></th>
<td class="colLast">
<div class="block">Used in Sequences as the unique identifier.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.sequence">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="package-summary.html">org.biojava.nbio.core.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/AccessionID.html#org.biojava.nbio.core.sequence">AccessionID</a></th>
<td class="colLast">
<div class="block">Used in Sequences as the unique identifier.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/CDSSequence.html#org.biojava.nbio.core.sequence">CDSSequence</a></th>
<td class="colLast">
<div class="block">Represents a exon or coding sequence in a gene.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ChromosomeSequence.html#org.biojava.nbio.core.sequence">ChromosomeSequence</a></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/DataSource.html#org.biojava.nbio.core.sequence">DataSource</a></th>
<td class="colLast">
<div class="block">
 GenBank                           gi|gi-number|gb|accession|locus
 ENA Data Library                  gi|gi-number|emb|accession|locus
 DDBJ, DNA Database of Japan       gi|gi-number|dbj|accession|locus
 NBRF PIR                          pir||entry
 Protein Research Foundation       prf||name
 SWISS-PROT UNIPROT                sp|accession|name
 Brookhaven Protein Data Bank (1)  pdb|entry|chain
 Brookhaven Protein Data Bank (2)  entry:chain|PDBID|CHAIN|SEQUENCE
 Patents                           pat|country|number
 GenInfo Backbone Id               bbs|number
 General database identifier       gnl|database|identifier
 NCBI Reference Sequence           ref|accession|locus
 Local Sequence identifier         lcl|identifier
 </pr></div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/DNASequence.html#org.biojava.nbio.core.sequence">DNASequence</a></th>
<td class="colLast">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/DNASequence.DNAType.html#org.biojava.nbio.core.sequence">DNASequence.DNAType</a></th>
<td class="colLast">
<div class="block">The type of DNA sequence</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ExonSequence.html#org.biojava.nbio.core.sequence">ExonSequence</a></th>
<td class="colLast">
<div class="block">A gene contains a collection of Exon sequences</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/GeneSequence.html#org.biojava.nbio.core.sequence">GeneSequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/IntronSequence.html#org.biojava.nbio.core.sequence">IntronSequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.core.sequence">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/RNASequence.html#org.biojava.nbio.core.sequence">RNASequence</a></th>
<td class="colLast">
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/SequenceOptimizationHints.SequenceCollection.html#org.biojava.nbio.core.sequence">SequenceOptimizationHints.SequenceCollection</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/SequenceOptimizationHints.SequenceUsage.html#org.biojava.nbio.core.sequence">SequenceOptimizationHints.SequenceUsage</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/StartCodonSequence.html#org.biojava.nbio.core.sequence">StartCodonSequence</a></th>
<td class="colLast">
<div class="block">Used to map the start codon feature on a gene</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/StopCodonSequence.html#org.biojava.nbio.core.sequence">StopCodonSequence</a></th>
<td class="colLast">
<div class="block">Used to map the stop codon sequence on a gene</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Strand.html#org.biojava.nbio.core.sequence">Strand</a></th>
<td class="colLast">
<div class="block">Provides a way of representing the strand of a sequence, location
 hit or feature.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/TranscriptSequence.html#org.biojava.nbio.core.sequence">TranscriptSequence</a></th>
<td class="colLast">
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.sequence.io">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="io/package-summary.html">org.biojava.nbio.core.sequence.io</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/DNASequence.html#org.biojava.nbio.core.sequence.io">DNASequence</a></th>
<td class="colLast">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/GeneSequence.html#org.biojava.nbio.core.sequence.io">GeneSequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.core.sequence.io">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/RNASequence.html#org.biojava.nbio.core.sequence.io">RNASequence</a></th>
<td class="colLast">
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.sequence.loader">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/AccessionID.html#org.biojava.nbio.core.sequence.loader">AccessionID</a></th>
<td class="colLast">
<div class="block">Used in Sequences as the unique identifier.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Strand.html#org.biojava.nbio.core.sequence.loader">Strand</a></th>
<td class="colLast">
<div class="block">Provides a way of representing the strand of a sequence, location
 hit or feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.sequence.location">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="location/package-summary.html">org.biojava.nbio.core.sequence.location</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/AccessionID.html#org.biojava.nbio.core.sequence.location">AccessionID</a></th>
<td class="colLast">
<div class="block">Used in Sequences as the unique identifier.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/DataSource.html#org.biojava.nbio.core.sequence.location">DataSource</a></th>
<td class="colLast">
<div class="block">
 GenBank                           gi|gi-number|gb|accession|locus
 ENA Data Library                  gi|gi-number|emb|accession|locus
 DDBJ, DNA Database of Japan       gi|gi-number|dbj|accession|locus
 NBRF PIR                          pir||entry
 Protein Research Foundation       prf||name
 SWISS-PROT UNIPROT                sp|accession|name
 Brookhaven Protein Data Bank (1)  pdb|entry|chain
 Brookhaven Protein Data Bank (2)  entry:chain|PDBID|CHAIN|SEQUENCE
 Patents                           pat|country|number
 GenInfo Backbone Id               bbs|number
 General database identifier       gnl|database|identifier
 NCBI Reference Sequence           ref|accession|locus
 Local Sequence identifier         lcl|identifier
 </pr></div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/Strand.html#org.biojava.nbio.core.sequence.location">Strand</a></th>
<td class="colLast">
<div class="block">Provides a way of representing the strand of a sequence, location
 hit or feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.sequence.location.template">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="location/template/package-summary.html">org.biojava.nbio.core.sequence.location.template</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/AccessionID.html#org.biojava.nbio.core.sequence.location.template">AccessionID</a></th>
<td class="colLast">
<div class="block">Used in Sequences as the unique identifier.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Strand.html#org.biojava.nbio.core.sequence.location.template">Strand</a></th>
<td class="colLast">
<div class="block">Provides a way of representing the strand of a sequence, location
 hit or feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.sequence.storage">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/AccessionID.html#org.biojava.nbio.core.sequence.storage">AccessionID</a></th>
<td class="colLast">
<div class="block">Used in Sequences as the unique identifier.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Strand.html#org.biojava.nbio.core.sequence.storage">Strand</a></th>
<td class="colLast">
<div class="block">Provides a way of representing the strand of a sequence, location
 hit or feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.sequence.template">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="template/package-summary.html">org.biojava.nbio.core.sequence.template</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/AccessionID.html#org.biojava.nbio.core.sequence.template">AccessionID</a></th>
<td class="colLast">
<div class="block">Used in Sequences as the unique identifier.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Strand.html#org.biojava.nbio.core.sequence.template">Strand</a></th>
<td class="colLast">
<div class="block">Provides a way of representing the strand of a sequence, location
 hit or feature.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/TaxonomyID.html#org.biojava.nbio.core.sequence.template">TaxonomyID</a></th>
<td class="colLast">
<div class="block">A sequence can be associated with a species or Taxonomy ID</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.sequence.transcription">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="transcription/package-summary.html">org.biojava.nbio.core.sequence.transcription</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/RNASequence.html#org.biojava.nbio.core.sequence.transcription">RNASequence</a></th>
<td class="colLast">
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.util">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../util/package-summary.html">org.biojava.nbio.core.util</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.core.util">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.genome">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../genome/package-summary.html">org.biojava.nbio.genome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ChromosomeSequence.html#org.biojava.nbio.genome">ChromosomeSequence</a></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/DNASequence.html#org.biojava.nbio.genome">DNASequence</a></th>
<td class="colLast">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/GeneSequence.html#org.biojava.nbio.genome">GeneSequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.genome">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.genome.homology">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../genome/homology/package-summary.html">org.biojava.nbio.genome.homology</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/GeneSequence.html#org.biojava.nbio.genome.homology">GeneSequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.genome.io.fastq">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/DNASequence.html#org.biojava.nbio.genome.io.fastq">DNASequence</a></th>
<td class="colLast">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.genome.parsers.geneid">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../genome/parsers/geneid/package-summary.html">org.biojava.nbio.genome.parsers.geneid</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/DNASequence.html#org.biojava.nbio.genome.parsers.geneid">DNASequence</a></th>
<td class="colLast">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.genome.parsers.geneid">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.genome.parsers.gff">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../genome/parsers/gff/package-summary.html">org.biojava.nbio.genome.parsers.gff</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ChromosomeSequence.html#org.biojava.nbio.genome.parsers.gff">ChromosomeSequence</a></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/DNASequence.html#org.biojava.nbio.genome.parsers.gff">DNASequence</a></th>
<td class="colLast">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.genome.util">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../genome/util/package-summary.html">org.biojava.nbio.genome.util</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/DNASequence.html#org.biojava.nbio.genome.util">DNASequence</a></th>
<td class="colLast">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.genome.util">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.phylo">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../phylo/package-summary.html">org.biojava.nbio.phylo</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/MultipleSequenceAlignment.html#org.biojava.nbio.phylo">MultipleSequenceAlignment</a></th>
<td class="colLast">
<div class="block">Implements a minimal data structure for reading and writing a sequence alignment.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.ronn">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../ronn/package-summary.html">org.biojava.nbio.ronn</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.ronn">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.structure.align.multiple.util">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/MultipleSequenceAlignment.html#org.biojava.nbio.structure.align.multiple.util">MultipleSequenceAlignment</a></th>
<td class="colLast">
<div class="block">Implements a minimal data structure for reading and writing a sequence alignment.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.structure.align.multiple.util">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.structure.cluster">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../structure/cluster/package-summary.html">org.biojava.nbio.structure.cluster</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.structure.cluster">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.structure.io">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../structure/io/package-summary.html">org.biojava.nbio.structure.io</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.structure.io">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.ws.hmmer">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence</a> used by <a href="../../ws/hmmer/package-summary.html">org.biojava.nbio.ws.hmmer</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/ProteinSequence.html#org.biojava.nbio.ws.hmmer">ProteinSequence</a></th>
<td class="colLast">
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2022 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
