<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.1) on Thu Jan 03 22:38:55 CET 2019 -->
<title>FastaStructureParser (biojava 5.2.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2019-01-03">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="FastaStructureParser (biojava 5.2.0 API)";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FastaStructureParser.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">org.biojava.nbio.structure.io</a></div>
<h2 title="Class FastaStructureParser" class="title">Class FastaStructureParser</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.nbio.structure.io.FastaStructureParser</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line.57">FastaStructureParser</a>
extends <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a></pre>
<div class="block">Reads a protein sequence from a fasta file and attempts to match it to a
 3D structure. Any gaps ('-') in the fasta file are preserved as null atoms in
 the output, allowing structural alignments to be read from fasta files.

 <p>Structures are loaded from an AtomCache. For this to work, the accession
 for each protein should be parsed from the fasta header line into a form
 understood by <a href="../align/util/AtomCache.html#getStructure(java.lang.String)"><code>AtomCache.getStructure(String)</code></a>.

 <p>Lowercase letters are sometimes used to specify unaligned residues.
 This information can be preserved by using a CasePreservingSequenceCreator,
 which allows the case of residues to be accessed through the
 <a href="../../core/sequence/template/AbstractSequence.html#getUserCollection()"><code>AbstractSequence.getUserCollection()</code></a> method.</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Spencer Bliven</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.io.File,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,org.biojava.nbio.structure.align.util.AtomCache)">FastaStructureParser</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file,
                    <a href="../../core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,&#8203;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
                    <a href="../../core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
                    <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.io.InputStream,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,org.biojava.nbio.structure.align.util.AtomCache)">FastaStructureParser</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">InputStream</a>&nbsp;is,
                    <a href="../../core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,&#8203;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
                    <a href="../../core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
                    <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(org.biojava.nbio.core.sequence.io.FastaReader,org.biojava.nbio.structure.align.util.AtomCache)">FastaStructureParser</a></span>&#8203;(<a href="../../core/sequence/io/FastaReader.html" title="class in org.biojava.nbio.core.sequence.io">FastaReader</a>&lt;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,&#8203;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;reader,
                    <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAccessions()">getAccessions</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets the protein accessions mapped from the Fasta file.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code><a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[][]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getResidues()">getResidues</a></span>()</code></th>
<td class="colLast">
<div class="block">For each residue in the fasta file, return the ResidueNumber in the
 corresponding structure.</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code><a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSequences()">getSequences</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets the protein sequences read from the Fasta file.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getStructures()">getStructures</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets the protein structures mapped from the Fasta file.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#process()">process</a></span>()</code></th>
<td class="colLast">
<div class="block">Parses the fasta file and loads it into memory.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a></h3>
<code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang" class="externalLink">clone</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang" class="externalLink">equals</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang" class="externalLink">finalize</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang" class="externalLink">getClass</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang" class="externalLink">hashCode</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang" class="externalLink">notify</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang" class="externalLink">notifyAll</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang" class="externalLink">toString</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang" class="externalLink">wait</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang" class="externalLink">wait</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#wait(long,int)" title="class or interface in java.lang" class="externalLink">wait</a></code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;(java.io.InputStream,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,org.biojava.nbio.structure.align.util.AtomCache)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>FastaStructureParser</h4>
<pre>public&nbsp;<a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line.69">FastaStructureParser</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">InputStream</a>&nbsp;is,
                            <a href="../../core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,&#8203;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
                            <a href="../../core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
                            <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</pre>
</li>
</ul>
<a id="&lt;init&gt;(java.io.File,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,org.biojava.nbio.structure.align.util.AtomCache)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>FastaStructureParser</h4>
<pre>public&nbsp;<a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line.78">FastaStructureParser</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file,
                            <a href="../../core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,&#8203;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
                            <a href="../../core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
                            <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)
                     throws <a href="https://docs.oracle.com/javase/8/docs/api/java/io/FileNotFoundException.html?is-external=true" title="class or interface in java.io" class="externalLink">FileNotFoundException</a></pre>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="https://docs.oracle.com/javase/8/docs/api/java/io/FileNotFoundException.html?is-external=true" title="class or interface in java.io" class="externalLink">FileNotFoundException</a></code></dd>
</dl>
</li>
</ul>
<a id="&lt;init&gt;(org.biojava.nbio.core.sequence.io.FastaReader,org.biojava.nbio.structure.align.util.AtomCache)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>FastaStructureParser</h4>
<pre>public&nbsp;<a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line.87">FastaStructureParser</a>&#8203;(<a href="../../core/sequence/io/FastaReader.html" title="class in org.biojava.nbio.core.sequence.io">FastaReader</a>&lt;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,&#8203;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;reader,
                            <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="process()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>process</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line.110">process</a>()
             throws <a href="https://docs.oracle.com/javase/8/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io" class="externalLink">IOException</a>,
                    <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></pre>
<div class="block">Parses the fasta file and loads it into memory.

 Information can be subsequently accessed through
 <a href="#getSequences()"><code>getSequences()</code></a>,
 <a href="#getStructures()"><code>getStructures()</code></a>,
 <a href="#getResidues()"><code>getResidues()</code></a>, and
 <a href="#getAccessions()"><code>getAccessions()</code></a>.</div>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="https://docs.oracle.com/javase/8/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io" class="externalLink">IOException</a></code></dd>
<dd><code><a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd>
</dl>
</li>
</ul>
<a id="getSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequences</h4>
<pre class="methodSignature">public&nbsp;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>[]&nbsp;<a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line.140">getSequences</a>()</pre>
<div class="block">Gets the protein sequences read from the Fasta file.
 Returns null if <a href="#process()"><code>process()</code></a> has not been called.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>An array ProteinSequences from
  parsing the fasta file, or null if process() hasn't been called.</dd>
</dl>
</li>
</ul>
<a id="getStructures()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStructures</h4>
<pre class="methodSignature">public&nbsp;<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>[]&nbsp;<a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line.150">getStructures</a>()</pre>
<div class="block">Gets the protein structures mapped from the Fasta file.
 Returns null if <a href="#process()"><code>process()</code></a> has not been called.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>An array of Structures for each protein
  in the fasta file, or null if process() hasn't been called.</dd>
</dl>
</li>
</ul>
<a id="getResidues()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getResidues</h4>
<pre class="methodSignature">public&nbsp;<a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[][]&nbsp;<a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line.163">getResidues</a>()</pre>
<div class="block">For each residue in the fasta file, return the ResidueNumber in the
 corresponding structure. If the residue cannot be found in the structure,
 that entry will be null. This can happen if that residue was not included
 in the PDB file (eg disordered residues), if the fasta sequence does not
 match the PDB sequence, or if errors occur during the matching process.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A 2D array of ResidueNumbers, or null if process() hasn't been called.</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="StructureSequenceMatcher.html#matchSequenceToStructure(org.biojava.nbio.core.sequence.ProteinSequence,org.biojava.nbio.structure.Structure)"><code>StructureSequenceMatcher.matchSequenceToStructure(ProteinSequence, Structure)</code></a></dd>
</dl>
</li>
</ul>
<a id="getAccessions()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getAccessions</h4>
<pre class="methodSignature">public&nbsp;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;<a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line.173">getAccessions</a>()</pre>
<div class="block">Gets the protein accessions mapped from the Fasta file.
 Returns null if <a href="#process()"><code>process()</code></a> has not been called.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>An array of Structures for each protein
  in the fasta file, or null if process() hasn't been called.</dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FastaStructureParser.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2019 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
