<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> * Created on 01-21-2010</a>
<span class="sourceLineNo">021</span><a id="line.21"> */</a>
<span class="sourceLineNo">022</span><a id="line.22">package org.biojava.nbio.core.sequence.io;</a>
<span class="sourceLineNo">023</span><a id="line.23"></a>
<span class="sourceLineNo">024</span><a id="line.24">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.nbio.core.sequence.ProteinSequence;</a>
<span class="sourceLineNo">026</span><a id="line.26">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</a>
<span class="sourceLineNo">027</span><a id="line.27">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</a>
<span class="sourceLineNo">028</span><a id="line.28">import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface;</a>
<span class="sourceLineNo">030</span><a id="line.30">import org.biojava.nbio.core.sequence.template.Compound;</a>
<span class="sourceLineNo">031</span><a id="line.31">import org.biojava.nbio.core.sequence.template.Sequence;</a>
<span class="sourceLineNo">032</span><a id="line.32">import org.slf4j.Logger;</a>
<span class="sourceLineNo">033</span><a id="line.33">import org.slf4j.LoggerFactory;</a>
<span class="sourceLineNo">034</span><a id="line.34"></a>
<span class="sourceLineNo">035</span><a id="line.35">import java.io.*;</a>
<span class="sourceLineNo">036</span><a id="line.36">import java.util.HashMap;</a>
<span class="sourceLineNo">037</span><a id="line.37">import java.util.LinkedHashMap;</a>
<span class="sourceLineNo">038</span><a id="line.38"></a>
<span class="sourceLineNo">039</span><a id="line.39">/**</a>
<span class="sourceLineNo">040</span><a id="line.40"> * Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be the</a>
<span class="sourceLineNo">041</span><a id="line.41"> * primary class used to read Fasta files</a>
<span class="sourceLineNo">042</span><a id="line.42"> * @author Scooter Willis ;lt;willishf at gmail dot com&amp;gt;</a>
<span class="sourceLineNo">043</span><a id="line.43"> */</a>
<span class="sourceLineNo">044</span><a id="line.44">public class FastaReader&lt;S extends Sequence&lt;?&gt;, C extends Compound&gt; {</a>
<span class="sourceLineNo">045</span><a id="line.45"></a>
<span class="sourceLineNo">046</span><a id="line.46">        private final static Logger logger = LoggerFactory.getLogger(FastaReader.class);</a>
<span class="sourceLineNo">047</span><a id="line.47"></a>
<span class="sourceLineNo">048</span><a id="line.48">        SequenceCreatorInterface&lt;C&gt; sequenceCreator;</a>
<span class="sourceLineNo">049</span><a id="line.49">        SequenceHeaderParserInterface&lt;S,C&gt; headerParser;</a>
<span class="sourceLineNo">050</span><a id="line.50">        BufferedReaderBytesRead br;</a>
<span class="sourceLineNo">051</span><a id="line.51">        InputStreamReader isr;</a>
<span class="sourceLineNo">052</span><a id="line.52">        FileInputStream fi = null;</a>
<span class="sourceLineNo">053</span><a id="line.53">        long fileIndex = 0;</a>
<span class="sourceLineNo">054</span><a id="line.54">        long sequenceIndex = 0;</a>
<span class="sourceLineNo">055</span><a id="line.55">        String line = "";</a>
<span class="sourceLineNo">056</span><a id="line.56">        String header= "";</a>
<span class="sourceLineNo">057</span><a id="line.57"></a>
<span class="sourceLineNo">058</span><a id="line.58">        /**</a>
<span class="sourceLineNo">059</span><a id="line.59">         * If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because we need details about</a>
<span class="sourceLineNo">060</span><a id="line.60">         * local file offsets for quick reads. InputStreams does not give you the name of the stream to access quickly via file seek. A seek in</a>
<span class="sourceLineNo">061</span><a id="line.61">         * an inputstream is forced to read all the data so you don't gain anything.</a>
<span class="sourceLineNo">062</span><a id="line.62">         * @param is inputStream</a>
<span class="sourceLineNo">063</span><a id="line.63">         * @param headerParser</a>
<span class="sourceLineNo">064</span><a id="line.64">         * @param sequenceCreator</a>
<span class="sourceLineNo">065</span><a id="line.65">         */</a>
<span class="sourceLineNo">066</span><a id="line.66">        public FastaReader(InputStream is, SequenceHeaderParserInterface&lt;S,C&gt; headerParser,</a>
<span class="sourceLineNo">067</span><a id="line.67">                                           SequenceCreatorInterface&lt;C&gt; sequenceCreator) {</a>
<span class="sourceLineNo">068</span><a id="line.68">                this.headerParser = headerParser;</a>
<span class="sourceLineNo">069</span><a id="line.69">                isr = new InputStreamReader(is);</a>
<span class="sourceLineNo">070</span><a id="line.70">                this.br = new BufferedReaderBytesRead(isr);</a>
<span class="sourceLineNo">071</span><a id="line.71">                this.sequenceCreator = sequenceCreator;</a>
<span class="sourceLineNo">072</span><a id="line.72">        }</a>
<span class="sourceLineNo">073</span><a id="line.73"></a>
<span class="sourceLineNo">074</span><a id="line.74">        /**</a>
<span class="sourceLineNo">075</span><a id="line.75">         * If you are going to use the FileProxyProteinSequenceCreator then you</a>
<span class="sourceLineNo">076</span><a id="line.76">         * need to use this constructor because we need details about</a>
<span class="sourceLineNo">077</span><a id="line.77">         * the location of the file.</a>
<span class="sourceLineNo">078</span><a id="line.78">         * @param file</a>
<span class="sourceLineNo">079</span><a id="line.79">         * @param headerParser</a>
<span class="sourceLineNo">080</span><a id="line.80">         * @param sequenceCreator</a>
<span class="sourceLineNo">081</span><a id="line.81">         * @throws FileNotFoundException if the file does not exist, is a directory</a>
<span class="sourceLineNo">082</span><a id="line.82">         *      rather than a regular file, or for some other reason cannot be opened</a>
<span class="sourceLineNo">083</span><a id="line.83">         *      for reading.</a>
<span class="sourceLineNo">084</span><a id="line.84">         * @throws SecurityException if a security manager exists and its checkRead</a>
<span class="sourceLineNo">085</span><a id="line.85">         *      method denies read access to the file.</a>
<span class="sourceLineNo">086</span><a id="line.86">         */</a>
<span class="sourceLineNo">087</span><a id="line.87">        public FastaReader(File file, SequenceHeaderParserInterface&lt;S,C&gt; headerParser,</a>
<span class="sourceLineNo">088</span><a id="line.88">                                           SequenceCreatorInterface&lt;C&gt; sequenceCreator) throws FileNotFoundException {</a>
<span class="sourceLineNo">089</span><a id="line.89">                this.headerParser = headerParser;</a>
<span class="sourceLineNo">090</span><a id="line.90">                fi = new FileInputStream(file);</a>
<span class="sourceLineNo">091</span><a id="line.91">                isr = new InputStreamReader(fi);</a>
<span class="sourceLineNo">092</span><a id="line.92">                this.br = new BufferedReaderBytesRead(isr);</a>
<span class="sourceLineNo">093</span><a id="line.93">                this.sequenceCreator = sequenceCreator;</a>
<span class="sourceLineNo">094</span><a id="line.94">        }</a>
<span class="sourceLineNo">095</span><a id="line.95"></a>
<span class="sourceLineNo">096</span><a id="line.96">        /**</a>
<span class="sourceLineNo">097</span><a id="line.97">         * The parsing is done in this method.&lt;br&gt;</a>
<span class="sourceLineNo">098</span><a id="line.98">         * This method tries to process all the available fasta records</a>
<span class="sourceLineNo">099</span><a id="line.99">         * in the File or InputStream, closes the underlying resource,</a>
<span class="sourceLineNo">100</span><a id="line.100">         * and return the results in {@link LinkedHashMap}.&lt;br&gt;</a>
<span class="sourceLineNo">101</span><a id="line.101">         * You don't need to call {@link #close()} after calling this method.</a>
<span class="sourceLineNo">102</span><a id="line.102">         * @see #process(int)</a>
<span class="sourceLineNo">103</span><a id="line.103">         * @return {@link HashMap} containing all the parsed fasta records</a>
<span class="sourceLineNo">104</span><a id="line.104">         * present, starting current fileIndex onwards.</a>
<span class="sourceLineNo">105</span><a id="line.105">         * @throws IOException if an error occurs reading the input file</a>
<span class="sourceLineNo">106</span><a id="line.106">         */</a>
<span class="sourceLineNo">107</span><a id="line.107">        public LinkedHashMap&lt;String,S&gt; process() throws IOException {</a>
<span class="sourceLineNo">108</span><a id="line.108">                LinkedHashMap&lt;String,S&gt; sequences = process(-1);</a>
<span class="sourceLineNo">109</span><a id="line.109">                close();</a>
<span class="sourceLineNo">110</span><a id="line.110"></a>
<span class="sourceLineNo">111</span><a id="line.111">                return sequences;</a>
<span class="sourceLineNo">112</span><a id="line.112">        }</a>
<span class="sourceLineNo">113</span><a id="line.113"></a>
<span class="sourceLineNo">114</span><a id="line.114">        /**</a>
<span class="sourceLineNo">115</span><a id="line.115">         * This method tries to parse maximum &lt;code&gt;max&lt;/code&gt; records from</a>
<span class="sourceLineNo">116</span><a id="line.116">         * the open File or InputStream, and leaves the underlying resource open.&lt;br&gt;</a>
<span class="sourceLineNo">117</span><a id="line.117">         * Subsequent calls to the same method continue parsing the rest of the file.&lt;br&gt;</a>
<span class="sourceLineNo">118</span><a id="line.118">         * This is particularly useful when dealing with very big data files,</a>
<span class="sourceLineNo">119</span><a id="line.119">         * (e.g. NCBI nr database), which can't fit into memory and will take long</a>
<span class="sourceLineNo">120</span><a id="line.120">         * time before the first result is available.&lt;br&gt;</a>
<span class="sourceLineNo">121</span><a id="line.121">         * &lt;b&gt;N.B.&lt;/b&gt;</a>
<span class="sourceLineNo">122</span><a id="line.122">         * &lt;ul&gt;</a>
<span class="sourceLineNo">123</span><a id="line.123">         * &lt;li&gt;This method can't be called after calling its NO-ARGUMENT twin.&lt;/li&gt;</a>
<span class="sourceLineNo">124</span><a id="line.124">         * &lt;li&gt;remember to close the underlying resource when you are done.&lt;/li&gt;</a>
<span class="sourceLineNo">125</span><a id="line.125">         * &lt;/ul&gt;</a>
<span class="sourceLineNo">126</span><a id="line.126">         * @see #process()</a>
<span class="sourceLineNo">127</span><a id="line.127">         * @author Amr AL-Hossary</a>
<span class="sourceLineNo">128</span><a id="line.128">         * @since 3.0.6</a>
<span class="sourceLineNo">129</span><a id="line.129">         * @param max maximum number of records to return, &lt;code&gt;-1&lt;/code&gt; for infinity.</a>
<span class="sourceLineNo">130</span><a id="line.130">         * @return {@link HashMap} containing maximum &lt;code&gt;max&lt;/code&gt; parsed fasta records</a>
<span class="sourceLineNo">131</span><a id="line.131">         * present, starting current fileIndex onwards.</a>
<span class="sourceLineNo">132</span><a id="line.132">         * @throws IOException if an error occurs reading the input file</a>
<span class="sourceLineNo">133</span><a id="line.133">         */</a>
<span class="sourceLineNo">134</span><a id="line.134">        public LinkedHashMap&lt;String,S&gt; process(int max) throws IOException {</a>
<span class="sourceLineNo">135</span><a id="line.135"></a>
<span class="sourceLineNo">136</span><a id="line.136"></a>
<span class="sourceLineNo">137</span><a id="line.137">                String line = "";</a>
<span class="sourceLineNo">138</span><a id="line.138">                if(this.line != null &amp;&amp; this.line.length() &gt; 0){</a>
<span class="sourceLineNo">139</span><a id="line.139">                        line=this.line;</a>
<span class="sourceLineNo">140</span><a id="line.140">                }</a>
<span class="sourceLineNo">141</span><a id="line.141">                String header = "";</a>
<span class="sourceLineNo">142</span><a id="line.142">                if(this.header != null &amp;&amp; this.header.length() &gt; 0){</a>
<span class="sourceLineNo">143</span><a id="line.143">                        header=this.header;</a>
<span class="sourceLineNo">144</span><a id="line.144">                }</a>
<span class="sourceLineNo">145</span><a id="line.145"></a>
<span class="sourceLineNo">146</span><a id="line.146">                StringBuilder sb = new StringBuilder();</a>
<span class="sourceLineNo">147</span><a id="line.147">                int processedSequences=0;</a>
<span class="sourceLineNo">148</span><a id="line.148">                boolean keepGoing = true;</a>
<span class="sourceLineNo">149</span><a id="line.149"></a>
<span class="sourceLineNo">150</span><a id="line.150"></a>
<span class="sourceLineNo">151</span><a id="line.151">                LinkedHashMap&lt;String,S&gt; sequences = new LinkedHashMap&lt;String,S&gt;();</a>
<span class="sourceLineNo">152</span><a id="line.152"></a>
<span class="sourceLineNo">153</span><a id="line.153">                do {</a>
<span class="sourceLineNo">154</span><a id="line.154">                        line = line.trim(); // nice to have but probably not needed</a>
<span class="sourceLineNo">155</span><a id="line.155">                        if (line.length() != 0) {</a>
<span class="sourceLineNo">156</span><a id="line.156">                                if (line.startsWith("&gt;")) {//start of new fasta record</a>
<span class="sourceLineNo">157</span><a id="line.157"></a>
<span class="sourceLineNo">158</span><a id="line.158">                                        if (sb.length() &gt; 0) {</a>
<span class="sourceLineNo">159</span><a id="line.159">                                                //i.e. if there is already a sequence before</a>
<span class="sourceLineNo">160</span><a id="line.160">                                                //logger.info("Sequence index=" + sequenceIndex);</a>
<span class="sourceLineNo">161</span><a id="line.161"></a>
<span class="sourceLineNo">162</span><a id="line.162">                                                try {</a>
<span class="sourceLineNo">163</span><a id="line.163">                                                        @SuppressWarnings("unchecked")</a>
<span class="sourceLineNo">164</span><a id="line.164">                                                        S sequence = (S)sequenceCreator.getSequence(sb.toString(), sequenceIndex);</a>
<span class="sourceLineNo">165</span><a id="line.165">                                                        headerParser.parseHeader(header, sequence);</a>
<span class="sourceLineNo">166</span><a id="line.166">                                                        sequences.put(sequence.getAccession().getID(),sequence);</a>
<span class="sourceLineNo">167</span><a id="line.167">                                                        processedSequences++;</a>
<span class="sourceLineNo">168</span><a id="line.168"></a>
<span class="sourceLineNo">169</span><a id="line.169">                                                } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">170</span><a id="line.170">                                                        logger.warn("Sequence with header '{}' has unrecognised compounds ({}), it will be ignored",</a>
<span class="sourceLineNo">171</span><a id="line.171">                                                                        header, e.getMessage());</a>
<span class="sourceLineNo">172</span><a id="line.172">                                                }</a>
<span class="sourceLineNo">173</span><a id="line.173"></a>
<span class="sourceLineNo">174</span><a id="line.174">                                                sb.setLength(0); //this is faster than allocating new buffers, better memory utilization (same buffer)</a>
<span class="sourceLineNo">175</span><a id="line.175">                                        }</a>
<span class="sourceLineNo">176</span><a id="line.176">                                        header = line.substring(1);</a>
<span class="sourceLineNo">177</span><a id="line.177">                                } else if (line.startsWith(";")) {</a>
<span class="sourceLineNo">178</span><a id="line.178">                                } else {</a>
<span class="sourceLineNo">179</span><a id="line.179">                                        //mark the start of the sequence with the fileIndex before the line was read</a>
<span class="sourceLineNo">180</span><a id="line.180">                                        if(sb.length() == 0){</a>
<span class="sourceLineNo">181</span><a id="line.181">                                                sequenceIndex = fileIndex;</a>
<span class="sourceLineNo">182</span><a id="line.182">                                        }</a>
<span class="sourceLineNo">183</span><a id="line.183">                                        sb.append(line);</a>
<span class="sourceLineNo">184</span><a id="line.184">                                }</a>
<span class="sourceLineNo">185</span><a id="line.185">                        }</a>
<span class="sourceLineNo">186</span><a id="line.186">                        fileIndex = br.getBytesRead();</a>
<span class="sourceLineNo">187</span><a id="line.187"></a>
<span class="sourceLineNo">188</span><a id="line.188">                        line = br.readLine();</a>
<span class="sourceLineNo">189</span><a id="line.189"></a>
<span class="sourceLineNo">190</span><a id="line.190">                        if (line == null) {</a>
<span class="sourceLineNo">191</span><a id="line.191">                                //i.e. EOF</a>
<span class="sourceLineNo">192</span><a id="line.192">                                if ( sb.length() == 0 &amp;&amp; header.length() != 0 ) {</a>
<span class="sourceLineNo">193</span><a id="line.193">                                        logger.warn("Can't parse sequence {}. Got sequence of length 0!", sequenceIndex);</a>
<span class="sourceLineNo">194</span><a id="line.194">                                        logger.warn("header: {}", header);</a>
<span class="sourceLineNo">195</span><a id="line.195">                                        header = null;</a>
<span class="sourceLineNo">196</span><a id="line.196">                                } else if ( sb.length() &gt; 0 ) {</a>
<span class="sourceLineNo">197</span><a id="line.197">                                        //logger.info("Sequence index=" + sequenceIndex + " " + fileIndex );</a>
<span class="sourceLineNo">198</span><a id="line.198">                                        try {</a>
<span class="sourceLineNo">199</span><a id="line.199">                                                @SuppressWarnings("unchecked")</a>
<span class="sourceLineNo">200</span><a id="line.200">                                                S sequence = (S)sequenceCreator.getSequence(sb.toString(), sequenceIndex);</a>
<span class="sourceLineNo">201</span><a id="line.201">                                                headerParser.parseHeader(header, sequence);</a>
<span class="sourceLineNo">202</span><a id="line.202">                                                sequences.put(sequence.getAccession().getID(),sequence);</a>
<span class="sourceLineNo">203</span><a id="line.203">                                                processedSequences++;</a>
<span class="sourceLineNo">204</span><a id="line.204">                                                header = null;</a>
<span class="sourceLineNo">205</span><a id="line.205">                                        } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">206</span><a id="line.206">                                                logger.warn("Sequence with header '{}' has unrecognised compounds ({}), it will be ignored",</a>
<span class="sourceLineNo">207</span><a id="line.207">                                                                header, e.getMessage());</a>
<span class="sourceLineNo">208</span><a id="line.208">                                        }</a>
<span class="sourceLineNo">209</span><a id="line.209">                                }</a>
<span class="sourceLineNo">210</span><a id="line.210">                                keepGoing = false;</a>
<span class="sourceLineNo">211</span><a id="line.211">                        }</a>
<span class="sourceLineNo">212</span><a id="line.212">                        if (max &gt; -1 &amp;&amp; processedSequences&gt;=max) {</a>
<span class="sourceLineNo">213</span><a id="line.213">                                keepGoing=false;</a>
<span class="sourceLineNo">214</span><a id="line.214">                        }</a>
<span class="sourceLineNo">215</span><a id="line.215">                } while (keepGoing);</a>
<span class="sourceLineNo">216</span><a id="line.216"></a>
<span class="sourceLineNo">217</span><a id="line.217">                this.line  = line;</a>
<span class="sourceLineNo">218</span><a id="line.218">                this.header= header;</a>
<span class="sourceLineNo">219</span><a id="line.219"></a>
<span class="sourceLineNo">220</span><a id="line.220">                return max &gt; -1 &amp;&amp; sequences.isEmpty() ? null :  sequences;</a>
<span class="sourceLineNo">221</span><a id="line.221">        }</a>
<span class="sourceLineNo">222</span><a id="line.222"></a>
<span class="sourceLineNo">223</span><a id="line.223">        public void close() throws IOException {</a>
<span class="sourceLineNo">224</span><a id="line.224">                br.close();</a>
<span class="sourceLineNo">225</span><a id="line.225">                isr.close();</a>
<span class="sourceLineNo">226</span><a id="line.226">                //If stream was created from File object then we need to close it</a>
<span class="sourceLineNo">227</span><a id="line.227">                if (fi != null) {</a>
<span class="sourceLineNo">228</span><a id="line.228">                        fi.close();</a>
<span class="sourceLineNo">229</span><a id="line.229">                }</a>
<span class="sourceLineNo">230</span><a id="line.230">                this.line=this.header = null;</a>
<span class="sourceLineNo">231</span><a id="line.231">        }</a>
<span class="sourceLineNo">232</span><a id="line.232"></a>
<span class="sourceLineNo">233</span><a id="line.233">        public static void main(String[] args) {</a>
<span class="sourceLineNo">234</span><a id="line.234">                try {</a>
<span class="sourceLineNo">235</span><a id="line.235">                        String inputFile = "/PF00104_small.fasta";</a>
<span class="sourceLineNo">236</span><a id="line.236">                        InputStream is = FastaReader.class.getResourceAsStream(inputFile);</a>
<span class="sourceLineNo">237</span><a id="line.237"></a>
<span class="sourceLineNo">238</span><a id="line.238"></a>
<span class="sourceLineNo">239</span><a id="line.239">                        if ( is == null)</a>
<span class="sourceLineNo">240</span><a id="line.240">                                System.err.println("Could not get input file " + inputFile);</a>
<span class="sourceLineNo">241</span><a id="line.241">                        FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; fastaReader = new FastaReader&lt;ProteinSequence, AminoAcidCompound&gt;(is, new GenericFastaHeaderParser&lt;ProteinSequence,AminoAcidCompound&gt;(), new ProteinSequenceCreator(AminoAcidCompoundSet.getAminoAcidCompoundSet()));</a>
<span class="sourceLineNo">242</span><a id="line.242">                        LinkedHashMap&lt;String,ProteinSequence&gt; proteinSequences = fastaReader.process();</a>
<span class="sourceLineNo">243</span><a id="line.243">                        is.close();</a>
<span class="sourceLineNo">244</span><a id="line.244"></a>
<span class="sourceLineNo">245</span><a id="line.245"></a>
<span class="sourceLineNo">246</span><a id="line.246">                        //logger.info("Protein Sequences: {}", proteinSequences);</a>
<span class="sourceLineNo">247</span><a id="line.247"></a>
<span class="sourceLineNo">248</span><a id="line.248">                        File file = new File(inputFile);</a>
<span class="sourceLineNo">249</span><a id="line.249">                        FastaReader&lt;ProteinSequence,AminoAcidCompound&gt; fastaProxyReader =</a>
<span class="sourceLineNo">250</span><a id="line.250">                                        new FastaReader&lt;ProteinSequence,AminoAcidCompound&gt;(</a>
<span class="sourceLineNo">251</span><a id="line.251">                                                        file,</a>
<span class="sourceLineNo">252</span><a id="line.252">                                                        new GenericFastaHeaderParser&lt;ProteinSequence,AminoAcidCompound&gt;(),</a>
<span class="sourceLineNo">253</span><a id="line.253">                                                        new FileProxyProteinSequenceCreator(</a>
<span class="sourceLineNo">254</span><a id="line.254">                                                                        file,</a>
<span class="sourceLineNo">255</span><a id="line.255">                                                                        AminoAcidCompoundSet.getAminoAcidCompoundSet(),</a>
<span class="sourceLineNo">256</span><a id="line.256">                                                                        new FastaSequenceParser()</a>
<span class="sourceLineNo">257</span><a id="line.257">                                                        )</a>
<span class="sourceLineNo">258</span><a id="line.258">                                        );</a>
<span class="sourceLineNo">259</span><a id="line.259">                        LinkedHashMap&lt;String,ProteinSequence&gt; proteinProxySequences = fastaProxyReader.process();</a>
<span class="sourceLineNo">260</span><a id="line.260"></a>
<span class="sourceLineNo">261</span><a id="line.261">                        for(String key : proteinProxySequences.keySet()){</a>
<span class="sourceLineNo">262</span><a id="line.262">                                ProteinSequence proteinSequence = proteinProxySequences.get(key);</a>
<span class="sourceLineNo">263</span><a id="line.263">                                logger.info("Protein Proxy Sequence Key: {}", key);</a>
<span class="sourceLineNo">264</span><a id="line.264">//                if(key.equals("Q98SJ1_CHICK/15-61")){</a>
<span class="sourceLineNo">265</span><a id="line.265">//                    int dummy = 1;</a>
<span class="sourceLineNo">266</span><a id="line.266">//                }</a>
<span class="sourceLineNo">267</span><a id="line.267">                                logger.info("Protein Sequence: {}", proteinSequence.toString());</a>
<span class="sourceLineNo">268</span><a id="line.268"></a>
<span class="sourceLineNo">269</span><a id="line.269">                        }</a>
<span class="sourceLineNo">270</span><a id="line.270"></a>
<span class="sourceLineNo">271</span><a id="line.271">                } catch (Exception e) {</a>
<span class="sourceLineNo">272</span><a id="line.272">                        logger.warn("Exception: ", e);</a>
<span class="sourceLineNo">273</span><a id="line.273">                }</a>
<span class="sourceLineNo">274</span><a id="line.274">        }</a>
<span class="sourceLineNo">275</span><a id="line.275">}</a>




























































</pre>
</div>
</main>
</body>
</html>
