<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> * Created on 01-21-2010</a>
<span class="sourceLineNo">021</span><a id="line.21"> */</a>
<span class="sourceLineNo">022</span><a id="line.22">package org.biojava.nbio.core.sequence.io;</a>
<span class="sourceLineNo">023</span><a id="line.23"></a>
<span class="sourceLineNo">024</span><a id="line.24">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.nbio.core.sequence.DNASequence;</a>
<span class="sourceLineNo">026</span><a id="line.26">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</a>
<span class="sourceLineNo">027</span><a id="line.27">import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;</a>
<span class="sourceLineNo">028</span><a id="line.28">import org.biojava.nbio.core.sequence.io.template.SequenceParserInterface;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.biojava.nbio.core.sequence.loader.SequenceFileProxyLoader;</a>
<span class="sourceLineNo">030</span><a id="line.30">import org.biojava.nbio.core.sequence.template.AbstractSequence;</a>
<span class="sourceLineNo">031</span><a id="line.31">import org.biojava.nbio.core.sequence.template.CompoundSet;</a>
<span class="sourceLineNo">032</span><a id="line.32">import org.biojava.nbio.core.sequence.template.ProxySequenceReader;</a>
<span class="sourceLineNo">033</span><a id="line.33"></a>
<span class="sourceLineNo">034</span><a id="line.34">import java.io.File;</a>
<span class="sourceLineNo">035</span><a id="line.35">import java.io.IOException;</a>
<span class="sourceLineNo">036</span><a id="line.36">import java.util.List;</a>
<span class="sourceLineNo">037</span><a id="line.37"></a>
<span class="sourceLineNo">038</span><a id="line.38">/**</a>
<span class="sourceLineNo">039</span><a id="line.39"> * This class is a good example of using the SequenceCreatorInterface where during parsing of the stream</a>
<span class="sourceLineNo">040</span><a id="line.40"> * the sequence and the offset index are passed to create a Protein sequence that will be loaded in lazily.</a>
<span class="sourceLineNo">041</span><a id="line.41"> * This way you can load very large fasta files and store accession id and delay loading the sequence to save</a>
<span class="sourceLineNo">042</span><a id="line.42"> * memory. The index is the file stream offset so when a DNASequence has a call to getSequence() the</a>
<span class="sourceLineNo">043</span><a id="line.43"> * SequenceFileProxyLoader will open the file and offset to the index and retrieve the sequence.</a>
<span class="sourceLineNo">044</span><a id="line.44"> *</a>
<span class="sourceLineNo">045</span><a id="line.45"> * Same approach can be used for genome sequence data stored in a local fasta file, in a database or via http</a>
<span class="sourceLineNo">046</span><a id="line.46"> * interface to a remote server</a>
<span class="sourceLineNo">047</span><a id="line.47"> *</a>
<span class="sourceLineNo">048</span><a id="line.48"> * @author Scooter Willis &lt;willishf at gmail dot com&gt;</a>
<span class="sourceLineNo">049</span><a id="line.49"> */</a>
<span class="sourceLineNo">050</span><a id="line.50">public class FileProxyDNASequenceCreator implements</a>
<span class="sourceLineNo">051</span><a id="line.51">                SequenceCreatorInterface&lt;NucleotideCompound&gt; {</a>
<span class="sourceLineNo">052</span><a id="line.52"></a>
<span class="sourceLineNo">053</span><a id="line.53">        CompoundSet&lt;NucleotideCompound&gt; compoundSet = null;</a>
<span class="sourceLineNo">054</span><a id="line.54">        File file = null;</a>
<span class="sourceLineNo">055</span><a id="line.55">        SequenceParserInterface sequenceParser;</a>
<span class="sourceLineNo">056</span><a id="line.56"></a>
<span class="sourceLineNo">057</span><a id="line.57">        /**</a>
<span class="sourceLineNo">058</span><a id="line.58">         * Need File so that we can store full path name in SequenceFileProxyLoader for Random File access as a quick read</a>
<span class="sourceLineNo">059</span><a id="line.59">         * @param fastaFile</a>
<span class="sourceLineNo">060</span><a id="line.60">         * @param compoundSet</a>
<span class="sourceLineNo">061</span><a id="line.61">         */</a>
<span class="sourceLineNo">062</span><a id="line.62">        public FileProxyDNASequenceCreator(File file,</a>
<span class="sourceLineNo">063</span><a id="line.63">                        CompoundSet&lt;NucleotideCompound&gt; compoundSet,</a>
<span class="sourceLineNo">064</span><a id="line.64">                        SequenceParserInterface sequenceParser) {</a>
<span class="sourceLineNo">065</span><a id="line.65">                this.compoundSet = compoundSet;</a>
<span class="sourceLineNo">066</span><a id="line.66">                this.file = file;</a>
<span class="sourceLineNo">067</span><a id="line.67">                this.sequenceParser = sequenceParser;</a>
<span class="sourceLineNo">068</span><a id="line.68">        }</a>
<span class="sourceLineNo">069</span><a id="line.69"></a>
<span class="sourceLineNo">070</span><a id="line.70">        /**</a>
<span class="sourceLineNo">071</span><a id="line.71">         * Even though we are passing in the sequence we really only care about the length of the sequence and the offset</a>
<span class="sourceLineNo">072</span><a id="line.72">         * index in the fasta file.</a>
<span class="sourceLineNo">073</span><a id="line.73">         * @param sequence</a>
<span class="sourceLineNo">074</span><a id="line.74">         * @param index</a>
<span class="sourceLineNo">075</span><a id="line.75">         * @return</a>
<span class="sourceLineNo">076</span><a id="line.76">         * @throws CompoundNotFoundException</a>
<span class="sourceLineNo">077</span><a id="line.77">         * @throws IOException</a>
<span class="sourceLineNo">078</span><a id="line.78">         */</a>
<span class="sourceLineNo">079</span><a id="line.79">        @Override</a>
<span class="sourceLineNo">080</span><a id="line.80">        public AbstractSequence&lt;NucleotideCompound&gt; getSequence(String sequence, long index ) throws CompoundNotFoundException, IOException {</a>
<span class="sourceLineNo">081</span><a id="line.81">                SequenceFileProxyLoader&lt;NucleotideCompound&gt; sequenceFileProxyLoader = new SequenceFileProxyLoader&lt;NucleotideCompound&gt;(</a>
<span class="sourceLineNo">082</span><a id="line.82">                                file,</a>
<span class="sourceLineNo">083</span><a id="line.83">                                sequenceParser,</a>
<span class="sourceLineNo">084</span><a id="line.84">                                index,</a>
<span class="sourceLineNo">085</span><a id="line.85">                                sequence.length(),</a>
<span class="sourceLineNo">086</span><a id="line.86">                                compoundSet);</a>
<span class="sourceLineNo">087</span><a id="line.87">                return new DNASequence(sequenceFileProxyLoader, compoundSet);</a>
<span class="sourceLineNo">088</span><a id="line.88">        }</a>
<span class="sourceLineNo">089</span><a id="line.89"></a>
<span class="sourceLineNo">090</span><a id="line.90">        /**</a>
<span class="sourceLineNo">091</span><a id="line.91">         * Should be able to extend the same concept to a remote URL call or database connection. Not supported yet</a>
<span class="sourceLineNo">092</span><a id="line.92">         * @param proxyLoader</a>
<span class="sourceLineNo">093</span><a id="line.93">         * @param index</a>
<span class="sourceLineNo">094</span><a id="line.94">         * @return</a>
<span class="sourceLineNo">095</span><a id="line.95">         */</a>
<span class="sourceLineNo">096</span><a id="line.96">        @Override</a>
<span class="sourceLineNo">097</span><a id="line.97">        public AbstractSequence&lt;NucleotideCompound&gt; getSequence(</a>
<span class="sourceLineNo">098</span><a id="line.98">                        ProxySequenceReader&lt;NucleotideCompound&gt; proxyLoader, long index) {</a>
<span class="sourceLineNo">099</span><a id="line.99">                throw new UnsupportedOperationException("Not supported yet.");</a>
<span class="sourceLineNo">100</span><a id="line.100">        }</a>
<span class="sourceLineNo">101</span><a id="line.101"></a>
<span class="sourceLineNo">102</span><a id="line.102">        /**</a>
<span class="sourceLineNo">103</span><a id="line.103">         * Not sure of use case and currently not supported</a>
<span class="sourceLineNo">104</span><a id="line.104">         * @param list</a>
<span class="sourceLineNo">105</span><a id="line.105">         * @return</a>
<span class="sourceLineNo">106</span><a id="line.106">         */</a>
<span class="sourceLineNo">107</span><a id="line.107">        @Override</a>
<span class="sourceLineNo">108</span><a id="line.108">        public AbstractSequence&lt;NucleotideCompound&gt; getSequence(</a>
<span class="sourceLineNo">109</span><a id="line.109">                        List&lt;NucleotideCompound&gt; list) {</a>
<span class="sourceLineNo">110</span><a id="line.110">                throw new UnsupportedOperationException("Not supported yet.");</a>
<span class="sourceLineNo">111</span><a id="line.111">        }</a>
<span class="sourceLineNo">112</span><a id="line.112">}</a>




























































</pre>
</div>
</main>
</body>
</html>
