<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * @author Scooter Willis ;lt;willishf at gmail dot com&amp;gt;</a>
<span class="sourceLineNo">015</span><a id="line.15"> * @author Karl Nicholas &lt;github:karlnicholas&gt;</a>
<span class="sourceLineNo">016</span><a id="line.16"> * @author Paolo Pavan</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">019</span><a id="line.19"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">020</span><a id="line.20"> * at:</a>
<span class="sourceLineNo">021</span><a id="line.21"> *</a>
<span class="sourceLineNo">022</span><a id="line.22"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">023</span><a id="line.23"> *</a>
<span class="sourceLineNo">024</span><a id="line.24"> * Created on 01-21-2010</a>
<span class="sourceLineNo">025</span><a id="line.25"> */</a>
<span class="sourceLineNo">026</span><a id="line.26">package org.biojava.nbio.core.sequence.io;</a>
<span class="sourceLineNo">027</span><a id="line.27"></a>
<span class="sourceLineNo">028</span><a id="line.28">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.biojava.nbio.core.sequence.DNASequence;</a>
<span class="sourceLineNo">030</span><a id="line.30">import org.biojava.nbio.core.sequence.DataSource;</a>
<span class="sourceLineNo">031</span><a id="line.31">import org.biojava.nbio.core.sequence.ProteinSequence;</a>
<span class="sourceLineNo">032</span><a id="line.32">import org.biojava.nbio.core.sequence.TaxonomyID;</a>
<span class="sourceLineNo">033</span><a id="line.33">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</a>
<span class="sourceLineNo">034</span><a id="line.34">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</a>
<span class="sourceLineNo">035</span><a id="line.35">import org.biojava.nbio.core.sequence.compound.DNACompoundSet;</a>
<span class="sourceLineNo">036</span><a id="line.36">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</a>
<span class="sourceLineNo">037</span><a id="line.37">import org.biojava.nbio.core.sequence.features.AbstractFeature;</a>
<span class="sourceLineNo">038</span><a id="line.38">import org.biojava.nbio.core.sequence.features.DBReferenceInfo;</a>
<span class="sourceLineNo">039</span><a id="line.39">import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;</a>
<span class="sourceLineNo">040</span><a id="line.40">import org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface;</a>
<span class="sourceLineNo">041</span><a id="line.41">import org.biojava.nbio.core.sequence.template.AbstractSequence;</a>
<span class="sourceLineNo">042</span><a id="line.42">import org.biojava.nbio.core.sequence.template.Compound;</a>
<span class="sourceLineNo">043</span><a id="line.43">import org.slf4j.Logger;</a>
<span class="sourceLineNo">044</span><a id="line.44">import org.slf4j.LoggerFactory;</a>
<span class="sourceLineNo">045</span><a id="line.45"></a>
<span class="sourceLineNo">046</span><a id="line.46">import java.io.*;</a>
<span class="sourceLineNo">047</span><a id="line.47">import java.util.ArrayList;</a>
<span class="sourceLineNo">048</span><a id="line.48">import java.util.HashMap;</a>
<span class="sourceLineNo">049</span><a id="line.49">import java.util.LinkedHashMap;</a>
<span class="sourceLineNo">050</span><a id="line.50"></a>
<span class="sourceLineNo">051</span><a id="line.51">/**</a>
<span class="sourceLineNo">052</span><a id="line.52"> * Use GenbankReaderHelper as an example of how to use this class where GenbankReaderHelper should be the</a>
<span class="sourceLineNo">053</span><a id="line.53"> * primary class used to read Genbank files</a>
<span class="sourceLineNo">054</span><a id="line.54"> *</a>
<span class="sourceLineNo">055</span><a id="line.55"> */</a>
<span class="sourceLineNo">056</span><a id="line.56">public class GenbankReader&lt;S extends AbstractSequence&lt;C&gt;, C extends Compound&gt; {</a>
<span class="sourceLineNo">057</span><a id="line.57"></a>
<span class="sourceLineNo">058</span><a id="line.58">        private SequenceCreatorInterface&lt;C&gt; sequenceCreator;</a>
<span class="sourceLineNo">059</span><a id="line.59">        private GenbankSequenceParser&lt;S,C&gt; genbankParser;</a>
<span class="sourceLineNo">060</span><a id="line.60">        private BufferedReader bufferedReader;</a>
<span class="sourceLineNo">061</span><a id="line.61">        private boolean closed;</a>
<span class="sourceLineNo">062</span><a id="line.62">        private final Logger logger = LoggerFactory.getLogger(this.getClass());</a>
<span class="sourceLineNo">063</span><a id="line.63"></a>
<span class="sourceLineNo">064</span><a id="line.64">        public boolean isClosed() {</a>
<span class="sourceLineNo">065</span><a id="line.65">                return closed;</a>
<span class="sourceLineNo">066</span><a id="line.66">        }</a>
<span class="sourceLineNo">067</span><a id="line.67"></a>
<span class="sourceLineNo">068</span><a id="line.68">        /**</a>
<span class="sourceLineNo">069</span><a id="line.69">         * If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because we need details about</a>
<span class="sourceLineNo">070</span><a id="line.70">         * local file offsets for quick reads. InputStreams does not give you the name of the stream to access quickly via file seek. A seek in</a>
<span class="sourceLineNo">071</span><a id="line.71">         * an inputstream is forced to read all the data so you don't gain anything.</a>
<span class="sourceLineNo">072</span><a id="line.72">         * @param is</a>
<span class="sourceLineNo">073</span><a id="line.73">         * @param headerParser</a>
<span class="sourceLineNo">074</span><a id="line.74">         * @param sequenceCreator</a>
<span class="sourceLineNo">075</span><a id="line.75">         */</a>
<span class="sourceLineNo">076</span><a id="line.76">        public GenbankReader(final InputStream is, final SequenceHeaderParserInterface&lt;S,C&gt; headerParser,</a>
<span class="sourceLineNo">077</span><a id="line.77">                                                 final SequenceCreatorInterface&lt;C&gt; sequenceCreator) {</a>
<span class="sourceLineNo">078</span><a id="line.78">                this.sequenceCreator = sequenceCreator;</a>
<span class="sourceLineNo">079</span><a id="line.79">                bufferedReader = new BufferedReader(new InputStreamReader(is));</a>
<span class="sourceLineNo">080</span><a id="line.80">                genbankParser = new GenbankSequenceParser&lt;&gt;();</a>
<span class="sourceLineNo">081</span><a id="line.81">                closed = false;</a>
<span class="sourceLineNo">082</span><a id="line.82">        }</a>
<span class="sourceLineNo">083</span><a id="line.83"></a>
<span class="sourceLineNo">084</span><a id="line.84">        /**</a>
<span class="sourceLineNo">085</span><a id="line.85">         * If you are going to use the FileProxyProteinSequenceCreator then you</a>
<span class="sourceLineNo">086</span><a id="line.86">         * need to use this constructor because we need details about</a>
<span class="sourceLineNo">087</span><a id="line.87">         * the location of the file.</a>
<span class="sourceLineNo">088</span><a id="line.88">         * @param file</a>
<span class="sourceLineNo">089</span><a id="line.89">         * @param headerParser</a>
<span class="sourceLineNo">090</span><a id="line.90">         * @param sequenceCreator</a>
<span class="sourceLineNo">091</span><a id="line.91">         * @throws FileNotFoundException if the file does not exist, is a directory</a>
<span class="sourceLineNo">092</span><a id="line.92">         *      rather than a regular file, or for some other reason cannot be opened</a>
<span class="sourceLineNo">093</span><a id="line.93">         *      for reading.</a>
<span class="sourceLineNo">094</span><a id="line.94">         * @throws SecurityException if a security manager exists and its checkRead</a>
<span class="sourceLineNo">095</span><a id="line.95">         *      method denies read access to the file.</a>
<span class="sourceLineNo">096</span><a id="line.96">         */</a>
<span class="sourceLineNo">097</span><a id="line.97">        public GenbankReader(</a>
<span class="sourceLineNo">098</span><a id="line.98">                        final File file,</a>
<span class="sourceLineNo">099</span><a id="line.99">                        final SequenceHeaderParserInterface&lt;S,C&gt; headerParser,</a>
<span class="sourceLineNo">100</span><a id="line.100">                        final SequenceCreatorInterface&lt;C&gt; sequenceCreator</a>
<span class="sourceLineNo">101</span><a id="line.101">                        ) throws FileNotFoundException {</a>
<span class="sourceLineNo">102</span><a id="line.102"></a>
<span class="sourceLineNo">103</span><a id="line.103">                this.bufferedReader = new BufferedReader(new FileReader(file));</a>
<span class="sourceLineNo">104</span><a id="line.104">                this.sequenceCreator = sequenceCreator;</a>
<span class="sourceLineNo">105</span><a id="line.105">                genbankParser = new GenbankSequenceParser&lt;&gt;();</a>
<span class="sourceLineNo">106</span><a id="line.106">        }</a>
<span class="sourceLineNo">107</span><a id="line.107"></a>
<span class="sourceLineNo">108</span><a id="line.108">        /**</a>
<span class="sourceLineNo">109</span><a id="line.109">         * The parsing is done in this method.&lt;br&gt;</a>
<span class="sourceLineNo">110</span><a id="line.110">         * This method tries to process all the available Genbank records</a>
<span class="sourceLineNo">111</span><a id="line.111">         * in the File or InputStream, closes the underlying resource,</a>
<span class="sourceLineNo">112</span><a id="line.112">         * and return the results in {@link LinkedHashMap}.&lt;br&gt;</a>
<span class="sourceLineNo">113</span><a id="line.113">         * You don't need to call {@link #close()} after calling this method.</a>
<span class="sourceLineNo">114</span><a id="line.114">         * @see #process(int)</a>
<span class="sourceLineNo">115</span><a id="line.115">         * @return {@link HashMap} containing all the parsed Genbank records</a>
<span class="sourceLineNo">116</span><a id="line.116">         * present, starting current fileIndex onwards.</a>
<span class="sourceLineNo">117</span><a id="line.117">         * @throws IOException</a>
<span class="sourceLineNo">118</span><a id="line.118">         * @throws CompoundNotFoundException</a>
<span class="sourceLineNo">119</span><a id="line.119">         */</a>
<span class="sourceLineNo">120</span><a id="line.120">        public LinkedHashMap&lt;String,S&gt; process() throws IOException, CompoundNotFoundException {</a>
<span class="sourceLineNo">121</span><a id="line.121">                return process(-1);</a>
<span class="sourceLineNo">122</span><a id="line.122">        }</a>
<span class="sourceLineNo">123</span><a id="line.123"></a>
<span class="sourceLineNo">124</span><a id="line.124">        /**</a>
<span class="sourceLineNo">125</span><a id="line.125">         * This method tries to parse maximum &lt;code&gt;max&lt;/code&gt; records from</a>
<span class="sourceLineNo">126</span><a id="line.126">         * the open File or InputStream, and leaves the underlying resource open.&lt;br&gt;</a>
<span class="sourceLineNo">127</span><a id="line.127">         *</a>
<span class="sourceLineNo">128</span><a id="line.128">         * Subsequent calls to the same method continue parsing the rest of the file.&lt;br&gt;</a>
<span class="sourceLineNo">129</span><a id="line.129">         * This is particularly useful when dealing with very big data files,</a>
<span class="sourceLineNo">130</span><a id="line.130">         * (e.g. NCBI nr database), which can't fit into memory and will take long</a>
<span class="sourceLineNo">131</span><a id="line.131">         * time before the first result is available.&lt;br&gt;</a>
<span class="sourceLineNo">132</span><a id="line.132">         * &lt;b&gt;N.B.&lt;/b&gt;</a>
<span class="sourceLineNo">133</span><a id="line.133">         * &lt;ul&gt;</a>
<span class="sourceLineNo">134</span><a id="line.134">         * &lt;li&gt;This method can't be called after calling its NO-ARGUMENT twin.&lt;/li&gt;</a>
<span class="sourceLineNo">135</span><a id="line.135">         * &lt;li&gt;remember to close the underlying resource when you are done.&lt;/li&gt;</a>
<span class="sourceLineNo">136</span><a id="line.136">         * &lt;/ul&gt;</a>
<span class="sourceLineNo">137</span><a id="line.137">         * @see #process()</a>
<span class="sourceLineNo">138</span><a id="line.138">         * @author Amr AL-Hossary</a>
<span class="sourceLineNo">139</span><a id="line.139">         * @since 3.0.6</a>
<span class="sourceLineNo">140</span><a id="line.140">         * @param max maximum number of records to return, &lt;code&gt;-1&lt;/code&gt; for infinity.</a>
<span class="sourceLineNo">141</span><a id="line.141">         * @return {@link HashMap} containing maximum &lt;code&gt;max&lt;/code&gt; parsed Genbank records</a>
<span class="sourceLineNo">142</span><a id="line.142">         * present, starting current fileIndex onwards.</a>
<span class="sourceLineNo">143</span><a id="line.143">         * @throws IOException</a>
<span class="sourceLineNo">144</span><a id="line.144">         * @throws CompoundNotFoundException</a>
<span class="sourceLineNo">145</span><a id="line.145">         */</a>
<span class="sourceLineNo">146</span><a id="line.146">        public LinkedHashMap&lt;String,S&gt; process(final int max) throws IOException, CompoundNotFoundException {</a>
<span class="sourceLineNo">147</span><a id="line.147">                LinkedHashMap&lt;String,S&gt; sequences = new LinkedHashMap&lt;&gt;();</a>
<span class="sourceLineNo">148</span><a id="line.148">                @SuppressWarnings("unchecked")</a>
<span class="sourceLineNo">149</span><a id="line.149">                int i=0;</a>
<span class="sourceLineNo">150</span><a id="line.150">                while(true) {</a>
<span class="sourceLineNo">151</span><a id="line.151">                        if(max&gt;0 &amp;&amp; i&gt;=max) break;</a>
<span class="sourceLineNo">152</span><a id="line.152">                        i++;</a>
<span class="sourceLineNo">153</span><a id="line.153">                        String seqString = genbankParser.getSequence(bufferedReader, 0);</a>
<span class="sourceLineNo">154</span><a id="line.154">                        //reached end of file?</a>
<span class="sourceLineNo">155</span><a id="line.155">                        if(seqString==null) break;</a>
<span class="sourceLineNo">156</span><a id="line.156">                        @SuppressWarnings("unchecked")</a>
<span class="sourceLineNo">157</span><a id="line.157">                        S sequence = (S) sequenceCreator.getSequence(seqString, 0);</a>
<span class="sourceLineNo">158</span><a id="line.158">                        genbankParser.getSequenceHeaderParser().parseHeader(genbankParser.getHeader(), sequence);</a>
<span class="sourceLineNo">159</span><a id="line.159"></a>
<span class="sourceLineNo">160</span><a id="line.160">                        // add features to new sequence</a>
<span class="sourceLineNo">161</span><a id="line.161">                        for (String k: genbankParser.getFeatures().keySet()){</a>
<span class="sourceLineNo">162</span><a id="line.162">                                for (AbstractFeature f: genbankParser.getFeatures(k)){</a>
<span class="sourceLineNo">163</span><a id="line.163">                                        //f.getLocations().setSequence(sequence);  // can't set proper sequence source to features. It is actually needed? Don't think so...</a>
<span class="sourceLineNo">164</span><a id="line.164">                                        sequence.addFeature(f);</a>
<span class="sourceLineNo">165</span><a id="line.165">                                }</a>
<span class="sourceLineNo">166</span><a id="line.166">                        }</a>
<span class="sourceLineNo">167</span><a id="line.167"></a>
<span class="sourceLineNo">168</span><a id="line.168">                        // add taxonomy ID to new sequence</a>
<span class="sourceLineNo">169</span><a id="line.169">                        ArrayList&lt;DBReferenceInfo&gt; dbQualifier = genbankParser.getDatabaseReferences().get("db_xref");</a>
<span class="sourceLineNo">170</span><a id="line.170">                        if (dbQualifier != null){</a>
<span class="sourceLineNo">171</span><a id="line.171">                                DBReferenceInfo q = dbQualifier.get(0);</a>
<span class="sourceLineNo">172</span><a id="line.172">                                sequence.setTaxonomy(new TaxonomyID(q.getDatabase()+":"+q.getId(), DataSource.GENBANK));</a>
<span class="sourceLineNo">173</span><a id="line.173">                        }</a>
<span class="sourceLineNo">174</span><a id="line.174"></a>
<span class="sourceLineNo">175</span><a id="line.175">                        sequences.put(sequence.getAccession().getID(), sequence);</a>
<span class="sourceLineNo">176</span><a id="line.176">                }</a>
<span class="sourceLineNo">177</span><a id="line.177"></a>
<span class="sourceLineNo">178</span><a id="line.178">                if (max &lt; 0) {</a>
<span class="sourceLineNo">179</span><a id="line.179">                        close();</a>
<span class="sourceLineNo">180</span><a id="line.180">                }</a>
<span class="sourceLineNo">181</span><a id="line.181"></a>
<span class="sourceLineNo">182</span><a id="line.182">                return sequences;</a>
<span class="sourceLineNo">183</span><a id="line.183">        }</a>
<span class="sourceLineNo">184</span><a id="line.184"></a>
<span class="sourceLineNo">185</span><a id="line.185">        public void close() {</a>
<span class="sourceLineNo">186</span><a id="line.186">                try {</a>
<span class="sourceLineNo">187</span><a id="line.187">                        bufferedReader.close();</a>
<span class="sourceLineNo">188</span><a id="line.188">                        this.closed = true;</a>
<span class="sourceLineNo">189</span><a id="line.189">                } catch (IOException e) {</a>
<span class="sourceLineNo">190</span><a id="line.190">                        logger.error("Couldn't close the reader. {}", e.getMessage());</a>
<span class="sourceLineNo">191</span><a id="line.191">                        this.closed = false;</a>
<span class="sourceLineNo">192</span><a id="line.192">                }</a>
<span class="sourceLineNo">193</span><a id="line.193">        }</a>
<span class="sourceLineNo">194</span><a id="line.194"></a>
<span class="sourceLineNo">195</span><a id="line.195">        public static void main(String[] args) throws Exception {</a>
<span class="sourceLineNo">196</span><a id="line.196">                String proteinFile = "src/test/resources/BondFeature.gb";</a>
<span class="sourceLineNo">197</span><a id="line.197">                FileInputStream is = new FileInputStream(proteinFile);</a>
<span class="sourceLineNo">198</span><a id="line.198"></a>
<span class="sourceLineNo">199</span><a id="line.199">                GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt; proteinReader = new GenbankReader&lt;&gt;(is, new GenericGenbankHeaderParser&lt;&gt;(), new ProteinSequenceCreator(AminoAcidCompoundSet.getAminoAcidCompoundSet()));</a>
<span class="sourceLineNo">200</span><a id="line.200">                LinkedHashMap&lt;String,ProteinSequence&gt; proteinSequences = proteinReader.process();</a>
<span class="sourceLineNo">201</span><a id="line.201">                System.out.println(proteinSequences);</a>
<span class="sourceLineNo">202</span><a id="line.202"></a>
<span class="sourceLineNo">203</span><a id="line.203">                String inputFile = "src/test/resources/NM_000266.gb";</a>
<span class="sourceLineNo">204</span><a id="line.204">                is = new FileInputStream(inputFile);</a>
<span class="sourceLineNo">205</span><a id="line.205">                GenbankReader&lt;DNASequence, NucleotideCompound&gt; dnaReader = new GenbankReader&lt;&gt;(is, new GenericGenbankHeaderParser&lt;&gt;(), new DNASequenceCreator(DNACompoundSet.getDNACompoundSet()));</a>
<span class="sourceLineNo">206</span><a id="line.206">                LinkedHashMap&lt;String,DNASequence&gt; dnaSequences = dnaReader.process();</a>
<span class="sourceLineNo">207</span><a id="line.207">                System.out.println(dnaSequences);</a>
<span class="sourceLineNo">208</span><a id="line.208"></a>
<span class="sourceLineNo">209</span><a id="line.209">                String crazyFile = "src/test/resources/CraftedFeature.gb";</a>
<span class="sourceLineNo">210</span><a id="line.210">                is = new FileInputStream(crazyFile);</a>
<span class="sourceLineNo">211</span><a id="line.211">                GenbankReader&lt;DNASequence, NucleotideCompound&gt; crazyReader = new GenbankReader&lt;&gt;(is, new GenericGenbankHeaderParser&lt;&gt;(), new DNASequenceCreator(DNACompoundSet.getDNACompoundSet()));</a>
<span class="sourceLineNo">212</span><a id="line.212">                LinkedHashMap&lt;String,DNASequence&gt; crazyAnnotatedSequences = crazyReader.process();</a>
<span class="sourceLineNo">213</span><a id="line.213"></a>
<span class="sourceLineNo">214</span><a id="line.214">                is.close();</a>
<span class="sourceLineNo">215</span><a id="line.215">                System.out.println(crazyAnnotatedSequences);</a>
<span class="sourceLineNo">216</span><a id="line.216">        }</a>
<span class="sourceLineNo">217</span><a id="line.217"></a>
<span class="sourceLineNo">218</span><a id="line.218">}</a>
<span class="sourceLineNo">219</span><a id="line.219"></a>




























































</pre>
</div>
</main>
</body>
</html>
