<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> * Created on 25.04.2004<a name="line.20"></a>
<span class="sourceLineNo">021</span> * @author Andreas Prlic<a name="line.21"></a>
<span class="sourceLineNo">022</span> *<a name="line.22"></a>
<span class="sourceLineNo">023</span> */<a name="line.23"></a>
<span class="sourceLineNo">024</span>package org.biojava.nbio.structure;<a name="line.24"></a>
<span class="sourceLineNo">025</span><a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.core.sequence.template.Sequence;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.nbio.structure.io.FileParsingParameters;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.nbio.structure.io.mmcif.model.ChemComp;<a name="line.28"></a>
<span class="sourceLineNo">029</span><a name="line.29"></a>
<span class="sourceLineNo">030</span>import java.io.Serializable;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import java.util.List;<a name="line.31"></a>
<span class="sourceLineNo">032</span><a name="line.32"></a>
<span class="sourceLineNo">033</span>/**<a name="line.33"></a>
<span class="sourceLineNo">034</span> * &lt;p&gt;<a name="line.34"></a>
<span class="sourceLineNo">035</span> * Defines the interface for a Chain. A Chain corresponds to a Chain in a PDB file.<a name="line.35"></a>
<span class="sourceLineNo">036</span> * A chain consists out of a list of {@link Group} objects. A Group can either be<a name="line.36"></a>
<span class="sourceLineNo">037</span> * an {@link AminoAcid}, {@link HetatomImpl Hetatom} or {@link NucleotideImpl Nucleotide}.<a name="line.37"></a>
<span class="sourceLineNo">038</span> * &lt;/p&gt;<a name="line.38"></a>
<span class="sourceLineNo">039</span> *<a name="line.39"></a>
<span class="sourceLineNo">040</span> * &lt;p&gt;<a name="line.40"></a>
<span class="sourceLineNo">041</span> * The BioJava API provides access to both the ATOM and SEQRES records in a PDB file.<a name="line.41"></a>
<span class="sourceLineNo">042</span> * During parsing of a PDB file it aligns the ATOM and SEQRES groups and joins them.<a name="line.42"></a>
<span class="sourceLineNo">043</span> * The SEQRES sequence can be accessed via  {@link #getSeqResGroups()} and the<a name="line.43"></a>
<span class="sourceLineNo">044</span> * ATOM groups via {@link #getAtomGroups()}. Groups that have been observed<a name="line.44"></a>
<span class="sourceLineNo">045</span> * (i.e. they are in the ATOM records) can be detected by {@link Group}.has3D()<a name="line.45"></a>
<span class="sourceLineNo">046</span> *  &lt;/p&gt;<a name="line.46"></a>
<span class="sourceLineNo">047</span> *<a name="line.47"></a>
<span class="sourceLineNo">048</span> * @author Andreas Prlic<a name="line.48"></a>
<span class="sourceLineNo">049</span> * @version %I% %G%<a name="line.49"></a>
<span class="sourceLineNo">050</span> * @since 1.4<a name="line.50"></a>
<span class="sourceLineNo">051</span> */<a name="line.51"></a>
<span class="sourceLineNo">052</span>public interface Chain extends Serializable {<a name="line.52"></a>
<span class="sourceLineNo">053</span><a name="line.53"></a>
<span class="sourceLineNo">054</span>        /** returns an identical copy of this Chain.<a name="line.54"></a>
<span class="sourceLineNo">055</span>         * @return  an identical copy of this Chain<a name="line.55"></a>
<span class="sourceLineNo">056</span>         */<a name="line.56"></a>
<span class="sourceLineNo">057</span>        Object clone();<a name="line.57"></a>
<span class="sourceLineNo">058</span><a name="line.58"></a>
<span class="sourceLineNo">059</span>        /** add a group to the list of ATOM record group of this chain.<a name="line.59"></a>
<span class="sourceLineNo">060</span>         * To add SEQRES records a more complex alignment between ATOM and SEQRES residues<a name="line.60"></a>
<span class="sourceLineNo">061</span>         * is required, please see SeqRes2AtomAligner for more details on that.<a name="line.61"></a>
<span class="sourceLineNo">062</span>         * @param group  a Group object<a name="line.62"></a>
<span class="sourceLineNo">063</span>         */<a name="line.63"></a>
<span class="sourceLineNo">064</span>        void addGroup(Group group);<a name="line.64"></a>
<span class="sourceLineNo">065</span><a name="line.65"></a>
<span class="sourceLineNo">066</span>        /** Get the 'private' asymId (internal chain IDs in mmCif) for this chain.<a name="line.66"></a>
<span class="sourceLineNo">067</span>         *<a name="line.67"></a>
<span class="sourceLineNo">068</span>         * @return the asymId<a name="line.68"></a>
<span class="sourceLineNo">069</span>         * @see #setId(String)<a name="line.69"></a>
<span class="sourceLineNo">070</span>         * @see #getName()<a name="line.70"></a>
<span class="sourceLineNo">071</span>         */<a name="line.71"></a>
<span class="sourceLineNo">072</span>        String getId() ;<a name="line.72"></a>
<span class="sourceLineNo">073</span><a name="line.73"></a>
<span class="sourceLineNo">074</span><a name="line.74"></a>
<span class="sourceLineNo">075</span>        /**<a name="line.75"></a>
<span class="sourceLineNo">076</span>         * Set the 'private' asymId (internal chain IDs in mmCif) for this chain.<a name="line.76"></a>
<span class="sourceLineNo">077</span>         *<a name="line.77"></a>
<span class="sourceLineNo">078</span>         * @param asymId the internal chain Id<a name="line.78"></a>
<span class="sourceLineNo">079</span>         */<a name="line.79"></a>
<span class="sourceLineNo">080</span>        void setId(String asymId) ;<a name="line.80"></a>
<span class="sourceLineNo">081</span><a name="line.81"></a>
<span class="sourceLineNo">082</span><a name="line.82"></a>
<span class="sourceLineNo">083</span>        /**<a name="line.83"></a>
<span class="sourceLineNo">084</span>         * Set the 'public' authId (chain ID in PDB file)<a name="line.84"></a>
<span class="sourceLineNo">085</span>         *<a name="line.85"></a>
<span class="sourceLineNo">086</span>         * @param authId the 'public' authId (chain ID in PDB file)<a name="line.86"></a>
<span class="sourceLineNo">087</span>         * @see #getId()<a name="line.87"></a>
<span class="sourceLineNo">088</span>         */<a name="line.88"></a>
<span class="sourceLineNo">089</span>        void setName(String authId);<a name="line.89"></a>
<span class="sourceLineNo">090</span><a name="line.90"></a>
<span class="sourceLineNo">091</span>        /**<a name="line.91"></a>
<span class="sourceLineNo">092</span>         * Get the 'public' authId (chain ID in PDB file)<a name="line.92"></a>
<span class="sourceLineNo">093</span>         *<a name="line.93"></a>
<span class="sourceLineNo">094</span>         * @return the authId for this chain.<a name="line.94"></a>
<span class="sourceLineNo">095</span>         * @see #getId()<a name="line.95"></a>
<span class="sourceLineNo">096</span>         */<a name="line.96"></a>
<span class="sourceLineNo">097</span>        String getName();<a name="line.97"></a>
<span class="sourceLineNo">098</span><a name="line.98"></a>
<span class="sourceLineNo">099</span><a name="line.99"></a>
<span class="sourceLineNo">100</span>        /**<a name="line.100"></a>
<span class="sourceLineNo">101</span>         * Return the Group at given position,<a name="line.101"></a>
<span class="sourceLineNo">102</span>         * from within Groups with observed density in the chain, i.e.<a name="line.102"></a>
<span class="sourceLineNo">103</span>         * those with coordinates in ATOM and HETATMS (including waters) records.<a name="line.103"></a>
<span class="sourceLineNo">104</span>         * @param position  an int<a name="line.104"></a>
<span class="sourceLineNo">105</span>         * @return a Group object<a name="line.105"></a>
<span class="sourceLineNo">106</span>         * @see #getAtomLength()<a name="line.106"></a>
<span class="sourceLineNo">107</span>         * @see #getAtomGroups()<a name="line.107"></a>
<span class="sourceLineNo">108</span>         * @see #getSeqResGroup(int)<a name="line.108"></a>
<span class="sourceLineNo">109</span>         */<a name="line.109"></a>
<span class="sourceLineNo">110</span>        Group getAtomGroup (int position);<a name="line.110"></a>
<span class="sourceLineNo">111</span><a name="line.111"></a>
<span class="sourceLineNo">112</span>        /**<a name="line.112"></a>
<span class="sourceLineNo">113</span>         * Return the Group at given position,<a name="line.113"></a>
<span class="sourceLineNo">114</span>         * from within groups in the SEQRES records of the chain, i.e.<a name="line.114"></a>
<span class="sourceLineNo">115</span>         * the aminoacids/nucleotides in the construct.<a name="line.115"></a>
<span class="sourceLineNo">116</span>         * @param position  an int<a name="line.116"></a>
<span class="sourceLineNo">117</span>         * @return a Group object<a name="line.117"></a>
<span class="sourceLineNo">118</span>         * @see #getSeqResLength()<a name="line.118"></a>
<span class="sourceLineNo">119</span>         * @see #getSeqResGroups()<a name="line.119"></a>
<span class="sourceLineNo">120</span>         * @see #getAtomGroup(int)<a name="line.120"></a>
<span class="sourceLineNo">121</span>         */<a name="line.121"></a>
<span class="sourceLineNo">122</span>        Group getSeqResGroup (int position);<a name="line.122"></a>
<span class="sourceLineNo">123</span><a name="line.123"></a>
<span class="sourceLineNo">124</span><a name="line.124"></a>
<span class="sourceLineNo">125</span>        /**<a name="line.125"></a>
<span class="sourceLineNo">126</span>         * Return all Groups with observed density in the chain, i.e.<a name="line.126"></a>
<span class="sourceLineNo">127</span>         * those with coordinates in ATOM and HETATMS (including waters) records.<a name="line.127"></a>
<span class="sourceLineNo">128</span>         *<a name="line.128"></a>
<span class="sourceLineNo">129</span>         * @return a List object representing the Groups of this Chain.<a name="line.129"></a>
<span class="sourceLineNo">130</span>         * @see #setAtomGroups(List)<a name="line.130"></a>
<span class="sourceLineNo">131</span>         * @see #getAtomLength()<a name="line.131"></a>
<span class="sourceLineNo">132</span>         * @see #getSeqResGroups()<a name="line.132"></a>
<span class="sourceLineNo">133</span>         */<a name="line.133"></a>
<span class="sourceLineNo">134</span>        List&lt;Group&gt; getAtomGroups();<a name="line.134"></a>
<span class="sourceLineNo">135</span><a name="line.135"></a>
<span class="sourceLineNo">136</span>        /**<a name="line.136"></a>
<span class="sourceLineNo">137</span>         * Set all Groups with observed density in the chain, i.e.<a name="line.137"></a>
<span class="sourceLineNo">138</span>         * those with coordinates in ATOM and HETATMs (including waters) records.<a name="line.138"></a>
<span class="sourceLineNo">139</span>         * @param groups a List object representing the Groups of this Chain.<a name="line.139"></a>
<span class="sourceLineNo">140</span>         * @see #getAtomGroups()<a name="line.140"></a>
<span class="sourceLineNo">141</span>         */<a name="line.141"></a>
<span class="sourceLineNo">142</span>        void setAtomGroups(List&lt;Group&gt; groups);<a name="line.142"></a>
<span class="sourceLineNo">143</span><a name="line.143"></a>
<span class="sourceLineNo">144</span>        /**<a name="line.144"></a>
<span class="sourceLineNo">145</span>         * Return a List of all (observed) Groups of a special type, one of: {@link GroupType#AMINOACID},<a name="line.145"></a>
<span class="sourceLineNo">146</span>         * {@link GroupType#HETATM} or {@link GroupType#NUCLEOTIDE}.<a name="line.146"></a>
<span class="sourceLineNo">147</span>         * Note that if a standard aminoacid appears as a HETATM (because it is part of a ligand) then<a name="line.147"></a>
<span class="sourceLineNo">148</span>         * it is still considered as {@link GroupType#AMINOACID} and not as {@link GroupType#HETATM}.<a name="line.148"></a>
<span class="sourceLineNo">149</span>         * @param type  GroupType<a name="line.149"></a>
<span class="sourceLineNo">150</span>         * @return a List object<a name="line.150"></a>
<span class="sourceLineNo">151</span>         * @see #setAtomGroups(List)<a name="line.151"></a>
<span class="sourceLineNo">152</span>         */<a name="line.152"></a>
<span class="sourceLineNo">153</span>        List&lt;Group&gt; getAtomGroups (GroupType type);<a name="line.153"></a>
<span class="sourceLineNo">154</span><a name="line.154"></a>
<span class="sourceLineNo">155</span><a name="line.155"></a>
<span class="sourceLineNo">156</span>        /**<a name="line.156"></a>
<span class="sourceLineNo">157</span>         * Get a group by its PDB residue numbering. If the PDB residue number is not known,<a name="line.157"></a>
<span class="sourceLineNo">158</span>         * throws a StructureException.<a name="line.158"></a>
<span class="sourceLineNo">159</span>         *<a name="line.159"></a>
<span class="sourceLineNo">160</span>         * @param resNum the PDB residue number of the group<a name="line.160"></a>
<span class="sourceLineNo">161</span>         * @return the matching group<a name="line.161"></a>
<span class="sourceLineNo">162</span>         * @throws StructureException<a name="line.162"></a>
<span class="sourceLineNo">163</span>         */<a name="line.163"></a>
<span class="sourceLineNo">164</span>        Group getGroupByPDB(ResidueNumber resNum) throws StructureException;<a name="line.164"></a>
<span class="sourceLineNo">165</span><a name="line.165"></a>
<span class="sourceLineNo">166</span>        /**<a name="line.166"></a>
<span class="sourceLineNo">167</span>         * Get all groups that are located between two PDB residue numbers.<a name="line.167"></a>
<span class="sourceLineNo">168</span>         *<a name="line.168"></a>
<span class="sourceLineNo">169</span>         * @param pdbresnumStart PDB residue number of start. If null, defaults to the chain start.<a name="line.169"></a>
<span class="sourceLineNo">170</span>         * @param pdbresnumEnd PDB residue number of end. If null, defaults to the chain end.<a name="line.170"></a>
<span class="sourceLineNo">171</span>         * @return Groups in between. or throws a StructureException if either start or end can not be found,<a name="line.171"></a>
<span class="sourceLineNo">172</span>         * @throws StructureException<a name="line.172"></a>
<span class="sourceLineNo">173</span>         */<a name="line.173"></a>
<span class="sourceLineNo">174</span>        Group[] getGroupsByPDB(ResidueNumber pdbresnumStart, ResidueNumber pdbresnumEnd) throws StructureException;<a name="line.174"></a>
<span class="sourceLineNo">175</span><a name="line.175"></a>
<span class="sourceLineNo">176</span><a name="line.176"></a>
<span class="sourceLineNo">177</span>        /**<a name="line.177"></a>
<span class="sourceLineNo">178</span>         * Get all groups that are located between two PDB residue numbers. In contrast to getGroupsByPDB<a name="line.178"></a>
<span class="sourceLineNo">179</span>         * this method call ignores if the exact outer groups are not found. This is useful e.g. when requesting the range<a name="line.179"></a>
<span class="sourceLineNo">180</span>         * of groups as specified by the DBREF records - these frequently are rather inaccurate.<a name="line.180"></a>
<span class="sourceLineNo">181</span>         *<a name="line.181"></a>
<span class="sourceLineNo">182</span>         *<a name="line.182"></a>
<span class="sourceLineNo">183</span>         * @param pdbresnumStart PDB residue number of start. If null, defaults to the chain start.<a name="line.183"></a>
<span class="sourceLineNo">184</span>         * @param pdbresnumEnd PDB residue number of end. If null, defaults to the chain end.<a name="line.184"></a>
<span class="sourceLineNo">185</span>         * @param ignoreMissing ignore missing groups in this range.<a name="line.185"></a>
<span class="sourceLineNo">186</span>         * @return Groups in between. or throws a StructureException if either start or end can not be found,<a name="line.186"></a>
<span class="sourceLineNo">187</span>         * @throws StructureException<a name="line.187"></a>
<span class="sourceLineNo">188</span>         *<a name="line.188"></a>
<span class="sourceLineNo">189</span>         */<a name="line.189"></a>
<span class="sourceLineNo">190</span>        Group[] getGroupsByPDB(ResidueNumber pdbresnumStart, ResidueNumber pdbresnumEnd,boolean ignoreMissing) throws StructureException;<a name="line.190"></a>
<span class="sourceLineNo">191</span><a name="line.191"></a>
<span class="sourceLineNo">192</span><a name="line.192"></a>
<span class="sourceLineNo">193</span>        /**<a name="line.193"></a>
<span class="sourceLineNo">194</span>         * Returns the number of Groups with observed density in the chain, i.e.<a name="line.194"></a>
<span class="sourceLineNo">195</span>         * those with coordinates in ATOM and HETATMs (including waters) records<a name="line.195"></a>
<span class="sourceLineNo">196</span>         *<a name="line.196"></a>
<span class="sourceLineNo">197</span>         * @return the length<a name="line.197"></a>
<span class="sourceLineNo">198</span>         * @see #getAtomGroup(int)<a name="line.198"></a>
<span class="sourceLineNo">199</span>         * @see #getAtomGroups()<a name="line.199"></a>
<span class="sourceLineNo">200</span>         * @see #getSeqResLength())<a name="line.200"></a>
<span class="sourceLineNo">201</span>         */<a name="line.201"></a>
<span class="sourceLineNo">202</span>        int getAtomLength();<a name="line.202"></a>
<span class="sourceLineNo">203</span><a name="line.203"></a>
<span class="sourceLineNo">204</span>        /**<a name="line.204"></a>
<span class="sourceLineNo">205</span>         * Returns the number of groups in the SEQRES records of the chain, i.e.<a name="line.205"></a>
<span class="sourceLineNo">206</span>         * the number of aminoacids/nucleotides in the construct<a name="line.206"></a>
<span class="sourceLineNo">207</span>         *<a name="line.207"></a>
<span class="sourceLineNo">208</span>         * @return the length<a name="line.208"></a>
<span class="sourceLineNo">209</span>         * @see #getSeqResGroup(int)<a name="line.209"></a>
<span class="sourceLineNo">210</span>         * @see #getSeqResGroups()<a name="line.210"></a>
<span class="sourceLineNo">211</span>         * @see #getAtomLength()<a name="line.211"></a>
<span class="sourceLineNo">212</span>         */<a name="line.212"></a>
<span class="sourceLineNo">213</span>        int getSeqResLength();<a name="line.213"></a>
<span class="sourceLineNo">214</span><a name="line.214"></a>
<span class="sourceLineNo">215</span>        /**<a name="line.215"></a>
<span class="sourceLineNo">216</span>         * Sets the Entity information<a name="line.216"></a>
<span class="sourceLineNo">217</span>         * @param entityInfo the EntityInfo<a name="line.217"></a>
<span class="sourceLineNo">218</span>         * @see #getEntityInfo()<a name="line.218"></a>
<span class="sourceLineNo">219</span>         */<a name="line.219"></a>
<span class="sourceLineNo">220</span>        void setEntityInfo(EntityInfo entityInfo);<a name="line.220"></a>
<span class="sourceLineNo">221</span><a name="line.221"></a>
<span class="sourceLineNo">222</span>        /**<a name="line.222"></a>
<span class="sourceLineNo">223</span>         * Returns the EntityInfo for this chain.<a name="line.223"></a>
<span class="sourceLineNo">224</span>         *<a name="line.224"></a>
<span class="sourceLineNo">225</span>         * @return the EntityInfo object<a name="line.225"></a>
<span class="sourceLineNo">226</span>         * @see #setEntityInfo(EntityInfo)<a name="line.226"></a>
<span class="sourceLineNo">227</span>         */<a name="line.227"></a>
<span class="sourceLineNo">228</span>        EntityInfo getEntityInfo();<a name="line.228"></a>
<span class="sourceLineNo">229</span><a name="line.229"></a>
<span class="sourceLineNo">230</span>        /**<a name="line.230"></a>
<span class="sourceLineNo">231</span>         * Sets the 'private' asymId of this chain (Chain id in PDB file ).<a name="line.231"></a>
<span class="sourceLineNo">232</span>         * @param asymId  a String specifying the name value<a name="line.232"></a>
<span class="sourceLineNo">233</span>         * @see #getChainID()<a name="line.233"></a>
<span class="sourceLineNo">234</span>         * @deprecated  use {@link #setId(String asymId)} instead<a name="line.234"></a>
<span class="sourceLineNo">235</span>         */<a name="line.235"></a>
<span class="sourceLineNo">236</span>        @Deprecated<a name="line.236"></a>
<span class="sourceLineNo">237</span>        void setChainID(String asymId);<a name="line.237"></a>
<span class="sourceLineNo">238</span><a name="line.238"></a>
<span class="sourceLineNo">239</span><a name="line.239"></a>
<span class="sourceLineNo">240</span><a name="line.240"></a>
<span class="sourceLineNo">241</span>        /**<a name="line.241"></a>
<span class="sourceLineNo">242</span>         * Gets the 'private' asymId of this chain.<a name="line.242"></a>
<span class="sourceLineNo">243</span>         * @return a String representing the name value<a name="line.243"></a>
<span class="sourceLineNo">244</span>         * @see #setChainID(String)<a name="line.244"></a>
<span class="sourceLineNo">245</span>         * @deprecated  use getId() instead<a name="line.245"></a>
<span class="sourceLineNo">246</span>         */<a name="line.246"></a>
<span class="sourceLineNo">247</span>        @Deprecated<a name="line.247"></a>
<span class="sourceLineNo">248</span>        String getChainID();<a name="line.248"></a>
<span class="sourceLineNo">249</span><a name="line.249"></a>
<span class="sourceLineNo">250</span><a name="line.250"></a>
<span class="sourceLineNo">251</span>        /**<a name="line.251"></a>
<span class="sourceLineNo">252</span>         * If available, returns the internal chain ID that is used in mmCIF files (asym_id), otherwise null<a name="line.252"></a>
<span class="sourceLineNo">253</span>         *<a name="line.253"></a>
<span class="sourceLineNo">254</span>         * @return String or null<a name="line.254"></a>
<span class="sourceLineNo">255</span>         * @since 3.0.5<a name="line.255"></a>
<span class="sourceLineNo">256</span>         * @deprecated  use {@link #getId()} instead<a name="line.256"></a>
<span class="sourceLineNo">257</span>         */<a name="line.257"></a>
<span class="sourceLineNo">258</span>        String getInternalChainID();<a name="line.258"></a>
<span class="sourceLineNo">259</span><a name="line.259"></a>
<span class="sourceLineNo">260</span>        /**<a name="line.260"></a>
<span class="sourceLineNo">261</span>         * Sets the internal chain ID that is used in mmCif files<a name="line.261"></a>
<span class="sourceLineNo">262</span>         *<a name="line.262"></a>
<span class="sourceLineNo">263</span>         * @param internalChainID<a name="line.263"></a>
<span class="sourceLineNo">264</span>         * @since 3.0.5<a name="line.264"></a>
<span class="sourceLineNo">265</span>         * @deprecated use {@link #setId()} instead<a name="line.265"></a>
<span class="sourceLineNo">266</span>         */<a name="line.266"></a>
<span class="sourceLineNo">267</span>        void setInternalChainID(String internalChainID);<a name="line.267"></a>
<span class="sourceLineNo">268</span><a name="line.268"></a>
<span class="sourceLineNo">269</span><a name="line.269"></a>
<span class="sourceLineNo">270</span>        @Override<a name="line.270"></a>
<span class="sourceLineNo">271</span>        String toString();<a name="line.271"></a>
<span class="sourceLineNo">272</span><a name="line.272"></a>
<span class="sourceLineNo">273</span><a name="line.273"></a>
<span class="sourceLineNo">274</span>        /**<a name="line.274"></a>
<span class="sourceLineNo">275</span>         * Converts the SEQRES groups of a Chain to a Biojava Sequence object.<a name="line.275"></a>
<span class="sourceLineNo">276</span>         *<a name="line.276"></a>
<span class="sourceLineNo">277</span>         * @return the SEQRES groups of the Chain as a Sequence object.<a name="line.277"></a>
<span class="sourceLineNo">278</span>         */<a name="line.278"></a>
<span class="sourceLineNo">279</span>        Sequence&lt;?&gt; getBJSequence()  ;<a name="line.279"></a>
<span class="sourceLineNo">280</span><a name="line.280"></a>
<span class="sourceLineNo">281</span>        /**<a name="line.281"></a>
<span class="sourceLineNo">282</span>         * Returns the sequence of amino acids as it has been provided in the ATOM records.<a name="line.282"></a>
<span class="sourceLineNo">283</span>         * Non-standard residues will be present in the string only if the property<a name="line.283"></a>
<span class="sourceLineNo">284</span>         * {@value org.biojava.nbio.structure.io.PDBFileReader#LOAD_CHEM_COMP_PROPERTY} has been set.<a name="line.284"></a>
<span class="sourceLineNo">285</span>         * @return amino acid sequence as string<a name="line.285"></a>
<span class="sourceLineNo">286</span>         * @see #getSeqResSequence()<a name="line.286"></a>
<span class="sourceLineNo">287</span>         */<a name="line.287"></a>
<span class="sourceLineNo">288</span>        String getAtomSequence();<a name="line.288"></a>
<span class="sourceLineNo">289</span><a name="line.289"></a>
<span class="sourceLineNo">290</span>        /**<a name="line.290"></a>
<span class="sourceLineNo">291</span>         * Returns the PDB SEQRES sequence as a one-letter sequence string.<a name="line.291"></a>
<span class="sourceLineNo">292</span>         * Non-standard residues are represented by an "X".<a name="line.292"></a>
<span class="sourceLineNo">293</span>         * @return one-letter PDB SEQRES sequence as string<a name="line.293"></a>
<span class="sourceLineNo">294</span>         * @see #getAtomSequence()<a name="line.294"></a>
<span class="sourceLineNo">295</span>         */<a name="line.295"></a>
<span class="sourceLineNo">296</span>        String getSeqResSequence();<a name="line.296"></a>
<span class="sourceLineNo">297</span><a name="line.297"></a>
<span class="sourceLineNo">298</span>        /**<a name="line.298"></a>
<span class="sourceLineNo">299</span>         * Sets the Swissprot id of this chain.<a name="line.299"></a>
<span class="sourceLineNo">300</span>         * @param sp_id  a String specifying the swissprot id value<a name="line.300"></a>
<span class="sourceLineNo">301</span>         * @see #getSwissprotId()<a name="line.301"></a>
<span class="sourceLineNo">302</span>         */<a name="line.302"></a>
<span class="sourceLineNo">303</span>        void setSwissprotId(String sp_id);<a name="line.303"></a>
<span class="sourceLineNo">304</span><a name="line.304"></a>
<span class="sourceLineNo">305</span>        /**<a name="line.305"></a>
<span class="sourceLineNo">306</span>         * Gets the Swissprot id of this chain.<a name="line.306"></a>
<span class="sourceLineNo">307</span>         * @return a String representing the swissprot id value<a name="line.307"></a>
<span class="sourceLineNo">308</span>         * @see #setSwissprotId(String sp_id)<a name="line.308"></a>
<span class="sourceLineNo">309</span>         */<a name="line.309"></a>
<span class="sourceLineNo">310</span>        String getSwissprotId() ;<a name="line.310"></a>
<span class="sourceLineNo">311</span><a name="line.311"></a>
<span class="sourceLineNo">312</span><a name="line.312"></a>
<span class="sourceLineNo">313</span>        /**<a name="line.313"></a>
<span class="sourceLineNo">314</span>         * Returns a List of all SEQRES groups of a special type, one of: {@link GroupType#AMINOACID},<a name="line.314"></a>
<span class="sourceLineNo">315</span>         * {@link GroupType#HETATM} or {@link GroupType#NUCLEOTIDE}.<a name="line.315"></a>
<span class="sourceLineNo">316</span>         * @param type  a GroupType<a name="line.316"></a>
<span class="sourceLineNo">317</span>         * @return an List object<a name="line.317"></a>
<span class="sourceLineNo">318</span>         * @see #setSeqResGroups(List)<a name="line.318"></a>
<span class="sourceLineNo">319</span>         */<a name="line.319"></a>
<span class="sourceLineNo">320</span>        List&lt;Group&gt; getSeqResGroups (GroupType type);<a name="line.320"></a>
<span class="sourceLineNo">321</span><a name="line.321"></a>
<span class="sourceLineNo">322</span>        /**<a name="line.322"></a>
<span class="sourceLineNo">323</span>         * Returns a list of all groups in SEQRES records of the chain, i.e.<a name="line.323"></a>
<span class="sourceLineNo">324</span>         * the aminoacids/nucleotides in the construct.<a name="line.324"></a>
<span class="sourceLineNo">325</span>         * @return a List of all Group objects of this chain<a name="line.325"></a>
<span class="sourceLineNo">326</span>         * @see #setSeqResGroups(List)<a name="line.326"></a>
<span class="sourceLineNo">327</span>         * @see #getSeqResLength()<a name="line.327"></a>
<span class="sourceLineNo">328</span>         * @see #getAtomGroups()<a name="line.328"></a>
<span class="sourceLineNo">329</span>         */<a name="line.329"></a>
<span class="sourceLineNo">330</span>        List&lt;Group&gt; getSeqResGroups ();<a name="line.330"></a>
<span class="sourceLineNo">331</span><a name="line.331"></a>
<span class="sourceLineNo">332</span>        /**<a name="line.332"></a>
<span class="sourceLineNo">333</span>         * Sets the list of SeqResGroups for this chain.<a name="line.333"></a>
<span class="sourceLineNo">334</span>         *<a name="line.334"></a>
<span class="sourceLineNo">335</span>         * @param seqResGroups a List of Group objects that from the SEQRES groups of this chain.<a name="line.335"></a>
<span class="sourceLineNo">336</span>         * @see #getSeqResGroups()<a name="line.336"></a>
<span class="sourceLineNo">337</span>         */<a name="line.337"></a>
<span class="sourceLineNo">338</span>        void setSeqResGroups(List&lt;Group&gt; seqResGroups);<a name="line.338"></a>
<span class="sourceLineNo">339</span><a name="line.339"></a>
<span class="sourceLineNo">340</span>        /**<a name="line.340"></a>
<span class="sourceLineNo">341</span>         * Sets the back-reference to its parent Structure.<a name="line.341"></a>
<span class="sourceLineNo">342</span>         * @param parent the parent Structure object for this Chain<a name="line.342"></a>
<span class="sourceLineNo">343</span>         * @see #getStructure()<a name="line.343"></a>
<span class="sourceLineNo">344</span>         * @deprecated  use setStructure instead<a name="line.344"></a>
<span class="sourceLineNo">345</span>         *<a name="line.345"></a>
<span class="sourceLineNo">346</span>         */<a name="line.346"></a>
<span class="sourceLineNo">347</span>        @Deprecated<a name="line.347"></a>
<span class="sourceLineNo">348</span>         void setParent(Structure parent) ;<a name="line.348"></a>
<span class="sourceLineNo">349</span><a name="line.349"></a>
<span class="sourceLineNo">350</span>        /**<a name="line.350"></a>
<span class="sourceLineNo">351</span>         * Sets the back-reference to its parent Structure.<a name="line.351"></a>
<span class="sourceLineNo">352</span>         *<a name="line.352"></a>
<span class="sourceLineNo">353</span>         * @param parent<a name="line.353"></a>
<span class="sourceLineNo">354</span>         */<a name="line.354"></a>
<span class="sourceLineNo">355</span>        void setStructure(Structure parent) ;<a name="line.355"></a>
<span class="sourceLineNo">356</span><a name="line.356"></a>
<span class="sourceLineNo">357</span>        /**<a name="line.357"></a>
<span class="sourceLineNo">358</span>         * Returns the parent Structure of this chain.<a name="line.358"></a>
<span class="sourceLineNo">359</span>         *<a name="line.359"></a>
<span class="sourceLineNo">360</span>         * @return the parent Structure object<a name="line.360"></a>
<span class="sourceLineNo">361</span>         * @see #setStructure(Structure)<a name="line.361"></a>
<span class="sourceLineNo">362</span>         * @deprecated use getStructure(Structure) instead.<a name="line.362"></a>
<span class="sourceLineNo">363</span>         */<a name="line.363"></a>
<span class="sourceLineNo">364</span>        @Deprecated<a name="line.364"></a>
<span class="sourceLineNo">365</span>        Structure getParent() ;<a name="line.365"></a>
<span class="sourceLineNo">366</span><a name="line.366"></a>
<span class="sourceLineNo">367</span><a name="line.367"></a>
<span class="sourceLineNo">368</span>        /**<a name="line.368"></a>
<span class="sourceLineNo">369</span>         * Returns the parent Structure of this chain.<a name="line.369"></a>
<span class="sourceLineNo">370</span>         *<a name="line.370"></a>
<span class="sourceLineNo">371</span>         * @return the parent Structure object<a name="line.371"></a>
<span class="sourceLineNo">372</span>         * @see #setStructure(Structure)<a name="line.372"></a>
<span class="sourceLineNo">373</span>         */<a name="line.373"></a>
<span class="sourceLineNo">374</span>        Structure getStructure() ;<a name="line.374"></a>
<span class="sourceLineNo">375</span><a name="line.375"></a>
<span class="sourceLineNo">376</span>        /**<a name="line.376"></a>
<span class="sourceLineNo">377</span>         * Gets all groups that are not polymer groups and that are not solvent groups.<a name="line.377"></a>
<span class="sourceLineNo">378</span>         * Will automatically fetch Chemical Component files from the PDB web site, even if<a name="line.378"></a>
<span class="sourceLineNo">379</span>         * {@link FileParsingParameters#setLoadChemCompInfo(boolean)} has not been set to true.<a name="line.379"></a>
<span class="sourceLineNo">380</span>         * Otherwise the Ligands could not correctly be identified.<a name="line.380"></a>
<span class="sourceLineNo">381</span>         * @return list of Groups that are ligands<a name="line.381"></a>
<span class="sourceLineNo">382</span>         * @deprecated since biojava 5.0 this does not apply anymore. Chains contain either<a name="line.382"></a>
<span class="sourceLineNo">383</span>         * polymeric groups or non-polymeric groups<a name="line.383"></a>
<span class="sourceLineNo">384</span>         */<a name="line.384"></a>
<span class="sourceLineNo">385</span>        @Deprecated<a name="line.385"></a>
<span class="sourceLineNo">386</span>        List&lt;Group&gt; getAtomLigands();<a name="line.386"></a>
<span class="sourceLineNo">387</span><a name="line.387"></a>
<span class="sourceLineNo">388</span>        /**<a name="line.388"></a>
<span class="sourceLineNo">389</span>         * Convert this Chain to a String in PDB format<a name="line.389"></a>
<span class="sourceLineNo">390</span>         * @return<a name="line.390"></a>
<span class="sourceLineNo">391</span>         */<a name="line.391"></a>
<span class="sourceLineNo">392</span>        String toPDB();<a name="line.392"></a>
<span class="sourceLineNo">393</span><a name="line.393"></a>
<span class="sourceLineNo">394</span>        /**<a name="line.394"></a>
<span class="sourceLineNo">395</span>         * Convert this Chain to a String in mmCIF format<a name="line.395"></a>
<span class="sourceLineNo">396</span>         * @return<a name="line.396"></a>
<span class="sourceLineNo">397</span>         */<a name="line.397"></a>
<span class="sourceLineNo">398</span>        String toMMCIF();<a name="line.398"></a>
<span class="sourceLineNo">399</span><a name="line.399"></a>
<span class="sourceLineNo">400</span><a name="line.400"></a>
<span class="sourceLineNo">401</span>        /**<a name="line.401"></a>
<span class="sourceLineNo">402</span>         * Sets annotated sequence mismatches for this chain. This is based on the STRUCT_REF_SEQ_DIF mmCif category<a name="line.402"></a>
<span class="sourceLineNo">403</span>         *<a name="line.403"></a>
<span class="sourceLineNo">404</span>         * @param seqMisMatches<a name="line.404"></a>
<span class="sourceLineNo">405</span>         */<a name="line.405"></a>
<span class="sourceLineNo">406</span>        void setSeqMisMatches(List&lt;SeqMisMatch&gt; seqMisMatches);<a name="line.406"></a>
<span class="sourceLineNo">407</span><a name="line.407"></a>
<span class="sourceLineNo">408</span>        /**<a name="line.408"></a>
<span class="sourceLineNo">409</span>         * Gets annotated sequence mismatches for this chain. This is based on the STRUCT_REF_SEQ_DIF mmCif category<a name="line.409"></a>
<span class="sourceLineNo">410</span>         *<a name="line.410"></a>
<span class="sourceLineNo">411</span>         * @returns a list of sequence mismatches (or null if none found)<a name="line.411"></a>
<span class="sourceLineNo">412</span>         */<a name="line.412"></a>
<span class="sourceLineNo">413</span>        List&lt;SeqMisMatch&gt; getSeqMisMatches();<a name="line.413"></a>
<span class="sourceLineNo">414</span><a name="line.414"></a>
<span class="sourceLineNo">415</span>        /**<a name="line.415"></a>
<span class="sourceLineNo">416</span>         * Returns the EntityType of this chain. Equivalent to getEntityInfo().getType()<a name="line.416"></a>
<span class="sourceLineNo">417</span>         * @return<a name="line.417"></a>
<span class="sourceLineNo">418</span>         * @see EntityType<a name="line.418"></a>
<span class="sourceLineNo">419</span>         */<a name="line.419"></a>
<span class="sourceLineNo">420</span>        EntityType getEntityType();<a name="line.420"></a>
<span class="sourceLineNo">421</span><a name="line.421"></a>
<span class="sourceLineNo">422</span>        /** Tests if a chain is consisting of water molecules only<a name="line.422"></a>
<span class="sourceLineNo">423</span>         *<a name="line.423"></a>
<span class="sourceLineNo">424</span>         * @return true if there are only solvent molecules in this chain.<a name="line.424"></a>
<span class="sourceLineNo">425</span>         */<a name="line.425"></a>
<span class="sourceLineNo">426</span>        public boolean isWaterOnly();<a name="line.426"></a>
<span class="sourceLineNo">427</span><a name="line.427"></a>
<span class="sourceLineNo">428</span>        /**  Returns true if the given chain is composed of non-polymeric (including water) groups only.<a name="line.428"></a>
<span class="sourceLineNo">429</span>         *<a name="line.429"></a>
<span class="sourceLineNo">430</span>         * @return true if only non-polymeric groups in this chain.<a name="line.430"></a>
<span class="sourceLineNo">431</span>         */<a name="line.431"></a>
<span class="sourceLineNo">432</span>        public boolean isPureNonPolymer();<a name="line.432"></a>
<span class="sourceLineNo">433</span><a name="line.433"></a>
<span class="sourceLineNo">434</span>        /**<a name="line.434"></a>
<span class="sourceLineNo">435</span>         * Get the predominant {@link GroupType} for a given Chain, following these<a name="line.435"></a>
<span class="sourceLineNo">436</span>         * rules: &lt;li&gt;if the ratio of number of residues of a certain<a name="line.436"></a>
<span class="sourceLineNo">437</span>         * {@link GroupType} to total non-water residues is above the threshold<a name="line.437"></a>
<span class="sourceLineNo">438</span>         * {@value #org.biojava.nbio.structure.StructureTools.RATIO_RESIDUES_TO_TOTAL}, then that {@link GroupType} is<a name="line.438"></a>
<span class="sourceLineNo">439</span>         * returned&lt;/li&gt; &lt;li&gt;if there is no {@link GroupType} that is above the<a name="line.439"></a>
<span class="sourceLineNo">440</span>         * threshold then the {@link GroupType} with most members is chosen, logging<a name="line.440"></a>
<span class="sourceLineNo">441</span>         * it&lt;/li&gt;<a name="line.441"></a>
<span class="sourceLineNo">442</span>         * &lt;p&gt;<a name="line.442"></a>
<span class="sourceLineNo">443</span>         * See also {@link ChemComp#getPolymerType()} and<a name="line.443"></a>
<span class="sourceLineNo">444</span>         * {@link ChemComp#getResidueType()} which follow the PDB chemical component<a name="line.444"></a>
<span class="sourceLineNo">445</span>         * dictionary and provide a much more accurate description of groups and<a name="line.445"></a>
<span class="sourceLineNo">446</span>         * their linking.<a name="line.446"></a>
<span class="sourceLineNo">447</span>         * &lt;/p&gt;<a name="line.447"></a>
<span class="sourceLineNo">448</span>         *<a name="line.448"></a>
<span class="sourceLineNo">449</span>         * @return<a name="line.449"></a>
<span class="sourceLineNo">450</span>         */<a name="line.450"></a>
<span class="sourceLineNo">451</span>        public GroupType getPredominantGroupType();<a name="line.451"></a>
<span class="sourceLineNo">452</span><a name="line.452"></a>
<span class="sourceLineNo">453</span>        /**<a name="line.453"></a>
<span class="sourceLineNo">454</span>         * Tell whether given chain is a protein chain<a name="line.454"></a>
<span class="sourceLineNo">455</span>         *<a name="line.455"></a>
<span class="sourceLineNo">456</span>         * @return true if protein, false if nucleotide or ligand<a name="line.456"></a>
<span class="sourceLineNo">457</span>         * @see #getPredominantGroupType()<a name="line.457"></a>
<span class="sourceLineNo">458</span>         */<a name="line.458"></a>
<span class="sourceLineNo">459</span>        public  boolean isProtein();<a name="line.459"></a>
<span class="sourceLineNo">460</span><a name="line.460"></a>
<span class="sourceLineNo">461</span>        /**<a name="line.461"></a>
<span class="sourceLineNo">462</span>         * Tell whether given chain is DNA or RNA<a name="line.462"></a>
<span class="sourceLineNo">463</span>         *<a name="line.463"></a>
<span class="sourceLineNo">464</span>         * @return true if nucleic acid, false if protein or ligand<a name="line.464"></a>
<span class="sourceLineNo">465</span>         * @see #getPredominantGroupType()<a name="line.465"></a>
<span class="sourceLineNo">466</span>         */<a name="line.466"></a>
<span class="sourceLineNo">467</span>        public  boolean isNucleicAcid();<a name="line.467"></a>
<span class="sourceLineNo">468</span>}<a name="line.468"></a>




























































</pre>
</div>
</body>
</html>
