<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> * Created on 01-21-2010</a>
<span class="sourceLineNo">021</span><a id="line.21"> */</a>
<span class="sourceLineNo">022</span><a id="line.22">package org.biojava.nbio.core.sequence.transcription;</a>
<span class="sourceLineNo">023</span><a id="line.23"></a>
<span class="sourceLineNo">024</span><a id="line.24">import org.biojava.nbio.core.sequence.RNASequence;</a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</a>
<span class="sourceLineNo">026</span><a id="line.26">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</a>
<span class="sourceLineNo">027</span><a id="line.27">import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;</a>
<span class="sourceLineNo">028</span><a id="line.28">import org.biojava.nbio.core.sequence.template.AbstractCompoundTranslator;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.biojava.nbio.core.sequence.template.CompoundSet;</a>
<span class="sourceLineNo">030</span><a id="line.30">import org.biojava.nbio.core.sequence.template.Sequence;</a>
<span class="sourceLineNo">031</span><a id="line.31">import org.biojava.nbio.core.sequence.template.SequenceView;</a>
<span class="sourceLineNo">032</span><a id="line.32">import org.biojava.nbio.core.sequence.transcription.Table.Codon;</a>
<span class="sourceLineNo">033</span><a id="line.33">import org.biojava.nbio.core.sequence.views.WindowedSequence;</a>
<span class="sourceLineNo">034</span><a id="line.34"></a>
<span class="sourceLineNo">035</span><a id="line.35">import java.util.*;</a>
<span class="sourceLineNo">036</span><a id="line.36"></a>
<span class="sourceLineNo">037</span><a id="line.37">/**</a>
<span class="sourceLineNo">038</span><a id="line.38"> * Takes a {@link Sequence} of {@link NucleotideCompound} which should represent</a>
<span class="sourceLineNo">039</span><a id="line.39"> * an RNA sequence ({@link RNASequence} is good for this) and returns a list of</a>
<span class="sourceLineNo">040</span><a id="line.40"> * {@link Sequence} which hold {@link AminoAcidCompound}. The translator can</a>
<span class="sourceLineNo">041</span><a id="line.41"> * also trim stop codons as well as changing any valid start codon to an</a>
<span class="sourceLineNo">042</span><a id="line.42"> * initiating met.</a>
<span class="sourceLineNo">043</span><a id="line.43"> *</a>
<span class="sourceLineNo">044</span><a id="line.44"> * @author ayates</a>
<span class="sourceLineNo">045</span><a id="line.45"> */</a>
<span class="sourceLineNo">046</span><a id="line.46">public class RNAToAminoAcidTranslator extends</a>
<span class="sourceLineNo">047</span><a id="line.47">                AbstractCompoundTranslator&lt;NucleotideCompound, AminoAcidCompound&gt; {</a>
<span class="sourceLineNo">048</span><a id="line.48"></a>
<span class="sourceLineNo">049</span><a id="line.49">        private final boolean trimStops;</a>
<span class="sourceLineNo">050</span><a id="line.50">        private final boolean initMetOnly;</a>
<span class="sourceLineNo">051</span><a id="line.51">        private final Map&lt;Table.CaseInsensitiveTriplet, Codon&gt; quickLookup;</a>
<span class="sourceLineNo">052</span><a id="line.52">        private final Map&lt;AminoAcidCompound, List&lt;Codon&gt;&gt; aminoAcidToCodon;</a>
<span class="sourceLineNo">053</span><a id="line.53">        // Cheeky lookup which uses a hashing value; key is to switch to using this</a>
<span class="sourceLineNo">054</span><a id="line.54">        // all the time</a>
<span class="sourceLineNo">055</span><a id="line.55">        private final Codon[] codonArray = new Codon[64000];</a>
<span class="sourceLineNo">056</span><a id="line.56">        private final AminoAcidCompound unknownAminoAcidCompound;</a>
<span class="sourceLineNo">057</span><a id="line.57">        private final AminoAcidCompound methionineAminoAcidCompound;</a>
<span class="sourceLineNo">058</span><a id="line.58">        private final boolean translateNCodons;</a>
<span class="sourceLineNo">059</span><a id="line.59"></a>
<span class="sourceLineNo">060</span><a id="line.60">        // If true, then translation will stop at the first stop codon encountered</a>
<span class="sourceLineNo">061</span><a id="line.61">        // in the reading frame (the stop codon will be included as the last residue</a>
<span class="sourceLineNo">062</span><a id="line.62">        // in the resulting ProteinSequence, unless removed by #trimStops)</a>
<span class="sourceLineNo">063</span><a id="line.63">        private final boolean stopAtStopCodons;</a>
<span class="sourceLineNo">064</span><a id="line.64"></a>
<span class="sourceLineNo">065</span><a id="line.65">        // If true, then translation will not start until the first start codon</a>
<span class="sourceLineNo">066</span><a id="line.66">        // encountered in the reading frame. The start codon will be included as the</a>
<span class="sourceLineNo">067</span><a id="line.67">        // first residue in the resulting ProteinSequence</a>
<span class="sourceLineNo">068</span><a id="line.68">        private final boolean waitForStartCodon;</a>
<span class="sourceLineNo">069</span><a id="line.69"></a>
<span class="sourceLineNo">070</span><a id="line.70"></a>
<span class="sourceLineNo">071</span><a id="line.71">        public RNAToAminoAcidTranslator(</a>
<span class="sourceLineNo">072</span><a id="line.72">                        SequenceCreatorInterface&lt;AminoAcidCompound&gt; creator,</a>
<span class="sourceLineNo">073</span><a id="line.73">                        CompoundSet&lt;NucleotideCompound&gt; nucleotides,</a>
<span class="sourceLineNo">074</span><a id="line.74">                        CompoundSet&lt;Codon&gt; codons,</a>
<span class="sourceLineNo">075</span><a id="line.75">                        CompoundSet&lt;AminoAcidCompound&gt; aminoAcids, Table table,</a>
<span class="sourceLineNo">076</span><a id="line.76">                        boolean trimStops, boolean initMetOnly, boolean translateNCodons,</a>
<span class="sourceLineNo">077</span><a id="line.77">                        boolean stopAtStopCodons, boolean waitForStartCodon) {</a>
<span class="sourceLineNo">078</span><a id="line.78"></a>
<span class="sourceLineNo">079</span><a id="line.79">                super(creator, nucleotides, aminoAcids);</a>
<span class="sourceLineNo">080</span><a id="line.80">                this.trimStops = trimStops;</a>
<span class="sourceLineNo">081</span><a id="line.81">                this.initMetOnly = initMetOnly;</a>
<span class="sourceLineNo">082</span><a id="line.82">                this.translateNCodons = translateNCodons;</a>
<span class="sourceLineNo">083</span><a id="line.83"></a>
<span class="sourceLineNo">084</span><a id="line.84">                quickLookup = new HashMap&lt;Table.CaseInsensitiveTriplet, Codon&gt;(codons</a>
<span class="sourceLineNo">085</span><a id="line.85">                                .getAllCompounds().size());</a>
<span class="sourceLineNo">086</span><a id="line.86">                aminoAcidToCodon = new HashMap&lt;AminoAcidCompound, List&lt;Codon&gt;&gt;();</a>
<span class="sourceLineNo">087</span><a id="line.87"></a>
<span class="sourceLineNo">088</span><a id="line.88">                List&lt;Codon&gt; codonList = table.getCodons(nucleotides, aminoAcids);</a>
<span class="sourceLineNo">089</span><a id="line.89">                for (Codon codon : codonList) {</a>
<span class="sourceLineNo">090</span><a id="line.90">                        quickLookup.put(codon.getTriplet(), codon);</a>
<span class="sourceLineNo">091</span><a id="line.91">                        codonArray[codon.getTriplet().intValue()] = codon;</a>
<span class="sourceLineNo">092</span><a id="line.92"></a>
<span class="sourceLineNo">093</span><a id="line.93">                        List&lt;Codon&gt; codonL = aminoAcidToCodon.get(codon.getAminoAcid());</a>
<span class="sourceLineNo">094</span><a id="line.94">                        if (codonL == null) {</a>
<span class="sourceLineNo">095</span><a id="line.95">                                codonL = new ArrayList&lt;Codon&gt;();</a>
<span class="sourceLineNo">096</span><a id="line.96">                                aminoAcidToCodon.put(codon.getAminoAcid(), codonL);</a>
<span class="sourceLineNo">097</span><a id="line.97">                        }</a>
<span class="sourceLineNo">098</span><a id="line.98">                        codonL.add(codon);</a>
<span class="sourceLineNo">099</span><a id="line.99"></a>
<span class="sourceLineNo">100</span><a id="line.100">                }</a>
<span class="sourceLineNo">101</span><a id="line.101">                unknownAminoAcidCompound = aminoAcids.getCompoundForString("X");</a>
<span class="sourceLineNo">102</span><a id="line.102">                methionineAminoAcidCompound = aminoAcids.getCompoundForString("M");</a>
<span class="sourceLineNo">103</span><a id="line.103">                this.stopAtStopCodons = stopAtStopCodons;</a>
<span class="sourceLineNo">104</span><a id="line.104">                this.waitForStartCodon = waitForStartCodon;</a>
<span class="sourceLineNo">105</span><a id="line.105">        }</a>
<span class="sourceLineNo">106</span><a id="line.106"></a>
<span class="sourceLineNo">107</span><a id="line.107">        /**</a>
<span class="sourceLineNo">108</span><a id="line.108">         * Performs the core conversion of RNA to Peptide. It does this by walking a</a>
<span class="sourceLineNo">109</span><a id="line.109">         * windowed version of the given sequence. Any trailing DNA base pairs are</a>
<span class="sourceLineNo">110</span><a id="line.110">         * ignored according to the specification of {@link WindowedSequence}.</a>
<span class="sourceLineNo">111</span><a id="line.111">         */</a>
<span class="sourceLineNo">112</span><a id="line.112"></a>
<span class="sourceLineNo">113</span><a id="line.113">        @Override</a>
<span class="sourceLineNo">114</span><a id="line.114">        public List&lt;Sequence&lt;AminoAcidCompound&gt;&gt; createSequences(</a>
<span class="sourceLineNo">115</span><a id="line.115">                        Sequence&lt;NucleotideCompound&gt; originalSequence) {</a>
<span class="sourceLineNo">116</span><a id="line.116"></a>
<span class="sourceLineNo">117</span><a id="line.117">                List&lt;List&lt;AminoAcidCompound&gt;&gt; workingList = new ArrayList&lt;List&lt;AminoAcidCompound&gt;&gt;();</a>
<span class="sourceLineNo">118</span><a id="line.118"></a>
<span class="sourceLineNo">119</span><a id="line.119">                Iterable&lt;SequenceView&lt;NucleotideCompound&gt;&gt; iter = new WindowedSequence&lt;NucleotideCompound&gt;(</a>
<span class="sourceLineNo">120</span><a id="line.120">                                originalSequence, 3);</a>
<span class="sourceLineNo">121</span><a id="line.121"></a>
<span class="sourceLineNo">122</span><a id="line.122">                boolean first = true;</a>
<span class="sourceLineNo">123</span><a id="line.123"></a>
<span class="sourceLineNo">124</span><a id="line.124">                // If not waiting for a start codon, start translating immediately</a>
<span class="sourceLineNo">125</span><a id="line.125">                boolean doTranslate = !waitForStartCodon;</a>
<span class="sourceLineNo">126</span><a id="line.126"></a>
<span class="sourceLineNo">127</span><a id="line.127">                for (SequenceView&lt;NucleotideCompound&gt; element : iter) {</a>
<span class="sourceLineNo">128</span><a id="line.128">                        AminoAcidCompound aminoAcid = null;</a>
<span class="sourceLineNo">129</span><a id="line.129"></a>
<span class="sourceLineNo">130</span><a id="line.130">                        int i = 1;</a>
<span class="sourceLineNo">131</span><a id="line.131">                        Table.CaseInsensitiveTriplet triplet = new Table.CaseInsensitiveTriplet(</a>
<span class="sourceLineNo">132</span><a id="line.132">                                        element.getCompoundAt(i++), element.getCompoundAt(i++),</a>
<span class="sourceLineNo">133</span><a id="line.133">                                        element.getCompoundAt(i++));</a>
<span class="sourceLineNo">134</span><a id="line.134"></a>
<span class="sourceLineNo">135</span><a id="line.135">                        Codon target = null;</a>
<span class="sourceLineNo">136</span><a id="line.136"></a>
<span class="sourceLineNo">137</span><a id="line.137">                        target = quickLookup.get(triplet);</a>
<span class="sourceLineNo">138</span><a id="line.138"></a>
<span class="sourceLineNo">139</span><a id="line.139">                        // Check for a start</a>
<span class="sourceLineNo">140</span><a id="line.140">                        if (!doTranslate &amp;&amp; target.isStart()) {</a>
<span class="sourceLineNo">141</span><a id="line.141">                                doTranslate = true;</a>
<span class="sourceLineNo">142</span><a id="line.142">                        }</a>
<span class="sourceLineNo">143</span><a id="line.143"></a>
<span class="sourceLineNo">144</span><a id="line.144">                        if (doTranslate) {</a>
<span class="sourceLineNo">145</span><a id="line.145">                                if (target != null)</a>
<span class="sourceLineNo">146</span><a id="line.146">                                        aminoAcid = target.getAminoAcid();</a>
<span class="sourceLineNo">147</span><a id="line.147">                                if (aminoAcid == null &amp;&amp; translateNCodons()) {</a>
<span class="sourceLineNo">148</span><a id="line.148">                                        aminoAcid = unknownAminoAcidCompound;</a>
<span class="sourceLineNo">149</span><a id="line.149">                                } else {</a>
<span class="sourceLineNo">150</span><a id="line.150">                                        if (first &amp;&amp; initMetOnly &amp;&amp; target.isStart()) {</a>
<span class="sourceLineNo">151</span><a id="line.151">                                                aminoAcid = methionineAminoAcidCompound;</a>
<span class="sourceLineNo">152</span><a id="line.152">                                        }</a>
<span class="sourceLineNo">153</span><a id="line.153">                                }</a>
<span class="sourceLineNo">154</span><a id="line.154"></a>
<span class="sourceLineNo">155</span><a id="line.155">                                addCompoundsToList(Arrays.asList(aminoAcid), workingList);</a>
<span class="sourceLineNo">156</span><a id="line.156">                        }</a>
<span class="sourceLineNo">157</span><a id="line.157"></a>
<span class="sourceLineNo">158</span><a id="line.158">                        if (doTranslate &amp;&amp; stopAtStopCodons &amp;&amp; target.isStop()) {</a>
<span class="sourceLineNo">159</span><a id="line.159">                                // Check if we need to stop, but dont stop until started!</a>
<span class="sourceLineNo">160</span><a id="line.160">                                break;</a>
<span class="sourceLineNo">161</span><a id="line.161">                        }</a>
<span class="sourceLineNo">162</span><a id="line.162"></a>
<span class="sourceLineNo">163</span><a id="line.163">                        first = false;</a>
<span class="sourceLineNo">164</span><a id="line.164">                }</a>
<span class="sourceLineNo">165</span><a id="line.165">                postProcessCompoundLists(workingList);</a>
<span class="sourceLineNo">166</span><a id="line.166"></a>
<span class="sourceLineNo">167</span><a id="line.167">                return workingListToSequences(workingList);</a>
<span class="sourceLineNo">168</span><a id="line.168">        }</a>
<span class="sourceLineNo">169</span><a id="line.169"></a>
<span class="sourceLineNo">170</span><a id="line.170">        /**</a>
<span class="sourceLineNo">171</span><a id="line.171">         * Performs the trimming of stop codons and the conversion of a valid start</a>
<span class="sourceLineNo">172</span><a id="line.172">         * amino acid to M</a>
<span class="sourceLineNo">173</span><a id="line.173">         */</a>
<span class="sourceLineNo">174</span><a id="line.174">        @Override</a>
<span class="sourceLineNo">175</span><a id="line.175">        protected void postProcessCompoundLists(</a>
<span class="sourceLineNo">176</span><a id="line.176">                        List&lt;List&lt;AminoAcidCompound&gt;&gt; compoundLists) {</a>
<span class="sourceLineNo">177</span><a id="line.177">                for (List&lt;AminoAcidCompound&gt; compounds : compoundLists) {</a>
<span class="sourceLineNo">178</span><a id="line.178">                        if (trimStops) {</a>
<span class="sourceLineNo">179</span><a id="line.179">                                trimStop(compounds);</a>
<span class="sourceLineNo">180</span><a id="line.180">                        }</a>
<span class="sourceLineNo">181</span><a id="line.181">                }</a>
<span class="sourceLineNo">182</span><a id="line.182">        }</a>
<span class="sourceLineNo">183</span><a id="line.183"></a>
<span class="sourceLineNo">184</span><a id="line.184">        /**</a>
<span class="sourceLineNo">185</span><a id="line.185">         * Imperfect code. Checks the last amino acid to see if a codon could have</a>
<span class="sourceLineNo">186</span><a id="line.186">         * translated a stop for it. Left in for the moment</a>
<span class="sourceLineNo">187</span><a id="line.187">         */</a>
<span class="sourceLineNo">188</span><a id="line.188">        protected void trimStop(List&lt;AminoAcidCompound&gt; sequence) {</a>
<span class="sourceLineNo">189</span><a id="line.189">                AminoAcidCompound stop = sequence.get(sequence.size() - 1);</a>
<span class="sourceLineNo">190</span><a id="line.190">                boolean isStop = false;</a>
<span class="sourceLineNo">191</span><a id="line.191">                if (aminoAcidToCodon.containsKey(stop)) {</a>
<span class="sourceLineNo">192</span><a id="line.192">                        for (Codon c : aminoAcidToCodon.get(stop)) {</a>
<span class="sourceLineNo">193</span><a id="line.193">                                if (c.isStop()) {</a>
<span class="sourceLineNo">194</span><a id="line.194">                                        isStop = true;</a>
<span class="sourceLineNo">195</span><a id="line.195">                                        break;</a>
<span class="sourceLineNo">196</span><a id="line.196">                                }</a>
<span class="sourceLineNo">197</span><a id="line.197">                        }</a>
<span class="sourceLineNo">198</span><a id="line.198">                }</a>
<span class="sourceLineNo">199</span><a id="line.199"></a>
<span class="sourceLineNo">200</span><a id="line.200">                if (isStop) {</a>
<span class="sourceLineNo">201</span><a id="line.201">                        sequence.remove(sequence.size() - 1);</a>
<span class="sourceLineNo">202</span><a id="line.202">                }</a>
<span class="sourceLineNo">203</span><a id="line.203">        }</a>
<span class="sourceLineNo">204</span><a id="line.204"></a>
<span class="sourceLineNo">205</span><a id="line.205">        /**</a>
<span class="sourceLineNo">206</span><a id="line.206">         * Indicates if we want to force exact translation of compounds or not i.e.</a>
<span class="sourceLineNo">207</span><a id="line.207">         * those with internal N RNA bases. This will cause a translation to an X</a>
<span class="sourceLineNo">208</span><a id="line.208">         * amino acid</a>
<span class="sourceLineNo">209</span><a id="line.209">         */</a>
<span class="sourceLineNo">210</span><a id="line.210">        public boolean translateNCodons() {</a>
<span class="sourceLineNo">211</span><a id="line.211">                return translateNCodons;</a>
<span class="sourceLineNo">212</span><a id="line.212">        }</a>
<span class="sourceLineNo">213</span><a id="line.213">}</a>




























































</pre>
</div>
</main>
</body>
</html>
