<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> * @(#)SequenceUtil.java 1.0 September 2009</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * Copyright (c) 2009 Peter Troshin</a>
<span class="sourceLineNo">005</span><a id="line.5"> *</a>
<span class="sourceLineNo">006</span><a id="line.6"> *        BioJava development code</a>
<span class="sourceLineNo">007</span><a id="line.7"> *</a>
<span class="sourceLineNo">008</span><a id="line.8"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">009</span><a id="line.9"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">010</span><a id="line.10"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">011</span><a id="line.11"> * see:</a>
<span class="sourceLineNo">012</span><a id="line.12"> *</a>
<span class="sourceLineNo">013</span><a id="line.13"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">014</span><a id="line.14"> *</a>
<span class="sourceLineNo">015</span><a id="line.15"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">016</span><a id="line.16"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">019</span><a id="line.19"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">020</span><a id="line.20"> * at:</a>
<span class="sourceLineNo">021</span><a id="line.21"> *</a>
<span class="sourceLineNo">022</span><a id="line.22"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">023</span><a id="line.23"> *</a>
<span class="sourceLineNo">024</span><a id="line.24"> */</a>
<span class="sourceLineNo">025</span><a id="line.25"></a>
<span class="sourceLineNo">026</span><a id="line.26">package org.biojava.nbio.data.sequence;</a>
<span class="sourceLineNo">027</span><a id="line.27"></a>
<span class="sourceLineNo">028</span><a id="line.28">import org.slf4j.Logger;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.slf4j.LoggerFactory;</a>
<span class="sourceLineNo">030</span><a id="line.30"></a>
<span class="sourceLineNo">031</span><a id="line.31">import java.io.*;</a>
<span class="sourceLineNo">032</span><a id="line.32">import java.util.ArrayList;</a>
<span class="sourceLineNo">033</span><a id="line.33">import java.util.List;</a>
<span class="sourceLineNo">034</span><a id="line.34">import java.util.regex.Matcher;</a>
<span class="sourceLineNo">035</span><a id="line.35">import java.util.regex.Pattern;</a>
<span class="sourceLineNo">036</span><a id="line.36"></a>
<span class="sourceLineNo">037</span><a id="line.37">/**</a>
<span class="sourceLineNo">038</span><a id="line.38"> * Utility class for operations on sequences</a>
<span class="sourceLineNo">039</span><a id="line.39"> *</a>
<span class="sourceLineNo">040</span><a id="line.40"> * @author Peter Troshin</a>
<span class="sourceLineNo">041</span><a id="line.41"> * @version 1.0</a>
<span class="sourceLineNo">042</span><a id="line.42"> * @since 3.0.2</a>
<span class="sourceLineNo">043</span><a id="line.43"> */</a>
<span class="sourceLineNo">044</span><a id="line.44">public final class SequenceUtil {</a>
<span class="sourceLineNo">045</span><a id="line.45"></a>
<span class="sourceLineNo">046</span><a id="line.46">        private static final Logger logger = LoggerFactory.getLogger(SequenceUtil.class);</a>
<span class="sourceLineNo">047</span><a id="line.47"></a>
<span class="sourceLineNo">048</span><a id="line.48">        /**</a>
<span class="sourceLineNo">049</span><a id="line.49">         * A whitespace character: [\t\n\x0B\f\r]</a>
<span class="sourceLineNo">050</span><a id="line.50">         */</a>
<span class="sourceLineNo">051</span><a id="line.51">        public static final Pattern WHITE_SPACE = Pattern.compile("\\s");</a>
<span class="sourceLineNo">052</span><a id="line.52"></a>
<span class="sourceLineNo">053</span><a id="line.53">        /**</a>
<span class="sourceLineNo">054</span><a id="line.54">         * A digit</a>
<span class="sourceLineNo">055</span><a id="line.55">         */</a>
<span class="sourceLineNo">056</span><a id="line.56">        public static final Pattern DIGIT = Pattern.compile("\\d");</a>
<span class="sourceLineNo">057</span><a id="line.57"></a>
<span class="sourceLineNo">058</span><a id="line.58">        /**</a>
<span class="sourceLineNo">059</span><a id="line.59">         * Non word</a>
<span class="sourceLineNo">060</span><a id="line.60">         */</a>
<span class="sourceLineNo">061</span><a id="line.61">        public static final Pattern NONWORD = Pattern.compile("\\W");</a>
<span class="sourceLineNo">062</span><a id="line.62"></a>
<span class="sourceLineNo">063</span><a id="line.63">        /**</a>
<span class="sourceLineNo">064</span><a id="line.64">         * Valid Amino acids</a>
<span class="sourceLineNo">065</span><a id="line.65">         */</a>
<span class="sourceLineNo">066</span><a id="line.66">        public static final Pattern AA = Pattern.compile("[ARNDCQEGHILKMFPSTWYVUO]+",</a>
<span class="sourceLineNo">067</span><a id="line.67">                Pattern.CASE_INSENSITIVE);</a>
<span class="sourceLineNo">068</span><a id="line.68"></a>
<span class="sourceLineNo">069</span><a id="line.69">        /**</a>
<span class="sourceLineNo">070</span><a id="line.70">         * inversion of AA pattern</a>
<span class="sourceLineNo">071</span><a id="line.71">         */</a>
<span class="sourceLineNo">072</span><a id="line.72">        public static final Pattern NON_AA = Pattern.compile(</a>
<span class="sourceLineNo">073</span><a id="line.73">                "[^ARNDCQEGHILKMFPSTWYVXUO]+", Pattern.CASE_INSENSITIVE);</a>
<span class="sourceLineNo">074</span><a id="line.74"></a>
<span class="sourceLineNo">075</span><a id="line.75">        /**</a>
<span class="sourceLineNo">076</span><a id="line.76">         * Same as AA pattern but with one additional letters - X</a>
<span class="sourceLineNo">077</span><a id="line.77">         */</a>
<span class="sourceLineNo">078</span><a id="line.78">        public static final Pattern AMBIGUOUS_AA = Pattern.compile(</a>
<span class="sourceLineNo">079</span><a id="line.79">                "[ARNDCQEGHILKMFPSTWYVXUO]+", Pattern.CASE_INSENSITIVE);</a>
<span class="sourceLineNo">080</span><a id="line.80"></a>
<span class="sourceLineNo">081</span><a id="line.81">        /**</a>
<span class="sourceLineNo">082</span><a id="line.82">         * Nucleotides a, t, g, c, u</a>
<span class="sourceLineNo">083</span><a id="line.83">         */</a>
<span class="sourceLineNo">084</span><a id="line.84">        public static final Pattern NUCLEOTIDE = Pattern.compile("[AGTCU]+",</a>
<span class="sourceLineNo">085</span><a id="line.85">                Pattern.CASE_INSENSITIVE);</a>
<span class="sourceLineNo">086</span><a id="line.86"></a>
<span class="sourceLineNo">087</span><a id="line.87">        /**</a>
<span class="sourceLineNo">088</span><a id="line.88">         * Ambiguous nucleotide</a>
<span class="sourceLineNo">089</span><a id="line.89">         */</a>
<span class="sourceLineNo">090</span><a id="line.90">        public static final Pattern AMBIGUOUS_NUCLEOTIDE = Pattern.compile(</a>
<span class="sourceLineNo">091</span><a id="line.91">                "[AGTCRYMKSWHBVDNU]+", Pattern.CASE_INSENSITIVE); // see IUPAC</a>
<span class="sourceLineNo">092</span><a id="line.92">        /**</a>
<span class="sourceLineNo">093</span><a id="line.93">         * Non nucleotide</a>
<span class="sourceLineNo">094</span><a id="line.94">         */</a>
<span class="sourceLineNo">095</span><a id="line.95">        public static final Pattern NON_NUCLEOTIDE = Pattern.compile("[^AGTCU]+",</a>
<span class="sourceLineNo">096</span><a id="line.96">                Pattern.CASE_INSENSITIVE);</a>
<span class="sourceLineNo">097</span><a id="line.97"></a>
<span class="sourceLineNo">098</span><a id="line.98">        private SequenceUtil() {</a>
<span class="sourceLineNo">099</span><a id="line.99">        } // utility class, no instantiation</a>
<span class="sourceLineNo">100</span><a id="line.100"></a>
<span class="sourceLineNo">101</span><a id="line.101">        /*</a>
<span class="sourceLineNo">102</span><a id="line.102">         * public static void write_PirSeq(OutputStream os, FastaSequence seq)</a>
<span class="sourceLineNo">103</span><a id="line.103">         * throws IOException { BufferedWriter pir_out = new BufferedWriter(new</a>
<span class="sourceLineNo">104</span><a id="line.104">         * OutputStreamWriter(os)); pir_out.write("&gt;P1;" + seq.getId() +</a>
<span class="sourceLineNo">105</span><a id="line.105">         * SysPrefs.newlinechar); pir_out.write(seq.getSequence() +</a>
<span class="sourceLineNo">106</span><a id="line.106">         * SysPrefs.newlinechar); pir_out.close(); }</a>
<span class="sourceLineNo">107</span><a id="line.107">         *</a>
<span class="sourceLineNo">108</span><a id="line.108">         * public static void write_FastaSeq(OutputStream os, FastaSequence seq)</a>
<span class="sourceLineNo">109</span><a id="line.109">         * throws IOException { BufferedWriter fasta_out = new BufferedWriter( new</a>
<span class="sourceLineNo">110</span><a id="line.110">         * OutputStreamWriter(os)); fasta_out.write("&gt;" + seq.getId() +</a>
<span class="sourceLineNo">111</span><a id="line.111">         * SysPrefs.newlinechar); fasta_out.write(seq.getSequence() +</a>
<span class="sourceLineNo">112</span><a id="line.112">         * SysPrefs.newlinechar); fasta_out.close(); }</a>
<span class="sourceLineNo">113</span><a id="line.113">         */</a>
<span class="sourceLineNo">114</span><a id="line.114"></a>
<span class="sourceLineNo">115</span><a id="line.115">        /**</a>
<span class="sourceLineNo">116</span><a id="line.116">         * @return true is the sequence contains only letters a,c, t, g, u</a>
<span class="sourceLineNo">117</span><a id="line.117">         */</a>
<span class="sourceLineNo">118</span><a id="line.118">        public static boolean isNucleotideSequence(final FastaSequence s) {</a>
<span class="sourceLineNo">119</span><a id="line.119">        return SequenceUtil.isNonAmbNucleotideSequence(s.getSequence());</a>
<span class="sourceLineNo">120</span><a id="line.120">        }</a>
<span class="sourceLineNo">121</span><a id="line.121"></a>
<span class="sourceLineNo">122</span><a id="line.122">        /**</a>
<span class="sourceLineNo">123</span><a id="line.123">         * Ambiguous DNA chars : AGTCRYMKSWHBVDN // differs from protein in only one</a>
<span class="sourceLineNo">124</span><a id="line.124">         * (!) - B char</a>
<span class="sourceLineNo">125</span><a id="line.125">         */</a>
<span class="sourceLineNo">126</span><a id="line.126">        public static boolean isNonAmbNucleotideSequence(String sequence) {</a>
<span class="sourceLineNo">127</span><a id="line.127">        sequence = SequenceUtil.cleanSequence(sequence);</a>
<span class="sourceLineNo">128</span><a id="line.128">        if (SequenceUtil.DIGIT.matcher(sequence).find()) {</a>
<span class="sourceLineNo">129</span><a id="line.129">                return false;</a>
<span class="sourceLineNo">130</span><a id="line.130">        }</a>
<span class="sourceLineNo">131</span><a id="line.131">        if (SequenceUtil.NON_NUCLEOTIDE.matcher(sequence).find()) {</a>
<span class="sourceLineNo">132</span><a id="line.132">                return false;</a>
<span class="sourceLineNo">133</span><a id="line.133">                /*</a>
<span class="sourceLineNo">134</span><a id="line.134">                 * System.out.format("I found the text starting at " +</a>
<span class="sourceLineNo">135</span><a id="line.135">                 * "index %d and ending at index %d.%n", nonDNAmatcher .start(),</a>
<span class="sourceLineNo">136</span><a id="line.136">                 * nonDNAmatcher.end());</a>
<span class="sourceLineNo">137</span><a id="line.137">                 */</a>
<span class="sourceLineNo">138</span><a id="line.138">        }</a>
<span class="sourceLineNo">139</span><a id="line.139">        final Matcher DNAmatcher = SequenceUtil.NUCLEOTIDE.matcher(sequence);</a>
<span class="sourceLineNo">140</span><a id="line.140">        return DNAmatcher.find();</a>
<span class="sourceLineNo">141</span><a id="line.141">        }</a>
<span class="sourceLineNo">142</span><a id="line.142"></a>
<span class="sourceLineNo">143</span><a id="line.143">        /**</a>
<span class="sourceLineNo">144</span><a id="line.144">         * Removes all whitespace chars in the sequence string</a>
<span class="sourceLineNo">145</span><a id="line.145">         *</a>
<span class="sourceLineNo">146</span><a id="line.146">         * @param sequence</a>
<span class="sourceLineNo">147</span><a id="line.147">         * @return cleaned up sequence</a>
<span class="sourceLineNo">148</span><a id="line.148">         */</a>
<span class="sourceLineNo">149</span><a id="line.149">        public static String cleanSequence(String sequence) {</a>
<span class="sourceLineNo">150</span><a id="line.150">        assert sequence != null;</a>
<span class="sourceLineNo">151</span><a id="line.151">        final Matcher m = SequenceUtil.WHITE_SPACE.matcher(sequence);</a>
<span class="sourceLineNo">152</span><a id="line.152">        sequence = m.replaceAll("").toUpperCase();</a>
<span class="sourceLineNo">153</span><a id="line.153">        return sequence;</a>
<span class="sourceLineNo">154</span><a id="line.154">        }</a>
<span class="sourceLineNo">155</span><a id="line.155"></a>
<span class="sourceLineNo">156</span><a id="line.156">        /**</a>
<span class="sourceLineNo">157</span><a id="line.157">         * Removes all special characters and digits as well as whitespace chars</a>
<span class="sourceLineNo">158</span><a id="line.158">         * from the sequence</a>
<span class="sourceLineNo">159</span><a id="line.159">         *</a>
<span class="sourceLineNo">160</span><a id="line.160">         * @param sequence</a>
<span class="sourceLineNo">161</span><a id="line.161">         * @return cleaned up sequence</a>
<span class="sourceLineNo">162</span><a id="line.162">         */</a>
<span class="sourceLineNo">163</span><a id="line.163">        public static String deepCleanSequence(String sequence) {</a>
<span class="sourceLineNo">164</span><a id="line.164">        sequence = SequenceUtil.cleanSequence(sequence);</a>
<span class="sourceLineNo">165</span><a id="line.165">        sequence = SequenceUtil.DIGIT.matcher(sequence).replaceAll("");</a>
<span class="sourceLineNo">166</span><a id="line.166">        sequence = SequenceUtil.NONWORD.matcher(sequence).replaceAll("");</a>
<span class="sourceLineNo">167</span><a id="line.167">        final Pattern othernonSeqChars = Pattern.compile("[_-]+");</a>
<span class="sourceLineNo">168</span><a id="line.168">        sequence = othernonSeqChars.matcher(sequence).replaceAll("");</a>
<span class="sourceLineNo">169</span><a id="line.169">        return sequence;</a>
<span class="sourceLineNo">170</span><a id="line.170">        }</a>
<span class="sourceLineNo">171</span><a id="line.171"></a>
<span class="sourceLineNo">172</span><a id="line.172">        /**</a>
<span class="sourceLineNo">173</span><a id="line.173">         *</a>
<span class="sourceLineNo">174</span><a id="line.174">         * @param sequence</a>
<span class="sourceLineNo">175</span><a id="line.175">         * @return true is the sequence is a protein sequence, false overwise</a>
<span class="sourceLineNo">176</span><a id="line.176">         */</a>
<span class="sourceLineNo">177</span><a id="line.177">        public static boolean isProteinSequence(String sequence) {</a>
<span class="sourceLineNo">178</span><a id="line.178">        sequence = SequenceUtil.cleanSequence(sequence);</a>
<span class="sourceLineNo">179</span><a id="line.179">        if (SequenceUtil.isNonAmbNucleotideSequence(sequence)) {</a>
<span class="sourceLineNo">180</span><a id="line.180">                return false;</a>
<span class="sourceLineNo">181</span><a id="line.181">        }</a>
<span class="sourceLineNo">182</span><a id="line.182">        if (SequenceUtil.DIGIT.matcher(sequence).find()) {</a>
<span class="sourceLineNo">183</span><a id="line.183">                return false;</a>
<span class="sourceLineNo">184</span><a id="line.184">        }</a>
<span class="sourceLineNo">185</span><a id="line.185">        if (SequenceUtil.NON_AA.matcher(sequence).find()) {</a>
<span class="sourceLineNo">186</span><a id="line.186">                logger.info("Found non aa: {}", sequence);</a>
<span class="sourceLineNo">187</span><a id="line.187">                return false;</a>
<span class="sourceLineNo">188</span><a id="line.188">        }</a>
<span class="sourceLineNo">189</span><a id="line.189">        final Matcher protmatcher = SequenceUtil.AA.matcher(sequence);</a>
<span class="sourceLineNo">190</span><a id="line.190">        return protmatcher.find();</a>
<span class="sourceLineNo">191</span><a id="line.191">        }</a>
<span class="sourceLineNo">192</span><a id="line.192"></a>
<span class="sourceLineNo">193</span><a id="line.193">        /**</a>
<span class="sourceLineNo">194</span><a id="line.194">         * Check whether the sequence confirms to amboguous protein sequence</a>
<span class="sourceLineNo">195</span><a id="line.195">         *</a>
<span class="sourceLineNo">196</span><a id="line.196">         * @param sequence</a>
<span class="sourceLineNo">197</span><a id="line.197">         * @return return true only if the sequence if ambiguous protein sequence</a>
<span class="sourceLineNo">198</span><a id="line.198">         *         Return false otherwise. e.g. if the sequence is non-ambiguous</a>
<span class="sourceLineNo">199</span><a id="line.199">         *         protein or DNA</a>
<span class="sourceLineNo">200</span><a id="line.200">         */</a>
<span class="sourceLineNo">201</span><a id="line.201">        public static boolean isAmbiguosProtein(String sequence) {</a>
<span class="sourceLineNo">202</span><a id="line.202">        sequence = SequenceUtil.cleanSequence(sequence);</a>
<span class="sourceLineNo">203</span><a id="line.203">        if (SequenceUtil.isNonAmbNucleotideSequence(sequence)) {</a>
<span class="sourceLineNo">204</span><a id="line.204">                return false;</a>
<span class="sourceLineNo">205</span><a id="line.205">        }</a>
<span class="sourceLineNo">206</span><a id="line.206">        if (SequenceUtil.DIGIT.matcher(sequence).find()) {</a>
<span class="sourceLineNo">207</span><a id="line.207">                return false;</a>
<span class="sourceLineNo">208</span><a id="line.208">        }</a>
<span class="sourceLineNo">209</span><a id="line.209">        if (SequenceUtil.NON_AA.matcher(sequence).find()) {</a>
<span class="sourceLineNo">210</span><a id="line.210">                return false;</a>
<span class="sourceLineNo">211</span><a id="line.211">        }</a>
<span class="sourceLineNo">212</span><a id="line.212">        if (SequenceUtil.AA.matcher(sequence).find()) {</a>
<span class="sourceLineNo">213</span><a id="line.213">                return false;</a>
<span class="sourceLineNo">214</span><a id="line.214">        }</a>
<span class="sourceLineNo">215</span><a id="line.215">        final Matcher amb_prot = SequenceUtil.AMBIGUOUS_AA.matcher(sequence);</a>
<span class="sourceLineNo">216</span><a id="line.216">        return amb_prot.find();</a>
<span class="sourceLineNo">217</span><a id="line.217">        }</a>
<span class="sourceLineNo">218</span><a id="line.218"></a>
<span class="sourceLineNo">219</span><a id="line.219">        /**</a>
<span class="sourceLineNo">220</span><a id="line.220">         * Writes list of FastaSequeces into the outstream formatting the sequence</a>
<span class="sourceLineNo">221</span><a id="line.221">         * so that it contains width chars on each line</a>
<span class="sourceLineNo">222</span><a id="line.222">         *</a>
<span class="sourceLineNo">223</span><a id="line.223">         * @param outstream</a>
<span class="sourceLineNo">224</span><a id="line.224">         * @param sequences</a>
<span class="sourceLineNo">225</span><a id="line.225">         * @param width</a>
<span class="sourceLineNo">226</span><a id="line.226">         *            - the maximum number of characters to write in one line</a>
<span class="sourceLineNo">227</span><a id="line.227">         * @throws IOException</a>
<span class="sourceLineNo">228</span><a id="line.228">         */</a>
<span class="sourceLineNo">229</span><a id="line.229">        public static void writeFasta(final OutputStream outstream,</a>
<span class="sourceLineNo">230</span><a id="line.230">                final List&lt;FastaSequence&gt; sequences, final int width)</a>
<span class="sourceLineNo">231</span><a id="line.231">                throws IOException {</a>
<span class="sourceLineNo">232</span><a id="line.232">        final OutputStreamWriter writer = new OutputStreamWriter(outstream);</a>
<span class="sourceLineNo">233</span><a id="line.233">        final BufferedWriter fastawriter = new BufferedWriter(writer);</a>
<span class="sourceLineNo">234</span><a id="line.234">        for (final FastaSequence fs : sequences) {</a>
<span class="sourceLineNo">235</span><a id="line.235">                fastawriter.write(fs.getFormatedSequence(width));</a>
<span class="sourceLineNo">236</span><a id="line.236">        }</a>
<span class="sourceLineNo">237</span><a id="line.237">        outstream.flush();</a>
<span class="sourceLineNo">238</span><a id="line.238">        fastawriter.close();</a>
<span class="sourceLineNo">239</span><a id="line.239">        writer.close();</a>
<span class="sourceLineNo">240</span><a id="line.240">        }</a>
<span class="sourceLineNo">241</span><a id="line.241"></a>
<span class="sourceLineNo">242</span><a id="line.242">        /**</a>
<span class="sourceLineNo">243</span><a id="line.243">         * Reads fasta sequences from inStream into the list of FastaSequence</a>
<span class="sourceLineNo">244</span><a id="line.244">         * objects</a>
<span class="sourceLineNo">245</span><a id="line.245">         *</a>
<span class="sourceLineNo">246</span><a id="line.246">         * @param inStream</a>
<span class="sourceLineNo">247</span><a id="line.247">         *            from</a>
<span class="sourceLineNo">248</span><a id="line.248">         * @return list of FastaSequence objects</a>
<span class="sourceLineNo">249</span><a id="line.249">         * @throws IOException</a>
<span class="sourceLineNo">250</span><a id="line.250">         */</a>
<span class="sourceLineNo">251</span><a id="line.251">        public static List&lt;FastaSequence&gt; readFasta(final InputStream inStream)</a>
<span class="sourceLineNo">252</span><a id="line.252">                throws IOException {</a>
<span class="sourceLineNo">253</span><a id="line.253">        final List&lt;FastaSequence&gt; seqs = new ArrayList&lt;FastaSequence&gt;();</a>
<span class="sourceLineNo">254</span><a id="line.254"></a>
<span class="sourceLineNo">255</span><a id="line.255">        final BufferedReader infasta = new BufferedReader(</a>
<span class="sourceLineNo">256</span><a id="line.256">                new InputStreamReader(inStream, "UTF8"), 16000);</a>
<span class="sourceLineNo">257</span><a id="line.257">        final Pattern pattern = Pattern.compile("//s+");</a>
<span class="sourceLineNo">258</span><a id="line.258"></a>
<span class="sourceLineNo">259</span><a id="line.259">        String line;</a>
<span class="sourceLineNo">260</span><a id="line.260">        String sname = "", seqstr = null;</a>
<span class="sourceLineNo">261</span><a id="line.261">        do {</a>
<span class="sourceLineNo">262</span><a id="line.262">                line = infasta.readLine();</a>
<span class="sourceLineNo">263</span><a id="line.263">                if ((line == null) || line.startsWith("&gt;")) {</a>
<span class="sourceLineNo">264</span><a id="line.264">                if (seqstr != null) {</a>
<span class="sourceLineNo">265</span><a id="line.265">                        seqs.add(new FastaSequence(sname.substring(1), seqstr));</a>
<span class="sourceLineNo">266</span><a id="line.266">                }</a>
<span class="sourceLineNo">267</span><a id="line.267">                sname = line; // remove &gt;</a>
<span class="sourceLineNo">268</span><a id="line.268">                seqstr = "";</a>
<span class="sourceLineNo">269</span><a id="line.269">                } else {</a>
<span class="sourceLineNo">270</span><a id="line.270">                final String subseq = pattern.matcher(line).replaceAll("");</a>
<span class="sourceLineNo">271</span><a id="line.271">                seqstr += subseq;</a>
<span class="sourceLineNo">272</span><a id="line.272">                }</a>
<span class="sourceLineNo">273</span><a id="line.273">        } while (line != null);</a>
<span class="sourceLineNo">274</span><a id="line.274"></a>
<span class="sourceLineNo">275</span><a id="line.275">        infasta.close();</a>
<span class="sourceLineNo">276</span><a id="line.276">        return seqs;</a>
<span class="sourceLineNo">277</span><a id="line.277">        }</a>
<span class="sourceLineNo">278</span><a id="line.278"></a>
<span class="sourceLineNo">279</span><a id="line.279">        /**</a>
<span class="sourceLineNo">280</span><a id="line.280">         * Writes FastaSequence in the file, each sequence will take one line only</a>
<span class="sourceLineNo">281</span><a id="line.281">         *</a>
<span class="sourceLineNo">282</span><a id="line.282">         * @param os</a>
<span class="sourceLineNo">283</span><a id="line.283">         * @param sequences</a>
<span class="sourceLineNo">284</span><a id="line.284">         * @throws IOException</a>
<span class="sourceLineNo">285</span><a id="line.285">         */</a>
<span class="sourceLineNo">286</span><a id="line.286">        public static void writeFasta(final OutputStream os,</a>
<span class="sourceLineNo">287</span><a id="line.287">                final List&lt;FastaSequence&gt; sequences) throws IOException {</a>
<span class="sourceLineNo">288</span><a id="line.288">        final OutputStreamWriter outWriter = new OutputStreamWriter(os);</a>
<span class="sourceLineNo">289</span><a id="line.289">        final BufferedWriter fasta_out = new BufferedWriter(outWriter);</a>
<span class="sourceLineNo">290</span><a id="line.290">        for (final FastaSequence fs : sequences) {</a>
<span class="sourceLineNo">291</span><a id="line.291">                fasta_out.write(fs.getOnelineFasta());</a>
<span class="sourceLineNo">292</span><a id="line.292">        }</a>
<span class="sourceLineNo">293</span><a id="line.293">        fasta_out.close();</a>
<span class="sourceLineNo">294</span><a id="line.294">        outWriter.close();</a>
<span class="sourceLineNo">295</span><a id="line.295">        }</a>
<span class="sourceLineNo">296</span><a id="line.296"></a>
<span class="sourceLineNo">297</span><a id="line.297">}</a>




























































</pre>
</div>
</main>
</body>
</html>
