<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.16) on Fri Oct 28 16:15:48 PDT 2022 -->
<title>Uses of Package org.biojava.nbio.core.sequence.features (biojava 6.1.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2022-10-28">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Package org.biojava.nbio.core.sequence.features (biojava 6.1.0 API)";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h1 title="Uses of Package org.biojava.nbio.core.sequence.features" class="title">Uses of Package<br>org.biojava.nbio.core.sequence.features</h1>
</div>
<div class="contentContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.features">org.biojava.nbio.core.sequence.features</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.loader">org.biojava.nbio.core.sequence.loader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome.io.fastq">org.biojava.nbio.genome.io.fastq</a></th>
<td class="colLast">
<div class="block">FASTQ and variants sequence format I/O.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.sequence.features">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/AbstractFeature.html#org.biojava.nbio.core.sequence.features">AbstractFeature</a></th>
<td class="colLast">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/DBReferenceInfo.html#org.biojava.nbio.core.sequence.features">DBReferenceInfo</a></th>
<td class="colLast">
<div class="block">If you have a uniprot ID then it is possible to get a collection
 of other id(s) that the protein is known by.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/FeatureInterface.html#org.biojava.nbio.core.sequence.features">FeatureInterface</a></th>
<td class="colLast">
<div class="block">Interface class to handle describing arbitrary features.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/Qualifier.html#org.biojava.nbio.core.sequence.features">Qualifier</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.sequence.io">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../io/package-summary.html">org.biojava.nbio.core.sequence.io</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/AbstractFeature.html#org.biojava.nbio.core.sequence.io">AbstractFeature</a></th>
<td class="colLast">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/DBReferenceInfo.html#org.biojava.nbio.core.sequence.io">DBReferenceInfo</a></th>
<td class="colLast">
<div class="block">If you have a uniprot ID then it is possible to get a collection
 of other id(s) that the protein is known by.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/FeatureInterface.html#org.biojava.nbio.core.sequence.io">FeatureInterface</a></th>
<td class="colLast">
<div class="block">Interface class to handle describing arbitrary features.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.sequence.loader">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/AbstractFeature.html#org.biojava.nbio.core.sequence.loader">AbstractFeature</a></th>
<td class="colLast">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/DatabaseReferenceInterface.html#org.biojava.nbio.core.sequence.loader">DatabaseReferenceInterface</a></th>
<td class="colLast">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list of cross reference id(s)</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/DBReferenceInfo.html#org.biojava.nbio.core.sequence.loader">DBReferenceInfo</a></th>
<td class="colLast">
<div class="block">If you have a uniprot ID then it is possible to get a collection
 of other id(s) that the protein is known by.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/FeatureRetriever.html#org.biojava.nbio.core.sequence.loader">FeatureRetriever</a></th>
<td class="colLast">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list features</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/FeaturesKeyWordInterface.html#org.biojava.nbio.core.sequence.loader">FeaturesKeyWordInterface</a></th>
<td class="colLast">
<div class="block">Models the keywords that are annotated for a protein sequence at Uniprot.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.core.sequence.template">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../template/package-summary.html">org.biojava.nbio.core.sequence.template</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/DatabaseReferenceInterface.html#org.biojava.nbio.core.sequence.template">DatabaseReferenceInterface</a></th>
<td class="colLast">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list of cross reference id(s)</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/FeatureInterface.html#org.biojava.nbio.core.sequence.template">FeatureInterface</a></th>
<td class="colLast">
<div class="block">Interface class to handle describing arbitrary features.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/FeatureRetriever.html#org.biojava.nbio.core.sequence.template">FeatureRetriever</a></th>
<td class="colLast">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list features</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/FeaturesKeyWordInterface.html#org.biojava.nbio.core.sequence.template">FeaturesKeyWordInterface</a></th>
<td class="colLast">
<div class="block">Models the keywords that are annotated for a protein sequence at Uniprot.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a id="org.biojava.nbio.genome.io.fastq">
<!--   -->
</a>
<table class="useSummary">
<caption><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../../../genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="class-use/QualityFeature.html#org.biojava.nbio.genome.io.fastq">QualityFeature</a></th>
<td class="colLast">
<div class="block">DNA Sequences produced by modern sequencers usually have quality informaion
 attached to them.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="class-use/QuantityFeature.html#org.biojava.nbio.genome.io.fastq">QuantityFeature</a></th>
<td class="colLast">
<div class="block">It is common to have a numerical value or values associated with a feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2022 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
