<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> */</a>
<span class="sourceLineNo">021</span><a id="line.21">package demo;</a>
<span class="sourceLineNo">022</span><a id="line.22"></a>
<span class="sourceLineNo">023</span><a id="line.23">import java.io.InputStream;</a>
<span class="sourceLineNo">024</span><a id="line.24">import java.util.LinkedHashMap;</a>
<span class="sourceLineNo">025</span><a id="line.25"></a>
<span class="sourceLineNo">026</span><a id="line.26">import org.biojava.nbio.core.alignment.matrices.SubstitutionMatrixHelper;</a>
<span class="sourceLineNo">027</span><a id="line.27">import org.biojava.nbio.core.alignment.template.SubstitutionMatrix;</a>
<span class="sourceLineNo">028</span><a id="line.28">import org.biojava.nbio.core.sequence.MultipleSequenceAlignment;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.biojava.nbio.core.sequence.ProteinSequence;</a>
<span class="sourceLineNo">030</span><a id="line.30">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</a>
<span class="sourceLineNo">031</span><a id="line.31">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</a>
<span class="sourceLineNo">032</span><a id="line.32">import org.biojava.nbio.core.sequence.io.FastaReader;</a>
<span class="sourceLineNo">033</span><a id="line.33">import org.biojava.nbio.core.sequence.io.GenericFastaHeaderParser;</a>
<span class="sourceLineNo">034</span><a id="line.34">import org.biojava.nbio.core.sequence.io.ProteinSequenceCreator;</a>
<span class="sourceLineNo">035</span><a id="line.35">import org.biojava.nbio.phylo.DistanceMatrixCalculator;</a>
<span class="sourceLineNo">036</span><a id="line.36">import org.biojava.nbio.phylo.DistanceTreeEvaluator;</a>
<span class="sourceLineNo">037</span><a id="line.37">import org.biojava.nbio.phylo.ForesterWrapper;</a>
<span class="sourceLineNo">038</span><a id="line.38">import org.biojava.nbio.phylo.TreeConstructor;</a>
<span class="sourceLineNo">039</span><a id="line.39">import org.biojava.nbio.phylo.TreeConstructorType;</a>
<span class="sourceLineNo">040</span><a id="line.40">import org.forester.evoinference.matrix.distance.BasicSymmetricalDistanceMatrix;</a>
<span class="sourceLineNo">041</span><a id="line.41">import org.forester.evoinference.matrix.distance.DistanceMatrix;</a>
<span class="sourceLineNo">042</span><a id="line.42">import org.forester.phylogeny.Phylogeny;</a>
<span class="sourceLineNo">043</span><a id="line.43"></a>
<span class="sourceLineNo">044</span><a id="line.44">/**</a>
<span class="sourceLineNo">045</span><a id="line.45"> * This demo contains the CookBook example to create a phylogenetic tree from a</a>
<span class="sourceLineNo">046</span><a id="line.46"> * multiple sequence alignment (MSA).</a>
<span class="sourceLineNo">047</span><a id="line.47"> *</a>
<span class="sourceLineNo">048</span><a id="line.48"> * @author Scooter Willis</a>
<span class="sourceLineNo">049</span><a id="line.49"> * @author Aleix Lafita</a>
<span class="sourceLineNo">050</span><a id="line.50"> *</a>
<span class="sourceLineNo">051</span><a id="line.51"> */</a>
<span class="sourceLineNo">052</span><a id="line.52">public class DemoDistanceTree {</a>
<span class="sourceLineNo">053</span><a id="line.53"></a>
<span class="sourceLineNo">054</span><a id="line.54">        public static void main(String[] args) throws Exception {</a>
<span class="sourceLineNo">055</span><a id="line.55"></a>
<span class="sourceLineNo">056</span><a id="line.56">                // 0. This is just to load an example MSA from a FASTA file</a>
<span class="sourceLineNo">057</span><a id="line.57">                InputStream inStream = TreeConstructor.class</a>
<span class="sourceLineNo">058</span><a id="line.58">                                .getResourceAsStream("/PF00104_small.fasta");</a>
<span class="sourceLineNo">059</span><a id="line.59"></a>
<span class="sourceLineNo">060</span><a id="line.60">                FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; fastaReader =</a>
<span class="sourceLineNo">061</span><a id="line.61">                                new FastaReader&lt;ProteinSequence, AminoAcidCompound&gt;(</a>
<span class="sourceLineNo">062</span><a id="line.62">                                inStream,</a>
<span class="sourceLineNo">063</span><a id="line.63">                                new GenericFastaHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),</a>
<span class="sourceLineNo">064</span><a id="line.64">                                new ProteinSequenceCreator(AminoAcidCompoundSet</a>
<span class="sourceLineNo">065</span><a id="line.65">                                                .getAminoAcidCompoundSet()));</a>
<span class="sourceLineNo">066</span><a id="line.66"></a>
<span class="sourceLineNo">067</span><a id="line.67">                LinkedHashMap&lt;String, ProteinSequence&gt; proteinSequences =</a>
<span class="sourceLineNo">068</span><a id="line.68">                                fastaReader.process();</a>
<span class="sourceLineNo">069</span><a id="line.69"></a>
<span class="sourceLineNo">070</span><a id="line.70">                inStream.close();</a>
<span class="sourceLineNo">071</span><a id="line.71"></a>
<span class="sourceLineNo">072</span><a id="line.72">                MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; msa =</a>
<span class="sourceLineNo">073</span><a id="line.73">                                new MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt;();</a>
<span class="sourceLineNo">074</span><a id="line.74"></a>
<span class="sourceLineNo">075</span><a id="line.75">                for (ProteinSequence proteinSequence : proteinSequences.values()) {</a>
<span class="sourceLineNo">076</span><a id="line.76">                        msa.addAlignedSequence(proteinSequence);</a>
<span class="sourceLineNo">077</span><a id="line.77">                }</a>
<span class="sourceLineNo">078</span><a id="line.78"></a>
<span class="sourceLineNo">079</span><a id="line.79">                long readT = System.currentTimeMillis();</a>
<span class="sourceLineNo">080</span><a id="line.80"></a>
<span class="sourceLineNo">081</span><a id="line.81">                // 1. Calculate the evolutionary distance matrix (can take long)</a>
<span class="sourceLineNo">082</span><a id="line.82">                SubstitutionMatrix&lt;AminoAcidCompound&gt; M = SubstitutionMatrixHelper</a>
<span class="sourceLineNo">083</span><a id="line.83">                                .getBlosum62();</a>
<span class="sourceLineNo">084</span><a id="line.84">                DistanceMatrix DM = DistanceMatrixCalculator</a>
<span class="sourceLineNo">085</span><a id="line.85">                                .dissimilarityScore(msa, M);</a>
<span class="sourceLineNo">086</span><a id="line.86"></a>
<span class="sourceLineNo">087</span><a id="line.87">                // 2. Construct a distance tree using the NJ algorithm</a>
<span class="sourceLineNo">088</span><a id="line.88">                Phylogeny phylo = TreeConstructor.distanceTree(</a>
<span class="sourceLineNo">089</span><a id="line.89">                                (BasicSymmetricalDistanceMatrix) DM, TreeConstructorType.NJ);</a>
<span class="sourceLineNo">090</span><a id="line.90"></a>
<span class="sourceLineNo">091</span><a id="line.91">                long treeT = System.currentTimeMillis();</a>
<span class="sourceLineNo">092</span><a id="line.92">                String newick = ForesterWrapper.getNewickString(phylo, true);</a>
<span class="sourceLineNo">093</span><a id="line.93">                System.out.println(newick);</a>
<span class="sourceLineNo">094</span><a id="line.94">                System.out.println("Tree Construction: " + (treeT - readT) + " ms.");</a>
<span class="sourceLineNo">095</span><a id="line.95"></a>
<span class="sourceLineNo">096</span><a id="line.96">                // 3. Evaluate the goodness of fit of the tree</a>
<span class="sourceLineNo">097</span><a id="line.97">                double cv = DistanceTreeEvaluator.evaluate(phylo, DM);</a>
<span class="sourceLineNo">098</span><a id="line.98">                System.out.println("CV of the tree: " + (int) (cv * 100) + " %");</a>
<span class="sourceLineNo">099</span><a id="line.99"></a>
<span class="sourceLineNo">100</span><a id="line.100">        }</a>
<span class="sourceLineNo">101</span><a id="line.101">}</a>




























































</pre>
</div>
</main>
</body>
</html>
