<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> */</a>
<span class="sourceLineNo">021</span><a id="line.21">package org.biojava.nbio.structure.align.multiple.util;</a>
<span class="sourceLineNo">022</span><a id="line.22"></a>
<span class="sourceLineNo">023</span><a id="line.23">import java.io.IOException;</a>
<span class="sourceLineNo">024</span><a id="line.24">import java.util.ArrayList;</a>
<span class="sourceLineNo">025</span><a id="line.25">import java.util.Arrays;</a>
<span class="sourceLineNo">026</span><a id="line.26">import java.util.Collections;</a>
<span class="sourceLineNo">027</span><a id="line.27">import java.util.Comparator;</a>
<span class="sourceLineNo">028</span><a id="line.28">import java.util.HashMap;</a>
<span class="sourceLineNo">029</span><a id="line.29">import java.util.List;</a>
<span class="sourceLineNo">030</span><a id="line.30">import java.util.Map;</a>
<span class="sourceLineNo">031</span><a id="line.31">import java.util.SortedSet;</a>
<span class="sourceLineNo">032</span><a id="line.32">import java.util.TreeSet;</a>
<span class="sourceLineNo">033</span><a id="line.33"></a>
<span class="sourceLineNo">034</span><a id="line.34">import javax.vecmath.Matrix4d;</a>
<span class="sourceLineNo">035</span><a id="line.35"></a>
<span class="sourceLineNo">036</span><a id="line.36">import org.biojava.nbio.core.alignment.matrices.SubstitutionMatrixHelper;</a>
<span class="sourceLineNo">037</span><a id="line.37">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</a>
<span class="sourceLineNo">038</span><a id="line.38">import org.biojava.nbio.core.sequence.AccessionID;</a>
<span class="sourceLineNo">039</span><a id="line.39">import org.biojava.nbio.core.sequence.MultipleSequenceAlignment;</a>
<span class="sourceLineNo">040</span><a id="line.40">import org.biojava.nbio.core.sequence.ProteinSequence;</a>
<span class="sourceLineNo">041</span><a id="line.41">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</a>
<span class="sourceLineNo">042</span><a id="line.42">import org.biojava.nbio.phylo.DistanceMatrixCalculator;</a>
<span class="sourceLineNo">043</span><a id="line.43">import org.biojava.nbio.phylo.TreeConstructor;</a>
<span class="sourceLineNo">044</span><a id="line.44">import org.biojava.nbio.phylo.TreeConstructorType;</a>
<span class="sourceLineNo">045</span><a id="line.45">import org.biojava.nbio.structure.AminoAcid;</a>
<span class="sourceLineNo">046</span><a id="line.46">import org.biojava.nbio.structure.Atom;</a>
<span class="sourceLineNo">047</span><a id="line.47">import org.biojava.nbio.structure.Calc;</a>
<span class="sourceLineNo">048</span><a id="line.48">import org.biojava.nbio.structure.Chain;</a>
<span class="sourceLineNo">049</span><a id="line.49">import org.biojava.nbio.structure.Group;</a>
<span class="sourceLineNo">050</span><a id="line.50">import org.biojava.nbio.structure.PDBHeader;</a>
<span class="sourceLineNo">051</span><a id="line.51">import org.biojava.nbio.structure.Structure;</a>
<span class="sourceLineNo">052</span><a id="line.52">import org.biojava.nbio.structure.StructureException;</a>
<span class="sourceLineNo">053</span><a id="line.53">import org.biojava.nbio.structure.StructureIdentifier;</a>
<span class="sourceLineNo">054</span><a id="line.54">import org.biojava.nbio.structure.StructureImpl;</a>
<span class="sourceLineNo">055</span><a id="line.55">import org.biojava.nbio.structure.StructureTools;</a>
<span class="sourceLineNo">056</span><a id="line.56">import org.biojava.nbio.structure.align.multiple.Block;</a>
<span class="sourceLineNo">057</span><a id="line.57">import org.biojava.nbio.structure.align.multiple.BlockSet;</a>
<span class="sourceLineNo">058</span><a id="line.58">import org.biojava.nbio.structure.align.multiple.MultipleAlignment;</a>
<span class="sourceLineNo">059</span><a id="line.59">import org.biojava.nbio.structure.align.util.AlignmentTools;</a>
<span class="sourceLineNo">060</span><a id="line.60">import org.biojava.nbio.structure.jama.Matrix;</a>
<span class="sourceLineNo">061</span><a id="line.61">import org.forester.evoinference.matrix.distance.BasicSymmetricalDistanceMatrix;</a>
<span class="sourceLineNo">062</span><a id="line.62">import org.forester.phylogeny.Phylogeny;</a>
<span class="sourceLineNo">063</span><a id="line.63"></a>
<span class="sourceLineNo">064</span><a id="line.64">/**</a>
<span class="sourceLineNo">065</span><a id="line.65"> * Utility functions for working with {@link MultipleAlignment}.</a>
<span class="sourceLineNo">066</span><a id="line.66"> * &lt;p&gt;</a>
<span class="sourceLineNo">067</span><a id="line.67"> * Supported functions:</a>
<span class="sourceLineNo">068</span><a id="line.68"> * &lt;ul&gt;</a>
<span class="sourceLineNo">069</span><a id="line.69"> * &lt;li&gt;Obtain the alignment as sequence strings</a>
<span class="sourceLineNo">070</span><a id="line.70"> * &lt;li&gt;Map from sequence alignment position to structure Atom</a>
<span class="sourceLineNo">071</span><a id="line.71"> * &lt;li&gt;Map from sequence alignment position to Block number</a>
<span class="sourceLineNo">072</span><a id="line.72"> * &lt;li&gt;Transform the aligned Atoms of a MultipleAlignment</a>
<span class="sourceLineNo">073</span><a id="line.73"> * &lt;li&gt;Get all the core alignment positions of the alignment</a>
<span class="sourceLineNo">074</span><a id="line.74"> * &lt;li&gt;Calculate the average residue distance of all aligned positions</a>
<span class="sourceLineNo">075</span><a id="line.75"> * &lt;li&gt;Sort Blocks in a MultipleAlignment by a specified row</a>
<span class="sourceLineNo">076</span><a id="line.76"> * &lt;li&gt;Convert a MultipleAlignment to a MultipleSequenceAlignment</a>
<span class="sourceLineNo">077</span><a id="line.77"> * &lt;/ul&gt;</a>
<span class="sourceLineNo">078</span><a id="line.78"> *</a>
<span class="sourceLineNo">079</span><a id="line.79"> * @author Spencer Bliven</a>
<span class="sourceLineNo">080</span><a id="line.80"> * @author Aleix Lafita</a>
<span class="sourceLineNo">081</span><a id="line.81"> * @since 4.1.0</a>
<span class="sourceLineNo">082</span><a id="line.82"> *</a>
<span class="sourceLineNo">083</span><a id="line.83"> */</a>
<span class="sourceLineNo">084</span><a id="line.84">public class MultipleAlignmentTools {</a>
<span class="sourceLineNo">085</span><a id="line.85"></a>
<span class="sourceLineNo">086</span><a id="line.86">        /**</a>
<span class="sourceLineNo">087</span><a id="line.87">         * Calculate the sequence alignment Strings for the whole alignment. This</a>
<span class="sourceLineNo">088</span><a id="line.88">         * method creates a sequence alignment where aligned residues are in</a>
<span class="sourceLineNo">089</span><a id="line.89">         * uppercase and unaligned residues are in lowercase, thus providing a more</a>
<span class="sourceLineNo">090</span><a id="line.90">         * compact way to represent the alignment.</a>
<span class="sourceLineNo">091</span><a id="line.91">         * &lt;p&gt;</a>
<span class="sourceLineNo">092</span><a id="line.92">         * Blocks are concatenated in the order returned by</a>
<span class="sourceLineNo">093</span><a id="line.93">         * {@link MultipleAlignment#getBlocks()}, so sequences may not be</a>
<span class="sourceLineNo">094</span><a id="line.94">         * sequential. Gaps are represented by '-'. Separation between different</a>
<span class="sourceLineNo">095</span><a id="line.95">         * Blocks is indicated by a gap in all positions, meaning that there is a</a>
<span class="sourceLineNo">096</span><a id="line.96">         * possible discontinuity.</a>
<span class="sourceLineNo">097</span><a id="line.97">         *</a>
<span class="sourceLineNo">098</span><a id="line.98">         * @param alignment</a>
<span class="sourceLineNo">099</span><a id="line.99">         *            input MultipleAlignment</a>
<span class="sourceLineNo">100</span><a id="line.100">         * @param mapSeqToStruct</a>
<span class="sourceLineNo">101</span><a id="line.101">         *            provides a link from the sequence alignment position to the</a>
<span class="sourceLineNo">102</span><a id="line.102">         *            structure alignment position. Specially designed for the GUI.</a>
<span class="sourceLineNo">103</span><a id="line.103">         *            Has to be initialized previously and will be overwritten.</a>
<span class="sourceLineNo">104</span><a id="line.104">         * @return a string for each row in the alignment, giving the 1-letter code</a>
<span class="sourceLineNo">105</span><a id="line.105">         *         for each aligned residue.</a>
<span class="sourceLineNo">106</span><a id="line.106">         */</a>
<span class="sourceLineNo">107</span><a id="line.107">        public static List&lt;String&gt; getSequenceAlignment(</a>
<span class="sourceLineNo">108</span><a id="line.108">                        MultipleAlignment alignment, final List&lt;Integer&gt; mapSeqToStruct) {</a>
<span class="sourceLineNo">109</span><a id="line.109"></a>
<span class="sourceLineNo">110</span><a id="line.110">                // Initialize sequence variables</a>
<span class="sourceLineNo">111</span><a id="line.111">                List&lt;String&gt; alnSequences = new ArrayList&lt;String&gt;();</a>
<span class="sourceLineNo">112</span><a id="line.112">                for (int str = 0; str &lt; alignment.size(); str++)</a>
<span class="sourceLineNo">113</span><a id="line.113">                        alnSequences.add("");</a>
<span class="sourceLineNo">114</span><a id="line.114">                mapSeqToStruct.clear();</a>
<span class="sourceLineNo">115</span><a id="line.115">                List&lt;Atom[]&gt; atoms = alignment.getAtomArrays();</a>
<span class="sourceLineNo">116</span><a id="line.116">                int globalPos = -1;</a>
<span class="sourceLineNo">117</span><a id="line.117"></a>
<span class="sourceLineNo">118</span><a id="line.118">                // Initialize helper variables in constucting the sequence alignment</a>
<span class="sourceLineNo">119</span><a id="line.119">                List&lt;SortedSet&lt;Integer&gt;&gt; freePool = new ArrayList&lt;SortedSet&lt;Integer&gt;&gt;();</a>
<span class="sourceLineNo">120</span><a id="line.120">                List&lt;SortedSet&lt;Integer&gt;&gt; blockStarts = new ArrayList&lt;SortedSet&lt;Integer&gt;&gt;();</a>
<span class="sourceLineNo">121</span><a id="line.121">                List&lt;List&lt;Integer&gt;&gt; aligned = new ArrayList&lt;List&lt;Integer&gt;&gt;();</a>
<span class="sourceLineNo">122</span><a id="line.122"></a>
<span class="sourceLineNo">123</span><a id="line.123">                // Generate freePool residues from the ones not aligned</a>
<span class="sourceLineNo">124</span><a id="line.124">                for (int i = 0; i &lt; alignment.size(); i++) {</a>
<span class="sourceLineNo">125</span><a id="line.125">                        List&lt;Integer&gt; residues = new ArrayList&lt;Integer&gt;();</a>
<span class="sourceLineNo">126</span><a id="line.126">                        freePool.add(new TreeSet&lt;Integer&gt;());</a>
<span class="sourceLineNo">127</span><a id="line.127">                        blockStarts.add(new TreeSet&lt;Integer&gt;());</a>
<span class="sourceLineNo">128</span><a id="line.128">                        for (BlockSet bs : alignment.getBlockSets()) {</a>
<span class="sourceLineNo">129</span><a id="line.129">                                for (Block b : bs.getBlocks()) {</a>
<span class="sourceLineNo">130</span><a id="line.130">                                        boolean first = true;</a>
<span class="sourceLineNo">131</span><a id="line.131">                                        for (int l = 0; l &lt; b.length(); l++) {</a>
<span class="sourceLineNo">132</span><a id="line.132">                                                Integer residue = b.getAlignRes().get(i).get(l);</a>
<span class="sourceLineNo">133</span><a id="line.133">                                                if (residue != null) {</a>
<span class="sourceLineNo">134</span><a id="line.134">                                                        if (first)</a>
<span class="sourceLineNo">135</span><a id="line.135">                                                                blockStarts.get(i).add(residue);</a>
<span class="sourceLineNo">136</span><a id="line.136">                                                        residues.add(residue);</a>
<span class="sourceLineNo">137</span><a id="line.137">                                                        first = false;</a>
<span class="sourceLineNo">138</span><a id="line.138">                                                }</a>
<span class="sourceLineNo">139</span><a id="line.139">                                        }</a>
<span class="sourceLineNo">140</span><a id="line.140">                                }</a>
<span class="sourceLineNo">141</span><a id="line.141">                        }</a>
<span class="sourceLineNo">142</span><a id="line.142">                        aligned.add(residues);</a>
<span class="sourceLineNo">143</span><a id="line.143">                }</a>
<span class="sourceLineNo">144</span><a id="line.144">                // Add any residue not aligned to the free pool for every structure</a>
<span class="sourceLineNo">145</span><a id="line.145">                for (int i = 0; i &lt; alignment.size(); i++) {</a>
<span class="sourceLineNo">146</span><a id="line.146">                        for (int k = 0; k &lt; atoms.get(i).length; k++) {</a>
<span class="sourceLineNo">147</span><a id="line.147">                                if (!aligned.get(i).contains(k))</a>
<span class="sourceLineNo">148</span><a id="line.148">                                        freePool.get(i).add(k);</a>
<span class="sourceLineNo">149</span><a id="line.149">                        }</a>
<span class="sourceLineNo">150</span><a id="line.150">                }</a>
<span class="sourceLineNo">151</span><a id="line.151"></a>
<span class="sourceLineNo">152</span><a id="line.152">                for (int b = 0; b &lt; alignment.getBlocks().size(); b++) {</a>
<span class="sourceLineNo">153</span><a id="line.153">                        if (b != 0) {</a>
<span class="sourceLineNo">154</span><a id="line.154">                                // Add a gap to all structures to separate visually the Blocks</a>
<span class="sourceLineNo">155</span><a id="line.155">                                for (int str = 0; str &lt; alignment.size(); str++)</a>
<span class="sourceLineNo">156</span><a id="line.156">                                        alnSequences.set(str, alnSequences.get(str).concat("-"));</a>
<span class="sourceLineNo">157</span><a id="line.157">                                mapSeqToStruct.add(-1); // means unaligned position</a>
<span class="sourceLineNo">158</span><a id="line.158">                        }</a>
<span class="sourceLineNo">159</span><a id="line.159">                        // Store the previous position added to the sequence alignment</a>
<span class="sourceLineNo">160</span><a id="line.160">                        int[] previousPos = new int[alignment.size()];</a>
<span class="sourceLineNo">161</span><a id="line.161">                        Arrays.fill(previousPos, -1);</a>
<span class="sourceLineNo">162</span><a id="line.162">                        // Store provisional characters</a>
<span class="sourceLineNo">163</span><a id="line.163">                        char[] provisionalChar = new char[alignment.size()];</a>
<span class="sourceLineNo">164</span><a id="line.164">                        Arrays.fill(provisionalChar, '-');</a>
<span class="sourceLineNo">165</span><a id="line.165"></a>
<span class="sourceLineNo">166</span><a id="line.166">                        for (int pos = 0; pos &lt; alignment.getBlocks().get(b).length(); pos++) {</a>
<span class="sourceLineNo">167</span><a id="line.167">                                globalPos++;</a>
<span class="sourceLineNo">168</span><a id="line.168">                                boolean gaps = true; // true if consecutive with the previous</a>
<span class="sourceLineNo">169</span><a id="line.169">                                while (gaps) {</a>
<span class="sourceLineNo">170</span><a id="line.170">                                        gaps = false;</a>
<span class="sourceLineNo">171</span><a id="line.171">                                        // Loop through all the structures</a>
<span class="sourceLineNo">172</span><a id="line.172">                                        for (int str = 0; str &lt; alignment.size(); str++) {</a>
<span class="sourceLineNo">173</span><a id="line.173">                                                // If it is the first position or before it was null</a>
<span class="sourceLineNo">174</span><a id="line.174">                                                if (previousPos[str] == -1) {</a>
<span class="sourceLineNo">175</span><a id="line.175">                                                        Integer residue = alignment.getBlocks().get(b)</a>
<span class="sourceLineNo">176</span><a id="line.176">                                                                        .getAlignRes().get(str).get(pos);</a>
<span class="sourceLineNo">177</span><a id="line.177">                                                        if (residue == null)</a>
<span class="sourceLineNo">178</span><a id="line.178">                                                                provisionalChar[str] = '-';</a>
<span class="sourceLineNo">179</span><a id="line.179">                                                        else {</a>
<span class="sourceLineNo">180</span><a id="line.180">                                                                Atom a = atoms.get(str)[residue];</a>
<span class="sourceLineNo">181</span><a id="line.181">                                                                String group = a.getGroup().getPDBName();</a>
<span class="sourceLineNo">182</span><a id="line.182">                                                                provisionalChar[str] = StructureTools</a>
<span class="sourceLineNo">183</span><a id="line.183">                                                                                .get1LetterCode(group);</a>
<span class="sourceLineNo">184</span><a id="line.184">                                                        }</a>
<span class="sourceLineNo">185</span><a id="line.185">                                                } else {</a>
<span class="sourceLineNo">186</span><a id="line.186">                                                        Integer residue = alignment.getBlocks().get(b)</a>
<span class="sourceLineNo">187</span><a id="line.187">                                                                        .getAlignRes().get(str).get(pos);</a>
<span class="sourceLineNo">188</span><a id="line.188">                                                        int nextPos = previousPos[str] + 1;</a>
<span class="sourceLineNo">189</span><a id="line.189">                                                        if (residue == null) {</a>
<span class="sourceLineNo">190</span><a id="line.190">                                                                if (freePool.get(str).contains(nextPos)) {</a>
<span class="sourceLineNo">191</span><a id="line.191">                                                                        Atom a = atoms.get(str)[nextPos];</a>
<span class="sourceLineNo">192</span><a id="line.192">                                                                        String g = a.getGroup().getPDBName();</a>
<span class="sourceLineNo">193</span><a id="line.193">                                                                        char aa = StructureTools.get1LetterCode(g);</a>
<span class="sourceLineNo">194</span><a id="line.194">                                                                        provisionalChar[str] = Character</a>
<span class="sourceLineNo">195</span><a id="line.195">                                                                                        .toLowerCase(aa);</a>
<span class="sourceLineNo">196</span><a id="line.196">                                                                } else</a>
<span class="sourceLineNo">197</span><a id="line.197">                                                                        provisionalChar[str] = '-';</a>
<span class="sourceLineNo">198</span><a id="line.198">                                                        } else if (nextPos == residue) {</a>
<span class="sourceLineNo">199</span><a id="line.199">                                                                Atom a = atoms.get(str)[nextPos];</a>
<span class="sourceLineNo">200</span><a id="line.200">                                                                String group = a.getGroup().getPDBName();</a>
<span class="sourceLineNo">201</span><a id="line.201">                                                                provisionalChar[str] = StructureTools</a>
<span class="sourceLineNo">202</span><a id="line.202">                                                                                .get1LetterCode(group);</a>
<span class="sourceLineNo">203</span><a id="line.203">                                                        } else {</a>
<span class="sourceLineNo">204</span><a id="line.204">                                                                // This means non-consecutive</a>
<span class="sourceLineNo">205</span><a id="line.205">                                                                provisionalChar[str] = ' ';</a>
<span class="sourceLineNo">206</span><a id="line.206">                                                                gaps = true;</a>
<span class="sourceLineNo">207</span><a id="line.207">                                                        }</a>
<span class="sourceLineNo">208</span><a id="line.208">                                                }</a>
<span class="sourceLineNo">209</span><a id="line.209">                                        }// End all structure analysis</a>
<span class="sourceLineNo">210</span><a id="line.210"></a>
<span class="sourceLineNo">211</span><a id="line.211">                                        if (gaps) {</a>
<span class="sourceLineNo">212</span><a id="line.212">                                                for (int str = 0; str &lt; alignment.size(); str++) {</a>
<span class="sourceLineNo">213</span><a id="line.213">                                                        if (provisionalChar[str] == ' ') {</a>
<span class="sourceLineNo">214</span><a id="line.214">                                                                // It means this residue was non-consecutive</a>
<span class="sourceLineNo">215</span><a id="line.215">                                                                Atom a = atoms.get(str)[previousPos[str] + 1];</a>
<span class="sourceLineNo">216</span><a id="line.216">                                                                String group = a.getGroup().getPDBName();</a>
<span class="sourceLineNo">217</span><a id="line.217">                                                                char aa = StructureTools.get1LetterCode(group);</a>
<span class="sourceLineNo">218</span><a id="line.218">                                                                alnSequences</a>
<span class="sourceLineNo">219</span><a id="line.219">                                                                                .set(str,</a>
<span class="sourceLineNo">220</span><a id="line.220">                                                                                                alnSequences</a>
<span class="sourceLineNo">221</span><a id="line.221">                                                                                                                .get(str)</a>
<span class="sourceLineNo">222</span><a id="line.222">                                                                                                                .concat(String.valueOf(Character</a>
<span class="sourceLineNo">223</span><a id="line.223">                                                                                                                                                .toLowerCase(aa))) );</a>
<span class="sourceLineNo">224</span><a id="line.224">                                                                previousPos[str]++;</a>
<span class="sourceLineNo">225</span><a id="line.225">                                                        } else {</a>
<span class="sourceLineNo">226</span><a id="line.226">                                                                // Insert a gap otherwise</a>
<span class="sourceLineNo">227</span><a id="line.227">                                                                alnSequences.set(str, alnSequences.get(str)</a>
<span class="sourceLineNo">228</span><a id="line.228">                                                                                .concat("-"));</a>
<span class="sourceLineNo">229</span><a id="line.229">                                                        }</a>
<span class="sourceLineNo">230</span><a id="line.230">                                                }</a>
<span class="sourceLineNo">231</span><a id="line.231">                                                mapSeqToStruct.add(-1); // unaligned position</a>
<span class="sourceLineNo">232</span><a id="line.232">                                        } else {</a>
<span class="sourceLineNo">233</span><a id="line.233">                                                // Add provisional and update the indices</a>
<span class="sourceLineNo">234</span><a id="line.234">                                                for (int str = 0; str &lt; alignment.size(); str++) {</a>
<span class="sourceLineNo">235</span><a id="line.235">                                                        alnSequences.set(</a>
<span class="sourceLineNo">236</span><a id="line.236">                                                                        str,</a>
<span class="sourceLineNo">237</span><a id="line.237">                                                                        alnSequences.get(str).concat(</a>
<span class="sourceLineNo">238</span><a id="line.238">                                                                                        String.valueOf(provisionalChar[str])));</a>
<span class="sourceLineNo">239</span><a id="line.239"></a>
<span class="sourceLineNo">240</span><a id="line.240">                                                        if (provisionalChar[str] != '-') {</a>
<span class="sourceLineNo">241</span><a id="line.241">                                                                if (alignment.getBlocks().get(b).getAlignRes()</a>
<span class="sourceLineNo">242</span><a id="line.242">                                                                                .get(str).get(pos) == null) {</a>
<span class="sourceLineNo">243</span><a id="line.243">                                                                        previousPos[str]++;</a>
<span class="sourceLineNo">244</span><a id="line.244">                                                                } else {</a>
<span class="sourceLineNo">245</span><a id="line.245">                                                                        previousPos[str] = alignment.getBlocks()</a>
<span class="sourceLineNo">246</span><a id="line.246">                                                                                        .get(b).getAlignRes().get(str)</a>
<span class="sourceLineNo">247</span><a id="line.247">                                                                                        .get(pos);</a>
<span class="sourceLineNo">248</span><a id="line.248">                                                                }</a>
<span class="sourceLineNo">249</span><a id="line.249">                                                        }</a>
<span class="sourceLineNo">250</span><a id="line.250">                                                }</a>
<span class="sourceLineNo">251</span><a id="line.251">                                                mapSeqToStruct.add(globalPos); // alignment index</a>
<span class="sourceLineNo">252</span><a id="line.252">                                        }</a>
<span class="sourceLineNo">253</span><a id="line.253">                                }</a>
<span class="sourceLineNo">254</span><a id="line.254">                        } // All positions in the Block considered so far</a>
<span class="sourceLineNo">255</span><a id="line.255"></a>
<span class="sourceLineNo">256</span><a id="line.256">                        // Calculate the index of the next Block for every structure</a>
<span class="sourceLineNo">257</span><a id="line.257">                        int[] blockEnds = new int[alignment.size()];</a>
<span class="sourceLineNo">258</span><a id="line.258">                        for (int str = 0; str &lt; alignment.size(); str++) {</a>
<span class="sourceLineNo">259</span><a id="line.259">                                for (int res : blockStarts.get(str)) {</a>
<span class="sourceLineNo">260</span><a id="line.260">                                        if (previousPos[str] &gt; res)</a>
<span class="sourceLineNo">261</span><a id="line.261">                                                blockEnds[str] = res;</a>
<span class="sourceLineNo">262</span><a id="line.262">                                        else {</a>
<span class="sourceLineNo">263</span><a id="line.263">                                                blockEnds[str] = res;</a>
<span class="sourceLineNo">264</span><a id="line.264">                                                break;</a>
<span class="sourceLineNo">265</span><a id="line.265">                                        }</a>
<span class="sourceLineNo">266</span><a id="line.266">                                }</a>
<span class="sourceLineNo">267</span><a id="line.267">                        }</a>
<span class="sourceLineNo">268</span><a id="line.268"></a>
<span class="sourceLineNo">269</span><a id="line.269">                        // Add the unaligned residues in between Blocks (lowercase)</a>
<span class="sourceLineNo">270</span><a id="line.270">                        boolean allGaps = false; // true means no more residues to add</a>
<span class="sourceLineNo">271</span><a id="line.271">                        while (!allGaps) {</a>
<span class="sourceLineNo">272</span><a id="line.272">                                allGaps = true;</a>
<span class="sourceLineNo">273</span><a id="line.273">                                for (int str = 0; str &lt; alignment.size(); str++) {</a>
<span class="sourceLineNo">274</span><a id="line.274">                                        if (previousPos[str] + 1 &lt; blockEnds[str]) {</a>
<span class="sourceLineNo">275</span><a id="line.275">                                                Atom a = atoms.get(str)[previousPos[str] + 1];</a>
<span class="sourceLineNo">276</span><a id="line.276">                                                String group = a.getGroup().getPDBName();</a>
<span class="sourceLineNo">277</span><a id="line.277">                                                char letter = StructureTools.get1LetterCode(group);</a>
<span class="sourceLineNo">278</span><a id="line.278"></a>
<span class="sourceLineNo">279</span><a id="line.279">                                                provisionalChar[str] = Character.toLowerCase(letter);</a>
<span class="sourceLineNo">280</span><a id="line.280">                                                previousPos[str]++;</a>
<span class="sourceLineNo">281</span><a id="line.281">                                                allGaps = false;</a>
<span class="sourceLineNo">282</span><a id="line.282">                                        } else</a>
<span class="sourceLineNo">283</span><a id="line.283">                                                provisionalChar[str] = '-';</a>
<span class="sourceLineNo">284</span><a id="line.284">                                }</a>
<span class="sourceLineNo">285</span><a id="line.285">                                if (!allGaps) {</a>
<span class="sourceLineNo">286</span><a id="line.286">                                        for (int str = 0; str &lt; alignment.size(); str++) {</a>
<span class="sourceLineNo">287</span><a id="line.287">                                                alnSequences.set(</a>
<span class="sourceLineNo">288</span><a id="line.288">                                                                str,</a>
<span class="sourceLineNo">289</span><a id="line.289">                                                                alnSequences.get(str).concat(</a>
<span class="sourceLineNo">290</span><a id="line.290">                                                                                String.valueOf(provisionalChar[str])) );</a>
<span class="sourceLineNo">291</span><a id="line.291">                                        }</a>
<span class="sourceLineNo">292</span><a id="line.292">                                        mapSeqToStruct.add(-1); // unaligned position</a>
<span class="sourceLineNo">293</span><a id="line.293">                                }</a>
<span class="sourceLineNo">294</span><a id="line.294">                        }</a>
<span class="sourceLineNo">295</span><a id="line.295">                }</a>
<span class="sourceLineNo">296</span><a id="line.296">                return alnSequences;</a>
<span class="sourceLineNo">297</span><a id="line.297">        }</a>
<span class="sourceLineNo">298</span><a id="line.298"></a>
<span class="sourceLineNo">299</span><a id="line.299">        /**</a>
<span class="sourceLineNo">300</span><a id="line.300">         * Calculate the sequence alignment Strings for the whole alignment. This</a>
<span class="sourceLineNo">301</span><a id="line.301">         * method creates a sequence alignment where aligned residues are in</a>
<span class="sourceLineNo">302</span><a id="line.302">         * uppercase and unaligned residues are in lowercase, thus providing a more</a>
<span class="sourceLineNo">303</span><a id="line.303">         * compact way to represent the alignment.</a>
<span class="sourceLineNo">304</span><a id="line.304">         * &lt;p&gt;</a>
<span class="sourceLineNo">305</span><a id="line.305">         * Blocks are concatenated in the order returned by</a>
<span class="sourceLineNo">306</span><a id="line.306">         * {@link MultipleAlignment#getBlocks()}, so sequences may not be</a>
<span class="sourceLineNo">307</span><a id="line.307">         * sequential. Gaps are represented by '-'. Separation between different</a>
<span class="sourceLineNo">308</span><a id="line.308">         * Blocks is indicated by a gap in all positions, meaning that there is a</a>
<span class="sourceLineNo">309</span><a id="line.309">         * possible discontinuity.</a>
<span class="sourceLineNo">310</span><a id="line.310">         *</a>
<span class="sourceLineNo">311</span><a id="line.311">         * @param alignment</a>
<span class="sourceLineNo">312</span><a id="line.312">         *            input MultipleAlignment</a>
<span class="sourceLineNo">313</span><a id="line.313">         * @return String for each row in the alignment, giving the 1-letter code</a>
<span class="sourceLineNo">314</span><a id="line.314">         *         for each aligned residue.</a>
<span class="sourceLineNo">315</span><a id="line.315">         */</a>
<span class="sourceLineNo">316</span><a id="line.316">        public static List&lt;String&gt; getSequenceAlignment(MultipleAlignment msa) {</a>
<span class="sourceLineNo">317</span><a id="line.317">                return getSequenceAlignment(msa, new ArrayList&lt;Integer&gt;());</a>
<span class="sourceLineNo">318</span><a id="line.318">        }</a>
<span class="sourceLineNo">319</span><a id="line.319"></a>
<span class="sourceLineNo">320</span><a id="line.320">        /**</a>
<span class="sourceLineNo">321</span><a id="line.321">         * Calculate the sequence alignment Strings for the alignment Blocks in an</a>
<span class="sourceLineNo">322</span><a id="line.322">         * alignment. This method creates a sequence alignment where all residues</a>
<span class="sourceLineNo">323</span><a id="line.323">         * are in uppercase and a residue alone with gaps in all the other</a>
<span class="sourceLineNo">324</span><a id="line.324">         * structures represents unaligned residues. Because of this latter</a>
<span class="sourceLineNo">325</span><a id="line.325">         * constraint only the residues within the Blocks are represented, for a</a>
<span class="sourceLineNo">326</span><a id="line.326">         * more compact alignment. For a sequence alignment of the full protein use</a>
<span class="sourceLineNo">327</span><a id="line.327">         * {@link #getSequenceAlignment(MultipleAlignment)}.</a>
<span class="sourceLineNo">328</span><a id="line.328">         * &lt;p&gt;</a>
<span class="sourceLineNo">329</span><a id="line.329">         * Blocks are concatenated in the order returned by</a>
<span class="sourceLineNo">330</span><a id="line.330">         * {@link MultipleAlignment#getBlocks()}, so sequences may not be</a>
<span class="sourceLineNo">331</span><a id="line.331">         * sequential. Gaps between blocks are omitted, while gaps within blocks are</a>
<span class="sourceLineNo">332</span><a id="line.332">         * represented by '-'. Separation between different Blocks is indicated by a</a>
<span class="sourceLineNo">333</span><a id="line.333">         * gap in all positions, meaning that there is something unaligned</a>
<span class="sourceLineNo">334</span><a id="line.334">         * inbetween.</a>
<span class="sourceLineNo">335</span><a id="line.335">         *</a>
<span class="sourceLineNo">336</span><a id="line.336">         * @param alignment</a>
<span class="sourceLineNo">337</span><a id="line.337">         *            input MultipleAlignment</a>
<span class="sourceLineNo">338</span><a id="line.338">         * @param mapSeqToStruct</a>
<span class="sourceLineNo">339</span><a id="line.339">         *            provides a link from the sequence alignment position to the</a>
<span class="sourceLineNo">340</span><a id="line.340">         *            structure alignment position. Specially designed for the GUI.</a>
<span class="sourceLineNo">341</span><a id="line.341">         *            Has to be initialized previously and will be overwritten.</a>
<span class="sourceLineNo">342</span><a id="line.342">         * @return a string for each row in the alignment, giving the 1-letter code</a>
<span class="sourceLineNo">343</span><a id="line.343">         *         for each aligned residue.</a>
<span class="sourceLineNo">344</span><a id="line.344">         */</a>
<span class="sourceLineNo">345</span><a id="line.345">        public static List&lt;String&gt; getBlockSequenceAlignment(</a>
<span class="sourceLineNo">346</span><a id="line.346">                        MultipleAlignment alignment, List&lt;Integer&gt; mapSeqToStruct) {</a>
<span class="sourceLineNo">347</span><a id="line.347"></a>
<span class="sourceLineNo">348</span><a id="line.348">                // Initialize sequence variables</a>
<span class="sourceLineNo">349</span><a id="line.349">                List&lt;String&gt; alnSequences = new ArrayList&lt;String&gt;();</a>
<span class="sourceLineNo">350</span><a id="line.350">                for (int str = 0; str &lt; alignment.size(); str++)</a>
<span class="sourceLineNo">351</span><a id="line.351">                        alnSequences.add("");</a>
<span class="sourceLineNo">352</span><a id="line.352">                mapSeqToStruct.clear();</a>
<span class="sourceLineNo">353</span><a id="line.353">                List&lt;Atom[]&gt; atoms = alignment.getAtomArrays();</a>
<span class="sourceLineNo">354</span><a id="line.354">                int globalPos = -1;</a>
<span class="sourceLineNo">355</span><a id="line.355"></a>
<span class="sourceLineNo">356</span><a id="line.356">                // Loop through all the alignment Blocks in the order given</a>
<span class="sourceLineNo">357</span><a id="line.357">                for (int b = 0; b &lt; alignment.getBlocks().size(); b++) {</a>
<span class="sourceLineNo">358</span><a id="line.358">                        if (b != 0) {</a>
<span class="sourceLineNo">359</span><a id="line.359">                                // Add a gap to all structures to separate Blocks</a>
<span class="sourceLineNo">360</span><a id="line.360">                                for (int str = 0; str &lt; alignment.size(); str++)</a>
<span class="sourceLineNo">361</span><a id="line.361">                                        alnSequences.set(str, alnSequences.get(str).concat("-"));</a>
<span class="sourceLineNo">362</span><a id="line.362">                                mapSeqToStruct.add(-1); // means unaligned position</a>
<span class="sourceLineNo">363</span><a id="line.363">                        }</a>
<span class="sourceLineNo">364</span><a id="line.364"></a>
<span class="sourceLineNo">365</span><a id="line.365">                        // Store the previous position added to the sequence alignment</a>
<span class="sourceLineNo">366</span><a id="line.366">                        int[] previousPos = new int[alignment.size()];</a>
<span class="sourceLineNo">367</span><a id="line.367">                        Arrays.fill(previousPos, -1);</a>
<span class="sourceLineNo">368</span><a id="line.368">                        // Store provisional characters</a>
<span class="sourceLineNo">369</span><a id="line.369">                        char[] provisionalChar = new char[alignment.size()];</a>
<span class="sourceLineNo">370</span><a id="line.370">                        Arrays.fill(provisionalChar, '-');</a>
<span class="sourceLineNo">371</span><a id="line.371"></a>
<span class="sourceLineNo">372</span><a id="line.372">                        for (int pos = 0; pos &lt; alignment.getBlocks().get(b).length(); pos++) {</a>
<span class="sourceLineNo">373</span><a id="line.373">                                globalPos++;</a>
<span class="sourceLineNo">374</span><a id="line.374">                                boolean gaps = true;</a>
<span class="sourceLineNo">375</span><a id="line.375">                                while (gaps) {</a>
<span class="sourceLineNo">376</span><a id="line.376">                                        gaps = false;</a>
<span class="sourceLineNo">377</span><a id="line.377">                                        // Loop through all the structures</a>
<span class="sourceLineNo">378</span><a id="line.378">                                        for (int str = 0; str &lt; alignment.size(); str++) {</a>
<span class="sourceLineNo">379</span><a id="line.379">                                                // If it is the first position or before it was null</a>
<span class="sourceLineNo">380</span><a id="line.380">                                                if (previousPos[str] == -1) {</a>
<span class="sourceLineNo">381</span><a id="line.381">                                                        Integer residue = alignment.getBlocks().get(b)</a>
<span class="sourceLineNo">382</span><a id="line.382">                                                                        .getAlignRes().get(str).get(pos);</a>
<span class="sourceLineNo">383</span><a id="line.383">                                                        if (residue == null)</a>
<span class="sourceLineNo">384</span><a id="line.384">                                                                provisionalChar[str] = '-';</a>
<span class="sourceLineNo">385</span><a id="line.385">                                                        else {</a>
<span class="sourceLineNo">386</span><a id="line.386">                                                                Atom a = atoms.get(str)[residue];</a>
<span class="sourceLineNo">387</span><a id="line.387">                                                                String g = a.getGroup().getPDBName();</a>
<span class="sourceLineNo">388</span><a id="line.388">                                                                char aa = StructureTools.get1LetterCode(g);</a>
<span class="sourceLineNo">389</span><a id="line.389">                                                                provisionalChar[str] = aa;</a>
<span class="sourceLineNo">390</span><a id="line.390">                                                        }</a>
<span class="sourceLineNo">391</span><a id="line.391">                                                } else {</a>
<span class="sourceLineNo">392</span><a id="line.392">                                                        Integer residue = alignment.getBlocks().get(b)</a>
<span class="sourceLineNo">393</span><a id="line.393">                                                                        .getAlignRes().get(str).get(pos);</a>
<span class="sourceLineNo">394</span><a id="line.394">                                                        if (residue == null)</a>
<span class="sourceLineNo">395</span><a id="line.395">                                                                provisionalChar[str] = '-';</a>
<span class="sourceLineNo">396</span><a id="line.396">                                                        else if (previousPos[str] + 1 == residue) {</a>
<span class="sourceLineNo">397</span><a id="line.397">                                                                Atom a = atoms.get(str)[residue];</a>
<span class="sourceLineNo">398</span><a id="line.398">                                                                String g = a.getGroup().getPDBName();</a>
<span class="sourceLineNo">399</span><a id="line.399">                                                                char aa = StructureTools.get1LetterCode(g);</a>
<span class="sourceLineNo">400</span><a id="line.400">                                                                provisionalChar[str] = aa;</a>
<span class="sourceLineNo">401</span><a id="line.401">                                                        } else {</a>
<span class="sourceLineNo">402</span><a id="line.402">                                                                provisionalChar[str] = ' ';</a>
<span class="sourceLineNo">403</span><a id="line.403">                                                                gaps = true;</a>
<span class="sourceLineNo">404</span><a id="line.404">                                                        }</a>
<span class="sourceLineNo">405</span><a id="line.405">                                                }</a>
<span class="sourceLineNo">406</span><a id="line.406">                                        }// End all structures analysis</a>
<span class="sourceLineNo">407</span><a id="line.407"></a>
<span class="sourceLineNo">408</span><a id="line.408">                                        if (gaps) {</a>
<span class="sourceLineNo">409</span><a id="line.409">                                                for (int str = 0; str &lt; alignment.size(); str++) {</a>
<span class="sourceLineNo">410</span><a id="line.410">                                                        if (provisionalChar[str] == ' ') {</a>
<span class="sourceLineNo">411</span><a id="line.411">                                                                // It means this residue was non-consecutive</a>
<span class="sourceLineNo">412</span><a id="line.412">                                                                for (int s2 = 0; s2 &lt; alignment.size(); s2++) {</a>
<span class="sourceLineNo">413</span><a id="line.413">                                                                        if (str == s2) {</a>
<span class="sourceLineNo">414</span><a id="line.414">                                                                                int next = previousPos[str] + 1;</a>
<span class="sourceLineNo">415</span><a id="line.415">                                                                                Atom a = atoms.get(s2)[next];</a>
<span class="sourceLineNo">416</span><a id="line.416">                                                                                String g = a.getGroup().getPDBName();</a>
<span class="sourceLineNo">417</span><a id="line.417">                                                                                char aa = StructureTools</a>
<span class="sourceLineNo">418</span><a id="line.418">                                                                                                .get1LetterCode(g);</a>
<span class="sourceLineNo">419</span><a id="line.419">                                                                                alnSequences.set(</a>
<span class="sourceLineNo">420</span><a id="line.420">                                                                                                s2,</a>
<span class="sourceLineNo">421</span><a id="line.421">                                                                                                alnSequences.get(s2).concat(</a>
<span class="sourceLineNo">422</span><a id="line.422">                                                                                                                String.valueOf(aa)) );</a>
<span class="sourceLineNo">423</span><a id="line.423">                                                                        } else {</a>
<span class="sourceLineNo">424</span><a id="line.424">                                                                                alnSequences.set(s2,</a>
<span class="sourceLineNo">425</span><a id="line.425">                                                                                                alnSequences.get(s2)</a>
<span class="sourceLineNo">426</span><a id="line.426">                                                                                                                .concat("-"));</a>
<span class="sourceLineNo">427</span><a id="line.427">                                                                        }</a>
<span class="sourceLineNo">428</span><a id="line.428">                                                                }</a>
<span class="sourceLineNo">429</span><a id="line.429">                                                                mapSeqToStruct.add(-1); // unaligned</a>
<span class="sourceLineNo">430</span><a id="line.430">                                                                previousPos[str] += 1;</a>
<span class="sourceLineNo">431</span><a id="line.431">                                                        }</a>
<span class="sourceLineNo">432</span><a id="line.432">                                                }</a>
<span class="sourceLineNo">433</span><a id="line.433">                                        } else { // Append the provisional and update the indices</a>
<span class="sourceLineNo">434</span><a id="line.434">                                                for (int str = 0; str &lt; alignment.size(); str++) {</a>
<span class="sourceLineNo">435</span><a id="line.435">                                                        alnSequences.set(</a>
<span class="sourceLineNo">436</span><a id="line.436">                                                                        str,</a>
<span class="sourceLineNo">437</span><a id="line.437">                                                                        alnSequences.get(str).concat(</a>
<span class="sourceLineNo">438</span><a id="line.438">                                                                                        String.valueOf(provisionalChar[str])) );</a>
<span class="sourceLineNo">439</span><a id="line.439">                                                        if (provisionalChar[str] != '-') {</a>
<span class="sourceLineNo">440</span><a id="line.440">                                                                previousPos[str] = alignment.getBlocks().get(b)</a>
<span class="sourceLineNo">441</span><a id="line.441">                                                                                .getAlignRes().get(str).get(pos);</a>
<span class="sourceLineNo">442</span><a id="line.442">                                                        }</a>
<span class="sourceLineNo">443</span><a id="line.443">                                                }</a>
<span class="sourceLineNo">444</span><a id="line.444">                                                mapSeqToStruct.add(globalPos);</a>
<span class="sourceLineNo">445</span><a id="line.445">                                        }</a>
<span class="sourceLineNo">446</span><a id="line.446">                                }</a>
<span class="sourceLineNo">447</span><a id="line.447">                        }</a>
<span class="sourceLineNo">448</span><a id="line.448">                }</a>
<span class="sourceLineNo">449</span><a id="line.449">                return alnSequences;</a>
<span class="sourceLineNo">450</span><a id="line.450">        }</a>
<span class="sourceLineNo">451</span><a id="line.451"></a>
<span class="sourceLineNo">452</span><a id="line.452">        /**</a>
<span class="sourceLineNo">453</span><a id="line.453">         * Calculate the sequence alignment Strings for the alignment Blocks in an</a>
<span class="sourceLineNo">454</span><a id="line.454">         * alignment. This method creates a sequence alignment where all residues</a>
<span class="sourceLineNo">455</span><a id="line.455">         * are in uppercase and a residue alone with gaps in all the other</a>
<span class="sourceLineNo">456</span><a id="line.456">         * structures represents unaligned residues. Because of this latter</a>
<span class="sourceLineNo">457</span><a id="line.457">         * constraint only the residues within the Blocks are represented, for a</a>
<span class="sourceLineNo">458</span><a id="line.458">         * more compact alignment. For a sequence alignment of the full protein use</a>
<span class="sourceLineNo">459</span><a id="line.459">         * {@link #getSequenceAlignment(MultipleAlignment)}.</a>
<span class="sourceLineNo">460</span><a id="line.460">         * &lt;p&gt;</a>
<span class="sourceLineNo">461</span><a id="line.461">         * Blocks are concatenated in the order returned by</a>
<span class="sourceLineNo">462</span><a id="line.462">         * {@link MultipleAlignment#getBlocks()}, so sequences may not be</a>
<span class="sourceLineNo">463</span><a id="line.463">         * sequential. Gaps between blocks are omitted, while gaps within blocks are</a>
<span class="sourceLineNo">464</span><a id="line.464">         * represented by '-'. Separation between different Blocks is indicated by a</a>
<span class="sourceLineNo">465</span><a id="line.465">         * gap in all positions, meaning that there is something unaligned</a>
<span class="sourceLineNo">466</span><a id="line.466">         * inbetween.</a>
<span class="sourceLineNo">467</span><a id="line.467">         *</a>
<span class="sourceLineNo">468</span><a id="line.468">         * @param alignment</a>
<span class="sourceLineNo">469</span><a id="line.469">         *            input MultipleAlignment</a>
<span class="sourceLineNo">470</span><a id="line.470">         * @return String for each row in the alignment, giving the 1-letter code</a>
<span class="sourceLineNo">471</span><a id="line.471">         *         for each aligned residue.</a>
<span class="sourceLineNo">472</span><a id="line.472">         */</a>
<span class="sourceLineNo">473</span><a id="line.473">        public static List&lt;String&gt; getBlockSequenceAlignment(MultipleAlignment ma) {</a>
<span class="sourceLineNo">474</span><a id="line.474">                return getBlockSequenceAlignment(ma, new ArrayList&lt;Integer&gt;());</a>
<span class="sourceLineNo">475</span><a id="line.475">        }</a>
<span class="sourceLineNo">476</span><a id="line.476"></a>
<span class="sourceLineNo">477</span><a id="line.477">        /**</a>
<span class="sourceLineNo">478</span><a id="line.478">         * Returns the Atom of the specified structure that is aligned in the</a>
<span class="sourceLineNo">479</span><a id="line.479">         * sequence alignment position specified.</a>
<span class="sourceLineNo">480</span><a id="line.480">         *</a>
<span class="sourceLineNo">481</span><a id="line.481">         * @param multAln</a>
<span class="sourceLineNo">482</span><a id="line.482">         *            the MultipleAlignment object from where the sequence alignment</a>
<span class="sourceLineNo">483</span><a id="line.483">         *            has been generated</a>
<span class="sourceLineNo">484</span><a id="line.484">         * @param mapSeqToStruct</a>
<span class="sourceLineNo">485</span><a id="line.485">         *            the mapping between sequence and structure generated with the</a>
<span class="sourceLineNo">486</span><a id="line.486">         *            sequence alignment</a>
<span class="sourceLineNo">487</span><a id="line.487">         * @param str</a>
<span class="sourceLineNo">488</span><a id="line.488">         *            the structure index of the alignment (row)</a>
<span class="sourceLineNo">489</span><a id="line.489">         * @param sequencePos</a>
<span class="sourceLineNo">490</span><a id="line.490">         *            the sequence alignment position (column)</a>
<span class="sourceLineNo">491</span><a id="line.491">         * @return Atom the atom in that position or null if there is a gap</a>
<span class="sourceLineNo">492</span><a id="line.492">         */</a>
<span class="sourceLineNo">493</span><a id="line.493">        public static Atom getAtomForSequencePosition(MultipleAlignment msa,</a>
<span class="sourceLineNo">494</span><a id="line.494">                        List&lt;Integer&gt; mapSeqToStruct, int str, int sequencePos) {</a>
<span class="sourceLineNo">495</span><a id="line.495"></a>
<span class="sourceLineNo">496</span><a id="line.496">                int seqPos = mapSeqToStruct.get(sequencePos);</a>
<span class="sourceLineNo">497</span><a id="line.497"></a>
<span class="sourceLineNo">498</span><a id="line.498">                // Check if the position selected is an aligned position</a>
<span class="sourceLineNo">499</span><a id="line.499">                if (seqPos == -1)</a>
<span class="sourceLineNo">500</span><a id="line.500">                        return null;</a>
<span class="sourceLineNo">501</span><a id="line.501">                else {</a>
<span class="sourceLineNo">502</span><a id="line.502">                        Atom a = null;</a>
<span class="sourceLineNo">503</span><a id="line.503">                        // Calculate the corresponding structure position</a>
<span class="sourceLineNo">504</span><a id="line.504">                        int sum = 0;</a>
<span class="sourceLineNo">505</span><a id="line.505">                        for (Block b : msa.getBlocks()) {</a>
<span class="sourceLineNo">506</span><a id="line.506">                                if (sum + b.length() &lt;= seqPos) {</a>
<span class="sourceLineNo">507</span><a id="line.507">                                        sum += b.length();</a>
<span class="sourceLineNo">508</span><a id="line.508">                                        continue;</a>
<span class="sourceLineNo">509</span><a id="line.509">                                } else {</a>
<span class="sourceLineNo">510</span><a id="line.510">                                        for (Integer p : b.getAlignRes().get(str)) {</a>
<span class="sourceLineNo">511</span><a id="line.511">                                                if (sum == seqPos) {</a>
<span class="sourceLineNo">512</span><a id="line.512">                                                        if (p != null) {</a>
<span class="sourceLineNo">513</span><a id="line.513">                                                                a = msa.getAtomArrays().get(str)[p];</a>
<span class="sourceLineNo">514</span><a id="line.514">                                                        }</a>
<span class="sourceLineNo">515</span><a id="line.515">                                                        break;</a>
<span class="sourceLineNo">516</span><a id="line.516">                                                }</a>
<span class="sourceLineNo">517</span><a id="line.517">                                                sum++;</a>
<span class="sourceLineNo">518</span><a id="line.518">                                        }</a>
<span class="sourceLineNo">519</span><a id="line.519">                                        break;</a>
<span class="sourceLineNo">520</span><a id="line.520">                                }</a>
<span class="sourceLineNo">521</span><a id="line.521">                        }</a>
<span class="sourceLineNo">522</span><a id="line.522">                        return a;</a>
<span class="sourceLineNo">523</span><a id="line.523">                }</a>
<span class="sourceLineNo">524</span><a id="line.524">        }</a>
<span class="sourceLineNo">525</span><a id="line.525"></a>
<span class="sourceLineNo">526</span><a id="line.526">        /**</a>
<span class="sourceLineNo">527</span><a id="line.527">         * Returns the block number of a specified position in the sequence</a>
<span class="sourceLineNo">528</span><a id="line.528">         * alignment, given the mapping from structure to function.</a>
<span class="sourceLineNo">529</span><a id="line.529">         *</a>
<span class="sourceLineNo">530</span><a id="line.530">         * @param multAln</a>
<span class="sourceLineNo">531</span><a id="line.531">         *            the MultipleAlignment object from where the sequence alignment</a>
<span class="sourceLineNo">532</span><a id="line.532">         *            has been generated.</a>
<span class="sourceLineNo">533</span><a id="line.533">         * @param mapSeqToStruct</a>
<span class="sourceLineNo">534</span><a id="line.534">         *            the mapping between sequence and structure generated with the</a>
<span class="sourceLineNo">535</span><a id="line.535">         *            sequence alignment</a>
<span class="sourceLineNo">536</span><a id="line.536">         * @param sequencePos</a>
<span class="sourceLineNo">537</span><a id="line.537">         *            the position in the sequence alignment (column)</a>
<span class="sourceLineNo">538</span><a id="line.538">         * @return int the block index, or -1 if the position is not aligned</a>
<span class="sourceLineNo">539</span><a id="line.539">         */</a>
<span class="sourceLineNo">540</span><a id="line.540">        public static int getBlockForSequencePosition(MultipleAlignment multAln,</a>
<span class="sourceLineNo">541</span><a id="line.541">                        List&lt;Integer&gt; mapSeqToStruct, int sequencePos) {</a>
<span class="sourceLineNo">542</span><a id="line.542"></a>
<span class="sourceLineNo">543</span><a id="line.543">                int seqPos = mapSeqToStruct.get(sequencePos);</a>
<span class="sourceLineNo">544</span><a id="line.544">                // Check if the position selected is an aligned position</a>
<span class="sourceLineNo">545</span><a id="line.545">                if (seqPos == -1)</a>
<span class="sourceLineNo">546</span><a id="line.546">                        return -1;</a>
<span class="sourceLineNo">547</span><a id="line.547">                else {</a>
<span class="sourceLineNo">548</span><a id="line.548">                        // Calculate the corresponding block (by iterating all Blocks)</a>
<span class="sourceLineNo">549</span><a id="line.549">                        int sum = 0;</a>
<span class="sourceLineNo">550</span><a id="line.550">                        int block = 0;</a>
<span class="sourceLineNo">551</span><a id="line.551">                        for (Block b : multAln.getBlocks()) {</a>
<span class="sourceLineNo">552</span><a id="line.552">                                if (sum + b.length() &lt;= seqPos) {</a>
<span class="sourceLineNo">553</span><a id="line.553">                                        sum += b.length();</a>
<span class="sourceLineNo">554</span><a id="line.554">                                        block++;</a>
<span class="sourceLineNo">555</span><a id="line.555">                                        continue;</a>
<span class="sourceLineNo">556</span><a id="line.556">                                } else</a>
<span class="sourceLineNo">557</span><a id="line.557">                                        break;</a>
<span class="sourceLineNo">558</span><a id="line.558">                        }</a>
<span class="sourceLineNo">559</span><a id="line.559">                        return block;</a>
<span class="sourceLineNo">560</span><a id="line.560">                }</a>
<span class="sourceLineNo">561</span><a id="line.561">        }</a>
<span class="sourceLineNo">562</span><a id="line.562"></a>
<span class="sourceLineNo">563</span><a id="line.563">        /**</a>
<span class="sourceLineNo">564</span><a id="line.564">         * The average residue distance Matrix contains the average distance from</a>
<span class="sourceLineNo">565</span><a id="line.565">         * each residue to all other residues aligned with it.</a>
<span class="sourceLineNo">566</span><a id="line.566">         * &lt;p&gt;</a>
<span class="sourceLineNo">567</span><a id="line.567">         * Complexity: T(n,l) = O(l*n^2), if n=number of structures and l=alignment</a>
<span class="sourceLineNo">568</span><a id="line.568">         * length.</a>
<span class="sourceLineNo">569</span><a id="line.569">         *</a>
<span class="sourceLineNo">570</span><a id="line.570">         * @param alignment</a>
<span class="sourceLineNo">571</span><a id="line.571">         *            MultipleAlignment</a>
<span class="sourceLineNo">572</span><a id="line.572">         * @return Matrix containing all average residue distances</a>
<span class="sourceLineNo">573</span><a id="line.573">         */</a>
<span class="sourceLineNo">574</span><a id="line.574">        public static Matrix getAverageResidueDistances(MultipleAlignment msa) {</a>
<span class="sourceLineNo">575</span><a id="line.575">                List&lt;Atom[]&gt; transformed = transformAtoms(msa);</a>
<span class="sourceLineNo">576</span><a id="line.576">                return getAverageResidueDistances(transformed);</a>
<span class="sourceLineNo">577</span><a id="line.577">        }</a>
<span class="sourceLineNo">578</span><a id="line.578"></a>
<span class="sourceLineNo">579</span><a id="line.579">        /**</a>
<span class="sourceLineNo">580</span><a id="line.580">         * The average residue distance Matrix contains the average distance from</a>
<span class="sourceLineNo">581</span><a id="line.581">         * each residue to all other residues aligned with it.</a>
<span class="sourceLineNo">582</span><a id="line.582">         * &lt;p&gt;</a>
<span class="sourceLineNo">583</span><a id="line.583">         * Complexity: T(n,l) = O(l*n^2), if n=number of structures and l=alignment</a>
<span class="sourceLineNo">584</span><a id="line.584">         * length.</a>
<span class="sourceLineNo">585</span><a id="line.585">         *</a>
<span class="sourceLineNo">586</span><a id="line.586">         * @param transformed</a>
<span class="sourceLineNo">587</span><a id="line.587">         *            List of Atom arrays containing only the aligned atoms of each</a>
<span class="sourceLineNo">588</span><a id="line.588">         *            structure, or null if there is a gap.</a>
<span class="sourceLineNo">589</span><a id="line.589">         * @return Matrix containing all average residue distances. Entry -1 means</a>
<span class="sourceLineNo">590</span><a id="line.590">         *         there is a gap in the position.</a>
<span class="sourceLineNo">591</span><a id="line.591">         */</a>
<span class="sourceLineNo">592</span><a id="line.592">        public static Matrix getAverageResidueDistances(List&lt;Atom[]&gt; transformed) {</a>
<span class="sourceLineNo">593</span><a id="line.593"></a>
<span class="sourceLineNo">594</span><a id="line.594">                int size = transformed.size();</a>
<span class="sourceLineNo">595</span><a id="line.595">                int length = transformed.get(0).length;</a>
<span class="sourceLineNo">596</span><a id="line.596">                Matrix resDist = new Matrix(size, length, -1);</a>
<span class="sourceLineNo">597</span><a id="line.597"></a>
<span class="sourceLineNo">598</span><a id="line.598">                // Calculate the average residue distances</a>
<span class="sourceLineNo">599</span><a id="line.599">                for (int r1 = 0; r1 &lt; size; r1++) {</a>
<span class="sourceLineNo">600</span><a id="line.600">                        for (int c = 0; c &lt; transformed.get(r1).length; c++) {</a>
<span class="sourceLineNo">601</span><a id="line.601">                                Atom refAtom = transformed.get(r1)[c];</a>
<span class="sourceLineNo">602</span><a id="line.602">                                if (refAtom == null)</a>
<span class="sourceLineNo">603</span><a id="line.603">                                        continue;</a>
<span class="sourceLineNo">604</span><a id="line.604"></a>
<span class="sourceLineNo">605</span><a id="line.605">                                for (int r2 = r1 + 1; r2 &lt; size; r2++) {</a>
<span class="sourceLineNo">606</span><a id="line.606">                                        Atom atom = transformed.get(r2)[c];</a>
<span class="sourceLineNo">607</span><a id="line.607">                                        if (atom != null) {</a>
<span class="sourceLineNo">608</span><a id="line.608">                                                double distance = Calc.getDistance(refAtom, atom);</a>
<span class="sourceLineNo">609</span><a id="line.609"></a>
<span class="sourceLineNo">610</span><a id="line.610">                                                if (resDist.get(r1, c) == -1) {</a>
<span class="sourceLineNo">611</span><a id="line.611">                                                        resDist.set(r1, c, 1 + distance);</a>
<span class="sourceLineNo">612</span><a id="line.612">                                                } else {</a>
<span class="sourceLineNo">613</span><a id="line.613">                                                        resDist.set(r1, c, resDist.get(r1, c) + distance);</a>
<span class="sourceLineNo">614</span><a id="line.614">                                                }</a>
<span class="sourceLineNo">615</span><a id="line.615"></a>
<span class="sourceLineNo">616</span><a id="line.616">                                                if (resDist.get(r2, c) == -1) {</a>
<span class="sourceLineNo">617</span><a id="line.617">                                                        resDist.set(r2, c, 1 + distance);</a>
<span class="sourceLineNo">618</span><a id="line.618">                                                } else {</a>
<span class="sourceLineNo">619</span><a id="line.619">                                                        resDist.set(r2, c, resDist.get(r2, c) + distance);</a>
<span class="sourceLineNo">620</span><a id="line.620">                                                }</a>
<span class="sourceLineNo">621</span><a id="line.621">                                        }</a>
<span class="sourceLineNo">622</span><a id="line.622">                                }</a>
<span class="sourceLineNo">623</span><a id="line.623">                        }</a>
<span class="sourceLineNo">624</span><a id="line.624">                }</a>
<span class="sourceLineNo">625</span><a id="line.625"></a>
<span class="sourceLineNo">626</span><a id="line.626">                for (int c = 0; c &lt; length; c++) {</a>
<span class="sourceLineNo">627</span><a id="line.627">                        int nonNullRes = 0;</a>
<span class="sourceLineNo">628</span><a id="line.628">                        for (int r = 0; r &lt; size; r++) {</a>
<span class="sourceLineNo">629</span><a id="line.629">                                if (resDist.get(r, c) != -1)</a>
<span class="sourceLineNo">630</span><a id="line.630">                                        nonNullRes++;</a>
<span class="sourceLineNo">631</span><a id="line.631">                        }</a>
<span class="sourceLineNo">632</span><a id="line.632">                        for (int r = 0; r &lt; size; r++) {</a>
<span class="sourceLineNo">633</span><a id="line.633">                                if (resDist.get(r, c) != -1) {</a>
<span class="sourceLineNo">634</span><a id="line.634">                                        resDist.set(r, c, resDist.get(r, c) / nonNullRes);</a>
<span class="sourceLineNo">635</span><a id="line.635">                                }</a>
<span class="sourceLineNo">636</span><a id="line.636">                        }</a>
<span class="sourceLineNo">637</span><a id="line.637">                }</a>
<span class="sourceLineNo">638</span><a id="line.638">                return resDist;</a>
<span class="sourceLineNo">639</span><a id="line.639">        }</a>
<span class="sourceLineNo">640</span><a id="line.640"></a>
<span class="sourceLineNo">641</span><a id="line.641">        /**</a>
<span class="sourceLineNo">642</span><a id="line.642">         * Transforms atoms according to the superposition stored in the alignment.</a>
<span class="sourceLineNo">643</span><a id="line.643">         * &lt;p&gt;</a>
<span class="sourceLineNo">644</span><a id="line.644">         * For each structure in the alignment, returns an atom for each</a>
<span class="sourceLineNo">645</span><a id="line.645">         * representative atom in the aligned columns, omitting unaligned residues</a>
<span class="sourceLineNo">646</span><a id="line.646">         * (i.e. an array of length &lt;tt&gt;alignment.length()&lt;/tt&gt; ).</a>
<span class="sourceLineNo">647</span><a id="line.647">         * &lt;p&gt;</a>
<span class="sourceLineNo">648</span><a id="line.648">         * All blocks are concatenated together, so Atoms may not appear in the same</a>
<span class="sourceLineNo">649</span><a id="line.649">         * order as in their parent structure. If the alignment blocks contain null</a>
<span class="sourceLineNo">650</span><a id="line.650">         * residues (gaps), then the returned array will also contain null Atoms in</a>
<span class="sourceLineNo">651</span><a id="line.651">         * the same positions.</a>
<span class="sourceLineNo">652</span><a id="line.652">         *</a>
<span class="sourceLineNo">653</span><a id="line.653">         * @param alignment</a>
<span class="sourceLineNo">654</span><a id="line.654">         *            MultipleAlignment</a>
<span class="sourceLineNo">655</span><a id="line.655">         * @return List of Atom arrays of only the aligned atoms of every structure</a>
<span class="sourceLineNo">656</span><a id="line.656">         *         (null Atom if a gap position)</a>
<span class="sourceLineNo">657</span><a id="line.657">         */</a>
<span class="sourceLineNo">658</span><a id="line.658">        public static List&lt;Atom[]&gt; transformAtoms(MultipleAlignment alignment) {</a>
<span class="sourceLineNo">659</span><a id="line.659">                if (alignment.getEnsemble() == null) {</a>
<span class="sourceLineNo">660</span><a id="line.660">                        throw new NullPointerException("No ensemble set for this alignment");</a>
<span class="sourceLineNo">661</span><a id="line.661">                }</a>
<span class="sourceLineNo">662</span><a id="line.662"></a>
<span class="sourceLineNo">663</span><a id="line.663">                List&lt;Atom[]&gt; atomArrays = alignment.getAtomArrays();</a>
<span class="sourceLineNo">664</span><a id="line.664">                List&lt;Atom[]&gt; transformed = new ArrayList&lt;Atom[]&gt;(atomArrays.size());</a>
<span class="sourceLineNo">665</span><a id="line.665"></a>
<span class="sourceLineNo">666</span><a id="line.666">                // Loop through structures</a>
<span class="sourceLineNo">667</span><a id="line.667">                for (int i = 0; i &lt; atomArrays.size(); i++) {</a>
<span class="sourceLineNo">668</span><a id="line.668"></a>
<span class="sourceLineNo">669</span><a id="line.669">                        Matrix4d transform = null;</a>
<span class="sourceLineNo">670</span><a id="line.670">                        Atom[] curr = atomArrays.get(i); // all CA atoms from structure</a>
<span class="sourceLineNo">671</span><a id="line.671"></a>
<span class="sourceLineNo">672</span><a id="line.672">                        // Concatenated list of all blocks for this structure</a>
<span class="sourceLineNo">673</span><a id="line.673">                        Atom[] transformedAtoms = new Atom[alignment.length()];</a>
<span class="sourceLineNo">674</span><a id="line.674">                        int transformedAtomsLength = 0;</a>
<span class="sourceLineNo">675</span><a id="line.675"></a>
<span class="sourceLineNo">676</span><a id="line.676">                        // Each blockset gets transformed independently</a>
<span class="sourceLineNo">677</span><a id="line.677">                        for (BlockSet bs : alignment.getBlockSets()) {</a>
<span class="sourceLineNo">678</span><a id="line.678"></a>
<span class="sourceLineNo">679</span><a id="line.679">                                Atom[] blocksetAtoms = new Atom[bs.length()];</a>
<span class="sourceLineNo">680</span><a id="line.680">                                int blockPos = 0;</a>
<span class="sourceLineNo">681</span><a id="line.681"></a>
<span class="sourceLineNo">682</span><a id="line.682">                                for (Block blk : bs.getBlocks()) {</a>
<span class="sourceLineNo">683</span><a id="line.683">                                        if (blk.size() != atomArrays.size()) {</a>
<span class="sourceLineNo">684</span><a id="line.684">                                                throw new IllegalStateException(String.format(</a>
<span class="sourceLineNo">685</span><a id="line.685">                                                                "Mismatched block size. Expected %d "</a>
<span class="sourceLineNo">686</span><a id="line.686">                                                                                + "structures, found %d.",</a>
<span class="sourceLineNo">687</span><a id="line.687">                                                                atomArrays.size(), blk.size()));</a>
<span class="sourceLineNo">688</span><a id="line.688">                                        }</a>
<span class="sourceLineNo">689</span><a id="line.689">                                        // Extract aligned atoms</a>
<span class="sourceLineNo">690</span><a id="line.690">                                        for (int j = 0; j &lt; blk.length(); j++) {</a>
<span class="sourceLineNo">691</span><a id="line.691">                                                Integer alignedPos = blk.getAlignRes().get(i).get(j);</a>
<span class="sourceLineNo">692</span><a id="line.692">                                                if (alignedPos != null) {</a>
<span class="sourceLineNo">693</span><a id="line.693">                                                        blocksetAtoms[blockPos] = (Atom) curr[alignedPos]</a>
<span class="sourceLineNo">694</span><a id="line.694">                                                                        .clone();</a>
<span class="sourceLineNo">695</span><a id="line.695">                                                }</a>
<span class="sourceLineNo">696</span><a id="line.696">                                                blockPos++;</a>
<span class="sourceLineNo">697</span><a id="line.697">                                        }</a>
<span class="sourceLineNo">698</span><a id="line.698">                                }</a>
<span class="sourceLineNo">699</span><a id="line.699"></a>
<span class="sourceLineNo">700</span><a id="line.700">                                // Transform according to the blockset or alignment matrix</a>
<span class="sourceLineNo">701</span><a id="line.701">                                Matrix4d blockTrans = null;</a>
<span class="sourceLineNo">702</span><a id="line.702">                                if (bs.getTransformations() != null)</a>
<span class="sourceLineNo">703</span><a id="line.703">                                        blockTrans = bs.getTransformations().get(i);</a>
<span class="sourceLineNo">704</span><a id="line.704">                                if (blockTrans == null) {</a>
<span class="sourceLineNo">705</span><a id="line.705">                                        blockTrans = transform;</a>
<span class="sourceLineNo">706</span><a id="line.706">                                }</a>
<span class="sourceLineNo">707</span><a id="line.707"></a>
<span class="sourceLineNo">708</span><a id="line.708">                                for (Atom a : blocksetAtoms) {</a>
<span class="sourceLineNo">709</span><a id="line.709">                                        if (a != null)</a>
<span class="sourceLineNo">710</span><a id="line.710">                                                Calc.transform(a, blockTrans);</a>
<span class="sourceLineNo">711</span><a id="line.711">                                        transformedAtoms[transformedAtomsLength] = a;</a>
<span class="sourceLineNo">712</span><a id="line.712">                                        transformedAtomsLength++;</a>
<span class="sourceLineNo">713</span><a id="line.713">                                }</a>
<span class="sourceLineNo">714</span><a id="line.714">                        }</a>
<span class="sourceLineNo">715</span><a id="line.715">                        assert (transformedAtomsLength == alignment.length());</a>
<span class="sourceLineNo">716</span><a id="line.716"></a>
<span class="sourceLineNo">717</span><a id="line.717">                        transformed.add(transformedAtoms);</a>
<span class="sourceLineNo">718</span><a id="line.718">                }</a>
<span class="sourceLineNo">719</span><a id="line.719">                return transformed;</a>
<span class="sourceLineNo">720</span><a id="line.720">        }</a>
<span class="sourceLineNo">721</span><a id="line.721"></a>
<span class="sourceLineNo">722</span><a id="line.722">        /**</a>
<span class="sourceLineNo">723</span><a id="line.723">         * Calculate a List of alignment indicies that correspond to the core of a</a>
<span class="sourceLineNo">724</span><a id="line.724">         * Block, which means that all structures have a residue in that positon.</a>
<span class="sourceLineNo">725</span><a id="line.725">         *</a>
<span class="sourceLineNo">726</span><a id="line.726">         * @param block</a>
<span class="sourceLineNo">727</span><a id="line.727">         *            alignment Block</a>
<span class="sourceLineNo">728</span><a id="line.728">         * @return List of positions in the core of the alignment</a>
<span class="sourceLineNo">729</span><a id="line.729">         */</a>
<span class="sourceLineNo">730</span><a id="line.730">        public static List&lt;Integer&gt; getCorePositions(Block block) {</a>
<span class="sourceLineNo">731</span><a id="line.731"></a>
<span class="sourceLineNo">732</span><a id="line.732">                List&lt;Integer&gt; corePositions = new ArrayList&lt;Integer&gt;();</a>
<span class="sourceLineNo">733</span><a id="line.733"></a>
<span class="sourceLineNo">734</span><a id="line.734">                for (int col = 0; col &lt; block.length(); col++) {</a>
<span class="sourceLineNo">735</span><a id="line.735">                        boolean core = true;</a>
<span class="sourceLineNo">736</span><a id="line.736">                        for (int str = 0; str &lt; block.size(); str++) {</a>
<span class="sourceLineNo">737</span><a id="line.737">                                if (block.getAlignRes().get(str).get(col) == null) {</a>
<span class="sourceLineNo">738</span><a id="line.738">                                        core = false;</a>
<span class="sourceLineNo">739</span><a id="line.739">                                        break;</a>
<span class="sourceLineNo">740</span><a id="line.740">                                }</a>
<span class="sourceLineNo">741</span><a id="line.741">                        }</a>
<span class="sourceLineNo">742</span><a id="line.742">                        if (core)</a>
<span class="sourceLineNo">743</span><a id="line.743">                                corePositions.add(col);</a>
<span class="sourceLineNo">744</span><a id="line.744">                }</a>
<span class="sourceLineNo">745</span><a id="line.745">                return corePositions;</a>
<span class="sourceLineNo">746</span><a id="line.746">        }</a>
<span class="sourceLineNo">747</span><a id="line.747"></a>
<span class="sourceLineNo">748</span><a id="line.748">        /**</a>
<span class="sourceLineNo">749</span><a id="line.749">         * Sort blocks so that the specified row is in sequential order. The sort</a>
<span class="sourceLineNo">750</span><a id="line.750">         * happens in place.</a>
<span class="sourceLineNo">751</span><a id="line.751">         *</a>
<span class="sourceLineNo">752</span><a id="line.752">         * @param blocks</a>
<span class="sourceLineNo">753</span><a id="line.753">         *            List of blocks</a>
<span class="sourceLineNo">754</span><a id="line.754">         * @param sortedIndex</a>
<span class="sourceLineNo">755</span><a id="line.755">         *            Index of the row to be sorted</a>
<span class="sourceLineNo">756</span><a id="line.756">         */</a>
<span class="sourceLineNo">757</span><a id="line.757">        public static void sortBlocks(List&lt;Block&gt; blocks, final int sortedIndex) {</a>
<span class="sourceLineNo">758</span><a id="line.758">                Collections.sort(blocks, new Comparator&lt;Block&gt;() {</a>
<span class="sourceLineNo">759</span><a id="line.759">                        @Override</a>
<span class="sourceLineNo">760</span><a id="line.760">                        public int compare(Block o1, Block o2) {</a>
<span class="sourceLineNo">761</span><a id="line.761">                                // Compare the first non-null residue of each block</a>
<span class="sourceLineNo">762</span><a id="line.762">                                List&lt;Integer&gt; alignres1 = o1.getAlignRes().get(sortedIndex);</a>
<span class="sourceLineNo">763</span><a id="line.763">                                List&lt;Integer&gt; alignres2 = o2.getAlignRes().get(sortedIndex);</a>
<span class="sourceLineNo">764</span><a id="line.764">                                Integer res1 = null;</a>
<span class="sourceLineNo">765</span><a id="line.765">                                Integer res2 = null;</a>
<span class="sourceLineNo">766</span><a id="line.766">                                for (Integer r : alignres1) {</a>
<span class="sourceLineNo">767</span><a id="line.767">                                        if (r != null) {</a>
<span class="sourceLineNo">768</span><a id="line.768">                                                res1 = r;</a>
<span class="sourceLineNo">769</span><a id="line.769">                                                break;</a>
<span class="sourceLineNo">770</span><a id="line.770">                                        }</a>
<span class="sourceLineNo">771</span><a id="line.771">                                }</a>
<span class="sourceLineNo">772</span><a id="line.772">                                for (Integer r : alignres2) {</a>
<span class="sourceLineNo">773</span><a id="line.773">                                        if (r != null) {</a>
<span class="sourceLineNo">774</span><a id="line.774">                                                res2 = r;</a>
<span class="sourceLineNo">775</span><a id="line.775">                                                break;</a>
<span class="sourceLineNo">776</span><a id="line.776">                                        }</a>
<span class="sourceLineNo">777</span><a id="line.777">                                }</a>
<span class="sourceLineNo">778</span><a id="line.778">                                return res1.compareTo(res2);</a>
<span class="sourceLineNo">779</span><a id="line.779">                        }</a>
<span class="sourceLineNo">780</span><a id="line.780">                });</a>
<span class="sourceLineNo">781</span><a id="line.781">        }</a>
<span class="sourceLineNo">782</span><a id="line.782"></a>
<span class="sourceLineNo">783</span><a id="line.783">        /**</a>
<span class="sourceLineNo">784</span><a id="line.784">         * Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid</a>
<span class="sourceLineNo">785</span><a id="line.785">         * residues. This method is only valid for protein structure alignments.</a>
<span class="sourceLineNo">786</span><a id="line.786">         *</a>
<span class="sourceLineNo">787</span><a id="line.787">         * @param msta</a>
<span class="sourceLineNo">788</span><a id="line.788">         *            Multiple Structure Alignment</a>
<span class="sourceLineNo">789</span><a id="line.789">         * @return MultipleSequenceAlignment of protein sequences</a>
<span class="sourceLineNo">790</span><a id="line.790">         * @throws CompoundNotFoundException</a>
<span class="sourceLineNo">791</span><a id="line.791">         */</a>
<span class="sourceLineNo">792</span><a id="line.792">        public static MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; toProteinMSA(</a>
<span class="sourceLineNo">793</span><a id="line.793">                        MultipleAlignment msta) throws CompoundNotFoundException {</a>
<span class="sourceLineNo">794</span><a id="line.794"></a>
<span class="sourceLineNo">795</span><a id="line.795">                // Check that the alignment is of protein structures</a>
<span class="sourceLineNo">796</span><a id="line.796">                Group g = msta.getAtomArrays().get(0)[0].getGroup();</a>
<span class="sourceLineNo">797</span><a id="line.797">                if (!(g instanceof AminoAcid)) {</a>
<span class="sourceLineNo">798</span><a id="line.798">                        throw new IllegalArgumentException(</a>
<span class="sourceLineNo">799</span><a id="line.799">                                        "Cannot convert to multiple sequence alignment: "</a>
<span class="sourceLineNo">800</span><a id="line.800">                                                        + "the structures aligned are not proteins");</a>
<span class="sourceLineNo">801</span><a id="line.801">                }</a>
<span class="sourceLineNo">802</span><a id="line.802"></a>
<span class="sourceLineNo">803</span><a id="line.803">                MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; msa = new MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt;();</a>
<span class="sourceLineNo">804</span><a id="line.804"></a>
<span class="sourceLineNo">805</span><a id="line.805">                Map&lt;String, Integer&gt; uniqueID = new HashMap&lt;String, Integer&gt;();</a>
<span class="sourceLineNo">806</span><a id="line.806">                List&lt;String&gt; seqs = getSequenceAlignment(msta);</a>
<span class="sourceLineNo">807</span><a id="line.807">                for (int i = 0; i &lt; msta.size(); i++) {</a>
<span class="sourceLineNo">808</span><a id="line.808">                        // Make sure the identifiers are unique (required by AccessionID)</a>
<span class="sourceLineNo">809</span><a id="line.809">                        String id = msta.getStructureIdentifier(i).toString();</a>
<span class="sourceLineNo">810</span><a id="line.810">                        if (uniqueID.containsKey(id)) {</a>
<span class="sourceLineNo">811</span><a id="line.811">                                uniqueID.put(id, uniqueID.get(id) + 1);</a>
<span class="sourceLineNo">812</span><a id="line.812">                                id += "_" + uniqueID.get(id);</a>
<span class="sourceLineNo">813</span><a id="line.813">                        } else</a>
<span class="sourceLineNo">814</span><a id="line.814">                                uniqueID.put(id, 1);</a>
<span class="sourceLineNo">815</span><a id="line.815"></a>
<span class="sourceLineNo">816</span><a id="line.816">                        AccessionID acc = new AccessionID(id);</a>
<span class="sourceLineNo">817</span><a id="line.817">                        ProteinSequence pseq = new ProteinSequence(seqs.get(i));</a>
<span class="sourceLineNo">818</span><a id="line.818">                        pseq.setAccession(acc);</a>
<span class="sourceLineNo">819</span><a id="line.819">                        msa.addAlignedSequence(pseq);</a>
<span class="sourceLineNo">820</span><a id="line.820">                }</a>
<span class="sourceLineNo">821</span><a id="line.821">                return msa;</a>
<span class="sourceLineNo">822</span><a id="line.822">        }</a>
<span class="sourceLineNo">823</span><a id="line.823"></a>
<span class="sourceLineNo">824</span><a id="line.824">        public static Structure toMultimodelStructure(MultipleAlignment multAln, List&lt;Atom[]&gt; transformedAtoms) throws StructureException {</a>
<span class="sourceLineNo">825</span><a id="line.825">                PDBHeader header = new PDBHeader();</a>
<span class="sourceLineNo">826</span><a id="line.826">                String title = multAln.getEnsemble().getAlgorithmName() + " V."</a>
<span class="sourceLineNo">827</span><a id="line.827">                                + multAln.getEnsemble().getVersion() + " : ";</a>
<span class="sourceLineNo">828</span><a id="line.828"></a>
<span class="sourceLineNo">829</span><a id="line.829">                for (StructureIdentifier name : multAln.getEnsemble()</a>
<span class="sourceLineNo">830</span><a id="line.830">                                .getStructureIdentifiers()) {</a>
<span class="sourceLineNo">831</span><a id="line.831">                        title += name.getIdentifier() + " ";</a>
<span class="sourceLineNo">832</span><a id="line.832">                }</a>
<span class="sourceLineNo">833</span><a id="line.833">                Structure artificial = getAlignedStructure(transformedAtoms);</a>
<span class="sourceLineNo">834</span><a id="line.834"></a>
<span class="sourceLineNo">835</span><a id="line.835">                artificial.setPDBHeader(header);</a>
<span class="sourceLineNo">836</span><a id="line.836">                header.setTitle(title);</a>
<span class="sourceLineNo">837</span><a id="line.837">                return artificial;</a>
<span class="sourceLineNo">838</span><a id="line.838">        }</a>
<span class="sourceLineNo">839</span><a id="line.839"></a>
<span class="sourceLineNo">840</span><a id="line.840">        /**</a>
<span class="sourceLineNo">841</span><a id="line.841">         * Get an artificial Structure containing a different model for every</a>
<span class="sourceLineNo">842</span><a id="line.842">         * input structure, so that the alignment result can be viewed in Jmol.</a>
<span class="sourceLineNo">843</span><a id="line.843">         * The Atoms have to be rotated beforehand.</a>
<span class="sourceLineNo">844</span><a id="line.844">         *</a>
<span class="sourceLineNo">845</span><a id="line.845">         * @param atomArrays an array of Atoms for every aligned structure</a>
<span class="sourceLineNo">846</span><a id="line.846">         * @return a structure object containing a set of models,</a>
<span class="sourceLineNo">847</span><a id="line.847">         *                      one for each input array of Atoms.</a>
<span class="sourceLineNo">848</span><a id="line.848">         * @throws StructureException</a>
<span class="sourceLineNo">849</span><a id="line.849">         */</a>
<span class="sourceLineNo">850</span><a id="line.850">        public static final Structure getAlignedStructure(List&lt;Atom[]&gt; atomArrays)</a>
<span class="sourceLineNo">851</span><a id="line.851">                        throws StructureException {</a>
<span class="sourceLineNo">852</span><a id="line.852"></a>
<span class="sourceLineNo">853</span><a id="line.853">                Structure s = new StructureImpl();</a>
<span class="sourceLineNo">854</span><a id="line.854">                for (int i=0; i&lt;atomArrays.size(); i++){</a>
<span class="sourceLineNo">855</span><a id="line.855">                        List&lt;Chain&gt; model = AlignmentTools.getAlignedModel(atomArrays.get(i));</a>
<span class="sourceLineNo">856</span><a id="line.856">                        s.addModel(model);</a>
<span class="sourceLineNo">857</span><a id="line.857">                }</a>
<span class="sourceLineNo">858</span><a id="line.858">                return s;</a>
<span class="sourceLineNo">859</span><a id="line.859">        }</a>
<span class="sourceLineNo">860</span><a id="line.860"></a>
<span class="sourceLineNo">861</span><a id="line.861">        /**</a>
<span class="sourceLineNo">862</span><a id="line.862">         * Calculate the RMSD matrix of a MultipleAlignment, that is, entry (i,j) of</a>
<span class="sourceLineNo">863</span><a id="line.863">         * the matrix contains the RMSD between structures i and j.</a>
<span class="sourceLineNo">864</span><a id="line.864">         *</a>
<span class="sourceLineNo">865</span><a id="line.865">         * @param msa</a>
<span class="sourceLineNo">866</span><a id="line.866">         *            Multiple Structure Alignment</a>
<span class="sourceLineNo">867</span><a id="line.867">         * @return Matrix of RMSD with size the number of structures squared</a>
<span class="sourceLineNo">868</span><a id="line.868">         */</a>
<span class="sourceLineNo">869</span><a id="line.869">        public static Matrix getRMSDMatrix(MultipleAlignment msa) {</a>
<span class="sourceLineNo">870</span><a id="line.870"></a>
<span class="sourceLineNo">871</span><a id="line.871">                Matrix rmsdMat = new Matrix(msa.size(), msa.size());</a>
<span class="sourceLineNo">872</span><a id="line.872">                List&lt;Atom[]&gt; superposed = transformAtoms(msa);</a>
<span class="sourceLineNo">873</span><a id="line.873"></a>
<span class="sourceLineNo">874</span><a id="line.874">                for (int i = 0; i &lt; msa.size(); i++) {</a>
<span class="sourceLineNo">875</span><a id="line.875">                        for (int j = i; j &lt; msa.size(); j++) {</a>
<span class="sourceLineNo">876</span><a id="line.876">                                if (i == j)</a>
<span class="sourceLineNo">877</span><a id="line.877">                                        rmsdMat.set(i, j, 0.0);</a>
<span class="sourceLineNo">878</span><a id="line.878">                                List&lt;Atom[]&gt; compared = new ArrayList&lt;Atom[]&gt;();</a>
<span class="sourceLineNo">879</span><a id="line.879">                                compared.add(superposed.get(i));</a>
<span class="sourceLineNo">880</span><a id="line.880">                                compared.add(superposed.get(j));</a>
<span class="sourceLineNo">881</span><a id="line.881">                                double rmsd = MultipleAlignmentScorer.getRMSD(compared);</a>
<span class="sourceLineNo">882</span><a id="line.882">                                rmsdMat.set(i, j, rmsd);</a>
<span class="sourceLineNo">883</span><a id="line.883">                                rmsdMat.set(j, i, rmsd);</a>
<span class="sourceLineNo">884</span><a id="line.884">                        }</a>
<span class="sourceLineNo">885</span><a id="line.885">                }</a>
<span class="sourceLineNo">886</span><a id="line.886">                return rmsdMat;</a>
<span class="sourceLineNo">887</span><a id="line.887">        }</a>
<span class="sourceLineNo">888</span><a id="line.888"></a>
<span class="sourceLineNo">889</span><a id="line.889">        /**</a>
<span class="sourceLineNo">890</span><a id="line.890">         * Calculate a phylogenetic tree of the MultipleAlignment using Kimura</a>
<span class="sourceLineNo">891</span><a id="line.891">         * distances and the Neighbor Joining algorithm from forester.</a>
<span class="sourceLineNo">892</span><a id="line.892">         *</a>
<span class="sourceLineNo">893</span><a id="line.893">         * @param msta</a>
<span class="sourceLineNo">894</span><a id="line.894">         *            MultipleAlignment of protein structures</a>
<span class="sourceLineNo">895</span><a id="line.895">         * @return Phylogeny phylogenetic tree</a>
<span class="sourceLineNo">896</span><a id="line.896">         * @throws CompoundNotFoundException</a>
<span class="sourceLineNo">897</span><a id="line.897">         * @throws IOException</a>
<span class="sourceLineNo">898</span><a id="line.898">         */</a>
<span class="sourceLineNo">899</span><a id="line.899">        public static Phylogeny getKimuraTree(MultipleAlignment msta)</a>
<span class="sourceLineNo">900</span><a id="line.900">                        throws CompoundNotFoundException, IOException {</a>
<span class="sourceLineNo">901</span><a id="line.901">                MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; msa = MultipleAlignmentTools</a>
<span class="sourceLineNo">902</span><a id="line.902">                                .toProteinMSA(msta);</a>
<span class="sourceLineNo">903</span><a id="line.903">                BasicSymmetricalDistanceMatrix distmat = (BasicSymmetricalDistanceMatrix) DistanceMatrixCalculator</a>
<span class="sourceLineNo">904</span><a id="line.904">                                .kimuraDistance(msa);</a>
<span class="sourceLineNo">905</span><a id="line.905">                Phylogeny tree = TreeConstructor.distanceTree(distmat,</a>
<span class="sourceLineNo">906</span><a id="line.906">                                TreeConstructorType.NJ);</a>
<span class="sourceLineNo">907</span><a id="line.907">                tree.setName("Kimura Tree");</a>
<span class="sourceLineNo">908</span><a id="line.908">                return tree;</a>
<span class="sourceLineNo">909</span><a id="line.909">        }</a>
<span class="sourceLineNo">910</span><a id="line.910"></a>
<span class="sourceLineNo">911</span><a id="line.911">        /**</a>
<span class="sourceLineNo">912</span><a id="line.912">         * Calculate a phylogenetic tree of the MultipleAlignment using</a>
<span class="sourceLineNo">913</span><a id="line.913">         * dissimilarity scores (DS), based in SDM Substitution Matrix (ideal for</a>
<span class="sourceLineNo">914</span><a id="line.914">         * distantly related proteins, structure-derived) and the Neighbor Joining</a>
<span class="sourceLineNo">915</span><a id="line.915">         * algorithm from forester.</a>
<span class="sourceLineNo">916</span><a id="line.916">         *</a>
<span class="sourceLineNo">917</span><a id="line.917">         * @param msta</a>
<span class="sourceLineNo">918</span><a id="line.918">         *            MultipleAlignment of protein structures</a>
<span class="sourceLineNo">919</span><a id="line.919">         * @return Phylogeny phylogenetic tree</a>
<span class="sourceLineNo">920</span><a id="line.920">         * @throws CompoundNotFoundException</a>
<span class="sourceLineNo">921</span><a id="line.921">         * @throws IOException</a>
<span class="sourceLineNo">922</span><a id="line.922">         */</a>
<span class="sourceLineNo">923</span><a id="line.923">        public static Phylogeny getHSDMTree(MultipleAlignment msta)</a>
<span class="sourceLineNo">924</span><a id="line.924">                        throws CompoundNotFoundException, IOException {</a>
<span class="sourceLineNo">925</span><a id="line.925">                MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; msa = MultipleAlignmentTools</a>
<span class="sourceLineNo">926</span><a id="line.926">                                .toProteinMSA(msta);</a>
<span class="sourceLineNo">927</span><a id="line.927">                BasicSymmetricalDistanceMatrix distmat = (BasicSymmetricalDistanceMatrix) DistanceMatrixCalculator</a>
<span class="sourceLineNo">928</span><a id="line.928">                                .dissimilarityScore(msa, SubstitutionMatrixHelper.getAminoAcidSubstitutionMatrix("PRLA000102"));</a>
<span class="sourceLineNo">929</span><a id="line.929">                Phylogeny tree = TreeConstructor.distanceTree(distmat,</a>
<span class="sourceLineNo">930</span><a id="line.930">                                TreeConstructorType.NJ);</a>
<span class="sourceLineNo">931</span><a id="line.931">                tree.setName("HSDM Tree");</a>
<span class="sourceLineNo">932</span><a id="line.932">                return tree;</a>
<span class="sourceLineNo">933</span><a id="line.933">        }</a>
<span class="sourceLineNo">934</span><a id="line.934"></a>
<span class="sourceLineNo">935</span><a id="line.935">        /**</a>
<span class="sourceLineNo">936</span><a id="line.936">         * Calculate a phylogenetic tree of the MultipleAlignment using RMSD</a>
<span class="sourceLineNo">937</span><a id="line.937">         * distances and the Neighbor Joining algorithm from forester.</a>
<span class="sourceLineNo">938</span><a id="line.938">         *</a>
<span class="sourceLineNo">939</span><a id="line.939">         * @param msta</a>
<span class="sourceLineNo">940</span><a id="line.940">         *            MultipleAlignment of protein structures</a>
<span class="sourceLineNo">941</span><a id="line.941">         * @return Phylogeny phylogenetic tree</a>
<span class="sourceLineNo">942</span><a id="line.942">         * @throws CompoundNotFoundException</a>
<span class="sourceLineNo">943</span><a id="line.943">         */</a>
<span class="sourceLineNo">944</span><a id="line.944">        public static Phylogeny getStructuralTree(MultipleAlignment msta) {</a>
<span class="sourceLineNo">945</span><a id="line.945">                double[][] rmsdMat = MultipleAlignmentTools.getRMSDMatrix(msta)</a>
<span class="sourceLineNo">946</span><a id="line.946">                                .getArray();</a>
<span class="sourceLineNo">947</span><a id="line.947">                BasicSymmetricalDistanceMatrix rmsdDist = (BasicSymmetricalDistanceMatrix) DistanceMatrixCalculator</a>
<span class="sourceLineNo">948</span><a id="line.948">                                .structuralDistance(rmsdMat, 1, 5, 0.4);</a>
<span class="sourceLineNo">949</span><a id="line.949">                // Set the identifiers of the matrix</a>
<span class="sourceLineNo">950</span><a id="line.950">                Map&lt;String, Integer&gt; alreadySeen = new HashMap&lt;String, Integer&gt;();</a>
<span class="sourceLineNo">951</span><a id="line.951">                for (int i = 0; i &lt; msta.size(); i++) {</a>
<span class="sourceLineNo">952</span><a id="line.952">                        // Make sure the identifiers are unique</a>
<span class="sourceLineNo">953</span><a id="line.953">                        String id = msta.getStructureIdentifier(i).toString();</a>
<span class="sourceLineNo">954</span><a id="line.954">                        if (alreadySeen.containsKey(id)) {</a>
<span class="sourceLineNo">955</span><a id="line.955">                                alreadySeen.put(id, alreadySeen.get(id) + 1);</a>
<span class="sourceLineNo">956</span><a id="line.956">                                id += "_" + alreadySeen.get(id);</a>
<span class="sourceLineNo">957</span><a id="line.957">                        } else</a>
<span class="sourceLineNo">958</span><a id="line.958">                                alreadySeen.put(id, 1);</a>
<span class="sourceLineNo">959</span><a id="line.959">                        rmsdDist.setIdentifier(i, id);</a>
<span class="sourceLineNo">960</span><a id="line.960">                }</a>
<span class="sourceLineNo">961</span><a id="line.961">                Phylogeny tree = TreeConstructor.distanceTree(rmsdDist,</a>
<span class="sourceLineNo">962</span><a id="line.962">                                TreeConstructorType.NJ);</a>
<span class="sourceLineNo">963</span><a id="line.963">                tree.setName("Structural Tree");</a>
<span class="sourceLineNo">964</span><a id="line.964">                return tree;</a>
<span class="sourceLineNo">965</span><a id="line.965">        }</a>
<span class="sourceLineNo">966</span><a id="line.966"></a>
<span class="sourceLineNo">967</span><a id="line.967">}</a>




























































</pre>
</div>
</main>
</body>
</html>
