<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.16) on Fri Oct 28 16:15:48 PDT 2022 -->
<title>Uses of Class org.biojava.nbio.structure.StructureException (biojava 6.1.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2022-10-28">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class org.biojava.nbio.structure.StructureException (biojava 6.1.0 API)";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../StructureException.html" title="class in org.biojava.nbio.structure">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Class org.biojava.nbio.structure.StructureException" class="title">Uses of Class<br>org.biojava.nbio.structure.StructureException</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#demo">demo</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure">org.biojava.nbio.structure</a></th>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align">org.biojava.nbio.structure.align</a></th>
<td class="colLast">
<div class="block">Classes for the alignment of structures.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.ce">org.biojava.nbio.structure.align.ce</a></th>
<td class="colLast">
<div class="block">Classes related to the implementation of the CE alignment algorithm, here called jCE.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.client">org.biojava.nbio.structure.align.client</a></th>
<td class="colLast">
<div class="block">This package deals with the server communication for auto-downloading pre-calculated alignments.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.fatcat">org.biojava.nbio.structure.align.fatcat</a></th>
<td class="colLast">
<div class="block">Classes related to the implementation of the FATCAT alignment algorithm, here called jFATCAT.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.fatcat.calc">org.biojava.nbio.structure.align.fatcat.calc</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.gui">org.biojava.nbio.structure.align.gui</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.gui.aligpanel">org.biojava.nbio.structure.align.gui.aligpanel</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.multiple">org.biojava.nbio.structure.align.multiple</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.multiple.mc">org.biojava.nbio.structure.align.multiple.mc</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.pairwise">org.biojava.nbio.structure.align.pairwise</a></th>
<td class="colLast">
<div class="block">Classes for the pairwise alignment of structures.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.quaternary">org.biojava.nbio.structure.align.quaternary</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.seq">org.biojava.nbio.structure.align.seq</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.util">org.biojava.nbio.structure.align.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.xml">org.biojava.nbio.structure.align.xml</a></th>
<td class="colLast">
<div class="block">Serialization and deserialization for structure alignment results (as AFPChain object) to XML.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.cath">org.biojava.nbio.structure.cath</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.cluster">org.biojava.nbio.structure.cluster</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.domain">org.biojava.nbio.structure.domain</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.domain.pdp">org.biojava.nbio.structure.domain.pdp</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.ecod">org.biojava.nbio.structure.ecod</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.gui.util">org.biojava.nbio.structure.gui.util</a></th>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></th>
<td class="colLast">
<div class="block">Input and Output of Structures</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.scop">org.biojava.nbio.structure.scop</a></th>
<td class="colLast">
<div class="block">Parsers and API for <a href="http://scop.mrc-lmb.cam.ac.uk/scop/">SCOP, Structural Classification of Proteins</a>.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.secstruc">org.biojava.nbio.structure.secstruc</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.symmetry.gui">org.biojava.nbio.structure.symmetry.gui</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.symmetry.internal">org.biojava.nbio.structure.symmetry.internal</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.symmetry.utils">org.biojava.nbio.structure.symmetry.utils</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="demo">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../demo/package-summary.html">demo</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../../demo/package-summary.html">demo</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DemoAlignmentFromFasta.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoAlignmentFromFasta.html#getAlignmentFromFasta()">getAlignmentFromFasta</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPFromFasta.</span><code><span class="memberNameLink"><a href="../../../../../demo/AFPFromFasta.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DemoAlignmentFromFasta.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoAlignmentFromFasta.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DemoAsa.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoAsa.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DemoCeSymm.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoCeSymm.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DemoContacts.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoContacts.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DemoLoadSecStruc.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoLoadSecStruc.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DemoMmtfReader.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoMmtfReader.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">
<div class="block">Main function to run the demo</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DemoMmtfWriter.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoMmtfWriter.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DemoMultipleMC.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoMultipleMC.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DemoQsAlign.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoQsAlign.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DemoQuatSymmetryJmol.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoQuatSymmetryJmol.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DemoSecStrucCalc.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoSecStrucCalc.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../package-summary.html">org.biojava.nbio.structure</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#centerAtoms(org.biojava.nbio.structure.Atom%5B%5D)">centerAtoms</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></th>
<td class="colLast">
<div class="block">Center the atoms at the Centroid.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#centerAtoms(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom)">centerAtoms</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code></th>
<td class="colLast">
<div class="block">Center the atoms at the Centroid, if the centroid is already know.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#createVirtualCBAtom(org.biojava.nbio.structure.AminoAcid)">createVirtualCBAtom</a></span>&#8203;(<a href="../AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;amino)</code></th>
<td class="colLast">
<div class="block">creates a virtual C-beta atom. this might be needed when working with GLY

 thanks to Peter Lackner for a python template of this method.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Structure.</span><code><span class="memberNameLink"><a href="../Structure.html#findGroup(java.lang.String,java.lang.String)">findGroup</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;authId,
         <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbResnum)</code></th>
<td class="colLast">
<div class="block">Request a particular group from a structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Structure.</span><code><span class="memberNameLink"><a href="../Structure.html#findGroup(java.lang.String,java.lang.String,int)">findGroup</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;authId,
         <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbResnum,
         int&nbsp;modelnr)</code></th>
<td class="colLast">
<div class="block">Request a particular group from a structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureImpl.</span><code><span class="memberNameLink"><a href="../StructureImpl.html#findGroup(java.lang.String,java.lang.String)">findGroup</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chainName,
         <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbResnum)</code></th>
<td class="colLast">
<div class="block">Request a particular group from a structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureImpl.</span><code><span class="memberNameLink"><a href="../StructureImpl.html#findGroup(java.lang.String,java.lang.String,int)">findGroup</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chainName,
         <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbResnum,
         int&nbsp;modelnr)</code></th>
<td class="colLast">
<div class="block">Request a particular group from a structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../StructureIO.html#getBiologicalAssemblies(java.lang.String)">getBiologicalAssemblies</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbId)</code></th>
<td class="colLast">
<div class="block">Returns all biological assemblies for the given PDB id,
 using multiModel=<a href="../align/util/AtomCache.html#DEFAULT_BIOASSEMBLY_STYLE">false</a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../StructureIO.html#getBiologicalAssemblies(java.lang.String,boolean)">getBiologicalAssemblies</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbId,
                       boolean&nbsp;multiModel)</code></th>
<td class="colLast">
<div class="block">Returns all biological assemblies for the given PDB id.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../StructureIO.html#getBiologicalAssembly(java.lang.String)">getBiologicalAssembly</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbId)</code></th>
<td class="colLast">
<div class="block">Returns the first biological assembly that is available for the given PDB id,
 using multiModel=<a href="../align/util/AtomCache.html#DEFAULT_BIOASSEMBLY_STYLE">false</a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../StructureIO.html#getBiologicalAssembly(java.lang.String,boolean)">getBiologicalAssembly</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbId,
                     boolean&nbsp;multiModel)</code></th>
<td class="colLast">
<div class="block">Returns the first biological assembly that is available for the given PDB id.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../StructureIO.html#getBiologicalAssembly(java.lang.String,int)">getBiologicalAssembly</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbId,
                     int&nbsp;biolAssemblyNr)</code></th>
<td class="colLast">
<div class="block">Returns the biological assembly for the given PDB id and bioassembly identifier,
 using multiModel=<a href="../align/util/AtomCache.html#DEFAULT_BIOASSEMBLY_STYLE">false</a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../StructureIO.html#getBiologicalAssembly(java.lang.String,int,boolean)">getBiologicalAssembly</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbId,
                     int&nbsp;biolAssemblyNr,
                     boolean&nbsp;multiModel)</code></th>
<td class="colLast">
<div class="block">Returns the biological assembly for the given PDB id and bioassembly identifier.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chain.</span><code><span class="memberNameLink"><a href="../Chain.html#getGroupByPDB(org.biojava.nbio.structure.ResidueNumber)">getGroupByPDB</a></span>&#8203;(<a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;resNum)</code></th>
<td class="colLast">
<div class="block">Get a group by its PDB residue numbering.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChainImpl.</span><code><span class="memberNameLink"><a href="../ChainImpl.html#getGroupByPDB(org.biojava.nbio.structure.ResidueNumber)">getGroupByPDB</a></span>&#8203;(<a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;resNum)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getGroupByPDBResidueNumber(org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.ResidueNumber)">getGroupByPDBResidueNumber</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struc,
                          <a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbResNum)</code></th>
<td class="colLast">
<div class="block">Get a group represented by a ResidueNumber.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chain.</span><code><span class="memberNameLink"><a href="../Chain.html#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,org.biojava.nbio.structure.ResidueNumber)">getGroupsByPDB</a></span>&#8203;(<a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
              <a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd)</code></th>
<td class="colLast">
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Chain.</span><code><span class="memberNameLink"><a href="../Chain.html#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,org.biojava.nbio.structure.ResidueNumber,boolean)">getGroupsByPDB</a></span>&#8203;(<a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
              <a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd,
              boolean&nbsp;ignoreMissing)</code></th>
<td class="colLast">
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChainImpl.</span><code><span class="memberNameLink"><a href="../ChainImpl.html#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,org.biojava.nbio.structure.ResidueNumber)">getGroupsByPDB</a></span>&#8203;(<a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;start,
              <a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;end)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChainImpl.</span><code><span class="memberNameLink"><a href="../ChainImpl.html#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,org.biojava.nbio.structure.ResidueNumber,boolean)">getGroupsByPDB</a></span>&#8203;(<a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;start,
              <a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;end,
              boolean&nbsp;ignoreMissing)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getPhi(org.biojava.nbio.structure.AminoAcid,org.biojava.nbio.structure.AminoAcid)">getPhi</a></span>&#8203;(<a href="../AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;a,
      <a href="../AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">Calculate the phi angle.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getPsi(org.biojava.nbio.structure.AminoAcid,org.biojava.nbio.structure.AminoAcid)">getPsi</a></span>&#8203;(<a href="../AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;a,
      <a href="../AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">Calculate the psi angle.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PdbId.</span><code><span class="memberNameLink"><a href="../PdbId.html#getShortId()">getShortId</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the PDB Id in the short format.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../StructureIO.html#getStructure(java.lang.String)">getStructure</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Loads a structure based on a name.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getStructure(java.lang.String)">getStructure</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Short version of <a href="../StructureTools.html#getStructure(java.lang.String,org.biojava.nbio.structure.io.PDBFileParser,org.biojava.nbio.structure.align.util.AtomCache)"><code>StructureTools.getStructure(String, PDBFileParser, AtomCache)</code></a>
 which creates new parsers when needed</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getStructure(java.lang.String,org.biojava.nbio.structure.io.PDBFileParser,org.biojava.nbio.structure.align.util.AtomCache)">getStructure</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name,
            <a href="../io/PDBFileParser.html" title="class in org.biojava.nbio.structure.io">PDBFileParser</a>&nbsp;parser,
            <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">
<div class="block">Flexibly get a structure from an input String.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getTMScore(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int)">getTMScore</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2,
          int&nbsp;len1,
          int&nbsp;len2)</code></th>
<td class="colLast">
<div class="block">Calculate the TM-Score for the superposition.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getTMScore(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int,boolean)">getTMScore</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2,
          int&nbsp;len1,
          int&nbsp;len2,
          boolean&nbsp;normalizeMin)</code></th>
<td class="colLast">
<div class="block">Calculate the TM-Score for the superposition.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BioAssemblyIdentifier.</span><code><span class="memberNameLink"><a href="../BioAssemblyIdentifier.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></span>&#8203;(<a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PassthroughIdentifier.</span><code><span class="memberNameLink"><a href="../PassthroughIdentifier.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></span>&#8203;(<a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">
<div class="block">Passthrough identifiers don't know how to load a structure</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureIdentifier.</span><code><span class="memberNameLink"><a href="../StructureIdentifier.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></span>&#8203;(<a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">
<div class="block">Loads a structure encompassing the structure identified.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SubstructureIdentifier.</span><code><span class="memberNameLink"><a href="../SubstructureIdentifier.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></span>&#8203;(<a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">
<div class="block">Loads the complete structure based on <a href="../SubstructureIdentifier.html#getPdbId()"><code>SubstructureIdentifier.getPdbId()</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">URLIdentifier.</span><code><span class="memberNameLink"><a href="../URLIdentifier.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></span>&#8203;(<a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">
<div class="block">Load the structure from the URL</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BioAssemblyIdentifier.</span><code><span class="memberNameLink"><a href="../BioAssemblyIdentifier.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PassthroughIdentifier.</span><code><span class="memberNameLink"><a href="../PassthroughIdentifier.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureIdentifier.</span><code><span class="memberNameLink"><a href="../StructureIdentifier.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code></th>
<td class="colLast">
<div class="block">Takes a complete structure as input and reduces it to the substructure
 represented by this StructureIdentifier.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SubstructureIdentifier.</span><code><span class="memberNameLink"><a href="../SubstructureIdentifier.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code></th>
<td class="colLast">
<div class="block">Takes a complete structure as input and reduces it to residues present in
 the specified ranges</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">URLIdentifier.</span><code><span class="memberNameLink"><a href="../URLIdentifier.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#rotate(org.biojava.nbio.structure.Group,double%5B%5D%5B%5D)">rotate</a></span>&#8203;(<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group,
      double[][]&nbsp;rotationmatrix)</code></th>
<td class="colLast">
<div class="block">Rotate a Group.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#rotate(org.biojava.nbio.structure.Structure,double%5B%5D%5B%5D)">rotate</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
      double[][]&nbsp;rotationmatrix)</code></th>
<td class="colLast">
<div class="block">Rotate a structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../SubstructureIdentifier.html" title="class in org.biojava.nbio.structure">SubstructureIdentifier</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BioAssemblyIdentifier.</span><code><span class="memberNameLink"><a href="../BioAssemblyIdentifier.html#toCanonical()">toCanonical</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../SubstructureIdentifier.html" title="class in org.biojava.nbio.structure">SubstructureIdentifier</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureIdentifier.</span><code><span class="memberNameLink"><a href="../StructureIdentifier.html#toCanonical()">toCanonical</a></span>()</code></th>
<td class="colLast">
<div class="block">Convert to a canonical SubstructureIdentifier.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../SubstructureIdentifier.html" title="class in org.biojava.nbio.structure">SubstructureIdentifier</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">URLIdentifier.</span><code><span class="memberNameLink"><a href="../URLIdentifier.html#toCanonical()">toCanonical</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PdbId.</span><code><span class="memberNameLink"><a href="../PdbId.html#toExtendedId(java.lang.String)">toExtendedId</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;shortId)</code></th>
<td class="colLast">
<div class="block">Converts <code>shortId</code> to the PDB ID extended format.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PdbId.</span><code><span class="memberNameLink"><a href="../PdbId.html#toShortId(java.lang.String)">toShortId</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;extendedId)</code></th>
<td class="colLast">
<div class="block">Converts <code>extendedId</code> to the PDB ID short format.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/package-summary.html">org.biojava.nbio.structure.align</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/package-summary.html">org.biojava.nbio.structure.align</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractStructureAlignment.</span><code><span class="memberNameLink"><a href="../align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractStructureAlignment.</span><code><span class="memberNameLink"><a href="../align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BioJavaStructureAlignment.</span><code><span class="memberNameLink"><a href="../align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BioJavaStructureAlignment.</span><code><span class="memberNameLink"><a href="../align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleStructureAligner.</span><code><span class="memberNameLink"><a href="../align/MultipleStructureAligner.html#align(java.util.List)">align</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code></th>
<td class="colLast">
<div class="block">Run an alignment while specifying the atoms to be aligned.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleStructureAligner.</span><code><span class="memberNameLink"><a href="../align/MultipleStructureAligner.html#align(java.util.List,java.lang.Object)">align</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Run an alignment and also send a bean containing the parameters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureAlignment.</span><code><span class="memberNameLink"><a href="../align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Run an alignment while specifying the atoms to be aligned.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureAlignment.</span><code><span class="memberNameLink"><a href="../align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">run an alignment and also send a bean containing the parameters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructurePairAligner.</span><code><span class="memberNameLink"><a href="../align/StructurePairAligner.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.StrucAligParameters)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Calculate the protein structure superimposition, between two sets of
 atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructurePairAligner.</span><code><span class="memberNameLink"><a href="../align/StructurePairAligner.html#align(org.biojava.nbio.structure.Structure,java.lang.String,org.biojava.nbio.structure.Structure,java.lang.String)">align</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chainId1,
     <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chainId2)</code></th>
<td class="colLast">
<div class="block">Align two chains from the structures.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructurePairAligner.</span><code><span class="memberNameLink"><a href="../align/StructurePairAligner.html#align(org.biojava.nbio.structure.Structure,java.lang.String,org.biojava.nbio.structure.Structure,java.lang.String,org.biojava.nbio.structure.align.StrucAligParameters)">align</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chainId1,
     <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chainId2,
     <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Aligns two chains from the structures using user provided parameters.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructurePairAligner.</span><code><span class="memberNameLink"><a href="../align/StructurePairAligner.html#align(org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure)">align</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2)</code></th>
<td class="colLast">
<div class="block">Calculate the alignment between the two full structures with default
 parameters</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructurePairAligner.</span><code><span class="memberNameLink"><a href="../align/StructurePairAligner.html#align(org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.align.StrucAligParameters)">align</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2,
     <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Calculate the alignment between the two full structures with user
 provided parameters</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/StructureAlignment.html" title="interface in org.biojava.nbio.structure.align">StructureAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureAlignmentFactory.</span><code><span class="memberNameLink"><a href="../align/StructureAlignmentFactory.html#getAlgorithm(java.lang.String)">getAlgorithm</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPTwister.</span><code><span class="memberNameLink"><a href="../align/AFPTwister.html#twistOptimized(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">twistOptimized</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">superimposing according to the optimized alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPTwister.</span><code><span class="memberNameLink"><a href="../align/AFPTwister.html#twistPDB(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">twistPDB</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">calculate the total rmsd of the blocks output a merged pdb file for both
 proteins protein 1, in chain A protein 2 is twisted according to the
 twists detected, in chain B</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.ce">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCPMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeCPMain.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param)</code></th>
<td class="colLast">
<div class="block">Aligns ca1 and ca2 using a heuristic to check for CPs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param)</code></th>
<td class="colLast">
<div class="block">Align ca2 onto ca1.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">OptimalCECPMain.</span><code><span class="memberNameLink"><a href="../align/ce/OptimalCECPMain.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param)</code></th>
<td class="colLast">
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">OptimalCECPMain.</span><code><span class="memberNameLink"><a href="../align/ce/OptimalCECPMain.html#alignOptimal(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object,org.biojava.nbio.structure.align.model.AFPChain%5B%5D)">alignOptimal</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param,
            <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>[]&nbsp;alignments)</code></th>
<td class="colLast">
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">OptimalCECPMain.</span><code><span class="memberNameLink"><a href="../align/ce/OptimalCECPMain.html#alignPermuted(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object,int)">alignPermuted</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param,
             int&nbsp;cp)</code></th>
<td class="colLast">
<div class="block">Aligns ca1 with ca2 permuted by <i>cp</i> residues.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CECalculator.</span><code><span class="memberNameLink"><a href="../align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,boolean)">calc_rmsd</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform)</code></th>
<td class="colLast">
<div class="block">superimpose and get rmsd</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCalculatorEnhanced.</span><code><span class="memberNameLink"><a href="../align/ce/CeCalculatorEnhanced.html#calc_rmsd(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,boolean,boolean)">calc_rmsd</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform,
         boolean&nbsp;show)</code></th>
<td class="colLast">
<div class="block">superimpose and get rmsd</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CECalculator.</span><code><span class="memberNameLink"><a href="../align/ce/CECalculator.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">extractFragments</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCalculatorEnhanced.</span><code><span class="memberNameLink"><a href="../align/ce/CeCalculatorEnhanced.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">extractFragments</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCPMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">filterDuplicateAFPs</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated)</code></th>
<td class="colLast">
<div class="block">Takes as input an AFPChain where ca2 has been artificially duplicated.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCPMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CECPParameters)">filterDuplicateAFPs</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated,
                   <a href="../align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CECalculator.</span><code><span class="memberNameLink"><a href="../align/ce/CECalculator.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">nextStep</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCalculatorEnhanced.</span><code><span class="memberNameLink"><a href="../align/ce/CeCalculatorEnhanced.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">nextStep</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCPMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CECalculator)">postProcessAlignment</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator)</code></th>
<td class="colLast">
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCPMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.align.ce.CECPParameters)">postProcessAlignment</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                    <a href="../align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;param)</code></th>
<td class="colLast">
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.client">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/client/package-summary.html">org.biojava.nbio.structure.align.client</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/client/package-summary.html">org.biojava.nbio.structure.align.client</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureName.</span><code><span class="memberNameLink"><a href="../align/client/StructureName.html#getBaseIdentifier()">getBaseIdentifier</a></span>()</code></th>
<td class="colLast">
<div class="block">StructureName wraps another StructureIdentifier.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PdbPair.</span><code><span class="memberNameLink"><a href="../align/client/PdbPair.html#getPDBCode1()">getPDBCode1</a></span>()</code></th>
<td class="colLast">
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="deprecationComment">use <a href="../align/client/PdbPair.html#getPDBCode1()"><code>PdbPair.getPDBCode1()</code></a> instead</div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PdbPair.</span><code><span class="memberNameLink"><a href="../align/client/PdbPair.html#getPDBCode2()">getPDBCode2</a></span>()</code></th>
<td class="colLast">
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="deprecationComment">use <a href="../align/client/PdbPair.html#getPDBCode2()"><code>PdbPair.getPDBCode2()</code></a> instead</div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../PdbId.html" title="class in org.biojava.nbio.structure">PdbId</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureName.</span><code><span class="memberNameLink"><a href="../align/client/StructureName.html#getPdbId()">getPdbId</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the PDB ID for this name, if any.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../PdbId.html" title="class in org.biojava.nbio.structure">PdbId</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PdbPair.</span><code><span class="memberNameLink"><a href="../align/client/PdbPair.html#getPdbId1()">getPdbId1</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../PdbId.html" title="class in org.biojava.nbio.structure">PdbId</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PdbPair.</span><code><span class="memberNameLink"><a href="../align/client/PdbPair.html#getPdbId2()">getPdbId2</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureName.</span><code><span class="memberNameLink"><a href="../align/client/StructureName.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></span>&#8203;(<a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureName.</span><code><span class="memberNameLink"><a href="../align/client/StructureName.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../SubstructureIdentifier.html" title="class in org.biojava.nbio.structure">SubstructureIdentifier</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureName.</span><code><span class="memberNameLink"><a href="../align/client/StructureName.html#toCanonical()">toCanonical</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.fatcat">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/fatcat/package-summary.html">org.biojava.nbio.structure.align.fatcat</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/fatcat/package-summary.html">org.biojava.nbio.structure.align.fatcat</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCat.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCat.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,boolean)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
     boolean&nbsp;doRigid)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCatFlexible.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCatFlexible.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCatRigid.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCatRigid.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCat.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCat.html#alignFlexible(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">alignFlexible</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCat.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">alignRigid</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCat.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">alignRigid</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
          <a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.fatcat.calc">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCatAligner.</span><code><span class="memberNameLink"><a href="../align/fatcat/calc/FatCatAligner.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean,org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     boolean&nbsp;doRigid,
     <a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPCalculator.</span><code><span class="memberNameLink"><a href="../align/fatcat/calc/AFPCalculator.html#extractAFPChains(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">extractAFPChains</a></span>&#8203;(<a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
                <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPOptimizer.</span><code><span class="memberNameLink"><a href="../align/fatcat/calc/AFPOptimizer.html#optimizeAln(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">optimizeAln</a></span>&#8203;(<a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
           <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">optimize the alignment by dynamic programming</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureAlignmentOptimizer.</span><code><span class="memberNameLink"><a href="../align/fatcat/calc/StructureAlignmentOptimizer.html#runOptimization(int)">runOptimization</a></span>&#8203;(int&nbsp;maxi)</code></th>
<td class="colLast">
<div class="block">run the optimization</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../align/fatcat/calc/StructureAlignmentOptimizer.html#%3Cinit%3E(int,int,org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.Atom%5B%5D,int,int%5B%5D%5B%5D)">StructureAlignmentOptimizer</a></span>&#8203;(int&nbsp;b1,
                           int&nbsp;end1,
                           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;c1,
                           int&nbsp;b2,
                           int&nbsp;end2,
                           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;c2,
                           int&nbsp;iniLen,
                           int[][]&nbsp;iniSet)</code></th>
<td class="colLast">
<div class="block">optimize the structural alignment by update the equivalent residues
 and then run dynamic programming
 input: len1 the length of structure 1; c1: the structure information of 1
        len2 the length of structure 2; c2: the structure information of 2
        iniLen and iniSet is the length and list of initial equivalent residues</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.gui">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">createArtificalStructure</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Create a "fake" Structure objects that contains the two sets of atoms aligned on top of each other.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#display(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Group%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.util.List,java.util.List)">display</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;twistedGroups,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
       <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms1,
       <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms2)</code></th>
<td class="colLast">
<div class="block">Note: ca2, hetatoms2 and nucleotides2 should not be rotated.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MultipleAlignmentJmol</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentJmolDisplay.</span><code><span class="memberNameLink"><a href="../align/gui/MultipleAlignmentJmolDisplay.html#display(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">display</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code></th>
<td class="colLast">
<div class="block">Display a MultipleAlignment with a JmolPanel.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/gui/StructureAlignmentDisplay.html#display(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">display</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Display an AFPChain alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#getAtomArray(org.biojava.nbio.structure.Atom%5B%5D,java.util.List)">getAtomArray</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms)</code></th>
<td class="colLast">
<div class="block">Returns the first atom for each group</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#getAtomForAligPos(org.biojava.nbio.structure.align.model.AFPChain,int,int,org.biojava.nbio.structure.Atom%5B%5D,boolean)">getAtomForAligPos</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 int&nbsp;chainNr,
                 int&nbsp;aligPos,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
                 boolean&nbsp;getPrevious)</code></th>
<td class="colLast">
<div class="block">Return the atom at alignment position aligPos. at the present only works with block 0</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SelectPDBPanel.</span><code><span class="memberNameLink"><a href="../align/gui/SelectPDBPanel.html#getStructure1()">getStructure1</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SelectPDBPanel.</span><code><span class="memberNameLink"><a href="../align/gui/SelectPDBPanel.html#getStructure2()">getStructure2</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#showAlignmentPanel(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">showAlignmentPanel</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                  <a href="../align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentJmolDisplay.</span><code><span class="memberNameLink"><a href="../align/gui/MultipleAlignmentJmolDisplay.html#showMultipleAligmentPanel(org.biojava.nbio.structure.align.multiple.MultipleAlignment,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">showMultipleAligmentPanel</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                         <a href="../align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code></th>
<td class="colLast">
<div class="block">Creates a new Frame with the MultipleAlignment Sequence Panel.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.gui.aligpanel">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a></h3>
<table class="useSummary">
<caption><span>Constructors in <a href="../align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../align/gui/aligpanel/MultipleAligPanel.html#%3Cinit%3E(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">MultipleAligPanel</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                 <a href="../align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code></th>
<td class="colLast">
<div class="block">Constructor using an afpChain and the atom arrays for pairwise
 alignments.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.multiple">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentEnsembleImpl.</span><code><span class="memberNameLink"><a href="../align/multiple/MultipleAlignmentEnsembleImpl.html#updateAtomArrays()">updateAtomArrays</a></span>()</code></th>
<td class="colLast">
<div class="block">Force the atom arrays to regenerate based on
 <a href="../align/multiple/MultipleAlignmentEnsembleImpl.html#getStructureIdentifiers()"><code>MultipleAlignmentEnsembleImpl.getStructureIdentifiers()</code></a>.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.multiple.mc">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/multiple/mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/multiple/mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleMcMain.</span><code><span class="memberNameLink"><a href="../align/multiple/mc/MultipleMcMain.html#align(java.util.List)">align</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleMcMain.</span><code><span class="memberNameLink"><a href="../align/multiple/mc/MultipleMcMain.html#align(java.util.List,java.lang.Object)">align</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;parameters)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleMcOptimizer.</span><code><span class="memberNameLink"><a href="../align/multiple/mc/MultipleMcOptimizer.html#optimize()">optimize</a></span>()</code></th>
<td class="colLast">
<div class="block">Optimization method based in a Monte-Carlo approach.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.multiple.util">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentScorer.</span><code><span class="memberNameLink"><a href="../align/multiple/util/MultipleAlignmentScorer.html#calculateScores(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">calculateScores</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Calculates and puts the RMSD and the average TM-Score of the
 MultipleAlignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../align/multiple/util/MultipleAlignmentTools.html#getAlignedStructure(java.util.List)">getAlignedStructure</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code></th>
<td class="colLast">
<div class="block">Get an artificial Structure containing a different model for every
 input structure, so that the alignment result can be viewed in Jmol.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentScorer.</span><code><span class="memberNameLink"><a href="../align/multiple/util/MultipleAlignmentScorer.html#getAvgTMScore(java.util.List,java.util.List)">getAvgTMScore</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformed,
             <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;lengths)</code></th>
<td class="colLast">
<div class="block">Calculates the average TMScore all the possible pairwise structure
 comparisons of the given a set of superimposed Atoms and the original
 structure lengths.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentScorer.</span><code><span class="memberNameLink"><a href="../align/multiple/util/MultipleAlignmentScorer.html#getAvgTMScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAvgTMScore</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Calculates the average TMScore of all the possible pairwise structure
 comparisons of the given alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentScorer.</span><code><span class="memberNameLink"><a href="../align/multiple/util/MultipleAlignmentScorer.html#getMCScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment,double,double,double)">getMCScore</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
          double&nbsp;gapOpen,
          double&nbsp;gapExtension,
          double&nbsp;dCutoff)</code></th>
<td class="colLast">
<div class="block">Calculates the MC score, specific for the MultipleAlignment algorithm.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentScorer.</span><code><span class="memberNameLink"><a href="../align/multiple/util/MultipleAlignmentScorer.html#getRefTMScore(java.util.List,java.util.List,int)">getRefTMScore</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformed,
             <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;lengths,
             int&nbsp;reference)</code></th>
<td class="colLast">
<div class="block">Calculates the average TMScore from all structures to a reference
 structure, given a set of superimposed atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentScorer.</span><code><span class="memberNameLink"><a href="../align/multiple/util/MultipleAlignmentScorer.html#getRefTMScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment,int)">getRefTMScore</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
             int&nbsp;ref)</code></th>
<td class="colLast">
<div class="block">Calculates the average TMScore from all structures to a reference
 structure, given a set of superimposed atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/multiple/util/MultipleAlignmentDisplay.html#getRotatedAtoms(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getRotatedAtoms</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code></th>
<td class="colLast">
<div class="block">New structures are downloaded if they were not cached in the alignment
 and they are entirely transformed here with the superposition information
 in the Multiple Alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CoreSuperimposer.</span><code><span class="memberNameLink"><a href="../align/multiple/util/CoreSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">superimpose</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleSuperimposer.</span><code><span class="memberNameLink"><a href="../align/multiple/util/MultipleSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">superimpose</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Superimpose all structures from a <a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ReferenceSuperimposer.</span><code><span class="memberNameLink"><a href="../align/multiple/util/ReferenceSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">superimpose</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentWriter.</span><code><span class="memberNameLink"><a href="../align/multiple/util/MultipleAlignmentWriter.html#to3DFormat(org.biojava.nbio.structure.align.multiple.MultipleAlignment,int,int)">to3DFormat</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
          int&nbsp;queryIndex,
          int&nbsp;templateIndex)</code></th>
<td class="colLast">
<div class="block">Outputs a pairwise alignment in I-TASSER's 3D Format for target-template
 alignment. http://zhanglab.ccmb.med.umich.edu/I-TASSER/option4.html</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../align/multiple/util/MultipleAlignmentTools.html#toMultimodelStructure(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)">toMultimodelStructure</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformedAtoms)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.pairwise">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentJoiner.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentJoiner.html#approach_ap3(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.pairwise.FragmentPair%5B%5D,org.biojava.nbio.structure.align.StrucAligParameters)">approach_ap3</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            <a href="../align/pairwise/FragmentPair.html" title="class in org.biojava.nbio.structure.align.pairwise">FragmentPair</a>[]&nbsp;fraglst,
            <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlternativeAlignment.</span><code><span class="memberNameLink"><a href="../align/pairwise/AlternativeAlignment.html#calculateSuperpositionByIdx(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">calculateSuperpositionByIdx</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentJoiner.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentJoiner.html#extendFragments(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.helper.JointFragments%5B%5D,org.biojava.nbio.structure.align.StrucAligParameters)">extendFragments</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
               <a href="../align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>[]&nbsp;fragments,
               <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentJoiner.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentJoiner.html#extendFragments(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.helper.JointFragments,org.biojava.nbio.structure.align.StrucAligParameters)">extendFragments</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
               <a href="../align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>&nbsp;fragments,
               <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlternativeAlignment.</span><code><span class="memberNameLink"><a href="../align/pairwise/AlternativeAlignment.html#finish(org.biojava.nbio.structure.align.StrucAligParameters,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">finish</a></span>&#8203;(<a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentJoiner.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentJoiner.html#getRMS(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.helper.JointFragments)">getRMS</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
      <a href="../align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>&nbsp;frag)</code></th>
<td class="colLast">
<div class="block">Get the RMS of the JointFragments pair frag</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlternativeAlignment.</span><code><span class="memberNameLink"><a href="../align/pairwise/AlternativeAlignment.html#refine(org.biojava.nbio.structure.align.StrucAligParameters,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">refine</a></span>&#8203;(<a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Refinement procedure based on superposition and dynamic programming.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.quaternary">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/quaternary/QsAlignResult.html" title="class in org.biojava.nbio.structure.align.quaternary">QsAlignResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">QsAlign.</span><code><span class="memberNameLink"><a href="../align/quaternary/QsAlign.html#align(java.util.List,java.util.List,org.biojava.nbio.structure.cluster.SubunitClustererParameters,org.biojava.nbio.structure.align.quaternary.QsAlignParameters)">align</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster">Subunit</a>&gt;&nbsp;s1,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster">Subunit</a>&gt;&nbsp;s2,
     <a href="../cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a>&nbsp;cParams,
     <a href="../align/quaternary/QsAlignParameters.html" title="class in org.biojava.nbio.structure.align.quaternary">QsAlignParameters</a>&nbsp;aParams)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/quaternary/QsAlignResult.html" title="class in org.biojava.nbio.structure.align.quaternary">QsAlignResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">QsAlign.</span><code><span class="memberNameLink"><a href="../align/quaternary/QsAlign.html#align(org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.cluster.SubunitClustererParameters,org.biojava.nbio.structure.align.quaternary.QsAlignParameters)">align</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2,
     <a href="../cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a>&nbsp;cParams,
     <a href="../align/quaternary/QsAlignParameters.html" title="class in org.biojava.nbio.structure.align.quaternary">QsAlignParameters</a>&nbsp;aParams)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.seq">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/seq/package-summary.html">org.biojava.nbio.structure.align.seq</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/seq/package-summary.html">org.biojava.nbio.structure.align.seq</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SmithWaterman3Daligner.</span><code><span class="memberNameLink"><a href="../align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SmithWaterman3Daligner.</span><code><span class="memberNameLink"><a href="../align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;parameters)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.util">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>,&#8203;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#alignmentAsMap(org.biojava.nbio.structure.align.model.AFPChain)">alignmentAsMap</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code></th>
<td class="colLast">
<div class="block">Creates a Map specifying the alignment as a mapping between residue indices
 of protein 1 and residue indices of protein 2.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#createAFPChain(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.ResidueNumber%5B%5D,org.biojava.nbio.structure.ResidueNumber%5B%5D)">createAFPChain</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              <a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned1,
              <a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned2)</code></th>
<td class="colLast">
<div class="block">Fundamentally, an alignment is just a list of aligned residues in each
 protein.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">createArtificalStructure</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#deleteColumn(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int)">deleteColumn</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            int&nbsp;block,
            int&nbsp;pos)</code></th>
<td class="colLast">
<div class="block">Delete an alignment position from the original alignment object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#deleteHighestDistanceColumn(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">deleteHighestDistanceColumn</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Find the alignment position with the highest atomic distance between the
 equivalent atomic positions of the arrays and remove it from the
 alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#getAlignedStructure(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getAlignedStructure</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Get an artifical Structure containing both chains.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RotationAxis.</span><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#getAngle(org.biojava.nbio.structure.align.model.AFPChain)">getAngle</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code></th>
<td class="colLast">
<div class="block">Calculate the rotation angle for a structure</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getAtoms(java.lang.String)">getAtoms</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Returns the CA atoms for the provided name.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getAtoms(org.biojava.nbio.structure.StructureIdentifier)">getAtoms</a></span>&#8203;(<a href="../StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;name)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getBiologicalAssemblies(java.lang.String,boolean)">getBiologicalAssemblies</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbId,
                       boolean&nbsp;multiModel)</code></th>
<td class="colLast">
<div class="block">Returns all biological assemblies for given PDB id.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getBiologicalAssembly(java.lang.String,boolean)">getBiologicalAssembly</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbId,
                     boolean&nbsp;multiModel)</code></th>
<td class="colLast">
<div class="block">Returns the default biological unit (bioassemblyId=1, known in PDB as pdb1.gz).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getBiologicalAssembly(java.lang.String,int,boolean)">getBiologicalAssembly</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbId,
                     int&nbsp;bioAssemblyId,
                     boolean&nbsp;multiModel)</code></th>
<td class="colLast">
<div class="block">Returns the biological assembly for a given PDB ID and bioAssemblyId, by building the
 assembly from the biounit annotations found in <a href="../Structure.html#getPDBHeader()"><code>Structure.getPDBHeader()</code></a></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getBiologicalAssembly(org.biojava.nbio.structure.PdbId,int,boolean)">getBiologicalAssembly</a></span>&#8203;(<a href="../PdbId.html" title="class in org.biojava.nbio.structure">PdbId</a>&nbsp;pdbId,
                     int&nbsp;bioAssemblyId,
                     boolean&nbsp;multiModel)</code></th>
<td class="colLast">
<div class="block">Returns the biological assembly for a given PDB ID and bioAssemblyId, by building the
 assembly from the biounit annotations found in <a href="../Structure.html#getPDBHeader()"><code>Structure.getPDBHeader()</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getRepresentativeAtoms(java.lang.String)">getRepresentativeAtoms</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Returns the representative atoms for the provided name.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getRepresentativeAtoms(org.biojava.nbio.structure.StructureIdentifier)">getRepresentativeAtoms</a></span>&#8203;(<a href="../StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;name)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#getRotMax(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getRotMax</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
         <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
         <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getStructure(java.lang.String)">getStructure</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Request a Structure based on a <i>name</i>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getStructure(org.biojava.nbio.structure.StructureIdentifier)">getStructure</a></span>&#8203;(<a href="../StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;strucId)</code></th>
<td class="colLast">
<div class="block">Get the structure corresponding to the given <a href="../StructureIdentifier.html" title="interface in org.biojava.nbio.structure"><code>StructureIdentifier</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getStructureForCathDomain(org.biojava.nbio.structure.align.client.StructureName)">getStructureForCathDomain</a></span>&#8203;(<a href="../align/client/StructureName.html" title="class in org.biojava.nbio.structure.align.client">StructureName</a>&nbsp;structureName)</code></th>
<td class="colLast">
<div class="block">Returns a <a href="../Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> corresponding to the CATH identifier supplied in <code>structureName</code>, using the the <a href="../cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath"><code>CathDatabase</code></a>
 at <a href="../cath/CathFactory.html#getCathDatabase()"><code>CathFactory.getCathDatabase()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getStructureForCathDomain(org.biojava.nbio.structure.align.client.StructureName,org.biojava.nbio.structure.cath.CathDatabase)">getStructureForCathDomain</a></span>&#8203;(<a href="../align/client/StructureName.html" title="class in org.biojava.nbio.structure.align.client">StructureName</a>&nbsp;structureName,
                         <a href="../cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath">CathDatabase</a>&nbsp;cathInstall)</code></th>
<td class="colLast">
<div class="block">Returns a <a href="../Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> corresponding to the CATH identifier supplied in <code>structureName</code>, using the specified <a href="../cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath"><code>CathDatabase</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getStructureForDomain(java.lang.String)">getStructureForDomain</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;scopId)</code></th>
<td class="colLast">
<div class="block">Returns the representation of a <a href="../scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getStructureForDomain(java.lang.String,org.biojava.nbio.structure.scop.ScopDatabase)">getStructureForDomain</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;scopId,
                     <a href="../scop/ScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">ScopDatabase</a>&nbsp;scopDatabase)</code></th>
<td class="colLast">
<div class="block">Returns the representation of a <a href="../scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getStructureForDomain(org.biojava.nbio.structure.scop.ScopDomain)">getStructureForDomain</a></span>&#8203;(<a href="../scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop">ScopDomain</a>&nbsp;domain)</code></th>
<td class="colLast">
<div class="block">Returns the representation of a <a href="../scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getStructureForDomain(org.biojava.nbio.structure.scop.ScopDomain,org.biojava.nbio.structure.scop.ScopDatabase)">getStructureForDomain</a></span>&#8203;(<a href="../scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop">ScopDomain</a>&nbsp;domain,
                     <a href="../scop/ScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">ScopDatabase</a>&nbsp;scopDatabase)</code></th>
<td class="colLast">
<div class="block">Returns the representation of a <a href="../scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getStructureForDomain(org.biojava.nbio.structure.scop.ScopDomain,org.biojava.nbio.structure.scop.ScopDatabase,boolean)">getStructureForDomain</a></span>&#8203;(<a href="../scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop">ScopDomain</a>&nbsp;domain,
                     <a href="../scop/ScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">ScopDatabase</a>&nbsp;scopDatabase,
                     boolean&nbsp;strictLigandHandling)</code></th>
<td class="colLast">
<div class="block">Returns the representation of a <a href="../scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getStructureForPdbId(java.lang.String)">getStructureForPdbId</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;id)</code></th>
<td class="colLast">
<div class="block">Loads a structure directly by PDB ID</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#getSymmetryOrder(org.biojava.nbio.structure.align.model.AFPChain,int,float)">getSymmetryOrder</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                int&nbsp;maxSymmetry,
                float&nbsp;minimumMetricChange)</code></th>
<td class="colLast">
<div class="block">Guesses the order of symmetry in an alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainScorer.</span><code><span class="memberNameLink"><a href="../align/util/AFPChainScorer.html#getTMScore(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getTMScore</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;align,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainScorer.</span><code><span class="memberNameLink"><a href="../align/util/AFPChainScorer.html#getTMScore(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean)">getTMScore</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;align,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
          boolean&nbsp;normalizeMin)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#getTranslation(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getTranslation</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#prepareGroupsForDisplay(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">prepareGroupsForDisplay</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Rotate the Atoms/Groups so they are aligned for the 3D visualisation</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#replaceOptAln(int%5B%5D%5B%5D%5B%5D,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">replaceOptAln</a></span>&#8203;(int[][][]&nbsp;newAlgn,
             <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">It replaces an optimal alignment of an AFPChain and calculates all the new alignment scores and variables.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int%5B%5D,int%5B%5D%5B%5D%5B%5D)">replaceOptAln</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             int&nbsp;blockNum,
             int[]&nbsp;optLens,
             int[][][]&nbsp;optAln)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.util.Map)">replaceOptAln</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>,&#8203;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Takes an AFPChain and replaces the optimal alignment based on an alignment map</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#splitBlocksByTopology(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">splitBlocksByTopology</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;a,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#updateSuperposition(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">updateSuperposition</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">After the alignment changes (optAln, optLen, blockNum, at a minimum),
 many other properties which depend on the superposition will be invalid.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#%3Cinit%3E(org.biojava.nbio.structure.align.model.AFPChain)">RotationAxis</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code></th>
<td class="colLast">
<div class="block">Calculate the rotation axis for the first block of an AFPChain</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.xml">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainXMLParser.</span><code><span class="memberNameLink"><a href="../align/xml/AFPChainXMLParser.html#flipAlignment(java.lang.String)">flipAlignment</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;xml)</code></th>
<td class="colLast">
<div class="block">Takes an XML representation of the alignment and flips the positions of name1 and name2</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainFlipper.</span><code><span class="memberNameLink"><a href="../align/xml/AFPChainFlipper.html#flipChain(org.biojava.nbio.structure.align.model.AFPChain)">flipChain</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;o)</code></th>
<td class="colLast">
<div class="block">Flip the position of name1 and name2 (as well as all underlying data) in an AFPChain.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainXMLParser.</span><code><span class="memberNameLink"><a href="../align/xml/AFPChainXMLParser.html#fromXML(java.lang.String,java.lang.String,java.lang.String,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">fromXML</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;xml,
       <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name1,
       <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name2,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">new utility method that checks that the order of the pair in the XML alignment is correct and flips the direction if needed</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.cath">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../cath/package-summary.html">org.biojava.nbio.structure.cath</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../cath/package-summary.html">org.biojava.nbio.structure.cath</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util" class="externalLink">Set</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CathDomain.</span><code><span class="memberNameLink"><a href="../cath/CathDomain.html#getChains()">getChains</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the chains this domain is defined over; contains more than 1 element only if this domains is a multi-chain domain.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CathDomain.</span><code><span class="memberNameLink"><a href="../cath/CathDomain.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></span>&#8203;(<a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CathDomain.</span><code><span class="memberNameLink"><a href="../cath/CathDomain.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../SubstructureIdentifier.html" title="class in org.biojava.nbio.structure">SubstructureIdentifier</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CathDomain.</span><code><span class="memberNameLink"><a href="../cath/CathDomain.html#toCanonical()">toCanonical</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.cluster">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../cluster/package-summary.html">org.biojava.nbio.structure.cluster</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SubunitCluster.</span><code><span class="memberNameLink"><a href="../cluster/SubunitCluster.html#divideInternally(org.biojava.nbio.structure.cluster.SubunitClustererParameters)">divideInternally</a></span>&#8203;(<a href="../cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a>&nbsp;clusterParams)</code></th>
<td class="colLast">
<div class="block">Analyze the internal symmetry of the SubunitCluster and divide its
 <a href="../cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a> into the internal repeats (domains) if they are
 internally symmetric.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SubunitCluster.</span><code><span class="memberNameLink"><a href="../cluster/SubunitCluster.html#getMultipleAlignment()">getMultipleAlignment</a></span>()</code></th>
<td class="colLast">
<div class="block">The multiple alignment is calculated from the equivalent residues in the
 SubunitCluster.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SubunitCluster.</span><code><span class="memberNameLink"><a href="../cluster/SubunitCluster.html#mergeStructure(org.biojava.nbio.structure.cluster.SubunitCluster,org.biojava.nbio.structure.cluster.SubunitClustererParameters)">mergeStructure</a></span>&#8203;(<a href="../cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster">SubunitCluster</a>&nbsp;other,
              <a href="../cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Merges the other SubunitCluster into this one if their representative
 Atoms are structurally similar (according to the criteria in params).</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.domain">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../domain/package-summary.html">org.biojava.nbio.structure.domain</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../domain/package-summary.html">org.biojava.nbio.structure.domain</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/SortedSet.html?is-external=true" title="class or interface in java.util" class="externalLink">SortedSet</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DomainProvider.</span><code><span class="memberNameLink"><a href="../domain/DomainProvider.html#getDomainNames(java.lang.String)">getDomainNames</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Get a list of constituent domain identifiers</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">LocalProteinDomainParser.</span><code><span class="memberNameLink"><a href="../domain/LocalProteinDomainParser.html#suggestDomains(org.biojava.nbio.structure.Atom%5B%5D)">suggestDomains</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">
<div class="block">Suggest domains for a set of Calpha atoms</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">LocalProteinDomainParser.</span><code><span class="memberNameLink"><a href="../domain/LocalProteinDomainParser.html#suggestDomains(org.biojava.nbio.structure.Structure)">suggestDomains</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code></th>
<td class="colLast">
<div class="block">Suggest domains for a protein structure</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.domain.pdp">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GetDistanceMatrix.</span><code><span class="memberNameLink"><a href="../domain/pdp/GetDistanceMatrix.html#getDistanceMatrix(org.biojava.nbio.structure.Atom%5B%5D)">getDistanceMatrix</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;protein)</code></th>
<td class="colLast">
<div class="block">A set of Calpha atoms that are representing the protein</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.ecod">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../ecod/package-summary.html">org.biojava.nbio.structure.ecod</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../ecod/package-summary.html">org.biojava.nbio.structure.ecod</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">EcodDomain.</span><code><span class="memberNameLink"><a href="../ecod/EcodDomain.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></span>&#8203;(<a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">EcodDomain.</span><code><span class="memberNameLink"><a href="../ecod/EcodDomain.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.gui.util">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../gui/util/package-summary.html">org.biojava.nbio.structure.gui.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../gui/util/package-summary.html">org.biojava.nbio.structure.gui.util</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PDBDirPanel.</span><code><span class="memberNameLink"><a href="../gui/util/PDBDirPanel.html#getStructure1()">getStructure1</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PDBServerPanel.</span><code><span class="memberNameLink"><a href="../gui/util/PDBServerPanel.html#getStructure1()">getStructure1</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PDBUploadPanel.</span><code><span class="memberNameLink"><a href="../gui/util/PDBUploadPanel.html#getStructure1()">getStructure1</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ScopSelectPanel.</span><code><span class="memberNameLink"><a href="../gui/util/ScopSelectPanel.html#getStructure1()">getStructure1</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructurePairSelector.</span><code><span class="memberNameLink"><a href="../gui/util/StructurePairSelector.html#getStructure1()">getStructure1</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PDBDirPanel.</span><code><span class="memberNameLink"><a href="../gui/util/PDBDirPanel.html#getStructure2()">getStructure2</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PDBServerPanel.</span><code><span class="memberNameLink"><a href="../gui/util/PDBServerPanel.html#getStructure2()">getStructure2</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PDBUploadPanel.</span><code><span class="memberNameLink"><a href="../gui/util/PDBUploadPanel.html#getStructure2()">getStructure2</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ScopSelectPanel.</span><code><span class="memberNameLink"><a href="../gui/util/ScopSelectPanel.html#getStructure2()">getStructure2</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructurePairSelector.</span><code><span class="memberNameLink"><a href="../gui/util/StructurePairSelector.html#getStructure2()">getStructure2</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SelectMultiplePanel.</span><code><span class="memberNameLink"><a href="../gui/util/SelectMultiplePanel.html#getStructures()">getStructures</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.io">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../io/package-summary.html">org.biojava.nbio.structure.io</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../io/package-summary.html">org.biojava.nbio.structure.io</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../io/FastaAFPChainConverter.html#cpFastaToAfpChain(java.io.File,org.biojava.nbio.structure.Structure,int)">cpFastaToAfpChain</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;fastaFile,
                 <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code></th>
<td class="colLast">
<div class="block">Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site
 <code>cpSite</code> residues to the right.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../io/FastaAFPChainConverter.html#cpFastaToAfpChain(java.lang.String,java.lang.String,org.biojava.nbio.structure.Structure,int)">cpFastaToAfpChain</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;first,
                 <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;second,
                 <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../io/FastaAFPChainConverter.html#cpFastaToAfpChain(org.biojava.nbio.core.sequence.ProteinSequence,org.biojava.nbio.core.sequence.ProteinSequence,org.biojava.nbio.structure.Structure,int)">cpFastaToAfpChain</a></span>&#8203;(<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;first,
                 <a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;second,
                 <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code></th>
<td class="colLast">
<div class="block">Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site
 <code>cpSite</code> residues to the right.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../io/FastaAFPChainConverter.html#fastaFileToAfpChain(java.io.File,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure)">fastaFileToAfpChain</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;fastaFile,
                   <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
                   <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code></th>
<td class="colLast">
<div class="block">Reads the file <code>fastaFile</code>, expecting exactly two sequences which give a pairwise alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../io/FastaAFPChainConverter.html#fastaStringToAfpChain(java.lang.String,java.lang.String,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure)">fastaStringToAfpChain</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence1,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence2,
                     <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
                     <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code></th>
<td class="colLast">
<div class="block">Returns an AFPChain corresponding to the alignment between <code>structure1</code> and <code>structure2</code>, which is given by the gapped protein sequences <code>sequence1</code> and <code>sequence2</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../io/FastaAFPChainConverter.html#fastaToAfpChain(java.lang.String,java.lang.String,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence1,
               <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;sequence2,
               <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code></th>
<td class="colLast">
<div class="block">TODO Write comment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../io/FastaAFPChainConverter.html#fastaToAfpChain(java.util.Map,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;&nbsp;sequences,
               <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code></th>
<td class="colLast">
<div class="block">Uses two sequences each with a corresponding structure to create an AFPChain corresponding to the alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../io/FastaAFPChainConverter.html#fastaToAfpChain(org.biojava.nbio.core.alignment.template.SequencePair,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></span>&#8203;(<a href="../../core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;<a href="../../core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;,&#8203;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;alignment,
               <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code></th>
<td class="colLast">
<div class="block">Provided only for convenience.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../io/FastaAFPChainConverter.html#fastaToAfpChain(org.biojava.nbio.core.sequence.ProteinSequence,org.biojava.nbio.core.sequence.ProteinSequence,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></span>&#8203;(<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence1,
               <a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence2,
               <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code></th>
<td class="colLast">
<div class="block">Returns an AFPChain corresponding to the alignment between <code>structure1</code> and <code>structure2</code>, which is given by the gapped protein sequences <code>sequence1</code> and <code>sequence2</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureProvider.</span><code><span class="memberNameLink"><a href="../io/StructureProvider.html#getStructureById(java.lang.String)">getStructureById</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;pdbId)</code></th>
<td class="colLast">
<div class="block">Get the structure for a PDB ID</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureProvider.</span><code><span class="memberNameLink"><a href="../io/StructureProvider.html#getStructureById(org.biojava.nbio.structure.PdbId)">getStructureById</a></span>&#8203;(<a href="../PdbId.html" title="class in org.biojava.nbio.structure">PdbId</a>&nbsp;pdbId)</code></th>
<td class="colLast">
<div class="block">Get the structure for a PDB ID</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../io/FastaAFPChainConverter.html#main(java.lang.String%5B%5D)">main</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;args)</code></th>
<td class="colLast">
<div class="block">Prints out the XML representation of an AFPChain from a file containing exactly two FASTA sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaStructureParser.</span><code><span class="memberNameLink"><a href="../io/FastaStructureParser.html#process()">process</a></span>()</code></th>
<td class="colLast">
<div class="block">Parses the fasta file and loads it into memory.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.scop">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../scop/package-summary.html">org.biojava.nbio.structure.scop</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../scop/package-summary.html">org.biojava.nbio.structure.scop</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ScopDomain.</span><code><span class="memberNameLink"><a href="../scop/ScopDomain.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></span>&#8203;(<a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ScopDomain.</span><code><span class="memberNameLink"><a href="../scop/ScopDomain.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.secstruc">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../secstruc/package-summary.html">org.biojava.nbio.structure.secstruc</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../secstruc/package-summary.html">org.biojava.nbio.structure.secstruc</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SecStrucCalc.</span><code><span class="memberNameLink"><a href="../secstruc/SecStrucCalc.html#calculate(org.biojava.nbio.structure.Structure,boolean)">calculate</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
         boolean&nbsp;assign)</code></th>
<td class="colLast">
<div class="block">Predicts the secondary structure of this Structure object,
 using a DSSP implementation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DSSPParser.</span><code><span class="memberNameLink"><a href="../secstruc/DSSPParser.html#parseFile(java.lang.String,org.biojava.nbio.structure.Structure,boolean)">parseFile</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;dsspPath,
         <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
         boolean&nbsp;assign)</code></th>
<td class="colLast">
<div class="block">Parse a DSSP output file and return the secondary structure
 annotation as a List of <a href="../secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc"><code>SecStrucState</code></a> objects.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DSSPParser.</span><code><span class="memberNameLink"><a href="../secstruc/DSSPParser.html#parseInputStream(java.io.InputStream,org.biojava.nbio.structure.Structure,boolean)">parseInputStream</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">InputStream</a>&nbsp;dsspIs,
                <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                boolean&nbsp;assign)</code></th>
<td class="colLast">
<div class="block">Parse a DSSP output file and return the secondary structure
 annotation as a List of <a href="../secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc"><code>SecStrucState</code></a> objects.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DSSPParser.</span><code><span class="memberNameLink"><a href="../secstruc/DSSPParser.html#parseString(java.lang.String,org.biojava.nbio.structure.Structure,boolean)">parseString</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;dsspOut,
           <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
           boolean&nbsp;assign)</code></th>
<td class="colLast">
<div class="block">Parse a DSSP format String and return the secondary structure
 annotation as a List of <a href="../secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc"><code>SecStrucState</code></a> objects.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.symmetry.gui">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../symmetry/gui/package-summary.html">org.biojava.nbio.structure.symmetry.gui</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../symmetry/gui/package-summary.html">org.biojava.nbio.structure.symmetry.gui</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryDisplay.</span><code><span class="memberNameLink"><a href="../symmetry/gui/SymmetryDisplay.html#display(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">display</a></span>&#8203;(<a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symmResult)</code></th>
<td class="colLast">
<div class="block">Displays a single structure in a cartoon representation with each
 symmetric repeat colored differently.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MultipleAlignmentJmol</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryDisplay.</span><code><span class="memberNameLink"><a href="../symmetry/gui/SymmetryDisplay.html#displayFull(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">displayFull</a></span>&#8203;(<a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symm)</code></th>
<td class="colLast">
<div class="block">Displays a multiple alignment of the whole structure transformations
 colored by blocks, corresponding to the symmetric protodomains.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MultipleAlignmentJmol</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryDisplay.</span><code><span class="memberNameLink"><a href="../symmetry/gui/SymmetryDisplay.html#displayRepeats(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">displayRepeats</a></span>&#8203;(<a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symm)</code></th>
<td class="colLast">
<div class="block">Displays a multiple alignment of the symmetry repeats</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryDisplay.</span><code><span class="memberNameLink"><a href="../symmetry/gui/SymmetryDisplay.html#printSymmetryAxes(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">printSymmetryAxes</a></span>&#8203;(<a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symm)</code></th>
<td class="colLast">
<div class="block">Generates a String that displays the symmetry axes of a structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryDisplay.</span><code><span class="memberNameLink"><a href="../symmetry/gui/SymmetryDisplay.html#printSymmetryAxes(org.biojava.nbio.structure.symmetry.internal.CeSymmResult,boolean)">printSymmetryAxes</a></span>&#8203;(<a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symm,
                 boolean&nbsp;allAxes)</code></th>
<td class="colLast">
<div class="block">Generates a String that displays the symmetry axes of a structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryDisplay.</span><code><span class="memberNameLink"><a href="../symmetry/gui/SymmetryDisplay.html#printSymmetryGroup(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">printSymmetryGroup</a></span>&#8203;(<a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symm)</code></th>
<td class="colLast">
<div class="block">Given a symmetry alignment, it draws the symmetry group axes and the
 polyhedron box around the structure.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.symmetry.internal">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected static <a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymm.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymm.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.symmetry.internal.CESymmParameters)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
     <a href="../symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymm.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymm.html#analyze(org.biojava.nbio.structure.Atom%5B%5D)">analyze</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Analyze the symmetries of the input Atom array using the DEFAULT
 parameters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymm.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymm.html#analyze(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.symmetry.internal.CESymmParameters)">analyze</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
       <a href="../symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Analyze the symmetries of the input Atom array using the provided
 parameters.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymm.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymm.html#analyzeLevel(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.symmetry.internal.CESymmParameters)">analyzeLevel</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
            <a href="../symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Analyze a single level of symmetry.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymmIterative.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymmIterative.html#execute(org.biojava.nbio.structure.Atom%5B%5D)">execute</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">This method uses iteratively CeSymm to calculate all symmetries in the
 input array of atoms and organize them in a multiple alignment of the
 repeats.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymmResult.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymmResult.html#getRepeatsID()">getRepeatsID</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the symmetric repeats as structure identifiers, if the result is
 symmetric and it was refined, return null otherwise.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmOptimizer.</span><code><span class="memberNameLink"><a href="../symmetry/internal/SymmOptimizer.html#optimize()">optimize</a></span>()</code></th>
<td class="colLast">
<div class="block">Optimization method based in a Monte-Carlo approach.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GraphComponentRefiner.</span><code><span class="memberNameLink"><a href="../symmetry/internal/GraphComponentRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)">refine</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceFunctionRefiner.</span><code><span class="memberNameLink"><a href="../symmetry/internal/SequenceFunctionRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)">refine</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryRefiner.</span><code><span class="memberNameLink"><a href="../symmetry/internal/SymmetryRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)">refine</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code></th>
<td class="colLast">
<div class="block">Returns a refined symmetry alignment, where the repeat residues are
 aligned consistently in a MultipleAlignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>,&#8203;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceFunctionRefiner.</span><code><span class="memberNameLink"><a href="../symmetry/internal/SequenceFunctionRefiner.html#refineSymmetry(java.util.Map,int)">refineSymmetry</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>,&#8203;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;alignment,
              int&nbsp;k)</code></th>
<td class="colLast">
<div class="block">Refines a CE-Symm alignment so that it is perfectly symmetric.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceFunctionRefiner.</span><code><span class="memberNameLink"><a href="../symmetry/internal/SequenceFunctionRefiner.html#refineSymmetry(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int)">refineSymmetry</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              int&nbsp;k)</code></th>
<td class="colLast">
<div class="block">Refines a CE-Symm alignment so that it is perfectly symmetric.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.symmetry.utils">
<!--   -->
</a>
<h3>Uses of <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> that throw <a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#divideStructure(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">divideStructure</a></span>&#8203;(<a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symmetry)</code></th>
<td class="colLast">
<div class="block">Method that converts the symmetric units of a structure into different
 structures, so that they can be individually visualized.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#fromAFP(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)">fromAFP</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;symm,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Converts a refined symmetry AFPChain alignment into the standard
 representation of symmetry in a MultipleAlignment, that contains the
 entire Atom array of the strcuture and the symmetric repeats are orgaized
 in different rows in a single Block.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../symmetry/core/QuatSymmetryResults.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryResults</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#getQuaternarySymmetry(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">getQuaternarySymmetry</a></span>&#8203;(<a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;result)</code></th>
<td class="colLast">
<div class="block">Given a symmetry result, it calculates the overall global symmetry,
 factoring out the alignment and detection steps of
 <a href="../symmetry/core/QuatSymmetryDetector.html" title="class in org.biojava.nbio.structure.symmetry.core"><code>QuatSymmetryDetector</code></a> algorithm.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#toRepeatsAlignment(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">toRepeatsAlignment</a></span>&#8203;(<a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;result)</code></th>
<td class="colLast">
<div class="block">Method that converts a symmetry alignment into an alignment of the
 repeats only, as new independent structures.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#updateSymmetryScores(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">updateSymmetryScores</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;symm)</code></th>
<td class="colLast">
<div class="block">Update the scores (TM-score and RMSD) of a symmetry multiple alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#updateSymmetryTransformation(org.biojava.nbio.structure.symmetry.internal.SymmetryAxes,org.biojava.nbio.structure.align.multiple.MultipleAlignment)">updateSymmetryTransformation</a></span>&#8203;(<a href="../symmetry/internal/SymmetryAxes.html" title="class in org.biojava.nbio.structure.symmetry.internal">SymmetryAxes</a>&nbsp;axes,
                            <a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">Calculates the set of symmetry operation Matrices (transformations) of
 the new alignment, based on the symmetry relations in the SymmetryAxes
 object.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../StructureException.html" title="class in org.biojava.nbio.structure">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2022 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
