<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> */</a>
<span class="sourceLineNo">021</span><a id="line.21">package org.biojava.nbio.genome.homology;</a>
<span class="sourceLineNo">022</span><a id="line.22"></a>
<span class="sourceLineNo">023</span><a id="line.23">import org.biojava.nbio.genome.GeneFeatureHelper;</a>
<span class="sourceLineNo">024</span><a id="line.24">import org.biojava.nbio.alignment.Alignments;</a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType;</a>
<span class="sourceLineNo">026</span><a id="line.26">import org.biojava.nbio.alignment.SimpleGapPenalty;</a>
<span class="sourceLineNo">027</span><a id="line.27">import org.biojava.nbio.core.alignment.matrices.SimpleSubstitutionMatrix;</a>
<span class="sourceLineNo">028</span><a id="line.28">import org.biojava.nbio.core.alignment.template.SequencePair;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.biojava.nbio.core.sequence.*;</a>
<span class="sourceLineNo">030</span><a id="line.30">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</a>
<span class="sourceLineNo">031</span><a id="line.31">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</a>
<span class="sourceLineNo">032</span><a id="line.32">import org.biojava.nbio.core.sequence.features.DBReferenceInfo;</a>
<span class="sourceLineNo">033</span><a id="line.33">import org.biojava.nbio.core.sequence.features.DatabaseReferenceInterface;</a>
<span class="sourceLineNo">034</span><a id="line.34">import org.biojava.nbio.core.sequence.features.FeaturesKeyWordInterface;</a>
<span class="sourceLineNo">035</span><a id="line.35">import org.biojava.nbio.core.sequence.loader.UniprotProxySequenceReader;</a>
<span class="sourceLineNo">036</span><a id="line.36">import org.slf4j.Logger;</a>
<span class="sourceLineNo">037</span><a id="line.37">import org.slf4j.LoggerFactory;</a>
<span class="sourceLineNo">038</span><a id="line.38"></a>
<span class="sourceLineNo">039</span><a id="line.39">import java.io.File;</a>
<span class="sourceLineNo">040</span><a id="line.40">import java.io.FileOutputStream;</a>
<span class="sourceLineNo">041</span><a id="line.41">import java.io.OutputStream;</a>
<span class="sourceLineNo">042</span><a id="line.42">import java.util.ArrayList;</a>
<span class="sourceLineNo">043</span><a id="line.43">import java.util.LinkedHashMap;</a>
<span class="sourceLineNo">044</span><a id="line.44">import java.util.List;</a>
<span class="sourceLineNo">045</span><a id="line.45">import java.util.Map;</a>
<span class="sourceLineNo">046</span><a id="line.46"></a>
<span class="sourceLineNo">047</span><a id="line.47">/**</a>
<span class="sourceLineNo">048</span><a id="line.48"> *</a>
<span class="sourceLineNo">049</span><a id="line.49"> * @author Scooter Willis &lt;willishf at gmail dot com&gt;</a>
<span class="sourceLineNo">050</span><a id="line.50"> * @author Mark Chapman</a>
<span class="sourceLineNo">051</span><a id="line.51"> */</a>
<span class="sourceLineNo">052</span><a id="line.52">public class GFF3FromUniprotBlastHits {</a>
<span class="sourceLineNo">053</span><a id="line.53"></a>
<span class="sourceLineNo">054</span><a id="line.54">        private static final Logger logger = LoggerFactory.getLogger(GFF3FromUniprotBlastHits.class);</a>
<span class="sourceLineNo">055</span><a id="line.55"></a>
<span class="sourceLineNo">056</span><a id="line.56">        public void process(File xmlBlastHits, double ecutoff, LinkedHashMap&lt;String, GeneSequence&gt; geneSequenceHashMap, OutputStream gff3Output) throws Exception {</a>
<span class="sourceLineNo">057</span><a id="line.57">                LinkedHashMap&lt;String, ArrayList&lt;String&gt;&gt; hits = BlastHomologyHits.getMatches(xmlBlastHits, ecutoff);</a>
<span class="sourceLineNo">058</span><a id="line.58">                process(hits, geneSequenceHashMap, gff3Output);</a>
<span class="sourceLineNo">059</span><a id="line.59">        }</a>
<span class="sourceLineNo">060</span><a id="line.60"></a>
<span class="sourceLineNo">061</span><a id="line.61">        public void process(LinkedHashMap&lt;String, ArrayList&lt;String&gt;&gt; hits, LinkedHashMap&lt;String, GeneSequence&gt; geneSequenceHashMap, OutputStream gff3Output) throws Exception {</a>
<span class="sourceLineNo">062</span><a id="line.62">                int size = hits.size();</a>
<span class="sourceLineNo">063</span><a id="line.63">                int index = 0;</a>
<span class="sourceLineNo">064</span><a id="line.64">//              HashMap&lt;String, String&gt; scaffoldsReferencedHashMap = new HashMap&lt;String, String&gt;();</a>
<span class="sourceLineNo">065</span><a id="line.65">                for (String accessionid : hits.keySet()) {</a>
<span class="sourceLineNo">066</span><a id="line.66">                        index++;</a>
<span class="sourceLineNo">067</span><a id="line.67">                        if (index == 12) {</a>
<span class="sourceLineNo">068</span><a id="line.68">                                index = 12;</a>
<span class="sourceLineNo">069</span><a id="line.69">                        }</a>
<span class="sourceLineNo">070</span><a id="line.70">                        logger.error(accessionid + " " + index + "/" + size);</a>
<span class="sourceLineNo">071</span><a id="line.71">                        try {</a>
<span class="sourceLineNo">072</span><a id="line.72"></a>
<span class="sourceLineNo">073</span><a id="line.73">                                String[] data = accessionid.split(" ");</a>
<span class="sourceLineNo">074</span><a id="line.74">                                String id = data[0];</a>
<span class="sourceLineNo">075</span><a id="line.75">                                GeneSequence geneSequence = geneSequenceHashMap.get(id);</a>
<span class="sourceLineNo">076</span><a id="line.76">                                if (geneSequence == null) {</a>
<span class="sourceLineNo">077</span><a id="line.77">                                        logger.error("Not found " + id);</a>
<span class="sourceLineNo">078</span><a id="line.78">                                        continue;</a>
<span class="sourceLineNo">079</span><a id="line.79">                                }</a>
<span class="sourceLineNo">080</span><a id="line.80">                                ArrayList&lt;String&gt; uniprotProteinHits = hits.get(accessionid);</a>
<span class="sourceLineNo">081</span><a id="line.81">                                String uniprotBestHit = uniprotProteinHits.get(0);</a>
<span class="sourceLineNo">082</span><a id="line.82">                                UniprotProxySequenceReader&lt;AminoAcidCompound&gt; uniprotSequence = new UniprotProxySequenceReader&lt;AminoAcidCompound&gt;(uniprotBestHit, AminoAcidCompoundSet.getAminoAcidCompoundSet());</a>
<span class="sourceLineNo">083</span><a id="line.83"></a>
<span class="sourceLineNo">084</span><a id="line.84">                                ProteinSequence proteinSequence = new ProteinSequence(uniprotSequence);</a>
<span class="sourceLineNo">085</span><a id="line.85">                                String hitSequence = proteinSequence.getSequenceAsString();</a>
<span class="sourceLineNo">086</span><a id="line.86">                                for (TranscriptSequence transcriptSequence : geneSequence.getTranscripts().values()) {</a>
<span class="sourceLineNo">087</span><a id="line.87"></a>
<span class="sourceLineNo">088</span><a id="line.88"></a>
<span class="sourceLineNo">089</span><a id="line.89">                                        String predictedProteinSequence = transcriptSequence.getProteinSequence().getSequenceAsString();</a>
<span class="sourceLineNo">090</span><a id="line.90">                                        ArrayList&lt;ProteinSequence&gt; cdsProteinList = transcriptSequence.getProteinCDSSequences();</a>
<span class="sourceLineNo">091</span><a id="line.91"></a>
<span class="sourceLineNo">092</span><a id="line.92">                                        ArrayList&lt;CDSSequence&gt; cdsSequenceList = new ArrayList&lt;CDSSequence&gt;(transcriptSequence.getCDSSequences().values());</a>
<span class="sourceLineNo">093</span><a id="line.93">                                        String testSequence = "";</a>
<span class="sourceLineNo">094</span><a id="line.94">                                        for (ProteinSequence cdsProteinSequence : cdsProteinList) {</a>
<span class="sourceLineNo">095</span><a id="line.95">                                                testSequence = testSequence + cdsProteinSequence.getSequenceAsString();</a>
<span class="sourceLineNo">096</span><a id="line.96">                                        }</a>
<span class="sourceLineNo">097</span><a id="line.97">                                        if (!testSequence.equals(predictedProteinSequence) &amp;&amp; (!predictedProteinSequence.equals(testSequence.substring(0, testSequence.length() - 1)))) {</a>
<span class="sourceLineNo">098</span><a id="line.98">                                                DNASequence codingSequence = transcriptSequence.getDNACodingSequence();</a>
<span class="sourceLineNo">099</span><a id="line.99">                                                logger.info("Coding Sequence: {}", codingSequence.getSequenceAsString());</a>
<span class="sourceLineNo">100</span><a id="line.100">                                                logger.info("Sequence agreement error");</a>
<span class="sourceLineNo">101</span><a id="line.101">                                                logger.info("CDS seq={}", testSequence);</a>
<span class="sourceLineNo">102</span><a id="line.102">                                                logger.info("PRE seq={}", predictedProteinSequence);</a>
<span class="sourceLineNo">103</span><a id="line.103">                                                logger.info("UNI seq={}", hitSequence);</a>
<span class="sourceLineNo">104</span><a id="line.104">                                                //  throw new Exception("Protein Sequence compare error " + id);</a>
<span class="sourceLineNo">105</span><a id="line.105">                                        }</a>
<span class="sourceLineNo">106</span><a id="line.106"></a>
<span class="sourceLineNo">107</span><a id="line.107">                                        SequencePair&lt;ProteinSequence, AminoAcidCompound&gt; alignment = Alignments.getPairwiseAlignment(</a>
<span class="sourceLineNo">108</span><a id="line.108">                                                        transcriptSequence.getProteinSequence(), proteinSequence,</a>
<span class="sourceLineNo">109</span><a id="line.109">                                                        PairwiseSequenceAlignerType.LOCAL, new SimpleGapPenalty(),</a>
<span class="sourceLineNo">110</span><a id="line.110">                                                        SimpleSubstitutionMatrix.getBlosum62()</a>
<span class="sourceLineNo">111</span><a id="line.111">                                                        );</a>
<span class="sourceLineNo">112</span><a id="line.112">                                        // System.out.println();</a>
<span class="sourceLineNo">113</span><a id="line.113">                                        //    System.out.println(alignment.getSummary());</a>
<span class="sourceLineNo">114</span><a id="line.114">                                        //   System.out.println(new Pair().format(alignment));</a>
<span class="sourceLineNo">115</span><a id="line.115">                                        int proteinIndex = 0;</a>
<span class="sourceLineNo">116</span><a id="line.116">                                        int gff3Index = 0;</a>
<span class="sourceLineNo">117</span><a id="line.117">                                        for (int i = 0; i &lt; cdsProteinList.size(); i++) {</a>
<span class="sourceLineNo">118</span><a id="line.118">                                                ProteinSequence peptideSequence = cdsProteinList.get(i);</a>
<span class="sourceLineNo">119</span><a id="line.119">                                                String seq = peptideSequence.getSequenceAsString();</a>
<span class="sourceLineNo">120</span><a id="line.120">                                                Integer startIndex = null;</a>
<span class="sourceLineNo">121</span><a id="line.121">                                                int offsetStartIndex = 0;</a>
<span class="sourceLineNo">122</span><a id="line.122">                                                for (int s = 0; s &lt; seq.length(); s++) {</a>
<span class="sourceLineNo">123</span><a id="line.123">                                                        startIndex = alignment.getIndexInTargetForQueryAt(proteinIndex + s);</a>
<span class="sourceLineNo">124</span><a id="line.124">                                                        if (startIndex != null) {</a>
<span class="sourceLineNo">125</span><a id="line.125">                                                                startIndex = startIndex + 1;</a>
<span class="sourceLineNo">126</span><a id="line.126">                                                                offsetStartIndex = s;</a>
<span class="sourceLineNo">127</span><a id="line.127">                                                                break;</a>
<span class="sourceLineNo">128</span><a id="line.128">                                                        }</a>
<span class="sourceLineNo">129</span><a id="line.129">                                                }</a>
<span class="sourceLineNo">130</span><a id="line.130">                                                Integer endIndex = null;</a>
<span class="sourceLineNo">131</span><a id="line.131"></a>
<span class="sourceLineNo">132</span><a id="line.132">                                                int offsetEndIndex = 0;</a>
<span class="sourceLineNo">133</span><a id="line.133">                                                for (int e = 0; e &lt; seq.length(); e++) {</a>
<span class="sourceLineNo">134</span><a id="line.134">                                                        endIndex = alignment.getIndexInTargetForQueryAt(proteinIndex + seq.length() - 1 - e);</a>
<span class="sourceLineNo">135</span><a id="line.135">                                                        if (endIndex != null) {</a>
<span class="sourceLineNo">136</span><a id="line.136">                                                                endIndex = endIndex + 1;</a>
<span class="sourceLineNo">137</span><a id="line.137">                                                                offsetEndIndex = e;</a>
<span class="sourceLineNo">138</span><a id="line.138">                                                                break;</a>
<span class="sourceLineNo">139</span><a id="line.139">                                                        }</a>
<span class="sourceLineNo">140</span><a id="line.140">                                                }</a>
<span class="sourceLineNo">141</span><a id="line.141"></a>
<span class="sourceLineNo">142</span><a id="line.142">                                                proteinIndex = proteinIndex + seq.length();</a>
<span class="sourceLineNo">143</span><a id="line.143">                                                if (startIndex != null &amp;&amp; endIndex != null &amp;&amp; startIndex != endIndex) {</a>
<span class="sourceLineNo">144</span><a id="line.144">                                                        CDSSequence cdsSequence = cdsSequenceList.get(i);</a>
<span class="sourceLineNo">145</span><a id="line.145">                                                        String hitLabel = "";</a>
<span class="sourceLineNo">146</span><a id="line.146">                                                        if (transcriptSequence.getStrand() == Strand.POSITIVE) {</a>
<span class="sourceLineNo">147</span><a id="line.147">                                                                hitLabel = uniprotBestHit + "_" + startIndex + "_" + endIndex;</a>
<span class="sourceLineNo">148</span><a id="line.148">                                                        } else {</a>
<span class="sourceLineNo">149</span><a id="line.149">                                                                hitLabel = uniprotBestHit + "_" + endIndex + "_" + startIndex;</a>
<span class="sourceLineNo">150</span><a id="line.150">                                                        }</a>
<span class="sourceLineNo">151</span><a id="line.151">                                                        int dnaBeginIndex = cdsSequence.getBioBegin() + (3 * offsetStartIndex);</a>
<span class="sourceLineNo">152</span><a id="line.152">                                                        int dnaEndIndex = cdsSequence.getBioEnd() - (3 * offsetEndIndex);</a>
<span class="sourceLineNo">153</span><a id="line.153">                                                        String scaffold = geneSequence.getParentChromosomeSequence().getAccession().getID();</a>
<span class="sourceLineNo">154</span><a id="line.154">                                        //        if (scaffoldsReferencedHashMap.containsKey(scaffold) == false) {</a>
<span class="sourceLineNo">155</span><a id="line.155">                                        //            String gff3line = scaffold + "\t" + geneSequence.getSource() + "\t" + "size" + "\t" + "1" + "\t" + geneSequence.getParentChromosomeSequence().getBioEnd() + "\t.\t.\t.\tName=" + scaffold + "\r\n";</a>
<span class="sourceLineNo">156</span><a id="line.156">                                        //            gff3Output.write(gff3line.getBytes());</a>
<span class="sourceLineNo">157</span><a id="line.157">                                        //            scaffoldsReferencedHashMap.put(scaffold, scaffold);</a>
<span class="sourceLineNo">158</span><a id="line.158">                                        //        }</a>
<span class="sourceLineNo">159</span><a id="line.159"></a>
<span class="sourceLineNo">160</span><a id="line.160">                                                        String line = scaffold + "\t" + geneSequence.getSource() + "_" + "UNIPROT\tmatch\t" + dnaBeginIndex + "\t" + dnaEndIndex + "\t.\t" + transcriptSequence.getStrand().getStringRepresentation() + "\t.\t";</a>
<span class="sourceLineNo">161</span><a id="line.161">                                                        if (gff3Index == 0) {</a>
<span class="sourceLineNo">162</span><a id="line.162">                                                                FeaturesKeyWordInterface featureKeyWords = proteinSequence.getFeaturesKeyWord();</a>
<span class="sourceLineNo">163</span><a id="line.163">                                                                String notes = "";</a>
<span class="sourceLineNo">164</span><a id="line.164">                                                                if (featureKeyWords != null) {</a>
<span class="sourceLineNo">165</span><a id="line.165">                                                                        List&lt;String&gt; keyWords = featureKeyWords.getKeyWords();</a>
<span class="sourceLineNo">166</span><a id="line.166">                                                                        if (keyWords.size() &gt; 0) {</a>
<span class="sourceLineNo">167</span><a id="line.167">                                                                                notes = ";Note=";</a>
<span class="sourceLineNo">168</span><a id="line.168">                                                                                for (String note : keyWords) {</a>
<span class="sourceLineNo">169</span><a id="line.169">                                                                                        if (note.equals("Complete proteome")) {</a>
<span class="sourceLineNo">170</span><a id="line.170">                                                                                                continue;</a>
<span class="sourceLineNo">171</span><a id="line.171">                                                                                        }</a>
<span class="sourceLineNo">172</span><a id="line.172">                                                                                        if (note.equals("Direct protein sequencing")) {</a>
<span class="sourceLineNo">173</span><a id="line.173">                                                                                                continue;</a>
<span class="sourceLineNo">174</span><a id="line.174">                                                                                        }</a>
<span class="sourceLineNo">175</span><a id="line.175"></a>
<span class="sourceLineNo">176</span><a id="line.176">                                                                                        notes = notes + " " + note;</a>
<span class="sourceLineNo">177</span><a id="line.177">                                                                                        geneSequence.addNote(note); // add note/keyword which can be output in fasta header if needed</a>
<span class="sourceLineNo">178</span><a id="line.178">                                                                                }</a>
<span class="sourceLineNo">179</span><a id="line.179">                                                                        }</a>
<span class="sourceLineNo">180</span><a id="line.180"></a>
<span class="sourceLineNo">181</span><a id="line.181">                                                                }</a>
<span class="sourceLineNo">182</span><a id="line.182"></a>
<span class="sourceLineNo">183</span><a id="line.183">                                                                DatabaseReferenceInterface databaseReferences = proteinSequence.getDatabaseReferences();</a>
<span class="sourceLineNo">184</span><a id="line.184">                                                                if (databaseReferences != null) {</a>
<span class="sourceLineNo">185</span><a id="line.185">                                                                        Map&lt;String, List&lt;DBReferenceInfo&gt;&gt; databaseReferenceHashMap = databaseReferences.getDatabaseReferences();</a>
<span class="sourceLineNo">186</span><a id="line.186">                                                                        List&lt;DBReferenceInfo&gt; pfamList = databaseReferenceHashMap.get("Pfam");</a>
<span class="sourceLineNo">187</span><a id="line.187">                                                                        List&lt;DBReferenceInfo&gt; cazyList = databaseReferenceHashMap.get("CAZy");</a>
<span class="sourceLineNo">188</span><a id="line.188">                                                                        List&lt;DBReferenceInfo&gt; goList = databaseReferenceHashMap.get("GO");</a>
<span class="sourceLineNo">189</span><a id="line.189">                                                                        List&lt;DBReferenceInfo&gt; eccList = databaseReferenceHashMap.get("BRENDA");</a>
<span class="sourceLineNo">190</span><a id="line.190">                                                                        if (pfamList != null &amp;&amp; pfamList.size() &gt; 0) {</a>
<span class="sourceLineNo">191</span><a id="line.191">                                                                                if (notes.length() == 0) {</a>
<span class="sourceLineNo">192</span><a id="line.192">                                                                                        notes = ";Note=";</a>
<span class="sourceLineNo">193</span><a id="line.193">                                                                                }</a>
<span class="sourceLineNo">194</span><a id="line.194">                                                                                for (DBReferenceInfo note : pfamList) {</a>
<span class="sourceLineNo">195</span><a id="line.195">                                                                                        notes = notes + " " + note.getId();</a>
<span class="sourceLineNo">196</span><a id="line.196">                                                                                        geneSequence.addNote(note.getId()); // add note/keyword which can be output in fasta header if needed</a>
<span class="sourceLineNo">197</span><a id="line.197">                                                                                }</a>
<span class="sourceLineNo">198</span><a id="line.198">                                                                        }</a>
<span class="sourceLineNo">199</span><a id="line.199"></a>
<span class="sourceLineNo">200</span><a id="line.200">                                                                        if (cazyList != null &amp;&amp; cazyList.size() &gt; 0) {</a>
<span class="sourceLineNo">201</span><a id="line.201">                                                                                if (notes.length() == 0) {</a>
<span class="sourceLineNo">202</span><a id="line.202">                                                                                        notes = ";Note=";</a>
<span class="sourceLineNo">203</span><a id="line.203">                                                                                }</a>
<span class="sourceLineNo">204</span><a id="line.204">                                                                                for (DBReferenceInfo note : cazyList) {</a>
<span class="sourceLineNo">205</span><a id="line.205">                                                                                        notes = notes + " " + note.getId();</a>
<span class="sourceLineNo">206</span><a id="line.206">                                                                                        geneSequence.addNote(note.getId()); // add note/keyword which can be output in fasta header if needed</a>
<span class="sourceLineNo">207</span><a id="line.207">                                                                                        // System.out.println("CAZy=" + note);</a>
<span class="sourceLineNo">208</span><a id="line.208">                                                                                }</a>
<span class="sourceLineNo">209</span><a id="line.209">                                                                        }</a>
<span class="sourceLineNo">210</span><a id="line.210"></a>
<span class="sourceLineNo">211</span><a id="line.211">                                                                        if (eccList != null &amp;&amp; eccList.size() &gt; 0) {</a>
<span class="sourceLineNo">212</span><a id="line.212">                                                                                if (notes.length() == 0) {</a>
<span class="sourceLineNo">213</span><a id="line.213">                                                                                        notes = ";Note=";</a>
<span class="sourceLineNo">214</span><a id="line.214">                                                                                }</a>
<span class="sourceLineNo">215</span><a id="line.215">                                                                                for (DBReferenceInfo note : eccList) {</a>
<span class="sourceLineNo">216</span><a id="line.216">                                                                                        String dbid = note.getId();</a>
<span class="sourceLineNo">217</span><a id="line.217">                                                                                        dbid = dbid.replace(".", "_"); //replace . with _ to facilitate searching in gbrowse</a>
<span class="sourceLineNo">218</span><a id="line.218">                                                                                        notes = notes + " " + "EC:" + dbid;</a>
<span class="sourceLineNo">219</span><a id="line.219">                                                                                        geneSequence.addNote("EC:" + dbid); // add note/keyword which can be output in fasta header if needed</a>
<span class="sourceLineNo">220</span><a id="line.220"></a>
<span class="sourceLineNo">221</span><a id="line.221">                                                                                }</a>
<span class="sourceLineNo">222</span><a id="line.222">                                                                        }</a>
<span class="sourceLineNo">223</span><a id="line.223"></a>
<span class="sourceLineNo">224</span><a id="line.224">                                                                        if (goList != null &amp;&amp; goList.size() &gt; 0) {</a>
<span class="sourceLineNo">225</span><a id="line.225">                                                                                if (notes.length() == 0) {</a>
<span class="sourceLineNo">226</span><a id="line.226">                                                                                        notes = ";Note=";</a>
<span class="sourceLineNo">227</span><a id="line.227">                                                                                }</a>
<span class="sourceLineNo">228</span><a id="line.228">                                                                                for (DBReferenceInfo note : goList) {</a>
<span class="sourceLineNo">229</span><a id="line.229">                                                                                        notes = notes + " " + note.getId();</a>
<span class="sourceLineNo">230</span><a id="line.230">                                                                                        geneSequence.addNote(note.getId()); // add note/keyword which can be output in fasta header if needed</a>
<span class="sourceLineNo">231</span><a id="line.231">                                                                                        LinkedHashMap&lt;String, String&gt; properties = note.getProperties();</a>
<span class="sourceLineNo">232</span><a id="line.232">                                                                                        for (String propertytype : properties.keySet()) {</a>
<span class="sourceLineNo">233</span><a id="line.233">                                                                                                if (propertytype.equals("evidence")) {</a>
<span class="sourceLineNo">234</span><a id="line.234">                                                                                                        continue;</a>
<span class="sourceLineNo">235</span><a id="line.235">                                                                                                }</a>
<span class="sourceLineNo">236</span><a id="line.236">                                                                                                String property = properties.get(propertytype);</a>
<span class="sourceLineNo">237</span><a id="line.237"></a>
<span class="sourceLineNo">238</span><a id="line.238">                                                                                                if (property.startsWith("C:")) {</a>
<span class="sourceLineNo">239</span><a id="line.239">                                                                                                        continue; // skip over the location</a>
<span class="sourceLineNo">240</span><a id="line.240">                                                                                                }</a>
<span class="sourceLineNo">241</span><a id="line.241">                                                                                                if (property.endsWith("...")) {</a>
<span class="sourceLineNo">242</span><a id="line.242">                                                                                                        property = property.substring(0, property.length() - 3);</a>
<span class="sourceLineNo">243</span><a id="line.243">                                                                                                }</a>
<span class="sourceLineNo">244</span><a id="line.244">                                                                                                notes = notes + " " + property;</a>
<span class="sourceLineNo">245</span><a id="line.245">                                                                                                geneSequence.addNote(property);</a>
<span class="sourceLineNo">246</span><a id="line.246">                                                                                        }</a>
<span class="sourceLineNo">247</span><a id="line.247">                                                                                }</a>
<span class="sourceLineNo">248</span><a id="line.248">                                                                        }</a>
<span class="sourceLineNo">249</span><a id="line.249"></a>
<span class="sourceLineNo">250</span><a id="line.250">                                                                }</a>
<span class="sourceLineNo">251</span><a id="line.251"></a>
<span class="sourceLineNo">252</span><a id="line.252"></a>
<span class="sourceLineNo">253</span><a id="line.253">                                                                line = line + "Name=" + hitLabel + ";Alias=" + uniprotBestHit + notes + "\n";</a>
<span class="sourceLineNo">254</span><a id="line.254">                                                        } else {</a>
<span class="sourceLineNo">255</span><a id="line.255">                                                                line = line + "Name=" + hitLabel + "\n";</a>
<span class="sourceLineNo">256</span><a id="line.256">                                                        }</a>
<span class="sourceLineNo">257</span><a id="line.257">                                                        gff3Index++;</a>
<span class="sourceLineNo">258</span><a id="line.258"></a>
<span class="sourceLineNo">259</span><a id="line.259">                                                        gff3Output.write(line.getBytes());</a>
<span class="sourceLineNo">260</span><a id="line.260">                                                }</a>
<span class="sourceLineNo">261</span><a id="line.261">                                        }</a>
<span class="sourceLineNo">262</span><a id="line.262">                                }</a>
<span class="sourceLineNo">263</span><a id="line.263">                        } catch (Exception e) {</a>
<span class="sourceLineNo">264</span><a id="line.264">                                logger.info("Accession Id: {}", accessionid, e);</a>
<span class="sourceLineNo">265</span><a id="line.265">                        }</a>
<span class="sourceLineNo">266</span><a id="line.266">                }</a>
<span class="sourceLineNo">267</span><a id="line.267"></a>
<span class="sourceLineNo">268</span><a id="line.268"></a>
<span class="sourceLineNo">269</span><a id="line.269"></a>
<span class="sourceLineNo">270</span><a id="line.270">        }</a>
<span class="sourceLineNo">271</span><a id="line.271"></a>
<span class="sourceLineNo">272</span><a id="line.272"></a>
<span class="sourceLineNo">273</span><a id="line.273"></a>
<span class="sourceLineNo">274</span><a id="line.274"></a>
<span class="sourceLineNo">275</span><a id="line.275">        public static void main(String[] args) {</a>
<span class="sourceLineNo">276</span><a id="line.276">                /*</a>
<span class="sourceLineNo">277</span><a id="line.277">                        try {</a>
<span class="sourceLineNo">278</span><a id="line.278">                                LinkedHashMap&lt;String, ChromosomeSequence&gt; dnaSequenceList = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGeneMarkGTF(new File("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/454Scaffolds.fna"), new File("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/genemark_hmm.gtf"));</a>
<span class="sourceLineNo">279</span><a id="line.279">                                LinkedHashMap&lt;String, GeneSequence&gt; geneSequenceList = GeneFeatureHelper.getGeneSequences(dnaSequenceList.values());</a>
<span class="sourceLineNo">280</span><a id="line.280">                                FileOutputStream fo = new FileOutputStream("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/genemark_uniprot_match.gff3");</a>
<span class="sourceLineNo">281</span><a id="line.281"></a>
<span class="sourceLineNo">282</span><a id="line.282">                                GFF3FromUniprotBlastHits gff3FromUniprotBlastHits = new GFF3FromUniprotBlastHits();</a>
<span class="sourceLineNo">283</span><a id="line.283">                                gff3FromUniprotBlastHits.process(new File("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/c1-454Scaffolds-hits-uniprot_fungi.xml"), 1E-10, geneSequenceList, fo);</a>
<span class="sourceLineNo">284</span><a id="line.284">                                fo.close();</a>
<span class="sourceLineNo">285</span><a id="line.285"></a>
<span class="sourceLineNo">286</span><a id="line.286"></a>
<span class="sourceLineNo">287</span><a id="line.287">                        } catch (Exception e) {</a>
<span class="sourceLineNo">288</span><a id="line.288">                                logger.error("Exception: ", e);</a>
<span class="sourceLineNo">289</span><a id="line.289"></a>
<span class="sourceLineNo">290</span><a id="line.290"></a>
<span class="sourceLineNo">291</span><a id="line.291">                        }</a>
<span class="sourceLineNo">292</span><a id="line.292">                */</a>
<span class="sourceLineNo">293</span><a id="line.293"></a>
<span class="sourceLineNo">294</span><a id="line.294">                        try {</a>
<span class="sourceLineNo">295</span><a id="line.295">                                LinkedHashMap&lt;String, ChromosomeSequence&gt; dnaSequenceHashMap = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGlimmerGFF3(new File("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/454Scaffolds-16.fna"), new File("/Users/Scooter/scripps/dyadic/GlimmerHMM/c1_glimmerhmm-16.gff"));</a>
<span class="sourceLineNo">296</span><a id="line.296">                                LinkedHashMap&lt;String, GeneSequence&gt; geneSequenceList = GeneFeatureHelper.getGeneSequences(dnaSequenceHashMap.values());</a>
<span class="sourceLineNo">297</span><a id="line.297">                                FileOutputStream fo = new FileOutputStream("/Users/Scooter/scripps/dyadic/outputGlimmer/genemark_uniprot_match-16.gff3");</a>
<span class="sourceLineNo">298</span><a id="line.298">                                LinkedHashMap&lt;String, ArrayList&lt;String&gt;&gt; blasthits = BlastHomologyHits.getMatches(new File("/Users/Scooter/scripps/dyadic/blastresults/c1_glimmer_in_uniprot.xml"), 1E-10);</a>
<span class="sourceLineNo">299</span><a id="line.299">                                logger.error("Number of uniprot hits " + blasthits.size());</a>
<span class="sourceLineNo">300</span><a id="line.300"></a>
<span class="sourceLineNo">301</span><a id="line.301">                                GFF3FromUniprotBlastHits gff3FromUniprotBlastHits = new GFF3FromUniprotBlastHits();</a>
<span class="sourceLineNo">302</span><a id="line.302">                                gff3FromUniprotBlastHits.process(blasthits, geneSequenceList, fo);</a>
<span class="sourceLineNo">303</span><a id="line.303">                                fo.close();</a>
<span class="sourceLineNo">304</span><a id="line.304">                        } catch (Exception e) {</a>
<span class="sourceLineNo">305</span><a id="line.305">                                logger.error("Exception: ", e);</a>
<span class="sourceLineNo">306</span><a id="line.306">                        }</a>
<span class="sourceLineNo">307</span><a id="line.307">        }</a>
<span class="sourceLineNo">308</span><a id="line.308">}</a>




























































</pre>
</div>
</main>
</body>
</html>
