<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.11) on Fri Oct 29 16:26:12 PDT 2021 -->
<title>ChromosomeMappingTools (biojava 6.0.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2021-10-29">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="ChromosomeMappingTools (biojava 6.0.0 API)";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":9,"i1":9,"i2":9,"i3":9,"i4":9,"i5":9,"i6":9,"i7":9,"i8":9,"i9":9,"i10":9,"i11":9,"i12":9,"i13":9,"i14":9,"i15":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/ChromosomeMappingTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">org.biojava.nbio.genome.util</a></div>
<h2 title="Class ChromosomeMappingTools" class="title">Class ChromosomeMappingTools</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.nbio.genome.util.ChromosomeMappingTools</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<pre>public class <a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.43">ChromosomeMappingTools</a>
extends <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a></pre>
<div class="block">A class that can map chromosomal positions to mRNA (coding sequence) positions.</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Andreas Prlic</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#CDS">CDS</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#CHROMOSOME">CHROMOSOME</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">ChromosomeMappingTools</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#formatExonStructure(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">formatExonStructure</a></span>&#8203;(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromosomePosition)</code></th>
<td class="colLast">
<div class="block">Pretty print the details of a GeneChromosomePosition to a String</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCDSExonRanges(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">getCDSExonRanges</a></span>&#8203;(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</code></th>
<td class="colLast">
<div class="block">Extracts the exon boundaries in CDS coordinates.</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCDSLength(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">getCDSLength</a></span>&#8203;(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</code></th>
<td class="colLast">
<div class="block">Get the length of the CDS in nucleotides.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCDSLengthForward(java.util.List,java.util.List,int,int)">getCDSLengthForward</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                   <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                   int&nbsp;cdsStart,
                   int&nbsp;cdsEnd)</code></th>
<td class="colLast">
<div class="block">Get the length of the coding sequence</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCDSLengthReverse(java.util.List,java.util.List,int,int)">getCDSLengthReverse</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                   <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                   int&nbsp;cdsStart,
                   int&nbsp;cdsEnd)</code></th>
<td class="colLast">
<div class="block">Get the length of the coding sequence</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCDSPosForChromosomeCoordinate(int,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">getCDSPosForChromosomeCoordinate</a></span>&#8203;(int&nbsp;coordinate,
                                <a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromosomePosition)</code></th>
<td class="colLast">
<div class="block">I have a genomic coordinate, where is it on the mRNA</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCDSPosForward(int,java.util.List,java.util.List,int,int)">getCDSPosForward</a></span>&#8203;(int&nbsp;chromPos,
                <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                int&nbsp;cdsStart,
                int&nbsp;cdsEnd)</code></th>
<td class="colLast">
<div class="block">Converts the genetic coordinate to the position of the nucleotide on the mRNA sequence for a gene
 living on the forward DNA strand.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCDSPosReverse(int,java.util.List,java.util.List,int,int)">getCDSPosReverse</a></span>&#8203;(int&nbsp;chromPos,
                <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                int&nbsp;cdsStart,
                int&nbsp;cdsEnd)</code></th>
<td class="colLast">
<div class="block">Converts the genetic coordinate to the position of the nucleotide on the mRNA sequence for a gene
 living on the reverse DNA strand.</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getCDSRegions(java.util.List,java.util.List,int,int)">getCDSRegions</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;origExonStarts,
             <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;origExonEnds,
             int&nbsp;cdsStart,
             int&nbsp;cdsEnd)</code></th>
<td class="colLast">
<div class="block">Extracts the exons boundaries in CDS coordinates corresponding to the forward DNA strand.</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromosomalRangesForCDS(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">getChromosomalRangesForCDS</a></span>&#8203;(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</code></th>
<td class="colLast">
<div class="block">Extracts the boundaries of the coding regions in chromosomal coordinates</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>static <a href="../parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromosomePosForCDScoordinate(int,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">getChromosomePosForCDScoordinate</a></span>&#8203;(int&nbsp;cdsNucleotidePosition,
                                <a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</code></th>
<td class="colLast">
<div class="block">Maps the position of a CDS nucleotide back to the genome</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>static <a href="../parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromPosForward(int,java.util.List,java.util.List,int,int)">getChromPosForward</a></span>&#8203;(int&nbsp;cdsPos,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                  int&nbsp;cdsStart,
                  int&nbsp;cdsEnd)</code></th>
<td class="colLast">
<div class="block">Get the CDS position mapped onto the chromosome position</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>static <a href="../parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getChromPosReverse(int,java.util.List,java.util.List,int,int)">getChromPosReverse</a></span>&#8203;(int&nbsp;cdsPos,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                  int&nbsp;cdsStart,
                  int&nbsp;cdsEnd)</code></th>
<td class="colLast">
<div class="block">Get the CDS position mapped on the chromosome position</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>static <a href="../../core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTranscriptDNASequence(org.biojava.nbio.genome.parsers.twobit.TwoBitFacade,java.lang.String,java.util.List,java.util.List,int,int,java.lang.Character)">getTranscriptDNASequence</a></span>&#8203;(<a href="../parsers/twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a>&nbsp;twoBitFacade,
                        <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chromosome,
                        <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                        <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                        int&nbsp;cdsStart,
                        int&nbsp;cdsEnd,
                        <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Character.html?is-external=true" title="class or interface in java.lang" class="externalLink">Character</a>&nbsp;orientation)</code></th>
<td class="colLast">
<div class="block">Extracts the DNA sequence transcribed from the input genetic coordinates.</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>static <a href="../../core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTranscriptDNASequence(org.biojava.nbio.genome.parsers.twobit.TwoBitFacade,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">getTranscriptDNASequence</a></span>&#8203;(<a href="../parsers/twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a>&nbsp;twoBitFacade,
                        <a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;gcp)</code></th>
<td class="colLast">
<div class="block">Extracts the DNA sequence transcribed from the input genetic coordinates.</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#setCoordinateSystem(int)">setCoordinateSystem</a></span>&#8203;(int&nbsp;baseInt)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a></h3>
<code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang" class="externalLink">clone</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang" class="externalLink">equals</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang" class="externalLink">finalize</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang" class="externalLink">getClass</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang" class="externalLink">hashCode</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang" class="externalLink">notify</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang" class="externalLink">notifyAll</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang" class="externalLink">toString</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang" class="externalLink">wait</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang" class="externalLink">wait</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#wait(long,int)" title="class or interface in java.lang" class="externalLink">wait</a></code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="CHROMOSOME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CHROMOSOME</h4>
<pre>public static final&nbsp;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a> <a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.49">CHROMOSOME</a></pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../../../constant-values.html#org.biojava.nbio.genome.util.ChromosomeMappingTools.CHROMOSOME">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a id="CDS">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>CDS</h4>
<pre>public static final&nbsp;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a> <a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.50">CDS</a></pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../../../constant-values.html#org.biojava.nbio.genome.util.ChromosomeMappingTools.CDS">Constant Field Values</a></dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>ChromosomeMappingTools</h4>
<pre>public&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.43">ChromosomeMappingTools</a>()</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="setCoordinateSystem(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCoordinateSystem</h4>
<pre class="methodSignature">public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.53">setCoordinateSystem</a>&#8203;(int&nbsp;baseInt)</pre>
</li>
</ul>
<a id="formatExonStructure(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>formatExonStructure</h4>
<pre class="methodSignature">public static&nbsp;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.63">formatExonStructure</a>&#8203;(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromosomePosition)</pre>
<div class="block">Pretty print the details of a GeneChromosomePosition to a String</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>chromosomePosition</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getCDSLength(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCDSLength</h4>
<pre class="methodSignature">public static&nbsp;int&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.210">getCDSLength</a>&#8203;(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</pre>
<div class="block">Get the length of the CDS in nucleotides.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>chromPos</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>length of the CDS in nucleotides.</dd>
</dl>
</li>
</ul>
<a id="getChromosomePosForCDScoordinate(int,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomePosForCDScoordinate</h4>
<pre class="methodSignature">public static&nbsp;<a href="../parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.232">getChromosomePosForCDScoordinate</a>&#8203;(int&nbsp;cdsNucleotidePosition,
                                                        <a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</pre>
<div class="block">Maps the position of a CDS nucleotide back to the genome</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>cdsNucleotidePosition</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a ChromPos object</dd>
</dl>
</li>
</ul>
<a id="getChromPosReverse(int,java.util.List,java.util.List,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromPosReverse</h4>
<pre class="methodSignature">public static&nbsp;<a href="../parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.277">getChromPosReverse</a>&#8203;(int&nbsp;cdsPos,
                                          <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                                          <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                                          int&nbsp;cdsStart,
                                          int&nbsp;cdsEnd)</pre>
<div class="block">Get the CDS position mapped on the chromosome position</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>exonStarts</code> - </dd>
<dd><code>exonEnds</code> - </dd>
<dd><code>cdsStart</code> - </dd>
<dd><code>cdsEnd</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getChromPosForward(int,java.util.List,java.util.List,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromPosForward</h4>
<pre class="methodSignature">public static&nbsp;<a href="../parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.415">getChromPosForward</a>&#8203;(int&nbsp;cdsPos,
                                          <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                                          <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                                          int&nbsp;cdsStart,
                                          int&nbsp;cdsEnd)</pre>
<div class="block">Get the CDS position mapped onto the chromosome position</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>exonStarts</code> - </dd>
<dd><code>exonEnds</code> - </dd>
<dd><code>cdsStart</code> - </dd>
<dd><code>cdsEnd</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getCDSLengthReverse(java.util.List,java.util.List,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCDSLengthReverse</h4>
<pre class="methodSignature">public static&nbsp;int&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.501">getCDSLengthReverse</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                                      <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                                      int&nbsp;cdsStart,
                                      int&nbsp;cdsEnd)</pre>
<div class="block">Get the length of the coding sequence</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>exonStarts</code> - </dd>
<dd><code>exonEnds</code> - </dd>
<dd><code>cdsStart</code> - </dd>
<dd><code>cdsEnd</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getCDSLengthForward(java.util.List,java.util.List,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCDSLengthForward</h4>
<pre class="methodSignature">public static&nbsp;int&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.548">getCDSLengthForward</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                                      <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                                      int&nbsp;cdsStart,
                                      int&nbsp;cdsEnd)</pre>
<div class="block">Get the length of the coding sequence</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>exonStarts</code> - </dd>
<dd><code>exonEnds</code> - </dd>
<dd><code>cdsStart</code> - </dd>
<dd><code>cdsEnd</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getCDSExonRanges(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCDSExonRanges</h4>
<pre class="methodSignature">public static&nbsp;<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&gt;&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.577">getCDSExonRanges</a>&#8203;(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</pre>
<div class="block">Extracts the exon boundaries in CDS coordinates. (needs to be divided by 3 to get AA positions)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>chromPos</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getChromosomalRangesForCDS(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomalRangesForCDS</h4>
<pre class="methodSignature">public static&nbsp;<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&gt;&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.588">getChromosomalRangesForCDS</a>&#8203;(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</pre>
<div class="block">Extracts the boundaries of the coding regions in chromosomal coordinates</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>chromPos</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getCDSPosForChromosomeCoordinate(int,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCDSPosForChromosomeCoordinate</h4>
<pre class="methodSignature">public static&nbsp;int&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.779">getCDSPosForChromosomeCoordinate</a>&#8203;(int&nbsp;coordinate,
                                                   <a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromosomePosition)</pre>
<div class="block">I have a genomic coordinate, where is it on the mRNA</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>coordinate</code> - </dd>
<dd><code>chromosomePosition</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getCDSPosForward(int,java.util.List,java.util.List,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCDSPosForward</h4>
<pre class="methodSignature">public static&nbsp;int&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.809">getCDSPosForward</a>&#8203;(int&nbsp;chromPos,
                                   <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                                   <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                                   int&nbsp;cdsStart,
                                   int&nbsp;cdsEnd)</pre>
<div class="block">Converts the genetic coordinate to the position of the nucleotide on the mRNA sequence for a gene
 living on the forward DNA strand.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>chromPos</code> - The genetic coordinate on a chromosome</dd>
<dd><code>exonStarts</code> - The list holding the genetic coordinates pointing to the start positions of the exons (including UTR regions)</dd>
<dd><code>exonEnds</code> - The list holding the genetic coordinates pointing to the end positions of the exons (including UTR regions)</dd>
<dd><code>cdsStart</code> - The start position of a coding region</dd>
<dd><code>cdsEnd</code> - The end position of a coding region</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the position of the nucleotide base on the mRNA sequence corresponding to the input genetic coordinate (base 1)</dd>
</dl>
</li>
</ul>
<a id="getCDSPosReverse(int,java.util.List,java.util.List,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCDSPosReverse</h4>
<pre class="methodSignature">public static&nbsp;int&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.856">getCDSPosReverse</a>&#8203;(int&nbsp;chromPos,
                                   <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                                   <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                                   int&nbsp;cdsStart,
                                   int&nbsp;cdsEnd)</pre>
<div class="block">Converts the genetic coordinate to the position of the nucleotide on the mRNA sequence for a gene
 living on the reverse DNA strand.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>chromPos</code> - The genetic coordinate on a chromosome</dd>
<dd><code>exonStarts</code> - The list holding the genetic coordinates pointing to the start positions of the exons (including UTR regions)</dd>
<dd><code>exonEnds</code> - The list holding the genetic coordinates pointing to the end positions of the exons (including UTR regions)</dd>
<dd><code>cdsStart</code> - The start position of a coding region</dd>
<dd><code>cdsEnd</code> - The end position of a coding region</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the position of the nucleotide base on the mRNA sequence corresponding to the input genetic coordinate (base 1)</dd>
</dl>
</li>
</ul>
<a id="getCDSRegions(java.util.List,java.util.List,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCDSRegions</h4>
<pre class="methodSignature">public static&nbsp;<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&gt;&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.899">getCDSRegions</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;origExonStarts,
                                                                           <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;origExonEnds,
                                                                           int&nbsp;cdsStart,
                                                                           int&nbsp;cdsEnd)</pre>
<div class="block">Extracts the exons boundaries in CDS coordinates corresponding to the forward DNA strand.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>origExonStarts</code> - The list holding the genetic coordinates pointing to the start positions of the exons (including UTR regions)</dd>
<dd><code>origExonEnds</code> - The list holding the genetic coordinates pointing to the end positions of the exons (including UTR regions)</dd>
<dd><code>cdsStart</code> - The start position of a coding region</dd>
<dd><code>cdsEnd</code> - The end position of a coding region</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the list of genetic positions corresponding to the exons boundaries in CDS coordinates</dd>
</dl>
</li>
</ul>
<a id="getTranscriptDNASequence(org.biojava.nbio.genome.parsers.twobit.TwoBitFacade,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTranscriptDNASequence</h4>
<pre class="methodSignature">public static&nbsp;<a href="../../core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.939">getTranscriptDNASequence</a>&#8203;(<a href="../parsers/twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a>&nbsp;twoBitFacade,
                                                   <a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;gcp)
                                            throws <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang" class="externalLink">Exception</a></pre>
<div class="block">Extracts the DNA sequence transcribed from the input genetic coordinates.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>twoBitFacade</code> - the facade that provide an access to a 2bit file</dd>
<dd><code>gcp</code> - The container with chromosomal positions</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the DNA sequence transcribed from the input genetic coordinates</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang" class="externalLink">Exception</a></code></dd>
</dl>
</li>
</ul>
<a id="getTranscriptDNASequence(org.biojava.nbio.genome.parsers.twobit.TwoBitFacade,java.lang.String,java.util.List,java.util.List,int,int,java.lang.Character)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getTranscriptDNASequence</h4>
<pre class="methodSignature">public static&nbsp;<a href="../../core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line.955">getTranscriptDNASequence</a>&#8203;(<a href="../parsers/twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a>&nbsp;twoBitFacade,
                                                   <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chromosome,
                                                   <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                                                   <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                                                   int&nbsp;cdsStart,
                                                   int&nbsp;cdsEnd,
                                                   <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Character.html?is-external=true" title="class or interface in java.lang" class="externalLink">Character</a>&nbsp;orientation)
                                            throws <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang" class="externalLink">Exception</a></pre>
<div class="block">Extracts the DNA sequence transcribed from the input genetic coordinates.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>chromosome</code> - the name of the chromosome</dd>
<dd><code>exonStarts</code> - The list holding the genetic coordinates pointing to the start positions of the exons (including UTR regions)</dd>
<dd><code>exonEnds</code> - The list holding the genetic coordinates pointing to the end positions of the exons (including UTR regions)</dd>
<dd><code>cdsStart</code> - The start position of a coding region</dd>
<dd><code>cdsEnd</code> - The end position of a coding region</dd>
<dd><code>orientation</code> - The orientation of the strand where the gene is living</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the DNA sequence transcribed from the input genetic coordinates</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang" class="externalLink">Exception</a></code></dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/ChromosomeMappingTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2021 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
