<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_162) on Fri Apr 06 16:19:08 PDT 2018 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Interface org.biojava.nbio.structure.Structure (biojava 5.0.0 API)</title>
<meta name="date" content="2018-04-06">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Interface org.biojava.nbio.structure.Structure (biojava 5.0.0 API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/structure/class-use/Structure.html" target="_top">Frames</a></li>
<li><a href="Structure.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Interface org.biojava.nbio.structure.Structure" class="title">Uses of Interface<br>org.biojava.nbio.structure.Structure</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#demo">demo</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.protmod.structure">org.biojava.nbio.protmod.structure</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure">org.biojava.nbio.structure</a></td>
<td class="colLast">
<div class="block"><BODY>
 
 Interfaces and classes for protein structure (PDB).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align">org.biojava.nbio.structure.align</a></td>
<td class="colLast">
<div class="block">Classes for the alignment of structures.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.ce">org.biojava.nbio.structure.align.ce</a></td>
<td class="colLast">
<div class="block">Classes related to the implementation of the CE alignment algorithm, here called jCE.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.client">org.biojava.nbio.structure.align.client</a></td>
<td class="colLast">
<div class="block">This package deals with the server communication for auto-downloading pre-calculated alignments.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui">org.biojava.nbio.structure.align.gui</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui.jmol">org.biojava.nbio.structure.align.gui.jmol</a></td>
<td class="colLast">
<div class="block">Utility methods for better interaction with Jmol.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.pairwise">org.biojava.nbio.structure.align.pairwise</a></td>
<td class="colLast">
<div class="block">Classes for the pairwise alignment of structures.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.quaternary">org.biojava.nbio.structure.align.quaternary</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.util">org.biojava.nbio.structure.align.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.asa">org.biojava.nbio.structure.asa</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.basepairs">org.biojava.nbio.structure.basepairs</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.cath">org.biojava.nbio.structure.cath</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.cluster">org.biojava.nbio.structure.cluster</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.contact">org.biojava.nbio.structure.contact</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.domain">org.biojava.nbio.structure.domain</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.ecod">org.biojava.nbio.structure.ecod</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.gui">org.biojava.nbio.structure.gui</a></td>
<td class="colLast">
<div class="block"><BODY>
 
 A few convenience classes to view protein structures with Jmol (if it is on the classpath),
 to calculate a protein structure alignment and to investigate the internals of the protein structure alignment algorithm.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.gui.events">org.biojava.nbio.structure.gui.events</a></td>
<td class="colLast">
<div class="block"><BODY>
 
 Some event classes for the protein structure GUIs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.gui.util">org.biojava.nbio.structure.gui.util</a></td>
<td class="colLast">
<div class="block"><BODY>
 
 Some utility classes for the protein structure GUIs.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></td>
<td class="colLast">
<div class="block">Input and Output of Structures</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.io.mmcif">org.biojava.nbio.structure.io.mmcif</a></td>
<td class="colLast">
<div class="block">Input and Output of mmcif files.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.io.mmtf">org.biojava.nbio.structure.io.mmtf</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.quaternary">org.biojava.nbio.structure.quaternary</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.scop">org.biojava.nbio.structure.scop</a></td>
<td class="colLast">
<div class="block">Parsers and API for <a href="http://scop.mrc-lmb.cam.ac.uk/scop/">SCOP, Structural Classification of Proteins</a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.secstruc">org.biojava.nbio.structure.secstruc</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.core">org.biojava.nbio.structure.symmetry.core</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.gui">org.biojava.nbio.structure.symmetry.gui</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.utils">org.biojava.nbio.structure.symmetry.utils</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.xtal">org.biojava.nbio.structure.xtal</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="demo">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../demo/package-summary.html">demo</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../demo/package-summary.html">demo</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">DemoShowLargeAssembly.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoShowLargeAssembly.html#readStructure-java.lang.String-int-">readStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
             int&nbsp;bioAssemblyId)</code>
<div class="block">Load a specific biological assembly for a PDB entry</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../demo/package-summary.html">demo</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">DemoLoadSecStruc.</span><code><span class="memberNameLink"><a href="../../../../../demo/DemoLoadSecStruc.html#printSecStruc-org.biojava.nbio.structure.Structure-">printSecStruc</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.protmod.structure">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/protmod/structure/package-summary.html">org.biojava.nbio.protmod.structure</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/protmod/structure/package-summary.html">org.biojava.nbio.protmod.structure</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">ProteinModificationIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/protmod/structure/ProteinModificationIdentifier.html#identify-org.biojava.nbio.structure.Structure-">identify</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>
<div class="block">Identify all registered modifications in a structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">ProteinModificationIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/protmod/structure/ProteinModificationIdentifier.html#identify-org.biojava.nbio.structure.Structure-java.util.Set-">identify</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
        <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../org/biojava/nbio/protmod/ProteinModification.html" title="interface in org.biojava.nbio.protmod">ProteinModification</a>&gt;&nbsp;potentialModifications)</code>
<div class="block">Identify a set of modifications in a structure.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> that implement <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureImpl.html" title="class in org.biojava.nbio.structure">StructureImpl</a></span></code>
<div class="block">Implementation of a PDB Structure.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureImpl.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureImpl.html#clone--">clone</a></span>()</code>
<div class="block">returns an identical copy of this structure .</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">Structure.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Structure.html#clone--">clone</a></span>()</code>
<div class="block">Return an identical copy of this Structure object</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getBiologicalAssembly-java.lang.String-">getBiologicalAssembly</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>
<div class="block">Returns the first biological assembly that is available for the given PDB id, 
 using multiModel=
  
 For more documentation on quaternary structures see:
 <code>http://pdb101.rcsb.org/learn/guide-to-understanding-pdb-data/biological-assemblies</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getBiologicalAssembly-java.lang.String-boolean-">getBiologicalAssembly</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
                     boolean&nbsp;multiModel)</code>
<div class="block">Returns the first biological assembly that is available for the given PDB id.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getBiologicalAssembly-java.lang.String-int-">getBiologicalAssembly</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
                     int&nbsp;biolAssemblyNr)</code>
<div class="block">Returns the biological assembly for the given PDB id and bioassembly identifier,
 using multiModel=</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getBiologicalAssembly-java.lang.String-int-boolean-">getBiologicalAssembly</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
                     int&nbsp;biolAssemblyNr,
                     boolean&nbsp;multiModel)</code>
<div class="block">Returns the biological assembly for the given PDB id and bioassembly identifier.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">Chain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Chain.html#getParent--">getParent</a></span>()</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;
<div class="block"><span class="deprecationComment">use getStructure(Structure) instead.</span></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">DBRef.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/DBRef.html#getParent--">getParent</a></span>()</code>
<div class="block">Get the structure object that this DBRef relates to.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">ChainImpl.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/ChainImpl.html#getParent--">getParent</a></span>()</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;
<div class="block"><span class="deprecationComment">use getStructure instead.</span></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getReducedStructure-org.biojava.nbio.structure.Structure-java.lang.String-">getReducedStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                   <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId)</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;
<div class="block"><span class="deprecationComment">Use <a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html#reduce-org.biojava.nbio.structure.Structure-"><code>StructureIdentifier.reduce(Structure)</code></a> instead (v. 4.2.0)</span></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">Chain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Chain.html#getStructure--">getStructure</a></span>()</code>
<div class="block">Returns the parent Structure of this chain.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">ChainImpl.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/ChainImpl.html#getStructure--">getStructure</a></span>()</code>
<div class="block">Returns the parent Structure of this chain.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getStructure-java.lang.String-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Short version of <a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getStructure-java.lang.String-org.biojava.nbio.structure.io.PDBFileParser-org.biojava.nbio.structure.align.util.AtomCache-"><code>StructureTools.getStructure(String, PDBFileParser, AtomCache)</code></a>
 which creates new parsers when needed</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getStructure-java.lang.String-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Loads a structure based on a name.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getStructure-java.lang.String-org.biojava.nbio.structure.io.PDBFileParser-org.biojava.nbio.structure.align.util.AtomCache-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
            <a href="../../../../../org/biojava/nbio/structure/io/PDBFileParser.html" title="class in org.biojava.nbio.structure.io">PDBFileParser</a>&nbsp;parser,
            <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Flexibly get a structure from an input String.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PassthroughIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/PassthroughIdentifier.html#loadStructure-org.biojava.nbio.structure.align.util.AtomCache-">loadStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Passthrough identifiers don't know how to load a structure</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">SubstructureIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/SubstructureIdentifier.html#loadStructure-org.biojava.nbio.structure.align.util.AtomCache-">loadStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Loads the complete structure based on <a href="../../../../../org/biojava/nbio/structure/SubstructureIdentifier.html#getPdbId--"><code>SubstructureIdentifier.getPdbId()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html#loadStructure-org.biojava.nbio.structure.align.util.AtomCache-">loadStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Loads a structure encompassing the structure identified.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">URLIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/URLIdentifier.html#loadStructure-org.biojava.nbio.structure.align.util.AtomCache-">loadStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Load the structure from the URL</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">BioAssemblyIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/BioAssemblyIdentifier.html#loadStructure-org.biojava.nbio.structure.align.util.AtomCache-">loadStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">Mutator.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Mutator.html#mutate-org.biojava.nbio.structure.Structure-java.lang.String-java.lang.String-java.lang.String-">mutate</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struc,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbResnum,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newType)</code>
<div class="block">creates a new structure which is identical with the original one.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PassthroughIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/PassthroughIdentifier.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">SubstructureIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/SubstructureIdentifier.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Takes a complete structure as input and reduces it to residues present in
 the specified ranges

 The returned structure will be a shallow copy of the input, with shared
 Chains, Residues, etc.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>
<div class="block">Takes a complete structure as input and reduces it to the substructure
 represented by this StructureIdentifier.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">URLIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/URLIdentifier.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">BioAssemblyIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/BioAssemblyIdentifier.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#removeModels-org.biojava.nbio.structure.Structure-">removeModels</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Remove all models from a Structure and keep only the first</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> that return types with arguments of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getBiologicalAssemblies-java.lang.String-">getBiologicalAssemblies</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>
<div class="block">Returns all biological assemblies for the given PDB id,
 using multiModel=
 
 If only one biological assembly is required use <a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getBiologicalAssembly-java.lang.String-"><code>StructureIO.getBiologicalAssembly(String)</code></a> or <a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getBiologicalAssembly-java.lang.String-int-"><code>StructureIO.getBiologicalAssembly(String, int)</code></a> instead.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">StructureIO.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getBiologicalAssemblies-java.lang.String-boolean-">getBiologicalAssemblies</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
                       boolean&nbsp;multiModel)</code>
<div class="block">Returns all biological assemblies for the given PDB id.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#addGroupsToStructure-org.biojava.nbio.structure.Structure-java.util.Collection-int-boolean-">addGroupsToStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                    <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;groups,
                    int&nbsp;model,
                    boolean&nbsp;clone)</code>
<div class="block">Add a list of groups to a new structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#addGroupToStructure-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Group-int-org.biojava.nbio.structure.Chain-boolean-">addGroupToStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                   <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;g,
                   int&nbsp;model,
                   <a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;chainGuess,
                   boolean&nbsp;clone)</code>
<div class="block">Adds a particular group to a structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#cleanUpAltLocs-org.biojava.nbio.structure.Structure-">cleanUpAltLocs</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>
<div class="block">Cleans up the structure's alternate location groups.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected static void</code></td>
<td class="colLast"><span class="typeNameLabel">SubstructureIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/SubstructureIdentifier.html#copyLigandsByProximity-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">copyLigandsByProximity</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;full,
                      <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;reduced)</code>
<div class="block">Supplements the reduced structure with ligands from the full structure based on
 a distance cutoff.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected static void</code></td>
<td class="colLast"><span class="typeNameLabel">SubstructureIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/SubstructureIdentifier.html#copyLigandsByProximity-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-double-int-int-">copyLigandsByProximity</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;full,
                      <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;reduced,
                      double&nbsp;cutoff,
                      int&nbsp;fromModel,
                      int&nbsp;toModel)</code>
<div class="block">Supplements the reduced structure with ligands from the full structure based on
 a distance cutoff.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAllAtomArray-org.biojava.nbio.structure.Structure-">getAllAtomArray</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Convert all atoms of the structure (all models) into an Atom array</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAllAtomArray-org.biojava.nbio.structure.Structure-int-">getAllAtomArray</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
               int&nbsp;model)</code>
<div class="block">Convert all atoms of the structure (specified model) into an Atom array</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAllNonHAtomArray-org.biojava.nbio.structure.Structure-boolean-">getAllNonHAtomArray</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                   boolean&nbsp;hetAtoms)</code>
<div class="block">Returns and array of all non-Hydrogen atoms in the given Structure,
 optionally including HET atoms or not.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAllNonHAtomArray-org.biojava.nbio.structure.Structure-boolean-int-">getAllNonHAtomArray</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                   boolean&nbsp;hetAtoms,
                   int&nbsp;modelNr)</code>
<div class="block">Returns and array of all non-Hydrogen atoms in the given Structure,
 optionally including HET atoms or not.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAtomArray-org.biojava.nbio.structure.Structure-java.lang.String:A-">getAtomArray</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;atomNames)</code>
<div class="block">Returns an array of the requested Atoms from the Structure object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAtomArrayAllModels-org.biojava.nbio.structure.Structure-java.lang.String:A-">getAtomArrayAllModels</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;atomNames)</code>
<div class="block">Returns an array of the requested Atoms from the Structure object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAtomCAArray-org.biojava.nbio.structure.Structure-">getAtomCAArray</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Return an Atom array of the C-alpha atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getBackboneAtomArray-org.biojava.nbio.structure.Structure-">getBackboneAtomArray</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Return an Atom array of the main chain atoms: CA, C, N, O Any group that
 contains those atoms will be included, be it a standard aminoacid or not</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getGroupByPDBResidueNumber-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.ResidueNumber-">getGroupByPDBResidueNumber</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struc,
                          <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbResNum)</code>
<div class="block">Get a group represented by a ResidueNumber.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getGroupDistancesWithinShell-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Atom-java.util.Set-double-boolean-boolean-">getGroupDistancesWithinShell</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid,
                            <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&gt;&nbsp;excludeResidues,
                            double&nbsp;radius,
                            boolean&nbsp;includeWater,
                            boolean&nbsp;useAverageDistance)</code>
<div class="block">Finds Groups in <code>structure</code> that contain at least one Atom that is
 within <code>radius</code> Angstroms of <code>centroid</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getGroupsWithinShell-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Atom-java.util.Set-double-boolean-">getGroupsWithinShell</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
                    <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&gt;&nbsp;excludeResidues,
                    double&nbsp;distance,
                    boolean&nbsp;includeWater)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getGroupsWithinShell-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Group-double-boolean-">getGroupsWithinShell</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                    <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group,
                    double&nbsp;distance,
                    boolean&nbsp;includeWater)</code>
<div class="block">
 Returns a Set of Groups in a structure within the distance specified of a
 given group.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getNrAtoms-org.biojava.nbio.structure.Structure-">getNrAtoms</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Count how many Atoms are contained within a Structure object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getNrGroups-org.biojava.nbio.structure.Structure-">getNrGroups</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Count how many groups are contained within a structure object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getReducedStructure-org.biojava.nbio.structure.Structure-java.lang.String-">getReducedStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                   <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId)</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;
<div class="block"><span class="deprecationComment">Use <a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html#reduce-org.biojava.nbio.structure.Structure-"><code>StructureIdentifier.reduce(Structure)</code></a> instead (v. 4.2.0)</span></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getRepresentativeAtomArray-org.biojava.nbio.structure.Structure-">getRepresentativeAtomArray</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Gets a representative atom for each group that is part of the chain
 backbone.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">Mutator.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Mutator.html#mutate-org.biojava.nbio.structure.Structure-java.lang.String-java.lang.String-java.lang.String-">mutate</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struc,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbResnum,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newType)</code>
<div class="block">creates a new structure which is identical with the original one.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Calc.html#plus-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.jama.Matrix-">plus</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
    <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;matrix)</code>
<div class="block">calculate structure + Matrix coodinates ...</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PassthroughIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/PassthroughIdentifier.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">SubstructureIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/SubstructureIdentifier.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Takes a complete structure as input and reduces it to residues present in
 the specified ranges

 The returned structure will be a shallow copy of the input, with shared
 Chains, Residues, etc.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>
<div class="block">Takes a complete structure as input and reduces it to the substructure
 represented by this StructureIdentifier.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">URLIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/URLIdentifier.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">BioAssemblyIdentifier.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/BioAssemblyIdentifier.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#removeModels-org.biojava.nbio.structure.Structure-">removeModels</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Remove all models from a Structure and keep only the first</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Calc.html#rotate-org.biojava.nbio.structure.Structure-double:A:A-">rotate</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
      double[][]&nbsp;rotationmatrix)</code>
<div class="block">Rotate a structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Calc.html#rotate-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.jama.Matrix-">rotate</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
      <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;m)</code>
<div class="block">Rotate a structure object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">Chain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Chain.html#setParent-org.biojava.nbio.structure.Structure-">setParent</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;parent)</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;
<div class="block"><span class="deprecationComment">use setStructure instead</span></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">DBRef.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/DBRef.html#setParent-org.biojava.nbio.structure.Structure-">setParent</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Set the structure object that this DBRef relates to.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">ChainImpl.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/ChainImpl.html#setParent-org.biojava.nbio.structure.Structure-">setParent</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;parent)</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">Chain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Chain.html#setStructure-org.biojava.nbio.structure.Structure-">setStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;parent)</code>
<div class="block">Sets the back-reference to its parent Structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">ChainImpl.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/ChainImpl.html#setStructure-org.biojava.nbio.structure.Structure-">setStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;parent)</code>
<div class="block">Sets the back-reference to its parent Structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Calc.html#shift-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Atom-">shift</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code>
<div class="block">shift a structure with a vector.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Calc.html#transform-org.biojava.nbio.structure.Structure-javax.vecmath.Matrix4d-">transform</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
         javax.vecmath.Matrix4d&nbsp;m)</code>
<div class="block">Transforms a structure object, given a Matrix4d (i.e. the vecmath library
 double-precision 4x4 rotation+translation matrix).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/Calc.html#translate-org.biojava.nbio.structure.Structure-javax.vecmath.Vector3d-">translate</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
         javax.vecmath.Vector3d&nbsp;v)</code>
<div class="block">Translates a Structure object, given a Vector3d (i.e. the vecmath library
 double-precision 3-d vector)</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/AtomIterator.html#AtomIterator-org.biojava.nbio.structure.Structure-">AtomIterator</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struct)</code>
<div class="block">Constructs an AtomIterator object over all models</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/AtomIterator.html#AtomIterator-org.biojava.nbio.structure.Structure-int-">AtomIterator</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struct,
            int&nbsp;modelNr)</code>
<div class="block">Constructs an AtomIterator object over a single model</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/AtomPositionMap.html#AtomPositionMap-org.biojava.nbio.structure.Structure-">AtomPositionMap</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Creates a new AtomPositionMap containing representative atoms
 from a structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/GroupIterator.html#GroupIterator-org.biojava.nbio.structure.Structure-">GroupIterator</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struct)</code>
<div class="block">Constructs a GroupIterator object over all models</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/GroupIterator.html#GroupIterator-org.biojava.nbio.structure.Structure-int-">GroupIterator</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struct,
             int&nbsp;modelNr)</code>
<div class="block">Constructs a GroupIterator object over a specific model</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">StructurePairAligner.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#align-org.biojava.nbio.structure.Structure-java.lang.String-org.biojava.nbio.structure.Structure-java.lang.String-">align</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId1,
     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId2)</code>
<div class="block">Align two chains from the structures.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">StructurePairAligner.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#align-org.biojava.nbio.structure.Structure-java.lang.String-org.biojava.nbio.structure.Structure-java.lang.String-org.biojava.nbio.structure.align.StrucAligParameters-">align</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId1,
     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId2,
     <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>
<div class="block">Aligns two chains from the structures using user provided parameters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">StructurePairAligner.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#align-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">align</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2)</code>
<div class="block">Calculate the alignment between the two full structures with default
 parameters</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">StructurePairAligner.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#align-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.align.StrucAligParameters-">align</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2,
     <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>
<div class="block">Calculate the alignment between the two full structures with user
 provided parameters</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">StructurePairAligner.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#getAlignmentAtoms-org.biojava.nbio.structure.Structure-">getAlignmentAtoms</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Returns the atoms that are being used for the alignment.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/MultiThreadedDBSearch.html#MultiThreadedDBSearch-java.lang.String-org.biojava.nbio.structure.Structure-java.lang.String-org.biojava.nbio.structure.align.StructureAlignment-int-boolean-">MultiThreadedDBSearch</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;outFile,
                     <a href="../../../../../org/biojava/nbio/structure/align/StructureAlignment.html" title="interface in org.biojava.nbio.structure.align">StructureAlignment</a>&nbsp;algorithm,
                     int&nbsp;nrCPUs,
                     boolean&nbsp;domainSplit)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.ce">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">GuiWrapper.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#getAlignedStructure-org.biojava.nbio.structure.Atom:A-org.biojava.nbio.structure.Atom:A-">getAlignedStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">GuiWrapper.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#showStructure-org.biojava.nbio.structure.Structure-">showStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>
<div class="block">Shows a structure in Jmol</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.client">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/align/client/package-summary.html">org.biojava.nbio.structure.align.client</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/client/package-summary.html">org.biojava.nbio.structure.align.client</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureName.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/client/StructureName.html#loadStructure-org.biojava.nbio.structure.align.util.AtomCache-">loadStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureName.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/client/StructureName.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/client/package-summary.html">org.biojava.nbio.structure.align.client</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureName.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/client/StructureName.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#createArtificalStructure-org.biojava.nbio.structure.align.model.AFPChain-org.biojava.nbio.structure.Atom:A-org.biojava.nbio.structure.Atom:A-">createArtificalStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Create a "fake" Structure objects that contains the two sets of atoms aligned on top of each other.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">SelectPDBPanel.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/gui/SelectPDBPanel.html#getStructure1--">getStructure1</a></span>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">SelectPDBPanel.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/gui/SelectPDBPanel.html#getStructure2--">getStructure2</a></span>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/gui/AlignmentCalc.html#AlignmentCalc-org.biojava.nbio.structure.align.gui.AlignmentGui-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-java.lang.String-java.lang.String-">AlignmentCalc</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/gui/AlignmentGui.html" title="class in org.biojava.nbio.structure.align.gui">AlignmentGui</a>&nbsp;parent,
             <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
             <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2,
             <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
             <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2)</code>
<div class="block">requests an alignment of pdb1 vs pdb 2.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/gui/AlignmentCalcDB.html#AlignmentCalcDB-org.biojava.nbio.structure.align.gui.AlignmentGui-org.biojava.nbio.structure.Structure-java.lang.String-org.biojava.nbio.structure.align.util.UserConfiguration-java.lang.String-java.lang.Boolean-">AlignmentCalcDB</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/gui/AlignmentGui.html" title="class in org.biojava.nbio.structure.align.gui">AlignmentGui</a>&nbsp;parent,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
               <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
               <a href="../../../../../org/biojava/nbio/structure/align/util/UserConfiguration.html" title="class in org.biojava.nbio.structure.align.util">UserConfiguration</a>&nbsp;config,
               <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;outFile,
               <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Boolean.html?is-external=true" title="class or interface in java.lang">Boolean</a>&nbsp;domainSplit)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructor parameters in <a href="../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with type arguments of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/gui/MultipleAlignmentCalc.html#MultipleAlignmentCalc-org.biojava.nbio.structure.align.gui.MultipleAlignmentGUI-java.util.List-java.util.List-">MultipleAlignmentCalc</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/gui/MultipleAlignmentGUI.html" title="class in org.biojava.nbio.structure.align.gui">MultipleAlignmentGUI</a>&nbsp;parent,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&gt;&nbsp;structures,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&gt;&nbsp;names)</code>
<div class="block">Requests an alignment of the pdbs.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui.jmol">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> declared as <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AbstractAlignmentJmol.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html#structure">structure</a></span></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AbstractAlignmentJmol.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html#getStructure--">getStructure</a></span>()</code>
<div class="block">Return the current Structure in the AlignmentJmol instance.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">AbstractAlignmentJmol.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html#setStructure-org.biojava.nbio.structure.Structure-">setStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Set a new Structure to visualize in the AlignmentJmol window.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">JmolPanel.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/JmolPanel.html#setStructure-org.biojava.nbio.structure.Structure-">setStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.multiple.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getAlignedStructure-java.util.List-">getAlignedStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code>
<div class="block">Get an artificial Structure containing a different model for every
 input structure, so that the alignment result can be viewed in Jmol.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#toMultimodelStructure-org.biojava.nbio.structure.align.multiple.MultipleAlignment-java.util.List-">toMultimodelStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformedAtoms)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.pairwise">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AlternativeAlignment.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#getAlignedStructure-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">getAlignedStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
                   <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2)</code>
<div class="block">create an artifical Structure object that contains the two
 structures superimposed onto each other.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AlternativeAlignment.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#getAlignedStructure-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">getAlignedStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
                   <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2)</code>
<div class="block">create an artifical Structure object that contains the two
 structures superimposed onto each other.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="typeNameLabel">AlternativeAlignment.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#toPDB-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">toPDB</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2)</code>
<div class="block">converts the alignment to a PDB file
 each of the structures will be represented as a model.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.quaternary">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/align/quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/quaternary/QsAlignResult.html" title="class in org.biojava.nbio.structure.align.quaternary">QsAlignResult</a></code></td>
<td class="colLast"><span class="typeNameLabel">QsAlign.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/quaternary/QsAlign.html#align-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.cluster.SubunitClustererParameters-org.biojava.nbio.structure.align.quaternary.QsAlignParameters-">align</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2,
     <a href="../../../../../org/biojava/nbio/structure/cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a>&nbsp;cParams,
     <a href="../../../../../org/biojava/nbio/structure/align/quaternary/QsAlignParameters.html" title="class in org.biojava.nbio.structure.align.quaternary">QsAlignParameters</a>&nbsp;aParams)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#createArtificalStructure-org.biojava.nbio.structure.align.model.AFPChain-org.biojava.nbio.structure.Atom:A-org.biojava.nbio.structure.Atom:A-">createArtificalStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#getAlignedStructure-org.biojava.nbio.structure.Atom:A-org.biojava.nbio.structure.Atom:A-">getAlignedStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Get an artifical Structure containing both chains.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getBiologicalAssembly-java.lang.String-boolean-">getBiologicalAssembly</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
                     boolean&nbsp;multiModel)</code>
<div class="block">Returns the default biological unit (bioassemblyId=1, known in PDB as pdb1.gz).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getBiologicalAssembly-java.lang.String-int-boolean-">getBiologicalAssembly</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
                     int&nbsp;bioAssemblyId,
                     boolean&nbsp;multiModel)</code>
<div class="block">Returns the biological assembly for a given PDB ID and bioAssemblyId, by building the 
 assembly from the biounit annotations found in <a href="../../../../../org/biojava/nbio/structure/Structure.html#getPDBHeader--"><code>getPDBHeader()</code></a>
 
 Note, the number of available biological unit files
 varies.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructure-java.lang.String-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Request a Structure based on a <i>name</i>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructure-org.biojava.nbio.structure.StructureIdentifier-">getStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;strucId)</code>
<div class="block">Get the structure corresponding to the given <a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure"><code>StructureIdentifier</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForCathDomain-org.biojava.nbio.structure.align.client.StructureName-">getStructureForCathDomain</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/client/StructureName.html" title="class in org.biojava.nbio.structure.align.client">StructureName</a>&nbsp;structureName)</code>
<div class="block">Returns a <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> corresponding to the CATH identifier supplied in <code>structureName</code>, using the the <a href="../../../../../org/biojava/nbio/structure/cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath"><code>CathDatabase</code></a>
 at <a href="../../../../../org/biojava/nbio/structure/cath/CathFactory.html#getCathDatabase--"><code>CathFactory.getCathDatabase()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForCathDomain-org.biojava.nbio.structure.align.client.StructureName-org.biojava.nbio.structure.cath.CathDatabase-">getStructureForCathDomain</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/client/StructureName.html" title="class in org.biojava.nbio.structure.align.client">StructureName</a>&nbsp;structureName,
                         <a href="../../../../../org/biojava/nbio/structure/cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath">CathDatabase</a>&nbsp;cathInstall)</code>
<div class="block">Returns a <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> corresponding to the CATH identifier supplied in <code>structureName</code>, using the specified <a href="../../../../../org/biojava/nbio/structure/cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath"><code>CathDatabase</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForDomain-org.biojava.nbio.structure.scop.ScopDomain-">getStructureForDomain</a></span>(<a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop">ScopDomain</a>&nbsp;domain)</code>
<div class="block">Returns the representation of a <a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForDomain-org.biojava.nbio.structure.scop.ScopDomain-org.biojava.nbio.structure.scop.ScopDatabase-">getStructureForDomain</a></span>(<a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop">ScopDomain</a>&nbsp;domain,
                     <a href="../../../../../org/biojava/nbio/structure/scop/ScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">ScopDatabase</a>&nbsp;scopDatabase)</code>
<div class="block">Returns the representation of a <a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForDomain-org.biojava.nbio.structure.scop.ScopDomain-org.biojava.nbio.structure.scop.ScopDatabase-boolean-">getStructureForDomain</a></span>(<a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop">ScopDomain</a>&nbsp;domain,
                     <a href="../../../../../org/biojava/nbio/structure/scop/ScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">ScopDatabase</a>&nbsp;scopDatabase,
                     boolean&nbsp;strictLigandHandling)</code>
<div class="block">Returns the representation of a <a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForDomain-java.lang.String-">getStructureForDomain</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;scopId)</code>
<div class="block">Returns the representation of a <a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForDomain-java.lang.String-org.biojava.nbio.structure.scop.ScopDatabase-">getStructureForDomain</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;scopId,
                     <a href="../../../../../org/biojava/nbio/structure/scop/ScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">ScopDatabase</a>&nbsp;scopDatabase)</code>
<div class="block">Returns the representation of a <a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForPdbId-java.lang.String-">getStructureForPdbId</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>
<div class="block">Loads a structure directly by PDB ID</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#loadStructureFromCifByPdbId-java.lang.String-">loadStructureFromCifByPdbId</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#loadStructureFromPdbByPdbId-java.lang.String-">loadStructureFromPdbByPdbId</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> that return types with arguments of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getBiologicalAssemblies-java.lang.String-boolean-">getBiologicalAssemblies</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
                       boolean&nbsp;multiModel)</code>
<div class="block">Returns all biological assemblies for given PDB id.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.asa">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/asa/package-summary.html">org.biojava.nbio.structure.asa</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/asa/package-summary.html">org.biojava.nbio.structure.asa</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/asa/AsaCalculator.html#AsaCalculator-org.biojava.nbio.structure.Structure-double-int-int-boolean-">AsaCalculator</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
             double&nbsp;probe,
             int&nbsp;nSpherePoints,
             int&nbsp;nThreads,
             boolean&nbsp;hetAtoms)</code>
<div class="block">Constructs a new AsaCalculator.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.basepairs">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/basepairs/package-summary.html">org.biojava.nbio.structure.basepairs</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../../../org/biojava/nbio/structure/basepairs/package-summary.html">org.biojava.nbio.structure.basepairs</a> declared as <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">BasePairParameters.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/basepairs/BasePairParameters.html#structure">structure</a></span></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/basepairs/package-summary.html">org.biojava.nbio.structure.basepairs</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/basepairs/BasePairParameters.html#BasePairParameters-org.biojava.nbio.structure.Structure-">BasePairParameters</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>
<div class="block">This constructor takes a Structure object, finds base pair and base-pair step parameters
 for double-helical regions within the structure for only canonical DNA pairs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/basepairs/BasePairParameters.html#BasePairParameters-org.biojava.nbio.structure.Structure-boolean-">BasePairParameters</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                  boolean&nbsp;useRNA)</code>
<div class="block">This constructor takes a Structure object, and whether to use the RNA standard bases.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/basepairs/BasePairParameters.html#BasePairParameters-org.biojava.nbio.structure.Structure-boolean-boolean-">BasePairParameters</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                  boolean&nbsp;useRNA,
                  boolean&nbsp;removeDups)</code>
<div class="block">This constructor takes a Structure object, whether to use RNA, and whether to remove duplicate sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/basepairs/BasePairParameters.html#BasePairParameters-org.biojava.nbio.structure.Structure-boolean-boolean-boolean-">BasePairParameters</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                  boolean&nbsp;useRNA,
                  boolean&nbsp;removeDups,
                  boolean&nbsp;canonical)</code>
<div class="block">This constructor takes a Structure object, finds base pair and base-pair step parameters
 for double-helical regions within the structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/basepairs/MismatchedBasePairParameters.html#MismatchedBasePairParameters-org.biojava.nbio.structure.Structure-boolean-boolean-boolean-">MismatchedBasePairParameters</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                            boolean&nbsp;RNA,
                            boolean&nbsp;removeDups,
                            boolean&nbsp;canonical)</code>
<div class="block">This constructor is used to create the TertiaryBasePairParameters object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/basepairs/TertiaryBasePairParameters.html#TertiaryBasePairParameters-org.biojava.nbio.structure.Structure-boolean-boolean-">TertiaryBasePairParameters</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                          boolean&nbsp;RNA,
                          boolean&nbsp;removeDups)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.cath">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/cath/package-summary.html">org.biojava.nbio.structure.cath</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/cath/package-summary.html">org.biojava.nbio.structure.cath</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">CathDomain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/cath/CathDomain.html#loadStructure-org.biojava.nbio.structure.align.util.AtomCache-">loadStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">CathDomain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/cath/CathDomain.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/cath/package-summary.html">org.biojava.nbio.structure.cath</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">CathDomain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/cath/CathDomain.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.cluster">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/cluster/package-summary.html">org.biojava.nbio.structure.cluster</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">Subunit.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/cluster/Subunit.html#getStructure--">getStructure</a></span>()</code>
<div class="block">The parent Structure from which the Subunit atoms were obtained.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/symmetry/core/Stoichiometry.html" title="class in org.biojava.nbio.structure.symmetry.core">Stoichiometry</a></code></td>
<td class="colLast"><span class="typeNameLabel">SubunitClusterer.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/cluster/SubunitClusterer.html#cluster-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.cluster.SubunitClustererParameters-">cluster</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
       <a href="../../../../../org/biojava/nbio/structure/cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster">Subunit</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">SubunitExtractor.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/cluster/SubunitExtractor.html#extractSubunits-org.biojava.nbio.structure.Structure-int-double-int-">extractSubunits</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
               int&nbsp;absMinLen,
               double&nbsp;fraction,
               int&nbsp;minLen)</code>
<div class="block">Extract the information of each protein Chain in a Structure and converts
 them into a List of Subunit.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/cluster/Subunit.html#Subunit-org.biojava.nbio.structure.Atom:A-java.lang.String-org.biojava.nbio.structure.StructureIdentifier-org.biojava.nbio.structure.Structure-">Subunit</a></span>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;reprAtoms,
       <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
       <a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;identifier,
       <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>
<div class="block">A Subunit is solely defined by the coordinates of the representative
 Atoms of its residues.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.contact">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/contact/package-summary.html">org.biojava.nbio.structure.contact</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/contact/package-summary.html">org.biojava.nbio.structure.contact</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/contact/StructureInterfaceList.html" title="class in org.biojava.nbio.structure.contact">StructureInterfaceList</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureInterfaceList.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/contact/StructureInterfaceList.html#calculateInterfaces-org.biojava.nbio.structure.Structure-">calculateInterfaces</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struc)</code>
<div class="block">Calculates the interfaces for a structure using default parameters</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.domain">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/domain/package-summary.html">org.biojava.nbio.structure.domain</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/domain/package-summary.html">org.biojava.nbio.structure.domain</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDPProvider.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/domain/PDPProvider.html#getDomain-java.lang.String-org.biojava.nbio.structure.align.util.AtomCache-">getDomain</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdpDomainName,
         <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Get the structure for a particular PDP domain</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">RemotePDPProvider.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/domain/RemotePDPProvider.html#getDomain-java.lang.String-org.biojava.nbio.structure.align.util.AtomCache-">getDomain</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdpDomainName,
         <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Get the structure for a particular PDP domain</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDPDomain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/domain/PDPDomain.html#loadStructure-org.biojava.nbio.structure.align.util.AtomCache-">loadStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDPDomain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/domain/PDPDomain.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/domain/package-summary.html">org.biojava.nbio.structure.domain</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDPDomain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/domain/PDPDomain.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">LocalProteinDomainParser.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/domain/LocalProteinDomainParser.html#suggestDomains-org.biojava.nbio.structure.Structure-">suggestDomains</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Suggest domains for a protein structure</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.ecod">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/ecod/package-summary.html">org.biojava.nbio.structure.ecod</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/ecod/package-summary.html">org.biojava.nbio.structure.ecod</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">EcodDomain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/ecod/EcodDomain.html#loadStructure-org.biojava.nbio.structure.align.util.AtomCache-">loadStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">EcodDomain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/ecod/EcodDomain.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/ecod/package-summary.html">org.biojava.nbio.structure.ecod</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">EcodDomain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/ecod/EcodDomain.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.gui">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/gui/package-summary.html">org.biojava.nbio.structure.gui</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/gui/package-summary.html">org.biojava.nbio.structure.gui</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">BiojavaJmol.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/BiojavaJmol.html#setStructure-org.biojava.nbio.structure.Structure-">setStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">StructureViewer.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/StructureViewer.html#setStructure-org.biojava.nbio.structure.Structure-">setStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">JmolViewerImpl.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/JmolViewerImpl.html#setStructure-org.biojava.nbio.structure.Structure-">setStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">SequenceDisplay.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/SequenceDisplay.html#setStructure1-org.biojava.nbio.structure.Structure-">setStructure1</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">SequenceDisplay.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/SequenceDisplay.html#setStructure2-org.biojava.nbio.structure.Structure-">setStructure2</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.gui.events">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/gui/events/package-summary.html">org.biojava.nbio.structure.gui.events</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/gui/events/package-summary.html">org.biojava.nbio.structure.gui.events</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">JmolAlignedPositionListener.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/events/JmolAlignedPositionListener.html#setStructure1-org.biojava.nbio.structure.Structure-">setStructure1</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">JmolAlignedPositionListener.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/events/JmolAlignedPositionListener.html#setStructure2-org.biojava.nbio.structure.Structure-">setStructure2</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.gui.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/gui/util/package-summary.html">org.biojava.nbio.structure.gui.util</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/gui/util/package-summary.html">org.biojava.nbio.structure.gui.util</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDBUploadPanel.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/PDBUploadPanel.html#getStructure1--">getStructure1</a></span>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDBServerPanel.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/PDBServerPanel.html#getStructure1--">getStructure1</a></span>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDBDirPanel.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/PDBDirPanel.html#getStructure1--">getStructure1</a></span>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">ScopSelectPanel.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/ScopSelectPanel.html#getStructure1--">getStructure1</a></span>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructurePairSelector.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/StructurePairSelector.html#getStructure1--">getStructure1</a></span>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDBUploadPanel.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/PDBUploadPanel.html#getStructure2--">getStructure2</a></span>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDBServerPanel.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/PDBServerPanel.html#getStructure2--">getStructure2</a></span>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDBDirPanel.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/PDBDirPanel.html#getStructure2--">getStructure2</a></span>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">ScopSelectPanel.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/ScopSelectPanel.html#getStructure2--">getStructure2</a></span>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructurePairSelector.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/StructurePairSelector.html#getStructure2--">getStructure2</a></span>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/gui/util/package-summary.html">org.biojava.nbio.structure.gui.util</a> that return types with arguments of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">SelectMultiplePanel.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/SelectMultiplePanel.html#getStructures--">getStructures</a></span>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/gui/util/package-summary.html">org.biojava.nbio.structure.gui.util</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/AlignmentCalc.html#AlignmentCalc-org.biojava.nbio.structure.align.gui.AlignmentGui-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">AlignmentCalc</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/gui/AlignmentGui.html" title="class in org.biojava.nbio.structure.align.gui">AlignmentGui</a>&nbsp;parent,
             <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
             <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2)</code>
<div class="block">requests an alignment of pdb1 vs pdb 2.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/gui/util/AlternativeAlignmentFrame.html#AlternativeAlignmentFrame-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">AlternativeAlignmentFrame</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
                         <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">LocalPDBDirectory.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/LocalPDBDirectory.html#getStructure-java.io.File-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;filename)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureIOFile.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/StructureIOFile.html#getStructure-java.io.File-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file)</code>
<div class="block">Read file from File and returns
 a Structure object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">LocalPDBDirectory.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/LocalPDBDirectory.html#getStructure-java.io.InputStream-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;inStream)</code>
<div class="block">Handles the actual parsing of the file into a Structure object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">MMCIFFileReader.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/MMCIFFileReader.html#getStructure-java.io.InputStream-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;inStream)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">MMTFFileReader.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/MMTFFileReader.html#getStructure-java.io.InputStream-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;inStream)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDBFileReader.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/PDBFileReader.html#getStructure-java.io.InputStream-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;inStream)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">LocalPDBDirectory.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/LocalPDBDirectory.html#getStructure-java.lang.String-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;filename)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureIOFile.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/StructureIOFile.html#getStructure-java.lang.String-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;filename)</code>
<div class="block">Open filename and return a Structure object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">LocalPDBDirectory.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/LocalPDBDirectory.html#getStructure-java.net.URL-">getStructure</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/net/URL.html?is-external=true" title="class or interface in java.net">URL</a>&nbsp;u)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureProvider.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/StructureProvider.html#getStructureById-java.lang.String-">getStructureById</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>
<div class="block">Get the structure for a PDB ID</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">LocalPDBDirectory.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/LocalPDBDirectory.html#getStructureById-java.lang.String-">getStructureById</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">SandboxStyleStructureProvider.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/SandboxStyleStructureProvider.html#getStructureById-java.lang.String-">getStructureById</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">FastaStructureParser.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#getStructures--">getStructures</a></span>()</code>
<div class="block">Gets the protein structures mapped from the Fasta file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureSequenceMatcher.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/StructureSequenceMatcher.html#getSubstructureMatchingProteinSequence-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.structure.Structure-">getSubstructureMatchingProteinSequence</a></span>(<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                                      <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;wholeStructure)</code>
<div class="block">Get a substructure of <code>wholeStructure</code> containing only the <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure"><code>Groups</code></a> that are included in
 <code>sequence</code>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDBFileParser.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/PDBFileParser.html#parsePDBFile-java.io.BufferedReader-">parsePDBFile</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;buf)</code>
<div class="block">Parse a PDB file and return a datastructure implementing
 PDBStructure interface.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">PDBFileParser.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/PDBFileParser.html#parsePDBFile-java.io.InputStream-">parsePDBFile</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;inStream)</code>
<div class="block">Parse a PDB file and return a datastructure implementing
 PDBStructure interface.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">ChargeAdder.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/ChargeAdder.html#addCharges-org.biojava.nbio.structure.Structure-">addCharges</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>
<div class="block">Function to add the charges to a given structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">SeqRes2AtomAligner.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/SeqRes2AtomAligner.html#align-org.biojava.nbio.structure.Structure-java.util.List-">align</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&gt;&nbsp;seqResList)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#cpFastaToAfpChain-java.io.File-org.biojava.nbio.structure.Structure-int-">cpFastaToAfpChain</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;fastaFile,
                 <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code>
<div class="block">Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site
 <code>cpSite</code> residues to the right.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#cpFastaToAfpChain-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.structure.Structure-int-">cpFastaToAfpChain</a></span>(<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;first,
                 <a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;second,
                 <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code>
<div class="block">Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site
 <code>cpSite</code> residues to the right.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#cpFastaToAfpChain-java.lang.String-java.lang.String-org.biojava.nbio.structure.Structure-int-">cpFastaToAfpChain</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;first,
                 <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;second,
                 <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaFileToAfpChain-java.io.File-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">fastaFileToAfpChain</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;fastaFile,
                   <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
                   <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Reads the file <code>fastaFile</code>, expecting exactly two sequences which give a pairwise alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaStringToAfpChain-java.lang.String-java.lang.String-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">fastaStringToAfpChain</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence1,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence2,
                     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
                     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Returns an AFPChain corresponding to the alignment between <code>structure1</code> and <code>structure2</code>, which is given by the gapped protein sequences <code>sequence1</code> and <code>sequence2</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain-java.util.Map-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">fastaToAfpChain</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;&nbsp;sequences,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Uses two sequences each with a corresponding structure to create an AFPChain corresponding to the alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">fastaToAfpChain</a></span>(<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence1,
               <a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence2,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Returns an AFPChain corresponding to the alignment between <code>structure1</code> and <code>structure2</code>, which is given by the gapped protein sequences <code>sequence1</code> and <code>sequence2</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain-org.biojava.nbio.core.alignment.template.SequencePair-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">fastaToAfpChain</a></span>(<a href="../../../../../org/biojava/nbio/core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;,<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;alignment,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Provided only for convenience.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="typeNameLabel">FastaAFPChainConverter.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain-java.lang.String-java.lang.String-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.Structure-">fastaToAfpChain</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence1,
               <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence2,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">TODO Write comment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureSequenceMatcher.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/StructureSequenceMatcher.html#getProteinSequenceForStructure-org.biojava.nbio.structure.Structure-java.util.Map-">getProteinSequenceForStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struct,
                              <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>,<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;groupIndexPosition)</code>
<div class="block">Generates a ProteinSequence corresponding to the sequence of struct,
 and maintains a mapping from the sequence back to the original groups.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">StructureSequenceMatcher.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/StructureSequenceMatcher.html#getSubstructureMatchingProteinSequence-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.structure.Structure-">getSubstructureMatchingProteinSequence</a></span>(<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence,
                                      <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;wholeStructure)</code>
<div class="block">Get a substructure of <code>wholeStructure</code> containing only the <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure"><code>Groups</code></a> that are included in
 <code>sequence</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">StructureSequenceMatcher.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/StructureSequenceMatcher.html#matchSequenceToStructure-org.biojava.nbio.core.sequence.ProteinSequence-org.biojava.nbio.structure.Structure-">matchSequenceToStructure</a></span>(<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;seq,
                        <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struct)</code>
<div class="block">Given a sequence and the corresponding Structure, get the ResidueNumber
 for each residue in the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">SeqRes2AtomAligner.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/SeqRes2AtomAligner.html#storeUnAlignedSeqRes-org.biojava.nbio.structure.Structure-java.util.List-boolean-">storeUnAlignedSeqRes</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                    <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&gt;&nbsp;seqResChains,
                    boolean&nbsp;headerOnly)</code>
<div class="block">Storing unaligned SEQRES groups in a Structure.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/BondMaker.html#BondMaker-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.io.FileParsingParameters-">BondMaker</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
         <a href="../../../../../org/biojava/nbio/structure/io/FileParsingParameters.html" title="class in org.biojava.nbio.structure.io">FileParsingParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/FileConvert.html#FileConvert-org.biojava.nbio.structure.Structure-">FileConvert</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struc)</code>
<div class="block">Constructs a FileConvert object.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.io.mmcif">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/io/mmcif/package-summary.html">org.biojava.nbio.structure.io.mmcif</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/io/mmcif/package-summary.html">org.biojava.nbio.structure.io.mmcif</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">SimpleMMcifConsumer.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmcif/SimpleMMcifConsumer.html#getStructure--">getStructure</a></span>()</code>
<div class="block">This method will return the parsed protein structure, once the parsing has been finished</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/io/mmcif/package-summary.html">org.biojava.nbio.structure.io.mmcif</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/io/mmcif/model/AtomSite.html" title="class in org.biojava.nbio.structure.io.mmcif.model">AtomSite</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">MMCIFFileTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmcif/MMCIFFileTools.html#convertStructureToAtomSites-org.biojava.nbio.structure.Structure-">convertStructureToAtomSites</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Converts a Structure into a List of <a href="../../../../../org/biojava/nbio/structure/io/mmcif/model/AtomSite.html" title="class in org.biojava.nbio.structure.io.mmcif.model"><code>AtomSite</code></a> objects</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.io.mmtf">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/io/mmtf/package-summary.html">org.biojava.nbio.structure.io.mmtf</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/io/mmtf/package-summary.html">org.biojava.nbio.structure.io.mmtf</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">MmtfStructureReader.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmtf/MmtfStructureReader.html#getStructure--">getStructure</a></span>()</code>
<div class="block">Gets the structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">MmtfActions.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmtf/MmtfActions.html#readFromFile-java.nio.file.Path-">readFromFile</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html?is-external=true" title="class or interface in java.nio.file">Path</a>&nbsp;filePath)</code>
<div class="block">Get a Structure object from a mmtf file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">MmtfActions.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmtf/MmtfActions.html#readFromInputStream-java.io.InputStream-">readFromInputStream</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;inStream)</code>
<div class="block">Read a Biojava structure from an <a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io"><code>InputStream</code></a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">MmtfActions.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmtf/MmtfActions.html#readFromWeb-java.lang.String-">readFromWeb</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>
<div class="block">Get a Biojava structure from the mmtf REST service.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/io/mmtf/package-summary.html">org.biojava.nbio.structure.io.mmtf</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">MmtfUtils.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmtf/MmtfUtils.html#calculateDsspSecondaryStructure-org.biojava.nbio.structure.Structure-">calculateDsspSecondaryStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;bioJavaStruct)</code>
<div class="block">Generate the secondary structure for a Biojava structure object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">MmtfUtils.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmtf/MmtfUtils.html#fixMicroheterogenity-org.biojava.nbio.structure.Structure-">fixMicroheterogenity</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;bioJavaStruct)</code>
<div class="block">This sets all microheterogeneous groups 
 (previously alternate location groups) as separate groups.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><span class="typeNameLabel">MmtfUtils.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmtf/MmtfUtils.html#getNumGroups-org.biojava.nbio.structure.Structure-">getNumGroups</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>
<div class="block">Count the total number of groups in the structure</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/io/mmtf/MmtfSummaryDataBean.html" title="class in org.biojava.nbio.structure.io.mmtf">MmtfSummaryDataBean</a></code></td>
<td class="colLast"><span class="typeNameLabel">MmtfUtils.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmtf/MmtfUtils.html#getStructureInfo-org.biojava.nbio.structure.Structure-">getStructureInfo</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>
<div class="block">Get summary information for the structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">MmtfActions.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmtf/MmtfActions.html#writeToFile-org.biojava.nbio.structure.Structure-java.nio.file.Path-">writeToFile</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
           <a href="https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html?is-external=true" title="class or interface in java.nio.file">Path</a>&nbsp;path)</code>
<div class="block">Write a Structure object to a file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">MmtfActions.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmtf/MmtfActions.html#writeToOutputStream-org.biojava.nbio.structure.Structure-java.io.OutputStream-">writeToOutputStream</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                   <a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;outputStream)</code>
<div class="block">Write a Structure object to an <a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io"><code>OutputStream</code></a></div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/io/mmtf/package-summary.html">org.biojava.nbio.structure.io.mmtf</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/io/mmtf/MmtfStructureWriter.html#MmtfStructureWriter-org.biojava.nbio.structure.Structure-org.rcsb.mmtf.api.StructureAdapterInterface-">MmtfStructureWriter</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                   org.rcsb.mmtf.api.StructureAdapterInterface&nbsp;dataTransferInterface)</code>
<div class="block">Pass data from Biojava structure  to another generic output type.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.quaternary">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/quaternary/package-summary.html">org.biojava.nbio.structure.quaternary</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/quaternary/package-summary.html">org.biojava.nbio.structure.quaternary</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">BioAssemblyTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/quaternary/BioAssemblyTools.html#getReducedStructure-org.biojava.nbio.structure.Structure-">getReducedStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;orig)</code>
<div class="block">Reduce a structure to a single-atom representation (e.g.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">BiologicalAssemblyBuilder.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/quaternary/BiologicalAssemblyBuilder.html#rebuildQuaternaryStructure-org.biojava.nbio.structure.Structure-java.util.List-boolean-boolean-">rebuildQuaternaryStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;asymUnit,
                          <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/quaternary/BiologicalAssemblyTransformation.html" title="class in org.biojava.nbio.structure.quaternary">BiologicalAssemblyTransformation</a>&gt;&nbsp;transformations,
                          boolean&nbsp;useAsymIds,
                          boolean&nbsp;multiModel)</code>
<div class="block">Builds a Structure object containing the quaternary structure built from given asymUnit and transformations,
 by adding symmetry partners as new models.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/quaternary/package-summary.html">org.biojava.nbio.structure.quaternary</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static double[][]</code></td>
<td class="colLast"><span class="typeNameLabel">BioAssemblyTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/quaternary/BioAssemblyTools.html#getAtomCoordinateBounds-org.biojava.nbio.structure.Structure-">getAtomCoordinateBounds</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double[][]</code></td>
<td class="colLast"><span class="typeNameLabel">BioAssemblyTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/quaternary/BioAssemblyTools.html#getBiologicalMoleculeBounds-org.biojava.nbio.structure.Structure-java.util.List-">getBiologicalMoleculeBounds</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;asymStructure,
                           <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/quaternary/BiologicalAssemblyTransformation.html" title="class in org.biojava.nbio.structure.quaternary">BiologicalAssemblyTransformation</a>&gt;&nbsp;transformations)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double[]</code></td>
<td class="colLast"><span class="typeNameLabel">BioAssemblyTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/quaternary/BioAssemblyTools.html#getBiologicalMoleculeCentroid-org.biojava.nbio.structure.Structure-java.util.List-">getBiologicalMoleculeCentroid</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;asymUnit,
                             <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/quaternary/BiologicalAssemblyTransformation.html" title="class in org.biojava.nbio.structure.quaternary">BiologicalAssemblyTransformation</a>&gt;&nbsp;transformations)</code>
<div class="block">Returns the centroid of the biological molecule.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="typeNameLabel">BioAssemblyTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/quaternary/BioAssemblyTools.html#getBiologicalMoleculeMaximumExtend-org.biojava.nbio.structure.Structure-java.util.List-">getBiologicalMoleculeMaximumExtend</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                                  <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/quaternary/BiologicalAssemblyTransformation.html" title="class in org.biojava.nbio.structure.quaternary">BiologicalAssemblyTransformation</a>&gt;&nbsp;transformations)</code>
<div class="block">Returns the maximum extend of the biological molecule in the x, y, or z direction.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="typeNameLabel">BioAssemblyTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/quaternary/BioAssemblyTools.html#getMaximumExtend-org.biojava.nbio.structure.Structure-">getMaximumExtend</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>
<div class="block">Returns the maximum extend of the structure in the x, y, or z direction.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">BioAssemblyTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/quaternary/BioAssemblyTools.html#getReducedStructure-org.biojava.nbio.structure.Structure-">getReducedStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;orig)</code>
<div class="block">Reduce a structure to a single-atom representation (e.g.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">BiologicalAssemblyBuilder.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/quaternary/BiologicalAssemblyBuilder.html#rebuildQuaternaryStructure-org.biojava.nbio.structure.Structure-java.util.List-boolean-boolean-">rebuildQuaternaryStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;asymUnit,
                          <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/quaternary/BiologicalAssemblyTransformation.html" title="class in org.biojava.nbio.structure.quaternary">BiologicalAssemblyTransformation</a>&gt;&nbsp;transformations,
                          boolean&nbsp;useAsymIds,
                          boolean&nbsp;multiModel)</code>
<div class="block">Builds a Structure object containing the quaternary structure built from given asymUnit and transformations,
 by adding symmetry partners as new models.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.scop">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/scop/package-summary.html">org.biojava.nbio.structure.scop</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/scop/package-summary.html">org.biojava.nbio.structure.scop</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">ScopDomain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html#loadStructure-org.biojava.nbio.structure.align.util.AtomCache-">loadStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">ScopDomain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/scop/package-summary.html">org.biojava.nbio.structure.scop</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">ScopDomain.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html#reduce-org.biojava.nbio.structure.Structure-">reduce</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.secstruc">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/secstruc/package-summary.html">org.biojava.nbio.structure.secstruc</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/secstruc/package-summary.html">org.biojava.nbio.structure.secstruc</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">SecStrucTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucTools.html#assignSecStruc-org.biojava.nbio.structure.Structure-java.util.List-">assignSecStruc</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
              <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucInfo.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucInfo</a>&gt;&nbsp;listSSI)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">SecStrucCalc.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucCalc.html#calculate-org.biojava.nbio.structure.Structure-boolean-">calculate</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
         boolean&nbsp;assign)</code>
<div class="block">Predicts the secondary structure of this Structure object,
 using a DSSP implementation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">DSSPParser.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/secstruc/DSSPParser.html#fetch-java.lang.String-org.biojava.nbio.structure.Structure-boolean-">fetch</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdb,
     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
     boolean&nbsp;assign)</code>
<div class="block">Fetch and parse the DSSP file of the specified pdb code
 from the PDB web server and return the secondary structure
 annotation as a List of <a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc"><code>SecStrucState</code></a> objects.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucElement.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucElement</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">SecStrucTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucTools.html#getSecStrucElements-org.biojava.nbio.structure.Structure-">getSecStrucElements</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Obtain the List of secondary structure elements (SecStrucElement) of a
 Structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucInfo.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucInfo</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">SecStrucTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucTools.html#getSecStrucInfo-org.biojava.nbio.structure.Structure-">getSecStrucInfo</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Obtain the List of secondary structure information (SecStrucInfo) of a
 Structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">DSSPParser.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/secstruc/DSSPParser.html#parseFile-java.lang.String-org.biojava.nbio.structure.Structure-boolean-">parseFile</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dsspPath,
         <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
         boolean&nbsp;assign)</code>
<div class="block">Parse a DSSP output file and return the secondary structure
 annotation as a List of <a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc"><code>SecStrucState</code></a> objects.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">DSSPParser.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/secstruc/DSSPParser.html#parseInputStream-java.io.InputStream-org.biojava.nbio.structure.Structure-boolean-">parseInputStream</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;dsspIs,
                <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                boolean&nbsp;assign)</code>
<div class="block">Parse a DSSP output file and return the secondary structure
 annotation as a List of <a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc"><code>SecStrucState</code></a> objects.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">DSSPParser.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/secstruc/DSSPParser.html#parseString-java.lang.String-org.biojava.nbio.structure.Structure-boolean-">parseString</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dsspOut,
           <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
           boolean&nbsp;assign)</code>
<div class="block">Parse a DSSP format String and return the secondary structure
 annotation as a List of <a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc"><code>SecStrucState</code></a> objects.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.core">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/symmetry/core/package-summary.html">org.biojava.nbio.structure.symmetry.core</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/symmetry/core/package-summary.html">org.biojava.nbio.structure.symmetry.core</a> that return <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="typeNameLabel">QuatSymmetryResults.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html#getStructure--">getStructure</a></span>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/symmetry/core/package-summary.html">org.biojava.nbio.structure.symmetry.core</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryResults</a></code></td>
<td class="colLast"><span class="typeNameLabel">QuatSymmetryDetector.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/symmetry/core/QuatSymmetryDetector.html#calcGlobalSymmetry-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.symmetry.core.QuatSymmetryParameters-org.biojava.nbio.structure.cluster.SubunitClustererParameters-">calcGlobalSymmetry</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                  <a href="../../../../../org/biojava/nbio/structure/symmetry/core/QuatSymmetryParameters.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryParameters</a>&nbsp;symmParams,
                  <a href="../../../../../org/biojava/nbio/structure/cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a>&nbsp;clusterParams)</code>
<div class="block">Calculate GLOBAL symmetry results.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryResults</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">QuatSymmetryDetector.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/symmetry/core/QuatSymmetryDetector.html#calcLocalSymmetries-org.biojava.nbio.structure.Structure-org.biojava.nbio.structure.symmetry.core.QuatSymmetryParameters-org.biojava.nbio.structure.cluster.SubunitClustererParameters-">calcLocalSymmetries</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                   <a href="../../../../../org/biojava/nbio/structure/symmetry/core/QuatSymmetryParameters.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryParameters</a>&nbsp;symmParams,
                   <a href="../../../../../org/biojava/nbio/structure/cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a>&nbsp;clusterParams)</code>
<div class="block">Returns a List of LOCAL symmetry results.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">QuatSymmetryResults.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html#setStructure-org.biojava.nbio.structure.Structure-">setStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.gui">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/symmetry/gui/package-summary.html">org.biojava.nbio.structure.symmetry.gui</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/symmetry/gui/package-summary.html">org.biojava.nbio.structure.symmetry.gui</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/symmetry/gui/SymmetryCalc.html#SymmetryCalc-org.biojava.nbio.structure.symmetry.gui.SymmetryGui-org.biojava.nbio.structure.Structure-">SymmetryCalc</a></span>(<a href="../../../../../org/biojava/nbio/structure/symmetry/gui/SymmetryGui.html" title="class in org.biojava.nbio.structure.symmetry.gui">SymmetryGui</a>&nbsp;p,
            <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Requests for a structure to analyze.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.utils">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> that return types with arguments of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#divideStructure-org.biojava.nbio.structure.symmetry.internal.CeSymmResult-">divideStructure</a></span>(<a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symmetry)</code>
<div class="block">Method that converts the symmetric units of a structure into different
 structures, so that they can be individually visualized.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#getRepresentativeAtoms-org.biojava.nbio.structure.Structure-">getRepresentativeAtoms</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>
<div class="block">Returns the representative Atom Array of the first model, if the
 structure is NMR, or the Array for each model, if it is a biological
 assembly with multiple models.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.xtal">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a> in <a href="../../../../../org/biojava/nbio/structure/xtal/package-summary.html">org.biojava.nbio.structure.xtal</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/xtal/package-summary.html">org.biojava.nbio.structure.xtal</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">CrystalBuilder.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/xtal/CrystalBuilder.html#expandNcsOps-org.biojava.nbio.structure.Structure-java.util.Map-java.util.Map-">expandNcsOps</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;chainOrigNames,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,javax.vecmath.Matrix4d&gt;&nbsp;chainNcsOps)</code>
<div class="block">Apply the NCS operators in the given Structure adding new chains as needed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">UnitCellBoundingBox.</span><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/xtal/UnitCellBoundingBox.html#setBbs-org.biojava.nbio.structure.Structure-javax.vecmath.Matrix4d:A-boolean-">setBbs</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
      javax.vecmath.Matrix4d[]&nbsp;ops,
      boolean&nbsp;includeHetAtoms)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/xtal/package-summary.html">org.biojava.nbio.structure.xtal</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/xtal/CrystalBuilder.html#CrystalBuilder-org.biojava.nbio.structure.Structure-">CrystalBuilder</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/structure/xtal/CrystalBuilder.html#CrystalBuilder-org.biojava.nbio.structure.Structure-java.util.Map-java.util.Map-">CrystalBuilder</a></span>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
              <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;chainOrigNames,
              <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,javax.vecmath.Matrix4d&gt;&nbsp;chainNcsOps)</code>
<div class="block">Special constructor for NCS-aware CrystalBuilder.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/structure/class-use/Structure.html" target="_top">Frames</a></li>
<li><a href="Structure.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2018 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
