<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> */</a>
<span class="sourceLineNo">021</span><a id="line.21">package org.biojava.nbio.aaproperties;</a>
<span class="sourceLineNo">022</span><a id="line.22"></a>
<span class="sourceLineNo">023</span><a id="line.23">import org.biojava.nbio.aaproperties.xml.AminoAcidCompositionTable;</a>
<span class="sourceLineNo">024</span><a id="line.24">import org.biojava.nbio.aaproperties.xml.CaseFreeAminoAcidCompoundSet;</a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</a>
<span class="sourceLineNo">026</span><a id="line.26">import org.biojava.nbio.core.sequence.ProteinSequence;</a>
<span class="sourceLineNo">027</span><a id="line.27">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</a>
<span class="sourceLineNo">028</span><a id="line.28">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.biojava.nbio.core.sequence.io.*;</a>
<span class="sourceLineNo">030</span><a id="line.30">import org.biojava.nbio.core.sequence.template.CompoundSet;</a>
<span class="sourceLineNo">031</span><a id="line.31"></a>
<span class="sourceLineNo">032</span><a id="line.32">import java.io.File;</a>
<span class="sourceLineNo">033</span><a id="line.33">import java.io.FileInputStream;</a>
<span class="sourceLineNo">034</span><a id="line.34">import java.io.IOException;</a>
<span class="sourceLineNo">035</span><a id="line.35">import java.io.PrintStream;</a>
<span class="sourceLineNo">036</span><a id="line.36">import java.util.ArrayList;</a>
<span class="sourceLineNo">037</span><a id="line.37">import java.util.LinkedHashMap;</a>
<span class="sourceLineNo">038</span><a id="line.38">import java.util.List;</a>
<span class="sourceLineNo">039</span><a id="line.39">import java.util.Map;</a>
<span class="sourceLineNo">040</span><a id="line.40">import java.util.Map.Entry;</a>
<span class="sourceLineNo">041</span><a id="line.41"></a>
<span class="sourceLineNo">042</span><a id="line.42"></a>
<span class="sourceLineNo">043</span><a id="line.43">public class CommandPrompt {</a>
<span class="sourceLineNo">044</span><a id="line.44"></a>
<span class="sourceLineNo">045</span><a id="line.45">        /**</a>
<span class="sourceLineNo">046</span><a id="line.46">         * The main method</a>
<span class="sourceLineNo">047</span><a id="line.47">         * @param args</a>
<span class="sourceLineNo">048</span><a id="line.48">         *      See showHelp for a list of available arguments</a>
<span class="sourceLineNo">049</span><a id="line.49">         * @throws Exception</a>
<span class="sourceLineNo">050</span><a id="line.50">         *  To handle exception thrown by reading of XML files</a>
<span class="sourceLineNo">051</span><a id="line.51">         */</a>
<span class="sourceLineNo">052</span><a id="line.52">        public static void main(String[] args) throws Exception{</a>
<span class="sourceLineNo">053</span><a id="line.53">                run(args);</a>
<span class="sourceLineNo">054</span><a id="line.54">        }</a>
<span class="sourceLineNo">055</span><a id="line.55"></a>
<span class="sourceLineNo">056</span><a id="line.56">        private static AminoAcidCompositionTable checkForValidityAndObtainAATable(String inputLocation, int propertyListSize, String aminoAcidCompositionLocation,</a>
<span class="sourceLineNo">057</span><a id="line.57">                        String elementMassLocation) throws Exception{</a>
<span class="sourceLineNo">058</span><a id="line.58">                if(inputLocation == null) {</a>
<span class="sourceLineNo">059</span><a id="line.59">                        showHelp();</a>
<span class="sourceLineNo">060</span><a id="line.60">                        throw new Error("Please do provide location of input file.");</a>
<span class="sourceLineNo">061</span><a id="line.61">                }</a>
<span class="sourceLineNo">062</span><a id="line.62">                if(propertyListSize == 0){</a>
<span class="sourceLineNo">063</span><a id="line.63">                        showHelp();</a>
<span class="sourceLineNo">064</span><a id="line.64">                        throw new Error("Please at least specify a property to compute.");</a>
<span class="sourceLineNo">065</span><a id="line.65">                }</a>
<span class="sourceLineNo">066</span><a id="line.66">                AminoAcidCompositionTable aaTable = null;</a>
<span class="sourceLineNo">067</span><a id="line.67">                if(aminoAcidCompositionLocation != null &amp;&amp; elementMassLocation == null){</a>
<span class="sourceLineNo">068</span><a id="line.68">                        aaTable = PeptideProperties.obtainAminoAcidCompositionTable(new File(aminoAcidCompositionLocation));</a>
<span class="sourceLineNo">069</span><a id="line.69">                }else if(aminoAcidCompositionLocation != null &amp;&amp; elementMassLocation != null){</a>
<span class="sourceLineNo">070</span><a id="line.70">                        aaTable = PeptideProperties.obtainAminoAcidCompositionTable(new File(aminoAcidCompositionLocation, elementMassLocation));</a>
<span class="sourceLineNo">071</span><a id="line.71">                }else if(aminoAcidCompositionLocation == null &amp;&amp; elementMassLocation != null){</a>
<span class="sourceLineNo">072</span><a id="line.72">                        throw new Error("You have define the location of Element Mass XML file. Please also define the location of Amino Acid Composition XML file");</a>
<span class="sourceLineNo">073</span><a id="line.73">                }</a>
<span class="sourceLineNo">074</span><a id="line.74">                return aaTable;</a>
<span class="sourceLineNo">075</span><a id="line.75">        }</a>
<span class="sourceLineNo">076</span><a id="line.76"></a>
<span class="sourceLineNo">077</span><a id="line.77">        private static void readInputAndGenerateOutput(String outputLocation, List&lt;Character&gt; propertyList, List&lt;Character&gt; specificList,</a>
<span class="sourceLineNo">078</span><a id="line.78">                        String delimiter, String inputLocation, AminoAcidCompositionTable aaTable, int decimalPlace) throws Exception{</a>
<span class="sourceLineNo">079</span><a id="line.79">                PrintStream output;</a>
<span class="sourceLineNo">080</span><a id="line.80">                if(outputLocation != null)</a>
<span class="sourceLineNo">081</span><a id="line.81">                        output = new PrintStream(new File(outputLocation));</a>
<span class="sourceLineNo">082</span><a id="line.82">                else</a>
<span class="sourceLineNo">083</span><a id="line.83">                        output = System.out;</a>
<span class="sourceLineNo">084</span><a id="line.84">                printHeader(output, propertyList, specificList, delimiter);</a>
<span class="sourceLineNo">085</span><a id="line.85">                LinkedHashMap&lt;String, ProteinSequence&gt; a = readInputFile(inputLocation, aaTable);</a>
<span class="sourceLineNo">086</span><a id="line.86">                //Need for the last sequence</a>
<span class="sourceLineNo">087</span><a id="line.87">                for(Entry&lt;String, ProteinSequence&gt; entry:a.entrySet()){</a>
<span class="sourceLineNo">088</span><a id="line.88">                        compute(output, entry.getValue().getOriginalHeader(), entry.getValue().getSequenceAsString().trim(), delimiter, aaTable, propertyList, specificList,</a>
<span class="sourceLineNo">089</span><a id="line.89">                                        decimalPlace);</a>
<span class="sourceLineNo">090</span><a id="line.90">                }</a>
<span class="sourceLineNo">091</span><a id="line.91">                output.close();</a>
<span class="sourceLineNo">092</span><a id="line.92">        }</a>
<span class="sourceLineNo">093</span><a id="line.93"></a>
<span class="sourceLineNo">094</span><a id="line.94">        public static void run(String[] args) throws Exception{</a>
<span class="sourceLineNo">095</span><a id="line.95">                /*</a>
<span class="sourceLineNo">096</span><a id="line.96">                 * Parse input arguments</a>
<span class="sourceLineNo">097</span><a id="line.97">                 */</a>
<span class="sourceLineNo">098</span><a id="line.98">                List&lt;Character&gt; propertyList = new ArrayList&lt;Character&gt;();</a>
<span class="sourceLineNo">099</span><a id="line.99">                List&lt;Character&gt; specificList = new ArrayList&lt;Character&gt;();</a>
<span class="sourceLineNo">100</span><a id="line.100">                String inputLocation = null;</a>
<span class="sourceLineNo">101</span><a id="line.101">                String outputLocation = null;</a>
<span class="sourceLineNo">102</span><a id="line.102">                String aminoAcidCompositionLocation = null;</a>
<span class="sourceLineNo">103</span><a id="line.103">                String elementMassLocation = null;</a>
<span class="sourceLineNo">104</span><a id="line.104">                String delimiter = ",";</a>
<span class="sourceLineNo">105</span><a id="line.105">                int decimalPlace = 4;</a>
<span class="sourceLineNo">106</span><a id="line.106"></a>
<span class="sourceLineNo">107</span><a id="line.107">                for(int i = 0; i &lt; args.length; i++){</a>
<span class="sourceLineNo">108</span><a id="line.108">                        if(args[i].charAt(0) != '-' || args[i].length() != 2){</a>
<span class="sourceLineNo">109</span><a id="line.109">                                showHelp();</a>
<span class="sourceLineNo">110</span><a id="line.110">                                throw new Error("Unknown option: " + args[i]);</a>
<span class="sourceLineNo">111</span><a id="line.111">                        }else{</a>
<span class="sourceLineNo">112</span><a id="line.112">                                switch(args[i].charAt(1)){</a>
<span class="sourceLineNo">113</span><a id="line.113">                                //Required</a>
<span class="sourceLineNo">114</span><a id="line.114">                                case 'i': inputLocation = args[++i]; break;</a>
<span class="sourceLineNo">115</span><a id="line.115">                                //Optional</a>
<span class="sourceLineNo">116</span><a id="line.116">                                case 'o': outputLocation = args[++i]; break;</a>
<span class="sourceLineNo">117</span><a id="line.117">                                case 'f':</a>
<span class="sourceLineNo">118</span><a id="line.118">                                        i++;</a>
<span class="sourceLineNo">119</span><a id="line.119">                                        if(args[i].equalsIgnoreCase("csv")) delimiter = ",";</a>
<span class="sourceLineNo">120</span><a id="line.120">                                        else if(args[i].equalsIgnoreCase("tsv")) delimiter = "\t";</a>
<span class="sourceLineNo">121</span><a id="line.121">                                        else throw new Error("Invalid value for -f: " + args[i] + ". Please choose either csv or tsv only.");</a>
<span class="sourceLineNo">122</span><a id="line.122">                                        break;</a>
<span class="sourceLineNo">123</span><a id="line.123">                                case 'x': aminoAcidCompositionLocation = args[++i]; break;</a>
<span class="sourceLineNo">124</span><a id="line.124">                                case 'y': elementMassLocation = args[++i]; break;</a>
<span class="sourceLineNo">125</span><a id="line.125">                                case 'd': decimalPlace = Integer.parseInt(args[++i]); break;</a>
<span class="sourceLineNo">126</span><a id="line.126">                                //Properties</a>
<span class="sourceLineNo">127</span><a id="line.127">                                case 'a':</a>
<span class="sourceLineNo">128</span><a id="line.128">                                        propertyList.add('1');</a>
<span class="sourceLineNo">129</span><a id="line.129">                                        propertyList.add('2');</a>
<span class="sourceLineNo">130</span><a id="line.130">                                        propertyList.add('3');</a>
<span class="sourceLineNo">131</span><a id="line.131">                                        propertyList.add('4');</a>
<span class="sourceLineNo">132</span><a id="line.132">                                        propertyList.add('5');</a>
<span class="sourceLineNo">133</span><a id="line.133">                                        propertyList.add('6');</a>
<span class="sourceLineNo">134</span><a id="line.134">                                        propertyList.add('7');</a>
<span class="sourceLineNo">135</span><a id="line.135">                                        propertyList.add('8');</a>
<span class="sourceLineNo">136</span><a id="line.136">                                        propertyList.add('9');</a>
<span class="sourceLineNo">137</span><a id="line.137">                                        break;</a>
<span class="sourceLineNo">138</span><a id="line.138">                                case '1': propertyList.add('1'); break;</a>
<span class="sourceLineNo">139</span><a id="line.139">                                case '2': propertyList.add('2'); break;</a>
<span class="sourceLineNo">140</span><a id="line.140">                                case '3': propertyList.add('3'); break;</a>
<span class="sourceLineNo">141</span><a id="line.141">                                case '4': propertyList.add('4'); break;</a>
<span class="sourceLineNo">142</span><a id="line.142">                                case '5': propertyList.add('5'); break;</a>
<span class="sourceLineNo">143</span><a id="line.143">                                case '6': propertyList.add('6'); break;</a>
<span class="sourceLineNo">144</span><a id="line.144">                                case '7': propertyList.add('7'); break;</a>
<span class="sourceLineNo">145</span><a id="line.145">                                case '8': propertyList.add('8'); break;</a>
<span class="sourceLineNo">146</span><a id="line.146">                                case '9': propertyList.add('9'); break;</a>
<span class="sourceLineNo">147</span><a id="line.147">                                case '0':</a>
<span class="sourceLineNo">148</span><a id="line.148">                                        propertyList.add('0');</a>
<span class="sourceLineNo">149</span><a id="line.149">                                        i++;</a>
<span class="sourceLineNo">150</span><a id="line.150">                                        if(args[i].length() != 1) throw new Error("Invalid value: " + args[i] + ". Amino Acid Symbol should be of single character");</a>
<span class="sourceLineNo">151</span><a id="line.151">                                        specificList.add(args[i].toUpperCase().charAt(0));</a>
<span class="sourceLineNo">152</span><a id="line.152">                                        break;</a>
<span class="sourceLineNo">153</span><a id="line.153">                                default:</a>
<span class="sourceLineNo">154</span><a id="line.154">                                        showHelp();</a>
<span class="sourceLineNo">155</span><a id="line.155">                                        throw new Error("Unknown option: " + args[i]);</a>
<span class="sourceLineNo">156</span><a id="line.156">                                }</a>
<span class="sourceLineNo">157</span><a id="line.157">                        }</a>
<span class="sourceLineNo">158</span><a id="line.158">                }</a>
<span class="sourceLineNo">159</span><a id="line.159"></a>
<span class="sourceLineNo">160</span><a id="line.160">                /*</a>
<span class="sourceLineNo">161</span><a id="line.161">                 * Check for validity of input arguments</a>
<span class="sourceLineNo">162</span><a id="line.162">                 */</a>
<span class="sourceLineNo">163</span><a id="line.163">                AminoAcidCompositionTable aaTable = checkForValidityAndObtainAATable(inputLocation, propertyList.size(), aminoAcidCompositionLocation,</a>
<span class="sourceLineNo">164</span><a id="line.164">                                elementMassLocation);</a>
<span class="sourceLineNo">165</span><a id="line.165"></a>
<span class="sourceLineNo">166</span><a id="line.166">                /*</a>
<span class="sourceLineNo">167</span><a id="line.167">                 * Read input file and generate output</a>
<span class="sourceLineNo">168</span><a id="line.168">                 */</a>
<span class="sourceLineNo">169</span><a id="line.169">                readInputAndGenerateOutput(outputLocation, propertyList, specificList, delimiter, inputLocation, aaTable, decimalPlace);</a>
<span class="sourceLineNo">170</span><a id="line.170">        }</a>
<span class="sourceLineNo">171</span><a id="line.171"></a>
<span class="sourceLineNo">172</span><a id="line.172">        private static LinkedHashMap&lt;String, ProteinSequence&gt; readInputFile(String inputLocation, AminoAcidCompositionTable aaTable) throws Exception{</a>
<span class="sourceLineNo">173</span><a id="line.173">                FileInputStream inStream = new FileInputStream(inputLocation);</a>
<span class="sourceLineNo">174</span><a id="line.174">                CompoundSet&lt;AminoAcidCompound&gt;  set;</a>
<span class="sourceLineNo">175</span><a id="line.175">                if(aaTable == null){</a>
<span class="sourceLineNo">176</span><a id="line.176">                        set = CaseFreeAminoAcidCompoundSet.getAminoAcidCompoundSet();</a>
<span class="sourceLineNo">177</span><a id="line.177">                }else{</a>
<span class="sourceLineNo">178</span><a id="line.178">                        set = aaTable.getAminoAcidCompoundSet();</a>
<span class="sourceLineNo">179</span><a id="line.179">                }</a>
<span class="sourceLineNo">180</span><a id="line.180">                LinkedHashMap&lt;String, ProteinSequence&gt; ret;</a>
<span class="sourceLineNo">181</span><a id="line.181">                if ( inputLocation.toLowerCase().contains(".gb")) {</a>
<span class="sourceLineNo">182</span><a id="line.182">                        GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt; genbankReader = new GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt;(</a>
<span class="sourceLineNo">183</span><a id="line.183">                                        inStream, new GenericGenbankHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),</a>
<span class="sourceLineNo">184</span><a id="line.184">                                        new ProteinSequenceCreator(set));</a>
<span class="sourceLineNo">185</span><a id="line.185">                        ret = genbankReader.process();</a>
<span class="sourceLineNo">186</span><a id="line.186"></a>
<span class="sourceLineNo">187</span><a id="line.187"></a>
<span class="sourceLineNo">188</span><a id="line.188">                } else {</a>
<span class="sourceLineNo">189</span><a id="line.189">                        FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; fastaReader = new FastaReader&lt;ProteinSequence, AminoAcidCompound&gt;(</a>
<span class="sourceLineNo">190</span><a id="line.190">                                        inStream, new GenericFastaHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),</a>
<span class="sourceLineNo">191</span><a id="line.191">                                        new ProteinSequenceCreator(set));</a>
<span class="sourceLineNo">192</span><a id="line.192">                        ret = fastaReader.process();</a>
<span class="sourceLineNo">193</span><a id="line.193"></a>
<span class="sourceLineNo">194</span><a id="line.194">                }</a>
<span class="sourceLineNo">195</span><a id="line.195">                return ret;</a>
<span class="sourceLineNo">196</span><a id="line.196">        }</a>
<span class="sourceLineNo">197</span><a id="line.197"></a>
<span class="sourceLineNo">198</span><a id="line.198">        public enum PropertyName{MolecularWeight, Absorbance_True, Absorbance_False, ExtinctionCoefficient_True, ExtinctionCoefficient_False,</a>
<span class="sourceLineNo">199</span><a id="line.199">                InstabilityIndex, ApliphaticIndex, AverageHydropathyValue, IsoelectricPoint, NetCharge_pH_7, A, R,</a>
<span class="sourceLineNo">200</span><a id="line.200">                N, D, C, E, Q, G, H, I, L,</a>
<span class="sourceLineNo">201</span><a id="line.201">                K, M, F, P, S, T, W, Y, V};</a>
<span class="sourceLineNo">202</span><a id="line.202"></a>
<span class="sourceLineNo">203</span><a id="line.203">        private static void printHeader(PrintStream output, List&lt;Character&gt; propertyList, List&lt;Character&gt; specificList, String delimiter) throws IOException{</a>
<span class="sourceLineNo">204</span><a id="line.204">                int specificCount = 0;</a>
<span class="sourceLineNo">205</span><a id="line.205">                /*</a>
<span class="sourceLineNo">206</span><a id="line.206">                 * 1 Molecular weight</a>
<span class="sourceLineNo">207</span><a id="line.207">                 * 2 Absorbance (assumed Cys reduced and assume Cys to form cystines)</a>
<span class="sourceLineNo">208</span><a id="line.208">                 * 3 Extinction coefficient (assumed Cys reduced and assume Cys to form cystines)</a>
<span class="sourceLineNo">209</span><a id="line.209">                 * 4 Instability index</a>
<span class="sourceLineNo">210</span><a id="line.210">                 * 5 Apliphatic index</a>
<span class="sourceLineNo">211</span><a id="line.211">                 * 6 Average hydropathy value</a>
<span class="sourceLineNo">212</span><a id="line.212">                 * 7 Isoelectric point</a>
<span class="sourceLineNo">213</span><a id="line.213">                 * 8 Net charge at pH 7</a>
<span class="sourceLineNo">214</span><a id="line.214">                 * 9 Composition of the 20 standard amino acid</a>
<span class="sourceLineNo">215</span><a id="line.215">                 * 0 Composition of the specific amino acid</a>
<span class="sourceLineNo">216</span><a id="line.216">                 */</a>
<span class="sourceLineNo">217</span><a id="line.217">                List&lt;String&gt; sList = new ArrayList&lt;String&gt;();</a>
<span class="sourceLineNo">218</span><a id="line.218">                sList.add("SequenceName");</a>
<span class="sourceLineNo">219</span><a id="line.219">                for(Character c:propertyList){</a>
<span class="sourceLineNo">220</span><a id="line.220">                        switch(c){</a>
<span class="sourceLineNo">221</span><a id="line.221">                        case '1': sList.add(PropertyName.MolecularWeight.toString()); break;</a>
<span class="sourceLineNo">222</span><a id="line.222">                        case '2': sList.add(PropertyName.Absorbance_True.toString()); sList.add(PropertyName.Absorbance_False.toString()); break;</a>
<span class="sourceLineNo">223</span><a id="line.223">                        case '3': sList.add(PropertyName.ExtinctionCoefficient_True.toString()); sList.add(PropertyName.ExtinctionCoefficient_False.toString()); break;</a>
<span class="sourceLineNo">224</span><a id="line.224">                        case '4': sList.add(PropertyName.InstabilityIndex.toString()); break;</a>
<span class="sourceLineNo">225</span><a id="line.225">                        case '5': sList.add(PropertyName.ApliphaticIndex.toString()); break;</a>
<span class="sourceLineNo">226</span><a id="line.226">                        case '6': sList.add(PropertyName.AverageHydropathyValue.toString()); break;</a>
<span class="sourceLineNo">227</span><a id="line.227">                        case '7': sList.add(PropertyName.IsoelectricPoint.toString()); break;</a>
<span class="sourceLineNo">228</span><a id="line.228">                        case '8': sList.add(PropertyName.NetCharge_pH_7.toString()); break;</a>
<span class="sourceLineNo">229</span><a id="line.229">                        case '9':</a>
<span class="sourceLineNo">230</span><a id="line.230">                                sList.add(PropertyName.A.toString()); sList.add(PropertyName.R.toString());</a>
<span class="sourceLineNo">231</span><a id="line.231">                                sList.add(PropertyName.N.toString()); sList.add(PropertyName.D.toString());</a>
<span class="sourceLineNo">232</span><a id="line.232">                                sList.add(PropertyName.C.toString()); sList.add(PropertyName.E.toString());</a>
<span class="sourceLineNo">233</span><a id="line.233">                                sList.add(PropertyName.Q.toString()); sList.add(PropertyName.G.toString());</a>
<span class="sourceLineNo">234</span><a id="line.234">                                sList.add(PropertyName.H.toString()); sList.add(PropertyName.I.toString());</a>
<span class="sourceLineNo">235</span><a id="line.235">                                sList.add(PropertyName.L.toString()); sList.add(PropertyName.K.toString());</a>
<span class="sourceLineNo">236</span><a id="line.236">                                sList.add(PropertyName.M.toString()); sList.add(PropertyName.F.toString());</a>
<span class="sourceLineNo">237</span><a id="line.237">                                sList.add(PropertyName.P.toString()); sList.add(PropertyName.S.toString());</a>
<span class="sourceLineNo">238</span><a id="line.238">                                sList.add(PropertyName.T.toString()); sList.add(PropertyName.W.toString());</a>
<span class="sourceLineNo">239</span><a id="line.239">                                sList.add(PropertyName.Y.toString()); sList.add(PropertyName.V.toString());</a>
<span class="sourceLineNo">240</span><a id="line.240">                                break;</a>
<span class="sourceLineNo">241</span><a id="line.241">                        case '0': sList.add("" + specificList.get(specificCount++)); break;</a>
<span class="sourceLineNo">242</span><a id="line.242">                        }</a>
<span class="sourceLineNo">243</span><a id="line.243">                }</a>
<span class="sourceLineNo">244</span><a id="line.244">                for(int i = 0; i &lt; sList.size(); i++){</a>
<span class="sourceLineNo">245</span><a id="line.245">                        if(i != 0) output.print(delimiter);</a>
<span class="sourceLineNo">246</span><a id="line.246">                        output.print(sList.get(i));</a>
<span class="sourceLineNo">247</span><a id="line.247">                }</a>
<span class="sourceLineNo">248</span><a id="line.248">                output.println();</a>
<span class="sourceLineNo">249</span><a id="line.249">                output.flush();</a>
<span class="sourceLineNo">250</span><a id="line.250">        }</a>
<span class="sourceLineNo">251</span><a id="line.251"></a>
<span class="sourceLineNo">252</span><a id="line.252">        private static void compute(PrintStream output, String header, String sequence, String delimiter,</a>
<span class="sourceLineNo">253</span><a id="line.253">                        AminoAcidCompositionTable aaTable, List&lt;Character&gt; propertyList, List&lt;Character&gt; specificList, int decimalPlace) throws CompoundNotFoundException{</a>
<span class="sourceLineNo">254</span><a id="line.254">                /*</a>
<span class="sourceLineNo">255</span><a id="line.255">                 * 1 Molecular weight</a>
<span class="sourceLineNo">256</span><a id="line.256">                 * 2 Absorbance (assumed Cys reduced and assume Cys to form cystines)</a>
<span class="sourceLineNo">257</span><a id="line.257">                 * 3 Extinction coefficient</a>
<span class="sourceLineNo">258</span><a id="line.258">                 * 4 Instability index</a>
<span class="sourceLineNo">259</span><a id="line.259">                 * 5 Apliphatic index</a>
<span class="sourceLineNo">260</span><a id="line.260">                 * 6 Average hydropathy value</a>
<span class="sourceLineNo">261</span><a id="line.261">                 * 7 Isoelectric point</a>
<span class="sourceLineNo">262</span><a id="line.262">                 * 8 Net charge at pH 7</a>
<span class="sourceLineNo">263</span><a id="line.263">                 * 9 Composition of the 20 standard amino acid</a>
<span class="sourceLineNo">264</span><a id="line.264">                 * 0 Composition of the specific amino acid</a>
<span class="sourceLineNo">265</span><a id="line.265">                 */</a>
<span class="sourceLineNo">266</span><a id="line.266">                ProteinSequence pSequence;</a>
<span class="sourceLineNo">267</span><a id="line.267">                CompoundSet&lt;AminoAcidCompound&gt; aaSet;</a>
<span class="sourceLineNo">268</span><a id="line.268">                if(aaTable != null){</a>
<span class="sourceLineNo">269</span><a id="line.269">                        sequence = Utils.checkSequence(sequence, aaTable.getSymbolSet());</a>
<span class="sourceLineNo">270</span><a id="line.270">                        pSequence = new ProteinSequence(sequence, aaTable.getAminoAcidCompoundSet());</a>
<span class="sourceLineNo">271</span><a id="line.271">                        aaSet = aaTable.getAminoAcidCompoundSet();</a>
<span class="sourceLineNo">272</span><a id="line.272">                }else{</a>
<span class="sourceLineNo">273</span><a id="line.273">                        sequence = Utils.checkSequence(sequence);</a>
<span class="sourceLineNo">274</span><a id="line.274">                        pSequence = new ProteinSequence(sequence);</a>
<span class="sourceLineNo">275</span><a id="line.275">                        aaSet = AminoAcidCompoundSet.getAminoAcidCompoundSet();</a>
<span class="sourceLineNo">276</span><a id="line.276">                }</a>
<span class="sourceLineNo">277</span><a id="line.277">                IPeptideProperties pp = new PeptidePropertiesImpl();</a>
<span class="sourceLineNo">278</span><a id="line.278"></a>
<span class="sourceLineNo">279</span><a id="line.279">                int specificCount = 0;</a>
<span class="sourceLineNo">280</span><a id="line.280">                List&lt;Double&gt; dList = new ArrayList&lt;Double&gt;();</a>
<span class="sourceLineNo">281</span><a id="line.281">                for(Character c:propertyList){</a>
<span class="sourceLineNo">282</span><a id="line.282">                        switch(c){</a>
<span class="sourceLineNo">283</span><a id="line.283">                        case '1':</a>
<span class="sourceLineNo">284</span><a id="line.284">                                if(aaTable == null)</a>
<span class="sourceLineNo">285</span><a id="line.285">                                        dList.add(pp.getMolecularWeight(pSequence));</a>
<span class="sourceLineNo">286</span><a id="line.286">                                else</a>
<span class="sourceLineNo">287</span><a id="line.287">                                        dList.add(pp.getMolecularWeight(pSequence));</a>
<span class="sourceLineNo">288</span><a id="line.288">                                break;</a>
<span class="sourceLineNo">289</span><a id="line.289">                        case '2':</a>
<span class="sourceLineNo">290</span><a id="line.290">                                dList.add(pp.getAbsorbance(pSequence, true));</a>
<span class="sourceLineNo">291</span><a id="line.291">                                dList.add(pp.getAbsorbance(pSequence, false));</a>
<span class="sourceLineNo">292</span><a id="line.292">                                break;</a>
<span class="sourceLineNo">293</span><a id="line.293">                        case '3':</a>
<span class="sourceLineNo">294</span><a id="line.294">                                dList.add(pp.getExtinctionCoefficient(pSequence, true));</a>
<span class="sourceLineNo">295</span><a id="line.295">                                dList.add(pp.getExtinctionCoefficient(pSequence, false));</a>
<span class="sourceLineNo">296</span><a id="line.296">                                break;</a>
<span class="sourceLineNo">297</span><a id="line.297">                        case '4': dList.add(pp.getInstabilityIndex(pSequence)); break;</a>
<span class="sourceLineNo">298</span><a id="line.298">                        case '5': dList.add(pp.getApliphaticIndex(pSequence)); break;</a>
<span class="sourceLineNo">299</span><a id="line.299">                        case '6': dList.add(pp.getAvgHydropathy(pSequence)); break;</a>
<span class="sourceLineNo">300</span><a id="line.300">                        case '7': dList.add(pp.getIsoelectricPoint(pSequence)); break;</a>
<span class="sourceLineNo">301</span><a id="line.301">                        case '8': dList.add(pp.getNetCharge(pSequence)); break;</a>
<span class="sourceLineNo">302</span><a id="line.302">                        case '9':</a>
<span class="sourceLineNo">303</span><a id="line.303">                                Map&lt;AminoAcidCompound, Double&gt; aaCompound2Double = pp.getAAComposition(pSequence);</a>
<span class="sourceLineNo">304</span><a id="line.304">                                //(A, R, N, D, C, E, Q, G, H, I, L, K, M, F, P, S, T, W, Y, V)</a>
<span class="sourceLineNo">305</span><a id="line.305">                                dList.add(aaCompound2Double.get(Constraints.A));</a>
<span class="sourceLineNo">306</span><a id="line.306">                                dList.add(aaCompound2Double.get(Constraints.R));</a>
<span class="sourceLineNo">307</span><a id="line.307">                                dList.add(aaCompound2Double.get(Constraints.N));</a>
<span class="sourceLineNo">308</span><a id="line.308">                                dList.add(aaCompound2Double.get(Constraints.D));</a>
<span class="sourceLineNo">309</span><a id="line.309">                                dList.add(aaCompound2Double.get(Constraints.C));</a>
<span class="sourceLineNo">310</span><a id="line.310">                                dList.add(aaCompound2Double.get(Constraints.E));</a>
<span class="sourceLineNo">311</span><a id="line.311">                                dList.add(aaCompound2Double.get(Constraints.Q));</a>
<span class="sourceLineNo">312</span><a id="line.312">                                dList.add(aaCompound2Double.get(Constraints.G));</a>
<span class="sourceLineNo">313</span><a id="line.313">                                dList.add(aaCompound2Double.get(Constraints.H));</a>
<span class="sourceLineNo">314</span><a id="line.314">                                dList.add(aaCompound2Double.get(Constraints.I));</a>
<span class="sourceLineNo">315</span><a id="line.315">                                dList.add(aaCompound2Double.get(Constraints.L));</a>
<span class="sourceLineNo">316</span><a id="line.316">                                dList.add(aaCompound2Double.get(Constraints.K));</a>
<span class="sourceLineNo">317</span><a id="line.317">                                dList.add(aaCompound2Double.get(Constraints.M));</a>
<span class="sourceLineNo">318</span><a id="line.318">                                dList.add(aaCompound2Double.get(Constraints.F));</a>
<span class="sourceLineNo">319</span><a id="line.319">                                dList.add(aaCompound2Double.get(Constraints.P));</a>
<span class="sourceLineNo">320</span><a id="line.320">                                dList.add(aaCompound2Double.get(Constraints.S));</a>
<span class="sourceLineNo">321</span><a id="line.321">                                dList.add(aaCompound2Double.get(Constraints.T));</a>
<span class="sourceLineNo">322</span><a id="line.322">                                dList.add(aaCompound2Double.get(Constraints.W));</a>
<span class="sourceLineNo">323</span><a id="line.323">                                dList.add(aaCompound2Double.get(Constraints.Y));</a>
<span class="sourceLineNo">324</span><a id="line.324">                                dList.add(aaCompound2Double.get(Constraints.V));</a>
<span class="sourceLineNo">325</span><a id="line.325">                                break;</a>
<span class="sourceLineNo">326</span><a id="line.326">                        case '0': dList.add(pp.getEnrichment(pSequence, aaSet.getCompoundForString("" + specificList.get(specificCount++)))); break;</a>
<span class="sourceLineNo">327</span><a id="line.327">                        }</a>
<span class="sourceLineNo">328</span><a id="line.328">                }</a>
<span class="sourceLineNo">329</span><a id="line.329">                output.print(header.replace(delimiter, "_"));</a>
<span class="sourceLineNo">330</span><a id="line.330">                for(int i = 0; i &lt; dList.size(); i++){</a>
<span class="sourceLineNo">331</span><a id="line.331">                        output.print(delimiter + Utils.roundToDecimals(dList.get(i), decimalPlace));</a>
<span class="sourceLineNo">332</span><a id="line.332">                }</a>
<span class="sourceLineNo">333</span><a id="line.333">                output.println();</a>
<span class="sourceLineNo">334</span><a id="line.334">                output.flush();</a>
<span class="sourceLineNo">335</span><a id="line.335">        }</a>
<span class="sourceLineNo">336</span><a id="line.336"></a>
<span class="sourceLineNo">337</span><a id="line.337">        private static void showHelp(){</a>
<span class="sourceLineNo">338</span><a id="line.338">                System.err.println("NAME");</a>
<span class="sourceLineNo">339</span><a id="line.339">                System.err.println("\tAn executable to generate physico-chemical properties of protein sequences.");</a>
<span class="sourceLineNo">340</span><a id="line.340">                System.err.println();</a>
<span class="sourceLineNo">341</span><a id="line.341"></a>
<span class="sourceLineNo">342</span><a id="line.342">                System.err.println("EXAMPLES");</a>
<span class="sourceLineNo">343</span><a id="line.343">                System.err.println("\tjava -jar AAProperties.jar -i test.fasta -a");</a>
<span class="sourceLineNo">344</span><a id="line.344">                System.err.println("\t\tGenerates all possible properties.");</a>
<span class="sourceLineNo">345</span><a id="line.345">                System.err.println();</a>
<span class="sourceLineNo">346</span><a id="line.346">                System.err.println("\tjava -jar AAProperties.jar -i test.fasta -1 -3 -7");</a>
<span class="sourceLineNo">347</span><a id="line.347">                System.err.println("\t\tGenerates only molecular weight, extinction coefficient and isoelectric point.");</a>
<span class="sourceLineNo">348</span><a id="line.348">                System.err.println();</a>
<span class="sourceLineNo">349</span><a id="line.349">                System.err.println("\tjava -jar AAProperties.jar -i test.fasta -0 A -0 N -1");</a>
<span class="sourceLineNo">350</span><a id="line.350">                System.err.println("\t\tGenerates composition of two specific amino acid symbol and molecular weight.");</a>
<span class="sourceLineNo">351</span><a id="line.351">                System.err.println();</a>
<span class="sourceLineNo">352</span><a id="line.352"></a>
<span class="sourceLineNo">353</span><a id="line.353">                System.err.println("OPTIONS");</a>
<span class="sourceLineNo">354</span><a id="line.354">                System.err.println("\tRequired");</a>
<span class="sourceLineNo">355</span><a id="line.355">                System.err.println("\t\t-i location of input FASTA file");</a>
<span class="sourceLineNo">356</span><a id="line.356">                System.err.println();</a>
<span class="sourceLineNo">357</span><a id="line.357"></a>
<span class="sourceLineNo">358</span><a id="line.358">                System.err.println("\tOptional");</a>
<span class="sourceLineNo">359</span><a id="line.359">                System.err.println("\t\t-o location of output file [standard output (default)]");</a>
<span class="sourceLineNo">360</span><a id="line.360">                System.err.println("\t\t-f output format [csv (default) or tsv]");</a>
<span class="sourceLineNo">361</span><a id="line.361">                System.err.println("\t\t-x location of Amino Acid Composition XML file for defining amino acid composition");</a>
<span class="sourceLineNo">362</span><a id="line.362">                System.err.println("\t\t-y location of Element Mass XML file for defining mass of elements");</a>
<span class="sourceLineNo">363</span><a id="line.363">                System.err.println("\t\t-d number of decimals (int) [4 (default)]");</a>
<span class="sourceLineNo">364</span><a id="line.364">                System.err.println();</a>
<span class="sourceLineNo">365</span><a id="line.365"></a>
<span class="sourceLineNo">366</span><a id="line.366">                System.err.println("\tProvide at least one of them");</a>
<span class="sourceLineNo">367</span><a id="line.367">                System.err.println("\t\t-a compute properties of option 1-9");</a>
<span class="sourceLineNo">368</span><a id="line.368">                System.err.println("\t\t-1 compute molecular weight");</a>
<span class="sourceLineNo">369</span><a id="line.369">                System.err.println("\t\t-2 compute absorbance");</a>
<span class="sourceLineNo">370</span><a id="line.370">                System.err.println("\t\t-3 compute extinction coefficient");</a>
<span class="sourceLineNo">371</span><a id="line.371">                System.err.println("\t\t-4 compute instability index");</a>
<span class="sourceLineNo">372</span><a id="line.372">                System.err.println("\t\t-5 compute apliphatic index");</a>
<span class="sourceLineNo">373</span><a id="line.373">                System.err.println("\t\t-6 compute average hydropathy value");</a>
<span class="sourceLineNo">374</span><a id="line.374">                System.err.println("\t\t-7 compute isoelectric point");</a>
<span class="sourceLineNo">375</span><a id="line.375">                System.err.println("\t\t-8 compute net charge at pH 7");</a>
<span class="sourceLineNo">376</span><a id="line.376">                System.err.println("\t\t-9 compute composition of 20 standard amino acid (A, R, N, D, C, E, Q, G, H, I, L, K, M, F, P, S, T, W, Y, V)");</a>
<span class="sourceLineNo">377</span><a id="line.377">                System.err.println("\t\t-0 compute composition of specific amino acid symbol");</a>
<span class="sourceLineNo">378</span><a id="line.378">                System.err.println();</a>
<span class="sourceLineNo">379</span><a id="line.379">        }</a>
<span class="sourceLineNo">380</span><a id="line.380">}</a>




























































</pre>
</div>
</main>
</body>
</html>
