<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> */</a>
<span class="sourceLineNo">021</span><a id="line.21">package org.biojava.nbio.structure.io;</a>
<span class="sourceLineNo">022</span><a id="line.22"></a>
<span class="sourceLineNo">023</span><a id="line.23">import org.biojava.nbio.alignment.Alignments;</a>
<span class="sourceLineNo">024</span><a id="line.24">import org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType;</a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.nbio.alignment.SimpleGapPenalty;</a>
<span class="sourceLineNo">026</span><a id="line.26">import org.biojava.nbio.core.alignment.matrices.SubstitutionMatrixHelper;</a>
<span class="sourceLineNo">027</span><a id="line.27">import org.biojava.nbio.alignment.template.GapPenalty;</a>
<span class="sourceLineNo">028</span><a id="line.28">import org.biojava.nbio.alignment.template.PairwiseSequenceAligner;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.biojava.nbio.core.alignment.template.SequencePair;</a>
<span class="sourceLineNo">030</span><a id="line.30">import org.biojava.nbio.core.alignment.template.SubstitutionMatrix;</a>
<span class="sourceLineNo">031</span><a id="line.31">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</a>
<span class="sourceLineNo">032</span><a id="line.32">import org.biojava.nbio.core.sequence.DNASequence;</a>
<span class="sourceLineNo">033</span><a id="line.33">import org.biojava.nbio.core.sequence.ProteinSequence;</a>
<span class="sourceLineNo">034</span><a id="line.34">import org.biojava.nbio.core.sequence.RNASequence;</a>
<span class="sourceLineNo">035</span><a id="line.35">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</a>
<span class="sourceLineNo">036</span><a id="line.36">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</a>
<span class="sourceLineNo">037</span><a id="line.37">import org.biojava.nbio.structure.*;</a>
<span class="sourceLineNo">038</span><a id="line.38">import org.slf4j.Logger;</a>
<span class="sourceLineNo">039</span><a id="line.39">import org.slf4j.LoggerFactory;</a>
<span class="sourceLineNo">040</span><a id="line.40"></a>
<span class="sourceLineNo">041</span><a id="line.41">import java.util.ArrayList;</a>
<span class="sourceLineNo">042</span><a id="line.42">import java.util.Collections;</a>
<span class="sourceLineNo">043</span><a id="line.43">import java.util.Comparator;</a>
<span class="sourceLineNo">044</span><a id="line.44">import java.util.HashMap;</a>
<span class="sourceLineNo">045</span><a id="line.45">import java.util.List;</a>
<span class="sourceLineNo">046</span><a id="line.46">import java.util.Set;</a>
<span class="sourceLineNo">047</span><a id="line.47">import java.util.TreeMap;</a>
<span class="sourceLineNo">048</span><a id="line.48">import java.util.TreeSet;</a>
<span class="sourceLineNo">049</span><a id="line.49"></a>
<span class="sourceLineNo">050</span><a id="line.50">/**</a>
<span class="sourceLineNo">051</span><a id="line.51"> * Heuristical finding of Entities (called Compounds in legacy PDB format)</a>
<span class="sourceLineNo">052</span><a id="line.52"> * in a given Structure. Entities are the groups of sequence identical NCS-related polymer chains</a>
<span class="sourceLineNo">053</span><a id="line.53"> * in the Structure.</a>
<span class="sourceLineNo">054</span><a id="line.54"> *</a>
<span class="sourceLineNo">055</span><a id="line.55"> * This is related to {@link SeqRes2AtomAligner} but it is intended for raw PDB/mmCIF files where</a>
<span class="sourceLineNo">056</span><a id="line.56"> * possibly no SEQRES is given.</a>
<span class="sourceLineNo">057</span><a id="line.57"> *</a>
<span class="sourceLineNo">058</span><a id="line.58"> * @author Jose Duarte</a>
<span class="sourceLineNo">059</span><a id="line.59"> */</a>
<span class="sourceLineNo">060</span><a id="line.60">public class EntityFinder {</a>
<span class="sourceLineNo">061</span><a id="line.61"></a>
<span class="sourceLineNo">062</span><a id="line.62">        private static final Logger logger = LoggerFactory.getLogger(EntityFinder.class);</a>
<span class="sourceLineNo">063</span><a id="line.63"></a>
<span class="sourceLineNo">064</span><a id="line.64">        /**</a>
<span class="sourceLineNo">065</span><a id="line.65">         * Above this ratio of mismatching residue types for same residue numbers we</a>
<span class="sourceLineNo">066</span><a id="line.66">         * consider the 2 chains to have mismatching residue numbers and warn about it</a>
<span class="sourceLineNo">067</span><a id="line.67">         */</a>
<span class="sourceLineNo">068</span><a id="line.68">        public static final double RATIO_GAPS_FOR_MISMATCH = 0.50;</a>
<span class="sourceLineNo">069</span><a id="line.69"></a>
<span class="sourceLineNo">070</span><a id="line.70">        /**</a>
<span class="sourceLineNo">071</span><a id="line.71">         * Identity value for 2 chains to be considered part of same entity</a>
<span class="sourceLineNo">072</span><a id="line.72">         */</a>
<span class="sourceLineNo">073</span><a id="line.73">        public static final double IDENTITY_THRESHOLD = 0.99999;</a>
<span class="sourceLineNo">074</span><a id="line.74"></a>
<span class="sourceLineNo">075</span><a id="line.75">        /**</a>
<span class="sourceLineNo">076</span><a id="line.76">         * Gap coverage value (num gaps over length of sequence) for each chain of the match:</a>
<span class="sourceLineNo">077</span><a id="line.77">         * 2 chains with more gap coverage than this value will not be considered part of the same entity</a>
<span class="sourceLineNo">078</span><a id="line.78">         */</a>
<span class="sourceLineNo">079</span><a id="line.79">        public static final double GAP_COVERAGE_THRESHOLD = 0.3;</a>
<span class="sourceLineNo">080</span><a id="line.80"></a>
<span class="sourceLineNo">081</span><a id="line.81"></a>
<span class="sourceLineNo">082</span><a id="line.82"></a>
<span class="sourceLineNo">083</span><a id="line.83"></a>
<span class="sourceLineNo">084</span><a id="line.84">        /**</a>
<span class="sourceLineNo">085</span><a id="line.85">         * Utility method that employs some heuristics to find the {@link EntityInfo}s</a>
<span class="sourceLineNo">086</span><a id="line.86">         * for the polymeric chains given in constructor.</a>
<span class="sourceLineNo">087</span><a id="line.87">         * To be used in case the information is missing in PDB/mmCIF file</a>
<span class="sourceLineNo">088</span><a id="line.88">         * @return</a>
<span class="sourceLineNo">089</span><a id="line.89">         */</a>
<span class="sourceLineNo">090</span><a id="line.90">        public static List&lt;EntityInfo&gt; findPolyEntities(List&lt;List&lt;Chain&gt;&gt; polyModels) {</a>
<span class="sourceLineNo">091</span><a id="line.91">                TreeMap&lt;String,EntityInfo&gt; chainIds2entities = findEntitiesFromAlignment(polyModels);</a>
<span class="sourceLineNo">092</span><a id="line.92"></a>
<span class="sourceLineNo">093</span><a id="line.93">                return findUniqueEntities(chainIds2entities);</a>
<span class="sourceLineNo">094</span><a id="line.94">        }</a>
<span class="sourceLineNo">095</span><a id="line.95"></a>
<span class="sourceLineNo">096</span><a id="line.96">        /**</a>
<span class="sourceLineNo">097</span><a id="line.97">         * Utility method to obtain a list of unique entities from the chainIds2entities map</a>
<span class="sourceLineNo">098</span><a id="line.98">         * @return</a>
<span class="sourceLineNo">099</span><a id="line.99">         */</a>
<span class="sourceLineNo">100</span><a id="line.100">        private static List&lt;EntityInfo&gt; findUniqueEntities(TreeMap&lt;String,EntityInfo&gt; chainIds2entities) {</a>
<span class="sourceLineNo">101</span><a id="line.101"></a>
<span class="sourceLineNo">102</span><a id="line.102">                List&lt;EntityInfo&gt; list = new ArrayList&lt;&gt;();</a>
<span class="sourceLineNo">103</span><a id="line.103"></a>
<span class="sourceLineNo">104</span><a id="line.104">                for (EntityInfo cluster:chainIds2entities.values()) {</a>
<span class="sourceLineNo">105</span><a id="line.105">                        boolean present = false;</a>
<span class="sourceLineNo">106</span><a id="line.106">                        for (EntityInfo cl:list) {</a>
<span class="sourceLineNo">107</span><a id="line.107">                                if (cl==cluster) {</a>
<span class="sourceLineNo">108</span><a id="line.108">                                        present = true;</a>
<span class="sourceLineNo">109</span><a id="line.109">                                        break;</a>
<span class="sourceLineNo">110</span><a id="line.110">                                }</a>
<span class="sourceLineNo">111</span><a id="line.111">                        }</a>
<span class="sourceLineNo">112</span><a id="line.112">                        if (!present) list.add(cluster);</a>
<span class="sourceLineNo">113</span><a id="line.113">                }</a>
<span class="sourceLineNo">114</span><a id="line.114">                return list;</a>
<span class="sourceLineNo">115</span><a id="line.115">        }</a>
<span class="sourceLineNo">116</span><a id="line.116"></a>
<span class="sourceLineNo">117</span><a id="line.117">        /**</a>
<span class="sourceLineNo">118</span><a id="line.118">         * Given all chains of all models find entities for the nonpolymers and water chains within them,</a>
<span class="sourceLineNo">119</span><a id="line.119">         * assigning entity ids, types and descriptions to them. The result is written back to the passed entities List.</a>
<span class="sourceLineNo">120</span><a id="line.120">         * @param nonPolyModels</a>
<span class="sourceLineNo">121</span><a id="line.121">         * @param waterModels</a>
<span class="sourceLineNo">122</span><a id="line.122">         * @param entities</a>
<span class="sourceLineNo">123</span><a id="line.123">         */</a>
<span class="sourceLineNo">124</span><a id="line.124">        public static void createPurelyNonPolyEntities(List&lt;List&lt;Chain&gt;&gt; nonPolyModels, List&lt;List&lt;Chain&gt;&gt; waterModels, List&lt;EntityInfo&gt; entities) {</a>
<span class="sourceLineNo">125</span><a id="line.125"></a>
<span class="sourceLineNo">126</span><a id="line.126">                if (nonPolyModels.isEmpty()) return;</a>
<span class="sourceLineNo">127</span><a id="line.127"></a>
<span class="sourceLineNo">128</span><a id="line.128">                // let's find first the max entity id to assign entity ids to the newly found entities</a>
<span class="sourceLineNo">129</span><a id="line.129">                int maxMolId = 0;</a>
<span class="sourceLineNo">130</span><a id="line.130">                if (!entities.isEmpty()) {</a>
<span class="sourceLineNo">131</span><a id="line.131">                        maxMolId = Collections.max(entities, Comparator.comparingInt(EntityInfo::getMolId)).getMolId();</a>
<span class="sourceLineNo">132</span><a id="line.132">                }</a>
<span class="sourceLineNo">133</span><a id="line.133">                // we go one over the max</a>
<span class="sourceLineNo">134</span><a id="line.134">                int molId = maxMolId + 1;</a>
<span class="sourceLineNo">135</span><a id="line.135"></a>
<span class="sourceLineNo">136</span><a id="line.136">                if (!nonPolyModels.get(0).isEmpty()) {</a>
<span class="sourceLineNo">137</span><a id="line.137">                        List&lt;EntityInfo&gt; nonPolyEntities = new ArrayList&lt;&gt;();</a>
<span class="sourceLineNo">138</span><a id="line.138"></a>
<span class="sourceLineNo">139</span><a id="line.139">                        for (List&lt;Chain&gt; model:nonPolyModels) {</a>
<span class="sourceLineNo">140</span><a id="line.140">                                for (Chain c: model) {</a>
<span class="sourceLineNo">141</span><a id="line.141">                                        // we assume there's only 1 group per non-poly chain</a>
<span class="sourceLineNo">142</span><a id="line.142">                                        String molecPdbName = c.getAtomGroup(0).getPDBName();</a>
<span class="sourceLineNo">143</span><a id="line.143"></a>
<span class="sourceLineNo">144</span><a id="line.144">                                        EntityInfo nonPolyEntity = findNonPolyEntityWithDescription(molecPdbName, nonPolyEntities);</a>
<span class="sourceLineNo">145</span><a id="line.145">                                        if (nonPolyEntity == null) {</a>
<span class="sourceLineNo">146</span><a id="line.146">                                                nonPolyEntity = new EntityInfo();</a>
<span class="sourceLineNo">147</span><a id="line.147">                                                nonPolyEntity.setDescription(molecPdbName);</a>
<span class="sourceLineNo">148</span><a id="line.148">                                                nonPolyEntity.setType(EntityType.NONPOLYMER);</a>
<span class="sourceLineNo">149</span><a id="line.149">                                                nonPolyEntity.setMolId(molId++);</a>
<span class="sourceLineNo">150</span><a id="line.150">                                                nonPolyEntities.add(nonPolyEntity);</a>
<span class="sourceLineNo">151</span><a id="line.151"></a>
<span class="sourceLineNo">152</span><a id="line.152">                                        }</a>
<span class="sourceLineNo">153</span><a id="line.153">                                        nonPolyEntity.addChain(c);</a>
<span class="sourceLineNo">154</span><a id="line.154">                                        c.setEntityInfo(nonPolyEntity);</a>
<span class="sourceLineNo">155</span><a id="line.155">                                }</a>
<span class="sourceLineNo">156</span><a id="line.156">                        }</a>
<span class="sourceLineNo">157</span><a id="line.157">                        entities.addAll(nonPolyEntities);</a>
<span class="sourceLineNo">158</span><a id="line.158">                }</a>
<span class="sourceLineNo">159</span><a id="line.159"></a>
<span class="sourceLineNo">160</span><a id="line.160">                if (!waterModels.get(0).isEmpty()) {</a>
<span class="sourceLineNo">161</span><a id="line.161">                        EntityInfo waterEntity = new EntityInfo();</a>
<span class="sourceLineNo">162</span><a id="line.162">                        waterEntity.setType(EntityType.WATER);</a>
<span class="sourceLineNo">163</span><a id="line.163">                        waterEntity.setDescription("water");</a>
<span class="sourceLineNo">164</span><a id="line.164">                        waterEntity.setMolId(molId);</a>
<span class="sourceLineNo">165</span><a id="line.165"></a>
<span class="sourceLineNo">166</span><a id="line.166">                        for (List&lt;Chain&gt; model:waterModels) {</a>
<span class="sourceLineNo">167</span><a id="line.167">                                for (Chain waterChain:model) {</a>
<span class="sourceLineNo">168</span><a id="line.168">                                        waterEntity.addChain(waterChain);</a>
<span class="sourceLineNo">169</span><a id="line.169">                                        waterChain.setEntityInfo(waterEntity);</a>
<span class="sourceLineNo">170</span><a id="line.170">                                }</a>
<span class="sourceLineNo">171</span><a id="line.171">                        }</a>
<span class="sourceLineNo">172</span><a id="line.172">                        entities.add(waterEntity);</a>
<span class="sourceLineNo">173</span><a id="line.173"></a>
<span class="sourceLineNo">174</span><a id="line.174">                }</a>
<span class="sourceLineNo">175</span><a id="line.175"></a>
<span class="sourceLineNo">176</span><a id="line.176">        }</a>
<span class="sourceLineNo">177</span><a id="line.177"></a>
<span class="sourceLineNo">178</span><a id="line.178">        private static EntityInfo findNonPolyEntityWithDescription(String description, List&lt;EntityInfo&gt; nonPolyEntities) {</a>
<span class="sourceLineNo">179</span><a id="line.179">                for (EntityInfo e:nonPolyEntities) {</a>
<span class="sourceLineNo">180</span><a id="line.180">                        if (e.getDescription().equals(description)) return e;</a>
<span class="sourceLineNo">181</span><a id="line.181">                }</a>
<span class="sourceLineNo">182</span><a id="line.182">                return null;</a>
<span class="sourceLineNo">183</span><a id="line.183">        }</a>
<span class="sourceLineNo">184</span><a id="line.184"></a>
<span class="sourceLineNo">185</span><a id="line.185"></a>
<span class="sourceLineNo">186</span><a id="line.186">        private static boolean areResNumbersAligned(Chain c1, Chain c2) {</a>
<span class="sourceLineNo">187</span><a id="line.187"></a>
<span class="sourceLineNo">188</span><a id="line.188">                boolean isC1prot = c1.isProtein();</a>
<span class="sourceLineNo">189</span><a id="line.189">                boolean isC2prot = c2.isProtein();</a>
<span class="sourceLineNo">190</span><a id="line.190"></a>
<span class="sourceLineNo">191</span><a id="line.191">                // different kind of chain: we won't try to align them</a>
<span class="sourceLineNo">192</span><a id="line.192">                if (isC1prot != isC2prot ) return false;</a>
<span class="sourceLineNo">193</span><a id="line.193"></a>
<span class="sourceLineNo">194</span><a id="line.194">                List&lt;Group&gt; c1AtomGroups = null;</a>
<span class="sourceLineNo">195</span><a id="line.195">                if (isC1prot) {</a>
<span class="sourceLineNo">196</span><a id="line.196">                        c1AtomGroups = c1.getAtomGroups(GroupType.AMINOACID);</a>
<span class="sourceLineNo">197</span><a id="line.197">                }</a>
<span class="sourceLineNo">198</span><a id="line.198">                else {</a>
<span class="sourceLineNo">199</span><a id="line.199">                        c1AtomGroups = c1.getAtomGroups(GroupType.NUCLEOTIDE);</a>
<span class="sourceLineNo">200</span><a id="line.200">                }</a>
<span class="sourceLineNo">201</span><a id="line.201"></a>
<span class="sourceLineNo">202</span><a id="line.202">                int countNonExisting = 0;</a>
<span class="sourceLineNo">203</span><a id="line.203"></a>
<span class="sourceLineNo">204</span><a id="line.204">                for (Group g1:c1AtomGroups) {</a>
<span class="sourceLineNo">205</span><a id="line.205">                        try {</a>
<span class="sourceLineNo">206</span><a id="line.206">                                Group g2 = c2.getGroupByPDB(g1.getResidueNumber());</a>
<span class="sourceLineNo">207</span><a id="line.207">                                if (!g2.getPDBName().equals(g1.getPDBName())) {</a>
<span class="sourceLineNo">208</span><a id="line.208">                                        logger.debug("Mismatch of residues between chains {},{} for residue number {}: {} {}",</a>
<span class="sourceLineNo">209</span><a id="line.209">                                                        c1.getId(),c2.getId(),g1.getResidueNumber(), g1.getPDBName(), g2.getPDBName());</a>
<span class="sourceLineNo">210</span><a id="line.210">                                        return false;</a>
<span class="sourceLineNo">211</span><a id="line.211">                                }</a>
<span class="sourceLineNo">212</span><a id="line.212">                        } catch (StructureException e) {</a>
<span class="sourceLineNo">213</span><a id="line.213">                                // the group doesn't exist (no density) in the chain, go on</a>
<span class="sourceLineNo">214</span><a id="line.214">                                countNonExisting++;</a>
<span class="sourceLineNo">215</span><a id="line.215">                        }</a>
<span class="sourceLineNo">216</span><a id="line.216">                }</a>
<span class="sourceLineNo">217</span><a id="line.217"></a>
<span class="sourceLineNo">218</span><a id="line.218">                if ((double)countNonExisting/(double)c1AtomGroups.size() &gt; RATIO_GAPS_FOR_MISMATCH) {</a>
<span class="sourceLineNo">219</span><a id="line.219">                        logger.debug("More than {} of the residues ({} out of {}) are not present in chain {} when comparing by residue numbers to chain {}.",</a>
<span class="sourceLineNo">220</span><a id="line.220">                                        RATIO_GAPS_FOR_MISMATCH, countNonExisting, c1AtomGroups.size(), c2.getId(), c1.getId());</a>
<span class="sourceLineNo">221</span><a id="line.221">                        return false;</a>
<span class="sourceLineNo">222</span><a id="line.222">                }</a>
<span class="sourceLineNo">223</span><a id="line.223"></a>
<span class="sourceLineNo">224</span><a id="line.224">                return true;</a>
<span class="sourceLineNo">225</span><a id="line.225">        }</a>
<span class="sourceLineNo">226</span><a id="line.226"></a>
<span class="sourceLineNo">227</span><a id="line.227"></a>
<span class="sourceLineNo">228</span><a id="line.228"></a>
<span class="sourceLineNo">229</span><a id="line.229">        private static TreeMap&lt;String,EntityInfo&gt; findEntitiesFromAlignment(List&lt;List&lt;Chain&gt;&gt; polyModels) {</a>
<span class="sourceLineNo">230</span><a id="line.230"></a>
<span class="sourceLineNo">231</span><a id="line.231">                TreeMap&lt;String, EntityInfo&gt; chainIds2entities = new TreeMap&lt;&gt;();</a>
<span class="sourceLineNo">232</span><a id="line.232"></a>
<span class="sourceLineNo">233</span><a id="line.233">                if (polyModels.isEmpty()) return chainIds2entities;</a>
<span class="sourceLineNo">234</span><a id="line.234"></a>
<span class="sourceLineNo">235</span><a id="line.235">                Set&lt;Integer&gt; polyChainIndices = new TreeSet&lt;&gt;();</a>
<span class="sourceLineNo">236</span><a id="line.236">                for (int i=0;i&lt;polyModels.get(0).size();i++) {</a>
<span class="sourceLineNo">237</span><a id="line.237">                        polyChainIndices.add(i);</a>
<span class="sourceLineNo">238</span><a id="line.238">                }</a>
<span class="sourceLineNo">239</span><a id="line.239"></a>
<span class="sourceLineNo">240</span><a id="line.240">                int molId = 1;</a>
<span class="sourceLineNo">241</span><a id="line.241"></a>
<span class="sourceLineNo">242</span><a id="line.242">                outer:</a>
<span class="sourceLineNo">243</span><a id="line.243">                        for (int i:polyChainIndices) {</a>
<span class="sourceLineNo">244</span><a id="line.244">                                Chain c1 = polyModels.get(0).get(i);</a>
<span class="sourceLineNo">245</span><a id="line.245">                                // here we use false, which means that X will be used for unknown compounds</a>
<span class="sourceLineNo">246</span><a id="line.246">                                String str1 = SeqRes2AtomAligner.getFullAtomSequence(c1.getAtomGroups(), new HashMap&lt;&gt;(), false);</a>
<span class="sourceLineNo">247</span><a id="line.247"></a>
<span class="sourceLineNo">248</span><a id="line.248">                                for (int j:polyChainIndices) {</a>
<span class="sourceLineNo">249</span><a id="line.249"></a>
<span class="sourceLineNo">250</span><a id="line.250">                                        if (j&lt;=i) continue;</a>
<span class="sourceLineNo">251</span><a id="line.251"></a>
<span class="sourceLineNo">252</span><a id="line.252">                                        Chain c2 = polyModels.get(0).get(j);</a>
<span class="sourceLineNo">253</span><a id="line.253">                                        String str2 = SeqRes2AtomAligner.getFullAtomSequence(c2.getAtomGroups(), new HashMap&lt;&gt;(), false);</a>
<span class="sourceLineNo">254</span><a id="line.254"></a>
<span class="sourceLineNo">255</span><a id="line.255">                                        int seq1Length = 0;</a>
<span class="sourceLineNo">256</span><a id="line.256">                                        int seq2Length = 0;</a>
<span class="sourceLineNo">257</span><a id="line.257"></a>
<span class="sourceLineNo">258</span><a id="line.258">                                        SequencePair&lt;?,?&gt; pair;</a>
<span class="sourceLineNo">259</span><a id="line.259">                                        if (isProteinSequence(str1) &amp;&amp; isProteinSequence(str2)) {</a>
<span class="sourceLineNo">260</span><a id="line.260">                                                ProteinSequence s1 = getProteinSequence(str1);</a>
<span class="sourceLineNo">261</span><a id="line.261">                                                ProteinSequence s2 = getProteinSequence(str2);</a>
<span class="sourceLineNo">262</span><a id="line.262">                                                seq1Length = s1.getLength();</a>
<span class="sourceLineNo">263</span><a id="line.263">                                                seq2Length = s2.getLength();</a>
<span class="sourceLineNo">264</span><a id="line.264"></a>
<span class="sourceLineNo">265</span><a id="line.265">                                                pair = alignProtein(s1,s2);</a>
<span class="sourceLineNo">266</span><a id="line.266"></a>
<span class="sourceLineNo">267</span><a id="line.267">                                        } else if (isDNASequence(str1) &amp;&amp; isDNASequence(str2)) {</a>
<span class="sourceLineNo">268</span><a id="line.268">                                                DNASequence s1 = getDNASequence(str1);</a>
<span class="sourceLineNo">269</span><a id="line.269">                                                DNASequence s2 = getDNASequence(str2);</a>
<span class="sourceLineNo">270</span><a id="line.270">                                                seq1Length = s1.getLength();</a>
<span class="sourceLineNo">271</span><a id="line.271">                                                seq2Length = s2.getLength();</a>
<span class="sourceLineNo">272</span><a id="line.272"></a>
<span class="sourceLineNo">273</span><a id="line.273">                                                pair = alignDNA(s1,s2);</a>
<span class="sourceLineNo">274</span><a id="line.274"></a>
<span class="sourceLineNo">275</span><a id="line.275">                                        } else if (isRNASequence(str1) &amp;&amp; isRNASequence(str2)) {</a>
<span class="sourceLineNo">276</span><a id="line.276">                                                RNASequence s1 = getRNASequence(str1);</a>
<span class="sourceLineNo">277</span><a id="line.277">                                                RNASequence s2 = getRNASequence(str2);</a>
<span class="sourceLineNo">278</span><a id="line.278">                                                seq1Length = s1.getLength();</a>
<span class="sourceLineNo">279</span><a id="line.279">                                                seq2Length = s2.getLength();</a>
<span class="sourceLineNo">280</span><a id="line.280"></a>
<span class="sourceLineNo">281</span><a id="line.281">                                                pair = alignRNA(s1,s2);</a>
<span class="sourceLineNo">282</span><a id="line.282"></a>
<span class="sourceLineNo">283</span><a id="line.283">                                        } else {</a>
<span class="sourceLineNo">284</span><a id="line.284">                                                logger.debug("Chains {},{} are either different kind of polymers or could not be recognized as protein or nucleotide polymers", c1.getId(), c2.getId());</a>
<span class="sourceLineNo">285</span><a id="line.285">                                                continue;</a>
<span class="sourceLineNo">286</span><a id="line.286">                                        }</a>
<span class="sourceLineNo">287</span><a id="line.287"></a>
<span class="sourceLineNo">288</span><a id="line.288">                                        int numGaps = getNumGaps(pair);</a>
<span class="sourceLineNo">289</span><a id="line.289">                                        int numGaps1 = getNumGapsQuery(pair);</a>
<span class="sourceLineNo">290</span><a id="line.290">                                        int numGaps2 = getNumGapsTarget(pair);</a>
<span class="sourceLineNo">291</span><a id="line.291"></a>
<span class="sourceLineNo">292</span><a id="line.292">                                        int nonGaps = pair.getLength() - numGaps;</a>
<span class="sourceLineNo">293</span><a id="line.293"></a>
<span class="sourceLineNo">294</span><a id="line.294">                                        double identity = (double)pair.getNumIdenticals()/(double)nonGaps;</a>
<span class="sourceLineNo">295</span><a id="line.295">                                        double gapCov1 = (double) numGaps1 / (double) seq1Length;</a>
<span class="sourceLineNo">296</span><a id="line.296">                                        double gapCov2 = (double) numGaps2 / (double) seq2Length;</a>
<span class="sourceLineNo">297</span><a id="line.297"></a>
<span class="sourceLineNo">298</span><a id="line.298">                                        logger.debug("Alignment for chain pair {},{}: identity: {}, gap coverage 1: {}, gap coverage 2: {}",</a>
<span class="sourceLineNo">299</span><a id="line.299">                                                        c1.getId(), c2.getId(), String.format("%4.2f",identity), String.format("%4.2f",gapCov1), String.format("%4.2f",gapCov2));</a>
<span class="sourceLineNo">300</span><a id="line.300">                                        logger.debug("\n"+pair.toString(100));</a>
<span class="sourceLineNo">301</span><a id="line.301"></a>
<span class="sourceLineNo">302</span><a id="line.302">                                        if (identity &gt; IDENTITY_THRESHOLD &amp;&amp; gapCov1&lt;GAP_COVERAGE_THRESHOLD &amp;&amp; gapCov2&lt;GAP_COVERAGE_THRESHOLD) {</a>
<span class="sourceLineNo">303</span><a id="line.303">                                                if (    !chainIds2entities.containsKey(c1.getId()) &amp;&amp;</a>
<span class="sourceLineNo">304</span><a id="line.304">                                                                !chainIds2entities.containsKey(c2.getId())) {</a>
<span class="sourceLineNo">305</span><a id="line.305">                                                        logger.debug("Creating entity with chains {},{}",c1.getId(),c2.getId());</a>
<span class="sourceLineNo">306</span><a id="line.306"></a>
<span class="sourceLineNo">307</span><a id="line.307">                                                        EntityInfo ent = new EntityInfo();</a>
<span class="sourceLineNo">308</span><a id="line.308">                                                        ent.addChain(c1);</a>
<span class="sourceLineNo">309</span><a id="line.309">                                                        ent.addChain(c2);</a>
<span class="sourceLineNo">310</span><a id="line.310">                                                        ent.setMolId(molId++);</a>
<span class="sourceLineNo">311</span><a id="line.311">                                                        ent.setType(EntityType.POLYMER);</a>
<span class="sourceLineNo">312</span><a id="line.312">                                                        c1.setEntityInfo(ent);</a>
<span class="sourceLineNo">313</span><a id="line.313">                                                        c2.setEntityInfo(ent);</a>
<span class="sourceLineNo">314</span><a id="line.314">                                                        chainIds2entities.put(c1.getId(), ent);</a>
<span class="sourceLineNo">315</span><a id="line.315">                                                        chainIds2entities.put(c2.getId(), ent);</a>
<span class="sourceLineNo">316</span><a id="line.316"></a>
<span class="sourceLineNo">317</span><a id="line.317">                                                } else {</a>
<span class="sourceLineNo">318</span><a id="line.318">                                                        Chain chainToAdd;</a>
<span class="sourceLineNo">319</span><a id="line.319">                                                        EntityInfo ent = chainIds2entities.get(c1.getId());</a>
<span class="sourceLineNo">320</span><a id="line.320">                                                        if (ent==null) {</a>
<span class="sourceLineNo">321</span><a id="line.321">                                                                ent = chainIds2entities.get(c2.getId());</a>
<span class="sourceLineNo">322</span><a id="line.322">                                                                chainToAdd = c1;</a>
<span class="sourceLineNo">323</span><a id="line.323">                                                        } else {</a>
<span class="sourceLineNo">324</span><a id="line.324">                                                                chainToAdd = c2;</a>
<span class="sourceLineNo">325</span><a id="line.325">                                                        }</a>
<span class="sourceLineNo">326</span><a id="line.326">                                                        if (!chainIds2entities.containsKey(chainToAdd.getId())) {</a>
<span class="sourceLineNo">327</span><a id="line.327">                                                                logger.debug("Adding chain {} to entity {}", chainToAdd.getId(), ent.getId());</a>
<span class="sourceLineNo">328</span><a id="line.328">                                                                ent.addChain(chainToAdd);</a>
<span class="sourceLineNo">329</span><a id="line.329">                                                                chainToAdd.setEntityInfo(ent);</a>
<span class="sourceLineNo">330</span><a id="line.330">                                                                chainIds2entities.put(chainToAdd.getId(), ent);</a>
<span class="sourceLineNo">331</span><a id="line.331">                                                        }</a>
<span class="sourceLineNo">332</span><a id="line.332">                                                }</a>
<span class="sourceLineNo">333</span><a id="line.333">                                                if (!areResNumbersAligned(c1, c2)) {</a>
<span class="sourceLineNo">334</span><a id="line.334">                                                        logger.warn("Including 100% identical chains {},{} in same entity, although they have misaligned residue numbers",</a>
<span class="sourceLineNo">335</span><a id="line.335">                                                                        c1.getId(),c2.getId());</a>
<span class="sourceLineNo">336</span><a id="line.336">                                                }</a>
<span class="sourceLineNo">337</span><a id="line.337">                                        }</a>
<span class="sourceLineNo">338</span><a id="line.338"></a>
<span class="sourceLineNo">339</span><a id="line.339">                                        if (identity&gt;1) {</a>
<span class="sourceLineNo">340</span><a id="line.340">                                                logger.warn("Identity for chains {},{} above 1. {} identicals out of {} non-gap-aligned residues (identity {})",</a>
<span class="sourceLineNo">341</span><a id="line.341">                                                                c1.getId(),c2.getId(),pair.getNumIdenticals(),nonGaps,identity);</a>
<span class="sourceLineNo">342</span><a id="line.342">                                                logger.warn("\n"+pair.toString(100));</a>
<span class="sourceLineNo">343</span><a id="line.343">                                        }</a>
<span class="sourceLineNo">344</span><a id="line.344"></a>
<span class="sourceLineNo">345</span><a id="line.345">                                        if (chainIds2entities.size()==polyChainIndices.size()) // we've got all chains in entities</a>
<span class="sourceLineNo">346</span><a id="line.346">                                                break outer;</a>
<span class="sourceLineNo">347</span><a id="line.347">                                }</a>
<span class="sourceLineNo">348</span><a id="line.348">                        }</a>
<span class="sourceLineNo">349</span><a id="line.349"></a>
<span class="sourceLineNo">350</span><a id="line.350">                // anything not in an Entity will be its own Entity</a>
<span class="sourceLineNo">351</span><a id="line.351">                for (int i:polyChainIndices) {</a>
<span class="sourceLineNo">352</span><a id="line.352">                        Chain c = polyModels.get(0).get(i);</a>
<span class="sourceLineNo">353</span><a id="line.353">                        if (!chainIds2entities.containsKey(c.getId())) {</a>
<span class="sourceLineNo">354</span><a id="line.354">                                logger.debug("Creating a 1-member entity for chain {}",c.getId());</a>
<span class="sourceLineNo">355</span><a id="line.355"></a>
<span class="sourceLineNo">356</span><a id="line.356">                                EntityInfo ent = new EntityInfo();</a>
<span class="sourceLineNo">357</span><a id="line.357">                                ent.addChain(c);</a>
<span class="sourceLineNo">358</span><a id="line.358">                                ent.setMolId(molId++);</a>
<span class="sourceLineNo">359</span><a id="line.359">                                ent.setType(EntityType.POLYMER);</a>
<span class="sourceLineNo">360</span><a id="line.360">                                c.setEntityInfo(ent);</a>
<span class="sourceLineNo">361</span><a id="line.361"></a>
<span class="sourceLineNo">362</span><a id="line.362">                                chainIds2entities.put(c.getId(),ent);</a>
<span class="sourceLineNo">363</span><a id="line.363">                        }</a>
<span class="sourceLineNo">364</span><a id="line.364">                }</a>
<span class="sourceLineNo">365</span><a id="line.365"></a>
<span class="sourceLineNo">366</span><a id="line.366">                // now that we've found the entities for first model, let's do the same for the rest of the models</a>
<span class="sourceLineNo">367</span><a id="line.367"></a>
<span class="sourceLineNo">368</span><a id="line.368">                for (int i=1;i&lt;polyModels.size();i++) {</a>
<span class="sourceLineNo">369</span><a id="line.369">                        for (Chain chain : polyModels.get(i)) {</a>
<span class="sourceLineNo">370</span><a id="line.370">                                EntityInfo e = chainIds2entities.get(chain.getId());</a>
<span class="sourceLineNo">371</span><a id="line.371">                                chain.setEntityInfo(e);</a>
<span class="sourceLineNo">372</span><a id="line.372">                                e.addChain(chain);</a>
<span class="sourceLineNo">373</span><a id="line.373">                        }</a>
<span class="sourceLineNo">374</span><a id="line.374">                }</a>
<span class="sourceLineNo">375</span><a id="line.375"></a>
<span class="sourceLineNo">376</span><a id="line.376">                return chainIds2entities;</a>
<span class="sourceLineNo">377</span><a id="line.377">        }</a>
<span class="sourceLineNo">378</span><a id="line.378"></a>
<span class="sourceLineNo">379</span><a id="line.379">        private static SequencePair&lt;ProteinSequence, AminoAcidCompound&gt; alignProtein(ProteinSequence s1, ProteinSequence s2) {</a>
<span class="sourceLineNo">380</span><a id="line.380">                SubstitutionMatrix&lt;AminoAcidCompound&gt; matrix = SubstitutionMatrixHelper.getIdentity();</a>
<span class="sourceLineNo">381</span><a id="line.381"></a>
<span class="sourceLineNo">382</span><a id="line.382">                GapPenalty penalty = new SimpleGapPenalty(8, 1);</a>
<span class="sourceLineNo">383</span><a id="line.383"></a>
<span class="sourceLineNo">384</span><a id="line.384">                PairwiseSequenceAligner&lt;ProteinSequence, AminoAcidCompound&gt; nw =</a>
<span class="sourceLineNo">385</span><a id="line.385">                                Alignments.getPairwiseAligner(s1, s2, PairwiseSequenceAlignerType.GLOBAL, penalty, matrix);</a>
<span class="sourceLineNo">386</span><a id="line.386"></a>
<span class="sourceLineNo">387</span><a id="line.387">                return nw.getPair();</a>
<span class="sourceLineNo">388</span><a id="line.388">        }</a>
<span class="sourceLineNo">389</span><a id="line.389"></a>
<span class="sourceLineNo">390</span><a id="line.390">        private static SequencePair&lt;DNASequence, NucleotideCompound&gt; alignDNA(DNASequence s1, DNASequence s2) {</a>
<span class="sourceLineNo">391</span><a id="line.391">                SubstitutionMatrix&lt;NucleotideCompound&gt; matrix = SubstitutionMatrixHelper.getNuc4_4();</a>
<span class="sourceLineNo">392</span><a id="line.392"></a>
<span class="sourceLineNo">393</span><a id="line.393">                GapPenalty penalty = new SimpleGapPenalty(8, 1);</a>
<span class="sourceLineNo">394</span><a id="line.394"></a>
<span class="sourceLineNo">395</span><a id="line.395">                PairwiseSequenceAligner&lt;DNASequence, NucleotideCompound&gt; nw =</a>
<span class="sourceLineNo">396</span><a id="line.396">                                Alignments.getPairwiseAligner(s1, s2, PairwiseSequenceAlignerType.GLOBAL, penalty, matrix);</a>
<span class="sourceLineNo">397</span><a id="line.397"></a>
<span class="sourceLineNo">398</span><a id="line.398">                return nw.getPair();</a>
<span class="sourceLineNo">399</span><a id="line.399">        }</a>
<span class="sourceLineNo">400</span><a id="line.400"></a>
<span class="sourceLineNo">401</span><a id="line.401">        private static SequencePair&lt;RNASequence, NucleotideCompound&gt; alignRNA(RNASequence s1, RNASequence s2) {</a>
<span class="sourceLineNo">402</span><a id="line.402">                SubstitutionMatrix&lt;NucleotideCompound&gt; matrix = SubstitutionMatrixHelper.getNuc4_4();</a>
<span class="sourceLineNo">403</span><a id="line.403"></a>
<span class="sourceLineNo">404</span><a id="line.404">                GapPenalty penalty = new SimpleGapPenalty(8, 1);</a>
<span class="sourceLineNo">405</span><a id="line.405"></a>
<span class="sourceLineNo">406</span><a id="line.406">                PairwiseSequenceAligner&lt;RNASequence, NucleotideCompound&gt; nw =</a>
<span class="sourceLineNo">407</span><a id="line.407">                                Alignments.getPairwiseAligner(s1, s2, PairwiseSequenceAlignerType.GLOBAL, penalty, matrix);</a>
<span class="sourceLineNo">408</span><a id="line.408"></a>
<span class="sourceLineNo">409</span><a id="line.409">                return nw.getPair();</a>
<span class="sourceLineNo">410</span><a id="line.410">        }</a>
<span class="sourceLineNo">411</span><a id="line.411"></a>
<span class="sourceLineNo">412</span><a id="line.412"></a>
<span class="sourceLineNo">413</span><a id="line.413"></a>
<span class="sourceLineNo">414</span><a id="line.414">        private static int getNumGaps(SequencePair&lt;?, ?&gt; pair) {</a>
<span class="sourceLineNo">415</span><a id="line.415">                int numGaps = 0;</a>
<span class="sourceLineNo">416</span><a id="line.416">                for (int alignmentIndex=1;alignmentIndex&lt;=pair.getLength();alignmentIndex++) {</a>
<span class="sourceLineNo">417</span><a id="line.417">                        if (pair.hasGap(alignmentIndex)) numGaps++;</a>
<span class="sourceLineNo">418</span><a id="line.418">                }</a>
<span class="sourceLineNo">419</span><a id="line.419">                return numGaps;</a>
<span class="sourceLineNo">420</span><a id="line.420">        }</a>
<span class="sourceLineNo">421</span><a id="line.421"></a>
<span class="sourceLineNo">422</span><a id="line.422">        private static int getNumGapsQuery(SequencePair&lt;?, ?&gt; pair) {</a>
<span class="sourceLineNo">423</span><a id="line.423">                int numGaps = 0;</a>
<span class="sourceLineNo">424</span><a id="line.424">                for (int alignmentIndex=1;alignmentIndex&lt;=pair.getLength();alignmentIndex++) {</a>
<span class="sourceLineNo">425</span><a id="line.425">                        if (pair.getCompoundInQueryAt(alignmentIndex).getShortName().equals("-")) {</a>
<span class="sourceLineNo">426</span><a id="line.426">                                numGaps++;</a>
<span class="sourceLineNo">427</span><a id="line.427">                        }</a>
<span class="sourceLineNo">428</span><a id="line.428">                }</a>
<span class="sourceLineNo">429</span><a id="line.429">                return numGaps;</a>
<span class="sourceLineNo">430</span><a id="line.430">        }</a>
<span class="sourceLineNo">431</span><a id="line.431"></a>
<span class="sourceLineNo">432</span><a id="line.432">        private static int getNumGapsTarget(SequencePair&lt;?, ?&gt; pair) {</a>
<span class="sourceLineNo">433</span><a id="line.433">                int numGaps = 0;</a>
<span class="sourceLineNo">434</span><a id="line.434">                for (int alignmentIndex=1;alignmentIndex&lt;=pair.getLength();alignmentIndex++) {</a>
<span class="sourceLineNo">435</span><a id="line.435">                        if (pair.getCompoundInTargetAt(alignmentIndex).getShortName().equals("-")) {</a>
<span class="sourceLineNo">436</span><a id="line.436">                                numGaps++;</a>
<span class="sourceLineNo">437</span><a id="line.437">                        }</a>
<span class="sourceLineNo">438</span><a id="line.438">                }</a>
<span class="sourceLineNo">439</span><a id="line.439">                return numGaps;</a>
<span class="sourceLineNo">440</span><a id="line.440">        }</a>
<span class="sourceLineNo">441</span><a id="line.441"></a>
<span class="sourceLineNo">442</span><a id="line.442">        private static boolean isProteinSequence(String str) {</a>
<span class="sourceLineNo">443</span><a id="line.443">                try {</a>
<span class="sourceLineNo">444</span><a id="line.444">                        new ProteinSequence(str);</a>
<span class="sourceLineNo">445</span><a id="line.445">                } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">446</span><a id="line.446"></a>
<span class="sourceLineNo">447</span><a id="line.447">                        return false;</a>
<span class="sourceLineNo">448</span><a id="line.448">                }</a>
<span class="sourceLineNo">449</span><a id="line.449">                return true;</a>
<span class="sourceLineNo">450</span><a id="line.450">        }</a>
<span class="sourceLineNo">451</span><a id="line.451"></a>
<span class="sourceLineNo">452</span><a id="line.452">        private static boolean isDNASequence(String str) {</a>
<span class="sourceLineNo">453</span><a id="line.453">                try {</a>
<span class="sourceLineNo">454</span><a id="line.454">                        new DNASequence(str);</a>
<span class="sourceLineNo">455</span><a id="line.455">                } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">456</span><a id="line.456"></a>
<span class="sourceLineNo">457</span><a id="line.457">                        return false;</a>
<span class="sourceLineNo">458</span><a id="line.458">                }</a>
<span class="sourceLineNo">459</span><a id="line.459">                return true;</a>
<span class="sourceLineNo">460</span><a id="line.460">        }</a>
<span class="sourceLineNo">461</span><a id="line.461"></a>
<span class="sourceLineNo">462</span><a id="line.462">        private static boolean isRNASequence(String str) {</a>
<span class="sourceLineNo">463</span><a id="line.463">                try {</a>
<span class="sourceLineNo">464</span><a id="line.464">                        new RNASequence(str);</a>
<span class="sourceLineNo">465</span><a id="line.465">                } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">466</span><a id="line.466"></a>
<span class="sourceLineNo">467</span><a id="line.467">                        return false;</a>
<span class="sourceLineNo">468</span><a id="line.468">                }</a>
<span class="sourceLineNo">469</span><a id="line.469">                return true;</a>
<span class="sourceLineNo">470</span><a id="line.470"></a>
<span class="sourceLineNo">471</span><a id="line.471">        }</a>
<span class="sourceLineNo">472</span><a id="line.472"></a>
<span class="sourceLineNo">473</span><a id="line.473">        /**</a>
<span class="sourceLineNo">474</span><a id="line.474">         * Returns the ProteinSequence or null if one can't be created</a>
<span class="sourceLineNo">475</span><a id="line.475">         * @param str</a>
<span class="sourceLineNo">476</span><a id="line.476">         * @return</a>
<span class="sourceLineNo">477</span><a id="line.477">         */</a>
<span class="sourceLineNo">478</span><a id="line.478">        private static ProteinSequence getProteinSequence(String str) {</a>
<span class="sourceLineNo">479</span><a id="line.479">                try {</a>
<span class="sourceLineNo">480</span><a id="line.480">                        ProteinSequence s = new ProteinSequence(str);</a>
<span class="sourceLineNo">481</span><a id="line.481">                        return s;</a>
<span class="sourceLineNo">482</span><a id="line.482">                } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">483</span><a id="line.483"></a>
<span class="sourceLineNo">484</span><a id="line.484">                        logger.error("Unexpected error when creating ProteinSequence",e);</a>
<span class="sourceLineNo">485</span><a id="line.485">                }</a>
<span class="sourceLineNo">486</span><a id="line.486">                return null;</a>
<span class="sourceLineNo">487</span><a id="line.487">        }</a>
<span class="sourceLineNo">488</span><a id="line.488"></a>
<span class="sourceLineNo">489</span><a id="line.489">        /**</a>
<span class="sourceLineNo">490</span><a id="line.490">         * Returns the DNASequence or null if one can't be created</a>
<span class="sourceLineNo">491</span><a id="line.491">         * @param str</a>
<span class="sourceLineNo">492</span><a id="line.492">         * @return</a>
<span class="sourceLineNo">493</span><a id="line.493">         */</a>
<span class="sourceLineNo">494</span><a id="line.494">        private static DNASequence getDNASequence(String str) {</a>
<span class="sourceLineNo">495</span><a id="line.495">                try {</a>
<span class="sourceLineNo">496</span><a id="line.496">                        DNASequence s = new DNASequence(str);</a>
<span class="sourceLineNo">497</span><a id="line.497">                        return s;</a>
<span class="sourceLineNo">498</span><a id="line.498">                } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">499</span><a id="line.499"></a>
<span class="sourceLineNo">500</span><a id="line.500">                        logger.error("Unexpected error when creating DNASequence ",e);</a>
<span class="sourceLineNo">501</span><a id="line.501">                }</a>
<span class="sourceLineNo">502</span><a id="line.502">                return null;</a>
<span class="sourceLineNo">503</span><a id="line.503">        }</a>
<span class="sourceLineNo">504</span><a id="line.504"></a>
<span class="sourceLineNo">505</span><a id="line.505">        /**</a>
<span class="sourceLineNo">506</span><a id="line.506">         * Returns the RNASequence or null if one can't be created</a>
<span class="sourceLineNo">507</span><a id="line.507">         * @param str</a>
<span class="sourceLineNo">508</span><a id="line.508">         * @return</a>
<span class="sourceLineNo">509</span><a id="line.509">         */</a>
<span class="sourceLineNo">510</span><a id="line.510">        private static RNASequence getRNASequence(String str) {</a>
<span class="sourceLineNo">511</span><a id="line.511">                try {</a>
<span class="sourceLineNo">512</span><a id="line.512">                        RNASequence s = new RNASequence(str);</a>
<span class="sourceLineNo">513</span><a id="line.513">                        return s;</a>
<span class="sourceLineNo">514</span><a id="line.514">                } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">515</span><a id="line.515"></a>
<span class="sourceLineNo">516</span><a id="line.516">                        logger.error("Unexpected error when creating RNASequence ",e);</a>
<span class="sourceLineNo">517</span><a id="line.517">                }</a>
<span class="sourceLineNo">518</span><a id="line.518">                return null;</a>
<span class="sourceLineNo">519</span><a id="line.519">        }</a>
<span class="sourceLineNo">520</span><a id="line.520"></a>
<span class="sourceLineNo">521</span><a id="line.521">}</a>




























































</pre>
</div>
</main>
</body>
</html>
