<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/**</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> * Created on 2013-05-28</a>
<span class="sourceLineNo">021</span><a id="line.21"> * Created by Douglas Myers-Turnbull</a>
<span class="sourceLineNo">022</span><a id="line.22"> *</a>
<span class="sourceLineNo">023</span><a id="line.23"> * @since 3.0.6</a>
<span class="sourceLineNo">024</span><a id="line.24"> */</a>
<span class="sourceLineNo">025</span><a id="line.25">package org.biojava.nbio.structure.io;</a>
<span class="sourceLineNo">026</span><a id="line.26"></a>
<span class="sourceLineNo">027</span><a id="line.27">import org.biojava.nbio.core.alignment.template.AlignedSequence;</a>
<span class="sourceLineNo">028</span><a id="line.28">import org.biojava.nbio.core.alignment.template.SequencePair;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.biojava.nbio.structure.*;</a>
<span class="sourceLineNo">030</span><a id="line.30">import org.biojava.nbio.structure.align.model.AFPChain;</a>
<span class="sourceLineNo">031</span><a id="line.31">import org.biojava.nbio.structure.align.util.AlignmentTools;</a>
<span class="sourceLineNo">032</span><a id="line.32">import org.biojava.nbio.structure.align.xml.AFPChainXMLConverter;</a>
<span class="sourceLineNo">033</span><a id="line.33">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</a>
<span class="sourceLineNo">034</span><a id="line.34">import org.biojava.nbio.core.sequence.ProteinSequence;</a>
<span class="sourceLineNo">035</span><a id="line.35">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</a>
<span class="sourceLineNo">036</span><a id="line.36">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</a>
<span class="sourceLineNo">037</span><a id="line.37">import org.biojava.nbio.core.sequence.io.CasePreservingProteinSequenceCreator;</a>
<span class="sourceLineNo">038</span><a id="line.38">import org.biojava.nbio.core.sequence.io.FastaReader;</a>
<span class="sourceLineNo">039</span><a id="line.39">import org.biojava.nbio.core.sequence.io.GenericFastaHeaderParser;</a>
<span class="sourceLineNo">040</span><a id="line.40">import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;</a>
<span class="sourceLineNo">041</span><a id="line.41">import org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface;</a>
<span class="sourceLineNo">042</span><a id="line.42">import org.biojava.nbio.core.sequence.template.Sequence;</a>
<span class="sourceLineNo">043</span><a id="line.43">import org.biojava.nbio.core.util.SequenceTools;</a>
<span class="sourceLineNo">044</span><a id="line.44">import org.slf4j.Logger;</a>
<span class="sourceLineNo">045</span><a id="line.45">import org.slf4j.LoggerFactory;</a>
<span class="sourceLineNo">046</span><a id="line.46"></a>
<span class="sourceLineNo">047</span><a id="line.47">import java.io.File;</a>
<span class="sourceLineNo">048</span><a id="line.48">import java.io.FileInputStream;</a>
<span class="sourceLineNo">049</span><a id="line.49">import java.io.IOException;</a>
<span class="sourceLineNo">050</span><a id="line.50">import java.io.InputStream;</a>
<span class="sourceLineNo">051</span><a id="line.51">import java.util.*;</a>
<span class="sourceLineNo">052</span><a id="line.52"></a>
<span class="sourceLineNo">053</span><a id="line.53">/**</a>
<span class="sourceLineNo">054</span><a id="line.54"> * A collection of static utilities to convert between {@link AFPChain AFPChains} and {@link FastaSequence FastaSequences}.</a>
<span class="sourceLineNo">055</span><a id="line.55"> *</a>
<span class="sourceLineNo">056</span><a id="line.56"> * @author dmyersturnbull</a>
<span class="sourceLineNo">057</span><a id="line.57"> * @see StructureSequenceMatcher</a>
<span class="sourceLineNo">058</span><a id="line.58"> * @see FastaStructureParser</a>
<span class="sourceLineNo">059</span><a id="line.59"> * @see SeqRes2AtomAligner</a>
<span class="sourceLineNo">060</span><a id="line.60"> */</a>
<span class="sourceLineNo">061</span><a id="line.61">public class FastaAFPChainConverter {</a>
<span class="sourceLineNo">062</span><a id="line.62"></a>
<span class="sourceLineNo">063</span><a id="line.63">        private final static Logger logger = LoggerFactory.getLogger(FastaAFPChainConverter.class);</a>
<span class="sourceLineNo">064</span><a id="line.64"></a>
<span class="sourceLineNo">065</span><a id="line.65"></a>
<span class="sourceLineNo">066</span><a id="line.66">        public static AFPChain cpFastaToAfpChain(String first, String second, Structure structure, int cpSite) throws StructureException, CompoundNotFoundException {</a>
<span class="sourceLineNo">067</span><a id="line.67">                ProteinSequence s1 = new ProteinSequence(first);</a>
<span class="sourceLineNo">068</span><a id="line.68">                s1.setUserCollection(getAlignedUserCollection(first));</a>
<span class="sourceLineNo">069</span><a id="line.69">                ProteinSequence s2 = new ProteinSequence(second);</a>
<span class="sourceLineNo">070</span><a id="line.70">                s2.setUserCollection(getAlignedUserCollection(second));</a>
<span class="sourceLineNo">071</span><a id="line.71">                return cpFastaToAfpChain(s1, s2, structure, cpSite);</a>
<span class="sourceLineNo">072</span><a id="line.72">        }</a>
<span class="sourceLineNo">073</span><a id="line.73"></a>
<span class="sourceLineNo">074</span><a id="line.74">        /**</a>
<span class="sourceLineNo">075</span><a id="line.75">         * Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site</a>
<span class="sourceLineNo">076</span><a id="line.76">         * {@link cpSite} residues to the right.</a>
<span class="sourceLineNo">077</span><a id="line.77">         *</a>
<span class="sourceLineNo">078</span><a id="line.78">         * @param fastaFile A FASTA file containing exactly 2 sequences, the first unpermuted and the second permuted</a>
<span class="sourceLineNo">079</span><a id="line.79">         * @param cpSite</a>
<span class="sourceLineNo">080</span><a id="line.80">         *            The number of residues from the beginning of the sequence at which the circular permutation site occurs; can be positive or negative; values greater than the length of the sequence</a>
<span class="sourceLineNo">081</span><a id="line.81">         *            are acceptable</a>
<span class="sourceLineNo">082</span><a id="line.82">         * @throws IOException</a>
<span class="sourceLineNo">083</span><a id="line.83">         * @throws StructureException</a>
<span class="sourceLineNo">084</span><a id="line.84">         */</a>
<span class="sourceLineNo">085</span><a id="line.85">        public static AFPChain cpFastaToAfpChain(File fastaFile, Structure structure, int cpSite) throws IOException, StructureException {</a>
<span class="sourceLineNo">086</span><a id="line.86">                InputStream inStream = new FileInputStream(fastaFile);</a>
<span class="sourceLineNo">087</span><a id="line.87">                SequenceCreatorInterface&lt;AminoAcidCompound&gt; creator = new CasePreservingProteinSequenceCreator(AminoAcidCompoundSet.getAminoAcidCompoundSet());</a>
<span class="sourceLineNo">088</span><a id="line.88">                SequenceHeaderParserInterface&lt;ProteinSequence, AminoAcidCompound&gt; headerParser = new GenericFastaHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;();</a>
<span class="sourceLineNo">089</span><a id="line.89">                FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; fastaReader = new FastaReader&lt;ProteinSequence, AminoAcidCompound&gt;(inStream, headerParser, creator);</a>
<span class="sourceLineNo">090</span><a id="line.90">                LinkedHashMap&lt;String, ProteinSequence&gt; sequences = fastaReader.process();</a>
<span class="sourceLineNo">091</span><a id="line.91">                inStream.close();</a>
<span class="sourceLineNo">092</span><a id="line.92">                Iterator&lt;ProteinSequence&gt; iter = sequences.values().iterator();</a>
<span class="sourceLineNo">093</span><a id="line.93">                ProteinSequence first = iter.next();</a>
<span class="sourceLineNo">094</span><a id="line.94">                ProteinSequence second = iter.next();</a>
<span class="sourceLineNo">095</span><a id="line.95">                return cpFastaToAfpChain(first, second, structure, cpSite);</a>
<span class="sourceLineNo">096</span><a id="line.96">        }</a>
<span class="sourceLineNo">097</span><a id="line.97"></a>
<span class="sourceLineNo">098</span><a id="line.98">        /**</a>
<span class="sourceLineNo">099</span><a id="line.99">         * Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site</a>
<span class="sourceLineNo">100</span><a id="line.100">         * {@link cpSite} residues to the right.</a>
<span class="sourceLineNo">101</span><a id="line.101">         *</a>
<span class="sourceLineNo">102</span><a id="line.102">         * @param first The unpermuted sequence</a>
<span class="sourceLineNo">103</span><a id="line.103">         * @param second The sequence permuted by cpSite</a>
<span class="sourceLineNo">104</span><a id="line.104">         * @param cpSite</a>
<span class="sourceLineNo">105</span><a id="line.105">         *            The number of residues from the beginning of the sequence at which the circular permutation site occurs; can be positive or negative; values greater than the length of the sequence</a>
<span class="sourceLineNo">106</span><a id="line.106">         *            are acceptable</a>
<span class="sourceLineNo">107</span><a id="line.107">         * @throws StructureException</a>
<span class="sourceLineNo">108</span><a id="line.108">         */</a>
<span class="sourceLineNo">109</span><a id="line.109">        public static AFPChain cpFastaToAfpChain(ProteinSequence first, ProteinSequence second, Structure structure, int cpSite)</a>
<span class="sourceLineNo">110</span><a id="line.110">                        throws StructureException {</a>
<span class="sourceLineNo">111</span><a id="line.111"></a>
<span class="sourceLineNo">112</span><a id="line.112">                if (structure == null) {</a>
<span class="sourceLineNo">113</span><a id="line.113">                        throw new IllegalArgumentException("The structure is null");</a>
<span class="sourceLineNo">114</span><a id="line.114">                }</a>
<span class="sourceLineNo">115</span><a id="line.115"></a>
<span class="sourceLineNo">116</span><a id="line.116">                if (first == null) {</a>
<span class="sourceLineNo">117</span><a id="line.117">                        throw new IllegalArgumentException("The sequence is null");</a>
<span class="sourceLineNo">118</span><a id="line.118">                }</a>
<span class="sourceLineNo">119</span><a id="line.119"></a>
<span class="sourceLineNo">120</span><a id="line.120">                // we need to find the ungapped CP site</a>
<span class="sourceLineNo">121</span><a id="line.121">                int gappedCpShift = 0;</a>
<span class="sourceLineNo">122</span><a id="line.122">                int ungappedCpShift = 0;</a>
<span class="sourceLineNo">123</span><a id="line.123">                while (ungappedCpShift &lt; Math.abs(cpSite)) {</a>
<span class="sourceLineNo">124</span><a id="line.124">                        char c;</a>
<span class="sourceLineNo">125</span><a id="line.125">                        try {</a>
<span class="sourceLineNo">126</span><a id="line.126">                                if (cpSite &lt;= 0) {</a>
<span class="sourceLineNo">127</span><a id="line.127">                                        c = second.getSequenceAsString().charAt(gappedCpShift);</a>
<span class="sourceLineNo">128</span><a id="line.128">                                } else {</a>
<span class="sourceLineNo">129</span><a id="line.129">                                        c = second.getSequenceAsString().charAt(first.getLength()-1 - gappedCpShift);</a>
<span class="sourceLineNo">130</span><a id="line.130">                                }</a>
<span class="sourceLineNo">131</span><a id="line.131">                        } catch (StringIndexOutOfBoundsException e) {</a>
<span class="sourceLineNo">132</span><a id="line.132">                                throw new IllegalArgumentException("CP site of " + cpSite + " is wrong");</a>
<span class="sourceLineNo">133</span><a id="line.133">                        }</a>
<span class="sourceLineNo">134</span><a id="line.134">                        if (c != '-') {</a>
<span class="sourceLineNo">135</span><a id="line.135">                                ungappedCpShift++;</a>
<span class="sourceLineNo">136</span><a id="line.136">                        }</a>
<span class="sourceLineNo">137</span><a id="line.137">                        gappedCpShift++;</a>
<span class="sourceLineNo">138</span><a id="line.138">                }</a>
<span class="sourceLineNo">139</span><a id="line.139"></a>
<span class="sourceLineNo">140</span><a id="line.140">                Atom[] ca1 = StructureTools.getRepresentativeAtomArray(structure);</a>
<span class="sourceLineNo">141</span><a id="line.141">                Atom[] ca2 =  StructureTools.getRepresentativeAtomArray(structure); // can't use cloneCAArray because it doesn't set parent group.chain.structure</a>
<span class="sourceLineNo">142</span><a id="line.142"></a>
<span class="sourceLineNo">143</span><a id="line.143">                ProteinSequence antipermuted = null;</a>
<span class="sourceLineNo">144</span><a id="line.144">                try {</a>
<span class="sourceLineNo">145</span><a id="line.145">                        antipermuted = new ProteinSequence(SequenceTools.permuteCyclic(second.getSequenceAsString(), gappedCpShift));</a>
<span class="sourceLineNo">146</span><a id="line.146">                } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">147</span><a id="line.147">                        // this can't happen, the original sequence comes from a ProteinSequence</a>
<span class="sourceLineNo">148</span><a id="line.148">                        logger.error("Unexpected error while creating protein sequence: {}. This is most likely a bug.",e.getMessage() );</a>
<span class="sourceLineNo">149</span><a id="line.149">                }</a>
<span class="sourceLineNo">150</span><a id="line.150"></a>
<span class="sourceLineNo">151</span><a id="line.151">                ResidueNumber[] residues = StructureSequenceMatcher.matchSequenceToStructure(first, structure);</a>
<span class="sourceLineNo">152</span><a id="line.152">                ResidueNumber[] antipermutedResidues = StructureSequenceMatcher.matchSequenceToStructure(antipermuted, structure);</a>
<span class="sourceLineNo">153</span><a id="line.153"></a>
<span class="sourceLineNo">154</span><a id="line.154">                ResidueNumber[] nonpermutedResidues = new ResidueNumber[antipermutedResidues.length];</a>
<span class="sourceLineNo">155</span><a id="line.155">                SequenceTools.permuteCyclic(antipermutedResidues, nonpermutedResidues, -gappedCpShift);</a>
<span class="sourceLineNo">156</span><a id="line.156"></a>
<span class="sourceLineNo">157</span><a id="line.157">                // nullify ResidueNumbers that have a lowercase sequence character</a>
<span class="sourceLineNo">158</span><a id="line.158">                if (first.getUserCollection() != null) {</a>
<span class="sourceLineNo">159</span><a id="line.159">                        CasePreservingProteinSequenceCreator.setLowercaseToNull(first, residues);</a>
<span class="sourceLineNo">160</span><a id="line.160">                }</a>
<span class="sourceLineNo">161</span><a id="line.161">                if (second.getUserCollection() != null) {</a>
<span class="sourceLineNo">162</span><a id="line.162">                        CasePreservingProteinSequenceCreator.setLowercaseToNull(second, nonpermutedResidues);</a>
<span class="sourceLineNo">163</span><a id="line.163">                }</a>
<span class="sourceLineNo">164</span><a id="line.164"></a>
<span class="sourceLineNo">165</span><a id="line.165">//              for (int i = 0; i &lt; residues.length; i++) {</a>
<span class="sourceLineNo">166</span><a id="line.166">//                      if (residues[i] == null) {</a>
<span class="sourceLineNo">167</span><a id="line.167">//                              System.out.print("=");</a>
<span class="sourceLineNo">168</span><a id="line.168">//                      } else {</a>
<span class="sourceLineNo">169</span><a id="line.169">//                              System.out.print(sequence.getSequenceAsString().charAt(i));</a>
<span class="sourceLineNo">170</span><a id="line.170">//                      }</a>
<span class="sourceLineNo">171</span><a id="line.171">//              }</a>
<span class="sourceLineNo">172</span><a id="line.172">//              System.out.println();</a>
<span class="sourceLineNo">173</span><a id="line.173">//              for (int i = 0; i &lt; residues.length; i++) {</a>
<span class="sourceLineNo">174</span><a id="line.174">//                      if (nonpermutedResidues[i] == null) {</a>
<span class="sourceLineNo">175</span><a id="line.175">//                              System.out.print("=");</a>
<span class="sourceLineNo">176</span><a id="line.176">//                      } else {</a>
<span class="sourceLineNo">177</span><a id="line.177">//                              System.out.print(second.getSequenceAsString().charAt(i));</a>
<span class="sourceLineNo">178</span><a id="line.178">//                      }</a>
<span class="sourceLineNo">179</span><a id="line.179">//              }</a>
<span class="sourceLineNo">180</span><a id="line.180">//              System.out.println();</a>
<span class="sourceLineNo">181</span><a id="line.181"></a>
<span class="sourceLineNo">182</span><a id="line.182">                return buildAlignment(ca1, ca2, residues, nonpermutedResidues);</a>
<span class="sourceLineNo">183</span><a id="line.183"></a>
<span class="sourceLineNo">184</span><a id="line.184">        }</a>
<span class="sourceLineNo">185</span><a id="line.185"></a>
<span class="sourceLineNo">186</span><a id="line.186">        /**</a>
<span class="sourceLineNo">187</span><a id="line.187">         * Reads the file {@code fastaFile}, expecting exactly two sequences which give a pairwise alignment. Uses this and two structures to create an AFPChain corresponding to the alignment. Uses a</a>
<span class="sourceLineNo">188</span><a id="line.188">         * {@link CasePreservingProteinSequenceCreator} and assumes that a residue is aligned if and only if it is given by an uppercase letter.</a>
<span class="sourceLineNo">189</span><a id="line.189">         *</a>
<span class="sourceLineNo">190</span><a id="line.190">         * @see #fastaToAfpChain(ProteinSequence, ProteinSequence, Structure, Structure)</a>
<span class="sourceLineNo">191</span><a id="line.191">         * @throws IOException</a>
<span class="sourceLineNo">192</span><a id="line.192">         * @throws StructureException</a>
<span class="sourceLineNo">193</span><a id="line.193">         */</a>
<span class="sourceLineNo">194</span><a id="line.194">        public static AFPChain fastaFileToAfpChain(File fastaFile, Structure structure1, Structure structure2)</a>
<span class="sourceLineNo">195</span><a id="line.195">                        throws IOException, StructureException {</a>
<span class="sourceLineNo">196</span><a id="line.196">                InputStream inStream = new FileInputStream(fastaFile);</a>
<span class="sourceLineNo">197</span><a id="line.197">                SequenceCreatorInterface&lt;AminoAcidCompound&gt; creator = new CasePreservingProteinSequenceCreator(</a>
<span class="sourceLineNo">198</span><a id="line.198">                                AminoAcidCompoundSet.getAminoAcidCompoundSet());</a>
<span class="sourceLineNo">199</span><a id="line.199">                SequenceHeaderParserInterface&lt;ProteinSequence, AminoAcidCompound&gt; headerParser = new GenericFastaHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;();</a>
<span class="sourceLineNo">200</span><a id="line.200">                FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; fastaReader = new FastaReader&lt;ProteinSequence, AminoAcidCompound&gt;(</a>
<span class="sourceLineNo">201</span><a id="line.201">                                inStream, headerParser, creator);</a>
<span class="sourceLineNo">202</span><a id="line.202">                LinkedHashMap&lt;String, ProteinSequence&gt; sequences = fastaReader.process();</a>
<span class="sourceLineNo">203</span><a id="line.203">                inStream.close();</a>
<span class="sourceLineNo">204</span><a id="line.204">                return fastaToAfpChain(sequences, structure1, structure2);</a>
<span class="sourceLineNo">205</span><a id="line.205">        }</a>
<span class="sourceLineNo">206</span><a id="line.206"></a>
<span class="sourceLineNo">207</span><a id="line.207">        /**</a>
<span class="sourceLineNo">208</span><a id="line.208">         * Returns an AFPChain corresponding to the alignment between {@code structure1} and {@code structure2}, which is given by the gapped protein sequences {@code sequence1} and {@code sequence2}. The</a>
<span class="sourceLineNo">209</span><a id="line.209">         * sequences need not correspond to the entire structures, since local alignment is performed to match the sequences to structures.</a>
<span class="sourceLineNo">210</span><a id="line.210">         * @throws StructureException</a>
<span class="sourceLineNo">211</span><a id="line.211">         * @throws CompoundNotFoundException</a>
<span class="sourceLineNo">212</span><a id="line.212">         */</a>
<span class="sourceLineNo">213</span><a id="line.213">        public static AFPChain fastaStringToAfpChain(String sequence1, String sequence2, Structure structure1,</a>
<span class="sourceLineNo">214</span><a id="line.214">                        Structure structure2) throws StructureException, CompoundNotFoundException {</a>
<span class="sourceLineNo">215</span><a id="line.215">                ProteinSequence seq1 = new ProteinSequence(sequence1);</a>
<span class="sourceLineNo">216</span><a id="line.216">                ProteinSequence seq2 = new ProteinSequence(sequence2);</a>
<span class="sourceLineNo">217</span><a id="line.217">                return fastaToAfpChain(seq1, seq2, structure1, structure2);</a>
<span class="sourceLineNo">218</span><a id="line.218">        }</a>
<span class="sourceLineNo">219</span><a id="line.219"></a>
<span class="sourceLineNo">220</span><a id="line.220">        /**</a>
<span class="sourceLineNo">221</span><a id="line.221">         * Uses two sequences each with a corresponding structure to create an AFPChain corresponding to the alignment. Provided only for convenience since FastaReaders return such maps.</a>
<span class="sourceLineNo">222</span><a id="line.222">         *</a>
<span class="sourceLineNo">223</span><a id="line.223">         * @param sequences</a>
<span class="sourceLineNo">224</span><a id="line.224">         *            A Map containing exactly two entries from sequence names as Strings to gapped ProteinSequences; the name is ignored</a>
<span class="sourceLineNo">225</span><a id="line.225">         * @see #fastaToAfpChain(ProteinSequence, ProteinSequence, Structure, Structure)</a>
<span class="sourceLineNo">226</span><a id="line.226">         * @throws StructureException</a>
<span class="sourceLineNo">227</span><a id="line.227">         */</a>
<span class="sourceLineNo">228</span><a id="line.228">        public static AFPChain fastaToAfpChain(Map&lt;String, ProteinSequence&gt; sequences, Structure structure1,</a>
<span class="sourceLineNo">229</span><a id="line.229">                        Structure structure2) throws StructureException {</a>
<span class="sourceLineNo">230</span><a id="line.230"></a>
<span class="sourceLineNo">231</span><a id="line.231">                if (sequences.size() != 2) {</a>
<span class="sourceLineNo">232</span><a id="line.232">                        throw new IllegalArgumentException("There must be exactly 2 sequences, but there were " + sequences.size());</a>
<span class="sourceLineNo">233</span><a id="line.233">                }</a>
<span class="sourceLineNo">234</span><a id="line.234"></a>
<span class="sourceLineNo">235</span><a id="line.235">                if (structure1 == null || structure2 == null) {</a>
<span class="sourceLineNo">236</span><a id="line.236">                        throw new IllegalArgumentException("A structure is null");</a>
<span class="sourceLineNo">237</span><a id="line.237">                }</a>
<span class="sourceLineNo">238</span><a id="line.238"></a>
<span class="sourceLineNo">239</span><a id="line.239">                List&lt;ProteinSequence&gt; seqs = new ArrayList&lt;ProteinSequence&gt;();</a>
<span class="sourceLineNo">240</span><a id="line.240">                List&lt;String&gt; names = new ArrayList&lt;String&gt;(2);</a>
<span class="sourceLineNo">241</span><a id="line.241">                for (Map.Entry&lt;String, ProteinSequence&gt; entry : sequences.entrySet()) {</a>
<span class="sourceLineNo">242</span><a id="line.242">                        seqs.add(entry.getValue());</a>
<span class="sourceLineNo">243</span><a id="line.243">                        names.add(entry.getKey());</a>
<span class="sourceLineNo">244</span><a id="line.244">                }</a>
<span class="sourceLineNo">245</span><a id="line.245"></a>
<span class="sourceLineNo">246</span><a id="line.246">                return fastaToAfpChain(seqs.get(0), seqs.get(1), structure1, structure2);</a>
<span class="sourceLineNo">247</span><a id="line.247">        }</a>
<span class="sourceLineNo">248</span><a id="line.248"></a>
<span class="sourceLineNo">249</span><a id="line.249">        /**</a>
<span class="sourceLineNo">250</span><a id="line.250">         * TODO Write comment</a>
<span class="sourceLineNo">251</span><a id="line.251">         * @param sequence1</a>
<span class="sourceLineNo">252</span><a id="line.252">         * @param sequence2</a>
<span class="sourceLineNo">253</span><a id="line.253">         * @param structure1</a>
<span class="sourceLineNo">254</span><a id="line.254">         * @param structure2</a>
<span class="sourceLineNo">255</span><a id="line.255">         * @return</a>
<span class="sourceLineNo">256</span><a id="line.256">         * @throws StructureException</a>
<span class="sourceLineNo">257</span><a id="line.257">         * @throws CompoundNotFoundException</a>
<span class="sourceLineNo">258</span><a id="line.258">         */</a>
<span class="sourceLineNo">259</span><a id="line.259">        public static AFPChain fastaToAfpChain(String sequence1, String sequence2, Structure structure1,</a>
<span class="sourceLineNo">260</span><a id="line.260">                        Structure structure2) throws StructureException, CompoundNotFoundException {</a>
<span class="sourceLineNo">261</span><a id="line.261">                ProteinSequence s1 = new ProteinSequence(sequence1);</a>
<span class="sourceLineNo">262</span><a id="line.262">                s1.setUserCollection(getAlignedUserCollection(sequence1));</a>
<span class="sourceLineNo">263</span><a id="line.263">                ProteinSequence s2 = new ProteinSequence(sequence2);</a>
<span class="sourceLineNo">264</span><a id="line.264">                s2.setUserCollection(getAlignedUserCollection(sequence2));</a>
<span class="sourceLineNo">265</span><a id="line.265">                return fastaToAfpChain(s1, s2, structure1, structure2);</a>
<span class="sourceLineNo">266</span><a id="line.266">        }</a>
<span class="sourceLineNo">267</span><a id="line.267"></a>
<span class="sourceLineNo">268</span><a id="line.268">        /**</a>
<span class="sourceLineNo">269</span><a id="line.269">         * Returns an AFPChain corresponding to the alignment between {@code structure1} and {@code structure2}, which is given by the gapped protein sequences {@code sequence1} and {@code sequence2}. The</a>
<span class="sourceLineNo">270</span><a id="line.270">         * sequences need not correspond to the entire structures, since local alignment is performed to match the sequences to structures. Assumes that a residue is aligned if and only if it is given by</a>
<span class="sourceLineNo">271</span><a id="line.271">         * an uppercase letter.</a>
<span class="sourceLineNo">272</span><a id="line.272">         * @param sequence1 &lt;em&gt;Must&lt;/em&gt; have {@link ProteinSequence#getUserCollection()} set to document upper- and lower-case as aligned and unaligned; see {@link #getAlignedUserCollection(String)}</a>
<span class="sourceLineNo">273</span><a id="line.273">         * @throws StructureException</a>
<span class="sourceLineNo">274</span><a id="line.274">         */</a>
<span class="sourceLineNo">275</span><a id="line.275">        public static AFPChain fastaToAfpChain(ProteinSequence sequence1, ProteinSequence sequence2, Structure structure1,</a>
<span class="sourceLineNo">276</span><a id="line.276">                        Structure structure2) throws StructureException {</a>
<span class="sourceLineNo">277</span><a id="line.277"></a>
<span class="sourceLineNo">278</span><a id="line.278">                if (structure1 == null || structure2 == null) {</a>
<span class="sourceLineNo">279</span><a id="line.279">                        throw new IllegalArgumentException("A structure is null");</a>
<span class="sourceLineNo">280</span><a id="line.280">                }</a>
<span class="sourceLineNo">281</span><a id="line.281"></a>
<span class="sourceLineNo">282</span><a id="line.282">                if (sequence1 == null || sequence2 == null) {</a>
<span class="sourceLineNo">283</span><a id="line.283">                        throw new IllegalArgumentException("A sequence is null");</a>
<span class="sourceLineNo">284</span><a id="line.284">                }</a>
<span class="sourceLineNo">285</span><a id="line.285"></a>
<span class="sourceLineNo">286</span><a id="line.286">                Atom[] ca1 = StructureTools.getRepresentativeAtomArray(structure1);</a>
<span class="sourceLineNo">287</span><a id="line.287">                Atom[] ca2 = StructureTools.getRepresentativeAtomArray(structure2);</a>
<span class="sourceLineNo">288</span><a id="line.288"></a>
<span class="sourceLineNo">289</span><a id="line.289">                ResidueNumber[] residues1 = StructureSequenceMatcher.matchSequenceToStructure(sequence1, structure1);</a>
<span class="sourceLineNo">290</span><a id="line.290">                ResidueNumber[] residues2 = StructureSequenceMatcher.matchSequenceToStructure(sequence2, structure2);</a>
<span class="sourceLineNo">291</span><a id="line.291"></a>
<span class="sourceLineNo">292</span><a id="line.292">                // nullify ResidueNumbers that have a lowercase sequence character</a>
<span class="sourceLineNo">293</span><a id="line.293">                if (sequence1.getUserCollection() != null) {</a>
<span class="sourceLineNo">294</span><a id="line.294">                        CasePreservingProteinSequenceCreator.setLowercaseToNull(sequence1, residues1);</a>
<span class="sourceLineNo">295</span><a id="line.295">                }</a>
<span class="sourceLineNo">296</span><a id="line.296">                if (sequence2.getUserCollection() != null) {</a>
<span class="sourceLineNo">297</span><a id="line.297">                        CasePreservingProteinSequenceCreator.setLowercaseToNull(sequence2, residues2);</a>
<span class="sourceLineNo">298</span><a id="line.298">                }</a>
<span class="sourceLineNo">299</span><a id="line.299"></a>
<span class="sourceLineNo">300</span><a id="line.300">                return buildAlignment(ca1, ca2, residues1, residues2);</a>
<span class="sourceLineNo">301</span><a id="line.301"></a>
<span class="sourceLineNo">302</span><a id="line.302">        }</a>
<span class="sourceLineNo">303</span><a id="line.303"></a>
<span class="sourceLineNo">304</span><a id="line.304">        /**</a>
<span class="sourceLineNo">305</span><a id="line.305">         * Provided only for convenience.</a>
<span class="sourceLineNo">306</span><a id="line.306">         *</a>
<span class="sourceLineNo">307</span><a id="line.307">         * @see #fastaToAfpChain(ProteinSequence, ProteinSequence, Structure, Structure)</a>
<span class="sourceLineNo">308</span><a id="line.308">         * @throws StructureException</a>
<span class="sourceLineNo">309</span><a id="line.309">         */</a>
<span class="sourceLineNo">310</span><a id="line.310">        public static AFPChain fastaToAfpChain(SequencePair&lt;Sequence&lt;AminoAcidCompound&gt;, AminoAcidCompound&gt; alignment,</a>
<span class="sourceLineNo">311</span><a id="line.311">                        Structure structure1, Structure structure2) throws StructureException {</a>
<span class="sourceLineNo">312</span><a id="line.312">                List&lt;AlignedSequence&lt;Sequence&lt;AminoAcidCompound&gt;, AminoAcidCompound&gt;&gt; seqs = alignment.getAlignedSequences();</a>
<span class="sourceLineNo">313</span><a id="line.313">                StringBuilder sb1 = new StringBuilder();</a>
<span class="sourceLineNo">314</span><a id="line.314">                for (AminoAcidCompound a : seqs.get(0)) {</a>
<span class="sourceLineNo">315</span><a id="line.315">                        sb1.append(a.getBase());</a>
<span class="sourceLineNo">316</span><a id="line.316">                }</a>
<span class="sourceLineNo">317</span><a id="line.317">                try {</a>
<span class="sourceLineNo">318</span><a id="line.318">                        ProteinSequence seq1 = new ProteinSequence(sb1.toString());</a>
<span class="sourceLineNo">319</span><a id="line.319">                        StringBuilder sb2 = new StringBuilder();</a>
<span class="sourceLineNo">320</span><a id="line.320">                        for (AminoAcidCompound a : seqs.get(1)) {</a>
<span class="sourceLineNo">321</span><a id="line.321">                                sb1.append(a.getBase());</a>
<span class="sourceLineNo">322</span><a id="line.322">                        }</a>
<span class="sourceLineNo">323</span><a id="line.323">                        ProteinSequence seq2 = new ProteinSequence(sb2.toString());</a>
<span class="sourceLineNo">324</span><a id="line.324">                        LinkedHashMap&lt;String, ProteinSequence&gt; map = new LinkedHashMap&lt;String, ProteinSequence&gt;();</a>
<span class="sourceLineNo">325</span><a id="line.325">                        map.put(structure1.getName(), seq1);</a>
<span class="sourceLineNo">326</span><a id="line.326">                        map.put(structure2.getName(), seq2);</a>
<span class="sourceLineNo">327</span><a id="line.327">                        return fastaToAfpChain(map, structure1, structure2);</a>
<span class="sourceLineNo">328</span><a id="line.328">                } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">329</span><a id="line.329">                        logger.error("Unexpected error while creating protein sequences: {}. This is most likely a bug.",e.getMessage());</a>
<span class="sourceLineNo">330</span><a id="line.330">                        return null;</a>
<span class="sourceLineNo">331</span><a id="line.331">                }</a>
<span class="sourceLineNo">332</span><a id="line.332">        }</a>
<span class="sourceLineNo">333</span><a id="line.333"></a>
<span class="sourceLineNo">334</span><a id="line.334">        /**</a>
<span class="sourceLineNo">335</span><a id="line.335">         * Builds an {@link AFPChain} from already-matched arrays of atoms and residues.</a>
<span class="sourceLineNo">336</span><a id="line.336">         *</a>
<span class="sourceLineNo">337</span><a id="line.337">         * @param ca1</a>
<span class="sourceLineNo">338</span><a id="line.338">         *            An array of atoms in the first structure</a>
<span class="sourceLineNo">339</span><a id="line.339">         * @param ca2</a>
<span class="sourceLineNo">340</span><a id="line.340">         *            An array of atoms in the second structure</a>
<span class="sourceLineNo">341</span><a id="line.341">         * @param residues1</a>
<span class="sourceLineNo">342</span><a id="line.342">         *            An array of {@link ResidueNumber ResidueNumbers} in the first structure that are aligned. Only null ResidueNumbers are considered to be unaligned</a>
<span class="sourceLineNo">343</span><a id="line.343">         * @param residues2</a>
<span class="sourceLineNo">344</span><a id="line.344">         *            An array of {@link ResidueNumber ResidueNumbers} in the second structure that are aligned. Only null ResidueNumbers are considered to be unaligned</a>
<span class="sourceLineNo">345</span><a id="line.345">         * @throws StructureException</a>
<span class="sourceLineNo">346</span><a id="line.346">         */</a>
<span class="sourceLineNo">347</span><a id="line.347">        private static AFPChain buildAlignment(Atom[] ca1, Atom[] ca2, ResidueNumber[] residues1, ResidueNumber[] residues2)</a>
<span class="sourceLineNo">348</span><a id="line.348">                        throws StructureException {</a>
<span class="sourceLineNo">349</span><a id="line.349"></a>
<span class="sourceLineNo">350</span><a id="line.350">                // remove any gap</a>
<span class="sourceLineNo">351</span><a id="line.351">                // this includes the ones introduced by the nullifying above</a>
<span class="sourceLineNo">352</span><a id="line.352">                List&lt;ResidueNumber&gt; alignedResiduesList1 = new ArrayList&lt;ResidueNumber&gt;();</a>
<span class="sourceLineNo">353</span><a id="line.353">                List&lt;ResidueNumber&gt; alignedResiduesList2 = new ArrayList&lt;ResidueNumber&gt;();</a>
<span class="sourceLineNo">354</span><a id="line.354">                for (int i = 0; i &lt; residues1.length; i++) {</a>
<span class="sourceLineNo">355</span><a id="line.355">                        if (residues1[i] != null &amp;&amp; residues2[i] != null) {</a>
<span class="sourceLineNo">356</span><a id="line.356">                                alignedResiduesList1.add(residues1[i]);</a>
<span class="sourceLineNo">357</span><a id="line.357">                                alignedResiduesList2.add(residues2[i]);</a>
<span class="sourceLineNo">358</span><a id="line.358">                        }</a>
<span class="sourceLineNo">359</span><a id="line.359">                }</a>
<span class="sourceLineNo">360</span><a id="line.360"></a>
<span class="sourceLineNo">361</span><a id="line.361">                ResidueNumber[] alignedResidues1 = alignedResiduesList1.toArray(new ResidueNumber[alignedResiduesList1.size()]);</a>
<span class="sourceLineNo">362</span><a id="line.362">                ResidueNumber[] alignedResidues2 = alignedResiduesList2.toArray(new ResidueNumber[alignedResiduesList2.size()]);</a>
<span class="sourceLineNo">363</span><a id="line.363"></a>
<span class="sourceLineNo">364</span><a id="line.364">                AFPChain afpChain = AlignmentTools.createAFPChain(ca1, ca2, alignedResidues1, alignedResidues2);</a>
<span class="sourceLineNo">365</span><a id="line.365">                afpChain.setAlgorithmName("unknown");</a>
<span class="sourceLineNo">366</span><a id="line.366"></a>
<span class="sourceLineNo">367</span><a id="line.367">                AlignmentTools.updateSuperposition(afpChain, ca1, ca2);</a>
<span class="sourceLineNo">368</span><a id="line.368"></a>
<span class="sourceLineNo">369</span><a id="line.369">                afpChain.setBlockSize(new int[] {afpChain.getNrEQR()});</a>
<span class="sourceLineNo">370</span><a id="line.370">                afpChain.setBlockRmsd(new double[] {afpChain.getTotalRmsdOpt()});</a>
<span class="sourceLineNo">371</span><a id="line.371">                afpChain.setBlockGap(new int[] {afpChain.getGapLen()});</a>
<span class="sourceLineNo">372</span><a id="line.372"></a>
<span class="sourceLineNo">373</span><a id="line.373">                return afpChain;</a>
<span class="sourceLineNo">374</span><a id="line.374"></a>
<span class="sourceLineNo">375</span><a id="line.375">        }</a>
<span class="sourceLineNo">376</span><a id="line.376"></a>
<span class="sourceLineNo">377</span><a id="line.377">        /**</a>
<span class="sourceLineNo">378</span><a id="line.378">         * Takes a protein sequence string with capital and lowercase letters and sets its {@link ProteinSequence#getUserCollection() user collection} to record which letters are uppercase (aligned) and which are lowercase (unaligned).</a>
<span class="sourceLineNo">379</span><a id="line.379">         * @param sequence Make sure &lt;em&gt;not&lt;/em&gt; to use {@link ProteinSequence#getSequenceAsString()} for this, as it won't preserve upper- and lower-case</a>
<span class="sourceLineNo">380</span><a id="line.380">         */</a>
<span class="sourceLineNo">381</span><a id="line.381">        public static List&lt;Object&gt; getAlignedUserCollection(String sequence) {</a>
<span class="sourceLineNo">382</span><a id="line.382">                List&lt;Object&gt; aligned = new ArrayList&lt;Object&gt;(sequence.length());</a>
<span class="sourceLineNo">383</span><a id="line.383">                for (char c : sequence.toCharArray()) {</a>
<span class="sourceLineNo">384</span><a id="line.384">                        aligned.add(Character.isUpperCase(c));</a>
<span class="sourceLineNo">385</span><a id="line.385">                }</a>
<span class="sourceLineNo">386</span><a id="line.386">                return aligned;</a>
<span class="sourceLineNo">387</span><a id="line.387">        }</a>
<span class="sourceLineNo">388</span><a id="line.388"></a>
<span class="sourceLineNo">389</span><a id="line.389">        /**</a>
<span class="sourceLineNo">390</span><a id="line.390">         * Prints out the XML representation of an AFPChain from a file containing exactly two FASTA sequences.</a>
<span class="sourceLineNo">391</span><a id="line.391">         *</a>
<span class="sourceLineNo">392</span><a id="line.392">         * @param args</a>
<span class="sourceLineNo">393</span><a id="line.393">         *            A String array of fasta-file structure-1-name structure-2-name</a>
<span class="sourceLineNo">394</span><a id="line.394">         * @throws StructureException</a>
<span class="sourceLineNo">395</span><a id="line.395">         * @throws IOException</a>
<span class="sourceLineNo">396</span><a id="line.396">         */</a>
<span class="sourceLineNo">397</span><a id="line.397">        public static void main(String[] args) throws StructureException, IOException {</a>
<span class="sourceLineNo">398</span><a id="line.398">                if (args.length != 3) {</a>
<span class="sourceLineNo">399</span><a id="line.399">                        System.err.println("Usage: FastaAFPChainConverter fasta-file structure-1-name structure-2-name");</a>
<span class="sourceLineNo">400</span><a id="line.400">                        return;</a>
<span class="sourceLineNo">401</span><a id="line.401">                }</a>
<span class="sourceLineNo">402</span><a id="line.402">                File fasta = new File(args[0]);</a>
<span class="sourceLineNo">403</span><a id="line.403">                Structure structure1 = StructureTools.getStructure(args[1]);</a>
<span class="sourceLineNo">404</span><a id="line.404">                Structure structure2 = StructureTools.getStructure(args[2]);</a>
<span class="sourceLineNo">405</span><a id="line.405">                if (structure1 == null) throw new IllegalArgumentException("No structure for " + args[1] + " was found");</a>
<span class="sourceLineNo">406</span><a id="line.406">                if (structure2 == null) throw new IllegalArgumentException("No structure for " + args[2] + " was found");</a>
<span class="sourceLineNo">407</span><a id="line.407">                AFPChain afpChain = fastaFileToAfpChain(fasta, structure1, structure2);</a>
<span class="sourceLineNo">408</span><a id="line.408">                String xml = AFPChainXMLConverter.toXML(afpChain);</a>
<span class="sourceLineNo">409</span><a id="line.409">                System.out.println(xml);</a>
<span class="sourceLineNo">410</span><a id="line.410">        }</a>
<span class="sourceLineNo">411</span><a id="line.411"></a>
<span class="sourceLineNo">412</span><a id="line.412">}</a>




























































</pre>
</div>
</main>
</body>
</html>
