<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/**</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> * Created by Spencer Bliven</a>
<span class="sourceLineNo">021</span><a id="line.21"> *</a>
<span class="sourceLineNo">022</span><a id="line.22"> */</a>
<span class="sourceLineNo">023</span><a id="line.23">package org.biojava.nbio.structure.io;</a>
<span class="sourceLineNo">024</span><a id="line.24"></a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.nbio.alignment.Alignments;</a>
<span class="sourceLineNo">026</span><a id="line.26">import org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType;</a>
<span class="sourceLineNo">027</span><a id="line.27">import org.biojava.nbio.alignment.SimpleGapPenalty;</a>
<span class="sourceLineNo">028</span><a id="line.28">import org.biojava.nbio.core.alignment.matrices.SimpleSubstitutionMatrix;</a>
<span class="sourceLineNo">029</span><a id="line.29">import org.biojava.nbio.core.alignment.template.AlignedSequence;</a>
<span class="sourceLineNo">030</span><a id="line.30">import org.biojava.nbio.core.alignment.template.SequencePair;</a>
<span class="sourceLineNo">031</span><a id="line.31">import org.biojava.nbio.core.alignment.template.SubstitutionMatrix;</a>
<span class="sourceLineNo">032</span><a id="line.32">import org.biojava.nbio.structure.*;</a>
<span class="sourceLineNo">033</span><a id="line.33">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</a>
<span class="sourceLineNo">034</span><a id="line.34">import org.biojava.nbio.core.sequence.ProteinSequence;</a>
<span class="sourceLineNo">035</span><a id="line.35">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</a>
<span class="sourceLineNo">036</span><a id="line.36">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</a>
<span class="sourceLineNo">037</span><a id="line.37">import org.biojava.nbio.core.sequence.template.CompoundSet;</a>
<span class="sourceLineNo">038</span><a id="line.38">import org.slf4j.Logger;</a>
<span class="sourceLineNo">039</span><a id="line.39">import org.slf4j.LoggerFactory;</a>
<span class="sourceLineNo">040</span><a id="line.40"></a>
<span class="sourceLineNo">041</span><a id="line.41">import java.io.InputStreamReader;</a>
<span class="sourceLineNo">042</span><a id="line.42">import java.util.*;</a>
<span class="sourceLineNo">043</span><a id="line.43"></a>
<span class="sourceLineNo">044</span><a id="line.44"></a>
<span class="sourceLineNo">045</span><a id="line.45">/**</a>
<span class="sourceLineNo">046</span><a id="line.46"> * A utility class with methods for matching ProteinSequences with</a>
<span class="sourceLineNo">047</span><a id="line.47"> * Structures.</a>
<span class="sourceLineNo">048</span><a id="line.48"> * @author Spencer Bliven</a>
<span class="sourceLineNo">049</span><a id="line.49"> *</a>
<span class="sourceLineNo">050</span><a id="line.50"> */</a>
<span class="sourceLineNo">051</span><a id="line.51">public class StructureSequenceMatcher {</a>
<span class="sourceLineNo">052</span><a id="line.52"></a>
<span class="sourceLineNo">053</span><a id="line.53">        private static final Logger logger = LoggerFactory.getLogger(StructureSequenceMatcher.class);</a>
<span class="sourceLineNo">054</span><a id="line.54"></a>
<span class="sourceLineNo">055</span><a id="line.55">        /**</a>
<span class="sourceLineNo">056</span><a id="line.56">         * Get a substructure of {@code wholeStructure} containing only the {@link Group Groups} that are included in</a>
<span class="sourceLineNo">057</span><a id="line.57">         * {@code sequence}. The resulting structure will contain only {@code ATOM} residues; the SEQ-RES will be empty.</a>
<span class="sourceLineNo">058</span><a id="line.58">         * The {@link Chain Chains} of the Structure will be new instances (cloned), but the {@link Group Groups} will not.</a>
<span class="sourceLineNo">059</span><a id="line.59">         * @param sequence The input protein sequence</a>
<span class="sourceLineNo">060</span><a id="line.60">         * @param wholeStructure The structure from which to take a substructure</a>
<span class="sourceLineNo">061</span><a id="line.61">         * @return The resulting structure</a>
<span class="sourceLineNo">062</span><a id="line.62">         * @throws StructureException</a>
<span class="sourceLineNo">063</span><a id="line.63">         * @see {@link #matchSequenceToStructure(ProteinSequence, Structure)}</a>
<span class="sourceLineNo">064</span><a id="line.64">         */</a>
<span class="sourceLineNo">065</span><a id="line.65">        public static Structure getSubstructureMatchingProteinSequence(ProteinSequence sequence, Structure wholeStructure) {</a>
<span class="sourceLineNo">066</span><a id="line.66">                ResidueNumber[] rns = matchSequenceToStructure(sequence, wholeStructure);</a>
<span class="sourceLineNo">067</span><a id="line.67">                Structure structure = wholeStructure.clone();</a>
<span class="sourceLineNo">068</span><a id="line.68">                structure.setChains(new ArrayList&lt;&gt;());</a>
<span class="sourceLineNo">069</span><a id="line.69">//              structure.getHetGroups().clear();</a>
<span class="sourceLineNo">070</span><a id="line.70">                Chain currentChain = null;</a>
<span class="sourceLineNo">071</span><a id="line.71">                for (ResidueNumber rn : rns) {</a>
<span class="sourceLineNo">072</span><a id="line.72">                        if (rn == null) continue;</a>
<span class="sourceLineNo">073</span><a id="line.73">                        Group group; // note that we don't clone</a>
<span class="sourceLineNo">074</span><a id="line.74">                        try {</a>
<span class="sourceLineNo">075</span><a id="line.75">                                group = StructureTools.getGroupByPDBResidueNumber(wholeStructure, rn);</a>
<span class="sourceLineNo">076</span><a id="line.76">                        } catch (StructureException e) {</a>
<span class="sourceLineNo">077</span><a id="line.77">                                throw new IllegalArgumentException("Could not find residue " + rn + " in structure", e);</a>
<span class="sourceLineNo">078</span><a id="line.78">                        }</a>
<span class="sourceLineNo">079</span><a id="line.79">                        Chain chain = new ChainImpl();</a>
<span class="sourceLineNo">080</span><a id="line.80">                        chain.setName(group.getChain().getName());</a>
<span class="sourceLineNo">081</span><a id="line.81">                        chain.setId(group.getChain().getId());</a>
<span class="sourceLineNo">082</span><a id="line.82">                        if (currentChain == null || !currentChain.getId().equals(chain.getId())) {</a>
<span class="sourceLineNo">083</span><a id="line.83">                                structure.addChain(chain);</a>
<span class="sourceLineNo">084</span><a id="line.84">                                chain.setEntityInfo(group.getChain().getEntityInfo());</a>
<span class="sourceLineNo">085</span><a id="line.85">                                chain.setStructure(structure);</a>
<span class="sourceLineNo">086</span><a id="line.86">                                chain.setId(group.getChain().getId());</a>
<span class="sourceLineNo">087</span><a id="line.87">                                chain.setName(group.getChain().getName());</a>
<span class="sourceLineNo">088</span><a id="line.88">                                currentChain = chain;</a>
<span class="sourceLineNo">089</span><a id="line.89">                        }</a>
<span class="sourceLineNo">090</span><a id="line.90">                        currentChain.addGroup(group);</a>
<span class="sourceLineNo">091</span><a id="line.91">                }</a>
<span class="sourceLineNo">092</span><a id="line.92">                return structure;</a>
<span class="sourceLineNo">093</span><a id="line.93">        }</a>
<span class="sourceLineNo">094</span><a id="line.94"></a>
<span class="sourceLineNo">095</span><a id="line.95">        /**</a>
<span class="sourceLineNo">096</span><a id="line.96">         * Generates a ProteinSequence corresponding to the sequence of struct,</a>
<span class="sourceLineNo">097</span><a id="line.97">         * and maintains a mapping from the sequence back to the original groups.</a>
<span class="sourceLineNo">098</span><a id="line.98">         *</a>
<span class="sourceLineNo">099</span><a id="line.99">         * Chains are appended to one another. 'X' is used for heteroatoms.</a>
<span class="sourceLineNo">100</span><a id="line.100">         *</a>
<span class="sourceLineNo">101</span><a id="line.101">         * @param struct Input structure</a>
<span class="sourceLineNo">102</span><a id="line.102">         * @param groupIndexPosition An empty map, which will be populated with</a>
<span class="sourceLineNo">103</span><a id="line.103">         *  (residue index in returned ProteinSequence) -&gt; (Group within struct)</a>
<span class="sourceLineNo">104</span><a id="line.104">         * @return A ProteinSequence with the full sequence of struct. Chains are</a>
<span class="sourceLineNo">105</span><a id="line.105">         *  concatenated in the same order as the input structures</a>
<span class="sourceLineNo">106</span><a id="line.106">         *</a>
<span class="sourceLineNo">107</span><a id="line.107">         * @see {@link SeqRes2AtomAligner#getFullAtomSequence(List, Map)}, which</a>
<span class="sourceLineNo">108</span><a id="line.108">         *      does the heavy lifting.</a>
<span class="sourceLineNo">109</span><a id="line.109">         *</a>
<span class="sourceLineNo">110</span><a id="line.110">         */</a>
<span class="sourceLineNo">111</span><a id="line.111">        public static ProteinSequence getProteinSequenceForStructure(Structure struct, Map&lt;Integer,Group&gt; groupIndexPosition ) {</a>
<span class="sourceLineNo">112</span><a id="line.112"></a>
<span class="sourceLineNo">113</span><a id="line.113">                if( groupIndexPosition != null) {</a>
<span class="sourceLineNo">114</span><a id="line.114">                        groupIndexPosition.clear();</a>
<span class="sourceLineNo">115</span><a id="line.115">                }</a>
<span class="sourceLineNo">116</span><a id="line.116"></a>
<span class="sourceLineNo">117</span><a id="line.117">                StringBuilder seqStr = new StringBuilder();</a>
<span class="sourceLineNo">118</span><a id="line.118"></a>
<span class="sourceLineNo">119</span><a id="line.119">                for(Chain chain : struct.getChains()) {</a>
<span class="sourceLineNo">120</span><a id="line.120">                        List&lt;Group&gt; groups = chain.getAtomGroups();</a>
<span class="sourceLineNo">121</span><a id="line.121">                        Map&lt;Integer,Integer&gt; chainIndexPosition = new HashMap&lt;Integer, Integer&gt;();</a>
<span class="sourceLineNo">122</span><a id="line.122">                        int prevLen = seqStr.length();</a>
<span class="sourceLineNo">123</span><a id="line.123"></a>
<span class="sourceLineNo">124</span><a id="line.124">                        // get the sequence for this chain</a>
<span class="sourceLineNo">125</span><a id="line.125">                        String chainSeq = SeqRes2AtomAligner.getFullAtomSequence(groups, chainIndexPosition, false);</a>
<span class="sourceLineNo">126</span><a id="line.126">                        seqStr.append(chainSeq);</a>
<span class="sourceLineNo">127</span><a id="line.127"></a>
<span class="sourceLineNo">128</span><a id="line.128"></a>
<span class="sourceLineNo">129</span><a id="line.129">                        // fix up the position to include previous chains, and map the value back to a Group</a>
<span class="sourceLineNo">130</span><a id="line.130">                        for(Integer seqIndex : chainIndexPosition.keySet()) {</a>
<span class="sourceLineNo">131</span><a id="line.131">                                Integer groupIndex = chainIndexPosition.get(seqIndex);</a>
<span class="sourceLineNo">132</span><a id="line.132">                                groupIndexPosition.put(prevLen + seqIndex, groups.get(groupIndex));</a>
<span class="sourceLineNo">133</span><a id="line.133">                        }</a>
<span class="sourceLineNo">134</span><a id="line.134">                }</a>
<span class="sourceLineNo">135</span><a id="line.135"></a>
<span class="sourceLineNo">136</span><a id="line.136">                ProteinSequence s = null;</a>
<span class="sourceLineNo">137</span><a id="line.137">                try {</a>
<span class="sourceLineNo">138</span><a id="line.138">                        s = new ProteinSequence(seqStr.toString());</a>
<span class="sourceLineNo">139</span><a id="line.139">                } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">140</span><a id="line.140">                        // I believe this can't happen, please correct this if I'm wrong - JD 2014-10-24</a>
<span class="sourceLineNo">141</span><a id="line.141">                        // we can log an error if it does, it would mean there's a bad bug somewhere</a>
<span class="sourceLineNo">142</span><a id="line.142">                        logger.error("Could not create protein sequence, unknown compounds in string: {}", e.getMessage());</a>
<span class="sourceLineNo">143</span><a id="line.143">                }</a>
<span class="sourceLineNo">144</span><a id="line.144"></a>
<span class="sourceLineNo">145</span><a id="line.145">                return s;</a>
<span class="sourceLineNo">146</span><a id="line.146">        }</a>
<span class="sourceLineNo">147</span><a id="line.147"></a>
<span class="sourceLineNo">148</span><a id="line.148">        /**</a>
<span class="sourceLineNo">149</span><a id="line.149">         * Given a sequence and the corresponding Structure, get the ResidueNumber</a>
<span class="sourceLineNo">150</span><a id="line.150">         * for each residue in the sequence.</a>
<span class="sourceLineNo">151</span><a id="line.151">         *</a>
<span class="sourceLineNo">152</span><a id="line.152">         * &lt;p&gt;Smith-Waterman alignment is used to match the sequences. Residues</a>
<span class="sourceLineNo">153</span><a id="line.153">         * in the sequence but not the structure or mismatched between sequence</a>
<span class="sourceLineNo">154</span><a id="line.154">         * and structure will have a null atom, while</a>
<span class="sourceLineNo">155</span><a id="line.155">         * residues in the structure but not the sequence are ignored with a warning.</a>
<span class="sourceLineNo">156</span><a id="line.156">         * @param seq The protein sequence. Should match the sequence of struct very</a>
<span class="sourceLineNo">157</span><a id="line.157">         *      closely.</a>
<span class="sourceLineNo">158</span><a id="line.158">         * @param struct The corresponding protein structure</a>
<span class="sourceLineNo">159</span><a id="line.159">         * @return A list of ResidueNumbers of the same length as seq, containing</a>
<span class="sourceLineNo">160</span><a id="line.160">         *  either the corresponding residue or null.</a>
<span class="sourceLineNo">161</span><a id="line.161">         */</a>
<span class="sourceLineNo">162</span><a id="line.162">        public static ResidueNumber[] matchSequenceToStructure(ProteinSequence seq, Structure struct) {</a>
<span class="sourceLineNo">163</span><a id="line.163"></a>
<span class="sourceLineNo">164</span><a id="line.164">                //1. Create ProteinSequence for struct while remembering to which group each residue corresponds</a>
<span class="sourceLineNo">165</span><a id="line.165">                Map&lt;Integer,Group&gt; atomIndexPosition   = new HashMap&lt;Integer, Group&gt;();</a>
<span class="sourceLineNo">166</span><a id="line.166"></a>
<span class="sourceLineNo">167</span><a id="line.167">                ProteinSequence structSeq = getProteinSequenceForStructure(struct,atomIndexPosition);</a>
<span class="sourceLineNo">168</span><a id="line.168"></a>
<span class="sourceLineNo">169</span><a id="line.169">                // TODO This should really be semi-global alignment, though global for either sequence OR structure (we don't know which)</a>
<span class="sourceLineNo">170</span><a id="line.170">                //2. Run Smith-Waterman to get the alignment</a>
<span class="sourceLineNo">171</span><a id="line.171">                // Identity substitution matrix with +1 for match, -1 for mismatch</a>
<span class="sourceLineNo">172</span><a id="line.172">                // TODO</a>
<span class="sourceLineNo">173</span><a id="line.173">                SubstitutionMatrix&lt;AminoAcidCompound&gt; matrix =</a>
<span class="sourceLineNo">174</span><a id="line.174">                        new SimpleSubstitutionMatrix&lt;AminoAcidCompound&gt;(</a>
<span class="sourceLineNo">175</span><a id="line.175">                                        AminoAcidCompoundSet.getAminoAcidCompoundSet(),</a>
<span class="sourceLineNo">176</span><a id="line.176">                                        (short)1, (short)-1 );</a>
<span class="sourceLineNo">177</span><a id="line.177">                matrix = new SimpleSubstitutionMatrix&lt;AminoAcidCompound&gt;(</a>
<span class="sourceLineNo">178</span><a id="line.178">                                AminoAcidCompoundSet.getAminoAcidCompoundSet(),</a>
<span class="sourceLineNo">179</span><a id="line.179">                                new InputStreamReader(</a>
<span class="sourceLineNo">180</span><a id="line.180">                                                SimpleSubstitutionMatrix.class.getResourceAsStream("/matrices/blosum100.txt")),</a>
<span class="sourceLineNo">181</span><a id="line.181">                "blosum100");</a>
<span class="sourceLineNo">182</span><a id="line.182">                SequencePair&lt;ProteinSequence, AminoAcidCompound&gt; pair =</a>
<span class="sourceLineNo">183</span><a id="line.183">                        Alignments.getPairwiseAlignment(seq, structSeq,</a>
<span class="sourceLineNo">184</span><a id="line.184">                                        PairwiseSequenceAlignerType.GLOBAL, new SimpleGapPenalty(), matrix);</a>
<span class="sourceLineNo">185</span><a id="line.185"></a>
<span class="sourceLineNo">186</span><a id="line.186">                //System.out.print(pair.toString());</a>
<span class="sourceLineNo">187</span><a id="line.187"></a>
<span class="sourceLineNo">188</span><a id="line.188">                //3. Convert the alignment back to Atoms</a>
<span class="sourceLineNo">189</span><a id="line.189">                AlignedSequence&lt;ProteinSequence,AminoAcidCompound&gt; alignedSeq = pair.getQuery();</a>
<span class="sourceLineNo">190</span><a id="line.190">                AlignedSequence&lt;ProteinSequence,AminoAcidCompound&gt; alignedStruct = pair.getTarget();</a>
<span class="sourceLineNo">191</span><a id="line.191"></a>
<span class="sourceLineNo">192</span><a id="line.192"></a>
<span class="sourceLineNo">193</span><a id="line.193">                assert(alignedSeq.getLength() == alignedStruct.getLength());</a>
<span class="sourceLineNo">194</span><a id="line.194"></a>
<span class="sourceLineNo">195</span><a id="line.195">//              System.out.println(pair.toString(80));</a>
<span class="sourceLineNo">196</span><a id="line.196">//              System.out.format("%d/min{%d,%d}\n", pair.getNumIdenticals(),</a>
<span class="sourceLineNo">197</span><a id="line.197">//                              alignedSeq.getLength()-alignedSeq.getNumGaps(),</a>
<span class="sourceLineNo">198</span><a id="line.198">//                              alignedStruct.getLength()-alignedStruct.getNumGaps());</a>
<span class="sourceLineNo">199</span><a id="line.199"></a>
<span class="sourceLineNo">200</span><a id="line.200">                ResidueNumber[] ca = new ResidueNumber[seq.getLength()];</a>
<span class="sourceLineNo">201</span><a id="line.201"></a>
<span class="sourceLineNo">202</span><a id="line.202">                for( int pos = alignedSeq.getStart().getPosition(); pos &lt;= alignedSeq.getEnd().getPosition(); pos++ ) { // 1-indexed</a>
<span class="sourceLineNo">203</span><a id="line.203">                        //skip missing residues from sequence. These probably represent alignment errors</a>
<span class="sourceLineNo">204</span><a id="line.204">                        if(alignedSeq.isGap(pos)) {</a>
<span class="sourceLineNo">205</span><a id="line.205">                                int structIndex = alignedStruct.getSequenceIndexAt(pos)-1;</a>
<span class="sourceLineNo">206</span><a id="line.206">                                assert(structIndex &gt; 0);//should be defined since seq gap</a>
<span class="sourceLineNo">207</span><a id="line.207"></a>
<span class="sourceLineNo">208</span><a id="line.208">                                Group g = atomIndexPosition.get(structIndex);</a>
<span class="sourceLineNo">209</span><a id="line.209"></a>
<span class="sourceLineNo">210</span><a id="line.210">                                logger.warn("Chain {} residue {} in the Structure {} has no corresponding amino acid in the sequence.",</a>
<span class="sourceLineNo">211</span><a id="line.211">                                                g.getChainId(),</a>
<span class="sourceLineNo">212</span><a id="line.212">                                                g.getResidueNumber().toString(),</a>
<span class="sourceLineNo">213</span><a id="line.213">                                                g.getChain().getStructure().getPDBCode() );</a>
<span class="sourceLineNo">214</span><a id="line.214">                                continue;</a>
<span class="sourceLineNo">215</span><a id="line.215">                        }</a>
<span class="sourceLineNo">216</span><a id="line.216"></a>
<span class="sourceLineNo">217</span><a id="line.217">                        if(! alignedStruct.isGap(pos) ) {</a>
<span class="sourceLineNo">218</span><a id="line.218">                                int seqIndex = alignedSeq.getSequenceIndexAt(pos)-1;//1-indexed</a>
<span class="sourceLineNo">219</span><a id="line.219">                                int structIndex = alignedStruct.getSequenceIndexAt(pos)-1;//1-indexed</a>
<span class="sourceLineNo">220</span><a id="line.220">                                Group g = atomIndexPosition.get(structIndex);</a>
<span class="sourceLineNo">221</span><a id="line.221"></a>
<span class="sourceLineNo">222</span><a id="line.222">                                assert(0&lt;=seqIndex &amp;&amp; seqIndex &lt; ca.length);</a>
<span class="sourceLineNo">223</span><a id="line.223"></a>
<span class="sourceLineNo">224</span><a id="line.224">                                ca[seqIndex] = g.getResidueNumber(); //remains null for gaps</a>
<span class="sourceLineNo">225</span><a id="line.225">                        }</a>
<span class="sourceLineNo">226</span><a id="line.226">                }</a>
<span class="sourceLineNo">227</span><a id="line.227">                return ca;</a>
<span class="sourceLineNo">228</span><a id="line.228">        }</a>
<span class="sourceLineNo">229</span><a id="line.229"></a>
<span class="sourceLineNo">230</span><a id="line.230"></a>
<span class="sourceLineNo">231</span><a id="line.231">        /**</a>
<span class="sourceLineNo">232</span><a id="line.232">         * Removes all gaps ('-') from a protein sequence</a>
<span class="sourceLineNo">233</span><a id="line.233">         * @param gapped</a>
<span class="sourceLineNo">234</span><a id="line.234">         * @return</a>
<span class="sourceLineNo">235</span><a id="line.235">         */</a>
<span class="sourceLineNo">236</span><a id="line.236">        public static ProteinSequence removeGaps(ProteinSequence gapped) {</a>
<span class="sourceLineNo">237</span><a id="line.237">                final String[] gapStrings = {"-","."};</a>
<span class="sourceLineNo">238</span><a id="line.238"></a>
<span class="sourceLineNo">239</span><a id="line.239">                StringBuilder seq = new StringBuilder();</a>
<span class="sourceLineNo">240</span><a id="line.240"></a>
<span class="sourceLineNo">241</span><a id="line.241">                CompoundSet&lt;AminoAcidCompound&gt; aaSet = gapped.getCompoundSet();</a>
<span class="sourceLineNo">242</span><a id="line.242">                AminoAcidCompound[] gaps = new AminoAcidCompound[gapStrings.length];</a>
<span class="sourceLineNo">243</span><a id="line.243"></a>
<span class="sourceLineNo">244</span><a id="line.244">                for(int i=0;i&lt;gapStrings.length;i++) {</a>
<span class="sourceLineNo">245</span><a id="line.245">                        gaps[i] = aaSet.getCompoundForString(gapStrings[i]);</a>
<span class="sourceLineNo">246</span><a id="line.246">                }</a>
<span class="sourceLineNo">247</span><a id="line.247"></a>
<span class="sourceLineNo">248</span><a id="line.248">                for(int i=1; i&lt;=gapped.getLength();i++) { //1-indexed</a>
<span class="sourceLineNo">249</span><a id="line.249">                        AminoAcidCompound aa = gapped.getCompoundAt(i);</a>
<span class="sourceLineNo">250</span><a id="line.250">                        boolean isGap = false;</a>
<span class="sourceLineNo">251</span><a id="line.251">                        for(AminoAcidCompound gap : gaps) {</a>
<span class="sourceLineNo">252</span><a id="line.252">                                if( aa.equals(gap)) {</a>
<span class="sourceLineNo">253</span><a id="line.253">                                        isGap = true;</a>
<span class="sourceLineNo">254</span><a id="line.254">                                        break;</a>
<span class="sourceLineNo">255</span><a id="line.255">                                }</a>
<span class="sourceLineNo">256</span><a id="line.256">                        }</a>
<span class="sourceLineNo">257</span><a id="line.257">                        if(!isGap) {</a>
<span class="sourceLineNo">258</span><a id="line.258">                                seq.append(aa.getShortName());</a>
<span class="sourceLineNo">259</span><a id="line.259">                        }</a>
<span class="sourceLineNo">260</span><a id="line.260">                }</a>
<span class="sourceLineNo">261</span><a id="line.261"></a>
<span class="sourceLineNo">262</span><a id="line.262">                ProteinSequence ungapped = null;</a>
<span class="sourceLineNo">263</span><a id="line.263">                try {</a>
<span class="sourceLineNo">264</span><a id="line.264">                        ungapped = new ProteinSequence(seq.toString());</a>
<span class="sourceLineNo">265</span><a id="line.265">                } catch (CompoundNotFoundException e) {</a>
<span class="sourceLineNo">266</span><a id="line.266">                        // this can't happen, if it does there's a bug somewhere</a>
<span class="sourceLineNo">267</span><a id="line.267">                        logger.error("Could not create ungapped protein sequence, found unknown compounds: {}. This is most likely a bug.", e.getMessage());</a>
<span class="sourceLineNo">268</span><a id="line.268">                }</a>
<span class="sourceLineNo">269</span><a id="line.269"></a>
<span class="sourceLineNo">270</span><a id="line.270">                return ungapped;</a>
<span class="sourceLineNo">271</span><a id="line.271">        }</a>
<span class="sourceLineNo">272</span><a id="line.272"></a>
<span class="sourceLineNo">273</span><a id="line.273">        /**</a>
<span class="sourceLineNo">274</span><a id="line.274">         * Creates a new list consisting of all columns of gapped where no row</a>
<span class="sourceLineNo">275</span><a id="line.275">         * contained a null value.</a>
<span class="sourceLineNo">276</span><a id="line.276">         *</a>
<span class="sourceLineNo">277</span><a id="line.277">         * Here, "row" refers to the first index and "column" to the second, eg</a>
<span class="sourceLineNo">278</span><a id="line.278">         * gapped.get(row).get(column)</a>
<span class="sourceLineNo">279</span><a id="line.279">         * @param gapped A rectangular matrix containing null to mark gaps</a>
<span class="sourceLineNo">280</span><a id="line.280">         * @return A new List without columns containing nulls</a>
<span class="sourceLineNo">281</span><a id="line.281">         */</a>
<span class="sourceLineNo">282</span><a id="line.282">        public static &lt;T&gt; T[][] removeGaps(final T[][] gapped) {</a>
<span class="sourceLineNo">283</span><a id="line.283">                if(gapped == null ) return null;</a>
<span class="sourceLineNo">284</span><a id="line.284">                if(gapped.length &lt; 1) return Arrays.copyOf(gapped, gapped.length);</a>
<span class="sourceLineNo">285</span><a id="line.285"></a>
<span class="sourceLineNo">286</span><a id="line.286">                final int nProts = gapped.length;</a>
<span class="sourceLineNo">287</span><a id="line.287">                final int protLen = gapped[0].length; // length of gapped proteins</a>
<span class="sourceLineNo">288</span><a id="line.288"></a>
<span class="sourceLineNo">289</span><a id="line.289">                // Verify that input is rectangular</a>
<span class="sourceLineNo">290</span><a id="line.290">                for(int i=0;i&lt;nProts;i++) {</a>
<span class="sourceLineNo">291</span><a id="line.291">                        if(gapped[i].length != protLen) {</a>
<span class="sourceLineNo">292</span><a id="line.292">                                throw new IllegalArgumentException(String.format(</a>
<span class="sourceLineNo">293</span><a id="line.293">                                                "Expected a rectangular array, but row 0 has %d elements " +</a>
<span class="sourceLineNo">294</span><a id="line.294">                                                "while row %d has %d.", protLen,i,gapped[i].length));</a>
<span class="sourceLineNo">295</span><a id="line.295"></a>
<span class="sourceLineNo">296</span><a id="line.296">                        }</a>
<span class="sourceLineNo">297</span><a id="line.297">                }</a>
<span class="sourceLineNo">298</span><a id="line.298"></a>
<span class="sourceLineNo">299</span><a id="line.299">                // determine where gaps exist in any structures</a>
<span class="sourceLineNo">300</span><a id="line.300">                boolean[] isGap = new boolean[protLen];</a>
<span class="sourceLineNo">301</span><a id="line.301">                int gaps = 0;</a>
<span class="sourceLineNo">302</span><a id="line.302">                for(int j=0;j&lt;protLen;j++) {</a>
<span class="sourceLineNo">303</span><a id="line.303">                        for(int i=0;i&lt;nProts;i++) {</a>
<span class="sourceLineNo">304</span><a id="line.304">                                if(gapped[i][j] == null ) {</a>
<span class="sourceLineNo">305</span><a id="line.305">                                        isGap[j] = true;</a>
<span class="sourceLineNo">306</span><a id="line.306">                                        gaps++;</a>
<span class="sourceLineNo">307</span><a id="line.307">                                        break; //go to next position</a>
<span class="sourceLineNo">308</span><a id="line.308">                                }</a>
<span class="sourceLineNo">309</span><a id="line.309">                        }</a>
<span class="sourceLineNo">310</span><a id="line.310">                }</a>
<span class="sourceLineNo">311</span><a id="line.311"></a>
<span class="sourceLineNo">312</span><a id="line.312">                // Create ungapped array</a>
<span class="sourceLineNo">313</span><a id="line.313">                T[][] ungapped = Arrays.copyOf(gapped,nProts);</a>
<span class="sourceLineNo">314</span><a id="line.314">                final int ungappedLen = protLen-gaps;</a>
<span class="sourceLineNo">315</span><a id="line.315">                for(int i=0;i&lt;nProts;i++) {</a>
<span class="sourceLineNo">316</span><a id="line.316">                        ungapped[i] = Arrays.copyOf(gapped[i],ungappedLen);</a>
<span class="sourceLineNo">317</span><a id="line.317">                        int k = 0;</a>
<span class="sourceLineNo">318</span><a id="line.318">                        for(int j=0;j&lt;protLen;j++) {</a>
<span class="sourceLineNo">319</span><a id="line.319">                                if(!isGap[j]) { //skip gaps</a>
<span class="sourceLineNo">320</span><a id="line.320">                                        assert(gapped[i][j] != null);</a>
<span class="sourceLineNo">321</span><a id="line.321">                                        ungapped[i][k] = gapped[i][j];</a>
<span class="sourceLineNo">322</span><a id="line.322">                                        k++;</a>
<span class="sourceLineNo">323</span><a id="line.323">                                }</a>
<span class="sourceLineNo">324</span><a id="line.324">                        }</a>
<span class="sourceLineNo">325</span><a id="line.325">                        assert(k == ungappedLen);</a>
<span class="sourceLineNo">326</span><a id="line.326">                }</a>
<span class="sourceLineNo">327</span><a id="line.327"></a>
<span class="sourceLineNo">328</span><a id="line.328">                return ungapped;</a>
<span class="sourceLineNo">329</span><a id="line.329">        }</a>
<span class="sourceLineNo">330</span><a id="line.330">}</a>




























































</pre>
</div>
</main>
</body>
</html>
