<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_162) on Fri Apr 06 16:19:08 PDT 2018 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Class org.biojava.nbio.core.sequence.template.AbstractSequence (biojava 5.0.0 API)</title>
<meta name="date" content="2018-04-06">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class org.biojava.nbio.core.sequence.template.AbstractSequence (biojava 5.0.0 API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/core/sequence/template/class-use/AbstractSequence.html" target="_top">Frames</a></li>
<li><a href="AbstractSequence.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Class org.biojava.nbio.core.sequence.template.AbstractSequence" class="title">Uses of Class<br>org.biojava.nbio.core.sequence.template.AbstractSequence</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.alignment.io">org.biojava.nbio.alignment.io</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.features">org.biojava.nbio.core.sequence.features</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.io.template">org.biojava.nbio.core.sequence.io.template</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.loader">org.biojava.nbio.core.sequence.loader</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.location">org.biojava.nbio.core.sequence.location</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.genome.io.fastq">org.biojava.nbio.genome.io.fastq</a></td>
<td class="colLast">
<div class="block">FASTQ and variants sequence format I/O.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.nbio.alignment.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../../org/biojava/nbio/alignment/io/package-summary.html">org.biojava.nbio.alignment.io</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/alignment/io/package-summary.html">org.biojava.nbio.alignment.io</a> that return types with arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;? extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompound.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompound</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">StockholmStructure.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/alignment/io/StockholmStructure.html#getBioSequences--">getBioSequences</a></span>()</code>
<div class="block">used to retrieve sequences from the structure</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;? extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompound.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompound</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">StockholmStructure.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/alignment/io/StockholmStructure.html#getBioSequences-boolean-">getBioSequences</a></span>(boolean&nbsp;ignoreCase)</code>
<div class="block">Because some database files have incorrectly small letters (e.g.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;? extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompound.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompound</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">StockholmStructure.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/alignment/io/StockholmStructure.html#getBioSequences-boolean-java.lang.String-">getBioSequences</a></span>(boolean&nbsp;ignoreCase,
               <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;forcedSequenceType)</code>
<div class="block">This function tolerates mixed case letters, and allows for forcing the output biosequence type (PFAM/RFAM).</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/BasicSequence.html" title="class in org.biojava.nbio.core.sequence">BasicSequence</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">Bare bones version of the Sequence object to be used sparingly.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a></span></code>
<div class="block">Represents a exon or coding sequence in a gene.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a></span></code>
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span></code>
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a></span></code>
<div class="block">A gene contains a collection of Exon sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></span></code>
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/RNASequence.html" title="class in org.biojava.nbio.core.sequence">RNASequence</a></span></code>
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/StartCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StartCodonSequence</a></span></code>
<div class="block">Used to map the start codon feature on a gene</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/StopCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StopCodonSequence</a></span></code>
<div class="block">Used to map the stop codon sequence on a gene</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a></span></code>
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="typeNameLabel">SequenceComparator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/SequenceComparator.html#compare-org.biojava.nbio.core.sequence.template.AbstractSequence-org.biojava.nbio.core.sequence.template.AbstractSequence-">compare</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;o1,
       <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;o2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="typeNameLabel">SequenceComparator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/SequenceComparator.html#compare-org.biojava.nbio.core.sequence.template.AbstractSequence-org.biojava.nbio.core.sequence.template.AbstractSequence-">compare</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;o1,
       <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;o2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">ProteinSequence.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html#setParentDNASequence-org.biojava.nbio.core.sequence.template.AbstractSequence-java.lang.Integer-java.lang.Integer-">setParentDNASequence</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;parentDNASequence,
                    <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;begin,
                    <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;end)</code>
<div class="block">A Protein sequence can be stand alone or loaded from a transcript
 sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.features">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/features/package-summary.html">org.biojava.nbio.core.sequence.features</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/features/package-summary.html">org.biojava.nbio.core.sequence.features</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/features/AbstractFeature.html" title="class in org.biojava.nbio.core.sequence.features">AbstractFeature</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/features/FeatureDbReferenceInfo.html" title="class in org.biojava.nbio.core.sequence.features">FeatureDbReferenceInfo</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">It is <a href="../../../../../../../org/biojava/nbio/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava.nbio.core.sequence.features"><code>DBReferenceInfo</code></a> which implements <a href="../../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features"><code>FeatureInterface</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">Interface class to handle describing arbitrary features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QualityFeature</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">DNA Sequences produced by modern sequencers usually have quality informaion
 attached to them.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QuantityFeature</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">It is common to have a numerical value or values associated with a feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/features/TextFeature.html" title="class in org.biojava.nbio.core.sequence.features">TextFeature</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">A implmentation of AbstractFeature</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenbankReader.html" title="class in org.biojava.nbio.core.sequence.io">GenbankReader</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">Use GenbankReaderHelper as an example of how to use this class where GenbankReaderHelper should be the
 primary class used to read Genbank files</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenbankSequenceParser.html" title="class in org.biojava.nbio.core.sequence.io">GenbankSequenceParser</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenericFastaHeaderFormat.html" title="class in org.biojava.nbio.core.sequence.io">GenericFastaHeaderFormat</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">We store the original header if the sequence is parsed from a fasta file and will use that exact
 sequence if we write out the sequences to a fasta file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">GenericFastaHeaderParser</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">The default fasta header parser where some headers are well defined based on the source
 database which allows us to set the source of the protein sequence and the identifier
 that can be used in future implementations to load features from external sources

 If the user has a custom header with local data then they can create their own implementation
 of a FastaHeaderParserInterface

 GenBank                           gi|gi-number|gb|accession|locus
 ENA Data Library                  gi|gi-number|emb|accession|locus
 DDBJ, DNA Database of Japan       gi|gi-number|dbj|accession|locus
 NBRF PIR                          pir||entry
 Protein Research Foundation       prf||name
 SWISS-PROT                        sp|accession|name
 Brookhaven Protein Data Bank (1)  pdb|entry|chain
 Brookhaven Protein Data Bank (2)  entry:chain|PDBID|CHAIN|SEQUENCE
 PDB EBI                           PDB:1ECY_A mol:protein length:142  ECOTIN
 Patents                           pat|country|number
 GenInfo Backbone Id               bbs|number
 General database identifier       gnl|database|identifier
 NCBI Reference Sequence           ref|accession|locus
 Local Sequence identifier         lcl|identifier
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenericGenbankHeaderFormat.html" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderFormat</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenericGenbankHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderParser</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenericInsdcHeaderFormat.html" title="class in org.biojava.nbio.core.sequence.io">GenericInsdcHeaderFormat</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/PlainFastaHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">PlainFastaHeaderParser</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">The plain fasta header takes everything in the header as a single entity.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/ProteinSequenceCreator.html#getSequence-java.util.List-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/CasePreservingProteinSequenceCreator.html#getSequence-java.util.List-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code>
<div class="block">Assumes all compounds were uppercase</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FileProxyProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FileProxyProteinSequenceCreator.html#getSequence-java.util.List-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code>
<div class="block">Not sure of use case and currently not supported</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">DNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/DNASequenceCreator.html#getSequence-java.util.List-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;list)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">RNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/RNASequenceCreator.html#getSequence-java.util.List-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;list)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FileProxyDNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FileProxyDNASequenceCreator.html#getSequence-java.util.List-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;list)</code>
<div class="block">Not sure of use case and currently not supported</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FileProxyRNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FileProxyRNASequenceCreator.html#getSequence-java.util.List-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;list)</code>
<div class="block">Not sure of use case and currently not supported</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/ProteinSequenceCreator.html#getSequence-org.biojava.nbio.core.sequence.template.ProxySequenceReader-long-">getSequence</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/CasePreservingProteinSequenceCreator.html#getSequence-org.biojava.nbio.core.sequence.template.ProxySequenceReader-long-">getSequence</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FileProxyProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FileProxyProteinSequenceCreator.html#getSequence-org.biojava.nbio.core.sequence.template.ProxySequenceReader-long-">getSequence</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code>
<div class="block">Should be able to extend the same concept to a remote URL call or database connection.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">DNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/DNASequenceCreator.html#getSequence-org.biojava.nbio.core.sequence.template.ProxySequenceReader-long-">getSequence</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">RNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/RNASequenceCreator.html#getSequence-org.biojava.nbio.core.sequence.template.ProxySequenceReader-long-">getSequence</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FileProxyDNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FileProxyDNASequenceCreator.html#getSequence-org.biojava.nbio.core.sequence.template.ProxySequenceReader-long-">getSequence</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code>
<div class="block">Should be able to extend the same concept to a remote URL call or database connection.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FileProxyRNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FileProxyRNASequenceCreator.html#getSequence-org.biojava.nbio.core.sequence.template.ProxySequenceReader-long-">getSequence</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code>
<div class="block">Should be able to extend the same concept to a remote URL call or database connection.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">ProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/ProteinSequenceCreator.html#getSequence-java.lang.String-long-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence,
           long&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">CasePreservingProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/CasePreservingProteinSequenceCreator.html#getSequence-java.lang.String-long-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence,
           long&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">DNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/DNASequenceCreator.html#getSequence-java.lang.String-long-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence,
           long&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">RNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/RNASequenceCreator.html#getSequence-java.lang.String-long-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence,
           long&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FileProxyDNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FileProxyDNASequenceCreator.html#getSequence-java.lang.String-long-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence,
           long&nbsp;index)</code>
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence and the offset
 index in the fasta file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FileProxyRNASequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FileProxyRNASequenceCreator.html#getSequence-java.lang.String-long-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence,
           long&nbsp;index)</code>
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence and the offset
 index in the fasta file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FileProxyProteinSequenceCreator.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FileProxyProteinSequenceCreator.html#getSequence-java.lang.String-long-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence,
           long&nbsp;index)</code>
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence and the offset
 index in the fasta file.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">GenbankSequenceParser.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenbankSequenceParser.html#parseFeatures-org.biojava.nbio.core.sequence.template.AbstractSequence-">parseFeatures</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenbankSequenceParser.html" title="type parameter in GenbankSequenceParser">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with type arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="typeNameLabel">GenericInsdcHeaderFormat.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenericInsdcHeaderFormat.html#Z:Z_write_feature-org.biojava.nbio.core.sequence.features.FeatureInterface-int-">_write_feature</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenericInsdcHeaderFormat.html" title="type parameter in GenericInsdcHeaderFormat">C</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenericInsdcHeaderFormat.html" title="type parameter in GenericInsdcHeaderFormat">C</a>&gt;&nbsp;feature,
              int&nbsp;record_length)</code>
<div class="block">Write a single SeqFeature object to features table.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.io.template">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">SequenceCreatorInterface.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html#getSequence-java.util.List-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;&nbsp;list)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">SequenceCreatorInterface.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html#getSequence-org.biojava.nbio.core.sequence.template.ProxySequenceReader-long-">getSequence</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;&nbsp;proxyLoader,
           long&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">SequenceCreatorInterface.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html#getSequence-java.lang.String-long-">getSequence</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence,
           long&nbsp;index)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.loader">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a> that return types with arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenericGenbankHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderParser</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/loader/GenbankProxySequenceReader.html" title="type parameter in GenbankProxySequenceReader">C</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/loader/GenbankProxySequenceReader.html" title="type parameter in GenbankProxySequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">GenbankProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/loader/GenbankProxySequenceReader.html#getHeaderParser--">getHeaderParser</a></span>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.location">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/location/package-summary.html">org.biojava.nbio.core.sequence.location</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/location/package-summary.html">org.biojava.nbio.core.sequence.location</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/location/InsdcParser.html" title="class in org.biojava.nbio.core.sequence.location">InsdcParser</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">Parser for working with INSDC style locations.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/location/SequenceLocation.html" title="class in org.biojava.nbio.core.sequence.location">SequenceLocation</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code>
<div class="block">A location in a sequence that keeps a reference to its parent sequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.template">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getParentSequence--">getParentSequence</a></span>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> that return types with arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures--">getFeatures</a></span>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures-int-">getFeatures</a></span>(int&nbsp;bioSequencePosition)</code>
<div class="block">Return features at a sequence position</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures-java.lang.String-int-">getFeatures</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;featureType,
           int&nbsp;bioSequencePosition)</code>
<div class="block">Return features at a sequence position by type</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeaturesByType-java.lang.String-">getFeaturesByType</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setParentSequence-org.biojava.nbio.core.sequence.template.AbstractSequence-">setParentSequence</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;parentSequence)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> with type arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addFeature-org.biojava.nbio.core.sequence.features.FeatureInterface-">addFeature</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code>
<div class="block">Add a feature to this sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addFeature-int-int-org.biojava.nbio.core.sequence.features.FeatureInterface-">addFeature</a></span>(int&nbsp;bioStart,
          int&nbsp;bioEnd,
          <a href="../../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code>
<div class="block">Method to help set the proper details for a feature as it relates to a sequence
 where the feature needs to have a location on the sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#removeFeature-org.biojava.nbio.core.sequence.features.FeatureInterface-">removeFeature</a></span>(<a href="../../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code>
<div class="block">Remove a feature from the sequence</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.genome.io.fastq">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../../../../../../org/biojava/nbio/genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a> that return types with arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QuantityFeature</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FastqTools.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/genome/io/fastq/FastqTools.html#createErrorProbabilities-org.biojava.nbio.genome.io.fastq.Fastq-">createErrorProbabilities</a></span>(<a href="../../../../../../../org/biojava/nbio/genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Create and return a new <a href="../../../../../../../org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QuantityFeature</code></a> from the error probabilities of the specified
 FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QualityFeature</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">FastqTools.</span><code><span class="memberNameLink"><a href="../../../../../../../org/biojava/nbio/genome/io/fastq/FastqTools.html#createQualityScores-org.biojava.nbio.genome.io.fastq.Fastq-">createQualityScores</a></span>(<a href="../../../../../../../org/biojava/nbio/genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Create and return a new <a href="../../../../../../../org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QualityFeature</code></a> from the quality scores of the specified
 FASTQ formatted sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/core/sequence/template/class-use/AbstractSequence.html" target="_top">Frames</a></li>
<li><a href="AbstractSequence.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2018 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
