<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.16) on Fri Oct 28 16:15:48 PDT 2022 -->
<title>Uses of Class org.biojava.nbio.core.sequence.DNASequence (biojava 6.1.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2022-10-28">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class org.biojava.nbio.core.sequence.DNASequence (biojava 6.1.0 API)";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Class org.biojava.nbio.core.sequence.DNASequence" class="title">Uses of Class<br>org.biojava.nbio.core.sequence.DNASequence</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome">org.biojava.nbio.genome</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome.io.fastq">org.biojava.nbio.genome.io.fastq</a></th>
<td class="colLast">
<div class="block">FASTQ and variants sequence format I/O.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome.parsers.geneid">org.biojava.nbio.genome.parsers.geneid</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome.parsers.gff">org.biojava.nbio.genome.parsers.gff</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.genome.util">org.biojava.nbio.genome.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../package-summary.html">org.biojava.nbio.core.sequence</a></h3>
<table class="useSummary">
<caption><span>Subclasses of <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../package-summary.html">org.biojava.nbio.core.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a></span></code></th>
<td class="colLast">
<div class="block">Represents a exon or coding sequence in a gene.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a></span></code></th>
<td class="colLast">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a></span></code></th>
<td class="colLast">
<div class="block">A gene contains a collection of Exon sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../StartCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StartCodonSequence</a></span></code></th>
<td class="colLast">
<div class="block">Used to map the start codon feature on a gene</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../StopCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StopCodonSequence</a></span></code></th>
<td class="colLast">
<div class="block">Used to map the stop codon sequence on a gene</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a></span></code></th>
<td class="colLast">
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Fields in <a href="../package-summary.html">org.biojava.nbio.core.sequence</a> declared as <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StartCodonSequence.</span><code><span class="memberNameLink"><a href="../StartCodonSequence.html#parentGeneSequence">parentGeneSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StopCodonSequence.</span><code><span class="memberNameLink"><a href="../StopCodonSequence.html#parentGeneSequence">parentGeneSequence</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.core.sequence</a> that return <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TranscriptSequence.</span><code><span class="memberNameLink"><a href="../TranscriptSequence.html#getDNACodingSequence()">getDNACodingSequence</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the stitched together CDS sequences then maps to the cDNA</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneSequence.</span><code><span class="memberNameLink"><a href="../GeneSequence.html#getSequence5PrimeTo3Prime()">getSequence5PrimeTo3Prime</a></span>()</code></th>
<td class="colLast">
<div class="block">Try to give method clarity where you want a DNASequence coding in the 5' to 3' direction
 Returns the DNASequence representative of the 5' and 3' reading based on strand</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../io/package-summary.html">org.biojava.nbio.core.sequence.io</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../io/package-summary.html">org.biojava.nbio.core.sequence.io</a> that return types with arguments of type <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaReaderHelper.</span><code><span class="memberNameLink"><a href="../io/FastaReaderHelper.html#readFastaDNASequence(java.io.File)">readFastaDNASequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaReaderHelper.</span><code><span class="memberNameLink"><a href="../io/FastaReaderHelper.html#readFastaDNASequence(java.io.File,boolean)">readFastaDNASequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file,
                    boolean&nbsp;lazySequenceLoad)</code></th>
<td class="colLast">
<div class="block">Selecting lazySequenceLoad=true will parse the FASTA file and figure out the accessionid and offsets and return sequence objects
 that can in the future read the sequence from the disk.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaReaderHelper.</span><code><span class="memberNameLink"><a href="../io/FastaReaderHelper.html#readFastaDNASequence(java.io.InputStream)">readFastaDNASequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">InputStream</a>&nbsp;inStream)</code></th>
<td class="colLast">
<div class="block">Read a fasta DNA sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankReaderHelper.</span><code><span class="memberNameLink"><a href="../io/GenbankReaderHelper.html#readGenbankDNASequence(java.io.File)">readGenbankDNASequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankReaderHelper.</span><code><span class="memberNameLink"><a href="../io/GenbankReaderHelper.html#readGenbankDNASequence(java.io.File,boolean)">readGenbankDNASequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file,
                      boolean&nbsp;lazySequenceLoad)</code></th>
<td class="colLast">
<div class="block">Selecting lazySequenceLoad=true will parse the Genbank file and figure out the accessionid and offsets and return sequence objects
 that can in the future read the sequence from the disk.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankReaderHelper.</span><code><span class="memberNameLink"><a href="../io/GenbankReaderHelper.html#readGenbankDNASequence(java.io.InputStream)">readGenbankDNASequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">InputStream</a>&nbsp;inStream)</code></th>
<td class="colLast">
<div class="block">Read a Genbank DNA sequence</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with type arguments of type <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaWriterHelper.</span><code><span class="memberNameLink"><a href="../io/FastaWriterHelper.html#writeNucleotideSequence(java.io.File,java.util.Collection)">writeNucleotideSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file,
                       <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences)</code></th>
<td class="colLast">
<div class="block">Write a collection of NucleotideSequences to a file</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastaWriterHelper.</span><code><span class="memberNameLink"><a href="../io/FastaWriterHelper.html#writeNucleotideSequence(java.io.OutputStream,java.util.Collection)">writeNucleotideSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">OutputStream</a>&nbsp;outputStream,
                       <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences)</code></th>
<td class="colLast">
<div class="block">Write a collection of NucleotideSequences to a file</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankWriterHelper.</span><code><span class="memberNameLink"><a href="../io/GenbankWriterHelper.html#writeNucleotideSequence(java.io.File,java.util.Collection)">writeNucleotideSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file,
                       <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences)</code></th>
<td class="colLast">
<div class="block">Write a collection of NucleotideSequences to a file</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankWriterHelper.</span><code><span class="memberNameLink"><a href="../io/GenbankWriterHelper.html#writeNucleotideSequence(java.io.OutputStream,java.util.Collection)">writeNucleotideSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">OutputStream</a>&nbsp;outputStream,
                       <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences)</code></th>
<td class="colLast">
<div class="block">Write a collection of NucleotideSequences to a file</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankWriterHelper.</span><code><span class="memberNameLink"><a href="../io/GenbankWriterHelper.html#writeNucleotideSequence(java.io.OutputStream,java.util.Collection,java.lang.String)">writeNucleotideSequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">OutputStream</a>&nbsp;outputStream,
                       <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences,
                       <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;seqType)</code></th>
<td class="colLast">
<div class="block">Write a collection of NucleotideSequences to a file</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankWriterHelper.</span><code><span class="memberNameLink"><a href="../io/GenbankWriterHelper.html#writeNucleotideSequenceOriginal(java.io.OutputStream,java.util.Collection)">writeNucleotideSequenceOriginal</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">OutputStream</a>&nbsp;outputStream,
                               <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences)</code></th>
<td class="colLast">
<div class="block">Write a collection of NucleotideSequences to a file using the NucleotideSequences
 original header as the LOCUS line rather than generating it</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.genome">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../../../genome/package-summary.html">org.biojava.nbio.genome</a></h3>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../genome/package-summary.html">org.biojava.nbio.genome</a> with type arguments of type <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="../ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneFeatureHelper.</span><code><span class="memberNameLink"><a href="../../../genome/GeneFeatureHelper.html#getChromosomeSequenceFromDNASequence(java.util.LinkedHashMap)">getChromosomeSequenceFromDNASequence</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequenceList)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.genome.io.fastq">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../../../genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a> that return <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastqTools.</span><code><span class="memberNameLink"><a href="../../../genome/io/fastq/FastqTools.html#createDNASequence(org.biojava.nbio.genome.io.fastq.Fastq)">createDNASequence</a></span>&#8203;(<a href="../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></th>
<td class="colLast">
<div class="block">Create and return a new <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> from the specified FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastqTools.</span><code><span class="memberNameLink"><a href="../../../genome/io/fastq/FastqTools.html#createDNASequenceWithErrorProbabilities(org.biojava.nbio.genome.io.fastq.Fastq)">createDNASequenceWithErrorProbabilities</a></span>&#8203;(<a href="../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></th>
<td class="colLast">
<div class="block">Create and return a new <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with error probabilities from the specified
 FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastqTools.</span><code><span class="memberNameLink"><a href="../../../genome/io/fastq/FastqTools.html#createDNASequenceWithQualityScores(org.biojava.nbio.genome.io.fastq.Fastq)">createDNASequenceWithQualityScores</a></span>&#8203;(<a href="../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></th>
<td class="colLast">
<div class="block">Create and return a new <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores from the specified
 FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FastqTools.</span><code><span class="memberNameLink"><a href="../../../genome/io/fastq/FastqTools.html#createDNASequenceWithQualityScoresAndErrorProbabilities(org.biojava.nbio.genome.io.fastq.Fastq)">createDNASequenceWithQualityScoresAndErrorProbabilities</a></span>&#8203;(<a href="../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></th>
<td class="colLast">
<div class="block">Create and return a new <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores and error probabilities from the
 specified FASTQ formatted sequence.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.genome.parsers.geneid">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../../../genome/parsers/geneid/package-summary.html">org.biojava.nbio.genome.parsers.geneid</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genome/parsers/geneid/package-summary.html">org.biojava.nbio.genome.parsers.geneid</a> that return types with arguments of type <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedHashMap</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>,&#8203;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GeneIDXMLReader.</span><code><span class="memberNameLink"><a href="../../../genome/parsers/geneid/GeneIDXMLReader.html#getDNACodingSequences()">getDNACodingSequences</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.genome.parsers.gff">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../../../genome/parsers/gff/package-summary.html">org.biojava.nbio.genome.parsers.gff</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genome/parsers/gff/package-summary.html">org.biojava.nbio.genome.parsers.gff</a> with parameters of type <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FeatureList.</span><code><span class="memberNameLink"><a href="../../../genome/parsers/gff/FeatureList.html#splice(org.biojava.nbio.core.sequence.DNASequence)">splice</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Concatenate successive portions of the specified sequence
 using the feature locations in the list.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../genome/parsers/gff/package-summary.html">org.biojava.nbio.genome.parsers.gff</a> with type arguments of type <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GCStats.</span><code><span class="memberNameLink"><a href="../../../genome/parsers/gff/GCStats.html#getGCStats(java.util.Collection)">getGCStats</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;sequences)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.genome.util">
<!--   -->
</a>
<h3>Uses of <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../../../genome/util/package-summary.html">org.biojava.nbio.genome.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genome/util/package-summary.html">org.biojava.nbio.genome.util</a> that return <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeMappingTools.</span><code><span class="memberNameLink"><a href="../../../genome/util/ChromosomeMappingTools.html#getTranscriptDNASequence(org.biojava.nbio.genome.parsers.twobit.TwoBitFacade,java.lang.String,java.util.List,java.util.List,int,int,java.lang.Character)">getTranscriptDNASequence</a></span>&#8203;(<a href="../../../genome/parsers/twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a>&nbsp;twoBitFacade,
                        <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chromosome,
                        <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonStarts,
                        <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;exonEnds,
                        int&nbsp;cdsStart,
                        int&nbsp;cdsEnd,
                        <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Character.html?is-external=true" title="class or interface in java.lang" class="externalLink">Character</a>&nbsp;orientation)</code></th>
<td class="colLast">
<div class="block">Extracts the DNA sequence transcribed from the input genetic coordinates.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ChromosomeMappingTools.</span><code><span class="memberNameLink"><a href="../../../genome/util/ChromosomeMappingTools.html#getTranscriptDNASequence(org.biojava.nbio.genome.parsers.twobit.TwoBitFacade,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">getTranscriptDNASequence</a></span>&#8203;(<a href="../../../genome/parsers/twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a>&nbsp;twoBitFacade,
                        <a href="../../../genome/parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;gcp)</code></th>
<td class="colLast">
<div class="block">Extracts the DNA sequence transcribed from the input genetic coordinates.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../genome/util/package-summary.html">org.biojava.nbio.genome.util</a> with parameters of type <a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ProteinMappingTools.</span><code><span class="memberNameLink"><a href="../../../genome/util/ProteinMappingTools.html#convertDNAtoProteinSequence(org.biojava.nbio.core.sequence.DNASequence)">convertDNAtoProteinSequence</a></span>&#8203;(<a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;dnaSequence)</code></th>
<td class="colLast">
<div class="block">Converts the DNA sequence to protein sequence.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../DNASequence.html" title="class in org.biojava.nbio.core.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2022 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
