<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.11) on Fri Oct 29 16:26:12 PDT 2021 -->
<title>Uses of Interface org.biojava.nbio.core.sequence.template.Compound (biojava 6.0.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2021-10-29">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../../jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Interface org.biojava.nbio.core.sequence.template.Compound (biojava 6.0.0 API)";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Interface org.biojava.nbio.core.sequence.template.Compound" class="title">Uses of Interface<br>org.biojava.nbio.core.sequence.template.Compound</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.alignment">org.biojava.nbio.alignment</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.alignment.routines">org.biojava.nbio.alignment.routines</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.alignment.template">org.biojava.nbio.alignment.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.alignment">org.biojava.nbio.core.alignment</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.alignment.matrices">org.biojava.nbio.core.alignment.matrices</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.alignment.template">org.biojava.nbio.core.alignment.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.search.io">org.biojava.nbio.core.search.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.compound">org.biojava.nbio.core.sequence.compound</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.edits">org.biojava.nbio.core.sequence.edits</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.features">org.biojava.nbio.core.sequence.features</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.io.template">org.biojava.nbio.core.sequence.io.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.io.util">org.biojava.nbio.core.sequence.io.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.loader">org.biojava.nbio.core.sequence.loader</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.location">org.biojava.nbio.core.sequence.location</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.location.template">org.biojava.nbio.core.sequence.location.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.storage">org.biojava.nbio.core.sequence.storage</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.transcription">org.biojava.nbio.core.sequence.transcription</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.core.sequence.views">org.biojava.nbio.core.sequence.views</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.phylo">org.biojava.nbio.phylo</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.ws.alignment">org.biojava.nbio.ws.alignment</a></th>
<td class="colLast">
<div class="block">This package contains the interfaces that need to be implemented by discrete alignment services.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.ws.alignment.qblast">org.biojava.nbio.ws.alignment.qblast</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../../../alignment/package-summary.html">org.biojava.nbio.alignment</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../../alignment/package-summary.html">org.biojava.nbio.alignment</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/FractionalIdentityInProfileScorer.html" title="class in org.biojava.nbio.alignment">FractionalIdentityInProfileScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair picked from an alignment
 <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/FractionalIdentityScorer.html" title="class in org.biojava.nbio.alignment">FractionalIdentityScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/FractionalSimilarityInProfileScorer.html" title="class in org.biojava.nbio.alignment">FractionalSimilarityInProfileScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair picked from an alignment
 <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/FractionalSimilarityScorer.html" title="class in org.biojava.nbio.alignment">FractionalSimilarityScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/GuideTree.html" title="class in org.biojava.nbio.alignment">GuideTree</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a data structure for a guide tree used during progressive multiple sequence alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/NeedlemanWunsch.html" title="class in org.biojava.nbio.alignment">NeedlemanWunsch</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Needleman and Wunsch defined an algorithm for pairwise global sequence alignments (from the first until the last
 <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/SimpleProfileProfileAligner.html" title="class in org.biojava.nbio.alignment">SimpleProfileProfileAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a simple (naive) <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/SmithWaterman.html" title="class in org.biojava.nbio.alignment">SmithWaterman</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Smith and Waterman defined an algorithm for pairwise local sequence alignments (best match of sections from each
 <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/StandardRescoreRefiner.html" title="class in org.biojava.nbio.alignment">StandardRescoreRefiner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/SubstitutionMatrixScorer.html" title="class in org.biojava.nbio.alignment">SubstitutionMatrixScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Scores using a substitution matrix.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../alignment/package-summary.html">org.biojava.nbio.alignment</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;S,&#8203;C&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getAllPairsAlignments(java.util.List,org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getAllPairsAlignments</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences,
                     <a href="../../../../alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;type,
                     <a href="../../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                     <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code></th>
<td class="colLast">
<div class="block">Factory method which computes a sequence alignment for all <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pairs in the given <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../../alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceScorer</a>&lt;S,&#8203;C&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getAllPairsScorers(java.util.List,org.biojava.nbio.alignment.Alignments.PairwiseSequenceScorerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getAllPairsScorers</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences,
                  <a href="../../../../alignment/Alignments.PairwiseSequenceScorerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceScorerType</a>&nbsp;type,
                  <a href="../../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                  <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code></th>
<td class="colLast">
<div class="block">Factory method which sets up a sequence pair scorer for all <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pairs in the given <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>double[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getAllPairsScores(java.util.List,org.biojava.nbio.alignment.Alignments.PairwiseSequenceScorerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getAllPairsScores</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences,
                 <a href="../../../../alignment/Alignments.PairwiseSequenceScorerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceScorerType</a>&nbsp;type,
                 <a href="../../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                 <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code></th>
<td class="colLast">
<div class="block">Factory method which computes a sequence pair score for all <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pairs in the given <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;S,&#8203;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getMultipleSequenceAlignment(java.util.List,java.lang.Object...)">getMultipleSequenceAlignment</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences,
                            <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>...&nbsp;settings)</code></th>
<td class="colLast">
<div class="block">Factory method which computes a multiple sequence alignment for the given <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a> of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../../../../alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceAligner</a>&lt;S,&#8203;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getPairwiseAligner(S,S,org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getPairwiseAligner</a></span>&#8203;(S&nbsp;query,
                  S&nbsp;target,
                  <a href="../../../../alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;type,
                  <a href="../../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                  <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code></th>
<td class="colLast">
<div class="block">Factory method which constructs a pairwise sequence aligner.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../../../alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;S,&#8203;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getPairwiseAlignment(S,S,org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getPairwiseAlignment</a></span>&#8203;(S&nbsp;query,
                    S&nbsp;target,
                    <a href="../../../../alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;type,
                    <a href="../../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                    <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code></th>
<td class="colLast">
<div class="block">Factory method which computes a sequence alignment for the given <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pair.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;S,&#8203;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#getProgressiveAlignment(org.biojava.nbio.alignment.GuideTree,org.biojava.nbio.alignment.Alignments.ProfileProfileAlignerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getProgressiveAlignment</a></span>&#8203;(<a href="../../../../alignment/GuideTree.html" title="class in org.biojava.nbio.alignment">GuideTree</a>&lt;S,&#8203;C&gt;&nbsp;tree,
                       <a href="../../../../alignment/Alignments.ProfileProfileAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.ProfileProfileAlignerType</a>&nbsp;type,
                       <a href="../../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                       <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code></th>
<td class="colLast">
<div class="block">Factory method to run the profile-profile alignments of a progressive multiple sequence alignment concurrently.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>double[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Alignments.</span><code><span class="memberNameLink"><a href="../../../../alignment/Alignments.html#runPairwiseScorers(java.util.List)">runPairwiseScorers</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../../alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceScorer</a>&lt;S,&#8203;C&gt;&gt;&nbsp;scorers)</code></th>
<td class="colLast">
<div class="block">Factory method to run a list of scorers concurrently.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.alignment.routines">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../../../alignment/routines/package-summary.html">org.biojava.nbio.alignment.routines</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../../alignment/routines/package-summary.html">org.biojava.nbio.alignment.routines</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/routines/AnchoredPairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.routines">AnchoredPairwiseSequenceAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">This algorithm uses a divide-and-conquer approach to find optimal pairwise global sequence alignments (from the
 first until the last <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>) with the restriction that any alignment produced
 will connect the query sequence to the target sequence at the <em>anchors</em>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/routines/GuanUberbacher.html" title="class in org.biojava.nbio.alignment.routines">GuanUberbacher</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Guan and Uberbacher defined an algorithm for pairwise global sequence alignments (from the first until the last
 <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.alignment.template">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../../../alignment/template/package-summary.html">org.biojava.nbio.alignment.template</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../../alignment/template/package-summary.html">org.biojava.nbio.alignment.template</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/AbstractMatrixAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractMatrixAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements common code for an <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> which builds a score matrix during computation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/AbstractPairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractPairwiseSequenceAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements common code for an <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/AbstractProfileProfileAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractProfileProfileAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements common code for an <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template">Aligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an algorithm which computes an alignment <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a> from a list of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/CallablePairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.template">CallablePairwiseSequenceAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a concurrency wrapper for a <a href="../../../../alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template"><code>PairwiseSequenceAligner</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/CallablePairwiseSequenceScorer.html" title="class in org.biojava.nbio.alignment.template">CallablePairwiseSequenceScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a concurrency wrapper for a <a href="../../../../alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template"><code>PairwiseSequenceScorer</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/CallableProfileProfileAligner.html" title="class in org.biojava.nbio.alignment.template">CallableProfileProfileAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a concurrency wrapper for a <a href="../../../../alignment/template/ProfileProfileAligner.html" title="interface in org.biojava.nbio.alignment.template"><code>ProfileProfileAligner</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/GuideTreeNode.html" title="interface in org.biojava.nbio.alignment.template">GuideTreeNode</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for the node in a guide tree used during progressive multiple sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/HierarchicalClusterer.html" title="interface in org.biojava.nbio.alignment.template">HierarchicalClusterer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a clustering algorithm that converts a distance matrix into a tree.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/MatrixAligner.html" title="interface in org.biojava.nbio.alignment.template">MatrixAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> which builds a score matrix during computation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/PairInProfileScorer.html" title="interface in org.biojava.nbio.alignment.template">PairInProfileScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an algorithm which computes a score for a sequence alignment pair picked from an alignment <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an algorithm which computes a score for a pair of sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/PartitionRefiner.html" title="interface in org.biojava.nbio.alignment.template">PartitionRefiner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an algorithm which computes a new alignment <code>Profile</code> by splitting a current alignment and realigning.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/ProfileProfileAligner.html" title="interface in org.biojava.nbio.alignment.template">ProfileProfileAligner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an <a href="../../../../alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <code>Profile</code>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/ProfileProfileScorer.html" title="interface in org.biojava.nbio.alignment.template">ProfileProfileScorer</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an algorithm which computes a score for a pairing of alignment profiles.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../alignment/template/RescoreRefiner.html" title="interface in org.biojava.nbio.alignment.template">RescoreRefiner</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines an algorithm which computes a new alignment <code>Profile</code> by rescoring all pairs in an alignment and
 realigning.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../../alignment/package-summary.html">org.biojava.nbio.core.alignment</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../alignment/package-summary.html">org.biojava.nbio.core.alignment</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/SimpleAlignedSequence.html" title="class in org.biojava.nbio.core.alignment">SimpleAlignedSequence</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a data structure for a <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/SimpleProfile.html" title="class in org.biojava.nbio.core.alignment">SimpleProfile</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a data structure for the results of sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/SimpleProfilePair.html" title="class in org.biojava.nbio.core.alignment">SimpleProfilePair</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a data structure for the results of the alignment of a pair of <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/SimpleSequencePair.html" title="class in org.biojava.nbio.core.alignment">SimpleSequencePair</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a data structure for the results of pairwise sequence alignment.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../alignment/package-summary.html">org.biojava.nbio.core.alignment</a> with parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SimpleAlignedSequence.</span><code><span class="memberNameLink"><a href="../../../alignment/SimpleAlignedSequence.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../../alignment/SimpleAlignedSequence.html" title="type parameter in SimpleAlignedSequence">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.alignment.matrices">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../../alignment/matrices/package-summary.html">org.biojava.nbio.core.alignment.matrices</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../alignment/matrices/package-summary.html">org.biojava.nbio.core.alignment.matrices</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/matrices/SimpleSubstitutionMatrix.html" title="class in org.biojava.nbio.core.alignment.matrices">SimpleSubstitutionMatrix</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a data structure which holds the score (penalty or bonus) given during alignment for the exchange of one
 <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> in a sequence for another.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.alignment.template">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../../alignment/template/package-summary.html">org.biojava.nbio.core.alignment.template</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../alignment/template/package-summary.html">org.biojava.nbio.core.alignment.template</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">AlignedSequence</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for a <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/MutableAlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">MutableAlignedSequence</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a mutable (editable) data structure for an <a href="../../../alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template"><code>AlignedSequence</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/MutableProfile.html" title="interface in org.biojava.nbio.core.alignment.template">MutableProfile</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a mutable (editable) data structure for a <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/MutableProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">MutableProfilePair</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a mutable (editable) data structure for a <a href="../../../alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template"><code>ProfilePair</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/MutableSequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">MutableSequencePair</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a mutable (editable) data structure for the results of pairwise sequence alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for the results of sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for the results of the alignment of a pair of <a href="../../../alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/ProfileView.html" title="interface in org.biojava.nbio.core.alignment.template">ProfileView</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for a view of sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure for the results of pairwise sequence alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a data structure which holds the score (penalty or bonus) given during alignment for the exchange of one
 <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> in a sequence for another.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.search.io">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../../search/io/package-summary.html">org.biojava.nbio.core.search.io</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../search/io/package-summary.html">org.biojava.nbio.core.search.io</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../search/io/Hsp.html" title="class in org.biojava.nbio.core.search.io">Hsp</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">This class models a search Hsp.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../package-summary.html">org.biojava.nbio.core.sequence</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../package-summary.html">org.biojava.nbio.core.sequence</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../BasicSequence.html" title="class in org.biojava.nbio.core.sequence">BasicSequence</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Bare bones version of the Sequence object to be used sparingly.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implements a minimal data structure for reading and writing a sequence alignment.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.compound">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../compound/package-summary.html">org.biojava.nbio.core.sequence.compound</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../compound/package-summary.html">org.biojava.nbio.core.sequence.compound</a> that implement <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a></span></code></th>
<td class="colLast">
<div class="block">Used to describe an Amino Acid.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/CodonCompound.html" title="class in org.biojava.nbio.core.sequence.compound">CodonCompound</a></span></code></th>
<td class="colLast">
<div class="block">Define a codon</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../compound/package-summary.html">org.biojava.nbio.core.sequence.compound</a> with parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidCompound.</span><code><span class="memberNameLink"><a href="../../compound/AminoAcidCompound.html#equalsIgnoreCase(org.biojava.nbio.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CodonCompound.</span><code><span class="memberNameLink"><a href="../../compound/CodonCompound.html#equalsIgnoreCase(org.biojava.nbio.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NucleotideCompound.</span><code><span class="memberNameLink"><a href="../../compound/NucleotideCompound.html#equalsIgnoreCase(org.biojava.nbio.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.edits">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../edits/package-summary.html">org.biojava.nbio.core.sequence.edits</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../edits/package-summary.html">org.biojava.nbio.core.sequence.edits</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.html" title="interface in org.biojava.nbio.core.sequence.edits">Edit</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Interface for carrying out edit operations on a Sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.AbstractEdit.html" title="class in org.biojava.nbio.core.sequence.edits">Edit.AbstractEdit</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Abstract class which defines all edit operations as a call to discover
 what 5' and 3' ends of an editing Sequence should be joined together
 with a target Sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.Delete.html" title="class in org.biojava.nbio.core.sequence.edits">Edit.Delete</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Implementation which allows for the deletion of bases from a Sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.Insert.html" title="class in org.biojava.nbio.core.sequence.edits">Edit.Insert</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Edit implementation which allows us to insert a base at any position
 in a Sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../edits/Edit.Substitute.html" title="class in org.biojava.nbio.core.sequence.edits">Edit.Substitute</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Allows for the substitution of bases into an existing Sequence.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.features">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../features/package-summary.html">org.biojava.nbio.core.sequence.features</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../features/package-summary.html">org.biojava.nbio.core.sequence.features</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/AbstractFeature.html" title="class in org.biojava.nbio.core.sequence.features">AbstractFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/FeatureDbReferenceInfo.html" title="class in org.biojava.nbio.core.sequence.features">FeatureDbReferenceInfo</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">It is <a href="../../features/DBReferenceInfo.html" title="class in org.biojava.nbio.core.sequence.features"><code>DBReferenceInfo</code></a> which implements <a href="../../features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features"><code>FeatureInterface</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Interface class to handle describing arbitrary features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/FeatureRetriever.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureRetriever</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list features</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QualityFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">DNA Sequences produced by modern sequencers usually have quality informaion
 attached to them.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QuantityFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">It is common to have a numerical value or values associated with a feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../features/TextFeature.html" title="class in org.biojava.nbio.core.sequence.features">TextFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A implmentation of AbstractFeature</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../io/package-summary.html">org.biojava.nbio.core.sequence.io</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/FastaReader.html" title="class in org.biojava.nbio.core.sequence.io">FastaReader</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be the
 primary class used to read Fasta files</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/FastaWriter.html" title="class in org.biojava.nbio.core.sequence.io">FastaWriter</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The FastaWriter writes a collection of sequences to an outputStream.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenbankReader.html" title="class in org.biojava.nbio.core.sequence.io">GenbankReader</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Use <a href="../../io/GenbankReaderHelper.html" title="class in org.biojava.nbio.core.sequence.io"><code>GenbankReaderHelper</code></a> as an example of how to use this class where <a href="../../io/GenbankReaderHelper.html" title="class in org.biojava.nbio.core.sequence.io"><code>GenbankReaderHelper</code></a> should be the
 primary class used to read Genbank files</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenbankSequenceParser.html" title="class in org.biojava.nbio.core.sequence.io">GenbankSequenceParser</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenbankWriter.html" title="class in org.biojava.nbio.core.sequence.io">GenbankWriter</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenericFastaHeaderFormat.html" title="class in org.biojava.nbio.core.sequence.io">GenericFastaHeaderFormat</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">We store the original header if the sequence is parsed from a fasta file and will use that exact
 sequence if we write out the sequences to a fasta file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenericFastaHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">GenericFastaHeaderParser</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The default fasta header parser where some headers are well defined based on the source
 database which allows us to set the source of the protein sequence and the identifier
 that can be used in future implementations to load features from external sources

 If the user has a custom header with local data then they can create their own implementation
 of a FastaHeaderParserInterface</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenericGenbankHeaderFormat.html" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderFormat</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenericGenbankHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderParser</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/GenericInsdcHeaderFormat.html" title="class in org.biojava.nbio.core.sequence.io">GenericInsdcHeaderFormat</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/PlainFastaHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">PlainFastaHeaderParser</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The plain fasta header takes everything in the header as a single entity.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.io.template">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/template/FastaHeaderFormatInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">FastaHeaderFormatInterface</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/template/GenbankHeaderFormatInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">GenbankHeaderFormatInterface</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.io.util">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../io/util/package-summary.html">org.biojava.nbio.core.sequence.io.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/util/package-summary.html">org.biojava.nbio.core.sequence.io.util</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getGCGChecksum(java.util.List)">getGCGChecksum</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Calculates GCG checksum for entire list of sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getGCGChecksum(S)">getGCGChecksum</a></span>&#8203;(S&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Calculates GCG checksum for a given sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getGCGHeader(java.util.List)">getGCGHeader</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Assembles a GCG file header</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getGCGType(org.biojava.nbio.core.sequence.template.CompoundSet)">getGCGType</a></span>&#8203;(<a href="../CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;C&gt;&nbsp;cs)</code></th>
<td class="colLast">
<div class="block">Determines GCG type</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">IOUtils.</span><code><span class="memberNameLink"><a href="../../io/util/IOUtils.html#getIDFormat(java.util.List)">getIDFormat</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;S&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Creates format String for accession IDs</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.loader">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/ArrayListProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">ArrayListProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/GenbankProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">GenbankProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/SequenceFileProxyLoader.html" title="class in org.biojava.nbio.core.sequence.loader">SequenceFileProxyLoader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">This class represents the storage container of a sequence stored in a fasta file where
 the initial parsing of the file we store the offset and length of the sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/StringProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">StringProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">An example of a ProxySequenceReader that is created from a String.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../loader/UniprotProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">UniprotProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Pass in a Uniprot ID and this ProxySequenceReader when passed to a ProteinSequence will get the sequence data and other data elements
 associated with the ProteinSequence by Uniprot.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../../loader/UniprotProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">UniprotProxySequenceReader</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">UniprotProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../loader/UniprotProxySequenceReader.html#parseUniprotXMLString(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)">parseUniprotXMLString</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;xml,
                     <a href="../CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;C&gt;&nbsp;compoundSet)</code></th>
<td class="colLast">
<div class="block">The passed in xml is parsed as a DOM object so we know everything about the protein.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a> with parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceFileProxyLoader.</span><code><span class="memberNameLink"><a href="../../loader/SequenceFileProxyLoader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StringProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../loader/StringProxySequenceReader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">UniprotProxySequenceReader.</span><code><span class="memberNameLink"><a href="../../loader/UniprotProxySequenceReader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.location">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../location/package-summary.html">org.biojava.nbio.core.sequence.location</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../location/package-summary.html">org.biojava.nbio.core.sequence.location</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../location/SequenceLocation.html" title="class in org.biojava.nbio.core.sequence.location">SequenceLocation</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A location in a sequence that keeps a reference to its parent sequence</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.location.template">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../location/template/package-summary.html">org.biojava.nbio.core.sequence.location.template</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../location/template/package-summary.html">org.biojava.nbio.core.sequence.location.template</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#canComplement(org.biojava.nbio.core.sequence.template.Sequence)">canComplement</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Uses the Sequence's CompoundSet to decide if a compound can
 be assgined to ComplementCompound meaning it can complement</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#getRelevantSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getRelevantSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Location.</span><code><span class="memberNameLink"><a href="../../location/template/Location.html#getRelevantSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getRelevantSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Will return a SequenceReader object which offers a view of all resolved
 locations i.e. those locations which are not complex and define the
 true Sequence represented</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#getSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">If circular this will return the sequence represented by the sub
 locations joined.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Location.</span><code><span class="memberNameLink"><a href="../../location/template/Location.html#getSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Will return a SequenceReader object which represents the outer bounds
 of this Location</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AbstractLocation.html#reverseSequence(org.biojava.nbio.core.sequence.template.Sequence)">reverseSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Reverses and (if possible) complements the Sequence so as to represent
 the reverse strand (if one exists).</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../location/template/package-summary.html">org.biojava.nbio.core.sequence.location.template</a> that return types with arguments of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;? extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AccesionedLocation.</span><code><span class="memberNameLink"><a href="../../location/template/AccesionedLocation.html#getProxySequenceReader()">getProxySequenceReader</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the proxy reader used to get sequence for this location.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.storage">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/ArrayListSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">ArrayListSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Stores a Sequence as a collection of compounds in an ArrayList</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">BitSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">An implementation of the popular bit encodings.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava.nbio.core.sequence.storage">BitSequenceReader.BitArrayWorker</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The logic of working with a bit has been separated out into this class
 to help developers create the bit data structures without having to
 put the code into an intermediate format and to also use the format
 without the need to copy this code.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/FourBitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">FourBitSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Four bit encoding of the bit formats.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/FourBitSequenceReader.FourBitArrayWorker.html" title="class in org.biojava.nbio.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A four bit per compound implementation of the bit array worker code.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">JoiningSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">This reader actually proxies onto multiple types of sequence in order
 to allow a number of sequence objects to act as if they are one sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/SequenceAsStringHelper.html" title="class in org.biojava.nbio.core.sequence.storage">SequenceAsStringHelper</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">This is a common method that can be used across multiple storage/proxy implementations to
 handle Negative strand and other interesting elements of sequence data.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../storage/SingleCompoundSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">SingleCompoundSequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">An implementation of the SequenceReader interface which for every
 call will return only 1 compound (given to it during construction; a String
 is also valid but will require a CompoundSet).</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> with parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ArrayListSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/ArrayListSequenceReader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BitSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/BitSequenceReader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">
<div class="block">Counts the number of times a compound appears in this sequence store</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">JoiningSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/JoiningSequenceReader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SingleCompoundSequenceReader.</span><code><span class="memberNameLink"><a href="../../storage/SingleCompoundSequenceReader.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../../storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">
<div class="block">Delegates to <code>SequenceMixin#countCompounds(org.biojava.nbio.core.sequence.template.Sequence, C[])</code></div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.template">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractCompoundSet.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompoundSet</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompoundTranslator</a>&lt;F extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>,&#8203;T extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompoundTranslator</a>&lt;F extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>,&#8203;T extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">The base class for DNA, RNA and Protein sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../CompoundTranslator.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundTranslator</a>&lt;F extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>,&#8203;T extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../CompoundTranslator.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundTranslator</a>&lt;F extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>,&#8203;T extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../LightweightProfile.html" title="interface in org.biojava.nbio.core.sequence.template">LightweightProfile</a>&lt;S extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,&#8203;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Defines a minimal data structure for reading and writing a sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">Main interface for defining a collection of Compounds and accessing them
 using biological indexes</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceMixin.SequenceIterator.html" title="class in org.biojava.nbio.core.sequence.template">SequenceMixin.SequenceIterator</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A basic sequence iterator which iterates over the given Sequence by
 biological index.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceProxyView.html" title="class in org.biojava.nbio.core.sequence.template">SequenceProxyView</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Subinterfaces of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../ComplementCompound.html" title="interface in org.biojava.nbio.core.sequence.template">ComplementCompound</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Classes in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> that implement <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractCompound.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompound</a></span></code></th>
<td class="colLast">
<div class="block">The details of a Compound</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#checksum(org.biojava.nbio.core.sequence.template.Sequence)">checksum</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Performs a simple CRC64 checksum on any given sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#countCompounds(org.biojava.nbio.core.sequence.template.Sequence,C...)">countCompounds</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
              C...&nbsp;compounds)</code></th>
<td class="colLast">
<div class="block">For the given vargs of compounds this method counts the number of
 times those compounds appear in the given sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html?is-external=true" title="class or interface in java.util" class="externalLink">Iterator</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#createIterator(org.biojava.nbio.core.sequence.template.Sequence)">createIterator</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Creates a simple sequence iterator which moves through a sequence going
 from 1 to the length of the Sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#createSubSequence(org.biojava.nbio.core.sequence.template.Sequence,int,int)">createSubSequence</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                 int&nbsp;start,
                 int&nbsp;end)</code></th>
<td class="colLast">
<div class="block">Creates a simple sub sequence view delimited by the given start and end.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;C,&#8203;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#getComposition(org.biojava.nbio.core.sequence.template.Sequence)">getComposition</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Does a linear scan over the given Sequence and records the number of
 times each base appears.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;C,&#8203;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang" class="externalLink">Double</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#getDistribution(org.biojava.nbio.core.sequence.template.Sequence)">getDistribution</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Analogous to <a href="../SequenceMixin.html#getComposition(org.biojava.nbio.core.sequence.template.Sequence)"><code>SequenceMixin.getComposition(Sequence)</code></a> but returns the
 distribution of that <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> over the given sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#indexOf(org.biojava.nbio.core.sequence.template.Sequence,C)">indexOf</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
       C&nbsp;compound)</code></th>
<td class="colLast">
<div class="block">Performs a linear search of the given Sequence for the given compound.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#inverse(org.biojava.nbio.core.sequence.template.Sequence)">inverse</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">A method which attempts to do the right thing when is comes to a
 reverse/reverse complement</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#lastIndexOf(org.biojava.nbio.core.sequence.template.Sequence,C)">lastIndexOf</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
           C&nbsp;compound)</code></th>
<td class="colLast">
<div class="block">Performs a reversed linear search of the given Sequence by wrapping
 it in a <a href="../../views/ReversedSequenceView.html" title="class in org.biojava.nbio.core.sequence.views"><code>ReversedSequenceView</code></a> and passing it into
 <a href="../SequenceMixin.html#indexOf(org.biojava.nbio.core.sequence.template.Sequence,C)"><code>SequenceMixin.indexOf(Sequence, Compound)</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#nonOverlappingKmers(org.biojava.nbio.core.sequence.template.Sequence,int)">nonOverlappingKmers</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                   int&nbsp;kmer)</code></th>
<td class="colLast">
<div class="block">Produces kmers of the specified size e.g.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#overlappingKmers(org.biojava.nbio.core.sequence.template.Sequence,int)">overlappingKmers</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                int&nbsp;kmer)</code></th>
<td class="colLast">
<div class="block">Used to generate overlapping k-mers such i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#sequenceEquality(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.core.sequence.template.Sequence)">sequenceEquality</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;source,
                <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;target)</code></th>
<td class="colLast">
<div class="block">A case-sensitive manner of comparing two sequence objects together.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#sequenceEqualityIgnoreCase(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.core.sequence.template.Sequence)">sequenceEqualityIgnoreCase</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;source,
                          <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;target)</code></th>
<td class="colLast">
<div class="block">A case-insensitive manner of comparing two sequence objects together.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#shuffle(org.biojava.nbio.core.sequence.template.Sequence)">shuffle</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Implements sequence shuffling by first materializing the given
 <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> into a <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a>, applying
 <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html?is-external=true#shuffle(java.util.List)" title="class or interface in java.util" class="externalLink"><code>Collections.shuffle(List)</code></a> and then returning the shuffled
 elements in a new instance of <code>SequenceBackingStore</code> which behaves
 as a <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;C&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#toList(org.biojava.nbio.core.sequence.template.Sequence)">toList</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">For the given <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> this will return a <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink"><code>List</code></a> filled with
 the Compounds of that <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#toString(org.biojava.nbio.core.sequence.template.Sequence)">toString</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Shortcut to <a href="../SequenceMixin.html#toStringBuilder(org.biojava.nbio.core.sequence.template.Sequence)"><code>SequenceMixin.toStringBuilder(org.biojava.nbio.core.sequence.template.Sequence)</code></a>
 which calls toString() on the resulting object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html?is-external=true" title="class or interface in java.lang" class="externalLink">StringBuilder</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#toStringBuilder(org.biojava.nbio.core.sequence.template.Sequence)">toStringBuilder</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">For the given Sequence this will return a <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html?is-external=true" title="class or interface in java.lang" class="externalLink"><code>StringBuilder</code></a> object
 filled with the results of <code>Compound#toString()</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#write(java.lang.Appendable,org.biojava.nbio.core.sequence.template.Sequence)">write</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Appendable.html?is-external=true" title="class or interface in java.lang" class="externalLink">Appendable</a>&nbsp;appendable,
     <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Used as a way of sending a Sequence to a writer without the cost of
 converting to a full length String and then writing the data out</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> with parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractCompoundSet.</span><code><span class="memberNameLink"><a href="../AbstractCompoundSet.html#addCompound(C,C,C...)">addCompound</a></span>&#8203;(<a href="../AbstractCompoundSet.html" title="type parameter in AbstractCompoundSet">C</a>&nbsp;compound,
           <a href="../AbstractCompoundSet.html" title="type parameter in AbstractCompoundSet">C</a>&nbsp;lowerCasedCompound,
           <a href="../AbstractCompoundSet.html" title="type parameter in AbstractCompoundSet">C</a>...&nbsp;equivalents)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractCompoundTranslator.</span><code><span class="memberNameLink"><a href="../AbstractCompoundTranslator.html#addCompounds(F,T...)">addCompounds</a></span>&#8203;(<a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&nbsp;source,
            <a href="../AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>...&nbsp;targets)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractSequence.</span><code><span class="memberNameLink"><a href="../AbstractSequence.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Sequence.</span><code><span class="memberNameLink"><a href="../Sequence.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../Sequence.html" title="type parameter in Sequence">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">
<div class="block">Returns the number of times we found a compound in the Sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceMixin.</span><code><span class="memberNameLink"><a href="../SequenceMixin.html#countCompounds(org.biojava.nbio.core.sequence.template.Sequence,C...)">countCompounds</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
              C...&nbsp;compounds)</code></th>
<td class="colLast">
<div class="block">For the given vargs of compounds this method counts the number of
 times those compounds appear in the given sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceProxyView.</span><code><span class="memberNameLink"><a href="../SequenceProxyView.html#countCompounds(C...)">countCompounds</a></span>&#8203;(<a href="../SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a>...&nbsp;compounds)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractCompound.</span><code><span class="memberNameLink"><a href="../AbstractCompound.html#equalsIgnoreCase(org.biojava.nbio.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Compound.</span><code><span class="memberNameLink"><a href="../Compound.html#equalsIgnoreCase(org.biojava.nbio.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.transcription">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../transcription/package-summary.html">org.biojava.nbio.core.sequence.transcription</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../transcription/package-summary.html">org.biojava.nbio.core.sequence.transcription</a> that implement <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../transcription/CaseInsensitiveCompound.html" title="class in org.biojava.nbio.core.sequence.transcription">CaseInsensitiveCompound</a></span></code></th>
<td class="colLast">
<div class="block">Attempts to wrap compounds so it is possible to view them
 in a case insensitive manner</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../transcription/Table.Codon.html" title="class in org.biojava.nbio.core.sequence.transcription">Table.Codon</a></span></code></th>
<td class="colLast">
<div class="block">Instance of a Codon which is 3 <a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound"><code>NucleotideCompound</code></a>s, its
 corresponding <a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound"><code>AminoAcidCompound</code></a> and if it is a start or stop codon.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../transcription/package-summary.html">org.biojava.nbio.core.sequence.transcription</a> with parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CaseInsensitiveCompound.</span><code><span class="memberNameLink"><a href="../../transcription/CaseInsensitiveCompound.html#equalsIgnoreCase(org.biojava.nbio.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Table.Codon.</span><code><span class="memberNameLink"><a href="../../transcription/Table.Codon.html#equalsIgnoreCase(org.biojava.nbio.core.sequence.template.Compound)">equalsIgnoreCase</a></span>&#8203;(<a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&nbsp;compound)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.core.sequence.views">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../views/package-summary.html">org.biojava.nbio.core.sequence.views</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../views/package-summary.html">org.biojava.nbio.core.sequence.views</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/ReversedSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ReversedSequenceView</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">For a given sequence this class will return the base at the reversed
 position i.e. in a sequence of size 10, if you request base 2 you will get
 back the base at position 9.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../views/WindowedSequence.html" title="class in org.biojava.nbio.core.sequence.views">WindowedSequence</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span></code></th>
<td class="colLast">
<div class="block">A sliding window view of a sequence which does not implement any
 interfaces like <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> because they do not fit how this works.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.phylo">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../../../phylo/package-summary.html">org.biojava.nbio.phylo</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../../phylo/package-summary.html">org.biojava.nbio.phylo</a> with type parameters of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.msa.Msa</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ForesterWrapper.</span><code><span class="memberNameLink"><a href="../../../../phylo/ForesterWrapper.html#convert(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">convert</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">Convert a BioJava <a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence"><code>MultipleSequenceAlignment</code></a> to a forester
 <code>Msa</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#dissimilarityScore(org.biojava.nbio.core.sequence.MultipleSequenceAlignment,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">dissimilarityScore</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa,
                  <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;D&gt;&nbsp;M)</code></th>
<td class="colLast">
<div class="block">The dissimilarity score is the additive inverse of the similarity score
 (sum of scores) between two aligned sequences using a substitution model
 (Substitution Matrix).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#fractionalDissimilarity(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">fractionalDissimilarity</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">The fractional dissimilarity (D) is defined as the percentage of sites
 that differ between two aligned sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#fractionalDissimilarityScore(org.biojava.nbio.core.sequence.MultipleSequenceAlignment,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">fractionalDissimilarityScore</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa,
                            <a href="../../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;D&gt;&nbsp;M)</code></th>
<td class="colLast">
<div class="block">The fractional dissimilarity score (Ds) is a relative measure of the
 dissimilarity between two aligned sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#kimuraDistance(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">kimuraDistance</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">The Kimura evolutionary distance (d) is a correction of the fractional
 dissimilarity (D) specially needed for large evolutionary distances.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#pamMLdistance(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">pamMLdistance</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">The PAM (Point Accepted Mutations) distance is a measure of evolutionary
 distance in protein sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#percentageIdentity(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">percentageIdentity</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">BioJava implementation for percentage of identity (PID).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#poissonDistance(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">poissonDistance</a></span>&#8203;(<a href="../../MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,&#8203;D&gt;&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">The Poisson (correction) evolutionary distance (d) is a function of the
 fractional dissimilarity (D), given by:</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,&#8203;D extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DistanceMatrixCalculator.</span><code><span class="memberNameLink"><a href="../../../../phylo/DistanceMatrixCalculator.html#structuralDistance(double%5B%5D%5B%5D,double,double,double)">structuralDistance</a></span>&#8203;(double[][]&nbsp;rmsdMat,
                  double&nbsp;alpha,
                  double&nbsp;rmsdMax,
                  double&nbsp;rmsd0)</code></th>
<td class="colLast">
<div class="block">The structural distance (d<sub>S</sub>) uses the structural similarity
 (or dissimilarity) from a the structural alignment of two protein
 strutures.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.ws.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../../../ws/alignment/package-summary.html">org.biojava.nbio.ws.alignment</a></h3>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../../ws/alignment/package-summary.html">org.biojava.nbio.ws.alignment</a> with type arguments of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RemotePairwiseAlignmentService.</span><code><span class="memberNameLink"><a href="../../../../ws/alignment/RemotePairwiseAlignmentService.html#sendAlignmentRequest(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.ws.alignment.RemotePairwiseAlignmentProperties)">sendAlignmentRequest</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;seq,
                    <a href="../../../../ws/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojava.nbio.ws.alignment">RemotePairwiseAlignmentProperties</a>&nbsp;rpa)</code></th>
<td class="colLast">
<div class="block">Doing the actual analysis on the instantiated service using specified parameters and the RichSequence object</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.ws.alignment.qblast">
<!--   -->
</a>
<h3>Uses of <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a> in <a href="../../../../ws/alignment/qblast/package-summary.html">org.biojava.nbio.ws.alignment.qblast</a></h3>
<table class="useSummary">
<caption><span>Method parameters in <a href="../../../../ws/alignment/qblast/package-summary.html">org.biojava.nbio.ws.alignment.qblast</a> with type arguments of type <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NCBIQBlastService.</span><code><span class="memberNameLink"><a href="../../../../ws/alignment/qblast/NCBIQBlastService.html#sendAlignmentRequest(org.biojava.nbio.core.sequence.template.Sequence,org.biojava.nbio.ws.alignment.RemotePairwiseAlignmentProperties)">sendAlignmentRequest</a></span>&#8203;(<a href="../Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;seq,
                    <a href="../../../../ws/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojava.nbio.ws.alignment">RemotePairwiseAlignmentProperties</a>&nbsp;rpa)</code></th>
<td class="colLast">
<div class="block">Converts given sequence to String and calls
 <a href="../../../../ws/alignment/qblast/NCBIQBlastService.html#sendAlignmentRequest(java.lang.String,org.biojava.nbio.ws.alignment.RemotePairwiseAlignmentProperties)"><code>NCBIQBlastService.sendAlignmentRequest(String, RemotePairwiseAlignmentProperties)</code></a></div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2021 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
