<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span>package org.biojava.nbio.genome.uniprot;<a name="line.21"></a>
<span class="sourceLineNo">022</span><a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import org.biojava.nbio.core.sequence.AccessionID;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.nbio.core.sequence.ProteinSequence;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.core.sequence.io.FastaWriterHelper;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.slf4j.Logger;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.slf4j.LoggerFactory;<a name="line.28"></a>
<span class="sourceLineNo">029</span><a name="line.29"></a>
<span class="sourceLineNo">030</span>import java.io.BufferedReader;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import java.io.File;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import java.io.FileReader;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import java.util.ArrayList;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import java.util.HashMap;<a name="line.34"></a>
<span class="sourceLineNo">035</span><a name="line.35"></a>
<span class="sourceLineNo">036</span>/**<a name="line.36"></a>
<span class="sourceLineNo">037</span> *<a name="line.37"></a>
<span class="sourceLineNo">038</span> * @author Scooter<a name="line.38"></a>
<span class="sourceLineNo">039</span> */<a name="line.39"></a>
<span class="sourceLineNo">040</span>public class UniprotToFasta {<a name="line.40"></a>
<span class="sourceLineNo">041</span><a name="line.41"></a>
<span class="sourceLineNo">042</span>        private static final Logger logger = LoggerFactory.getLogger(UniprotToFasta.class);<a name="line.42"></a>
<span class="sourceLineNo">043</span><a name="line.43"></a>
<span class="sourceLineNo">044</span>        public static void main( String[] args ){<a name="line.44"></a>
<span class="sourceLineNo">045</span>                try{<a name="line.45"></a>
<span class="sourceLineNo">046</span>                        String uniprotDatFileName = "uniprot_trembl_fungi.dat";<a name="line.46"></a>
<span class="sourceLineNo">047</span>                        String fastaFileName = "uniprot__trembel_fungi.faa";<a name="line.47"></a>
<span class="sourceLineNo">048</span>                        UniprotToFasta uniprotToFasta = new UniprotToFasta();<a name="line.48"></a>
<span class="sourceLineNo">049</span>                        uniprotToFasta.process(uniprotDatFileName, fastaFileName);<a name="line.49"></a>
<span class="sourceLineNo">050</span>                }catch(Exception e){<a name="line.50"></a>
<span class="sourceLineNo">051</span>                        logger.error("Exception: ", e);<a name="line.51"></a>
<span class="sourceLineNo">052</span>                }<a name="line.52"></a>
<span class="sourceLineNo">053</span>        }<a name="line.53"></a>
<span class="sourceLineNo">054</span><a name="line.54"></a>
<span class="sourceLineNo">055</span>        /**<a name="line.55"></a>
<span class="sourceLineNo">056</span>         * Convert a Uniprot sequence file to a fasta file. Allows you to download all sequence data for a species<a name="line.56"></a>
<span class="sourceLineNo">057</span>         * and convert to fasta to be used in a blast database<a name="line.57"></a>
<span class="sourceLineNo">058</span>         * @param uniprotDatFileName<a name="line.58"></a>
<span class="sourceLineNo">059</span>         * @param fastaFileName<a name="line.59"></a>
<span class="sourceLineNo">060</span>         * @throws Exception<a name="line.60"></a>
<span class="sourceLineNo">061</span>         */<a name="line.61"></a>
<span class="sourceLineNo">062</span><a name="line.62"></a>
<span class="sourceLineNo">063</span>        public void process( String uniprotDatFileName,String fastaFileName ) throws Exception{<a name="line.63"></a>
<span class="sourceLineNo">064</span><a name="line.64"></a>
<span class="sourceLineNo">065</span>                        FileReader fr = new FileReader(uniprotDatFileName);<a name="line.65"></a>
<span class="sourceLineNo">066</span>                        BufferedReader br = new BufferedReader(fr);<a name="line.66"></a>
<span class="sourceLineNo">067</span>                        String line = br.readLine();<a name="line.67"></a>
<span class="sourceLineNo">068</span>                        String id = "";<a name="line.68"></a>
<span class="sourceLineNo">069</span>                        StringBuffer sequence = new StringBuffer();<a name="line.69"></a>
<span class="sourceLineNo">070</span>                        ArrayList&lt;ProteinSequence&gt; seqCodingRegionsList = new ArrayList&lt;ProteinSequence&gt;();<a name="line.70"></a>
<span class="sourceLineNo">071</span>                        int count = 0;<a name="line.71"></a>
<span class="sourceLineNo">072</span>                        HashMap&lt;String,String&gt; uniqueGenes = new HashMap&lt;String,String&gt;();<a name="line.72"></a>
<span class="sourceLineNo">073</span>                        HashMap&lt;String,String&gt; uniqueSpecies = new HashMap&lt;String,String&gt;();<a name="line.73"></a>
<span class="sourceLineNo">074</span>                        while(line != null){<a name="line.74"></a>
<span class="sourceLineNo">075</span>                                if(line.startsWith("ID")){<a name="line.75"></a>
<span class="sourceLineNo">076</span>                                        String[] data = line.split(" ");<a name="line.76"></a>
<span class="sourceLineNo">077</span>                                        id = data[3];<a name="line.77"></a>
<span class="sourceLineNo">078</span>                                }else if(line.startsWith("SQ")){<a name="line.78"></a>
<span class="sourceLineNo">079</span>                                        line = br.readLine();<a name="line.79"></a>
<span class="sourceLineNo">080</span>                                        while(!line.startsWith("//")){<a name="line.80"></a>
<span class="sourceLineNo">081</span><a name="line.81"></a>
<span class="sourceLineNo">082</span>                                                for(int i = 0; i &lt; line.length(); i++){<a name="line.82"></a>
<span class="sourceLineNo">083</span>                                                        char aa = line.charAt(i);<a name="line.83"></a>
<span class="sourceLineNo">084</span>                                                        if((aa &gt;= 'A' &amp;&amp; aa &lt;= 'Z') || (aa &gt;= 'a' &amp;&amp; aa &lt;= 'z' )){<a name="line.84"></a>
<span class="sourceLineNo">085</span>                                                                sequence.append(aa);<a name="line.85"></a>
<span class="sourceLineNo">086</span>                                                        }<a name="line.86"></a>
<span class="sourceLineNo">087</span>                                                }<a name="line.87"></a>
<span class="sourceLineNo">088</span>                                                line = br.readLine();<a name="line.88"></a>
<span class="sourceLineNo">089</span>                                        }<a name="line.89"></a>
<span class="sourceLineNo">090</span><a name="line.90"></a>
<span class="sourceLineNo">091</span>                                 //   System.out.println("&gt;" + id);<a name="line.91"></a>
<span class="sourceLineNo">092</span>                                 //   System.out.println(sequence.toString());<a name="line.92"></a>
<span class="sourceLineNo">093</span><a name="line.93"></a>
<span class="sourceLineNo">094</span>                                        ProteinSequence seq = new ProteinSequence(sequence.toString() );<a name="line.94"></a>
<span class="sourceLineNo">095</span>                                        seq.setAccession(new AccessionID(id));<a name="line.95"></a>
<span class="sourceLineNo">096</span><a name="line.96"></a>
<span class="sourceLineNo">097</span>                                        seqCodingRegionsList.add(seq);<a name="line.97"></a>
<span class="sourceLineNo">098</span>                                        sequence = new StringBuffer();<a name="line.98"></a>
<span class="sourceLineNo">099</span>                                        count++;<a name="line.99"></a>
<span class="sourceLineNo">100</span>                                        if(count % 100 == 0)<a name="line.100"></a>
<span class="sourceLineNo">101</span>                                                logger.info("Count: ", count);<a name="line.101"></a>
<span class="sourceLineNo">102</span>                                        String[] parts = id.split("_");<a name="line.102"></a>
<span class="sourceLineNo">103</span>                                        uniqueGenes.put(parts[0], "");<a name="line.103"></a>
<span class="sourceLineNo">104</span>                                        uniqueSpecies.put(parts[1],"");<a name="line.104"></a>
<span class="sourceLineNo">105</span>                                }<a name="line.105"></a>
<span class="sourceLineNo">106</span>                                line = br.readLine();<a name="line.106"></a>
<span class="sourceLineNo">107</span>                        }<a name="line.107"></a>
<span class="sourceLineNo">108</span>           //     System.out.println("Unique Genes=" + uniqueGenes.size());<a name="line.108"></a>
<span class="sourceLineNo">109</span>           //     System.out.println("Unique Species=" + uniqueSpecies.size());<a name="line.109"></a>
<span class="sourceLineNo">110</span>           //     System.out.println("Total sequences=" + seqCodingRegionsList.size());<a name="line.110"></a>
<span class="sourceLineNo">111</span>                        FastaWriterHelper.writeProteinSequence(new File(fastaFileName), seqCodingRegionsList);<a name="line.111"></a>
<span class="sourceLineNo">112</span><a name="line.112"></a>
<span class="sourceLineNo">113</span>                        br.close();<a name="line.113"></a>
<span class="sourceLineNo">114</span>                        fr.close();<a name="line.114"></a>
<span class="sourceLineNo">115</span><a name="line.115"></a>
<span class="sourceLineNo">116</span>          //      System.out.println(uniqueGenes.keySet());<a name="line.116"></a>
<span class="sourceLineNo">117</span>          //      System.out.println("====================");<a name="line.117"></a>
<span class="sourceLineNo">118</span>          //      System.out.println(uniqueSpecies.keySet());<a name="line.118"></a>
<span class="sourceLineNo">119</span><a name="line.119"></a>
<span class="sourceLineNo">120</span><a name="line.120"></a>
<span class="sourceLineNo">121</span>        }<a name="line.121"></a>
<span class="sourceLineNo">122</span><a name="line.122"></a>
<span class="sourceLineNo">123</span>}<a name="line.123"></a>




























































</pre>
</div>
</body>
</html>
