<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.1) on Thu Jan 03 22:38:55 CET 2019 -->
<title>Jronn (biojava 5.2.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2019-01-03">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Jronn (biojava 5.2.0 API)";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":9,"i1":9,"i2":9,"i3":9,"i4":9,"i5":9,"i6":9,"i7":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Jronn.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">org.biojava.nbio.ronn</a></div>
<h2 title="Class Jronn" class="title">Class Jronn</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.nbio.ronn.Jronn</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code><a href="https://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io" class="externalLink">Serializable</a></code></dd>
</dl>
<hr>
<pre>public class <a href="../../../../src-html/org/biojava/nbio/ronn/Jronn.html#line.50">Jronn</a>
extends <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>
implements <a href="https://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io" class="externalLink">Serializable</a></pre>
<div class="block">This class gives public API to RONN functions.
 It is build on top of the command line client. Due to this fact a few things
 could be improved and extended pending the refactoring of the command line client.

 The input sequence limitations - the input sequence must not contain any ambiguous characters,
 and have a minimum length of 19 amino acids.</div>
<dl>
<dt><span class="simpleTagLabel">Since:</span></dt>
<dd>3.0.2</dd>
<dt><span class="simpleTagLabel">Version:</span></dt>
<dd>1.0</dd>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Peter Troshin</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../../serialized-form.html#org.biojava.nbio.ronn.Jronn">Serialized Form</a></dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="memberSummary">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a></span></code></th>
<td class="colLast">
<div class="block">Holder for the ranges, contain pointers to starting and ending position
 on the sequence which comprises a disordered region.</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">Jronn</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static <a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#convertProteinSequencetoFasta(org.biojava.nbio.core.sequence.ProteinSequence)">convertProteinSequencetoFasta</a></span>&#8203;(<a href="../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Utility method to convert a BioJava ProteinSequence object to the FastaSequence
  object used internally in JRonn.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>,&#8203;<a href="Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDisorder(java.lang.String)">getDisorder</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;fastaFile)</code></th>
<td class="colLast">
<div class="block">Calculates the disordered regions of the protein sequence.</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>,&#8203;<a href="Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDisorder(java.util.List)">getDisorder</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Calculates the disordered regions of the sequence for many sequences in the input.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static <a href="Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDisorder(org.biojava.nbio.data.sequence.FastaSequence)">getDisorder</a></span>&#8203;(<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Calculates the disordered regions of the sequence.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>,&#8203;float[]&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDisorderScores(java.util.List)">getDisorderScores</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&gt;&nbsp;sequences)</code></th>
<td class="colLast">
<div class="block">Calculates the probability of disorder scores for each residue in the sequence for
 many sequences in the input.</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static float[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDisorderScores(org.biojava.nbio.core.sequence.ProteinSequence)">getDisorderScores</a></span>&#8203;(<a href="../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Calculates the probability value for each residue in the protein sequence,
 telling the probability that the residue belongs to disordered region.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>static float[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDisorderScores(org.biojava.nbio.data.sequence.FastaSequence)">getDisorderScores</a></span>&#8203;(<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&nbsp;sequence)</code></th>
<td class="colLast">
<div class="block">Calculates the probability value for each residue in the protein sequence,
 telling the probability that the residue belongs to disordered region.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>static <a href="Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#scoresToRanges(float%5B%5D,float)">scoresToRanges</a></span>&#8203;(float[]&nbsp;scores,
              float&nbsp;probability)</code></th>
<td class="colLast">
<div class="block">Convert raw scores to ranges.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a></h3>
<code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang" class="externalLink">clone</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang" class="externalLink">equals</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang" class="externalLink">finalize</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang" class="externalLink">getClass</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang" class="externalLink">hashCode</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang" class="externalLink">notify</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang" class="externalLink">notifyAll</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang" class="externalLink">toString</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang" class="externalLink">wait</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang" class="externalLink">wait</a>, <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true#wait(long,int)" title="class or interface in java.lang" class="externalLink">wait</a></code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>Jronn</h4>
<pre>public&nbsp;<a href="../../../../src-html/org/biojava/nbio/ronn/Jronn.html#line.50">Jronn</a>()</pre>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="getDisorderScores(org.biojava.nbio.data.sequence.FastaSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisorderScores</h4>
<pre class="methodSignature">public static&nbsp;float[]&nbsp;<a href="../../../../src-html/org/biojava/nbio/ronn/Jronn.html#line.134">getDisorderScores</a>&#8203;(<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&nbsp;sequence)</pre>
<div class="block">Calculates the probability value for each residue in the protein sequence,
 telling the probability that the residue belongs to disordered region.
 In general, values greater than 0.5 considered to be in the disordered regions.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequence</code> - an instance of FastaSequence object, holding the name and the sequence.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the probability scores for each residue in the sequence</dd>
</dl>
</li>
</ul>
<a id="getDisorderScores(org.biojava.nbio.core.sequence.ProteinSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisorderScores</h4>
<pre class="methodSignature">public static&nbsp;float[]&nbsp;<a href="../../../../src-html/org/biojava/nbio/ronn/Jronn.html#line.146">getDisorderScores</a>&#8203;(<a href="../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</pre>
<div class="block">Calculates the probability value for each residue in the protein sequence,
 telling the probability that the residue belongs to disordered region.
 In general, values greater than 0.5 considered to be in the disordered regions.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequence</code> - an instance of FastaSequence object, holding the name and the sequence.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the probability scores for each residue in the sequence</dd>
</dl>
</li>
</ul>
<a id="convertProteinSequencetoFasta(org.biojava.nbio.core.sequence.ProteinSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>convertProteinSequencetoFasta</h4>
<pre class="methodSignature">public static&nbsp;<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&nbsp;<a href="../../../../src-html/org/biojava/nbio/ronn/Jronn.html#line.159">convertProteinSequencetoFasta</a>&#8203;(<a href="../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</pre>
<div class="block">Utility method to convert a BioJava ProteinSequence object to the FastaSequence
  object used internally in JRonn.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequence</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a id="getDisorder(org.biojava.nbio.data.sequence.FastaSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisorder</h4>
<pre class="methodSignature">public static&nbsp;<a href="Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]&nbsp;<a href="../../../../src-html/org/biojava/nbio/ronn/Jronn.html#line.200">getDisorder</a>&#8203;(<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&nbsp;sequence)</pre>
<div class="block">Calculates the disordered regions of the sequence. More formally, the regions for which the
 probability of disorder is greater then 0.50.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequence</code> - an instance of FastaSequence object, holding the name and the sequence.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the array of ranges if there are any residues predicted to have the
 probability of disorder greater then 0.5, null otherwise.</dd>
</dl>
</li>
</ul>
<a id="scoresToRanges(float[],float)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>scoresToRanges</h4>
<pre class="methodSignature">public static&nbsp;<a href="Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]&nbsp;<a href="../../../../src-html/org/biojava/nbio/ronn/Jronn.html#line.212">scoresToRanges</a>&#8203;(float[]&nbsp;scores,
                                           float&nbsp;probability)</pre>
<div class="block">Convert raw scores to ranges. Gives ranges for given probability of disorder value</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>scores</code> - the raw probability of disorder scores for each residue in the sequence.</dd>
<dd><code>probability</code> - the cut off threshold. Include all residues with the probability of disorder greater then this value</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the array of ranges if there are any residues predicted to have the
 probability of disorder greater then <code>probability</code>, null otherwise.</dd>
</dl>
</li>
</ul>
<a id="getDisorderScores(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisorderScores</h4>
<pre class="methodSignature">public static&nbsp;<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>,&#8203;float[]&gt;&nbsp;<a href="../../../../src-html/org/biojava/nbio/ronn/Jronn.html#line.248">getDisorderScores</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&gt;&nbsp;sequences)</pre>
<div class="block">Calculates the probability of disorder scores for each residue in the sequence for
 many sequences in the input.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequences</code> - the list of the FastaSequence objects</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the Map with key->FastaSequence, value->probability of disorder for each residue</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getDisorder(org.biojava.nbio.data.sequence.FastaSequence)"><code>getDisorder(FastaSequence)</code></a></dd>
</dl>
</li>
</ul>
<a id="getDisorder(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisorder</h4>
<pre class="methodSignature">public static&nbsp;<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>,&#8203;<a href="Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]&gt;&nbsp;<a href="../../../../src-html/org/biojava/nbio/ronn/Jronn.html#line.263">getDisorder</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&gt;&nbsp;sequences)</pre>
<div class="block">Calculates the disordered regions of the sequence for many sequences in the input.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>sequences</code> - sequences the list of the FastaSequence objects</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getDisorder(org.biojava.nbio.data.sequence.FastaSequence)"><code>getDisorder(FastaSequence)</code></a></dd>
</dl>
</li>
</ul>
<a id="getDisorder(java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getDisorder</h4>
<pre class="methodSignature">public static&nbsp;<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="../data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>,&#8203;<a href="Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]&gt;&nbsp;<a href="../../../../src-html/org/biojava/nbio/ronn/Jronn.html#line.280">getDisorder</a>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;fastaFile)
                                                          throws <a href="https://docs.oracle.com/javase/8/docs/api/java/io/FileNotFoundException.html?is-external=true" title="class or interface in java.io" class="externalLink">FileNotFoundException</a>,
                                                                 <a href="https://docs.oracle.com/javase/8/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io" class="externalLink">IOException</a></pre>
<div class="block">Calculates the disordered regions of the protein sequence.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fastaFile</code> - input file name containing the sequence in FASTA</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the Map with key->FastaSequence, value->the list of disordered regions for each sequence</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="https://docs.oracle.com/javase/8/docs/api/java/io/FileNotFoundException.html?is-external=true" title="class or interface in java.io" class="externalLink">FileNotFoundException</a></code> - if the input file cannot be found</dd>
<dd><code><a href="https://docs.oracle.com/javase/8/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io" class="externalLink">IOException</a></code> - of the system cannot access or read from the input file</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getDisorder(org.biojava.nbio.data.sequence.FastaSequence)"><code>getDisorder(FastaSequence)</code></a></dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Jronn.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2019 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
