<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * @author Scooter Willis ;lt;willishf at gmail dot com&amp;gt;<a name="line.14"></a>
<span class="sourceLineNo">015</span> * @author Karl Nicholas &lt;github:karlnicholas&gt;<a name="line.15"></a>
<span class="sourceLineNo">016</span> * @author Paolo Pavan<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> * For more information on the BioJava project and its aims,<a name="line.18"></a>
<span class="sourceLineNo">019</span> * or to join the biojava-l mailing list, visit the home page<a name="line.19"></a>
<span class="sourceLineNo">020</span> * at:<a name="line.20"></a>
<span class="sourceLineNo">021</span> *<a name="line.21"></a>
<span class="sourceLineNo">022</span> *      http://www.biojava.org/<a name="line.22"></a>
<span class="sourceLineNo">023</span> *<a name="line.23"></a>
<span class="sourceLineNo">024</span> * Created on 01-21-2010<a name="line.24"></a>
<span class="sourceLineNo">025</span> */<a name="line.25"></a>
<span class="sourceLineNo">026</span>package org.biojava.nbio.core.sequence.io;<a name="line.26"></a>
<span class="sourceLineNo">027</span><a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.nbio.core.exceptions.CompoundNotFoundException;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.biojava.nbio.core.sequence.DNASequence;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import org.biojava.nbio.core.sequence.DataSource;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import org.biojava.nbio.core.sequence.ProteinSequence;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import org.biojava.nbio.core.sequence.TaxonomyID;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import org.biojava.nbio.core.sequence.compound.DNACompoundSet;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import org.biojava.nbio.core.sequence.features.AbstractFeature;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import org.biojava.nbio.core.sequence.features.DBReferenceInfo;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;<a name="line.39"></a>
<span class="sourceLineNo">040</span>import org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface;<a name="line.40"></a>
<span class="sourceLineNo">041</span>import org.biojava.nbio.core.sequence.template.AbstractSequence;<a name="line.41"></a>
<span class="sourceLineNo">042</span>import org.biojava.nbio.core.sequence.template.Compound;<a name="line.42"></a>
<span class="sourceLineNo">043</span>import org.slf4j.Logger;<a name="line.43"></a>
<span class="sourceLineNo">044</span>import org.slf4j.LoggerFactory;<a name="line.44"></a>
<span class="sourceLineNo">045</span><a name="line.45"></a>
<span class="sourceLineNo">046</span>import java.io.*;<a name="line.46"></a>
<span class="sourceLineNo">047</span>import java.util.ArrayList;<a name="line.47"></a>
<span class="sourceLineNo">048</span>import java.util.HashMap;<a name="line.48"></a>
<span class="sourceLineNo">049</span>import java.util.LinkedHashMap;<a name="line.49"></a>
<span class="sourceLineNo">050</span><a name="line.50"></a>
<span class="sourceLineNo">051</span>/**<a name="line.51"></a>
<span class="sourceLineNo">052</span> * Use GenbankReaderHelper as an example of how to use this class where GenbankReaderHelper should be the<a name="line.52"></a>
<span class="sourceLineNo">053</span> * primary class used to read Genbank files<a name="line.53"></a>
<span class="sourceLineNo">054</span> *<a name="line.54"></a>
<span class="sourceLineNo">055</span> */<a name="line.55"></a>
<span class="sourceLineNo">056</span>public class GenbankReader&lt;S extends AbstractSequence&lt;C&gt;, C extends Compound&gt; {<a name="line.56"></a>
<span class="sourceLineNo">057</span><a name="line.57"></a>
<span class="sourceLineNo">058</span>        private SequenceCreatorInterface&lt;C&gt; sequenceCreator;<a name="line.58"></a>
<span class="sourceLineNo">059</span>        private GenbankSequenceParser&lt;S,C&gt; genbankParser;<a name="line.59"></a>
<span class="sourceLineNo">060</span>        private BufferedReader bufferedReader;<a name="line.60"></a>
<span class="sourceLineNo">061</span>        private boolean closed;<a name="line.61"></a>
<span class="sourceLineNo">062</span>        private final Logger logger = LoggerFactory.getLogger(this.getClass());<a name="line.62"></a>
<span class="sourceLineNo">063</span><a name="line.63"></a>
<span class="sourceLineNo">064</span>        public boolean isClosed() {<a name="line.64"></a>
<span class="sourceLineNo">065</span>                return closed;<a name="line.65"></a>
<span class="sourceLineNo">066</span>        }<a name="line.66"></a>
<span class="sourceLineNo">067</span><a name="line.67"></a>
<span class="sourceLineNo">068</span>        /**<a name="line.68"></a>
<span class="sourceLineNo">069</span>         * If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because we need details about<a name="line.69"></a>
<span class="sourceLineNo">070</span>         * local file offsets for quick reads. InputStreams does not give you the name of the stream to access quickly via file seek. A seek in<a name="line.70"></a>
<span class="sourceLineNo">071</span>         * an inputstream is forced to read all the data so you don't gain anything.<a name="line.71"></a>
<span class="sourceLineNo">072</span>         * @param is<a name="line.72"></a>
<span class="sourceLineNo">073</span>         * @param headerParser<a name="line.73"></a>
<span class="sourceLineNo">074</span>         * @param sequenceCreator<a name="line.74"></a>
<span class="sourceLineNo">075</span>         */<a name="line.75"></a>
<span class="sourceLineNo">076</span>        public GenbankReader(final InputStream is, final SequenceHeaderParserInterface&lt;S,C&gt; headerParser,<a name="line.76"></a>
<span class="sourceLineNo">077</span>                                                 final SequenceCreatorInterface&lt;C&gt; sequenceCreator) {<a name="line.77"></a>
<span class="sourceLineNo">078</span>                this.sequenceCreator = sequenceCreator;<a name="line.78"></a>
<span class="sourceLineNo">079</span>                bufferedReader = new BufferedReader(new InputStreamReader(is));<a name="line.79"></a>
<span class="sourceLineNo">080</span>                genbankParser = new GenbankSequenceParser&lt;&gt;();<a name="line.80"></a>
<span class="sourceLineNo">081</span>                closed = false;<a name="line.81"></a>
<span class="sourceLineNo">082</span>        }<a name="line.82"></a>
<span class="sourceLineNo">083</span><a name="line.83"></a>
<span class="sourceLineNo">084</span>        /**<a name="line.84"></a>
<span class="sourceLineNo">085</span>         * If you are going to use the FileProxyProteinSequenceCreator then you<a name="line.85"></a>
<span class="sourceLineNo">086</span>         * need to use this constructor because we need details about<a name="line.86"></a>
<span class="sourceLineNo">087</span>         * the location of the file.<a name="line.87"></a>
<span class="sourceLineNo">088</span>         * @param file<a name="line.88"></a>
<span class="sourceLineNo">089</span>         * @param headerParser<a name="line.89"></a>
<span class="sourceLineNo">090</span>         * @param sequenceCreator<a name="line.90"></a>
<span class="sourceLineNo">091</span>         * @throws FileNotFoundException if the file does not exist, is a directory<a name="line.91"></a>
<span class="sourceLineNo">092</span>         *      rather than a regular file, or for some other reason cannot be opened<a name="line.92"></a>
<span class="sourceLineNo">093</span>         *      for reading.<a name="line.93"></a>
<span class="sourceLineNo">094</span>         * @throws SecurityException if a security manager exists and its checkRead<a name="line.94"></a>
<span class="sourceLineNo">095</span>         *      method denies read access to the file.<a name="line.95"></a>
<span class="sourceLineNo">096</span>         */<a name="line.96"></a>
<span class="sourceLineNo">097</span>        public GenbankReader(<a name="line.97"></a>
<span class="sourceLineNo">098</span>                        final File file,<a name="line.98"></a>
<span class="sourceLineNo">099</span>                        final SequenceHeaderParserInterface&lt;S,C&gt; headerParser,<a name="line.99"></a>
<span class="sourceLineNo">100</span>                        final SequenceCreatorInterface&lt;C&gt; sequenceCreator<a name="line.100"></a>
<span class="sourceLineNo">101</span>                        ) throws FileNotFoundException {<a name="line.101"></a>
<span class="sourceLineNo">102</span><a name="line.102"></a>
<span class="sourceLineNo">103</span>                this.bufferedReader = new BufferedReader(new FileReader(file));<a name="line.103"></a>
<span class="sourceLineNo">104</span>                this.sequenceCreator = sequenceCreator;<a name="line.104"></a>
<span class="sourceLineNo">105</span>                genbankParser = new GenbankSequenceParser&lt;&gt;();<a name="line.105"></a>
<span class="sourceLineNo">106</span>        }<a name="line.106"></a>
<span class="sourceLineNo">107</span><a name="line.107"></a>
<span class="sourceLineNo">108</span>        /**<a name="line.108"></a>
<span class="sourceLineNo">109</span>         * The parsing is done in this method.&lt;br&gt;<a name="line.109"></a>
<span class="sourceLineNo">110</span>         * This method tries to process all the available Genbank records<a name="line.110"></a>
<span class="sourceLineNo">111</span>         * in the File or InputStream, closes the underlying resource,<a name="line.111"></a>
<span class="sourceLineNo">112</span>         * and return the results in {@link LinkedHashMap}.&lt;br&gt;<a name="line.112"></a>
<span class="sourceLineNo">113</span>         * You don't need to call {@link #close()} after calling this method.<a name="line.113"></a>
<span class="sourceLineNo">114</span>         * @see #process(int)<a name="line.114"></a>
<span class="sourceLineNo">115</span>         * @return {@link HashMap} containing all the parsed Genbank records<a name="line.115"></a>
<span class="sourceLineNo">116</span>         * present, starting current fileIndex onwards.<a name="line.116"></a>
<span class="sourceLineNo">117</span>         * @throws IOException<a name="line.117"></a>
<span class="sourceLineNo">118</span>         * @throws CompoundNotFoundException<a name="line.118"></a>
<span class="sourceLineNo">119</span>         */<a name="line.119"></a>
<span class="sourceLineNo">120</span>        public LinkedHashMap&lt;String,S&gt; process() throws IOException, CompoundNotFoundException {<a name="line.120"></a>
<span class="sourceLineNo">121</span>                return process(-1);<a name="line.121"></a>
<span class="sourceLineNo">122</span>        }<a name="line.122"></a>
<span class="sourceLineNo">123</span><a name="line.123"></a>
<span class="sourceLineNo">124</span>        /**<a name="line.124"></a>
<span class="sourceLineNo">125</span>         * This method tries to parse maximum &lt;code&gt;max&lt;/code&gt; records from<a name="line.125"></a>
<span class="sourceLineNo">126</span>         * the open File or InputStream, and leaves the underlying resource open.&lt;br&gt;<a name="line.126"></a>
<span class="sourceLineNo">127</span>         *<a name="line.127"></a>
<span class="sourceLineNo">128</span>         * Subsequent calls to the same method continue parsing the rest of the file.&lt;br&gt;<a name="line.128"></a>
<span class="sourceLineNo">129</span>         * This is particularly useful when dealing with very big data files,<a name="line.129"></a>
<span class="sourceLineNo">130</span>         * (e.g. NCBI nr database), which can't fit into memory and will take long<a name="line.130"></a>
<span class="sourceLineNo">131</span>         * time before the first result is available.&lt;br&gt;<a name="line.131"></a>
<span class="sourceLineNo">132</span>         * &lt;b&gt;N.B.&lt;/b&gt;<a name="line.132"></a>
<span class="sourceLineNo">133</span>         * &lt;ul&gt;<a name="line.133"></a>
<span class="sourceLineNo">134</span>         * &lt;li&gt;This method can't be called after calling its NO-ARGUMENT twin.&lt;/li&gt;<a name="line.134"></a>
<span class="sourceLineNo">135</span>         * &lt;li&gt;remember to close the underlying resource when you are done.&lt;/li&gt;<a name="line.135"></a>
<span class="sourceLineNo">136</span>         * &lt;/ul&gt;<a name="line.136"></a>
<span class="sourceLineNo">137</span>         * @see #process()<a name="line.137"></a>
<span class="sourceLineNo">138</span>         * @author Amr AL-Hossary<a name="line.138"></a>
<span class="sourceLineNo">139</span>         * @since 3.0.6<a name="line.139"></a>
<span class="sourceLineNo">140</span>         * @param max maximum number of records to return, &lt;code&gt;-1&lt;/code&gt; for infinity.<a name="line.140"></a>
<span class="sourceLineNo">141</span>         * @return {@link HashMap} containing maximum &lt;code&gt;max&lt;/code&gt; parsed Genbank records<a name="line.141"></a>
<span class="sourceLineNo">142</span>         * present, starting current fileIndex onwards.<a name="line.142"></a>
<span class="sourceLineNo">143</span>         * @throws IOException<a name="line.143"></a>
<span class="sourceLineNo">144</span>         * @throws CompoundNotFoundException<a name="line.144"></a>
<span class="sourceLineNo">145</span>         */<a name="line.145"></a>
<span class="sourceLineNo">146</span>        public LinkedHashMap&lt;String,S&gt; process(final int max) throws IOException, CompoundNotFoundException {<a name="line.146"></a>
<span class="sourceLineNo">147</span>                LinkedHashMap&lt;String,S&gt; sequences = new LinkedHashMap&lt;&gt;();<a name="line.147"></a>
<span class="sourceLineNo">148</span>                @SuppressWarnings("unchecked")<a name="line.148"></a>
<span class="sourceLineNo">149</span>                int i=0;<a name="line.149"></a>
<span class="sourceLineNo">150</span>                while(true) {<a name="line.150"></a>
<span class="sourceLineNo">151</span>                        if(max&gt;0 &amp;&amp; i&gt;=max) break;<a name="line.151"></a>
<span class="sourceLineNo">152</span>                        i++;<a name="line.152"></a>
<span class="sourceLineNo">153</span>                        String seqString = genbankParser.getSequence(bufferedReader, 0);<a name="line.153"></a>
<span class="sourceLineNo">154</span>                        //reached end of file?<a name="line.154"></a>
<span class="sourceLineNo">155</span>                        if(seqString==null) break;<a name="line.155"></a>
<span class="sourceLineNo">156</span>            @SuppressWarnings("unchecked")<a name="line.156"></a>
<span class="sourceLineNo">157</span>                        S sequence = (S) sequenceCreator.getSequence(seqString, 0);<a name="line.157"></a>
<span class="sourceLineNo">158</span>                        genbankParser.getSequenceHeaderParser().parseHeader(genbankParser.getHeader(), sequence);<a name="line.158"></a>
<span class="sourceLineNo">159</span><a name="line.159"></a>
<span class="sourceLineNo">160</span>                        // add features to new sequence<a name="line.160"></a>
<span class="sourceLineNo">161</span>                        for (String k: genbankParser.getFeatures().keySet()){<a name="line.161"></a>
<span class="sourceLineNo">162</span>                                for (AbstractFeature f: genbankParser.getFeatures(k)){<a name="line.162"></a>
<span class="sourceLineNo">163</span>                                        //f.getLocations().setSequence(sequence);  // can't set proper sequence source to features. It is actually needed? Don't think so...<a name="line.163"></a>
<span class="sourceLineNo">164</span>                                        sequence.addFeature(f);<a name="line.164"></a>
<span class="sourceLineNo">165</span>                                }<a name="line.165"></a>
<span class="sourceLineNo">166</span>                        }<a name="line.166"></a>
<span class="sourceLineNo">167</span><a name="line.167"></a>
<span class="sourceLineNo">168</span>                        // add taxonomy ID to new sequence<a name="line.168"></a>
<span class="sourceLineNo">169</span>                        ArrayList&lt;DBReferenceInfo&gt; dbQualifier = genbankParser.getDatabaseReferences().get("db_xref");<a name="line.169"></a>
<span class="sourceLineNo">170</span>                        if (dbQualifier != null){<a name="line.170"></a>
<span class="sourceLineNo">171</span>                                DBReferenceInfo q = dbQualifier.get(0);<a name="line.171"></a>
<span class="sourceLineNo">172</span>                                sequence.setTaxonomy(new TaxonomyID(q.getDatabase()+":"+q.getId(), DataSource.GENBANK));<a name="line.172"></a>
<span class="sourceLineNo">173</span>                        }<a name="line.173"></a>
<span class="sourceLineNo">174</span><a name="line.174"></a>
<span class="sourceLineNo">175</span>                        sequences.put(sequence.getAccession().getID(), sequence);<a name="line.175"></a>
<span class="sourceLineNo">176</span>                }<a name="line.176"></a>
<span class="sourceLineNo">177</span><a name="line.177"></a>
<span class="sourceLineNo">178</span>                if (max &lt; 0) {<a name="line.178"></a>
<span class="sourceLineNo">179</span>                        close();<a name="line.179"></a>
<span class="sourceLineNo">180</span>                }<a name="line.180"></a>
<span class="sourceLineNo">181</span><a name="line.181"></a>
<span class="sourceLineNo">182</span>                return sequences;<a name="line.182"></a>
<span class="sourceLineNo">183</span>        }<a name="line.183"></a>
<span class="sourceLineNo">184</span><a name="line.184"></a>
<span class="sourceLineNo">185</span>        public void close() {<a name="line.185"></a>
<span class="sourceLineNo">186</span>                try {<a name="line.186"></a>
<span class="sourceLineNo">187</span>                        bufferedReader.close();<a name="line.187"></a>
<span class="sourceLineNo">188</span>                        this.closed = true;<a name="line.188"></a>
<span class="sourceLineNo">189</span>                } catch (IOException e) {<a name="line.189"></a>
<span class="sourceLineNo">190</span>                        logger.error("Couldn't close the reader. {}", e.getMessage());<a name="line.190"></a>
<span class="sourceLineNo">191</span>                        this.closed = false;<a name="line.191"></a>
<span class="sourceLineNo">192</span>                }<a name="line.192"></a>
<span class="sourceLineNo">193</span>        }<a name="line.193"></a>
<span class="sourceLineNo">194</span><a name="line.194"></a>
<span class="sourceLineNo">195</span>        public static void main(String[] args) throws Exception {<a name="line.195"></a>
<span class="sourceLineNo">196</span>                String proteinFile = "src/test/resources/BondFeature.gb";<a name="line.196"></a>
<span class="sourceLineNo">197</span>                FileInputStream is = new FileInputStream(proteinFile);<a name="line.197"></a>
<span class="sourceLineNo">198</span><a name="line.198"></a>
<span class="sourceLineNo">199</span>                GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt; proteinReader = new GenbankReader&lt;&gt;(is, new GenericGenbankHeaderParser&lt;&gt;(), new ProteinSequenceCreator(AminoAcidCompoundSet.getAminoAcidCompoundSet()));<a name="line.199"></a>
<span class="sourceLineNo">200</span>                LinkedHashMap&lt;String,ProteinSequence&gt; proteinSequences = proteinReader.process();<a name="line.200"></a>
<span class="sourceLineNo">201</span>                System.out.println(proteinSequences);<a name="line.201"></a>
<span class="sourceLineNo">202</span><a name="line.202"></a>
<span class="sourceLineNo">203</span>                String inputFile = "src/test/resources/NM_000266.gb";<a name="line.203"></a>
<span class="sourceLineNo">204</span>                is = new FileInputStream(inputFile);<a name="line.204"></a>
<span class="sourceLineNo">205</span>                GenbankReader&lt;DNASequence, NucleotideCompound&gt; dnaReader = new GenbankReader&lt;&gt;(is, new GenericGenbankHeaderParser&lt;&gt;(), new DNASequenceCreator(DNACompoundSet.getDNACompoundSet()));<a name="line.205"></a>
<span class="sourceLineNo">206</span>                LinkedHashMap&lt;String,DNASequence&gt; dnaSequences = dnaReader.process();<a name="line.206"></a>
<span class="sourceLineNo">207</span>                System.out.println(dnaSequences);<a name="line.207"></a>
<span class="sourceLineNo">208</span><a name="line.208"></a>
<span class="sourceLineNo">209</span>                String crazyFile = "src/test/resources/CraftedFeature.gb";<a name="line.209"></a>
<span class="sourceLineNo">210</span>                is = new FileInputStream(crazyFile);<a name="line.210"></a>
<span class="sourceLineNo">211</span>                GenbankReader&lt;DNASequence, NucleotideCompound&gt; crazyReader = new GenbankReader&lt;&gt;(is, new GenericGenbankHeaderParser&lt;&gt;(), new DNASequenceCreator(DNACompoundSet.getDNACompoundSet()));<a name="line.211"></a>
<span class="sourceLineNo">212</span>                LinkedHashMap&lt;String,DNASequence&gt; crazyAnnotatedSequences = crazyReader.process();<a name="line.212"></a>
<span class="sourceLineNo">213</span><a name="line.213"></a>
<span class="sourceLineNo">214</span>                is.close();<a name="line.214"></a>
<span class="sourceLineNo">215</span>                System.out.println(crazyAnnotatedSequences);<a name="line.215"></a>
<span class="sourceLineNo">216</span>        }<a name="line.216"></a>
<span class="sourceLineNo">217</span><a name="line.217"></a>
<span class="sourceLineNo">218</span>}<a name="line.218"></a>
<span class="sourceLineNo">219</span><a name="line.219"></a>




























































</pre>
</div>
</body>
</html>
