<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> */</a>
<span class="sourceLineNo">021</span><a id="line.21">package org.biojava.nbio.genome.parsers.gff;</a>
<span class="sourceLineNo">022</span><a id="line.22"></a>
<span class="sourceLineNo">023</span><a id="line.23">import org.biojava.nbio.genome.GeneFeatureHelper;</a>
<span class="sourceLineNo">024</span><a id="line.24">import org.biojava.nbio.core.sequence.*;</a>
<span class="sourceLineNo">025</span><a id="line.25"></a>
<span class="sourceLineNo">026</span><a id="line.26">import java.io.File;</a>
<span class="sourceLineNo">027</span><a id="line.27">import java.io.FileOutputStream;</a>
<span class="sourceLineNo">028</span><a id="line.28">import java.io.OutputStream;</a>
<span class="sourceLineNo">029</span><a id="line.29">import java.util.ArrayList;</a>
<span class="sourceLineNo">030</span><a id="line.30">import java.util.Collections;</a>
<span class="sourceLineNo">031</span><a id="line.31">import java.util.LinkedHashMap;</a>
<span class="sourceLineNo">032</span><a id="line.32"></a>
<span class="sourceLineNo">033</span><a id="line.33">/**</a>
<span class="sourceLineNo">034</span><a id="line.34"> *</a>
<span class="sourceLineNo">035</span><a id="line.35"> * @author Scooter Willis &lt;willishf at gmail dot com&gt;</a>
<span class="sourceLineNo">036</span><a id="line.36"> */</a>
<span class="sourceLineNo">037</span><a id="line.37">public class GFF3Writer {</a>
<span class="sourceLineNo">038</span><a id="line.38"></a>
<span class="sourceLineNo">039</span><a id="line.39">        /**</a>
<span class="sourceLineNo">040</span><a id="line.40">         * Output gff3 format for a DNA Sequence</a>
<span class="sourceLineNo">041</span><a id="line.41">         * @param fileName</a>
<span class="sourceLineNo">042</span><a id="line.42">         * @param chromosomeSequence</a>
<span class="sourceLineNo">043</span><a id="line.43">         * @throws Exception</a>
<span class="sourceLineNo">044</span><a id="line.44">         */</a>
<span class="sourceLineNo">045</span><a id="line.45">        public void write(OutputStream outputStream, LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList) throws Exception {</a>
<span class="sourceLineNo">046</span><a id="line.46"></a>
<span class="sourceLineNo">047</span><a id="line.47">                outputStream.write("##gff-version 3\n".getBytes());</a>
<span class="sourceLineNo">048</span><a id="line.48">                for (String key : chromosomeSequenceList.keySet()) {</a>
<span class="sourceLineNo">049</span><a id="line.49">                        ChromosomeSequence chromosomeSequence = chromosomeSequenceList.get(key);</a>
<span class="sourceLineNo">050</span><a id="line.50">                        String gff3line = "";</a>
<span class="sourceLineNo">051</span><a id="line.51">        //         if(source.length() == 0){</a>
<span class="sourceLineNo">052</span><a id="line.52">        //             Collection&lt;GeneSequence&gt; genes = chromosomeSequence.getGeneSequences().values();</a>
<span class="sourceLineNo">053</span><a id="line.53">        //             for(GeneSequence gene : genes){</a>
<span class="sourceLineNo">054</span><a id="line.54">        //                 source = gene.getSource();</a>
<span class="sourceLineNo">055</span><a id="line.55">        //                 break;</a>
<span class="sourceLineNo">056</span><a id="line.56">        //             }</a>
<span class="sourceLineNo">057</span><a id="line.57">        //         }</a>
<span class="sourceLineNo">058</span><a id="line.58">        //         gff3line = key + "\t" + source + "\t" + "size" + "\t" + "1" + "\t" + chromosomeSequence.getBioEnd() + "\t.\t.\t.\tName=" + key + "\r\n";</a>
<span class="sourceLineNo">059</span><a id="line.59">        //         outputStream.write(gff3line.getBytes());</a>
<span class="sourceLineNo">060</span><a id="line.60"></a>
<span class="sourceLineNo">061</span><a id="line.61">                        for (GeneSequence geneSequence : chromosomeSequence.getGeneSequences().values()) {</a>
<span class="sourceLineNo">062</span><a id="line.62">                                gff3line = key + "\t" + geneSequence.getSource() + "\t" + "gene" + "\t" + geneSequence.getBioBegin() + "\t" + geneSequence.getBioEnd() + "\t";</a>
<span class="sourceLineNo">063</span><a id="line.63">                                Double score = geneSequence.getSequenceScore();</a>
<span class="sourceLineNo">064</span><a id="line.64">                                if (score == null) {</a>
<span class="sourceLineNo">065</span><a id="line.65">                                        gff3line = gff3line + ".\t";</a>
<span class="sourceLineNo">066</span><a id="line.66">                                } else {</a>
<span class="sourceLineNo">067</span><a id="line.67">                                        gff3line = gff3line + score + "\t";</a>
<span class="sourceLineNo">068</span><a id="line.68">                                }</a>
<span class="sourceLineNo">069</span><a id="line.69">                                gff3line = gff3line + geneSequence.getStrand().getStringRepresentation() + "\t";</a>
<span class="sourceLineNo">070</span><a id="line.70">                                gff3line = gff3line + ".\t";</a>
<span class="sourceLineNo">071</span><a id="line.71">                                gff3line = gff3line + "ID=" + geneSequence.getAccession().getID() + ";Name=" + geneSequence.getAccession().getID();</a>
<span class="sourceLineNo">072</span><a id="line.72">                                gff3line = gff3line + getGFF3Note(geneSequence.getNotesList());</a>
<span class="sourceLineNo">073</span><a id="line.73">                                gff3line = gff3line + "\n";</a>
<span class="sourceLineNo">074</span><a id="line.74">                                outputStream.write(gff3line.getBytes());</a>
<span class="sourceLineNo">075</span><a id="line.75"></a>
<span class="sourceLineNo">076</span><a id="line.76">                                int transcriptIndex = 0;</a>
<span class="sourceLineNo">077</span><a id="line.77">                                for (TranscriptSequence transcriptSequence : geneSequence.getTranscripts().values()) {</a>
<span class="sourceLineNo">078</span><a id="line.78">                                        transcriptIndex++;</a>
<span class="sourceLineNo">079</span><a id="line.79"></a>
<span class="sourceLineNo">080</span><a id="line.80">                                        gff3line = key + "\t" + transcriptSequence.getSource() + "\t" + "mRNA" + "\t" + transcriptSequence.getBioBegin() + "\t" + transcriptSequence.getBioEnd() + "\t";</a>
<span class="sourceLineNo">081</span><a id="line.81">                                        score = transcriptSequence.getSequenceScore();</a>
<span class="sourceLineNo">082</span><a id="line.82">                                        if (score == null) {</a>
<span class="sourceLineNo">083</span><a id="line.83">                                                gff3line = gff3line + ".\t";</a>
<span class="sourceLineNo">084</span><a id="line.84">                                        } else {</a>
<span class="sourceLineNo">085</span><a id="line.85">                                                gff3line = gff3line + score + "\t";</a>
<span class="sourceLineNo">086</span><a id="line.86">                                        }</a>
<span class="sourceLineNo">087</span><a id="line.87">                                        gff3line = gff3line + transcriptSequence.getStrand().getStringRepresentation() + "\t";</a>
<span class="sourceLineNo">088</span><a id="line.88">                                        gff3line = gff3line + ".\t";</a>
<span class="sourceLineNo">089</span><a id="line.89">                                        String id = geneSequence.getAccession().getID() + "." + transcriptIndex;</a>
<span class="sourceLineNo">090</span><a id="line.90">                                        gff3line = gff3line + "ID=" + id + ";Parent=" + geneSequence.getAccession().getID() + ";Name=" + id;</a>
<span class="sourceLineNo">091</span><a id="line.91">                                        gff3line = gff3line + getGFF3Note(transcriptSequence.getNotesList());</a>
<span class="sourceLineNo">092</span><a id="line.92"></a>
<span class="sourceLineNo">093</span><a id="line.93">                                        gff3line = gff3line + "\n";</a>
<span class="sourceLineNo">094</span><a id="line.94">                                        outputStream.write(gff3line.getBytes());</a>
<span class="sourceLineNo">095</span><a id="line.95"></a>
<span class="sourceLineNo">096</span><a id="line.96">                                        String transcriptParentName = geneSequence.getAccession().getID() + "." + transcriptIndex;</a>
<span class="sourceLineNo">097</span><a id="line.97">                                        ArrayList&lt;CDSSequence&gt; cdsSequenceList = new ArrayList&lt;CDSSequence&gt;(transcriptSequence.getCDSSequences().values());</a>
<span class="sourceLineNo">098</span><a id="line.98">                                        Collections.sort(cdsSequenceList, new SequenceComparator());</a>
<span class="sourceLineNo">099</span><a id="line.99">                                        for (CDSSequence cdsSequence : cdsSequenceList) {</a>
<span class="sourceLineNo">100</span><a id="line.100">                                                gff3line = key + "\t" + cdsSequence.getSource() + "\t" + "CDS" + "\t" + cdsSequence.getBioBegin() + "\t" + cdsSequence.getBioEnd() + "\t";</a>
<span class="sourceLineNo">101</span><a id="line.101">                                                score = cdsSequence.getSequenceScore();</a>
<span class="sourceLineNo">102</span><a id="line.102">                                                if (score == null) {</a>
<span class="sourceLineNo">103</span><a id="line.103">                                                        gff3line = gff3line + ".\t";</a>
<span class="sourceLineNo">104</span><a id="line.104">                                                } else {</a>
<span class="sourceLineNo">105</span><a id="line.105">                                                        gff3line = gff3line + score + "\t";</a>
<span class="sourceLineNo">106</span><a id="line.106">                                                }</a>
<span class="sourceLineNo">107</span><a id="line.107">                                                gff3line = gff3line + cdsSequence.getStrand().getStringRepresentation() + "\t";</a>
<span class="sourceLineNo">108</span><a id="line.108">                                                gff3line = gff3line + cdsSequence.getPhase() + "\t";</a>
<span class="sourceLineNo">109</span><a id="line.109">                                                gff3line = gff3line + "ID=" + cdsSequence.getAccession().getID() + ";Parent=" + transcriptParentName;</a>
<span class="sourceLineNo">110</span><a id="line.110">                                                gff3line = gff3line + getGFF3Note(cdsSequence.getNotesList());</a>
<span class="sourceLineNo">111</span><a id="line.111"></a>
<span class="sourceLineNo">112</span><a id="line.112">                                                gff3line = gff3line + "\n";</a>
<span class="sourceLineNo">113</span><a id="line.113">                                                outputStream.write(gff3line.getBytes());</a>
<span class="sourceLineNo">114</span><a id="line.114">                                        }</a>
<span class="sourceLineNo">115</span><a id="line.115"></a>
<span class="sourceLineNo">116</span><a id="line.116">                                }</a>
<span class="sourceLineNo">117</span><a id="line.117">                        }</a>
<span class="sourceLineNo">118</span><a id="line.118"></a>
<span class="sourceLineNo">119</span><a id="line.119">                }</a>
<span class="sourceLineNo">120</span><a id="line.120"></a>
<span class="sourceLineNo">121</span><a id="line.121"></a>
<span class="sourceLineNo">122</span><a id="line.122">        }</a>
<span class="sourceLineNo">123</span><a id="line.123"></a>
<span class="sourceLineNo">124</span><a id="line.124">        private String getGFF3Note(ArrayList&lt;String&gt; notesList) {</a>
<span class="sourceLineNo">125</span><a id="line.125">                String notes = "";</a>
<span class="sourceLineNo">126</span><a id="line.126"></a>
<span class="sourceLineNo">127</span><a id="line.127">                if (notesList.size() &gt; 0) {</a>
<span class="sourceLineNo">128</span><a id="line.128">                        notes = ";Note=";</a>
<span class="sourceLineNo">129</span><a id="line.129">                        int noteindex = 1;</a>
<span class="sourceLineNo">130</span><a id="line.130">                        for (String note : notesList) {</a>
<span class="sourceLineNo">131</span><a id="line.131">                                notes = notes + note;</a>
<span class="sourceLineNo">132</span><a id="line.132">                                if (noteindex &lt; notesList.size() - 1) {</a>
<span class="sourceLineNo">133</span><a id="line.133">                                        notes = notes + " ";</a>
<span class="sourceLineNo">134</span><a id="line.134">                                }</a>
<span class="sourceLineNo">135</span><a id="line.135">                        }</a>
<span class="sourceLineNo">136</span><a id="line.136"></a>
<span class="sourceLineNo">137</span><a id="line.137">                }</a>
<span class="sourceLineNo">138</span><a id="line.138">                return notes;</a>
<span class="sourceLineNo">139</span><a id="line.139">        }</a>
<span class="sourceLineNo">140</span><a id="line.140"></a>
<span class="sourceLineNo">141</span><a id="line.141">        public static void main(String[] args) throws Exception {</a>
<span class="sourceLineNo">142</span><a id="line.142"></a>
<span class="sourceLineNo">143</span><a id="line.143">                if (true) {</a>
<span class="sourceLineNo">144</span><a id="line.144">                        FileOutputStream fo = new FileOutputStream("/Users/Scooter/scripps/dyadic/geneid/geneid/c1-geneid.gff3");//-16</a>
<span class="sourceLineNo">145</span><a id="line.145">                        LinkedHashMap&lt;String, ChromosomeSequence&gt; dnaSequenceList = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGeneIDGFF2(new File("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/454Scaffolds.fna"), new File("/Users/Scooter/scripps/dyadic/geneid/geneid/c1_geneid.gff"));</a>
<span class="sourceLineNo">146</span><a id="line.146">                        GFF3Writer gff3Writer = new GFF3Writer();</a>
<span class="sourceLineNo">147</span><a id="line.147">                        gff3Writer.write(fo, dnaSequenceList);</a>
<span class="sourceLineNo">148</span><a id="line.148"></a>
<span class="sourceLineNo">149</span><a id="line.149"></a>
<span class="sourceLineNo">150</span><a id="line.150">         //       LinkedHashMap&lt;String, ProteinSequence&gt; proteinSequenceList = GeneFeatureHelper.getProteinSequences(chromosomeSequenceList.values());</a>
<span class="sourceLineNo">151</span><a id="line.151">         //       for(String id : proteinSequenceList.keySet()){</a>
<span class="sourceLineNo">152</span><a id="line.152">         //           ProteinSequence sequence = proteinSequenceList.get(id);</a>
<span class="sourceLineNo">153</span><a id="line.153">         //           System.out.println(id + " " + sequence.getSequenceAsString());</a>
<span class="sourceLineNo">154</span><a id="line.154"></a>
<span class="sourceLineNo">155</span><a id="line.155">         //       }</a>
<span class="sourceLineNo">156</span><a id="line.156">                        fo.close();</a>
<span class="sourceLineNo">157</span><a id="line.157">                }</a>
<span class="sourceLineNo">158</span><a id="line.158">/*</a>
<span class="sourceLineNo">159</span><a id="line.159">                if (false) {</a>
<span class="sourceLineNo">160</span><a id="line.160">                        FileOutputStream fo = new FileOutputStream("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/genemark_hmm.gff3");//-16</a>
<span class="sourceLineNo">161</span><a id="line.161">                        LinkedHashMap&lt;String, ChromosomeSequence&gt; dnaSequenceList = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGeneMarkGTF(new File("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/454Scaffolds.fna"), new File("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/genemark_hmm.gtf"));</a>
<span class="sourceLineNo">162</span><a id="line.162">                        GFF3Writer gff3Writer = new GFF3Writer();</a>
<span class="sourceLineNo">163</span><a id="line.163">                        gff3Writer.write(fo, dnaSequenceList);</a>
<span class="sourceLineNo">164</span><a id="line.164">                        fo.close();</a>
<span class="sourceLineNo">165</span><a id="line.165">                }</a>
<span class="sourceLineNo">166</span><a id="line.166"></a>
<span class="sourceLineNo">167</span><a id="line.167">                if (false) {</a>
<span class="sourceLineNo">168</span><a id="line.168">                        LinkedHashMap&lt;String, ChromosomeSequence&gt; dnaSequenceList = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGlimmerGFF3(new File("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/454Scaffolds-16.fna"), new File("/Users/Scooter/scripps/dyadic/GlimmerHMM/c1_glimmerhmm-16.gff"));</a>
<span class="sourceLineNo">169</span><a id="line.169">                        GFF3Writer gff3Writer = new GFF3Writer();</a>
<span class="sourceLineNo">170</span><a id="line.170">                        gff3Writer.write(System.out, dnaSequenceList);</a>
<span class="sourceLineNo">171</span><a id="line.171">                }</a>
<span class="sourceLineNo">172</span><a id="line.172">                */</a>
<span class="sourceLineNo">173</span><a id="line.173">//        System.out.println(listGenes);</a>
<span class="sourceLineNo">174</span><a id="line.174">                //      GeneMarkGTF.write( list, args[1] );</a>
<span class="sourceLineNo">175</span><a id="line.175">        }</a>
<span class="sourceLineNo">176</span><a id="line.176">}</a>




























































</pre>
</div>
</main>
</body>
</html>
