<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span>package org.biojava.nbio.structure.align.multiple.util;<a name="line.21"></a>
<span class="sourceLineNo">022</span><a name="line.22"></a>
<span class="sourceLineNo">023</span>import java.io.IOException;<a name="line.23"></a>
<span class="sourceLineNo">024</span>import java.util.ArrayList;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import java.util.Arrays;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import java.util.Collections;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import java.util.Comparator;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import java.util.HashMap;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import java.util.List;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import java.util.Map;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import java.util.SortedSet;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import java.util.TreeSet;<a name="line.32"></a>
<span class="sourceLineNo">033</span><a name="line.33"></a>
<span class="sourceLineNo">034</span>import javax.vecmath.Matrix4d;<a name="line.34"></a>
<span class="sourceLineNo">035</span><a name="line.35"></a>
<span class="sourceLineNo">036</span>import org.biojava.nbio.core.alignment.matrices.SubstitutionMatrixHelper;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import org.biojava.nbio.core.exceptions.CompoundNotFoundException;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import org.biojava.nbio.core.sequence.AccessionID;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import org.biojava.nbio.core.sequence.MultipleSequenceAlignment;<a name="line.39"></a>
<span class="sourceLineNo">040</span>import org.biojava.nbio.core.sequence.ProteinSequence;<a name="line.40"></a>
<span class="sourceLineNo">041</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;<a name="line.41"></a>
<span class="sourceLineNo">042</span>import org.biojava.nbio.phylo.DistanceMatrixCalculator;<a name="line.42"></a>
<span class="sourceLineNo">043</span>import org.biojava.nbio.phylo.TreeConstructor;<a name="line.43"></a>
<span class="sourceLineNo">044</span>import org.biojava.nbio.phylo.TreeConstructorType;<a name="line.44"></a>
<span class="sourceLineNo">045</span>import org.biojava.nbio.structure.AminoAcid;<a name="line.45"></a>
<span class="sourceLineNo">046</span>import org.biojava.nbio.structure.Atom;<a name="line.46"></a>
<span class="sourceLineNo">047</span>import org.biojava.nbio.structure.Calc;<a name="line.47"></a>
<span class="sourceLineNo">048</span>import org.biojava.nbio.structure.Chain;<a name="line.48"></a>
<span class="sourceLineNo">049</span>import org.biojava.nbio.structure.Group;<a name="line.49"></a>
<span class="sourceLineNo">050</span>import org.biojava.nbio.structure.PDBHeader;<a name="line.50"></a>
<span class="sourceLineNo">051</span>import org.biojava.nbio.structure.Structure;<a name="line.51"></a>
<span class="sourceLineNo">052</span>import org.biojava.nbio.structure.StructureException;<a name="line.52"></a>
<span class="sourceLineNo">053</span>import org.biojava.nbio.structure.StructureIdentifier;<a name="line.53"></a>
<span class="sourceLineNo">054</span>import org.biojava.nbio.structure.StructureImpl;<a name="line.54"></a>
<span class="sourceLineNo">055</span>import org.biojava.nbio.structure.StructureTools;<a name="line.55"></a>
<span class="sourceLineNo">056</span>import org.biojava.nbio.structure.align.multiple.Block;<a name="line.56"></a>
<span class="sourceLineNo">057</span>import org.biojava.nbio.structure.align.multiple.BlockSet;<a name="line.57"></a>
<span class="sourceLineNo">058</span>import org.biojava.nbio.structure.align.multiple.MultipleAlignment;<a name="line.58"></a>
<span class="sourceLineNo">059</span>import org.biojava.nbio.structure.align.util.AlignmentTools;<a name="line.59"></a>
<span class="sourceLineNo">060</span>import org.biojava.nbio.structure.jama.Matrix;<a name="line.60"></a>
<span class="sourceLineNo">061</span>import org.forester.evoinference.matrix.distance.BasicSymmetricalDistanceMatrix;<a name="line.61"></a>
<span class="sourceLineNo">062</span>import org.forester.phylogeny.Phylogeny;<a name="line.62"></a>
<span class="sourceLineNo">063</span><a name="line.63"></a>
<span class="sourceLineNo">064</span>/**<a name="line.64"></a>
<span class="sourceLineNo">065</span> * Utility functions for working with {@link MultipleAlignment}.<a name="line.65"></a>
<span class="sourceLineNo">066</span> * &lt;p&gt;<a name="line.66"></a>
<span class="sourceLineNo">067</span> * Supported functions:<a name="line.67"></a>
<span class="sourceLineNo">068</span> * &lt;ul&gt;<a name="line.68"></a>
<span class="sourceLineNo">069</span> * &lt;li&gt;Obtain the alignment as sequence strings<a name="line.69"></a>
<span class="sourceLineNo">070</span> * &lt;li&gt;Map from sequence alignment position to structure Atom<a name="line.70"></a>
<span class="sourceLineNo">071</span> * &lt;li&gt;Map from sequence alignment position to Block number<a name="line.71"></a>
<span class="sourceLineNo">072</span> * &lt;li&gt;Transform the aligned Atoms of a MultipleAlignment<a name="line.72"></a>
<span class="sourceLineNo">073</span> * &lt;li&gt;Get all the core alignment positions of the alignment<a name="line.73"></a>
<span class="sourceLineNo">074</span> * &lt;li&gt;Calculate the average residue distance of all aligned positions<a name="line.74"></a>
<span class="sourceLineNo">075</span> * &lt;li&gt;Sort Blocks in a MultipleAlignment by a specified row<a name="line.75"></a>
<span class="sourceLineNo">076</span> * &lt;li&gt;Convert a MultipleAlignment to a MultipleSequenceAlignment<a name="line.76"></a>
<span class="sourceLineNo">077</span> * &lt;/ul&gt;<a name="line.77"></a>
<span class="sourceLineNo">078</span> *<a name="line.78"></a>
<span class="sourceLineNo">079</span> * @author Spencer Bliven<a name="line.79"></a>
<span class="sourceLineNo">080</span> * @author Aleix Lafita<a name="line.80"></a>
<span class="sourceLineNo">081</span> * @since 4.1.0<a name="line.81"></a>
<span class="sourceLineNo">082</span> *<a name="line.82"></a>
<span class="sourceLineNo">083</span> */<a name="line.83"></a>
<span class="sourceLineNo">084</span>public class MultipleAlignmentTools {<a name="line.84"></a>
<span class="sourceLineNo">085</span><a name="line.85"></a>
<span class="sourceLineNo">086</span>        /**<a name="line.86"></a>
<span class="sourceLineNo">087</span>         * Calculate the sequence alignment Strings for the whole alignment. This<a name="line.87"></a>
<span class="sourceLineNo">088</span>         * method creates a sequence alignment where aligned residues are in<a name="line.88"></a>
<span class="sourceLineNo">089</span>         * uppercase and unaligned residues are in lowercase, thus providing a more<a name="line.89"></a>
<span class="sourceLineNo">090</span>         * compact way to represent the alignment.<a name="line.90"></a>
<span class="sourceLineNo">091</span>         * &lt;p&gt;<a name="line.91"></a>
<span class="sourceLineNo">092</span>         * Blocks are concatenated in the order returned by<a name="line.92"></a>
<span class="sourceLineNo">093</span>         * {@link MultipleAlignment#getBlocks()}, so sequences may not be<a name="line.93"></a>
<span class="sourceLineNo">094</span>         * sequential. Gaps are represented by '-'. Separation between different<a name="line.94"></a>
<span class="sourceLineNo">095</span>         * Blocks is indicated by a gap in all positions, meaning that there is a<a name="line.95"></a>
<span class="sourceLineNo">096</span>         * possible discontinuity.<a name="line.96"></a>
<span class="sourceLineNo">097</span>         *<a name="line.97"></a>
<span class="sourceLineNo">098</span>         * @param alignment<a name="line.98"></a>
<span class="sourceLineNo">099</span>         *            input MultipleAlignment<a name="line.99"></a>
<span class="sourceLineNo">100</span>         * @param mapSeqToStruct<a name="line.100"></a>
<span class="sourceLineNo">101</span>         *            provides a link from the sequence alignment position to the<a name="line.101"></a>
<span class="sourceLineNo">102</span>         *            structure alignment position. Specially designed for the GUI.<a name="line.102"></a>
<span class="sourceLineNo">103</span>         *            Has to be initialized previously and will be overwritten.<a name="line.103"></a>
<span class="sourceLineNo">104</span>         * @return a string for each row in the alignment, giving the 1-letter code<a name="line.104"></a>
<span class="sourceLineNo">105</span>         *         for each aligned residue.<a name="line.105"></a>
<span class="sourceLineNo">106</span>         */<a name="line.106"></a>
<span class="sourceLineNo">107</span>        public static List&lt;String&gt; getSequenceAlignment(<a name="line.107"></a>
<span class="sourceLineNo">108</span>                        MultipleAlignment alignment, final List&lt;Integer&gt; mapSeqToStruct) {<a name="line.108"></a>
<span class="sourceLineNo">109</span><a name="line.109"></a>
<span class="sourceLineNo">110</span>                // Initialize sequence variables<a name="line.110"></a>
<span class="sourceLineNo">111</span>                List&lt;String&gt; alnSequences = new ArrayList&lt;String&gt;();<a name="line.111"></a>
<span class="sourceLineNo">112</span>                for (int str = 0; str &lt; alignment.size(); str++)<a name="line.112"></a>
<span class="sourceLineNo">113</span>                        alnSequences.add("");<a name="line.113"></a>
<span class="sourceLineNo">114</span>                mapSeqToStruct.clear();<a name="line.114"></a>
<span class="sourceLineNo">115</span>                List&lt;Atom[]&gt; atoms = alignment.getAtomArrays();<a name="line.115"></a>
<span class="sourceLineNo">116</span>                int globalPos = -1;<a name="line.116"></a>
<span class="sourceLineNo">117</span><a name="line.117"></a>
<span class="sourceLineNo">118</span>                // Initialize helper variables in constucting the sequence alignment<a name="line.118"></a>
<span class="sourceLineNo">119</span>                List&lt;SortedSet&lt;Integer&gt;&gt; freePool = new ArrayList&lt;SortedSet&lt;Integer&gt;&gt;();<a name="line.119"></a>
<span class="sourceLineNo">120</span>                List&lt;SortedSet&lt;Integer&gt;&gt; blockStarts = new ArrayList&lt;SortedSet&lt;Integer&gt;&gt;();<a name="line.120"></a>
<span class="sourceLineNo">121</span>                List&lt;List&lt;Integer&gt;&gt; aligned = new ArrayList&lt;List&lt;Integer&gt;&gt;();<a name="line.121"></a>
<span class="sourceLineNo">122</span><a name="line.122"></a>
<span class="sourceLineNo">123</span>                // Generate freePool residues from the ones not aligned<a name="line.123"></a>
<span class="sourceLineNo">124</span>                for (int i = 0; i &lt; alignment.size(); i++) {<a name="line.124"></a>
<span class="sourceLineNo">125</span>                        List&lt;Integer&gt; residues = new ArrayList&lt;Integer&gt;();<a name="line.125"></a>
<span class="sourceLineNo">126</span>                        freePool.add(new TreeSet&lt;Integer&gt;());<a name="line.126"></a>
<span class="sourceLineNo">127</span>                        blockStarts.add(new TreeSet&lt;Integer&gt;());<a name="line.127"></a>
<span class="sourceLineNo">128</span>                        for (BlockSet bs : alignment.getBlockSets()) {<a name="line.128"></a>
<span class="sourceLineNo">129</span>                                for (Block b : bs.getBlocks()) {<a name="line.129"></a>
<span class="sourceLineNo">130</span>                                        boolean first = true;<a name="line.130"></a>
<span class="sourceLineNo">131</span>                                        for (int l = 0; l &lt; b.length(); l++) {<a name="line.131"></a>
<span class="sourceLineNo">132</span>                                                Integer residue = b.getAlignRes().get(i).get(l);<a name="line.132"></a>
<span class="sourceLineNo">133</span>                                                if (residue != null) {<a name="line.133"></a>
<span class="sourceLineNo">134</span>                                                        if (first)<a name="line.134"></a>
<span class="sourceLineNo">135</span>                                                                blockStarts.get(i).add(residue);<a name="line.135"></a>
<span class="sourceLineNo">136</span>                                                        residues.add(residue);<a name="line.136"></a>
<span class="sourceLineNo">137</span>                                                        first = false;<a name="line.137"></a>
<span class="sourceLineNo">138</span>                                                }<a name="line.138"></a>
<span class="sourceLineNo">139</span>                                        }<a name="line.139"></a>
<span class="sourceLineNo">140</span>                                }<a name="line.140"></a>
<span class="sourceLineNo">141</span>                        }<a name="line.141"></a>
<span class="sourceLineNo">142</span>                        aligned.add(residues);<a name="line.142"></a>
<span class="sourceLineNo">143</span>                }<a name="line.143"></a>
<span class="sourceLineNo">144</span>                // Add any residue not aligned to the free pool for every structure<a name="line.144"></a>
<span class="sourceLineNo">145</span>                for (int i = 0; i &lt; alignment.size(); i++) {<a name="line.145"></a>
<span class="sourceLineNo">146</span>                        for (int k = 0; k &lt; atoms.get(i).length; k++) {<a name="line.146"></a>
<span class="sourceLineNo">147</span>                                if (!aligned.get(i).contains(k))<a name="line.147"></a>
<span class="sourceLineNo">148</span>                                        freePool.get(i).add(k);<a name="line.148"></a>
<span class="sourceLineNo">149</span>                        }<a name="line.149"></a>
<span class="sourceLineNo">150</span>                }<a name="line.150"></a>
<span class="sourceLineNo">151</span><a name="line.151"></a>
<span class="sourceLineNo">152</span>                for (int b = 0; b &lt; alignment.getBlocks().size(); b++) {<a name="line.152"></a>
<span class="sourceLineNo">153</span>                        if (b != 0) {<a name="line.153"></a>
<span class="sourceLineNo">154</span>                                // Add a gap to all structures to separate visually the Blocks<a name="line.154"></a>
<span class="sourceLineNo">155</span>                                for (int str = 0; str &lt; alignment.size(); str++)<a name="line.155"></a>
<span class="sourceLineNo">156</span>                                        alnSequences.set(str, alnSequences.get(str).concat("-"));<a name="line.156"></a>
<span class="sourceLineNo">157</span>                                mapSeqToStruct.add(-1); // means unaligned position<a name="line.157"></a>
<span class="sourceLineNo">158</span>                        }<a name="line.158"></a>
<span class="sourceLineNo">159</span>                        // Store the previous position added to the sequence alignment<a name="line.159"></a>
<span class="sourceLineNo">160</span>                        int[] previousPos = new int[alignment.size()];<a name="line.160"></a>
<span class="sourceLineNo">161</span>                        Arrays.fill(previousPos, -1);<a name="line.161"></a>
<span class="sourceLineNo">162</span>                        // Store provisional characters<a name="line.162"></a>
<span class="sourceLineNo">163</span>                        char[] provisionalChar = new char[alignment.size()];<a name="line.163"></a>
<span class="sourceLineNo">164</span>                        Arrays.fill(provisionalChar, '-');<a name="line.164"></a>
<span class="sourceLineNo">165</span><a name="line.165"></a>
<span class="sourceLineNo">166</span>                        for (int pos = 0; pos &lt; alignment.getBlocks().get(b).length(); pos++) {<a name="line.166"></a>
<span class="sourceLineNo">167</span>                                globalPos++;<a name="line.167"></a>
<span class="sourceLineNo">168</span>                                boolean gaps = true; // true if consecutive with the previous<a name="line.168"></a>
<span class="sourceLineNo">169</span>                                while (gaps) {<a name="line.169"></a>
<span class="sourceLineNo">170</span>                                        gaps = false;<a name="line.170"></a>
<span class="sourceLineNo">171</span>                                        // Loop through all the structures<a name="line.171"></a>
<span class="sourceLineNo">172</span>                                        for (int str = 0; str &lt; alignment.size(); str++) {<a name="line.172"></a>
<span class="sourceLineNo">173</span>                                                // If it is the first position or before it was null<a name="line.173"></a>
<span class="sourceLineNo">174</span>                                                if (previousPos[str] == -1) {<a name="line.174"></a>
<span class="sourceLineNo">175</span>                                                        Integer residue = alignment.getBlocks().get(b)<a name="line.175"></a>
<span class="sourceLineNo">176</span>                                                                        .getAlignRes().get(str).get(pos);<a name="line.176"></a>
<span class="sourceLineNo">177</span>                                                        if (residue == null)<a name="line.177"></a>
<span class="sourceLineNo">178</span>                                                                provisionalChar[str] = '-';<a name="line.178"></a>
<span class="sourceLineNo">179</span>                                                        else {<a name="line.179"></a>
<span class="sourceLineNo">180</span>                                                                Atom a = atoms.get(str)[residue];<a name="line.180"></a>
<span class="sourceLineNo">181</span>                                                                String group = a.getGroup().getPDBName();<a name="line.181"></a>
<span class="sourceLineNo">182</span>                                                                provisionalChar[str] = StructureTools<a name="line.182"></a>
<span class="sourceLineNo">183</span>                                                                                .get1LetterCode(group);<a name="line.183"></a>
<span class="sourceLineNo">184</span>                                                        }<a name="line.184"></a>
<span class="sourceLineNo">185</span>                                                } else {<a name="line.185"></a>
<span class="sourceLineNo">186</span>                                                        Integer residue = alignment.getBlocks().get(b)<a name="line.186"></a>
<span class="sourceLineNo">187</span>                                                                        .getAlignRes().get(str).get(pos);<a name="line.187"></a>
<span class="sourceLineNo">188</span>                                                        int nextPos = previousPos[str] + 1;<a name="line.188"></a>
<span class="sourceLineNo">189</span>                                                        if (residue == null) {<a name="line.189"></a>
<span class="sourceLineNo">190</span>                                                                if (freePool.get(str).contains(nextPos)) {<a name="line.190"></a>
<span class="sourceLineNo">191</span>                                                                        Atom a = atoms.get(str)[nextPos];<a name="line.191"></a>
<span class="sourceLineNo">192</span>                                                                        String g = a.getGroup().getPDBName();<a name="line.192"></a>
<span class="sourceLineNo">193</span>                                                                        char aa = StructureTools.get1LetterCode(g);<a name="line.193"></a>
<span class="sourceLineNo">194</span>                                                                        provisionalChar[str] = Character<a name="line.194"></a>
<span class="sourceLineNo">195</span>                                                                                        .toLowerCase(aa);<a name="line.195"></a>
<span class="sourceLineNo">196</span>                                                                } else<a name="line.196"></a>
<span class="sourceLineNo">197</span>                                                                        provisionalChar[str] = '-';<a name="line.197"></a>
<span class="sourceLineNo">198</span>                                                        } else if (nextPos == residue) {<a name="line.198"></a>
<span class="sourceLineNo">199</span>                                                                Atom a = atoms.get(str)[nextPos];<a name="line.199"></a>
<span class="sourceLineNo">200</span>                                                                String group = a.getGroup().getPDBName();<a name="line.200"></a>
<span class="sourceLineNo">201</span>                                                                provisionalChar[str] = StructureTools<a name="line.201"></a>
<span class="sourceLineNo">202</span>                                                                                .get1LetterCode(group);<a name="line.202"></a>
<span class="sourceLineNo">203</span>                                                        } else {<a name="line.203"></a>
<span class="sourceLineNo">204</span>                                                                // This means non-consecutive<a name="line.204"></a>
<span class="sourceLineNo">205</span>                                                                provisionalChar[str] = ' ';<a name="line.205"></a>
<span class="sourceLineNo">206</span>                                                                gaps = true;<a name="line.206"></a>
<span class="sourceLineNo">207</span>                                                        }<a name="line.207"></a>
<span class="sourceLineNo">208</span>                                                }<a name="line.208"></a>
<span class="sourceLineNo">209</span>                                        }// End all structure analysis<a name="line.209"></a>
<span class="sourceLineNo">210</span><a name="line.210"></a>
<span class="sourceLineNo">211</span>                                        if (gaps) {<a name="line.211"></a>
<span class="sourceLineNo">212</span>                                                for (int str = 0; str &lt; alignment.size(); str++) {<a name="line.212"></a>
<span class="sourceLineNo">213</span>                                                        if (provisionalChar[str] == ' ') {<a name="line.213"></a>
<span class="sourceLineNo">214</span>                                                                // It means this residue was non-consecutive<a name="line.214"></a>
<span class="sourceLineNo">215</span>                                                                Atom a = atoms.get(str)[previousPos[str] + 1];<a name="line.215"></a>
<span class="sourceLineNo">216</span>                                                                String group = a.getGroup().getPDBName();<a name="line.216"></a>
<span class="sourceLineNo">217</span>                                                                char aa = StructureTools.get1LetterCode(group);<a name="line.217"></a>
<span class="sourceLineNo">218</span>                                                                alnSequences<a name="line.218"></a>
<span class="sourceLineNo">219</span>                                                                                .set(str,<a name="line.219"></a>
<span class="sourceLineNo">220</span>                                                                                                alnSequences<a name="line.220"></a>
<span class="sourceLineNo">221</span>                                                                                                                .get(str)<a name="line.221"></a>
<span class="sourceLineNo">222</span>                                                                                                                .concat(String.valueOf(Character<a name="line.222"></a>
<span class="sourceLineNo">223</span>                                                                                                                                                .toLowerCase(aa))) );<a name="line.223"></a>
<span class="sourceLineNo">224</span>                                                                previousPos[str]++;<a name="line.224"></a>
<span class="sourceLineNo">225</span>                                                        } else {<a name="line.225"></a>
<span class="sourceLineNo">226</span>                                                                // Insert a gap otherwise<a name="line.226"></a>
<span class="sourceLineNo">227</span>                                                                alnSequences.set(str, alnSequences.get(str)<a name="line.227"></a>
<span class="sourceLineNo">228</span>                                                                                .concat("-"));<a name="line.228"></a>
<span class="sourceLineNo">229</span>                                                        }<a name="line.229"></a>
<span class="sourceLineNo">230</span>                                                }<a name="line.230"></a>
<span class="sourceLineNo">231</span>                                                mapSeqToStruct.add(-1); // unaligned position<a name="line.231"></a>
<span class="sourceLineNo">232</span>                                        } else {<a name="line.232"></a>
<span class="sourceLineNo">233</span>                                                // Add provisional and update the indices<a name="line.233"></a>
<span class="sourceLineNo">234</span>                                                for (int str = 0; str &lt; alignment.size(); str++) {<a name="line.234"></a>
<span class="sourceLineNo">235</span>                                                        alnSequences.set(<a name="line.235"></a>
<span class="sourceLineNo">236</span>                                                                        str,<a name="line.236"></a>
<span class="sourceLineNo">237</span>                                                                        alnSequences.get(str).concat(<a name="line.237"></a>
<span class="sourceLineNo">238</span>                                                                                        String.valueOf(provisionalChar[str])));<a name="line.238"></a>
<span class="sourceLineNo">239</span><a name="line.239"></a>
<span class="sourceLineNo">240</span>                                                        if (provisionalChar[str] != '-') {<a name="line.240"></a>
<span class="sourceLineNo">241</span>                                                                if (alignment.getBlocks().get(b).getAlignRes()<a name="line.241"></a>
<span class="sourceLineNo">242</span>                                                                                .get(str).get(pos) == null) {<a name="line.242"></a>
<span class="sourceLineNo">243</span>                                                                        previousPos[str]++;<a name="line.243"></a>
<span class="sourceLineNo">244</span>                                                                } else {<a name="line.244"></a>
<span class="sourceLineNo">245</span>                                                                        previousPos[str] = alignment.getBlocks()<a name="line.245"></a>
<span class="sourceLineNo">246</span>                                                                                        .get(b).getAlignRes().get(str)<a name="line.246"></a>
<span class="sourceLineNo">247</span>                                                                                        .get(pos);<a name="line.247"></a>
<span class="sourceLineNo">248</span>                                                                }<a name="line.248"></a>
<span class="sourceLineNo">249</span>                                                        }<a name="line.249"></a>
<span class="sourceLineNo">250</span>                                                }<a name="line.250"></a>
<span class="sourceLineNo">251</span>                                                mapSeqToStruct.add(globalPos); // alignment index<a name="line.251"></a>
<span class="sourceLineNo">252</span>                                        }<a name="line.252"></a>
<span class="sourceLineNo">253</span>                                }<a name="line.253"></a>
<span class="sourceLineNo">254</span>                        } // All positions in the Block considered so far<a name="line.254"></a>
<span class="sourceLineNo">255</span><a name="line.255"></a>
<span class="sourceLineNo">256</span>                        // Calculate the index of the next Block for every structure<a name="line.256"></a>
<span class="sourceLineNo">257</span>                        int[] blockEnds = new int[alignment.size()];<a name="line.257"></a>
<span class="sourceLineNo">258</span>                        for (int str = 0; str &lt; alignment.size(); str++) {<a name="line.258"></a>
<span class="sourceLineNo">259</span>                                for (int res : blockStarts.get(str)) {<a name="line.259"></a>
<span class="sourceLineNo">260</span>                                        if (previousPos[str] &gt; res)<a name="line.260"></a>
<span class="sourceLineNo">261</span>                                                blockEnds[str] = res;<a name="line.261"></a>
<span class="sourceLineNo">262</span>                                        else {<a name="line.262"></a>
<span class="sourceLineNo">263</span>                                                blockEnds[str] = res;<a name="line.263"></a>
<span class="sourceLineNo">264</span>                                                break;<a name="line.264"></a>
<span class="sourceLineNo">265</span>                                        }<a name="line.265"></a>
<span class="sourceLineNo">266</span>                                }<a name="line.266"></a>
<span class="sourceLineNo">267</span>                        }<a name="line.267"></a>
<span class="sourceLineNo">268</span><a name="line.268"></a>
<span class="sourceLineNo">269</span>                        // Add the unaligned residues in between Blocks (lowercase)<a name="line.269"></a>
<span class="sourceLineNo">270</span>                        boolean allGaps = false; // true means no more residues to add<a name="line.270"></a>
<span class="sourceLineNo">271</span>                        while (!allGaps) {<a name="line.271"></a>
<span class="sourceLineNo">272</span>                                allGaps = true;<a name="line.272"></a>
<span class="sourceLineNo">273</span>                                for (int str = 0; str &lt; alignment.size(); str++) {<a name="line.273"></a>
<span class="sourceLineNo">274</span>                                        if (previousPos[str] + 1 &lt; blockEnds[str]) {<a name="line.274"></a>
<span class="sourceLineNo">275</span>                                                Atom a = atoms.get(str)[previousPos[str] + 1];<a name="line.275"></a>
<span class="sourceLineNo">276</span>                                                String group = a.getGroup().getPDBName();<a name="line.276"></a>
<span class="sourceLineNo">277</span>                                                char letter = StructureTools.get1LetterCode(group);<a name="line.277"></a>
<span class="sourceLineNo">278</span><a name="line.278"></a>
<span class="sourceLineNo">279</span>                                                provisionalChar[str] = Character.toLowerCase(letter);<a name="line.279"></a>
<span class="sourceLineNo">280</span>                                                previousPos[str]++;<a name="line.280"></a>
<span class="sourceLineNo">281</span>                                                allGaps = false;<a name="line.281"></a>
<span class="sourceLineNo">282</span>                                        } else<a name="line.282"></a>
<span class="sourceLineNo">283</span>                                                provisionalChar[str] = '-';<a name="line.283"></a>
<span class="sourceLineNo">284</span>                                }<a name="line.284"></a>
<span class="sourceLineNo">285</span>                                if (!allGaps) {<a name="line.285"></a>
<span class="sourceLineNo">286</span>                                        for (int str = 0; str &lt; alignment.size(); str++) {<a name="line.286"></a>
<span class="sourceLineNo">287</span>                                                alnSequences.set(<a name="line.287"></a>
<span class="sourceLineNo">288</span>                                                                str,<a name="line.288"></a>
<span class="sourceLineNo">289</span>                                                                alnSequences.get(str).concat(<a name="line.289"></a>
<span class="sourceLineNo">290</span>                                                                                String.valueOf(provisionalChar[str])) );<a name="line.290"></a>
<span class="sourceLineNo">291</span>                                        }<a name="line.291"></a>
<span class="sourceLineNo">292</span>                                        mapSeqToStruct.add(-1); // unaligned position<a name="line.292"></a>
<span class="sourceLineNo">293</span>                                }<a name="line.293"></a>
<span class="sourceLineNo">294</span>                        }<a name="line.294"></a>
<span class="sourceLineNo">295</span>                }<a name="line.295"></a>
<span class="sourceLineNo">296</span>                return alnSequences;<a name="line.296"></a>
<span class="sourceLineNo">297</span>        }<a name="line.297"></a>
<span class="sourceLineNo">298</span><a name="line.298"></a>
<span class="sourceLineNo">299</span>        /**<a name="line.299"></a>
<span class="sourceLineNo">300</span>         * Calculate the sequence alignment Strings for the whole alignment. This<a name="line.300"></a>
<span class="sourceLineNo">301</span>         * method creates a sequence alignment where aligned residues are in<a name="line.301"></a>
<span class="sourceLineNo">302</span>         * uppercase and unaligned residues are in lowercase, thus providing a more<a name="line.302"></a>
<span class="sourceLineNo">303</span>         * compact way to represent the alignment.<a name="line.303"></a>
<span class="sourceLineNo">304</span>         * &lt;p&gt;<a name="line.304"></a>
<span class="sourceLineNo">305</span>         * Blocks are concatenated in the order returned by<a name="line.305"></a>
<span class="sourceLineNo">306</span>         * {@link MultipleAlignment#getBlocks()}, so sequences may not be<a name="line.306"></a>
<span class="sourceLineNo">307</span>         * sequential. Gaps are represented by '-'. Separation between different<a name="line.307"></a>
<span class="sourceLineNo">308</span>         * Blocks is indicated by a gap in all positions, meaning that there is a<a name="line.308"></a>
<span class="sourceLineNo">309</span>         * possible discontinuity.<a name="line.309"></a>
<span class="sourceLineNo">310</span>         *<a name="line.310"></a>
<span class="sourceLineNo">311</span>         * @param alignment<a name="line.311"></a>
<span class="sourceLineNo">312</span>         *            input MultipleAlignment<a name="line.312"></a>
<span class="sourceLineNo">313</span>         * @return String for each row in the alignment, giving the 1-letter code<a name="line.313"></a>
<span class="sourceLineNo">314</span>         *         for each aligned residue.<a name="line.314"></a>
<span class="sourceLineNo">315</span>         */<a name="line.315"></a>
<span class="sourceLineNo">316</span>        public static List&lt;String&gt; getSequenceAlignment(MultipleAlignment msa) {<a name="line.316"></a>
<span class="sourceLineNo">317</span>                return getSequenceAlignment(msa, new ArrayList&lt;Integer&gt;());<a name="line.317"></a>
<span class="sourceLineNo">318</span>        }<a name="line.318"></a>
<span class="sourceLineNo">319</span><a name="line.319"></a>
<span class="sourceLineNo">320</span>        /**<a name="line.320"></a>
<span class="sourceLineNo">321</span>         * Calculate the sequence alignment Strings for the alignment Blocks in an<a name="line.321"></a>
<span class="sourceLineNo">322</span>         * alignment. This method creates a sequence alignment where all residues<a name="line.322"></a>
<span class="sourceLineNo">323</span>         * are in uppercase and a residue alone with gaps in all the other<a name="line.323"></a>
<span class="sourceLineNo">324</span>         * structures represents unaligned residues. Because of this latter<a name="line.324"></a>
<span class="sourceLineNo">325</span>         * constraint only the residues within the Blocks are represented, for a<a name="line.325"></a>
<span class="sourceLineNo">326</span>         * more compact alignment. For a sequence alignment of the full protein use<a name="line.326"></a>
<span class="sourceLineNo">327</span>         * {@link #getSequenceAlignment(MultipleAlignment)}.<a name="line.327"></a>
<span class="sourceLineNo">328</span>         * &lt;p&gt;<a name="line.328"></a>
<span class="sourceLineNo">329</span>         * Blocks are concatenated in the order returned by<a name="line.329"></a>
<span class="sourceLineNo">330</span>         * {@link MultipleAlignment#getBlocks()}, so sequences may not be<a name="line.330"></a>
<span class="sourceLineNo">331</span>         * sequential. Gaps between blocks are omitted, while gaps within blocks are<a name="line.331"></a>
<span class="sourceLineNo">332</span>         * represented by '-'. Separation between different Blocks is indicated by a<a name="line.332"></a>
<span class="sourceLineNo">333</span>         * gap in all positions, meaning that there is something unaligned<a name="line.333"></a>
<span class="sourceLineNo">334</span>         * inbetween.<a name="line.334"></a>
<span class="sourceLineNo">335</span>         *<a name="line.335"></a>
<span class="sourceLineNo">336</span>         * @param alignment<a name="line.336"></a>
<span class="sourceLineNo">337</span>         *            input MultipleAlignment<a name="line.337"></a>
<span class="sourceLineNo">338</span>         * @param mapSeqToStruct<a name="line.338"></a>
<span class="sourceLineNo">339</span>         *            provides a link from the sequence alignment position to the<a name="line.339"></a>
<span class="sourceLineNo">340</span>         *            structure alignment position. Specially designed for the GUI.<a name="line.340"></a>
<span class="sourceLineNo">341</span>         *            Has to be initialized previously and will be overwritten.<a name="line.341"></a>
<span class="sourceLineNo">342</span>         * @return a string for each row in the alignment, giving the 1-letter code<a name="line.342"></a>
<span class="sourceLineNo">343</span>         *         for each aligned residue.<a name="line.343"></a>
<span class="sourceLineNo">344</span>         */<a name="line.344"></a>
<span class="sourceLineNo">345</span>        public static List&lt;String&gt; getBlockSequenceAlignment(<a name="line.345"></a>
<span class="sourceLineNo">346</span>                        MultipleAlignment alignment, List&lt;Integer&gt; mapSeqToStruct) {<a name="line.346"></a>
<span class="sourceLineNo">347</span><a name="line.347"></a>
<span class="sourceLineNo">348</span>                // Initialize sequence variables<a name="line.348"></a>
<span class="sourceLineNo">349</span>                List&lt;String&gt; alnSequences = new ArrayList&lt;String&gt;();<a name="line.349"></a>
<span class="sourceLineNo">350</span>                for (int str = 0; str &lt; alignment.size(); str++)<a name="line.350"></a>
<span class="sourceLineNo">351</span>                        alnSequences.add("");<a name="line.351"></a>
<span class="sourceLineNo">352</span>                mapSeqToStruct.clear();<a name="line.352"></a>
<span class="sourceLineNo">353</span>                List&lt;Atom[]&gt; atoms = alignment.getAtomArrays();<a name="line.353"></a>
<span class="sourceLineNo">354</span>                int globalPos = -1;<a name="line.354"></a>
<span class="sourceLineNo">355</span><a name="line.355"></a>
<span class="sourceLineNo">356</span>                // Loop through all the alignment Blocks in the order given<a name="line.356"></a>
<span class="sourceLineNo">357</span>                for (int b = 0; b &lt; alignment.getBlocks().size(); b++) {<a name="line.357"></a>
<span class="sourceLineNo">358</span>                        if (b != 0) {<a name="line.358"></a>
<span class="sourceLineNo">359</span>                                // Add a gap to all structures to separate Blocks<a name="line.359"></a>
<span class="sourceLineNo">360</span>                                for (int str = 0; str &lt; alignment.size(); str++)<a name="line.360"></a>
<span class="sourceLineNo">361</span>                                        alnSequences.set(str, alnSequences.get(str).concat("-"));<a name="line.361"></a>
<span class="sourceLineNo">362</span>                                mapSeqToStruct.add(-1); // means unaligned position<a name="line.362"></a>
<span class="sourceLineNo">363</span>                        }<a name="line.363"></a>
<span class="sourceLineNo">364</span><a name="line.364"></a>
<span class="sourceLineNo">365</span>                        // Store the previous position added to the sequence alignment<a name="line.365"></a>
<span class="sourceLineNo">366</span>                        int[] previousPos = new int[alignment.size()];<a name="line.366"></a>
<span class="sourceLineNo">367</span>                        Arrays.fill(previousPos, -1);<a name="line.367"></a>
<span class="sourceLineNo">368</span>                        // Store provisional characters<a name="line.368"></a>
<span class="sourceLineNo">369</span>                        char[] provisionalChar = new char[alignment.size()];<a name="line.369"></a>
<span class="sourceLineNo">370</span>                        Arrays.fill(provisionalChar, '-');<a name="line.370"></a>
<span class="sourceLineNo">371</span><a name="line.371"></a>
<span class="sourceLineNo">372</span>                        for (int pos = 0; pos &lt; alignment.getBlocks().get(b).length(); pos++) {<a name="line.372"></a>
<span class="sourceLineNo">373</span>                                globalPos++;<a name="line.373"></a>
<span class="sourceLineNo">374</span>                                boolean gaps = true;<a name="line.374"></a>
<span class="sourceLineNo">375</span>                                while (gaps) {<a name="line.375"></a>
<span class="sourceLineNo">376</span>                                        gaps = false;<a name="line.376"></a>
<span class="sourceLineNo">377</span>                                        // Loop through all the structures<a name="line.377"></a>
<span class="sourceLineNo">378</span>                                        for (int str = 0; str &lt; alignment.size(); str++) {<a name="line.378"></a>
<span class="sourceLineNo">379</span>                                                // If it is the first position or before it was null<a name="line.379"></a>
<span class="sourceLineNo">380</span>                                                if (previousPos[str] == -1) {<a name="line.380"></a>
<span class="sourceLineNo">381</span>                                                        Integer residue = alignment.getBlocks().get(b)<a name="line.381"></a>
<span class="sourceLineNo">382</span>                                                                        .getAlignRes().get(str).get(pos);<a name="line.382"></a>
<span class="sourceLineNo">383</span>                                                        if (residue == null)<a name="line.383"></a>
<span class="sourceLineNo">384</span>                                                                provisionalChar[str] = '-';<a name="line.384"></a>
<span class="sourceLineNo">385</span>                                                        else {<a name="line.385"></a>
<span class="sourceLineNo">386</span>                                                                Atom a = atoms.get(str)[residue];<a name="line.386"></a>
<span class="sourceLineNo">387</span>                                                                String g = a.getGroup().getPDBName();<a name="line.387"></a>
<span class="sourceLineNo">388</span>                                                                char aa = StructureTools.get1LetterCode(g);<a name="line.388"></a>
<span class="sourceLineNo">389</span>                                                                provisionalChar[str] = aa;<a name="line.389"></a>
<span class="sourceLineNo">390</span>                                                        }<a name="line.390"></a>
<span class="sourceLineNo">391</span>                                                } else {<a name="line.391"></a>
<span class="sourceLineNo">392</span>                                                        Integer residue = alignment.getBlocks().get(b)<a name="line.392"></a>
<span class="sourceLineNo">393</span>                                                                        .getAlignRes().get(str).get(pos);<a name="line.393"></a>
<span class="sourceLineNo">394</span>                                                        if (residue == null)<a name="line.394"></a>
<span class="sourceLineNo">395</span>                                                                provisionalChar[str] = '-';<a name="line.395"></a>
<span class="sourceLineNo">396</span>                                                        else if (previousPos[str] + 1 == residue) {<a name="line.396"></a>
<span class="sourceLineNo">397</span>                                                                Atom a = atoms.get(str)[residue];<a name="line.397"></a>
<span class="sourceLineNo">398</span>                                                                String g = a.getGroup().getPDBName();<a name="line.398"></a>
<span class="sourceLineNo">399</span>                                                                char aa = StructureTools.get1LetterCode(g);<a name="line.399"></a>
<span class="sourceLineNo">400</span>                                                                provisionalChar[str] = aa;<a name="line.400"></a>
<span class="sourceLineNo">401</span>                                                        } else {<a name="line.401"></a>
<span class="sourceLineNo">402</span>                                                                provisionalChar[str] = ' ';<a name="line.402"></a>
<span class="sourceLineNo">403</span>                                                                gaps = true;<a name="line.403"></a>
<span class="sourceLineNo">404</span>                                                        }<a name="line.404"></a>
<span class="sourceLineNo">405</span>                                                }<a name="line.405"></a>
<span class="sourceLineNo">406</span>                                        }// End all structures analysis<a name="line.406"></a>
<span class="sourceLineNo">407</span><a name="line.407"></a>
<span class="sourceLineNo">408</span>                                        if (gaps) {<a name="line.408"></a>
<span class="sourceLineNo">409</span>                                                for (int str = 0; str &lt; alignment.size(); str++) {<a name="line.409"></a>
<span class="sourceLineNo">410</span>                                                        if (provisionalChar[str] == ' ') {<a name="line.410"></a>
<span class="sourceLineNo">411</span>                                                                // It means this residue was non-consecutive<a name="line.411"></a>
<span class="sourceLineNo">412</span>                                                                for (int s2 = 0; s2 &lt; alignment.size(); s2++) {<a name="line.412"></a>
<span class="sourceLineNo">413</span>                                                                        if (str == s2) {<a name="line.413"></a>
<span class="sourceLineNo">414</span>                                                                                int next = previousPos[str] + 1;<a name="line.414"></a>
<span class="sourceLineNo">415</span>                                                                                Atom a = atoms.get(s2)[next];<a name="line.415"></a>
<span class="sourceLineNo">416</span>                                                                                String g = a.getGroup().getPDBName();<a name="line.416"></a>
<span class="sourceLineNo">417</span>                                                                                char aa = StructureTools<a name="line.417"></a>
<span class="sourceLineNo">418</span>                                                                                                .get1LetterCode(g);<a name="line.418"></a>
<span class="sourceLineNo">419</span>                                                                                alnSequences.set(<a name="line.419"></a>
<span class="sourceLineNo">420</span>                                                                                                s2,<a name="line.420"></a>
<span class="sourceLineNo">421</span>                                                                                                alnSequences.get(s2).concat(<a name="line.421"></a>
<span class="sourceLineNo">422</span>                                                                                                                String.valueOf(aa)) );<a name="line.422"></a>
<span class="sourceLineNo">423</span>                                                                        } else {<a name="line.423"></a>
<span class="sourceLineNo">424</span>                                                                                alnSequences.set(s2,<a name="line.424"></a>
<span class="sourceLineNo">425</span>                                                                                                alnSequences.get(s2)<a name="line.425"></a>
<span class="sourceLineNo">426</span>                                                                                                                .concat("-"));<a name="line.426"></a>
<span class="sourceLineNo">427</span>                                                                        }<a name="line.427"></a>
<span class="sourceLineNo">428</span>                                                                }<a name="line.428"></a>
<span class="sourceLineNo">429</span>                                                                mapSeqToStruct.add(-1); // unaligned<a name="line.429"></a>
<span class="sourceLineNo">430</span>                                                                previousPos[str] += 1;<a name="line.430"></a>
<span class="sourceLineNo">431</span>                                                        }<a name="line.431"></a>
<span class="sourceLineNo">432</span>                                                }<a name="line.432"></a>
<span class="sourceLineNo">433</span>                                        } else { // Append the provisional and update the indices<a name="line.433"></a>
<span class="sourceLineNo">434</span>                                                for (int str = 0; str &lt; alignment.size(); str++) {<a name="line.434"></a>
<span class="sourceLineNo">435</span>                                                        alnSequences.set(<a name="line.435"></a>
<span class="sourceLineNo">436</span>                                                                        str,<a name="line.436"></a>
<span class="sourceLineNo">437</span>                                                                        alnSequences.get(str).concat(<a name="line.437"></a>
<span class="sourceLineNo">438</span>                                                                                        String.valueOf(provisionalChar[str])) );<a name="line.438"></a>
<span class="sourceLineNo">439</span>                                                        if (provisionalChar[str] != '-') {<a name="line.439"></a>
<span class="sourceLineNo">440</span>                                                                previousPos[str] = alignment.getBlocks().get(b)<a name="line.440"></a>
<span class="sourceLineNo">441</span>                                                                                .getAlignRes().get(str).get(pos);<a name="line.441"></a>
<span class="sourceLineNo">442</span>                                                        }<a name="line.442"></a>
<span class="sourceLineNo">443</span>                                                }<a name="line.443"></a>
<span class="sourceLineNo">444</span>                                                mapSeqToStruct.add(globalPos);<a name="line.444"></a>
<span class="sourceLineNo">445</span>                                        }<a name="line.445"></a>
<span class="sourceLineNo">446</span>                                }<a name="line.446"></a>
<span class="sourceLineNo">447</span>                        }<a name="line.447"></a>
<span class="sourceLineNo">448</span>                }<a name="line.448"></a>
<span class="sourceLineNo">449</span>                return alnSequences;<a name="line.449"></a>
<span class="sourceLineNo">450</span>        }<a name="line.450"></a>
<span class="sourceLineNo">451</span><a name="line.451"></a>
<span class="sourceLineNo">452</span>        /**<a name="line.452"></a>
<span class="sourceLineNo">453</span>         * Calculate the sequence alignment Strings for the alignment Blocks in an<a name="line.453"></a>
<span class="sourceLineNo">454</span>         * alignment. This method creates a sequence alignment where all residues<a name="line.454"></a>
<span class="sourceLineNo">455</span>         * are in uppercase and a residue alone with gaps in all the other<a name="line.455"></a>
<span class="sourceLineNo">456</span>         * structures represents unaligned residues. Because of this latter<a name="line.456"></a>
<span class="sourceLineNo">457</span>         * constraint only the residues within the Blocks are represented, for a<a name="line.457"></a>
<span class="sourceLineNo">458</span>         * more compact alignment. For a sequence alignment of the full protein use<a name="line.458"></a>
<span class="sourceLineNo">459</span>         * {@link #getSequenceAlignment(MultipleAlignment)}.<a name="line.459"></a>
<span class="sourceLineNo">460</span>         * &lt;p&gt;<a name="line.460"></a>
<span class="sourceLineNo">461</span>         * Blocks are concatenated in the order returned by<a name="line.461"></a>
<span class="sourceLineNo">462</span>         * {@link MultipleAlignment#getBlocks()}, so sequences may not be<a name="line.462"></a>
<span class="sourceLineNo">463</span>         * sequential. Gaps between blocks are omitted, while gaps within blocks are<a name="line.463"></a>
<span class="sourceLineNo">464</span>         * represented by '-'. Separation between different Blocks is indicated by a<a name="line.464"></a>
<span class="sourceLineNo">465</span>         * gap in all positions, meaning that there is something unaligned<a name="line.465"></a>
<span class="sourceLineNo">466</span>         * inbetween.<a name="line.466"></a>
<span class="sourceLineNo">467</span>         *<a name="line.467"></a>
<span class="sourceLineNo">468</span>         * @param alignment<a name="line.468"></a>
<span class="sourceLineNo">469</span>         *            input MultipleAlignment<a name="line.469"></a>
<span class="sourceLineNo">470</span>         * @return String for each row in the alignment, giving the 1-letter code<a name="line.470"></a>
<span class="sourceLineNo">471</span>         *         for each aligned residue.<a name="line.471"></a>
<span class="sourceLineNo">472</span>         */<a name="line.472"></a>
<span class="sourceLineNo">473</span>        public static List&lt;String&gt; getBlockSequenceAlignment(MultipleAlignment ma) {<a name="line.473"></a>
<span class="sourceLineNo">474</span>                return getBlockSequenceAlignment(ma, new ArrayList&lt;Integer&gt;());<a name="line.474"></a>
<span class="sourceLineNo">475</span>        }<a name="line.475"></a>
<span class="sourceLineNo">476</span><a name="line.476"></a>
<span class="sourceLineNo">477</span>        /**<a name="line.477"></a>
<span class="sourceLineNo">478</span>         * Returns the Atom of the specified structure that is aligned in the<a name="line.478"></a>
<span class="sourceLineNo">479</span>         * sequence alignment position specified.<a name="line.479"></a>
<span class="sourceLineNo">480</span>         *<a name="line.480"></a>
<span class="sourceLineNo">481</span>         * @param multAln<a name="line.481"></a>
<span class="sourceLineNo">482</span>         *            the MultipleAlignment object from where the sequence alignment<a name="line.482"></a>
<span class="sourceLineNo">483</span>         *            has been generated<a name="line.483"></a>
<span class="sourceLineNo">484</span>         * @param mapSeqToStruct<a name="line.484"></a>
<span class="sourceLineNo">485</span>         *            the mapping between sequence and structure generated with the<a name="line.485"></a>
<span class="sourceLineNo">486</span>         *            sequence alignment<a name="line.486"></a>
<span class="sourceLineNo">487</span>         * @param str<a name="line.487"></a>
<span class="sourceLineNo">488</span>         *            the structure index of the alignment (row)<a name="line.488"></a>
<span class="sourceLineNo">489</span>         * @param sequencePos<a name="line.489"></a>
<span class="sourceLineNo">490</span>         *            the sequence alignment position (column)<a name="line.490"></a>
<span class="sourceLineNo">491</span>         * @return Atom the atom in that position or null if there is a gap<a name="line.491"></a>
<span class="sourceLineNo">492</span>         */<a name="line.492"></a>
<span class="sourceLineNo">493</span>        public static Atom getAtomForSequencePosition(MultipleAlignment msa,<a name="line.493"></a>
<span class="sourceLineNo">494</span>                        List&lt;Integer&gt; mapSeqToStruct, int str, int sequencePos) {<a name="line.494"></a>
<span class="sourceLineNo">495</span><a name="line.495"></a>
<span class="sourceLineNo">496</span>                int seqPos = mapSeqToStruct.get(sequencePos);<a name="line.496"></a>
<span class="sourceLineNo">497</span><a name="line.497"></a>
<span class="sourceLineNo">498</span>                // Check if the position selected is an aligned position<a name="line.498"></a>
<span class="sourceLineNo">499</span>                if (seqPos == -1)<a name="line.499"></a>
<span class="sourceLineNo">500</span>                        return null;<a name="line.500"></a>
<span class="sourceLineNo">501</span>                else {<a name="line.501"></a>
<span class="sourceLineNo">502</span>                        Atom a = null;<a name="line.502"></a>
<span class="sourceLineNo">503</span>                        // Calculate the corresponding structure position<a name="line.503"></a>
<span class="sourceLineNo">504</span>                        int sum = 0;<a name="line.504"></a>
<span class="sourceLineNo">505</span>                        for (Block b : msa.getBlocks()) {<a name="line.505"></a>
<span class="sourceLineNo">506</span>                                if (sum + b.length() &lt;= seqPos) {<a name="line.506"></a>
<span class="sourceLineNo">507</span>                                        sum += b.length();<a name="line.507"></a>
<span class="sourceLineNo">508</span>                                        continue;<a name="line.508"></a>
<span class="sourceLineNo">509</span>                                } else {<a name="line.509"></a>
<span class="sourceLineNo">510</span>                                        for (Integer p : b.getAlignRes().get(str)) {<a name="line.510"></a>
<span class="sourceLineNo">511</span>                                                if (sum == seqPos) {<a name="line.511"></a>
<span class="sourceLineNo">512</span>                                                        if (p != null) {<a name="line.512"></a>
<span class="sourceLineNo">513</span>                                                                a = msa.getAtomArrays().get(str)[p];<a name="line.513"></a>
<span class="sourceLineNo">514</span>                                                        }<a name="line.514"></a>
<span class="sourceLineNo">515</span>                                                        break;<a name="line.515"></a>
<span class="sourceLineNo">516</span>                                                }<a name="line.516"></a>
<span class="sourceLineNo">517</span>                                                sum++;<a name="line.517"></a>
<span class="sourceLineNo">518</span>                                        }<a name="line.518"></a>
<span class="sourceLineNo">519</span>                                        break;<a name="line.519"></a>
<span class="sourceLineNo">520</span>                                }<a name="line.520"></a>
<span class="sourceLineNo">521</span>                        }<a name="line.521"></a>
<span class="sourceLineNo">522</span>                        return a;<a name="line.522"></a>
<span class="sourceLineNo">523</span>                }<a name="line.523"></a>
<span class="sourceLineNo">524</span>        }<a name="line.524"></a>
<span class="sourceLineNo">525</span><a name="line.525"></a>
<span class="sourceLineNo">526</span>        /**<a name="line.526"></a>
<span class="sourceLineNo">527</span>         * Returns the block number of a specified position in the sequence<a name="line.527"></a>
<span class="sourceLineNo">528</span>         * alignment, given the mapping from structure to function.<a name="line.528"></a>
<span class="sourceLineNo">529</span>         *<a name="line.529"></a>
<span class="sourceLineNo">530</span>         * @param multAln<a name="line.530"></a>
<span class="sourceLineNo">531</span>         *            the MultipleAlignment object from where the sequence alignment<a name="line.531"></a>
<span class="sourceLineNo">532</span>         *            has been generated.<a name="line.532"></a>
<span class="sourceLineNo">533</span>         * @param mapSeqToStruct<a name="line.533"></a>
<span class="sourceLineNo">534</span>         *            the mapping between sequence and structure generated with the<a name="line.534"></a>
<span class="sourceLineNo">535</span>         *            sequence alignment<a name="line.535"></a>
<span class="sourceLineNo">536</span>         * @param sequencePos<a name="line.536"></a>
<span class="sourceLineNo">537</span>         *            the position in the sequence alignment (column)<a name="line.537"></a>
<span class="sourceLineNo">538</span>         * @return int the block index, or -1 if the position is not aligned<a name="line.538"></a>
<span class="sourceLineNo">539</span>         */<a name="line.539"></a>
<span class="sourceLineNo">540</span>        public static int getBlockForSequencePosition(MultipleAlignment multAln,<a name="line.540"></a>
<span class="sourceLineNo">541</span>                        List&lt;Integer&gt; mapSeqToStruct, int sequencePos) {<a name="line.541"></a>
<span class="sourceLineNo">542</span><a name="line.542"></a>
<span class="sourceLineNo">543</span>                int seqPos = mapSeqToStruct.get(sequencePos);<a name="line.543"></a>
<span class="sourceLineNo">544</span>                // Check if the position selected is an aligned position<a name="line.544"></a>
<span class="sourceLineNo">545</span>                if (seqPos == -1)<a name="line.545"></a>
<span class="sourceLineNo">546</span>                        return -1;<a name="line.546"></a>
<span class="sourceLineNo">547</span>                else {<a name="line.547"></a>
<span class="sourceLineNo">548</span>                        // Calculate the corresponding block (by iterating all Blocks)<a name="line.548"></a>
<span class="sourceLineNo">549</span>                        int sum = 0;<a name="line.549"></a>
<span class="sourceLineNo">550</span>                        int block = 0;<a name="line.550"></a>
<span class="sourceLineNo">551</span>                        for (Block b : multAln.getBlocks()) {<a name="line.551"></a>
<span class="sourceLineNo">552</span>                                if (sum + b.length() &lt;= seqPos) {<a name="line.552"></a>
<span class="sourceLineNo">553</span>                                        sum += b.length();<a name="line.553"></a>
<span class="sourceLineNo">554</span>                                        block++;<a name="line.554"></a>
<span class="sourceLineNo">555</span>                                        continue;<a name="line.555"></a>
<span class="sourceLineNo">556</span>                                } else<a name="line.556"></a>
<span class="sourceLineNo">557</span>                                        break;<a name="line.557"></a>
<span class="sourceLineNo">558</span>                        }<a name="line.558"></a>
<span class="sourceLineNo">559</span>                        return block;<a name="line.559"></a>
<span class="sourceLineNo">560</span>                }<a name="line.560"></a>
<span class="sourceLineNo">561</span>        }<a name="line.561"></a>
<span class="sourceLineNo">562</span><a name="line.562"></a>
<span class="sourceLineNo">563</span>        /**<a name="line.563"></a>
<span class="sourceLineNo">564</span>         * The average residue distance Matrix contains the average distance from<a name="line.564"></a>
<span class="sourceLineNo">565</span>         * each residue to all other residues aligned with it.<a name="line.565"></a>
<span class="sourceLineNo">566</span>         * &lt;p&gt;<a name="line.566"></a>
<span class="sourceLineNo">567</span>         * Complexity: T(n,l) = O(l*n^2), if n=number of structures and l=alignment<a name="line.567"></a>
<span class="sourceLineNo">568</span>         * length.<a name="line.568"></a>
<span class="sourceLineNo">569</span>         *<a name="line.569"></a>
<span class="sourceLineNo">570</span>         * @param alignment<a name="line.570"></a>
<span class="sourceLineNo">571</span>         *            MultipleAlignment<a name="line.571"></a>
<span class="sourceLineNo">572</span>         * @return Matrix containing all average residue distances<a name="line.572"></a>
<span class="sourceLineNo">573</span>         */<a name="line.573"></a>
<span class="sourceLineNo">574</span>        public static Matrix getAverageResidueDistances(MultipleAlignment msa) {<a name="line.574"></a>
<span class="sourceLineNo">575</span>                List&lt;Atom[]&gt; transformed = transformAtoms(msa);<a name="line.575"></a>
<span class="sourceLineNo">576</span>                return getAverageResidueDistances(transformed);<a name="line.576"></a>
<span class="sourceLineNo">577</span>        }<a name="line.577"></a>
<span class="sourceLineNo">578</span><a name="line.578"></a>
<span class="sourceLineNo">579</span>        /**<a name="line.579"></a>
<span class="sourceLineNo">580</span>         * The average residue distance Matrix contains the average distance from<a name="line.580"></a>
<span class="sourceLineNo">581</span>         * each residue to all other residues aligned with it.<a name="line.581"></a>
<span class="sourceLineNo">582</span>         * &lt;p&gt;<a name="line.582"></a>
<span class="sourceLineNo">583</span>         * Complexity: T(n,l) = O(l*n^2), if n=number of structures and l=alignment<a name="line.583"></a>
<span class="sourceLineNo">584</span>         * length.<a name="line.584"></a>
<span class="sourceLineNo">585</span>         *<a name="line.585"></a>
<span class="sourceLineNo">586</span>         * @param transformed<a name="line.586"></a>
<span class="sourceLineNo">587</span>         *            List of Atom arrays containing only the aligned atoms of each<a name="line.587"></a>
<span class="sourceLineNo">588</span>         *            structure, or null if there is a gap.<a name="line.588"></a>
<span class="sourceLineNo">589</span>         * @return Matrix containing all average residue distances. Entry -1 means<a name="line.589"></a>
<span class="sourceLineNo">590</span>         *         there is a gap in the position.<a name="line.590"></a>
<span class="sourceLineNo">591</span>         */<a name="line.591"></a>
<span class="sourceLineNo">592</span>        public static Matrix getAverageResidueDistances(List&lt;Atom[]&gt; transformed) {<a name="line.592"></a>
<span class="sourceLineNo">593</span><a name="line.593"></a>
<span class="sourceLineNo">594</span>                int size = transformed.size();<a name="line.594"></a>
<span class="sourceLineNo">595</span>                int length = transformed.get(0).length;<a name="line.595"></a>
<span class="sourceLineNo">596</span>                Matrix resDist = new Matrix(size, length, -1);<a name="line.596"></a>
<span class="sourceLineNo">597</span><a name="line.597"></a>
<span class="sourceLineNo">598</span>                // Calculate the average residue distances<a name="line.598"></a>
<span class="sourceLineNo">599</span>                for (int r1 = 0; r1 &lt; size; r1++) {<a name="line.599"></a>
<span class="sourceLineNo">600</span>                        for (int c = 0; c &lt; transformed.get(r1).length; c++) {<a name="line.600"></a>
<span class="sourceLineNo">601</span>                                Atom refAtom = transformed.get(r1)[c];<a name="line.601"></a>
<span class="sourceLineNo">602</span>                                if (refAtom == null)<a name="line.602"></a>
<span class="sourceLineNo">603</span>                                        continue;<a name="line.603"></a>
<span class="sourceLineNo">604</span><a name="line.604"></a>
<span class="sourceLineNo">605</span>                                for (int r2 = r1 + 1; r2 &lt; size; r2++) {<a name="line.605"></a>
<span class="sourceLineNo">606</span>                                        Atom atom = transformed.get(r2)[c];<a name="line.606"></a>
<span class="sourceLineNo">607</span>                                        if (atom != null) {<a name="line.607"></a>
<span class="sourceLineNo">608</span>                                                double distance = Calc.getDistance(refAtom, atom);<a name="line.608"></a>
<span class="sourceLineNo">609</span><a name="line.609"></a>
<span class="sourceLineNo">610</span>                                                if (resDist.get(r1, c) == -1) {<a name="line.610"></a>
<span class="sourceLineNo">611</span>                                                        resDist.set(r1, c, 1 + distance);<a name="line.611"></a>
<span class="sourceLineNo">612</span>                                                } else {<a name="line.612"></a>
<span class="sourceLineNo">613</span>                                                        resDist.set(r1, c, resDist.get(r1, c) + distance);<a name="line.613"></a>
<span class="sourceLineNo">614</span>                                                }<a name="line.614"></a>
<span class="sourceLineNo">615</span><a name="line.615"></a>
<span class="sourceLineNo">616</span>                                                if (resDist.get(r2, c) == -1) {<a name="line.616"></a>
<span class="sourceLineNo">617</span>                                                        resDist.set(r2, c, 1 + distance);<a name="line.617"></a>
<span class="sourceLineNo">618</span>                                                } else {<a name="line.618"></a>
<span class="sourceLineNo">619</span>                                                        resDist.set(r2, c, resDist.get(r2, c) + distance);<a name="line.619"></a>
<span class="sourceLineNo">620</span>                                                }<a name="line.620"></a>
<span class="sourceLineNo">621</span>                                        }<a name="line.621"></a>
<span class="sourceLineNo">622</span>                                }<a name="line.622"></a>
<span class="sourceLineNo">623</span>                        }<a name="line.623"></a>
<span class="sourceLineNo">624</span>                }<a name="line.624"></a>
<span class="sourceLineNo">625</span><a name="line.625"></a>
<span class="sourceLineNo">626</span>                for (int c = 0; c &lt; length; c++) {<a name="line.626"></a>
<span class="sourceLineNo">627</span>                        int nonNullRes = 0;<a name="line.627"></a>
<span class="sourceLineNo">628</span>                        for (int r = 0; r &lt; size; r++) {<a name="line.628"></a>
<span class="sourceLineNo">629</span>                                if (resDist.get(r, c) != -1)<a name="line.629"></a>
<span class="sourceLineNo">630</span>                                        nonNullRes++;<a name="line.630"></a>
<span class="sourceLineNo">631</span>                        }<a name="line.631"></a>
<span class="sourceLineNo">632</span>                        for (int r = 0; r &lt; size; r++) {<a name="line.632"></a>
<span class="sourceLineNo">633</span>                                if (resDist.get(r, c) != -1) {<a name="line.633"></a>
<span class="sourceLineNo">634</span>                                        resDist.set(r, c, resDist.get(r, c) / nonNullRes);<a name="line.634"></a>
<span class="sourceLineNo">635</span>                                }<a name="line.635"></a>
<span class="sourceLineNo">636</span>                        }<a name="line.636"></a>
<span class="sourceLineNo">637</span>                }<a name="line.637"></a>
<span class="sourceLineNo">638</span>                return resDist;<a name="line.638"></a>
<span class="sourceLineNo">639</span>        }<a name="line.639"></a>
<span class="sourceLineNo">640</span><a name="line.640"></a>
<span class="sourceLineNo">641</span>        /**<a name="line.641"></a>
<span class="sourceLineNo">642</span>         * Transforms atoms according to the superposition stored in the alignment.<a name="line.642"></a>
<span class="sourceLineNo">643</span>         * &lt;p&gt;<a name="line.643"></a>
<span class="sourceLineNo">644</span>         * For each structure in the alignment, returns an atom for each<a name="line.644"></a>
<span class="sourceLineNo">645</span>         * representative atom in the aligned columns, omitting unaligned residues<a name="line.645"></a>
<span class="sourceLineNo">646</span>         * (i.e. an array of length &lt;tt&gt;alignment.length()&lt;/tt&gt; ).<a name="line.646"></a>
<span class="sourceLineNo">647</span>         * &lt;p&gt;<a name="line.647"></a>
<span class="sourceLineNo">648</span>         * All blocks are concatenated together, so Atoms may not appear in the same<a name="line.648"></a>
<span class="sourceLineNo">649</span>         * order as in their parent structure. If the alignment blocks contain null<a name="line.649"></a>
<span class="sourceLineNo">650</span>         * residues (gaps), then the returned array will also contain null Atoms in<a name="line.650"></a>
<span class="sourceLineNo">651</span>         * the same positions.<a name="line.651"></a>
<span class="sourceLineNo">652</span>         *<a name="line.652"></a>
<span class="sourceLineNo">653</span>         * @param alignment<a name="line.653"></a>
<span class="sourceLineNo">654</span>         *            MultipleAlignment<a name="line.654"></a>
<span class="sourceLineNo">655</span>         * @return List of Atom arrays of only the aligned atoms of every structure<a name="line.655"></a>
<span class="sourceLineNo">656</span>         *         (null Atom if a gap position)<a name="line.656"></a>
<span class="sourceLineNo">657</span>         */<a name="line.657"></a>
<span class="sourceLineNo">658</span>        public static List&lt;Atom[]&gt; transformAtoms(MultipleAlignment alignment) {<a name="line.658"></a>
<span class="sourceLineNo">659</span>                if (alignment.getEnsemble() == null) {<a name="line.659"></a>
<span class="sourceLineNo">660</span>                        throw new NullPointerException("No ensemble set for this alignment");<a name="line.660"></a>
<span class="sourceLineNo">661</span>                }<a name="line.661"></a>
<span class="sourceLineNo">662</span><a name="line.662"></a>
<span class="sourceLineNo">663</span>                List&lt;Atom[]&gt; atomArrays = alignment.getAtomArrays();<a name="line.663"></a>
<span class="sourceLineNo">664</span>                List&lt;Atom[]&gt; transformed = new ArrayList&lt;Atom[]&gt;(atomArrays.size());<a name="line.664"></a>
<span class="sourceLineNo">665</span><a name="line.665"></a>
<span class="sourceLineNo">666</span>                // Loop through structures<a name="line.666"></a>
<span class="sourceLineNo">667</span>                for (int i = 0; i &lt; atomArrays.size(); i++) {<a name="line.667"></a>
<span class="sourceLineNo">668</span><a name="line.668"></a>
<span class="sourceLineNo">669</span>                        Matrix4d transform = null;<a name="line.669"></a>
<span class="sourceLineNo">670</span>                        Atom[] curr = atomArrays.get(i); // all CA atoms from structure<a name="line.670"></a>
<span class="sourceLineNo">671</span><a name="line.671"></a>
<span class="sourceLineNo">672</span>                        // Concatenated list of all blocks for this structure<a name="line.672"></a>
<span class="sourceLineNo">673</span>                        Atom[] transformedAtoms = new Atom[alignment.length()];<a name="line.673"></a>
<span class="sourceLineNo">674</span>                        int transformedAtomsLength = 0;<a name="line.674"></a>
<span class="sourceLineNo">675</span><a name="line.675"></a>
<span class="sourceLineNo">676</span>                        // Each blockset gets transformed independently<a name="line.676"></a>
<span class="sourceLineNo">677</span>                        for (BlockSet bs : alignment.getBlockSets()) {<a name="line.677"></a>
<span class="sourceLineNo">678</span><a name="line.678"></a>
<span class="sourceLineNo">679</span>                                Atom[] blocksetAtoms = new Atom[bs.length()];<a name="line.679"></a>
<span class="sourceLineNo">680</span>                                int blockPos = 0;<a name="line.680"></a>
<span class="sourceLineNo">681</span><a name="line.681"></a>
<span class="sourceLineNo">682</span>                                for (Block blk : bs.getBlocks()) {<a name="line.682"></a>
<span class="sourceLineNo">683</span>                                        if (blk.size() != atomArrays.size()) {<a name="line.683"></a>
<span class="sourceLineNo">684</span>                                                throw new IllegalStateException(String.format(<a name="line.684"></a>
<span class="sourceLineNo">685</span>                                                                "Mismatched block size. Expected %d "<a name="line.685"></a>
<span class="sourceLineNo">686</span>                                                                                + "structures, found %d.",<a name="line.686"></a>
<span class="sourceLineNo">687</span>                                                                atomArrays.size(), blk.size()));<a name="line.687"></a>
<span class="sourceLineNo">688</span>                                        }<a name="line.688"></a>
<span class="sourceLineNo">689</span>                                        // Extract aligned atoms<a name="line.689"></a>
<span class="sourceLineNo">690</span>                                        for (int j = 0; j &lt; blk.length(); j++) {<a name="line.690"></a>
<span class="sourceLineNo">691</span>                                                Integer alignedPos = blk.getAlignRes().get(i).get(j);<a name="line.691"></a>
<span class="sourceLineNo">692</span>                                                if (alignedPos != null) {<a name="line.692"></a>
<span class="sourceLineNo">693</span>                                                        blocksetAtoms[blockPos] = (Atom) curr[alignedPos]<a name="line.693"></a>
<span class="sourceLineNo">694</span>                                                                        .clone();<a name="line.694"></a>
<span class="sourceLineNo">695</span>                                                }<a name="line.695"></a>
<span class="sourceLineNo">696</span>                                                blockPos++;<a name="line.696"></a>
<span class="sourceLineNo">697</span>                                        }<a name="line.697"></a>
<span class="sourceLineNo">698</span>                                }<a name="line.698"></a>
<span class="sourceLineNo">699</span><a name="line.699"></a>
<span class="sourceLineNo">700</span>                                // Transform according to the blockset or alignment matrix<a name="line.700"></a>
<span class="sourceLineNo">701</span>                                Matrix4d blockTrans = null;<a name="line.701"></a>
<span class="sourceLineNo">702</span>                                if (bs.getTransformations() != null)<a name="line.702"></a>
<span class="sourceLineNo">703</span>                                        blockTrans = bs.getTransformations().get(i);<a name="line.703"></a>
<span class="sourceLineNo">704</span>                                if (blockTrans == null) {<a name="line.704"></a>
<span class="sourceLineNo">705</span>                                        blockTrans = transform;<a name="line.705"></a>
<span class="sourceLineNo">706</span>                                }<a name="line.706"></a>
<span class="sourceLineNo">707</span><a name="line.707"></a>
<span class="sourceLineNo">708</span>                                for (Atom a : blocksetAtoms) {<a name="line.708"></a>
<span class="sourceLineNo">709</span>                                        if (a != null)<a name="line.709"></a>
<span class="sourceLineNo">710</span>                                                Calc.transform(a, blockTrans);<a name="line.710"></a>
<span class="sourceLineNo">711</span>                                        transformedAtoms[transformedAtomsLength] = a;<a name="line.711"></a>
<span class="sourceLineNo">712</span>                                        transformedAtomsLength++;<a name="line.712"></a>
<span class="sourceLineNo">713</span>                                }<a name="line.713"></a>
<span class="sourceLineNo">714</span>                        }<a name="line.714"></a>
<span class="sourceLineNo">715</span>                        assert (transformedAtomsLength == alignment.length());<a name="line.715"></a>
<span class="sourceLineNo">716</span><a name="line.716"></a>
<span class="sourceLineNo">717</span>                        transformed.add(transformedAtoms);<a name="line.717"></a>
<span class="sourceLineNo">718</span>                }<a name="line.718"></a>
<span class="sourceLineNo">719</span>                return transformed;<a name="line.719"></a>
<span class="sourceLineNo">720</span>        }<a name="line.720"></a>
<span class="sourceLineNo">721</span><a name="line.721"></a>
<span class="sourceLineNo">722</span>        /**<a name="line.722"></a>
<span class="sourceLineNo">723</span>         * Calculate a List of alignment indicies that correspond to the core of a<a name="line.723"></a>
<span class="sourceLineNo">724</span>         * Block, which means that all structures have a residue in that positon.<a name="line.724"></a>
<span class="sourceLineNo">725</span>         *<a name="line.725"></a>
<span class="sourceLineNo">726</span>         * @param block<a name="line.726"></a>
<span class="sourceLineNo">727</span>         *            alignment Block<a name="line.727"></a>
<span class="sourceLineNo">728</span>         * @return List of positions in the core of the alignment<a name="line.728"></a>
<span class="sourceLineNo">729</span>         */<a name="line.729"></a>
<span class="sourceLineNo">730</span>        public static List&lt;Integer&gt; getCorePositions(Block block) {<a name="line.730"></a>
<span class="sourceLineNo">731</span><a name="line.731"></a>
<span class="sourceLineNo">732</span>                List&lt;Integer&gt; corePositions = new ArrayList&lt;Integer&gt;();<a name="line.732"></a>
<span class="sourceLineNo">733</span><a name="line.733"></a>
<span class="sourceLineNo">734</span>                for (int col = 0; col &lt; block.length(); col++) {<a name="line.734"></a>
<span class="sourceLineNo">735</span>                        boolean core = true;<a name="line.735"></a>
<span class="sourceLineNo">736</span>                        for (int str = 0; str &lt; block.size(); str++) {<a name="line.736"></a>
<span class="sourceLineNo">737</span>                                if (block.getAlignRes().get(str).get(col) == null) {<a name="line.737"></a>
<span class="sourceLineNo">738</span>                                        core = false;<a name="line.738"></a>
<span class="sourceLineNo">739</span>                                        break;<a name="line.739"></a>
<span class="sourceLineNo">740</span>                                }<a name="line.740"></a>
<span class="sourceLineNo">741</span>                        }<a name="line.741"></a>
<span class="sourceLineNo">742</span>                        if (core)<a name="line.742"></a>
<span class="sourceLineNo">743</span>                                corePositions.add(col);<a name="line.743"></a>
<span class="sourceLineNo">744</span>                }<a name="line.744"></a>
<span class="sourceLineNo">745</span>                return corePositions;<a name="line.745"></a>
<span class="sourceLineNo">746</span>        }<a name="line.746"></a>
<span class="sourceLineNo">747</span><a name="line.747"></a>
<span class="sourceLineNo">748</span>        /**<a name="line.748"></a>
<span class="sourceLineNo">749</span>         * Sort blocks so that the specified row is in sequential order. The sort<a name="line.749"></a>
<span class="sourceLineNo">750</span>         * happens in place.<a name="line.750"></a>
<span class="sourceLineNo">751</span>         *<a name="line.751"></a>
<span class="sourceLineNo">752</span>         * @param blocks<a name="line.752"></a>
<span class="sourceLineNo">753</span>         *            List of blocks<a name="line.753"></a>
<span class="sourceLineNo">754</span>         * @param sortedIndex<a name="line.754"></a>
<span class="sourceLineNo">755</span>         *            Index of the row to be sorted<a name="line.755"></a>
<span class="sourceLineNo">756</span>         */<a name="line.756"></a>
<span class="sourceLineNo">757</span>        public static void sortBlocks(List&lt;Block&gt; blocks, final int sortedIndex) {<a name="line.757"></a>
<span class="sourceLineNo">758</span>                Collections.sort(blocks, new Comparator&lt;Block&gt;() {<a name="line.758"></a>
<span class="sourceLineNo">759</span>                        @Override<a name="line.759"></a>
<span class="sourceLineNo">760</span>                        public int compare(Block o1, Block o2) {<a name="line.760"></a>
<span class="sourceLineNo">761</span>                                // Compare the first non-null residue of each block<a name="line.761"></a>
<span class="sourceLineNo">762</span>                                List&lt;Integer&gt; alignres1 = o1.getAlignRes().get(sortedIndex);<a name="line.762"></a>
<span class="sourceLineNo">763</span>                                List&lt;Integer&gt; alignres2 = o2.getAlignRes().get(sortedIndex);<a name="line.763"></a>
<span class="sourceLineNo">764</span>                                Integer res1 = null;<a name="line.764"></a>
<span class="sourceLineNo">765</span>                                Integer res2 = null;<a name="line.765"></a>
<span class="sourceLineNo">766</span>                                for (Integer r : alignres1) {<a name="line.766"></a>
<span class="sourceLineNo">767</span>                                        if (r != null) {<a name="line.767"></a>
<span class="sourceLineNo">768</span>                                                res1 = r;<a name="line.768"></a>
<span class="sourceLineNo">769</span>                                                break;<a name="line.769"></a>
<span class="sourceLineNo">770</span>                                        }<a name="line.770"></a>
<span class="sourceLineNo">771</span>                                }<a name="line.771"></a>
<span class="sourceLineNo">772</span>                                for (Integer r : alignres2) {<a name="line.772"></a>
<span class="sourceLineNo">773</span>                                        if (r != null) {<a name="line.773"></a>
<span class="sourceLineNo">774</span>                                                res2 = r;<a name="line.774"></a>
<span class="sourceLineNo">775</span>                                                break;<a name="line.775"></a>
<span class="sourceLineNo">776</span>                                        }<a name="line.776"></a>
<span class="sourceLineNo">777</span>                                }<a name="line.777"></a>
<span class="sourceLineNo">778</span>                                return res1.compareTo(res2);<a name="line.778"></a>
<span class="sourceLineNo">779</span>                        }<a name="line.779"></a>
<span class="sourceLineNo">780</span>                });<a name="line.780"></a>
<span class="sourceLineNo">781</span>        }<a name="line.781"></a>
<span class="sourceLineNo">782</span><a name="line.782"></a>
<span class="sourceLineNo">783</span>        /**<a name="line.783"></a>
<span class="sourceLineNo">784</span>         * Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid<a name="line.784"></a>
<span class="sourceLineNo">785</span>         * residues. This method is only valid for protein structure alignments.<a name="line.785"></a>
<span class="sourceLineNo">786</span>         *<a name="line.786"></a>
<span class="sourceLineNo">787</span>         * @param msta<a name="line.787"></a>
<span class="sourceLineNo">788</span>         *            Multiple Structure Alignment<a name="line.788"></a>
<span class="sourceLineNo">789</span>         * @return MultipleSequenceAlignment of protein sequences<a name="line.789"></a>
<span class="sourceLineNo">790</span>         * @throws CompoundNotFoundException<a name="line.790"></a>
<span class="sourceLineNo">791</span>         */<a name="line.791"></a>
<span class="sourceLineNo">792</span>        public static MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; toProteinMSA(<a name="line.792"></a>
<span class="sourceLineNo">793</span>                        MultipleAlignment msta) throws CompoundNotFoundException {<a name="line.793"></a>
<span class="sourceLineNo">794</span><a name="line.794"></a>
<span class="sourceLineNo">795</span>                // Check that the alignment is of protein structures<a name="line.795"></a>
<span class="sourceLineNo">796</span>                Group g = msta.getAtomArrays().get(0)[0].getGroup();<a name="line.796"></a>
<span class="sourceLineNo">797</span>                if (!(g instanceof AminoAcid)) {<a name="line.797"></a>
<span class="sourceLineNo">798</span>                        throw new IllegalArgumentException(<a name="line.798"></a>
<span class="sourceLineNo">799</span>                                        "Cannot convert to multiple sequence alignment: "<a name="line.799"></a>
<span class="sourceLineNo">800</span>                                                        + "the structures aligned are not proteins");<a name="line.800"></a>
<span class="sourceLineNo">801</span>                }<a name="line.801"></a>
<span class="sourceLineNo">802</span><a name="line.802"></a>
<span class="sourceLineNo">803</span>                MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; msa = new MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt;();<a name="line.803"></a>
<span class="sourceLineNo">804</span><a name="line.804"></a>
<span class="sourceLineNo">805</span>                Map&lt;String, Integer&gt; uniqueID = new HashMap&lt;String, Integer&gt;();<a name="line.805"></a>
<span class="sourceLineNo">806</span>                List&lt;String&gt; seqs = getSequenceAlignment(msta);<a name="line.806"></a>
<span class="sourceLineNo">807</span>                for (int i = 0; i &lt; msta.size(); i++) {<a name="line.807"></a>
<span class="sourceLineNo">808</span>                        // Make sure the identifiers are unique (required by AccessionID)<a name="line.808"></a>
<span class="sourceLineNo">809</span>                        String id = msta.getStructureIdentifier(i).toString();<a name="line.809"></a>
<span class="sourceLineNo">810</span>                        if (uniqueID.containsKey(id)) {<a name="line.810"></a>
<span class="sourceLineNo">811</span>                                uniqueID.put(id, uniqueID.get(id) + 1);<a name="line.811"></a>
<span class="sourceLineNo">812</span>                                id += "_" + uniqueID.get(id);<a name="line.812"></a>
<span class="sourceLineNo">813</span>                        } else<a name="line.813"></a>
<span class="sourceLineNo">814</span>                                uniqueID.put(id, 1);<a name="line.814"></a>
<span class="sourceLineNo">815</span><a name="line.815"></a>
<span class="sourceLineNo">816</span>                        AccessionID acc = new AccessionID(id);<a name="line.816"></a>
<span class="sourceLineNo">817</span>                        ProteinSequence pseq = new ProteinSequence(seqs.get(i));<a name="line.817"></a>
<span class="sourceLineNo">818</span>                        pseq.setAccession(acc);<a name="line.818"></a>
<span class="sourceLineNo">819</span>                        msa.addAlignedSequence(pseq);<a name="line.819"></a>
<span class="sourceLineNo">820</span>                }<a name="line.820"></a>
<span class="sourceLineNo">821</span>                return msa;<a name="line.821"></a>
<span class="sourceLineNo">822</span>        }<a name="line.822"></a>
<span class="sourceLineNo">823</span><a name="line.823"></a>
<span class="sourceLineNo">824</span>        public static Structure toMultimodelStructure(MultipleAlignment multAln, List&lt;Atom[]&gt; transformedAtoms) throws StructureException {<a name="line.824"></a>
<span class="sourceLineNo">825</span>                PDBHeader header = new PDBHeader();<a name="line.825"></a>
<span class="sourceLineNo">826</span>                String title = multAln.getEnsemble().getAlgorithmName() + " V."<a name="line.826"></a>
<span class="sourceLineNo">827</span>                                + multAln.getEnsemble().getVersion() + " : ";<a name="line.827"></a>
<span class="sourceLineNo">828</span><a name="line.828"></a>
<span class="sourceLineNo">829</span>                for (StructureIdentifier name : multAln.getEnsemble()<a name="line.829"></a>
<span class="sourceLineNo">830</span>                                .getStructureIdentifiers()) {<a name="line.830"></a>
<span class="sourceLineNo">831</span>                        title += name.getIdentifier() + " ";<a name="line.831"></a>
<span class="sourceLineNo">832</span>                }<a name="line.832"></a>
<span class="sourceLineNo">833</span>                Structure artificial = getAlignedStructure(transformedAtoms);<a name="line.833"></a>
<span class="sourceLineNo">834</span><a name="line.834"></a>
<span class="sourceLineNo">835</span>                artificial.setPDBHeader(header);<a name="line.835"></a>
<span class="sourceLineNo">836</span>                header.setTitle(title);<a name="line.836"></a>
<span class="sourceLineNo">837</span>                return artificial;<a name="line.837"></a>
<span class="sourceLineNo">838</span>        }<a name="line.838"></a>
<span class="sourceLineNo">839</span><a name="line.839"></a>
<span class="sourceLineNo">840</span>        /**<a name="line.840"></a>
<span class="sourceLineNo">841</span>         * Get an artificial Structure containing a different model for every<a name="line.841"></a>
<span class="sourceLineNo">842</span>         * input structure, so that the alignment result can be viewed in Jmol.<a name="line.842"></a>
<span class="sourceLineNo">843</span>         * The Atoms have to be rotated beforehand.<a name="line.843"></a>
<span class="sourceLineNo">844</span>         *<a name="line.844"></a>
<span class="sourceLineNo">845</span>         * @param atomArrays an array of Atoms for every aligned structure<a name="line.845"></a>
<span class="sourceLineNo">846</span>         * @return a structure object containing a set of models,<a name="line.846"></a>
<span class="sourceLineNo">847</span>         *                      one for each input array of Atoms.<a name="line.847"></a>
<span class="sourceLineNo">848</span>         * @throws StructureException<a name="line.848"></a>
<span class="sourceLineNo">849</span>         */<a name="line.849"></a>
<span class="sourceLineNo">850</span>        public static final Structure getAlignedStructure(List&lt;Atom[]&gt; atomArrays)<a name="line.850"></a>
<span class="sourceLineNo">851</span>                        throws StructureException {<a name="line.851"></a>
<span class="sourceLineNo">852</span><a name="line.852"></a>
<span class="sourceLineNo">853</span>                Structure s = new StructureImpl();<a name="line.853"></a>
<span class="sourceLineNo">854</span>                for (int i=0; i&lt;atomArrays.size(); i++){<a name="line.854"></a>
<span class="sourceLineNo">855</span>                        List&lt;Chain&gt; model = AlignmentTools.getAlignedModel(atomArrays.get(i));<a name="line.855"></a>
<span class="sourceLineNo">856</span>                        s.addModel(model);<a name="line.856"></a>
<span class="sourceLineNo">857</span>                }<a name="line.857"></a>
<span class="sourceLineNo">858</span>                return s;<a name="line.858"></a>
<span class="sourceLineNo">859</span>        }<a name="line.859"></a>
<span class="sourceLineNo">860</span><a name="line.860"></a>
<span class="sourceLineNo">861</span>        /**<a name="line.861"></a>
<span class="sourceLineNo">862</span>         * Calculate the RMSD matrix of a MultipleAlignment, that is, entry (i,j) of<a name="line.862"></a>
<span class="sourceLineNo">863</span>         * the matrix contains the RMSD between structures i and j.<a name="line.863"></a>
<span class="sourceLineNo">864</span>         *<a name="line.864"></a>
<span class="sourceLineNo">865</span>         * @param msa<a name="line.865"></a>
<span class="sourceLineNo">866</span>         *            Multiple Structure Alignment<a name="line.866"></a>
<span class="sourceLineNo">867</span>         * @return Matrix of RMSD with size the number of structures squared<a name="line.867"></a>
<span class="sourceLineNo">868</span>         */<a name="line.868"></a>
<span class="sourceLineNo">869</span>        public static Matrix getRMSDMatrix(MultipleAlignment msa) {<a name="line.869"></a>
<span class="sourceLineNo">870</span><a name="line.870"></a>
<span class="sourceLineNo">871</span>                Matrix rmsdMat = new Matrix(msa.size(), msa.size());<a name="line.871"></a>
<span class="sourceLineNo">872</span>                List&lt;Atom[]&gt; superposed = transformAtoms(msa);<a name="line.872"></a>
<span class="sourceLineNo">873</span><a name="line.873"></a>
<span class="sourceLineNo">874</span>                for (int i = 0; i &lt; msa.size(); i++) {<a name="line.874"></a>
<span class="sourceLineNo">875</span>                        for (int j = i; j &lt; msa.size(); j++) {<a name="line.875"></a>
<span class="sourceLineNo">876</span>                                if (i == j)<a name="line.876"></a>
<span class="sourceLineNo">877</span>                                        rmsdMat.set(i, j, 0.0);<a name="line.877"></a>
<span class="sourceLineNo">878</span>                                List&lt;Atom[]&gt; compared = new ArrayList&lt;Atom[]&gt;();<a name="line.878"></a>
<span class="sourceLineNo">879</span>                                compared.add(superposed.get(i));<a name="line.879"></a>
<span class="sourceLineNo">880</span>                                compared.add(superposed.get(j));<a name="line.880"></a>
<span class="sourceLineNo">881</span>                                double rmsd = MultipleAlignmentScorer.getRMSD(compared);<a name="line.881"></a>
<span class="sourceLineNo">882</span>                                rmsdMat.set(i, j, rmsd);<a name="line.882"></a>
<span class="sourceLineNo">883</span>                                rmsdMat.set(j, i, rmsd);<a name="line.883"></a>
<span class="sourceLineNo">884</span>                        }<a name="line.884"></a>
<span class="sourceLineNo">885</span>                }<a name="line.885"></a>
<span class="sourceLineNo">886</span>                return rmsdMat;<a name="line.886"></a>
<span class="sourceLineNo">887</span>        }<a name="line.887"></a>
<span class="sourceLineNo">888</span><a name="line.888"></a>
<span class="sourceLineNo">889</span>        /**<a name="line.889"></a>
<span class="sourceLineNo">890</span>         * Calculate a phylogenetic tree of the MultipleAlignment using Kimura<a name="line.890"></a>
<span class="sourceLineNo">891</span>         * distances and the Neighbor Joining algorithm from forester.<a name="line.891"></a>
<span class="sourceLineNo">892</span>         *<a name="line.892"></a>
<span class="sourceLineNo">893</span>         * @param msta<a name="line.893"></a>
<span class="sourceLineNo">894</span>         *            MultipleAlignment of protein structures<a name="line.894"></a>
<span class="sourceLineNo">895</span>         * @return Phylogeny phylogenetic tree<a name="line.895"></a>
<span class="sourceLineNo">896</span>         * @throws CompoundNotFoundException<a name="line.896"></a>
<span class="sourceLineNo">897</span>         * @throws IOException<a name="line.897"></a>
<span class="sourceLineNo">898</span>         */<a name="line.898"></a>
<span class="sourceLineNo">899</span>        public static Phylogeny getKimuraTree(MultipleAlignment msta)<a name="line.899"></a>
<span class="sourceLineNo">900</span>                        throws CompoundNotFoundException, IOException {<a name="line.900"></a>
<span class="sourceLineNo">901</span>                MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; msa = MultipleAlignmentTools<a name="line.901"></a>
<span class="sourceLineNo">902</span>                                .toProteinMSA(msta);<a name="line.902"></a>
<span class="sourceLineNo">903</span>                BasicSymmetricalDistanceMatrix distmat = (BasicSymmetricalDistanceMatrix) DistanceMatrixCalculator<a name="line.903"></a>
<span class="sourceLineNo">904</span>                                .kimuraDistance(msa);<a name="line.904"></a>
<span class="sourceLineNo">905</span>                Phylogeny tree = TreeConstructor.distanceTree(distmat,<a name="line.905"></a>
<span class="sourceLineNo">906</span>                                TreeConstructorType.NJ);<a name="line.906"></a>
<span class="sourceLineNo">907</span>                tree.setName("Kimura Tree");<a name="line.907"></a>
<span class="sourceLineNo">908</span>                return tree;<a name="line.908"></a>
<span class="sourceLineNo">909</span>        }<a name="line.909"></a>
<span class="sourceLineNo">910</span><a name="line.910"></a>
<span class="sourceLineNo">911</span>        /**<a name="line.911"></a>
<span class="sourceLineNo">912</span>         * Calculate a phylogenetic tree of the MultipleAlignment using<a name="line.912"></a>
<span class="sourceLineNo">913</span>         * dissimilarity scores (DS), based in SDM Substitution Matrix (ideal for<a name="line.913"></a>
<span class="sourceLineNo">914</span>         * distantly related proteins, structure-derived) and the Neighbor Joining<a name="line.914"></a>
<span class="sourceLineNo">915</span>         * algorithm from forester.<a name="line.915"></a>
<span class="sourceLineNo">916</span>         *<a name="line.916"></a>
<span class="sourceLineNo">917</span>         * @param msta<a name="line.917"></a>
<span class="sourceLineNo">918</span>         *            MultipleAlignment of protein structures<a name="line.918"></a>
<span class="sourceLineNo">919</span>         * @return Phylogeny phylogenetic tree<a name="line.919"></a>
<span class="sourceLineNo">920</span>         * @throws CompoundNotFoundException<a name="line.920"></a>
<span class="sourceLineNo">921</span>         * @throws IOException<a name="line.921"></a>
<span class="sourceLineNo">922</span>         */<a name="line.922"></a>
<span class="sourceLineNo">923</span>        public static Phylogeny getHSDMTree(MultipleAlignment msta)<a name="line.923"></a>
<span class="sourceLineNo">924</span>                        throws CompoundNotFoundException, IOException {<a name="line.924"></a>
<span class="sourceLineNo">925</span>                MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; msa = MultipleAlignmentTools<a name="line.925"></a>
<span class="sourceLineNo">926</span>                                .toProteinMSA(msta);<a name="line.926"></a>
<span class="sourceLineNo">927</span>                BasicSymmetricalDistanceMatrix distmat = (BasicSymmetricalDistanceMatrix) DistanceMatrixCalculator<a name="line.927"></a>
<span class="sourceLineNo">928</span>                                .dissimilarityScore(msa, SubstitutionMatrixHelper.getAminoAcidSubstitutionMatrix("PRLA000102"));<a name="line.928"></a>
<span class="sourceLineNo">929</span>                Phylogeny tree = TreeConstructor.distanceTree(distmat,<a name="line.929"></a>
<span class="sourceLineNo">930</span>                                TreeConstructorType.NJ);<a name="line.930"></a>
<span class="sourceLineNo">931</span>                tree.setName("HSDM Tree");<a name="line.931"></a>
<span class="sourceLineNo">932</span>                return tree;<a name="line.932"></a>
<span class="sourceLineNo">933</span>        }<a name="line.933"></a>
<span class="sourceLineNo">934</span><a name="line.934"></a>
<span class="sourceLineNo">935</span>        /**<a name="line.935"></a>
<span class="sourceLineNo">936</span>         * Calculate a phylogenetic tree of the MultipleAlignment using RMSD<a name="line.936"></a>
<span class="sourceLineNo">937</span>         * distances and the Neighbor Joining algorithm from forester.<a name="line.937"></a>
<span class="sourceLineNo">938</span>         *<a name="line.938"></a>
<span class="sourceLineNo">939</span>         * @param msta<a name="line.939"></a>
<span class="sourceLineNo">940</span>         *            MultipleAlignment of protein structures<a name="line.940"></a>
<span class="sourceLineNo">941</span>         * @return Phylogeny phylogenetic tree<a name="line.941"></a>
<span class="sourceLineNo">942</span>         * @throws CompoundNotFoundException<a name="line.942"></a>
<span class="sourceLineNo">943</span>         */<a name="line.943"></a>
<span class="sourceLineNo">944</span>        public static Phylogeny getStructuralTree(MultipleAlignment msta) {<a name="line.944"></a>
<span class="sourceLineNo">945</span>                double[][] rmsdMat = MultipleAlignmentTools.getRMSDMatrix(msta)<a name="line.945"></a>
<span class="sourceLineNo">946</span>                                .getArray();<a name="line.946"></a>
<span class="sourceLineNo">947</span>                BasicSymmetricalDistanceMatrix rmsdDist = (BasicSymmetricalDistanceMatrix) DistanceMatrixCalculator<a name="line.947"></a>
<span class="sourceLineNo">948</span>                                .structuralDistance(rmsdMat, 1, 5, 0.4);<a name="line.948"></a>
<span class="sourceLineNo">949</span>                // Set the identifiers of the matrix<a name="line.949"></a>
<span class="sourceLineNo">950</span>                Map&lt;String, Integer&gt; alreadySeen = new HashMap&lt;String, Integer&gt;();<a name="line.950"></a>
<span class="sourceLineNo">951</span>                for (int i = 0; i &lt; msta.size(); i++) {<a name="line.951"></a>
<span class="sourceLineNo">952</span>                        // Make sure the identifiers are unique<a name="line.952"></a>
<span class="sourceLineNo">953</span>                        String id = msta.getStructureIdentifier(i).toString();<a name="line.953"></a>
<span class="sourceLineNo">954</span>                        if (alreadySeen.containsKey(id)) {<a name="line.954"></a>
<span class="sourceLineNo">955</span>                                alreadySeen.put(id, alreadySeen.get(id) + 1);<a name="line.955"></a>
<span class="sourceLineNo">956</span>                                id += "_" + alreadySeen.get(id);<a name="line.956"></a>
<span class="sourceLineNo">957</span>                        } else<a name="line.957"></a>
<span class="sourceLineNo">958</span>                                alreadySeen.put(id, 1);<a name="line.958"></a>
<span class="sourceLineNo">959</span>                        rmsdDist.setIdentifier(i, id);<a name="line.959"></a>
<span class="sourceLineNo">960</span>                }<a name="line.960"></a>
<span class="sourceLineNo">961</span>                Phylogeny tree = TreeConstructor.distanceTree(rmsdDist,<a name="line.961"></a>
<span class="sourceLineNo">962</span>                                TreeConstructorType.NJ);<a name="line.962"></a>
<span class="sourceLineNo">963</span>                tree.setName("Structural Tree");<a name="line.963"></a>
<span class="sourceLineNo">964</span>                return tree;<a name="line.964"></a>
<span class="sourceLineNo">965</span>        }<a name="line.965"></a>
<span class="sourceLineNo">966</span><a name="line.966"></a>
<span class="sourceLineNo">967</span>}<a name="line.967"></a>




























































</pre>
</div>
</body>
</html>
