<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*        BioJava development code</a>
<span class="sourceLineNo">002</span><a id="line.2"> *</a>
<span class="sourceLineNo">003</span><a id="line.3"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">004</span><a id="line.4"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">005</span><a id="line.5"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">006</span><a id="line.6"> * see:</a>
<span class="sourceLineNo">007</span><a id="line.7"> *</a>
<span class="sourceLineNo">008</span><a id="line.8"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">009</span><a id="line.9"> *</a>
<span class="sourceLineNo">010</span><a id="line.10"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">011</span><a id="line.11"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">012</span><a id="line.12"> *</a>
<span class="sourceLineNo">013</span><a id="line.13"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">014</span><a id="line.14"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">015</span><a id="line.15"> * at:</a>
<span class="sourceLineNo">016</span><a id="line.16"> *</a>
<span class="sourceLineNo">017</span><a id="line.17"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">018</span><a id="line.18"> *</a>
<span class="sourceLineNo">019</span><a id="line.19"> */</a>
<span class="sourceLineNo">020</span><a id="line.20">package org.biojava.nbio.ronn;</a>
<span class="sourceLineNo">021</span><a id="line.21"></a>
<span class="sourceLineNo">022</span><a id="line.22">import org.biojava.nbio.core.sequence.ProteinSequence;</a>
<span class="sourceLineNo">023</span><a id="line.23">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</a>
<span class="sourceLineNo">024</span><a id="line.24">import org.biojava.nbio.data.sequence.FastaSequence;</a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.nbio.data.sequence.SequenceUtil;</a>
<span class="sourceLineNo">026</span><a id="line.26"></a>
<span class="sourceLineNo">027</span><a id="line.27">import java.io.FileInputStream;</a>
<span class="sourceLineNo">028</span><a id="line.28">import java.io.FileNotFoundException;</a>
<span class="sourceLineNo">029</span><a id="line.29">import java.io.IOException;</a>
<span class="sourceLineNo">030</span><a id="line.30">import java.io.Serializable;</a>
<span class="sourceLineNo">031</span><a id="line.31">import java.util.ArrayList;</a>
<span class="sourceLineNo">032</span><a id="line.32">import java.util.List;</a>
<span class="sourceLineNo">033</span><a id="line.33">import java.util.Map;</a>
<span class="sourceLineNo">034</span><a id="line.34">import java.util.TreeMap;</a>
<span class="sourceLineNo">035</span><a id="line.35"></a>
<span class="sourceLineNo">036</span><a id="line.36"></a>
<span class="sourceLineNo">037</span><a id="line.37">/**</a>
<span class="sourceLineNo">038</span><a id="line.38"> * This class gives public API to RONN functions.</a>
<span class="sourceLineNo">039</span><a id="line.39"> * It is build on top of the command line client. Due to this fact a few things</a>
<span class="sourceLineNo">040</span><a id="line.40"> * could be improved and extended pending the refactoring of the command line client.</a>
<span class="sourceLineNo">041</span><a id="line.41"> *</a>
<span class="sourceLineNo">042</span><a id="line.42"> * The input sequence limitations - the input sequence must not contain any ambiguous characters,</a>
<span class="sourceLineNo">043</span><a id="line.43"> * and have a minimum length of 19 amino acids.</a>
<span class="sourceLineNo">044</span><a id="line.44"> *</a>
<span class="sourceLineNo">045</span><a id="line.45"> * @author Peter Troshin</a>
<span class="sourceLineNo">046</span><a id="line.46"> * @version 1.0</a>
<span class="sourceLineNo">047</span><a id="line.47"> * @since 3.0.2</a>
<span class="sourceLineNo">048</span><a id="line.48"> *</a>
<span class="sourceLineNo">049</span><a id="line.49"> */</a>
<span class="sourceLineNo">050</span><a id="line.50">public class Jronn implements Serializable {</a>
<span class="sourceLineNo">051</span><a id="line.51"></a>
<span class="sourceLineNo">052</span><a id="line.52">        /**</a>
<span class="sourceLineNo">053</span><a id="line.53">         *</a>
<span class="sourceLineNo">054</span><a id="line.54">         */</a>
<span class="sourceLineNo">055</span><a id="line.55">        private static final long serialVersionUID = 8104272449130849946L;</a>
<span class="sourceLineNo">056</span><a id="line.56">        // Load models</a>
<span class="sourceLineNo">057</span><a id="line.57">        private static final ModelLoader loader = new ModelLoader();</a>
<span class="sourceLineNo">058</span><a id="line.58">        static {</a>
<span class="sourceLineNo">059</span><a id="line.59">                try {</a>
<span class="sourceLineNo">060</span><a id="line.60">                        loader.loadModels();</a>
<span class="sourceLineNo">061</span><a id="line.61">                } catch (NumberFormatException e) {</a>
<span class="sourceLineNo">062</span><a id="line.62">                        throw new RuntimeException("Fails to load models!" + e.getMessage(), e);</a>
<span class="sourceLineNo">063</span><a id="line.63">                } catch (IOException e) {</a>
<span class="sourceLineNo">064</span><a id="line.64">                        throw new RuntimeException("Fails to load models!" + e.getMessage(), e);</a>
<span class="sourceLineNo">065</span><a id="line.65">                }</a>
<span class="sourceLineNo">066</span><a id="line.66">        }</a>
<span class="sourceLineNo">067</span><a id="line.67"></a>
<span class="sourceLineNo">068</span><a id="line.68"></a>
<span class="sourceLineNo">069</span><a id="line.69">        /**</a>
<span class="sourceLineNo">070</span><a id="line.70">         * Holder for the ranges, contain pointers to starting and ending position</a>
<span class="sourceLineNo">071</span><a id="line.71">         * on the sequence which comprises a disordered region. Immutable.</a>
<span class="sourceLineNo">072</span><a id="line.72">         * @author pvtroshin</a>
<span class="sourceLineNo">073</span><a id="line.73">         */</a>
<span class="sourceLineNo">074</span><a id="line.74">        public static class Range {</a>
<span class="sourceLineNo">075</span><a id="line.75">                /**</a>
<span class="sourceLineNo">076</span><a id="line.76">                 * Range starting position counts from 1 (the first position on the sequence is 1)</a>
<span class="sourceLineNo">077</span><a id="line.77">                 */</a>
<span class="sourceLineNo">078</span><a id="line.78">                public final int from;</a>
<span class="sourceLineNo">079</span><a id="line.79">                /**</a>
<span class="sourceLineNo">080</span><a id="line.80">                 * The range ending position includes the last residue.</a>
<span class="sourceLineNo">081</span><a id="line.81">                 */</a>
<span class="sourceLineNo">082</span><a id="line.82">                public final int to;</a>
<span class="sourceLineNo">083</span><a id="line.83"></a>
<span class="sourceLineNo">084</span><a id="line.84">                public final float score;</a>
<span class="sourceLineNo">085</span><a id="line.85">                public Range(int from, int to, float score) {</a>
<span class="sourceLineNo">086</span><a id="line.86">                        assert from&gt;=0;</a>
<span class="sourceLineNo">087</span><a id="line.87">                        assert from&lt;to;</a>
<span class="sourceLineNo">088</span><a id="line.88">                        this.from = from;</a>
<span class="sourceLineNo">089</span><a id="line.89">                        this.to = to;</a>
<span class="sourceLineNo">090</span><a id="line.90">                        this.score = score;</a>
<span class="sourceLineNo">091</span><a id="line.91">                }</a>
<span class="sourceLineNo">092</span><a id="line.92"></a>
<span class="sourceLineNo">093</span><a id="line.93">                @Override</a>
<span class="sourceLineNo">094</span><a id="line.94">                public String toString() {</a>
<span class="sourceLineNo">095</span><a id="line.95">                        return "Range" + " From:" + from + "\t" + "to: " + to + "\n";</a>
<span class="sourceLineNo">096</span><a id="line.96">                }</a>
<span class="sourceLineNo">097</span><a id="line.97"></a>
<span class="sourceLineNo">098</span><a id="line.98">                @Override</a>
<span class="sourceLineNo">099</span><a id="line.99">                public int hashCode() {</a>
<span class="sourceLineNo">100</span><a id="line.100">                        final int prime = 31;</a>
<span class="sourceLineNo">101</span><a id="line.101">                        int result = 1;</a>
<span class="sourceLineNo">102</span><a id="line.102">                        result = prime * result + from;</a>
<span class="sourceLineNo">103</span><a id="line.103">                        result = prime * result + to;</a>
<span class="sourceLineNo">104</span><a id="line.104">                        return result;</a>
<span class="sourceLineNo">105</span><a id="line.105">                }</a>
<span class="sourceLineNo">106</span><a id="line.106"></a>
<span class="sourceLineNo">107</span><a id="line.107">                @Override</a>
<span class="sourceLineNo">108</span><a id="line.108">                public boolean equals(Object obj) {</a>
<span class="sourceLineNo">109</span><a id="line.109">                        if (this == obj)</a>
<span class="sourceLineNo">110</span><a id="line.110">                                return true;</a>
<span class="sourceLineNo">111</span><a id="line.111">                        if (obj == null)</a>
<span class="sourceLineNo">112</span><a id="line.112">                                return false;</a>
<span class="sourceLineNo">113</span><a id="line.113">                        if (getClass() != obj.getClass())</a>
<span class="sourceLineNo">114</span><a id="line.114">                                return false;</a>
<span class="sourceLineNo">115</span><a id="line.115">                        Range other = (Range) obj;</a>
<span class="sourceLineNo">116</span><a id="line.116">                        if (from != other.from)</a>
<span class="sourceLineNo">117</span><a id="line.117">                                return false;</a>
<span class="sourceLineNo">118</span><a id="line.118">                        if (to != other.to)</a>
<span class="sourceLineNo">119</span><a id="line.119">                                return false;</a>
<span class="sourceLineNo">120</span><a id="line.120">                        return true;</a>
<span class="sourceLineNo">121</span><a id="line.121">                }</a>
<span class="sourceLineNo">122</span><a id="line.122"></a>
<span class="sourceLineNo">123</span><a id="line.123"></a>
<span class="sourceLineNo">124</span><a id="line.124">        }</a>
<span class="sourceLineNo">125</span><a id="line.125"></a>
<span class="sourceLineNo">126</span><a id="line.126">        /**</a>
<span class="sourceLineNo">127</span><a id="line.127">         * Calculates the probability value for each residue in the protein sequence,</a>
<span class="sourceLineNo">128</span><a id="line.128">         * telling the probability that the residue belongs to disordered region.</a>
<span class="sourceLineNo">129</span><a id="line.129">         * In general, values greater than 0.5 considered to be in the disordered regions.</a>
<span class="sourceLineNo">130</span><a id="line.130">         *</a>
<span class="sourceLineNo">131</span><a id="line.131">         * @param sequence an instance of FastaSequence object, holding the name and the sequence.</a>
<span class="sourceLineNo">132</span><a id="line.132">         * @return the probability scores for each residue in the sequence</a>
<span class="sourceLineNo">133</span><a id="line.133">         */</a>
<span class="sourceLineNo">134</span><a id="line.134">        public static float[] getDisorderScores(FastaSequence sequence) {</a>
<span class="sourceLineNo">135</span><a id="line.135">                    return predictSerial(sequence);</a>
<span class="sourceLineNo">136</span><a id="line.136">        }</a>
<span class="sourceLineNo">137</span><a id="line.137"></a>
<span class="sourceLineNo">138</span><a id="line.138">        /**</a>
<span class="sourceLineNo">139</span><a id="line.139">         * Calculates the probability value for each residue in the protein sequence,</a>
<span class="sourceLineNo">140</span><a id="line.140">         * telling the probability that the residue belongs to disordered region.</a>
<span class="sourceLineNo">141</span><a id="line.141">         * In general, values greater than 0.5 considered to be in the disordered regions.</a>
<span class="sourceLineNo">142</span><a id="line.142">         *</a>
<span class="sourceLineNo">143</span><a id="line.143">         * @param sequence an instance of FastaSequence object, holding the name and the sequence.</a>
<span class="sourceLineNo">144</span><a id="line.144">         * @return the probability scores for each residue in the sequence</a>
<span class="sourceLineNo">145</span><a id="line.145">         */</a>
<span class="sourceLineNo">146</span><a id="line.146">        public static float[] getDisorderScores(ProteinSequence sequence) {</a>
<span class="sourceLineNo">147</span><a id="line.147"></a>
<span class="sourceLineNo">148</span><a id="line.148">                FastaSequence seq = convertProteinSequencetoFasta(sequence);</a>
<span class="sourceLineNo">149</span><a id="line.149"></a>
<span class="sourceLineNo">150</span><a id="line.150">                return predictSerial(seq);</a>
<span class="sourceLineNo">151</span><a id="line.151">        }</a>
<span class="sourceLineNo">152</span><a id="line.152"></a>
<span class="sourceLineNo">153</span><a id="line.153">        /** Utility method to convert a BioJava ProteinSequence object to the FastaSequence</a>
<span class="sourceLineNo">154</span><a id="line.154">         *  object used internally in JRonn.</a>
<span class="sourceLineNo">155</span><a id="line.155">         *</a>
<span class="sourceLineNo">156</span><a id="line.156">         * @param sequence</a>
<span class="sourceLineNo">157</span><a id="line.157">         * @return</a>
<span class="sourceLineNo">158</span><a id="line.158">         */</a>
<span class="sourceLineNo">159</span><a id="line.159">        public static FastaSequence convertProteinSequencetoFasta(ProteinSequence sequence){</a>
<span class="sourceLineNo">160</span><a id="line.160">                StringBuffer buf = new StringBuffer();</a>
<span class="sourceLineNo">161</span><a id="line.161">                for (AminoAcidCompound compound : sequence) {</a>
<span class="sourceLineNo">162</span><a id="line.162"></a>
<span class="sourceLineNo">163</span><a id="line.163">                        String c = compound.getShortName();</a>
<span class="sourceLineNo">164</span><a id="line.164"></a>
<span class="sourceLineNo">165</span><a id="line.165">                        if (! SequenceUtil.NON_AA.matcher(c).find()) {</a>
<span class="sourceLineNo">166</span><a id="line.166">                                buf.append(c);</a>
<span class="sourceLineNo">167</span><a id="line.167">                        } else {</a>
<span class="sourceLineNo">168</span><a id="line.168">                                buf.append("X");</a>
<span class="sourceLineNo">169</span><a id="line.169">                        }</a>
<span class="sourceLineNo">170</span><a id="line.170">                }</a>
<span class="sourceLineNo">171</span><a id="line.171"></a>
<span class="sourceLineNo">172</span><a id="line.172">                return new FastaSequence(sequence.getAccession().getID(),buf.toString());</a>
<span class="sourceLineNo">173</span><a id="line.173">        }</a>
<span class="sourceLineNo">174</span><a id="line.174"></a>
<span class="sourceLineNo">175</span><a id="line.175">        private static float[] predictSerial(FastaSequence fsequence) {</a>
<span class="sourceLineNo">176</span><a id="line.176">                ORonn.validateSequenceForRonn(fsequence);</a>
<span class="sourceLineNo">177</span><a id="line.177">                ORonn ronn;</a>
<span class="sourceLineNo">178</span><a id="line.178">                float[] disorder = null;</a>
<span class="sourceLineNo">179</span><a id="line.179">                try {</a>
<span class="sourceLineNo">180</span><a id="line.180">                        ronn = new ORonn(fsequence, loader);</a>
<span class="sourceLineNo">181</span><a id="line.181">                        disorder = ronn.call().getMeanScores();</a>
<span class="sourceLineNo">182</span><a id="line.182">                } catch (NumberFormatException e) {</a>
<span class="sourceLineNo">183</span><a id="line.183">                        throw new RuntimeException("Jronn fails to load models " + e.getLocalizedMessage(), e);</a>
<span class="sourceLineNo">184</span><a id="line.184">                } catch (IOException e) {</a>
<span class="sourceLineNo">185</span><a id="line.185">                        throw new RuntimeException("Jronn fails to load models " + e.getLocalizedMessage(), e);</a>
<span class="sourceLineNo">186</span><a id="line.186">                }</a>
<span class="sourceLineNo">187</span><a id="line.187">                return disorder;</a>
<span class="sourceLineNo">188</span><a id="line.188">        }</a>
<span class="sourceLineNo">189</span><a id="line.189"></a>
<span class="sourceLineNo">190</span><a id="line.190">        /**</a>
<span class="sourceLineNo">191</span><a id="line.191">         * Calculates the disordered regions of the sequence. More formally, the regions for which the</a>
<span class="sourceLineNo">192</span><a id="line.192">         * probability of disorder is greater then 0.50.</a>
<span class="sourceLineNo">193</span><a id="line.193">         *</a>
<span class="sourceLineNo">194</span><a id="line.194">         *</a>
<span class="sourceLineNo">195</span><a id="line.195">         * @param sequence an instance of FastaSequence object, holding the name and the sequence.</a>
<span class="sourceLineNo">196</span><a id="line.196">         * @return the array of ranges if there are any residues predicted to have the</a>
<span class="sourceLineNo">197</span><a id="line.197">         * probability of disorder greater then 0.5, null otherwise.</a>
<span class="sourceLineNo">198</span><a id="line.198">         *</a>
<span class="sourceLineNo">199</span><a id="line.199">         */</a>
<span class="sourceLineNo">200</span><a id="line.200">        public static Range[] getDisorder(FastaSequence sequence) {</a>
<span class="sourceLineNo">201</span><a id="line.201">                float[] scores = getDisorderScores(sequence);</a>
<span class="sourceLineNo">202</span><a id="line.202">                return scoresToRanges(scores, RonnConstraint.DEFAULT_RANGE_PROBABILITY_THRESHOLD);</a>
<span class="sourceLineNo">203</span><a id="line.203">        }</a>
<span class="sourceLineNo">204</span><a id="line.204"></a>
<span class="sourceLineNo">205</span><a id="line.205">        /**</a>
<span class="sourceLineNo">206</span><a id="line.206">         * Convert raw scores to ranges. Gives ranges for given probability of disorder value</a>
<span class="sourceLineNo">207</span><a id="line.207">         * @param scores the raw probability of disorder scores for each residue in the sequence.</a>
<span class="sourceLineNo">208</span><a id="line.208">         * @param probability the cut off threshold. Include all residues with the probability of disorder greater then this value</a>
<span class="sourceLineNo">209</span><a id="line.209">         * @return the array of ranges if there are any residues predicted to have the</a>
<span class="sourceLineNo">210</span><a id="line.210">         * probability of disorder greater then {@code probability}, null otherwise.</a>
<span class="sourceLineNo">211</span><a id="line.211">         */</a>
<span class="sourceLineNo">212</span><a id="line.212">        public static Range[] scoresToRanges(float[] scores, float probability)  {</a>
<span class="sourceLineNo">213</span><a id="line.213">                assert scores!=null &amp;&amp; scores.length&gt;0;</a>
<span class="sourceLineNo">214</span><a id="line.214">                assert probability&gt;0 &amp;&amp; probability&lt;1;</a>
<span class="sourceLineNo">215</span><a id="line.215"></a>
<span class="sourceLineNo">216</span><a id="line.216">                int count=0;</a>
<span class="sourceLineNo">217</span><a id="line.217">                int regionLen=0;</a>
<span class="sourceLineNo">218</span><a id="line.218">                List&lt;Range&gt; ranges = new ArrayList&lt;Range&gt;();</a>
<span class="sourceLineNo">219</span><a id="line.219">                for(float score: scores) {</a>
<span class="sourceLineNo">220</span><a id="line.220">                        count++;</a>
<span class="sourceLineNo">221</span><a id="line.221">                        // Round to 2 decimal points before comparison</a>
<span class="sourceLineNo">222</span><a id="line.222">                        score = (float) (Math.round(score*100.0)/100.0);</a>
<span class="sourceLineNo">223</span><a id="line.223">                        if(score&gt;probability) {</a>
<span class="sourceLineNo">224</span><a id="line.224">                                regionLen++;</a>
<span class="sourceLineNo">225</span><a id="line.225">                        } else {</a>
<span class="sourceLineNo">226</span><a id="line.226">                                if(regionLen&gt;0) {</a>
<span class="sourceLineNo">227</span><a id="line.227">                                        ranges.add(new Range(count-regionLen, count-1,score));</a>
<span class="sourceLineNo">228</span><a id="line.228">                                }</a>
<span class="sourceLineNo">229</span><a id="line.229">                                regionLen=0;</a>
<span class="sourceLineNo">230</span><a id="line.230">                        }</a>
<span class="sourceLineNo">231</span><a id="line.231">                }</a>
<span class="sourceLineNo">232</span><a id="line.232">                // In case of the range to boundary runs to the very end of the sequence</a>
<span class="sourceLineNo">233</span><a id="line.233">                if(regionLen&gt;1) {</a>
<span class="sourceLineNo">234</span><a id="line.234">                        ranges.add(new Range(count-regionLen+1, count,scores[scores.length-1]));</a>
<span class="sourceLineNo">235</span><a id="line.235">                }</a>
<span class="sourceLineNo">236</span><a id="line.236">                return ranges.toArray(new Range[ranges.size()]);</a>
<span class="sourceLineNo">237</span><a id="line.237"></a>
<span class="sourceLineNo">238</span><a id="line.238">        }</a>
<span class="sourceLineNo">239</span><a id="line.239"></a>
<span class="sourceLineNo">240</span><a id="line.240">        /**</a>
<span class="sourceLineNo">241</span><a id="line.241">         * Calculates the probability of disorder scores for each residue in the sequence for</a>
<span class="sourceLineNo">242</span><a id="line.242">         * many sequences in the input.</a>
<span class="sourceLineNo">243</span><a id="line.243">         *</a>
<span class="sourceLineNo">244</span><a id="line.244">         * @param sequences the list of the FastaSequence objects</a>
<span class="sourceLineNo">245</span><a id="line.245">         * @return the Map with key-&gt;FastaSequence, value-&gt;probability of disorder for each residue</a>
<span class="sourceLineNo">246</span><a id="line.246">         * @see #getDisorder(FastaSequence)</a>
<span class="sourceLineNo">247</span><a id="line.247">         */</a>
<span class="sourceLineNo">248</span><a id="line.248">        public static Map&lt;FastaSequence,float[]&gt; getDisorderScores(List&lt;FastaSequence&gt; sequences) {</a>
<span class="sourceLineNo">249</span><a id="line.249">                Map&lt;FastaSequence,float[]&gt; results = new TreeMap&lt;FastaSequence, float[]&gt;();</a>
<span class="sourceLineNo">250</span><a id="line.250">                for(FastaSequence fsequence : sequences) {</a>
<span class="sourceLineNo">251</span><a id="line.251">                        results.put(fsequence, predictSerial(fsequence));</a>
<span class="sourceLineNo">252</span><a id="line.252">                }</a>
<span class="sourceLineNo">253</span><a id="line.253">                return results;</a>
<span class="sourceLineNo">254</span><a id="line.254">        }</a>
<span class="sourceLineNo">255</span><a id="line.255"></a>
<span class="sourceLineNo">256</span><a id="line.256">        /**</a>
<span class="sourceLineNo">257</span><a id="line.257">         * Calculates the disordered regions of the sequence for many sequences in the input.</a>
<span class="sourceLineNo">258</span><a id="line.258">         *</a>
<span class="sourceLineNo">259</span><a id="line.259">         * @param sequences sequences the list of the FastaSequence objects</a>
<span class="sourceLineNo">260</span><a id="line.260">         * @return</a>
<span class="sourceLineNo">261</span><a id="line.261">         * @see #getDisorder(FastaSequence)</a>
<span class="sourceLineNo">262</span><a id="line.262">         */</a>
<span class="sourceLineNo">263</span><a id="line.263">        public static Map&lt;FastaSequence,Range[]&gt; getDisorder(List&lt;FastaSequence&gt; sequences) {</a>
<span class="sourceLineNo">264</span><a id="line.264">                Map&lt;FastaSequence,Range[]&gt; disorderRanges = new TreeMap&lt;FastaSequence,Range[]&gt;();</a>
<span class="sourceLineNo">265</span><a id="line.265">                for(FastaSequence fs: sequences) {</a>
<span class="sourceLineNo">266</span><a id="line.266">                        disorderRanges.put(fs, getDisorder(fs));</a>
<span class="sourceLineNo">267</span><a id="line.267">                }</a>
<span class="sourceLineNo">268</span><a id="line.268">                return disorderRanges;</a>
<span class="sourceLineNo">269</span><a id="line.269">        }</a>
<span class="sourceLineNo">270</span><a id="line.270"></a>
<span class="sourceLineNo">271</span><a id="line.271">        /**</a>
<span class="sourceLineNo">272</span><a id="line.272">         * Calculates the disordered regions of the protein sequence.</a>
<span class="sourceLineNo">273</span><a id="line.273">         * @param fastaFile input file name containing the sequence in FASTA</a>
<span class="sourceLineNo">274</span><a id="line.274">         * @return the Map with key-&gt;FastaSequence, value-&gt;the list of disordered regions for each sequence</a>
<span class="sourceLineNo">275</span><a id="line.275">         * @throws FileNotFoundException if the input file cannot be found</a>
<span class="sourceLineNo">276</span><a id="line.276">         * @throws IOException of the system cannot access or read from the input file</a>
<span class="sourceLineNo">277</span><a id="line.277">         * @see #getDisorder(FastaSequence)</a>
<span class="sourceLineNo">278</span><a id="line.278">         * @see #Jronn.Range</a>
<span class="sourceLineNo">279</span><a id="line.279">         */</a>
<span class="sourceLineNo">280</span><a id="line.280">        public static Map&lt;FastaSequence,Range[]&gt; getDisorder(String fastaFile) throws FileNotFoundException, IOException {</a>
<span class="sourceLineNo">281</span><a id="line.281">                final List&lt;FastaSequence&gt; sequences = SequenceUtil.readFasta(new FileInputStream(fastaFile));</a>
<span class="sourceLineNo">282</span><a id="line.282">                return getDisorder(sequences);</a>
<span class="sourceLineNo">283</span><a id="line.283">        }</a>
<span class="sourceLineNo">284</span><a id="line.284"></a>
<span class="sourceLineNo">285</span><a id="line.285">        /**</a>
<span class="sourceLineNo">286</span><a id="line.286">         * TODO</a>
<span class="sourceLineNo">287</span><a id="line.287">         *</a>
<span class="sourceLineNo">288</span><a id="line.288">         * High performance method for calculating disorder. Use multiple threads to achieve the speedup.</a>
<span class="sourceLineNo">289</span><a id="line.289">         *</a>
<span class="sourceLineNo">290</span><a id="line.290">         * @param fastaFile  fully qualified path to the input FASTA file</a>
<span class="sourceLineNo">291</span><a id="line.291">         * @param outputFile file name of the file for the results</a>
<span class="sourceLineNo">292</span><a id="line.292">         * @param threadNumber the number of threads to use, default</a>
<span class="sourceLineNo">293</span><a id="line.293">         * @param controls the format of the result file</a>
<span class="sourceLineNo">294</span><a id="line.294">         * @throws FileNotFoundException if input file in not found</a>
<span class="sourceLineNo">295</span><a id="line.295">         * @throws IOException if the input or the output files cannot be accessed</a>
<span class="sourceLineNo">296</span><a id="line.296">         * @see ORonn.ResultLayout</a>
<span class="sourceLineNo">297</span><a id="line.297"></a>
<span class="sourceLineNo">298</span><a id="line.298">        public static void calculateDisorder(String fastaFile, String outputFile, int threadNumber, ResultLayout layout) throws FileNotFoundException, IOException {</a>
<span class="sourceLineNo">299</span><a id="line.299">                final List&lt;FastaSequence&gt; sequences = SequenceUtil.readFasta(new FileInputStream(fastaFile));</a>
<span class="sourceLineNo">300</span><a id="line.300">                InputParameters in = new InputParameters();</a>
<span class="sourceLineNo">301</span><a id="line.301">                in.setFilePrm(fastaFile, InputParameters.inputKey);</a>
<span class="sourceLineNo">302</span><a id="line.302">                in.setFilePrm(outputFile, InputParameters.outputKey);</a>
<span class="sourceLineNo">303</span><a id="line.303">                //in.setThreadNum(Integer.toString(threadNumber));</a>
<span class="sourceLineNo">304</span><a id="line.304">                ORonn.predictParallel(sequences, in, loader);</a>
<span class="sourceLineNo">305</span><a id="line.305">        }</a>
<span class="sourceLineNo">306</span><a id="line.306">        */</a>
<span class="sourceLineNo">307</span><a id="line.307">}</a>




























































</pre>
</div>
</main>
</body>
</html>
