<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.1) on Thu Jan 03 22:38:55 CET 2019 -->
<title>Uses of Interface org.biojava.nbio.structure.Atom (biojava 5.2.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2019-01-03">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Interface org.biojava.nbio.structure.Atom (biojava 5.2.0 API)";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Interface org.biojava.nbio.structure.Atom" class="title">Uses of Interface<br>org.biojava.nbio.structure.Atom</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.protmod.structure">org.biojava.nbio.protmod.structure</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure">org.biojava.nbio.structure</a></th>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align">org.biojava.nbio.structure.align</a></th>
<td class="colLast">
<div class="block">Classes for the alignment of structures.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.ce">org.biojava.nbio.structure.align.ce</a></th>
<td class="colLast">
<div class="block">Classes related to the implementation of the CE alignment algorithm, here called jCE.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.client">org.biojava.nbio.structure.align.client</a></th>
<td class="colLast">
<div class="block">This package deals with the server communication for auto-downloading pre-calculated alignments.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.fatcat">org.biojava.nbio.structure.align.fatcat</a></th>
<td class="colLast">
<div class="block">Classes related to the implementation of the FATCAT alignment algorithm, here called jFATCAT.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.fatcat.calc">org.biojava.nbio.structure.align.fatcat.calc</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.gui">org.biojava.nbio.structure.align.gui</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.gui.aligpanel">org.biojava.nbio.structure.align.gui.aligpanel</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.gui.jmol">org.biojava.nbio.structure.align.gui.jmol</a></th>
<td class="colLast">
<div class="block">Utility methods for better interaction with Jmol.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.helper">org.biojava.nbio.structure.align.helper</a></th>
<td class="colLast">
<div class="block">Helper classes for structural alignment.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.model">org.biojava.nbio.structure.align.model</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.multiple">org.biojava.nbio.structure.align.multiple</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.pairwise">org.biojava.nbio.structure.align.pairwise</a></th>
<td class="colLast">
<div class="block">Classes for the pairwise alignment of structures.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.quaternary">org.biojava.nbio.structure.align.quaternary</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.seq">org.biojava.nbio.structure.align.seq</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.util">org.biojava.nbio.structure.align.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.xml">org.biojava.nbio.structure.align.xml</a></th>
<td class="colLast">
<div class="block">Serialization and deserialization for structure alignment results (as AFPChain object) to XML.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.asa">org.biojava.nbio.structure.asa</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.cluster">org.biojava.nbio.structure.cluster</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.contact">org.biojava.nbio.structure.contact</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.domain">org.biojava.nbio.structure.domain</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.domain.pdp">org.biojava.nbio.structure.domain.pdp</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></th>
<td class="colLast">
<div class="block">Input and Output of Structures</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.io.mmcif">org.biojava.nbio.structure.io.mmcif</a></th>
<td class="colLast">
<div class="block">Input and Output of mmcif files.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.io.mmtf">org.biojava.nbio.structure.io.mmtf</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.symmetry.internal">org.biojava.nbio.structure.symmetry.internal</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.symmetry.utils">org.biojava.nbio.structure.symmetry.utils</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.protmod.structure">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../protmod/structure/package-summary.html">org.biojava.nbio.protmod.structure</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../protmod/structure/package-summary.html">org.biojava.nbio.protmod.structure</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureUtil.</span><code><span class="memberNameLink"><a href="../../protmod/structure/StructureUtil.html#findLinkage(org.biojava.nbio.structure.Group,org.biojava.nbio.structure.Group,java.lang.String,java.lang.String,double)">findLinkage</a></span>&#8203;(<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group1,
           <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group2,
           <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;nameOfAtomOnGroup1,
           <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;nameOfAtomOnGroup2,
           double&nbsp;bondLengthTolerance)</code></th>
<td class="colLast">
<div class="block">Find a linkage between two groups within tolerance of bond length.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureUtil.</span><code><span class="memberNameLink"><a href="../../protmod/structure/StructureUtil.html#findNearestAtomLinkage(org.biojava.nbio.structure.Group,org.biojava.nbio.structure.Group,java.util.List,java.util.List,boolean,double)">findNearestAtomLinkage</a></span>&#8203;(<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group1,
                      <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group2,
                      <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;&nbsp;potentialNamesOfAtomOnGroup1,
                      <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;&nbsp;potentialNamesOfAtomOnGroup2,
                      boolean&nbsp;ignoreNCLinkage,
                      double&nbsp;bondLengthTolerance)</code></th>
<td class="colLast">
<div class="block">Find a linkage between two groups within tolerance of bond length,
 from potential atoms.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../protmod/structure/package-summary.html">org.biojava.nbio.protmod.structure</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureUtil.</span><code><span class="memberNameLink"><a href="../../protmod/structure/StructureUtil.html#getAtomDistance(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">getAtomDistance</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom1,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../protmod/structure/StructureAtom.html" title="class in org.biojava.nbio.protmod.structure">StructureAtom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureUtil.</span><code><span class="memberNameLink"><a href="../../protmod/structure/StructureUtil.html#getStructureAtom(org.biojava.nbio.structure.Atom,boolean)">getStructureAtom</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
                boolean&nbsp;isParentAminoAcid)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../protmod/structure/StructureAtomLinkage.html" title="class in org.biojava.nbio.protmod.structure">StructureAtomLinkage</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureUtil.</span><code><span class="memberNameLink"><a href="../../protmod/structure/StructureUtil.html#getStructureAtomLinkage(org.biojava.nbio.structure.Atom,boolean,org.biojava.nbio.structure.Atom,boolean)">getStructureAtomLinkage</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom1,
                       boolean&nbsp;isParentAminoAcid1,
                       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom2,
                       boolean&nbsp;isParentAminoAcid2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../package-summary.html">org.biojava.nbio.structure</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../package-summary.html">org.biojava.nbio.structure</a> that implement <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AtomImpl.html" title="class in org.biojava.nbio.structure">AtomImpl</a></span></code></th>
<td class="colLast">
<div class="block">Implementation of an Atom of a PDB file.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Fields in <a href="../package-summary.html">org.biojava.nbio.structure</a> with type parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">HetatomImpl.</span><code><span class="memberNameLink"><a href="../HetatomImpl.html#atoms">atoms</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#add(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">add</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">add two atoms ( a + b).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#centerAtoms(org.biojava.nbio.structure.Atom%5B%5D)">centerAtoms</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></th>
<td class="colLast">
<div class="block">Center the atoms at the Centroid.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#centerAtoms(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom)">centerAtoms</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code></th>
<td class="colLast">
<div class="block">Center the atoms at the Centroid, if the centroid is already know.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#centerOfMass(org.biojava.nbio.structure.Atom%5B%5D)">centerOfMass</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;points)</code></th>
<td class="colLast">
<div class="block">Returns the center of mass of the set of atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#cloneAtomArray(org.biojava.nbio.structure.Atom%5B%5D)">cloneAtomArray</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">
<div class="block">Provides an equivalent copy of Atoms in a new array.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#createVirtualCBAtom(org.biojava.nbio.structure.AminoAcid)">createVirtualCBAtom</a></span>&#8203;(<a href="../AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;amino)</code></th>
<td class="colLast">
<div class="block">creates a virtual C-beta atom. this might be needed when working with GLY

 thanks to Peter Lackner for a python template of this method.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#duplicateCA2(org.biojava.nbio.structure.Atom%5B%5D)">duplicateCA2</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Utility method for working with circular permutations.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAllAtomArray(org.biojava.nbio.structure.Chain)">getAllAtomArray</a></span>&#8203;(<a href="../Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c)</code></th>
<td class="colLast">
<div class="block">Returns and array of all atoms of the chain, including
 Hydrogens (if present) and all HETATOMs.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAllAtomArray(org.biojava.nbio.structure.Structure)">getAllAtomArray</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code></th>
<td class="colLast">
<div class="block">Convert all atoms of the structure (all models) into an Atom array</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAllAtomArray(org.biojava.nbio.structure.Structure,int)">getAllAtomArray</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
               int&nbsp;model)</code></th>
<td class="colLast">
<div class="block">Convert all atoms of the structure (specified model) into an Atom array</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAllNonHAtomArray(org.biojava.nbio.structure.Chain,boolean)">getAllNonHAtomArray</a></span>&#8203;(<a href="../Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c,
                   boolean&nbsp;hetAtoms)</code></th>
<td class="colLast">
<div class="block">Returns and array of all non-Hydrogen atoms in the given Chain,
 optionally including HET atoms or not Waters are not included.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAllNonHAtomArray(org.biojava.nbio.structure.Structure,boolean)">getAllNonHAtomArray</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                   boolean&nbsp;hetAtoms)</code></th>
<td class="colLast">
<div class="block">Returns and array of all non-Hydrogen atoms in the given Structure,
 optionally including HET atoms or not.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAllNonHAtomArray(org.biojava.nbio.structure.Structure,boolean,int)">getAllNonHAtomArray</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                   boolean&nbsp;hetAtoms,
                   int&nbsp;modelNr)</code></th>
<td class="colLast">
<div class="block">Returns and array of all non-Hydrogen atoms in the given Structure,
 optionally including HET atoms or not.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Group.</span><code><span class="memberNameLink"><a href="../Group.html#getAtom(int)">getAtom</a></span>&#8203;(int&nbsp;position)</code></th>
<td class="colLast">
<div class="block">Get at atom by position.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Group.</span><code><span class="memberNameLink"><a href="../Group.html#getAtom(java.lang.String)">getAtom</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Get an atom given its PDB name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">HetatomImpl.</span><code><span class="memberNameLink"><a href="../HetatomImpl.html#getAtom(int)">getAtom</a></span>&#8203;(int&nbsp;position)</code></th>
<td class="colLast">
<div class="block">Get at atom by position.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">HetatomImpl.</span><code><span class="memberNameLink"><a href="../HetatomImpl.html#getAtom(java.lang.String)">getAtom</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Get an atom given its PDB name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Bond.</span><code><span class="memberNameLink"><a href="../Bond.html#getAtomA()">getAtomA</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets atom 'A' of this bond.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BondImpl.</span><code><span class="memberNameLink"><a href="../BondImpl.html#getAtomA()">getAtomA</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets atom 'A' of this bond.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAtomArray(org.biojava.nbio.structure.Chain,java.lang.String%5B%5D)">getAtomArray</a></span>&#8203;(<a href="../Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;atomNames)</code></th>
<td class="colLast">
<div class="block">Returns an array of the requested Atoms from the Chain object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAtomArray(org.biojava.nbio.structure.Structure,java.lang.String%5B%5D)">getAtomArray</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;atomNames)</code></th>
<td class="colLast">
<div class="block">Returns an array of the requested Atoms from the Structure object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAtomArrayAllModels(org.biojava.nbio.structure.Structure,java.lang.String%5B%5D)">getAtomArrayAllModels</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>[]&nbsp;atomNames)</code></th>
<td class="colLast">
<div class="block">Returns an array of the requested Atoms from the Structure object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Bond.</span><code><span class="memberNameLink"><a href="../Bond.html#getAtomB()">getAtomB</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets atom 'B' of this bond.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BondImpl.</span><code><span class="memberNameLink"><a href="../BondImpl.html#getAtomB()">getAtomB</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets atom 'B' of this bond.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAtomCAArray(org.biojava.nbio.structure.Chain)">getAtomCAArray</a></span>&#8203;(<a href="../Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c)</code></th>
<td class="colLast">
<div class="block">Returns an Atom array of the C-alpha atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAtomCAArray(org.biojava.nbio.structure.Structure)">getAtomCAArray</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code></th>
<td class="colLast">
<div class="block">Return an Atom array of the C-alpha atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getBackboneAtomArray(org.biojava.nbio.structure.Structure)">getBackboneAtomArray</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code></th>
<td class="colLast">
<div class="block">Return an Atom array of the main chain atoms: CA, C, N, O Any group that
 contains those atoms will be included, be it a standard aminoacid or not</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcid.</span><code><span class="memberNameLink"><a href="../AminoAcid.html#getC()">getC</a></span>()</code></th>
<td class="colLast">
<div class="block">Get C atom.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidImpl.</span><code><span class="memberNameLink"><a href="../AminoAcidImpl.html#getC()">getC</a></span>()</code></th>
<td class="colLast">
<div class="block">Get C atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcid.</span><code><span class="memberNameLink"><a href="../AminoAcid.html#getCA()">getCA</a></span>()</code></th>
<td class="colLast">
<div class="block">Get CA atom.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidImpl.</span><code><span class="memberNameLink"><a href="../AminoAcidImpl.html#getCA()">getCA</a></span>()</code></th>
<td class="colLast">
<div class="block">Get CA atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcid.</span><code><span class="memberNameLink"><a href="../AminoAcid.html#getCB()">getCB</a></span>()</code></th>
<td class="colLast">
<div class="block">Get CB atom.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidImpl.</span><code><span class="memberNameLink"><a href="../AminoAcidImpl.html#getCB()">getCB</a></span>()</code></th>
<td class="colLast">
<div class="block">Get CB atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getCenterVector(org.biojava.nbio.structure.Atom%5B%5D)">getCenterVector</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></th>
<td class="colLast">
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms to
 the Centroid.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getCenterVector(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom)">getCenterVector</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code></th>
<td class="colLast">
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms to
 the Centroid, if the centroid is already known</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getCentroid(org.biojava.nbio.structure.Atom%5B%5D)">getCentroid</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></th>
<td class="colLast">
<div class="block">Returns the centroid of the set of atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcid.</span><code><span class="memberNameLink"><a href="../AminoAcid.html#getN()">getN</a></span>()</code></th>
<td class="colLast">
<div class="block">Get N atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidImpl.</span><code><span class="memberNameLink"><a href="../AminoAcidImpl.html#getN()">getN</a></span>()</code></th>
<td class="colLast">
<div class="block">Get N atom.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcid.</span><code><span class="memberNameLink"><a href="../AminoAcid.html#getO()">getO</a></span>()</code></th>
<td class="colLast">
<div class="block">Get O atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AminoAcidImpl.</span><code><span class="memberNameLink"><a href="../AminoAcidImpl.html#getO()">getO</a></span>()</code></th>
<td class="colLast">
<div class="block">Get O atom.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NucleotideImpl.</span><code><span class="memberNameLink"><a href="../NucleotideImpl.html#getO3Prime()">getO3Prime</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the O3' atom if present, otherwise null</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NucleotideImpl.</span><code><span class="memberNameLink"><a href="../NucleotideImpl.html#getO5Prime()">getO5Prime</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the O5' atom if present, otherwise null</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Bond.</span><code><span class="memberNameLink"><a href="../Bond.html#getOther(org.biojava.nbio.structure.Atom)">getOther</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;exclude)</code></th>
<td class="colLast">
<div class="block">A utility method to get the other atom in a bond, given one of its atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BondImpl.</span><code><span class="memberNameLink"><a href="../BondImpl.html#getOther(org.biojava.nbio.structure.Atom)">getOther</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;exclude)</code></th>
<td class="colLast">
<div class="block">A utility method to get the other atom in a bond, given one of its atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NucleotideImpl.</span><code><span class="memberNameLink"><a href="../NucleotideImpl.html#getP()">getP</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the P atom if present, otherwise null</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getRepresentativeAtomArray(org.biojava.nbio.structure.Chain)">getRepresentativeAtomArray</a></span>&#8203;(<a href="../Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c)</code></th>
<td class="colLast">
<div class="block">Gets a representative atom for each group that is part of the chain
 backbone.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getRepresentativeAtomArray(org.biojava.nbio.structure.Structure)">getRepresentativeAtomArray</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code></th>
<td class="colLast">
<div class="block">Gets a representative atom for each group that is part of the chain
 backbone.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getTranslationVector(javax.vecmath.Matrix4d)">getTranslationVector</a></span>&#8203;(javax.vecmath.Matrix4d&nbsp;transform)</code></th>
<td class="colLast">
<div class="block">Extract the translational vector as an Atom of a transformation matrix.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#invert(org.biojava.nbio.structure.Atom)">invert</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomIterator.</span><code><span class="memberNameLink"><a href="../AtomIterator.html#next()">next</a></span>()</code></th>
<td class="colLast">
<div class="block">Return next atom.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#scale(org.biojava.nbio.structure.Atom,double)">scale</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     double&nbsp;s)</code></th>
<td class="colLast">
<div class="block">Multiply elements of a by s</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#scaleAdd(double,org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">scaleAdd</a></span>&#8203;(double&nbsp;s,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;x,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">Perform linear transformation s*X+B, and store the result in b</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#scaleEquals(org.biojava.nbio.structure.Atom,double)">scaleEquals</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
           double&nbsp;s)</code></th>
<td class="colLast">
<div class="block">Multiply elements of a by s (in place)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#subtract(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">subtract</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">subtract two atoms ( a - b).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#unitVector(org.biojava.nbio.structure.Atom)">unitVector</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></th>
<td class="colLast">
<div class="block">Returns the unit vector of vector a .</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#vectorProduct(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">vectorProduct</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">Vector product (cross product).</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure</a> that return types with arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Group.</span><code><span class="memberNameLink"><a href="../Group.html#getAtoms()">getAtoms</a></span>()</code></th>
<td class="colLast">
<div class="block">Get list of atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">HetatomImpl.</span><code><span class="memberNameLink"><a href="../HetatomImpl.html#getAtoms()">getAtoms</a></span>()</code></th>
<td class="colLast">
<div class="block">Get list of atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html?is-external=true" title="class or interface in java.util" class="externalLink">Iterator</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Group.</span><code><span class="memberNameLink"><a href="../Group.html#iterator()">iterator</a></span>()</code></th>
<td class="colLast">
<div class="block">Get an Atom Iterator.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html?is-external=true" title="class or interface in java.util" class="externalLink">Iterator</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">HetatomImpl.</span><code><span class="memberNameLink"><a href="../HetatomImpl.html#iterator()">iterator</a></span>()</code></th>
<td class="colLast">
<div class="block">return an AtomIterator.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#add(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">add</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">add two atoms ( a + b).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Group.</span><code><span class="memberNameLink"><a href="../Group.html#addAtom(org.biojava.nbio.structure.Atom)">addAtom</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom)</code></th>
<td class="colLast">
<div class="block">Add an atom to this group.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">HetatomImpl.</span><code><span class="memberNameLink"><a href="../HetatomImpl.html#addAtom(org.biojava.nbio.structure.Atom)">addAtom</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom)</code></th>
<td class="colLast">
<div class="block">Add an atom to this group.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#amount(org.biojava.nbio.structure.Atom)">amount</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></th>
<td class="colLast">
<div class="block">Gets the length of the vector (2-norm)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#angle(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">angle</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">Gets the angle between two vectors</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static javax.vecmath.Point3d[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#atomsToPoints(org.biojava.nbio.structure.Atom%5B%5D)">atomsToPoints</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Convert an array of atoms into an array of vecmath points</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#calcRotationAngleInDegrees(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">calcRotationAngleInDegrees</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centerPt,
                          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;targetPt)</code></th>
<td class="colLast">
<div class="block">Calculates the angle from centerPt to targetPt in degrees.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#centerAtoms(org.biojava.nbio.structure.Atom%5B%5D)">centerAtoms</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></th>
<td class="colLast">
<div class="block">Center the atoms at the Centroid.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#centerAtoms(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom)">centerAtoms</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code></th>
<td class="colLast">
<div class="block">Center the atoms at the Centroid, if the centroid is already know.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#centerOfMass(org.biojava.nbio.structure.Atom%5B%5D)">centerOfMass</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;points)</code></th>
<td class="colLast">
<div class="block">Returns the center of mass of the set of atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#cloneAtomArray(org.biojava.nbio.structure.Atom%5B%5D)">cloneAtomArray</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">
<div class="block">Provides an equivalent copy of Atoms in a new array.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#cloneGroups(org.biojava.nbio.structure.Atom%5B%5D)">cloneGroups</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">
<div class="block">Clone a set of representative Atoms, but returns the parent groups</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#convertAtomsToSeq(org.biojava.nbio.structure.Atom%5B%5D)">convertAtomsToSeq</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#duplicateCA2(org.biojava.nbio.structure.Atom%5B%5D)">duplicateCA2</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Utility method for working with circular permutations.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util" class="externalLink">Set</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAllGroupsFromSubset(org.biojava.nbio.structure.Atom%5B%5D)">getAllGroupsFromSubset</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Expand a set of atoms into all groups from the same structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util" class="externalLink">Set</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getAllGroupsFromSubset(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.GroupType)">getAllGroupsFromSubset</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
                      <a href="../GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;types)</code></th>
<td class="colLast">
<div class="block">Expand a set of atoms into all groups from the same structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getCenterVector(org.biojava.nbio.structure.Atom%5B%5D)">getCenterVector</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></th>
<td class="colLast">
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms to
 the Centroid.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getCenterVector(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom)">getCenterVector</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code></th>
<td class="colLast">
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms to
 the Centroid, if the centroid is already known</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getCentroid(org.biojava.nbio.structure.Atom%5B%5D)">getCentroid</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></th>
<td class="colLast">
<div class="block">Returns the centroid of the set of atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getDistance(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">getDistance</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">calculate distance between two atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getDistanceFast(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">getDistanceFast</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">Will calculate the square of distances between two atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>,&#8203;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang" class="externalLink">Double</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getGroupDistancesWithinShell(org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Atom,java.util.Set,double,boolean,boolean)">getGroupDistancesWithinShell</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid,
                            <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util" class="externalLink">Set</a>&lt;<a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&gt;&nbsp;excludeResidues,
                            double&nbsp;radius,
                            boolean&nbsp;includeWater,
                            boolean&nbsp;useAverageDistance)</code></th>
<td class="colLast">
<div class="block">Finds Groups in <code>structure</code> that contain at least one Atom that is
 within <code>radius</code> Angstroms of <code>centroid</code>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util" class="externalLink">Set</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getGroupsWithinShell(org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Atom,java.util.Set,double,boolean)">getGroupsWithinShell</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
                    <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util" class="externalLink">Set</a>&lt;<a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&gt;&nbsp;excludeResidues,
                    double&nbsp;distance,
                    boolean&nbsp;includeWater)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getLigandsByProximity(java.util.Collection,org.biojava.nbio.structure.Atom%5B%5D,double)">getLigandsByProximity</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;target,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;query,
                     double&nbsp;cutoff)</code></th>
<td class="colLast">
<div class="block">Finds all ligand groups from the target which fall within the cutoff distance
 of some atom from the query set.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Bond.</span><code><span class="memberNameLink"><a href="../Bond.html#getOther(org.biojava.nbio.structure.Atom)">getOther</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;exclude)</code></th>
<td class="colLast">
<div class="block">A utility method to get the other atom in a bond, given one of its atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BondImpl.</span><code><span class="memberNameLink"><a href="../BondImpl.html#getOther(org.biojava.nbio.structure.Atom)">getOther</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;exclude)</code></th>
<td class="colLast">
<div class="block">A utility method to get the other atom in a bond, given one of its atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getTMScore(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int)">getTMScore</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2,
          int&nbsp;len1,
          int&nbsp;len2)</code></th>
<td class="colLast">
<div class="block">Calculate the TM-Score for the superposition.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getTMScore(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int,boolean)">getTMScore</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2,
          int&nbsp;len1,
          int&nbsp;len2,
          boolean&nbsp;normalizeMin)</code></th>
<td class="colLast">
<div class="block">Calculate the TM-Score for the superposition.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static javax.vecmath.Matrix4d</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#getTransformation(org.biojava.nbio.structure.jama.Matrix,org.biojava.nbio.structure.Atom)">getTransformation</a></span>&#8203;(<a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;rot,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;trans)</code></th>
<td class="colLast">
<div class="block">Convert JAMA rotation and translation to a Vecmath transformation matrix.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#getUnalignedGroups(org.biojava.nbio.structure.Atom%5B%5D)">getUnalignedGroups</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">
<div class="block">List of groups from the structure not included in ca (e.g. ligands).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Atom.</span><code><span class="memberNameLink"><a href="../Atom.html#hasBond(org.biojava.nbio.structure.Atom)">hasBond</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;other)</code></th>
<td class="colLast">
<div class="block">Test if another atom has a bond to this atom</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomImpl.</span><code><span class="memberNameLink"><a href="../AtomImpl.html#hasBond(org.biojava.nbio.structure.Atom)">hasBond</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;other)</code></th>
<td class="colLast">
<div class="block">Test if another atom has a bond to this atom</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#hasDeuteratedEquiv(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Group)">hasDeuteratedEquiv</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
                  <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;currentGroup)</code></th>
<td class="colLast">
<div class="block">Check to see if a Hydrogen has a  Deuterated brother in the group.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureTools.</span><code><span class="memberNameLink"><a href="../StructureTools.html#hasNonDeuteratedEquiv(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Group)">hasNonDeuteratedEquiv</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
                     <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;currentGroup)</code></th>
<td class="colLast">
<div class="block">Check to see if an Deuterated atom has a non deuterated brother in the group.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#invert(org.biojava.nbio.structure.Atom)">invert</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#rmsd(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">rmsd</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;x,
    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;y)</code></th>
<td class="colLast">
<div class="block">Calculate the RMSD of two Atom arrays, already superposed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#rotate(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.jama.Matrix)">rotate</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
      <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;matrix)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#rotate(org.biojava.nbio.structure.Atom,double%5B%5D%5B%5D)">rotate</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
      double[][]&nbsp;m)</code></th>
<td class="colLast">
<div class="block">Rotate a single Atom aroud a rotation matrix.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#rotate(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.jama.Matrix)">rotate</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
      <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;m)</code></th>
<td class="colLast">
<div class="block">Rotate an Atom around a Matrix object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#scalarProduct(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">scalarProduct</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">Scalar product (dot product).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#scale(org.biojava.nbio.structure.Atom,double)">scale</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     double&nbsp;s)</code></th>
<td class="colLast">
<div class="block">Multiply elements of a by s</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#scaleAdd(double,org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">scaleAdd</a></span>&#8203;(double&nbsp;s,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;x,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">Perform linear transformation s*X+B, and store the result in b</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#scaleEquals(org.biojava.nbio.structure.Atom,double)">scaleEquals</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
           double&nbsp;s)</code></th>
<td class="colLast">
<div class="block">Multiply elements of a by s (in place)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#shift(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom)">shift</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">Shift an array of atoms at once.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#shift(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">shift</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">Shift a vector.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#shift(org.biojava.nbio.structure.Group,org.biojava.nbio.structure.Atom)">shift</a></span>&#8203;(<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></th>
<td class="colLast">
<div class="block">Shift a Group with a vector.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#shift(org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Atom)">shift</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></th>
<td class="colLast">
<div class="block">shift a structure with a vector.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#subtract(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">subtract</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">subtract two atoms ( a - b).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#torsionAngle(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">torsionAngle</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;c,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;d)</code></th>
<td class="colLast">
<div class="block">Calculate the torsion angle, i.e. the angle between the normal vectors of
 the two plains a-b-c and b-c-d.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#transform(org.biojava.nbio.structure.Atom%5B%5D,javax.vecmath.Matrix4d)">transform</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
         javax.vecmath.Matrix4d&nbsp;t)</code></th>
<td class="colLast">
<div class="block">Transform an array of atoms at once.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#transform(org.biojava.nbio.structure.Atom,javax.vecmath.Matrix4d)">transform</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
         javax.vecmath.Matrix4d&nbsp;m)</code></th>
<td class="colLast">
<div class="block">Transforms an atom object, given a Matrix4d (i.e. the vecmath library
 double-precision 4x4 rotation+translation matrix).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#translate(org.biojava.nbio.structure.Atom,javax.vecmath.Vector3d)">translate</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
         javax.vecmath.Vector3d&nbsp;v)</code></th>
<td class="colLast">
<div class="block">Translates an atom object, given a Vector3d (i.e. the vecmath library
 double-precision 3-d vector)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#unitVector(org.biojava.nbio.structure.Atom)">unitVector</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></th>
<td class="colLast">
<div class="block">Returns the unit vector of vector a .</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#vectorProduct(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">vectorProduct</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></th>
<td class="colLast">
<div class="block">Vector product (cross product).</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../package-summary.html">org.biojava.nbio.structure</a> with type arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;javax.vecmath.Point3d&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Calc.</span><code><span class="memberNameLink"><a href="../Calc.html#atomsToPoints(java.util.Collection)">atomsToPoints</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Convert an array of atoms into an array of vecmath points</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Group.</span><code><span class="memberNameLink"><a href="../Group.html#setAtoms(java.util.List)">setAtoms</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Set the atoms of this group.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">HetatomImpl.</span><code><span class="memberNameLink"><a href="../HetatomImpl.html#setAtoms(java.util.List)">setAtoms</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Set the atoms of this group.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../package-summary.html">org.biojava.nbio.structure</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AtomPositionMap.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D)">AtomPositionMap</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Creates a new AtomPositionMap containing peptide alpha-carbon atoms</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AtomPositionMap.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.AtomPositionMap.GroupMatcher)">AtomPositionMap</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
               <a href="../AtomPositionMap.GroupMatcher.html" title="interface in org.biojava.nbio.structure">AtomPositionMap.GroupMatcher</a>&nbsp;matcher)</code></th>
<td class="colLast">
<div class="block">Creates a new AtomPositionMap containing only atoms matched by <code>matcher</code>.</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../BondImpl.html#%3Cinit%3E(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom,int)">BondImpl</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atomA,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atomB,
        int&nbsp;bondOrder)</code></th>
<td class="colLast">
<div class="block">Constructs a new bond from a pair of atoms and the bond order of the bond
 between them.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../BondImpl.html#%3Cinit%3E(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom,int,boolean)">BondImpl</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atomA,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atomB,
        int&nbsp;bondOrder,
        boolean&nbsp;addSelfToAtoms)</code></th>
<td class="colLast">
<div class="block">Constructs a new bond from a pair of atoms and the bond order of the bond
 between them.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/package-summary.html">org.biojava.nbio.structure.align</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/package-summary.html">org.biojava.nbio.structure.align</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructurePairAligner.</span><code><span class="memberNameLink"><a href="../align/StructurePairAligner.html#getAlignmentAtoms(org.biojava.nbio.structure.Structure)">getAlignmentAtoms</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code></th>
<td class="colLast">
<div class="block">Returns the atoms that are being used for the alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CallableStructureAlignment.</span><code><span class="memberNameLink"><a href="../align/CallableStructureAlignment.html#getCa1()">getCa1</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../align/package-summary.html">org.biojava.nbio.structure.align</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractStructureAlignment.</span><code><span class="memberNameLink"><a href="../align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractStructureAlignment.</span><code><span class="memberNameLink"><a href="../align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BioJavaStructureAlignment.</span><code><span class="memberNameLink"><a href="../align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BioJavaStructureAlignment.</span><code><span class="memberNameLink"><a href="../align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureAlignment.</span><code><span class="memberNameLink"><a href="../align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Run an alignment while specifying the atoms to be aligned.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureAlignment.</span><code><span class="memberNameLink"><a href="../align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">run an alignment and also send a bean containing the parameters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructurePairAligner.</span><code><span class="memberNameLink"><a href="../align/StructurePairAligner.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.StrucAligParameters)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">calculate the protein structure superimposition, between two sets of
 atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CallableStructureAlignment.</span><code><span class="memberNameLink"><a href="../align/CallableStructureAlignment.html#setCa1(org.biojava.nbio.structure.Atom%5B%5D)">setCa1</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPTwister.</span><code><span class="memberNameLink"><a href="../align/AFPTwister.html#twistOptimized(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">twistOptimized</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">superimposing according to the optimized alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPTwister.</span><code><span class="memberNameLink"><a href="../align/AFPTwister.html#twistPDB(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">twistPDB</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">calculate the total rmsd of the blocks output a merged pdb file for both
 proteins protein 1, in chain A protein 2 is twisted according to the
 twists detected, in chain B</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../align/package-summary.html">org.biojava.nbio.structure.align</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../align/CallableStructureAlignment.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.String,org.biojava.nbio.structure.align.ce.ConfigStrucAligParams)">CallableStructureAlignment</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                          <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;algorithmName,
                          <a href="../align/ce/ConfigStrucAligParams.html" title="interface in org.biojava.nbio.structure.align.ce">ConfigStrucAligParams</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Constructor for all-to-all alignment calculation.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.ce">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GuiWrapper.</span><code><span class="memberNameLink"><a href="../align/ce/GuiWrapper.html#getAtomArray(org.biojava.nbio.structure.Atom%5B%5D,java.util.List,java.util.List)">getAtomArray</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatoms,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;nucs)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CECalculator.</span><code><span class="memberNameLink"><a href="../align/ce/CECalculator.html#getShift()">getShift</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets the shift from the last call to
 <a href="../align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,boolean)"><code>calc_rmsd</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCalculatorEnhanced.</span><code><span class="memberNameLink"><a href="../align/ce/CeCalculatorEnhanced.html#getShift()">getShift</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets the shift from the last call to
 <a href="../align/ce/CeCalculatorEnhanced.html#calc_rmsd(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,boolean,boolean)"><code>calc_rmsd</code></a>.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCPMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeCPMain.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param)</code></th>
<td class="colLast">
<div class="block">Aligns ca1 and ca2 using a heuristic to check for CPs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param)</code></th>
<td class="colLast">
<div class="block">Align ca2 onto ca1.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">OptimalCECPMain.</span><code><span class="memberNameLink"><a href="../align/ce/OptimalCECPMain.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param)</code></th>
<td class="colLast">
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">OptimalCECPMain.</span><code><span class="memberNameLink"><a href="../align/ce/OptimalCECPMain.html#alignOptimal(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object,org.biojava.nbio.structure.align.model.AFPChain%5B%5D)">alignOptimal</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param,
            <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>[]&nbsp;alignments)</code></th>
<td class="colLast">
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">OptimalCECPMain.</span><code><span class="memberNameLink"><a href="../align/ce/OptimalCECPMain.html#alignPermuted(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object,int)">alignPermuted</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param,
             int&nbsp;cp)</code></th>
<td class="colLast">
<div class="block">Aligns ca1 with ca2 permuted by <i>cp</i> residues.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CECalculator.</span><code><span class="memberNameLink"><a href="../align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,boolean)">calc_rmsd</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform)</code></th>
<td class="colLast">
<div class="block">superimpose and get rmsd</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCalculatorEnhanced.</span><code><span class="memberNameLink"><a href="../align/ce/CeCalculatorEnhanced.html#calc_rmsd(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,boolean,boolean)">calc_rmsd</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform,
         boolean&nbsp;show)</code></th>
<td class="colLast">
<div class="block">superimpose and get rmsd</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CECalculator.</span><code><span class="memberNameLink"><a href="../align/ce/CECalculator.html#convertAfpChain(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">convertAfpChain</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">copy data from this class into AFPChain container object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCalculatorEnhanced.</span><code><span class="memberNameLink"><a href="../align/ce/CeCalculatorEnhanced.html#convertAfpChain(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">convertAfpChain</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">copy data from this class into AFPChain container object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GuiWrapper.</span><code><span class="memberNameLink"><a href="../align/ce/GuiWrapper.html#display(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">display</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CECalculator.</span><code><span class="memberNameLink"><a href="../align/ce/CECalculator.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">extractFragments</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCalculatorEnhanced.</span><code><span class="memberNameLink"><a href="../align/ce/CeCalculatorEnhanced.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">extractFragments</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCPMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">filterDuplicateAFPs</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated)</code></th>
<td class="colLast">
<div class="block">Takes as input an AFPChain where ca2 has been artificially duplicated.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCPMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CECPParameters)">filterDuplicateAFPs</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated,
                   <a href="../align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GuiWrapper.</span><code><span class="memberNameLink"><a href="../align/ce/GuiWrapper.html#getAlignedStructure(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getAlignedStructure</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GuiWrapper.</span><code><span class="memberNameLink"><a href="../align/ce/GuiWrapper.html#getAtomArray(org.biojava.nbio.structure.Atom%5B%5D,java.util.List,java.util.List)">getAtomArray</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatoms,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;nucs)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double[][]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CECalculator.</span><code><span class="memberNameLink"><a href="../align/ce/CECalculator.html#initSumOfDistances(int,int,int,int,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">initSumOfDistances</a></span>&#8203;(int&nbsp;nse1,
                  int&nbsp;nse2,
                  int&nbsp;winSize,
                  int&nbsp;winSizeComb1,
                  <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double[][]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCalculatorEnhanced.</span><code><span class="memberNameLink"><a href="../align/ce/CeCalculatorEnhanced.html#initSumOfDistances(int,int,int,int,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">initSumOfDistances</a></span>&#8203;(int&nbsp;nse1,
                  int&nbsp;nse2,
                  int&nbsp;winSize,
                  int&nbsp;winSizeComb1,
                  <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CECalculator.</span><code><span class="memberNameLink"><a href="../align/ce/CECalculator.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">nextStep</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCalculatorEnhanced.</span><code><span class="memberNameLink"><a href="../align/ce/CeCalculatorEnhanced.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">nextStep</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCPMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CECalculator)">postProcessAlignment</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator)</code></th>
<td class="colLast">
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCPMain.</span><code><span class="memberNameLink"><a href="../align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.align.ce.CECPParameters)">postProcessAlignment</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                    <a href="../align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;param)</code></th>
<td class="colLast">
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GuiWrapper.</span><code><span class="memberNameLink"><a href="../align/ce/GuiWrapper.html#showAlignmentImage(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">showAlignmentImage</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;jmol)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CECalculator.</span><code><span class="memberNameLink"><a href="../align/ce/CECalculator.html#traceFragmentMatrix(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">traceFragmentMatrix</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCalculatorEnhanced.</span><code><span class="memberNameLink"><a href="../align/ce/CeCalculatorEnhanced.html#traceFragmentMatrix(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">traceFragmentMatrix</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double[][]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CECalculator.</span><code><span class="memberNameLink"><a href="../align/ce/CECalculator.html#updateMatrixWithSequenceConservation(double%5B%5D%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CeParameters)">updateMatrixWithSequenceConservation</a></span>&#8203;(double[][]&nbsp;max,
                                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                                    <a href="../align/ce/CeParameters.html" title="class in org.biojava.nbio.structure.align.ce">CeParameters</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Modifies an alignment matrix by favoring the alignment of similar and identical amino acids and penalizing the alignment of unrelated ones.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double[][]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeCalculatorEnhanced.</span><code><span class="memberNameLink"><a href="../align/ce/CeCalculatorEnhanced.html#updateMatrixWithSequenceConservation(double%5B%5D%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CeParameters)">updateMatrixWithSequenceConservation</a></span>&#8203;(double[][]&nbsp;max,
                                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                                    <a href="../align/ce/CeParameters.html" title="class in org.biojava.nbio.structure.align.ce">CeParameters</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Modifies an alignment matrix by favoring the alignment of similar and identical amino acids and penalizing the alignment of unrelated ones.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.client">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/client/package-summary.html">org.biojava.nbio.structure.align.client</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/client/package-summary.html">org.biojava.nbio.structure.align.client</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">JFatCatClient.</span><code><span class="memberNameLink"><a href="../align/client/JFatCatClient.html#getAFPChainFromServer(java.lang.String,java.lang.String,java.lang.String,java.lang.String,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int)">getAFPChainFromServer</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;serverLocation,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;method,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name1,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name2,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                     int&nbsp;timeout)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">JFatCatClient.</span><code><span class="memberNameLink"><a href="../align/client/JFatCatClient.html#getAFPChainFromServer(java.lang.String,java.lang.String,java.lang.String,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getAFPChainFromServer</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;serverLocation,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name1,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name2,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">JFatCatClient.</span><code><span class="memberNameLink"><a href="../align/client/JFatCatClient.html#sendAFPChainToServer(java.lang.String,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">sendAFPChainToServer</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;serverLocation,
                    <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.fatcat">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/fatcat/package-summary.html">org.biojava.nbio.structure.align.fatcat</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/fatcat/package-summary.html">org.biojava.nbio.structure.align.fatcat</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCat.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCat.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,boolean)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
     boolean&nbsp;doRigid)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCatFlexible.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCatFlexible.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCatRigid.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCatRigid.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;param)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCat.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCat.html#alignFlexible(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">alignFlexible</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCat.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">alignRigid</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCat.</span><code><span class="memberNameLink"><a href="../align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">alignRigid</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
          <a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.fatcat.calc">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FatCatAligner.</span><code><span class="memberNameLink"><a href="../align/fatcat/calc/FatCatAligner.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean,org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     boolean&nbsp;doRigid,
     <a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainer.</span><code><span class="memberNameLink"><a href="../align/fatcat/calc/AFPChainer.html#calAfpRmsd(int,int%5B%5D,int,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">calAfpRmsd</a></span>&#8203;(int&nbsp;afpn,
          int[]&nbsp;afpPositions,
          int&nbsp;listStart,
          <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">//return the rmsd of the residues from the segments that form the given AFP list
        //this value can be a measurement (1) for the connectivity of the AFPs</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainer.</span><code><span class="memberNameLink"><a href="../align/fatcat/calc/AFPChainer.html#doChainAfp(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">doChainAfp</a></span>&#8203;(<a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
          <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">// Key function: chain (assembly) the AFPs
        // a AFP (k) is defined as (i, j, k), with i and j are staring points
        // AFP extension (eg.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPCalculator.</span><code><span class="memberNameLink"><a href="../align/fatcat/calc/AFPCalculator.html#extractAFPChains(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">extractAFPChains</a></span>&#8203;(<a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
                <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPOptimizer.</span><code><span class="memberNameLink"><a href="../align/fatcat/calc/AFPOptimizer.html#optimizeAln(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">optimizeAln</a></span>&#8203;(<a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
           <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">optimize the alignment by dynamic programming</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPPostProcessor.</span><code><span class="memberNameLink"><a href="../align/fatcat/calc/AFPPostProcessor.html#postProcess(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">postProcess</a></span>&#8203;(<a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
           <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPCalculator.</span><code><span class="memberNameLink"><a href="../align/fatcat/calc/AFPCalculator.html#sortAfps(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">sortAfps</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../align/fatcat/calc/StructureAlignmentOptimizer.html#%3Cinit%3E(int,int,org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.Atom%5B%5D,int,int%5B%5D%5B%5D)">StructureAlignmentOptimizer</a></span>&#8203;(int&nbsp;b1,
                           int&nbsp;end1,
                           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;c1,
                           int&nbsp;b2,
                           int&nbsp;end2,
                           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;c2,
                           int&nbsp;iniLen,
                           int[][]&nbsp;iniSet)</code></th>
<td class="colLast">
<div class="block">optimize the structural alignment by update the equivalent residues
 and then run dynamic programming
 input: len1 the length of structure 1; c1: the structure information of 1
        len2 the length of structure 2; c2: the structure information of 2
        iniLen and iniSet is the length and list of initial equivalent residues</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.gui">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#getAtomArray(org.biojava.nbio.structure.Atom%5B%5D,java.util.List)">getAtomArray</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms)</code></th>
<td class="colLast">
<div class="block">Returns the first atom for each group</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#getAtomForAligPos(org.biojava.nbio.structure.align.model.AFPChain,int,int,org.biojava.nbio.structure.Atom%5B%5D,boolean)">getAtomForAligPos</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 int&nbsp;chainNr,
                 int&nbsp;aligPos,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
                 boolean&nbsp;getPrevious)</code></th>
<td class="colLast">
<div class="block">Return the atom at alignment position aligPos. at the present only works with block 0</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">createArtificalStructure</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Create a "fake" Structure objects that contains the two sets of atoms aligned on top of each other.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#display(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Group%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.util.List,java.util.List)">display</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;twistedGroups,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
       <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms1,
       <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms2)</code></th>
<td class="colLast">
<div class="block">Note: ca2, hetatoms2 and nucleotides2 should not be rotated.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/gui/StructureAlignmentDisplay.html#display(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">display</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Display an AFPChain alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#getAtomArray(org.biojava.nbio.structure.Atom%5B%5D,java.util.List)">getAtomArray</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
            <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms)</code></th>
<td class="colLast">
<div class="block">Returns the first atom for each group</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#getAtomForAligPos(org.biojava.nbio.structure.align.model.AFPChain,int,int,org.biojava.nbio.structure.Atom%5B%5D,boolean)">getAtomForAligPos</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 int&nbsp;chainNr,
                 int&nbsp;aligPos,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
                 boolean&nbsp;getPrevious)</code></th>
<td class="colLast">
<div class="block">Return the atom at alignment position aligPos. at the present only works with block 0</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#getPDBresnum(int,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)">getPDBresnum</a></span>&#8203;(int&nbsp;aligPos,
            <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">
<div class="block">Return a list of pdb Strings corresponding to the aligned positions of the molecule.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentJmolDisplay.</span><code><span class="memberNameLink"><a href="../align/gui/MultipleAlignmentJmolDisplay.html#getPDBresnum(int,org.biojava.nbio.structure.align.multiple.MultipleAlignment,org.biojava.nbio.structure.Atom%5B%5D)">getPDBresnum</a></span>&#8203;(int&nbsp;structNum,
            <a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">
<div class="block">Utility method used in the <a href="../align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol"><code>MultipleAlignmentJmol</code></a> Frame,
 when the aligned residues of a structure in the alignment have
 to be selected for formatting them (coloring and style).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DisplayAFP.</span><code><span class="memberNameLink"><a href="../align/gui/DisplayAFP.html#showAlignmentPanel(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">showAlignmentPanel</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                  <a href="../align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.gui.aligpanel">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AligPanel.</span><code><span class="memberNameLink"><a href="../align/gui/aligpanel/AligPanel.html#getCa1()">getCa1</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StatusDisplay.</span><code><span class="memberNameLink"><a href="../align/gui/aligpanel/StatusDisplay.html#getCa1()">getCa1</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AligPanel.</span><code><span class="memberNameLink"><a href="../align/gui/aligpanel/AligPanel.html#getCa2()">getCa2</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StatusDisplay.</span><code><span class="memberNameLink"><a href="../align/gui/aligpanel/StatusDisplay.html#getCa2()">getCa2</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AligPanel.</span><code><span class="memberNameLink"><a href="../align/gui/aligpanel/AligPanel.html#setCa1(org.biojava.nbio.structure.Atom%5B%5D)">setCa1</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StatusDisplay.</span><code><span class="memberNameLink"><a href="../align/gui/aligpanel/StatusDisplay.html#setCa1(org.biojava.nbio.structure.Atom%5B%5D)">setCa1</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AligPanel.</span><code><span class="memberNameLink"><a href="../align/gui/aligpanel/AligPanel.html#setCa2(org.biojava.nbio.structure.Atom%5B%5D)">setCa2</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StatusDisplay.</span><code><span class="memberNameLink"><a href="../align/gui/aligpanel/StatusDisplay.html#setCa2(org.biojava.nbio.structure.Atom%5B%5D)">setCa2</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../align/gui/aligpanel/MultipleAligPanel.html#%3Cinit%3E(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">MultipleAligPanel</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                 <a href="../align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code></th>
<td class="colLast">
<div class="block">Constructor using an afpChain and the atom arrays for pairwise
 alignments.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.gui.jmol">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureAlignmentJmol.</span><code><span class="memberNameLink"><a href="../align/gui/jmol/StructureAlignmentJmol.html#getJmolScript4Block(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int)">getJmolScript4Block</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                   int&nbsp;blockNr)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureAlignmentJmol.</span><code><span class="memberNameLink"><a href="../align/gui/jmol/StructureAlignmentJmol.html#getJmolString(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getJmolString</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">JmolTools.</span><code><span class="memberNameLink"><a href="../align/gui/jmol/JmolTools.html#getPdbInfo(org.biojava.nbio.structure.Atom)">getPdbInfo</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></th>
<td class="colLast">
<div class="block">get jmol style info:
  jmol style: [MET]508:A.CA/1 #3918
  insertion code: [ASP]1^A:A.CA/1 #2</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">JmolTools.</span><code><span class="memberNameLink"><a href="../align/gui/jmol/JmolTools.html#getPdbInfo(org.biojava.nbio.structure.Atom,boolean)">getPdbInfo</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
          boolean&nbsp;printResName)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractAlignmentJmol.</span><code><span class="memberNameLink"><a href="../align/gui/jmol/AbstractAlignmentJmol.html#setAtoms(org.biojava.nbio.structure.Atom%5B%5D)">setAtoms</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Create and set a new structure from a given atom array.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../align/gui/jmol/StructureAlignmentJmol.html#%3Cinit%3E(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">StructureAlignmentJmol</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.helper">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/helper/package-summary.html">org.biojava.nbio.structure.align.helper</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/helper/package-summary.html">org.biojava.nbio.structure.align.helper</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignUtils.</span><code><span class="memberNameLink"><a href="../align/helper/AlignUtils.html#getCenter(org.biojava.nbio.structure.Atom%5B%5D,int,int)">getCenter</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
         int&nbsp;pos,
         int&nbsp;fragmentLength)</code></th>
<td class="colLast">
<div class="block">Get the centroid for the set of atoms starting from position pos, length fragmentLenght</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignUtils.</span><code><span class="memberNameLink"><a href="../align/helper/AlignUtils.html#getFragment(org.biojava.nbio.structure.Atom%5B%5D,int,int)">getFragment</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
           int&nbsp;pos,
           int&nbsp;fragmentLength)</code></th>
<td class="colLast">
<div class="block">Get a continue subset of Atoms based by the starting position and the length</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignUtils.</span><code><span class="memberNameLink"><a href="../align/helper/AlignUtils.html#getFragmentFromIdxList(org.biojava.nbio.structure.Atom%5B%5D,int%5B%5D)">getFragmentFromIdxList</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
                      int[]&nbsp;idx)</code></th>
<td class="colLast">
<div class="block">Get a subset of Atoms based by their positions</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignUtils.</span><code><span class="memberNameLink"><a href="../align/helper/AlignUtils.html#getFragmentNoClone(org.biojava.nbio.structure.Atom%5B%5D,int,int)">getFragmentNoClone</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
                  int&nbsp;pos,
                  int&nbsp;fragmentLength)</code></th>
<td class="colLast">
<div class="block">Get a continue subset of Atoms based by the starting position and the length
 does not clone the original atoms.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../align/helper/package-summary.html">org.biojava.nbio.structure.align.helper</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignUtils.</span><code><span class="memberNameLink"><a href="../align/helper/AlignUtils.html#getCenter(org.biojava.nbio.structure.Atom%5B%5D,int,int)">getCenter</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
         int&nbsp;pos,
         int&nbsp;fragmentLength)</code></th>
<td class="colLast">
<div class="block">Get the centroid for the set of atoms starting from position pos, length fragmentLenght</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignUtils.</span><code><span class="memberNameLink"><a href="../align/helper/AlignUtils.html#getDiagonalAtK(org.biojava.nbio.structure.Atom%5B%5D,int)">getDiagonalAtK</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
              int&nbsp;k)</code></th>
<td class="colLast">
<div class="block">Get distances along diagonal k from coordinate array coords.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignUtils.</span><code><span class="memberNameLink"><a href="../align/helper/AlignUtils.html#getDistanceMatrix(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getDistanceMatrix</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Matrix of all distances between two sets of Atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignUtils.</span><code><span class="memberNameLink"><a href="../align/helper/AlignUtils.html#getFragment(org.biojava.nbio.structure.Atom%5B%5D,int,int)">getFragment</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
           int&nbsp;pos,
           int&nbsp;fragmentLength)</code></th>
<td class="colLast">
<div class="block">Get a continue subset of Atoms based by the starting position and the length</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignUtils.</span><code><span class="memberNameLink"><a href="../align/helper/AlignUtils.html#getFragmentFromIdxList(org.biojava.nbio.structure.Atom%5B%5D,int%5B%5D)">getFragmentFromIdxList</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
                      int[]&nbsp;idx)</code></th>
<td class="colLast">
<div class="block">Get a subset of Atoms based by their positions</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignUtils.</span><code><span class="memberNameLink"><a href="../align/helper/AlignUtils.html#getFragmentNoClone(org.biojava.nbio.structure.Atom%5B%5D,int,int)">getFragmentNoClone</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
                  int&nbsp;pos,
                  int&nbsp;fragmentLength)</code></th>
<td class="colLast">
<div class="block">Get a continue subset of Atoms based by the starting position and the length
 does not clone the original atoms.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.model">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/model/package-summary.html">org.biojava.nbio.structure.align.model</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/model/package-summary.html">org.biojava.nbio.structure.align.model</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChain.</span><code><span class="memberNameLink"><a href="../align/model/AFPChain.html#getBlockShiftVector()">getBlockShiftVector</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../align/model/package-summary.html">org.biojava.nbio.structure.align.model</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChain.</span><code><span class="memberNameLink"><a href="../align/model/AFPChain.html#setBlockShiftVector(org.biojava.nbio.structure.Atom%5B%5D)">setBlockShiftVector</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;blockShiftVector)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AfpChainWriter.</span><code><span class="memberNameLink"><a href="../align/model/AfpChainWriter.html#toAlignedPairs(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">toAlignedPairs</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Prints the alignment in the simplest form: a list of aligned residues.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChain.</span><code><span class="memberNameLink"><a href="../align/model/AFPChain.html#toCE(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">toCE</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AfpChainWriter.</span><code><span class="memberNameLink"><a href="../align/model/AfpChainWriter.html#toCE(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">toCE</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
    <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChain.</span><code><span class="memberNameLink"><a href="../align/model/AFPChain.html#toFatcat(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">toFatcat</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AfpChainWriter.</span><code><span class="memberNameLink"><a href="../align/model/AfpChainWriter.html#toFatCat(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">toFatCat</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AfpChainWriter.</span><code><span class="memberNameLink"><a href="../align/model/AfpChainWriter.html#toFatCatCore(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean,boolean,boolean,boolean)">toFatCatCore</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            boolean&nbsp;printLegend,
            boolean&nbsp;longHeader,
            boolean&nbsp;showHTML,
            boolean&nbsp;showAlignmentBlock)</code></th>
<td class="colLast">
<div class="block">Output in FatCatCore format</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AfpChainWriter.</span><code><span class="memberNameLink"><a href="../align/model/AfpChainWriter.html#toPrettyAlignment(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean,boolean)">toPrettyAlignment</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                 boolean&nbsp;showHTML,
                 boolean&nbsp;showAlignmentBlock)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AfpChainWriter.</span><code><span class="memberNameLink"><a href="../align/model/AfpChainWriter.html#toWebSiteDisplay(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">toWebSiteDisplay</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Prints the afpChain as a nicely formatted alignment, including alignment
 statistics, the aligned sequences themselves, and information about the
 superposition.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AfpChainWriter.</span><code><span class="memberNameLink"><a href="../align/model/AfpChainWriter.html#toWebSiteDisplay(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean)">toWebSiteDisplay</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                boolean&nbsp;showAlignmentBlock)</code></th>
<td class="colLast">
<div class="block">Prints the afpChain as a nicely formatted alignment, including alignment
 statistics, the aligned sequences themselves, and information about the
 superposition.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.multiple">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a></h3>
<table class="useSummary">
<caption><span>Constructors in <a href="../align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../align/multiple/MultipleAlignmentEnsembleImpl.html#%3Cinit%3E(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean)">MultipleAlignmentEnsembleImpl</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afp,
                             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                             boolean&nbsp;flexible)</code></th>
<td class="colLast">
<div class="block">Constructor from an AFPChain instance.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.multiple.util">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../align/multiple/util/MultipleAlignmentTools.html#getAtomForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,int,int)">getAtomForSequencePosition</a></span>&#8203;(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                          <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;mapSeqToStruct,
                          int&nbsp;str,
                          int&nbsp;sequencePos)</code></th>
<td class="colLast">
<div class="block">Returns the Atom of the specified structure that is aligned in the
 sequence alignment position specified.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.pairwise">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentPair.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentPair.html#getCenter1()">getCenter1</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentPair.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentPair.html#getCenter2()">getCenter2</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlternativeAlignment.</span><code><span class="memberNameLink"><a href="../align/pairwise/AlternativeAlignment.html#getShift()">getShift</a></span>()</code></th>
<td class="colLast">
<div class="block">returns the shift vector that has to be applied on structure to to shift on structure one</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentPair.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentPair.html#getTrans()">getTrans</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentPair.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentPair.html#getUnitv()">getUnitv</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentJoiner.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentJoiner.html#approach_ap3(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.pairwise.FragmentPair%5B%5D,org.biojava.nbio.structure.align.StrucAligParameters)">approach_ap3</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            <a href="../align/pairwise/FragmentPair.html" title="class in org.biojava.nbio.structure.align.pairwise">FragmentPair</a>[]&nbsp;fraglst,
            <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlternativeAlignment.</span><code><span class="memberNameLink"><a href="../align/pairwise/AlternativeAlignment.html#calcScores(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">calcScores</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">calculates  scores for this alignment ( %id )</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlternativeAlignment.</span><code><span class="memberNameLink"><a href="../align/pairwise/AlternativeAlignment.html#calculateSuperpositionByIdx(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">calculateSuperpositionByIdx</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentJoiner.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentJoiner.html#extendFragments(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.helper.JointFragments%5B%5D,org.biojava.nbio.structure.align.StrucAligParameters)">extendFragments</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
               <a href="../align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>[]&nbsp;fragments,
               <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentJoiner.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentJoiner.html#extendFragments(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.helper.JointFragments,org.biojava.nbio.structure.align.StrucAligParameters)">extendFragments</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
               <a href="../align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>&nbsp;fragments,
               <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlternativeAlignment.</span><code><span class="memberNameLink"><a href="../align/pairwise/AlternativeAlignment.html#finish(org.biojava.nbio.structure.align.StrucAligParameters,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">finish</a></span>&#8203;(<a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlternativeAlignment.</span><code><span class="memberNameLink"><a href="../align/pairwise/AlternativeAlignment.html#getDistanceMatrix(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getDistanceMatrix</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentJoiner.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentJoiner.html#getRMS(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.helper.JointFragments)">getRMS</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
      <a href="../align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>&nbsp;frag)</code></th>
<td class="colLast">
<div class="block">Get the RMS of the JointFragments pair frag</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlternativeAlignment.</span><code><span class="memberNameLink"><a href="../align/pairwise/AlternativeAlignment.html#refine(org.biojava.nbio.structure.align.StrucAligParameters,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">refine</a></span>&#8203;(<a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Refinement procedure based on superposition and dynamic programming.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentPair.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentPair.html#setCenter1(org.biojava.nbio.structure.Atom)">setCenter1</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;center1)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentPair.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentPair.html#setCenter2(org.biojava.nbio.structure.Atom)">setCenter2</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;center2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentPair.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentPair.html#setTrans(org.biojava.nbio.structure.Atom)">setTrans</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;trans)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FragmentPair.</span><code><span class="memberNameLink"><a href="../align/pairwise/FragmentPair.html#setUnitv(org.biojava.nbio.structure.Atom)">setUnitv</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;unitv)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentProgressListener.</span><code><span class="memberNameLink"><a href="../align/pairwise/AlignmentProgressListener.html#startingAlignment(java.lang.String,org.biojava.nbio.structure.Atom%5B%5D,java.lang.String,org.biojava.nbio.structure.Atom%5B%5D)">startingAlignment</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name1,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name2,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.quaternary">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">QsAlignResult.</span><code><span class="memberNameLink"><a href="../align/quaternary/QsAlignResult.html#getAlignedAtomsForSubunits1(int)">getAlignedAtomsForSubunits1</a></span>&#8203;(int&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">QsAlignResult.</span><code><span class="memberNameLink"><a href="../align/quaternary/QsAlignResult.html#getAlignedAtomsForSubunits2(int)">getAlignedAtomsForSubunits2</a></span>&#8203;(int&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.seq">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/seq/package-summary.html">org.biojava.nbio.structure.align.seq</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/seq/package-summary.html">org.biojava.nbio.structure.align.seq</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SmithWaterman3Daligner.</span><code><span class="memberNameLink"><a href="../align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SmithWaterman3Daligner.</span><code><span class="memberNameLink"><a href="../align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;parameters)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.util">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#getAlignedAtoms1(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)">getAlignedAtoms1</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#getAlignedAtoms2(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)">getAlignedAtoms2</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getAtoms(java.lang.String)">getAtoms</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Returns the CA atoms for the provided name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getAtoms(org.biojava.nbio.structure.StructureIdentifier)">getAtoms</a></span>&#8203;(<a href="../StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;name)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RotationAxis.</span><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#getProjectedPoint(org.biojava.nbio.structure.Atom)">getProjectedPoint</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;point)</code></th>
<td class="colLast">
<div class="block">Projects a given point onto the axis of rotation</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getRepresentativeAtoms(java.lang.String)">getRepresentativeAtoms</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name)</code></th>
<td class="colLast">
<div class="block">Returns the representative atoms for the provided name.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomCache.</span><code><span class="memberNameLink"><a href="../align/util/AtomCache.html#getRepresentativeAtoms(org.biojava.nbio.structure.StructureIdentifier)">getRepresentativeAtoms</a></span>&#8203;(<a href="../StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;name)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RotationAxis.</span><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#getRotationAxis()">getRotationAxis</a></span>()</code></th>
<td class="colLast">
<div class="block">Get a unit vector along the rotation axis</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RotationAxis.</span><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#getRotationPos()">getRotationPos</a></span>()</code></th>
<td class="colLast">
<div class="block">Get a position on the rotation axis.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RotationAxis.</span><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#getScrewTranslation()">getScrewTranslation</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the component of translation parallel to the axis of rotation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#getTranslation(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getTranslation</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> that return types with arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RotationAxis.</span><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#getAxisEnds(org.biojava.nbio.structure.Atom%5B%5D)">getAxisEnds</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Find a segment of the axis that covers the specified set of atoms.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#alignmentToSIF(java.io.Writer,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.String,java.lang.String)">alignmentToSIF</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/Writer.html?is-external=true" title="class or interface in java.io" class="externalLink">Writer</a>&nbsp;out,
              <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;backboneInteraction,
              <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;alignmentInteraction)</code></th>
<td class="colLast">
<div class="block">Creates a simple interaction format (SIF) file for an alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#createAFPChain(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.ResidueNumber%5B%5D,org.biojava.nbio.structure.ResidueNumber%5B%5D)">createAFPChain</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              <a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned1,
              <a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned2)</code></th>
<td class="colLast">
<div class="block">Fundamentally, an alignment is just a list of aligned residues in each
 protein.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">createArtificalStructure</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#deleteColumn(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int)">deleteColumn</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            int&nbsp;block,
            int&nbsp;pos)</code></th>
<td class="colLast">
<div class="block">Delete an alignment position from the original alignment object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#deleteHighestDistanceColumn(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">deleteHighestDistanceColumn</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Find the alignment position with the highest atomic distance between the
 equivalent atomic positions of the arrays and remove it from the
 alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#fillAlignedAtomArrays(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">fillAlignedAtomArrays</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1aligned,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2aligned)</code></th>
<td class="colLast">
<div class="block">Fill the aligned Atom arrays with the equivalent residues in the afpChain.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#getAlign(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getAlign</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Extract the alignment output</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#getAlign(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean)">getAlign</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
        boolean&nbsp;showSeq)</code></th>
<td class="colLast">
<div class="block">Sets the following properties:
 
 The alignment strings <a href="../align/model/AFPChain.html#setAlnseq1(char%5B%5D)"><code>alnseq1</code></a>,
  <a href="../align/model/AFPChain.html#setAlnseq2(char%5B%5D)"><code>alnseq2</code></a>,
  and <a href="../align/model/AFPChain.html#setAlnsymb(char%5B%5D)"><code>alnsymb</code></a>
 <a href="../align/model/AFPChain.html#setAlnbeg1(int)"><code>alnbeg1</code></a> and 2
 <a href="../align/model/AFPChain.html#setAlnLength(int)"><code>alnLength</code></a> and
  <a href="../align/model/AFPChain.html#setGapLen(int)"><code>gapLen</code></a>
 </div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#getAlignedAtoms1(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)">getAlignedAtoms1</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#getAlignedAtoms2(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)">getAlignedAtoms2</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#getAlignedModel(org.biojava.nbio.structure.Atom%5B%5D)">getAlignedModel</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">
<div class="block">get an artificial List of chains containing the Atoms and groups.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#getAlignedStructure(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getAlignedStructure</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Get an artifical Structure containing both chains.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RotationAxis.</span><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#getAxisEnds(org.biojava.nbio.structure.Atom%5B%5D)">getAxisEnds</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Find a segment of the axis that covers the specified set of atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RotationAxis.</span><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#getJmolScript(org.biojava.nbio.structure.Atom%5B%5D)">getJmolScript</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Returns a Jmol script which will display the axis of rotation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RotationAxis.</span><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#getJmolScript(org.biojava.nbio.structure.Atom%5B%5D,int)">getJmolScript</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
             int&nbsp;axisID)</code></th>
<td class="colLast">
<div class="block">Returns a Jmol script which will display the axis of rotation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RotationAxis.</span><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#getProjectedDistance(org.biojava.nbio.structure.Atom)">getProjectedDistance</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;point)</code></th>
<td class="colLast">
<div class="block">Get the distance from a point to the axis of rotation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RotationAxis.</span><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#getProjectedPoint(org.biojava.nbio.structure.Atom)">getProjectedPoint</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;point)</code></th>
<td class="colLast">
<div class="block">Projects a given point onto the axis of rotation</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#getRotMax(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getRotMax</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
         <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
         <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainScorer.</span><code><span class="memberNameLink"><a href="../align/util/AFPChainScorer.html#getTMScore(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getTMScore</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;align,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainScorer.</span><code><span class="memberNameLink"><a href="../align/util/AFPChainScorer.html#getTMScore(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean)">getTMScore</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;align,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
          boolean&nbsp;normalizeMin)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../align/util/AFPAlignmentDisplay.html#getTranslation(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getTranslation</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#prepareGroupsForDisplay(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">prepareGroupsForDisplay</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Rotate the Atoms/Groups so they are aligned for the 3D visualisation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#replaceOptAln(int%5B%5D%5B%5D%5B%5D,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">replaceOptAln</a></span>&#8203;(int[][][]&nbsp;newAlgn,
             <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">It replaces an optimal alignment of an AFPChain and calculates all the new alignment scores and variables.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int%5B%5D,int%5B%5D%5B%5D%5B%5D)">replaceOptAln</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             int&nbsp;blockNum,
             int[]&nbsp;optLens,
             int[][][]&nbsp;optAln)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.util.Map)">replaceOptAln</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>,&#8203;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Takes an AFPChain and replaces the optimal alignment based on an alignment map</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">RotationAxis.</span><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#rotate(org.biojava.nbio.structure.Atom%5B%5D,double)">rotate</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      double&nbsp;theta)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#shiftCA2(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.jama.Matrix,org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Group%5B%5D)">shiftCA2</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
        <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;m,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;shift,
        <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;twistedGroups)</code></th>
<td class="colLast">
<div class="block">only shift CA positions.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#splitBlocksByTopology(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">splitBlocksByTopology</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;a,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentTools.</span><code><span class="memberNameLink"><a href="../align/util/AlignmentTools.html#updateSuperposition(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">updateSuperposition</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">After the alignment changes (optAln, optLen, blockNum, at a minimum),
 many other properties which depend on the superposition will be invalid.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#%3Cinit%3E(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom,double)">RotationAxis</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;axis,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;pos,
            double&nbsp;theta)</code></th>
<td class="colLast">
<div class="block">Create a rotation axis from a vector, a point, and an angle.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../align/util/RotationAxis.html#%3Cinit%3E(org.biojava.nbio.structure.jama.Matrix,org.biojava.nbio.structure.Atom)">RotationAxis</a></span>&#8203;(<a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;rotation,
            <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;translation)</code></th>
<td class="colLast">
<div class="block">Determine the location of the rotation axis based on a rotation matrix and a translation vector</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.xml">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainXMLParser.</span><code><span class="memberNameLink"><a href="../align/xml/AFPChainXMLParser.html#fromXML(java.lang.String,java.lang.String,java.lang.String,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">fromXML</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;xml,
       <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name1,
       <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name2,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">new utility method that checks that the order of the pair in the XML alignment is correct and flips the direction if needed</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainXMLParser.</span><code><span class="memberNameLink"><a href="../align/xml/AFPChainXMLParser.html#fromXML(java.lang.String,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">fromXML</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;xml,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainXMLConverter.</span><code><span class="memberNameLink"><a href="../align/xml/AFPChainXMLConverter.html#printXMLEQRInferPositions(org.biojava.nbio.core.util.PrettyXMLWriter,org.biojava.nbio.structure.align.model.AFPChain,int,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">printXMLEQRInferPositions</a></span>&#8203;(<a href="../../core/util/PrettyXMLWriter.html" title="class in org.biojava.nbio.core.util">PrettyXMLWriter</a>&nbsp;xml,
                         <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                         int&nbsp;bk,
                         <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                         <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainXMLParser.</span><code><span class="memberNameLink"><a href="../align/xml/AFPChainXMLParser.html#rebuildAFPChain(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">rebuildAFPChain</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">replace the PDB res nums with atom positions:</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainXMLConverter.</span><code><span class="memberNameLink"><a href="../align/xml/AFPChainXMLConverter.html#toXML(org.biojava.nbio.structure.align.model.AFPChain,java.io.StringWriter,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">toXML</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/io/StringWriter.html?is-external=true" title="class or interface in java.io" class="externalLink">StringWriter</a>&nbsp;swriter,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Write the XML representation to a StringWriter</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AFPChainXMLConverter.</span><code><span class="memberNameLink"><a href="../align/xml/AFPChainXMLConverter.html#toXML(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">toXML</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Convert an afpChain to a simple XML representation</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.asa">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../asa/package-summary.html">org.biojava.nbio.structure.asa</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../asa/package-summary.html">org.biojava.nbio.structure.asa</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AsaCalculator.</span><code><span class="memberNameLink"><a href="../asa/AsaCalculator.html#getRadius(org.biojava.nbio.structure.Atom)">getRadius</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom)</code></th>
<td class="colLast">
<div class="block">Gets the van der Waals radius of the given atom following the values defined by
 Chothia (1976) J.Mol.Biol.105,1-14
 NOTE: the vdw values defined by the paper assume no Hydrogens and thus "inflates"
 slightly the heavy atoms to account for Hydrogens.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../asa/package-summary.html">org.biojava.nbio.structure.asa</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../asa/AsaCalculator.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D,double,int,int)">AsaCalculator</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
             double&nbsp;probe,
             int&nbsp;nSpherePoints,
             int&nbsp;nThreads)</code></th>
<td class="colLast">
<div class="block">Constructs a new AsaCalculator.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.cluster">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../cluster/package-summary.html">org.biojava.nbio.structure.cluster</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SubunitCluster.</span><code><span class="memberNameLink"><a href="../cluster/SubunitCluster.html#getAlignedAtomsSubunit(int)">getAlignedAtomsSubunit</a></span>&#8203;(int&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Subunit.</span><code><span class="memberNameLink"><a href="../cluster/Subunit.html#getRepresentativeAtoms()">getRepresentativeAtoms</a></span>()</code></th>
<td class="colLast">
<div class="block">Get all the representative Atoms of the Subunit.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../cluster/Subunit.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D,java.lang.String,org.biojava.nbio.structure.StructureIdentifier,org.biojava.nbio.structure.Structure)">Subunit</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;reprAtoms,
       <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;name,
       <a href="../StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;identifier,
       <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code></th>
<td class="colLast">
<div class="block">A Subunit is solely defined by the coordinates of the representative
 Atoms of its residues.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.contact">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureInterface.</span><code><span class="memberNameLink"><a href="../contact/StructureInterface.html#getAtomsForAsa(int)">getAtomsForAsa</a></span>&#8203;(int&nbsp;cofactorSizeToUse)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureInterface.</span><code><span class="memberNameLink"><a href="../contact/StructureInterface.html#getFirstAtomsForAsa(int)">getFirstAtomsForAsa</a></span>&#8203;(int&nbsp;cofactorSizeToUse)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">StructureInterface.</span><code><span class="memberNameLink"><a href="../contact/StructureInterface.html#getSecondAtomsForAsa(int)">getSecondAtomsForAsa</a></span>&#8203;(int&nbsp;cofactorSizeToUse)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a> that return types with arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomContact.</span><code><span class="memberNameLink"><a href="../contact/AtomContact.html#getPair()">getPair</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Grid.</span><code><span class="memberNameLink"><a href="../contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom%5B%5D)">addAtoms</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Adds a set of atoms, subsequent call to <a href="../contact/Grid.html#getIndicesContacts()"><code>Grid.getIndicesContacts()</code></a> or <a href="../contact/Grid.html#getAtomContacts()"><code>Grid.getAtomContacts()</code></a> will produce the interatomic contacts.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Grid.</span><code><span class="memberNameLink"><a href="../contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">addAtoms</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;iAtoms,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;jAtoms)</code></th>
<td class="colLast">
<div class="block">Adds the i and j atoms and fills the grid.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Grid.</span><code><span class="memberNameLink"><a href="../contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.contact.BoundingBox)">addAtoms</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
        <a href="../contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a>&nbsp;bounds)</code></th>
<td class="colLast">
<div class="block">Adds a set of atoms, subsequent call to <a href="../contact/Grid.html#getIndicesContacts()"><code>Grid.getIndicesContacts()</code></a> or <a href="../contact/Grid.html#getAtomContacts()"><code>Grid.getAtomContacts()</code></a> will produce the interatomic contacts.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Grid.</span><code><span class="memberNameLink"><a href="../contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.contact.BoundingBox,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.contact.BoundingBox)">addAtoms</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;iAtoms,
        <a href="../contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a>&nbsp;icoordbounds,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;jAtoms,
        <a href="../contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a>&nbsp;jcoordbounds)</code></th>
<td class="colLast">
<div class="block">Adds the i and j atoms and fills the grid, passing their bounds (array of size 6 with x,y,z minima and x,y,z maxima)
 This way the bounds don't need to be recomputed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../contact/AtomContact.html" title="class in org.biojava.nbio.structure.contact">AtomContact</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomContactSet.</span><code><span class="memberNameLink"><a href="../contact/AtomContactSet.html#getContact(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">getContact</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom2)</code></th>
<td class="colLast">
<div class="block">Returns the corresponding AtomContact or null if no contact exists between the 2 given atoms</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomContactSet.</span><code><span class="memberNameLink"><a href="../contact/AtomContactSet.html#hasContact(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)">hasContact</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom1,
          <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom2)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a> with type arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AtomContact.</span><code><span class="memberNameLink"><a href="../contact/AtomContact.html#setPair(org.biojava.nbio.structure.contact.Pair)">setPair</a></span>&#8203;(<a href="../contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;pair)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../contact/StructureInterface.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.String,java.lang.String,org.biojava.nbio.structure.contact.AtomContactSet,org.biojava.nbio.structure.xtal.CrystalTransform,org.biojava.nbio.structure.xtal.CrystalTransform)">StructureInterface</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;firstMolecule,
                  <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;secondMolecule,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;firstMoleculeId,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;secondMoleculeId,
                  <a href="../contact/AtomContactSet.html" title="class in org.biojava.nbio.structure.contact">AtomContactSet</a>&nbsp;contacts,
                  <a href="../xtal/CrystalTransform.html" title="class in org.biojava.nbio.structure.xtal">CrystalTransform</a>&nbsp;firstTransf,
                  <a href="../xtal/CrystalTransform.html" title="class in org.biojava.nbio.structure.xtal">CrystalTransform</a>&nbsp;secondTransf)</code></th>
<td class="colLast">
<div class="block">Constructs a StructureInterface</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructor parameters in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a> with type arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../contact/AtomContact.html#%3Cinit%3E(org.biojava.nbio.structure.contact.Pair,double)">AtomContact</a></span>&#8203;(<a href="../contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;pair,
           double&nbsp;distance)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.domain">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../domain/package-summary.html">org.biojava.nbio.structure.domain</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../domain/package-summary.html">org.biojava.nbio.structure.domain</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">LocalProteinDomainParser.</span><code><span class="memberNameLink"><a href="../domain/LocalProteinDomainParser.html#suggestDomains(org.biojava.nbio.structure.Atom%5B%5D)">suggestDomains</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">
<div class="block">Suggest domains for a set of Calpha atoms</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.domain.pdp">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">Cut.</span><code><span class="memberNameLink"><a href="../domain/pdp/Cut.html#cut(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.domain.pdp.Domain,org.biojava.nbio.structure.domain.pdp.CutValues,int%5B%5D%5B%5D,org.biojava.nbio.structure.domain.pdp.PDPDistanceMatrix)">cut</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
   <a href="../domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a>&nbsp;dom,
   <a href="../domain/pdp/CutValues.html" title="class in org.biojava.nbio.structure.domain.pdp">CutValues</a>&nbsp;val,
   int[][]&nbsp;dist,
   <a href="../domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a>&nbsp;pdpMatrix)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GetDistanceMatrix.</span><code><span class="memberNameLink"><a href="../domain/pdp/GetDistanceMatrix.html#getDistanceMatrix(org.biojava.nbio.structure.Atom%5B%5D)">getDistanceMatrix</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;protein)</code></th>
<td class="colLast">
<div class="block">A set of Calpha atoms that are representing the protein</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../domain/pdp/CutDomain.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.domain.pdp.PDPDistanceMatrix)">CutDomain</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
         <a href="../domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a>&nbsp;pdpMatrix)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.io">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../io/package-summary.html">org.biojava.nbio.structure.io</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../io/package-summary.html">org.biojava.nbio.structure.io</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileConvert.</span><code><span class="memberNameLink"><a href="../io/FileConvert.html#toPDB(org.biojava.nbio.structure.Atom)">toPDB</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></th>
<td class="colLast">
<div class="block">Prints the content of an Atom object as a PDB formatted line.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileConvert.</span><code><span class="memberNameLink"><a href="../io/FileConvert.html#toPDB(org.biojava.nbio.structure.Atom,java.lang.String)">toPDB</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chainId)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileConvert.</span><code><span class="memberNameLink"><a href="../io/FileConvert.html#toPDB(org.biojava.nbio.structure.Atom,java.lang.StringBuffer)">toPDB</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuffer.html?is-external=true" title="class or interface in java.lang" class="externalLink">StringBuffer</a>&nbsp;str)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FileConvert.</span><code><span class="memberNameLink"><a href="../io/FileConvert.html#toPDB(org.biojava.nbio.structure.Atom,java.lang.StringBuffer,java.lang.String)">toPDB</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuffer.html?is-external=true" title="class or interface in java.lang" class="externalLink">StringBuffer</a>&nbsp;str,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chainID)</code></th>
<td class="colLast">
<div class="block">Print ATOM record in the following syntax</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.io.mmcif">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../io/mmcif/package-summary.html">org.biojava.nbio.structure.io.mmcif</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../io/mmcif/package-summary.html">org.biojava.nbio.structure.io.mmcif</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../io/mmcif/model/AtomSite.html" title="class in org.biojava.nbio.structure.io.mmcif.model">AtomSite</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MMCIFFileTools.</span><code><span class="memberNameLink"><a href="../io/mmcif/MMCIFFileTools.html#convertAtomToAtomSite(org.biojava.nbio.structure.Atom,int,java.lang.String,java.lang.String)">convertAtomToAtomSite</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
                     int&nbsp;model,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chainName,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chainId)</code></th>
<td class="colLast">
<div class="block">Converts an Atom object to an <a href="../io/mmcif/model/AtomSite.html" title="class in org.biojava.nbio.structure.io.mmcif.model"><code>AtomSite</code></a> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../io/mmcif/model/AtomSite.html" title="class in org.biojava.nbio.structure.io.mmcif.model">AtomSite</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MMCIFFileTools.</span><code><span class="memberNameLink"><a href="../io/mmcif/MMCIFFileTools.html#convertAtomToAtomSite(org.biojava.nbio.structure.Atom,int,java.lang.String,java.lang.String,int)">convertAtomToAtomSite</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
                     int&nbsp;model,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chainName,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;chainId,
                     int&nbsp;atomId)</code></th>
<td class="colLast">
<div class="block">Converts an Atom object to an <a href="../io/mmcif/model/AtomSite.html" title="class in org.biojava.nbio.structure.io.mmcif.model"><code>AtomSite</code></a> object.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.io.mmtf">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../io/mmtf/package-summary.html">org.biojava.nbio.structure.io.mmtf</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../io/mmtf/package-summary.html">org.biojava.nbio.structure.io.mmtf</a> that return types with arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MmtfSummaryDataBean.</span><code><span class="memberNameLink"><a href="../io/mmtf/MmtfSummaryDataBean.html#getAllAtoms()">getAllAtoms</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MmtfUtils.</span><code><span class="memberNameLink"><a href="../io/mmtf/MmtfUtils.html#getAtomsForGroup(org.biojava.nbio.structure.Group)">getAtomsForGroup</a></span>&#8203;(<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;inputGroup)</code></th>
<td class="colLast">
<div class="block">Get a list of atoms for a group.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../io/mmtf/package-summary.html">org.biojava.nbio.structure.io.mmtf</a> with type arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MmtfUtils.</span><code><span class="memberNameLink"><a href="../io/mmtf/MmtfUtils.html#getNumBondsInGroup(java.util.List)">getNumBondsInGroup</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;atomsInGroup)</code></th>
<td class="colLast">
<div class="block">Find the number of bonds in a group</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MmtfSummaryDataBean.</span><code><span class="memberNameLink"><a href="../io/mmtf/MmtfSummaryDataBean.html#setAllAtoms(java.util.List)">setAllAtoms</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;allAtoms)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.symmetry.internal">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymmResult.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymmResult.html#getAtoms()">getAtoms</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected static <a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymm.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymm.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.symmetry.internal.CESymmParameters)">align</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
     <a href="../symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymm.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymm.html#analyze(org.biojava.nbio.structure.Atom%5B%5D)">analyze</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Analyze the symmetries of the input Atom array using the DEFAULT
 parameters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymm.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymm.html#analyze(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.symmetry.internal.CESymmParameters)">analyze</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
       <a href="../symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Analyze the symmetries of the input Atom array using the provided
 parameters.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymm.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymm.html#analyzeLevel(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.symmetry.internal.CESymmParameters)">analyzeLevel</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
            <a href="../symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Analyze a single level of symmetry.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AngleOrderDetectorPlus.</span><code><span class="memberNameLink"><a href="../symmetry/internal/AngleOrderDetectorPlus.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)">calculateOrder</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GraphComponentOrderDetector.</span><code><span class="memberNameLink"><a href="../symmetry/internal/GraphComponentOrderDetector.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)">calculateOrder</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">OrderDetector.</span><code><span class="memberNameLink"><a href="../symmetry/internal/OrderDetector.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)">calculateOrder</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceFunctionOrderDetector.</span><code><span class="memberNameLink"><a href="../symmetry/internal/SequenceFunctionOrderDetector.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)">calculateOrder</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymmIterative.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymmIterative.html#execute(org.biojava.nbio.structure.Atom%5B%5D)">execute</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">This method uses iteratively CeSymm to calculate all symmetries in the
 input array of atoms and organize them in a multiple alignment of the
 repeats.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GraphComponentRefiner.</span><code><span class="memberNameLink"><a href="../symmetry/internal/GraphComponentRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)">refine</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceFunctionRefiner.</span><code><span class="memberNameLink"><a href="../symmetry/internal/SequenceFunctionRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)">refine</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryRefiner.</span><code><span class="memberNameLink"><a href="../symmetry/internal/SymmetryRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)">refine</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code></th>
<td class="colLast">
<div class="block">Returns a refined symmetry alignment, where the repeat residues are
 aligned consistently in a MultipleAlignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceFunctionRefiner.</span><code><span class="memberNameLink"><a href="../symmetry/internal/SequenceFunctionRefiner.html#refineSymmetry(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int)">refineSymmetry</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              int&nbsp;k)</code></th>
<td class="colLast">
<div class="block">Refines a CE-Symm alignment so that it is perfectly symmetric.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymmResult.</span><code><span class="memberNameLink"><a href="../symmetry/internal/CeSymmResult.html#setAtoms(org.biojava.nbio.structure.Atom%5B%5D)">setAtoms</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.symmetry.utils">
<!--   -->
</a>
<h3>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#getRepresentativeAtoms(org.biojava.nbio.structure.Structure)">getRepresentativeAtoms</a></span>&#8203;(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code></th>
<td class="colLast">
<div class="block">Returns the representative Atom Array of the first model, if the
 structure is NMR, or the Array for each model, if it is a biological
 assembly with multiple models.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static boolean[][]</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#blankOutBreakFlag(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.align.ce.CECalculator,boolean%5B%5D%5B%5D,int)">blankOutBreakFlag</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                 int&nbsp;rows,
                 int&nbsp;cols,
                 <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                 boolean[][]&nbsp;breakFlag,
                 int&nbsp;blankWindowSize)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#blankOutCEOrig(org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.jama.Matrix,int)">blankOutCEOrig</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              int&nbsp;rows,
              int&nbsp;cols,
              <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
              <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;origM,
              int&nbsp;blankWindowSize)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#blankOutPreviousAlignment(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.jama.Matrix,int)">blankOutPreviousAlignment</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                         <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                         int&nbsp;rows,
                         int&nbsp;cols,
                         <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                         <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;max,
                         int&nbsp;blankWindowSize)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#buildSymmetryGraph(java.util.List,org.biojava.nbio.structure.Atom%5B%5D,boolean)">buildSymmetryGraph</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&gt;&nbsp;afps,
                  <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
                  boolean&nbsp;undirected)</code></th>
<td class="colLast">
<div class="block">Converts a set of AFP alignments into a Graph of aligned residues, where
 each vertex is a residue and each edge means the connection between the
 two residues in one of the alignments.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#fromAFP(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)">fromAFP</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;symm,
       <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Converts a refined symmetry AFPChain alignment into the standard
 representation of symmetry in a MultipleAlignment, that contains the
 entire Atom array of the strcuture and the symmetric repeats are orgaized
 in different rows in a single Block.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#getAngle(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)">getAngle</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></th>
<td class="colLast">
<div class="block">Returns the <em>magnitude</em> of the angle between the first and second
 blocks of <code>afpChain</code>, measured in degrees.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#getDkMatrix(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,double%5B%5D,double%5B%5D,int,int)">getDkMatrix</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
           int&nbsp;fragmentLength,
           double[]&nbsp;dist1,
           double[]&nbsp;dist2,
           int&nbsp;rows,
           int&nbsp;cols)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#getDkMatrix(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int)">getDkMatrix</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
           int&nbsp;k,
           int&nbsp;fragmentLength)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#getGroups(org.biojava.nbio.structure.Atom%5B%5D)">getGroups</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;rAtoms)</code></th>
<td class="colLast">
<div class="block">Returns the List of Groups of the corresponding representative Atom
 array.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#grayOutCEOrig(org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.jama.Matrix,int,double%5B%5D,double)">grayOutCEOrig</a></span>&#8203;(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             int&nbsp;rows,
             int&nbsp;cols,
             <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
             <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;origM,
             int&nbsp;blankWindowSize,
             double[]&nbsp;gradientPolyCoeff,
             double&nbsp;gradientExpCoeff)</code></th>
<td class="colLast">
<div class="block">Grays out the main diagonal of a duplicated distance matrix.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../symmetry/utils/SymmetryTools.html#grayOutPreviousAlignment(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.jama.Matrix,int,double%5B%5D,double)">grayOutPreviousAlignment</a></span>&#8203;(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                        int&nbsp;rows,
                        int&nbsp;cols,
                        <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                        <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;max,
                        int&nbsp;blankWindowSize,
                        double[]&nbsp;gradientPolyCoeff,
                        double&nbsp;gradientExpCoeff)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2019 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
