<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_162) on Fri Apr 06 16:19:07 PDT 2018 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>org.biojava.nbio.structure.io (biojava 5.0.0 API)</title>
<meta name="date" content="2018-04-06">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="org.biojava.nbio.structure.io (biojava 5.0.0 API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li class="navBarCell1Rev">Package</li>
<li>Class</li>
<li><a href="package-use.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/nbio/structure/gui/util/color/package-summary.html">Prev&nbsp;Package</a></li>
<li><a href="../../../../../org/biojava/nbio/structure/io/mmcif/package-summary.html">Next&nbsp;Package</a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/structure/io/package-summary.html" target="_top">Frames</a></li>
<li><a href="package-summary.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h1 title="Package" class="title">Package&nbsp;org.biojava.nbio.structure.io</h1>
<div class="docSummary">
<div class="block">Input and Output of Structures</div>
</div>
<p>See:&nbsp;<a href="#package.description">Description</a></p>
</div>
<div class="contentContainer">
<ul class="blockList">
<li class="blockList">
<table class="typeSummary" border="0" cellpadding="3" cellspacing="0" summary="Interface Summary table, listing interfaces, and an explanation">
<caption><span>Interface Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/StructureIOFile.html" title="interface in org.biojava.nbio.structure.io">StructureIOFile</a></td>
<td class="colLast">
<div class="block">StructureIOFile extends StructureProvider with methods specific to
  parsing files from the filesystem.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/StructureProvider.html" title="interface in org.biojava.nbio.structure.io">StructureProvider</a></td>
<td class="colLast">
<div class="block">A class that can provide a protein structure object from somewhere.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<table class="typeSummary" border="0" cellpadding="3" cellspacing="0" summary="Class Summary table, listing classes, and an explanation">
<caption><span>Class Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/BondMaker.html" title="class in org.biojava.nbio.structure.io">BondMaker</a></td>
<td class="colLast">
<div class="block">Adds polymer bonds for peptides and nucleotides based on distance cutoffs and
 intra-group (residue) bonds based on data from the Chemical Component Dictionary
 to the Structure object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/CAConverter.html" title="class in org.biojava.nbio.structure.io">CAConverter</a></td>
<td class="colLast">
<div class="block">Converts full atom representations to Calpha only ones.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/ChargeAdder.html" title="class in org.biojava.nbio.structure.io">ChargeAdder</a></td>
<td class="colLast">
<div class="block">A class to add appropriate charge information to a structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/EntityFinder.html" title="class in org.biojava.nbio.structure.io">EntityFinder</a></td>
<td class="colLast">
<div class="block">Heuristical finding of Entities (called Compounds in legacy PDB format)
 in a given Structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html" title="class in org.biojava.nbio.structure.io">FastaAFPChainConverter</a></td>
<td class="colLast">
<div class="block">A collection of static utilities to convert between <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model"><code>AFPChains</code></a> and <code>FastaSequences</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html" title="class in org.biojava.nbio.structure.io">FastaStructureParser</a></td>
<td class="colLast">
<div class="block">Reads a protein sequence from a fasta file and attempts to match it to a
 3D structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/FileConvert.html" title="class in org.biojava.nbio.structure.io">FileConvert</a></td>
<td class="colLast">
<div class="block">Methods to convert a structure object into different file formats.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/FileParsingParameters.html" title="class in org.biojava.nbio.structure.io">FileParsingParameters</a></td>
<td class="colLast">
<div class="block">A class that configures parameters that can be sent to the PDB file parsers

 
  <a href="../../../../../org/biojava/nbio/structure/io/FileParsingParameters.html#setParseCAOnly-boolean-"><code>FileParsingParameters.setParseCAOnly(boolean)</code></a> - parse only the Atom records for C-alpha atoms
  <a href="../../../../../org/biojava/nbio/structure/io/FileParsingParameters.html#setParseSecStruc-boolean-"><code>FileParsingParameters.setParseSecStruc(boolean)</code></a> - a flag if the secondary structure information from the PDB file (author's assignment) should be parsed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/GroupToSDF.html" title="class in org.biojava.nbio.structure.io">GroupToSDF</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/LocalPDBDirectory.html" title="class in org.biojava.nbio.structure.io">LocalPDBDirectory</a></td>
<td class="colLast">
<div class="block">Superclass for classes which download and interact with the PDB's FTP server,
 specifically <a href="../../../../../org/biojava/nbio/structure/io/PDBFileReader.html" title="class in org.biojava.nbio.structure.io"><code>PDBFileReader</code></a> and <a href="../../../../../org/biojava/nbio/structure/io/MMCIFFileReader.html" title="class in org.biojava.nbio.structure.io"><code>MMCIFFileReader</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/MMCIFFileReader.html" title="class in org.biojava.nbio.structure.io">MMCIFFileReader</a></td>
<td class="colLast">
<div class="block">How to parse an mmCif file:
 
public static void main(String[] args) throws Exception {
        String filename =  "/path/to/something.cif.gz" ;

        StructureIOFile reader = new MMCIFFileReader();

        Structure struc = reader.getStructure(filename);
        System.out.println(struc);
}
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/MMTFFileReader.html" title="class in org.biojava.nbio.structure.io">MMTFFileReader</a></td>
<td class="colLast">
<div class="block">A class to read MMTF files and cache them locally.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/PDBBioAssemblyParser.html" title="class in org.biojava.nbio.structure.io">PDBBioAssemblyParser</a></td>
<td class="colLast">
<div class="block">Parses REMARK 350 records in a PDB file and creates transformations to
 construct the quaternary structure of a protein from an asymmetric unit</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/PDBFileParser.html" title="class in org.biojava.nbio.structure.io">PDBFileParser</a></td>
<td class="colLast">
<div class="block">This class implements the actual PDB file parsing.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/PDBFileReader.html" title="class in org.biojava.nbio.structure.io">PDBFileReader</a></td>
<td class="colLast">
<div class="block">
  The wrapper class for parsing a PDB file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/SandboxStyleStructureProvider.html" title="class in org.biojava.nbio.structure.io">SandboxStyleStructureProvider</a></td>
<td class="colLast">
<div class="block">The "Sandbox" style of organizing files is  to have a directory structure like below, i.e. the files are organized into
 
        directory with two characters, based on the two middle characters of a PDB ID
        directory of PDB ID
        several files that are available for this PDB ID
 

 
a1/2a1v/2a1v.cif.gz
a1/2a1v/2a1v.dssp.gz
a1/2a1v/2a1v.pdb-250.jpg.gz
a1/2a1v/2a1v.pdb-500.jpg.gz
a1/2a1v/2a1v.pdb-65.jpg.gz
a1/2a1v/2a1v.pdb-80.jpg.gz
a1/2a1v/2a1v.pdb1-250.jpg.gz
a1/2a1v/2a1v.pdb1-500.jpg.gz
a1/2a1v/2a1v.pdb1-65.jpg.gz
a1/2a1v/2a1v.pdb1-80.jpg.gz
a1/2a1v/2a1v.pdb1.gz
a1/2a1v/2a1v.stride.gz
a1/2a1v/2a1v.xml.gz
a1/2a1v/pdb2a1v.ent.gz
a1/2a1v/r2a1vsf.ent.gz
a1/2a1w/2a1w-deriv.cif.gz
a1/2a1w/2a1w-extatom.xml.gz
a1/2a1w/2a1w-noatom.xml.gz
a1/2a1w/2a1w.cif.gz
a1/2a1w/2a1w.dssp.gz
a1/2a1w/2a1w.pdb-250.jpg.gz
a1/2a1w/2a1w.pdb-500.jpg.gz
a1/2a1w/2a1w.pdb-65.jpg.gz
a1/2a1w/2a1w.pdb-80.jpg.gz
a1/2a1w/2a1w.pdb1-250.jpg.gz
a1/2a1w/2a1w.pdb1-500.jpg.gz
a1/2a1w/2a1w.pdb1-65.jpg.gz
a1/2a1w/2a1w.pdb1-80.jpg.gz
a1/2a1w/2a1w.pdb1.gz
a1/2a1w/2a1w.pdb2-250.jpg.gz
a1/2a1w/2a1w.pdb2-500.jpg.gz
a1/2a1w/2a1w.pdb2-65.jpg.gz
a1/2a1w/2a1w.pdb2-80.jpg.gz
a1/2a1w/2a1w.pdb2.gz
a1/2a1w/2a1w.pdb3-250.jpg.gz
a1/2a1w/2a1w.pdb3-500.jpg.gz
a1/2a1w/2a1w.pdb3-65.jpg.gz
a1/2a1w/2a1w.pdb3-80.jpg.gz
a1/2a1w/2a1w.pdb3.gz
a1/2a1w/2a1w.pdb4-250.jpg.gz
a1/2a1w/2a1w.pdb4-500.jpg.gz
a1/2a1w/2a1w.pdb4-65.jpg.gz
a1/2a1w/2a1w.pdb4-80.jpg.gz
a1/2a1w/2a1w.pdb4.gz
a1/2a1w/2a1w.pdb5-250.jpg.gz
a1/2a1w/2a1w.pdb5-500.jpg.gz
a1/2a1w/2a1w.pdb5-65.jpg.gz
a1/2a1w/2a1w.pdb5-80.jpg.gz
a1/2a1w/2a1w.pdb5.gz
a1/2a1w/2a1w.pdb6-250.jpg.gz
a1/2a1w/2a1w.pdb6-500.jpg.gz
a1/2a1w/2a1w.pdb6-65.jpg.gz
a1/2a1w/2a1w.pdb6-80.jpg.gz
a1/2a1w/2a1w.pdb6.gz
a1/2a1w/2a1w.stride.gz
a1/2a1w/2a1w.xml.gz
a1/2a1w/pdb2a1w.ent.gz
a1/2a1w/r2a1wsf.ent.gz
a1/2a1x/2a1x-deriv.cif.gz
a1/2a1x/2a1x-extatom.xml.gz
a1/2a1x/2a1x-noatom.xml.gz
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/SeqRes2AtomAligner.html" title="class in org.biojava.nbio.structure.io">SeqRes2AtomAligner</a></td>
<td class="colLast">
<div class="block">Aligns the SEQRES residues to the ATOM residues.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/SSBondImpl.html" title="class in org.biojava.nbio.structure.io">SSBondImpl</a></td>
<td class="colLast">
<div class="block">A simple bean to store disulfide bridge information, the SSBOND records in the PDB files.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/StructureSequenceMatcher.html" title="class in org.biojava.nbio.structure.io">StructureSequenceMatcher</a></td>
<td class="colLast">
<div class="block">A utility class with methods for matching ProteinSequences with
 Structures.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<table class="typeSummary" border="0" cellpadding="3" cellspacing="0" summary="Enum Summary table, listing enums, and an explanation">
<caption><span>Enum Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Enum</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/LocalPDBDirectory.FetchBehavior.html" title="enum in org.biojava.nbio.structure.io">LocalPDBDirectory.FetchBehavior</a></td>
<td class="colLast">
<div class="block">Controls when the class should fetch files from the ftp server</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/LocalPDBDirectory.ObsoleteBehavior.html" title="enum in org.biojava.nbio.structure.io">LocalPDBDirectory.ObsoleteBehavior</a></td>
<td class="colLast">
<div class="block">Behaviors for when an obsolete structure is requested.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<table class="typeSummary" border="0" cellpadding="3" cellspacing="0" summary="Exception Summary table, listing exceptions, and an explanation">
<caption><span>Exception Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Exception</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="../../../../../org/biojava/nbio/structure/io/PDBParseException.html" title="class in org.biojava.nbio.structure.io">PDBParseException</a></td>
<td class="colLast">
<div class="block">An exception during the parsing of a PDB file.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
<a name="package.description">
<!--   -->
</a>
<h2 title="Package org.biojava.nbio.structure.io Description">Package org.biojava.nbio.structure.io Description</h2>
<div class="block">Input and Output of Structures</div>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li class="navBarCell1Rev">Package</li>
<li>Class</li>
<li><a href="package-use.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/nbio/structure/gui/util/color/package-summary.html">Prev&nbsp;Package</a></li>
<li><a href="../../../../../org/biojava/nbio/structure/io/mmcif/package-summary.html">Next&nbsp;Package</a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/structure/io/package-summary.html" target="_top">Frames</a></li>
<li><a href="package-summary.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2018 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
