<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> */</a>
<span class="sourceLineNo">021</span><a id="line.21">package org.biojava.nbio.genome;</a>
<span class="sourceLineNo">022</span><a id="line.22"></a>
<span class="sourceLineNo">023</span><a id="line.23">import org.biojava.nbio.genome.parsers.gff.*;</a>
<span class="sourceLineNo">024</span><a id="line.24">import org.biojava.nbio.core.sequence.*;</a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.nbio.core.sequence.io.FastaReaderHelper;</a>
<span class="sourceLineNo">026</span><a id="line.26">import org.slf4j.Logger;</a>
<span class="sourceLineNo">027</span><a id="line.27">import org.slf4j.LoggerFactory;</a>
<span class="sourceLineNo">028</span><a id="line.28"></a>
<span class="sourceLineNo">029</span><a id="line.29">import java.io.File;</a>
<span class="sourceLineNo">030</span><a id="line.30">import java.io.FileWriter;</a>
<span class="sourceLineNo">031</span><a id="line.31">import java.util.ArrayList;</a>
<span class="sourceLineNo">032</span><a id="line.32">import java.util.Collection;</a>
<span class="sourceLineNo">033</span><a id="line.33">import java.util.Collections;</a>
<span class="sourceLineNo">034</span><a id="line.34">import java.util.LinkedHashMap;</a>
<span class="sourceLineNo">035</span><a id="line.35"></a>
<span class="sourceLineNo">036</span><a id="line.36">/**</a>
<span class="sourceLineNo">037</span><a id="line.37"> *</a>
<span class="sourceLineNo">038</span><a id="line.38"> * @author Scooter Willis &lt;willishf at gmail dot com&gt;</a>
<span class="sourceLineNo">039</span><a id="line.39"> */</a>
<span class="sourceLineNo">040</span><a id="line.40">public class GeneFeatureHelper {</a>
<span class="sourceLineNo">041</span><a id="line.41"></a>
<span class="sourceLineNo">042</span><a id="line.42">        private static final Logger logger = LoggerFactory.getLogger(GeneFeatureHelper.class);</a>
<span class="sourceLineNo">043</span><a id="line.43"></a>
<span class="sourceLineNo">044</span><a id="line.44">        static public LinkedHashMap&lt;String, ChromosomeSequence&gt; loadFastaAddGeneFeaturesFromUpperCaseExonFastaFile(File fastaSequenceFile, File uppercaseFastaFile, boolean throwExceptionGeneNotFound) throws Exception {</a>
<span class="sourceLineNo">045</span><a id="line.45">                LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = new LinkedHashMap&lt;String, ChromosomeSequence&gt;();</a>
<span class="sourceLineNo">046</span><a id="line.46">                LinkedHashMap&lt;String, DNASequence&gt; dnaSequenceList = FastaReaderHelper.readFastaDNASequence(fastaSequenceFile);</a>
<span class="sourceLineNo">047</span><a id="line.47">                for (String accession : dnaSequenceList.keySet()) {</a>
<span class="sourceLineNo">048</span><a id="line.48">                        DNASequence contigSequence = dnaSequenceList.get(accession);</a>
<span class="sourceLineNo">049</span><a id="line.49">                        ChromosomeSequence chromsomeSequence = new ChromosomeSequence(contigSequence.getSequenceAsString());</a>
<span class="sourceLineNo">050</span><a id="line.50">                        chromsomeSequence.setAccession(contigSequence.getAccession());</a>
<span class="sourceLineNo">051</span><a id="line.51">                        chromosomeSequenceList.put(accession, chromsomeSequence);</a>
<span class="sourceLineNo">052</span><a id="line.52">                }</a>
<span class="sourceLineNo">053</span><a id="line.53"></a>
<span class="sourceLineNo">054</span><a id="line.54"></a>
<span class="sourceLineNo">055</span><a id="line.55">                LinkedHashMap&lt;String, DNASequence&gt; geneSequenceList = FastaReaderHelper.readFastaDNASequence(uppercaseFastaFile);</a>
<span class="sourceLineNo">056</span><a id="line.56">                for (DNASequence dnaSequence : geneSequenceList.values()) {</a>
<span class="sourceLineNo">057</span><a id="line.57">                        String geneSequence = dnaSequence.getSequenceAsString();</a>
<span class="sourceLineNo">058</span><a id="line.58">                        String lcGeneSequence = geneSequence.toLowerCase();</a>
<span class="sourceLineNo">059</span><a id="line.59">                        String reverseGeneSequence = dnaSequence.getReverse().getSequenceAsString();</a>
<span class="sourceLineNo">060</span><a id="line.60">                        String lcReverseGeneSequence = reverseGeneSequence.toLowerCase();</a>
<span class="sourceLineNo">061</span><a id="line.61">                        Integer bioStart = null;</a>
<span class="sourceLineNo">062</span><a id="line.62">                        Integer bioEnd = null;</a>
<span class="sourceLineNo">063</span><a id="line.63">                        Strand strand = Strand.POSITIVE;</a>
<span class="sourceLineNo">064</span><a id="line.64">                        boolean geneFound = false;</a>
<span class="sourceLineNo">065</span><a id="line.65">                        String accession = "";</a>
<span class="sourceLineNo">066</span><a id="line.66">                        DNASequence contigDNASequence = null;</a>
<span class="sourceLineNo">067</span><a id="line.67">                        for (String id : dnaSequenceList.keySet()) {</a>
<span class="sourceLineNo">068</span><a id="line.68">                                accession = id;</a>
<span class="sourceLineNo">069</span><a id="line.69">                                contigDNASequence = dnaSequenceList.get(id);</a>
<span class="sourceLineNo">070</span><a id="line.70">                                String contigSequence = contigDNASequence.getSequenceAsString().toLowerCase();</a>
<span class="sourceLineNo">071</span><a id="line.71">                                bioStart = contigSequence.indexOf(lcGeneSequence);</a>
<span class="sourceLineNo">072</span><a id="line.72">                                if (bioStart != -1) {</a>
<span class="sourceLineNo">073</span><a id="line.73">                                        bioStart = bioStart + 1;</a>
<span class="sourceLineNo">074</span><a id="line.74">                                        bioEnd = bioStart + geneSequence.length() - 1;</a>
<span class="sourceLineNo">075</span><a id="line.75">                                        geneFound = true;</a>
<span class="sourceLineNo">076</span><a id="line.76">                                        break;</a>
<span class="sourceLineNo">077</span><a id="line.77">                                } else {</a>
<span class="sourceLineNo">078</span><a id="line.78">                                        bioStart = contigSequence.indexOf(lcReverseGeneSequence);</a>
<span class="sourceLineNo">079</span><a id="line.79">                                        if (bioStart != -1) {</a>
<span class="sourceLineNo">080</span><a id="line.80">                                                bioStart = bioStart + 1;</a>
<span class="sourceLineNo">081</span><a id="line.81">                                                bioEnd = bioStart - geneSequence.length() - 1;</a>
<span class="sourceLineNo">082</span><a id="line.82">                                                strand = Strand.NEGATIVE;</a>
<span class="sourceLineNo">083</span><a id="line.83">                                                geneFound = true;</a>
<span class="sourceLineNo">084</span><a id="line.84">                                                break;</a>
<span class="sourceLineNo">085</span><a id="line.85">                                        }</a>
<span class="sourceLineNo">086</span><a id="line.86">                                }</a>
<span class="sourceLineNo">087</span><a id="line.87">                        }</a>
<span class="sourceLineNo">088</span><a id="line.88"></a>
<span class="sourceLineNo">089</span><a id="line.89">                        if (geneFound) {</a>
<span class="sourceLineNo">090</span><a id="line.90">                                logger.info("Gene {} found at {} {} {} {}",</a>
<span class="sourceLineNo">091</span><a id="line.91">                                                dnaSequence.getAccession().toString(), contigDNASequence.getAccession().toString(), bioStart, bioEnd, strand);</a>
<span class="sourceLineNo">092</span><a id="line.92">                                ChromosomeSequence chromosomeSequence = chromosomeSequenceList.get(accession);</a>
<span class="sourceLineNo">093</span><a id="line.93"></a>
<span class="sourceLineNo">094</span><a id="line.94">                                ArrayList&lt;Integer&gt; exonBoundries = new ArrayList&lt;Integer&gt;();</a>
<span class="sourceLineNo">095</span><a id="line.95"></a>
<span class="sourceLineNo">096</span><a id="line.96">                                //look for transitions from lowercase to upper case</a>
<span class="sourceLineNo">097</span><a id="line.97">                                for (int i = 0; i &lt; geneSequence.length(); i++) {</a>
<span class="sourceLineNo">098</span><a id="line.98">                                        if (i == 0 &amp;&amp; Character.isUpperCase(geneSequence.charAt(i))) {</a>
<span class="sourceLineNo">099</span><a id="line.99">                                                exonBoundries.add(i);</a>
<span class="sourceLineNo">100</span><a id="line.100">                                        } else if (i == geneSequence.length() - 1) {</a>
<span class="sourceLineNo">101</span><a id="line.101">                                                exonBoundries.add(i);</a>
<span class="sourceLineNo">102</span><a id="line.102">                                        } else if (Character.isUpperCase(geneSequence.charAt(i)) &amp;&amp; Character.isLowerCase(geneSequence.charAt(i - 1))) {</a>
<span class="sourceLineNo">103</span><a id="line.103">                                                exonBoundries.add(i);</a>
<span class="sourceLineNo">104</span><a id="line.104">                                        } else if (Character.isUpperCase(geneSequence.charAt(i)) &amp;&amp; Character.isLowerCase(geneSequence.charAt(i + 1))) {</a>
<span class="sourceLineNo">105</span><a id="line.105">                                                exonBoundries.add(i);</a>
<span class="sourceLineNo">106</span><a id="line.106">                                        }</a>
<span class="sourceLineNo">107</span><a id="line.107">                                }</a>
<span class="sourceLineNo">108</span><a id="line.108">                                if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">109</span><a id="line.109">                                        Collections.reverse(exonBoundries);</a>
<span class="sourceLineNo">110</span><a id="line.110">                                }</a>
<span class="sourceLineNo">111</span><a id="line.111"></a>
<span class="sourceLineNo">112</span><a id="line.112"></a>
<span class="sourceLineNo">113</span><a id="line.113">                                String geneaccession = dnaSequence.getAccession().getID();</a>
<span class="sourceLineNo">114</span><a id="line.114">                                String note = geneaccession;</a>
<span class="sourceLineNo">115</span><a id="line.115">                                String[] values = geneaccession.split(" ");</a>
<span class="sourceLineNo">116</span><a id="line.116">                                geneaccession = values[0];</a>
<span class="sourceLineNo">117</span><a id="line.117"></a>
<span class="sourceLineNo">118</span><a id="line.118"></a>
<span class="sourceLineNo">119</span><a id="line.119"></a>
<span class="sourceLineNo">120</span><a id="line.120">                                GeneSequence geneSeq = chromosomeSequence.addGene(new AccessionID(geneaccession), bioStart, bioEnd, strand);</a>
<span class="sourceLineNo">121</span><a id="line.121">                                geneSeq.addNote(note);</a>
<span class="sourceLineNo">122</span><a id="line.122">                                geneSeq.setSource(uppercaseFastaFile.getName());</a>
<span class="sourceLineNo">123</span><a id="line.123">                                //String transcriptName = geneaccession + "-transcript";</a>
<span class="sourceLineNo">124</span><a id="line.124">                                //TranscriptSequence transcriptSequence = geneSeq.addTranscript(new AccessionID(transcriptName), bioStart, bioEnd);</a>
<span class="sourceLineNo">125</span><a id="line.125"></a>
<span class="sourceLineNo">126</span><a id="line.126">                                int runningFrameLength = 0;</a>
<span class="sourceLineNo">127</span><a id="line.127">                                for (int i = 0; i &lt; exonBoundries.size() - 1; i = i + 2) {</a>
<span class="sourceLineNo">128</span><a id="line.128">                                        int cdsBioStart = exonBoundries.get(i) + bioStart;</a>
<span class="sourceLineNo">129</span><a id="line.129">                                        int cdsBioEnd = exonBoundries.get(i + 1) + bioStart;</a>
<span class="sourceLineNo">130</span><a id="line.130">                                        runningFrameLength = runningFrameLength + Math.abs(cdsBioEnd - cdsBioStart) + 1;</a>
<span class="sourceLineNo">131</span><a id="line.131">                                        //String cdsName = transcriptName + "-cds-" + cdsBioStart + "-" + cdsBioEnd;</a>
<span class="sourceLineNo">132</span><a id="line.132"></a>
<span class="sourceLineNo">133</span><a id="line.133">                                        //AccessionID cdsAccessionID = new AccessionID(cdsName);</a>
<span class="sourceLineNo">134</span><a id="line.134">                                        //ExonSequence exonSequence = geneSeq.addExon(cdsAccessionID, cdsBioStart, cdsBioEnd);</a>
<span class="sourceLineNo">135</span><a id="line.135">                                        int remainder = runningFrameLength % 3;</a>
<span class="sourceLineNo">136</span><a id="line.136">                                        //int frame = 0;</a>
<span class="sourceLineNo">137</span><a id="line.137">                                        if (remainder == 1) {</a>
<span class="sourceLineNo">138</span><a id="line.138">                                                //frame = 2; // borrow 2 from next CDS region</a>
<span class="sourceLineNo">139</span><a id="line.139">                                        } else if (remainder == 2) {</a>
<span class="sourceLineNo">140</span><a id="line.140">                                                //frame = 1;</a>
<span class="sourceLineNo">141</span><a id="line.141">                                        }</a>
<span class="sourceLineNo">142</span><a id="line.142">                                        //CDSSequence cdsSequence = transcriptSequence.addCDS(cdsAccessionID, cdsBioStart, cdsBioEnd, frame);</a>
<span class="sourceLineNo">143</span><a id="line.143">                                }</a>
<span class="sourceLineNo">144</span><a id="line.144"></a>
<span class="sourceLineNo">145</span><a id="line.145"></a>
<span class="sourceLineNo">146</span><a id="line.146">                        } else {</a>
<span class="sourceLineNo">147</span><a id="line.147">                                if (throwExceptionGeneNotFound) {</a>
<span class="sourceLineNo">148</span><a id="line.148">                                        throw new Exception(dnaSequence.getAccession().toString() + " not found");</a>
<span class="sourceLineNo">149</span><a id="line.149">                                }</a>
<span class="sourceLineNo">150</span><a id="line.150">                                logger.info("Gene not found {}", dnaSequence.getAccession().toString());</a>
<span class="sourceLineNo">151</span><a id="line.151">                        }</a>
<span class="sourceLineNo">152</span><a id="line.152"></a>
<span class="sourceLineNo">153</span><a id="line.153">                }</a>
<span class="sourceLineNo">154</span><a id="line.154">                return chromosomeSequenceList;</a>
<span class="sourceLineNo">155</span><a id="line.155">        }</a>
<span class="sourceLineNo">156</span><a id="line.156"></a>
<span class="sourceLineNo">157</span><a id="line.157">        /**</a>
<span class="sourceLineNo">158</span><a id="line.158">         * Output a gff3 feature file that will give the length of each scaffold/chromosome in the fasta file.</a>
<span class="sourceLineNo">159</span><a id="line.159">         * Used for gbrowse so it knows length.</a>
<span class="sourceLineNo">160</span><a id="line.160">         * @param fastaSequenceFile</a>
<span class="sourceLineNo">161</span><a id="line.161">         * @param gffFile</a>
<span class="sourceLineNo">162</span><a id="line.162">         * @throws Exception</a>
<span class="sourceLineNo">163</span><a id="line.163">         */</a>
<span class="sourceLineNo">164</span><a id="line.164">        static public void outputFastaSequenceLengthGFF3(File fastaSequenceFile, File gffFile) throws Exception {</a>
<span class="sourceLineNo">165</span><a id="line.165">                LinkedHashMap&lt;String, DNASequence&gt; dnaSequenceList = FastaReaderHelper.readFastaDNASequence(fastaSequenceFile);</a>
<span class="sourceLineNo">166</span><a id="line.166">                String fileName = fastaSequenceFile.getName();</a>
<span class="sourceLineNo">167</span><a id="line.167">                FileWriter fw = new FileWriter(gffFile);</a>
<span class="sourceLineNo">168</span><a id="line.168">                String newLine = System.getProperty("line.separator");</a>
<span class="sourceLineNo">169</span><a id="line.169">                fw.write("##gff-version 3" + newLine);</a>
<span class="sourceLineNo">170</span><a id="line.170">                for (DNASequence dnaSequence : dnaSequenceList.values()) {</a>
<span class="sourceLineNo">171</span><a id="line.171">                        String gff3line = dnaSequence.getAccession().getID() + "\t" + fileName + "\t" + "contig" + "\t" + "1" + "\t" + dnaSequence.getBioEnd() + "\t.\t.\t.\tName=" + dnaSequence.getAccession().getID() + newLine;</a>
<span class="sourceLineNo">172</span><a id="line.172">                        fw.write(gff3line);</a>
<span class="sourceLineNo">173</span><a id="line.173">                }</a>
<span class="sourceLineNo">174</span><a id="line.174">                fw.close();</a>
<span class="sourceLineNo">175</span><a id="line.175">        }</a>
<span class="sourceLineNo">176</span><a id="line.176"></a>
<span class="sourceLineNo">177</span><a id="line.177">        /**</a>
<span class="sourceLineNo">178</span><a id="line.178">         * Loads Fasta file and GFF2 feature file generated from the geneid prediction algorithm</a>
<span class="sourceLineNo">179</span><a id="line.179">         *</a>
<span class="sourceLineNo">180</span><a id="line.180">         * @param fastaSequenceFile</a>
<span class="sourceLineNo">181</span><a id="line.181">         * @param gffFile</a>
<span class="sourceLineNo">182</span><a id="line.182">         * @return</a>
<span class="sourceLineNo">183</span><a id="line.183">         * @throws Exception</a>
<span class="sourceLineNo">184</span><a id="line.184">         */</a>
<span class="sourceLineNo">185</span><a id="line.185">        static public LinkedHashMap&lt;String, ChromosomeSequence&gt; loadFastaAddGeneFeaturesFromGeneIDGFF2(File fastaSequenceFile, File gffFile) throws Exception {</a>
<span class="sourceLineNo">186</span><a id="line.186">                LinkedHashMap&lt;String, DNASequence&gt; dnaSequenceList = FastaReaderHelper.readFastaDNASequence(fastaSequenceFile);</a>
<span class="sourceLineNo">187</span><a id="line.187">                LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = GeneFeatureHelper.getChromosomeSequenceFromDNASequence(dnaSequenceList);</a>
<span class="sourceLineNo">188</span><a id="line.188">                FeatureList listGenes = GeneIDGFF2Reader.read(gffFile.getAbsolutePath());</a>
<span class="sourceLineNo">189</span><a id="line.189">                addGeneIDGFF2GeneFeatures(chromosomeSequenceList, listGenes);</a>
<span class="sourceLineNo">190</span><a id="line.190">                return chromosomeSequenceList;</a>
<span class="sourceLineNo">191</span><a id="line.191">        }</a>
<span class="sourceLineNo">192</span><a id="line.192"></a>
<span class="sourceLineNo">193</span><a id="line.193">        /**</a>
<span class="sourceLineNo">194</span><a id="line.194">         * Load GFF2 feature file generated from the geneid prediction algorithm and map features onto the chromosome sequences</a>
<span class="sourceLineNo">195</span><a id="line.195">         *</a>
<span class="sourceLineNo">196</span><a id="line.196">         * @param chromosomeSequenceList</a>
<span class="sourceLineNo">197</span><a id="line.197">         * @param listGenes</a>
<span class="sourceLineNo">198</span><a id="line.198">         * @throws Exception</a>
<span class="sourceLineNo">199</span><a id="line.199">         */</a>
<span class="sourceLineNo">200</span><a id="line.200">        static public void addGeneIDGFF2GeneFeatures(LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList, FeatureList listGenes) throws Exception {</a>
<span class="sourceLineNo">201</span><a id="line.201">                Collection&lt;String&gt; geneIds = listGenes.attributeValues("gene_id");</a>
<span class="sourceLineNo">202</span><a id="line.202">                for (String geneid : geneIds) {</a>
<span class="sourceLineNo">203</span><a id="line.203">                        FeatureList gene = listGenes.selectByAttribute("gene_id", geneid);</a>
<span class="sourceLineNo">204</span><a id="line.204">                        FeatureI geneFeature = gene.get(0);</a>
<span class="sourceLineNo">205</span><a id="line.205">                        ChromosomeSequence seq = chromosomeSequenceList.get(geneFeature.seqname());</a>
<span class="sourceLineNo">206</span><a id="line.206">                        geneid = geneid.replaceAll("_", ".G");</a>
<span class="sourceLineNo">207</span><a id="line.207">                        AccessionID geneAccessionID = new AccessionID(geneid);</a>
<span class="sourceLineNo">208</span><a id="line.208">                        GeneSequence geneSequence = null;</a>
<span class="sourceLineNo">209</span><a id="line.209">                        Collection&lt;String&gt; transcriptids = gene.attributeValues("gene_id");</a>
<span class="sourceLineNo">210</span><a id="line.210">                        for (String transcriptid : transcriptids) {</a>
<span class="sourceLineNo">211</span><a id="line.211">                                // get all the individual features (exons, CDS regions, etc.) of this gene</a>
<span class="sourceLineNo">212</span><a id="line.212">                                FeatureList transcriptFeature = listGenes.selectByAttribute("gene_id", transcriptid);</a>
<span class="sourceLineNo">213</span><a id="line.213">                                transcriptid = transcriptid.replaceAll("_", ".G");</a>
<span class="sourceLineNo">214</span><a id="line.214"></a>
<span class="sourceLineNo">215</span><a id="line.215"></a>
<span class="sourceLineNo">216</span><a id="line.216"></a>
<span class="sourceLineNo">217</span><a id="line.217"></a>
<span class="sourceLineNo">218</span><a id="line.218">                                //      String seqName = feature.seqname();</a>
<span class="sourceLineNo">219</span><a id="line.219">                                //FeatureI startCodon = null;</a>
<span class="sourceLineNo">220</span><a id="line.220">                                //FeatureI stopCodon = null;</a>
<span class="sourceLineNo">221</span><a id="line.221">                                Integer startCodonBegin = null;</a>
<span class="sourceLineNo">222</span><a id="line.222">                                Integer stopCodonEnd = null;</a>
<span class="sourceLineNo">223</span><a id="line.223">                                //String startCodonName = "";</a>
<span class="sourceLineNo">224</span><a id="line.224">                                //String stopCodonName = "";</a>
<span class="sourceLineNo">225</span><a id="line.225"></a>
<span class="sourceLineNo">226</span><a id="line.226"></a>
<span class="sourceLineNo">227</span><a id="line.227">                                // now select only the coding regions of this gene</a>
<span class="sourceLineNo">228</span><a id="line.228">                                FeatureList firstFeatures = transcriptFeature.selectByType("First");</a>
<span class="sourceLineNo">229</span><a id="line.229">                                FeatureList terminalFeatures = transcriptFeature.selectByType("Terminal");</a>
<span class="sourceLineNo">230</span><a id="line.230">                                FeatureList internalFeatures = transcriptFeature.selectByType("Internal");</a>
<span class="sourceLineNo">231</span><a id="line.231">                                FeatureList singleFeatures = transcriptFeature.selectByType("Single");</a>
<span class="sourceLineNo">232</span><a id="line.232">                                FeatureList cdsFeatures = new FeatureList();</a>
<span class="sourceLineNo">233</span><a id="line.233">                                cdsFeatures.add(firstFeatures);</a>
<span class="sourceLineNo">234</span><a id="line.234">                                cdsFeatures.add(terminalFeatures);</a>
<span class="sourceLineNo">235</span><a id="line.235">                                cdsFeatures.add(internalFeatures);</a>
<span class="sourceLineNo">236</span><a id="line.236">                                cdsFeatures.add(singleFeatures);</a>
<span class="sourceLineNo">237</span><a id="line.237">                                // sort them</a>
<span class="sourceLineNo">238</span><a id="line.238">                                cdsFeatures = cdsFeatures.sortByStart();</a>
<span class="sourceLineNo">239</span><a id="line.239">                                Strand strand = Strand.POSITIVE;</a>
<span class="sourceLineNo">240</span><a id="line.240">                                FeatureI feature = cdsFeatures.get(0);</a>
<span class="sourceLineNo">241</span><a id="line.241">                                if (feature.location().isNegative()) {</a>
<span class="sourceLineNo">242</span><a id="line.242">                                        strand = Strand.NEGATIVE;</a>
<span class="sourceLineNo">243</span><a id="line.243">                                }</a>
<span class="sourceLineNo">244</span><a id="line.244">                                if (startCodonBegin == null) {</a>
<span class="sourceLineNo">245</span><a id="line.245">                                        FeatureI firstFeature = cdsFeatures.get(0);</a>
<span class="sourceLineNo">246</span><a id="line.246">                                        if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">247</span><a id="line.247">                                                startCodonBegin = firstFeature.location().bioEnd();</a>
<span class="sourceLineNo">248</span><a id="line.248">                                        } else {</a>
<span class="sourceLineNo">249</span><a id="line.249">                                                startCodonBegin = firstFeature.location().bioStart();</a>
<span class="sourceLineNo">250</span><a id="line.250">                                        }</a>
<span class="sourceLineNo">251</span><a id="line.251">                                }</a>
<span class="sourceLineNo">252</span><a id="line.252"></a>
<span class="sourceLineNo">253</span><a id="line.253">                                if (stopCodonEnd == null) {</a>
<span class="sourceLineNo">254</span><a id="line.254"></a>
<span class="sourceLineNo">255</span><a id="line.255">                                        FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</a>
<span class="sourceLineNo">256</span><a id="line.256">                                        if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">257</span><a id="line.257">                                                stopCodonEnd = lastFeature.location().bioStart();</a>
<span class="sourceLineNo">258</span><a id="line.258">                                        } else {</a>
<span class="sourceLineNo">259</span><a id="line.259">                                                stopCodonEnd = lastFeature.location().bioEnd();</a>
<span class="sourceLineNo">260</span><a id="line.260">                                        }</a>
<span class="sourceLineNo">261</span><a id="line.261">                                }</a>
<span class="sourceLineNo">262</span><a id="line.262">                                //for gtf ordering can be strand based so first is last and last is first</a>
<span class="sourceLineNo">263</span><a id="line.263">                                if (startCodonBegin &gt; stopCodonEnd) {</a>
<span class="sourceLineNo">264</span><a id="line.264">                                        int temp = startCodonBegin;</a>
<span class="sourceLineNo">265</span><a id="line.265">                                        startCodonBegin = stopCodonEnd;</a>
<span class="sourceLineNo">266</span><a id="line.266">                                        stopCodonEnd = temp;</a>
<span class="sourceLineNo">267</span><a id="line.267">                                }</a>
<span class="sourceLineNo">268</span><a id="line.268"></a>
<span class="sourceLineNo">269</span><a id="line.269">                                AccessionID transcriptAccessionID = new AccessionID(transcriptid);</a>
<span class="sourceLineNo">270</span><a id="line.270">                                if (geneSequence == null) {</a>
<span class="sourceLineNo">271</span><a id="line.271">                                        geneSequence = seq.addGene(geneAccessionID, startCodonBegin, stopCodonEnd, strand);</a>
<span class="sourceLineNo">272</span><a id="line.272">                                        geneSequence.setSource(((Feature) feature).source());</a>
<span class="sourceLineNo">273</span><a id="line.273">                                } else {</a>
<span class="sourceLineNo">274</span><a id="line.274">                                        //if multiple transcripts for one gene make sure the gene is defined as the min and max start/end</a>
<span class="sourceLineNo">275</span><a id="line.275"></a>
<span class="sourceLineNo">276</span><a id="line.276">                                        if (startCodonBegin &lt; geneSequence.getBioBegin()) {</a>
<span class="sourceLineNo">277</span><a id="line.277">                                                geneSequence.setBioBegin(startCodonBegin);</a>
<span class="sourceLineNo">278</span><a id="line.278">                                        }</a>
<span class="sourceLineNo">279</span><a id="line.279">                                        if (stopCodonEnd &gt; geneSequence.getBioBegin()) {</a>
<span class="sourceLineNo">280</span><a id="line.280">                                                geneSequence.setBioEnd(stopCodonEnd);</a>
<span class="sourceLineNo">281</span><a id="line.281">                                        }</a>
<span class="sourceLineNo">282</span><a id="line.282"></a>
<span class="sourceLineNo">283</span><a id="line.283">                                }</a>
<span class="sourceLineNo">284</span><a id="line.284">                                TranscriptSequence transcriptSequence = geneSequence.addTranscript(transcriptAccessionID, startCodonBegin, stopCodonEnd);</a>
<span class="sourceLineNo">285</span><a id="line.285">                                /*</a>
<span class="sourceLineNo">286</span><a id="line.286">                                if (startCodon != null) {</a>
<span class="sourceLineNo">287</span><a id="line.287">                                        if (startCodonName == null || startCodonName.length() == 0) {</a>
<span class="sourceLineNo">288</span><a id="line.288">                                                startCodonName = transcriptid + "-start_codon-" + startCodon.location().bioStart() + "-" + startCodon.location().bioEnd();</a>
<span class="sourceLineNo">289</span><a id="line.289">                                        }</a>
<span class="sourceLineNo">290</span><a id="line.290">                                        transcriptSequence.addStartCodonSequence(new AccessionID(startCodonName), startCodon.location().bioStart(), startCodon.location().bioEnd());</a>
<span class="sourceLineNo">291</span><a id="line.291">                                }</a>
<span class="sourceLineNo">292</span><a id="line.292">                                if (stopCodon != null) {</a>
<span class="sourceLineNo">293</span><a id="line.293">                                        if (stopCodonName == null || stopCodonName.length() == 0) {</a>
<span class="sourceLineNo">294</span><a id="line.294">                                                stopCodonName = transcriptid + "-stop_codon-" + stopCodon.location().bioStart() + "-" + stopCodon.location().bioEnd();</a>
<span class="sourceLineNo">295</span><a id="line.295">                                        }</a>
<span class="sourceLineNo">296</span><a id="line.296">                                        transcriptSequence.addStopCodonSequence(new AccessionID(stopCodonName), stopCodon.location().bioStart(), stopCodon.location().bioEnd());</a>
<span class="sourceLineNo">297</span><a id="line.297">                                }</a>
<span class="sourceLineNo">298</span><a id="line.298">                                */</a>
<span class="sourceLineNo">299</span><a id="line.299"></a>
<span class="sourceLineNo">300</span><a id="line.300">                                for (FeatureI cdsFeature : cdsFeatures) {</a>
<span class="sourceLineNo">301</span><a id="line.301">                                        Feature cds = (Feature) cdsFeature;</a>
<span class="sourceLineNo">302</span><a id="line.302">                                        String cdsName = cds.getAttribute("transcript_name");</a>
<span class="sourceLineNo">303</span><a id="line.303">                                        if (cdsName == null || cdsName.length() == 0) {</a>
<span class="sourceLineNo">304</span><a id="line.304">                                                cdsName = transcriptid + "-cds-" + cds.location().bioStart() + "-" + cds.location().bioEnd();</a>
<span class="sourceLineNo">305</span><a id="line.305">                                        }</a>
<span class="sourceLineNo">306</span><a id="line.306">                                        AccessionID cdsAccessionID = new AccessionID(cdsName);</a>
<span class="sourceLineNo">307</span><a id="line.307">                                        //ExonSequence exonSequence = geneSequence.addExon(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd());</a>
<span class="sourceLineNo">308</span><a id="line.308">                                        CDSSequence cdsSequence = transcriptSequence.addCDS(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd(), cds.frame());</a>
<span class="sourceLineNo">309</span><a id="line.309">                                        cdsSequence.setSequenceScore(cds.score());</a>
<span class="sourceLineNo">310</span><a id="line.310">                                }</a>
<span class="sourceLineNo">311</span><a id="line.311">                        }</a>
<span class="sourceLineNo">312</span><a id="line.312">                }</a>
<span class="sourceLineNo">313</span><a id="line.313"></a>
<span class="sourceLineNo">314</span><a id="line.314">        }</a>
<span class="sourceLineNo">315</span><a id="line.315"></a>
<span class="sourceLineNo">316</span><a id="line.316">        static public LinkedHashMap&lt;String, ChromosomeSequence&gt; getChromosomeSequenceFromDNASequence(LinkedHashMap&lt;String, DNASequence&gt; dnaSequenceList) {</a>
<span class="sourceLineNo">317</span><a id="line.317">                LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = new LinkedHashMap&lt;String, ChromosomeSequence&gt;();</a>
<span class="sourceLineNo">318</span><a id="line.318">                for (String key : dnaSequenceList.keySet()) {</a>
<span class="sourceLineNo">319</span><a id="line.319">                        DNASequence dnaSequence = dnaSequenceList.get(key);</a>
<span class="sourceLineNo">320</span><a id="line.320">                        ChromosomeSequence chromosomeSequence = new ChromosomeSequence(dnaSequence.getProxySequenceReader()); //we want the underlying sequence but don't need storage</a>
<span class="sourceLineNo">321</span><a id="line.321">                        chromosomeSequence.setAccession(dnaSequence.getAccession());</a>
<span class="sourceLineNo">322</span><a id="line.322">                        chromosomeSequenceList.put(key, chromosomeSequence);</a>
<span class="sourceLineNo">323</span><a id="line.323">                }</a>
<span class="sourceLineNo">324</span><a id="line.324">                return chromosomeSequenceList;</a>
<span class="sourceLineNo">325</span><a id="line.325">        }</a>
<span class="sourceLineNo">326</span><a id="line.326"></a>
<span class="sourceLineNo">327</span><a id="line.327">        /**</a>
<span class="sourceLineNo">328</span><a id="line.328">         * Lots of variations in the ontology or descriptors that can be used in GFF3 which requires writing a custom parser to handle a GFF3 generated or used</a>
<span class="sourceLineNo">329</span><a id="line.329">         * by a specific application. Probably could be abstracted out but for now easier to handle with custom code to deal with gff3 elements that are not</a>
<span class="sourceLineNo">330</span><a id="line.330">         * included but can be extracted from other data elements.</a>
<span class="sourceLineNo">331</span><a id="line.331">         * @param fastaSequenceFile</a>
<span class="sourceLineNo">332</span><a id="line.332">         * @param gffFile</a>
<span class="sourceLineNo">333</span><a id="line.333">         * @param lazyloadsequences If set to true then the fasta file will be parsed for accession id but sequences will be read from disk when needed to save memory</a>
<span class="sourceLineNo">334</span><a id="line.334">         * @return</a>
<span class="sourceLineNo">335</span><a id="line.335">         * @throws Exception</a>
<span class="sourceLineNo">336</span><a id="line.336">         */</a>
<span class="sourceLineNo">337</span><a id="line.337">        static public LinkedHashMap&lt;String, ChromosomeSequence&gt; loadFastaAddGeneFeaturesFromGmodGFF3(File fastaSequenceFile, File gffFile,boolean lazyloadsequences) throws Exception {</a>
<span class="sourceLineNo">338</span><a id="line.338">                LinkedHashMap&lt;String, DNASequence&gt; dnaSequenceList = FastaReaderHelper.readFastaDNASequence(fastaSequenceFile,lazyloadsequences);</a>
<span class="sourceLineNo">339</span><a id="line.339">                LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = GeneFeatureHelper.getChromosomeSequenceFromDNASequence(dnaSequenceList);</a>
<span class="sourceLineNo">340</span><a id="line.340">                FeatureList listGenes = GFF3Reader.read(gffFile.getAbsolutePath());</a>
<span class="sourceLineNo">341</span><a id="line.341">                addGmodGFF3GeneFeatures(chromosomeSequenceList, listGenes);</a>
<span class="sourceLineNo">342</span><a id="line.342">                return chromosomeSequenceList;</a>
<span class="sourceLineNo">343</span><a id="line.343">        }</a>
<span class="sourceLineNo">344</span><a id="line.344"></a>
<span class="sourceLineNo">345</span><a id="line.345">        /**</a>
<span class="sourceLineNo">346</span><a id="line.346">         * Load GFF3 file using mRNA as the gene feature as not all GFF3 files are complete</a>
<span class="sourceLineNo">347</span><a id="line.347">         * @param chromosomeSequenceList</a>
<span class="sourceLineNo">348</span><a id="line.348">         * @param listGenes</a>
<span class="sourceLineNo">349</span><a id="line.349">         * @throws Exception</a>
<span class="sourceLineNo">350</span><a id="line.350">         */</a>
<span class="sourceLineNo">351</span><a id="line.351">        static public void addGmodGFF3GeneFeatures(LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList, FeatureList listGenes) throws Exception {</a>
<span class="sourceLineNo">352</span><a id="line.352"></a>
<span class="sourceLineNo">353</span><a id="line.353"></a>
<span class="sourceLineNo">354</span><a id="line.354">                // key off mRNA as being a known feature that may or may not have a parent gene</a>
<span class="sourceLineNo">355</span><a id="line.355"></a>
<span class="sourceLineNo">356</span><a id="line.356"></a>
<span class="sourceLineNo">357</span><a id="line.357">                FeatureList mRNAFeatures = listGenes.selectByType("mRNA");</a>
<span class="sourceLineNo">358</span><a id="line.358">                LinkedHashMap&lt;String,FeatureList&gt; featureIDHashMap = FeatureHelper.buildFeatureAtrributeIndex("ID", listGenes);</a>
<span class="sourceLineNo">359</span><a id="line.359">                LinkedHashMap&lt;String,FeatureList&gt; featureParentHashMap = FeatureHelper.buildFeatureAtrributeIndex("Parent", listGenes);</a>
<span class="sourceLineNo">360</span><a id="line.360"></a>
<span class="sourceLineNo">361</span><a id="line.361">                for (FeatureI f : mRNAFeatures) {</a>
<span class="sourceLineNo">362</span><a id="line.362">                        String geneID;</a>
<span class="sourceLineNo">363</span><a id="line.363">                        String geneNote = null;</a>
<span class="sourceLineNo">364</span><a id="line.364">                        String geneSource = null;</a>
<span class="sourceLineNo">365</span><a id="line.365">                        String sequenceName = null;</a>
<span class="sourceLineNo">366</span><a id="line.366">                        ChromosomeSequence seq = null;</a>
<span class="sourceLineNo">367</span><a id="line.367">                        GeneSequence geneSequence = null;</a>
<span class="sourceLineNo">368</span><a id="line.368"></a>
<span class="sourceLineNo">369</span><a id="line.369">                        Feature mRNAFeature = (Feature) f;</a>
<span class="sourceLineNo">370</span><a id="line.370">                        String mRNAID = mRNAFeature.getAttribute("ID");</a>
<span class="sourceLineNo">371</span><a id="line.371">                        String mRNAsource = mRNAFeature.source();</a>
<span class="sourceLineNo">372</span><a id="line.372">                        String mRNANote = mRNAFeature.getAttribute("Note");</a>
<span class="sourceLineNo">373</span><a id="line.373">                        String mRNAParent = mRNAFeature.getAttribute("Parent");</a>
<span class="sourceLineNo">374</span><a id="line.374">                        if (mRNAParent != null &amp;&amp; mRNAParent.length() &gt; 0) {</a>
<span class="sourceLineNo">375</span><a id="line.375">                           // FeatureList geneFeatureList = listGenes.selectByAttribute("ID", mRNAParent);</a>
<span class="sourceLineNo">376</span><a id="line.376">                                FeatureList geneFeatureList = featureIDHashMap.get(mRNAParent);</a>
<span class="sourceLineNo">377</span><a id="line.377">                                Feature geneFeature = (Feature) geneFeatureList.get(0);</a>
<span class="sourceLineNo">378</span><a id="line.378">                                geneID = geneFeature.getAttribute("ID");</a>
<span class="sourceLineNo">379</span><a id="line.379">                                geneNote = geneFeature.getAttribute("Note");</a>
<span class="sourceLineNo">380</span><a id="line.380">                                geneSource = geneFeature.source();</a>
<span class="sourceLineNo">381</span><a id="line.381">                                sequenceName = geneFeature.seqname();</a>
<span class="sourceLineNo">382</span><a id="line.382"></a>
<span class="sourceLineNo">383</span><a id="line.383">                                //</a>
<span class="sourceLineNo">384</span><a id="line.384">                        } else {</a>
<span class="sourceLineNo">385</span><a id="line.385">                                //deal with cases where no parent gene is given</a>
<span class="sourceLineNo">386</span><a id="line.386">                                geneID = mRNAID;</a>
<span class="sourceLineNo">387</span><a id="line.387">                                geneSource = mRNAsource;</a>
<span class="sourceLineNo">388</span><a id="line.388">                                sequenceName = mRNAFeature.seqname();</a>
<span class="sourceLineNo">389</span><a id="line.389">                        }</a>
<span class="sourceLineNo">390</span><a id="line.390"></a>
<span class="sourceLineNo">391</span><a id="line.391">                        seq = chromosomeSequenceList.get(sequenceName);</a>
<span class="sourceLineNo">392</span><a id="line.392"></a>
<span class="sourceLineNo">393</span><a id="line.393">                        AccessionID geneAccessionID = new AccessionID(geneID);</a>
<span class="sourceLineNo">394</span><a id="line.394"></a>
<span class="sourceLineNo">395</span><a id="line.395">                  //  FeatureList mRNAChildren = listGenes.selectByAttribute("Parent", mRNAID);</a>
<span class="sourceLineNo">396</span><a id="line.396">                        FeatureList mRNAChildren = featureParentHashMap.get(mRNAID);</a>
<span class="sourceLineNo">397</span><a id="line.397">                        FeatureList cdsFeatures = mRNAChildren.selectByType("CDS");</a>
<span class="sourceLineNo">398</span><a id="line.398">                        FeatureI feature = cdsFeatures.get(0);</a>
<span class="sourceLineNo">399</span><a id="line.399">                        Strand strand = Strand.POSITIVE;</a>
<span class="sourceLineNo">400</span><a id="line.400"></a>
<span class="sourceLineNo">401</span><a id="line.401">                        if (feature.location().isNegative()) {</a>
<span class="sourceLineNo">402</span><a id="line.402">                                strand = Strand.NEGATIVE;</a>
<span class="sourceLineNo">403</span><a id="line.403">                        }</a>
<span class="sourceLineNo">404</span><a id="line.404">                        cdsFeatures = cdsFeatures.sortByStart();</a>
<span class="sourceLineNo">405</span><a id="line.405"></a>
<span class="sourceLineNo">406</span><a id="line.406"></a>
<span class="sourceLineNo">407</span><a id="line.407"></a>
<span class="sourceLineNo">408</span><a id="line.408"></a>
<span class="sourceLineNo">409</span><a id="line.409"></a>
<span class="sourceLineNo">410</span><a id="line.410"></a>
<span class="sourceLineNo">411</span><a id="line.411"></a>
<span class="sourceLineNo">412</span><a id="line.412">                        //String seqName = feature.seqname();</a>
<span class="sourceLineNo">413</span><a id="line.413">                        FeatureI startCodon = null;</a>
<span class="sourceLineNo">414</span><a id="line.414">                        FeatureI stopCodon = null;</a>
<span class="sourceLineNo">415</span><a id="line.415">                        Integer startCodonBegin = null;</a>
<span class="sourceLineNo">416</span><a id="line.416">                        Integer stopCodonEnd = null;</a>
<span class="sourceLineNo">417</span><a id="line.417">                        String startCodonName = "";</a>
<span class="sourceLineNo">418</span><a id="line.418">                        String stopCodonName = "";</a>
<span class="sourceLineNo">419</span><a id="line.419">                        FeatureList startCodonList = mRNAChildren.selectByType("five_prime_UTR");</a>
<span class="sourceLineNo">420</span><a id="line.420">                        if (startCodonList != null &amp;&amp; startCodonList.size() &gt; 0) {</a>
<span class="sourceLineNo">421</span><a id="line.421">                                startCodon = startCodonList.get(0);</a>
<span class="sourceLineNo">422</span><a id="line.422">                                if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">423</span><a id="line.423">                                        startCodonBegin = startCodon.location().bioEnd();</a>
<span class="sourceLineNo">424</span><a id="line.424">                                } else {</a>
<span class="sourceLineNo">425</span><a id="line.425">                                        startCodonBegin = startCodon.location().bioStart();</a>
<span class="sourceLineNo">426</span><a id="line.426">                                }</a>
<span class="sourceLineNo">427</span><a id="line.427">                                startCodonName = startCodon.getAttribute("ID");</a>
<span class="sourceLineNo">428</span><a id="line.428">                        }</a>
<span class="sourceLineNo">429</span><a id="line.429"></a>
<span class="sourceLineNo">430</span><a id="line.430">                        FeatureList stopCodonList = mRNAChildren.selectByType("three_prime_UTR");</a>
<span class="sourceLineNo">431</span><a id="line.431"></a>
<span class="sourceLineNo">432</span><a id="line.432">                        if (stopCodonList != null &amp;&amp; stopCodonList.size() &gt; 0) {</a>
<span class="sourceLineNo">433</span><a id="line.433">                                stopCodon = stopCodonList.get(0);</a>
<span class="sourceLineNo">434</span><a id="line.434">                                if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">435</span><a id="line.435">                                        stopCodonEnd = stopCodon.location().bioStart();</a>
<span class="sourceLineNo">436</span><a id="line.436">                                } else {</a>
<span class="sourceLineNo">437</span><a id="line.437">                                        stopCodonEnd = stopCodon.location().bioEnd();</a>
<span class="sourceLineNo">438</span><a id="line.438">                                }</a>
<span class="sourceLineNo">439</span><a id="line.439">                                stopCodonName = stopCodon.getAttribute("ID");</a>
<span class="sourceLineNo">440</span><a id="line.440"></a>
<span class="sourceLineNo">441</span><a id="line.441">                        }</a>
<span class="sourceLineNo">442</span><a id="line.442"></a>
<span class="sourceLineNo">443</span><a id="line.443"></a>
<span class="sourceLineNo">444</span><a id="line.444"></a>
<span class="sourceLineNo">445</span><a id="line.445"></a>
<span class="sourceLineNo">446</span><a id="line.446">                        if (startCodonBegin == null) {</a>
<span class="sourceLineNo">447</span><a id="line.447">                                if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">448</span><a id="line.448">                                        FeatureI firstFeature = cdsFeatures.get(0);</a>
<span class="sourceLineNo">449</span><a id="line.449"></a>
<span class="sourceLineNo">450</span><a id="line.450">                                        startCodonBegin = firstFeature.location().bioEnd();</a>
<span class="sourceLineNo">451</span><a id="line.451">                                } else {</a>
<span class="sourceLineNo">452</span><a id="line.452">                                        FeatureI firstFeature = cdsFeatures.get(0);</a>
<span class="sourceLineNo">453</span><a id="line.453"></a>
<span class="sourceLineNo">454</span><a id="line.454">                                        startCodonBegin = firstFeature.location().bioStart();</a>
<span class="sourceLineNo">455</span><a id="line.455">                                }</a>
<span class="sourceLineNo">456</span><a id="line.456">                        }</a>
<span class="sourceLineNo">457</span><a id="line.457"></a>
<span class="sourceLineNo">458</span><a id="line.458">                        if (stopCodonEnd == null) {</a>
<span class="sourceLineNo">459</span><a id="line.459">                                if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">460</span><a id="line.460">                                        FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</a>
<span class="sourceLineNo">461</span><a id="line.461">                                        stopCodonEnd = lastFeature.location().bioStart();</a>
<span class="sourceLineNo">462</span><a id="line.462">                                } else {</a>
<span class="sourceLineNo">463</span><a id="line.463">                                        FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</a>
<span class="sourceLineNo">464</span><a id="line.464">                                        stopCodonEnd = lastFeature.location().bioEnd();</a>
<span class="sourceLineNo">465</span><a id="line.465">                                }</a>
<span class="sourceLineNo">466</span><a id="line.466">                        }</a>
<span class="sourceLineNo">467</span><a id="line.467">                        //for gtf ordering can be strand based so first is last and last is first</a>
<span class="sourceLineNo">468</span><a id="line.468">                        if (startCodonBegin &gt; stopCodonEnd) {</a>
<span class="sourceLineNo">469</span><a id="line.469">                                int temp = startCodonBegin;</a>
<span class="sourceLineNo">470</span><a id="line.470">                                startCodonBegin = stopCodonEnd;</a>
<span class="sourceLineNo">471</span><a id="line.471">                                stopCodonEnd = temp;</a>
<span class="sourceLineNo">472</span><a id="line.472">                        }</a>
<span class="sourceLineNo">473</span><a id="line.473"></a>
<span class="sourceLineNo">474</span><a id="line.474"></a>
<span class="sourceLineNo">475</span><a id="line.475"></a>
<span class="sourceLineNo">476</span><a id="line.476">                        AccessionID transcriptAccessionID = new AccessionID(mRNAID);</a>
<span class="sourceLineNo">477</span><a id="line.477">                        geneSequence = seq.getGene(geneID);</a>
<span class="sourceLineNo">478</span><a id="line.478">                        if (geneSequence == null) {</a>
<span class="sourceLineNo">479</span><a id="line.479">                                geneSequence = seq.addGene(geneAccessionID, startCodonBegin, stopCodonEnd, strand);</a>
<span class="sourceLineNo">480</span><a id="line.480">                                geneSequence.setSource(geneSource);</a>
<span class="sourceLineNo">481</span><a id="line.481">                                if (geneNote != null &amp;&amp; geneNote.length() &gt; 0) {</a>
<span class="sourceLineNo">482</span><a id="line.482">                                        geneSequence.addNote(geneNote);</a>
<span class="sourceLineNo">483</span><a id="line.483">                                }</a>
<span class="sourceLineNo">484</span><a id="line.484">                        } else {</a>
<span class="sourceLineNo">485</span><a id="line.485"></a>
<span class="sourceLineNo">486</span><a id="line.486">                                if (startCodonBegin &lt; geneSequence.getBioBegin()) {</a>
<span class="sourceLineNo">487</span><a id="line.487">                                        geneSequence.setBioBegin(startCodonBegin);</a>
<span class="sourceLineNo">488</span><a id="line.488">                                }</a>
<span class="sourceLineNo">489</span><a id="line.489">                                if (stopCodonEnd &gt; geneSequence.getBioBegin()) {</a>
<span class="sourceLineNo">490</span><a id="line.490">                                        geneSequence.setBioEnd(stopCodonEnd);</a>
<span class="sourceLineNo">491</span><a id="line.491">                                }</a>
<span class="sourceLineNo">492</span><a id="line.492"></a>
<span class="sourceLineNo">493</span><a id="line.493">                        }</a>
<span class="sourceLineNo">494</span><a id="line.494">                        TranscriptSequence transcriptSequence = geneSequence.addTranscript(transcriptAccessionID, startCodonBegin, stopCodonEnd);</a>
<span class="sourceLineNo">495</span><a id="line.495">                        transcriptSequence.setSource(mRNAsource);</a>
<span class="sourceLineNo">496</span><a id="line.496">                        if (mRNANote != null &amp;&amp; mRNANote.length() &gt; 0) {</a>
<span class="sourceLineNo">497</span><a id="line.497">                                transcriptSequence.addNote(mRNANote);</a>
<span class="sourceLineNo">498</span><a id="line.498"></a>
<span class="sourceLineNo">499</span><a id="line.499">                        }</a>
<span class="sourceLineNo">500</span><a id="line.500">                        if (startCodon != null) {</a>
<span class="sourceLineNo">501</span><a id="line.501">                                if (startCodonName == null || startCodonName.length() == 0) {</a>
<span class="sourceLineNo">502</span><a id="line.502">                                        startCodonName = mRNAID + "-start_codon-" + startCodon.location().bioStart() + "-" + startCodon.location().bioEnd();</a>
<span class="sourceLineNo">503</span><a id="line.503">                                }</a>
<span class="sourceLineNo">504</span><a id="line.504">                                transcriptSequence.addStartCodonSequence(new AccessionID(startCodonName), startCodon.location().bioStart(), startCodon.location().bioEnd());</a>
<span class="sourceLineNo">505</span><a id="line.505">                        }</a>
<span class="sourceLineNo">506</span><a id="line.506">                        if (stopCodon != null) {</a>
<span class="sourceLineNo">507</span><a id="line.507">                                if (stopCodonName == null || stopCodonName.length() == 0) {</a>
<span class="sourceLineNo">508</span><a id="line.508">                                        stopCodonName = mRNAID + "-stop_codon-" + stopCodon.location().bioStart() + "-" + stopCodon.location().bioEnd();</a>
<span class="sourceLineNo">509</span><a id="line.509">                                }</a>
<span class="sourceLineNo">510</span><a id="line.510">                                transcriptSequence.addStopCodonSequence(new AccessionID(stopCodonName), stopCodon.location().bioStart(), stopCodon.location().bioEnd());</a>
<span class="sourceLineNo">511</span><a id="line.511">                        }</a>
<span class="sourceLineNo">512</span><a id="line.512"></a>
<span class="sourceLineNo">513</span><a id="line.513">                        for (FeatureI cdsFeature : cdsFeatures) {</a>
<span class="sourceLineNo">514</span><a id="line.514">                                Feature cds = (Feature) cdsFeature;</a>
<span class="sourceLineNo">515</span><a id="line.515">                                String cdsNote = cdsFeature.getAttribute("Note");</a>
<span class="sourceLineNo">516</span><a id="line.516">                                String cdsSource = cds.source();</a>
<span class="sourceLineNo">517</span><a id="line.517">                                String cdsName = cds.getAttribute("ID");</a>
<span class="sourceLineNo">518</span><a id="line.518">                                if (cdsName == null || cdsName.length() == 0) {</a>
<span class="sourceLineNo">519</span><a id="line.519">                                        cdsName = mRNAID + "-cds-" + cds.location().bioStart() + "-" + cds.location().bioEnd();</a>
<span class="sourceLineNo">520</span><a id="line.520">                                }</a>
<span class="sourceLineNo">521</span><a id="line.521">                                AccessionID cdsAccessionID = new AccessionID(cdsName);</a>
<span class="sourceLineNo">522</span><a id="line.522">                                ExonSequence exonSequence = geneSequence.addExon(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd());</a>
<span class="sourceLineNo">523</span><a id="line.523">                                exonSequence.setSource(cdsSource);</a>
<span class="sourceLineNo">524</span><a id="line.524">                                if (cdsNote != null &amp;&amp; cdsNote.length() &gt; 0) {</a>
<span class="sourceLineNo">525</span><a id="line.525">                                        exonSequence.addNote(cdsNote);</a>
<span class="sourceLineNo">526</span><a id="line.526">                                }</a>
<span class="sourceLineNo">527</span><a id="line.527">                                transcriptSequence.addCDS(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd(), cds.frame());</a>
<span class="sourceLineNo">528</span><a id="line.528">                        }</a>
<span class="sourceLineNo">529</span><a id="line.529">                        geneSequence.addIntronsUsingExons();</a>
<span class="sourceLineNo">530</span><a id="line.530"></a>
<span class="sourceLineNo">531</span><a id="line.531">                }</a>
<span class="sourceLineNo">532</span><a id="line.532"></a>
<span class="sourceLineNo">533</span><a id="line.533">        }</a>
<span class="sourceLineNo">534</span><a id="line.534"></a>
<span class="sourceLineNo">535</span><a id="line.535">        static public LinkedHashMap&lt;String, ChromosomeSequence&gt; loadFastaAddGeneFeaturesFromGlimmerGFF3(File fastaSequenceFile, File gffFile) throws Exception {</a>
<span class="sourceLineNo">536</span><a id="line.536">                LinkedHashMap&lt;String, DNASequence&gt; dnaSequenceList = FastaReaderHelper.readFastaDNASequence(fastaSequenceFile);</a>
<span class="sourceLineNo">537</span><a id="line.537">                LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = GeneFeatureHelper.getChromosomeSequenceFromDNASequence(dnaSequenceList);</a>
<span class="sourceLineNo">538</span><a id="line.538">                FeatureList listGenes = GFF3Reader.read(gffFile.getAbsolutePath());</a>
<span class="sourceLineNo">539</span><a id="line.539">                addGlimmerGFF3GeneFeatures(chromosomeSequenceList, listGenes);</a>
<span class="sourceLineNo">540</span><a id="line.540">                return chromosomeSequenceList;</a>
<span class="sourceLineNo">541</span><a id="line.541">        }</a>
<span class="sourceLineNo">542</span><a id="line.542"></a>
<span class="sourceLineNo">543</span><a id="line.543">        static public void addGlimmerGFF3GeneFeatures(LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList, FeatureList listGenes) throws Exception {</a>
<span class="sourceLineNo">544</span><a id="line.544">                FeatureList mRNAFeatures = listGenes.selectByType("mRNA");</a>
<span class="sourceLineNo">545</span><a id="line.545">                for (FeatureI f : mRNAFeatures) {</a>
<span class="sourceLineNo">546</span><a id="line.546">                        Feature mRNAFeature = (Feature) f;</a>
<span class="sourceLineNo">547</span><a id="line.547">                        String geneid = mRNAFeature.getAttribute("ID");</a>
<span class="sourceLineNo">548</span><a id="line.548">                        String source = mRNAFeature.source();</a>
<span class="sourceLineNo">549</span><a id="line.549"></a>
<span class="sourceLineNo">550</span><a id="line.550">                        FeatureList gene = listGenes.selectByAttribute("Parent", geneid);</a>
<span class="sourceLineNo">551</span><a id="line.551">                        FeatureI geneFeature = gene.get(0);</a>
<span class="sourceLineNo">552</span><a id="line.552">                        ChromosomeSequence seq = chromosomeSequenceList.get(geneFeature.seqname());</a>
<span class="sourceLineNo">553</span><a id="line.553">                        AccessionID geneAccessionID = new AccessionID(geneid);</a>
<span class="sourceLineNo">554</span><a id="line.554">                        GeneSequence geneSequence = null;</a>
<span class="sourceLineNo">555</span><a id="line.555"></a>
<span class="sourceLineNo">556</span><a id="line.556">                        FeatureList cdsFeatures = gene.selectByType("CDS");</a>
<span class="sourceLineNo">557</span><a id="line.557">                        FeatureI feature = cdsFeatures.get(0);</a>
<span class="sourceLineNo">558</span><a id="line.558">                        Strand strand = Strand.POSITIVE;</a>
<span class="sourceLineNo">559</span><a id="line.559"></a>
<span class="sourceLineNo">560</span><a id="line.560">                        if (feature.location().isNegative()) {</a>
<span class="sourceLineNo">561</span><a id="line.561">                                strand = Strand.NEGATIVE;</a>
<span class="sourceLineNo">562</span><a id="line.562">                        }</a>
<span class="sourceLineNo">563</span><a id="line.563">                        cdsFeatures = cdsFeatures.sortByStart();</a>
<span class="sourceLineNo">564</span><a id="line.564"></a>
<span class="sourceLineNo">565</span><a id="line.565"></a>
<span class="sourceLineNo">566</span><a id="line.566"></a>
<span class="sourceLineNo">567</span><a id="line.567"></a>
<span class="sourceLineNo">568</span><a id="line.568"></a>
<span class="sourceLineNo">569</span><a id="line.569"></a>
<span class="sourceLineNo">570</span><a id="line.570"></a>
<span class="sourceLineNo">571</span><a id="line.571">                        //String seqName = feature.seqname();</a>
<span class="sourceLineNo">572</span><a id="line.572">                        FeatureI startCodon = null;</a>
<span class="sourceLineNo">573</span><a id="line.573">                        FeatureI stopCodon = null;</a>
<span class="sourceLineNo">574</span><a id="line.574">                        Integer startCodonBegin = null;</a>
<span class="sourceLineNo">575</span><a id="line.575">                        Integer stopCodonEnd = null;</a>
<span class="sourceLineNo">576</span><a id="line.576">                        String startCodonName = "";</a>
<span class="sourceLineNo">577</span><a id="line.577">                        String stopCodonName = "";</a>
<span class="sourceLineNo">578</span><a id="line.578">                        FeatureList startCodonList = gene.selectByAttribute("Note", "initial-exon");</a>
<span class="sourceLineNo">579</span><a id="line.579">                        if (startCodonList != null &amp;&amp; startCodonList.size() &gt; 0) {</a>
<span class="sourceLineNo">580</span><a id="line.580">                                startCodon = startCodonList.get(0);</a>
<span class="sourceLineNo">581</span><a id="line.581">                                if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">582</span><a id="line.582">                                        startCodonBegin = startCodon.location().bioEnd();</a>
<span class="sourceLineNo">583</span><a id="line.583">                                } else {</a>
<span class="sourceLineNo">584</span><a id="line.584">                                        startCodonBegin = startCodon.location().bioStart();</a>
<span class="sourceLineNo">585</span><a id="line.585">                                }</a>
<span class="sourceLineNo">586</span><a id="line.586">                                startCodonName = startCodon.getAttribute("ID");</a>
<span class="sourceLineNo">587</span><a id="line.587">                        }</a>
<span class="sourceLineNo">588</span><a id="line.588"></a>
<span class="sourceLineNo">589</span><a id="line.589">                        FeatureList stopCodonList = gene.selectByAttribute("Note", "final-exon");</a>
<span class="sourceLineNo">590</span><a id="line.590"></a>
<span class="sourceLineNo">591</span><a id="line.591">                        if (stopCodonList != null &amp;&amp; stopCodonList.size() &gt; 0) {</a>
<span class="sourceLineNo">592</span><a id="line.592">                                stopCodon = stopCodonList.get(0);</a>
<span class="sourceLineNo">593</span><a id="line.593">                                if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">594</span><a id="line.594">                                        stopCodonEnd = stopCodon.location().bioStart();</a>
<span class="sourceLineNo">595</span><a id="line.595">                                } else {</a>
<span class="sourceLineNo">596</span><a id="line.596">                                        stopCodonEnd = stopCodon.location().bioEnd();</a>
<span class="sourceLineNo">597</span><a id="line.597">                                }</a>
<span class="sourceLineNo">598</span><a id="line.598">                                stopCodonName = stopCodon.getAttribute("ID");</a>
<span class="sourceLineNo">599</span><a id="line.599"></a>
<span class="sourceLineNo">600</span><a id="line.600">                        }</a>
<span class="sourceLineNo">601</span><a id="line.601"></a>
<span class="sourceLineNo">602</span><a id="line.602"></a>
<span class="sourceLineNo">603</span><a id="line.603"></a>
<span class="sourceLineNo">604</span><a id="line.604"></a>
<span class="sourceLineNo">605</span><a id="line.605">                        if (startCodonBegin == null) {</a>
<span class="sourceLineNo">606</span><a id="line.606">                                if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">607</span><a id="line.607">                                        FeatureI firstFeature = cdsFeatures.get(0);</a>
<span class="sourceLineNo">608</span><a id="line.608"></a>
<span class="sourceLineNo">609</span><a id="line.609">                                        startCodonBegin = firstFeature.location().bioEnd();</a>
<span class="sourceLineNo">610</span><a id="line.610">                                } else {</a>
<span class="sourceLineNo">611</span><a id="line.611">                                        FeatureI firstFeature = cdsFeatures.get(0);</a>
<span class="sourceLineNo">612</span><a id="line.612"></a>
<span class="sourceLineNo">613</span><a id="line.613">                                        startCodonBegin = firstFeature.location().bioStart();</a>
<span class="sourceLineNo">614</span><a id="line.614">                                }</a>
<span class="sourceLineNo">615</span><a id="line.615">                        }</a>
<span class="sourceLineNo">616</span><a id="line.616"></a>
<span class="sourceLineNo">617</span><a id="line.617">                        if (stopCodonEnd == null) {</a>
<span class="sourceLineNo">618</span><a id="line.618">                                if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">619</span><a id="line.619">                                        FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</a>
<span class="sourceLineNo">620</span><a id="line.620">                                        stopCodonEnd = lastFeature.location().bioStart();</a>
<span class="sourceLineNo">621</span><a id="line.621">                                } else {</a>
<span class="sourceLineNo">622</span><a id="line.622">                                        FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</a>
<span class="sourceLineNo">623</span><a id="line.623">                                        stopCodonEnd = lastFeature.location().bioEnd();</a>
<span class="sourceLineNo">624</span><a id="line.624">                                }</a>
<span class="sourceLineNo">625</span><a id="line.625">                        }</a>
<span class="sourceLineNo">626</span><a id="line.626">                        //for gtf ordering can be strand based so first is last and last is first</a>
<span class="sourceLineNo">627</span><a id="line.627">                        if (startCodonBegin &gt; stopCodonEnd) {</a>
<span class="sourceLineNo">628</span><a id="line.628">                                int temp = startCodonBegin;</a>
<span class="sourceLineNo">629</span><a id="line.629">                                startCodonBegin = stopCodonEnd;</a>
<span class="sourceLineNo">630</span><a id="line.630">                                stopCodonEnd = temp;</a>
<span class="sourceLineNo">631</span><a id="line.631">                        }</a>
<span class="sourceLineNo">632</span><a id="line.632"></a>
<span class="sourceLineNo">633</span><a id="line.633"></a>
<span class="sourceLineNo">634</span><a id="line.634"></a>
<span class="sourceLineNo">635</span><a id="line.635">                        AccessionID transcriptAccessionID = new AccessionID(geneid);</a>
<span class="sourceLineNo">636</span><a id="line.636">                        if (geneSequence == null) {</a>
<span class="sourceLineNo">637</span><a id="line.637">                                geneSequence = seq.addGene(geneAccessionID, startCodonBegin, stopCodonEnd, strand);</a>
<span class="sourceLineNo">638</span><a id="line.638">                                geneSequence.setSource(source);</a>
<span class="sourceLineNo">639</span><a id="line.639">                        }</a>
<span class="sourceLineNo">640</span><a id="line.640">                        /*</a>
<span class="sourceLineNo">641</span><a id="line.641">                        else {</a>
<span class="sourceLineNo">642</span><a id="line.642"></a>
<span class="sourceLineNo">643</span><a id="line.643">                                if (startCodonBegin &lt; geneSequence.getBioBegin()) {</a>
<span class="sourceLineNo">644</span><a id="line.644">                                        geneSequence.setBioBegin(startCodonBegin);</a>
<span class="sourceLineNo">645</span><a id="line.645">                                }</a>
<span class="sourceLineNo">646</span><a id="line.646">                                if (stopCodonEnd &gt; geneSequence.getBioBegin()) {</a>
<span class="sourceLineNo">647</span><a id="line.647">                                        geneSequence.setBioEnd(stopCodonEnd);</a>
<span class="sourceLineNo">648</span><a id="line.648">                                }</a>
<span class="sourceLineNo">649</span><a id="line.649">                        }</a>
<span class="sourceLineNo">650</span><a id="line.650">                        */</a>
<span class="sourceLineNo">651</span><a id="line.651">                        TranscriptSequence transcriptSequence = geneSequence.addTranscript(transcriptAccessionID, startCodonBegin, stopCodonEnd);</a>
<span class="sourceLineNo">652</span><a id="line.652">                        if (startCodon != null) {</a>
<span class="sourceLineNo">653</span><a id="line.653">                                if (startCodonName == null || startCodonName.length() == 0) {</a>
<span class="sourceLineNo">654</span><a id="line.654">                                        startCodonName = geneid + "-start_codon-" + startCodon.location().bioStart() + "-" + startCodon.location().bioEnd();</a>
<span class="sourceLineNo">655</span><a id="line.655">                                }</a>
<span class="sourceLineNo">656</span><a id="line.656">                                transcriptSequence.addStartCodonSequence(new AccessionID(startCodonName), startCodon.location().bioStart(), startCodon.location().bioEnd());</a>
<span class="sourceLineNo">657</span><a id="line.657">                        }</a>
<span class="sourceLineNo">658</span><a id="line.658">                        if (stopCodon != null) {</a>
<span class="sourceLineNo">659</span><a id="line.659">                                if (stopCodonName == null || stopCodonName.length() == 0) {</a>
<span class="sourceLineNo">660</span><a id="line.660">                                        stopCodonName = geneid + "-stop_codon-" + stopCodon.location().bioStart() + "-" + stopCodon.location().bioEnd();</a>
<span class="sourceLineNo">661</span><a id="line.661">                                }</a>
<span class="sourceLineNo">662</span><a id="line.662">                                transcriptSequence.addStopCodonSequence(new AccessionID(stopCodonName), stopCodon.location().bioStart(), stopCodon.location().bioEnd());</a>
<span class="sourceLineNo">663</span><a id="line.663">                        }</a>
<span class="sourceLineNo">664</span><a id="line.664"></a>
<span class="sourceLineNo">665</span><a id="line.665">                        for (FeatureI cdsFeature : cdsFeatures) {</a>
<span class="sourceLineNo">666</span><a id="line.666">                                Feature cds = (Feature) cdsFeature;</a>
<span class="sourceLineNo">667</span><a id="line.667">                                String cdsName = cds.getAttribute("ID");</a>
<span class="sourceLineNo">668</span><a id="line.668">                                if (cdsName == null || cdsName.length() == 0) {</a>
<span class="sourceLineNo">669</span><a id="line.669">                                        cdsName = geneid + "-cds-" + cds.location().bioStart() + "-" + cds.location().bioEnd();</a>
<span class="sourceLineNo">670</span><a id="line.670">                                }</a>
<span class="sourceLineNo">671</span><a id="line.671">                                AccessionID cdsAccessionID = new AccessionID(cdsName);</a>
<span class="sourceLineNo">672</span><a id="line.672">                                //ExonSequence exonSequence = geneSequence.addExon(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd());</a>
<span class="sourceLineNo">673</span><a id="line.673">                                transcriptSequence.addCDS(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd(), cds.frame());</a>
<span class="sourceLineNo">674</span><a id="line.674">                        }</a>
<span class="sourceLineNo">675</span><a id="line.675"></a>
<span class="sourceLineNo">676</span><a id="line.676">                }</a>
<span class="sourceLineNo">677</span><a id="line.677"></a>
<span class="sourceLineNo">678</span><a id="line.678">        }</a>
<span class="sourceLineNo">679</span><a id="line.679"></a>
<span class="sourceLineNo">680</span><a id="line.680">        static public LinkedHashMap&lt;String, ChromosomeSequence&gt; loadFastaAddGeneFeaturesFromGeneMarkGTF(File fastaSequenceFile, File gffFile) throws Exception {</a>
<span class="sourceLineNo">681</span><a id="line.681">                LinkedHashMap&lt;String, DNASequence&gt; dnaSequenceList = FastaReaderHelper.readFastaDNASequence(fastaSequenceFile);</a>
<span class="sourceLineNo">682</span><a id="line.682">                LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = GeneFeatureHelper.getChromosomeSequenceFromDNASequence(dnaSequenceList);</a>
<span class="sourceLineNo">683</span><a id="line.683">                FeatureList listGenes = GeneMarkGTFReader.read(gffFile.getAbsolutePath());</a>
<span class="sourceLineNo">684</span><a id="line.684">                addGeneMarkGTFGeneFeatures(chromosomeSequenceList, listGenes);</a>
<span class="sourceLineNo">685</span><a id="line.685">                return chromosomeSequenceList;</a>
<span class="sourceLineNo">686</span><a id="line.686">        }</a>
<span class="sourceLineNo">687</span><a id="line.687"></a>
<span class="sourceLineNo">688</span><a id="line.688">        static public void addGeneMarkGTFGeneFeatures(LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList, FeatureList listGenes) throws Exception {</a>
<span class="sourceLineNo">689</span><a id="line.689">                Collection&lt;String&gt; geneIds = listGenes.attributeValues("gene_id");</a>
<span class="sourceLineNo">690</span><a id="line.690">                for (String geneid : geneIds) {</a>
<span class="sourceLineNo">691</span><a id="line.691">                        //       if(geneid.equals("45_g")){</a>
<span class="sourceLineNo">692</span><a id="line.692">                        //           int dummy =1;</a>
<span class="sourceLineNo">693</span><a id="line.693">                        //       }</a>
<span class="sourceLineNo">694</span><a id="line.694">                        FeatureList gene = listGenes.selectByAttribute("gene_id", geneid);</a>
<span class="sourceLineNo">695</span><a id="line.695">                        FeatureI geneFeature = gene.get(0);</a>
<span class="sourceLineNo">696</span><a id="line.696">                        ChromosomeSequence seq = chromosomeSequenceList.get(geneFeature.seqname());</a>
<span class="sourceLineNo">697</span><a id="line.697">                        AccessionID geneAccessionID = new AccessionID(geneid);</a>
<span class="sourceLineNo">698</span><a id="line.698">                        GeneSequence geneSequence = null;</a>
<span class="sourceLineNo">699</span><a id="line.699">                        Collection&lt;String&gt; transcriptids = gene.attributeValues("transcript_id");</a>
<span class="sourceLineNo">700</span><a id="line.700">                        for (String transcriptid : transcriptids) {</a>
<span class="sourceLineNo">701</span><a id="line.701">                                // get all the individual features (exons, CDS regions, etc.) of this gene</a>
<span class="sourceLineNo">702</span><a id="line.702"></a>
<span class="sourceLineNo">703</span><a id="line.703"></a>
<span class="sourceLineNo">704</span><a id="line.704">                                FeatureList transcriptFeature = listGenes.selectByAttribute("transcript_id", transcriptid);</a>
<span class="sourceLineNo">705</span><a id="line.705">                                // now select only the coding regions of this gene</a>
<span class="sourceLineNo">706</span><a id="line.706">                                FeatureList cdsFeatures = transcriptFeature.selectByType("CDS");</a>
<span class="sourceLineNo">707</span><a id="line.707">                                // sort them</a>
<span class="sourceLineNo">708</span><a id="line.708">                                cdsFeatures = cdsFeatures.sortByStart();</a>
<span class="sourceLineNo">709</span><a id="line.709"></a>
<span class="sourceLineNo">710</span><a id="line.710">                                FeatureI feature = cdsFeatures.get(0);</a>
<span class="sourceLineNo">711</span><a id="line.711">                                Strand strand = Strand.POSITIVE;</a>
<span class="sourceLineNo">712</span><a id="line.712"></a>
<span class="sourceLineNo">713</span><a id="line.713">                                if (feature.location().isNegative()) {</a>
<span class="sourceLineNo">714</span><a id="line.714">                                        strand = Strand.NEGATIVE;</a>
<span class="sourceLineNo">715</span><a id="line.715">                                }</a>
<span class="sourceLineNo">716</span><a id="line.716"></a>
<span class="sourceLineNo">717</span><a id="line.717">                                //String seqName = feature.seqname();</a>
<span class="sourceLineNo">718</span><a id="line.718">                                FeatureI startCodon = null;</a>
<span class="sourceLineNo">719</span><a id="line.719">                                FeatureI stopCodon = null;</a>
<span class="sourceLineNo">720</span><a id="line.720">                                Integer startCodonBegin = null;</a>
<span class="sourceLineNo">721</span><a id="line.721">                                Integer stopCodonEnd = null;</a>
<span class="sourceLineNo">722</span><a id="line.722">                                String startCodonName = "";</a>
<span class="sourceLineNo">723</span><a id="line.723">                                String stopCodonName = "";</a>
<span class="sourceLineNo">724</span><a id="line.724">                                FeatureList startCodonList = transcriptFeature.selectByType("start_codon");</a>
<span class="sourceLineNo">725</span><a id="line.725">                                if (startCodonList != null &amp;&amp; startCodonList.size() &gt; 0) {</a>
<span class="sourceLineNo">726</span><a id="line.726">                                        startCodon = startCodonList.get(0);</a>
<span class="sourceLineNo">727</span><a id="line.727">                                        if (strand == Strand.POSITIVE) {</a>
<span class="sourceLineNo">728</span><a id="line.728">                                                startCodonBegin = startCodon.location().bioStart();</a>
<span class="sourceLineNo">729</span><a id="line.729">                                        } else {</a>
<span class="sourceLineNo">730</span><a id="line.730">                                                startCodonBegin = startCodon.location().bioEnd();</a>
<span class="sourceLineNo">731</span><a id="line.731">                                        }</a>
<span class="sourceLineNo">732</span><a id="line.732">                                        startCodonName = startCodon.getAttribute("transcript_name");</a>
<span class="sourceLineNo">733</span><a id="line.733">                                }</a>
<span class="sourceLineNo">734</span><a id="line.734"></a>
<span class="sourceLineNo">735</span><a id="line.735">                                FeatureList stopCodonList = transcriptFeature.selectByType("stop_codon");</a>
<span class="sourceLineNo">736</span><a id="line.736"></a>
<span class="sourceLineNo">737</span><a id="line.737">                                if (stopCodonList != null &amp;&amp; stopCodonList.size() &gt; 0) {</a>
<span class="sourceLineNo">738</span><a id="line.738">                                        stopCodon = stopCodonList.get(0);</a>
<span class="sourceLineNo">739</span><a id="line.739">                                        if (strand == Strand.POSITIVE) {</a>
<span class="sourceLineNo">740</span><a id="line.740">                                                stopCodonEnd = stopCodon.location().bioEnd();</a>
<span class="sourceLineNo">741</span><a id="line.741">                                        } else {</a>
<span class="sourceLineNo">742</span><a id="line.742">                                                stopCodonEnd = stopCodon.location().bioStart();</a>
<span class="sourceLineNo">743</span><a id="line.743">                                        }</a>
<span class="sourceLineNo">744</span><a id="line.744"></a>
<span class="sourceLineNo">745</span><a id="line.745">                                        stopCodonName = stopCodon.getAttribute("transcript_name");</a>
<span class="sourceLineNo">746</span><a id="line.746"></a>
<span class="sourceLineNo">747</span><a id="line.747">                                }</a>
<span class="sourceLineNo">748</span><a id="line.748"></a>
<span class="sourceLineNo">749</span><a id="line.749"></a>
<span class="sourceLineNo">750</span><a id="line.750"></a>
<span class="sourceLineNo">751</span><a id="line.751"></a>
<span class="sourceLineNo">752</span><a id="line.752">                                if (startCodonBegin == null) {</a>
<span class="sourceLineNo">753</span><a id="line.753">                                        if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">754</span><a id="line.754">                                                FeatureI firstFeature = cdsFeatures.get(0);</a>
<span class="sourceLineNo">755</span><a id="line.755"></a>
<span class="sourceLineNo">756</span><a id="line.756">                                                startCodonBegin = firstFeature.location().bioEnd();</a>
<span class="sourceLineNo">757</span><a id="line.757">                                        } else {</a>
<span class="sourceLineNo">758</span><a id="line.758">                                                FeatureI firstFeature = cdsFeatures.get(0);</a>
<span class="sourceLineNo">759</span><a id="line.759"></a>
<span class="sourceLineNo">760</span><a id="line.760">                                                startCodonBegin = firstFeature.location().bioStart();</a>
<span class="sourceLineNo">761</span><a id="line.761">                                        }</a>
<span class="sourceLineNo">762</span><a id="line.762">                                }</a>
<span class="sourceLineNo">763</span><a id="line.763"></a>
<span class="sourceLineNo">764</span><a id="line.764">                                if (stopCodonEnd == null) {</a>
<span class="sourceLineNo">765</span><a id="line.765">                                        if (strand == Strand.NEGATIVE) {</a>
<span class="sourceLineNo">766</span><a id="line.766">                                                FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</a>
<span class="sourceLineNo">767</span><a id="line.767">                                                stopCodonEnd = lastFeature.location().bioStart();</a>
<span class="sourceLineNo">768</span><a id="line.768">                                        } else {</a>
<span class="sourceLineNo">769</span><a id="line.769">                                                FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</a>
<span class="sourceLineNo">770</span><a id="line.770">                                                stopCodonEnd = lastFeature.location().bioEnd();</a>
<span class="sourceLineNo">771</span><a id="line.771">                                        }</a>
<span class="sourceLineNo">772</span><a id="line.772">                                }</a>
<span class="sourceLineNo">773</span><a id="line.773">                                //for gtf ordering can be strand based so first is last and last is first</a>
<span class="sourceLineNo">774</span><a id="line.774">                                if (startCodonBegin &gt; stopCodonEnd) {</a>
<span class="sourceLineNo">775</span><a id="line.775">                                        int temp = startCodonBegin;</a>
<span class="sourceLineNo">776</span><a id="line.776">                                        startCodonBegin = stopCodonEnd;</a>
<span class="sourceLineNo">777</span><a id="line.777">                                        stopCodonEnd = temp;</a>
<span class="sourceLineNo">778</span><a id="line.778">                                }</a>
<span class="sourceLineNo">779</span><a id="line.779"></a>
<span class="sourceLineNo">780</span><a id="line.780">                                AccessionID transcriptAccessionID = new AccessionID(transcriptid);</a>
<span class="sourceLineNo">781</span><a id="line.781">                                if (geneSequence == null) {</a>
<span class="sourceLineNo">782</span><a id="line.782">                                        geneSequence = seq.addGene(geneAccessionID, startCodonBegin, stopCodonEnd, strand);</a>
<span class="sourceLineNo">783</span><a id="line.783">                                        geneSequence.setSource(((Feature) feature).source());</a>
<span class="sourceLineNo">784</span><a id="line.784">                                } else {</a>
<span class="sourceLineNo">785</span><a id="line.785">                                        //if multiple transcripts for one gene make sure the gene is defined as the min and max start/end</a>
<span class="sourceLineNo">786</span><a id="line.786"></a>
<span class="sourceLineNo">787</span><a id="line.787">                                        if (startCodonBegin &lt; geneSequence.getBioBegin()) {</a>
<span class="sourceLineNo">788</span><a id="line.788">                                                geneSequence.setBioBegin(startCodonBegin);</a>
<span class="sourceLineNo">789</span><a id="line.789">                                        }</a>
<span class="sourceLineNo">790</span><a id="line.790">                                        if (stopCodonEnd &gt; geneSequence.getBioBegin()) {</a>
<span class="sourceLineNo">791</span><a id="line.791">                                                geneSequence.setBioEnd(stopCodonEnd);</a>
<span class="sourceLineNo">792</span><a id="line.792">                                        }</a>
<span class="sourceLineNo">793</span><a id="line.793"></a>
<span class="sourceLineNo">794</span><a id="line.794">                                }</a>
<span class="sourceLineNo">795</span><a id="line.795">                                TranscriptSequence transcriptSequence = geneSequence.addTranscript(transcriptAccessionID, startCodonBegin, stopCodonEnd);</a>
<span class="sourceLineNo">796</span><a id="line.796">                                if (startCodon != null) {</a>
<span class="sourceLineNo">797</span><a id="line.797">                                        if (startCodonName == null || startCodonName.length() == 0) {</a>
<span class="sourceLineNo">798</span><a id="line.798">                                                startCodonName = transcriptid + "-start_codon-" + startCodon.location().bioStart() + "-" + startCodon.location().bioEnd();</a>
<span class="sourceLineNo">799</span><a id="line.799">                                        }</a>
<span class="sourceLineNo">800</span><a id="line.800">                                        transcriptSequence.addStartCodonSequence(new AccessionID(startCodonName), startCodon.location().bioStart(), startCodon.location().bioEnd());</a>
<span class="sourceLineNo">801</span><a id="line.801">                                }</a>
<span class="sourceLineNo">802</span><a id="line.802">                                if (stopCodon != null) {</a>
<span class="sourceLineNo">803</span><a id="line.803">                                        if (stopCodonName == null || stopCodonName.length() == 0) {</a>
<span class="sourceLineNo">804</span><a id="line.804">                                                stopCodonName = transcriptid + "-stop_codon-" + stopCodon.location().bioStart() + "-" + stopCodon.location().bioEnd();</a>
<span class="sourceLineNo">805</span><a id="line.805">                                        }</a>
<span class="sourceLineNo">806</span><a id="line.806">                                        transcriptSequence.addStopCodonSequence(new AccessionID(stopCodonName), stopCodon.location().bioStart(), stopCodon.location().bioEnd());</a>
<span class="sourceLineNo">807</span><a id="line.807">                                }</a>
<span class="sourceLineNo">808</span><a id="line.808"></a>
<span class="sourceLineNo">809</span><a id="line.809">                                for (FeatureI cdsFeature : cdsFeatures) {</a>
<span class="sourceLineNo">810</span><a id="line.810">                                        Feature cds = (Feature) cdsFeature;</a>
<span class="sourceLineNo">811</span><a id="line.811">                                        // for genemark it appears frame of 2 =1 and frame of 1 = 2</a>
<span class="sourceLineNo">812</span><a id="line.812">                                        // doesn't matter when you string cds regions together as one block</a>
<span class="sourceLineNo">813</span><a id="line.813">                                        // but does make a difference when you try to make a protein sequence for each CDS region where</a>
<span class="sourceLineNo">814</span><a id="line.814">                                        // you give up or borrow based on the frame value</a>
<span class="sourceLineNo">815</span><a id="line.815">                                        // compared with gff like files and docs for geneid and glimmer where geneid and glimmer both do it the same</a>
<span class="sourceLineNo">816</span><a id="line.816">                                        // way that appears to match the gff3 docs.</a>
<span class="sourceLineNo">817</span><a id="line.817">                                        int frame = cds.frame();</a>
<span class="sourceLineNo">818</span><a id="line.818">                                        if (frame == 1) {</a>
<span class="sourceLineNo">819</span><a id="line.819">                                                frame = 2;</a>
<span class="sourceLineNo">820</span><a id="line.820">                                        } else if (frame == 2) {</a>
<span class="sourceLineNo">821</span><a id="line.821">                                                frame = 1;</a>
<span class="sourceLineNo">822</span><a id="line.822">                                        } else {</a>
<span class="sourceLineNo">823</span><a id="line.823">                                                frame = 0;</a>
<span class="sourceLineNo">824</span><a id="line.824">                                        }</a>
<span class="sourceLineNo">825</span><a id="line.825">                                        String cdsName = cds.getAttribute("transcript_name");</a>
<span class="sourceLineNo">826</span><a id="line.826">                                        if (cdsName == null || cdsName.length() == 0) {</a>
<span class="sourceLineNo">827</span><a id="line.827">                                                cdsName = transcriptid + "-cds-" + cds.location().bioStart() + "-" + cds.location().bioEnd();</a>
<span class="sourceLineNo">828</span><a id="line.828">                                        }</a>
<span class="sourceLineNo">829</span><a id="line.829">                                        AccessionID cdsAccessionID = new AccessionID(cdsName);</a>
<span class="sourceLineNo">830</span><a id="line.830">                                        //ExonSequence exonSequence = geneSequence.addExon(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd());</a>
<span class="sourceLineNo">831</span><a id="line.831">                                        transcriptSequence.addCDS(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd(), frame);</a>
<span class="sourceLineNo">832</span><a id="line.832">                                }</a>
<span class="sourceLineNo">833</span><a id="line.833">                        }</a>
<span class="sourceLineNo">834</span><a id="line.834">                }</a>
<span class="sourceLineNo">835</span><a id="line.835"></a>
<span class="sourceLineNo">836</span><a id="line.836">        }</a>
<span class="sourceLineNo">837</span><a id="line.837"></a>
<span class="sourceLineNo">838</span><a id="line.838">        static public LinkedHashMap&lt;String, ProteinSequence&gt; getProteinSequences(Collection&lt;ChromosomeSequence&gt; chromosomeSequences) throws Exception {</a>
<span class="sourceLineNo">839</span><a id="line.839">                LinkedHashMap&lt;String, ProteinSequence&gt; proteinSequenceHashMap = new LinkedHashMap&lt;String, ProteinSequence&gt;();</a>
<span class="sourceLineNo">840</span><a id="line.840">                for (ChromosomeSequence dnaSequence : chromosomeSequences) {</a>
<span class="sourceLineNo">841</span><a id="line.841">                        for (GeneSequence geneSequence : dnaSequence.getGeneSequences().values()) {</a>
<span class="sourceLineNo">842</span><a id="line.842">                                for (TranscriptSequence transcriptSequence : geneSequence.getTranscripts().values()) {</a>
<span class="sourceLineNo">843</span><a id="line.843">                                        //TODO remove?</a>
<span class="sourceLineNo">844</span><a id="line.844">//                    DNASequence dnaCodingSequence = transcriptSequence.getDNACodingSequence();</a>
<span class="sourceLineNo">845</span><a id="line.845">//                    logger.info("CDS={}", dnaCodingSequence.getSequenceAsString());</a>
<span class="sourceLineNo">846</span><a id="line.846"></a>
<span class="sourceLineNo">847</span><a id="line.847">                                        try {</a>
<span class="sourceLineNo">848</span><a id="line.848">                                                ProteinSequence proteinSequence = transcriptSequence.getProteinSequence();</a>
<span class="sourceLineNo">849</span><a id="line.849"></a>
<span class="sourceLineNo">850</span><a id="line.850">//                        logger.info("{} {}", proteinSequence.getAccession().getID(), proteinSequence);</a>
<span class="sourceLineNo">851</span><a id="line.851">                                                if (proteinSequenceHashMap.containsKey(proteinSequence.getAccession().getID())) {</a>
<span class="sourceLineNo">852</span><a id="line.852">                                                        throw new Exception("Duplicate protein sequence id=" + proteinSequence.getAccession().getID() + " found at Gene id=" + geneSequence.getAccession().getID());</a>
<span class="sourceLineNo">853</span><a id="line.853">                                                } else {</a>
<span class="sourceLineNo">854</span><a id="line.854">                                                        proteinSequenceHashMap.put(proteinSequence.getAccession().getID(), proteinSequence);</a>
<span class="sourceLineNo">855</span><a id="line.855">                                                }</a>
<span class="sourceLineNo">856</span><a id="line.856">                                        } catch (Exception e) {</a>
<span class="sourceLineNo">857</span><a id="line.857">                                                logger.error("Exception: ", e);</a>
<span class="sourceLineNo">858</span><a id="line.858">                                        }</a>
<span class="sourceLineNo">859</span><a id="line.859"></a>
<span class="sourceLineNo">860</span><a id="line.860">                                }</a>
<span class="sourceLineNo">861</span><a id="line.861"></a>
<span class="sourceLineNo">862</span><a id="line.862">                        }</a>
<span class="sourceLineNo">863</span><a id="line.863">                }</a>
<span class="sourceLineNo">864</span><a id="line.864">                return proteinSequenceHashMap;</a>
<span class="sourceLineNo">865</span><a id="line.865">        }</a>
<span class="sourceLineNo">866</span><a id="line.866"></a>
<span class="sourceLineNo">867</span><a id="line.867">        static public LinkedHashMap&lt;String, GeneSequence&gt; getGeneSequences(Collection&lt;ChromosomeSequence&gt; chromosomeSequences) throws Exception {</a>
<span class="sourceLineNo">868</span><a id="line.868">                LinkedHashMap&lt;String, GeneSequence&gt; geneSequenceHashMap = new LinkedHashMap&lt;String, GeneSequence&gt;();</a>
<span class="sourceLineNo">869</span><a id="line.869">                for (ChromosomeSequence chromosomeSequence : chromosomeSequences) {</a>
<span class="sourceLineNo">870</span><a id="line.870">                        for (GeneSequence geneSequence : chromosomeSequence.getGeneSequences().values()) {</a>
<span class="sourceLineNo">871</span><a id="line.871">                                geneSequenceHashMap.put(geneSequence.getAccession().getID(), geneSequence);</a>
<span class="sourceLineNo">872</span><a id="line.872">                        }</a>
<span class="sourceLineNo">873</span><a id="line.873">                }</a>
<span class="sourceLineNo">874</span><a id="line.874"></a>
<span class="sourceLineNo">875</span><a id="line.875">                return geneSequenceHashMap;</a>
<span class="sourceLineNo">876</span><a id="line.876">        }</a>
<span class="sourceLineNo">877</span><a id="line.877"></a>
<span class="sourceLineNo">878</span><a id="line.878">        public static void main(String[] args) throws Exception {</a>
<span class="sourceLineNo">879</span><a id="line.879">/*        if (false) {</a>
<span class="sourceLineNo">880</span><a id="line.880">                        LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGeneMarkGTF(new File("Scaffolds.fna"), new File("genemark_hmm.gtf"));</a>
<span class="sourceLineNo">881</span><a id="line.881">                        LinkedHashMap&lt;String, ProteinSequence&gt; proteinSequenceList = GeneFeatureHelper.getProteinSequences(chromosomeSequenceList.values());</a>
<span class="sourceLineNo">882</span><a id="line.882">                }</a>
<span class="sourceLineNo">883</span><a id="line.883"></a>
<span class="sourceLineNo">884</span><a id="line.884">                if (false) {</a>
<span class="sourceLineNo">885</span><a id="line.885">                        LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGlimmerGFF3(new File("Scaffolds.fna"), new File("glimmerhmm.gff"));</a>
<span class="sourceLineNo">886</span><a id="line.886">                        LinkedHashMap&lt;String, ProteinSequence&gt; proteinSequenceList = GeneFeatureHelper.getProteinSequences(chromosomeSequenceList.values());</a>
<span class="sourceLineNo">887</span><a id="line.887">                        //  for (ProteinSequence proteinSequence : proteinSequenceList.values()) {</a>
<span class="sourceLineNo">888</span><a id="line.888">                        //      logger.info(proteinSequence.getAccession().getID() + " " + proteinSequence);</a>
<span class="sourceLineNo">889</span><a id="line.889">                        //  }</a>
<span class="sourceLineNo">890</span><a id="line.890">                        FastaWriterHelper.writeProteinSequence(new File("predicted_glimmer.faa"), proteinSequenceList.values());</a>
<span class="sourceLineNo">891</span><a id="line.891"></a>
<span class="sourceLineNo">892</span><a id="line.892">                }</a>
<span class="sourceLineNo">893</span><a id="line.893">                if (false) {</a>
<span class="sourceLineNo">894</span><a id="line.894">                        GeneFeatureHelper.outputFastaSequenceLengthGFF3(new File("Scaffolds.fna"), new File("scaffolds.gff3"));</a>
<span class="sourceLineNo">895</span><a id="line.895">                }</a>
<span class="sourceLineNo">896</span><a id="line.896"></a>
<span class="sourceLineNo">897</span><a id="line.897"> */</a>
<span class="sourceLineNo">898</span><a id="line.898"></a>
<span class="sourceLineNo">899</span><a id="line.899">                try {</a>
<span class="sourceLineNo">900</span><a id="line.900"></a>
<span class="sourceLineNo">901</span><a id="line.901">                        if (true) {</a>
<span class="sourceLineNo">902</span><a id="line.902">                                //File fastaSequenceFile = new File("Scaffolds.fna");</a>
<span class="sourceLineNo">903</span><a id="line.903">                                //File gff3File = new File("geneid-v6.gff3");</a>
<span class="sourceLineNo">904</span><a id="line.904">                                //LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceHashMap = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGmodGFF3(fastaSequenceFile, gff3File,true);</a>
<span class="sourceLineNo">905</span><a id="line.905">                        }</a>
<span class="sourceLineNo">906</span><a id="line.906"></a>
<span class="sourceLineNo">907</span><a id="line.907">                } catch (Exception e) {</a>
<span class="sourceLineNo">908</span><a id="line.908">                        logger.error("Exception: ", e);</a>
<span class="sourceLineNo">909</span><a id="line.909">                }</a>
<span class="sourceLineNo">910</span><a id="line.910"></a>
<span class="sourceLineNo">911</span><a id="line.911">        }</a>
<span class="sourceLineNo">912</span><a id="line.912">}</a>




























































</pre>
</div>
</main>
</body>
</html>
