<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.16) on Fri Oct 28 16:15:48 PDT 2022 -->
<title>Uses of Interface org.biojava.nbio.structure.align.multiple.MultipleAlignment (biojava 6.1.0 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2022-10-28">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../../jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Interface org.biojava.nbio.structure.align.multiple.MultipleAlignment (biojava 6.1.0 API)";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Interface org.biojava.nbio.structure.align.multiple.MultipleAlignment" class="title">Uses of Interface<br>org.biojava.nbio.structure.align.multiple.MultipleAlignment</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align">org.biojava.nbio.structure.align</a></th>
<td class="colLast">
<div class="block">Classes for the alignment of structures.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.gui">org.biojava.nbio.structure.align.gui</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.gui.aligpanel">org.biojava.nbio.structure.align.gui.aligpanel</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.gui.jmol">org.biojava.nbio.structure.align.gui.jmol</a></th>
<td class="colLast">
<div class="block">Utility methods for better interaction with Jmol.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.multiple">org.biojava.nbio.structure.align.multiple</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.multiple.mc">org.biojava.nbio.structure.align.multiple.mc</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.quaternary">org.biojava.nbio.structure.align.quaternary</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.align.xml">org.biojava.nbio.structure.align.xml</a></th>
<td class="colLast">
<div class="block">Serialization and deserialization for structure alignment results (as AFPChain object) to XML.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.cluster">org.biojava.nbio.structure.cluster</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.symmetry.internal">org.biojava.nbio.structure.symmetry.internal</a></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.nbio.structure.symmetry.utils">org.biojava.nbio.structure.symmetry.utils</a></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align">
<!--   -->
</a>
<h3>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../package-summary.html">org.biojava.nbio.structure.align</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../package-summary.html">org.biojava.nbio.structure.align</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleStructureAligner.</span><code><span class="memberNameLink"><a href="../../MultipleStructureAligner.html#align(java.util.List)">align</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code></th>
<td class="colLast">
<div class="block">Run an alignment while specifying the atoms to be aligned.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleStructureAligner.</span><code><span class="memberNameLink"><a href="../../MultipleStructureAligner.html#align(java.util.List,java.lang.Object)">align</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;params)</code></th>
<td class="colLast">
<div class="block">Run an alignment and also send a bean containing the parameters.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.gui">
<!--   -->
</a>
<h3>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../gui/package-summary.html">org.biojava.nbio.structure.align.gui</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MultipleAlignmentJmol</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentJmolDisplay.</span><code><span class="memberNameLink"><a href="../../gui/MultipleAlignmentJmolDisplay.html#display(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">display</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code></th>
<td class="colLast">
<div class="block">Display a MultipleAlignment with a JmolPanel.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/javax/swing/JMenuBar.html?is-external=true" title="class or interface in javax.swing" class="externalLink">JMenuBar</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MenuCreator.</span><code><span class="memberNameLink"><a href="../../gui/MenuCreator.html#getAlignmentPanelMenu(javax.swing.JFrame,java.awt.event.ActionListener,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAlignmentPanelMenu</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/javax/swing/JFrame.html?is-external=true" title="class or interface in javax.swing" class="externalLink">JFrame</a>&nbsp;frame,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/awt/event/ActionListener.html?is-external=true" title="class or interface in java.awt.event" class="externalLink">ActionListener</a>&nbsp;actionListener,
                     <a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                     <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">Create the menu for the Alignment Panel representation of
 Structural Alignments.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/javax/swing/JMenuBar.html?is-external=true" title="class or interface in javax.swing" class="externalLink">JMenuBar</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MenuCreator.</span><code><span class="memberNameLink"><a href="../../gui/MenuCreator.html#getAlignmentTextMenu(javax.swing.JFrame,java.awt.event.ActionListener,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAlignmentTextMenu</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/javax/swing/JFrame.html?is-external=true" title="class or interface in javax.swing" class="externalLink">JFrame</a>&nbsp;frame,
                    <a href="https://docs.oracle.com/javase/8/docs/api/java/awt/event/ActionListener.html?is-external=true" title="class or interface in java.awt.event" class="externalLink">ActionListener</a>&nbsp;actionListener,
                    <a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">Create the menu for the Text representations of Structural Alignments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentJmolDisplay.</span><code><span class="memberNameLink"><a href="../../gui/MultipleAlignmentJmolDisplay.html#getPDBresnum(int,org.biojava.nbio.structure.align.multiple.MultipleAlignment,org.biojava.nbio.structure.Atom%5B%5D)">getPDBresnum</a></span>&#8203;(int&nbsp;structNum,
            <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
            <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></th>
<td class="colLast">
<div class="block">Utility method used in the <a href="../../gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol"><code>MultipleAlignmentJmol</code></a> Frame,
 when the aligned residues of a structure in the alignment have
 to be selected for formatting them (coloring and style).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/javax/swing/JMenuItem.html?is-external=true" title="class or interface in javax.swing" class="externalLink">JMenuItem</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MenuCreator.</span><code><span class="memberNameLink"><a href="../../gui/MenuCreator.html#getSaveAlignmentMenuItem(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getSaveAlignmentMenuItem</a></span>&#8203;(<a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/javax/swing/JMenuBar.html?is-external=true" title="class or interface in javax.swing" class="externalLink">JMenuBar</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MenuCreator.</span><code><span class="memberNameLink"><a href="../../gui/MenuCreator.html#initJmolMenu(javax.swing.JFrame,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.multiple.MultipleAlignment)">initJmolMenu</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/javax/swing/JFrame.html?is-external=true" title="class or interface in javax.swing" class="externalLink">JFrame</a>&nbsp;frame,
            <a href="../../gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;parent,
            <a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
            <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">Provide a JMenuBar that can be added to a JFrame containing
 a JmolPanel.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentJmolDisplay.</span><code><span class="memberNameLink"><a href="../../gui/MultipleAlignmentJmolDisplay.html#showAlignmentImage(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.lang.String)">showAlignmentImage</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                  <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;result)</code></th>
<td class="colLast">
<div class="block">Creates a new Frame with the String output representation of the
 <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentJmolDisplay.</span><code><span class="memberNameLink"><a href="../../gui/MultipleAlignmentJmolDisplay.html#showMultipleAligmentPanel(org.biojava.nbio.structure.align.multiple.MultipleAlignment,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">showMultipleAligmentPanel</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                         <a href="../../gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code></th>
<td class="colLast">
<div class="block">Creates a new Frame with the MultipleAlignment Sequence Panel.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../gui/MySaveFileListener.html#%3Cinit%3E(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.multiple.MultipleAlignment)">MySaveFileListener</a></span>&#8203;(<a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">Constructor to avoid checking which of the two is null before
 instantiating this class.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../gui/MySaveFileListener.html#%3Cinit%3E(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">MySaveFileListener</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.gui.aligpanel">
<!--   -->
</a>
<h3>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAligPanel.</span><code><span class="memberNameLink"><a href="../../gui/aligpanel/MultipleAligPanel.html#getMultipleAlignment()">getMultipleAlignment</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../gui/aligpanel/MultipleAligPanel.html#%3Cinit%3E(org.biojava.nbio.structure.align.multiple.MultipleAlignment,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">MultipleAligPanel</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                 <a href="../../gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jm)</code></th>
<td class="colLast">
<div class="block">Constructor using a MultipleAlignment.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.gui.jmol">
<!--   -->
</a>
<h3>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentJmol.</span><code><span class="memberNameLink"><a href="../../gui/jmol/MultipleAlignmentJmol.html#getMultipleAlignment()">getMultipleAlignment</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentJmol.</span><code><span class="memberNameLink"><a href="../../gui/jmol/MultipleAlignmentJmol.html#getJmolString(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,org.jcolorbrewer.ColorBrewer,boolean)">getJmolString</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
             <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformedAtoms,
             org.jcolorbrewer.ColorBrewer&nbsp;colorPalette,
             boolean&nbsp;colorByBlocks)</code></th>
<td class="colLast">
<div class="block">Generate a Jmol command String that colors the aligned residues of every
 structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentJmol.</span><code><span class="memberNameLink"><a href="../../gui/jmol/MultipleAlignmentJmol.html#getMultiBlockJmolString(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,org.jcolorbrewer.ColorBrewer,boolean)">getMultiBlockJmolString</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                       <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformedAtoms,
                       org.jcolorbrewer.ColorBrewer&nbsp;colorPalette,
                       boolean&nbsp;colorByBlocks)</code></th>
<td class="colLast">
<div class="block">Colors every Block of the structures with a different color, following
 the palette.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../gui/jmol/MultipleAlignmentJmol.html#%3Cinit%3E(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)">MultipleAlignmentJmol</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;rotatedAtoms)</code></th>
<td class="colLast">
<div class="block">The constructor displays the Mutltiple Alignment in a new JmolPanel
 Frame.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.multiple">
<!--   -->
</a>
<h3>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a> that implement <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../MultipleAlignmentImpl.html" title="class in org.biojava.nbio.structure.align.multiple">MultipleAlignmentImpl</a></span></code></th>
<td class="colLast">
<div class="block">A general implementation of a <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignment.</span><code><span class="memberNameLink"><a href="../MultipleAlignment.html#clone()">clone</a></span>()</code></th>
<td class="colLast">
<div class="block">Creates and returns an identical copy of this alignment, including a deep
 copy of all constituent BlockSets.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BlockSet.</span><code><span class="memberNameLink"><a href="../BlockSet.html#getMultipleAlignment()">getMultipleAlignment</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the parent MultipleAlignment of the BlockSet.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BlockSetImpl.</span><code><span class="memberNameLink"><a href="../BlockSetImpl.html#getMultipleAlignment()">getMultipleAlignment</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentEnsemble.</span><code><span class="memberNameLink"><a href="../MultipleAlignmentEnsemble.html#getMultipleAlignment(int)">getMultipleAlignment</a></span>&#8203;(int&nbsp;index)</code></th>
<td class="colLast">
<div class="block">Returns the MultipleAlignments at the specified index
 in the ensemble.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentEnsembleImpl.</span><code><span class="memberNameLink"><a href="../MultipleAlignmentEnsembleImpl.html#getMultipleAlignment(int)">getMultipleAlignment</a></span>&#8203;(int&nbsp;index)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a> that return types with arguments of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentEnsemble.</span><code><span class="memberNameLink"><a href="../MultipleAlignmentEnsemble.html#getMultipleAlignments()">getMultipleAlignments</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the List of MultipleAlignments in the ensemble.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentEnsembleImpl.</span><code><span class="memberNameLink"><a href="../MultipleAlignmentEnsembleImpl.html#getMultipleAlignments()">getMultipleAlignments</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentEnsemble.</span><code><span class="memberNameLink"><a href="../MultipleAlignmentEnsemble.html#addMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">addMultipleAlignment</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Add a new MultipleAlignment to the end of the ensemble and set its
 parent ensemble to this.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentEnsembleImpl.</span><code><span class="memberNameLink"><a href="../MultipleAlignmentEnsembleImpl.html#addMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">addMultipleAlignment</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BlockSet.</span><code><span class="memberNameLink"><a href="../BlockSet.html#setMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">setMultipleAlignment</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;parent)</code></th>
<td class="colLast">
<div class="block">Set the back-reference to its parent MultipleAlignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BlockSetImpl.</span><code><span class="memberNameLink"><a href="../BlockSetImpl.html#setMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">setMultipleAlignment</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;parent)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Method parameters in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a> with type arguments of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentEnsemble.</span><code><span class="memberNameLink"><a href="../MultipleAlignmentEnsemble.html#setMultipleAlignments(java.util.List)">setMultipleAlignments</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&gt;&nbsp;alignments)</code></th>
<td class="colLast">
<div class="block">Set the List of MultipleAlignments in the ensemble.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentEnsembleImpl.</span><code><span class="memberNameLink"><a href="../MultipleAlignmentEnsembleImpl.html#setMultipleAlignments(java.util.List)">setMultipleAlignments</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&gt;&nbsp;alignments)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../BlockSetImpl.html#%3Cinit%3E(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">BlockSetImpl</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Constructor.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.multiple.mc">
<!--   -->
</a>
<h3>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleMcMain.</span><code><span class="memberNameLink"><a href="../mc/MultipleMcMain.html#align(java.util.List)">align</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleMcMain.</span><code><span class="memberNameLink"><a href="../mc/MultipleMcMain.html#align(java.util.List,java.lang.Object)">align</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays,
     <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a>&nbsp;parameters)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleMcOptimizer.</span><code><span class="memberNameLink"><a href="../mc/MultipleMcOptimizer.html#call()">call</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleMcOptimizer.</span><code><span class="memberNameLink"><a href="../mc/MultipleMcOptimizer.html#optimize()">optimize</a></span>()</code></th>
<td class="colLast">
<div class="block">Optimization method based in a Monte-Carlo approach.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../mc/MultipleMcOptimizer.html#%3Cinit%3E(org.biojava.nbio.structure.align.multiple.MultipleAlignment,org.biojava.nbio.structure.align.multiple.mc.MultipleMcParameters,int)">MultipleMcOptimizer</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;seedAln,
                   <a href="../mc/MultipleMcParameters.html" title="class in org.biojava.nbio.structure.align.multiple.mc">MultipleMcParameters</a>&nbsp;params,
                   int&nbsp;reference)</code></th>
<td class="colLast">
<div class="block">Constructor.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.multiple.util">
<!--   -->
</a>
<h3>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentScorer.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentScorer.html#calculateScores(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">calculateScores</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Calculates and puts the RMSD and the average TM-Score of the
 MultipleAlignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#getAtomForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,int,int)">getAtomForSequencePosition</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                          <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;mapSeqToStruct,
                          int&nbsp;str,
                          int&nbsp;sequencePos)</code></th>
<td class="colLast">
<div class="block">Returns the Atom of the specified structure that is aligned in the
 sequence alignment position specified.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#getAverageResidueDistances(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAverageResidueDistances</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">The average residue distance Matrix contains the average distance from
 each residue to all other residues aligned with it.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentScorer.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentScorer.html#getAvgTMScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAvgTMScore</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Calculates the average TMScore of all the possible pairwise structure
 comparisons of the given alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#getBlockForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,int)">getBlockForSequencePosition</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                           <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;mapSeqToStruct,
                           int&nbsp;sequencePos)</code></th>
<td class="colLast">
<div class="block">Returns the block number of a specified position in the sequence
 alignment, given the mapping from structure to function.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getBlockSequenceAlignment</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;ma)</code></th>
<td class="colLast">
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)">getBlockSequenceAlignment</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
                         <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;mapSeqToStruct)</code></th>
<td class="colLast">
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static org.forester.phylogeny.Phylogeny</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#getHSDMTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getHSDMTree</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></th>
<td class="colLast">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using
 dissimilarity scores (DS), based in SDM Substitution Matrix (ideal for
 distantly related proteins, structure-derived) and the Neighbor Joining
 algorithm from forester.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static org.forester.phylogeny.Phylogeny</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#getKimuraTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getKimuraTree</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></th>
<td class="colLast">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using Kimura
 distances and the Neighbor Joining algorithm from forester.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentScorer.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentScorer.html#getMCScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment,double,double,double)">getMCScore</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
          double&nbsp;gapOpen,
          double&nbsp;gapExtension,
          double&nbsp;dCutoff)</code></th>
<td class="colLast">
<div class="block">Calculates the MC score, specific for the MultipleAlignment algorithm.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentScorer.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentScorer.html#getRefRMSD(org.biojava.nbio.structure.align.multiple.MultipleAlignment,int)">getRefRMSD</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
          int&nbsp;ref)</code></th>
<td class="colLast">
<div class="block">/** Calculates the average RMSD from all structures to a reference s
 tructure, given a set of superimposed atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentScorer.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentScorer.html#getRefTMScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment,int)">getRefTMScore</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
             int&nbsp;ref)</code></th>
<td class="colLast">
<div class="block">Calculates the average TMScore from all structures to a reference
 structure, given a set of superimposed atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentScorer.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentScorer.html#getRMSD(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getRMSD</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Calculates the RMSD of all-to-all structure comparisons (distances) of
 the given MultipleAlignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#getRMSDMatrix(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getRMSDMatrix</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">Calculate the RMSD matrix of a MultipleAlignment, that is, entry (i,j) of
 the matrix contains the RMSD between structures i and j.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentDisplay.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentDisplay.html#getRotatedAtoms(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getRotatedAtoms</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code></th>
<td class="colLast">
<div class="block">New structures are downloaded if they were not cached in the alignment
 and they are entirely transformed here with the superposition information
 in the Multiple Alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getSequenceAlignment</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">Calculate the sequence alignment Strings for the whole alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)">getSequenceAlignment</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
                    <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;&nbsp;mapSeqToStruct)</code></th>
<td class="colLast">
<div class="block">Calculate the sequence alignment Strings for the whole alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static org.forester.phylogeny.Phylogeny</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#getStructuralTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getStructuralTree</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></th>
<td class="colLast">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using RMSD
 distances and the Neighbor Joining algorithm from forester.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CoreSuperimposer.</span><code><span class="memberNameLink"><a href="../util/CoreSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">superimpose</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleSuperimposer.</span><code><span class="memberNameLink"><a href="../util/MultipleSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">superimpose</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Superimpose all structures from a <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ReferenceSuperimposer.</span><code><span class="memberNameLink"><a href="../util/ReferenceSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">superimpose</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentWriter.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentWriter.html#to3DFormat(org.biojava.nbio.structure.align.multiple.MultipleAlignment,int,int)">to3DFormat</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
          int&nbsp;queryIndex,
          int&nbsp;templateIndex)</code></th>
<td class="colLast">
<div class="block">Outputs a pairwise alignment in I-TASSER's 3D Format for target-template
 alignment. http://zhanglab.ccmb.med.umich.edu/I-TASSER/option4.html</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentWriter.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentWriter.html#toAlignedResidues(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">toAlignedResidues</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code></th>
<td class="colLast">
<div class="block">Converts the alignment to its simplest form: a list of groups of aligned
 residues.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentWriter.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentWriter.html#toFASTA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">toFASTA</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Converts the <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> into a multiple sequence alignment
 String in FASTA format.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentWriter.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentWriter.html#toFatCat(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">toFatCat</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Converts the <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> into a FatCat String format.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#toMultimodelStructure(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)">toMultimodelStructure</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                     <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformedAtoms)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;<a href="../../../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,&#8203;<a href="../../../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#toProteinMSA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">toProteinMSA</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></th>
<td class="colLast">
<div class="block">Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid
 residues.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentWriter.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentWriter.html#toTransformMatrices(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">toTransformMatrices</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Converts the transformation Matrices of the alignment into a String
 output.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentTools.</span><code><span class="memberNameLink"><a href="../util/MultipleAlignmentTools.html#transformAtoms(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">transformAtoms</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">Transforms atoms according to the superposition stored in the alignment.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.quaternary">
<!--   -->
</a>
<h3>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">QsAlignResult.</span><code><span class="memberNameLink"><a href="../../quaternary/QsAlignResult.html#getAlignment()">getAlignment</a></span>()</code></th>
<td class="colLast">
<div class="block">The alignment that specifies the residue equivalencies of the equivalent
 Subunits.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">QsAlignResult.</span><code><span class="memberNameLink"><a href="../../quaternary/QsAlignResult.html#setAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">setAlignment</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></th>
<td class="colLast">
<div class="block">The alignment that specifies the residue equivalencies of the equivalent
 Subunits.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.align.xml">
<!--   -->
</a>
<h3>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../xml/package-summary.html">org.biojava.nbio.structure.align.xml</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../xml/package-summary.html">org.biojava.nbio.structure.align.xml</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MultipleAlignmentXMLConverter.</span><code><span class="memberNameLink"><a href="../../xml/MultipleAlignmentXMLConverter.html#printXMLalignment(org.biojava.nbio.core.util.PrettyXMLWriter,org.biojava.nbio.structure.align.multiple.MultipleAlignment)">printXMLalignment</a></span>&#8203;(<a href="../../../../core/util/PrettyXMLWriter.html" title="class in org.biojava.nbio.core.util">PrettyXMLWriter</a>&nbsp;xml,
                 <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.cluster">
<!--   -->
</a>
<h3>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../cluster/package-summary.html">org.biojava.nbio.structure.cluster</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SubunitCluster.</span><code><span class="memberNameLink"><a href="../../../cluster/SubunitCluster.html#getMultipleAlignment()">getMultipleAlignment</a></span>()</code></th>
<td class="colLast">
<div class="block">The multiple alignment is calculated from the equivalent residues in the
 SubunitCluster.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.symmetry.internal">
<!--   -->
</a>
<h3>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymmResult.</span><code><span class="memberNameLink"><a href="../../../symmetry/internal/CeSymmResult.html#getMultipleAlignment()">getMultipleAlignment</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmOptimizer.</span><code><span class="memberNameLink"><a href="../../../symmetry/internal/SymmOptimizer.html#optimize()">optimize</a></span>()</code></th>
<td class="colLast">
<div class="block">Optimization method based in a Monte-Carlo approach.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GraphComponentRefiner.</span><code><span class="memberNameLink"><a href="../../../symmetry/internal/GraphComponentRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)">refine</a></span>&#8203;(<a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceFunctionRefiner.</span><code><span class="memberNameLink"><a href="../../../symmetry/internal/SequenceFunctionRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)">refine</a></span>&#8203;(<a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryRefiner.</span><code><span class="memberNameLink"><a href="../../../symmetry/internal/SymmetryRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)">refine</a></span>&#8203;(<a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code></th>
<td class="colLast">
<div class="block">Returns a refined symmetry alignment, where the repeat residues are
 aligned consistently in a MultipleAlignment.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CeSymmResult.</span><code><span class="memberNameLink"><a href="../../../symmetry/internal/CeSymmResult.html#setMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">setMultipleAlignment</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multipleAlignment)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.nbio.structure.symmetry.utils">
<!--   -->
</a>
<h3>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../../../symmetry/utils/SymmetryTools.html#fromAFP(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)">fromAFP</a></span>&#8203;(<a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;symm,
       <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></th>
<td class="colLast">
<div class="block">Converts a refined symmetry AFPChain alignment into the standard
 representation of symmetry in a MultipleAlignment, that contains the
 entire Atom array of the strcuture and the symmetric repeats are orgaized
 in different rows in a single Block.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../../../symmetry/utils/SymmetryTools.html#toFullAlignment(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">toFullAlignment</a></span>&#8203;(<a href="../../../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symm)</code></th>
<td class="colLast">
<div class="block">Method that converts a repeats symmetric alignment into an alignment of
 whole structures.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../../../symmetry/utils/SymmetryTools.html#toRepeatsAlignment(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">toRepeatsAlignment</a></span>&#8203;(<a href="../../../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;result)</code></th>
<td class="colLast">
<div class="block">Method that converts a symmetry alignment into an alignment of the
 repeats only, as new independent structures.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../../../symmetry/utils/SymmetryTools.html#updateSymmetryScores(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">updateSymmetryScores</a></span>&#8203;(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;symm)</code></th>
<td class="colLast">
<div class="block">Update the scores (TM-score and RMSD) of a symmetry multiple alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SymmetryTools.</span><code><span class="memberNameLink"><a href="../../../symmetry/utils/SymmetryTools.html#updateSymmetryTransformation(org.biojava.nbio.structure.symmetry.internal.SymmetryAxes,org.biojava.nbio.structure.align.multiple.MultipleAlignment)">updateSymmetryTransformation</a></span>&#8203;(<a href="../../../symmetry/internal/SymmetryAxes.html" title="class in org.biojava.nbio.structure.symmetry.internal">SymmetryAxes</a>&nbsp;axes,
                            <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></th>
<td class="colLast">
<div class="block">Calculates the set of symmetry operation Matrices (transformations) of
 the new alignment, based on the symmetry relations in the SymmetryAxes
 object.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2022 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
