<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span>package org.biojava.nbio.aaproperties;<a name="line.21"></a>
<span class="sourceLineNo">022</span><a name="line.22"></a>
<span class="sourceLineNo">023</span>import org.biojava.nbio.aaproperties.xml.AminoAcidCompositionTable;<a name="line.23"></a>
<span class="sourceLineNo">024</span>import org.biojava.nbio.aaproperties.xml.CaseFreeAminoAcidCompoundSet;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.nbio.core.exceptions.CompoundNotFoundException;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.core.sequence.ProteinSequence;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.biojava.nbio.core.sequence.io.*;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import org.biojava.nbio.core.sequence.template.CompoundSet;<a name="line.30"></a>
<span class="sourceLineNo">031</span><a name="line.31"></a>
<span class="sourceLineNo">032</span>import java.io.File;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import java.io.FileInputStream;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import java.io.IOException;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import java.io.PrintStream;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import java.util.ArrayList;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import java.util.LinkedHashMap;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import java.util.List;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import java.util.Map;<a name="line.39"></a>
<span class="sourceLineNo">040</span>import java.util.Map.Entry;<a name="line.40"></a>
<span class="sourceLineNo">041</span><a name="line.41"></a>
<span class="sourceLineNo">042</span><a name="line.42"></a>
<span class="sourceLineNo">043</span>public class CommandPrompt {<a name="line.43"></a>
<span class="sourceLineNo">044</span><a name="line.44"></a>
<span class="sourceLineNo">045</span>        /**<a name="line.45"></a>
<span class="sourceLineNo">046</span>         * The main method<a name="line.46"></a>
<span class="sourceLineNo">047</span>         * @param args<a name="line.47"></a>
<span class="sourceLineNo">048</span>         *      See showHelp for a list of available arguments<a name="line.48"></a>
<span class="sourceLineNo">049</span>         * @throws Exception<a name="line.49"></a>
<span class="sourceLineNo">050</span>         *  To handle exception thrown by reading of XML files<a name="line.50"></a>
<span class="sourceLineNo">051</span>         */<a name="line.51"></a>
<span class="sourceLineNo">052</span>        public static void main(String[] args) throws Exception{<a name="line.52"></a>
<span class="sourceLineNo">053</span>                run(args);<a name="line.53"></a>
<span class="sourceLineNo">054</span>        }<a name="line.54"></a>
<span class="sourceLineNo">055</span><a name="line.55"></a>
<span class="sourceLineNo">056</span>        private static AminoAcidCompositionTable checkForValidityAndObtainAATable(String inputLocation, int propertyListSize, String aminoAcidCompositionLocation,<a name="line.56"></a>
<span class="sourceLineNo">057</span>                        String elementMassLocation) throws Exception{<a name="line.57"></a>
<span class="sourceLineNo">058</span>                if(inputLocation == null) {<a name="line.58"></a>
<span class="sourceLineNo">059</span>                        showHelp();<a name="line.59"></a>
<span class="sourceLineNo">060</span>                        throw new Error("Please do provide location of input file.");<a name="line.60"></a>
<span class="sourceLineNo">061</span>                }<a name="line.61"></a>
<span class="sourceLineNo">062</span>                if(propertyListSize == 0){<a name="line.62"></a>
<span class="sourceLineNo">063</span>                        showHelp();<a name="line.63"></a>
<span class="sourceLineNo">064</span>                        throw new Error("Please at least specify a property to compute.");<a name="line.64"></a>
<span class="sourceLineNo">065</span>                }<a name="line.65"></a>
<span class="sourceLineNo">066</span>                AminoAcidCompositionTable aaTable = null;<a name="line.66"></a>
<span class="sourceLineNo">067</span>                if(aminoAcidCompositionLocation != null &amp;&amp; elementMassLocation == null){<a name="line.67"></a>
<span class="sourceLineNo">068</span>                        aaTable = PeptideProperties.obtainAminoAcidCompositionTable(new File(aminoAcidCompositionLocation));<a name="line.68"></a>
<span class="sourceLineNo">069</span>                }else if(aminoAcidCompositionLocation != null &amp;&amp; elementMassLocation != null){<a name="line.69"></a>
<span class="sourceLineNo">070</span>                        aaTable = PeptideProperties.obtainAminoAcidCompositionTable(new File(aminoAcidCompositionLocation, elementMassLocation));<a name="line.70"></a>
<span class="sourceLineNo">071</span>                }else if(aminoAcidCompositionLocation == null &amp;&amp; elementMassLocation != null){<a name="line.71"></a>
<span class="sourceLineNo">072</span>                        throw new Error("You have define the location of Element Mass XML file. Please also define the location of Amino Acid Composition XML file");<a name="line.72"></a>
<span class="sourceLineNo">073</span>                }<a name="line.73"></a>
<span class="sourceLineNo">074</span>                return aaTable;<a name="line.74"></a>
<span class="sourceLineNo">075</span>        }<a name="line.75"></a>
<span class="sourceLineNo">076</span><a name="line.76"></a>
<span class="sourceLineNo">077</span>        private static void readInputAndGenerateOutput(String outputLocation, List&lt;Character&gt; propertyList, List&lt;Character&gt; specificList,<a name="line.77"></a>
<span class="sourceLineNo">078</span>                        String delimiter, String inputLocation, AminoAcidCompositionTable aaTable, int decimalPlace) throws Exception{<a name="line.78"></a>
<span class="sourceLineNo">079</span>                PrintStream output;<a name="line.79"></a>
<span class="sourceLineNo">080</span>                if(outputLocation != null)<a name="line.80"></a>
<span class="sourceLineNo">081</span>                        output = new PrintStream(new File(outputLocation));<a name="line.81"></a>
<span class="sourceLineNo">082</span>                else<a name="line.82"></a>
<span class="sourceLineNo">083</span>                        output = System.out;<a name="line.83"></a>
<span class="sourceLineNo">084</span>                printHeader(output, propertyList, specificList, delimiter);<a name="line.84"></a>
<span class="sourceLineNo">085</span>                LinkedHashMap&lt;String, ProteinSequence&gt; a = readInputFile(inputLocation, aaTable);<a name="line.85"></a>
<span class="sourceLineNo">086</span>                //Need for the last sequence<a name="line.86"></a>
<span class="sourceLineNo">087</span>                for(Entry&lt;String, ProteinSequence&gt; entry:a.entrySet()){<a name="line.87"></a>
<span class="sourceLineNo">088</span>                        compute(output, entry.getValue().getOriginalHeader(), entry.getValue().getSequenceAsString().trim(), delimiter, aaTable, propertyList, specificList,<a name="line.88"></a>
<span class="sourceLineNo">089</span>                                        decimalPlace);<a name="line.89"></a>
<span class="sourceLineNo">090</span>                }<a name="line.90"></a>
<span class="sourceLineNo">091</span>                output.close();<a name="line.91"></a>
<span class="sourceLineNo">092</span>        }<a name="line.92"></a>
<span class="sourceLineNo">093</span><a name="line.93"></a>
<span class="sourceLineNo">094</span>        public static void run(String[] args) throws Exception{<a name="line.94"></a>
<span class="sourceLineNo">095</span>                /*<a name="line.95"></a>
<span class="sourceLineNo">096</span>                 * Parse input arguments<a name="line.96"></a>
<span class="sourceLineNo">097</span>                 */<a name="line.97"></a>
<span class="sourceLineNo">098</span>                List&lt;Character&gt; propertyList = new ArrayList&lt;Character&gt;();<a name="line.98"></a>
<span class="sourceLineNo">099</span>                List&lt;Character&gt; specificList = new ArrayList&lt;Character&gt;();<a name="line.99"></a>
<span class="sourceLineNo">100</span>                String inputLocation = null;<a name="line.100"></a>
<span class="sourceLineNo">101</span>                String outputLocation = null;<a name="line.101"></a>
<span class="sourceLineNo">102</span>                String aminoAcidCompositionLocation = null;<a name="line.102"></a>
<span class="sourceLineNo">103</span>                String elementMassLocation = null;<a name="line.103"></a>
<span class="sourceLineNo">104</span>                String delimiter = ",";<a name="line.104"></a>
<span class="sourceLineNo">105</span>                int decimalPlace = 4;<a name="line.105"></a>
<span class="sourceLineNo">106</span><a name="line.106"></a>
<span class="sourceLineNo">107</span>                for(int i = 0; i &lt; args.length; i++){<a name="line.107"></a>
<span class="sourceLineNo">108</span>                        if(args[i].charAt(0) != '-' || args[i].length() != 2){<a name="line.108"></a>
<span class="sourceLineNo">109</span>                                showHelp();<a name="line.109"></a>
<span class="sourceLineNo">110</span>                                throw new Error("Unknown option: " + args[i]);<a name="line.110"></a>
<span class="sourceLineNo">111</span>                        }else{<a name="line.111"></a>
<span class="sourceLineNo">112</span>                                switch(args[i].charAt(1)){<a name="line.112"></a>
<span class="sourceLineNo">113</span>                                //Required<a name="line.113"></a>
<span class="sourceLineNo">114</span>                                case 'i': inputLocation = args[++i]; break;<a name="line.114"></a>
<span class="sourceLineNo">115</span>                                //Optional<a name="line.115"></a>
<span class="sourceLineNo">116</span>                                case 'o': outputLocation = args[++i]; break;<a name="line.116"></a>
<span class="sourceLineNo">117</span>                                case 'f':<a name="line.117"></a>
<span class="sourceLineNo">118</span>                                        i++;<a name="line.118"></a>
<span class="sourceLineNo">119</span>                                        if(args[i].equalsIgnoreCase("csv")) delimiter = ",";<a name="line.119"></a>
<span class="sourceLineNo">120</span>                                        else if(args[i].equalsIgnoreCase("tsv")) delimiter = "\t";<a name="line.120"></a>
<span class="sourceLineNo">121</span>                                        else throw new Error("Invalid value for -f: " + args[i] + ". Please choose either csv or tsv only.");<a name="line.121"></a>
<span class="sourceLineNo">122</span>                                        break;<a name="line.122"></a>
<span class="sourceLineNo">123</span>                                case 'x': aminoAcidCompositionLocation = args[++i]; break;<a name="line.123"></a>
<span class="sourceLineNo">124</span>                                case 'y': elementMassLocation = args[++i]; break;<a name="line.124"></a>
<span class="sourceLineNo">125</span>                                case 'd': decimalPlace = Integer.parseInt(args[++i]); break;<a name="line.125"></a>
<span class="sourceLineNo">126</span>                                //Properties<a name="line.126"></a>
<span class="sourceLineNo">127</span>                                case 'a':<a name="line.127"></a>
<span class="sourceLineNo">128</span>                                        propertyList.add('1');<a name="line.128"></a>
<span class="sourceLineNo">129</span>                                        propertyList.add('2');<a name="line.129"></a>
<span class="sourceLineNo">130</span>                                        propertyList.add('3');<a name="line.130"></a>
<span class="sourceLineNo">131</span>                                        propertyList.add('4');<a name="line.131"></a>
<span class="sourceLineNo">132</span>                                        propertyList.add('5');<a name="line.132"></a>
<span class="sourceLineNo">133</span>                                        propertyList.add('6');<a name="line.133"></a>
<span class="sourceLineNo">134</span>                                        propertyList.add('7');<a name="line.134"></a>
<span class="sourceLineNo">135</span>                                        propertyList.add('8');<a name="line.135"></a>
<span class="sourceLineNo">136</span>                                        propertyList.add('9');<a name="line.136"></a>
<span class="sourceLineNo">137</span>                                        break;<a name="line.137"></a>
<span class="sourceLineNo">138</span>                                case '1': propertyList.add('1'); break;<a name="line.138"></a>
<span class="sourceLineNo">139</span>                                case '2': propertyList.add('2'); break;<a name="line.139"></a>
<span class="sourceLineNo">140</span>                                case '3': propertyList.add('3'); break;<a name="line.140"></a>
<span class="sourceLineNo">141</span>                                case '4': propertyList.add('4'); break;<a name="line.141"></a>
<span class="sourceLineNo">142</span>                                case '5': propertyList.add('5'); break;<a name="line.142"></a>
<span class="sourceLineNo">143</span>                                case '6': propertyList.add('6'); break;<a name="line.143"></a>
<span class="sourceLineNo">144</span>                                case '7': propertyList.add('7'); break;<a name="line.144"></a>
<span class="sourceLineNo">145</span>                                case '8': propertyList.add('8'); break;<a name="line.145"></a>
<span class="sourceLineNo">146</span>                                case '9': propertyList.add('9'); break;<a name="line.146"></a>
<span class="sourceLineNo">147</span>                                case '0':<a name="line.147"></a>
<span class="sourceLineNo">148</span>                                        propertyList.add('0');<a name="line.148"></a>
<span class="sourceLineNo">149</span>                                        i++;<a name="line.149"></a>
<span class="sourceLineNo">150</span>                                        if(args[i].length() != 1) throw new Error("Invalid value: " + args[i] + ". Amino Acid Symbol should be of single character");<a name="line.150"></a>
<span class="sourceLineNo">151</span>                                        specificList.add(args[i].toUpperCase().charAt(0));<a name="line.151"></a>
<span class="sourceLineNo">152</span>                                        break;<a name="line.152"></a>
<span class="sourceLineNo">153</span>                                default:<a name="line.153"></a>
<span class="sourceLineNo">154</span>                                        showHelp();<a name="line.154"></a>
<span class="sourceLineNo">155</span>                                        throw new Error("Unknown option: " + args[i]);<a name="line.155"></a>
<span class="sourceLineNo">156</span>                                }<a name="line.156"></a>
<span class="sourceLineNo">157</span>                        }<a name="line.157"></a>
<span class="sourceLineNo">158</span>                }<a name="line.158"></a>
<span class="sourceLineNo">159</span><a name="line.159"></a>
<span class="sourceLineNo">160</span>                /*<a name="line.160"></a>
<span class="sourceLineNo">161</span>                 * Check for validity of input arguments<a name="line.161"></a>
<span class="sourceLineNo">162</span>                 */<a name="line.162"></a>
<span class="sourceLineNo">163</span>                AminoAcidCompositionTable aaTable = checkForValidityAndObtainAATable(inputLocation, propertyList.size(), aminoAcidCompositionLocation,<a name="line.163"></a>
<span class="sourceLineNo">164</span>                                elementMassLocation);<a name="line.164"></a>
<span class="sourceLineNo">165</span><a name="line.165"></a>
<span class="sourceLineNo">166</span>                /*<a name="line.166"></a>
<span class="sourceLineNo">167</span>                 * Read input file and generate output<a name="line.167"></a>
<span class="sourceLineNo">168</span>                 */<a name="line.168"></a>
<span class="sourceLineNo">169</span>                readInputAndGenerateOutput(outputLocation, propertyList, specificList, delimiter, inputLocation, aaTable, decimalPlace);<a name="line.169"></a>
<span class="sourceLineNo">170</span>        }<a name="line.170"></a>
<span class="sourceLineNo">171</span><a name="line.171"></a>
<span class="sourceLineNo">172</span>        private static LinkedHashMap&lt;String, ProteinSequence&gt; readInputFile(String inputLocation, AminoAcidCompositionTable aaTable) throws Exception{<a name="line.172"></a>
<span class="sourceLineNo">173</span>                FileInputStream inStream = new FileInputStream(inputLocation);<a name="line.173"></a>
<span class="sourceLineNo">174</span>                CompoundSet&lt;AminoAcidCompound&gt;  set;<a name="line.174"></a>
<span class="sourceLineNo">175</span>                if(aaTable == null){<a name="line.175"></a>
<span class="sourceLineNo">176</span>                        set = CaseFreeAminoAcidCompoundSet.getAminoAcidCompoundSet();<a name="line.176"></a>
<span class="sourceLineNo">177</span>                }else{<a name="line.177"></a>
<span class="sourceLineNo">178</span>                        set = aaTable.getAminoAcidCompoundSet();<a name="line.178"></a>
<span class="sourceLineNo">179</span>                }<a name="line.179"></a>
<span class="sourceLineNo">180</span>                LinkedHashMap&lt;String, ProteinSequence&gt; ret;<a name="line.180"></a>
<span class="sourceLineNo">181</span>                if ( inputLocation.toLowerCase().contains(".gb")) {<a name="line.181"></a>
<span class="sourceLineNo">182</span>                        GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt; genbankReader = new GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt;(<a name="line.182"></a>
<span class="sourceLineNo">183</span>                                        inStream, new GenericGenbankHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),<a name="line.183"></a>
<span class="sourceLineNo">184</span>                                        new ProteinSequenceCreator(set));<a name="line.184"></a>
<span class="sourceLineNo">185</span>                        ret = genbankReader.process();<a name="line.185"></a>
<span class="sourceLineNo">186</span><a name="line.186"></a>
<span class="sourceLineNo">187</span><a name="line.187"></a>
<span class="sourceLineNo">188</span>                } else {<a name="line.188"></a>
<span class="sourceLineNo">189</span>                        FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; fastaReader = new FastaReader&lt;ProteinSequence, AminoAcidCompound&gt;(<a name="line.189"></a>
<span class="sourceLineNo">190</span>                                        inStream, new GenericFastaHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),<a name="line.190"></a>
<span class="sourceLineNo">191</span>                                        new ProteinSequenceCreator(set));<a name="line.191"></a>
<span class="sourceLineNo">192</span>                        ret = fastaReader.process();<a name="line.192"></a>
<span class="sourceLineNo">193</span><a name="line.193"></a>
<span class="sourceLineNo">194</span>                }<a name="line.194"></a>
<span class="sourceLineNo">195</span>                return ret;<a name="line.195"></a>
<span class="sourceLineNo">196</span>        }<a name="line.196"></a>
<span class="sourceLineNo">197</span><a name="line.197"></a>
<span class="sourceLineNo">198</span>        public enum PropertyName{MolecularWeight, Absorbance_True, Absorbance_False, ExtinctionCoefficient_True, ExtinctionCoefficient_False,<a name="line.198"></a>
<span class="sourceLineNo">199</span>                InstabilityIndex, ApliphaticIndex, AverageHydropathyValue, IsoelectricPoint, NetCharge_pH_7, A, R,<a name="line.199"></a>
<span class="sourceLineNo">200</span>                N, D, C, E, Q, G, H, I, L,<a name="line.200"></a>
<span class="sourceLineNo">201</span>                K, M, F, P, S, T, W, Y, V};<a name="line.201"></a>
<span class="sourceLineNo">202</span><a name="line.202"></a>
<span class="sourceLineNo">203</span>        private static void printHeader(PrintStream output, List&lt;Character&gt; propertyList, List&lt;Character&gt; specificList, String delimiter) throws IOException{<a name="line.203"></a>
<span class="sourceLineNo">204</span>                int specificCount = 0;<a name="line.204"></a>
<span class="sourceLineNo">205</span>                /*<a name="line.205"></a>
<span class="sourceLineNo">206</span>                 * 1 Molecular weight<a name="line.206"></a>
<span class="sourceLineNo">207</span>                 * 2 Absorbance (assumed Cys reduced and assume Cys to form cystines)<a name="line.207"></a>
<span class="sourceLineNo">208</span>                 * 3 Extinction coefficient (assumed Cys reduced and assume Cys to form cystines)<a name="line.208"></a>
<span class="sourceLineNo">209</span>                 * 4 Instability index<a name="line.209"></a>
<span class="sourceLineNo">210</span>                 * 5 Apliphatic index<a name="line.210"></a>
<span class="sourceLineNo">211</span>                 * 6 Average hydropathy value<a name="line.211"></a>
<span class="sourceLineNo">212</span>                 * 7 Isoelectric point<a name="line.212"></a>
<span class="sourceLineNo">213</span>                 * 8 Net charge at pH 7<a name="line.213"></a>
<span class="sourceLineNo">214</span>                 * 9 Composition of the 20 standard amino acid<a name="line.214"></a>
<span class="sourceLineNo">215</span>                 * 0 Composition of the specific amino acid<a name="line.215"></a>
<span class="sourceLineNo">216</span>                 */<a name="line.216"></a>
<span class="sourceLineNo">217</span>                List&lt;String&gt; sList = new ArrayList&lt;String&gt;();<a name="line.217"></a>
<span class="sourceLineNo">218</span>                sList.add("SequenceName");<a name="line.218"></a>
<span class="sourceLineNo">219</span>                for(Character c:propertyList){<a name="line.219"></a>
<span class="sourceLineNo">220</span>                        switch(c){<a name="line.220"></a>
<span class="sourceLineNo">221</span>                        case '1': sList.add(PropertyName.MolecularWeight.toString()); break;<a name="line.221"></a>
<span class="sourceLineNo">222</span>                        case '2': sList.add(PropertyName.Absorbance_True.toString()); sList.add(PropertyName.Absorbance_False.toString()); break;<a name="line.222"></a>
<span class="sourceLineNo">223</span>                        case '3': sList.add(PropertyName.ExtinctionCoefficient_True.toString()); sList.add(PropertyName.ExtinctionCoefficient_False.toString()); break;<a name="line.223"></a>
<span class="sourceLineNo">224</span>                        case '4': sList.add(PropertyName.InstabilityIndex.toString()); break;<a name="line.224"></a>
<span class="sourceLineNo">225</span>                        case '5': sList.add(PropertyName.ApliphaticIndex.toString()); break;<a name="line.225"></a>
<span class="sourceLineNo">226</span>                        case '6': sList.add(PropertyName.AverageHydropathyValue.toString()); break;<a name="line.226"></a>
<span class="sourceLineNo">227</span>                        case '7': sList.add(PropertyName.IsoelectricPoint.toString()); break;<a name="line.227"></a>
<span class="sourceLineNo">228</span>                        case '8': sList.add(PropertyName.NetCharge_pH_7.toString()); break;<a name="line.228"></a>
<span class="sourceLineNo">229</span>                        case '9':<a name="line.229"></a>
<span class="sourceLineNo">230</span>                                sList.add(PropertyName.A.toString()); sList.add(PropertyName.R.toString());<a name="line.230"></a>
<span class="sourceLineNo">231</span>                                sList.add(PropertyName.N.toString()); sList.add(PropertyName.D.toString());<a name="line.231"></a>
<span class="sourceLineNo">232</span>                                sList.add(PropertyName.C.toString()); sList.add(PropertyName.E.toString());<a name="line.232"></a>
<span class="sourceLineNo">233</span>                                sList.add(PropertyName.Q.toString()); sList.add(PropertyName.G.toString());<a name="line.233"></a>
<span class="sourceLineNo">234</span>                                sList.add(PropertyName.H.toString()); sList.add(PropertyName.I.toString());<a name="line.234"></a>
<span class="sourceLineNo">235</span>                                sList.add(PropertyName.L.toString()); sList.add(PropertyName.K.toString());<a name="line.235"></a>
<span class="sourceLineNo">236</span>                                sList.add(PropertyName.M.toString()); sList.add(PropertyName.F.toString());<a name="line.236"></a>
<span class="sourceLineNo">237</span>                                sList.add(PropertyName.P.toString()); sList.add(PropertyName.S.toString());<a name="line.237"></a>
<span class="sourceLineNo">238</span>                                sList.add(PropertyName.T.toString()); sList.add(PropertyName.W.toString());<a name="line.238"></a>
<span class="sourceLineNo">239</span>                                sList.add(PropertyName.Y.toString()); sList.add(PropertyName.V.toString());<a name="line.239"></a>
<span class="sourceLineNo">240</span>                                break;<a name="line.240"></a>
<span class="sourceLineNo">241</span>                        case '0': sList.add("" + specificList.get(specificCount++)); break;<a name="line.241"></a>
<span class="sourceLineNo">242</span>                        }<a name="line.242"></a>
<span class="sourceLineNo">243</span>                }<a name="line.243"></a>
<span class="sourceLineNo">244</span>                for(int i = 0; i &lt; sList.size(); i++){<a name="line.244"></a>
<span class="sourceLineNo">245</span>                        if(i != 0) output.print(delimiter);<a name="line.245"></a>
<span class="sourceLineNo">246</span>                        output.print(sList.get(i));<a name="line.246"></a>
<span class="sourceLineNo">247</span>                }<a name="line.247"></a>
<span class="sourceLineNo">248</span>                output.println();<a name="line.248"></a>
<span class="sourceLineNo">249</span>                output.flush();<a name="line.249"></a>
<span class="sourceLineNo">250</span>        }<a name="line.250"></a>
<span class="sourceLineNo">251</span><a name="line.251"></a>
<span class="sourceLineNo">252</span>        private static void compute(PrintStream output, String header, String sequence, String delimiter,<a name="line.252"></a>
<span class="sourceLineNo">253</span>                        AminoAcidCompositionTable aaTable, List&lt;Character&gt; propertyList, List&lt;Character&gt; specificList, int decimalPlace) throws CompoundNotFoundException{<a name="line.253"></a>
<span class="sourceLineNo">254</span>                /*<a name="line.254"></a>
<span class="sourceLineNo">255</span>                 * 1 Molecular weight<a name="line.255"></a>
<span class="sourceLineNo">256</span>                 * 2 Absorbance (assumed Cys reduced and assume Cys to form cystines)<a name="line.256"></a>
<span class="sourceLineNo">257</span>                 * 3 Extinction coefficient<a name="line.257"></a>
<span class="sourceLineNo">258</span>                 * 4 Instability index<a name="line.258"></a>
<span class="sourceLineNo">259</span>                 * 5 Apliphatic index<a name="line.259"></a>
<span class="sourceLineNo">260</span>                 * 6 Average hydropathy value<a name="line.260"></a>
<span class="sourceLineNo">261</span>                 * 7 Isoelectric point<a name="line.261"></a>
<span class="sourceLineNo">262</span>                 * 8 Net charge at pH 7<a name="line.262"></a>
<span class="sourceLineNo">263</span>                 * 9 Composition of the 20 standard amino acid<a name="line.263"></a>
<span class="sourceLineNo">264</span>                 * 0 Composition of the specific amino acid<a name="line.264"></a>
<span class="sourceLineNo">265</span>                 */<a name="line.265"></a>
<span class="sourceLineNo">266</span>                ProteinSequence pSequence;<a name="line.266"></a>
<span class="sourceLineNo">267</span>                CompoundSet&lt;AminoAcidCompound&gt; aaSet;<a name="line.267"></a>
<span class="sourceLineNo">268</span>                if(aaTable != null){<a name="line.268"></a>
<span class="sourceLineNo">269</span>                        sequence = Utils.checkSequence(sequence, aaTable.getSymbolSet());<a name="line.269"></a>
<span class="sourceLineNo">270</span>                        pSequence = new ProteinSequence(sequence, aaTable.getAminoAcidCompoundSet());<a name="line.270"></a>
<span class="sourceLineNo">271</span>                        aaSet = aaTable.getAminoAcidCompoundSet();<a name="line.271"></a>
<span class="sourceLineNo">272</span>                }else{<a name="line.272"></a>
<span class="sourceLineNo">273</span>                        sequence = Utils.checkSequence(sequence);<a name="line.273"></a>
<span class="sourceLineNo">274</span>                        pSequence = new ProteinSequence(sequence);<a name="line.274"></a>
<span class="sourceLineNo">275</span>                        aaSet = AminoAcidCompoundSet.getAminoAcidCompoundSet();<a name="line.275"></a>
<span class="sourceLineNo">276</span>                }<a name="line.276"></a>
<span class="sourceLineNo">277</span>                IPeptideProperties pp = new PeptidePropertiesImpl();<a name="line.277"></a>
<span class="sourceLineNo">278</span><a name="line.278"></a>
<span class="sourceLineNo">279</span>                int specificCount = 0;<a name="line.279"></a>
<span class="sourceLineNo">280</span>                List&lt;Double&gt; dList = new ArrayList&lt;Double&gt;();<a name="line.280"></a>
<span class="sourceLineNo">281</span>                for(Character c:propertyList){<a name="line.281"></a>
<span class="sourceLineNo">282</span>                        switch(c){<a name="line.282"></a>
<span class="sourceLineNo">283</span>                        case '1':<a name="line.283"></a>
<span class="sourceLineNo">284</span>                                if(aaTable == null)<a name="line.284"></a>
<span class="sourceLineNo">285</span>                                        dList.add(pp.getMolecularWeight(pSequence));<a name="line.285"></a>
<span class="sourceLineNo">286</span>                                else<a name="line.286"></a>
<span class="sourceLineNo">287</span>                                        dList.add(pp.getMolecularWeight(pSequence));<a name="line.287"></a>
<span class="sourceLineNo">288</span>                                break;<a name="line.288"></a>
<span class="sourceLineNo">289</span>                        case '2':<a name="line.289"></a>
<span class="sourceLineNo">290</span>                                dList.add(pp.getAbsorbance(pSequence, true));<a name="line.290"></a>
<span class="sourceLineNo">291</span>                                dList.add(pp.getAbsorbance(pSequence, false));<a name="line.291"></a>
<span class="sourceLineNo">292</span>                                break;<a name="line.292"></a>
<span class="sourceLineNo">293</span>                        case '3':<a name="line.293"></a>
<span class="sourceLineNo">294</span>                                dList.add(pp.getExtinctionCoefficient(pSequence, true));<a name="line.294"></a>
<span class="sourceLineNo">295</span>                                dList.add(pp.getExtinctionCoefficient(pSequence, false));<a name="line.295"></a>
<span class="sourceLineNo">296</span>                                break;<a name="line.296"></a>
<span class="sourceLineNo">297</span>                        case '4': dList.add(pp.getInstabilityIndex(pSequence)); break;<a name="line.297"></a>
<span class="sourceLineNo">298</span>                        case '5': dList.add(pp.getApliphaticIndex(pSequence)); break;<a name="line.298"></a>
<span class="sourceLineNo">299</span>                        case '6': dList.add(pp.getAvgHydropathy(pSequence)); break;<a name="line.299"></a>
<span class="sourceLineNo">300</span>                        case '7': dList.add(pp.getIsoelectricPoint(pSequence)); break;<a name="line.300"></a>
<span class="sourceLineNo">301</span>                        case '8': dList.add(pp.getNetCharge(pSequence)); break;<a name="line.301"></a>
<span class="sourceLineNo">302</span>                        case '9':<a name="line.302"></a>
<span class="sourceLineNo">303</span>                                Map&lt;AminoAcidCompound, Double&gt; aaCompound2Double = pp.getAAComposition(pSequence);<a name="line.303"></a>
<span class="sourceLineNo">304</span>                                //(A, R, N, D, C, E, Q, G, H, I, L, K, M, F, P, S, T, W, Y, V)<a name="line.304"></a>
<span class="sourceLineNo">305</span>                                dList.add(aaCompound2Double.get(Constraints.A));<a name="line.305"></a>
<span class="sourceLineNo">306</span>                                dList.add(aaCompound2Double.get(Constraints.R));<a name="line.306"></a>
<span class="sourceLineNo">307</span>                                dList.add(aaCompound2Double.get(Constraints.N));<a name="line.307"></a>
<span class="sourceLineNo">308</span>                                dList.add(aaCompound2Double.get(Constraints.D));<a name="line.308"></a>
<span class="sourceLineNo">309</span>                                dList.add(aaCompound2Double.get(Constraints.C));<a name="line.309"></a>
<span class="sourceLineNo">310</span>                                dList.add(aaCompound2Double.get(Constraints.E));<a name="line.310"></a>
<span class="sourceLineNo">311</span>                                dList.add(aaCompound2Double.get(Constraints.Q));<a name="line.311"></a>
<span class="sourceLineNo">312</span>                                dList.add(aaCompound2Double.get(Constraints.G));<a name="line.312"></a>
<span class="sourceLineNo">313</span>                                dList.add(aaCompound2Double.get(Constraints.H));<a name="line.313"></a>
<span class="sourceLineNo">314</span>                                dList.add(aaCompound2Double.get(Constraints.I));<a name="line.314"></a>
<span class="sourceLineNo">315</span>                                dList.add(aaCompound2Double.get(Constraints.L));<a name="line.315"></a>
<span class="sourceLineNo">316</span>                                dList.add(aaCompound2Double.get(Constraints.K));<a name="line.316"></a>
<span class="sourceLineNo">317</span>                                dList.add(aaCompound2Double.get(Constraints.M));<a name="line.317"></a>
<span class="sourceLineNo">318</span>                                dList.add(aaCompound2Double.get(Constraints.F));<a name="line.318"></a>
<span class="sourceLineNo">319</span>                                dList.add(aaCompound2Double.get(Constraints.P));<a name="line.319"></a>
<span class="sourceLineNo">320</span>                                dList.add(aaCompound2Double.get(Constraints.S));<a name="line.320"></a>
<span class="sourceLineNo">321</span>                                dList.add(aaCompound2Double.get(Constraints.T));<a name="line.321"></a>
<span class="sourceLineNo">322</span>                                dList.add(aaCompound2Double.get(Constraints.W));<a name="line.322"></a>
<span class="sourceLineNo">323</span>                                dList.add(aaCompound2Double.get(Constraints.Y));<a name="line.323"></a>
<span class="sourceLineNo">324</span>                                dList.add(aaCompound2Double.get(Constraints.V));<a name="line.324"></a>
<span class="sourceLineNo">325</span>                                break;<a name="line.325"></a>
<span class="sourceLineNo">326</span>                        case '0': dList.add(pp.getEnrichment(pSequence, aaSet.getCompoundForString("" + specificList.get(specificCount++)))); break;<a name="line.326"></a>
<span class="sourceLineNo">327</span>                        }<a name="line.327"></a>
<span class="sourceLineNo">328</span>                }<a name="line.328"></a>
<span class="sourceLineNo">329</span>                output.print(header.replace(delimiter, "_"));<a name="line.329"></a>
<span class="sourceLineNo">330</span>                for(int i = 0; i &lt; dList.size(); i++){<a name="line.330"></a>
<span class="sourceLineNo">331</span>                        output.print(delimiter + Utils.roundToDecimals(dList.get(i), decimalPlace));<a name="line.331"></a>
<span class="sourceLineNo">332</span>                }<a name="line.332"></a>
<span class="sourceLineNo">333</span>                output.println();<a name="line.333"></a>
<span class="sourceLineNo">334</span>                output.flush();<a name="line.334"></a>
<span class="sourceLineNo">335</span>        }<a name="line.335"></a>
<span class="sourceLineNo">336</span><a name="line.336"></a>
<span class="sourceLineNo">337</span>        private static void showHelp(){<a name="line.337"></a>
<span class="sourceLineNo">338</span>                System.err.println("NAME");<a name="line.338"></a>
<span class="sourceLineNo">339</span>                System.err.println("\tAn executable to generate physico-chemical properties of protein sequences.");<a name="line.339"></a>
<span class="sourceLineNo">340</span>                System.err.println();<a name="line.340"></a>
<span class="sourceLineNo">341</span><a name="line.341"></a>
<span class="sourceLineNo">342</span>                System.err.println("EXAMPLES");<a name="line.342"></a>
<span class="sourceLineNo">343</span>                System.err.println("\tjava -jar AAProperties.jar -i test.fasta -a");<a name="line.343"></a>
<span class="sourceLineNo">344</span>                System.err.println("\t\tGenerates all possible properties.");<a name="line.344"></a>
<span class="sourceLineNo">345</span>                System.err.println();<a name="line.345"></a>
<span class="sourceLineNo">346</span>                System.err.println("\tjava -jar AAProperties.jar -i test.fasta -1 -3 -7");<a name="line.346"></a>
<span class="sourceLineNo">347</span>                System.err.println("\t\tGenerates only molecular weight, extinction coefficient and isoelectric point.");<a name="line.347"></a>
<span class="sourceLineNo">348</span>                System.err.println();<a name="line.348"></a>
<span class="sourceLineNo">349</span>                System.err.println("\tjava -jar AAProperties.jar -i test.fasta -0 A -0 N -1");<a name="line.349"></a>
<span class="sourceLineNo">350</span>                System.err.println("\t\tGenerates composition of two specific amino acid symbol and molecular weight.");<a name="line.350"></a>
<span class="sourceLineNo">351</span>                System.err.println();<a name="line.351"></a>
<span class="sourceLineNo">352</span><a name="line.352"></a>
<span class="sourceLineNo">353</span>                System.err.println("OPTIONS");<a name="line.353"></a>
<span class="sourceLineNo">354</span>                System.err.println("\tRequired");<a name="line.354"></a>
<span class="sourceLineNo">355</span>                System.err.println("\t\t-i location of input FASTA file");<a name="line.355"></a>
<span class="sourceLineNo">356</span>                System.err.println();<a name="line.356"></a>
<span class="sourceLineNo">357</span><a name="line.357"></a>
<span class="sourceLineNo">358</span>                System.err.println("\tOptional");<a name="line.358"></a>
<span class="sourceLineNo">359</span>                System.err.println("\t\t-o location of output file [standard output (default)]");<a name="line.359"></a>
<span class="sourceLineNo">360</span>                System.err.println("\t\t-f output format [csv (default) or tsv]");<a name="line.360"></a>
<span class="sourceLineNo">361</span>                System.err.println("\t\t-x location of Amino Acid Composition XML file for defining amino acid composition");<a name="line.361"></a>
<span class="sourceLineNo">362</span>                System.err.println("\t\t-y location of Element Mass XML file for defining mass of elements");<a name="line.362"></a>
<span class="sourceLineNo">363</span>                System.err.println("\t\t-d number of decimals (int) [4 (default)]");<a name="line.363"></a>
<span class="sourceLineNo">364</span>                System.err.println();<a name="line.364"></a>
<span class="sourceLineNo">365</span><a name="line.365"></a>
<span class="sourceLineNo">366</span>                System.err.println("\tProvide at least one of them");<a name="line.366"></a>
<span class="sourceLineNo">367</span>                System.err.println("\t\t-a compute properties of option 1-9");<a name="line.367"></a>
<span class="sourceLineNo">368</span>                System.err.println("\t\t-1 compute molecular weight");<a name="line.368"></a>
<span class="sourceLineNo">369</span>                System.err.println("\t\t-2 compute absorbance");<a name="line.369"></a>
<span class="sourceLineNo">370</span>                System.err.println("\t\t-3 compute extinction coefficient");<a name="line.370"></a>
<span class="sourceLineNo">371</span>                System.err.println("\t\t-4 compute instability index");<a name="line.371"></a>
<span class="sourceLineNo">372</span>                System.err.println("\t\t-5 compute apliphatic index");<a name="line.372"></a>
<span class="sourceLineNo">373</span>                System.err.println("\t\t-6 compute average hydropathy value");<a name="line.373"></a>
<span class="sourceLineNo">374</span>                System.err.println("\t\t-7 compute isoelectric point");<a name="line.374"></a>
<span class="sourceLineNo">375</span>                System.err.println("\t\t-8 compute net charge at pH 7");<a name="line.375"></a>
<span class="sourceLineNo">376</span>                System.err.println("\t\t-9 compute composition of 20 standard amino acid (A, R, N, D, C, E, Q, G, H, I, L, K, M, F, P, S, T, W, Y, V)");<a name="line.376"></a>
<span class="sourceLineNo">377</span>                System.err.println("\t\t-0 compute composition of specific amino acid symbol");<a name="line.377"></a>
<span class="sourceLineNo">378</span>                System.err.println();<a name="line.378"></a>
<span class="sourceLineNo">379</span>        }<a name="line.379"></a>
<span class="sourceLineNo">380</span>}<a name="line.380"></a>




























































</pre>
</div>
</body>
</html>
